% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
  a4paper,
]{scrbook}

\usepackage{amsmath,amssymb}
\usepackage{setspace}
\usepackage{iftex}
\ifPDFTeX
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
\usepackage{lmodern}
\ifPDFTeX\else  
    % xetex/luatex font selection
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\usepackage[left=2.54cm,right=2.54cm]{geometry}
\setlength{\emergencystretch}{3em} % prevent overfull lines
\setcounter{secnumdepth}{5}
% Make \paragraph and \subparagraph free-standing
\makeatletter
\ifx\paragraph\undefined\else
  \let\oldparagraph\paragraph
  \renewcommand{\paragraph}{
    \@ifstar
      \xxxParagraphStar
      \xxxParagraphNoStar
  }
  \newcommand{\xxxParagraphStar}[1]{\oldparagraph*{#1}\mbox{}}
  \newcommand{\xxxParagraphNoStar}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
  \let\oldsubparagraph\subparagraph
  \renewcommand{\subparagraph}{
    \@ifstar
      \xxxSubParagraphStar
      \xxxSubParagraphNoStar
  }
  \newcommand{\xxxSubParagraphStar}[1]{\oldsubparagraph*{#1}\mbox{}}
  \newcommand{\xxxSubParagraphNoStar}[1]{\oldsubparagraph{#1}\mbox{}}
\fi
\makeatother


\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}\usepackage{longtable,booktabs,array}
\usepackage{calc} % for calculating minipage widths
% Correct order of tables after \paragraph or \subparagraph
\usepackage{etoolbox}
\makeatletter
\patchcmd\longtable{\par}{\if@noskipsec\mbox{}\fi\par}{}{}
\makeatother
% Allow footnotes in longtable head/foot
\IfFileExists{footnotehyper.sty}{\usepackage{footnotehyper}}{\usepackage{footnote}}
\makesavenoteenv{longtable}
\usepackage{graphicx}
\makeatletter
\newsavebox\pandoc@box
\newcommand*\pandocbounded[1]{% scales image to fit in text height/width
  \sbox\pandoc@box{#1}%
  \Gscale@div\@tempa{\textheight}{\dimexpr\ht\pandoc@box+\dp\pandoc@box\relax}%
  \Gscale@div\@tempb{\linewidth}{\wd\pandoc@box}%
  \ifdim\@tempb\p@<\@tempa\p@\let\@tempa\@tempb\fi% select the smaller of both
  \ifdim\@tempa\p@<\p@\scalebox{\@tempa}{\usebox\pandoc@box}%
  \else\usebox{\pandoc@box}%
  \fi%
}
% Set default figure placement to htbp
\def\fps@figure{htbp}
\makeatother

% Palatino font:
\usepackage[T1]{fontenc}
\usepackage[utf8]{inputenc}
\usepackage{palatino}

\usepackage{mathpazo}

\usepackage{upquote}

% front page
\usepackage[colorlinks]{hyperref} % for text color on title
\usepackage{titling} % makes \thetitle etc available...
\usepackage{lipsum}
\usepackage{tikz}
\tikzstyle{bag} = [align=center] % allows new lines in text below
\AtBeginDocument{\thispagestyle{empty}
\definecolor{titlecolor}{HTML}{1C2404}
\begin{tikzpicture}[remember picture,overlay]
\draw (current page.center) node[inner sep=0, opacity=0.5] {\resizebox{!}{350mm}{\includegraphics{highres_banner.pdf}}};
\draw (current page.center) node[inner sep=0, opacity=0.5] {\resizebox{!}{100mm}{\includegraphics{ausegl_sort.pdf}}};
% \draw (current page.center) node[bag] {  
        \draw (17, 1.7) node[bag, align=right, anchor=north east] {  
    % \mytitlefont 
    \color{titlecolor} \Large \textbf{Thesis} \\ 
    \\ 
    % \mytitlefont 
    \color{titlecolor} \Huge \textbf{\thetitle} \\ 
    \\ 
    % \mytitlefont
    \color{titlecolor} \huge \theauthor \\
    \\
    % \mytitlefont 
    \color{titlecolor} \Large \thedate 
    };
\end{tikzpicture}
\clearpage}

% back side
\AtEndDocument{\clearpage\thispagestyle{empty}
\begin{tikzpicture}[remember picture,overlay]
\draw (current page.center) node[inner sep=0, opacity=0.5] {\resizebox{!}{350mm}{\includegraphics{highres_banner.pdf}}};
% \draw (current page.center) node[bag] {  
    \draw (17, 1.7) node[bag, align=right, anchor=north east] {  
        % \mytitlefont 
        \color{titlecolor} \large Bioinformatics Research Centre \\ 
        % \mytitlefont 
        \color{titlecolor} \large Department of Molecular Biology and Genetics \\ 
        % \mytitlefont 
        \color{titlecolor} \large Aarhus University \\
        % \mytitlefont 
        \color{titlecolor} \large Universitetsbyen 81 \\
        % \mytitlefont 
        \color{titlecolor} \large 8000 Aarhus C \\
        % \mytitlefont 
        \color{titlecolor} \large Denmark
        };

\end{tikzpicture}}


% % picture (AU logo in title page)
% \usepackage{titlepic}
% % \usepackage{graphicx}
% \titlepic{\resizebox{!}{50mm}{\includegraphics[width=\textwidth]{ausegl_sort.pdf}}}



% \pagestyle{plain} % no running header

\usepackage[many]{tcolorbox}
\definecolor{quotegray}{HTML}{505050}
\newtcolorbox{myquote}{%
    enhanced jigsaw, 
    breakable,      % allow page breaks
    frame hidden,   % hide the default frame
    left=1cm,       % left margin
    right=1cm,      % right margin
    colback=white,
    fontupper=\color{quotegray},
    overlay={%
        \node [scale=4,
            text=lightgray,
            inner sep=0pt,] at ([xshift=0.5cm,yshift=-0.7cm]frame.north west){``}; 
        \node [scale=4,
            text=lightgray,
            inner sep=0pt,] at ([xshift=-0.5cm]frame.south east){''};  
            },
        % paragraph skips obeyed within tcolorbox
                parbox=false,
}
% redefine the 'quote' environment to use this 'myquote' environment
\renewenvironment{quote}{\begin{myquote}}{\end{myquote}}

% add a dummy abstract environment that is defined in quarto manuscript 
% but not in the quarto book that we run first to give it a slot in the 
% side bar
\ifcsmacro{abstract}{}{
  \let\endmyenvironment\undefined%
  \newenvironment{abstract}{\chapter*{Abstract}}{}
}

% \usepackage{framed} % not sure i need this anymore

% \usepackage[T1]{fontenc}
% \usepackage{inconsolata}

% % I have to only define Shaded if it is already defined.
% % The reason is that pandoc does not define the macro if there are not code blocks in the a markdown file.
% \ifx \@Shaded \@empty

% \renewcommand{\KeywordTok}[1]{\textcolor[rgb]{0, 0, 0}{\textbf{{#1}}}} % def and or not reg
% \renewcommand{\BuiltInTok}[1]{\textcolor[rgb]{0.373, 0.298, 0.580}{\textbf{{#1}}}} % print open 
% \renewcommand{\VariableTok}[1]{\textcolor[rgb]{0.141, 0.392, 0.824}{\textbf{{#1}}}}
% \renewcommand{\OperatorTok}[1]{\textcolor[rgb]{0, 0, 0}{{#1}}} % def and or not reg
% \renewcommand{\DataTypeTok}[1]{\textcolor[rgb]{1.0,0.13,0.00}{{#1}}}
% \renewcommand{\DecValTok}[1]{\textcolor[rgb]{0.655, 0.498, 0.161}{{#1}}}
% \renewcommand{\BaseNTok}[1]{\textcolor[rgb]{0.259, 0.592, 0.596}{{#1}}}
% \renewcommand{\FloatTok}[1]{\textcolor[rgb]{0.655, 0.498, 0.161}{{#1}}}
% \renewcommand{\CharTok}[1]{\textcolor[rgb]{0.678,0.141,0.098}{{#1}}}
% \renewcommand{\StringTok}[1]{\textcolor[rgb]{0.678,0.141,0.098}{{#1}}}
% \renewcommand{\CommentTok}[1]{\textcolor[rgb]{0.135, 0.134, 0.133}{{#1}}}
% \renewcommand{\OtherTok}[1]{\textcolor[rgb]{0.00,0.44,0.13}{{#1}}}
% \renewcommand{\AlertTok}[1]{\textcolor[rgb]{1.00,0.00,0.00}{\textbf{{#1}}}}
% \renewcommand{\FunctionTok}[1]{\textcolor[rgb]{0.549, 0.102, 0.063}{\textbf{{#1}}}}  % function name
% \renewcommand{\RegionMarkerTok}[1]{{#1}}
% \renewcommand{\ErrorTok}[1]{\textcolor[rgb]{1.00,0.00,0.00}{\textbf{{#1}}}}
% \renewcommand{\NormalTok}[1]{\textcolor[rgb]{0, 0, 0}{{#1}}}


% \else
%   % no code blocks with markup...
% \fi



\usepackage{etoolbox}
\makeatletter
\g@addto@macro{\appendix}{%
  \patchcmd{\@@makechapterhead}% <cmd>
    {\endgraf\nobreak\vskip.5\baselineskip}% <search>
    {\hspace*{-.5em}:\space}% <replace>
    {}{}% <success><failure>
  \patchcmd{\@chapter}% <cmd>
    {\addchaptertocentry{\thechapter}}% <search>
    {\addchaptertocentry{Appendix~\thechapter:}}% <replace>
    {}{}% <success><failure>
  \addtocontents{toc}{%
    \protect\patchcmd{\protect\l@chapter}% <cmd>
      {1.5em}% <search>
      {6.5em}% <replace>
      {}{}}% <success><failure>
}
\renewcommand{\autodot}{}% Remove all end-of-counter dots
\makeatother


% restart chapter numbers in each part
\makeatletter
\@addtoreset{chapter}{part}
\makeatother


% \usepackage{chngcntr}
% \counterwithin*{subsubsection}{chapter}
% \counterwithout*{subsubsection}{section}
% \counterwithout*{subsubsection}{subsection}

% %  % KMT only use subsubsection number (this increments though the book 
% %  % when we supress numbering with  {.unnumbered} after each header except Exerisices
% \renewcommand\thesubsubsection{\arabic{section}}
% \renewcommand\thesubsubsection{\arabic{subsection}}
% \renewcommand\thesubsubsection{\arabic{chapter}-\arabic{subsubsection}}


\renewcommand*{\chapterformat}{%
\textcolor[rgb]{0.7, 0.7, 0.7}{\thechapter}\autodot\enskip%
}
\renewcommand*{\sectionformat}{%
\textcolor[rgb]{0.7, 0.7, 0.7}{\thesection}\autodot\enskip%
}
\renewcommand*{\subsectionformat}{%
\textcolor[rgb]{0.7, 0.7, 0.7}{\thesubsection}\autodot\enskip%
}
\renewcommand*{\subsubsectionformat}{%
\textcolor[rgb]{0.7, 0.7, 0.7}{\thesubsubsection}\autodot\enskip%
}



% % prevent latex from "floating" the figures
% \usepackage{float}
% \let\origfigure\figure
% \let\endorigfigure\endfigure
% \renewenvironment{figure}[1][2] {
%     \expandafter\origfigure\expandafter[H]
% } {
%     \endorigfigure
% }

\usepackage{xcolor}
% \let\oldtextit\textit 
% \renewcommand\textit[1]{\oldtextit{\color{blue}#1}}
\let\oldemph\emph
\renewcommand\emph[1]{\oldemph{\color{gray}#1}}

\makeatletter
\@ifpackageloaded{caption}{}{\usepackage{caption}}
\AtBeginDocument{%
\ifdefined\contentsname
  \renewcommand*\contentsname{Table of contents}
\else
  \newcommand\contentsname{Table of contents}
\fi
\ifdefined\listfigurename
  \renewcommand*\listfigurename{List of Figures}
\else
  \newcommand\listfigurename{List of Figures}
\fi
\ifdefined\listtablename
  \renewcommand*\listtablename{List of Tables}
\else
  \newcommand\listtablename{List of Tables}
\fi
\ifdefined\figurename
  \renewcommand*\figurename{Figure}
\else
  \newcommand\figurename{Figure}
\fi
\ifdefined\tablename
  \renewcommand*\tablename{Table}
\else
  \newcommand\tablename{Table}
\fi
}
\@ifpackageloaded{float}{}{\usepackage{float}}
\floatstyle{ruled}
\@ifundefined{c@chapter}{\newfloat{codelisting}{h}{lop}}{\newfloat{codelisting}{h}{lop}[chapter]}
\floatname{codelisting}{Listing}
\newcommand*\listoflistings{\listof{codelisting}{List of Listings}}
\makeatother
\makeatletter
\makeatother
\makeatletter
\@ifpackageloaded{caption}{}{\usepackage{caption}}
\@ifpackageloaded{subcaption}{}{\usepackage{subcaption}}
\makeatother

\usepackage[numbers]{natbib}
\bibliographystyle{plainnat}
\usepackage{bookmark}

\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\urlstyle{same} % disable monospaced font for URLs
\hypersetup{
  pdftitle={Chromatin Compartments and Selection on X},
  pdfkeywords={Hi-C, Chromatin Compartments, Selection},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}


\title{Chromatin Compartments and Selection on X}
\usepackage{etoolbox}
\makeatletter
\providecommand{\subtitle}[1]{% add subtitle to \maketitle
  \apptocmd{\@title}{\par {\large #1 \par}}{}{}
}
\makeatother
\subtitle{How Edges of Active Chromatin Align with Selection Regions in
Primates}
\author{Søren Jørgensen}
\date{2024-11-30}

\begin{document}
\frontmatter
\cleardoublepage
\thispagestyle{empty}
{\centering
\hbox{}\vskip 0cm plus 1fill

{%\mytitlefont 
\Huge\bfseries Chromatin Compartments and Selection on X \par}
\vspace{3ex}
{\Large\bfseries How Edges of Active Chromatin Align with Selection
Regions in Primates \par}
\vspace{6ex}

    {\Large\bfseries Søren Jørgensen \par}
    %
%
\vskip 0cm plus 2fill

{\resizebox{!}{50mm}{\includegraphics[width=\textwidth]{ausegl_sort.pdf}} \par}
\vskip 0cm plus 2fill

{\bfseries\Large\textit{} \par}
\vspace{3ex}

{\bfseries\large 2024-11-30 \par}
\vspace{3ex}

\vspace{12ex}
{\small Submitted in fulfillment of the requirements
of the degree of  \par}
\pagebreak

\begin{quote}
\raggedright    
3-dimensional structure of chromatin brings light onto the mystery of
selfish genes. \par
\raggedleft    
    {- Søren Jørgensen \par}
%    
\end{quote}
}

\renewcommand*\contentsname{Table of contents}
{
\setcounter{tocdepth}{1}
\tableofcontents
}

\setstretch{1}
\mainmatter
\chapter{Introduction}\label{introduction}

\section{Sexual reproduction (spermatogenesis,
meiosis)}\label{sexual-reproduction-spermatogenesis-meiosis}

The production of gametes in a sexually reproducing organism is a highly
complex process that involves numeruous elements. Spermatogenesis, the
process of forming male gametes, involves four stages of differentiation
from a germ cell through \emph{spermatogonia}, \emph{pachytene
spermatocyte}, and \emph{round spermatids} to \emph{spermatozoa}, or
\emph{sperm} \citep{wang_reprogramming_2019}, and it is the very basis
of male reproduction. The specialized cell division of meiosis neatly
handles the pairing, recombination, and segregation of homologous
chromosomes, thereby ensuring proper genetic distribution. Deeply
understanding the steps of molecular steps of reproduction and how our
genetic material is inherited is essential in biology, bringing insight
to areas such as speciation, population diversity, and (male)
infertility.

\section{Selfish genes (and
randomness)}\label{selfish-genes-and-randomness}

The conventional story of meiosis in gametogenesis is one of random
segregation of the sex chromosomes. They split into haploid gametes,
where each chromosome has an equal chance of being passed on to a
gamete. That seems like a fair game, but what if some genes are cheating
the system by making others less viable. A meiotic driver is a selfish
gene element that modulates meiosis and preferentially transmits its own
allele through meiosis, regardless of the downstream fitness effects it
may have (good or bad) on the organism it is part of. This phenomenon
challenges the traditional understanding of selection, extending its
scope beyond the fitness effects on an organism to include selective
pressures at the molecular level. For example, if some genes on the X
chromosome create a disadvantage for gametes that \emph{do not} contain
those genes, making sure the Y chromosome is not as viable as the X,
resulting in a sex imbalance and possibly numerous other downstream
effects. That is exactly what is coined \emph{sex chromosome meiotic
drive} \citep{jaenike_sex_2001}, a result of selfish genetic elements.

Motivated by previous results in the Munch Research group
\citep{munch_group_2024} on hybrid incompatibility and extended common
haplotypes \citep{skov_extraordinary_2023, sorensen_genome_wide_2023}
that could be explained by meiotic drive, we wanted to investigate how
these patterns correlate with chromatin compartments.

\section{High-Throughput Chromosome Conformation Capture
(Hi-C)}\label{high-throughput-chromosome-conformation-capture-hi-c}

Our DNA can be divided into different orders of structure. \emph{3C}
focus on identifying the highest orders of organization inside the
nucleus, that is, when the 30 nm thick coil of chromatin fibers folds
into loops, Topologically Associating Domains (TADs), and chromatin
compartments. Here, we narrow our focus on the largest of the
structures, \emph{compartments}, that is known to determine availability
to transcription factors, thus making an \emph{A} compartment
\emph{active}---and the \emph{B} compartment \emph{inactive}. The
introduction of the Hi-C (high-throughput 3C) method
\citep{lieberman_aiden_comprehensive_2009} opened new possibilities for
exploring the three-dimensional organization of the genome.

\newpage{}

\chapter{Methods}\label{methods}

In this project, we formulate two objectives:

\textbf{A}: Reproduce the Hi-C interaction maps and eigendecomposition
from \citep{wang_reprogramming_2019}, with some modifications. We
briefly use \emph{HiCExplorer}, but change the analyses to use the
\emph{Open2C Ecosystem} \citep{open2c} which have a Pyton API as well as
command-line functions, which can be paired very well with Jupyter
Notebooks. The majority of the data analysis was run with a \emph{gwf}
workflow, and the commands that were visually inspected were run in
Jupyter Notebooks.

\textbf{B} Compare with regions of selection that are found in
\emph{papio anubis}, and maybe in \emph{human} too. Investigate the
biological meaning of the results.

All computations were performed on GenomeDK (GDK) {[}ref{]}, an HPC
cluster located on Aarhus Uninversity, and most of the processing of the
data was nested into a \emph{gwf} workflow {[}ref{]}, a workflow manager
developed at GDK. I would like to thank GDK and Aarhus University for
providing computational resources and support that contributed to these
research results.

The whole of this project is carried out with reproducibility in mind,
so an effort (and quite a significant amount of time) has been put into
documenting code and organizing the project for readbility and
transparency through a Quarto project {[}ref{]}. Therefore, all code,
virtual environments and text is made available as a Quarto book,
rendered directly from the GitHub repository with GitHub Pages
{[}ref{]}. To make this possible, the Quarto documentation has been
extensively studied and discussed with \emph{KMT} {[}ref, aknowledge{]}.

\section{Initial Exploration with
HiCExplorer}\label{initial-exploration-with-hicexplorer}

Here moves most of the text about HiCExplorer\ldots{}

\section{Downloading Data and Project
Structure}\label{downloading-data-and-project-structure}

To reproduce the results from \citep{wang_reprogramming_2019}, I chose
to use their raw data directly from the SRA portal {[}ref{]}. I filtered
the data to contain all their paired-end Hi-C reads, and included only
macaque samples. The data set also contains RNAseq data, and the same
tissues for both macaque and mouse. The meta data for the data set was
extracted into a runtable \texttt{SRA-runtable.tsv}. To get an overview
of the data accessions used in this analysis, we will first summarize
the runtable that contains the accession numbers and some metadata for
each sample (Table~\ref{tbl-runtable-summary}). It adds up to
\textasciitilde1Tb of compressed \texttt{fastq} files, holding
\textasciitilde9.5 billion reads, roughly evenly spread on the 5 tissue
types.

\begin{longtable}[]{@{}lllll@{}}

\toprule\noalign{}
~ & source\_name & GB & Bases & Reads \\
\midrule\noalign{}
\endfirsthead
\toprule\noalign{}
~ & source\_name & GB & Bases & Reads \\
\midrule\noalign{}
\endhead
\bottomrule\noalign{}
\tabularnewline
\caption{}\label{T_55add}\tabularnewline
\endlastfoot
0 & fibroblast & 211.403275 & 553,968,406,500 & 1,846,561,355 \\
1 & pachytene spermatocyte & 274.835160 & 715,656,614,700 &
2,385,522,049 \\
2 & round spermatid & 243.128044 & 655,938,457,200 & 2,186,461,524 \\
3 & sperm & 164.131640 & 428,913,635,400 & 1,429,712,118 \\
4 & spermatogonia & 192.794420 & 518,665,980,300 & 1,728,886,601 \\


\caption{\label{tbl-runtable-summary}Summary of the data accessions used
in this analysis}

\tabularnewline
\end{longtable}

\section{Handling coolers (Or: preparing
coolers)}\label{handling-coolers-or-preparing-coolers}

\begin{figure}

\centering{

\pandocbounded{\includegraphics[keepaspectratio]{illustrations/placeholder2000x360.png}}

}

\caption{\label{fig-flowchart-handling-coolers}A flowchart showing the
pipeline from \texttt{.fastq} to \texttt{.mcool}. The first 6 steps were
done with a Probably BioRender or Inkscape.}

\end{figure}%

\subsection{\texorpdfstring{The \emph{gwf} workflow
targets}{The gwf workflow targets}}\label{the-gwf-workflow-targets}

A \emph{gwf} workflow was created to handle the first part of the data
processing, and each accesion number (read pair, mate pair) from the
Hi-C sequencing was processed in parallel, so their execution was
independent from each other.

\subsubsection{Downloading the reads}\label{downloading-the-reads}

The reads were downloaded from NCBI SRA portal {[}ref{]} directly to GDK
using \texttt{sra-downloader} {[}ref{]} through docker {[}ref{]} as
\texttt{.fastq.gz} files.

\subsubsection{Handling the reference}\label{handling-the-reference}

Needs:

\begin{itemize}
\tightlist
\item[$\boxtimes$]
  \emph{rheMac10}, \texttt{bwa} indexing
\item[$\boxtimes$]
  Wang et al.~used \emph{rheMac2}, the first assembly of rhesus
\item[$\boxtimes$]
  Bowtie2 indexing
\end{itemize}

The latest reference genome for rhesus macaque (\emph{macaca mulata}),
\emph{rheMac10} (or \emph{Mmul\_10}, UCSC or NCBI naming conventions,
respectively) was downloaded to GDK from UCSC web servers with
\texttt{wget} {[}ref{]}. To use \texttt{bwa} (Burrow Wheeler's Aligner)
{[}ref{]} for mapping, rheMac10 needs to be indexed with both
\texttt{bwa\ index} with the \texttt{-\/-bwtsw} option and
\texttt{samtools\ faidx}, which results in six indexing files for
\texttt{bwa\ mem} to use.

Since \citeyearpar{wang_reprogramming_2019}, the reference genome for
rhesus macaque has changed several times from \emph{rheMac2} to
\emph{rheMac10}, each time resulting in a much less fragmented reference
assembly. Part of the reasoning for reproducing their results was doing
so on the latest assembly of the Macaca mulata genome, which arguably
will result in a more accurate mapping of the reads, and a better
inference of the chromatin compartments as well.

Several mappers were used in different configurations (described in
below), and \texttt{bowtie2} requires its own indexing of the reference,
using \texttt{bowtie2-build\ -\/-large-index}, which creates six index
files for \texttt{bowtie2} to use. \texttt{-\/-large-index} creates the
special indexing format required for large genomes such as macaque.

\subsubsection{Mapping Hi-C reads}\label{mapping-hi-c-reads}

Needs:

\begin{itemize}
\tightlist
\item[$\boxtimes$]
  Genereal problems from standard PE reads to Hi-C reads (rescue
  chimeric fragments)
\item[$\boxtimes$]
  HiCExplorer, bwa, mapping reads seperately
\item[$\boxtimes$]
  Bowtie2 (local and end-to-end)
\item[$\square$]
  Open2C formats and \texttt{bwa}
\end{itemize}

\textbf{General Approach} The main difference between Hi-C libraries and
standard paired-end libraries is the high fraction of chimeric reads in
Hi-C. As a contact pair is crosslinked and ligated before sequencing,
chimeric reads occur as a feature, and standard mapping techniques seeks
to filter out this type of reads {[}ref{]}. Thus, we need specialized
tools for rescuing chimeric reads.

\textbf{Reproduction} It was not feasible to follow the same approach as
\citet{wang_reprogramming_2019} with either \emph{HiCExplorer} or
\emph{Open2C}, as they use a third software, \emph{HiC-Pro}. Hic-Pro
uses bowtie2 in end-to-end mode, followed by remapping of 5'-ends of the
unmapped reads to rescue chimeric fragments along with another approach.
I argue that even when trying to reproduce results, it is nonsensical to
use methods that are not state-of-the-art. The HiC-Pro pipeline stops at
a normalized contact map, and is thus not sufficient for downstream
analysis.

\begin{itemize}
\tightlist
\item
  Hi-C is a rapidly evolving concept and a lot has changed in 5 years
  \citep[since][]{wang_reprogramming_2019}
\end{itemize}

\textbf{HiCExplorer} Initially, recommendations from HiCExplorer were
used. According to their documentation {[}ref{]} it is crucial to 1)
align reads locally, and 2) map mates separately. They recommend either
of \texttt{bwa} or \texttt{bowtie2}, so I tested both with their
recommended settings. \texttt{bowtie2} turned out to be a lot more
resource-intensive and to produce almost no mapped reads {[}ref
sup-fig-bowtie2-stats{]}, so I suspect some settings was not set
correctly. The mapped reads was converted to a Hi-C Matrix with
\emph{HiCExplorers} \texttt{hicBuildMatrix}, which is extremely
memory-intensive.

However, the resulting files were incompatible with the \emph{open2C}
ecosystem, and I therefore followed \emph{HiCExplorer} pipeline to plot
and explore the matrices created from this mapping. However, the work
was laborious for experimentation, as the provided functions all write
plots to files. I did not manage to make an efficient implementation for
plotting the \texttt{.h5} files produced by the pipeline, and I
relatively quickly shifted to \emph{Open2C} for their promises of the
greener grass.

\textbf{Open2C} Suspiciously, \citet{open2c} never mentions any problems
with aligning the Hi-C reads, they just provide an example using
\texttt{bwa\ mem} with the \texttt{-P} option set, which activates the
Smith-Waterman {[}ref{]} algorithm to rescue missing hits. The
documentation of \texttt{bwa}
\href{https://bio-bwa.sourceforge.net}{ref} state that both
\texttt{bwa-mem} and \texttt{bwa-sw} will rescue chimeric reads.

\subsubsection{Pair and sort the reads}\label{pair-and-sort-the-reads}

Needs:

\begin{itemize}
\tightlist
\item[$\square$]
  mapping mates separately vs.~as paired-end reads
\item[$\square$]
  \texttt{pairtools\ parse} and \texttt{pairtools\ sort}
\item[$\square$]
  discuss the use of default parameters:
  \href{https://pairtools.readthedocs.io/en/latest/protocols_pipelines.html\#recommended-pairtools-parameters-for-standard-hi-c-protocols}{docs}
\end{itemize}

\subsubsection{Filter (deduplicate)
pairs}\label{filter-deduplicate-pairs}

At this point we will remove all reads that are mapped to an unplaced
scaffold. Even though the publication of \emph{rhemac10} assembly states
they have closed 99\% of the gaps since \emph{rhemac8}, \emph{rheMac10}
still contain more than 2,500 unplaced scaffolds, which are all
uninformative when calculating the chromatin compartments as is the goal
of this analysis. Therefore, we simply only include the list of
conventional chromosomes (1..22, X, Y) when doing the deduplication.
Initially, the default values were used to remove duplicates, where
pairs with both sides mapped within 3 base pairs from each other are
considered duplicates.

\texttt{cooler} recommend to store the most comprehensive and unfilteres
list of pairs, and then applying a filter on it on the fly by piping
from \texttt{pairtools\ select} I have missed this step, so I have not
filtered for mapping quality. I will make a histogram showing the
distribution of mapq scores to see the significance of this. Or just
rerun that part of the analysis.

\subsubsection{Create interaction matrices
(coolers)}\label{create-interaction-matrices-coolers}

The final part of the \emph{gwf} workflow takes \texttt{.pairs} as input
and outputs a \texttt{.cool} file (\emph{cooler}). Initially, we read
directly from the newly generated deduplicated pairs without additional
filtering, but here, the official recommendation is to filter out
everything below \(mapq = 30\) by piping the pairs through
\texttt{pairtools\ select\ "(mapq1\textgreater{}=30)\ and\ (mapq2\textgreater{}=30)"}
to \texttt{cooler\ cload\ pairs}.

We should have plenty of data to do the filtering, but I argue it is not
strictly necessary. I will show a histogram of the \emph{mapq} scores to
convince you {[}ref{]}. Otherwise, I will have fixed this issue.

\subsection{Notebook edits}\label{notebook-edits}

As \texttt{cooler} and \texttt{cooltools} have a Python API, the more
experimental parts of the analysis were moved to Jupyter Notebooks
(still running on GenomeDK). \texttt{cooltools} comes with a helper
library for operations on genomic intervals called \texttt{bioframe}.

\subsubsection{Pooling samples (Merging
coolers)}\label{pooling-samples-merging-coolers}

The samples are grouped into \emph{replicates} with a unique
\textbf{BioSample} ID, but we chose to pool all the interaction matrices
for each cell type. We argue that when \citet{wang_reprogramming_2019}
determine compartments to be highly reproducible between replicates, by
merging the replicates we can get a more robust signal.

\texttt{cooler\ merge} was used to merge all samples in each sub-folder
(cell type) to just one interaction matrix for each cell type. The
function merges matrices of the same dimensions by simply adding the
interaction frequencies of each genomic position together, resulting in
less empty positions by chance.

\subsubsection{Create multi-resolution coolers
(zoomify)}\label{create-multi-resolution-coolers-zoomify}

A feature of working inside the ecosystem of \emph{Open2C} {[}ref{]} is
that it natively provides support for storing sparse interaction
matrices in multiple resolutions in the same file by adding groups to
the cooler {[}ref{]}. We can then efficiently store resolutions (i.e.,
different bin sizes) that is multiples of the smallest bin size. We
chose to use 10kb, 50kb, 100kb, and 500kb bins, and the resolutions are
made by recursively binning the base resolution. We call this process
zoomifying.

\subsubsection{Matrix balancing (Iterative
correction)}\label{matrix-balancing-iterative-correction}

Finally, we balance the matrices using the cooler CLI. We use
\texttt{cooler\ balance} with the default options which iteratively
balances the matrix (Iterative Correction). It is first described as a
method for bias correction of Hi-C matrices in
\citep{imakaev_iterative_2012}, where it is paired with eigenvector
decomposition, coining the combined analysis \textbf{ICE}. Here, the
eigenvector decomposition of the obtained maps is experimentally
validated to provide insights into local chromatin states.

{[}According to \texttt{cooler} documentation{]} We have to balance the
matrices on each resolution, and thus it cannot be done prior to
zoomifying. They state that the balancing weights are
resolution-specific and will no longer retain its biological meaning
when binned with other weights. Therefore, we apply
\texttt{cooler\ balance} to each resolution separately.
\texttt{cooler\ balance} will create a new column in the \texttt{bins}
group of each cooler , \texttt{weight}, which can then be included or
not in the downstream analysis. This means we will have access to both
the balanced and the unbalanced matrix.

The default mode uses genome-wide data to calculate the weights for each
bin. It would maybe be more suitable to calculate the weights for
\emph{cis} contacts only, and that is possible through the
\texttt{-\/-cis-only} flag, and that can be added to another column, so
that we can compare the difference between the two methods easily.
However, we will only use the default mode for now.

\subsubsection{Eigendecomposition}\label{eigendecomposition}

The eigendecomposition of a Hi-C interaction matrix is performed in
multiple steps. As value of the eigenvector is only \emph{significant}
up to a sign, it is convention {[}ref{]} to use GC content as a phasing
track to orient the vector. E1 is arbitrarily defined to be positively
correlated with GC content, meaning a positive E1 value signifies an
active chromatin state, which we denote a A-type compartment (or simply
A-compartment). We performed eigendecomposition of two resolutions, 100
Kbp and 500 Kbp.

First, we calculate the GC content of each bin of the reference genome,
\emph{rheMac10}, which is binned to the resolution of the Hi-C matrix we
are handling. It is done with \texttt{bioframe.frac\_gc}
(\emph{Open2C}). To calculate the E1 compartments, we use only
within-chromosome contacts (\emph{cis}), as we are not interested in the
genome-wide contacts. \texttt{cooltools.eigs\_cis} will decorrelate the
contact-frequency by distance before performing the eigendecomposition.
\texttt{eigs\_cis} needs a \emph{viewframe} (view) to calculate E1
values, the simplest view being the full chromosome. However, when there
is more variance between chromosome arms than within arms, the sign of
the first eigenvector will be determined largely by the chromosome arm
it sits on, and not by the chromatin compartments. To mitigate this, we
apply a chromosome-arm-partitioned view of the chromosome (as a bedlike
format, described in \texttt{bioframe} docs {[}ref{]}).

\subsubsection{Plotting}\label{plotting}

Needs:

\begin{itemize}
\tightlist
\item[$\square$]
  HiCExplorer: plot with CLI, writes to pngs
\item[$\square$]
  cooler: plot through python API (fetch matrix from cooler)
\end{itemize}

\newpage{}

\chapter{Results}\label{results}

Here are the glorious results

\newpage{}

\chapter{Discussion}\label{discussion}

Here is the discussion

\newpage{}

\chapter*{Bibliography}\label{bibliography}
\addcontentsline{toc}{chapter}{Bibliography}

\begingroup
\raggedright

\renewcommand{\bibsection}{}
\bibliography{../references.bib}

\endgroup


\backmatter



\end{document}
