# %% [markdown]
# ---
# title: "gwf_map_reads"
# author: Søren Jørgensen
# date: last-modified
# execute: 
#   enabled: false
# ---

# %%
from gwf import *
import glob
import os

#######################################
#
# GWF workflow to map Hi-C reads to pairs with `bwa`, `samtools` and `HiCExplorer`.
#
# How to run:
# conda activate gwf
# gwf -f gwf_map_reads.py status
#
# Workflow:
#   1a bwa_index        : [Reference genome] Index the reference genome with `bwa index`
#   1b sam_index        : [Reference genome] Index the fasta again with `samtools faidx``
#   1c find_rest_sites  : [Reference genome] Find the restriction sites in the reference genome with `hicFindRestSites` from HiCExplorer
#   2  bwa_map          : Map reads (individually) to the reference with `bwa mem`
#   3  build_hic_matrix : Build the interaction matrix from mapped reads (.bam) with `hicBuildMatrix` from HiCExplorer
#
#######################################

# Create a workflow object
gwf = Workflow(defaults={'nodes': 1, 'queue':"normal", 'account':"hic-spermatogenesis"})

#############################################
############### Templates ###################
#############################################

def bwa_index(ref_genome):
    """Template for indexing the reference genome with bwa"""
    inputs  = [ref_genome]
    outputs = [f"{ref_genome}.amb", 
               f"{ref_genome}.ann", 
               f"{ref_genome}.bwt", 
               f"{ref_genome}.pac", 
               f"{ref_genome}.sa"]
    options = {'cores':1, 'memory': "5g", 'walltime':"02:00:00"}
    spec = f'''
source $(conda info --base)/etc/profile.d/conda.sh
conda activate hic
bwa index -p {ref_genome} -a bwtsw {ref_genome}
'''
    return AnonymousTarget(inputs=inputs, outputs=outputs, options=options, spec=spec)

def sam_index(ref_genome):
    """Creating a Fasta index. `bwa mem` also needs a fasta index generated by samtools"""
    inputs = [ref_genome]
    outputs = [f"{ref_genome}.fai"]
    options = {'cores':1, 'memory':"5g", 'walltime':"00:20:00"}
    spec="""
source $(conda info --base)/etc/profile.d/conda.sh
conda activate hic
samtools faidx {ref_genome}
"""
    return AnonymousTarget(inputs=inputs, outputs=outputs, options=options, spec=spec)

def find_rest_sites(ref_genome, rest_seq, out_bed):
    """Find the restriction sites in the reference genome.
    Used by `hicBuildMatrix` when building the interaction matrix"""
    inputs = [ref_genome]
    outputs = [out_bed]
    options = {'cores':1, 'memory':"5g", 'walltime':"00:20:00"}
    spec = f"""
source $(conda info --base)/etc/profile.d/conda.sh
conda activate hic
hicFindRestSites --fasta {ref_genome} --searchPattern {rest_seq} --outFile {out_bed}
"""
    return AnonymousTarget(inputs=inputs, outputs=outputs, options=options, spec=spec)

def bwa_map(ref_genome, in_fastq, out_bam):
    """
    Template for mapping reads to a reference genome using `bwa` and `samtools`. 
    NB! For Hi-C reads, pairs are mapped individually to avoid heuristics made for standard PE libs. (https://hicexplorer.readthedocs.io/en/latest/content/mES-HiC_analysis.html)
    bwa mem should be able to operate on .gz files directly
    """
    threads = 16
    inputs = [f"{ref_genome}.amb", 
              f"{ref_genome}.ann", 
              f"{ref_genome}.bwt", 
              f"{ref_genome}.pac", 
              f"{ref_genome}.sa", 
              f"{ref_genome}.fai",
              in_fastq]
    outputs = [out_bam]
    options = {'cores':threads, 'memory': "16g", 'walltime':"03:00:00"}
    spec = f"""
source $(conda info --base)/etc/profile.d/conda.sh
conda activate hic
bwa mem -t {threads} -A 1 -B 4 -E 50 -L 0 {ref_genome} {in_fastq} | samtools view -Shb - > {out_bam}
"""
    return AnonymousTarget(inputs=inputs, outputs=outputs, options=options, spec=spec)

def build_hic_matrix(bam1, bam2, rest_seq, rest_site_positions, out_matrix, out_qc_folder):
    threads = 8
    inputs = [bam1, bam2, rest_site_positions]
    outputs = [out_matrix, out_qc_folder]
    options = {'cores':threads, 'memory': "64g", 'walltime':"02:00:00"}
    spec = f"""
source $(conda info --base)/etc/profile.d/conda.sh
conda activate hic
hicBuildMatrix --samFiles {bam1} {bam2} \
    --binSize 10000 \
    --restrictionSequence {rest_seq} \
    --danglingSequence {rest_seq} \
    --restrictionCutFile {rest_site_positions} \
    --threads {threads} \
    --inputBufferSize 100000 \
    -o {out_matrix} \
    --QCfolder ./{out_qc_folder}
"""
    return AnonymousTarget(inputs=inputs, outputs=outputs, options=options, spec=spec)



#############################################
############### Create targets ##############
#############################################

# Do stuff with the reference genome
ref_genome = "../hicdata/Mmul10/ncbi_dataset/data/GCF_003339765.1/GCF_003339765.1_Mmul_10_genomic.fna"
rest_seq = "GATC"
rest_sites_out = "../hicdata/test/rest_site_positions.bed"

T1a = gwf.target_from_template(f"bwa_index_{os.path.basename(ref_genome)}", bwa_index(ref_genome=ref_genome))
T1b = gwf.target_from_template(f"sam_index_{os.path.basename(ref_genome)}", sam_index(ref_genome=ref_genome))
T1c = gwf.target_from_template(f"find_rest_sites_{os.path.basename(ref_genome)}", find_rest_sites(ref_genome=ref_genome, rest_seq=rest_seq, out_bed=rest_sites_out))


# Map Hi-C reads 
fastq_folder = "data/links/macaque_fastq/"
fastq_files = glob.glob(os.path.join(fastq_folder, "*.fastq.gz"))


# Pair the files (make sure they have the same base name prefix):
fastq_files.sort()
paired_fastq_files = list(zip(fastq_files[::2], fastq_files[1::2]))

for f1,f2 in paired_fastq_files:
    # Get the base names
    basename_1 = os.path.basename(f1).split('.fast')[0]
    basename_2 = os.path.basename(f2).split('.fast')[0]
    
    # Create the output bam filenames
    out_bam_1 = f"steps/{basename_1}.bam"
    out_bam_2 = f"steps/{basename_2}.bam"

    # Create targets for mapping
    T2a = gwf.target_from_template(f"bwa_map_{basename_1}", bwa_map(ref_genome=ref_genome, in_fastq=f1, out_bam=out_bam_1))
    T2b = gwf.target_from_template(f"bwa_map_{basename_2}", bwa_map(ref_genome=ref_genome, in_fastq=f2, out_bam=out_bam_2))

    # Combine pair names
    pairname = os.path.commonprefix([basename_1, basename_2]).split('_')[0]

    # Create the target for building the matrix (uses out_bam_1 and out_bam_2)
    out_matrix = f"steps/{pairname}_matrix.h5"
    out_qc_folder = f"steps/{pairname}_QC"
    T3 = gwf.target_from_template(f"build_hic_matrix_{pairname}", 
                                  build_hic_matrix(bam1=out_bam_1, bam2=out_bam_2, 
                                                   rest_seq=rest_seq,
                                                   rest_site_positions=rest_sites_out, 
                                                   out_matrix=out_matrix, 
                                                   out_qc_folder=out_qc_folder))



    

