<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Søren Jørgensen">

<title>Søren’s MSc project about Hi-C and spermatogenesis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./thoughts.html" rel="next">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./index.html"><span class="chapter-title">Project</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Søren’s MSc project about Hi-C and spermatogenesis</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link active"><span class="chapter-title">Project</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./thoughts.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Thoughts and Prayers</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">Notebooks</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./notebooks/001_weather.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Danish weather</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./notebooks/002_interaction.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Workplace interaction</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./notebooks/01_hicexplorer.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">HiCExplorer using macaque data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./notebooks/02_open2c_framework.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Open2C framework</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">Workflows</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./gwf_map_reads.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">gwf_map_reads</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./gwf_bowtie.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">gwf_bowtie</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./gwf_bowtie_local.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">gwf_bowtie_local</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./gwf_hiccorrect.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">gwf_hiccorrect</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./gwf_bwamem.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">gwf_bwamem</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./gwf_pair_alignments.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">gwf_pair_alignments</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true">
 <span class="menu-text">Reports</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./results/Readme.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Result files</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./reports/manuscript.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Correlating Chromatin Compartments with Selection</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./reports/supplementary.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Supplementary info</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./references.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">References</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul class="collapse">
  <li><a href="#project" id="toc-project" class="nav-link active" data-scroll-target="#project">Project</a>
  <ul class="collapse">
  <li><a href="#outline" id="toc-outline" class="nav-link" data-scroll-target="#outline">Outline</a></li>
  <li><a href="#notes-on-litterature-as-i-read-it" id="toc-notes-on-litterature-as-i-read-it" class="nav-link" data-scroll-target="#notes-on-litterature-as-i-read-it">Notes on litterature as I read it</a></li>
  <li><a href="#schedule-work-plan" id="toc-schedule-work-plan" class="nav-link" data-scroll-target="#schedule-work-plan">Schedule / Work plan</a></li>
  </ul></li>
  </ul>
<div class="quarto-code-links"><h2>Code Links</h2><ul><li><a href="https://github.com/munch-group/hic-spermatogenesis/" target="_blank"><i class="bi bi-github"></i>GitHub Repo</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Søren’s MSc project about Hi-C and spermatogenesis</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Søren Jørgensen </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">October 9, 2024 <br> 13:26:16 (UTC +00:00)</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<section id="project" class="level1 unnumbered">
<h1 class="unnumbered">Project</h1>
<ul>
<li>Visit the <a href="ttps://munch-group.org/project-guide/">project guide</a> for help.</li>
<li>Visit Open Chromosome Collective (<a href="https://github.com/open2c/">Open2C</a>) for tools and tutorials on 3D Chromosome biology</li>
<li>Visit <a href="https://hicexplorer.readthedocs.io/en/latest/">HiCExplorer docs</a></li>
<li>Raw data from Wang2019 can be download <a href="https://www.ncbi.nlm.nih.gov/Traces/study/?acc=PRJNA430777&amp;o=acc_s%3Aa">here</a>, but requires <a href="https://github.com/ncbi/sra-tools/wiki/01.-Downloading-SRA-Toolkit"><code>SRA toolkit</code></a></li>
</ul>
<section id="outline" class="level2">
<h2 class="anchored" data-anchor-id="outline">Outline</h2>
<p>Hi-C data is often generated for whole-genome sequencing (WGS) and the assembly of genomes due to its ability to capture the three-dimensional organization of the genome. This spatial information helps in correctly assembling contigs into chromosomes by providing contacts between distant genomic regions, thus resolving ambiguities in the linear sequence data. Thus, Hi-C data is often available, and leveraging this existing data can provide valuable insights to other areas without the need for additional sequencing. The aim of this project is to analyze chromatin compartments on the X chromosome during spermatogenesis in baboons, macaques, and humans using Hi-C data. The project involves generating interaction maps and identifying PC1 compartments. A key objective is to compare transition zones with regions undergoing positive selection to uncover how chromatin architecture reflects evolutionary pressures. The ultimate goal is to deepen our understanding of chromatin organization and the conservation of non-advantageous alleles, providing insights into the role of selfish genes in shaping genome evolution.</p>
</section>
<section id="notes-on-litterature-as-i-read-it" class="level2">
<h2 class="anchored" data-anchor-id="notes-on-litterature-as-i-read-it">Notes on litterature as I read it</h2>
<p><span class="citation" data-cites="wang_reprogramming_2019">(<a href="references.html#ref-wang_reprogramming_2019" role="doc-biblioref">Wang et al. 2019</a>)</span> Macaque/Mouse Hi-C paper. They use HiC-Pro and incorporates RNAseq data analyzed by TopHat and CuffLinks. Introduce small-scale <em>in situ</em> Hi-C (sisHi-C). They analyze both A/B compartments and TADs. Also introduce ‘Refined A/B compartments’ with ‘Local PCA’. They cite <span class="citation" data-cites="servant_hic-pro_2015">(<a href="references.html#ref-servant_hic-pro_2015" role="doc-biblioref">Servant et al. 2015</a>)</span> for their PCA method.</p>
<p><span class="citation" data-cites="servant_hic-pro_2015">(<a href="references.html#ref-servant_hic-pro_2015" role="doc-biblioref">Servant et al. 2015</a>)</span> describes the software <a href="https://github.com/nservant/HiC-Pro"><code>HiC-Pro</code></a>. Is a pipeline for all the steps until analysis of the contact maps. Includes (iterative correction) normalization steps, which is also available as standalone <a href="https://github.com/hiclib/iced"><code>hiclib</code>/<code>iced</code></a> tool. They refer to <span class="citation" data-cites="lieberman-aiden_comprehensive_2009">(<a href="references.html#ref-lieberman-aiden_comprehensive_2009" role="doc-biblioref">Lieberman-Aiden et al. 2009</a>)</span> for PCA method.</p>
<p><span class="citation" data-cites="lieberman-aiden_comprehensive_2009">(<a href="references.html#ref-lieberman-aiden_comprehensive_2009" role="doc-biblioref">Lieberman-Aiden et al. 2009</a>)</span> Introduces the Hi-C method and some ways of analyzing the data. E.g. the eigenvector (PCA) method, where they refer back to <span class="citation" data-cites="price_principal_2006">(<a href="references.html#ref-price_principal_2006" role="doc-biblioref">Price et al. 2006</a>)</span></p>
<p><span class="citation" data-cites="price_principal_2006">(<a href="references.html#ref-price_principal_2006" role="doc-biblioref">Price et al. 2006</a>)</span> use PCA to account for stratification in GWAS. The paper was cited by the Hi-C paper (2009) which was cited by the HiC-Pro paper as their PCA method. No real use for this paper, and they also refer back to a general guide on PCA from 2003. <em>End of citation chain</em>.</p>
<p><span class="citation" data-cites="zuo_stage-resolved_2021">(<a href="references.html#ref-zuo_stage-resolved_2021" role="doc-biblioref">Zuo et al. 2021</a>)</span> Stage-resolved Hi-C analysis in <strong>mouse spermatogenesis</strong>, using mm10 reference genome. They use <a href="https://github.com/mirnylab/hiclib-legacy"><code>hiclib</code></a> (now replaced by <a href="https://github.com/open2c/distiller-nf"><code>distiller</code></a>) and <a href="https://github.com/open2c/cooler"><code>cooler</code></a>/<a href="https://github.com/open2c/cooltools"><code>cooltools</code></a> from mirnylab. They focus on <strong>genome-wide</strong> chromatin structure, especially the weakening of TADs in prophase I (pachytene stage) while there is persistent CTCF binding (loops). They suggest that chromosome organization may provide an infrastructure for the modulation of meiotic recombination in higher eukaryotes.</p>
<blockquote class="blockquote">
<p><strong>Q:</strong> Could alterations in TAD organization during specific meiotic stages create opportunities for selfish genes to proliferate or influence transmission?</p>
</blockquote>
<p><span class="citation" data-cites="batra_accurate_2020">(<a href="references.html#ref-batra_accurate_2020" role="doc-biblioref">Batra et al. 2020</a>)</span> Baboon paper. They announce a new <em>de novo</em>-assembled reference genome, <em>Panubis1.0</em>, more accurate than the previous <em>Panu3.0</em>, which was a highly fragmented assembly based on short-read seq-tech and reference guided assembly (ref: <em>rhesus macaque</em>). They made the <em>de novo</em> assembly by combining 10x Genomics (physically linked short-reads) with Nanopore long-reads, and finally ordering and orienting the reads with Hi-C seq. They state no specific conclusions about the Hi-C data separate from the assemmbly.</p>
<p><span class="citation" data-cites="shami_single-cell_2020">(<a href="references.html#ref-shami_single-cell_2020" role="doc-biblioref">Shami et al. 2020</a>)</span> Divergent features of mammalian spermatogenesis. They use scRNA-seq to study spermatogenesis in human, macaque, mice. They compare conserved and divergent features of spermatogenesis across species. Spermatogonia, spermatocytes, spermatids. Identified a gene <em>TSPAN33</em> that was expressed in primate spermatogonia populations, but not in mice. They found both conserved patterns across species and species-specific patterns, notably in genes involved in immune system and blood-testis barrier, also including signalling pathways. Identify a genetic region that may be of interest, but otherwise I don’t know how to use this paper, unless they have Hi-C data they can send us.</p>
<p><span class="citation" data-cites="chakraborty_dchic_2022">(<a href="references.html#ref-chakraborty_dchic_2022" role="doc-biblioref">Chakraborty, Wang, and Ay 2022</a>)</span> dcHiC paper/<a href="https://github.com/ay-lab/dcHiC"><code>dcHiC</code> tool</a>. Differential compartments analysis. Utilize a multivariate distance measure, Mahalanobis (opposed to only pairwise comparisons) to identify and analyze changes across multiple Hi-C maps. They found compartmental changes in regions containing genes involved in cell identity (regulating pluripottency, cell adhesion, signaling pathways) during neuronal development. <code>dcHiC</code> can also identify sub-compartmental changes, and integrate compartmentalization data from other datasets (gene expression, histone modification, replication timing). Multivariata seems to be a better way of considering the Hi-C data.</p>
<p><span class="citation" data-cites="bicciato_hi-c_2022">(<a href="references.html#ref-bicciato_hi-c_2022" role="doc-biblioref">Bicciato and Ferrari 2022</a>)</span> Book on Hi-C data analysis. Might use to find conventional use of thresholds, etc. Might follow a couple of these pipelines if they are not already outdated.</p>
<p><span class="citation" data-cites="skov_extraordinary_2023">(<a href="references.html#ref-skov_extraordinary_2023" role="doc-biblioref">Skov et al. 2023</a>)</span> Paper from Munch-group. <em>Cautiously</em> hypothesize that meiotic drive on X chr could be responsible for their observations. Examining the selective sweeps in the context of (primate) spermatogenesis could be insightful. Integrating gene expression and Hi-C maps during spermatogenesis, could potentially uncover if the sweeps preferentially affect active elements/gene during spermatogenesis at i.e.&nbsp;specific stages.</p>
<p><span class="citation" data-cites="nunez_genetic-villains-2018">(<a href="references.html#ref-nunez_genetic-villains-2018" role="doc-biblioref">Bravo Núñez, Nuckolls, and Zanders 2018</a>)</span>. Paper on Genetic Villains/meiotic drivers. Describes the difficulty of detecting drive, and different types of drive. Killer-Target, Poison-Antidote. They have nice illustrations of meiotic drive.</p>
<p><span class="citation" data-cites="HiC_human_cancer_2021">(<a href="references.html#ref-HiC_human_cancer_2021" role="doc-biblioref">Gong et al. 2021</a>)</span> Most recent paper I could find on Hi-C in human. But it is about cancer and other omics than Hi-C.</p>
</section>
<section id="schedule-work-plan" class="level2">
<h2 class="anchored" data-anchor-id="schedule-work-plan">Schedule / Work plan</h2>
<p><strong>Semester:</strong> Monday, Aug 26th – Wednesday, Jan 15th (Week 35-3)<br>
<strong>Hand-in:</strong> Wednesday, Jan 15th (Week 3)<br>
<strong>Holiday:</strong> Wednesday, Oct 23rd – Monday, Oct 28th (Week 43)<br>
<strong>Christmas Holidays:</strong> Monday, Dec 23rd – Friday, Jan 3rd (Week 52-1)<br>
</p>
<p><strong>Week 35-36 (Aug 26 - Sep 7):</strong><br>
Conduct a literature review on chromatin architecture, Hi-C methodology, and the concept of selfish genes. Get an overview of (and obtain) data sets to be used.</p>
<p><strong>Week 37-38 (Sep 9 - Sep 21):</strong><br>
Assess the status of existing Hi-C data (raw or QC’ed). If necessary, perform initial quality control. Align reads to reference genomes.</p>
<p><strong>Week 39-40 (Sep 23 - Oct 5):</strong><br>
Process Hi-C interaction data. Perform quality assessment.</p>
<p><strong>Week 41-42 (Oct 7 - Oct 19):</strong><br>
Generate and normalize interaction matrices for the X chromosome.</p>
<p><strong>Week 43 (Oct 21 - Oct 26):</strong><br>
Holiday from Wednesday (no major activities planned). Tidy notes and do some writing.</p>
<p><strong>Week 44-45 (Oct 28 - Nov 9):</strong><br>
Perform PCA to obtain PC1 compartments. Visualize results.</p>
<p><strong>Week 46-47 (Nov 11 - Nov 23):</strong><br>
Identify A/B compartment transitions. Overlay with positive selection regions.</p>
<p><strong>Week 48-49 (Nov 25 - Dec 7):</strong><br>
Conduct cross-species analysis by converting genomic coordinates.</p>
<p><strong>Week 50 (Dec 9 - Dec 14):</strong><br>
Finish analyses. Compare results.</p>
<p><strong>Week 51 (Dec 16 - Dec 21):</strong><br>
Finalize analyses and begin writing initial thesis draft, incorporating analysis results and discussions.</p>
<p><strong>Week 52 (Dec 23 - Dec 28):</strong><br>
Christmas holidays (no major activities planned). Will work a couple of days Dec 27-29.</p>
<p><strong>Week 1 (Dec 30 - Jan 4):</strong><br>
Christmas holidays until Jan 2 (just performing modifications to analyses, etc.).</p>
<p><strong>Week 2 (Jan 6 - Jan 11):</strong><br>
Revise thesis draft based on feedback. Continue refinement of analyses if necessary.</p>
<p><strong>Week 3 (Jan 13 - Jan 15):</strong><br>
Final revisions and hand-in on Wednesday, January 15th.</p>


<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list" style="display: none">
<div id="ref-batra_accurate_2020" class="csl-entry" role="listitem">
Batra, Sanjit Singh, Michal Levy-Sakin, Jacqueline Robinson, Joseph Guillory, Steffen Durinck, Tauras P Vilgalys, Pui-Yan Kwok, et al. 2020. <span>“Accurate Assembly of the Olive Baboon ( <em><span>Papio</span> Anubis</em> ) Genome Using Long-Read and <span>Hi</span>-<span>C</span> Data.”</span> <em>GigaScience</em> 9 (12): giaa134. <a href="https://doi.org/10.1093/gigascience/giaa134">https://doi.org/10.1093/gigascience/giaa134</a>.
</div>
<div id="ref-bicciato_hi-c_2022" class="csl-entry" role="listitem">
Bicciato, Silvio, and Francesco Ferrari, eds. 2022. <em>Hi-<span>C</span> <span>Data</span> <span>Analysis</span>: <span>Methods</span> and <span>Protocols</span></em>. Vol. 2301. Methods in <span>Molecular</span> <span>Biology</span>. New York, NY: Springer US. <a href="https://doi.org/10.1007/978-1-0716-1390-0">https://doi.org/10.1007/978-1-0716-1390-0</a>.
</div>
<div id="ref-nunez_genetic-villains-2018" class="csl-entry" role="listitem">
Bravo Núñez, María Angélica, Nicole L. Nuckolls, and Sarah E. Zanders. 2018. <span>“Genetic Villains: Killer Meiotic Drivers.”</span> <em>Trends in Genetics</em> 34 (6): 424–33. https://doi.org/<a href="https://doi.org/10.1016/j.tig.2018.02.003">https://doi.org/10.1016/j.tig.2018.02.003</a>.
</div>
<div id="ref-chakraborty_dchic_2022" class="csl-entry" role="listitem">
Chakraborty, Abhijit, Jeffrey G. Wang, and Ferhat Ay. 2022. <span>“<span class="nocase">dcHiC</span> Detects Differential Compartments Across Multiple <span>Hi</span>-<span>C</span> Datasets.”</span> <em>Nature Communications</em> 13 (1): 6827. <a href="https://doi.org/10.1038/s41467-022-34626-6">https://doi.org/10.1038/s41467-022-34626-6</a>.
</div>
<div id="ref-HiC_human_cancer_2021" class="csl-entry" role="listitem">
Gong, Haiyan, Yi Yang, Sichen Zhang, Minghong Li, and Xiaotong Zhang. 2021. <span>“Application of Hi-c and Other Omics Data Analysis in Human Cancer and Cell Differentiation Research.”</span> <em>Computational and Structural Biotechnology Journal</em> 19: 2070–83. https://doi.org/<a href="https://doi.org/10.1016/j.csbj.2021.04.016">https://doi.org/10.1016/j.csbj.2021.04.016</a>.
</div>
<div id="ref-lieberman-aiden_comprehensive_2009" class="csl-entry" role="listitem">
Lieberman-Aiden, Erez, Nynke L. Van Berkum, Louise Williams, Maxim Imakaev, Tobias Ragoczy, Agnes Telling, Ido Amit, et al. 2009. <span>“Comprehensive <span>Mapping</span> of <span>Long</span>-<span>Range</span> <span>Interactions</span> <span>Reveals</span> <span>Folding</span> <span>Principles</span> of the <span>Human</span> <span>Genome</span>.”</span> <em>Science</em> 326 (5950): 289–93. <a href="https://doi.org/10.1126/science.1181369">https://doi.org/10.1126/science.1181369</a>.
</div>
<div id="ref-price_principal_2006" class="csl-entry" role="listitem">
Price, Alkes L, Nick J Patterson, Robert M Plenge, Michael E Weinblatt, Nancy A Shadick, and David Reich. 2006. <span>“Principal Components Analysis Corrects for Stratification in Genome-Wide Association Studies.”</span> <em>Nature Genetics</em> 38 (8): 904–9. <a href="https://doi.org/10.1038/ng1847">https://doi.org/10.1038/ng1847</a>.
</div>
<div id="ref-servant_hic-pro_2015" class="csl-entry" role="listitem">
Servant, Nicolas, Nelle Varoquaux, Bryan R. Lajoie, Eric Viara, Chong-Jian Chen, Jean-Philippe Vert, Edith Heard, Job Dekker, and Emmanuel Barillot. 2015. <span>“<span>HiC</span>-<span>Pro</span>: An Optimized and Flexible Pipeline for <span>Hi</span>-<span>C</span> Data Processing.”</span> <em>Genome Biology</em> 16 (1): 259. <a href="https://doi.org/10.1186/s13059-015-0831-x">https://doi.org/10.1186/s13059-015-0831-x</a>.
</div>
<div id="ref-shami_single-cell_2020" class="csl-entry" role="listitem">
Shami, Adrienne Niederriter, Xianing Zheng, Sarah K. Munyoki, Qianyi Ma, Gabriel L. Manske, Christopher D. Green, Meena Sukhwani, Kyle E. Orwig, Jun Z. Li, and Saher Sue Hammoud. 2020. <span>“Single-<span>Cell</span> <span>RNA</span> <span>Sequencing</span> of <span>Human</span>, <span>Macaque</span>, and <span>Mouse</span> <span>Testes</span> <span>Uncovers</span> <span>Conserved</span> and <span>Divergent</span> <span>Features</span> of <span>Mammalian</span> <span>Spermatogenesis</span>.”</span> <em>Developmental Cell</em> 54 (4): 529–547.e12. <a href="https://doi.org/10.1016/j.devcel.2020.05.010">https://doi.org/10.1016/j.devcel.2020.05.010</a>.
</div>
<div id="ref-skov_extraordinary_2023" class="csl-entry" role="listitem">
Skov, Laurits, Moisès Coll Macià, Elise Anne Lucotte, Maria Izabel Alves Cavassim, David Castellano, Mikkel Heide Schierup, and Kasper Munch. 2023. <span>“Extraordinary Selection on the Human <span>X</span> Chromosome Associated with Archaic Admixture.”</span> <em>Cell Genomics</em> 3 (3): 100274. <a href="https://doi.org/10.1016/j.xgen.2023.100274">https://doi.org/10.1016/j.xgen.2023.100274</a>.
</div>
<div id="ref-wang_reprogramming_2019" class="csl-entry" role="listitem">
Wang, Yao, Hanben Wang, Yu Zhang, Zhenhai Du, Wei Si, Suixing Fan, Dongdong Qin, et al. 2019. <span>“Reprogramming of <span>Meiotic</span> <span>Chromatin</span> <span>Architecture</span> During <span>Spermatogenesis</span>.”</span> <em>Molecular Cell</em> 73 (3): 547–561.e6. <a href="https://doi.org/10.1016/j.molcel.2018.11.019">https://doi.org/10.1016/j.molcel.2018.11.019</a>.
</div>
<div id="ref-zuo_stage-resolved_2021" class="csl-entry" role="listitem">
Zuo, Wu, Guangming Chen, Zhimei Gao, Shuai Li, Yanyan Chen, Chenhui Huang, Juan Chen, Zhengjun Chen, Ming Lei, and Qian Bian. 2021. <span>“Stage-Resolved <span>Hi</span>-<span>C</span> Analyses Reveal Meiotic Chromosome Organizational Features Influencing Homolog Alignment.”</span> <em>Nature Communications</em> 12 (1): 5827. <a href="https://doi.org/10.1038/s41467-021-26033-0">https://doi.org/10.1038/s41467-021-26033-0</a>.
</div>
</div>
</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
  </div>
  <div class="nav-page nav-page-next">
      <a href="./thoughts.html" class="pagination-link" aria-label="Thoughts and Prayers">
        <span class="nav-page-text"><span class="chapter-title">Thoughts and Prayers</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->




</body></html>