---
editor:
  render-on-save: true
title: "Chromatin Architecture and Selection on X"
subtitle: "Thesis Defence: Msc. Project 2024"
author: 
    - "Søren Jørgensen" 
    - "Kasper Munch (Supervisor)"
institute: 
    - "Stud. MSc. Bioinformatics, BiRC"
    - "Associate Professor, BiRC"
date: last-modified
title-slide-attributes: 
  data-notes: "*Opening:* Welcome everyone, and thank you all for joining in to listen to what I've spent my time doing since September. Let's get started. The X chromosome stands out in our genome due to its unique inheritance and hemizygosity in males, which is a probable cause of reduced diversity. We then ask a fundamental question: Could chromatin architecture help explain observed patterns of reduced diversity on X? [45s]"
---

## Overview 

:::: {layout="[[1,1],[1,1]]" style="margin-top:30pt;"}

::: {#always-visible}
* Research question
* Reduced Diversity on X
* Rationale
:::

::: {.fragment}
* Chromosome Architecture
* Compartments
* TADs
:::

::: {.fragment}
* Compartment Inference
* Genomic Comparison
* Regions under selection
:::

::: {.fragment}
* Compartment calls
* Test results
* Implications
* Future
:::

::::

::: {.fragment .alert style="text-align:center;"}
All code, figures, analyses, and slides are available at <br> [https://munch-group.org/hic-spermatogenesis/](https://munch-group.org/hic-spermatogenesis/)
:::

::: notes
Now, let me briefly give an overview of this talk. First, I will give a brief introduction to the analysis I did, expand a bit about reduced diversity on X, and connecting the two.
(click) I will introduce you to the hierarchical structures of our DNA, especially the higher-order organization, Topologically Associating Domains and A/B compartments. 
(click) I will show how I generate Hi-C interaction matrices and infer the compartments on macaque Hi-C data, and how I correlate the genomic coordinates with the regions under selection in baboons and human. 
(click) I will go through some compartment calls and the test results from correlating the genomic intervals. Finally, I will discuss the meaning of the results and what to focus on in the future.
(click) A major part of this project is reproducibility, which I will talk a bit about later on, but for now, I will let you know that all code, figures, analyses and these slides are available at munch-group.org/hic-spermatogenesis. Let's move on.
[2min]
:::

## Introduction and Background

::: fragment
### What have we done

Analyzed the 3D chromatin structure on X during spermatogenesis in rhesus macaque and correlating to low-diversity regions in baboons and humans, leveraging a comprehensive reproducible computational framework.
:::

::: fragment
### Reduced Diversity on X

Megabase-spanning regions of reduced diversity across great apes indicate strong selection, reflecting evolutionary dynamics that possibly interplay with structural features.
:::

::: fragment
### Rationale of analyzing chromatin architecture

Architecture modulates gene regulation and meiosis. Could these high-level structural features underlie the low-diversity regions?
:::


::: notes
In this study, we explore the X chromosome in rhesus macaque (Macaca mulatta), combining Hi-C analysis with genomic tools to unravel its 3D chromatin structure. Our goal? To investigate how chromatin architecture intertwines with evolutionary forces, particularly in regions under strong selection.
What are those regions under selection
This forms the rationale. I ask...
[1min]
:::


## Molecular Biology: Chromosome Architecture {auto-animate=true}

::: {.fragment fragment-index="1"}
### Chromosome Architecture in the Nucleus

-   Balance [compact storage]{.alert} with [functional accessibility]{.alert} for essential processes: replication and gene expression.\
:::

:::::::::: {.fragment fragment-index="2"}
:::::::: columns
::::::: {.column width="45%"}
### Levels of Organization

::: incremental
-   Double Helix
-   Nucleosomes
-   [Chromatin]{.alert}
:::

::::: {.fragment layout="[-0.01,0.3,0.6]" style="font-size:20pt;"}
::: {#col1 data-id="fiber-loop"}
-   Fibers
-   Loops
:::

::: {#col2 data-id="dom-comp"}
-   Domains (TADs)
-   Compartments
:::
:::::
:::::::
::::::::

::: {.absolute bottom="30" right="0" width="77%" height="77%"}
![](images/chromatin-intro.jpg){.absolute right=0 bottom=0 width="75%" height="75%"}
:::

::: {.tinynote .absolute bottom="32" right="5"}
Figure from [@misteli2020selforganizinggenomeprinciples]
:::
::::::::::

::: notes
chromatin must balance structure and function. nucleosomes makwa DNA compact yet accessible.
- **DNA Double Helix**: Fundamental structure of DNA.
- **Nucleosomes**: DNA wrapped around histone proteins, forming the "beads-on-a-string" structure.
- **Chromatin Fibers**: Higher-order coiling of nucleosomes for compact organization.
[1min]
:::


## Molecular Biology: Chromatin Architecture {auto-animate=true}

### Chromosome Architecture in the Nucleus

::::: columns
:::: {.column width="28%"}
-   Compact storage
-   Accessibility

### Chromatin

:::: {layout="[1]"}
::: {data-id="fiber-loop"}
-   Fibers
-   Loops
:::
::: {data-id="dom-comp"}
-   Domains (TADs)
-   Compartments
:::
::::
::::
:::::

::: {.absolute bottom="30" right="0" width="77%" height="77%" style="margin:0; padding:0;"}
![](images/chromatin-intro.jpg){.absolute width="100%" height="100%" right=0 bottom=0}
::: 

::: {.tinynote .absolute bottom="32" right="5"}
Figure from [@misteli2020selforganizinggenomeprinciples]
:::

::: notes
[45s]
:::

## Molecular Biology: Compartments {auto-animate=true}

:::: {layout="[[50, 50],[40,-60]]"}
::: {#non-fragment}

### Topologically Associating Domains (TADs)

-   Self-interacting chromatin domains that insulate regulatory interactions.\
-   Act as boundaries to regulate enhancer-promoter interactions.
:::
::: {#non-fragment2}
### A/B Compartments

-   A-Compartments: Gene-rich, active, euchromatin regions.\
-   B-Compartments: Gene-poor, inactive, heterochromatin regions.

:::

::: {.fragment .alert}

### Key differences

-   Method of inference
-   Size

:::
::::

::: {.absolute bottom="30" right="0" width="77%" height="77%"}
![](images/chromatin-intro.jpg){.absolute width="65%" height="60%" right=0 bottom=0}
:::

::: {.tinynote .absolute bottom="32" right="5"}
Figure from [@misteli2020selforganizinggenomeprinciples]
:::

::: notes
- Self-interacting chromatin domains, often referred to as TADs, play a key role in insulating regulatory interactions. They act as boundaries, ensuring that enhancers only interact with their target promoters, which helps maintain precise gene regulation and prevents unwanted cross-talk between adjacent genes. TAD boundaries are typically enriched in structural proteins (CTCF motif), and specific histone modifications (e.g. H3K4me3) marking promoter/enhancers. 
- Now, A/B compartments are large genomic regions that organize the nucleus into functional zones. A compartments are gene-rich and transcriptionally active, while B compartments are gene-poor and largely inactive. This separation allows the genome to balance accessibility for active genes with compact storage for inactive regions, optimizing nuclear function. 
- The key difference between the TADs and A/B compartments is the method of inference, where TADs are notoriously hard to call, but initially, the size was the separator: TADs were sub-megabase and A/B compartments were multiple megabases. 
[2m15s]
:::


## Hi-C (Chromosome Conformation Capture) {auto-animate=true auto-animate-id="chrom-cap"}

### How to capture chromosome conformation:


![](images/hic-schematic-cutout.png){.fragment .absolute fragment-index=1 bottom=30 right=50 height="75%" data-id="schematic"}

::: {.tinynote .absolute .fragment fragment-index=1 right=55 bottom=33}
From [@wikimedia:HiCschematic]
:::

::: notes 
*So, how is the 3D organization of DNA in the nucleus captured?*
To understand the data, we must know a bit about how the samples are generated. Let us briefly visit the wet-lab. Here, the cells are fixed and proximal DNA strands are crosslinked. 
[2m20s]
:::


## Hi-C (Chromosome Conformation Capture) {auto-animate=true auto-animate-id="chrom-cap"}

![](images/fig-hic-data-analysis.png){data-id="schematic"}

::: notes
[5m]
:::


## Hi-C (Chromosome Conformation Capture) {auto-animate=true auto-animate-id="chrom-cap"}

<!--{{< embed ../notebooks/07_various plotting.ipynb#hic-example >}}-->
<!-- add the saved .png in stead for auto-animation to work-->

![](images/hic-example.png){.absolute left=0 right=0 bottom=30 height="86%" style="margin: auto auto;" data-id="schematic"}

::: notes
Here is the heat map, showing the checkerboard pattern. Darker spots signify an interaction between two places in the genome. Dark squaresa along the diagonal would be TADs. 
But how do we get from matrix to the 1D track of compartment type?
[1m15]
:::


## Inference Methods: A/B compartments

::: {.incremental}
* [PCA]{.alert}: Captures compartmentalization by minimizing the error between observed and predicted frequencies
* [PC1]{.alert}: Reflects A/B compartment assignments, sign (+/-) indicates type (A/B).
  * [How?]{.alert}
:::

::: {#rationale .incremental layout="[-2.5,95,-2.5]" style="margin-top:-10pt;font-size:0.79em;font-family:math;color:#6f4747;"}
1. Assign each bin $x$ a compartment value $c(x)$.  
2. Define interaction score as $I(x,y) = c(x)c(y)$, with $I(x,y) > 0$ for same-compartment loci.  
3. Optimize $c(x)$ via PCA to minimize MSE between observed interactions and $c(x)c(y)$.  
4. Use GC content to orient the eigenvector, assigning loci to A ($c(x) > 0$) or B ($c(x) < 0$).  
5. Constrain MSE minimization to a genomic range (e.g., chromosome arms).  
:::

:::notes
1. **Compartment Assignment:** Each genomic bin $x$ is assigned a compartment value $c(x)$, representing whether it belongs to compartment A or B. The goal is to determine $c(x)$ for all bins
2. **Interaction Score:** The interaction between two loci, $x$ and $y$, is modeled as $I(x,y) = c(x)c(y)$. If they belong to the same compartment, their interaction score is positive ($I(x,y) > 0$), indicating enriched interactions.
3. **PCA Optimization:** Principal Component Analysis (PCA) finds $c(x)$ by minimizing the mean squared error (MSE) between observed Hi-C interaction frequencies and the predicted values $c(x)c(y)$. If the compartment pattern is strong, PCA effectively captures it.
4. **Eigenvector Orientation:** Since PCA eigenvectors are only determined up to a sign, an external biological feature—such as GC content—is used to orient the compartment assignments. This ensures that positive $c(x)$ values correspond to compartment A and negative values to compartment B.
5. **Genomic Range Restriction:** The MSE minimization can be restricted to specific genomic regions, such as chromosome arms, to focus on local compartment structures rather than genome-wide patterns.

[3m15s]
:::

## Methods {auto-animate-reset=true}

::: {.incremental style="margin-bottom:30px;"}
* Infer compartments using *HiCExplorer* or *Open2C*
* Try to reproduce the compartments from @wang2019reprogrammingmeioticchromatin in fibroblast
* Success with *Open2c*:
:::

![](images/overlay-smoothed.png){.fragment}


::: notes 
Now, I think we are ready to move on to ***my*** analysis.
First, I wanted to reproduce the compartments from Wang, but it was not possible with HiCExplorer. As I mentioned earlier, it is sometimes necessary to restrict the viewframe of the PCA, if the variance between chrom.arms is larger than within each arm. 
However, I think I succeeded with Open2C (applause). (Zoom in)
This result took 9 weeks to obtain.
[1m30]
:::


## Results: Compartments {auto-animate=true auto-animate-id="compartments"}

* PC1 compartments were obtained
  * Tissue: fibroblast, spermatogonia, pachytene spermatocyte, round spermatid, sperm
  * Resolutions: 100kb, 500kb, ps500kb
  * Viewframe: full, arms, 10Mb windows
* PC1 tracks for each tissue type: [9]{.alert} 

::: notes 
[1m35s]
:::

## Results: Compartments {.smaller auto-animate=true auto-animate-id="compartments"}

* PC1 compartments were obtained
  * Tissue: fibroblast, spermatogonia, pachytene spermatocyte, round spermatid, sperm
  * Resolutions: 100kb, 500kb, ps500kb
  * Viewframe: full, arms, 10Mb windows

::: {style="margin-top:-20pt;"}
{{< embed ../notebooks/05_rec_compartments.ipynb#e1-fullmerged-all-100kb >}}
:::

::: notes 
[2m10s]
:::

## Results: Compartments {auto-animate=true auto-animate-id="compartments"}

* PC1 compartments were obtained
  * Tissue: fibroblast, spermatogonia, pachytene spermatocyte, round spermatid, sperm
  * Resolutions: 100kb, 500kb, ps500kb
  * Viewframe: full, arms, 10Mb windows
* Define sets of intervals
  * A 200kb transition zone between A/B compartments
  * A 1bp *limit* between A/B compartments

::: notes 
[1m25s]
:::

## Genomic Intervals

### Introducing the regions of selection

* Baboons: Disproportionate parent ancestry on X
  * Either olive baboon (*P. anubis*) or hamadryas (*P. hamadryas*)
* Human: ECH90 regions
  * Extraordinary selective sweeps on X (Extended Common Haplotypes 90% peak). From @skov2023extraordinaryselectionhuman

::: {style="margin-top:-15pt;"}
{{< embed ../notebooks/07_various_plotting.ipynb#introduce-selected-regions-slides >}}
:::

::: notes
It's time to talk more about the regions of interest: baboon regions are from unplished analyses (Kasper, Erik), showing disproportionate ancestry of either olive or hamadryas baboons indicating strong selection. 
The ECH90 regions are from a published paper (dare I say it's a BiRC paper involving Kasper), identifying extraordinarily long sweeps on X. One of them double the size of the sweep associated with lactase persistence gene. Well, here they are, the regions we use. We also use a merged version of the baboons (they do not overlap).
[1m35s]
:::

## Methods: Comparing Genomic Intervals

### How to compare intervals

:::: {layout-ncol=2 .fragment .incremental}

#### Jaccard test

1. Calculate the Jaccard index between two sets of segments, $Q$ and $A$
  $$
  J(Q, A) = \frac{|Q \cap A|}{|Q \cup A|}
  $$
2. Bootstrap $Q$: $shuffle+reorder$ while conserving segment $lengths$

#### Proximity test

1. Remove overlapping segments between $Q$ and $A$
2. Calculate a proximity index between two sets of segments, $Q$ and $A$.
  $$
  P(Q, A) = \frac{1}{|Q|} \sum_{q \in Q} \min_{a \in A} d(q, a)
  $$
3. Bootstrap $Q$ in the same manner

::::

::: notes
**Jaccard:** Calculate intersection over union
**Proximity:** Calculate the mean of the distance from each segment in $Q$ to the nearest segment in $A$
[2m00s]
:::

## Results: Comparing Genomic Intervals

::: {.incremental}
* ECH90 overlaps with fibroblast and round spermatid 200kb transition zones ($p \sim 0.01$)
* What happens when only using the *limits*?
  * $Q$: full ECH, baboon limits (separate and concatenated groups)
  * $A$: PC1 limits (arms, 10Mb)
:::

::: {.fragment .border style="font-size:20pt;"}
{{< embed ../notebooks/09_defence.ipynb#tbl-test-results-uncorrected >}}
:::

::: {.absolute .alert .border .fragment style="top:232px; left:500px; border-left:2pt dotted grey; padding:20px;font-size:0.8em;background-color:rgba(255,255,255,0.9);"}
$\leftarrow 2\times 40$ tests. What about multiple testing?
:::


::: notes
When first testing the transition zones, I found overlap: ECH with fibroblast and round spermatid. Initially, I also found significant proximity, but when I updated the test, it dissapeared, sadly. I started testing the transition zones as I thought it would allow for some uncertainty in the positions, and had only small hopes for testing the *limits*.
Here, we use Let's see what came out of it (click) --> boom. \*Head scratching\*. Let's unpack. 
Still, fibroblast and round spermatid limits *overlap* the ECH regions, but the p-value is lower for the 200kb transition zone.
Additionally, spermatogonia and pachytene spermatocyte limits overlap ECH, so now all tissues except sperm overlaps ECH in their compartment limits. 
What is surprising is that in some cases, the compartment limits are both significantly proximal to the concatenated baboon set *and* overlaps the ECH90 regions. 
Also note the very low p-value for proximity between baboon limits and sperm samples.
(click) We observe 7 and 5 significant p-values out of 40 for Jaccard and Proximity tests, respectively. That is more than double the amount expected by change. But we still have to talk a bit about multiple testing. The tests are not independent as the different viewframes still originate from the same data, and the compartments called from the two viewframes are highly similar. That means that multiple testing correction gets complicated, and naively correcting the p-values is too conservative. However, I made the corrections to get an idea of the impact of multiple tests. Here is what I got: (click)
[6m00s] 
:::

 
## Results: Comparing Genomic Intervals

### Multiple testing correction

::: {layout="[[1,3],[1,3]]" style="font-size:20pt;"}
[<br>Uncorrected:]{style="margin-top:20pt;"}

{{< embed ../notebooks/09_defence.ipynb#tbl-test-results-uncorrected >}}

[<br>Benjamini-Hochberg (FDR) corrected]{style="margin-top:20pt;"}

{{< embed ../notebooks/09_defence.ipynb#tbl-test-results-BH-corrected >}}

:::

::: notes
Now, only the baboon regions are significantly proximal to sperm, but there is still overlap between ECH and the limits of all tissue types except sperm.
So, even with a multiple testing correction (that I deem too conservative), there are still a lot of correlation between chromatin compartments in macaque in both fibroblasts and through spermatogenesis, either by intersection or by edge/limit proximity. 
[40s]
:::

## Implications

* Assuming the results are real
* What does it *mean* that the regions correlate
  * The ECH paper: hinting to possible mechanisms aiding a fitness-based model
  * Structural features could aid in preserving certain genetic elements
  * Some tissues are connected to both ECH and baboon regions
  * Does fibroblast rule out the implication in spermatogenesis?
* Macaque is often used as an outgroup, but not here
  * Reproduce in great ape data



::: notes 
Let's for a moment assume that this explorative analysis is sound, and the regions are actually correlating. *What does it mean* that they do? 
Well, an extraordinary selection coefficient has been reported for segments in ECH, and one that is more than double the size of the sweep associated with *lactase persistence*.
They hint that other mechanisms could aid the apparent selection by structurally linking regions across the chromosome. 
Then, the compartment limits were in some tissues associated with both ECH regions and baboon regions, indicating some indirect link. Maybe, some regions are more prone to selection as a result of structural features?
Additionally, fibroblast showed overlap to ECH - does that rule out the role of spermatogenesis?
Another point that I've avoided until now is: macaque is often used as an outgroup to great apes, but here, we find strong correlations between the elements. Can we strengthen our results by reproducing the Hi-C analysis in other (more closely related) species?
[2m10s]
:::



## Reproducibility

:::: {columns}
::: {.column width="50%"}

* Can we reproduce in other organisms?
  * The question I have been waiting for

![](images/sponsors.png)

:::

::: {.column width="50%"}
::: {#tbl-reproducibility .striped style="font-size:17pt"}
| Tool | Description |
|-------------:|:---------------------------------------------------------|
| Jupyter | Interactive coding environment for analysis and development (notebooks are natively rendered with Quarto) |
| Quarto | A Quarto Manuscript project nested inside a Quarto Book for rendering html (website) and PDF (manuscript) from Markdown via Pandoc. Supports direct embedding of output from Jupyter Notebook cells (plots, tables). |
| Conda | For managing software requirements and dependency versions reproducibly. |
| git | Version control and `gh-pages` branch for automated render of Quarto project |
| GitHub | Action was triggered `on push` to render the project and host on [munch-group.org](https://munch-group.org/hic-spermatogenesis/) |
| *gwf* | Workflow manager to automate the analysis on a HPC cluster, wrapped in Python code. `workflow.py` currently does everything from `.fastq` to `.cool`, but notebooks can be set to run sequentially as part of the workflow as well. |

Overview of the tools used for reproducibility of this thesis.
:::
:::

::::

::: notes 
[2m00s]
:::


## Conclusive thoughts and perspectives

* Previous results
  * No such comparison has been performed before (I might be deceived)
* Find a way to weight the compartments based on metadata
* We should investigate TADs as well
* We should analyze the genes at the coordinates of correlation 
* Chromosome X is an obvious target
* I have only looked at *intra*-chromosomal interactions

::: notes 

:::

# Extra slides

## Inference Methods: TADs 

Call TADs using interaction data summarized into a 1D genomic profile

::: {.incremental}
* [Insulation Score]{.alert}: Sliding window along the main diagonal
* [Boundaries]{.alert}: Find local minima along the genome
* [Define threshold]{.alert}: define a threshold for *peak prominence*
  * Borrow *Otsu* or *Li* boundaries from image analysis
  * Optional: use CTCF enrichment to guide the thresholding
:::

::: {.fragment}
*Many methods exist for calling TADs $\rightarrow$ different results from different approaches*
:::

:::{.fragment style="font-size:0.8em;"}
> *"[...] an example of how adjacent boundaries calculated with cooltools* 
> *can specify a set of intervals that could be analyzed as TADs, [...]"*
> <br> [---cooltools documentation]{style="margin-left:50pt;font-style:italic"}
:::

::: notes
* **insulation score:** draw a sliding window on the diagonal of a matrix, draw insulation profile
* **boundaries:** draw local minima, 
* **threshold:** draw and explain prominence, mark strong/weak boundaries
  we can borrow thresholding methods from image analysis
  as CTCF is shown to be enriched at TAD boundaries, we can use CTCF enrichment to guide the thresholds as well
* There is not concensus about the correct method
* Healthy vagueness
:::

## References

::: {#refs}
:::