<!DOCTYPE html>
<html lang="en"><head>
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js" integrity="sha512-bLT0Qm9VnAYZDflyKcBaQ2gg0hSYNQrJ8RilYldYQ1FxQYoCLtUjuuRuZo+fjqhx/qtq/1itJ0C2ejDxltZVFg==" crossorigin="anonymous"></script><script src="index_files/libs/clipboard/clipboard.min.js"></script>
<script src="index_files/libs/quarto-html/tabby.min.js"></script>
<script src="index_files/libs/quarto-html/popper.min.js"></script>
<script src="index_files/libs/quarto-html/tippy.umd.min.js"></script>
<link href="index_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="index_files/libs/quarto-html/light-border.css" rel="stylesheet">
<link href="index_files/libs/quarto-html/quarto-syntax-highlighting-e26003cea8cd680ca0c55a263523d882.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.6.39">

  <meta name="author" content="Søren Jørgensen">
  <meta name="author" content="Kasper Munch (Supervisor)">
  <meta name="dcterms.date" content="2025-01-02">
  <title>Project Progress Day: Msc. Project 2024</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="index_files/libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="index_files/libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
  </style>
  <link rel="stylesheet" href="index_files/libs/revealjs/dist/theme/quarto-e6ec3429f00f3dae4ab744fc8c423248.css">
  <link href="index_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="index_files/libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="index_files/libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="index_files/libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" integrity="sha512-c3Nl8+7g4LMSTdrm621y7kf9v3SDPnhxLNhcjFJbKECVnmZHTdo+IRO05sNLTH/D3vA6u1X32ehoLC7WFVdheg==" crossorigin="anonymous"></script>
  
  <script type="application/javascript">define('jquery', [],function() {return window.jQuery;})</script>
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" data-notes="Hi, I’m Søren. I’m doing my MSc thesis with Kasper about Chromatin Structures and Selection." class="quarto-title-block center">
  <h1 class="title">Project Progress Day: Msc. Project 2024</h1>
  <p class="subtitle">Correlating Chromatin Compartments with Selection</p>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div class="quarto-title-author-name">
Søren Jørgensen 
</div>
        <p class="quarto-title-affiliation">
            Stud. MSc. Bioinformatics, BiRC
          </p>
    </div>
<div class="quarto-title-author">
<div class="quarto-title-author-name">
Kasper Munch (Supervisor) 
</div>
        <p class="quarto-title-affiliation">
            Associate Professor, BiRC
          </p>
    </div>
</div>

  <p class="date">January 2, 2025</p>
</section><section id="TOC">
<nav role="doc-toc"> 
<h2 id="toc-title">Overview</h2>
<ul>
<li><a href="#/problem-1-chromatin-compartments" id="/toc-problem-1-chromatin-compartments">Problem 1: Chromatin Compartments</a>
<ul>

<li><a href="#/d-organization-of-chromatin-into-compartments" id="/toc-d-organization-of-chromatin-into-compartments">3D organization of Chromatin into Compartments</a></li>
<li><a href="#/hi-c-pipeline" id="/toc-hi-c-pipeline">Hi-C pipeline</a></li>
<li><a href="#/hi-c-pipeline-1" id="/toc-hi-c-pipeline-1">Hi-C pipeline</a></li>
<li><a href="#/reconstructing-figures-from-wang_reprogramming_2019" id="/toc-reconstructing-figures-from-wang_reprogramming_2019">Reconstructing figures from <span class="citation" data-cites="wang_reprogramming_2019">(<strong>wang_reprogramming_2019?</strong>)</span></a></li>
<li><a href="#/reconstructing-figures-from-wang_reprogramming_2019-1" id="/toc-reconstructing-figures-from-wang_reprogramming_2019-1">Reconstructing figures from <span class="citation" data-cites="wang_reprogramming_2019">(<strong>wang_reprogramming_2019?</strong>)</span></a></li>
<li><a href="#/partition-the-genome" id="/toc-partition-the-genome">Partition the genome</a></li>
<li><a href="#/partition-the-genome-and-smooth" id="/toc-partition-the-genome-and-smooth">Partition the genome and smooth</a></li>
<li><a href="#/check-with-the-matrix" id="/toc-check-with-the-matrix">Check with the matrix</a></li>
</ul></li>
<li><a href="#/problem-2-correlating-with-regions-of-interest" id="/toc-problem-2-correlating-with-regions-of-interest">Problem 2: Correlating with regions of interest</a>
<ul>
<li><a href="#/what-regions" id="/toc-what-regions">What regions</a></li>
<li><a href="#/macaque-vs-baboon-vs-human" id="/toc-macaque-vs-baboon-vs-human"><em>Macaque</em> vs <em>Baboon</em> vs <em>human</em></a></li>
<li><a href="#/references" id="/toc-references">References</a></li>
</ul></li>
</ul>
</nav>
</section>
<section>
<section id="problem-1-chromatin-compartments" class="title-slide slide level1 center" data-background-image="images/chromatin-intro.jpg" data-background-opacity="0.2">
<h1>Problem 1: Chromatin Compartments</h1>

</section>
<section id="section" class="slide level2" data-background-image="images/chromatin-intro.jpg">
<h2></h2>
<aside class="notes">
<p>Ok, very briefly: The DNA is organized into higher-order structures by the histones/nucleosomes. This illustration is a nice view of the higher-order structures of DNA inside the nucleus; Nucleosome, Chromatin fiber, Chromatin loops, Chromatin domains, and Chromatin compartments</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="d-organization-of-chromatin-into-compartments" class="slide level2">
<h2>3D organization of Chromatin into Compartments</h2>
<div class="columns">
<div class="column" style="width:40%;">
<ul>
<li>Data
<ul>
<li>Special library construction</li>
<li>Sequencing (Hi-C)</li>
</ul></li>
</ul>
</div><div class="column" style="width:60%;">
<p><img data-src="images/hic-intro.jpg"></p>
</div></div>
<aside class="notes">
<p>To infer this 3-dimensional structure, we need a specialized sequence library, which is made by cross-linking close-proximity DNA (at a specific site), digestion, proximity-ligation, amplification. Now the amplified fragments are pairs of sequences that are close in physical space, but not in necessarily in bp position. All this lab-work is done by others, and now it’s our turn.</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="hi-c-pipeline" class="slide level2" data-auto-animate="true">
<h2 data-id="quarto-animate-title">Hi-C pipeline</h2>
<div class="columns">
<div class="column" style="width:80%;">
<p><img data-src="images/HiCschematic.png"></p>
</div><div class="column" style="width: 20%; text-align: right; margin-left: auto;">
<p><strong>The <em>plaid</em> pattern is reflected by the E1</strong></p>
<div class="{.data-id='boxright'}">
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/03_compartments.ipynb" data-notebook-title="Compartments Analysis" data-notebook-cellid="cell-fig-matrix_e1_500kb">
<div id="cell-fig-matrix_e1_500kb" class="cell" data-execution_count="17">
<div class="cell-output cell-output-display" data-execution_count="17">
<div id="fig-matrix_e1_500kb" class="quarto-float quarto-figure quarto-figure-center">
<figure class="quarto-float quarto-float-fig">
<div aria-describedby="fig-matrix_e1_500kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img data-src="index_files/figure-revealjs/..-notebooks-03_compartments-fig-matrix_e1_500kb-output-1.png">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-matrix_e1_500kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1: chrX interaction matrix with E1 eigenvector values in 500kb resolution. The sign change of the E1 is overlayed as thin black lines, making it more easily interpretable. We should qualitatively determine how well the E1 values capture the <em>plaid</em> pattern in the matrix.
</figcaption>
</figure>
</div>
</div>
</div>
</div>
</div>
</div></div>
<aside class="notes">
<p>When the pairs are sequenced, we map (align) them back onto the reference genome (to get a coordinate for each of the pairs). After filtering out poorly mapped and unmapped reads and matrix-balancing, we construct a heat map at some chosen resolution. We should see <em>plaid</em> patterns like these, which indicate that regions of the genome interact more closely than others, marked by a sharp increase in interaction frequency.</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="hi-c-pipeline-1" class="slide level2" data-auto-animate="true">
<h2 data-id="quarto-animate-title">Hi-C pipeline</h2>
<div class="columns">
<div class="column" style="width:38%;">
<p><img data-src="images/HiCschematic.png"></p>
</div><div class="column" style="width: 62%; text-align: right; margin-left: auto;">
<div class="{.data-id='boxright'}">
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/03_compartments.ipynb" data-notebook-title="Compartments Analysis" data-notebook-cellid="cell-fig-matrix_e1_500kb">
<div id="cell-fig-matrix_e1_500kb" class="cell" data-execution_count="17">
<div class="cell-output cell-output-display" data-execution_count="17">
<div id="fig-matrix_e1_500kb" class="quarto-float quarto-figure quarto-figure-center">
<figure class="quarto-float quarto-float-fig">
<div aria-describedby="fig-matrix_e1_500kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img data-src="index_files/figure-revealjs/..-notebooks-03_compartments-fig-matrix_e1_500kb-output-1.png">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-matrix_e1_500kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2: chrX interaction matrix with E1 eigenvector values in 500kb resolution. The sign change of the E1 is overlayed as thin black lines, making it more easily interpretable. We should qualitatively determine how well the E1 values capture the <em>plaid</em> pattern in the matrix.
</figcaption>
</figure>
</div>
</div>
</div>
</div>
</div>
</div></div>
<aside class="notes">
<p>Now, it has been shown that the first eigenvector (the vector of PC1) can well predict these compartments, and we label them ‘A’ for active and ‘B’ for inactive compartments. We use (binned) GC content to <em>phase</em> or calibrate the sign. It is convention to eyeball how well we capture the plaid pattern as this point.</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="reconstructing-figures-from-wang_reprogramming_2019" class="slide level2">
<h2>Reconstructing figures from <span class="citation" data-cites="wang_reprogramming_2019">(<a href="" role="doc-biblioref" onclick="return false;"><strong>wang_reprogramming_2019?</strong></a>)</span></h2>
<div>
<ul>
<li class="fragment"><span class="citation" data-cites="wang_reprogramming_2019">(<a href="" role="doc-biblioref" onclick="return false;"><strong>wang_reprogramming_2019?</strong></a>)</span> identified compartments in <em>rhesus macaque</em> in five different stages of spermatogenesis:
<ul>
<li class="fragment">fibroblast, spermatogonia, pachytene spermatocyte, round spermatids, and sperm</li>
</ul></li>
<li class="fragment">The edges of some compartments look suspiciously close to some of the regions of weirdly selected genes of <em>papio anubis</em> (olive baboons)</li>
</ul>
</div>
<div class="fragment fade-in" data-id="box1">
<div style="top-margin: 0px; margin-left:auto; margin-right:auto;">
<figure>
<center>
<figcaption>
Wang2019 PC1 on chrX
</figcaption>
<img src="../figures/E1_chrX_fib_wang.png" alt="E1_chrX_fib_wang" style="height: 150px; width: 880px;">
</center>
</figure>
</div>
<div style="top-margin: 0px; margin-left:auto; margin-right:auto; height: 150px;">
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/02_open2c_framework.ipynb" data-notebook-title="Open2C framework" data-notebook-cellid="cell-chrX-e1-plot">
<div id="cell-chrX-e1-plot" class="cell" data-fig-height="2" data-execution_count="18">
<div class="cell-output cell-output-display">
<div id="chrx-e1-plot" class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="index_files/figure-revealjs/..-notebooks-02_open2c_framework-chrx-e1-plot-output-1.png"></p>
<figcaption>100kb binned E1 eigenvector values for chrX. Freshly calculated from the cooler file.</figcaption>
</figure>
</div>
</div>
</div>
</div>
</div>
</div>
</section>
<section id="reconstructing-figures-from-wang_reprogramming_2019-1" class="slide level2">
<h2>Reconstructing figures from <span class="citation" data-cites="wang_reprogramming_2019">(<a href="" role="doc-biblioref" onclick="return false;"><strong>wang_reprogramming_2019?</strong></a>)</span></h2>
<ul>
<li><span class="citation" data-cites="wang_reprogramming_2019">(<a href="" role="doc-biblioref" onclick="return false;"><strong>wang_reprogramming_2019?</strong></a>)</span> identified compartments in <em>rhesus macaque</em> in five different stages of spermatogenesis:
<ul>
<li>fibroblast, spermatogonia, pachytene spermatocyte, round spermatids, and sperm</li>
</ul></li>
<li>The edges of some compartments look suspiciously close to some of the regions of weirdly selected genes of <em>papio anubis</em> (olive baboons)</li>
</ul>
<div data-id="box1">
<div class="r-stretch">
<p><img data-src="images/overlay-smoothed.png" style="height:200px; width:1200px"></p>
</div>
</div>
<aside class="notes">
<p>The initial results were good for fibroblasts, but they were not so good for the rest of the cell types (next slide)</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="partition-the-genome" class="slide level2">
<h2>Partition the genome</h2>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/03_compartments.ipynb" data-notebook-title="Compartments Analysis" data-notebook-cellid="cell-e1-fullmerged-all-500kb">
<div id="cell-e1-fullmerged-all-500kb" class="cell" data-execution_count="31">
<div class="cell-output cell-output-display column-page-inset">
<div id="e1-fullmerged-all-500kb" class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="index_files/figure-revealjs/..-notebooks-03_compartments-e1-fullmerged-all-500kb-output-1.png"></p>
<figcaption>E1 eigenvector values for all merged samples at 500kb resolution. Left: Chromosome-arm restricted E1. Right: 10Mb window restricted E1.</figcaption>
</figure>
</div>
</div>
</div>
</div>
</section>
<section id="partition-the-genome-and-smooth" class="slide level2">
<h2>Partition the genome and smooth</h2>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/03_compartments.ipynb" data-notebook-title="Compartments Analysis" data-notebook-cellid="cell-e1-fullmerged-all-100kb-5smoothed">
<div id="cell-e1-fullmerged-all-100kb-5smoothed" class="cell" data-execution_count="40">
<div class="cell-output cell-output-display column-page-inset">
<div id="e1-fullmerged-all-100kb-5smoothed" class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="index_files/figure-revealjs/..-notebooks-03_compartments-e1-fullmerged-all-100kb-5smoothed-output-1.png"></p>
<figcaption>E1 eigenvector values for all merged samples at 100kb resolution. Left: Chromosomes (not restricted) Middle: Chromosome-arm restricted E1. Right: 10Mb window restricted E1.Values are smoothed with a sliding window of 5 bins, step size 1 bin.</figcaption>
</figure>
</div>
</div>
</div>
</div>
<aside class="notes">
<p>It seems it has a big effect how we partition the genome when calculating the E1. What it actually means (biologically), I don’t know. Do we remove noise or actual information?</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="check-with-the-matrix" class="slide level2 nostretch">
<h2>Check with the matrix</h2>
<div class="columns">
<div class="column fragment" style="width:30%;">
<ul>
<li>As stated, we have to eyeball how well the E1 captures the plaid pattern</li>
<li>How does it look?</li>
</ul>
</div><div class="column fragment" style="width:70%;">
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/03_compartments.ipynb" data-notebook-title="Compartments Analysis" data-notebook-cellid="cell-e1-matrix-500kb-full-arms-10mb-round_spermatid">
<div id="cell-e1-matrix-500kb-full-arms-10mb-round_spermatid" class="cell" data-execution_count="32">
<div class="cell-output cell-output-display">
<div id="e1-matrix-500kb-full-arms-10mb-round_spermatid" class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="index_files/figure-revealjs/..-notebooks-03_compartments-e1-matrix-500kb-full-arms-10mb-round_spermatid-output-1.png"></p>
<figcaption>E1 eigenvector values for all merged samples at 500kb resolution for round spermatid, as well as the interaction matrix. E1 was restricted to top: Full-chromosome, middle: Chromosome-arms, bottom: 10Mb window.</figcaption>
</figure>
</div>
</div>
</div>
</div>
</div></div>
</section></section>
<section>
<section id="problem-2-correlating-with-regions-of-interest" class="title-slide slide level1 center" data-background-image="https://upload.wikimedia.org/wikipedia/commons/3/35/Olive_baboon_Ngorongoro.jpg" data-background-opacity="0.3">
<h1>Problem 2: Correlating with regions of interest</h1>
<div class="footer">
<p>By <a href="https://en.wikipedia.org/wiki/User:Muhammad_Mahdi_Karim" class="extiw" title="w:User:Muhammad Mahdi Karim">Muhammad Mahdi Karim</a> - <span class="int-own-work" lang="en">Own work</span>, <a href="http://www.gnu.org/licenses/old-licenses/fdl-1.2.html" title="GNU Free Documentation License 1.2">GFDL 1.2</a>, <a href="https://commons.wikimedia.org/w/index.php?curid=11193550">Link</a></p>
</div>
</section>
<section id="what-regions" class="slide level2">
<h2>What regions</h2>
<div>
<ul>
<li class="fragment">Some genes behave weirdly during spermatogenesis</li>
<li class="fragment">Why are they selected (seemingly) against fitness?</li>
</ul>
</div>
</section>
<section id="macaque-vs-baboon-vs-human" class="slide level2">
<h2><em>Macaque</em> vs <em>Baboon</em> vs <em>human</em></h2>
<div class="&nbsp;{.incremental}">
<ul>
<li>Do they correlate?</li>
<li>Do we have to work on sperm samples (laborious - hard to get data)</li>
</ul>
</div>
</section>
<section id="references" class="slide level2">
<h2>References</h2>

</section></section>
    </div>
  <div class="quarto-auto-generated-content" style="display: none;">
<div class="footer footer-default">

</div>
</div></div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="index_files/libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="index_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="index_files/libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="index_files/libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="index_files/libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="index_files/libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="index_files/libs/revealjs/plugin/notes/notes.js"></script>
  <script src="index_files/libs/revealjs/plugin/search/search.js"></script>
  <script src="index_files/libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="index_files/libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'jumpToSlide': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleScrollView(event)\"><kbd>r</kbd> Scroll View Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"5\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'smaller': true,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: 'c/t',

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'none',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1080,

        height: 720,

        // Factor of the display size that should remain empty around the content
        margin: 5.0e-2,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    <script id="quarto-html-after-body" type="application/javascript">
    window.document.addEventListener("DOMContentLoaded", function (event) {
      const toggleBodyColorMode = (bsSheetEl) => {
        const mode = bsSheetEl.getAttribute("data-mode");
        const bodyEl = window.document.querySelector("body");
        if (mode === "dark") {
          bodyEl.classList.add("quarto-dark");
          bodyEl.classList.remove("quarto-light");
        } else {
          bodyEl.classList.add("quarto-light");
          bodyEl.classList.remove("quarto-dark");
        }
      }
      const toggleBodyColorPrimary = () => {
        const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
        if (bsSheetEl) {
          toggleBodyColorMode(bsSheetEl);
        }
      }
      toggleBodyColorPrimary();  
      const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
      tabsets.forEach(function(tabset) {
        const tabby = new Tabby('#' + tabset.id);
      });
      const isCodeAnnotation = (el) => {
        for (const clz of el.classList) {
          if (clz.startsWith('code-annotation-')) {                     
            return true;
          }
        }
        return false;
      }
      const onCopySuccess = function(e) {
        // button target
        const button = e.trigger;
        // don't keep focus
        button.blur();
        // flash "checked"
        button.classList.add('code-copy-button-checked');
        var currentTitle = button.getAttribute("title");
        button.setAttribute("title", "Copied!");
        let tooltip;
        if (window.bootstrap) {
          button.setAttribute("data-bs-toggle", "tooltip");
          button.setAttribute("data-bs-placement", "left");
          button.setAttribute("data-bs-title", "Copied!");
          tooltip = new bootstrap.Tooltip(button, 
            { trigger: "manual", 
              customClass: "code-copy-button-tooltip",
              offset: [0, -8]});
          tooltip.show();    
        }
        setTimeout(function() {
          if (tooltip) {
            tooltip.hide();
            button.removeAttribute("data-bs-title");
            button.removeAttribute("data-bs-toggle");
            button.removeAttribute("data-bs-placement");
          }
          button.setAttribute("title", currentTitle);
          button.classList.remove('code-copy-button-checked');
        }, 1000);
        // clear code selection
        e.clearSelection();
      }
      const getTextToCopy = function(trigger) {
          const codeEl = trigger.previousElementSibling.cloneNode(true);
          for (const childEl of codeEl.children) {
            if (isCodeAnnotation(childEl)) {
              childEl.remove();
            }
          }
          return codeEl.innerText;
      }
      const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
        text: getTextToCopy
      });
      clipboard.on('success', onCopySuccess);
      if (window.document.getElementById('quarto-embedded-source-code-modal')) {
        const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
          text: getTextToCopy,
          container: window.document.getElementById('quarto-embedded-source-code-modal')
        });
        clipboardModal.on('success', onCopySuccess);
      }
        var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
        var mailtoRegex = new RegExp(/^mailto:/);
          var filterRegex = new RegExp('/' + window.location.host + '/');
        var isInternal = (href) => {
            return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
        }
        // Inspect non-navigation links and adorn them if external
     	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
        for (var i=0; i<links.length; i++) {
          const link = links[i];
          if (!isInternal(link.href)) {
            // undo the damage that might have been done by quarto-nav.js in the case of
            // links that we want to consider external
            if (link.dataset.originalHref !== undefined) {
              link.href = link.dataset.originalHref;
            }
          }
        }
      function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
        const config = {
          allowHTML: true,
          maxWidth: 500,
          delay: 100,
          arrow: false,
          appendTo: function(el) {
              return el.closest('section.slide') || el.parentElement;
          },
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start',
        };
        if (contentFn) {
          config.content = contentFn;
        }
        if (onTriggerFn) {
          config.onTrigger = onTriggerFn;
        }
        if (onUntriggerFn) {
          config.onUntrigger = onUntriggerFn;
        }
          config['offset'] = [0,0];
          config['maxWidth'] = 700;
        window.tippy(el, config); 
      }
      const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
      for (var i=0; i<noterefs.length; i++) {
        const ref = noterefs[i];
        tippyHover(ref, function() {
          // use id or data attribute instead here
          let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
          try { href = new URL(href).hash; } catch {}
          const id = href.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note) {
            return note.innerHTML;
          } else {
            return "";
          }
        });
      }
      const findCites = (el) => {
        const parentEl = el.parentElement;
        if (parentEl) {
          const cites = parentEl.dataset.cites;
          if (cites) {
            return {
              el,
              cites: cites.split(' ')
            };
          } else {
            return findCites(el.parentElement)
          }
        } else {
          return undefined;
        }
      };
      var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
      for (var i=0; i<bibliorefs.length; i++) {
        const ref = bibliorefs[i];
        const citeInfo = findCites(ref);
        if (citeInfo) {
          tippyHover(citeInfo.el, function() {
            var popup = window.document.createElement('div');
            citeInfo.cites.forEach(function(cite) {
              var citeDiv = window.document.createElement('div');
              citeDiv.classList.add('hanging-indent');
              citeDiv.classList.add('csl-entry');
              var biblioDiv = window.document.getElementById('ref-' + cite);
              if (biblioDiv) {
                citeDiv.innerHTML = biblioDiv.innerHTML;
              }
              popup.appendChild(citeDiv);
            });
            return popup.innerHTML;
          });
        }
      }
    });
    </script>
    

</body></html>