<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.40">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Thoughts and Prayers – Søren's MSc project about Hi-C and spermatogenesis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js" integrity="sha512-bLT0Qm9VnAYZDflyKcBaQ2gg0hSYNQrJ8RilYldYQ1FxQYoCLtUjuuRuZo+fjqhx/qtq/1itJ0C2ejDxltZVFg==" crossorigin="anonymous"></script><script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../results/Readme.html" rel="next">
<link href="../workflow.html" rel="prev">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-549806ee2085284f45b00abea8c6df48.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap-6a801e7181b8b24ef9cb8ae289dd8d09.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script src="../site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="../site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="../site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://hypothes.is/embed.js"></script>
<script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" integrity="sha512-c3Nl8+7g4LMSTdrm621y7kf9v3SDPnhxLNhcjFJbKECVnmZHTdo+IRO05sNLTH/D3vA6u1X32ehoLC7WFVdheg==" crossorigin="anonymous"></script>

<script type="application/javascript">define('jquery', [],function() {return window.jQuery;})</script>


</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../reports/thoughts.html">Reports</a></li><li class="breadcrumb-item"><a href="../reports/thoughts.html"><span class="chapter-title">Thoughts and Prayers</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">Søren’s MSc project about Hi-C and spermatogenesis</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/munch-group/hic-spermatogenesis" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Project</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">Notebooks</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/01_hicexplorer.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">HiCExplorer using macaque data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/02_open2c_framework.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Open2C framework</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/03_compartments.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Compartments Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/04_genomicintervals.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">E1 vs.&nbsp;ECH</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/05_rec_compartments.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Compartments Analysis w. recommended parameters</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/06_rec_genomicintervals.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">E1 vs.&nbsp;ECH (recommended params)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/07_various_plotting.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Various Comparisons</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">Workflows</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../workflow.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">master_workflow</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true">
 <span class="menu-text">Reports</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../reports/thoughts.html" class="sidebar-item-text sidebar-link active"><span class="chapter-title">Thoughts and Prayers</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../results/Readme.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Result files</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true">
 <span class="menu-text">Slides</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../slides/index.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Project Progress Day: Msc. Project 2024</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="true">
 <span class="menu-text">Thesis</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../thesis/index.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Chromatin Compartments and Selection on X</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#notes-on-methods" id="toc-notes-on-methods" class="nav-link active" data-scroll-target="#notes-on-methods">Notes on methods</a></li>
  <li><a href="#notes-on-quarto" id="toc-notes-on-quarto" class="nav-link" data-scroll-target="#notes-on-quarto">Notes on Quarto</a></li>
  <li><a href="#notes-on-hicexplorer" id="toc-notes-on-hicexplorer" class="nav-link" data-scroll-target="#notes-on-hicexplorer">Notes on <code>HiCExplorer</code></a>
  <ul>
  <li><a href="#plotting-.cool-files-with-h5py-and-pybigwig" id="toc-plotting-.cool-files-with-h5py-and-pybigwig" class="nav-link" data-scroll-target="#plotting-.cool-files-with-h5py-and-pybigwig">Plotting .cool files with <code>h5py</code> and <code>pyBigWig</code></a></li>
  </ul></li>
  <li><a href="#notes-on-open2c-framework" id="toc-notes-on-open2c-framework" class="nav-link" data-scroll-target="#notes-on-open2c-framework">Notes on Open2C framework</a>
  <ul>
  <li><a href="#pairtools" id="toc-pairtools" class="nav-link" data-scroll-target="#pairtools">pairtools</a></li>
  <li><a href="#cooler" id="toc-cooler" class="nav-link" data-scroll-target="#cooler">cooler</a>
  <ul class="collapse">
  <li><a href="#update-08-10-2024" id="toc-update-08-10-2024" class="nav-link" data-scroll-target="#update-08-10-2024">Update: 08-10-2024</a></li>
  <li><a href="#update-10-10-2024" id="toc-update-10-10-2024" class="nav-link" data-scroll-target="#update-10-10-2024">Update 10-10-2024</a></li>
  <li><a href="#compartments" id="toc-compartments" class="nav-link" data-scroll-target="#compartments">Compartments</a></li>
  <li><a href="#compartments-500kb-resolution-looks-smoother" id="toc-compartments-500kb-resolution-looks-smoother" class="nav-link" data-scroll-target="#compartments-500kb-resolution-looks-smoother">Compartments 500kb resolution looks smoother</a></li>
  </ul></li>
  <li><a href="#compartments-on-all-cell-types-by-merging-all-samples-from-each-type" id="toc-compartments-on-all-cell-types-by-merging-all-samples-from-each-type" class="nav-link" data-scroll-target="#compartments-on-all-cell-types-by-merging-all-samples-from-each-type">Compartments on all cell types (By merging all samples from each type)</a></li>
  </ul></li>
  <li><a href="#calculating-the-eigenvectors-again" id="toc-calculating-the-eigenvectors-again" class="nav-link" data-scroll-target="#calculating-the-eigenvectors-again">Calculating the eigenvectors again</a></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.com/munch-group/hic-spermatogenesis/edit/main/reports/thoughts.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/munch-group/hic-spermatogenesis/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li><li><a href="https://github.com/munch-group/hic-spermatogenesis/blob/main/reports/thoughts.qmd" class="toc-action"><i class="bi empty"></i>View source</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content page-columns page-full" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../reports/thoughts.html">Reports</a></li><li class="breadcrumb-item"><a href="../reports/thoughts.html"><span class="chapter-title">Thoughts and Prayers</span></a></li></ol></nav>
<div class="quarto-title">
<h1 class="title"><span class="chapter-title">Thoughts and Prayers</span></h1>
</div>



<div class="quarto-title-meta">

    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">January 21, 2025 <br> 13:22:42 (UTC +00:00)</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<p><em>Here goes my thoughts and frustrations about the tools, the results, or the project in general.</em></p>
<section id="notes-on-methods" class="level2">
<h2 class="anchored" data-anchor-id="notes-on-methods">Notes on methods</h2>
<ul>
<li><span class="citation" data-cites="wang2019reprogrammingmeioticchromatin">(<a href="../thesis/index.html#ref-wang2019reprogrammingmeioticchromatin" role="doc-biblioref">Wang et al. 2019</a>)</span> uses a sliding window approach <span class="citation" data-cites="dixon2015chromatinarchitecturereorganization">(<a href="../thesis/index.html#ref-dixon2015chromatinarchitecturereorganization" role="doc-biblioref">Dixon et al. 2015</a>)</span> to calculate the obs/exp matrices for the PCA. I will try that next, and work within the <code>cooler</code>/<code>cooltools</code> framework which has a Python API and is thus more suitable to run in Jupyter Notebooks.</li>
<li><code>HiCExplorer</code> use a genome-wide contact-distance-based normalization method (<code>dist_norm</code>) to generate <em>obs/exp</em> matrices for PCA (similar to <span class="citation" data-cites="aiden2009comprehensivemappinglongrange">(<a href="../thesis/index.html#ref-aiden2009comprehensivemappinglongrange" role="doc-biblioref"><strong>aiden2009comprehensivemappinglongrange?</strong></a>)</span>) that may introduce biases. I don’t know why, but it looks more like <span class="citation" data-cites="wang2019reprogrammingmeioticchromatin">(<a href="../thesis/index.html#ref-wang2019reprogrammingmeioticchromatin" role="doc-biblioref">Wang et al. 2019</a>)</span> when subtracting the running average [triangular windows, <em>Kasper</em> did this] from the computed PCs.</li>
</ul>
</section>
<section id="notes-on-quarto" class="level2">
<h2 class="anchored" data-anchor-id="notes-on-quarto">Notes on Quarto</h2>
<p>I feel like i have the Markdown/Quarto thing under control, which is pretty nice. It is going to be really cool when actually starting to write the manus (I think).</p>
</section>
<section id="notes-on-hicexplorer" class="level2">
<h2 class="anchored" data-anchor-id="notes-on-hicexplorer">Notes on <code>HiCExplorer</code></h2>
<p>It feels very old school and unnecessary that I can’t just output the <code>HiCExplorer</code> commands [plots] to the Jupyter display. I <em>have</em> to save the plots to <em>.png</em>s, which is flooding my <code>figures</code> directory. I should probably clean it up.</p>
<p>I have created plots to mimic the <span class="citation" data-cites="wang2019reprogrammingmeioticchromatin">(<a href="../thesis/index.html#ref-wang2019reprogrammingmeioticchromatin" role="doc-biblioref">Wang et al. 2019</a>)</span> for the <code>chr12</code> as well as <code>chrX</code>. They don’t compare very well with Wang2019, and the both PC1 and PC2 plot looks like they’re highly biased (only intersects 0 one time). PC3 looks more reasonable, but I don’t know if it is reasonable at all to use the 3rd PC in stead of the 1st. I guess it means that the compartmentalization is not the most variable feature, and there are some biases that we have to get rid of. That, or the X chromosome is divided only into two compartments in fibroblasts (unlikely).</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/01_hicexplorer.ipynb" data-notebook-title="HiCExplorer using macaque data" data-notebook-cellid="cell-chrx-50kb-pca">
<div id="cell-chrx-50kb-pca" class="cell" data-execution_count="8">
<div class="cell-output cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><a href="thoughts_files/figure-html/..-notebooks-01_hicexplorer-chrx-50kb-pca-output-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-1" title="The X chromosome in 50kb bins and the first 3 PCs below. Made with HicExplorer."><img src="thoughts_files/figure-html/..-notebooks-01_hicexplorer-chrx-50kb-pca-output-1.png" class="img-fluid figure-img" alt="The X chromosome in 50kb bins and the first 3 PCs below. Made with HicExplorer."></a></p>
<figcaption>The X chromosome in 50kb bins and the first 3 PCs below. Made with HicExplorer.</figcaption>
</figure>
</div>
</div>
</div>
</div>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/01_hicexplorer.ipynb" data-notebook-title="HiCExplorer using macaque data" data-notebook-cellid="cell-chrx-100kb-pca">
<div id="cell-chrx-100kb-pca" class="cell" data-execution_count="7">
<div class="cell-output cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><a href="thoughts_files/figure-html/..-notebooks-01_hicexplorer-chrx-100kb-pca-output-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-2" title="The X chromosome in 100kb bins and the first 3 PCs below. Made with HicExplorer."><img src="thoughts_files/figure-html/..-notebooks-01_hicexplorer-chrx-100kb-pca-output-1.png" class="img-fluid figure-img" alt="The X chromosome in 100kb bins and the first 3 PCs below. Made with HicExplorer."></a></p>
<figcaption>The X chromosome in 100kb bins and the first 3 PCs below. Made with HicExplorer.</figcaption>
</figure>
</div>
</div>
</div>
</div>
<section id="plotting-.cool-files-with-h5py-and-pybigwig" class="level3">
<h3 class="anchored" data-anchor-id="plotting-.cool-files-with-h5py-and-pybigwig">Plotting .cool files with <code>h5py</code> and <code>pyBigWig</code></h3>
<p>It does not work as I want. To read both the matrix and PC components, it requires an abnormal amount of memory (I don’t think it is strictly necessary to load 75 Gb of data into memory to make a 5x8” plot). However, <code>HicExplorer</code> advise against merging bins after correction and normalization, but I just don’t get that. I see no reason it could mess up the correction. Jupyter Display and, ultimately, my screen resolution makes that binning, so It would be completely fine to just bin the matrices after the fact.</p>
</section>
</section>
<section id="notes-on-open2c-framework" class="level2">
<h2 class="anchored" data-anchor-id="notes-on-open2c-framework">Notes on Open2C framework</h2>
<section id="pairtools" class="level3">
<h3 class="anchored" data-anchor-id="pairtools">pairtools</h3>
<p>I’m considering to just map the reads again with bwa, mates together to create a pairs file in the first place. Alternatively, I will use <code>samtools sort | samtools fixmate</code> to convert them back into a paired-end alignment for inputting into <code>cooler</code>. Their own recommendation is to map with <code>bwa mem</code> with paired-end mates.</p>
<blockquote class="blockquote">
<p>Ended up mapping them again</p>
</blockquote>
</section>
<section id="cooler" class="level3">
<h3 class="anchored" data-anchor-id="cooler">cooler</h3>
<p>I’ve decided that this is a much better tool than HiCExplorer, the overall experience is way better. The Python API makes it optimal for Jupyter Notebooks, in stead of doing shell escapes in every cell. It can plot 10kb bins without any issue.</p>
<section id="update-08-10-2024" class="level4">
<h4 class="anchored" data-anchor-id="update-08-10-2024">Update: 08-10-2024</h4>
<p>It took more than a week to get a valid <code>.pairs</code> file and to get that processed into a <code>.cool</code> matrix. But now it seems to work pretty smoothly, and I believe it is more customizable. The pairs are split up into .nodups, .dups, and .unmapped pairs, meaning all the reads are still saved. A <code>dedup.stats</code> is also generated and visualized with <code>MultiQC</code>.</p>
<p>I made these plots of both uncorrected and corrected (ICE) matrices (made with <code>cooler</code>)</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/02_open2c_framework.ipynb" data-notebook-title="Open2C framework" data-notebook-cellid="cell-rep1-ncol3-cooltools-uncorrected">
<div id="cell-rep1-ncol3-cooltools-uncorrected" class="cell" data-execution_count="9">
<div class="cell-output cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><a href="thoughts_files/figure-html/..-notebooks-02_open2c_framework-rep1-ncol3-cooltools-uncorrected-output-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-3" title="Raw matrix. Left: All chromosomes plotted in 100kb bins. Middle: chrX in 100kb bins. Right: chrX:60Mb-70Mb in 50 kb bins"><img src="thoughts_files/figure-html/..-notebooks-02_open2c_framework-rep1-ncol3-cooltools-uncorrected-output-1.png" class="img-fluid figure-img" alt="Raw matrix. Left: All chromosomes plotted in 100kb bins. Middle: chrX in 100kb bins. Right: chrX:60Mb-70Mb in 50 kb bins"></a></p>
<figcaption>Raw matrix. Left: All chromosomes plotted in 100kb bins. Middle: chrX in 100kb bins. Right: chrX:60Mb-70Mb in 50 kb bins</figcaption>
</figure>
</div>
</div>
</div>
</div>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/02_open2c_framework.ipynb" data-notebook-title="Open2C framework" data-notebook-cellid="cell-rep1-ncol3-cooltools-corrected">
<div id="cell-rep1-ncol3-cooltools-corrected" class="cell" data-execution_count="10">
<div class="cell-output cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><a href="thoughts_files/figure-html/..-notebooks-02_open2c_framework-rep1-ncol3-cooltools-corrected-output-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-4" title="Corrected matrix. Left: All chromosomes plotted in 100kb bins. Middle: chrX in 100kb bins. Right: chrX:60Mb-70Mb in 50 kb bins"><img src="thoughts_files/figure-html/..-notebooks-02_open2c_framework-rep1-ncol3-cooltools-corrected-output-1.png" class="img-fluid figure-img" alt="Corrected matrix. Left: All chromosomes plotted in 100kb bins. Middle: chrX in 100kb bins. Right: chrX:60Mb-70Mb in 50 kb bins"></a></p>
<figcaption>Corrected matrix. Left: All chromosomes plotted in 100kb bins. Middle: chrX in 100kb bins. Right: chrX:60Mb-70Mb in 50 kb bins</figcaption>
</figure>
</div>
</div>
</div>
</div>
<p>Neither of them really resembles the ones from <span class="citation" data-cites="wang2019reprogrammingmeioticchromatin">(<a href="../thesis/index.html#ref-wang2019reprogrammingmeioticchromatin" role="doc-biblioref">Wang et al. 2019</a>)</span>, but we have some confidence in our method still. Next step is to do the default compartmentalization analysis from <code>cooler</code> as well.</p>
</section>
<section id="update-10-10-2024" class="level4">
<h4 class="anchored" data-anchor-id="update-10-10-2024">Update 10-10-2024</h4>
<p>I plotted the <code>weights</code> variable below the interaction maps. Maybe it will be useful.</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/02_open2c_framework.ipynb" data-notebook-title="Open2C framework" data-notebook-cellid="cell-rep1-matrix-weights">
<div id="cell-rep1-matrix-weights" class="cell" data-execution_count="18">
<div class="cell-output cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><a href="thoughts_files/figure-html/..-notebooks-02_open2c_framework-rep1-matrix-weights-output-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-5" title="Rep1 interaction matrix with weights visualized. Top: raw counts. Bottom: corrected frequencies. Left: whole chrX. Right: chrX60:70Mb"><img src="thoughts_files/figure-html/..-notebooks-02_open2c_framework-rep1-matrix-weights-output-1.png" class="img-fluid figure-img" alt="Rep1 interaction matrix with weights visualized. Top: raw counts. Bottom: corrected frequencies. Left: whole chrX. Right: chrX60:70Mb"></a></p>
<figcaption>Rep1 interaction matrix with weights visualized. Top: raw counts. Bottom: corrected frequencies. Left: whole chrX. Right: chrX60:70Mb</figcaption>
</figure>
</div>
</div>
</div>
</div>
<p>Also, we can plot the coverage divided into cis and all contacts, as well as the cis/total coverage ratio:</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/02_open2c_framework.ipynb" data-notebook-title="Open2C framework" data-notebook-cellid="cell-full-X-coverage-weights">
<div id="cell-full-X-coverage-weights" class="cell" data-execution_count="35">
<div class="cell-output cell-output-display" data-execution_count="35">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><a href="thoughts_files/figure-html/..-notebooks-02_open2c_framework-full-x-coverage-weights-output-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-6" title="Rep1 chrX interaction matrix with coverage for cis and total as well as cis/total coverage ratio."><img src="thoughts_files/figure-html/..-notebooks-02_open2c_framework-full-x-coverage-weights-output-1.png" class="img-fluid figure-img" alt="Rep1 chrX interaction matrix with coverage for cis and total as well as cis/total coverage ratio."></a></p>
<figcaption>Rep1 chrX interaction matrix with coverage for cis and total as well as cis/total coverage ratio.</figcaption>
</figure>
</div>
</div>
</div>
</div>
<p>Finally, before moving on, we can do some visual improvements to the plot (it does not alter the data), with <em>adaptive coarsegraining</em> and <em>interpolating</em> the matrix. It looks pretty cool:</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/02_open2c_framework.ipynb" data-notebook-title="Open2C framework" data-notebook-cellid="cell-chrX_sub_smooth_interpolate">
<div id="cell-chrX_sub_smooth_interpolate" class="cell" data-execution_count="40">
<div class="cell-output cell-output-display" data-execution_count="40">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><a href="thoughts_files/figure-html/..-notebooks-02_open2c_framework-chrx_sub_smooth_interpolate-output-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-7" title="Normal, smoothed, and interpolated views of the interaction matrix. Functions provided by cooltools."><img src="thoughts_files/figure-html/..-notebooks-02_open2c_framework-chrx_sub_smooth_interpolate-output-1.png" class="img-fluid figure-img" alt="Normal, smoothed, and interpolated views of the interaction matrix. Functions provided by cooltools."></a></p>
<figcaption>Normal, smoothed, and interpolated views of the interaction matrix. Functions provided by <code>cooltools</code>.</figcaption>
</figure>
</div>
</div>
</div>
</div>
</section>
<section id="compartments" class="level4">
<h4 class="anchored" data-anchor-id="compartments">Compartments</h4>
<p>Finally, I managed to map the compartments with <code>cooler</code>. First, the depencies are imported: <code>numpy</code>, <code>pandas</code>, <code>cooler</code>, <code>cooltools</code>, <code>bioframe</code>, and some more plot-specific packages.</p>
<ol type="1">
<li>Load the cooler at a selected binsize.</li>
<li>Load the reference genome<br>
(<code>bioframe.load_fasta()</code>)</li>
<li>Calculate the GC% in each of the bins<br>
(<code>bioframe.frac_gc()</code>)
<ul>
<li>NB I found some missing values, but only 1 on the <em>X</em>. I don’t know why.</li>
</ul></li>
<li>Calculate the <em>cis</em> eigenvectors to each bin (<code>cooltools.eigs_cis()</code>), where the GC coverage is used as a ‘phasing’ track.<br>
It is saved as a E1-track, with <code>[['chrom','start','end','E1']]</code></li>
<li>Plot with <code>matplotlib</code> and submodules. Optionally, we can mark the E1 transitions on the matrix. Here, I removed the horisontal lines.</li>
</ol>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/02_open2c_framework.ipynb" data-notebook-title="Open2C framework" data-notebook-cellid="cell-chrX-matrix-e1">
<div id="cell-chrX-matrix-e1" class="cell" data-execution_count="13">
<div class="cell-output cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><a href="thoughts_files/figure-html/..-notebooks-02_open2c_framework-chrx-matrix-e1-output-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-8" title="chrX interaction matrix with E1 eigenvector values."><img src="thoughts_files/figure-html/..-notebooks-02_open2c_framework-chrx-matrix-e1-output-1.png" class="img-fluid figure-img" alt="chrX interaction matrix with E1 eigenvector values."></a></p>
<figcaption>chrX interaction matrix with E1 eigenvector values.</figcaption>
</figure>
</div>
</div>
</div>
</div>
<p>A note from the authors of <code>cooltools</code> is that we always have to inspect the E1 values to see how well it captures the compartments in the matrix. Here, it actually doesn’t do very well, So i’m not sure what went wrong. The E1 values seem to have a lot of sign changes, that are not visible in the plot. Maybe it looks better with 500kb resolution.</p>
<p>Here, comparing to the Wang paper:</p>
<figure class="figure">
<center>
<figcaption>
Wang2019 PC1 on chrX
</figcaption>
<p><img src="../figures/E1_chrX_fib_wang.png" alt="E1_chrX_fib_wang" width="93%" class="figure-img"></p>
</center>
</figure>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/02_open2c_framework.ipynb" data-notebook-title="Open2C framework" data-notebook-cellid="cell-chrX-e1-plot">
<div id="cell-chrX-e1-plot" class="cell" data-fig-height="2" data-execution_count="18">
<div class="cell-output cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><a href="thoughts_files/figure-html/..-notebooks-02_open2c_framework-chrx-e1-plot-output-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-9" title="100kb binned E1 eigenvector values for chrX. Freshly calculated from the cooler file."><img src="thoughts_files/figure-html/..-notebooks-02_open2c_framework-chrx-e1-plot-output-1.png" class="img-fluid figure-img" alt="100kb binned E1 eigenvector values for chrX. Freshly calculated from the cooler file."></a></p>
<figcaption>100kb binned E1 eigenvector values for chrX. Freshly calculated from the cooler file.</figcaption>
</figure>
</div>
</div>
</div>
</div>
<p>I’m very positive about the results. The compartments do not match completely, but we have to take into account another mapping algorithm and a new (updated) reference genome <em>rheMac10</em>, where more than 99% of the segmentation is removed, compared to <em>rheMac2</em>. It is evident in the <span class="citation" data-cites="wang2019reprogrammingmeioticchromatin">(<a href="../thesis/index.html#ref-wang2019reprogrammingmeioticchromatin" role="doc-biblioref">Wang et al. 2019</a>)</span> PC1 plot, that a chunk in the middle is missing, possibly due to unmmapped reads. In any case, here is higher-resolution of PC compartments, as more Hi-C contacts can be mapped.</p>
</section>
<section id="compartments-500kb-resolution-looks-smoother" class="level4">
<h4 class="anchored" data-anchor-id="compartments-500kb-resolution-looks-smoother">Compartments 500kb resolution looks smoother</h4>
<p>I made the same plot as above, but using a 500kb bin size in stead. It looks more smooth, but it is not obvious if we a removing valid information or noise. It looks more like it captures only the visible compartments in the matrix, whcih is good</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/03_compartments.ipynb" data-notebook-title="Compartments Analysis" data-notebook-cellid="cell-fig-matrix_e1_500kb">
<div id="cell-fig-matrix_e1_500kb" class="cell" data-execution_count="17">
<div class="cell-output cell-output-display" data-execution_count="17">
<div id="fig-matrix_e1_500kb" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-matrix_e1_500kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="thoughts_files/figure-html/..-notebooks-03_compartments-fig-matrix_e1_500kb-output-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-10" title="Figure&nbsp;10.1: chrX interaction matrix with E1 eigenvector values in 500kb resolution. The sign change of the E1 is overlayed as thin black lines, making it more easily interpretable. We should qualitatively determine how well the E1 values capture the plaid pattern in the matrix."><img src="thoughts_files/figure-html/..-notebooks-03_compartments-fig-matrix_e1_500kb-output-1.png" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-matrix_e1_500kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;10.1: chrX interaction matrix with E1 eigenvector values in 500kb resolution. The sign change of the E1 is overlayed as thin black lines, making it more easily interpretable. We should qualitatively determine how well the E1 values capture the <em>plaid</em> pattern in the matrix.
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<p>Or the full size E1 plot:</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/03_compartments.ipynb" data-notebook-title="Compartments Analysis" data-notebook-cellid="cell-fig-e1_500kb_plot">
<div id="cell-fig-e1_500kb_plot" class="cell" data-execution_count="19">
<div class="cell-output cell-output-display">
<div id="fig-e1_500kb_plot" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-e1_500kb_plot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="thoughts_files/figure-html/..-notebooks-03_compartments-fig-e1_500kb_plot-output-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-11" title="Figure&nbsp;10.2: 500kb binned E1 eigenvector values for chrX. Freshly calculated from the cooler file."><img src="thoughts_files/figure-html/..-notebooks-03_compartments-fig-e1_500kb_plot-output-1.png" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-e1_500kb_plot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;10.2: 500kb binned E1 eigenvector values for chrX. Freshly calculated from the cooler file.
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<p>To ease the eye, here are the 100kb E1 and 500kb E1 plots next to each other:</p>
<p><strong>500kb resolution:</strong></p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/03_compartments.ipynb" data-notebook-title="Compartments Analysis" data-notebook-cellid="cell-fig-e1_500kb_plot">
<div id="cell-fig-e1_500kb_plot" class="cell" data-execution_count="19">
<div class="cell-output cell-output-display">
<div id="fig-e1_500kb_plot" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-e1_500kb_plot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="thoughts_files/figure-html/..-notebooks-03_compartments-fig-e1_500kb_plot-output-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-12" title="Figure&nbsp;10.3: 500kb binned E1 eigenvector values for chrX. Freshly calculated from the cooler file."><img src="thoughts_files/figure-html/..-notebooks-03_compartments-fig-e1_500kb_plot-output-1.png" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-e1_500kb_plot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;10.3: 500kb binned E1 eigenvector values for chrX. Freshly calculated from the cooler file.
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<p><strong>100kb resolution</strong></p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/02_open2c_framework.ipynb" data-notebook-title="Open2C framework" data-notebook-cellid="cell-chrX-e1-plot">
<div id="cell-chrX-e1-plot" class="cell" data-fig-height="2" data-execution_count="18">
<div class="cell-output cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><a href="thoughts_files/figure-html/..-notebooks-02_open2c_framework-chrx-e1-plot-output-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-13" title="100kb binned E1 eigenvector values for chrX. Freshly calculated from the cooler file."><img src="thoughts_files/figure-html/..-notebooks-02_open2c_framework-chrx-e1-plot-output-1.png" class="img-fluid figure-img" alt="100kb binned E1 eigenvector values for chrX. Freshly calculated from the cooler file."></a></p>
<figcaption>100kb binned E1 eigenvector values for chrX. Freshly calculated from the cooler file.</figcaption>
</figure>
</div>
</div>
</div>
</div>
</section>
</section>
<section id="compartments-on-all-cell-types-by-merging-all-samples-from-each-type" class="level3">
<h3 class="anchored" data-anchor-id="compartments-on-all-cell-types-by-merging-all-samples-from-each-type">Compartments on all cell types (By merging all samples from each type)</h3>
<p>Now it is time for the analysis of the other cell types as well. Here, we trust <span class="citation" data-cites="wang2019reprogrammingmeioticchromatin">(<a href="../thesis/index.html#ref-wang2019reprogrammingmeioticchromatin" role="doc-biblioref">Wang et al. 2019</a>)</span> when they state that their compartments were highly reproducible between replicates.</p>
<p>Thus, we merge all samples from each cell type, leaving us with 5 <em>coolers</em> at 10 kb bins. We then <code>zoomify</code> the coolers to include the resolutions 50, 100, and 500kb binned data as well.</p>
<p>We continue with the analysis with the 500kb binned data, as it runs faster, uses less memory, while still being sufficient for chromosome-wide analysis.</p>
</section>
</section>
<section id="calculating-the-eigenvectors-again" class="level2 page-columns page-full">
<h2 class="anchored" data-anchor-id="calculating-the-eigenvectors-again">Calculating the eigenvectors again</h2>
<p>Initially, the results were not as expected, as the E1 did not reflect the plaid pattern from the matrix from all cell types. In stead, it showed two compartments.</p>
<p>As stated in <span class="citation" data-cites="wang2019reprogrammingmeioticchromatin">(<a href="../thesis/index.html#ref-aiden2009comprehensivemappinglongrange" role="doc-biblioref"><strong>aiden2009comprehensivemappinglongrange?</strong></a>)</span> and <code>cooltools</code> documentation, the E1 occasionally <em>does not</em> reflect the plaid pattern from the interaction matrix, but rather a two-compartment chromosome. That happens when the difference in interaction frequencies differ more between the chromosome arms than it does within the arms. To mitigate such cases (or get the correct compartments), we can either use the next eigenvector, or (a better option, in our opinion) calculate the eigenvectors from a chromosome partitioned in the two arms (separated by the centromere).</p>
<p>I determined the region of the centromere from the <a href="https://www.ncbi.nlm.nih.gov/gdv/browser/genome/?id=GCF_003339765.1">NCBI browser</a>, in a region from 58-61Mb that have no gene annotations. That region corresponds well with <code>NaN</code> values in the calculated eigenvector (values removed because of low-count bins), as seen below:</p>
<div class="quarto-embed-nb-cell page-columns page-full" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/03_compartments.ipynb" data-notebook-title="Compartments Analysis" data-notebook-cellid="cell-e1_nan_hist">
<div id="cell-e1_nan_hist" class="cell page-columns page-full" data-execution_count="30">
<div class="cell-output cell-output-display page-columns page-full">
<div class="quarto-figure quarto-figure-center page-columns page-full">
<figure class="figure page-columns page-full">
<p class="page-columns page-full"><a href="thoughts_files/figure-html/..-notebooks-03_compartments-e1_nan_hist-output-1.png" class="lightbox page-columns page-full" data-gallery="quarto-lightbox-gallery-14" title="Histogram of NaN values in the E1 eigenvector for each cell type."><img src="thoughts_files/figure-html/..-notebooks-03_compartments-e1_nan_hist-output-1.png" class="img-fluid figure-img column-page-inset" alt="Histogram of NaN values in the E1 eigenvector for each cell type."></a></p>
<figcaption>Histogram of NaN values in the E1 eigenvector for each cell type.</figcaption>
</figure>
</div>
</div>
</div>
</div>
<p>Then, the E1 for each cell type was plotted and looks like we have succesfully reproduced the A/B compartments from <span class="citation" data-cites="wang2019reprogrammingmeioticchromatin">(<a href="../thesis/index.html#ref-wang2019reprogrammingmeioticchromatin" role="doc-biblioref">Wang et al. 2019</a>)</span>. Also, I noticed after all this, that they also partition the chromosomes at the centromere. The more you know…</p>
<div class="quarto-embed-nb-cell page-columns page-full" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/03_compartments.ipynb" data-notebook-title="Compartments Analysis" data-notebook-cellid="cell-e1-fullmerged-all-500kb">
<div id="cell-e1-fullmerged-all-500kb" class="cell page-columns page-full" data-execution_count="31">
<div class="cell-output cell-output-display page-columns page-full">
<div class="quarto-figure quarto-figure-center page-columns page-full">
<figure class="figure page-columns page-full">
<p class="page-columns page-full"><a href="thoughts_files/figure-html/..-notebooks-03_compartments-e1-fullmerged-all-500kb-output-1.png" class="lightbox page-columns page-full" data-gallery="quarto-lightbox-gallery-15" title="E1 eigenvector values for all merged samples at 500kb resolution. Left: Chromosome-arm restricted E1. Right: 10Mb window restricted E1."><img src="thoughts_files/figure-html/..-notebooks-03_compartments-e1-fullmerged-all-500kb-output-1.png" class="img-fluid figure-img column-page-inset" alt="E1 eigenvector values for all merged samples at 500kb resolution. Left: Chromosome-arm restricted E1. Right: 10Mb window restricted E1."></a></p>
<figcaption>E1 eigenvector values for all merged samples at 500kb resolution. Left: Chromosome-arm restricted E1. Right: 10Mb window restricted E1.</figcaption>
</figure>
</div>
</div>
</div>
</div>
<p>Furthermoere, we wanted to see if we could apply the same smoothing as they did: They calculate the obs/exp matrix on a smoothed 100kb matrix: they slide a 400kb window with a step size of 100kb, summing the values.</p>
<p>It turned out to be practically very hard to do in the <em>Open2c</em> ecosystem, so I decided to apply the smoothing directly on the calculated E1 values.</p>
<div class="quarto-embed-nb-cell page-columns page-full" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/03_compartments.ipynb" data-notebook-title="Compartments Analysis" data-notebook-cellid="cell-e1-fullmerged-all-100kb-5smoothed">
<div id="cell-e1-fullmerged-all-100kb-5smoothed" class="cell page-columns page-full" data-execution_count="40">
<div class="cell-output cell-output-display page-columns page-full">
<div class="quarto-figure quarto-figure-center page-columns page-full">
<figure class="figure page-columns page-full">
<p class="page-columns page-full"><a href="thoughts_files/figure-html/..-notebooks-03_compartments-e1-fullmerged-all-100kb-5smoothed-output-1.png" class="lightbox page-columns page-full" data-gallery="quarto-lightbox-gallery-16" title="E1 eigenvector values for all merged samples at 100kb resolution. Left: Chromosomes (not restricted) Middle: Chromosome-arm restricted E1. Right: 10Mb window restricted E1.Values are smoothed with a sliding window of 5 bins, step size 1 bin."><img src="thoughts_files/figure-html/..-notebooks-03_compartments-e1-fullmerged-all-100kb-5smoothed-output-1.png" class="img-fluid figure-img column-page-inset" alt="E1 eigenvector values for all merged samples at 100kb resolution. Left: Chromosomes (not restricted) Middle: Chromosome-arm restricted E1. Right: 10Mb window restricted E1.Values are smoothed with a sliding window of 5 bins, step size 1 bin."></a></p>
<figcaption>E1 eigenvector values for all merged samples at 100kb resolution. Left: Chromosomes (not restricted) Middle: Chromosome-arm restricted E1. Right: 10Mb window restricted E1.Values are smoothed with a sliding window of 5 bins, step size 1 bin.</figcaption>
</figure>
</div>
</div>
</div>
</div>
<p>Now we will do some planning, tidying, and writing before continuing the analysis.</p>


<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list" style="display: none">
<div id="ref-dixon2015chromatinarchitecturereorganization" class="csl-entry" role="listitem">
Dixon, Jesse R., Inkyung Jung, Siddarth Selvaraj, Yin Shen, Jessica E. Antosiewicz-Bourget, Ah Young Lee, Zhen Ye, et al. 2015. <span>“Chromatin Architecture Reorganization During Stem Cell Differentiation.”</span> <em>Nature</em> 518 (7539): 331–36. <a href="https://doi.org/10.1038/nature14222">https://doi.org/10.1038/nature14222</a>.
</div>
<div id="ref-wang2019reprogrammingmeioticchromatin" class="csl-entry" role="listitem">
Wang, Yao, Hanben Wang, Yu Zhang, Zhenhai Du, Wei Si, Suixing Fan, Dongdong Qin, et al. 2019. <span>“Reprogramming of <span>Meiotic Chromatin Architecture</span> During <span>Spermatogenesis</span>.”</span> <em>Molecular Cell</em> 73 (3): 547–561.e6. <a href="https://doi.org/10.1016/j.molcel.2018.11.019">https://doi.org/10.1016/j.molcel.2018.11.019</a>.
</div>
</div>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../workflow.html" class="pagination-link" aria-label="master_workflow">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-title">master_workflow</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../results/Readme.html" class="pagination-link" aria-label="Result files">
        <span class="nav-page-text"><span class="chapter-title">Result files</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer"><div class="nav-footer-center"><div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/munch-group/hic-spermatogenesis/edit/main/reports/thoughts.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/munch-group/hic-spermatogenesis/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li><li><a href="https://github.com/munch-group/hic-spermatogenesis/blob/main/reports/thoughts.qmd" class="toc-action"><i class="bi empty"></i>View source</a></li></ul></div></div></div></footer><script>var lightboxQuarto = GLightbox({"closeEffect":"zoom","descPosition":"bottom","loop":false,"openEffect":"zoom","selector":".lightbox"});
(function() {
  let previousOnload = window.onload;
  window.onload = () => {
    if (previousOnload) {
      previousOnload();
    }
    lightboxQuarto.on('slide_before_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      const href = trigger.getAttribute('href');
      if (href !== null) {
        const imgEl = window.document.querySelector(`a[href="${href}"] img`);
        if (imgEl !== null) {
          const srcAttr = imgEl.getAttribute("src");
          if (srcAttr && srcAttr.startsWith("data:")) {
            slideConfig.href = srcAttr;
          }
        }
      } 
    });
  
    lightboxQuarto.on('slide_after_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(slideNode);
      }
    });
  
  };
  
})();
          </script>




</body></html>