% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
\PassOptionsToPackage{dvipsnames,svgnames,x11names}{xcolor}
%
\documentclass[
  a4paper,
  DIV=11,
  numbers=noendperiod]{scrarticle}

\usepackage{amsmath,amssymb}
\usepackage{iftex}
\ifPDFTeX
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
\usepackage{lmodern}
\ifPDFTeX\else  
    % xetex/luatex font selection
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\usepackage[textheight=9in,textwidth=6.5in,top=1in,headheight=14pt,headsep=25pt,footskip=30pt]{geometry}
\setlength{\emergencystretch}{3em} % prevent overfull lines
\setcounter{secnumdepth}{-\maxdimen} % remove section numbering
% Make \paragraph and \subparagraph free-standing
\makeatletter
\ifx\paragraph\undefined\else
  \let\oldparagraph\paragraph
  \renewcommand{\paragraph}{
    \@ifstar
      \xxxParagraphStar
      \xxxParagraphNoStar
  }
  \newcommand{\xxxParagraphStar}[1]{\oldparagraph*{#1}\mbox{}}
  \newcommand{\xxxParagraphNoStar}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
  \let\oldsubparagraph\subparagraph
  \renewcommand{\subparagraph}{
    \@ifstar
      \xxxSubParagraphStar
      \xxxSubParagraphNoStar
  }
  \newcommand{\xxxSubParagraphStar}[1]{\oldsubparagraph*{#1}\mbox{}}
  \newcommand{\xxxSubParagraphNoStar}[1]{\oldsubparagraph{#1}\mbox{}}
\fi
\makeatother


\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}\usepackage{longtable,booktabs,array}
\usepackage{calc} % for calculating minipage widths
% Correct order of tables after \paragraph or \subparagraph
\usepackage{etoolbox}
\makeatletter
\patchcmd\longtable{\par}{\if@noskipsec\mbox{}\fi\par}{}{}
\makeatother
% Allow footnotes in longtable head/foot
\IfFileExists{footnotehyper.sty}{\usepackage{footnotehyper}}{\usepackage{footnote}}
\makesavenoteenv{longtable}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother

%% Palatino font:
\usepackage[T1]{fontenc}
\usepackage[utf8]{inputenc}
\usepackage{palatino}

\usepackage{pifont}

\usepackage{upquote}

% \pagestyle{plain} % no running header

% \usepackage[many]{tcolorbox}
% \definecolor{quotegray}{HTML}{505050}
% \newtcolorbox{myquote}{%
%     enhanced jigsaw, 
%     breakable,      % allow page breaks
%     frame hidden,   % hide the default frame
%     left=1cm,       % left margin
%     right=1cm,      % right margin
%     colback=white,
%     fontupper=\color{quotegray},
%     overlay={%
%         \node [scale=4,
%             text=lightgray,
%             inner sep=0pt,] at ([xshift=0.5cm,yshift=-0.7cm]frame.north west){``}; 
%         \node [scale=4,
%             text=lightgray,
%             inner sep=0pt,] at ([xshift=-0.5cm]frame.south east){''};  
%             },
%         % paragraph skips obeyed within tcolorbox
%                 parbox=false,
% }
% % redefine the 'quote' environment to use this 'myquote' environment
% \renewenvironment{quote}{\begin{myquote}}{\end{myquote}}


% \usepackage{framed} % not sure i need this anymore

% \usepackage[T1]{fontenc}
% \usepackage{inconsolata}

% % I have to only define Shaded if it is already defined.
% % The reason is that pandoc does not define the macro if there are not code blocks in the a markdown file.
% \ifx \@Shaded \@empty

% \renewcommand{\KeywordTok}[1]{\textcolor[rgb]{0, 0, 0}{\textbf{{#1}}}} % def and or not reg
% \renewcommand{\BuiltInTok}[1]{\textcolor[rgb]{0.373, 0.298, 0.580}{\textbf{{#1}}}} % print open 
% \renewcommand{\VariableTok}[1]{\textcolor[rgb]{0.141, 0.392, 0.824}{\textbf{{#1}}}}
% \renewcommand{\OperatorTok}[1]{\textcolor[rgb]{0, 0, 0}{{#1}}} % def and or not reg
% \renewcommand{\DataTypeTok}[1]{\textcolor[rgb]{1.0,0.13,0.00}{{#1}}}
% \renewcommand{\DecValTok}[1]{\textcolor[rgb]{0.655, 0.498, 0.161}{{#1}}}
% \renewcommand{\BaseNTok}[1]{\textcolor[rgb]{0.259, 0.592, 0.596}{{#1}}}
% \renewcommand{\FloatTok}[1]{\textcolor[rgb]{0.655, 0.498, 0.161}{{#1}}}
% \renewcommand{\CharTok}[1]{\textcolor[rgb]{0.678,0.141,0.098}{{#1}}}
% \renewcommand{\StringTok}[1]{\textcolor[rgb]{0.678,0.141,0.098}{{#1}}}
% \renewcommand{\CommentTok}[1]{\textcolor[rgb]{0.135, 0.134, 0.133}{{#1}}}
% \renewcommand{\OtherTok}[1]{\textcolor[rgb]{0.00,0.44,0.13}{{#1}}}
% \renewcommand{\AlertTok}[1]{\textcolor[rgb]{1.00,0.00,0.00}{\textbf{{#1}}}}
% \renewcommand{\FunctionTok}[1]{\textcolor[rgb]{0.549, 0.102, 0.063}{\textbf{{#1}}}}  % function name
% \renewcommand{\RegionMarkerTok}[1]{{#1}}
% \renewcommand{\ErrorTok}[1]{\textcolor[rgb]{1.00,0.00,0.00}{\textbf{{#1}}}}
% \renewcommand{\NormalTok}[1]{\textcolor[rgb]{0, 0, 0}{{#1}}}


% \else
%   % no code blocks with markup...
% \fi



% \usepackage{etoolbox}
% \makeatletter
% \g@addto@macro{\appendix}{%
%   \patchcmd{\@@makechapterhead}% <cmd>
%     {\endgraf\nobreak\vskip.5\baselineskip}% <search>
%     {\hspace*{-.5em}:\space}% <replace>
%     {}{}% <success><failure>
%   \patchcmd{\@chapter}% <cmd>
%     {\addchaptertocentry{\thechapter}}% <search>
%     {\addchaptertocentry{Appendix~\thechapter:}}% <replace>
%     {}{}% <success><failure>
%   \addtocontents{toc}{%
%     \protect\patchcmd{\protect\l@chapter}% <cmd>
%       {1.5em}% <search>
%       {6.5em}% <replace>
%       {}{}}% <success><failure>
% }
% \renewcommand{\autodot}{}% Remove all end-of-counter dots
% \makeatother

% \makeatletter
% \@addtoreset{chapter}{part}
% \makeatother


% \usepackage{chngcntr}
% \counterwithin*{subsubsection}{chapter}
% \counterwithout*{subsubsection}{section}
% \counterwithout*{subsubsection}{subsection}

% %  % KMT only use subsubsection number (this increments though the book 
% %  % when we supress numbering with  {.unnumbered} after each header except Exerisices
% \renewcommand\thesubsubsection{\arabic{section}}
% \renewcommand\thesubsubsection{\arabic{subsection}}
% \renewcommand\thesubsubsection{\arabic{chapter}-\arabic{subsubsection}}


% \renewcommand*{\chapterformat}{%
%   \textcolor[rgb]{0.8, 0.8, 0.8}{\thechapter}\autodot\enskip%
% }


% % prevent latex from "floating" the figures
% \usepackage{float}
% \let\origfigure\figure
% \let\endorigfigure\endfigure
% \renewenvironment{figure}[1][2] {
%     \expandafter\origfigure\expandafter[H]
% } {
%     \endorigfigure
% }

\KOMAoption{captions}{tableheading}
\makeatletter
\@ifpackageloaded{caption}{}{\usepackage{caption}}
\AtBeginDocument{%
\ifdefined\contentsname
  \renewcommand*\contentsname{Table of contents}
\else
  \newcommand\contentsname{Table of contents}
\fi
\ifdefined\listfigurename
  \renewcommand*\listfigurename{List of Figures}
\else
  \newcommand\listfigurename{List of Figures}
\fi
\ifdefined\listtablename
  \renewcommand*\listtablename{List of Tables}
\else
  \newcommand\listtablename{List of Tables}
\fi
\ifdefined\figurename
  \renewcommand*\figurename{Figure}
\else
  \newcommand\figurename{Figure}
\fi
\ifdefined\tablename
  \renewcommand*\tablename{Table}
\else
  \newcommand\tablename{Table}
\fi
}
\@ifpackageloaded{float}{}{\usepackage{float}}
\floatstyle{ruled}
\@ifundefined{c@chapter}{\newfloat{codelisting}{h}{lop}}{\newfloat{codelisting}{h}{lop}[chapter]}
\floatname{codelisting}{Listing}
\newcommand*\listoflistings{\listof{codelisting}{List of Listings}}
\makeatother
\makeatletter
\makeatother
\makeatletter
\@ifpackageloaded{caption}{}{\usepackage{caption}}
\@ifpackageloaded{subcaption}{}{\usepackage{subcaption}}
\makeatother

\ifLuaTeX
  \usepackage{selnolig}  % disable illegal ligatures
\fi
\usepackage{bookmark}

\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\urlstyle{same} % disable monospaced font for URLs
\hypersetup{
  pdftitle={Manuscript},
  pdfauthor={S√∏ren J√∏rgensen},
  pdfkeywords={Hi-C, Chromatin Compartments, Selection},
  colorlinks=true,
  linkcolor={blue},
  filecolor={Maroon},
  citecolor={Blue},
  urlcolor={Blue},
  pdfcreator={LaTeX via pandoc}}


\title{Manuscript}
\author{S√∏ren J√∏rgensen}
\date{November 19, 2024 13:23:43 (UTC +01:00)}

\begin{document}
\maketitle
\begin{abstract}
This is a dummy abstract, dreamt up by chatGPT. This thesis investigates
the 3D chromatin architecture of the X chromosome in baboons, macaques,
and humans, focusing on chromatin compartments during spermatogenesis.
Using publicly available Hi-C data, interaction maps were created to
identify Principal Component 1 (PC1) compartments, revealing distinct
compartmentalization patterns among species. The analysis included
transition zones, where chromatin shifts between compartment types, and
their correlation with positively selected regions. By comparing these
zones with evolutionarily significant regions, the study explores how
chromatin structure influences evolutionary pressures. Key findings
include conserved chromatin features that may help retain
non-advantageous alleles, suggesting a role for selfish genetic elements
in genome evolution. This research offers new insights into the
relationship between chromatin architecture and evolutionary dynamics
across primate species.
\end{abstract}

\renewcommand*\contentsname{Table of contents}
{
\hypersetup{linkcolor=}
\setcounter{tocdepth}{2}
\tableofcontents
}

\section{\texorpdfstring{Chromatin Compartments and Selection on
\emph{X}}{Chromatin Compartments and Selection on X}}\label{chromatin-compartments-and-selection-on-x}

\subsection{Introduction}\label{introduction}

{üìù} = \emph{draft block}

\subsubsection{Sexual reproduction (spermatogenesis,
meiosis)}\label{sexual-reproduction-spermatogenesis-meiosis}

\begin{quote}
{üìù} The production of gametes in a sexually reproducing organism is a
highly complex process that involves numeruous elements.
Spermatogenesis, the process of forming male gametes, involves four
stages of differentiation from a germ cell through \emph{spermatogonia},
\emph{pachytene spermatocyte}, and \emph{round spermatids} to
\emph{spermatozoa}, or \emph{sperm}
(\textbf{wang\_reprogramming\_2019?}).
\end{quote}

\subsubsection{Selfish genes}\label{selfish-genes}

\begin{quote}
{üìù} The conventional story of meiosis in gametogenesis is one of random
segregation of the sex chromosomes. They split into haploid gametes that
are each responsible for their own survival, and nothing more. That
seems like a fair game, but what if some genes are cheating the system
by making other participants less viable. Say, some genes on the X
chromosome creates a disadvantage for gametes that \emph{do not} contain
those genes, making sure the Y chromosome is not as viable as the X,
resulting in a sex imbalance and possibly numerous other downstream
effects. That is exactly what is coined \emph{sex chromosome meiotic
drive} (\textbf{jaenike\_sex\_2001?}), a result of selfish genetic
elements.
\end{quote}

\begin{quote}
{üìù} Motivated by previous results in the Munch Research group
(\textbf{munch\_group\_2024?}) on hybrid incompatibility and extended
common haplotypes (\textbf{skov\_extraordinary\_2023?};
\textbf{sorensen\_genome\_wide\_2023?}) that could be explained by
meiotic drive, we wanted to investigate how these patterns correlate
with chromatin compartments.
\end{quote}

\subsubsection{High-Throughput Chromosome conformation capture
(Hi-C)}\label{high-throughput-chromosome-conformation-capture-hi-c}

Our DNA can be divided into different orders of structure. \emph{3C}
focus on identifying the highest orders of organization inside the
nucleus, that is, when the 30 nm thick coil of chromatin fibers folds
into loops, Topologically Associating Domains (TADs), and chromatin
compartments. Here, we narrow our focus on the largest of the
structures, \emph{compartments}, that is known to determine availability
to transcription factors, thus making an \emph{A} compartment
\emph{active}---and the \emph{B} compartment \emph{inactive}. The
introduction of the Hi-C method
(\textbf{lieberman-aiden\_comprehensive\_2009?}) (high-throughput 3C)
opened new possibilities for exploring the three-dimensional
organization of the genome.

\subsection{Methods}\label{methods}

In this project, we formulate two objectives:

\textbf{A}: Reproduce the Hi-C interaction maps and eigendecomposition
from (\textbf{wang\_reprogramming\_2019?}), with some modifications. We
briefly use \emph{HiCExplorer}, but change the analyses to use the
\emph{Open2C Ecosystem} (\textbf{open2c?}) which have a Pyton API as
well as command-line functions, which can be paired very well with
Jupyter Notebooks. The majority of the data analysis was run with a
\emph{gwf} workflow, and the commands that were visually inspected were
run in Jupyter Notebooks.

\textbf{B} Compare with regions of selection that are found in
\emph{papio anubis}, and maybe in \emph{human} too. Investigate the
biological meaning of the results.

\subsubsection{Data and structure}\label{data-and-structure}

\phantomsection\label{md-data-accessions}
To get an overview of the data accessions used in this analysis, we will
first summarize the \texttt{SRA-runtable.tsv} that contains the
accession numbers and some metadata for each sample
(Table~\ref{tbl-runtable-summary}).

\begin{longtable}[]{@{}lllll@{}}

\caption{\label{tbl-runtable-summary}Summary of the data accessions used
in this analysis}

\tabularnewline

\caption{}\label{T_0aa9c}\tabularnewline
\toprule\noalign{}
~ & source\_name & GB & Bases & Reads \\
\midrule\noalign{}
\endfirsthead
\toprule\noalign{}
~ & source\_name & GB & Bases & Reads \\
\midrule\noalign{}
\endhead
\bottomrule\noalign{}
\endlastfoot
0 & fibroblast & 211.403275 & 553,968,406,500 & 1,846,561,355 \\
1 & pachytene spermatocyte & 274.835160 & 715,656,614,700 &
2,385,522,049 \\
2 & round spermatid & 243.128044 & 655,938,457,200 & 2,186,461,524 \\
3 & sperm & 164.131640 & 428,913,635,400 & 1,429,712,118 \\
4 & spermatogonia & 192.794420 & 518,665,980,300 & 1,728,886,601 \\

\end{longtable}

\phantomsection\label{md-folder-structure}
For ease of mind, here is the folder structure of the project.
\texttt{../steps/bwa/PE/} is the base directory artificially defined in
the \texttt{master\_workflow.py}. It ccould be any other directory
inside \texttt{steps}. It is defined relative to the
\texttt{master\_workflow.py} file (inside the worklow), and converted to
an absolute path by python.

\begin{figure}

\centering{

\begin{verbatim}
../steps/bwa/PE
‚îú‚îÄ‚îÄ bamfiles
‚îÇ¬†¬† ‚îú‚îÄ‚îÄ fibroblast
‚îÇ¬†¬† ‚îú‚îÄ‚îÄ pachytene_spermatocyte
‚îÇ¬†¬† ‚îú‚îÄ‚îÄ round_spermatid
‚îÇ¬†¬† ‚îú‚îÄ‚îÄ sperm
‚îÇ¬†¬† ‚îî‚îÄ‚îÄ spermatogonia
‚îú‚îÄ‚îÄ cool
‚îÇ¬†¬† ‚îú‚îÄ‚îÄ fibroblast
‚îÇ¬†¬† ‚îú‚îÄ‚îÄ pachytene_spermatocyte
‚îÇ¬†¬† ‚îú‚îÄ‚îÄ round_spermatid
‚îÇ¬†¬† ‚îú‚îÄ‚îÄ sperm
‚îÇ¬†¬† ‚îî‚îÄ‚îÄ spermatogonia
‚îî‚îÄ‚îÄ pairs
    ‚îú‚îÄ‚îÄ fibroblast
    ‚îú‚îÄ‚îÄ pachytene_spermatocyte
    ‚îú‚îÄ‚îÄ round_spermatid
    ‚îú‚îÄ‚îÄ sperm
    ‚îî‚îÄ‚îÄ spermatogonia

18 directories
\end{verbatim}

}

\caption{\label{fig-folder-structure}}

\end{figure}%

\phantomsection\label{md-plotting-main}
The following section contains the visualization of the matrices and
includes the calculation of the E1 compartments and their visualization.
I discuss differents methods for construction and try to get both
methodologically and result-wise close to
(\textbf{wang\_reprogramming\_2019?}).

First, we will work on matrices in 500kb resolution, as documentation
{[}maybe it was HiCExplorer??{]} states it is sufficient for
chromosome-wide analysis and plotting. We will follow the
\texttt{cooltools}-recommended pipeline (except that they use a 100kb
cooler) for
\href{https://cooltools.readthedocs.io/en/latest/notebooks/viz.html}{visualization}
and
\href{https://cooltools.readthedocs.io/en/latest/notebooks/compartments_and_saddles.html}{compartment
calling} with one example cooler (the merged fibroblast cooler at 500kb
resolution).

Then, the most relevant parts will be generalized to run in a loop over
all the coolers. First at 500kb resolution, then at 100kb resolution.

To accomodate the approach (barely) described in the paper, we will
discuss a smoothing step to the observed/expected matrices before
compartment calling, and we will apply a smoothing step to the E1
compartments and compare to the raw compartments.

\phantomsection\label{md-plotting-500kb-single}
First, I will explore the visualization pipeline for a single cooler at
500kb resolution. I will modify the plot to be `stairs' in stead of just
a regular line plot, as it is both a more accurate representation of the
data and it is more aesthetically pleasing with less spiky lines and
holes.

In practice, the length of the dataframe is doubled, as it now contains
an E1 value for both the start and end position for each bin in stead of
only for the start. However, I first make the regular line plot to show
the difference.

\begin{figure}[H]

\centering{

\includegraphics{index_files/figure-latex/..-..-notebooks-03_compartments-fig-matrix_e1_500kb-output-1.png}

}

\caption{\label{fig-matrix_e1_500kb}chrX interaction matrix with E1
eigenvector values in 500kb resolution. The sign change of the E1 is
overlayed as thin black lines, making it more easily interpretable. We
should qualitatively determine how well the E1 values capture the
\emph{plaid} pattern in the matrix.}

\end{figure}%

Here we see that Figure~\ref{fig-matrix_e1_500kb}\ldots{}

\subsection{Results}\label{results}

Here are the glorious results

\subsection{Discussion}\label{discussion}

Here is the discussion

\subsection*{Bibliography}\label{bibliography}
\addcontentsline{toc}{subsection}{Bibliography}




\end{document}
