<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>HiCExplorer using macaque data – Søren's MSc project about Hi-C and spermatogenesis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js" integrity="sha512-bLT0Qm9VnAYZDflyKcBaQ2gg0hSYNQrJ8RilYldYQ1FxQYoCLtUjuuRuZo+fjqhx/qtq/1itJ0C2ejDxltZVFg==" crossorigin="anonymous"></script><script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../gwf_map_reads.html" rel="next">
<link href="../notebooks/002_interaction.html" rel="prev">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" integrity="sha512-c3Nl8+7g4LMSTdrm621y7kf9v3SDPnhxLNhcjFJbKECVnmZHTdo+IRO05sNLTH/D3vA6u1X32ehoLC7WFVdheg==" crossorigin="anonymous"></script>

<script type="application/javascript">define('jquery', [],function() {return window.jQuery;})</script>


</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../notebooks/001_weather.html">Notebooks</a></li><li class="breadcrumb-item"><a href="../notebooks/01_hicexplorer.html"><span class="chapter-title">HiCExplorer using macaque data</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">Søren’s MSc project about Hi-C and spermatogenesis</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Project</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">Notebooks</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/001_weather.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Danish weather</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/002_interaction.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Workplace interaction</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/01_hicexplorer.html" class="sidebar-item-text sidebar-link active"><span class="chapter-title">HiCExplorer using macaque data</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">Workflows</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../gwf_map_reads.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">gwf_map_reads</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true">
 <span class="menu-text">Reports</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../results/Readme.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Result files</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../reports/manuscript.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Correlating Chromatin Compartments with Selection</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../reports/supplementary.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Supplementary info</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../references.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">References</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul class="collapse">
  <li><a href="#getting-started-with-an-overview" id="toc-getting-started-with-an-overview" class="nav-link active" data-scroll-target="#getting-started-with-an-overview">Getting started with an overview</a></li>
  <li><a href="#hicexplorer" id="toc-hicexplorer" class="nav-link" data-scroll-target="#hicexplorer">HiCExplorer</a></li>
  </ul>
<div class="quarto-code-links"><h2>Code Links</h2><ul><li><a href="https://github.com/munch-group/hic-spermatogenesis/" target="_blank"><i class="bi bi-github"></i>GitHub Repo</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../notebooks/001_weather.html">Notebooks</a></li><li class="breadcrumb-item"><a href="../notebooks/01_hicexplorer.html"><span class="chapter-title">HiCExplorer using macaque data</span></a></li></ol></nav>
<div class="quarto-title">
<h1 class="title"><span class="chapter-title">HiCExplorer using macaque data</span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<p><strong><em>Before using this notebook, make sure you use the ‘hic’ kernel.</em></strong></p>
<p>I might change the env into a seperate HiC env for each of the tools, such as HICExplorer, Cooler/cooltools, etc.</p>
<p>As of this update, there is only one environment in the repo, and it contains both Python, R, and CLI tools for use in HI-C data analysis.</p>
<section id="getting-started-with-an-overview" class="level2">
<h2 class="anchored" data-anchor-id="getting-started-with-an-overview">Getting started with an overview</h2>
<p>In this notebook, we will use HiCExplorer to analysis Hi-C reads from macaque monkeys. We will follow an <a href="https://hicexplorer.readthedocs.io/en/latest/content/example_usage.html#how-we-use-hicexplorer">example</a> from the HiCExplorer documentation.</p>
<p>First, let’s look at the files we have. They are from <a href="https://www.ncbi.nlm.nih.gov/Traces/study/?acc=PRJNA430777&amp;o=acc_s%3Aahttps://www.ncbi.nlm.nih.gov/Traces/study/?acc=PRJNA430777&amp;o=acc_s%3Aa">SRA</a>, and are Hi-C sequencing data from macaque fibroblasts. The data is generated for the Wang2019 paper.</p>
<div id="cell-2" class="cell" data-execution_count="11">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="op">%%</span>bash </span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>tree ..<span class="op">/</span>data<span class="op">/</span>links<span class="op">/</span>macaque_fastq</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>tree ..<span class="op">/</span>..<span class="op">/</span>..<span class="op">/</span>..<span class="op">/</span>data<span class="op">/</span>macaque_raw<span class="op">/</span>downloaded<span class="op">/</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>ls ..<span class="op">/</span>..<span class="op">/</span>..<span class="op">/</span>..<span class="op">/</span>data<span class="op">/</span>macaque_raw<span class="op">/</span>downloaded<span class="op">/</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>../data/links/macaque_fastq
├── SRR6502335_1.fastq.gz -&gt; ../../../../../../data/macaque_raw/downloaded/SRR6502335_1.fastq.gz
├── SRR6502335_2.fastq.gz -&gt; ../../../../../../data/macaque_raw/downloaded/SRR6502335_2.fastq.gz
├── SRR6502336_1.fastq.gz -&gt; ../../../../../../data/macaque_raw/downloaded/SRR6502336_1.fastq.gz
├── SRR6502336_2.fastq.gz -&gt; ../../../../../../data/macaque_raw/downloaded/SRR6502336_2.fastq.gz
├── SRR6502337_1.fastq.gz -&gt; ../../../../../../data/macaque_raw/downloaded/SRR6502337_1.fastq.gz
├── SRR6502337_2.fastq.gz -&gt; ../../../../../../data/macaque_raw/downloaded/SRR6502337_2.fastq.gz
├── SRR6502338_1.fastq.gz -&gt; ../../../../../../data/macaque_raw/downloaded/SRR6502338_1.fastq.gz
├── SRR6502338_2.fastq.gz -&gt; ../../../../../../data/macaque_raw/downloaded/SRR6502338_2.fastq.gz
├── SRR6502339_1.fastq.gz -&gt; ../../../../../../data/macaque_raw/downloaded/SRR6502339_1.fastq.gz
└── SRR6502339_2.fastq.gz -&gt; ../../../../../../data/macaque_raw/downloaded/SRR6502339_2.fastq.gz

0 directories, 10 files
../../../../data/macaque_raw/downloaded/
├── rheMac10_Mmul10.zip
├── SraRunTable.txt
├── SRR6502335_1.fastq.gz
├── SRR6502335_2.fastq.gz
├── SRR6502336_1.fastq.gz
├── SRR6502336_2.fastq.gz
├── SRR6502337_1.fastq.gz
├── SRR6502337_2.fastq.gz
├── SRR6502338_1.fastq.gz
├── SRR6502338_2.fastq.gz
├── SRR6502339_1.fastq.gz
└── SRR6502339_2.fastq.gz

0 directories, 12 files
rheMac10_Mmul10.zip
SraRunTable.txt
SRR6502335_1.fastq.gz
SRR6502335_2.fastq.gz
SRR6502336_1.fastq.gz
SRR6502336_2.fastq.gz
SRR6502337_1.fastq.gz
SRR6502337_2.fastq.gz
SRR6502338_1.fastq.gz
SRR6502338_2.fastq.gz
SRR6502339_1.fastq.gz
SRR6502339_2.fastq.gz</code></pre>
</div>
</div>
<section id="an-overview-of-the-sequences" class="level3">
<h3 class="anchored" data-anchor-id="an-overview-of-the-sequences">An overview of the sequences</h3>
<p>We can see summary statistics for our sequences in the <code>SraRunTable.txt</code></p>
<div id="cell-4" class="cell" data-execution_count="13">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>pd.read_csv(<span class="st">"../../../../data/macaque_raw/downloaded/SraRunTable.txt"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="13">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">Run</th>
<th data-quarto-table-cell-role="th">Assay Type</th>
<th data-quarto-table-cell-role="th">AvgSpotLen</th>
<th data-quarto-table-cell-role="th">Bases</th>
<th data-quarto-table-cell-role="th">BioProject</th>
<th data-quarto-table-cell-role="th">BioSample</th>
<th data-quarto-table-cell-role="th">Bytes</th>
<th data-quarto-table-cell-role="th">Center Name</th>
<th data-quarto-table-cell-role="th">Consent</th>
<th data-quarto-table-cell-role="th">DATASTORE filetype</th>
<th data-quarto-table-cell-role="th">...</th>
<th data-quarto-table-cell-role="th">ReleaseDate</th>
<th data-quarto-table-cell-role="th">create_date</th>
<th data-quarto-table-cell-role="th">version</th>
<th data-quarto-table-cell-role="th">Sample Name</th>
<th data-quarto-table-cell-role="th">source_name</th>
<th data-quarto-table-cell-role="th">SRA Study</th>
<th data-quarto-table-cell-role="th">cell_type</th>
<th data-quarto-table-cell-role="th">tissue</th>
<th data-quarto-table-cell-role="th">agent</th>
<th data-quarto-table-cell-role="th">genotype</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>SRR6502335</td>
<td>Hi-C</td>
<td>300</td>
<td>73201141800</td>
<td>PRJNA430777</td>
<td>SAMN08375237</td>
<td>31966430779</td>
<td>GEO</td>
<td>public</td>
<td>fastq,run.zq,sra</td>
<td>...</td>
<td>2019-02-12T00:00:00Z</td>
<td>2018-01-23T23:05:00Z</td>
<td>1</td>
<td>GSM2940099</td>
<td>fibroblast</td>
<td>SRP131117</td>
<td>fibroblast</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>SRR6502336</td>
<td>Hi-C</td>
<td>300</td>
<td>65119970100</td>
<td>PRJNA430777</td>
<td>SAMN08375237</td>
<td>24433383054</td>
<td>GEO</td>
<td>public</td>
<td>fastq,run.zq,sra</td>
<td>...</td>
<td>2019-02-12T00:00:00Z</td>
<td>2018-01-24T00:16:00Z</td>
<td>1</td>
<td>GSM2940099</td>
<td>fibroblast</td>
<td>SRP131117</td>
<td>fibroblast</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>SRR6502337</td>
<td>Hi-C</td>
<td>300</td>
<td>52769196300</td>
<td>PRJNA430777</td>
<td>SAMN08375236</td>
<td>23015357755</td>
<td>GEO</td>
<td>public</td>
<td>fastq,run.zq,sra</td>
<td>...</td>
<td>2019-02-12T00:00:00Z</td>
<td>2018-01-24T07:21:00Z</td>
<td>1</td>
<td>GSM2940100</td>
<td>fibroblast</td>
<td>SRP131117</td>
<td>fibroblast</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3</td>
<td>SRR6502338</td>
<td>Hi-C</td>
<td>300</td>
<td>52378949100</td>
<td>PRJNA430777</td>
<td>SAMN08375236</td>
<td>22999581685</td>
<td>GEO</td>
<td>public</td>
<td>fastq,run.zq,sra</td>
<td>...</td>
<td>2019-02-12T00:00:00Z</td>
<td>2018-01-23T22:14:00Z</td>
<td>1</td>
<td>GSM2940100</td>
<td>fibroblast</td>
<td>SRP131117</td>
<td>fibroblast</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">4</td>
<td>SRR6502339</td>
<td>Hi-C</td>
<td>300</td>
<td>28885941600</td>
<td>PRJNA430777</td>
<td>SAMN08375236</td>
<td>10960123150</td>
<td>GEO</td>
<td>public</td>
<td>fastq,run.zq,sra</td>
<td>...</td>
<td>2019-02-12T00:00:00Z</td>
<td>2018-01-23T21:03:00Z</td>
<td>1</td>
<td>GSM2940100</td>
<td>fibroblast</td>
<td>SRP131117</td>
<td>fibroblast</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
</tr>
</tbody>
</table>

<p>5 rows × 30 columns</p>
</div>
</div>
</div>
</section>
<section id="preparation-of-the-.fastq-files" class="level3">
<h3 class="anchored" data-anchor-id="preparation-of-the-.fastq-files">Preparation of the <code>.fastq</code> files</h3>
<p>The <code>.fastq</code> files were all prepared using this <a href="../gwf_map_reads.html">workflow</a>. Briefly, it does three things on the reference genome (Mmul_10/rheMac10):</p>
<ol type="1">
<li>indexing with <code>bwa index</code><br>
</li>
<li>indexing with <code>samtools faidx</code></li>
<li>indexing of restriction sites of the used restriction enzyme used (<code>hicFindRestSites</code>)</li>
</ol>
<p>Then, it maps all the <code>.fastq</code> files (each mate individually) back to the reference genome with <code>bwa mem</code>.</p>
<p>Finally, it builds the <code>.h5</code> Hi-C matrix with <code>hicBuildMatrix</code>, which includes some quality control of the Hi-C library. The results from the quality control can be viewed <a href="../results/Readme.html">here</a>.</p>
</section>
</section>
<section id="hicexplorer" class="level2">
<h2 class="anchored" data-anchor-id="hicexplorer">HiCExplorer</h2>
<div id="cell-8" class="cell" data-execution_count="15">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="op">!</span>hicexplorer <span class="op">--</span><span class="bu">help</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>usage: hicexplorer [-h] [--version]

HiCExplorer addresses the common tasks of Hi-C analysis from processing to visualization.
Each tool should be called by its own name as in the following example:

 $ hicPlotMatrix -m hic_matrix.h5 -o plot.pdf

If you find HiCExplorer useful for your research please cite as:

Fidel Ramirez, Vivek Bhardwaj, Jose Villaveces, Laura Arrigoni, Bjoern A Gruening, Kin Chung Lam,
Bianca Habermann, Asifa Akhtar, Thomas Manke.
"High-resolution TADs reveal DNA sequences underlying genome organization in flies".
Nature Communications, Volume 9, Article number: 189 (2018), doi: https://doi.org/10.1038/s41467-017-02525-w

Joachim Wolff, Vivek Bhardwaj, Stephan Nothjunge, Gautier Richard, Gina Renschler, Ralf Gilsbach,
Thomas Manke, Rolf Backofen, Fidel Ramírez, Björn A Grüning.
"Galaxy HiCExplorer: a web server for reproducible Hi-C data analysis, quality control and visualization",
Nucleic Acids Research, Volume 46, Issue W1, 2 July 2018, Pages W11–W16, doi: https://doi.org/10.1093/nar/gky504

The following is the list of tools:

findRestSite                 Identifies the genomic locations of restriction sites
hicBuildMatrix               Creates a Hi-C matrix using the aligned BAM files of the Hi-C sequencing reads
hicQuickQC                   Estimates the quality of Hi-C dataset
hicQC                        Plots QC measures from the output of hicBuildMatrix
hicCorrectMatrix             Uses iterative correction to remove biases from a Hi-C matrix
hicDetectLoops               Identifies enriched Hi-C contacts
hicCorrelate                 Computes and visualizes the correlation of Hi-C matrices
hicFindTADs                  Identifies Topologically Associating Domains (TADs)
hicPCA                       Computes for A / B compartments the eigenvectors
hicTransform                 Computes a obs_exp matrix like Lieberman-Aiden (2009), a pearson correlation matrix and or a covariance matrix. These matrices can be used for plotting.
hicMergeMatrixBins           Merges consecutive bins on a Hi-C matrix to reduce resolution
hicMergeTADbins              Uses a BED file of domains or TAD boundaries to merge the bin counts of a Hi-C matrix.
hicPlotDistVsCounts          Plot the decay in interaction frequency with distance
hicPlotMatrix                Plots a Hi-C matrix as a heatmap
hicPlotTADs                  Plots TADs as a track that can be combined with other tracks (genes, signal, interactions)
hicPlotViewpoint             A plot with the interactions around a reference point or region.
hicAggregateContacts         A tool that allows plotting of aggregated Hi-C sub-matrices of a specified list of positions.
hicSumMatrices               Adds Hi-C matrices of the same size
hicPlotDistVsCounts          Plots distance vs. Hi-C counts of corrected data
hicInfo                      Shows information about a Hi-C matrix file (no. of bins, bin length, sum, max, min, etc)
hicCompareMatrices           Computes difference or ratio between two matrices
hicAverageRegions            Computes the average of multiple given regions, usually TAD regions
hicPlotAverageRegions        visualization of hicAverageRegions
hicNormalize                 Normalizes the given matrices to 0-1 range or the smallest read coverage
hicConvertFormat             Converts between different Hi-C interaction matrices
hicAdjustMatrix              Keeps, removes or masks regions in a Hi-C matrix
hicValidateLocations         Compare the loops with known peak protein locations
hicMergeLoops                Merges loops of different resolutions
hicCompartmentsPolarization  Compute the global compartmentalization signal
chicQualityControl           Quality control for cHi-C data
chicViewpointBackgroundModel Background model computation for cHi-C analysis
chicViewpoint                Computation of all viewpoints based on background model for cHi-C analysis
chicSignificantInteractions  Detection of significant interactions per viewpoint based on background model
chicAggregateStatistic       Compiling of target regions for two samples as input for differential analysis
chicDifferentialTest         Differential analysis of interactions of two samples
chicPlotViewpoint            Plotting of viewpoint with background model and highlighting of significant and differential regions
hicPlotSVL                   Computing short vs long range contacts and plotting the results

For more information visit: http://hicexplorer.readthedocs.org

options:
  -h, --help  show this help message and exit
  --version   show program's version number and exit</code></pre>
</div>
</div>
<div id="cell-9" class="cell" data-execution_count="20">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="op">!</span>hicexplorer hicBuildMatrix </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>usage: hicexplorer [-h] [--version]

HiCExplorer addresses the common tasks of Hi-C analysis from processing to visualization.
Each tool should be called by its own name as in the following example:

 $ hicPlotMatrix -m hic_matrix.h5 -o plot.pdf

If you find HiCExplorer useful for your research please cite as:

Fidel Ramirez, Vivek Bhardwaj, Jose Villaveces, Laura Arrigoni, Bjoern A Gruening, Kin Chung Lam,
Bianca Habermann, Asifa Akhtar, Thomas Manke.
"High-resolution TADs reveal DNA sequences underlying genome organization in flies".
Nature Communications, Volume 9, Article number: 189 (2018), doi: https://doi.org/10.1038/s41467-017-02525-w

Joachim Wolff, Vivek Bhardwaj, Stephan Nothjunge, Gautier Richard, Gina Renschler, Ralf Gilsbach,
Thomas Manke, Rolf Backofen, Fidel Ramírez, Björn A Grüning.
"Galaxy HiCExplorer: a web server for reproducible Hi-C data analysis, quality control and visualization",
Nucleic Acids Research, Volume 46, Issue W1, 2 July 2018, Pages W11–W16, doi: https://doi.org/10.1093/nar/gky504

The following is the list of tools:

findRestSite                 Identifies the genomic locations of restriction sites
hicBuildMatrix               Creates a Hi-C matrix using the aligned BAM files of the Hi-C sequencing reads
hicQuickQC                   Estimates the quality of Hi-C dataset
hicQC                        Plots QC measures from the output of hicBuildMatrix
hicCorrectMatrix             Uses iterative correction to remove biases from a Hi-C matrix
hicDetectLoops               Identifies enriched Hi-C contacts
hicCorrelate                 Computes and visualizes the correlation of Hi-C matrices
hicFindTADs                  Identifies Topologically Associating Domains (TADs)
hicPCA                       Computes for A / B compartments the eigenvectors
hicTransform                 Computes a obs_exp matrix like Lieberman-Aiden (2009), a pearson correlation matrix and or a covariance matrix. These matrices can be used for plotting.
hicMergeMatrixBins           Merges consecutive bins on a Hi-C matrix to reduce resolution
hicMergeTADbins              Uses a BED file of domains or TAD boundaries to merge the bin counts of a Hi-C matrix.
hicPlotDistVsCounts          Plot the decay in interaction frequency with distance
hicPlotMatrix                Plots a Hi-C matrix as a heatmap
hicPlotTADs                  Plots TADs as a track that can be combined with other tracks (genes, signal, interactions)
hicPlotViewpoint             A plot with the interactions around a reference point or region.
hicAggregateContacts         A tool that allows plotting of aggregated Hi-C sub-matrices of a specified list of positions.
hicSumMatrices               Adds Hi-C matrices of the same size
hicPlotDistVsCounts          Plots distance vs. Hi-C counts of corrected data
hicInfo                      Shows information about a Hi-C matrix file (no. of bins, bin length, sum, max, min, etc)
hicCompareMatrices           Computes difference or ratio between two matrices
hicAverageRegions            Computes the average of multiple given regions, usually TAD regions
hicPlotAverageRegions        visualization of hicAverageRegions
hicNormalize                 Normalizes the given matrices to 0-1 range or the smallest read coverage
hicConvertFormat             Converts between different Hi-C interaction matrices
hicAdjustMatrix              Keeps, removes or masks regions in a Hi-C matrix
hicValidateLocations         Compare the loops with known peak protein locations
hicMergeLoops                Merges loops of different resolutions
hicCompartmentsPolarization  Compute the global compartmentalization signal
chicQualityControl           Quality control for cHi-C data
chicViewpointBackgroundModel Background model computation for cHi-C analysis
chicViewpoint                Computation of all viewpoints based on background model for cHi-C analysis
chicSignificantInteractions  Detection of significant interactions per viewpoint based on background model
chicAggregateStatistic       Compiling of target regions for two samples as input for differential analysis
chicDifferentialTest         Differential analysis of interactions of two samples
chicPlotViewpoint            Plotting of viewpoint with background model and highlighting of significant and differential regions
hicPlotSVL                   Computing short vs long range contacts and plotting the results

For more information visit: http://hicexplorer.readthedocs.org

options:
  -h, --help  show this help message and exit
  --version   show program's version number and exit</code></pre>
</div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../notebooks/002_interaction.html" class="pagination-link" aria-label="Workplace interaction">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-title">Workplace interaction</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../gwf_map_reads.html" class="pagination-link" aria-label="gwf_map_reads">
        <span class="nav-page-text"><span class="chapter-title">gwf_map_reads</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->




</body></html>