{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# HiCExplorer using macaque data\n",
    "\n",
    "***Before using this notebook, make sure you use the 'hic' kernel.*** \n",
    "\n",
    "I might change the env into a seperate HiC env for each of the tools, such as HICExplorer, Cooler/cooltools, etc. \n",
    "\n",
    "As of this update, there is only one environment in the repo, and it contains both Python, R, and CLI tools for use in HI-C data analysis. \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Getting started\n",
    "\n",
    "In this notebook, we will use HiCExplorer to analysis Hi-C reads from macaque monkeys. We will follow an [example](https://hicexplorer.readthedocs.io/en/latest/content/example_usage.html#how-we-use-hicexplorer) from the HiCExplorer documentation.\n",
    "\n",
    "First, let's look at the files we have:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../../../kmt/hicmaps/macaque/GSE109344_data/data\n",
      "├── GSE109344_monkey_fibro.allValidPairs -> GSE109344_monkey_fibro_allValidPairs.txt\n",
      "├── GSE109344_monkey_fibro_allValidPairs.txt\n",
      "├── GSE109344_monkey_fibro_pac_rs_gene.fpkm -> GSE109344_monkey_fibro_pac_rs_gene.fpkm.txt\n",
      "├── GSE109344_monkey_fibro_pac_rs_gene.fpkm.txt\n",
      "├── GSE109344_monkey_pac.allValidPairs -> GSE109344_monkey_pac_allValidPairs.txt\n",
      "├── GSE109344_monkey_pac_allValidPairs.txt\n",
      "├── GSE109344_monkey_rs.allValidPairs -> GSE109344_monkey_rs_allValidPairs.txt\n",
      "├── GSE109344_monkey_rs_allValidPairs.txt\n",
      "├── GSE109344_monkey_spa.allValidPairs -> GSE109344_monkey_spa_allValidPairs.txt\n",
      "├── GSE109344_monkey_spa_allValidPairs.txt\n",
      "├── GSE109344_monkey_sperm.allValidPairs -> GSE109344_monkey_sperm_allValidPairs.txt\n",
      "└── GSE109344_monkey_sperm_allValidPairs.txt\n",
      "\n",
      "0 directories, 12 files\n",
      "../data/links\n",
      "└── macaque_data -> ../../../../kmt/hicmaps/macaque/GSE109344_data/data\n",
      "\n",
      "1 directory, 0 files\n"
     ]
    }
   ],
   "source": [
    "!tree ../../../kmt/hicmaps/macaque/GSE109344_data/data\n",
    "\n",
    "!tree ../data/links"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### What is inside the files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 306G\n",
      "lrwxrwxrwx 1 kmt baboondiversity   40 May  2 16:08 GSE109344_monkey_fibro.allValidPairs -> GSE109344_monkey_fibro_allValidPairs.txt\n",
      "-rw-rw-r-- 1 kmt baboondiversity  61G May  1 14:28 GSE109344_monkey_fibro_allValidPairs.txt\n",
      "lrwxrwxrwx 1 kmt baboondiversity   43 May  2 16:09 GSE109344_monkey_fibro_pac_rs_gene.fpkm -> GSE109344_monkey_fibro_pac_rs_gene.fpkm.txt\n",
      "-rw-rw-r-- 1 kmt baboondiversity 1.1M May  1 14:28 GSE109344_monkey_fibro_pac_rs_gene.fpkm.txt\n",
      "lrwxrwxrwx 1 kmt baboondiversity   38 May  2 16:09 GSE109344_monkey_pac.allValidPairs -> GSE109344_monkey_pac_allValidPairs.txt\n",
      "-rw-rw-r-- 1 kmt baboondiversity  82G May  1 14:43 GSE109344_monkey_pac_allValidPairs.txt\n",
      "lrwxrwxrwx 1 kmt baboondiversity   37 May  2 16:10 GSE109344_monkey_rs.allValidPairs -> GSE109344_monkey_rs_allValidPairs.txt\n",
      "-rw-rw-r-- 1 kmt baboondiversity  62G May  1 14:55 GSE109344_monkey_rs_allValidPairs.txt\n",
      "lrwxrwxrwx 1 kmt baboondiversity   38 May  2 16:10 GSE109344_monkey_spa.allValidPairs -> GSE109344_monkey_spa_allValidPairs.txt\n",
      "-rw-rw-r-- 1 kmt baboondiversity  58G May  1 15:05 GSE109344_monkey_spa_allValidPairs.txt\n",
      "lrwxrwxrwx 1 kmt baboondiversity   40 May  2 16:10 GSE109344_monkey_sperm.allValidPairs -> GSE109344_monkey_sperm_allValidPairs.txt\n",
      "-rw-rw-r-- 1 kmt baboondiversity  46G May  1 15:13 GSE109344_monkey_sperm_allValidPairs.txt\n",
      "\n",
      ">> head -n2 GSE109344_monkey_fibro.allValidPairs\n",
      "ST-E00545:100:HCNYYALXX:4:2115:5142:66162\tchr1\t31\t-\tchr5\t214251\t-\t93\t\n",
      "ST-E00545:100:HCNYYALXX:4:2116:23947:59182\tchr1\t32\t-\tchr15\t57886364\t-\t178\t\n",
      "\n",
      ">> head -n2 GSE109344_monkey_fibro_allValidPairs.txt\n",
      "ST-E00545:100:HCNYYALXX:4:2115:5142:66162\tchr1\t31\t-\tchr5\t214251\t-\t93\t\n",
      "ST-E00545:100:HCNYYALXX:4:2116:23947:59182\tchr1\t32\t-\tchr15\t57886364\t-\t178\t\n",
      "\n",
      ">> head -n2 GSE109344_monkey_fibro_pac_rs_gene.fpkm\n",
      "Gene\tfibroblast\tpac\trs\n",
      "IMMRNTR1522\t0\t38716.9\t81725.3\n",
      "\n",
      ">> head -n2 GSE109344_monkey_fibro_pac_rs_gene.fpkm.txt\n",
      "Gene\tfibroblast\tpac\trs\n",
      "IMMRNTR1522\t0\t38716.9\t81725.3\n",
      "\n",
      ">> head -n2 GSE109344_monkey_pac.allValidPairs\n",
      "ST-E00545:324:HL3M7CCXY:8:1110:12053:15619\tchr1\t32\t-\tchr5\t47163\t+\t270\tHIC_chr1_1\tHIC_chr5_88\t34\t23\t\n",
      "ST-E00545:324:HL3M7CCXY:8:1203:26058:21667\tchr1\t32\t-\tchr5\t234437\t-\t308\tHIC_chr1_1\tHIC_chr5_521\t34\t40\t\n",
      "\n",
      ">> head -n2 GSE109344_monkey_pac_allValidPairs.txt\n",
      "ST-E00545:324:HL3M7CCXY:8:1110:12053:15619\tchr1\t32\t-\tchr5\t47163\t+\t270\tHIC_chr1_1\tHIC_chr5_88\t34\t23\t\n",
      "ST-E00545:324:HL3M7CCXY:8:1203:26058:21667\tchr1\t32\t-\tchr5\t234437\t-\t308\tHIC_chr1_1\tHIC_chr5_521\t34\t40\t\n",
      "\n",
      ">> head -n2 GSE109344_monkey_rs.allValidPairs\n",
      "ST-E00205:521:HCM3LCCXY:7:2222:12784:40337\tchr1\t31\t-\tchr17\t4099589\t-\t281\t\n",
      "ST-E00318:369:H33LLCCXY:4:1121:22414:51852\tchr1\t33\t-\tchr6\t16939862\t+\t186\t\n",
      "\n",
      ">> head -n2 GSE109344_monkey_rs_allValidPairs.txt\n",
      "ST-E00205:521:HCM3LCCXY:7:2222:12784:40337\tchr1\t31\t-\tchr17\t4099589\t-\t281\t\n",
      "ST-E00318:369:H33LLCCXY:4:1121:22414:51852\tchr1\t33\t-\tchr6\t16939862\t+\t186\t\n",
      "\n",
      ">> head -n2 GSE109344_monkey_spa.allValidPairs\n",
      "ST-E00251:501:HK75NCCXY:6:1116:10358:22845\tchr1\t31\t-\tchr3\t47548612\t-\t112\t\n",
      "ST-E00493:390:HL53FCCXY:1:1204:16112:11505\tchr1\t31\t-\tchr7\t121241748\t-\t122\t\n",
      "\n",
      ">> head -n2 GSE109344_monkey_spa_allValidPairs.txt\n",
      "ST-E00251:501:HK75NCCXY:6:1116:10358:22845\tchr1\t31\t-\tchr3\t47548612\t-\t112\t\n",
      "ST-E00493:390:HL53FCCXY:1:1204:16112:11505\tchr1\t31\t-\tchr7\t121241748\t-\t122\t\n",
      "\n",
      ">> head -n2 GSE109344_monkey_sperm.allValidPairs\n",
      "ST-E00545:312:HL3M3CCXY:1:1211:18000:72403\tchr1\t31\t-\tchr1\t9603\t-\t171\tHIC_chr1_1\tHIC_chr1_24\t34\t42\t\n",
      "ST-E00545:96:HCTC2ALXX:1:1119:21704:24532\tchr1\t31\t-\tchr5\t15448278\t-\t228\t\n",
      "\n",
      ">> head -n2 GSE109344_monkey_sperm_allValidPairs.txt\n",
      "ST-E00545:312:HL3M3CCXY:1:1211:18000:72403\tchr1\t31\t-\tchr1\t9603\t-\t171\tHIC_chr1_1\tHIC_chr1_24\t34\t42\t\n",
      "ST-E00545:96:HCTC2ALXX:1:1119:21704:24532\tchr1\t31\t-\tchr5\t15448278\t-\t228\t\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "ls -lh ../../../kmt/hicmaps/macaque/GSE109344_data/data/$FILE\n",
    "\n",
    "echo \"\"\n",
    "\n",
    "for FILE in $(ls ../../../kmt/hicmaps/macaque/GSE109344_data/data)\n",
    "do \n",
    "    echo \">> head -n2 $FILE\"\n",
    "    head -n2 ../../../kmt/hicmaps/macaque/GSE109344_data/data/$FILE\n",
    "    echo \"\"\n",
    "done\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Convert the format"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1 31 32 chr5 214251 214252\n",
      "chr1 32 33 chr15 57886364 57886365\n",
      "chr1 34 35 chr10 61122746 61122747\n",
      "chr1 34 35 chr14 44036807 44036808\n",
      "chr1 35 36 chr2 82076213 82076214\n",
      "chr1 43 44 chr10 52711984 52711985\n",
      "chr1 44 45 chr1 2602668 2602669\n",
      "chr1 45 46 chr5 11863259 11863260\n",
      "chr1 45 46 chr9 99351902 99351903\n",
      "chr1 48 49 chr5 23829 23830\n",
      "chr1 48 49 chr9 5814466 5814467\n",
      "chr1 49 50 chr1 17555 17556\n",
      "chr1 50 51 chr5 138691168 138691169\n",
      "chr1 51 52 chr1 2784680 2784681\n",
      "chr1 51 52 chr14 1291765 1291766\n",
      "chr1 53 54 chr1 2276683 2276684\n",
      "chr1 53 54 chr14 14068817 14068818\n",
      "chr1 55 56 chr5 39602073 39602074\n",
      "chr1 56 57 chr16 55377362 55377363\n",
      "chr1 57 58 chr1 1388389 1388390\n",
      "chr1 58 59 chrX 84752078 84752079\n",
      "chr1 61 62 chr9 114926426 114926427\n",
      "chr1 63 64 chr1 121156298 121156299\n",
      "chr1 63 64 chr14 10232915 10232916\n",
      "chr1 63 64 chr14 27098869 27098870\n",
      "chr1 65 66 chr6 154971391 154971392\n",
      "chr1 66 67 chr17 10250744 10250745\n",
      "chr1 67 68 chr4 83604512 83604513\n",
      "chr1 67 68 chr16 32192925 32192926\n",
      "chr1 69 70 chr1 36374 36375\n",
      "chr1 69 70 chr14 70671050 70671051\n",
      "chr1 72 73 chrX 61272785 61272786\n",
      "chr1 73 74 chr5 80153 80154\n",
      "chr1 73 74 chr15 73165341 73165342\n",
      "chr1 73 74 chr19 3587963 3587964\n",
      "chr1 74 75 chr1 226514 226515\n",
      "chr1 80 81 chr1 1453 1454\n",
      "chr1 81 82 chr19 18824065 18824066\n",
      "chr1 82 83 chr2 93115808 93115809\n",
      "chr1 82 83 chr5 137645497 137645498\n",
      "chr1 83 84 chr11 46075640 46075641\n",
      "chr1 85 86 chr14 106936129 106936130\n",
      "chr1 87 88 chr4 41770319 41770320\n",
      "chr1 90 91 chr13 10866932 10866933\n",
      "chr1 93 94 chr11 107900175 107900176\n",
      "chr1 95 96 chr2 91918116 91918117\n",
      "chr1 96 97 chr5 242148 242149\n",
      "chr1 96 97 chr6 22864665 22864666\n",
      "chr1 97 98 chr1 1633436 1633437\n",
      "chr1 105 106 chr5 65613241 65613242\n",
      "chr1 119 120 chr2 81695434 81695435\n",
      "chr1 121 122 chr5 2765811 2765812\n",
      "chr1 121 122 chr5 4529396 4529397\n",
      "chr1 129 130 chr1 165715785 165715786\n",
      "chr1 134 135 chr2 39744639 39744640\n",
      "chr1 136 137 chr1 165849335 165849336\n",
      "chr1 138 139 chr20 14091309 14091310\n",
      "chr1 145 146 chr10 63202461 63202462\n",
      "chr1 150 151 chr8 22306836 22306837\n",
      "chr1 151 152 chr5 16869112 16869113\n",
      "chr1 160 161 chr5 2096513 2096514\n",
      "chr1 169 170 chr19 12697547 12697548\n",
      "chr1 170 171 chr16 11872842 11872843\n",
      "chr1 174 175 chr13 12843260 12843261\n",
      "chr1 175 176 chr1 230273 230274\n",
      "chr1 176 177 chr3 85479576 85479577\n",
      "chr1 184 185 chr5 138050309 138050310\n",
      "chr1 206 207 chr14 53446585 53446586\n",
      "chr1 207 208 chr5 34003 34004\n",
      "chr1 220 221 chr1 2327 2328\n",
      "chr1 221 222 chr1 3886 3887\n",
      "chr1 238 239 chr1 1228118 1228119\n",
      "chr1 246 247 chr5 312802 312803\n",
      "chr1 259 260 chr1 17619475 17619476\n",
      "chr1 267 268 chr13 97219444 97219445\n",
      "chr1 269 270 chr20 82744470 82744471\n",
      "chr1 275 276 chr1 1523424 1523425\n",
      "chr1 278 279 chr12 104205714 104205715\n",
      "chr1 287 288 chr1 177215 177216\n",
      "chr1 289 290 chr5 357418 357419\n",
      "chr1 293 294 chr3 809699 809700\n",
      "chr1 293 294 chr8 93911893 93911894\n",
      "chr1 300 301 chr1 23204 23205\n",
      "chr1 303 304 chr1 97899710 97899711\n",
      "chr1 304 305 chr2 182661650 182661651\n",
      "chr1 304 305 chr5 1822771 1822772\n",
      "chr1 311 312 chr1 253438 253439\n",
      "chr1 312 313 chr1 1250 1251\n",
      "chr1 324 325 chr1 1272 1273\n",
      "chr1 324 325 chr7 134171137 134171138\n",
      "chr1 327 328 chr5 38878505 38878506\n",
      "chr1 328 329 chr1 1254 1255\n",
      "chr1 328 329 chr5 79810529 79810530\n",
      "chr1 330 331 chr9 62360904 62360905\n",
      "chr1 330 331 chr12 55226467 55226468\n",
      "chr1 355 356 chr8 136291178 136291179\n",
      "chr1 363 364 chr1 105043773 105043774\n",
      "chr1 366 367 chr5 9410430 9410431\n",
      "chr1 366 367 chr7 59105724 59105725\n",
      "chr1 371 372 chr13 67808706 67808707\n"
     ]
    }
   ],
   "source": [
    "!head -n 100 ../data/links/macaque_data/GSE109344_monkey_fibro.allValidPairs | awk '{print $2, $3, $3+1, $5, $6, $6+1}'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## HiCExplorer\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "usage: hicexplorer [-h] [--version]\n",
      "\n",
      "HiCExplorer addresses the common tasks of Hi-C analysis from processing to visualization.\n",
      "Each tool should be called by its own name as in the following example:\n",
      "\n",
      " $ hicPlotMatrix -m hic_matrix.h5 -o plot.pdf\n",
      "\n",
      "If you find HiCExplorer useful for your research please cite as:\n",
      "\n",
      "Fidel Ramirez, Vivek Bhardwaj, Jose Villaveces, Laura Arrigoni, Bjoern A Gruening, Kin Chung Lam,\n",
      "Bianca Habermann, Asifa Akhtar, Thomas Manke.\n",
      "\"High-resolution TADs reveal DNA sequences underlying genome organization in flies\".\n",
      "Nature Communications, Volume 9, Article number: 189 (2018), doi: https://doi.org/10.1038/s41467-017-02525-w\n",
      "\n",
      "Joachim Wolff, Vivek Bhardwaj, Stephan Nothjunge, Gautier Richard, Gina Renschler, Ralf Gilsbach,\n",
      "Thomas Manke, Rolf Backofen, Fidel Ramírez, Björn A Grüning.\n",
      "\"Galaxy HiCExplorer: a web server for reproducible Hi-C data analysis, quality control and visualization\",\n",
      "Nucleic Acids Research, Volume 46, Issue W1, 2 July 2018, Pages W11–W16, doi: https://doi.org/10.1093/nar/gky504\n",
      "\n",
      "The following is the list of tools:\n",
      "\n",
      "findRestSite                 Identifies the genomic locations of restriction sites\n",
      "hicBuildMatrix               Creates a Hi-C matrix using the aligned BAM files of the Hi-C sequencing reads\n",
      "hicQuickQC                   Estimates the quality of Hi-C dataset\n",
      "hicQC                        Plots QC measures from the output of hicBuildMatrix\n",
      "hicCorrectMatrix             Uses iterative correction to remove biases from a Hi-C matrix\n",
      "hicDetectLoops               Identifies enriched Hi-C contacts\n",
      "hicCorrelate                 Computes and visualizes the correlation of Hi-C matrices\n",
      "hicFindTADs                  Identifies Topologically Associating Domains (TADs)\n",
      "hicPCA                       Computes for A / B compartments the eigenvectors\n",
      "hicTransform                 Computes a obs_exp matrix like Lieberman-Aiden (2009), a pearson correlation matrix and or a covariance matrix. These matrices can be used for plotting.\n",
      "hicMergeMatrixBins           Merges consecutive bins on a Hi-C matrix to reduce resolution\n",
      "hicMergeTADbins              Uses a BED file of domains or TAD boundaries to merge the bin counts of a Hi-C matrix.\n",
      "hicPlotDistVsCounts          Plot the decay in interaction frequency with distance\n",
      "hicPlotMatrix                Plots a Hi-C matrix as a heatmap\n",
      "hicPlotTADs                  Plots TADs as a track that can be combined with other tracks (genes, signal, interactions)\n",
      "hicPlotViewpoint             A plot with the interactions around a reference point or region.\n",
      "hicAggregateContacts         A tool that allows plotting of aggregated Hi-C sub-matrices of a specified list of positions.\n",
      "hicSumMatrices               Adds Hi-C matrices of the same size\n",
      "hicPlotDistVsCounts          Plots distance vs. Hi-C counts of corrected data\n",
      "hicInfo                      Shows information about a Hi-C matrix file (no. of bins, bin length, sum, max, min, etc)\n",
      "hicCompareMatrices           Computes difference or ratio between two matrices\n",
      "hicAverageRegions            Computes the average of multiple given regions, usually TAD regions\n",
      "hicPlotAverageRegions        visualization of hicAverageRegions\n",
      "hicNormalize                 Normalizes the given matrices to 0-1 range or the smallest read coverage\n",
      "hicConvertFormat             Converts between different Hi-C interaction matrices\n",
      "hicAdjustMatrix              Keeps, removes or masks regions in a Hi-C matrix\n",
      "hicValidateLocations         Compare the loops with known peak protein locations\n",
      "hicMergeLoops                Merges loops of different resolutions\n",
      "hicCompartmentsPolarization  Compute the global compartmentalization signal\n",
      "chicQualityControl           Quality control for cHi-C data\n",
      "chicViewpointBackgroundModel Background model computation for cHi-C analysis\n",
      "chicViewpoint                Computation of all viewpoints based on background model for cHi-C analysis\n",
      "chicSignificantInteractions  Detection of significant interactions per viewpoint based on background model\n",
      "chicAggregateStatistic       Compiling of target regions for two samples as input for differential analysis\n",
      "chicDifferentialTest         Differential analysis of interactions of two samples\n",
      "chicPlotViewpoint            Plotting of viewpoint with background model and highlighting of significant and differential regions\n",
      "hicPlotSVL                   Computing short vs long range contacts and plotting the results\n",
      "\n",
      "For more information visit: http://hicexplorer.readthedocs.org\n",
      "\n",
      "options:\n",
      "  -h, --help  show this help message and exit\n",
      "  --version   show program's version number and exit\n"
     ]
    }
   ],
   "source": [
    "!hicexplorer --help"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "usage: hicexplorer [-h] [--version]\n",
      "\n",
      "HiCExplorer addresses the common tasks of Hi-C analysis from processing to visualization.\n",
      "Each tool should be called by its own name as in the following example:\n",
      "\n",
      " $ hicPlotMatrix -m hic_matrix.h5 -o plot.pdf\n",
      "\n",
      "If you find HiCExplorer useful for your research please cite as:\n",
      "\n",
      "Fidel Ramirez, Vivek Bhardwaj, Jose Villaveces, Laura Arrigoni, Bjoern A Gruening, Kin Chung Lam,\n",
      "Bianca Habermann, Asifa Akhtar, Thomas Manke.\n",
      "\"High-resolution TADs reveal DNA sequences underlying genome organization in flies\".\n",
      "Nature Communications, Volume 9, Article number: 189 (2018), doi: https://doi.org/10.1038/s41467-017-02525-w\n",
      "\n",
      "Joachim Wolff, Vivek Bhardwaj, Stephan Nothjunge, Gautier Richard, Gina Renschler, Ralf Gilsbach,\n",
      "Thomas Manke, Rolf Backofen, Fidel Ramírez, Björn A Grüning.\n",
      "\"Galaxy HiCExplorer: a web server for reproducible Hi-C data analysis, quality control and visualization\",\n",
      "Nucleic Acids Research, Volume 46, Issue W1, 2 July 2018, Pages W11–W16, doi: https://doi.org/10.1093/nar/gky504\n",
      "\n",
      "The following is the list of tools:\n",
      "\n",
      "findRestSite                 Identifies the genomic locations of restriction sites\n",
      "hicBuildMatrix               Creates a Hi-C matrix using the aligned BAM files of the Hi-C sequencing reads\n",
      "hicQuickQC                   Estimates the quality of Hi-C dataset\n",
      "hicQC                        Plots QC measures from the output of hicBuildMatrix\n",
      "hicCorrectMatrix             Uses iterative correction to remove biases from a Hi-C matrix\n",
      "hicDetectLoops               Identifies enriched Hi-C contacts\n",
      "hicCorrelate                 Computes and visualizes the correlation of Hi-C matrices\n",
      "hicFindTADs                  Identifies Topologically Associating Domains (TADs)\n",
      "hicPCA                       Computes for A / B compartments the eigenvectors\n",
      "hicTransform                 Computes a obs_exp matrix like Lieberman-Aiden (2009), a pearson correlation matrix and or a covariance matrix. These matrices can be used for plotting.\n",
      "hicMergeMatrixBins           Merges consecutive bins on a Hi-C matrix to reduce resolution\n",
      "hicMergeTADbins              Uses a BED file of domains or TAD boundaries to merge the bin counts of a Hi-C matrix.\n",
      "hicPlotDistVsCounts          Plot the decay in interaction frequency with distance\n",
      "hicPlotMatrix                Plots a Hi-C matrix as a heatmap\n",
      "hicPlotTADs                  Plots TADs as a track that can be combined with other tracks (genes, signal, interactions)\n",
      "hicPlotViewpoint             A plot with the interactions around a reference point or region.\n",
      "hicAggregateContacts         A tool that allows plotting of aggregated Hi-C sub-matrices of a specified list of positions.\n",
      "hicSumMatrices               Adds Hi-C matrices of the same size\n",
      "hicPlotDistVsCounts          Plots distance vs. Hi-C counts of corrected data\n",
      "hicInfo                      Shows information about a Hi-C matrix file (no. of bins, bin length, sum, max, min, etc)\n",
      "hicCompareMatrices           Computes difference or ratio between two matrices\n",
      "hicAverageRegions            Computes the average of multiple given regions, usually TAD regions\n",
      "hicPlotAverageRegions        visualization of hicAverageRegions\n",
      "hicNormalize                 Normalizes the given matrices to 0-1 range or the smallest read coverage\n",
      "hicConvertFormat             Converts between different Hi-C interaction matrices\n",
      "hicAdjustMatrix              Keeps, removes or masks regions in a Hi-C matrix\n",
      "hicValidateLocations         Compare the loops with known peak protein locations\n",
      "hicMergeLoops                Merges loops of different resolutions\n",
      "hicCompartmentsPolarization  Compute the global compartmentalization signal\n",
      "chicQualityControl           Quality control for cHi-C data\n",
      "chicViewpointBackgroundModel Background model computation for cHi-C analysis\n",
      "chicViewpoint                Computation of all viewpoints based on background model for cHi-C analysis\n",
      "chicSignificantInteractions  Detection of significant interactions per viewpoint based on background model\n",
      "chicAggregateStatistic       Compiling of target regions for two samples as input for differential analysis\n",
      "chicDifferentialTest         Differential analysis of interactions of two samples\n",
      "chicPlotViewpoint            Plotting of viewpoint with background model and highlighting of significant and differential regions\n",
      "hicPlotSVL                   Computing short vs long range contacts and plotting the results\n",
      "\n",
      "For more information visit: http://hicexplorer.readthedocs.org\n",
      "\n",
      "options:\n",
      "  -h, --help  show this help message and exit\n",
      "  --version   show program's version number and exit\n"
     ]
    }
   ],
   "source": [
    "!hicexplorer hicBuildMatrix "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python (hic)",
   "language": "python",
   "name": "hic"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
