{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# HiCExplorer using macaque data\n",
    "\n",
    "***Before using this notebook, make sure you use the 'hic' kernel.*** \n",
    "\n",
    "I might change the env into a seperate HiC env for each of the tools, such as HICExplorer, Cooler/cooltools, etc. \n",
    "\n",
    "As of this update, there is only one environment in the repo, and it contains both Python, R, and CLI tools for use in HI-C data analysis. \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Getting started\n",
    "\n",
    "In this notebook, we will use HiCExplorer to analysis Hi-C reads from macaque monkeys. We will follow an [example](https://hicexplorer.readthedocs.io/en/latest/content/example_usage.html#how-we-use-hicexplorer) from the HiCExplorer documentation.\n",
    "\n",
    "First, let's look at the files we have. They are from [SRA](https://www.ncbi.nlm.nih.gov/Traces/study/?acc=PRJNA430777&o=acc_s%3Aahttps://www.ncbi.nlm.nih.gov/Traces/study/?acc=PRJNA430777&o=acc_s%3Aa), and are Hi-C sequencing data from macaque fibroblasts. The data is generated for the Wang2019 paper. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[01;34m../data/links/macaque_fastq\u001b[0m\n",
      "├── \u001b[01;36mSRR6502335_1.fastq.gz\u001b[0m -> \u001b[01;31m../../../../../../data/macaque_raw/downloaded/SRR6502335_1.fastq.gz\u001b[0m\n",
      "├── \u001b[01;36mSRR6502335_2.fastq.gz\u001b[0m -> \u001b[01;31m../../../../../../data/macaque_raw/downloaded/SRR6502335_2.fastq.gz\u001b[0m\n",
      "├── \u001b[01;36mSRR6502336_1.fastq.gz\u001b[0m -> \u001b[01;31m../../../../../../data/macaque_raw/downloaded/SRR6502336_1.fastq.gz\u001b[0m\n",
      "├── \u001b[01;36mSRR6502336_2.fastq.gz\u001b[0m -> \u001b[01;31m../../../../../../data/macaque_raw/downloaded/SRR6502336_2.fastq.gz\u001b[0m\n",
      "├── \u001b[01;36mSRR6502337_1.fastq.gz\u001b[0m -> \u001b[01;31m../../../../../../data/macaque_raw/downloaded/SRR6502337_1.fastq.gz\u001b[0m\n",
      "├── \u001b[01;36mSRR6502337_2.fastq.gz\u001b[0m -> \u001b[01;31m../../../../../../data/macaque_raw/downloaded/SRR6502337_2.fastq.gz\u001b[0m\n",
      "├── \u001b[01;36mSRR6502338_1.fastq.gz\u001b[0m -> \u001b[01;31m../../../../../../data/macaque_raw/downloaded/SRR6502338_1.fastq.gz\u001b[0m\n",
      "├── \u001b[01;36mSRR6502338_2.fastq.gz\u001b[0m -> \u001b[01;31m../../../../../../data/macaque_raw/downloaded/SRR6502338_2.fastq.gz\u001b[0m\n",
      "├── \u001b[01;36mSRR6502339_1.fastq.gz\u001b[0m -> \u001b[01;31m../../../../../../data/macaque_raw/downloaded/SRR6502339_1.fastq.gz\u001b[0m\n",
      "└── \u001b[01;36mSRR6502339_2.fastq.gz\u001b[0m -> \u001b[01;31m../../../../../../data/macaque_raw/downloaded/SRR6502339_2.fastq.gz\u001b[0m\n",
      "\n",
      "0 directories, 10 files\n",
      "\u001b[01;34m../../../../data/macaque_raw/downloaded/\u001b[0m\n",
      "├── \u001b[01;31mrheMac10_Mmul10.zip\u001b[0m\n",
      "├── SraRunTable.txt\n",
      "├── \u001b[01;31mSRR6502335_1.fastq.gz\u001b[0m\n",
      "├── \u001b[01;31mSRR6502335_2.fastq.gz\u001b[0m\n",
      "├── \u001b[01;31mSRR6502336_1.fastq.gz\u001b[0m\n",
      "├── \u001b[01;31mSRR6502336_2.fastq.gz\u001b[0m\n",
      "├── \u001b[01;31mSRR6502337_1.fastq.gz\u001b[0m\n",
      "├── \u001b[01;31mSRR6502337_2.fastq.gz\u001b[0m\n",
      "├── \u001b[01;31mSRR6502338_1.fastq.gz\u001b[0m\n",
      "├── \u001b[01;31mSRR6502338_2.fastq.gz\u001b[0m\n",
      "├── \u001b[01;31mSRR6502339_1.fastq.gz\u001b[0m\n",
      "└── \u001b[01;31mSRR6502339_2.fastq.gz\u001b[0m\n",
      "\n",
      "0 directories, 12 files\n",
      "rheMac10_Mmul10.zip\n",
      "SraRunTable.txt\n",
      "SRR6502335_1.fastq.gz\n",
      "SRR6502335_2.fastq.gz\n",
      "SRR6502336_1.fastq.gz\n",
      "SRR6502336_2.fastq.gz\n",
      "SRR6502337_1.fastq.gz\n",
      "SRR6502337_2.fastq.gz\n",
      "SRR6502338_1.fastq.gz\n",
      "SRR6502338_2.fastq.gz\n",
      "SRR6502339_1.fastq.gz\n",
      "SRR6502339_2.fastq.gz\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "\n",
    "tree ../data/links/macaque_fastq\n",
    "\n",
    "tree ../../../../data/macaque_raw/downloaded/\n",
    "\n",
    "ls ../../../../data/macaque_raw/downloaded/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### An overview of the sequences\n",
    "\n",
    "We can see summary statistics for our sequences in the `SraRunTable.txt`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Run</th>\n",
       "      <th>Assay Type</th>\n",
       "      <th>AvgSpotLen</th>\n",
       "      <th>Bases</th>\n",
       "      <th>BioProject</th>\n",
       "      <th>BioSample</th>\n",
       "      <th>Bytes</th>\n",
       "      <th>Center Name</th>\n",
       "      <th>Consent</th>\n",
       "      <th>DATASTORE filetype</th>\n",
       "      <th>...</th>\n",
       "      <th>ReleaseDate</th>\n",
       "      <th>create_date</th>\n",
       "      <th>version</th>\n",
       "      <th>Sample Name</th>\n",
       "      <th>source_name</th>\n",
       "      <th>SRA Study</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>tissue</th>\n",
       "      <th>agent</th>\n",
       "      <th>genotype</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SRR6502335</td>\n",
       "      <td>Hi-C</td>\n",
       "      <td>300</td>\n",
       "      <td>73201141800</td>\n",
       "      <td>PRJNA430777</td>\n",
       "      <td>SAMN08375237</td>\n",
       "      <td>31966430779</td>\n",
       "      <td>GEO</td>\n",
       "      <td>public</td>\n",
       "      <td>fastq,run.zq,sra</td>\n",
       "      <td>...</td>\n",
       "      <td>2019-02-12T00:00:00Z</td>\n",
       "      <td>2018-01-23T23:05:00Z</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM2940099</td>\n",
       "      <td>fibroblast</td>\n",
       "      <td>SRP131117</td>\n",
       "      <td>fibroblast</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SRR6502336</td>\n",
       "      <td>Hi-C</td>\n",
       "      <td>300</td>\n",
       "      <td>65119970100</td>\n",
       "      <td>PRJNA430777</td>\n",
       "      <td>SAMN08375237</td>\n",
       "      <td>24433383054</td>\n",
       "      <td>GEO</td>\n",
       "      <td>public</td>\n",
       "      <td>fastq,run.zq,sra</td>\n",
       "      <td>...</td>\n",
       "      <td>2019-02-12T00:00:00Z</td>\n",
       "      <td>2018-01-24T00:16:00Z</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM2940099</td>\n",
       "      <td>fibroblast</td>\n",
       "      <td>SRP131117</td>\n",
       "      <td>fibroblast</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SRR6502337</td>\n",
       "      <td>Hi-C</td>\n",
       "      <td>300</td>\n",
       "      <td>52769196300</td>\n",
       "      <td>PRJNA430777</td>\n",
       "      <td>SAMN08375236</td>\n",
       "      <td>23015357755</td>\n",
       "      <td>GEO</td>\n",
       "      <td>public</td>\n",
       "      <td>fastq,run.zq,sra</td>\n",
       "      <td>...</td>\n",
       "      <td>2019-02-12T00:00:00Z</td>\n",
       "      <td>2018-01-24T07:21:00Z</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM2940100</td>\n",
       "      <td>fibroblast</td>\n",
       "      <td>SRP131117</td>\n",
       "      <td>fibroblast</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SRR6502338</td>\n",
       "      <td>Hi-C</td>\n",
       "      <td>300</td>\n",
       "      <td>52378949100</td>\n",
       "      <td>PRJNA430777</td>\n",
       "      <td>SAMN08375236</td>\n",
       "      <td>22999581685</td>\n",
       "      <td>GEO</td>\n",
       "      <td>public</td>\n",
       "      <td>fastq,run.zq,sra</td>\n",
       "      <td>...</td>\n",
       "      <td>2019-02-12T00:00:00Z</td>\n",
       "      <td>2018-01-23T22:14:00Z</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM2940100</td>\n",
       "      <td>fibroblast</td>\n",
       "      <td>SRP131117</td>\n",
       "      <td>fibroblast</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SRR6502339</td>\n",
       "      <td>Hi-C</td>\n",
       "      <td>300</td>\n",
       "      <td>28885941600</td>\n",
       "      <td>PRJNA430777</td>\n",
       "      <td>SAMN08375236</td>\n",
       "      <td>10960123150</td>\n",
       "      <td>GEO</td>\n",
       "      <td>public</td>\n",
       "      <td>fastq,run.zq,sra</td>\n",
       "      <td>...</td>\n",
       "      <td>2019-02-12T00:00:00Z</td>\n",
       "      <td>2018-01-23T21:03:00Z</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM2940100</td>\n",
       "      <td>fibroblast</td>\n",
       "      <td>SRP131117</td>\n",
       "      <td>fibroblast</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Run Assay Type  AvgSpotLen        Bases   BioProject     BioSample  \\\n",
       "0  SRR6502335       Hi-C         300  73201141800  PRJNA430777  SAMN08375237   \n",
       "1  SRR6502336       Hi-C         300  65119970100  PRJNA430777  SAMN08375237   \n",
       "2  SRR6502337       Hi-C         300  52769196300  PRJNA430777  SAMN08375236   \n",
       "3  SRR6502338       Hi-C         300  52378949100  PRJNA430777  SAMN08375236   \n",
       "4  SRR6502339       Hi-C         300  28885941600  PRJNA430777  SAMN08375236   \n",
       "\n",
       "         Bytes Center Name Consent DATASTORE filetype  ...  \\\n",
       "0  31966430779         GEO  public   fastq,run.zq,sra  ...   \n",
       "1  24433383054         GEO  public   fastq,run.zq,sra  ...   \n",
       "2  23015357755         GEO  public   fastq,run.zq,sra  ...   \n",
       "3  22999581685         GEO  public   fastq,run.zq,sra  ...   \n",
       "4  10960123150         GEO  public   fastq,run.zq,sra  ...   \n",
       "\n",
       "            ReleaseDate           create_date version Sample Name source_name  \\\n",
       "0  2019-02-12T00:00:00Z  2018-01-23T23:05:00Z       1  GSM2940099  fibroblast   \n",
       "1  2019-02-12T00:00:00Z  2018-01-24T00:16:00Z       1  GSM2940099  fibroblast   \n",
       "2  2019-02-12T00:00:00Z  2018-01-24T07:21:00Z       1  GSM2940100  fibroblast   \n",
       "3  2019-02-12T00:00:00Z  2018-01-23T22:14:00Z       1  GSM2940100  fibroblast   \n",
       "4  2019-02-12T00:00:00Z  2018-01-23T21:03:00Z       1  GSM2940100  fibroblast   \n",
       "\n",
       "   SRA Study   cell_type tissue agent genotype  \n",
       "0  SRP131117  fibroblast    NaN   NaN      NaN  \n",
       "1  SRP131117  fibroblast    NaN   NaN      NaN  \n",
       "2  SRP131117  fibroblast    NaN   NaN      NaN  \n",
       "3  SRP131117  fibroblast    NaN   NaN      NaN  \n",
       "4  SRP131117  fibroblast    NaN   NaN      NaN  \n",
       "\n",
       "[5 rows x 30 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "pd.read_csv(\"../../../../data/macaque_raw/downloaded/SraRunTable.txt\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "## HiCExplorer\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "usage: hicexplorer [-h] [--version]\n",
      "\n",
      "HiCExplorer addresses the common tasks of Hi-C analysis from processing to visualization.\n",
      "Each tool should be called by its own name as in the following example:\n",
      "\n",
      " $ hicPlotMatrix -m hic_matrix.h5 -o plot.pdf\n",
      "\n",
      "If you find HiCExplorer useful for your research please cite as:\n",
      "\n",
      "Fidel Ramirez, Vivek Bhardwaj, Jose Villaveces, Laura Arrigoni, Bjoern A Gruening, Kin Chung Lam,\n",
      "Bianca Habermann, Asifa Akhtar, Thomas Manke.\n",
      "\"High-resolution TADs reveal DNA sequences underlying genome organization in flies\".\n",
      "Nature Communications, Volume 9, Article number: 189 (2018), doi: https://doi.org/10.1038/s41467-017-02525-w\n",
      "\n",
      "Joachim Wolff, Vivek Bhardwaj, Stephan Nothjunge, Gautier Richard, Gina Renschler, Ralf Gilsbach,\n",
      "Thomas Manke, Rolf Backofen, Fidel Ramírez, Björn A Grüning.\n",
      "\"Galaxy HiCExplorer: a web server for reproducible Hi-C data analysis, quality control and visualization\",\n",
      "Nucleic Acids Research, Volume 46, Issue W1, 2 July 2018, Pages W11–W16, doi: https://doi.org/10.1093/nar/gky504\n",
      "\n",
      "The following is the list of tools:\n",
      "\n",
      "findRestSite                 Identifies the genomic locations of restriction sites\n",
      "hicBuildMatrix               Creates a Hi-C matrix using the aligned BAM files of the Hi-C sequencing reads\n",
      "hicQuickQC                   Estimates the quality of Hi-C dataset\n",
      "hicQC                        Plots QC measures from the output of hicBuildMatrix\n",
      "hicCorrectMatrix             Uses iterative correction to remove biases from a Hi-C matrix\n",
      "hicDetectLoops               Identifies enriched Hi-C contacts\n",
      "hicCorrelate                 Computes and visualizes the correlation of Hi-C matrices\n",
      "hicFindTADs                  Identifies Topologically Associating Domains (TADs)\n",
      "hicPCA                       Computes for A / B compartments the eigenvectors\n",
      "hicTransform                 Computes a obs_exp matrix like Lieberman-Aiden (2009), a pearson correlation matrix and or a covariance matrix. These matrices can be used for plotting.\n",
      "hicMergeMatrixBins           Merges consecutive bins on a Hi-C matrix to reduce resolution\n",
      "hicMergeTADbins              Uses a BED file of domains or TAD boundaries to merge the bin counts of a Hi-C matrix.\n",
      "hicPlotDistVsCounts          Plot the decay in interaction frequency with distance\n",
      "hicPlotMatrix                Plots a Hi-C matrix as a heatmap\n",
      "hicPlotTADs                  Plots TADs as a track that can be combined with other tracks (genes, signal, interactions)\n",
      "hicPlotViewpoint             A plot with the interactions around a reference point or region.\n",
      "hicAggregateContacts         A tool that allows plotting of aggregated Hi-C sub-matrices of a specified list of positions.\n",
      "hicSumMatrices               Adds Hi-C matrices of the same size\n",
      "hicPlotDistVsCounts          Plots distance vs. Hi-C counts of corrected data\n",
      "hicInfo                      Shows information about a Hi-C matrix file (no. of bins, bin length, sum, max, min, etc)\n",
      "hicCompareMatrices           Computes difference or ratio between two matrices\n",
      "hicAverageRegions            Computes the average of multiple given regions, usually TAD regions\n",
      "hicPlotAverageRegions        visualization of hicAverageRegions\n",
      "hicNormalize                 Normalizes the given matrices to 0-1 range or the smallest read coverage\n",
      "hicConvertFormat             Converts between different Hi-C interaction matrices\n",
      "hicAdjustMatrix              Keeps, removes or masks regions in a Hi-C matrix\n",
      "hicValidateLocations         Compare the loops with known peak protein locations\n",
      "hicMergeLoops                Merges loops of different resolutions\n",
      "hicCompartmentsPolarization  Compute the global compartmentalization signal\n",
      "chicQualityControl           Quality control for cHi-C data\n",
      "chicViewpointBackgroundModel Background model computation for cHi-C analysis\n",
      "chicViewpoint                Computation of all viewpoints based on background model for cHi-C analysis\n",
      "chicSignificantInteractions  Detection of significant interactions per viewpoint based on background model\n",
      "chicAggregateStatistic       Compiling of target regions for two samples as input for differential analysis\n",
      "chicDifferentialTest         Differential analysis of interactions of two samples\n",
      "chicPlotViewpoint            Plotting of viewpoint with background model and highlighting of significant and differential regions\n",
      "hicPlotSVL                   Computing short vs long range contacts and plotting the results\n",
      "\n",
      "For more information visit: http://hicexplorer.readthedocs.org\n",
      "\n",
      "options:\n",
      "  -h, --help  show this help message and exit\n",
      "  --version   show program's version number and exit\n"
     ]
    }
   ],
   "source": [
    "!hicexplorer --help"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "usage: hicexplorer [-h] [--version]\n",
      "\n",
      "HiCExplorer addresses the common tasks of Hi-C analysis from processing to visualization.\n",
      "Each tool should be called by its own name as in the following example:\n",
      "\n",
      " $ hicPlotMatrix -m hic_matrix.h5 -o plot.pdf\n",
      "\n",
      "If you find HiCExplorer useful for your research please cite as:\n",
      "\n",
      "Fidel Ramirez, Vivek Bhardwaj, Jose Villaveces, Laura Arrigoni, Bjoern A Gruening, Kin Chung Lam,\n",
      "Bianca Habermann, Asifa Akhtar, Thomas Manke.\n",
      "\"High-resolution TADs reveal DNA sequences underlying genome organization in flies\".\n",
      "Nature Communications, Volume 9, Article number: 189 (2018), doi: https://doi.org/10.1038/s41467-017-02525-w\n",
      "\n",
      "Joachim Wolff, Vivek Bhardwaj, Stephan Nothjunge, Gautier Richard, Gina Renschler, Ralf Gilsbach,\n",
      "Thomas Manke, Rolf Backofen, Fidel Ramírez, Björn A Grüning.\n",
      "\"Galaxy HiCExplorer: a web server for reproducible Hi-C data analysis, quality control and visualization\",\n",
      "Nucleic Acids Research, Volume 46, Issue W1, 2 July 2018, Pages W11–W16, doi: https://doi.org/10.1093/nar/gky504\n",
      "\n",
      "The following is the list of tools:\n",
      "\n",
      "findRestSite                 Identifies the genomic locations of restriction sites\n",
      "hicBuildMatrix               Creates a Hi-C matrix using the aligned BAM files of the Hi-C sequencing reads\n",
      "hicQuickQC                   Estimates the quality of Hi-C dataset\n",
      "hicQC                        Plots QC measures from the output of hicBuildMatrix\n",
      "hicCorrectMatrix             Uses iterative correction to remove biases from a Hi-C matrix\n",
      "hicDetectLoops               Identifies enriched Hi-C contacts\n",
      "hicCorrelate                 Computes and visualizes the correlation of Hi-C matrices\n",
      "hicFindTADs                  Identifies Topologically Associating Domains (TADs)\n",
      "hicPCA                       Computes for A / B compartments the eigenvectors\n",
      "hicTransform                 Computes a obs_exp matrix like Lieberman-Aiden (2009), a pearson correlation matrix and or a covariance matrix. These matrices can be used for plotting.\n",
      "hicMergeMatrixBins           Merges consecutive bins on a Hi-C matrix to reduce resolution\n",
      "hicMergeTADbins              Uses a BED file of domains or TAD boundaries to merge the bin counts of a Hi-C matrix.\n",
      "hicPlotDistVsCounts          Plot the decay in interaction frequency with distance\n",
      "hicPlotMatrix                Plots a Hi-C matrix as a heatmap\n",
      "hicPlotTADs                  Plots TADs as a track that can be combined with other tracks (genes, signal, interactions)\n",
      "hicPlotViewpoint             A plot with the interactions around a reference point or region.\n",
      "hicAggregateContacts         A tool that allows plotting of aggregated Hi-C sub-matrices of a specified list of positions.\n",
      "hicSumMatrices               Adds Hi-C matrices of the same size\n",
      "hicPlotDistVsCounts          Plots distance vs. Hi-C counts of corrected data\n",
      "hicInfo                      Shows information about a Hi-C matrix file (no. of bins, bin length, sum, max, min, etc)\n",
      "hicCompareMatrices           Computes difference or ratio between two matrices\n",
      "hicAverageRegions            Computes the average of multiple given regions, usually TAD regions\n",
      "hicPlotAverageRegions        visualization of hicAverageRegions\n",
      "hicNormalize                 Normalizes the given matrices to 0-1 range or the smallest read coverage\n",
      "hicConvertFormat             Converts between different Hi-C interaction matrices\n",
      "hicAdjustMatrix              Keeps, removes or masks regions in a Hi-C matrix\n",
      "hicValidateLocations         Compare the loops with known peak protein locations\n",
      "hicMergeLoops                Merges loops of different resolutions\n",
      "hicCompartmentsPolarization  Compute the global compartmentalization signal\n",
      "chicQualityControl           Quality control for cHi-C data\n",
      "chicViewpointBackgroundModel Background model computation for cHi-C analysis\n",
      "chicViewpoint                Computation of all viewpoints based on background model for cHi-C analysis\n",
      "chicSignificantInteractions  Detection of significant interactions per viewpoint based on background model\n",
      "chicAggregateStatistic       Compiling of target regions for two samples as input for differential analysis\n",
      "chicDifferentialTest         Differential analysis of interactions of two samples\n",
      "chicPlotViewpoint            Plotting of viewpoint with background model and highlighting of significant and differential regions\n",
      "hicPlotSVL                   Computing short vs long range contacts and plotting the results\n",
      "\n",
      "For more information visit: http://hicexplorer.readthedocs.org\n",
      "\n",
      "options:\n",
      "  -h, --help  show this help message and exit\n",
      "  --version   show program's version number and exit\n"
     ]
    }
   ],
   "source": [
    "!hicexplorer hicBuildMatrix "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python (hic)",
   "language": "python",
   "name": "hic"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
