<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>HiCExplorer using macaque data – Søren's MSc project about Hi-C and spermatogenesis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../notebooks/01_weather.html" rel="next">
<link href="../index.html" rel="prev">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../notebooks/01_hicexplorer.html">Notebooks</a></li><li class="breadcrumb-item"><a href="../notebooks/01_hicexplorer.html"><span class="chapter-title">HiCExplorer using macaque data</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">Søren’s MSc project about Hi-C and spermatogenesis</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Project</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">Notebooks</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/01_hicexplorer.html" class="sidebar-item-text sidebar-link active"><span class="chapter-title">HiCExplorer using macaque data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/01_weather.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Danish weather</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/02_interaction.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Workplace interaction</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">Workflows</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../gwf_map_reads.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">gwf_map_reads</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true">
 <span class="menu-text">Reports</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../reports/manuscript.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Correlating Chromatin Compartments with Selection</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../reports/supplementary.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Supplementary info</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../references.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">References</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul class="collapse">
  <li><a href="#getting-started" id="toc-getting-started" class="nav-link active" data-scroll-target="#getting-started">Getting started</a></li>
  <li><a href="#convert-the-format" id="toc-convert-the-format" class="nav-link" data-scroll-target="#convert-the-format">Convert the format</a></li>
  <li><a href="#hicexplorer" id="toc-hicexplorer" class="nav-link" data-scroll-target="#hicexplorer">HiCExplorer</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../notebooks/01_hicexplorer.html">Notebooks</a></li><li class="breadcrumb-item"><a href="../notebooks/01_hicexplorer.html"><span class="chapter-title">HiCExplorer using macaque data</span></a></li></ol></nav>
<div class="quarto-title">
<h1 class="title"><span class="chapter-title">HiCExplorer using macaque data</span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<p><strong><em>Before using this notebook, make sure you use the ‘hic’ kernel.</em></strong></p>
<p>I might change the env into a seperate HiC env for each of the tools, such as HICExplorer, Cooler/cooltools, etc.</p>
<p>As of this update, there is only one environment in the repo, and it contains both Python, R, and CLI tools for use in HI-C data analysis.</p>
<section id="getting-started" class="level2">
<h2 class="anchored" data-anchor-id="getting-started">Getting started</h2>
<p>In this notebook, we will use HiCExplorer to analysis Hi-C reads from macaque monkeys. We will follow an <a href="https://hicexplorer.readthedocs.io/en/latest/content/example_usage.html#how-we-use-hicexplorer">example</a> from the HiCExplorer documentation.</p>
<p>First, let’s look at the files we have:</p>
<div id="cell-2" class="cell" data-execution_count="21">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="op">!</span>tree ..<span class="op">/</span>..<span class="op">/</span>..<span class="op">/</span>kmt<span class="op">/</span>hicmaps<span class="op">/</span>macaque<span class="op">/</span>GSE109344_data<span class="op">/</span>data</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="op">!</span>tree ..<span class="op">/</span>data<span class="op">/</span>links</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>../../../kmt/hicmaps/macaque/GSE109344_data/data
├── GSE109344_monkey_fibro.allValidPairs -&gt; GSE109344_monkey_fibro_allValidPairs.txt
├── GSE109344_monkey_fibro_allValidPairs.txt
├── GSE109344_monkey_fibro_pac_rs_gene.fpkm -&gt; GSE109344_monkey_fibro_pac_rs_gene.fpkm.txt
├── GSE109344_monkey_fibro_pac_rs_gene.fpkm.txt
├── GSE109344_monkey_pac.allValidPairs -&gt; GSE109344_monkey_pac_allValidPairs.txt
├── GSE109344_monkey_pac_allValidPairs.txt
├── GSE109344_monkey_rs.allValidPairs -&gt; GSE109344_monkey_rs_allValidPairs.txt
├── GSE109344_monkey_rs_allValidPairs.txt
├── GSE109344_monkey_spa.allValidPairs -&gt; GSE109344_monkey_spa_allValidPairs.txt
├── GSE109344_monkey_spa_allValidPairs.txt
├── GSE109344_monkey_sperm.allValidPairs -&gt; GSE109344_monkey_sperm_allValidPairs.txt
└── GSE109344_monkey_sperm_allValidPairs.txt

0 directories, 12 files
../data/links
└── macaque_data -&gt; ../../../../kmt/hicmaps/macaque/GSE109344_data/data

1 directory, 0 files</code></pre>
</div>
</div>
<section id="what-is-inside-the-files" class="level3">
<h3 class="anchored" data-anchor-id="what-is-inside-the-files">What is inside the files</h3>
<div id="cell-4" class="cell" data-execution_count="18">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="op">%%</span>bash</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>ls <span class="op">-</span>lh ..<span class="op">/</span>..<span class="op">/</span>..<span class="op">/</span>kmt<span class="op">/</span>hicmaps<span class="op">/</span>macaque<span class="op">/</span>GSE109344_data<span class="op">/</span>data<span class="op">/</span>$FILE</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>echo <span class="st">""</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> FILE <span class="kw">in</span> $(ls ..<span class="op">/</span>..<span class="op">/</span>..<span class="op">/</span>kmt<span class="op">/</span>hicmaps<span class="op">/</span>macaque<span class="op">/</span>GSE109344_data<span class="op">/</span>data)</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>do </span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>    echo <span class="st">"&gt;&gt; head -n2 $FILE"</span></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>    head <span class="op">-</span>n2 ..<span class="op">/</span>..<span class="op">/</span>..<span class="op">/</span>kmt<span class="op">/</span>hicmaps<span class="op">/</span>macaque<span class="op">/</span>GSE109344_data<span class="op">/</span>data<span class="op">/</span>$FILE</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>    echo <span class="st">""</span></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>done</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>total 306G
lrwxrwxrwx 1 kmt baboondiversity   40 May  2 16:08 GSE109344_monkey_fibro.allValidPairs -&gt; GSE109344_monkey_fibro_allValidPairs.txt
-rw-rw-r-- 1 kmt baboondiversity  61G May  1 14:28 GSE109344_monkey_fibro_allValidPairs.txt
lrwxrwxrwx 1 kmt baboondiversity   43 May  2 16:09 GSE109344_monkey_fibro_pac_rs_gene.fpkm -&gt; GSE109344_monkey_fibro_pac_rs_gene.fpkm.txt
-rw-rw-r-- 1 kmt baboondiversity 1.1M May  1 14:28 GSE109344_monkey_fibro_pac_rs_gene.fpkm.txt
lrwxrwxrwx 1 kmt baboondiversity   38 May  2 16:09 GSE109344_monkey_pac.allValidPairs -&gt; GSE109344_monkey_pac_allValidPairs.txt
-rw-rw-r-- 1 kmt baboondiversity  82G May  1 14:43 GSE109344_monkey_pac_allValidPairs.txt
lrwxrwxrwx 1 kmt baboondiversity   37 May  2 16:10 GSE109344_monkey_rs.allValidPairs -&gt; GSE109344_monkey_rs_allValidPairs.txt
-rw-rw-r-- 1 kmt baboondiversity  62G May  1 14:55 GSE109344_monkey_rs_allValidPairs.txt
lrwxrwxrwx 1 kmt baboondiversity   38 May  2 16:10 GSE109344_monkey_spa.allValidPairs -&gt; GSE109344_monkey_spa_allValidPairs.txt
-rw-rw-r-- 1 kmt baboondiversity  58G May  1 15:05 GSE109344_monkey_spa_allValidPairs.txt
lrwxrwxrwx 1 kmt baboondiversity   40 May  2 16:10 GSE109344_monkey_sperm.allValidPairs -&gt; GSE109344_monkey_sperm_allValidPairs.txt
-rw-rw-r-- 1 kmt baboondiversity  46G May  1 15:13 GSE109344_monkey_sperm_allValidPairs.txt

&gt;&gt; head -n2 GSE109344_monkey_fibro.allValidPairs
ST-E00545:100:HCNYYALXX:4:2115:5142:66162   chr1    31  -   chr5    214251  -   93  
ST-E00545:100:HCNYYALXX:4:2116:23947:59182  chr1    32  -   chr15   57886364    -   178 

&gt;&gt; head -n2 GSE109344_monkey_fibro_allValidPairs.txt
ST-E00545:100:HCNYYALXX:4:2115:5142:66162   chr1    31  -   chr5    214251  -   93  
ST-E00545:100:HCNYYALXX:4:2116:23947:59182  chr1    32  -   chr15   57886364    -   178 

&gt;&gt; head -n2 GSE109344_monkey_fibro_pac_rs_gene.fpkm
Gene    fibroblast  pac rs
IMMRNTR1522 0   38716.9 81725.3

&gt;&gt; head -n2 GSE109344_monkey_fibro_pac_rs_gene.fpkm.txt
Gene    fibroblast  pac rs
IMMRNTR1522 0   38716.9 81725.3

&gt;&gt; head -n2 GSE109344_monkey_pac.allValidPairs
ST-E00545:324:HL3M7CCXY:8:1110:12053:15619  chr1    32  -   chr5    47163   +   270 HIC_chr1_1  HIC_chr5_88 34  23  
ST-E00545:324:HL3M7CCXY:8:1203:26058:21667  chr1    32  -   chr5    234437  -   308 HIC_chr1_1  HIC_chr5_521    34  40  

&gt;&gt; head -n2 GSE109344_monkey_pac_allValidPairs.txt
ST-E00545:324:HL3M7CCXY:8:1110:12053:15619  chr1    32  -   chr5    47163   +   270 HIC_chr1_1  HIC_chr5_88 34  23  
ST-E00545:324:HL3M7CCXY:8:1203:26058:21667  chr1    32  -   chr5    234437  -   308 HIC_chr1_1  HIC_chr5_521    34  40  

&gt;&gt; head -n2 GSE109344_monkey_rs.allValidPairs
ST-E00205:521:HCM3LCCXY:7:2222:12784:40337  chr1    31  -   chr17   4099589 -   281 
ST-E00318:369:H33LLCCXY:4:1121:22414:51852  chr1    33  -   chr6    16939862    +   186 

&gt;&gt; head -n2 GSE109344_monkey_rs_allValidPairs.txt
ST-E00205:521:HCM3LCCXY:7:2222:12784:40337  chr1    31  -   chr17   4099589 -   281 
ST-E00318:369:H33LLCCXY:4:1121:22414:51852  chr1    33  -   chr6    16939862    +   186 

&gt;&gt; head -n2 GSE109344_monkey_spa.allValidPairs
ST-E00251:501:HK75NCCXY:6:1116:10358:22845  chr1    31  -   chr3    47548612    -   112 
ST-E00493:390:HL53FCCXY:1:1204:16112:11505  chr1    31  -   chr7    121241748   -   122 

&gt;&gt; head -n2 GSE109344_monkey_spa_allValidPairs.txt
ST-E00251:501:HK75NCCXY:6:1116:10358:22845  chr1    31  -   chr3    47548612    -   112 
ST-E00493:390:HL53FCCXY:1:1204:16112:11505  chr1    31  -   chr7    121241748   -   122 

&gt;&gt; head -n2 GSE109344_monkey_sperm.allValidPairs
ST-E00545:312:HL3M3CCXY:1:1211:18000:72403  chr1    31  -   chr1    9603    -   171 HIC_chr1_1  HIC_chr1_24 34  42  
ST-E00545:96:HCTC2ALXX:1:1119:21704:24532   chr1    31  -   chr5    15448278    -   228 

&gt;&gt; head -n2 GSE109344_monkey_sperm_allValidPairs.txt
ST-E00545:312:HL3M3CCXY:1:1211:18000:72403  chr1    31  -   chr1    9603    -   171 HIC_chr1_1  HIC_chr1_24 34  42  
ST-E00545:96:HCTC2ALXX:1:1119:21704:24532   chr1    31  -   chr5    15448278    -   228 
</code></pre>
</div>
</div>
</section>
</section>
<section id="convert-the-format" class="level2">
<h2 class="anchored" data-anchor-id="convert-the-format">Convert the format</h2>
<div id="cell-6" class="cell" data-execution_count="3">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="op">!</span>head <span class="op">-</span>n <span class="dv">100</span> ..<span class="op">/</span>data<span class="op">/</span>links<span class="op">/</span>macaque_data<span class="op">/</span>GSE109344_monkey_fibro.allValidPairs <span class="op">|</span> awk <span class="st">'{print $2, $3, $3+1, $5, $6, $6+1}'</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>chr1 31 32 chr5 214251 214252
chr1 32 33 chr15 57886364 57886365
chr1 34 35 chr10 61122746 61122747
chr1 34 35 chr14 44036807 44036808
chr1 35 36 chr2 82076213 82076214
chr1 43 44 chr10 52711984 52711985
chr1 44 45 chr1 2602668 2602669
chr1 45 46 chr5 11863259 11863260
chr1 45 46 chr9 99351902 99351903
chr1 48 49 chr5 23829 23830
chr1 48 49 chr9 5814466 5814467
chr1 49 50 chr1 17555 17556
chr1 50 51 chr5 138691168 138691169
chr1 51 52 chr1 2784680 2784681
chr1 51 52 chr14 1291765 1291766
chr1 53 54 chr1 2276683 2276684
chr1 53 54 chr14 14068817 14068818
chr1 55 56 chr5 39602073 39602074
chr1 56 57 chr16 55377362 55377363
chr1 57 58 chr1 1388389 1388390
chr1 58 59 chrX 84752078 84752079
chr1 61 62 chr9 114926426 114926427
chr1 63 64 chr1 121156298 121156299
chr1 63 64 chr14 10232915 10232916
chr1 63 64 chr14 27098869 27098870
chr1 65 66 chr6 154971391 154971392
chr1 66 67 chr17 10250744 10250745
chr1 67 68 chr4 83604512 83604513
chr1 67 68 chr16 32192925 32192926
chr1 69 70 chr1 36374 36375
chr1 69 70 chr14 70671050 70671051
chr1 72 73 chrX 61272785 61272786
chr1 73 74 chr5 80153 80154
chr1 73 74 chr15 73165341 73165342
chr1 73 74 chr19 3587963 3587964
chr1 74 75 chr1 226514 226515
chr1 80 81 chr1 1453 1454
chr1 81 82 chr19 18824065 18824066
chr1 82 83 chr2 93115808 93115809
chr1 82 83 chr5 137645497 137645498
chr1 83 84 chr11 46075640 46075641
chr1 85 86 chr14 106936129 106936130
chr1 87 88 chr4 41770319 41770320
chr1 90 91 chr13 10866932 10866933
chr1 93 94 chr11 107900175 107900176
chr1 95 96 chr2 91918116 91918117
chr1 96 97 chr5 242148 242149
chr1 96 97 chr6 22864665 22864666
chr1 97 98 chr1 1633436 1633437
chr1 105 106 chr5 65613241 65613242
chr1 119 120 chr2 81695434 81695435
chr1 121 122 chr5 2765811 2765812
chr1 121 122 chr5 4529396 4529397
chr1 129 130 chr1 165715785 165715786
chr1 134 135 chr2 39744639 39744640
chr1 136 137 chr1 165849335 165849336
chr1 138 139 chr20 14091309 14091310
chr1 145 146 chr10 63202461 63202462
chr1 150 151 chr8 22306836 22306837
chr1 151 152 chr5 16869112 16869113
chr1 160 161 chr5 2096513 2096514
chr1 169 170 chr19 12697547 12697548
chr1 170 171 chr16 11872842 11872843
chr1 174 175 chr13 12843260 12843261
chr1 175 176 chr1 230273 230274
chr1 176 177 chr3 85479576 85479577
chr1 184 185 chr5 138050309 138050310
chr1 206 207 chr14 53446585 53446586
chr1 207 208 chr5 34003 34004
chr1 220 221 chr1 2327 2328
chr1 221 222 chr1 3886 3887
chr1 238 239 chr1 1228118 1228119
chr1 246 247 chr5 312802 312803
chr1 259 260 chr1 17619475 17619476
chr1 267 268 chr13 97219444 97219445
chr1 269 270 chr20 82744470 82744471
chr1 275 276 chr1 1523424 1523425
chr1 278 279 chr12 104205714 104205715
chr1 287 288 chr1 177215 177216
chr1 289 290 chr5 357418 357419
chr1 293 294 chr3 809699 809700
chr1 293 294 chr8 93911893 93911894
chr1 300 301 chr1 23204 23205
chr1 303 304 chr1 97899710 97899711
chr1 304 305 chr2 182661650 182661651
chr1 304 305 chr5 1822771 1822772
chr1 311 312 chr1 253438 253439
chr1 312 313 chr1 1250 1251
chr1 324 325 chr1 1272 1273
chr1 324 325 chr7 134171137 134171138
chr1 327 328 chr5 38878505 38878506
chr1 328 329 chr1 1254 1255
chr1 328 329 chr5 79810529 79810530
chr1 330 331 chr9 62360904 62360905
chr1 330 331 chr12 55226467 55226468
chr1 355 356 chr8 136291178 136291179
chr1 363 364 chr1 105043773 105043774
chr1 366 367 chr5 9410430 9410431
chr1 366 367 chr7 59105724 59105725
chr1 371 372 chr13 67808706 67808707</code></pre>
</div>
</div>
</section>
<section id="hicexplorer" class="level2">
<h2 class="anchored" data-anchor-id="hicexplorer">HiCExplorer</h2>
<div id="cell-8" class="cell" data-execution_count="15">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="op">!</span>hicexplorer <span class="op">--</span><span class="bu">help</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>usage: hicexplorer [-h] [--version]

HiCExplorer addresses the common tasks of Hi-C analysis from processing to visualization.
Each tool should be called by its own name as in the following example:

 $ hicPlotMatrix -m hic_matrix.h5 -o plot.pdf

If you find HiCExplorer useful for your research please cite as:

Fidel Ramirez, Vivek Bhardwaj, Jose Villaveces, Laura Arrigoni, Bjoern A Gruening, Kin Chung Lam,
Bianca Habermann, Asifa Akhtar, Thomas Manke.
"High-resolution TADs reveal DNA sequences underlying genome organization in flies".
Nature Communications, Volume 9, Article number: 189 (2018), doi: https://doi.org/10.1038/s41467-017-02525-w

Joachim Wolff, Vivek Bhardwaj, Stephan Nothjunge, Gautier Richard, Gina Renschler, Ralf Gilsbach,
Thomas Manke, Rolf Backofen, Fidel Ramírez, Björn A Grüning.
"Galaxy HiCExplorer: a web server for reproducible Hi-C data analysis, quality control and visualization",
Nucleic Acids Research, Volume 46, Issue W1, 2 July 2018, Pages W11–W16, doi: https://doi.org/10.1093/nar/gky504

The following is the list of tools:

findRestSite                 Identifies the genomic locations of restriction sites
hicBuildMatrix               Creates a Hi-C matrix using the aligned BAM files of the Hi-C sequencing reads
hicQuickQC                   Estimates the quality of Hi-C dataset
hicQC                        Plots QC measures from the output of hicBuildMatrix
hicCorrectMatrix             Uses iterative correction to remove biases from a Hi-C matrix
hicDetectLoops               Identifies enriched Hi-C contacts
hicCorrelate                 Computes and visualizes the correlation of Hi-C matrices
hicFindTADs                  Identifies Topologically Associating Domains (TADs)
hicPCA                       Computes for A / B compartments the eigenvectors
hicTransform                 Computes a obs_exp matrix like Lieberman-Aiden (2009), a pearson correlation matrix and or a covariance matrix. These matrices can be used for plotting.
hicMergeMatrixBins           Merges consecutive bins on a Hi-C matrix to reduce resolution
hicMergeTADbins              Uses a BED file of domains or TAD boundaries to merge the bin counts of a Hi-C matrix.
hicPlotDistVsCounts          Plot the decay in interaction frequency with distance
hicPlotMatrix                Plots a Hi-C matrix as a heatmap
hicPlotTADs                  Plots TADs as a track that can be combined with other tracks (genes, signal, interactions)
hicPlotViewpoint             A plot with the interactions around a reference point or region.
hicAggregateContacts         A tool that allows plotting of aggregated Hi-C sub-matrices of a specified list of positions.
hicSumMatrices               Adds Hi-C matrices of the same size
hicPlotDistVsCounts          Plots distance vs. Hi-C counts of corrected data
hicInfo                      Shows information about a Hi-C matrix file (no. of bins, bin length, sum, max, min, etc)
hicCompareMatrices           Computes difference or ratio between two matrices
hicAverageRegions            Computes the average of multiple given regions, usually TAD regions
hicPlotAverageRegions        visualization of hicAverageRegions
hicNormalize                 Normalizes the given matrices to 0-1 range or the smallest read coverage
hicConvertFormat             Converts between different Hi-C interaction matrices
hicAdjustMatrix              Keeps, removes or masks regions in a Hi-C matrix
hicValidateLocations         Compare the loops with known peak protein locations
hicMergeLoops                Merges loops of different resolutions
hicCompartmentsPolarization  Compute the global compartmentalization signal
chicQualityControl           Quality control for cHi-C data
chicViewpointBackgroundModel Background model computation for cHi-C analysis
chicViewpoint                Computation of all viewpoints based on background model for cHi-C analysis
chicSignificantInteractions  Detection of significant interactions per viewpoint based on background model
chicAggregateStatistic       Compiling of target regions for two samples as input for differential analysis
chicDifferentialTest         Differential analysis of interactions of two samples
chicPlotViewpoint            Plotting of viewpoint with background model and highlighting of significant and differential regions
hicPlotSVL                   Computing short vs long range contacts and plotting the results

For more information visit: http://hicexplorer.readthedocs.org

options:
  -h, --help  show this help message and exit
  --version   show program's version number and exit</code></pre>
</div>
</div>
<div id="cell-9" class="cell" data-execution_count="20">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="op">!</span>hicexplorer hicBuildMatrix </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>usage: hicexplorer [-h] [--version]

HiCExplorer addresses the common tasks of Hi-C analysis from processing to visualization.
Each tool should be called by its own name as in the following example:

 $ hicPlotMatrix -m hic_matrix.h5 -o plot.pdf

If you find HiCExplorer useful for your research please cite as:

Fidel Ramirez, Vivek Bhardwaj, Jose Villaveces, Laura Arrigoni, Bjoern A Gruening, Kin Chung Lam,
Bianca Habermann, Asifa Akhtar, Thomas Manke.
"High-resolution TADs reveal DNA sequences underlying genome organization in flies".
Nature Communications, Volume 9, Article number: 189 (2018), doi: https://doi.org/10.1038/s41467-017-02525-w

Joachim Wolff, Vivek Bhardwaj, Stephan Nothjunge, Gautier Richard, Gina Renschler, Ralf Gilsbach,
Thomas Manke, Rolf Backofen, Fidel Ramírez, Björn A Grüning.
"Galaxy HiCExplorer: a web server for reproducible Hi-C data analysis, quality control and visualization",
Nucleic Acids Research, Volume 46, Issue W1, 2 July 2018, Pages W11–W16, doi: https://doi.org/10.1093/nar/gky504

The following is the list of tools:

findRestSite                 Identifies the genomic locations of restriction sites
hicBuildMatrix               Creates a Hi-C matrix using the aligned BAM files of the Hi-C sequencing reads
hicQuickQC                   Estimates the quality of Hi-C dataset
hicQC                        Plots QC measures from the output of hicBuildMatrix
hicCorrectMatrix             Uses iterative correction to remove biases from a Hi-C matrix
hicDetectLoops               Identifies enriched Hi-C contacts
hicCorrelate                 Computes and visualizes the correlation of Hi-C matrices
hicFindTADs                  Identifies Topologically Associating Domains (TADs)
hicPCA                       Computes for A / B compartments the eigenvectors
hicTransform                 Computes a obs_exp matrix like Lieberman-Aiden (2009), a pearson correlation matrix and or a covariance matrix. These matrices can be used for plotting.
hicMergeMatrixBins           Merges consecutive bins on a Hi-C matrix to reduce resolution
hicMergeTADbins              Uses a BED file of domains or TAD boundaries to merge the bin counts of a Hi-C matrix.
hicPlotDistVsCounts          Plot the decay in interaction frequency with distance
hicPlotMatrix                Plots a Hi-C matrix as a heatmap
hicPlotTADs                  Plots TADs as a track that can be combined with other tracks (genes, signal, interactions)
hicPlotViewpoint             A plot with the interactions around a reference point or region.
hicAggregateContacts         A tool that allows plotting of aggregated Hi-C sub-matrices of a specified list of positions.
hicSumMatrices               Adds Hi-C matrices of the same size
hicPlotDistVsCounts          Plots distance vs. Hi-C counts of corrected data
hicInfo                      Shows information about a Hi-C matrix file (no. of bins, bin length, sum, max, min, etc)
hicCompareMatrices           Computes difference or ratio between two matrices
hicAverageRegions            Computes the average of multiple given regions, usually TAD regions
hicPlotAverageRegions        visualization of hicAverageRegions
hicNormalize                 Normalizes the given matrices to 0-1 range or the smallest read coverage
hicConvertFormat             Converts between different Hi-C interaction matrices
hicAdjustMatrix              Keeps, removes or masks regions in a Hi-C matrix
hicValidateLocations         Compare the loops with known peak protein locations
hicMergeLoops                Merges loops of different resolutions
hicCompartmentsPolarization  Compute the global compartmentalization signal
chicQualityControl           Quality control for cHi-C data
chicViewpointBackgroundModel Background model computation for cHi-C analysis
chicViewpoint                Computation of all viewpoints based on background model for cHi-C analysis
chicSignificantInteractions  Detection of significant interactions per viewpoint based on background model
chicAggregateStatistic       Compiling of target regions for two samples as input for differential analysis
chicDifferentialTest         Differential analysis of interactions of two samples
chicPlotViewpoint            Plotting of viewpoint with background model and highlighting of significant and differential regions
hicPlotSVL                   Computing short vs long range contacts and plotting the results

For more information visit: http://hicexplorer.readthedocs.org

options:
  -h, --help  show this help message and exit
  --version   show program's version number and exit</code></pre>
</div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../index.html" class="pagination-link" aria-label="Project">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-title">Project</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../notebooks/01_weather.html" class="pagination-link" aria-label="Danish weather">
        <span class="nav-page-text"><span class="chapter-title">Danish weather</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->




</body></html>