<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Working with coolers – Søren's MSc project about Hi-C and spermatogenesis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js" integrity="sha512-bLT0Qm9VnAYZDflyKcBaQ2gg0hSYNQrJ8RilYldYQ1FxQYoCLtUjuuRuZo+fjqhx/qtq/1itJ0C2ejDxltZVFg==" crossorigin="anonymous"></script><script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../master_workflow.html" rel="next">
<link href="../notebooks/02_open2c_framework.html" rel="prev">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://hypothes.is/embed.js"></script>
<script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" integrity="sha512-c3Nl8+7g4LMSTdrm621y7kf9v3SDPnhxLNhcjFJbKECVnmZHTdo+IRO05sNLTH/D3vA6u1X32ehoLC7WFVdheg==" crossorigin="anonymous"></script>

<script type="application/javascript">define('jquery', [],function() {return window.jQuery;})</script>


</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../notebooks/01_hicexplorer.html">Notebooks</a></li><li class="breadcrumb-item"><a href="../notebooks/03_compartments.html"><span class="chapter-title">Working with coolers</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">Søren’s MSc project about Hi-C and spermatogenesis</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/munch-group/hic-spermatogenesis" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Project</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../thoughts.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Thoughts and Prayers</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">Notebooks</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/01_hicexplorer.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">HiCExplorer using macaque data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/02_open2c_framework.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Open2C framework</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/03_compartments.html" class="sidebar-item-text sidebar-link active"><span class="chapter-title">Working with coolers</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false">
 <span class="menu-text">Workflows</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../master_workflow.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">master_workflow</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../gwf_map_reads.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">gwf_map_reads</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../gwf_bowtie.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">gwf_bowtie</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../gwf_bowtie_local.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">gwf_bowtie_local</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../gwf_hiccorrect.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">gwf_hiccorrect</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../gwf_bwamem.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">gwf_bwamem</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../gwf_pair_alignments.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">gwf_pair_alignments</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false">
 <span class="menu-text">Reports</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../results/Readme.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Result files</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../reports/manuscript.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Correlating Chromatin Compartments with Selection</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../reports/supplementary.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Supplementary info</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false">
 <span class="menu-text">Slides</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../slides/slides.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Main talk</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../slides/suppl_slides.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Suppl. slides</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../references.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">References</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul class="collapse">
  <li><a href="#overview" id="toc-overview" class="nav-link active" data-scroll-target="#overview">Overview</a></li>
  <li><a href="#merge-all-data-from-each-cell-type" id="toc-merge-all-data-from-each-cell-type" class="nav-link" data-scroll-target="#merge-all-data-from-each-cell-type">Merge all data from each cell type</a></li>
  <li><a href="#zoomify-the-merged-cooler-files" id="toc-zoomify-the-merged-cooler-files" class="nav-link" data-scroll-target="#zoomify-the-merged-cooler-files">Zoomify the merged cooler files</a></li>
  <li><a href="#merge-coolers-based-on-the-biosample-id" id="toc-merge-coolers-based-on-the-biosample-id" class="nav-link" data-scroll-target="#merge-coolers-based-on-the-biosample-id">Merge coolers based on the BioSample ID</a></li>
  <li><a href="#plotting" id="toc-plotting" class="nav-link" data-scroll-target="#plotting">Plotting</a>
  <ul class="collapse">
  <li><a href="#kb-resolution" id="toc-kb-resolution" class="nav-link" data-scroll-target="#kb-resolution">500kb resolution</a></li>
  <li><a href="#stairs-plot-of-the-e1-compartments" id="toc-stairs-plot-of-the-e1-compartments" class="nav-link" data-scroll-target="#stairs-plot-of-the-e1-compartments">Stairs plot of the E1 compartments</a></li>
  </ul></li>
  </ul>
<div class="toc-actions"><ul class="collapse"><li><a href="https://github.dev/munch-group/hic-spermatogenesis/blob/main/notebooks/03_compartments.ipynb" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/munch-group/hic-spermatogenesis/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li><li><a href="https://github.com/munch-group/hic-spermatogenesis/blob/main/notebooks/03_compartments.ipynb" class="toc-action"><i class="bi empty"></i>View source</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../notebooks/01_hicexplorer.html">Notebooks</a></li><li class="breadcrumb-item"><a href="../notebooks/03_compartments.html"><span class="chapter-title">Working with coolers</span></a></li></ol></nav>
<div class="quarto-title">
<h1 class="title"><span class="chapter-title">Working with coolers</span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<p>In this notebook, we use files generated by the workflow <code>master_workflow</code> that, in short does:</p>
<ol type="1">
<li>Download the data from the source</li>
<li>Index the reference genome with <code>bwa index</code> and <code>samtools faidx</code></li>
<li>Align the reads to the reference genome with <code>bwa mem</code></li>
<li>Pair and sort the reads to <code>.pairs</code> files with <code>pairtools parse | pairtools sort</code></li>
<li>Deduplicate the pairs with <code>pairtools dedup</code></li>
<li>Convert the .pairs to cooler files with <code>cooler cload pairs</code></li>
</ol>
<p>We will:</p>
<ol type="1">
<li>Load the cooler files</li>
<li>Merge the coolers from the same BioSample ID –&gt; Create ‘replicates’</li>
<li>Zoomify the merged cooler files (coarsen) to create a multicooler (.mcool) file</li>
<li>Balance the matrices (use commandline <code>cooler balance</code>)</li>
<li>Calculate E1 compartments with <code>cooltools compute_cis_eig</code></li>
</ol>
<section id="overview" class="level2">
<h2 class="anchored" data-anchor-id="overview">Overview</h2>
<section id="data-accessions" class="level3">
<h3 class="anchored" data-anchor-id="data-accessions">Data (Accessions)</h3>
<div id="cell-3" class="cell" data-execution_count="86">
<div class="cell-output cell-output-display" data-execution_count="86">
<style type="text/css">
</style>

<table id="T_1515a" class="caption-top table table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th class="blank level0" data-quarto-table-cell-role="th">&nbsp;</th>
<th id="T_1515a_level0_col0" class="col_heading level0 col0" data-quarto-table-cell-role="th">source_name</th>
<th id="T_1515a_level0_col1" class="col_heading level0 col1" data-quarto-table-cell-role="th">BioSample</th>
<th id="T_1515a_level0_col2" class="col_heading level0 col2" data-quarto-table-cell-role="th">Run</th>
<th id="T_1515a_level0_col3" class="col_heading level0 col3" data-quarto-table-cell-role="th">GB</th>
<th id="T_1515a_level0_col4" class="col_heading level0 col4" data-quarto-table-cell-role="th">Bases</th>
<th id="T_1515a_level0_col5" class="col_heading level0 col5" data-quarto-table-cell-role="th">Reads</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td id="T_1515a_level0_row0" class="row_heading level0 row0" data-quarto-table-cell-role="th">16</td>
<td id="T_1515a_row0_col0" class="data row0 col0">fibroblast</td>
<td id="T_1515a_row0_col1" class="data row0 col1">SAMN08375237</td>
<td id="T_1515a_row0_col2" class="data row0 col2">SRR6502335</td>
<td id="T_1515a_row0_col3" class="data row0 col3">29.771059</td>
<td id="T_1515a_row0_col4" class="data row0 col4">73,201,141,800</td>
<td id="T_1515a_row0_col5" class="data row0 col5">244,003,806</td>
</tr>
<tr class="even">
<td id="T_1515a_level0_row1" class="row_heading level0 row1" data-quarto-table-cell-role="th">17</td>
<td id="T_1515a_row1_col0" class="data row1 col0">fibroblast</td>
<td id="T_1515a_row1_col1" class="data row1 col1">SAMN08375237</td>
<td id="T_1515a_row1_col2" class="data row1 col2">SRR6502336</td>
<td id="T_1515a_row1_col3" class="data row1 col3">22.755361</td>
<td id="T_1515a_row1_col4" class="data row1 col4">65,119,970,100</td>
<td id="T_1515a_row1_col5" class="data row1 col5">217,066,567</td>
</tr>
<tr class="odd">
<td id="T_1515a_level0_row2" class="row_heading level0 row2" data-quarto-table-cell-role="th">18</td>
<td id="T_1515a_row2_col0" class="data row2 col0">fibroblast</td>
<td id="T_1515a_row2_col1" class="data row2 col1">SAMN08375236</td>
<td id="T_1515a_row2_col2" class="data row2 col2">SRR6502337</td>
<td id="T_1515a_row2_col3" class="data row2 col3">21.434722</td>
<td id="T_1515a_row2_col4" class="data row2 col4">52,769,196,300</td>
<td id="T_1515a_row2_col5" class="data row2 col5">175,897,321</td>
</tr>
<tr class="even">
<td id="T_1515a_level0_row3" class="row_heading level0 row3" data-quarto-table-cell-role="th">19</td>
<td id="T_1515a_row3_col0" class="data row3 col0">fibroblast</td>
<td id="T_1515a_row3_col1" class="data row3 col1">SAMN08375236</td>
<td id="T_1515a_row3_col2" class="data row3 col2">SRR6502338</td>
<td id="T_1515a_row3_col3" class="data row3 col3">21.420030</td>
<td id="T_1515a_row3_col4" class="data row3 col4">52,378,949,100</td>
<td id="T_1515a_row3_col5" class="data row3 col5">174,596,497</td>
</tr>
<tr class="odd">
<td id="T_1515a_level0_row4" class="row_heading level0 row4" data-quarto-table-cell-role="th">20</td>
<td id="T_1515a_row4_col0" class="data row4 col0">fibroblast</td>
<td id="T_1515a_row4_col1" class="data row4 col1">SAMN08375236</td>
<td id="T_1515a_row4_col2" class="data row4 col2">SRR6502339</td>
<td id="T_1515a_row4_col3" class="data row4 col3">10.207410</td>
<td id="T_1515a_row4_col4" class="data row4 col4">28,885,941,600</td>
<td id="T_1515a_row4_col5" class="data row4 col5">96,286,472</td>
</tr>
<tr class="even">
<td id="T_1515a_level0_row5" class="row_heading level0 row5" data-quarto-table-cell-role="th">9</td>
<td id="T_1515a_row5_col0" class="data row5 col0">fibroblast</td>
<td id="T_1515a_row5_col1" class="data row5 col1">SAMN08375237</td>
<td id="T_1515a_row5_col2" class="data row5 col2">SRR7349189</td>
<td id="T_1515a_row5_col3" class="data row5 col3">52.729173</td>
<td id="T_1515a_row5_col4" class="data row5 col4">139,604,854,200</td>
<td id="T_1515a_row5_col5" class="data row5 col5">465,349,514</td>
</tr>
<tr class="odd">
<td id="T_1515a_level0_row6" class="row_heading level0 row6" data-quarto-table-cell-role="th">10</td>
<td id="T_1515a_row6_col0" class="data row6 col0">fibroblast</td>
<td id="T_1515a_row6_col1" class="data row6 col1">SAMN08375236</td>
<td id="T_1515a_row6_col2" class="data row6 col2">SRR7349190</td>
<td id="T_1515a_row6_col3" class="data row6 col3">53.085520</td>
<td id="T_1515a_row6_col4" class="data row6 col4">142,008,353,400</td>
<td id="T_1515a_row6_col5" class="data row6 col5">473,361,178</td>
</tr>
<tr class="even">
<td id="T_1515a_level0_row7" class="row_heading level0 row7" data-quarto-table-cell-role="th">21</td>
<td id="T_1515a_row7_col0" class="data row7 col0">pachytene spermatocyte</td>
<td id="T_1515a_row7_col1" class="data row7 col1">SAMN08375234</td>
<td id="T_1515a_row7_col2" class="data row7 col2">SRR6502342</td>
<td id="T_1515a_row7_col3" class="data row7 col3">60.258880</td>
<td id="T_1515a_row7_col4" class="data row7 col4">150,370,993,500</td>
<td id="T_1515a_row7_col5" class="data row7 col5">501,236,645</td>
</tr>
<tr class="odd">
<td id="T_1515a_level0_row8" class="row_heading level0 row8" data-quarto-table-cell-role="th">22</td>
<td id="T_1515a_row8_col0" class="data row8 col0">pachytene spermatocyte</td>
<td id="T_1515a_row8_col1" class="data row8 col1">SAMN08375234</td>
<td id="T_1515a_row8_col2" class="data row8 col2">SRR6502344</td>
<td id="T_1515a_row8_col3" class="data row8 col3">27.146048</td>
<td id="T_1515a_row8_col4" class="data row8 col4">65,697,684,300</td>
<td id="T_1515a_row8_col5" class="data row8 col5">218,992,281</td>
</tr>
<tr class="even">
<td id="T_1515a_level0_row9" class="row_heading level0 row9" data-quarto-table-cell-role="th">23</td>
<td id="T_1515a_row9_col0" class="data row9 col0">pachytene spermatocyte</td>
<td id="T_1515a_row9_col1" class="data row9 col1">SAMN08375234</td>
<td id="T_1515a_row9_col2" class="data row9 col2">SRR6502345</td>
<td id="T_1515a_row9_col3" class="data row9 col3">26.202707</td>
<td id="T_1515a_row9_col4" class="data row9 col4">63,490,538,700</td>
<td id="T_1515a_row9_col5" class="data row9 col5">211,635,129</td>
</tr>
<tr class="odd">
<td id="T_1515a_level0_row10" class="row_heading level0 row10" data-quarto-table-cell-role="th">0</td>
<td id="T_1515a_row10_col0" class="data row10 col0">pachytene spermatocyte</td>
<td id="T_1515a_row10_col1" class="data row10 col1">SAMN09427370</td>
<td id="T_1515a_row10_col2" class="data row10 col2">SRR7345458</td>
<td id="T_1515a_row10_col3" class="data row10 col3">55.970557</td>
<td id="T_1515a_row10_col4" class="data row10 col4">153,281,577,900</td>
<td id="T_1515a_row10_col5" class="data row10 col5">510,938,593</td>
</tr>
<tr class="even">
<td id="T_1515a_level0_row11" class="row_heading level0 row11" data-quarto-table-cell-role="th">1</td>
<td id="T_1515a_row11_col0" class="data row11 col0">pachytene spermatocyte</td>
<td id="T_1515a_row11_col1" class="data row11 col1">SAMN09427370</td>
<td id="T_1515a_row11_col2" class="data row11 col2">SRR7345459</td>
<td id="T_1515a_row11_col3" class="data row11 col3">53.982492</td>
<td id="T_1515a_row11_col4" class="data row11 col4">144,993,841,200</td>
<td id="T_1515a_row11_col5" class="data row11 col5">483,312,804</td>
</tr>
<tr class="odd">
<td id="T_1515a_level0_row12" class="row_heading level0 row12" data-quarto-table-cell-role="th">11</td>
<td id="T_1515a_row12_col0" class="data row12 col0">pachytene spermatocyte</td>
<td id="T_1515a_row12_col1" class="data row12 col1">SAMN08375235</td>
<td id="T_1515a_row12_col2" class="data row12 col2">SRR7349191</td>
<td id="T_1515a_row12_col3" class="data row12 col3">51.274476</td>
<td id="T_1515a_row12_col4" class="data row12 col4">137,821,979,100</td>
<td id="T_1515a_row12_col5" class="data row12 col5">459,406,597</td>
</tr>
<tr class="even">
<td id="T_1515a_level0_row13" class="row_heading level0 row13" data-quarto-table-cell-role="th">24</td>
<td id="T_1515a_row13_col0" class="data row13 col0">round spermatid</td>
<td id="T_1515a_row13_col1" class="data row13 col1">SAMN08375232</td>
<td id="T_1515a_row13_col2" class="data row13 col2">SRR6502351</td>
<td id="T_1515a_row13_col3" class="data row13 col3">20.924497</td>
<td id="T_1515a_row13_col4" class="data row13 col4">55,095,075,300</td>
<td id="T_1515a_row13_col5" class="data row13 col5">183,650,251</td>
</tr>
<tr class="odd">
<td id="T_1515a_level0_row14" class="row_heading level0 row14" data-quarto-table-cell-role="th">25</td>
<td id="T_1515a_row14_col0" class="data row14 col0">round spermatid</td>
<td id="T_1515a_row14_col1" class="data row14 col1">SAMN08375232</td>
<td id="T_1515a_row14_col2" class="data row14 col2">SRR6502352</td>
<td id="T_1515a_row14_col3" class="data row14 col3">41.133960</td>
<td id="T_1515a_row14_col4" class="data row14 col4">115,578,475,800</td>
<td id="T_1515a_row14_col5" class="data row14 col5">385,261,586</td>
</tr>
<tr class="even">
<td id="T_1515a_level0_row15" class="row_heading level0 row15" data-quarto-table-cell-role="th">26</td>
<td id="T_1515a_row15_col0" class="data row15 col0">round spermatid</td>
<td id="T_1515a_row15_col1" class="data row15 col1">SAMN08375232</td>
<td id="T_1515a_row15_col2" class="data row15 col2">SRR6502353</td>
<td id="T_1515a_row15_col3" class="data row15 col3">36.444117</td>
<td id="T_1515a_row15_col4" class="data row15 col4">96,195,161,400</td>
<td id="T_1515a_row15_col5" class="data row15 col5">320,650,538</td>
</tr>
<tr class="odd">
<td id="T_1515a_level0_row16" class="row_heading level0 row16" data-quarto-table-cell-role="th">2</td>
<td id="T_1515a_row16_col0" class="data row16 col0">round spermatid</td>
<td id="T_1515a_row16_col1" class="data row16 col1">SAMN09427369</td>
<td id="T_1515a_row16_col2" class="data row16 col2">SRR7345460</td>
<td id="T_1515a_row16_col3" class="data row16 col3">38.244654</td>
<td id="T_1515a_row16_col4" class="data row16 col4">104,105,827,200</td>
<td id="T_1515a_row16_col5" class="data row16 col5">347,019,424</td>
</tr>
<tr class="even">
<td id="T_1515a_level0_row17" class="row_heading level0 row17" data-quarto-table-cell-role="th">3</td>
<td id="T_1515a_row17_col0" class="data row17 col0">round spermatid</td>
<td id="T_1515a_row17_col1" class="data row17 col1">SAMN09427369</td>
<td id="T_1515a_row17_col2" class="data row17 col2">SRR7345461</td>
<td id="T_1515a_row17_col3" class="data row17 col3">53.996261</td>
<td id="T_1515a_row17_col4" class="data row17 col4">144,532,309,500</td>
<td id="T_1515a_row17_col5" class="data row17 col5">481,774,365</td>
</tr>
<tr class="odd">
<td id="T_1515a_level0_row18" class="row_heading level0 row18" data-quarto-table-cell-role="th">12</td>
<td id="T_1515a_row18_col0" class="data row18 col0">round spermatid</td>
<td id="T_1515a_row18_col1" class="data row18 col1">SAMN08375232</td>
<td id="T_1515a_row18_col2" class="data row18 col2">SRR7349192</td>
<td id="T_1515a_row18_col3" class="data row18 col3">52.384556</td>
<td id="T_1515a_row18_col4" class="data row18 col4">140,431,608,000</td>
<td id="T_1515a_row18_col5" class="data row18 col5">468,105,360</td>
</tr>
<tr class="even">
<td id="T_1515a_level0_row19" class="row_heading level0 row19" data-quarto-table-cell-role="th">29</td>
<td id="T_1515a_row19_col0" class="data row19 col0">sperm</td>
<td id="T_1515a_row19_col1" class="data row19 col1">SAMN08375229</td>
<td id="T_1515a_row19_col2" class="data row19 col2">SRR6502360</td>
<td id="T_1515a_row19_col3" class="data row19 col3">26.653940</td>
<td id="T_1515a_row19_col4" class="data row19 col4">64,752,370,800</td>
<td id="T_1515a_row19_col5" class="data row19 col5">215,841,236</td>
</tr>
<tr class="odd">
<td id="T_1515a_level0_row20" class="row_heading level0 row20" data-quarto-table-cell-role="th">30</td>
<td id="T_1515a_row20_col0" class="data row20 col0">sperm</td>
<td id="T_1515a_row20_col1" class="data row20 col1">SAMN08375228</td>
<td id="T_1515a_row20_col2" class="data row20 col2">SRR6502362</td>
<td id="T_1515a_row20_col3" class="data row20 col3">23.973440</td>
<td id="T_1515a_row20_col4" class="data row20 col4">58,369,232,700</td>
<td id="T_1515a_row20_col5" class="data row20 col5">194,564,109</td>
</tr>
<tr class="even">
<td id="T_1515a_level0_row21" class="row_heading level0 row21" data-quarto-table-cell-role="th">13</td>
<td id="T_1515a_row21_col0" class="data row21 col0">sperm</td>
<td id="T_1515a_row21_col1" class="data row21 col1">SAMN08375229</td>
<td id="T_1515a_row21_col2" class="data row21 col2">SRR7349193</td>
<td id="T_1515a_row21_col3" class="data row21 col3">52.806276</td>
<td id="T_1515a_row21_col4" class="data row21 col4">141,148,572,300</td>
<td id="T_1515a_row21_col5" class="data row21 col5">470,495,241</td>
</tr>
<tr class="odd">
<td id="T_1515a_level0_row22" class="row_heading level0 row22" data-quarto-table-cell-role="th">14</td>
<td id="T_1515a_row22_col0" class="data row22 col0">sperm</td>
<td id="T_1515a_row22_col1" class="data row22 col1">SAMN08375229</td>
<td id="T_1515a_row22_col2" class="data row22 col2">SRR7349195</td>
<td id="T_1515a_row22_col3" class="data row22 col3">22.444378</td>
<td id="T_1515a_row22_col4" class="data row22 col4">60,523,788,600</td>
<td id="T_1515a_row22_col5" class="data row22 col5">201,745,962</td>
</tr>
<tr class="even">
<td id="T_1515a_level0_row23" class="row_heading level0 row23" data-quarto-table-cell-role="th">15</td>
<td id="T_1515a_row23_col0" class="data row23 col0">sperm</td>
<td id="T_1515a_row23_col1" class="data row23 col1">SAMN08375229</td>
<td id="T_1515a_row23_col2" class="data row23 col2">SRR7349196</td>
<td id="T_1515a_row23_col3" class="data row23 col3">38.253606</td>
<td id="T_1515a_row23_col4" class="data row23 col4">104,119,671,000</td>
<td id="T_1515a_row23_col5" class="data row23 col5">347,065,570</td>
</tr>
<tr class="odd">
<td id="T_1515a_level0_row24" class="row_heading level0 row24" data-quarto-table-cell-role="th">27</td>
<td id="T_1515a_row24_col0" class="data row24 col0">spermatogonia</td>
<td id="T_1515a_row24_col1" class="data row24 col1">SAMN08375231</td>
<td id="T_1515a_row24_col2" class="data row24 col2">SRR6502356</td>
<td id="T_1515a_row24_col3" class="data row24 col3">22.845286</td>
<td id="T_1515a_row24_col4" class="data row24 col4">58,909,579,800</td>
<td id="T_1515a_row24_col5" class="data row24 col5">196,365,266</td>
</tr>
<tr class="even">
<td id="T_1515a_level0_row25" class="row_heading level0 row25" data-quarto-table-cell-role="th">28</td>
<td id="T_1515a_row25_col0" class="data row25 col0">spermatogonia</td>
<td id="T_1515a_row25_col1" class="data row25 col1">SAMN08375231</td>
<td id="T_1515a_row25_col2" class="data row25 col2">SRR6502357</td>
<td id="T_1515a_row25_col3" class="data row25 col3">17.947471</td>
<td id="T_1515a_row25_col4" class="data row25 col4">46,888,332,900</td>
<td id="T_1515a_row25_col5" class="data row25 col5">156,294,443</td>
</tr>
<tr class="odd">
<td id="T_1515a_level0_row26" class="row_heading level0 row26" data-quarto-table-cell-role="th">4</td>
<td id="T_1515a_row26_col0" class="data row26 col0">spermatogonia</td>
<td id="T_1515a_row26_col1" class="data row26 col1">SAMN09427379</td>
<td id="T_1515a_row26_col2" class="data row26 col2">SRR7345462</td>
<td id="T_1515a_row26_col3" class="data row26 col3">18.686342</td>
<td id="T_1515a_row26_col4" class="data row26 col4">52,032,780,000</td>
<td id="T_1515a_row26_col5" class="data row26 col5">173,442,600</td>
</tr>
<tr class="even">
<td id="T_1515a_level0_row27" class="row_heading level0 row27" data-quarto-table-cell-role="th">5</td>
<td id="T_1515a_row27_col0" class="data row27 col0">spermatogonia</td>
<td id="T_1515a_row27_col1" class="data row27 col1">SAMN09427379</td>
<td id="T_1515a_row27_col2" class="data row27 col2">SRR7345463</td>
<td id="T_1515a_row27_col3" class="data row27 col3">29.956561</td>
<td id="T_1515a_row27_col4" class="data row27 col4">82,384,836,000</td>
<td id="T_1515a_row27_col5" class="data row27 col5">274,616,120</td>
</tr>
<tr class="odd">
<td id="T_1515a_level0_row28" class="row_heading level0 row28" data-quarto-table-cell-role="th">6</td>
<td id="T_1515a_row28_col0" class="data row28 col0">spermatogonia</td>
<td id="T_1515a_row28_col1" class="data row28 col1">SAMN09427379</td>
<td id="T_1515a_row28_col2" class="data row28 col2">SRR7345464</td>
<td id="T_1515a_row28_col3" class="data row28 col3">39.145759</td>
<td id="T_1515a_row28_col4" class="data row28 col4">105,153,716,100</td>
<td id="T_1515a_row28_col5" class="data row28 col5">350,512,387</td>
</tr>
<tr class="even">
<td id="T_1515a_level0_row29" class="row_heading level0 row29" data-quarto-table-cell-role="th">7</td>
<td id="T_1515a_row29_col0" class="data row29 col0">spermatogonia</td>
<td id="T_1515a_row29_col1" class="data row29 col1">SAMN09427378</td>
<td id="T_1515a_row29_col2" class="data row29 col2">SRR7345465</td>
<td id="T_1515a_row29_col3" class="data row29 col3">35.816184</td>
<td id="T_1515a_row29_col4" class="data row29 col4">96,048,594,600</td>
<td id="T_1515a_row29_col5" class="data row29 col5">320,161,982</td>
</tr>
<tr class="odd">
<td id="T_1515a_level0_row30" class="row_heading level0 row30" data-quarto-table-cell-role="th">8</td>
<td id="T_1515a_row30_col0" class="data row30 col0">spermatogonia</td>
<td id="T_1515a_row30_col1" class="data row30 col1">SAMN09427378</td>
<td id="T_1515a_row30_col2" class="data row30 col2">SRR7345467</td>
<td id="T_1515a_row30_col3" class="data row30 col3">28.396816</td>
<td id="T_1515a_row30_col4" class="data row30 col4">77,248,140,900</td>
<td id="T_1515a_row30_col5" class="data row30 col5">257,493,803</td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
<section id="folder-structure" class="level3">
<h3 class="anchored" data-anchor-id="folder-structure">Folder structure</h3>
<div id="cell-5" class="cell" data-execution_count="39">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="op">!</span>tree <span class="op">-</span>d <span class="op">-</span>L <span class="dv">2</span> ..<span class="op">/</span>steps<span class="op">/</span>bwa<span class="op">/</span>PE</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>../steps/bwa/PE
├── bamfiles
│&nbsp;&nbsp; ├── fibroblast
│&nbsp;&nbsp; ├── pachytene_spermatocyte
│&nbsp;&nbsp; ├── round_spermatid
│&nbsp;&nbsp; ├── sperm
│&nbsp;&nbsp; └── spermatogonia
├── cool
│&nbsp;&nbsp; ├── fibroblast
│&nbsp;&nbsp; ├── pachytene_spermatocyte
│&nbsp;&nbsp; ├── round_spermatid
│&nbsp;&nbsp; ├── sperm
│&nbsp;&nbsp; └── spermatogonia
└── pairs
    ├── fibroblast
    ├── pachytene_spermatocyte
    ├── round_spermatid
    ├── sperm
    └── spermatogonia

18 directories</code></pre>
</div>
</div>
</section>
</section>
<section id="merge-all-data-from-each-cell-type" class="level2">
<h2 class="anchored" data-anchor-id="merge-all-data-from-each-cell-type">Merge all data from each cell type</h2>
<p>With <code>cooler merge</code></p>
<div id="cell-7" class="cell" data-execution_count="24">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> glob</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os.path <span class="im">as</span> op</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> pprint <span class="im">import</span> pprint <span class="im">as</span> pp</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the list of cell type dirs</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>base_dir <span class="op">=</span> <span class="st">'../steps/bwa/PE/cool'</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>folders <span class="op">=</span> glob.glob(op.join(base_dir, <span class="st">'*'</span>))</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>files_dict <span class="op">=</span> {f:glob.glob(<span class="ss">f"</span><span class="sc">{</span>f<span class="sc">}</span><span class="ss">/*.nodups.*"</span>) <span class="cf">for</span> f <span class="kw">in</span> folders}</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>cooler_dict <span class="op">=</span> {op.basename(k): [op.basename(f) <span class="cf">for</span> f <span class="kw">in</span> v] <span class="cf">for</span> k,v <span class="kw">in</span> files_dict.items()}</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>cooler_dict_no_fib <span class="op">=</span> {k:v <span class="cf">for</span> k,v <span class="kw">in</span> cooler_dict.items() <span class="cf">if</span> <span class="st">'fib'</span> <span class="kw">not</span> <span class="kw">in</span> k}</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>pp(cooler_dict_no_fib)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>{'pachytene_spermatocyte': ['SRR7345459.nodups.10000.cool',
                            'SRR6502344.nodups.10000.cool',
                            'SRR6502342.nodups.10000.cool',
                            'SRR7345458.nodups.10000.cool',
                            'SRR6502345.nodups.10000.cool',
                            'SRR7349191.nodups.10000.cool'],
 'round_spermatid': ['SRR7349192.nodups.10000.cool',
                     'SRR6502353.nodups.10000.cool',
                     'SRR6502352.nodups.10000.cool',
                     'SRR6502351.nodups.10000.cool',
                     'SRR7345460.nodups.10000.cool',
                     'SRR7345461.nodups.10000.cool'],
 'sperm': ['SRR7349196.nodups.10000.cool',
           'SRR6502362.nodups.10000.cool',
           'SRR7349193.nodups.10000.cool',
           'SRR6502360.nodups.10000.cool',
           'SRR7349195.nodups.10000.cool'],
 'spermatogonia': ['SRR6502357.nodups.10000.cool',
                   'SRR7345467.nodups.10000.cool',
                   'SRR6502356.nodups.10000.cool',
                   'SRR7345464.nodups.10000.cool',
                   'SRR7345462.nodups.10000.cool',
                   'SRR7345465.nodups.10000.cool',
                   'SRR7345463.nodups.10000.cool']}</code></pre>
</div>
</div>
<div id="cell-8" class="cell" data-execution_count="40">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># NB adjust mergebuf if you don't have 32G of RAM</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="co"># I used </span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooler</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> folder,cooler_list <span class="kw">in</span> cooler_dict_no_fib.items():</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>    in_uris <span class="op">=</span> [op.join(base_dir, folder, <span class="bu">file</span>) <span class="cf">for</span> <span class="bu">file</span> <span class="kw">in</span> cooler_list]</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>    out_uri <span class="op">=</span> op.join(base_dir, folder, <span class="ss">f'</span><span class="sc">{</span>folder<span class="sc">}</span><span class="ss">.fullmerge.cool'</span>)</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> op.exists(out_uri):</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Skipping </span><span class="sc">{</span>out_uri<span class="sc">}</span><span class="ss">: exists..."</span>)</span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>        <span class="cf">continue</span></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Creating </span><span class="sc">{</span>out_uri<span class="sc">}</span><span class="ss"> by </span><span class="ch">\n</span><span class="ss">Merging </span><span class="sc">{</span><span class="bu">len</span>(cooler_list)<span class="sc">}</span><span class="ss"> coolers into one:"</span>, end<span class="op">=</span><span class="st">" "</span>)</span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>    <span class="co">#print("\t",[file.split('.')[0] for file in cooler_list])</span></span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>    cooler.merge_coolers(output_uri<span class="op">=</span>out_uri,</span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a>                         input_uris<span class="op">=</span>in_uris,</span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a>                         mergebuf<span class="op">=</span><span class="bu">int</span>(<span class="fl">4e9</span>),</span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>                         )</span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"Done!"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Creating ../steps/bwa/PE/cool/round_spermatid/round_spermatid.fullmerge.cool by 
Merging 6 coolers into one: </code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/dask/dataframe/_pyarrow_compat.py:15: FutureWarning: Minimal version of pyarrow will soon be increased to 14.0.1. You are using 13.0.0. Please consider upgrading.
  warnings.warn(</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Done!
Creating ../steps/bwa/PE/cool/spermatogonia/spermatogonia.fullmerge.cool by 
Merging 7 coolers into one: Done!
Creating ../steps/bwa/PE/cool/sperm/sperm.fullmerge.cool by 
Merging 5 coolers into one: Done!
Creating ../steps/bwa/PE/cool/pachytene_spermatocyte/pachytene_spermatocyte.fullmerge.cool by 
Merging 6 coolers into one: Done!</code></pre>
</div>
</div>
</section>
<section id="zoomify-the-merged-cooler-files" class="level2">
<h2 class="anchored" data-anchor-id="zoomify-the-merged-cooler-files">Zoomify the merged cooler files</h2>
<div id="cell-10" class="cell" data-execution_count="3">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># NB 8 cores and 32G of RAM was used</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> glob</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooler</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os.path <span class="im">as</span> op</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>base_dir <span class="op">=</span> <span class="st">'../steps/bwa/PE/cool'</span></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>merged_coolers <span class="op">=</span> glob.glob(op.join(base_dir, <span class="st">'*/*.fullmerge.cool'</span>))</span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> clr <span class="kw">in</span> merged_coolers:</span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>    out_uri <span class="op">=</span> clr.replace(<span class="st">'.fullmerge.cool'</span>, <span class="st">'.mcool'</span>)</span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> op.exists(out_uri):</span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Skipping </span><span class="sc">{</span>out_uri<span class="sc">}</span><span class="ss">: exists..."</span>)</span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a>        <span class="cf">continue</span></span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Zoomifying cooler: </span><span class="ch">\n\t</span><span class="ss">   </span><span class="sc">{</span>clr<span class="sc">}</span><span class="ch">\n\t</span><span class="ss">-&gt; </span><span class="sc">{</span>out_uri<span class="sc">}</span><span class="ss">"</span>, end<span class="op">=</span><span class="st">""</span>)</span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a>    cooler.zoomify_cooler(base_uris <span class="op">=</span> clr,</span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a>                          outfile <span class="op">=</span> out_uri,</span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a>                          resolutions <span class="op">=</span> [<span class="dv">10000</span>,<span class="dv">50000</span>,<span class="dv">100000</span>,<span class="dv">500000</span>],</span>
<span id="cb9-22"><a href="#cb9-22" aria-hidden="true" tabindex="-1"></a>                          chunksize <span class="op">=</span> <span class="dv">400000000</span>,</span>
<span id="cb9-23"><a href="#cb9-23" aria-hidden="true" tabindex="-1"></a>                          nproc <span class="op">=</span> <span class="dv">8</span>)</span>
<span id="cb9-24"><a href="#cb9-24" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">" --&gt; done"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Zoomifying cooler: 
       ../steps/bwa/PE/cool/round_spermatid/round_spermatid.fullmerge.cool
    -&gt; ../steps/bwa/PE/cool/round_spermatid/round_spermatid.mcool</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/dask/dataframe/_pyarrow_compat.py:15: FutureWarning: Minimal version of pyarrow will soon be increased to 14.0.1. You are using 13.0.0. Please consider upgrading.
  warnings.warn(</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code> --&gt; done
Zoomifying cooler: 
       ../steps/bwa/PE/cool/spermatogonia/spermatogonia.fullmerge.cool
    -&gt; ../steps/bwa/PE/cool/spermatogonia/spermatogonia.mcool --&gt; done
Zoomifying cooler: 
       ../steps/bwa/PE/cool/sperm/sperm.fullmerge.cool
    -&gt; ../steps/bwa/PE/cool/sperm/sperm.mcool --&gt; done
Zoomifying cooler: 
       ../steps/bwa/PE/cool/pachytene_spermatocyte/pachytene_spermatocyte.fullmerge.cool
    -&gt; ../steps/bwa/PE/cool/pachytene_spermatocyte/pachytene_spermatocyte.mcool --&gt; done</code></pre>
</div>
</div>
</section>
<section id="merge-coolers-based-on-the-biosample-id" class="level2">
<h2 class="anchored" data-anchor-id="merge-coolers-based-on-the-biosample-id">Merge coolers based on the BioSample ID</h2>
<p>Merge into the original ‘replicates’</p>
<div id="cell-12" class="cell" data-execution_count="90">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>df.groupby([<span class="st">'source_name'</span>,<span class="st">'BioSample'</span>])[<span class="st">'Reads'</span>].<span class="bu">sum</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="90">
<pre><code>source_name             BioSample   
fibroblast              SAMN08375236    9.201415e+08
                        SAMN08375237    9.264199e+08
pachytene spermatocyte  SAMN08375234    9.318641e+08
                        SAMN08375235    4.594066e+08
                        SAMN09427370    9.942514e+08
round spermatid         SAMN08375232    1.357668e+09
                        SAMN09427369    8.287938e+08
sperm                   SAMN08375228    1.945641e+08
                        SAMN08375229    1.235148e+09
spermatogonia           SAMN08375231    3.526597e+08
                        SAMN09427378    5.776558e+08
                        SAMN09427379    7.985711e+08
Name: Reads, dtype: float64</code></pre>
</div>
</div>
<div id="cell-13" class="cell" data-execution_count="84">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> pprintpp <span class="im">import</span> pprint <span class="im">as</span> pp</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>grouped_df <span class="op">=</span> df.groupby([<span class="st">'source_name'</span>,<span class="st">'BioSample'</span>])</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize an empty dictionary</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>rep_dict <span class="op">=</span> {}</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Iterate over each group</span></span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (source_name, BioSample), group <span class="kw">in</span> grouped_df:</span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Extract the 'Run' column and convert it to a list</span></span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>    run_list <span class="op">=</span> group[<span class="st">'Run'</span>].tolist()</span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Populate the nested dictionary</span></span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> source_name <span class="kw">not</span> <span class="kw">in</span> rep_dict:</span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a>        rep_dict[source_name] <span class="op">=</span> {}</span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a>    rep_dict[source_name][BioSample] <span class="op">=</span> run_list</span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a>pp(rep_dict)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>{
    'fibroblast': {
        'SAMN08375236': [
            'SRR6502337',
            'SRR6502338',
            'SRR6502339',
            'SRR7349190',
        ],
        'SAMN08375237': ['SRR6502335', 'SRR6502336', 'SRR7349189'],
    },
    'pachytene spermatocyte': {
        'SAMN08375234': ['SRR6502342', 'SRR6502344', 'SRR6502345'],
        'SAMN08375235': ['SRR7349191'],
        'SAMN09427370': ['SRR7345458', 'SRR7345459'],
    },
    'round spermatid': {
        'SAMN08375232': [
            'SRR6502351',
            'SRR6502352',
            'SRR6502353',
            'SRR7349192',
        ],
        'SAMN09427369': ['SRR7345460', 'SRR7345461'],
    },
    'sperm': {
        'SAMN08375228': ['SRR6502362'],
        'SAMN08375229': [
            'SRR6502360',
            'SRR7349193',
            'SRR7349195',
            'SRR7349196',
        ],
    },
    'spermatogonia': {
        'SAMN08375231': ['SRR6502356', 'SRR6502357'],
        'SAMN09427378': ['SRR7345465', 'SRR7345467'],
        'SAMN09427379': ['SRR7345462', 'SRR7345463', 'SRR7345464'],
    },
}</code></pre>
</div>
</div>
<div id="cell-14" class="cell" data-execution_count="92">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> source_name, BioSample_dict <span class="kw">in</span> rep_dict.items():</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"source_name: </span><span class="sc">{</span>source_name<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Changing working dir: </span><span class="sc">{</span>source_name<span class="sc">}</span><span class="ss">/"</span>)</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> BioSample, run_list <span class="kw">in</span> BioSample_dict.items():</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Merging samples for BioSample: </span><span class="sc">{</span>BioSample<span class="sc">}</span><span class="ss">: </span><span class="sc">{</span>run_list<span class="sc">}</span><span class="ss">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>source_name: fibroblast
Changing working dir: fibroblast/
Merging samples for BioSample: SAMN08375236: ['SRR6502337', 'SRR6502338', 'SRR6502339', 'SRR7349190']
Merging samples for BioSample: SAMN08375237: ['SRR6502335', 'SRR6502336', 'SRR7349189']
source_name: pachytene spermatocyte
Changing working dir: pachytene spermatocyte/
Merging samples for BioSample: SAMN08375234: ['SRR6502342', 'SRR6502344', 'SRR6502345']
Merging samples for BioSample: SAMN08375235: ['SRR7349191']
Merging samples for BioSample: SAMN09427370: ['SRR7345458', 'SRR7345459']
source_name: round spermatid
Changing working dir: round spermatid/
Merging samples for BioSample: SAMN08375232: ['SRR6502351', 'SRR6502352', 'SRR6502353', 'SRR7349192']
Merging samples for BioSample: SAMN09427369: ['SRR7345460', 'SRR7345461']
source_name: sperm
Changing working dir: sperm/
Merging samples for BioSample: SAMN08375228: ['SRR6502362']
Merging samples for BioSample: SAMN08375229: ['SRR6502360', 'SRR7349193', 'SRR7349195', 'SRR7349196']
source_name: spermatogonia
Changing working dir: spermatogonia/
Merging samples for BioSample: SAMN08375231: ['SRR6502356', 'SRR6502357']
Merging samples for BioSample: SAMN09427378: ['SRR7345465', 'SRR7345467']
Merging samples for BioSample: SAMN09427379: ['SRR7345462', 'SRR7345463', 'SRR7345464']</code></pre>
</div>
</div>
</section>
<section id="plotting" class="level1">
<h1>Plotting</h1>
<section id="kb-resolution" class="level2">
<h2 class="anchored" data-anchor-id="kb-resolution">500kb resolution</h2>
<div id="cell-17" class="cell" data-execution_count="1">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># import standard python libraries</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a><span class="op">%</span>matplotlib inline</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os, subprocess</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div id="cell-18" class="cell" data-execution_count="2">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Import python package for working with cooler files and tools for analysis</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooler</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooltools.lib.plotting</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooltools</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div id="cell-19" class="cell" data-execution_count="3">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>cool_file <span class="op">=</span> <span class="st">"../steps/bwa/PE/cool/fibroblast/fib_rep1.mcool"</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>clr <span class="op">=</span> cooler.Cooler(<span class="ss">f"</span><span class="sc">{</span>cool_file<span class="sc">}</span><span class="ss">::resolutions/500000"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div id="cell-20" class="cell" data-execution_count="4">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> bioframe</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>bins <span class="op">=</span> clr.bins()[:]</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>rheMac10 <span class="op">=</span> bioframe.load_fasta(<span class="st">'../data/links/ucsc_ref/rheMac10.fa'</span>)</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> <span class="kw">not</span> os.path.exists(<span class="st">'rheMac10_gc_cov_500kb.tsv'</span>):</span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">'Calculate the fraction of GC basepairs for each bin'</span>)</span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>    gc_cov <span class="op">=</span> bioframe.frac_gc(bins[[<span class="st">'chrom'</span>, <span class="st">'start'</span>, <span class="st">'end'</span>]], rheMac10)</span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>    gc_cov.to_csv(<span class="st">'rheMac10_gc_cov_500kb.tsv'</span>, index<span class="op">=</span><span class="va">False</span>,sep<span class="op">=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>)</span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>    display(gc_cov)</span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>: </span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"Already exists, just read from file:"</span>)</span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a>    gc_cov <span class="op">=</span> pd.read_csv(<span class="st">"rheMac10_gc_cov_500kb.tsv"</span>, sep<span class="op">=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>)</span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a>    display(gc_cov)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Already exists, just read from file:</code></pre>
</div>
<div class="cell-output cell-output-display">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">chrom</th>
<th data-quarto-table-cell-role="th">start</th>
<th data-quarto-table-cell-role="th">end</th>
<th data-quarto-table-cell-role="th">GC</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>chr1</td>
<td>0</td>
<td>500000</td>
<td>0.424508</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>chr1</td>
<td>500000</td>
<td>1000000</td>
<td>0.378836</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>chr1</td>
<td>1000000</td>
<td>1500000</td>
<td>0.388272</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3</td>
<td>chr1</td>
<td>1500000</td>
<td>2000000</td>
<td>0.445226</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">4</td>
<td>chr1</td>
<td>2000000</td>
<td>2500000</td>
<td>0.439485</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">5715</td>
<td>chrY</td>
<td>9500000</td>
<td>10000000</td>
<td>0.399518</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">5716</td>
<td>chrY</td>
<td>10000000</td>
<td>10500000</td>
<td>0.396028</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">5717</td>
<td>chrY</td>
<td>10500000</td>
<td>11000000</td>
<td>0.395280</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">5718</td>
<td>chrY</td>
<td>11000000</td>
<td>11500000</td>
<td>0.382006</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">5719</td>
<td>chrY</td>
<td>11500000</td>
<td>11753682</td>
<td>NaN</td>
</tr>
</tbody>
</table>

<p>5720 rows × 4 columns</p>
</div>
</div>
</div>
<div id="cell-21" class="cell" data-execution_count="5">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Apparently, we have som NaN values, I don't know why (and only 1 on the X)</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>gc_cov[gc_cov[<span class="st">'GC'</span>].isna()].groupby(<span class="st">'chrom'</span>).size()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="5">
<pre><code>chrom
chr1     1
chr10    1
chr11    1
chr12    1
chr13    1
chr14    1
chr15    1
chr16    1
chr17    1
chr18    1
chr19    1
chr2     1
chr20    1
chr3     1
chr4     1
chr5     1
chr6     1
chr7     1
chr8     1
chr9     1
chrY     1
dtype: int64</code></pre>
</div>
</div>
<div id="cell-22" class="cell" data-execution_count="6">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>view_df <span class="op">=</span> pd.DataFrame(</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>    {</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>    <span class="st">'chrom'</span>: clr.chromnames,</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>    <span class="st">'start'</span>: <span class="dv">0</span>,</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>    <span class="st">'end'</span>: clr.chromsizes.values,</span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>    <span class="st">'name'</span>: clr.chromnames</span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>                      )</span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a><span class="co">#display(view_df)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div id="cell-23" class="cell" data-execution_count="7">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co"># obtain first 3 eigenvectors</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>cis_eigs <span class="op">=</span> cooltools.eigs_cis(</span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>                        clr,</span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>                        gc_cov,</span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>                        view_df<span class="op">=</span>view_df,</span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>                        n_eigs<span class="op">=</span><span class="dv">3</span>,</span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a>                        )</span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a><span class="co"># cis_eigs[0] returns eigenvalues, here we focus on eigenvectors</span></span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a>eigenvector_track <span class="op">=</span> cis_eigs[<span class="dv">1</span>][[<span class="st">'chrom'</span>,<span class="st">'start'</span>,<span class="st">'end'</span>,<span class="st">'E1'</span>]]</span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a><span class="bu">len</span>(eigenvector_track)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="7">
<pre><code>5720</code></pre>
</div>
</div>
<div id="cell-24" class="cell" data-execution_count="8">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co"># full track</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>eigenvector_track</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="8">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">chrom</th>
<th data-quarto-table-cell-role="th">start</th>
<th data-quarto-table-cell-role="th">end</th>
<th data-quarto-table-cell-role="th">E1</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>chr1</td>
<td>0</td>
<td>500000</td>
<td>-0.818737</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>chr1</td>
<td>500000</td>
<td>1000000</td>
<td>-1.052494</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>chr1</td>
<td>1000000</td>
<td>1500000</td>
<td>-0.946036</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3</td>
<td>chr1</td>
<td>1500000</td>
<td>2000000</td>
<td>-0.550397</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">4</td>
<td>chr1</td>
<td>2000000</td>
<td>2500000</td>
<td>0.108430</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">5715</td>
<td>chrY</td>
<td>9500000</td>
<td>10000000</td>
<td>-0.523549</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">5716</td>
<td>chrY</td>
<td>10000000</td>
<td>10500000</td>
<td>-0.554934</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">5717</td>
<td>chrY</td>
<td>10500000</td>
<td>11000000</td>
<td>-1.022690</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">5718</td>
<td>chrY</td>
<td>11000000</td>
<td>11500000</td>
<td>-0.491319</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">5719</td>
<td>chrY</td>
<td>11500000</td>
<td>11753682</td>
<td>NaN</td>
</tr>
</tbody>
</table>

<p>5720 rows × 4 columns</p>
</div>
</div>
</div>
<div id="cell-25" class="cell" data-execution_count="9">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co"># subset the chrX   </span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>eigenvector_track_chrX <span class="op">=</span> eigenvector_track.loc[eigenvector_track[<span class="st">'chrom'</span>] <span class="op">==</span> <span class="st">'chrX'</span>]</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>nbins <span class="op">=</span> <span class="bu">len</span>(eigenvector_track_chrX)</span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>eigenvector_track_chrX</span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>e1X_values <span class="op">=</span> eigenvector_track_chrX[<span class="st">'E1'</span>].values</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div id="cell-26" class="cell" data-execution_count="10">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="op">%%</span>capture chrX_matrix_e1_500kb</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> matplotlib.colors <span class="im">import</span> LogNorm</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> mpl_toolkits.axes_grid1 <span class="im">import</span> make_axes_locatable</span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>f, ax <span class="op">=</span> plt.subplots(</span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a>    figsize<span class="op">=</span>(<span class="dv">15</span>, <span class="dv">10</span>),</span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-10"><a href="#cb31-10" aria-hidden="true" tabindex="-1"></a>norm <span class="op">=</span> LogNorm(vmax<span class="op">=</span><span class="fl">0.1</span>)</span>
<span id="cb31-11"><a href="#cb31-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-12"><a href="#cb31-12" aria-hidden="true" tabindex="-1"></a>im <span class="op">=</span> ax.matshow(</span>
<span id="cb31-13"><a href="#cb31-13" aria-hidden="true" tabindex="-1"></a>    clr.matrix().fetch(<span class="st">'chrX'</span>),</span>
<span id="cb31-14"><a href="#cb31-14" aria-hidden="true" tabindex="-1"></a>    norm<span class="op">=</span>norm,</span>
<span id="cb31-15"><a href="#cb31-15" aria-hidden="true" tabindex="-1"></a>    cmap<span class="op">=</span><span class="st">'fall'</span>,</span>
<span id="cb31-16"><a href="#cb31-16" aria-hidden="true" tabindex="-1"></a>)<span class="op">;</span></span>
<span id="cb31-17"><a href="#cb31-17" aria-hidden="true" tabindex="-1"></a>plt.axis([<span class="dv">0</span>,nbins,nbins,<span class="dv">0</span>])</span>
<span id="cb31-18"><a href="#cb31-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-19"><a href="#cb31-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-20"><a href="#cb31-20" aria-hidden="true" tabindex="-1"></a>divider <span class="op">=</span> make_axes_locatable(ax)</span>
<span id="cb31-21"><a href="#cb31-21" aria-hidden="true" tabindex="-1"></a>cax <span class="op">=</span> divider.append_axes(<span class="st">"right"</span>, size<span class="op">=</span><span class="st">"5%"</span>, pad<span class="op">=</span><span class="fl">0.1</span>)</span>
<span id="cb31-22"><a href="#cb31-22" aria-hidden="true" tabindex="-1"></a>plt.colorbar(im, cax<span class="op">=</span>cax, label<span class="op">=</span><span class="st">'corrected frequencies'</span>)<span class="op">;</span></span>
<span id="cb31-23"><a href="#cb31-23" aria-hidden="true" tabindex="-1"></a>ax.set_ylabel(<span class="st">'chrX:500kb bins, #bin'</span>)</span>
<span id="cb31-24"><a href="#cb31-24" aria-hidden="true" tabindex="-1"></a>ax.xaxis.set_visible(<span class="va">False</span>)</span>
<span id="cb31-25"><a href="#cb31-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-26"><a href="#cb31-26" aria-hidden="true" tabindex="-1"></a>ax1 <span class="op">=</span> divider.append_axes(<span class="st">"top"</span>, size<span class="op">=</span><span class="st">"20%"</span>, pad<span class="op">=</span><span class="fl">0.25</span>, sharex<span class="op">=</span>ax)</span>
<span id="cb31-27"><a href="#cb31-27" aria-hidden="true" tabindex="-1"></a>weights <span class="op">=</span> clr.bins()[:][<span class="st">'weight'</span>].values</span>
<span id="cb31-28"><a href="#cb31-28" aria-hidden="true" tabindex="-1"></a><span class="co">#ax1.plot([0,nbins],[0,0],'k',lw=0.25)</span></span>
<span id="cb31-29"><a href="#cb31-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-30"><a href="#cb31-30" aria-hidden="true" tabindex="-1"></a><span class="co">#ax1.plot(e1X_values, label='E1')</span></span>
<span id="cb31-31"><a href="#cb31-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-32"><a href="#cb31-32" aria-hidden="true" tabindex="-1"></a><span class="co"># Fill between the line and 0</span></span>
<span id="cb31-33"><a href="#cb31-33" aria-hidden="true" tabindex="-1"></a>ax1.fill_between(<span class="bu">range</span>(<span class="bu">len</span>(e1X_values)), e1X_values, <span class="dv">0</span>, where<span class="op">=</span>(e1X_values <span class="op">&gt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:red'</span>)</span>
<span id="cb31-34"><a href="#cb31-34" aria-hidden="true" tabindex="-1"></a>ax1.fill_between(<span class="bu">range</span>(<span class="bu">len</span>(e1X_values)), e1X_values, <span class="dv">0</span>, where<span class="op">=</span>(e1X_values <span class="op">&lt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'C0'</span>)</span>
<span id="cb31-35"><a href="#cb31-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-36"><a href="#cb31-36" aria-hidden="true" tabindex="-1"></a>ax1.set_ylabel(<span class="st">'E1'</span>)</span>
<span id="cb31-37"><a href="#cb31-37" aria-hidden="true" tabindex="-1"></a>ax1.set_xticks([])<span class="op">;</span></span>
<span id="cb31-38"><a href="#cb31-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-39"><a href="#cb31-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-40"><a href="#cb31-40" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i <span class="kw">in</span> np.where(np.diff( (cis_eigs[<span class="dv">1</span>][<span class="st">'E1'</span>]<span class="op">&gt;</span><span class="dv">0</span>).astype(<span class="bu">int</span>)))[<span class="dv">0</span>]:</span>
<span id="cb31-41"><a href="#cb31-41" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Horisontal lines where E1 intersects 0</span></span>
<span id="cb31-42"><a href="#cb31-42" aria-hidden="true" tabindex="-1"></a>    <span class="co">#ax.plot([0,nbins],[i,i],'k',lw=0.5)</span></span>
<span id="cb31-43"><a href="#cb31-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-44"><a href="#cb31-44" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Vertical lines where E1 intersects 0</span></span>
<span id="cb31-45"><a href="#cb31-45" aria-hidden="true" tabindex="-1"></a>    ax.plot([i,i],[<span class="dv">0</span>,nbins],<span class="st">'k'</span>,lw<span class="op">=</span><span class="fl">0.3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div id="cell-chrX-matrix-e1-500kb" class="cell" data-execution_count="11">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>chrX_matrix_e1_500kb()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div id="chrx-matrix-e1-500kb" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="03_compartments_files/figure-html/chrx-matrix-e1-500kb-output-1.png" class="img-fluid figure-img"></p>
<figcaption>chrX interaction matrix with E1 eigenvector values. 500kb resolution</figcaption>
</figure>
</div>
</div>
</div>
<div id="cell-chrX-e1-500kb-plot" class="cell" data-fig-height="1" data-execution_count="38">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>f, ax1 <span class="op">=</span> plt.subplots(</span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>    figsize<span class="op">=</span>(<span class="dv">15</span>, <span class="dv">5</span>),</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Fill between the line and 0</span></span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a>ax1.fill_between(<span class="bu">range</span>(<span class="bu">len</span>(e1X_values)), e1X_values, <span class="dv">0</span>, where<span class="op">=</span>(e1X_values <span class="op">&gt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:red'</span>)</span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a>ax1.fill_between(<span class="bu">range</span>(<span class="bu">len</span>(e1X_values)), e1X_values, <span class="dv">0</span>, where<span class="op">=</span>(e1X_values <span class="op">&lt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'C0'</span>)</span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a><span class="co">#ax1.set_ylabel('E1')</span></span>
<span id="cb33-12"><a href="#cb33-12" aria-hidden="true" tabindex="-1"></a>ax1.set_xticks([])</span>
<span id="cb33-13"><a href="#cb33-13" aria-hidden="true" tabindex="-1"></a>ax1.set_yticks([])</span>
<span id="cb33-14"><a href="#cb33-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-15"><a href="#cb33-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove borders</span></span>
<span id="cb33-16"><a href="#cb33-16" aria-hidden="true" tabindex="-1"></a>ax1.spines[<span class="st">'top'</span>].set_visible(<span class="va">False</span>)</span>
<span id="cb33-17"><a href="#cb33-17" aria-hidden="true" tabindex="-1"></a>ax1.spines[<span class="st">'right'</span>].set_visible(<span class="va">False</span>)</span>
<span id="cb33-18"><a href="#cb33-18" aria-hidden="true" tabindex="-1"></a>ax1.spines[<span class="st">'left'</span>].set_visible(<span class="va">False</span>)</span>
<span id="cb33-19"><a href="#cb33-19" aria-hidden="true" tabindex="-1"></a>ax1.spines[<span class="st">'bottom'</span>].set_visible(<span class="va">False</span>)</span>
<span id="cb33-20"><a href="#cb33-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-21"><a href="#cb33-21" aria-hidden="true" tabindex="-1"></a>plt.tight_layout()</span>
<span id="cb33-22"><a href="#cb33-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-23"><a href="#cb33-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the plot as a high-resolution PNG file</span></span>
<span id="cb33-24"><a href="#cb33-24" aria-hidden="true" tabindex="-1"></a>plt.savefig(<span class="st">'e1_plot.png'</span>, dpi<span class="op">=</span><span class="dv">320</span>, bbox_inches<span class="op">=</span><span class="st">'tight'</span>)</span>
<span id="cb33-25"><a href="#cb33-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-26"><a href="#cb33-26" aria-hidden="true" tabindex="-1"></a>plt.show()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div id="chrx-e1-500kb-plot" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="03_compartments_files/figure-html/chrx-e1-500kb-plot-output-1.png" class="img-fluid figure-img"></p>
<figcaption>500kb binned E1 eigenvector values for chrX. Freshly calculated from the cooler file.</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="stairs-plot-of-the-e1-compartments" class="level2">
<h2 class="anchored" data-anchor-id="stairs-plot-of-the-e1-compartments">Stairs plot of the E1 compartments</h2>
<p>(Less spiky, more smooth)</p>
<div id="cell-chrX-e1-500kb-stairs" class="cell" data-fig-height="2" data-execution_count="36">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>f, (ax1, ax2) <span class="op">=</span> plt.subplots(<span class="dv">2</span>,<span class="dv">1</span>,</span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>    figsize<span class="op">=</span>(<span class="dv">16</span>, <span class="dv">8</span>)</span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a>chrom_start <span class="op">=</span> eigenvector_track_chrX[<span class="st">'start'</span>].values</span>
<span id="cb34-8"><a href="#cb34-8" aria-hidden="true" tabindex="-1"></a>window_size <span class="op">=</span> chrom_start[<span class="dv">1</span>] <span class="op">-</span> chrom_start[<span class="dv">0</span>]</span>
<span id="cb34-9"><a href="#cb34-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-10"><a href="#cb34-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Fill between the line and 0</span></span>
<span id="cb34-11"><a href="#cb34-11" aria-hidden="true" tabindex="-1"></a>ax1.fill_between(<span class="bu">range</span>(<span class="bu">len</span>(e1X_values)), e1X_values, <span class="dv">0</span>, where<span class="op">=</span>(e1X_values <span class="op">&gt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:red'</span>)</span>
<span id="cb34-12"><a href="#cb34-12" aria-hidden="true" tabindex="-1"></a>ax1.fill_between(<span class="bu">range</span>(<span class="bu">len</span>(e1X_values)), e1X_values, <span class="dv">0</span>, where<span class="op">=</span>(e1X_values <span class="op">&lt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:blue'</span>)</span>
<span id="cb34-13"><a href="#cb34-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-14"><a href="#cb34-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Create stairs</span></span>
<span id="cb34-15"><a href="#cb34-15" aria-hidden="true" tabindex="-1"></a>x <span class="op">=</span> np.zeros(<span class="dv">2</span><span class="op">*</span>chrom_start.size)</span>
<span id="cb34-16"><a href="#cb34-16" aria-hidden="true" tabindex="-1"></a>y <span class="op">=</span> np.zeros(<span class="dv">2</span><span class="op">*</span>chrom_start.size)</span>
<span id="cb34-17"><a href="#cb34-17" aria-hidden="true" tabindex="-1"></a>x[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> chrom_start</span>
<span id="cb34-18"><a href="#cb34-18" aria-hidden="true" tabindex="-1"></a>x[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> chrom_start <span class="op">+</span> window_size</span>
<span id="cb34-19"><a href="#cb34-19" aria-hidden="true" tabindex="-1"></a>y[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> e1X_values</span>
<span id="cb34-20"><a href="#cb34-20" aria-hidden="true" tabindex="-1"></a>y[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> e1X_values</span>
<span id="cb34-21"><a href="#cb34-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-22"><a href="#cb34-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Layout</span></span>
<span id="cb34-23"><a href="#cb34-23" aria-hidden="true" tabindex="-1"></a>ax1.set_ylabel(<span class="st">'Spiky'</span>)</span>
<span id="cb34-24"><a href="#cb34-24" aria-hidden="true" tabindex="-1"></a>ax2.set_ylabel(<span class="st">'Stairs'</span>)</span>
<span id="cb34-25"><a href="#cb34-25" aria-hidden="true" tabindex="-1"></a><span class="co">#ax1.set_xticks([])</span></span>
<span id="cb34-26"><a href="#cb34-26" aria-hidden="true" tabindex="-1"></a><span class="co">#ax1.set_yticks([])</span></span>
<span id="cb34-27"><a href="#cb34-27" aria-hidden="true" tabindex="-1"></a>ax2.set_ylim(<span class="op">-</span><span class="dv">1</span>, <span class="dv">1</span>)</span>
<span id="cb34-28"><a href="#cb34-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-29"><a href="#cb34-29" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove borders</span></span>
<span id="cb34-30"><a href="#cb34-30" aria-hidden="true" tabindex="-1"></a>ax1.spines[:].set_visible(<span class="va">False</span>)</span>
<span id="cb34-31"><a href="#cb34-31" aria-hidden="true" tabindex="-1"></a>ax2.spines[:].set_visible(<span class="va">False</span>)</span>
<span id="cb34-32"><a href="#cb34-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-33"><a href="#cb34-33" aria-hidden="true" tabindex="-1"></a>ax2.fill_between(x, y, <span class="dv">0</span>, where<span class="op">=</span>(y <span class="op">&gt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:red'</span>)</span>
<span id="cb34-34"><a href="#cb34-34" aria-hidden="true" tabindex="-1"></a>ax2.fill_between(x, y, <span class="dv">0</span>, where<span class="op">=</span>(y <span class="op">&lt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:blue'</span>)</span>
<span id="cb34-35"><a href="#cb34-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-36"><a href="#cb34-36" aria-hidden="true" tabindex="-1"></a>plt.tight_layout()</span>
<span id="cb34-37"><a href="#cb34-37" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the plot as a high-resolution PNG file</span></span>
<span id="cb34-38"><a href="#cb34-38" aria-hidden="true" tabindex="-1"></a><span class="co">#plt.savefig('e1_plot.png', dpi=320, bbox_inches='tight')</span></span>
<span id="cb34-39"><a href="#cb34-39" aria-hidden="true" tabindex="-1"></a></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div id="chrx-e1-500kb-stairs" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="03_compartments_files/figure-html/chrx-e1-500kb-stairs-output-1.png" class="img-fluid figure-img"></p>
<figcaption>500kb binned E1 eigenvector values for chrX.</figcaption>
</figure>
</div>
</div>
</div>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../notebooks/02_open2c_framework.html" class="pagination-link" aria-label="Open2C framework">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-title">Open2C framework</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../master_workflow.html" class="pagination-link" aria-label="master_workflow">
        <span class="nav-page-text"><span class="chapter-title">master_workflow</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->




<footer class="footer"><div class="nav-footer"><div class="nav-footer-center"><div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.dev/munch-group/hic-spermatogenesis/blob/main/notebooks/03_compartments.ipynb" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/munch-group/hic-spermatogenesis/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li><li><a href="https://github.com/munch-group/hic-spermatogenesis/blob/main/notebooks/03_compartments.ipynb" class="toc-action"><i class="bi empty"></i>View source</a></li></ul></div></div></div></footer></body></html>