<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Compartments Analysis – Søren's MSc project about Hi-C and spermatogenesis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js" integrity="sha512-bLT0Qm9VnAYZDflyKcBaQ2gg0hSYNQrJ8RilYldYQ1FxQYoCLtUjuuRuZo+fjqhx/qtq/1itJ0C2ejDxltZVFg==" crossorigin="anonymous"></script><script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../master_workflow.html" rel="next">
<link href="../notebooks/02_open2c_framework.html" rel="prev">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://hypothes.is/embed.js"></script>
<script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" integrity="sha512-c3Nl8+7g4LMSTdrm621y7kf9v3SDPnhxLNhcjFJbKECVnmZHTdo+IRO05sNLTH/D3vA6u1X32ehoLC7WFVdheg==" crossorigin="anonymous"></script>

<script type="application/javascript">define('jquery', [],function() {return window.jQuery;})</script>


</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../notebooks/01_hicexplorer.html">Notebooks</a></li><li class="breadcrumb-item"><a href="../notebooks/03_compartments.html"><span class="chapter-title">Compartments Analysis</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">Søren’s MSc project about Hi-C and spermatogenesis</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/munch-group/hic-spermatogenesis" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Project</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../thoughts.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Thoughts and Prayers</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">Notebooks</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/01_hicexplorer.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">HiCExplorer using macaque data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/02_open2c_framework.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Open2C framework</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/03_compartments.html" class="sidebar-item-text sidebar-link active"><span class="chapter-title">Compartments Analysis</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false">
 <span class="menu-text">Workflows</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../master_workflow.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">master_workflow</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../gwf_map_reads.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">gwf_map_reads</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../gwf_bowtie.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">gwf_bowtie</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../gwf_bowtie_local.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">gwf_bowtie_local</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../gwf_hiccorrect.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">gwf_hiccorrect</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../gwf_bwamem.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">gwf_bwamem</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../gwf_pair_alignments.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">gwf_pair_alignments</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false">
 <span class="menu-text">Reports</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../reports/manus/main.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Manuscript</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../reports/manus/supplementary.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Supplementary material</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false">
 <span class="menu-text">Slides</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../slides/project_progress.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Project Progress Day: Msc. Project 2024</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../references.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">References</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul class="collapse">
  <li><a href="#working-with-coolers" id="toc-working-with-coolers" class="nav-link active" data-scroll-target="#working-with-coolers">Working with coolers</a>
  <ul class="collapse">
  <li><a href="#overview" id="toc-overview" class="nav-link" data-scroll-target="#overview">Overview</a></li>
  <li><a href="#fullmerge-pool-all-from-each-source_name" id="toc-fullmerge-pool-all-from-each-source_name" class="nav-link" data-scroll-target="#fullmerge-pool-all-from-each-source_name">FullMerge (pool all from each source_name)</a></li>
  <li><a href="#repmerge-pool-all-from-each-biosample-id" id="toc-repmerge-pool-all-from-each-biosample-id" class="nav-link" data-scroll-target="#repmerge-pool-all-from-each-biosample-id">RepMerge (pool all from each BioSample ID)</a></li>
  </ul></li>
  <li><a href="#plotting" id="toc-plotting" class="nav-link" data-scroll-target="#plotting">Plotting</a>
  <ul class="collapse">
  <li><a href="#kb-resolution" id="toc-kb-resolution" class="nav-link" data-scroll-target="#kb-resolution">500kb resolution</a></li>
  <li><a href="#kb-resolution-1" id="toc-kb-resolution-1" class="nav-link" data-scroll-target="#kb-resolution-1">100kb resolution</a></li>
  </ul></li>
  </ul>
<div class="toc-actions"><ul class="collapse"><li><a href="https://github.dev/munch-group/hic-spermatogenesis/blob/main/notebooks/03_compartments.ipynb" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/munch-group/hic-spermatogenesis/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li><li><a href="https://github.com/munch-group/hic-spermatogenesis/blob/main/notebooks/03_compartments.ipynb" class="toc-action"><i class="bi empty"></i>View source</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content page-columns page-full" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../notebooks/01_hicexplorer.html">Notebooks</a></li><li class="breadcrumb-item"><a href="../notebooks/03_compartments.html"><span class="chapter-title">Compartments Analysis</span></a></li></ol></nav>
<div class="quarto-title">
<h1 class="title"><span class="chapter-title">Compartments Analysis</span></h1>
<p class="subtitle lead">Analyses of Chromatin Compartments using the Open2C Ecosystem</p>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="working-with-coolers" class="level1">
<h1>Working with coolers</h1>
<p>In this notebook, we use files generated by the workflow <code>master_workflow</code> that, in short, does the following:</p>
<ol type="1">
<li>Download the data from the source</li>
<li>Index the reference genome with <code>bwa index</code> and <code>samtools faidx</code></li>
<li>Align the reads to the reference genome with <code>bwa mem</code></li>
<li>Pair and sort the reads to <code>.pairs</code> files with <code>pairtools parse | pairtools sort</code></li>
<li>Deduplicate the pairs with <code>pairtools dedup</code></li>
<li>Convert the .pairs to cooler files with <code>cooler cload pairs</code></li>
</ol>
<p>We will:</p>
<ol type="1">
<li>Load the cooler files</li>
<li>Merge the coolers from the same BioSample ID –&gt; Create ‘replicates’</li>
<li>Zoomify the merged cooler files (coarsen) to create a multicooler (.mcool) file</li>
<li>Balance the matrices (use commandline <code>cooler balance</code>)</li>
<li>Calculate E1 compartments with <code>cooltools compute_cis_eig</code></li>
</ol>
<section id="overview" class="level2">
<h2 class="anchored" data-anchor-id="overview">Overview</h2>
<section id="data-accessions" class="level3">
<h3 class="anchored" data-anchor-id="data-accessions">Data (Accessions)</h3>
<div id="cell-md-data-accessions" class="cell" data-execution_count="2">
<div id="md-data-accessions" class="cell-output cell-output-display cell-output-markdown" data-execution_count="2">
<p>To get an overview of the data accessions used in this analysis, we will first summarize the <code>SRA-runtable.tsv</code> that contains the accession numbers and some metadata for each sample.</p>
</div>
</div>
<div class="cell" data-execution_count="12">
<div id="tbl-runtable" class="cell quarto-float quarto-figure quarto-figure-center anchored" data-execution_count="12">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-runtable-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;6.1: The most relevant columns of the <code>SRA-runtable.tsv</code> file
</figcaption>
<div aria-describedby="tbl-runtable-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output cell-output-display">
<style type="text/css">
</style>

<table id="T_25f78" class="do-not-create-environment cell caption-top table table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th class="blank level0" data-quarto-table-cell-role="th">&nbsp;</th>
<th id="T_25f78_level0_col0" class="col_heading level0 col0" data-quarto-table-cell-role="th">source_name</th>
<th id="T_25f78_level0_col1" class="col_heading level0 col1" data-quarto-table-cell-role="th">BioSample</th>
<th id="T_25f78_level0_col2" class="col_heading level0 col2" data-quarto-table-cell-role="th">Run</th>
<th id="T_25f78_level0_col3" class="col_heading level0 col3" data-quarto-table-cell-role="th">GB</th>
<th id="T_25f78_level0_col4" class="col_heading level0 col4" data-quarto-table-cell-role="th">Bases</th>
<th id="T_25f78_level0_col5" class="col_heading level0 col5" data-quarto-table-cell-role="th">Reads</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td id="T_25f78_level0_row0" class="row_heading level0 row0" data-quarto-table-cell-role="th">16</td>
<td id="T_25f78_row0_col0" class="data row0 col0">fibroblast</td>
<td id="T_25f78_row0_col1" class="data row0 col1">SAMN08375237</td>
<td id="T_25f78_row0_col2" class="data row0 col2">SRR6502335</td>
<td id="T_25f78_row0_col3" class="data row0 col3">29.771059</td>
<td id="T_25f78_row0_col4" class="data row0 col4">73,201,141,800</td>
<td id="T_25f78_row0_col5" class="data row0 col5">244,003,806</td>
</tr>
<tr class="even">
<td id="T_25f78_level0_row1" class="row_heading level0 row1" data-quarto-table-cell-role="th">17</td>
<td id="T_25f78_row1_col0" class="data row1 col0">fibroblast</td>
<td id="T_25f78_row1_col1" class="data row1 col1">SAMN08375237</td>
<td id="T_25f78_row1_col2" class="data row1 col2">SRR6502336</td>
<td id="T_25f78_row1_col3" class="data row1 col3">22.755361</td>
<td id="T_25f78_row1_col4" class="data row1 col4">65,119,970,100</td>
<td id="T_25f78_row1_col5" class="data row1 col5">217,066,567</td>
</tr>
<tr class="odd">
<td id="T_25f78_level0_row2" class="row_heading level0 row2" data-quarto-table-cell-role="th">18</td>
<td id="T_25f78_row2_col0" class="data row2 col0">fibroblast</td>
<td id="T_25f78_row2_col1" class="data row2 col1">SAMN08375236</td>
<td id="T_25f78_row2_col2" class="data row2 col2">SRR6502337</td>
<td id="T_25f78_row2_col3" class="data row2 col3">21.434722</td>
<td id="T_25f78_row2_col4" class="data row2 col4">52,769,196,300</td>
<td id="T_25f78_row2_col5" class="data row2 col5">175,897,321</td>
</tr>
<tr class="even">
<td id="T_25f78_level0_row3" class="row_heading level0 row3" data-quarto-table-cell-role="th">19</td>
<td id="T_25f78_row3_col0" class="data row3 col0">fibroblast</td>
<td id="T_25f78_row3_col1" class="data row3 col1">SAMN08375236</td>
<td id="T_25f78_row3_col2" class="data row3 col2">SRR6502338</td>
<td id="T_25f78_row3_col3" class="data row3 col3">21.420030</td>
<td id="T_25f78_row3_col4" class="data row3 col4">52,378,949,100</td>
<td id="T_25f78_row3_col5" class="data row3 col5">174,596,497</td>
</tr>
<tr class="odd">
<td id="T_25f78_level0_row4" class="row_heading level0 row4" data-quarto-table-cell-role="th">20</td>
<td id="T_25f78_row4_col0" class="data row4 col0">fibroblast</td>
<td id="T_25f78_row4_col1" class="data row4 col1">SAMN08375236</td>
<td id="T_25f78_row4_col2" class="data row4 col2">SRR6502339</td>
<td id="T_25f78_row4_col3" class="data row4 col3">10.207410</td>
<td id="T_25f78_row4_col4" class="data row4 col4">28,885,941,600</td>
<td id="T_25f78_row4_col5" class="data row4 col5">96,286,472</td>
</tr>
<tr class="even">
<td id="T_25f78_level0_row5" class="row_heading level0 row5" data-quarto-table-cell-role="th">9</td>
<td id="T_25f78_row5_col0" class="data row5 col0">fibroblast</td>
<td id="T_25f78_row5_col1" class="data row5 col1">SAMN08375237</td>
<td id="T_25f78_row5_col2" class="data row5 col2">SRR7349189</td>
<td id="T_25f78_row5_col3" class="data row5 col3">52.729173</td>
<td id="T_25f78_row5_col4" class="data row5 col4">139,604,854,200</td>
<td id="T_25f78_row5_col5" class="data row5 col5">465,349,514</td>
</tr>
<tr class="odd">
<td id="T_25f78_level0_row6" class="row_heading level0 row6" data-quarto-table-cell-role="th">10</td>
<td id="T_25f78_row6_col0" class="data row6 col0">fibroblast</td>
<td id="T_25f78_row6_col1" class="data row6 col1">SAMN08375236</td>
<td id="T_25f78_row6_col2" class="data row6 col2">SRR7349190</td>
<td id="T_25f78_row6_col3" class="data row6 col3">53.085520</td>
<td id="T_25f78_row6_col4" class="data row6 col4">142,008,353,400</td>
<td id="T_25f78_row6_col5" class="data row6 col5">473,361,178</td>
</tr>
<tr class="even">
<td id="T_25f78_level0_row7" class="row_heading level0 row7" data-quarto-table-cell-role="th">21</td>
<td id="T_25f78_row7_col0" class="data row7 col0">pachytene spermatocyte</td>
<td id="T_25f78_row7_col1" class="data row7 col1">SAMN08375234</td>
<td id="T_25f78_row7_col2" class="data row7 col2">SRR6502342</td>
<td id="T_25f78_row7_col3" class="data row7 col3">60.258880</td>
<td id="T_25f78_row7_col4" class="data row7 col4">150,370,993,500</td>
<td id="T_25f78_row7_col5" class="data row7 col5">501,236,645</td>
</tr>
<tr class="odd">
<td id="T_25f78_level0_row8" class="row_heading level0 row8" data-quarto-table-cell-role="th">22</td>
<td id="T_25f78_row8_col0" class="data row8 col0">pachytene spermatocyte</td>
<td id="T_25f78_row8_col1" class="data row8 col1">SAMN08375234</td>
<td id="T_25f78_row8_col2" class="data row8 col2">SRR6502344</td>
<td id="T_25f78_row8_col3" class="data row8 col3">27.146048</td>
<td id="T_25f78_row8_col4" class="data row8 col4">65,697,684,300</td>
<td id="T_25f78_row8_col5" class="data row8 col5">218,992,281</td>
</tr>
<tr class="even">
<td id="T_25f78_level0_row9" class="row_heading level0 row9" data-quarto-table-cell-role="th">23</td>
<td id="T_25f78_row9_col0" class="data row9 col0">pachytene spermatocyte</td>
<td id="T_25f78_row9_col1" class="data row9 col1">SAMN08375234</td>
<td id="T_25f78_row9_col2" class="data row9 col2">SRR6502345</td>
<td id="T_25f78_row9_col3" class="data row9 col3">26.202707</td>
<td id="T_25f78_row9_col4" class="data row9 col4">63,490,538,700</td>
<td id="T_25f78_row9_col5" class="data row9 col5">211,635,129</td>
</tr>
<tr class="odd">
<td id="T_25f78_level0_row10" class="row_heading level0 row10" data-quarto-table-cell-role="th">0</td>
<td id="T_25f78_row10_col0" class="data row10 col0">pachytene spermatocyte</td>
<td id="T_25f78_row10_col1" class="data row10 col1">SAMN09427370</td>
<td id="T_25f78_row10_col2" class="data row10 col2">SRR7345458</td>
<td id="T_25f78_row10_col3" class="data row10 col3">55.970557</td>
<td id="T_25f78_row10_col4" class="data row10 col4">153,281,577,900</td>
<td id="T_25f78_row10_col5" class="data row10 col5">510,938,593</td>
</tr>
<tr class="even">
<td id="T_25f78_level0_row11" class="row_heading level0 row11" data-quarto-table-cell-role="th">1</td>
<td id="T_25f78_row11_col0" class="data row11 col0">pachytene spermatocyte</td>
<td id="T_25f78_row11_col1" class="data row11 col1">SAMN09427370</td>
<td id="T_25f78_row11_col2" class="data row11 col2">SRR7345459</td>
<td id="T_25f78_row11_col3" class="data row11 col3">53.982492</td>
<td id="T_25f78_row11_col4" class="data row11 col4">144,993,841,200</td>
<td id="T_25f78_row11_col5" class="data row11 col5">483,312,804</td>
</tr>
<tr class="odd">
<td id="T_25f78_level0_row12" class="row_heading level0 row12" data-quarto-table-cell-role="th">11</td>
<td id="T_25f78_row12_col0" class="data row12 col0">pachytene spermatocyte</td>
<td id="T_25f78_row12_col1" class="data row12 col1">SAMN08375235</td>
<td id="T_25f78_row12_col2" class="data row12 col2">SRR7349191</td>
<td id="T_25f78_row12_col3" class="data row12 col3">51.274476</td>
<td id="T_25f78_row12_col4" class="data row12 col4">137,821,979,100</td>
<td id="T_25f78_row12_col5" class="data row12 col5">459,406,597</td>
</tr>
<tr class="even">
<td id="T_25f78_level0_row13" class="row_heading level0 row13" data-quarto-table-cell-role="th">24</td>
<td id="T_25f78_row13_col0" class="data row13 col0">round spermatid</td>
<td id="T_25f78_row13_col1" class="data row13 col1">SAMN08375232</td>
<td id="T_25f78_row13_col2" class="data row13 col2">SRR6502351</td>
<td id="T_25f78_row13_col3" class="data row13 col3">20.924497</td>
<td id="T_25f78_row13_col4" class="data row13 col4">55,095,075,300</td>
<td id="T_25f78_row13_col5" class="data row13 col5">183,650,251</td>
</tr>
<tr class="odd">
<td id="T_25f78_level0_row14" class="row_heading level0 row14" data-quarto-table-cell-role="th">25</td>
<td id="T_25f78_row14_col0" class="data row14 col0">round spermatid</td>
<td id="T_25f78_row14_col1" class="data row14 col1">SAMN08375232</td>
<td id="T_25f78_row14_col2" class="data row14 col2">SRR6502352</td>
<td id="T_25f78_row14_col3" class="data row14 col3">41.133960</td>
<td id="T_25f78_row14_col4" class="data row14 col4">115,578,475,800</td>
<td id="T_25f78_row14_col5" class="data row14 col5">385,261,586</td>
</tr>
<tr class="even">
<td id="T_25f78_level0_row15" class="row_heading level0 row15" data-quarto-table-cell-role="th">26</td>
<td id="T_25f78_row15_col0" class="data row15 col0">round spermatid</td>
<td id="T_25f78_row15_col1" class="data row15 col1">SAMN08375232</td>
<td id="T_25f78_row15_col2" class="data row15 col2">SRR6502353</td>
<td id="T_25f78_row15_col3" class="data row15 col3">36.444117</td>
<td id="T_25f78_row15_col4" class="data row15 col4">96,195,161,400</td>
<td id="T_25f78_row15_col5" class="data row15 col5">320,650,538</td>
</tr>
<tr class="odd">
<td id="T_25f78_level0_row16" class="row_heading level0 row16" data-quarto-table-cell-role="th">2</td>
<td id="T_25f78_row16_col0" class="data row16 col0">round spermatid</td>
<td id="T_25f78_row16_col1" class="data row16 col1">SAMN09427369</td>
<td id="T_25f78_row16_col2" class="data row16 col2">SRR7345460</td>
<td id="T_25f78_row16_col3" class="data row16 col3">38.244654</td>
<td id="T_25f78_row16_col4" class="data row16 col4">104,105,827,200</td>
<td id="T_25f78_row16_col5" class="data row16 col5">347,019,424</td>
</tr>
<tr class="even">
<td id="T_25f78_level0_row17" class="row_heading level0 row17" data-quarto-table-cell-role="th">3</td>
<td id="T_25f78_row17_col0" class="data row17 col0">round spermatid</td>
<td id="T_25f78_row17_col1" class="data row17 col1">SAMN09427369</td>
<td id="T_25f78_row17_col2" class="data row17 col2">SRR7345461</td>
<td id="T_25f78_row17_col3" class="data row17 col3">53.996261</td>
<td id="T_25f78_row17_col4" class="data row17 col4">144,532,309,500</td>
<td id="T_25f78_row17_col5" class="data row17 col5">481,774,365</td>
</tr>
<tr class="odd">
<td id="T_25f78_level0_row18" class="row_heading level0 row18" data-quarto-table-cell-role="th">12</td>
<td id="T_25f78_row18_col0" class="data row18 col0">round spermatid</td>
<td id="T_25f78_row18_col1" class="data row18 col1">SAMN08375232</td>
<td id="T_25f78_row18_col2" class="data row18 col2">SRR7349192</td>
<td id="T_25f78_row18_col3" class="data row18 col3">52.384556</td>
<td id="T_25f78_row18_col4" class="data row18 col4">140,431,608,000</td>
<td id="T_25f78_row18_col5" class="data row18 col5">468,105,360</td>
</tr>
<tr class="even">
<td id="T_25f78_level0_row19" class="row_heading level0 row19" data-quarto-table-cell-role="th">29</td>
<td id="T_25f78_row19_col0" class="data row19 col0">sperm</td>
<td id="T_25f78_row19_col1" class="data row19 col1">SAMN08375229</td>
<td id="T_25f78_row19_col2" class="data row19 col2">SRR6502360</td>
<td id="T_25f78_row19_col3" class="data row19 col3">26.653940</td>
<td id="T_25f78_row19_col4" class="data row19 col4">64,752,370,800</td>
<td id="T_25f78_row19_col5" class="data row19 col5">215,841,236</td>
</tr>
<tr class="odd">
<td id="T_25f78_level0_row20" class="row_heading level0 row20" data-quarto-table-cell-role="th">30</td>
<td id="T_25f78_row20_col0" class="data row20 col0">sperm</td>
<td id="T_25f78_row20_col1" class="data row20 col1">SAMN08375228</td>
<td id="T_25f78_row20_col2" class="data row20 col2">SRR6502362</td>
<td id="T_25f78_row20_col3" class="data row20 col3">23.973440</td>
<td id="T_25f78_row20_col4" class="data row20 col4">58,369,232,700</td>
<td id="T_25f78_row20_col5" class="data row20 col5">194,564,109</td>
</tr>
<tr class="even">
<td id="T_25f78_level0_row21" class="row_heading level0 row21" data-quarto-table-cell-role="th">13</td>
<td id="T_25f78_row21_col0" class="data row21 col0">sperm</td>
<td id="T_25f78_row21_col1" class="data row21 col1">SAMN08375229</td>
<td id="T_25f78_row21_col2" class="data row21 col2">SRR7349193</td>
<td id="T_25f78_row21_col3" class="data row21 col3">52.806276</td>
<td id="T_25f78_row21_col4" class="data row21 col4">141,148,572,300</td>
<td id="T_25f78_row21_col5" class="data row21 col5">470,495,241</td>
</tr>
<tr class="odd">
<td id="T_25f78_level0_row22" class="row_heading level0 row22" data-quarto-table-cell-role="th">14</td>
<td id="T_25f78_row22_col0" class="data row22 col0">sperm</td>
<td id="T_25f78_row22_col1" class="data row22 col1">SAMN08375229</td>
<td id="T_25f78_row22_col2" class="data row22 col2">SRR7349195</td>
<td id="T_25f78_row22_col3" class="data row22 col3">22.444378</td>
<td id="T_25f78_row22_col4" class="data row22 col4">60,523,788,600</td>
<td id="T_25f78_row22_col5" class="data row22 col5">201,745,962</td>
</tr>
<tr class="even">
<td id="T_25f78_level0_row23" class="row_heading level0 row23" data-quarto-table-cell-role="th">15</td>
<td id="T_25f78_row23_col0" class="data row23 col0">sperm</td>
<td id="T_25f78_row23_col1" class="data row23 col1">SAMN08375229</td>
<td id="T_25f78_row23_col2" class="data row23 col2">SRR7349196</td>
<td id="T_25f78_row23_col3" class="data row23 col3">38.253606</td>
<td id="T_25f78_row23_col4" class="data row23 col4">104,119,671,000</td>
<td id="T_25f78_row23_col5" class="data row23 col5">347,065,570</td>
</tr>
<tr class="odd">
<td id="T_25f78_level0_row24" class="row_heading level0 row24" data-quarto-table-cell-role="th">27</td>
<td id="T_25f78_row24_col0" class="data row24 col0">spermatogonia</td>
<td id="T_25f78_row24_col1" class="data row24 col1">SAMN08375231</td>
<td id="T_25f78_row24_col2" class="data row24 col2">SRR6502356</td>
<td id="T_25f78_row24_col3" class="data row24 col3">22.845286</td>
<td id="T_25f78_row24_col4" class="data row24 col4">58,909,579,800</td>
<td id="T_25f78_row24_col5" class="data row24 col5">196,365,266</td>
</tr>
<tr class="even">
<td id="T_25f78_level0_row25" class="row_heading level0 row25" data-quarto-table-cell-role="th">28</td>
<td id="T_25f78_row25_col0" class="data row25 col0">spermatogonia</td>
<td id="T_25f78_row25_col1" class="data row25 col1">SAMN08375231</td>
<td id="T_25f78_row25_col2" class="data row25 col2">SRR6502357</td>
<td id="T_25f78_row25_col3" class="data row25 col3">17.947471</td>
<td id="T_25f78_row25_col4" class="data row25 col4">46,888,332,900</td>
<td id="T_25f78_row25_col5" class="data row25 col5">156,294,443</td>
</tr>
<tr class="odd">
<td id="T_25f78_level0_row26" class="row_heading level0 row26" data-quarto-table-cell-role="th">4</td>
<td id="T_25f78_row26_col0" class="data row26 col0">spermatogonia</td>
<td id="T_25f78_row26_col1" class="data row26 col1">SAMN09427379</td>
<td id="T_25f78_row26_col2" class="data row26 col2">SRR7345462</td>
<td id="T_25f78_row26_col3" class="data row26 col3">18.686342</td>
<td id="T_25f78_row26_col4" class="data row26 col4">52,032,780,000</td>
<td id="T_25f78_row26_col5" class="data row26 col5">173,442,600</td>
</tr>
<tr class="even">
<td id="T_25f78_level0_row27" class="row_heading level0 row27" data-quarto-table-cell-role="th">5</td>
<td id="T_25f78_row27_col0" class="data row27 col0">spermatogonia</td>
<td id="T_25f78_row27_col1" class="data row27 col1">SAMN09427379</td>
<td id="T_25f78_row27_col2" class="data row27 col2">SRR7345463</td>
<td id="T_25f78_row27_col3" class="data row27 col3">29.956561</td>
<td id="T_25f78_row27_col4" class="data row27 col4">82,384,836,000</td>
<td id="T_25f78_row27_col5" class="data row27 col5">274,616,120</td>
</tr>
<tr class="odd">
<td id="T_25f78_level0_row28" class="row_heading level0 row28" data-quarto-table-cell-role="th">6</td>
<td id="T_25f78_row28_col0" class="data row28 col0">spermatogonia</td>
<td id="T_25f78_row28_col1" class="data row28 col1">SAMN09427379</td>
<td id="T_25f78_row28_col2" class="data row28 col2">SRR7345464</td>
<td id="T_25f78_row28_col3" class="data row28 col3">39.145759</td>
<td id="T_25f78_row28_col4" class="data row28 col4">105,153,716,100</td>
<td id="T_25f78_row28_col5" class="data row28 col5">350,512,387</td>
</tr>
<tr class="even">
<td id="T_25f78_level0_row29" class="row_heading level0 row29" data-quarto-table-cell-role="th">7</td>
<td id="T_25f78_row29_col0" class="data row29 col0">spermatogonia</td>
<td id="T_25f78_row29_col1" class="data row29 col1">SAMN09427378</td>
<td id="T_25f78_row29_col2" class="data row29 col2">SRR7345465</td>
<td id="T_25f78_row29_col3" class="data row29 col3">35.816184</td>
<td id="T_25f78_row29_col4" class="data row29 col4">96,048,594,600</td>
<td id="T_25f78_row29_col5" class="data row29 col5">320,161,982</td>
</tr>
<tr class="odd">
<td id="T_25f78_level0_row30" class="row_heading level0 row30" data-quarto-table-cell-role="th">8</td>
<td id="T_25f78_row30_col0" class="data row30 col0">spermatogonia</td>
<td id="T_25f78_row30_col1" class="data row30 col1">SAMN09427378</td>
<td id="T_25f78_row30_col2" class="data row30 col2">SRR7345467</td>
<td id="T_25f78_row30_col3" class="data row30 col3">28.396816</td>
<td id="T_25f78_row30_col4" class="data row30 col4">77,248,140,900</td>
<td id="T_25f78_row30_col5" class="data row30 col5">257,493,803</td>
</tr>
</tbody>
</table>
</div>
</div>
</figure>
</div>
</div>
<div class="cell" data-execution_count="13">
<div id="tbl-runtable-summary" class="cell quarto-float quarto-figure quarto-figure-center anchored" data-execution_count="13">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-runtable-summary-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;6.2: Summary of the data accessions used in this analysis
</figcaption>
<div aria-describedby="tbl-runtable-summary-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output cell-output-display">
<style type="text/css">
</style>

<table id="T_543d4" class="do-not-create-environment cell caption-top table table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th class="blank level0" data-quarto-table-cell-role="th">&nbsp;</th>
<th id="T_543d4_level0_col0" class="col_heading level0 col0" data-quarto-table-cell-role="th">source_name</th>
<th id="T_543d4_level0_col1" class="col_heading level0 col1" data-quarto-table-cell-role="th">GB</th>
<th id="T_543d4_level0_col2" class="col_heading level0 col2" data-quarto-table-cell-role="th">Bases</th>
<th id="T_543d4_level0_col3" class="col_heading level0 col3" data-quarto-table-cell-role="th">Reads</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td id="T_543d4_level0_row0" class="row_heading level0 row0" data-quarto-table-cell-role="th">0</td>
<td id="T_543d4_row0_col0" class="data row0 col0">fibroblast</td>
<td id="T_543d4_row0_col1" class="data row0 col1">211.403275</td>
<td id="T_543d4_row0_col2" class="data row0 col2">553,968,406,500</td>
<td id="T_543d4_row0_col3" class="data row0 col3">1,846,561,355</td>
</tr>
<tr class="even">
<td id="T_543d4_level0_row1" class="row_heading level0 row1" data-quarto-table-cell-role="th">1</td>
<td id="T_543d4_row1_col0" class="data row1 col0">pachytene spermatocyte</td>
<td id="T_543d4_row1_col1" class="data row1 col1">274.835160</td>
<td id="T_543d4_row1_col2" class="data row1 col2">715,656,614,700</td>
<td id="T_543d4_row1_col3" class="data row1 col3">2,385,522,049</td>
</tr>
<tr class="odd">
<td id="T_543d4_level0_row2" class="row_heading level0 row2" data-quarto-table-cell-role="th">2</td>
<td id="T_543d4_row2_col0" class="data row2 col0">round spermatid</td>
<td id="T_543d4_row2_col1" class="data row2 col1">243.128044</td>
<td id="T_543d4_row2_col2" class="data row2 col2">655,938,457,200</td>
<td id="T_543d4_row2_col3" class="data row2 col3">2,186,461,524</td>
</tr>
<tr class="even">
<td id="T_543d4_level0_row3" class="row_heading level0 row3" data-quarto-table-cell-role="th">3</td>
<td id="T_543d4_row3_col0" class="data row3 col0">sperm</td>
<td id="T_543d4_row3_col1" class="data row3 col1">164.131640</td>
<td id="T_543d4_row3_col2" class="data row3 col2">428,913,635,400</td>
<td id="T_543d4_row3_col3" class="data row3 col3">1,429,712,118</td>
</tr>
<tr class="odd">
<td id="T_543d4_level0_row4" class="row_heading level0 row4" data-quarto-table-cell-role="th">4</td>
<td id="T_543d4_row4_col0" class="data row4 col0">spermatogonia</td>
<td id="T_543d4_row4_col1" class="data row4 col1">192.794420</td>
<td id="T_543d4_row4_col2" class="data row4 col2">518,665,980,300</td>
<td id="T_543d4_row4_col3" class="data row4 col3">1,728,886,601</td>
</tr>
</tbody>
</table>
</div>
</div>
</figure>
</div>
</div>
</section>
<section id="folder-structure" class="level3">
<h3 class="anchored" data-anchor-id="folder-structure">Folder structure</h3>
<div id="cell-md-folder-structure" class="cell" data-execution_count="15">
<div id="md-folder-structure" class="cell-output cell-output-display cell-output-markdown" data-execution_count="15">
<p>For ease of mind, here is the folder structure of the project. <code>../steps/bwa/PE/</code> is the base directory artificially defined in the <code>master_workflow.py</code>. It ccould be any other directory inside <code>steps</code>. It is defined relative to the <code>master_workflow.py</code> file (inside the workdlow), and converted to an absolute path by python.</p>
</div>
</div>
<div id="folder-structure" class="cell" data-execution_count="28">
<div class="cell-output cell-output-stdout">
<pre><code>../steps/bwa/PE
├── bamfiles
│&nbsp;&nbsp; ├── fibroblast
│&nbsp;&nbsp; ├── pachytene_spermatocyte
│&nbsp;&nbsp; ├── round_spermatid
│&nbsp;&nbsp; ├── sperm
│&nbsp;&nbsp; └── spermatogonia
├── cool
│&nbsp;&nbsp; ├── fibroblast
│&nbsp;&nbsp; ├── pachytene_spermatocyte
│&nbsp;&nbsp; ├── round_spermatid
│&nbsp;&nbsp; ├── sperm
│&nbsp;&nbsp; └── spermatogonia
└── pairs
    ├── fibroblast
    ├── pachytene_spermatocyte
    ├── round_spermatid
    ├── sperm
    └── spermatogonia

18 directories</code></pre>
</div>
</div>
</section>
</section>
<section id="fullmerge-pool-all-from-each-source_name" class="level2">
<h2 class="anchored" data-anchor-id="fullmerge-pool-all-from-each-source_name">FullMerge (pool all from each source_name)</h2>
<div id="cell-md-fullmerge-coolers" class="cell" data-execution_count="32">
<div id="md-fullmerge-coolers" class="cell-output cell-output-display cell-output-markdown" data-execution_count="32">
<p>We will use <code>cooler merge</code> to merge all samples in each sub-folder (cell type) to just one interaction matrix for each cell type. The reason for that is that we choose to trust <span class="citation" data-cites="wang_reprogramming_2019">(<a href="../references.html#ref-wang_reprogramming_2019" role="doc-biblioref">Wang et al. 2019</a>)</span> when they say that compartments are highly reproducible between replicates, and by merging all replicates, we will have a more robust signal.</p>
<p>Thus, we will merge all samples from the same source_name into a single cooler file.</p>
<p><strong>Overview of this section:</strong></p>
<ul>
<li>Locate the coolers (<code>glob</code> –&gt; dictionary)</li>
<li>Merge the coolers (<code>cooler.merge_coolers</code>)</li>
<li>Zoomify the merged cooler (<code>cooler.zoomify_cooler</code>) to resolutions: 10kb, 50kb, 100kb, 500kb.</li>
<li>Balance the matrices (<code>!cooler balance</code>) (use the CLI, as it is more easily parallelized)</li>
</ul>
</div>
</div>
<section id="create-cooler-dictionary-glob" class="level3">
<h3 class="anchored" data-anchor-id="create-cooler-dictionary-glob">Create cooler dictionary (<code>glob</code>)</h3>
<div id="cell-md-create-cooler-dict" class="cell" data-execution_count="33">
<div id="md-create-cooler-dict" class="cell-output cell-output-display cell-output-markdown" data-execution_count="33">
<p>First, we will create a dictionary with the paths to the coolers for each sample. We use <code>glob.glob</code> to fetch all the coolers in each sub-folder that remain after filtering and (automatic) quality control (those are the ones with <code>nodups</code> in their names).</p>
</div>
</div>
<div id="cell-16" class="cell" data-execution_count="1">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> glob</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os.path <span class="im">as</span> op</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> pprint <span class="im">import</span> pprint <span class="im">as</span> pp</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the list of cell type dirs</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>base_dir <span class="op">=</span> <span class="st">'../steps/bwa/PE/cool'</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>folders <span class="op">=</span> glob.glob(op.join(base_dir, <span class="st">'*'</span>))</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>files_dict <span class="op">=</span> {f:glob.glob(<span class="ss">f"</span><span class="sc">{</span>f<span class="sc">}</span><span class="ss">/*.nodups.*"</span>) <span class="cf">for</span> f <span class="kw">in</span> folders}</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>cooler_dict <span class="op">=</span> {op.basename(k): [op.basename(f) <span class="cf">for</span> f <span class="kw">in</span> v] <span class="cf">for</span> k,v <span class="kw">in</span> files_dict.items()}</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>pp(cooler_dict)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>{'fibroblast': ['SRR6502339.nodups.10000.cool',
                'SRR7349190.nodups.10000.cool',
                'SRR7349189.nodups.10000.cool',
                'SRR6502335.nodups.10000.cool',
                'SRR6502338.nodups.10000.cool',
                'SRR6502336.nodups.10000.cool',
                'SRR6502337.nodups.10000.cool'],
 'pachytene_spermatocyte': ['SRR7345459.nodups.10000.cool',
                            'SRR6502344.nodups.10000.cool',
                            'SRR6502342.nodups.10000.cool',
                            'SRR7345458.nodups.10000.cool',
                            'SRR6502345.nodups.10000.cool',
                            'SRR7349191.nodups.10000.cool'],
 'round_spermatid': ['SRR7349192.nodups.10000.cool',
                     'SRR6502353.nodups.10000.cool',
                     'SRR6502352.nodups.10000.cool',
                     'SRR6502351.nodups.10000.cool',
                     'SRR7345460.nodups.10000.cool',
                     'SRR7345461.nodups.10000.cool'],
 'sperm': ['SRR7349196.nodups.10000.cool',
           'SRR6502362.nodups.10000.cool',
           'SRR7349193.nodups.10000.cool',
           'SRR6502360.nodups.10000.cool',
           'SRR7349195.nodups.10000.cool'],
 'spermatogonia': ['SRR6502357.nodups.10000.cool',
                   'SRR7345467.nodups.10000.cool',
                   'SRR6502356.nodups.10000.cool',
                   'SRR7345464.nodups.10000.cool',
                   'SRR7345462.nodups.10000.cool',
                   'SRR7345465.nodups.10000.cool',
                   'SRR7345463.nodups.10000.cool']}</code></pre>
</div>
</div>
</section>
<section id="merge-coolers" class="level3">
<h3 class="anchored" data-anchor-id="merge-coolers">Merge coolers</h3>
<div id="cell-md-merge-coolers" class="cell" data-execution_count="35">
<div id="md-merge-coolers" class="cell-output cell-output-display cell-output-markdown" data-execution_count="35">
<p>The coolers are merged by summing each bin in the matrices, meaning we can only merge matrices with same dimensions. We iterate through the dictionary and merge the coolers with <code>cooler merge</code>. The <code>mergebuf</code> parameter should be adjusted if you don’t have 32G memory. Default: <code>mergebuf = 20000000</code>. Below, we also check if the output file already exists. If it does, we skip the merge.</p>
</div>
</div>
<div id="cell-19" class="cell" data-execution_count="5">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># NB adjust `mergebuf` if you don't have 32G of RAM</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooler</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> folder,cooler_list <span class="kw">in</span> cooler_dict.items():</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>    in_uris <span class="op">=</span> [op.join(base_dir, folder, <span class="bu">file</span>) <span class="cf">for</span> <span class="bu">file</span> <span class="kw">in</span> cooler_list]</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>    out_uri <span class="op">=</span> op.join(base_dir, folder, <span class="ss">f'</span><span class="sc">{</span>folder<span class="sc">}</span><span class="ss">.fullmerge.cool'</span>)</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> op.exists(out_uri):</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Skipping </span><span class="sc">{</span>out_uri<span class="sc">}</span><span class="ss">: exists..."</span>)</span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>        <span class="cf">continue</span></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Creating </span><span class="sc">{</span>out_uri<span class="sc">}</span><span class="ss"> by </span><span class="ch">\n</span><span class="ss">Merging </span><span class="sc">{</span><span class="bu">len</span>(cooler_list)<span class="sc">}</span><span class="ss"> coolers into one:"</span>, end<span class="op">=</span><span class="st">" "</span>)</span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"</span><span class="ch">\t</span><span class="st">"</span>,[<span class="bu">file</span>.split(<span class="st">'.'</span>)[<span class="dv">0</span>] <span class="cf">for</span> <span class="bu">file</span> <span class="kw">in</span> cooler_list])</span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>    cooler.merge_coolers(output_uri<span class="op">=</span>out_uri,</span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>                         input_uris<span class="op">=</span>in_uris,</span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>                         mergebuf<span class="op">=</span><span class="bu">int</span>(<span class="fl">5e7</span>),</span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>                         )</span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"... Done!"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Skipping ../steps/bwa/PE/cool/round_spermatid/round_spermatid.fullmerge.cool: exists...
Skipping ../steps/bwa/PE/cool/spermatogonia/spermatogonia.fullmerge.cool: exists...
Skipping ../steps/bwa/PE/cool/sperm/sperm.fullmerge.cool: exists...
Skipping ../steps/bwa/PE/cool/fibroblast/fibroblast.fullmerge.cool: exists...
Skipping ../steps/bwa/PE/cool/pachytene_spermatocyte/pachytene_spermatocyte.fullmerge.cool: exists...</code></pre>
</div>
</div>
</section>
<section id="zoomify-the-merged-cooler-files" class="level3">
<h3 class="anchored" data-anchor-id="zoomify-the-merged-cooler-files">Zoomify the merged cooler files</h3>
<p>Here, we zoomify the merged cooler files to resolutions: 10kb, 50kb, 100kb, 500kb.</p>
<div id="cell-21" class="cell" data-execution_count="7">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># NB 8 cores and 32G of RAM was used</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> glob</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooler</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os.path <span class="im">as</span> op</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>base_dir <span class="op">=</span> <span class="st">'../steps/bwa/PE/cool'</span></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>merged_coolers <span class="op">=</span> glob.glob(op.join(base_dir, <span class="st">'*/*.fullmerge.cool'</span>))</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> clr <span class="kw">in</span> merged_coolers:</span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>    out_uri <span class="op">=</span> clr.replace(<span class="st">'.fullmerge.cool'</span>, <span class="st">'.mcool'</span>)</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> op.exists(out_uri):</span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Skipping </span><span class="sc">{</span>out_uri<span class="sc">}</span><span class="ss">: exists..."</span>)</span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>        <span class="cf">continue</span></span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Zoomifying cooler: </span><span class="ch">\n\t</span><span class="ss">   </span><span class="sc">{</span>clr<span class="sc">}</span><span class="ch">\n\t</span><span class="ss">-&gt; </span><span class="sc">{</span>out_uri<span class="sc">}</span><span class="ss">"</span>, end<span class="op">=</span><span class="st">""</span>)</span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>    cooler.zoomify_cooler(base_uris <span class="op">=</span> clr,</span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a>                          outfile <span class="op">=</span> out_uri,</span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a>                          resolutions <span class="op">=</span> [<span class="dv">10000</span>,<span class="dv">50000</span>,<span class="dv">100000</span>,<span class="dv">500000</span>],</span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a>                          chunksize <span class="op">=</span> <span class="dv">10000000</span>,</span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a>                          nproc <span class="op">=</span> <span class="dv">8</span>)</span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">" --&gt; done"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Skipping ../steps/bwa/PE/cool/round_spermatid/round_spermatid.mcool: exists...
Skipping ../steps/bwa/PE/cool/spermatogonia/spermatogonia.mcool: exists...
Skipping ../steps/bwa/PE/cool/sperm/sperm.mcool: exists...
Zoomifying cooler: 
       ../steps/bwa/PE/cool/fibroblast/fibroblast.fullmerge.cool
    -&gt; ../steps/bwa/PE/cool/fibroblast/fibroblast.mcool --&gt; done
Skipping ../steps/bwa/PE/cool/pachytene_spermatocyte/pachytene_spermatocyte.mcool: exists...</code></pre>
</div>
</div>
<div id="cell-22" class="cell" data-execution_count="19">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> glob</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>mcools <span class="op">=</span> glob.glob(<span class="st">"../steps/bwa/PE/cool/*/*.mcool"</span>)</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> mcool <span class="kw">in</span> mcools:</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"</span><span class="sc">{</span>mcool<span class="sc">}</span><span class="ss">:"</span>)</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(cooler.fileops.list_coolers(mcool))</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>()</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>../steps/bwa/PE/cool/round_spermatid/round_spermatid.mcool:
['/resolutions/10000', '/resolutions/50000', '/resolutions/100000', '/resolutions/500000']

../steps/bwa/PE/cool/spermatogonia/spermatogonia.mcool:
['/resolutions/10000', '/resolutions/50000', '/resolutions/100000', '/resolutions/500000']

../steps/bwa/PE/cool/sperm/sperm.mcool:
['/resolutions/10000', '/resolutions/50000', '/resolutions/100000', '/resolutions/500000']

../steps/bwa/PE/cool/fibroblast/fibroblast.mcool:
['/resolutions/10000', '/resolutions/50000', '/resolutions/100000', '/resolutions/500000']

../steps/bwa/PE/cool/pachytene_spermatocyte/pachytene_spermatocyte.mcool:
['/resolutions/10000', '/resolutions/50000', '/resolutions/100000', '/resolutions/500000']
</code></pre>
</div>
<div class="cell-output cell-output-display">
<pre><code>0    chr1
Name: chrom, dtype: category
Categories (22, object): ['chr1' &lt; 'chr2' &lt; 'chr3' &lt; 'chr4' ... 'chr19' &lt; 'chr20' &lt; 'chrX' &lt; 'chrY']</code></pre>
</div>
<div class="cell-output cell-output-display">
<pre><code>0    0
Name: start, dtype: int32</code></pre>
</div>
<div class="cell-output cell-output-display">
<pre><code>0    500000
Name: end, dtype: int32</code></pre>
</div>
<div class="cell-output cell-output-display">
<pre><code>0    0.002814
Name: weight, dtype: float64</code></pre>
</div>
</div>
</section>
<section id="balance-the-matrices" class="level3">
<h3 class="anchored" data-anchor-id="balance-the-matrices">Balance the matrices</h3>
<div id="cell-24" class="cell" data-execution_count="2">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="op">%</span>capture <span class="op">--</span>no<span class="op">-</span>stdout</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> glob</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>mcools <span class="op">=</span> glob.glob(<span class="st">"../steps/bwa/PE/cool/*/*.mcool"</span>)</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>resolutions <span class="op">=</span> [<span class="dv">10000</span>, <span class="dv">50000</span>, <span class="dv">100000</span>, <span class="dv">500000</span>]</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> mcool <span class="kw">in</span> mcools:</span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Balancing </span><span class="sc">{</span>mcool<span class="sc">}</span><span class="ss">:"</span>)</span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> res <span class="kw">in</span> resolutions:</span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>        full_name <span class="op">=</span> <span class="ss">f"</span><span class="sc">{</span>mcool<span class="sc">}</span><span class="ss">::resolutions/</span><span class="sc">{</span>res<span class="sc">}</span><span class="ss">"</span></span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\t</span><span class="ss">resolution </span><span class="sc">{</span>res<span class="sc">}</span><span class="ss">..."</span>, end<span class="op">=</span><span class="st">" "</span>)</span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>        <span class="co"># First, just default values</span></span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>        <span class="co">#!cooler balance -p 32 {full_name}</span></span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a>        <span class="co"># With more filtering</span></span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a>        <span class="op">!</span>cooler balance <span class="op">-</span>p <span class="dv">32</span> <span class="op">-</span>c <span class="dv">20000000</span> <span class="op">--</span>cis<span class="op">-</span>only <span class="op">-</span>n cis_weights {full_name}</span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"--&gt; Done!"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Balancing ../steps/bwa/PE/cool/round_spermatid/round_spermatid.mcool:
    resolution 10000... INFO:cooler.cli.balance:Balancing "../steps/bwa/PE/cool/round_spermatid/round_spermatid.mcool::resolutions/10000"
INFO:cooler.balance:variance is 2079661.4027216053
INFO:cooler.balance:variance is 453553.06527791504
INFO:cooler.balance:variance is 122090.58632047936
INFO:cooler.balance:variance is 129190.84140421994
INFO:cooler.balance:variance is 48191.01921909741
INFO:cooler.balance:variance is 40764.260723976775
INFO:cooler.balance:variance is 18356.248248008043
INFO:cooler.balance:variance is 13507.74990364802
INFO:cooler.balance:variance is 6807.56897366601
INFO:cooler.balance:variance is 4602.949620233886
INFO:cooler.balance:variance is 2481.4615818257016
INFO:cooler.balance:variance is 1594.394282252964
INFO:cooler.balance:variance is 894.8193883950739
INFO:cooler.balance:variance is 557.6097513200839
INFO:cooler.balance:variance is 320.5395715985926
INFO:cooler.balance:variance is 196.1207825649443
INFO:cooler.balance:variance is 114.360455989492
INFO:cooler.balance:variance is 69.20983704916726
INFO:cooler.balance:variance is 40.701663195238446
INFO:cooler.balance:variance is 24.472032570532765
INFO:cooler.balance:variance is 14.464800443947336
INFO:cooler.balance:variance is 8.663243224655139
INFO:cooler.balance:variance is 5.136077178262542
INFO:cooler.balance:variance is 3.0689686912415834
INFO:cooler.balance:variance is 1.8227311404002953
INFO:cooler.balance:variance is 1.0876356471857458
INFO:cooler.balance:variance is 0.6466621996220416
INFO:cooler.balance:variance is 0.38555008168429644
INFO:cooler.balance:variance is 0.2293776232379284
INFO:cooler.balance:variance is 0.13669147465356649
INFO:cooler.balance:variance is 0.08135348118033474
INFO:cooler.balance:variance is 0.048466280530093726
INFO:cooler.balance:variance is 0.02885176157871944
INFO:cooler.balance:variance is 0.01718542784928682
INFO:cooler.balance:variance is 0.010231782639168413
INFO:cooler.balance:variance is 0.00609388612696222
INFO:cooler.balance:variance is 0.0036284405804380727
INFO:cooler.balance:variance is 0.0021609078863241
INFO:cooler.balance:variance is 0.0012867157377500968
INFO:cooler.balance:variance is 0.000766271896733645
INFO:cooler.balance:variance is 0.000456290649742504
INFO:cooler.balance:variance is 0.000271726724574162
INFO:cooler.balance:variance is 0.00016180739177129168
INFO:cooler.balance:variance is 9.635704748080421e-05
INFO:cooler.balance:variance is 5.737911611869424e-05
INFO:cooler.balance:variance is 3.4169262997483374e-05
INFO:cooler.balance:variance is 2.0347384009202282e-05
INFO:cooler.balance:variance is 1.211681096961127e-05
INFO:cooler.balance:variance is 7.2154405187102805e-06
--&gt; Done!
    resolution 50000... INFO:cooler.cli.balance:Balancing "../steps/bwa/PE/cool/round_spermatid/round_spermatid.mcool::resolutions/50000"
INFO:cooler.balance:variance is 23715051.788399633
INFO:cooler.balance:variance is 9587615.484821318
INFO:cooler.balance:variance is 2506901.69052377
INFO:cooler.balance:variance is 2490202.6786175943
INFO:cooler.balance:variance is 914697.0341084832
INFO:cooler.balance:variance is 722552.8777210371
INFO:cooler.balance:variance is 320506.1035810594
INFO:cooler.balance:variance is 220802.11194116462
INFO:cooler.balance:variance is 109122.83454572913
INFO:cooler.balance:variance is 69412.43424470197
INFO:cooler.balance:variance is 36495.31748751918
INFO:cooler.balance:variance is 22169.116790904776
INFO:cooler.balance:variance is 12075.855789381112
INFO:cooler.balance:variance is 7143.820928171591
INFO:cooler.balance:variance is 3970.795302176272
INFO:cooler.balance:variance is 2313.6665451446643
INFO:cooler.balance:variance is 1300.9444369499479
INFO:cooler.balance:variance is 751.4698428623118
INFO:cooler.balance:variance is 425.3340683171334
INFO:cooler.balance:variance is 244.47099190785718
INFO:cooler.balance:variance is 138.89257297811258
INFO:cooler.balance:variance is 79.60573861549905
INFO:cooler.balance:variance is 45.32401694919133
INFO:cooler.balance:variance is 25.935227148474347
INFO:cooler.balance:variance is 14.784490535695745
INFO:cooler.balance:variance is 8.45212880649952
INFO:cooler.balance:variance is 4.821546087745567
INFO:cooler.balance:variance is 2.7549678284476977
INFO:cooler.balance:variance is 1.572208954528203
INFO:cooler.balance:variance is 0.8980683651992448
INFO:cooler.balance:variance is 0.5126278599647546
INFO:cooler.balance:variance is 0.2927699442195732
INFO:cooler.balance:variance is 0.1671382553936957
INFO:cooler.balance:variance is 0.09544592915150751
INFO:cooler.balance:variance is 0.05449279496762292
INFO:cooler.balance:variance is 0.031116892889047974
INFO:cooler.balance:variance is 0.017766273378872162
INFO:cooler.balance:variance is 0.010144707907233161
INFO:cooler.balance:variance is 0.005792288604359692
INFO:cooler.balance:variance is 0.003307390256508526
INFO:cooler.balance:variance is 0.001888435259902268
INFO:cooler.balance:variance is 0.0010782831223879019
INFO:cooler.balance:variance is 0.0006156769558235292
INFO:cooler.balance:variance is 0.0003515450696321683
INFO:cooler.balance:variance is 0.00020072573771498175
INFO:cooler.balance:variance is 0.00011461189526818916
INFO:cooler.balance:variance is 6.5441442661249e-05
INFO:cooler.balance:variance is 3.736617561807919e-05
INFO:cooler.balance:variance is 2.1335482079086216e-05
INFO:cooler.balance:variance is 1.2182257166140531e-05
INFO:cooler.balance:variance is 6.955877703132534e-06
--&gt; Done!
    resolution 100000... INFO:cooler.cli.balance:Balancing "../steps/bwa/PE/cool/round_spermatid/round_spermatid.mcool::resolutions/100000"
INFO:cooler.balance:variance is 60497905.344961695
INFO:cooler.balance:variance is 34621992.49789496
INFO:cooler.balance:variance is 8927334.258448513
INFO:cooler.balance:variance is 8524481.211809449
INFO:cooler.balance:variance is 3105920.579568889
INFO:cooler.balance:variance is 2355673.1910981555
INFO:cooler.balance:variance is 1035322.3478044458
INFO:cooler.balance:variance is 686403.3433162335
INFO:cooler.balance:variance is 334999.2662678754
INFO:cooler.balance:variance is 205731.95946154604
INFO:cooler.balance:variance is 106455.17145908358
INFO:cooler.balance:variance is 62618.79328173612
INFO:cooler.balance:variance is 33475.397869860026
INFO:cooler.balance:variance is 19221.11122020354
INFO:cooler.balance:variance is 10463.66717701297
INFO:cooler.balance:variance is 5927.6050265636895
INFO:cooler.balance:variance is 3259.682332523258
INFO:cooler.balance:variance is 1832.7485266512251
INFO:cooler.balance:variance is 1013.5470152859881
INFO:cooler.balance:variance is 567.4790152929131
INFO:cooler.balance:variance is 314.8128293753999
INFO:cooler.balance:variance is 175.85032879214742
INFO:cooler.balance:variance is 97.72464125027004
INFO:cooler.balance:variance is 54.51666529996015
INFO:cooler.balance:variance is 30.32581544842157
INFO:cooler.balance:variance is 16.90529045646383
INFO:cooler.balance:variance is 9.408947303931592
INFO:cooler.balance:variance is 5.242949810832167
INFO:cooler.balance:variance is 2.9189394625225566
INFO:cooler.balance:variance is 1.6261552656417484
INFO:cooler.balance:variance is 0.9054914421707447
INFO:cooler.balance:variance is 0.5043904235709779
INFO:cooler.balance:variance is 0.2808858122865027
INFO:cooler.balance:variance is 0.15645230462328624
INFO:cooler.balance:variance is 0.08712997004397402
INFO:cooler.balance:variance is 0.04852916726662521
INFO:cooler.balance:variance is 0.027027199166828775
INFO:cooler.balance:variance is 0.015053133425002377
INFO:cooler.balance:variance is 0.008383630581428353
INFO:cooler.balance:variance is 0.004669310599853651
INFO:cooler.balance:variance is 0.002600530162654482
INFO:cooler.balance:variance is 0.0014483702930882083
INFO:cooler.balance:variance is 0.0008066607342970502
INFO:cooler.balance:variance is 0.0004492695719175469
INFO:cooler.balance:variance is 0.0002502185638808936
INFO:cooler.balance:variance is 0.00013935890028261635
INFO:cooler.balance:variance is 7.761540130791976e-05
INFO:cooler.balance:variance is 4.322774557922796e-05
INFO:cooler.balance:variance is 2.4075546807187287e-05
INFO:cooler.balance:variance is 1.3408819921770884e-05
INFO:cooler.balance:variance is 7.468000539129476e-06
--&gt; Done!
    resolution 500000... INFO:cooler.cli.balance:Balancing "../steps/bwa/PE/cool/round_spermatid/round_spermatid.mcool::resolutions/500000"
INFO:cooler.balance:variance is 752097810.1092529
INFO:cooler.balance:variance is 616091810.0783108
INFO:cooler.balance:variance is 151447603.4460279
INFO:cooler.balance:variance is 124595765.53655349
INFO:cooler.balance:variance is 43950655.218647
INFO:cooler.balance:variance is 28714108.330988254
INFO:cooler.balance:variance is 12110748.781926733
INFO:cooler.balance:variance is 6992744.074571425
INFO:cooler.balance:variance is 3229176.7088141437
INFO:cooler.balance:variance is 1749168.555860552
INFO:cooler.balance:variance is 845758.0640346411
INFO:cooler.balance:variance is 443426.8374600871
INFO:cooler.balance:variance is 219464.49580185706
INFO:cooler.balance:variance is 113172.02250178311
INFO:cooler.balance:variance is 56678.329755405095
INFO:cooler.balance:variance is 28982.57741465774
INFO:cooler.balance:variance is 14602.222314954442
INFO:cooler.balance:variance is 7435.071956263811
INFO:cooler.balance:variance is 3757.3993345815275
INFO:cooler.balance:variance is 1909.033816827551
INFO:cooler.balance:variance is 966.2402498413427
INFO:cooler.balance:variance is 490.38236203755054
INFO:cooler.balance:variance is 248.39661103297976
INFO:cooler.balance:variance is 125.99513821225486
INFO:cooler.balance:variance is 63.84643210507502
INFO:cooler.balance:variance is 32.375931739790126
INFO:cooler.balance:variance is 16.409385837722745
INFO:cooler.balance:variance is 8.319857340272586
INFO:cooler.balance:variance is 4.217257692727053
INFO:cooler.balance:variance is 2.13807145554647
INFO:cooler.balance:variance is 1.0838243058802508
INFO:cooler.balance:variance is 0.5494586387521827
INFO:cooler.balance:variance is 0.2785370925254058
INFO:cooler.balance:variance is 0.14120532292445875
INFO:cooler.balance:variance is 0.07158217001928041
INFO:cooler.balance:variance is 0.036288481008308573
INFO:cooler.balance:variance is 0.018396088952525008
INFO:cooler.balance:variance is 0.009325826932954376
INFO:cooler.balance:variance is 0.0047276524600972395
INFO:cooler.balance:variance is 0.0023966595167263165
INFO:cooler.balance:variance is 0.0012149692374305524
INFO:cooler.balance:variance is 0.0006159217556187857
INFO:cooler.balance:variance is 0.00031223736031748124
INFO:cooler.balance:variance is 0.0001582868580206506
INFO:cooler.balance:variance is 8.024248571301605e-05
INFO:cooler.balance:variance is 4.067843426110775e-05
INFO:cooler.balance:variance is 2.0621670399740046e-05
INFO:cooler.balance:variance is 1.045402721372966e-05
INFO:cooler.balance:variance is 5.29960242360679e-06
--&gt; Done!
Balancing ../steps/bwa/PE/cool/spermatogonia/spermatogonia.mcool:
    resolution 10000... INFO:cooler.cli.balance:Balancing "../steps/bwa/PE/cool/spermatogonia/spermatogonia.mcool::resolutions/10000"
INFO:cooler.balance:variance is 1308046.8986688335
INFO:cooler.balance:variance is 316673.30240057554
INFO:cooler.balance:variance is 82785.5248728855
INFO:cooler.balance:variance is 78768.27969422366
INFO:cooler.balance:variance is 28056.132700292994
INFO:cooler.balance:variance is 22399.17426980566
INFO:cooler.balance:variance is 9672.539897303917
INFO:cooler.balance:variance is 6839.308546255572
INFO:cooler.balance:variance is 3326.2125837620065
INFO:cooler.balance:variance is 2177.657946649484
INFO:cooler.balance:variance is 1138.6127739458539
INFO:cooler.balance:variance is 711.015655051492
INFO:cooler.balance:variance is 388.3569673740054
INFO:cooler.balance:variance is 235.6899027449793
INFO:cooler.balance:variance is 132.14299968952943
INFO:cooler.balance:variance is 78.84358322685917
INFO:cooler.balance:variance is 44.89793252767011
INFO:cooler.balance:variance is 26.520683012624694
INFO:cooler.balance:variance is 15.242341119779086
INFO:cooler.balance:variance is 8.950563839507556
INFO:cooler.balance:variance is 5.172351848354103
INFO:cooler.balance:variance is 3.026869840291717
INFO:cooler.balance:variance is 1.7548185957552003
INFO:cooler.balance:variance is 1.0248753551646965
INFO:cooler.balance:variance is 0.5953013737100321
INFO:cooler.balance:variance is 0.3472757240874453
INFO:cooler.balance:variance is 0.20194309112705913
INFO:cooler.balance:variance is 0.11772745169053997
INFO:cooler.balance:variance is 0.06850483459787624
INFO:cooler.balance:variance is 0.039921255509448354
INFO:cooler.balance:variance is 0.023239063859600777
INFO:cooler.balance:variance is 0.013539630088010603
INFO:cooler.balance:variance is 0.007883563025672202
INFO:cooler.balance:variance is 0.004592579923071141
INFO:cooler.balance:variance is 0.002674438406769844
INFO:cooler.balance:variance is 0.0015578877775208686
INFO:cooler.balance:variance is 0.0009072935216964874
INFO:cooler.balance:variance is 0.0005284867670857317
INFO:cooler.balance:variance is 0.00030779890628106
INFO:cooler.balance:variance is 0.00017928489663283752
INFO:cooler.balance:variance is 0.0001044213719188654
INFO:cooler.balance:variance is 6.0821992899236746e-05
INFO:cooler.balance:variance is 3.542533486248425e-05
INFO:cooler.balance:variance is 2.063394410509228e-05
INFO:cooler.balance:variance is 1.2018220012279698e-05
INFO:cooler.balance:variance is 7.000141523505563e-06
--&gt; Done!
    resolution 50000... INFO:cooler.cli.balance:Balancing "../steps/bwa/PE/cool/spermatogonia/spermatogonia.mcool::resolutions/50000"
INFO:cooler.balance:variance is 12664970.643345702
INFO:cooler.balance:variance is 6119316.049040368
INFO:cooler.balance:variance is 1524725.0734942225
INFO:cooler.balance:variance is 1317383.720278999
INFO:cooler.balance:variance is 457749.7138024568
INFO:cooler.balance:variance is 330139.33925419877
INFO:cooler.balance:variance is 139313.30823029307
INFO:cooler.balance:variance is 89388.48672410581
INFO:cooler.balance:variance is 42212.031890871396
INFO:cooler.balance:variance is 25265.071622190768
INFO:cooler.balance:variance is 12722.676102383617
INFO:cooler.balance:variance is 7315.885090671071
INFO:cooler.balance:variance is 3820.22527127937
INFO:cooler.balance:variance is 2147.590745464749
INFO:cooler.balance:variance is 1144.4874115113412
INFO:cooler.balance:variance is 635.3276839396618
INFO:cooler.balance:variance is 342.44141946240575
INFO:cooler.balance:variance is 188.77786890941866
INFO:cooler.balance:variance is 102.39536334082007
INFO:cooler.balance:variance is 56.23277208013894
INFO:cooler.balance:variance is 30.608372241848254
INFO:cooler.balance:variance is 16.774425152298623
INFO:cooler.balance:variance is 9.148347629937652
INFO:cooler.balance:variance is 5.007964580634879
INFO:cooler.balance:variance is 2.7341659296457403
INFO:cooler.balance:variance is 1.495821063440017
INFO:cooler.balance:variance is 0.8171533830298441
INFO:cooler.balance:variance is 0.4469064129157515
INFO:cooler.balance:variance is 0.24422230597840425
INFO:cooler.balance:variance is 0.13354341430001962
INFO:cooler.balance:variance is 0.07299143420977716
INFO:cooler.balance:variance is 0.03990879849603416
INFO:cooler.balance:variance is 0.021815396811108154
INFO:cooler.balance:variance is 0.011927199439084947
INFO:cooler.balance:variance is 0.0065201584267363475
INFO:cooler.balance:variance is 0.003564694008018788
INFO:cooler.balance:variance is 0.0019487496970339556
INFO:cooler.balance:variance is 0.0010654039205419844
INFO:cooler.balance:variance is 0.000582446510079603
INFO:cooler.balance:variance is 0.0003184279912398374
INFO:cooler.balance:variance is 0.00017408344814996056
INFO:cooler.balance:variance is 9.517242089222104e-05
INFO:cooler.balance:variance is 5.2030727582799864e-05
INFO:cooler.balance:variance is 2.8445446398979426e-05
INFO:cooler.balance:variance is 1.5551166746672267e-05
INFO:cooler.balance:variance is 8.501889337010656e-06
--&gt; Done!
    resolution 100000... INFO:cooler.cli.balance:Balancing "../steps/bwa/PE/cool/spermatogonia/spermatogonia.mcool::resolutions/100000"
INFO:cooler.balance:variance is 35417979.414851815
INFO:cooler.balance:variance is 20978398.36316186
INFO:cooler.balance:variance is 5082067.645745064
INFO:cooler.balance:variance is 4117030.0830009854
INFO:cooler.balance:variance is 1407664.1289004453
INFO:cooler.balance:variance is 955052.5352853398
INFO:cooler.balance:variance is 396409.00953044463
INFO:cooler.balance:variance is 240460.57128696173
INFO:cooler.balance:variance is 111211.88522093414
INFO:cooler.balance:variance is 63235.17231681437
INFO:cooler.balance:variance is 31035.646033697943
INFO:cooler.balance:variance is 17022.615961850992
INFO:cooler.balance:variance is 8627.91387797452
INFO:cooler.balance:variance is 4640.384336348682
INFO:cooler.balance:variance is 2392.9907235534715
INFO:cooler.balance:variance is 1273.5562197517847
INFO:cooler.balance:variance is 662.850268163825
INFO:cooler.balance:variance is 350.8027403705651
INFO:cooler.balance:variance is 183.48285521762452
INFO:cooler.balance:variance is 96.81891150271014
INFO:cooler.balance:variance is 50.77238920315966
INFO:cooler.balance:variance is 26.74965195772774
INFO:cooler.balance:variance is 14.047152776124742
INFO:cooler.balance:variance is 7.394789114260982
INFO:cooler.balance:variance is 3.886118782168093
INFO:cooler.balance:variance is 2.0448864673765055
INFO:cooler.balance:variance is 1.0750521479351027
INFO:cooler.balance:variance is 0.5655703446353354
INFO:cooler.balance:variance is 0.29739752741042164
INFO:cooler.balance:variance is 0.15643882466925124
INFO:cooler.balance:variance is 0.08227035258617608
INFO:cooler.balance:variance is 0.0432737640735207
INFO:cooler.balance:variance is 0.022758786269366824
INFO:cooler.balance:variance is 0.011970630555550617
INFO:cooler.balance:variance is 0.0062958606419361825
INFO:cooler.balance:variance is 0.0033114342192394285
INFO:cooler.balance:variance is 0.0017416524731905152
INFO:cooler.balance:variance is 0.0009160495712062265
INFO:cooler.balance:variance is 0.00048180160411278857
INFO:cooler.balance:variance is 0.00025341008612315963
INFO:cooler.balance:variance is 0.00013328315032802725
INFO:cooler.balance:variance is 7.010193004233756e-05
INFO:cooler.balance:variance is 3.687079037920824e-05
INFO:cooler.balance:variance is 1.9392629540567188e-05
INFO:cooler.balance:variance is 1.0199755824066701e-05
INFO:cooler.balance:variance is 5.364679614387041e-06
--&gt; Done!
    resolution 500000... INFO:cooler.cli.balance:Balancing "../steps/bwa/PE/cool/spermatogonia/spermatogonia.mcool::resolutions/500000"
INFO:cooler.balance:variance is 502908794.55910784
INFO:cooler.balance:variance is 302885012.9395799
INFO:cooler.balance:variance is 64769841.08097244
INFO:cooler.balance:variance is 39738788.34995444
INFO:cooler.balance:variance is 12477710.26016655
INFO:cooler.balance:variance is 6488039.9554140605
INFO:cooler.balance:variance is 2442496.433927255
INFO:cooler.balance:variance is 1161186.926207534
INFO:cooler.balance:variance is 475560.9205520366
INFO:cooler.balance:variance is 216424.55078923996
INFO:cooler.balance:variance is 92084.54215766255
INFO:cooler.balance:variance is 41066.4536084368
INFO:cooler.balance:variance is 17773.39503533346
INFO:cooler.balance:variance is 7854.067579895398
INFO:cooler.balance:variance is 3424.997246168138
INFO:cooler.balance:variance is 1507.3502280588384
INFO:cooler.balance:variance is 659.5241944600145
INFO:cooler.balance:variance is 289.736727452119
INFO:cooler.balance:variance is 126.95829088056419
INFO:cooler.balance:variance is 55.730295610759974
INFO:cooler.balance:variance is 24.436117555785817
INFO:cooler.balance:variance is 10.722919007547057
INFO:cooler.balance:variance is 4.703047349242043
INFO:cooler.balance:variance is 2.0634562680269304
INFO:cooler.balance:variance is 0.9051431946700886
INFO:cooler.balance:variance is 0.39710483238150596
INFO:cooler.balance:variance is 0.17420154340792685
INFO:cooler.balance:variance is 0.07642366207589198
INFO:cooler.balance:variance is 0.03352631055910466
INFO:cooler.balance:variance is 0.014708096942917404
INFO:cooler.balance:variance is 0.0064523718632841405
INFO:cooler.balance:variance is 0.0028306615682161147
INFO:cooler.balance:variance is 0.0012418041540445696
INFO:cooler.balance:variance is 0.000544779517517622
INFO:cooler.balance:variance is 0.00023899397775345836
INFO:cooler.balance:variance is 0.00010484657661802518
INFO:cooler.balance:variance is 4.599608985809808e-05
INFO:cooler.balance:variance is 2.0178462242009167e-05
INFO:cooler.balance:variance is 8.852275971975331e-06
--&gt; Done!
Balancing ../steps/bwa/PE/cool/sperm/sperm.mcool:
    resolution 10000... INFO:cooler.cli.balance:Balancing "../steps/bwa/PE/cool/sperm/sperm.mcool::resolutions/10000"
INFO:cooler.balance:variance is 741858.3253318394
INFO:cooler.balance:variance is 125654.14409759178
INFO:cooler.balance:variance is 29707.895428666514
INFO:cooler.balance:variance is 26308.618812100405
INFO:cooler.balance:variance is 9126.995700064954
INFO:cooler.balance:variance is 6361.814778481835
INFO:cooler.balance:variance is 2686.284798559173
INFO:cooler.balance:variance is 1636.497024370329
INFO:cooler.balance:variance is 765.6520583213421
INFO:cooler.balance:variance is 434.1217240131892
INFO:cooler.balance:variance is 214.3041980711702
INFO:cooler.balance:variance is 116.99534062646569
INFO:cooler.balance:variance is 59.399390573895516
INFO:cooler.balance:variance is 31.789714525994775
INFO:cooler.balance:variance is 16.378852155670593
INFO:cooler.balance:variance is 8.674909436374353
INFO:cooler.balance:variance is 4.504070684761752
INFO:cooler.balance:variance is 2.372576487749128
INFO:cooler.balance:variance is 1.236838088007083
INFO:cooler.balance:variance is 0.6496674692760012
INFO:cooler.balance:variance is 0.3393930711935677
INFO:cooler.balance:variance is 0.17800637363404917
INFO:cooler.balance:variance is 0.09309579192340094
INFO:cooler.balance:variance is 0.04878944758246891
INFO:cooler.balance:variance is 0.025531369470789055
INFO:cooler.balance:variance is 0.013375024478619869
INFO:cooler.balance:variance is 0.00700126236976874
INFO:cooler.balance:variance is 0.0036669555191961374
INFO:cooler.balance:variance is 0.0019198081963679966
INFO:cooler.balance:variance is 0.0010054021765131787
INFO:cooler.balance:variance is 0.0005264162010217427
INFO:cooler.balance:variance is 0.0002756682514123214
INFO:cooler.balance:variance is 0.00014434306022155716
INFO:cooler.balance:variance is 7.558590204617383e-05
INFO:cooler.balance:variance is 3.957860234444607e-05
INFO:cooler.balance:variance is 2.0725207108927694e-05
INFO:cooler.balance:variance is 1.0852358744258687e-05
INFO:cooler.balance:variance is 5.682759487137995e-06
--&gt; Done!
    resolution 50000... INFO:cooler.cli.balance:Balancing "../steps/bwa/PE/cool/sperm/sperm.mcool::resolutions/50000"
INFO:cooler.balance:variance is 11226892.111312442
INFO:cooler.balance:variance is 2529665.7059934894
INFO:cooler.balance:variance is 581999.9893026546
INFO:cooler.balance:variance is 472337.4957996669
INFO:cooler.balance:variance is 160916.8113716224
INFO:cooler.balance:variance is 102738.72916850327
INFO:cooler.balance:variance is 42303.55508531462
INFO:cooler.balance:variance is 23790.49695299119
INFO:cooler.balance:variance is 10754.782610265629
INFO:cooler.balance:variance is 5674.121620221377
INFO:cooler.balance:variance is 2686.275368890727
INFO:cooler.balance:variance is 1372.901968646645
INFO:cooler.balance:variance is 664.9952951360676
INFO:cooler.balance:variance is 334.54863270733176
INFO:cooler.balance:variance is 163.89168521364684
INFO:cooler.balance:variance is 81.81033452545914
INFO:cooler.balance:variance is 40.30371761726792
INFO:cooler.balance:variance is 20.041084396155842
INFO:cooler.balance:variance is 9.900772754225764
INFO:cooler.balance:variance is 4.913810025834852
INFO:cooler.balance:variance is 2.43090567547462
INFO:cooler.balance:variance is 1.2053406944852152
INFO:cooler.balance:variance is 0.5967042824060315
INFO:cooler.balance:variance is 0.2957333634754577
INFO:cooler.balance:variance is 0.1464532671647815
INFO:cooler.balance:variance is 0.07256743562122547
INFO:cooler.balance:variance is 0.035943073386419166
INFO:cooler.balance:variance is 0.017807778013246334
INFO:cooler.balance:variance is 0.008821056926579725
INFO:cooler.balance:variance is 0.004370102903910446
INFO:cooler.balance:variance is 0.0021648181404164153
INFO:cooler.balance:variance is 0.0010724599416357206
INFO:cooler.balance:variance is 0.0005312762293403381
INFO:cooler.balance:variance is 0.00026319314873834233
INFO:cooler.balance:variance is 0.00013038233492911421
INFO:cooler.balance:variance is 6.459075409998573e-05
INFO:cooler.balance:variance is 3.199757264582162e-05
INFO:cooler.balance:variance is 1.5851392698324252e-05
INFO:cooler.balance:variance is 7.85263491025088e-06
--&gt; Done!
    resolution 100000... INFO:cooler.cli.balance:Balancing "../steps/bwa/PE/cool/sperm/sperm.mcool::resolutions/100000"
INFO:cooler.balance:variance is 26978149.012366746
INFO:cooler.balance:variance is 8900156.94370849
INFO:cooler.balance:variance is 2016627.4174504378
INFO:cooler.balance:variance is 1551364.6412996298
INFO:cooler.balance:variance is 522185.826549854
INFO:cooler.balance:variance is 316195.7758177847
INFO:cooler.balance:variance is 127946.05016421733
INFO:cooler.balance:variance is 68595.88279107968
INFO:cooler.balance:variance is 30298.426967140596
INFO:cooler.balance:variance is 15312.260407260588
INFO:cooler.balance:variance is 7052.450566192081
INFO:cooler.balance:variance is 3464.4487069581824
INFO:cooler.balance:variance is 1627.9182174001055
INFO:cooler.balance:variance is 788.9090050328566
INFO:cooler.balance:variance is 374.27557326339814
INFO:cooler.balance:variance is 180.20523376289722
INFO:cooler.balance:variance is 85.88745006028755
INFO:cooler.balance:variance is 41.225053856778686
INFO:cooler.balance:variance is 19.691676811548774
INFO:cooler.balance:variance is 9.437868113551502
INFO:cooler.balance:variance is 4.512909280544654
INFO:cooler.balance:variance is 2.1614410205498022
INFO:cooler.balance:variance is 1.0340661666257085
INFO:cooler.balance:variance is 0.49509750833033705
INFO:cooler.balance:variance is 0.23692072309544868
INFO:cooler.balance:variance is 0.11341675591170837
INFO:cooler.balance:variance is 0.05428021834383655
INFO:cooler.balance:variance is 0.02598266272898889
INFO:cooler.balance:variance is 0.012435789969899271
INFO:cooler.balance:variance is 0.005952514338789774
INFO:cooler.balance:variance is 0.0028490666047004906
INFO:cooler.balance:variance is 0.0013637122309606209
INFO:cooler.balance:variance is 0.000652726177053505
INFO:cooler.balance:variance is 0.000312426598847558
INFO:cooler.balance:variance is 0.00014954068420420272
INFO:cooler.balance:variance is 7.157723621330367e-05
INFO:cooler.balance:variance is 3.4260038024799415e-05
INFO:cooler.balance:variance is 1.639844988840405e-05
INFO:cooler.balance:variance is 7.849039453995556e-06
--&gt; Done!
    resolution 500000... INFO:cooler.cli.balance:Balancing "../steps/bwa/PE/cool/sperm/sperm.mcool::resolutions/500000"
INFO:cooler.balance:variance is 295807137.88910925
INFO:cooler.balance:variance is 147065535.97677138
INFO:cooler.balance:variance is 31558040.255049814
INFO:cooler.balance:variance is 19890269.304988354
INFO:cooler.balance:variance is 6349983.4655896835
INFO:cooler.balance:variance is 3187143.129140095
INFO:cooler.balance:variance is 1195162.4464936242
INFO:cooler.balance:variance is 542271.0324039235
INFO:cooler.balance:variance is 217534.60078393028
INFO:cooler.balance:variance is 94560.20726960694
INFO:cooler.balance:variance is 39026.631537067966
INFO:cooler.balance:variance is 16661.191138130707
INFO:cooler.balance:variance is 6959.364478323574
INFO:cooler.balance:variance is 2948.668468787632
INFO:cooler.balance:variance is 1237.9229952184448
INFO:cooler.balance:variance is 522.8449040589231
INFO:cooler.balance:variance is 219.97583682881836
INFO:cooler.balance:variance is 92.78529788050602
INFO:cooler.balance:variance is 39.07316829430628
INFO:cooler.balance:variance is 16.471897403186045
INFO:cooler.balance:variance is 6.939254403667243
INFO:cooler.balance:variance is 2.924683737405765
INFO:cooler.balance:variance is 1.232312992525181
INFO:cooler.balance:variance is 0.5193338143610416
INFO:cooler.balance:variance is 0.21883678787107338
INFO:cooler.balance:variance is 0.09222093672936361
INFO:cooler.balance:variance is 0.038861282155901095
INFO:cooler.balance:variance is 0.016376453389006302
INFO:cooler.balance:variance is 0.006901025442559217
INFO:cooler.balance:variance is 0.0029081305336677815
INFO:cooler.balance:variance is 0.0012254920082762333
INFO:cooler.balance:variance is 0.0005164281160424181
INFO:cooler.balance:variance is 0.00021762449681160079
INFO:cooler.balance:variance is 9.170793393983562e-05
INFO:cooler.balance:variance is 3.864607967040314e-05
INFO:cooler.balance:variance is 1.628562856899402e-05
INFO:cooler.balance:variance is 6.8628321656597144e-06
--&gt; Done!
Balancing ../steps/bwa/PE/cool/fibroblast/fibroblast.mcool:
    resolution 10000... INFO:cooler.cli.balance:Balancing "../steps/bwa/PE/cool/fibroblast/fibroblast.mcool::resolutions/10000"
INFO:cooler.balance:variance is 2243118.7382447985
INFO:cooler.balance:variance is 991747.1270330226
INFO:cooler.balance:variance is 259316.25655305735
INFO:cooler.balance:variance is 297362.15539104934
INFO:cooler.balance:variance is 102962.78773562737
INFO:cooler.balance:variance is 102701.15505492924
INFO:cooler.balance:variance is 43666.08528448372
INFO:cooler.balance:variance is 38199.37739623126
INFO:cooler.balance:variance is 18730.73347933698
INFO:cooler.balance:variance is 14868.41888183261
INFO:cooler.balance:variance is 8024.57046327179
INFO:cooler.balance:variance is 5955.281527207283
INFO:cooler.balance:variance is 3425.2504241500937
INFO:cooler.balance:variance is 2429.1699095430486
INFO:cooler.balance:variance is 1456.9955185049366
INFO:cooler.balance:variance is 1002.5369516783766
INFO:cooler.balance:variance is 618.1013880145684
INFO:cooler.balance:variance is 416.901837877862
INFO:cooler.balance:variance is 261.7159985714534
INFO:cooler.balance:variance is 174.22418275561708
INFO:cooler.balance:variance is 110.67098057074581
INFO:cooler.balance:variance is 73.0437133984367
INFO:cooler.balance:variance is 46.758613700403615
INFO:cooler.balance:variance is 30.688536728391497
INFO:cooler.balance:variance is 19.744512308732116
INFO:cooler.balance:variance is 12.911454912353367
INFO:cooler.balance:variance is 8.33445392717434
INFO:cooler.balance:variance is 5.437195953949247
INFO:cooler.balance:variance is 3.5173239231320452
INFO:cooler.balance:variance is 2.2910847014813087
INFO:cooler.balance:variance is 1.4841911208997762
INFO:cooler.balance:variance is 0.9657953991007048
INFO:cooler.balance:variance is 0.6262294743793554
INFO:cooler.balance:variance is 0.4072381318491516
INFO:cooler.balance:variance is 0.2642155208627828
INFO:cooler.balance:variance is 0.1717482186714724
INFO:cooler.balance:variance is 0.11147395417236386
INFO:cooler.balance:variance is 0.07244204217556938
INFO:cooler.balance:variance is 0.047030994356778696
INFO:cooler.balance:variance is 0.030558107589759093
INFO:cooler.balance:variance is 0.019842376614872696
INFO:cooler.balance:variance is 0.012891038536355119
INFO:cooler.balance:variance is 0.00837150738892597
INFO:cooler.balance:variance is 0.005438350033562759
INFO:cooler.balance:variance is 0.0035319529539791115
INFO:cooler.balance:variance is 0.0022943456242462295
INFO:cooler.balance:variance is 0.001490142303741579
INFO:cooler.balance:variance is 0.0009679641714111494
INFO:cooler.balance:variance is 0.0006286981869746439
INFO:cooler.balance:variance is 0.000408381389160917
INFO:cooler.balance:variance is 0.00026525171318165196
INFO:cooler.balance:variance is 0.00017229673790517122
INFO:cooler.balance:variance is 0.00011191171295225644
INFO:cooler.balance:variance is 7.269279771398023e-05
INFO:cooler.balance:variance is 4.721652868051391e-05
INFO:cooler.balance:variance is 3.066959590871441e-05
INFO:cooler.balance:variance is 1.9921110622070587e-05
INFO:cooler.balance:variance is 1.2939766583058893e-05
INFO:cooler.balance:variance is 8.404925158333893e-06
--&gt; Done!
    resolution 50000... INFO:cooler.cli.balance:Balancing "../steps/bwa/PE/cool/fibroblast/fibroblast.mcool::resolutions/50000"
INFO:cooler.balance:variance is 33098650.12736815
INFO:cooler.balance:variance is 18861473.70590618
INFO:cooler.balance:variance is 4706882.272186386
INFO:cooler.balance:variance is 5056251.080239619
INFO:cooler.balance:variance is 1732226.31225507
INFO:cooler.balance:variance is 1591524.2524037405
INFO:cooler.balance:variance is 675725.862301692
INFO:cooler.balance:variance is 542970.3032262143
INFO:cooler.balance:variance is 265117.16505824047
INFO:cooler.balance:variance is 194129.30281686314
INFO:cooler.balance:variance is 103611.1407531792
INFO:cooler.balance:variance is 71384.30110114903
INFO:cooler.balance:variance is 40299.12513092778
INFO:cooler.balance:variance is 26701.007148300923
INFO:cooler.balance:variance is 15614.365401067733
INFO:cooler.balance:variance is 10092.640847951732
INFO:cooler.balance:variance is 6033.701907998682
INFO:cooler.balance:variance is 3839.7159030501757
INFO:cooler.balance:variance is 2327.3842697590917
INFO:cooler.balance:variance is 1466.720347857387
INFO:cooler.balance:variance is 896.7198467733905
INFO:cooler.balance:variance is 561.685522837574
INFO:cooler.balance:variance is 345.25271449991817
INFO:cooler.balance:variance is 215.4415485198332
INFO:cooler.balance:variance is 132.8709251792976
INFO:cooler.balance:variance is 82.7183708577876
INFO:cooler.balance:variance is 51.12243400736155
INFO:cooler.balance:variance is 31.779837309483067
INFO:cooler.balance:variance is 19.66657886496113
INFO:cooler.balance:variance is 12.214578902528666
INFO:cooler.balance:variance is 7.56502432930727
INFO:cooler.balance:variance is 4.695904455075625
INFO:cooler.balance:variance is 2.90986582762248
INFO:cooler.balance:variance is 1.8056509347302994
INFO:cooler.balance:variance is 1.1192484958544657
INFO:cooler.balance:variance is 0.6943788999825237
INFO:cooler.balance:variance is 0.4305033045473406
INFO:cooler.balance:variance is 0.26704899743299676
INFO:cooler.balance:variance is 0.1655868389925652
INFO:cooler.balance:variance is 0.10270851830622812
INFO:cooler.balance:variance is 0.06369069014162039
INFO:cooler.balance:variance is 0.03950354929276463
INFO:cooler.balance:variance is 0.024497824865218596
INFO:cooler.balance:variance is 0.015194113166564159
INFO:cooler.balance:variance is 0.009422815595873709
INFO:cooler.balance:variance is 0.005844147404713019
INFO:cooler.balance:variance is 0.0036243938805720615
INFO:cooler.balance:variance is 0.0022478716546752184
INFO:cooler.balance:variance is 0.001394092034278237
INFO:cooler.balance:variance is 0.0008646195268219724
INFO:cooler.balance:variance is 0.0005362269796344828
INFO:cooler.balance:variance is 0.00033256827258229814
INFO:cooler.balance:variance is 0.00020625611589813352
INFO:cooler.balance:variance is 0.00012791992404197544
INFO:cooler.balance:variance is 7.933517547413656e-05
INFO:cooler.balance:variance is 4.920358259911646e-05
INFO:cooler.balance:variance is 3.051584221673408e-05
INFO:cooler.balance:variance is 1.8925880658521028e-05
INFO:cooler.balance:variance is 1.1737765481087798e-05
INFO:cooler.balance:variance is 7.279743592053885e-06
--&gt; Done!
    resolution 100000... INFO:cooler.cli.balance:Balancing "../steps/bwa/PE/cool/fibroblast/fibroblast.mcool::resolutions/100000"
INFO:cooler.balance:variance is 95762729.34928331
INFO:cooler.balance:variance is 61076831.455153994
INFO:cooler.balance:variance is 14830747.465030745
INFO:cooler.balance:variance is 15122278.342628365
INFO:cooler.balance:variance is 5157013.049074703
INFO:cooler.balance:variance is 4445643.853411706
INFO:cooler.balance:variance is 1884098.6260467838
INFO:cooler.balance:variance is 1420554.9772676432
INFO:cooler.balance:variance is 689248.1740502883
INFO:cooler.balance:variance is 475613.5341070679
INFO:cooler.balance:variance is 250664.28814100937
INFO:cooler.balance:variance is 163585.53459444997
INFO:cooler.balance:variance is 90652.19649284013
INFO:cooler.balance:variance is 57159.24871875445
INFO:cooler.balance:variance is 32650.79780667571
INFO:cooler.balance:variance is 20159.773422835788
INFO:cooler.balance:variance is 11728.384556004832
INFO:cooler.balance:variance is 7150.028295973408
INFO:cooler.balance:variance is 4205.774093516145
INFO:cooler.balance:variance is 2544.4066869126173
INFO:cooler.balance:variance is 1506.6253771907323
INFO:cooler.balance:variance is 907.2904975519776
INFO:cooler.balance:variance is 539.3847808311697
INFO:cooler.balance:variance is 323.92342692741084
INFO:cooler.balance:variance is 193.03597409149418
INFO:cooler.balance:variance is 115.73553817267903
INFO:cooler.balance:variance is 69.07028429977203
INFO:cooler.balance:variance is 41.37077525235102
INFO:cooler.balance:variance is 24.711393035696165
INFO:cooler.balance:variance is 14.79266761484528
INFO:cooler.balance:variance is 8.84054386078467
INFO:cooler.balance:variance is 5.2902756683441305
INFO:cooler.balance:variance is 3.1626366384383524
INFO:cooler.balance:variance is 1.8921698104407556
INFO:cooler.balance:variance is 1.1313972518478432
INFO:cooler.balance:variance is 0.6768211393030208
INFO:cooler.balance:variance is 0.40474364883419806
INFO:cooler.balance:variance is 0.24210760683314492
INFO:cooler.balance:variance is 0.14479240485417574
INFO:cooler.balance:variance is 0.0866077045681562
INFO:cooler.balance:variance is 0.05179800437623845
INFO:cooler.balance:variance is 0.03098229814859465
INFO:cooler.balance:variance is 0.01853027839478892
INFO:cooler.balance:variance is 0.011083496394368623
INFO:cooler.balance:variance is 0.006629067422611634
INFO:cooler.balance:variance is 0.0039650079525874425
INFO:cooler.balance:variance is 0.002371506422022685
INFO:cooler.balance:variance is 0.001418450498367986
INFO:cooler.balance:variance is 0.000848393484188623
INFO:cooler.balance:variance is 0.0005074418598329213
INFO:cooler.balance:variance is 0.0003035087933700132
INFO:cooler.balance:variance is 0.00018153476061006293
INFO:cooler.balance:variance is 0.00010857903473919804
INFO:cooler.balance:variance is 6.4943292513953e-05
INFO:cooler.balance:variance is 3.8843759848746485e-05
INFO:cooler.balance:variance is 2.3233226606969482e-05
INFO:cooler.balance:variance is 1.3896230223045618e-05
INFO:cooler.balance:variance is 8.311612290465695e-06
--&gt; Done!
    resolution 500000... INFO:cooler.cli.balance:Balancing "../steps/bwa/PE/cool/fibroblast/fibroblast.mcool::resolutions/500000"
INFO:cooler.balance:variance is 1263029871.4028523
INFO:cooler.balance:variance is 713041615.7020518
INFO:cooler.balance:variance is 155459168.18479806
INFO:cooler.balance:variance is 120278348.19536278
INFO:cooler.balance:variance is 39400048.456381015
INFO:cooler.balance:variance is 25341702.48208014
INFO:cooler.balance:variance is 10236908.14941692
INFO:cooler.balance:variance is 5861237.265242753
INFO:cooler.balance:variance is 2635567.014828291
INFO:cooler.balance:variance is 1415968.571919777
INFO:cooler.balance:variance is 672291.4270227149
INFO:cooler.balance:variance is 349341.9293198957
INFO:cooler.balance:variance is 170491.7031776903
INFO:cooler.balance:variance is 87087.79818905548
INFO:cooler.balance:variance is 43097.219213253054
INFO:cooler.balance:variance is 21823.669277186564
INFO:cooler.balance:variance is 10876.110953072413
INFO:cooler.balance:variance is 5483.416832221197
INFO:cooler.balance:variance is 2742.4813595640435
INFO:cooler.balance:variance is 1379.6551287130035
INFO:cooler.balance:variance is 691.2707283828129
INFO:cooler.balance:variance is 347.3781850937816
INFO:cooler.balance:variance is 174.21289833852526
INFO:cooler.balance:variance is 87.49870459095322
INFO:cooler.balance:variance is 43.902003566759035
INFO:cooler.balance:variance is 22.044070235766075
INFO:cooler.balance:variance is 11.063185697240737
INFO:cooler.balance:variance is 5.5543434789971435
INFO:cooler.balance:variance is 2.787893318470167
INFO:cooler.balance:variance is 1.3995960250779331
INFO:cooler.balance:variance is 0.7025464959204487
INFO:cooler.balance:variance is 0.3526871907124563
INFO:cooler.balance:variance is 0.17704249565441085
INFO:cooler.balance:variance is 0.08887649483283891
INFO:cooler.balance:variance is 0.04461523276604731
INFO:cooler.balance:variance is 0.02239703226666938
INFO:cooler.balance:variance is 0.011243237652625441
INFO:cooler.balance:variance is 0.005644143306841245
INFO:cooler.balance:variance is 0.0028333589950725762
INFO:cooler.balance:variance is 0.001422355546012453
INFO:cooler.balance:variance is 0.0007140247985879804
INFO:cooler.balance:variance is 0.00035844292003720646
INFO:cooler.balance:variance is 0.00017993931562735287
INFO:cooler.balance:variance is 9.033020102705829e-05
INFO:cooler.balance:variance is 4.5346065396008476e-05
INFO:cooler.balance:variance is 2.276390232581904e-05
INFO:cooler.balance:variance is 1.1427565283906855e-05
INFO:cooler.balance:variance is 5.736684616241224e-06
--&gt; Done!
Balancing ../steps/bwa/PE/cool/pachytene_spermatocyte/pachytene_spermatocyte.mcool:
    resolution 10000... INFO:cooler.cli.balance:Balancing "../steps/bwa/PE/cool/pachytene_spermatocyte/pachytene_spermatocyte.mcool::resolutions/10000"
INFO:cooler.balance:variance is 2259035.2579729445
INFO:cooler.balance:variance is 862660.4307984081
INFO:cooler.balance:variance is 250327.04099481337
INFO:cooler.balance:variance is 406382.6280304705
INFO:cooler.balance:variance is 150013.8890621128
INFO:cooler.balance:variance is 201040.05651255383
INFO:cooler.balance:variance is 88584.40036468355
INFO:cooler.balance:variance is 101622.17951879876
INFO:cooler.balance:variance is 51134.30170765518
INFO:cooler.balance:variance is 52164.477419668634
INFO:cooler.balance:variance is 28977.910886383866
INFO:cooler.balance:variance is 27097.454813591165
INFO:cooler.balance:variance is 16199.113698563177
INFO:cooler.balance:variance is 14206.413809249761
INFO:cooler.balance:variance is 8966.201299516742
INFO:cooler.balance:variance is 7500.764350758503
INFO:cooler.balance:variance is 4927.310220243256
INFO:cooler.balance:variance is 3981.5157030759005
INFO:cooler.balance:variance is 2693.736241758525
INFO:cooler.balance:variance is 2121.9506854016436
INFO:cooler.balance:variance is 1467.110150317689
INFO:cooler.balance:variance is 1134.2892880066738
INFO:cooler.balance:variance is 796.8546510267557
INFO:cooler.balance:variance is 607.6864497622404
INFO:cooler.balance:variance is 431.94361970524625
INFO:cooler.balance:variance is 326.1005562253113
INFO:cooler.balance:variance is 233.79801950386639
INFO:cooler.balance:variance is 175.20750668366983
INFO:cooler.balance:variance is 126.41273873998765
INFO:cooler.balance:variance is 94.2202318840865
INFO:cooler.balance:variance is 68.29696045544898
INFO:cooler.balance:variance is 50.70176977355159
INFO:cooler.balance:variance is 36.87764743579143
INFO:cooler.balance:variance is 27.296919868826205
INFO:cooler.balance:variance is 19.904108419276437
INFO:cooler.balance:variance is 14.701436594018055
INFO:cooler.balance:variance is 10.739610888069826
INFO:cooler.balance:variance is 7.9199108074436975
INFO:cooler.balance:variance is 5.793436849874688
INFO:cooler.balance:variance is 4.267414827864594
INFO:cooler.balance:variance is 3.124726655072868
INFO:cooler.balance:variance is 2.2997000036899973
INFO:cooler.balance:variance is 1.6851360181855328
INFO:cooler.balance:variance is 1.2394324697348535
INFO:cooler.balance:variance is 0.9086970437848186
INFO:cooler.balance:variance is 0.6680483047314373
INFO:cooler.balance:variance is 0.4899759957636673
INFO:cooler.balance:variance is 0.36009519610775675
INFO:cooler.balance:variance is 0.2641858761777497
INFO:cooler.balance:variance is 0.19410858950408966
INFO:cooler.balance:variance is 0.14243904225290555
INFO:cooler.balance:variance is 0.10463702203865335
INFO:cooler.balance:variance is 0.07679575763162953
INFO:cooler.balance:variance is 0.05640734703739417
INFO:cooler.balance:variance is 0.041403508357366677
INFO:cooler.balance:variance is 0.030408366383655402
INFO:cooler.balance:variance is 0.022321891038197296
INFO:cooler.balance:variance is 0.01639289759975234
INFO:cooler.balance:variance is 0.012034287035015587
INFO:cooler.balance:variance is 0.008837352872941892
INFO:cooler.balance:variance is 0.006487934755384124
INFO:cooler.balance:variance is 0.004764216425740425
INFO:cooler.balance:variance is 0.0034977613771051777
INFO:cooler.balance:variance is 0.002568401024039522
INFO:cooler.balance:variance is 0.0018856978973954277
INFO:cooler.balance:variance is 0.0013846362717872865
INFO:cooler.balance:variance is 0.0010166062132771095
INFO:cooler.balance:variance is 0.0007464654100640905
INFO:cooler.balance:variance is 0.0005480654826329996
INFO:cooler.balance:variance is 0.0004024245712191185
INFO:cooler.balance:variance is 0.00029546867414217775
INFO:cooler.balance:variance is 0.00021695011892046702
INFO:cooler.balance:variance is 0.000159290517333741
INFO:cooler.balance:variance is 0.00011695956275391548
INFO:cooler.balance:variance is 8.587525217595095e-05
INFO:cooler.balance:variance is 6.30538926913886e-05
INFO:cooler.balance:variance is 4.629625393001586e-05
INFO:cooler.balance:variance is 3.399290717390147e-05
INFO:cooler.balance:variance is 2.4958786975855485e-05
INFO:cooler.balance:variance is 1.8325882085958815e-05
INFO:cooler.balance:variance is 1.3455534314563483e-05
INFO:cooler.balance:variance is 9.879650822684964e-06
--&gt; Done!
    resolution 50000... INFO:cooler.cli.balance:Balancing "../steps/bwa/PE/cool/pachytene_spermatocyte/pachytene_spermatocyte.mcool::resolutions/50000"
INFO:cooler.balance:variance is 18431848.298031054
INFO:cooler.balance:variance is 17447671.849013276
INFO:cooler.balance:variance is 4981344.05431578
INFO:cooler.balance:variance is 7693287.105934558
INFO:cooler.balance:variance is 2838726.4255396393
INFO:cooler.balance:variance is 3576316.9968233746
INFO:cooler.balance:variance is 1585495.623638183
INFO:cooler.balance:variance is 1703746.0556875316
INFO:cooler.balance:variance is 863398.1117170398
INFO:cooler.balance:variance is 825978.1989153951
INFO:cooler.balance:variance is 461127.79159515153
INFO:cooler.balance:variance is 405736.014310267
INFO:cooler.balance:variance is 242874.60519403912
INFO:cooler.balance:variance is 201273.7695367369
INFO:cooler.balance:variance is 126668.53470854613
INFO:cooler.balance:variance is 100573.43846005795
INFO:cooler.balance:variance is 65605.83323716617
INFO:cooler.balance:variance is 50522.49624256342
INFO:cooler.balance:variance is 33813.46430959407
INFO:cooler.balance:variance is 25477.694111100456
INFO:cooler.balance:variance is 17367.322716809605
INFO:cooler.balance:variance is 12883.816008011141
INFO:cooler.balance:variance is 8898.363836406606
INFO:cooler.balance:variance is 6528.2854864242845
INFO:cooler.balance:variance is 4551.249999195831
INFO:cooler.balance:variance is 3312.6726107340996
INFO:cooler.balance:variance is 2324.947925672288
INFO:cooler.balance:variance is 1682.6958210843459
INFO:cooler.balance:variance is 1186.6233597877042
INFO:cooler.balance:variance is 855.3683200754018
INFO:cooler.balance:variance is 605.2570602842542
INFO:cooler.balance:variance is 435.04066530126056
INFO:cooler.balance:variance is 308.58336062896547
INFO:cooler.balance:variance is 221.34530046582938
INFO:cooler.balance:variance is 157.27750120616236
INFO:cooler.balance:variance is 112.64910181686078
INFO:cooler.balance:variance is 80.14231708344705
INFO:cooler.balance:variance is 57.34145309739457
INFO:cooler.balance:variance is 40.83068946039579
INFO:cooler.balance:variance is 29.192365876690122
INFO:cooler.balance:variance is 20.799900362644202
INFO:cooler.balance:variance is 14.863204902438035
INFO:cooler.balance:variance is 10.594974687543148
INFO:cooler.balance:variance is 7.568085378615812
INFO:cooler.balance:variance is 5.396510124041308
INFO:cooler.balance:variance is 3.8537298590445253
INFO:cooler.balance:variance is 2.7485763415856677
INFO:cooler.balance:variance is 1.9624201326615418
INFO:cooler.balance:variance is 1.399876032539985
INFO:cooler.balance:variance is 0.9993411230574059
INFO:cooler.balance:variance is 0.7129548878919737
INFO:cooler.balance:variance is 0.5089128419429257
INFO:cooler.balance:variance is 0.36310136872476706
INFO:cooler.balance:variance is 0.2591663940838513
INFO:cooler.balance:variance is 0.18492217138142833
INFO:cooler.balance:variance is 0.13198299419526735
INFO:cooler.balance:variance is 0.09417740124469144
INFO:cooler.balance:variance is 0.06721406036876992
INFO:cooler.balance:variance is 0.0479625215897152
INFO:cooler.balance:variance is 0.03422979758887808
INFO:cooler.balance:variance is 0.024426182158008933
INFO:cooler.balance:variance is 0.017432111261062637
INFO:cooler.balance:variance is 0.012439643853501497
INFO:cooler.balance:variance is 0.008877622797358035
INFO:cooler.balance:variance is 0.0063351867998939
INFO:cooler.balance:variance is 0.004521100179079953
INFO:cooler.balance:variance is 0.0032263410781954915
INFO:cooler.balance:variance is 0.0023024600444497972
INFO:cooler.balance:variance is 0.001643087469795459
INFO:cooler.balance:variance is 0.0011725745203987215
INFO:cooler.balance:variance is 0.0008367789972726538
INFO:cooler.balance:variance is 0.0005971577504135824
INFO:cooler.balance:variance is 0.0004261481731548678
INFO:cooler.balance:variance is 0.00030411503137932696
INFO:cooler.balance:variance is 0.00021702528171900417
INFO:cooler.balance:variance is 0.00015487696764933953
INFO:cooler.balance:variance is 0.0001105248439641671
INFO:cooler.balance:variance is 7.887436676642943e-05
INFO:cooler.balance:variance is 5.628716944850075e-05
INFO:cooler.balance:variance is 4.016844357520983e-05
INFO:cooler.balance:variance is 2.8665455667531923e-05
INFO:cooler.balance:variance is 2.0456633899571525e-05
INFO:cooler.balance:variance is 1.4598500457332114e-05
INFO:cooler.balance:variance is 1.041797658945766e-05
INFO:cooler.balance:variance is 7.434600142721613e-06
--&gt; Done!
    resolution 100000... INFO:cooler.cli.balance:Balancing "../steps/bwa/PE/cool/pachytene_spermatocyte/pachytene_spermatocyte.mcool::resolutions/100000"
INFO:cooler.balance:variance is 53616608.142429814
INFO:cooler.balance:variance is 62270240.87549858
INFO:cooler.balance:variance is 17673584.536443096
INFO:cooler.balance:variance is 26540910.062633596
INFO:cooler.balance:variance is 9793931.507898387
INFO:cooler.balance:variance is 11936280.982396986
INFO:cooler.balance:variance is 5305360.820377936
INFO:cooler.balance:variance is 5510844.229498362
INFO:cooler.balance:variance is 2799768.25823349
INFO:cooler.balance:variance is 2592147.1300255563
INFO:cooler.balance:variance is 1448669.2782955666
INFO:cooler.balance:variance is 1236163.0897219577
INFO:cooler.balance:variance is 739186.0340420571
INFO:cooler.balance:variance is 595482.2457245848
INFO:cooler.balance:variance is 373514.62111090764
INFO:cooler.balance:variance is 288956.4223052109
INFO:cooler.balance:variance is 187463.75001945812
INFO:cooler.balance:variance is 140952.27381906184
INFO:cooler.balance:variance is 93643.23451826241
INFO:cooler.balance:variance is 69013.73704575999
INFO:cooler.balance:variance is 46623.49245950063
INFO:cooler.balance:variance is 33880.677166747766
INFO:cooler.balance:variance is 23159.716369041696
INFO:cooler.balance:variance is 16664.21656971125
INFO:cooler.balance:variance is 11485.812878165652
INFO:cooler.balance:variance is 8207.186707568459
INFO:cooler.balance:variance is 5689.835933978325
INFO:cooler.balance:variance is 4045.8557009680217
INFO:cooler.balance:variance is 2816.3960565648817
INFO:cooler.balance:variance is 1995.7814169375108
INFO:cooler.balance:variance is 1393.3052225559989
INFO:cooler.balance:variance is 984.9569914127245
INFO:cooler.balance:variance is 689.0159400361586
INFO:cooler.balance:variance is 486.2543421662657
INFO:cooler.balance:variance is 340.638079209245
INFO:cooler.balance:variance is 240.10961643487337
INFO:cooler.balance:variance is 168.37339249965183
INFO:cooler.balance:variance is 118.58392572852415
INFO:cooler.balance:variance is 83.21372155793466
INFO:cooler.balance:variance is 58.572189344967796
INFO:cooler.balance:variance is 41.122084170481855
INFO:cooler.balance:variance is 28.93288871373499
INFO:cooler.balance:variance is 20.320119460282545
INFO:cooler.balance:variance is 14.292774189255743
INFO:cooler.balance:variance is 10.040538139727042
INFO:cooler.balance:variance is 7.060873285877933
INFO:cooler.balance:variance is 4.961047452113731
INFO:cooler.balance:variance is 3.4882881919217574
INFO:cooler.balance:variance is 2.4512053215719316
INFO:cooler.balance:variance is 1.7233550903080506
INFO:cooler.balance:variance is 1.2110969616475535
INFO:cooler.balance:variance is 0.8514186137929988
INFO:cooler.balance:variance is 0.5983746102758688
INFO:cooler.balance:variance is 0.42064496610775565
INFO:cooler.balance:variance is 0.2956404806488803
INFO:cooler.balance:variance is 0.20782184400752798
INFO:cooler.balance:variance is 0.1460670239629719
INFO:cooler.balance:variance is 0.1026759571729071
INFO:cooler.balance:variance is 0.0721670133397838
INFO:cooler.balance:variance is 0.05072800465040654
INFO:cooler.balance:variance is 0.03565529776827899
INFO:cooler.balance:variance is 0.025062698035080534
INFO:cooler.balance:variance is 0.017616050579021866
INFO:cooler.balance:variance is 0.012382506977478764
INFO:cooler.balance:variance is 0.008703469839450206
INFO:cooler.balance:variance is 0.00611772352079245
INFO:cooler.balance:variance is 0.004300073575113891
INFO:cooler.balance:variance is 0.0030225358824295433
INFO:cooler.balance:variance is 0.002124511310476646
INFO:cooler.balance:variance is 0.0014933215559001201
INFO:cooler.balance:variance is 0.0010496439331096595
INFO:cooler.balance:variance is 0.0007377944402278002
INFO:cooler.balance:variance is 0.000518590797232676
INFO:cooler.balance:variance is 0.00036451679698182304
INFO:cooler.balance:variance is 0.00025621674422765815
INFO:cooler.balance:variance is 0.00018009423023698608
INFO:cooler.balance:variance is 0.0001265873004781543
INFO:cooler.balance:variance is 8.897789232968615e-05
INFO:cooler.balance:variance is 6.254213717821953e-05
INFO:cooler.balance:variance is 4.3960687094673345e-05
INFO:cooler.balance:variance is 3.089977102815868e-05
INFO:cooler.balance:variance is 2.1719351774475946e-05
INFO:cooler.balance:variance is 1.5266440157551614e-05
INFO:cooler.balance:variance is 1.0730730017598774e-05
INFO:cooler.balance:variance is 7.542586167459442e-06
--&gt; Done!
    resolution 500000... INFO:cooler.cli.balance:Balancing "../steps/bwa/PE/cool/pachytene_spermatocyte/pachytene_spermatocyte.mcool::resolutions/500000"
INFO:cooler.balance:variance is 834280253.4737167
INFO:cooler.balance:variance is 1061416258.2212846
INFO:cooler.balance:variance is 292880402.39824384
INFO:cooler.balance:variance is 385071108.6814928
INFO:cooler.balance:variance is 141703015.0753609
INFO:cooler.balance:variance is 150124311.50685197
INFO:cooler.balance:variance is 67023391.70471464
INFO:cooler.balance:variance is 60602333.346214965
INFO:cooler.balance:variance is 30831881.977075275
INFO:cooler.balance:variance is 25034554.17668421
INFO:cooler.balance:variance is 13897395.760384617
INFO:cooler.balance:variance is 10504594.83095645
INFO:cooler.balance:variance is 6178752.087875902
INFO:cooler.balance:variance is 4454457.137107904
INFO:cooler.balance:variance is 2722245.574047635
INFO:cooler.balance:variance is 1902271.6825734612
INFO:cooler.balance:variance is 1192245.60600815
INFO:cooler.balance:variance is 816184.5116627672
INFO:cooler.balance:variance is 520123.12105649576
INFO:cooler.balance:variance is 351281.54170832597
INFO:cooler.balance:variance is 226324.7993936284
INFO:cooler.balance:variance is 151501.0941272102
INFO:cooler.balance:variance is 98316.45322067304
INFO:cooler.balance:variance is 65428.351117088554
INFO:cooler.balance:variance is 42661.809420574355
INFO:cooler.balance:variance is 28281.66920854915
INFO:cooler.balance:variance is 18498.482655117874
INFO:cooler.balance:variance is 12232.079570857899
INFO:cooler.balance:variance is 8017.241440998923
INFO:cooler.balance:variance is 5292.541467311659
INFO:cooler.balance:variance is 3473.5775941885627
INFO:cooler.balance:variance is 2290.54753530696
INFO:cooler.balance:variance is 1504.6622816771903
INFO:cooler.balance:variance is 991.4879787898274
INFO:cooler.balance:variance is 651.6911714977425
INFO:cooler.balance:variance is 429.2237139989607
INFO:cooler.balance:variance is 282.23162024641846
INFO:cooler.balance:variance is 185.82820143817116
INFO:cooler.balance:variance is 122.220443528926
INFO:cooler.balance:variance is 80.4563585718504
INFO:cooler.balance:variance is 52.92552396318983
INFO:cooler.balance:variance is 34.83556275320123
INFO:cooler.balance:variance is 22.917929124955013
INFO:cooler.balance:variance is 15.083228472460018
INFO:cooler.balance:variance is 9.923806100100139
INFO:cooler.balance:variance is 6.530880300978679
INFO:cooler.balance:variance is 4.297108892552986
INFO:cooler.balance:variance is 2.82782835043848
INFO:cooler.balance:variance is 1.8606782621639915
INFO:cooler.balance:variance is 1.2244383807497863
INFO:cooler.balance:variance is 0.8056828629215431
INFO:cooler.balance:variance is 0.5301789926107803
INFO:cooler.balance:variance is 0.3488635518806641
INFO:cooler.balance:variance is 0.2295668682310631
INFO:cooler.balance:variance is 0.15105884800587205
INFO:cooler.balance:variance is 0.09940234632879694
INFO:cooler.balance:variance is 0.06540879452938436
INFO:cooler.balance:variance is 0.043041217112726
INFO:cooler.balance:variance is 0.02832211824549304
INFO:cooler.balance:variance is 0.018636861113787605
INFO:cooler.balance:variance is 0.012263517668078972
INFO:cooler.balance:variance is 0.008069770845208508
INFO:cooler.balance:variance is 0.005310118604675879
INFO:cooler.balance:variance is 0.0034942162028774554
INFO:cooler.balance:variance is 0.002299287459075082
INFO:cooler.balance:variance is 0.0015129982807701346
INFO:cooler.balance:variance is 0.0009955939447996328
INFO:cooler.balance:variance is 0.0006551294990723688
INFO:cooler.balance:variance is 0.0004310932018127833
INFO:cooler.balance:variance is 0.0002836716370018215
INFO:cooler.balance:variance is 0.00018666378601855788
INFO:cooler.balance:variance is 0.0001228300703023587
INFO:cooler.balance:variance is 8.082560151970415e-05
INFO:cooler.balance:variance is 5.318553136687267e-05
INFO:cooler.balance:variance is 3.499756319916865e-05
INFO:cooler.balance:variance is 2.3029383703449055e-05
INFO:cooler.balance:variance is 1.5153978220299725e-05
INFO:cooler.balance:variance is 9.971744383977664e-06
--&gt; Done!</code></pre>
</div>
</div>
</section>
</section>
<section id="repmerge-pool-all-from-each-biosample-id" class="level2">
<h2 class="anchored" data-anchor-id="repmerge-pool-all-from-each-biosample-id">RepMerge (pool all from each BioSample ID)</h2>
<p>Merge into the original ‘replicates’</p>
<div id="cell-26" class="cell" data-execution_count="1">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co">#df.groupby(['source_name','BioSample'])['Reads'].sum()</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div id="cell-27" class="cell" data-execution_count="2">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># from pprintpp import pprint as pp</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="co"># grouped_df = df.groupby(['source_name','BioSample'])</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a><span class="co"># # Initialize an empty dictionary</span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a><span class="co"># rep_dict = {}</span></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a><span class="co"># # Iterate over each group</span></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a><span class="co"># for (source_name, BioSample), group in grouped_df:</span></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a><span class="co">#     # Extract the 'Run' column and convert it to a list</span></span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a><span class="co">#     run_list = group['Run'].tolist()</span></span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a><span class="co">#     # Populate the nested dictionary</span></span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a><span class="co">#     if source_name not in rep_dict:</span></span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a><span class="co">#         rep_dict[source_name] = {}</span></span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a><span class="co">#     rep_dict[source_name][BioSample] = run_list</span></span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a><span class="co"># pp(rep_dict)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div id="cell-28" class="cell" data-execution_count="3">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># for source_name, BioSample_dict in rep_dict.items():</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="co">#     print(f"source_name: {source_name}")</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="co">#     print(f"Changing working dir: {source_name}/")</span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a><span class="co">#     for BioSample, run_list in BioSample_dict.items():</span></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a><span class="co">#         print(f"Merging samples for BioSample: {BioSample}: {run_list}")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="plotting" class="level1 page-columns page-full">
<h1>Plotting</h1>
<section id="kb-resolution" class="level2 page-columns page-full">
<h2 class="anchored" data-anchor-id="kb-resolution">500kb resolution</h2>
<section id="example-with-a-single-sample" class="level3">
<h3 class="anchored" data-anchor-id="example-with-a-single-sample">Example with a single sample</h3>
<section id="imports" class="level4">
<h4 class="anchored" data-anchor-id="imports">Imports</h4>
<div id="cell-33" class="cell" data-execution_count="1">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># import standard python libraries</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a><span class="op">%</span>matplotlib inline</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os, subprocess</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div id="cell-34" class="cell" data-execution_count="2">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Import python package for working with cooler files and tools for analysis</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooler</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooltools.lib.plotting</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooltools</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-cooler" class="level4">
<h4 class="anchored" data-anchor-id="load-cooler">Load cooler</h4>
<div id="cell-36" class="cell" data-execution_count="3">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>cool_file <span class="op">=</span> <span class="st">"../steps/bwa/PE/cool/fibroblast/fibroblast.mcool"</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>clr <span class="op">=</span> cooler.Cooler(<span class="ss">f"</span><span class="sc">{</span>cool_file<span class="sc">}</span><span class="ss">::resolutions/500000"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="calculate-gc-covariance-from-the-reference-genome" class="level4">
<h4 class="anchored" data-anchor-id="calculate-gc-covariance-from-the-reference-genome">Calculate gc covariance (from the reference genome)</h4>
<div id="cell-38" class="cell" data-execution_count="4">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> bioframe</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>bins <span class="op">=</span> clr.bins()[:]</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>rheMac10 <span class="op">=</span> bioframe.load_fasta(<span class="st">'../data/links/ucsc_ref/rheMac10.fa'</span>)</span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>gc_cov_csv <span class="op">=</span> <span class="st">'../steps/rheMac10_gc_cov_500kb.tsv'</span></span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> <span class="kw">not</span> os.path.exists(gc_cov_csv):</span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">'Calculate the fraction of GC basepairs for each bin'</span>)</span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>    gc_cov <span class="op">=</span> bioframe.frac_gc(bins[[<span class="st">'chrom'</span>, <span class="st">'start'</span>, <span class="st">'end'</span>]], rheMac10)</span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>    gc_cov.to_csv(gc_cov_csv, index<span class="op">=</span><span class="va">False</span>, sep<span class="op">=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>)</span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a>    display(gc_cov)</span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>: </span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"Already exists, just read from file:"</span>)</span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a>    gc_cov <span class="op">=</span> pd.read_csv(gc_cov_csv, sep<span class="op">=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>)</span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a>    display(gc_cov)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Already exists, just read from file:</code></pre>
</div>
<div class="cell-output cell-output-display">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">chrom</th>
<th data-quarto-table-cell-role="th">start</th>
<th data-quarto-table-cell-role="th">end</th>
<th data-quarto-table-cell-role="th">GC</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>chr1</td>
<td>0</td>
<td>500000</td>
<td>0.424508</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>chr1</td>
<td>500000</td>
<td>1000000</td>
<td>0.378836</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>chr1</td>
<td>1000000</td>
<td>1500000</td>
<td>0.388272</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3</td>
<td>chr1</td>
<td>1500000</td>
<td>2000000</td>
<td>0.445226</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">4</td>
<td>chr1</td>
<td>2000000</td>
<td>2500000</td>
<td>0.439485</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">5715</td>
<td>chrY</td>
<td>9500000</td>
<td>10000000</td>
<td>0.399518</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">5716</td>
<td>chrY</td>
<td>10000000</td>
<td>10500000</td>
<td>0.396028</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">5717</td>
<td>chrY</td>
<td>10500000</td>
<td>11000000</td>
<td>0.395280</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">5718</td>
<td>chrY</td>
<td>11000000</td>
<td>11500000</td>
<td>0.382006</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">5719</td>
<td>chrY</td>
<td>11500000</td>
<td>11753682</td>
<td>NaN</td>
</tr>
</tbody>
</table>

<p>5720 rows × 4 columns</p>
</div>
</div>
</div>
<div id="cell-39" class="cell" data-execution_count="5">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Apparently, we have som NaN values, I don't know why (and only 1 on the X)</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>gc_cov[gc_cov[<span class="st">'GC'</span>].isna()].groupby(<span class="st">'chrom'</span>).size()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="5">
<pre><code>chrom
chr1     1
chr10    1
chr11    1
chr12    1
chr13    1
chr14    1
chr15    1
chr16    1
chr17    1
chr18    1
chr19    1
chr2     1
chr20    1
chr3     1
chr4     1
chr5     1
chr6     1
chr7     1
chr8     1
chr9     1
chrY     1
dtype: int64</code></pre>
</div>
</div>
<div id="cell-40" class="cell" data-execution_count="6">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Make the full view frame</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>view_df <span class="op">=</span> pd.DataFrame(</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>    {</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>    <span class="st">'chrom'</span>: clr.chromnames,</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>    <span class="st">'start'</span>: <span class="dv">0</span>,</span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>    <span class="st">'end'</span>: clr.chromsizes.values,</span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>    <span class="st">'name'</span>: clr.chromnames</span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>                      )</span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a><span class="co">#display(view_df)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div id="cell-41" class="cell" data-execution_count="7">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co"># obtain first 3 eigenvectors</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>cis_eigs <span class="op">=</span> cooltools.eigs_cis(</span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>                        clr,</span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>                        gc_cov,</span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>                        view_df<span class="op">=</span>view_df,</span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>                        n_eigs<span class="op">=</span><span class="dv">3</span>,</span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a>                        )</span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a><span class="co"># cis_eigs[0] returns eigenvalues, here we focus on eigenvectors</span></span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a>eigenvector_track <span class="op">=</span> cis_eigs[<span class="dv">1</span>][[<span class="st">'chrom'</span>,<span class="st">'start'</span>,<span class="st">'end'</span>,<span class="st">'E1'</span>]]</span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a><span class="bu">len</span>(eigenvector_track)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="7">
<pre><code>5720</code></pre>
</div>
</div>
<div id="cell-42" class="cell" data-execution_count="8">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co"># full track</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>eigenvector_track</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="8">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">chrom</th>
<th data-quarto-table-cell-role="th">start</th>
<th data-quarto-table-cell-role="th">end</th>
<th data-quarto-table-cell-role="th">E1</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>chr1</td>
<td>0</td>
<td>500000</td>
<td>-0.756052</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>chr1</td>
<td>500000</td>
<td>1000000</td>
<td>-1.001063</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>chr1</td>
<td>1000000</td>
<td>1500000</td>
<td>-0.928042</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3</td>
<td>chr1</td>
<td>1500000</td>
<td>2000000</td>
<td>-0.507579</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">4</td>
<td>chr1</td>
<td>2000000</td>
<td>2500000</td>
<td>0.123708</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">5715</td>
<td>chrY</td>
<td>9500000</td>
<td>10000000</td>
<td>-0.499798</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">5716</td>
<td>chrY</td>
<td>10000000</td>
<td>10500000</td>
<td>-0.588820</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">5717</td>
<td>chrY</td>
<td>10500000</td>
<td>11000000</td>
<td>-0.988677</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">5718</td>
<td>chrY</td>
<td>11000000</td>
<td>11500000</td>
<td>-0.566585</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">5719</td>
<td>chrY</td>
<td>11500000</td>
<td>11753682</td>
<td>NaN</td>
</tr>
</tbody>
</table>

<p>5720 rows × 4 columns</p>
</div>
</div>
</div>
<div id="cell-43" class="cell" data-execution_count="9">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co"># subset the chrX   </span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>eigenvector_track_chrX <span class="op">=</span> eigenvector_track.loc[eigenvector_track[<span class="st">'chrom'</span>] <span class="op">==</span> <span class="st">'chrX'</span>]</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>nbins <span class="op">=</span> <span class="bu">len</span>(eigenvector_track_chrX)</span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>eigenvector_track_chrX</span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>e1X_values <span class="op">=</span> eigenvector_track_chrX[<span class="st">'E1'</span>].values</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div id="cell-44" class="cell" data-execution_count="10">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Where does the E1 change sign</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>np.where(np.diff( (cis_eigs[<span class="dv">1</span>][cis_eigs[<span class="dv">1</span>][<span class="st">'chrom'</span>]<span class="op">==</span><span class="st">'chrX'</span>][<span class="st">'E1'</span>]<span class="op">&gt;</span><span class="dv">0</span>).astype(<span class="bu">int</span>)))[<span class="dv">0</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="10">
<pre><code>array([  1,   6,  16,  26,  29,  40,  42,  49,  54,  55,  59,  60,  61,
        62,  74,  83,  86, 101, 102, 103, 104, 108, 123, 124, 129, 138,
       141, 142, 147, 149, 185, 188, 194, 196, 197, 201, 206, 209, 211,
       215, 222, 224, 228, 233, 238, 240, 250, 253, 256, 265, 273, 274,
       288, 289, 290, 293, 296, 302])</code></pre>
</div>
</div>
</section>
<section id="plot-e1-and-matrix" class="level4">
<h4 class="anchored" data-anchor-id="plot-e1-and-matrix">Plot E1 and matrix</h4>
<div id="cell-46" class="cell" data-execution_count="56">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="op">%%</span>capture chrX_matrix_e1_500kb</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> matplotlib.colors <span class="im">import</span> LogNorm</span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> mpl_toolkits.axes_grid1 <span class="im">import</span> make_axes_locatable</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>f, ax <span class="op">=</span> plt.subplots(</span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a>    figsize<span class="op">=</span>(<span class="dv">15</span>, <span class="dv">10</span>),</span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a>norm <span class="op">=</span> LogNorm(vmax<span class="op">=</span><span class="fl">0.1</span>)</span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-12"><a href="#cb33-12" aria-hidden="true" tabindex="-1"></a>im <span class="op">=</span> ax.matshow(</span>
<span id="cb33-13"><a href="#cb33-13" aria-hidden="true" tabindex="-1"></a>    clr.matrix().fetch(<span class="st">'chrX'</span>),</span>
<span id="cb33-14"><a href="#cb33-14" aria-hidden="true" tabindex="-1"></a>    norm<span class="op">=</span>norm,</span>
<span id="cb33-15"><a href="#cb33-15" aria-hidden="true" tabindex="-1"></a>    cmap<span class="op">=</span><span class="st">'fall'</span>,</span>
<span id="cb33-16"><a href="#cb33-16" aria-hidden="true" tabindex="-1"></a>)<span class="op">;</span></span>
<span id="cb33-17"><a href="#cb33-17" aria-hidden="true" tabindex="-1"></a>plt.axis([<span class="dv">0</span>,nbins,nbins,<span class="dv">0</span>])</span>
<span id="cb33-18"><a href="#cb33-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-19"><a href="#cb33-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-20"><a href="#cb33-20" aria-hidden="true" tabindex="-1"></a>divider <span class="op">=</span> make_axes_locatable(ax)</span>
<span id="cb33-21"><a href="#cb33-21" aria-hidden="true" tabindex="-1"></a>cax <span class="op">=</span> divider.append_axes(<span class="st">"right"</span>, size<span class="op">=</span><span class="st">"5%"</span>, pad<span class="op">=</span><span class="fl">0.05</span>)</span>
<span id="cb33-22"><a href="#cb33-22" aria-hidden="true" tabindex="-1"></a>plt.colorbar(im, cax<span class="op">=</span>cax, label<span class="op">=</span><span class="st">'corrected frequencies'</span>)<span class="op">;</span></span>
<span id="cb33-23"><a href="#cb33-23" aria-hidden="true" tabindex="-1"></a>ax.set_ylabel(<span class="st">'chrX:500kb bins, #bin'</span>)</span>
<span id="cb33-24"><a href="#cb33-24" aria-hidden="true" tabindex="-1"></a>ax.xaxis.set_visible(<span class="va">False</span>)</span>
<span id="cb33-25"><a href="#cb33-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-26"><a href="#cb33-26" aria-hidden="true" tabindex="-1"></a>ax1 <span class="op">=</span> divider.append_axes(<span class="st">"top"</span>, size<span class="op">=</span><span class="st">"20%"</span>, pad<span class="op">=</span><span class="fl">0.05</span>, sharex<span class="op">=</span>ax)</span>
<span id="cb33-27"><a href="#cb33-27" aria-hidden="true" tabindex="-1"></a><span class="co">#weights = clr.bins()[:]['weight'].values</span></span>
<span id="cb33-28"><a href="#cb33-28" aria-hidden="true" tabindex="-1"></a><span class="co">#ax1.plot([0,nbins],[0,0],'k',lw=0.25)</span></span>
<span id="cb33-29"><a href="#cb33-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-30"><a href="#cb33-30" aria-hidden="true" tabindex="-1"></a><span class="co">#ax1.plot(e1X_values, label='E1')</span></span>
<span id="cb33-31"><a href="#cb33-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-32"><a href="#cb33-32" aria-hidden="true" tabindex="-1"></a><span class="co"># Fill between the line and 0</span></span>
<span id="cb33-33"><a href="#cb33-33" aria-hidden="true" tabindex="-1"></a>ax1.fill_between(<span class="bu">range</span>(<span class="bu">len</span>(e1X_values)), e1X_values, <span class="dv">0</span>, where<span class="op">=</span>(e1X_values <span class="op">&gt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:red'</span>)</span>
<span id="cb33-34"><a href="#cb33-34" aria-hidden="true" tabindex="-1"></a>ax1.fill_between(<span class="bu">range</span>(<span class="bu">len</span>(e1X_values)), e1X_values, <span class="dv">0</span>, where<span class="op">=</span>(e1X_values <span class="op">&lt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:blue'</span>)</span>
<span id="cb33-35"><a href="#cb33-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-36"><a href="#cb33-36" aria-hidden="true" tabindex="-1"></a>ax1.set_ylabel(<span class="st">'E1'</span>)</span>
<span id="cb33-37"><a href="#cb33-37" aria-hidden="true" tabindex="-1"></a>ax1.set_xticks([])<span class="op">;</span></span>
<span id="cb33-38"><a href="#cb33-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-39"><a href="#cb33-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-40"><a href="#cb33-40" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i <span class="kw">in</span> np.where(np.diff( (cis_eigs[<span class="dv">1</span>][cis_eigs[<span class="dv">1</span>][<span class="st">'chrom'</span>]<span class="op">==</span><span class="st">'chrX'</span>][<span class="st">'E1'</span>]<span class="op">&gt;</span><span class="dv">0</span>).astype(<span class="bu">int</span>)))[<span class="dv">0</span>]:</span>
<span id="cb33-41"><a href="#cb33-41" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Horisontal lines where E1 intersects 0</span></span>
<span id="cb33-42"><a href="#cb33-42" aria-hidden="true" tabindex="-1"></a>    ax.plot([<span class="dv">0</span>,nbins],[i,i],<span class="st">'k'</span>,lw<span class="op">=</span><span class="fl">0.4</span>)</span>
<span id="cb33-43"><a href="#cb33-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-44"><a href="#cb33-44" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Vertical lines where E1 intersects 0</span></span>
<span id="cb33-45"><a href="#cb33-45" aria-hidden="true" tabindex="-1"></a>    ax.plot([i,i],[<span class="dv">0</span>,nbins],<span class="st">'k'</span>,lw<span class="op">=</span><span class="fl">0.4</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div id="cell-chrX-matrix-e1-500kb" class="cell" data-execution_count="57">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>chrX_matrix_e1_500kb()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div id="chrx-matrix-e1-500kb" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="03_compartments_files/figure-html/chrx-matrix-e1-500kb-output-1.png" class="img-fluid figure-img"></p>
<figcaption>chrX interaction matrix with E1 eigenvector values. 500kb resolution</figcaption>
</figure>
</div>
</div>
</div>
<div id="cell-chrX-e1-500kb-plot" class="cell" data-fig-height="1" data-execution_count="13">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>f, ax1 <span class="op">=</span> plt.subplots(</span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>    figsize<span class="op">=</span>(<span class="dv">10</span>, <span class="dv">2</span>),</span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Fill between the line and 0</span></span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a>ax1.fill_between(<span class="bu">range</span>(<span class="bu">len</span>(e1X_values)), e1X_values, <span class="dv">0</span>, where<span class="op">=</span>(e1X_values <span class="op">&gt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:red'</span>)</span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a>ax1.fill_between(<span class="bu">range</span>(<span class="bu">len</span>(e1X_values)), e1X_values, <span class="dv">0</span>, where<span class="op">=</span>(e1X_values <span class="op">&lt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'C0'</span>)</span>
<span id="cb35-10"><a href="#cb35-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-11"><a href="#cb35-11" aria-hidden="true" tabindex="-1"></a><span class="co">#ax1.set_ylabel('E1')</span></span>
<span id="cb35-12"><a href="#cb35-12" aria-hidden="true" tabindex="-1"></a>ax1.set_xticks([])</span>
<span id="cb35-13"><a href="#cb35-13" aria-hidden="true" tabindex="-1"></a>ax1.set_yticks([])</span>
<span id="cb35-14"><a href="#cb35-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-15"><a href="#cb35-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove borders</span></span>
<span id="cb35-16"><a href="#cb35-16" aria-hidden="true" tabindex="-1"></a>ax1.spines[<span class="st">'top'</span>].set_visible(<span class="va">False</span>)</span>
<span id="cb35-17"><a href="#cb35-17" aria-hidden="true" tabindex="-1"></a>ax1.spines[<span class="st">'right'</span>].set_visible(<span class="va">False</span>)</span>
<span id="cb35-18"><a href="#cb35-18" aria-hidden="true" tabindex="-1"></a>ax1.spines[<span class="st">'left'</span>].set_visible(<span class="va">False</span>)</span>
<span id="cb35-19"><a href="#cb35-19" aria-hidden="true" tabindex="-1"></a>ax1.spines[<span class="st">'bottom'</span>].set_visible(<span class="va">False</span>)</span>
<span id="cb35-20"><a href="#cb35-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-21"><a href="#cb35-21" aria-hidden="true" tabindex="-1"></a>plt.tight_layout()</span>
<span id="cb35-22"><a href="#cb35-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-23"><a href="#cb35-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the plot as a SVG file</span></span>
<span id="cb35-24"><a href="#cb35-24" aria-hidden="true" tabindex="-1"></a><span class="co">#plt.savefig('e1_plot.svg', bbox_inches='tight')</span></span>
<span id="cb35-25"><a href="#cb35-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-26"><a href="#cb35-26" aria-hidden="true" tabindex="-1"></a>plt.show()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div id="chrx-e1-500kb-plot" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="03_compartments_files/figure-html/chrx-e1-500kb-plot-output-1.png" class="img-fluid figure-img"></p>
<figcaption>500kb binned E1 eigenvector values for chrX. Freshly calculated from the cooler file.</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="stairs-plot-of-the-e1-compartments" class="level4">
<h4 class="anchored" data-anchor-id="stairs-plot-of-the-e1-compartments">Stairs plot of the E1 compartments</h4>
<p>(Less spiky, more smooth)</p>
<div id="cell-chrX-e1-500kb-stairs" class="cell" data-fig-height="2">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>f, (ax1, ax2) <span class="op">=</span> plt.subplots(<span class="dv">2</span>,<span class="dv">1</span>,</span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>    figsize<span class="op">=</span>(<span class="dv">10</span>, <span class="dv">4</span>)</span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a>chrom_start <span class="op">=</span> eigenvector_track_chrX[<span class="st">'start'</span>].values</span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a>window_size <span class="op">=</span> chrom_start[<span class="dv">1</span>] <span class="op">-</span> chrom_start[<span class="dv">0</span>]</span>
<span id="cb36-9"><a href="#cb36-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-10"><a href="#cb36-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Fill between the line and 0</span></span>
<span id="cb36-11"><a href="#cb36-11" aria-hidden="true" tabindex="-1"></a>ax1.fill_between(<span class="bu">range</span>(<span class="bu">len</span>(e1X_values)), e1X_values, <span class="dv">0</span>, where<span class="op">=</span>(e1X_values <span class="op">&gt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:red'</span>)</span>
<span id="cb36-12"><a href="#cb36-12" aria-hidden="true" tabindex="-1"></a>ax1.fill_between(<span class="bu">range</span>(<span class="bu">len</span>(e1X_values)), e1X_values, <span class="dv">0</span>, where<span class="op">=</span>(e1X_values <span class="op">&lt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:blue'</span>)</span>
<span id="cb36-13"><a href="#cb36-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-14"><a href="#cb36-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Create stairs</span></span>
<span id="cb36-15"><a href="#cb36-15" aria-hidden="true" tabindex="-1"></a>x <span class="op">=</span> np.zeros(<span class="dv">2</span><span class="op">*</span>chrom_start.size)</span>
<span id="cb36-16"><a href="#cb36-16" aria-hidden="true" tabindex="-1"></a>y <span class="op">=</span> np.zeros(<span class="dv">2</span><span class="op">*</span>chrom_start.size)</span>
<span id="cb36-17"><a href="#cb36-17" aria-hidden="true" tabindex="-1"></a>x[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> chrom_start</span>
<span id="cb36-18"><a href="#cb36-18" aria-hidden="true" tabindex="-1"></a>x[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> chrom_start <span class="op">+</span> window_size</span>
<span id="cb36-19"><a href="#cb36-19" aria-hidden="true" tabindex="-1"></a>y[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> e1X_values</span>
<span id="cb36-20"><a href="#cb36-20" aria-hidden="true" tabindex="-1"></a>y[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> e1X_values</span>
<span id="cb36-21"><a href="#cb36-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-22"><a href="#cb36-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Layout</span></span>
<span id="cb36-23"><a href="#cb36-23" aria-hidden="true" tabindex="-1"></a>ax1.set_ylabel(<span class="st">'Spiky E1'</span>)</span>
<span id="cb36-24"><a href="#cb36-24" aria-hidden="true" tabindex="-1"></a>ax2.set_ylabel(<span class="st">'Stairs E1'</span>)</span>
<span id="cb36-25"><a href="#cb36-25" aria-hidden="true" tabindex="-1"></a>ax1.set_xticks([])</span>
<span id="cb36-26"><a href="#cb36-26" aria-hidden="true" tabindex="-1"></a>ax1.set_yticks([])</span>
<span id="cb36-27"><a href="#cb36-27" aria-hidden="true" tabindex="-1"></a>ax2.set_xticks([])</span>
<span id="cb36-28"><a href="#cb36-28" aria-hidden="true" tabindex="-1"></a>ax2.set_yticks([])</span>
<span id="cb36-29"><a href="#cb36-29" aria-hidden="true" tabindex="-1"></a>ax2.set_ylim(<span class="op">-</span><span class="dv">1</span>, <span class="dv">1</span>)</span>
<span id="cb36-30"><a href="#cb36-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-31"><a href="#cb36-31" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove borders</span></span>
<span id="cb36-32"><a href="#cb36-32" aria-hidden="true" tabindex="-1"></a>ax1.spines[:].set_visible(<span class="va">False</span>)</span>
<span id="cb36-33"><a href="#cb36-33" aria-hidden="true" tabindex="-1"></a>ax2.spines[:].set_visible(<span class="va">False</span>)</span>
<span id="cb36-34"><a href="#cb36-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-35"><a href="#cb36-35" aria-hidden="true" tabindex="-1"></a>ax2.fill_between(x, y, <span class="dv">0</span>, where<span class="op">=</span>(y <span class="op">&gt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:red'</span>)</span>
<span id="cb36-36"><a href="#cb36-36" aria-hidden="true" tabindex="-1"></a>ax2.fill_between(x, y, <span class="dv">0</span>, where<span class="op">=</span>(y <span class="op">&lt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:blue'</span>)</span>
<span id="cb36-37"><a href="#cb36-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-38"><a href="#cb36-38" aria-hidden="true" tabindex="-1"></a>plt.tight_layout()</span>
<span id="cb36-39"><a href="#cb36-39" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the plot as a high-resolution PNG file</span></span>
<span id="cb36-40"><a href="#cb36-40" aria-hidden="true" tabindex="-1"></a><span class="co">#plt.savefig('../steps/e1_plot.png', dpi=320, bbox_inches='tight')</span></span>
<span id="cb36-41"><a href="#cb36-41" aria-hidden="true" tabindex="-1"></a></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div id="chrx-e1-500kb-stairs" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="03_compartments_files/figure-html/chrx-e1-500kb-stairs-output-1.png" class="img-fluid figure-img"></p>
<figcaption>500kb binned E1 eigenvector values for chrX.</figcaption>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="all-5-full-merges" class="level3 page-columns page-full">
<h3 class="anchored" data-anchor-id="all-5-full-merges">All 5 full merges</h3>
<section id="load-coolers" class="level4">
<h4 class="anchored" data-anchor-id="load-coolers">Load coolers</h4>
<div id="cell-53" class="cell" data-execution_count="41">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> glob</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os.path <span class="im">as</span> op</span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooler</span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a>mcools <span class="op">=</span> glob.glob(<span class="st">"../steps/bwa/PE/cool/*/*.mcool"</span>)</span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a>res <span class="op">=</span> <span class="st">"::resolutions/500000"</span></span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-8"><a href="#cb37-8" aria-hidden="true" tabindex="-1"></a>clrs <span class="op">=</span> {op.basename(op.dirname(mcool)): cooler.Cooler(mcool<span class="op">+</span>res) <span class="cf">for</span> mcool <span class="kw">in</span> mcools}</span>
<span id="cb37-9"><a href="#cb37-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-10"><a href="#cb37-10" aria-hidden="true" tabindex="-1"></a>chron_order <span class="op">=</span> [<span class="st">'fibroblast'</span>, <span class="st">'spermatogonia'</span>, <span class="st">'pachytene_spermatocyte'</span>, <span class="st">'round_spermatid'</span>, <span class="st">'sperm'</span>]</span>
<span id="cb37-11"><a href="#cb37-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-12"><a href="#cb37-12" aria-hidden="true" tabindex="-1"></a>clrs <span class="op">=</span> {key: clrs[key] <span class="cf">for</span> key <span class="kw">in</span> chron_order}</span>
<span id="cb37-13"><a href="#cb37-13" aria-hidden="true" tabindex="-1"></a>clrs</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="41">
<pre><code>{'fibroblast': &lt;Cooler "fibroblast.mcool::/resolutions/500000"&gt;,
 'spermatogonia': &lt;Cooler "spermatogonia.mcool::/resolutions/500000"&gt;,
 'pachytene_spermatocyte': &lt;Cooler "pachytene_spermatocyte.mcool::/resolutions/500000"&gt;,
 'round_spermatid': &lt;Cooler "round_spermatid.mcool::/resolutions/500000"&gt;,
 'sperm': &lt;Cooler "sperm.mcool::/resolutions/500000"&gt;}</code></pre>
</div>
</div>
</section>
<section id="calculate-gc-covariance-from-the-reference-genome-1" class="level4">
<h4 class="anchored" data-anchor-id="calculate-gc-covariance-from-the-reference-genome-1">Calculate gc covariance (from the reference genome)</h4>
<p>Do this with any of the clrs - it just needs the bins positions.</p>
<div id="cell-55" class="cell" data-execution_count="42">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Try with only the gc_cov for chrX</span></span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> bioframe</span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os.path <span class="im">as</span> op</span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a>bins <span class="op">=</span> clrs[<span class="st">'fibroblast'</span>].bins().fetch(<span class="st">'chrX'</span>)[:]</span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a>out_name <span class="op">=</span> <span class="st">'../steps/rheMac10_gc_cov_X_500kb.tsv'</span></span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-10"><a href="#cb39-10" aria-hidden="true" tabindex="-1"></a>rheMac10 <span class="op">=</span> bioframe.load_fasta(<span class="st">'../data/links/ucsc_ref/rheMac10.fa'</span>)</span>
<span id="cb39-11"><a href="#cb39-11" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> <span class="kw">not</span> op.exists(out_name):</span>
<span id="cb39-12"><a href="#cb39-12" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">'Calculate the fraction of GC basepairs for each bin'</span>)</span>
<span id="cb39-13"><a href="#cb39-13" aria-hidden="true" tabindex="-1"></a>    gc_cov <span class="op">=</span> bioframe.frac_gc(bins[[<span class="st">'chrom'</span>, <span class="st">'start'</span>, <span class="st">'end'</span>]], rheMac10)</span>
<span id="cb39-14"><a href="#cb39-14" aria-hidden="true" tabindex="-1"></a>    gc_cov.to_csv(out_name, index<span class="op">=</span><span class="va">False</span>,sep<span class="op">=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>)</span>
<span id="cb39-15"><a href="#cb39-15" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(gc_cov.info())</span>
<span id="cb39-16"><a href="#cb39-16" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>: </span>
<span id="cb39-17"><a href="#cb39-17" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"Already exists, read from file"</span>)</span>
<span id="cb39-18"><a href="#cb39-18" aria-hidden="true" tabindex="-1"></a>    gc_cov <span class="op">=</span> pd.read_csv(out_name, sep<span class="op">=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>)</span>
<span id="cb39-19"><a href="#cb39-19" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(gc_cov.info())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Already exists, read from file
&lt;class 'pandas.core.frame.DataFrame'&gt;
RangeIndex: 307 entries, 0 to 306
Data columns (total 4 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   chrom   307 non-null    object 
 1   start   307 non-null    int64  
 2   end     307 non-null    int64  
 3   GC      307 non-null    float64
dtypes: float64(1), int64(2), object(1)
memory usage: 9.7+ KB
None</code></pre>
</div>
</div>
</section>
<section id="calculate-the-e1-compartments" class="level4 page-columns page-full">
<h4 class="anchored" data-anchor-id="calculate-the-e1-compartments">Calculate the E1 compartments</h4>
<p>Loop: view_df, cis_eigs, e1_values</p>
<section id="plot-gc-covariance" class="level5">
<h5 class="anchored" data-anchor-id="plot-gc-covariance">Plot GC covariance</h5>
<div id="cell-58" class="cell" data-execution_count="43">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>f, ax <span class="op">=</span> plt.subplots(</span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a>    figsize<span class="op">=</span>(<span class="dv">10</span>, <span class="dv">2</span>),</span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a>ax.plot(gc_cov[<span class="st">'start'</span>],gc_cov[<span class="st">'GC'</span>])</span>
<span id="cb41-8"><a href="#cb41-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-9"><a href="#cb41-9" aria-hidden="true" tabindex="-1"></a>plt.tight_layout()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="03_compartments_files/figure-html/cell-36-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="chromosome-restricted-e1-compartments" class="level5">
<h5 class="anchored" data-anchor-id="chromosome-restricted-e1-compartments">Chromosome restricted E1 compartments</h5>
<div id="cell-60" class="cell" data-execution_count="44">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Use gc_cov to calculate eigenvectors with cooltools.eigs_cis</span></span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooltools</span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a>eigs_full <span class="op">=</span> {}</span>
<span id="cb42-7"><a href="#cb42-7" aria-hidden="true" tabindex="-1"></a>e1_values_full <span class="op">=</span> {}</span>
<span id="cb42-8"><a href="#cb42-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-9"><a href="#cb42-9" aria-hidden="true" tabindex="-1"></a>chrX_size <span class="op">=</span> clrs[<span class="st">'fibroblast'</span>].chromsizes[<span class="st">'chrX'</span>]</span>
<span id="cb42-10"><a href="#cb42-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-11"><a href="#cb42-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Divide into chromosome arms</span></span>
<span id="cb42-12"><a href="#cb42-12" aria-hidden="true" tabindex="-1"></a>clr <span class="op">=</span> clrs[<span class="st">'fibroblast'</span>]</span>
<span id="cb42-13"><a href="#cb42-13" aria-hidden="true" tabindex="-1"></a>view_df_full <span class="op">=</span> pd.DataFrame(</span>
<span id="cb42-14"><a href="#cb42-14" aria-hidden="true" tabindex="-1"></a>    {</span>
<span id="cb42-15"><a href="#cb42-15" aria-hidden="true" tabindex="-1"></a>    <span class="st">'chrom'</span>: <span class="st">'chrX'</span>,</span>
<span id="cb42-16"><a href="#cb42-16" aria-hidden="true" tabindex="-1"></a>    <span class="st">'start'</span>: <span class="dv">0</span>,</span>
<span id="cb42-17"><a href="#cb42-17" aria-hidden="true" tabindex="-1"></a>    <span class="st">'end'</span>: chrX_size,</span>
<span id="cb42-18"><a href="#cb42-18" aria-hidden="true" tabindex="-1"></a>    <span class="st">'name'</span>: <span class="st">'chrX'</span></span>
<span id="cb42-19"><a href="#cb42-19" aria-hidden="true" tabindex="-1"></a>    }, index<span class="op">=</span>[<span class="dv">0</span>]</span>
<span id="cb42-20"><a href="#cb42-20" aria-hidden="true" tabindex="-1"></a>                      )</span>
<span id="cb42-21"><a href="#cb42-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-22"><a href="#cb42-22" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> name, clr <span class="kw">in</span> clrs.items():</span>
<span id="cb42-23"><a href="#cb42-23" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Calculating eigenvectors for </span><span class="sc">{</span>name<span class="sc">}</span><span class="ss">, size </span><span class="sc">{</span>clr<span class="sc">.</span>binsize<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb42-24"><a href="#cb42-24" aria-hidden="true" tabindex="-1"></a>    cis_eigs_full <span class="op">=</span> cooltools.eigs_cis(</span>
<span id="cb42-25"><a href="#cb42-25" aria-hidden="true" tabindex="-1"></a>                        clr,</span>
<span id="cb42-26"><a href="#cb42-26" aria-hidden="true" tabindex="-1"></a>                        gc_cov,</span>
<span id="cb42-27"><a href="#cb42-27" aria-hidden="true" tabindex="-1"></a>                        view_df<span class="op">=</span>view_df_full,</span>
<span id="cb42-28"><a href="#cb42-28" aria-hidden="true" tabindex="-1"></a>                        n_eigs<span class="op">=</span><span class="dv">3</span>,</span>
<span id="cb42-29"><a href="#cb42-29" aria-hidden="true" tabindex="-1"></a>                        )</span>
<span id="cb42-30"><a href="#cb42-30" aria-hidden="true" tabindex="-1"></a>    eigs_full[name] <span class="op">=</span> cis_eigs_full[<span class="dv">1</span>]</span>
<span id="cb42-31"><a href="#cb42-31" aria-hidden="true" tabindex="-1"></a>    e1_track_full <span class="op">=</span> cis_eigs_full[<span class="dv">1</span>][[<span class="st">'chrom'</span>,<span class="st">'start'</span>,<span class="st">'end'</span>,<span class="st">'E1'</span>]]</span>
<span id="cb42-32"><a href="#cb42-32" aria-hidden="true" tabindex="-1"></a>    e1_values_full[name] <span class="op">=</span> e1_track_full[<span class="st">'E1'</span>].values</span>
<span id="cb42-33"><a href="#cb42-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-34"><a href="#cb42-34" aria-hidden="true" tabindex="-1"></a><span class="co">#eigs</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for fibroblast, size 500000</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for spermatogonia, size 500000</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for pachytene_spermatocyte, size 500000</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for round_spermatid, size 500000</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for sperm, size 500000</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
</div>
</section>
<section id="chromosome-arms-restricted" class="level5">
<h5 class="anchored" data-anchor-id="chromosome-arms-restricted">Chromosome arms restricted</h5>
<div id="cell-62" class="cell" data-execution_count="45">
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Use gc_cov to calculate eigenvectors with cooltools.eigs_cis</span></span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooltools</span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb53-5"><a href="#cb53-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-6"><a href="#cb53-6" aria-hidden="true" tabindex="-1"></a>eigs <span class="op">=</span> {}</span>
<span id="cb53-7"><a href="#cb53-7" aria-hidden="true" tabindex="-1"></a>e1_values <span class="op">=</span> {}</span>
<span id="cb53-8"><a href="#cb53-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-9"><a href="#cb53-9" aria-hidden="true" tabindex="-1"></a>chrX_size <span class="op">=</span> clrs[<span class="st">'fibroblast'</span>].chromsizes[<span class="st">'chrX'</span>]</span>
<span id="cb53-10"><a href="#cb53-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-11"><a href="#cb53-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Divide into chromosome arms</span></span>
<span id="cb53-12"><a href="#cb53-12" aria-hidden="true" tabindex="-1"></a>view_df <span class="op">=</span> pd.DataFrame(</span>
<span id="cb53-13"><a href="#cb53-13" aria-hidden="true" tabindex="-1"></a>    {</span>
<span id="cb53-14"><a href="#cb53-14" aria-hidden="true" tabindex="-1"></a>    <span class="st">'chrom'</span>: <span class="st">'chrX'</span>,</span>
<span id="cb53-15"><a href="#cb53-15" aria-hidden="true" tabindex="-1"></a>    <span class="st">'start'</span>: [<span class="dv">0</span>, <span class="dv">59_000_001</span>],</span>
<span id="cb53-16"><a href="#cb53-16" aria-hidden="true" tabindex="-1"></a>    <span class="st">'end'</span>: [<span class="dv">59_000_000</span>, chrX_size],</span>
<span id="cb53-17"><a href="#cb53-17" aria-hidden="true" tabindex="-1"></a>    <span class="st">'name'</span>: [<span class="st">'X_short'</span>, <span class="st">'X_long'</span>]</span>
<span id="cb53-18"><a href="#cb53-18" aria-hidden="true" tabindex="-1"></a>    }, index<span class="op">=</span>[<span class="dv">0</span>,<span class="dv">1</span>]</span>
<span id="cb53-19"><a href="#cb53-19" aria-hidden="true" tabindex="-1"></a>                      )</span>
<span id="cb53-20"><a href="#cb53-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-21"><a href="#cb53-21" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> name, clr <span class="kw">in</span> clrs.items():</span>
<span id="cb53-22"><a href="#cb53-22" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Calculating eigenvectors for </span><span class="sc">{</span>name<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb53-23"><a href="#cb53-23" aria-hidden="true" tabindex="-1"></a>    cis_eigs <span class="op">=</span> cooltools.eigs_cis(</span>
<span id="cb53-24"><a href="#cb53-24" aria-hidden="true" tabindex="-1"></a>                        clr,</span>
<span id="cb53-25"><a href="#cb53-25" aria-hidden="true" tabindex="-1"></a>                        gc_cov,</span>
<span id="cb53-26"><a href="#cb53-26" aria-hidden="true" tabindex="-1"></a>                        view_df<span class="op">=</span>view_df,</span>
<span id="cb53-27"><a href="#cb53-27" aria-hidden="true" tabindex="-1"></a>                        n_eigs<span class="op">=</span><span class="dv">3</span>,</span>
<span id="cb53-28"><a href="#cb53-28" aria-hidden="true" tabindex="-1"></a>                        )</span>
<span id="cb53-29"><a href="#cb53-29" aria-hidden="true" tabindex="-1"></a>    eigs[name] <span class="op">=</span> cis_eigs[<span class="dv">1</span>]</span>
<span id="cb53-30"><a href="#cb53-30" aria-hidden="true" tabindex="-1"></a>    e1_track <span class="op">=</span> cis_eigs[<span class="dv">1</span>][[<span class="st">'chrom'</span>,<span class="st">'start'</span>,<span class="st">'end'</span>,<span class="st">'E1'</span>]]</span>
<span id="cb53-31"><a href="#cb53-31" aria-hidden="true" tabindex="-1"></a>    e1_values[name] <span class="op">=</span> e1_track[<span class="st">'E1'</span>].values</span>
<span id="cb53-32"><a href="#cb53-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-33"><a href="#cb53-33" aria-hidden="true" tabindex="-1"></a><span class="co">#eigs</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for fibroblast
Calculating eigenvectors for spermatogonia</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")
/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for pachytene_spermatocyte
Calculating eigenvectors for round_spermatid</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")
/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for sperm</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
</div>
</section>
<section id="refined-10mb-windows-restricted" class="level5">
<h5 class="anchored" data-anchor-id="refined-10mb-windows-restricted">Refined: 10Mb windows restricted</h5>
<div id="cell-64" class="cell" data-execution_count="46">
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Use gc_cov to calculate eigenvectors with cooltools.eigs_cis</span></span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooltools</span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-5"><a href="#cb60-5" aria-hidden="true" tabindex="-1"></a>eigs_10mb <span class="op">=</span> {}</span>
<span id="cb60-6"><a href="#cb60-6" aria-hidden="true" tabindex="-1"></a>e1_values_10mb <span class="op">=</span> {}</span>
<span id="cb60-7"><a href="#cb60-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-8"><a href="#cb60-8" aria-hidden="true" tabindex="-1"></a>chrX_size <span class="op">=</span> clrs[<span class="st">'fibroblast'</span>].chromsizes[<span class="st">'chrX'</span>]</span>
<span id="cb60-9"><a href="#cb60-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-10"><a href="#cb60-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate in 1Mb windows</span></span>
<span id="cb60-11"><a href="#cb60-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the window size (1Mb)</span></span>
<span id="cb60-12"><a href="#cb60-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-13"><a href="#cb60-13" aria-hidden="true" tabindex="-1"></a>window_size <span class="op">=</span> <span class="dv">10_000_000</span></span>
<span id="cb60-14"><a href="#cb60-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-15"><a href="#cb60-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate the start and end positions for each window</span></span>
<span id="cb60-16"><a href="#cb60-16" aria-hidden="true" tabindex="-1"></a>start_positions <span class="op">=</span> <span class="bu">list</span>(<span class="bu">range</span>(<span class="dv">0</span>, chrX_size, window_size))</span>
<span id="cb60-17"><a href="#cb60-17" aria-hidden="true" tabindex="-1"></a>end_positions <span class="op">=</span> [<span class="bu">min</span>(start <span class="op">+</span> window_size, chrX_size) <span class="cf">for</span> start <span class="kw">in</span> start_positions]</span>
<span id="cb60-18"><a href="#cb60-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the DataFrame</span></span>
<span id="cb60-19"><a href="#cb60-19" aria-hidden="true" tabindex="-1"></a>view_df_10mb <span class="op">=</span> pd.DataFrame({</span>
<span id="cb60-20"><a href="#cb60-20" aria-hidden="true" tabindex="-1"></a>    <span class="st">'chrom'</span>: [<span class="st">'chrX'</span>] <span class="op">*</span> <span class="bu">len</span>(start_positions),</span>
<span id="cb60-21"><a href="#cb60-21" aria-hidden="true" tabindex="-1"></a>    <span class="st">'start'</span>: start_positions,</span>
<span id="cb60-22"><a href="#cb60-22" aria-hidden="true" tabindex="-1"></a>    <span class="st">'end'</span>: end_positions,</span>
<span id="cb60-23"><a href="#cb60-23" aria-hidden="true" tabindex="-1"></a>    <span class="st">'name'</span>: [<span class="ss">f'X_</span><span class="sc">{</span>i<span class="sc">}</span><span class="ss">'</span> <span class="cf">for</span> i <span class="kw">in</span> <span class="bu">range</span>(<span class="bu">len</span>(start_positions))]</span>
<span id="cb60-24"><a href="#cb60-24" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb60-25"><a href="#cb60-25" aria-hidden="true" tabindex="-1"></a><span class="co">#display(view_df)</span></span>
<span id="cb60-26"><a href="#cb60-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-27"><a href="#cb60-27" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> name, clr <span class="kw">in</span> clrs.items():</span>
<span id="cb60-28"><a href="#cb60-28" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Calculating eigenvectors for </span><span class="sc">{</span>name<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb60-29"><a href="#cb60-29" aria-hidden="true" tabindex="-1"></a>    cis_eigs_10mb <span class="op">=</span> cooltools.eigs_cis(</span>
<span id="cb60-30"><a href="#cb60-30" aria-hidden="true" tabindex="-1"></a>                        clr,</span>
<span id="cb60-31"><a href="#cb60-31" aria-hidden="true" tabindex="-1"></a>                        gc_cov,</span>
<span id="cb60-32"><a href="#cb60-32" aria-hidden="true" tabindex="-1"></a>                        view_df<span class="op">=</span>view_df_10mb,</span>
<span id="cb60-33"><a href="#cb60-33" aria-hidden="true" tabindex="-1"></a>                        n_eigs<span class="op">=</span><span class="dv">3</span>,</span>
<span id="cb60-34"><a href="#cb60-34" aria-hidden="true" tabindex="-1"></a>                        )</span>
<span id="cb60-35"><a href="#cb60-35" aria-hidden="true" tabindex="-1"></a>    eigs_10mb[name] <span class="op">=</span> cis_eigs_10mb[<span class="dv">1</span>]</span>
<span id="cb60-36"><a href="#cb60-36" aria-hidden="true" tabindex="-1"></a>    e1_track_10mb <span class="op">=</span> cis_eigs_10mb[<span class="dv">1</span>][[<span class="st">'chrom'</span>,<span class="st">'start'</span>,<span class="st">'end'</span>,<span class="st">'E1'</span>]]</span>
<span id="cb60-37"><a href="#cb60-37" aria-hidden="true" tabindex="-1"></a>    e1_values_10mb[name] <span class="op">=</span> e1_track_10mb[<span class="st">'E1'</span>].values</span>
<span id="cb60-38"><a href="#cb60-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-39"><a href="#cb60-39" aria-hidden="true" tabindex="-1"></a><span class="co">#eigs</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for fibroblast</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for spermatogonia</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for pachytene_spermatocyte</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for round_spermatid</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for sperm</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
</div>
</section>
<section id="plot-nan-histogram" class="level5 page-columns page-full">
<h5 class="anchored" data-anchor-id="plot-nan-histogram">Plot NaN histogram</h5>
<div id="cell-e1_nan_hist" class="cell page-columns page-full" data-execution_count="47">
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Check the number of NaN values in the E1 column</span></span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb71-3"><a href="#cb71-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-4"><a href="#cb71-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Check the number of NaN values in the E1 column and create a DataFrame</span></span>
<span id="cb71-5"><a href="#cb71-5" aria-hidden="true" tabindex="-1"></a>nan_counts <span class="op">=</span> {k: {<span class="st">'length'</span>: <span class="bu">len</span>(v), <span class="st">'NaNs'</span>: np.isnan(v).<span class="bu">sum</span>()} <span class="cf">for</span> k, v <span class="kw">in</span> e1_values.items()}</span>
<span id="cb71-6"><a href="#cb71-6" aria-hidden="true" tabindex="-1"></a><span class="co">#display(pd.DataFrame.from_dict(nan_counts, orient='index'))</span></span>
<span id="cb71-7"><a href="#cb71-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-8"><a href="#cb71-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Locate the NaN values (histogram)</span></span>
<span id="cb71-9"><a href="#cb71-9" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb71-10"><a href="#cb71-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-11"><a href="#cb71-11" aria-hidden="true" tabindex="-1"></a>f, ax <span class="op">=</span> plt.subplots(<span class="dv">1</span>, <span class="dv">5</span>, figsize<span class="op">=</span>(<span class="dv">20</span>,<span class="dv">2</span>))</span>
<span id="cb71-12"><a href="#cb71-12" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i, (name, track) <span class="kw">in</span> <span class="bu">enumerate</span>(eigs.items()):</span>
<span id="cb71-13"><a href="#cb71-13" aria-hidden="true" tabindex="-1"></a>    e1 <span class="op">=</span> track[<span class="st">'E1'</span>].values</span>
<span id="cb71-14"><a href="#cb71-14" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Locate NaN values</span></span>
<span id="cb71-15"><a href="#cb71-15" aria-hidden="true" tabindex="-1"></a>    e1_nan <span class="op">=</span> np.where(np.isnan(e1))</span>
<span id="cb71-16"><a href="#cb71-16" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Plot histogram</span></span>
<span id="cb71-17"><a href="#cb71-17" aria-hidden="true" tabindex="-1"></a>    ax[i].hist(e1_nan, bins<span class="op">=</span><span class="dv">100</span>)</span>
<span id="cb71-18"><a href="#cb71-18" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb71-19"><a href="#cb71-19" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Plot median line</span></span>
<span id="cb71-20"><a href="#cb71-20" aria-hidden="true" tabindex="-1"></a>    median_pos <span class="op">=</span> np.median(e1_nan)</span>
<span id="cb71-21"><a href="#cb71-21" aria-hidden="true" tabindex="-1"></a>    ax[i].axvline(median_pos, color<span class="op">=</span><span class="st">'r'</span>)</span>
<span id="cb71-22"><a href="#cb71-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-23"><a href="#cb71-23" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Layout</span></span>
<span id="cb71-24"><a href="#cb71-24" aria-hidden="true" tabindex="-1"></a>    ax[i].set_title(name)</span>
<span id="cb71-25"><a href="#cb71-25" aria-hidden="true" tabindex="-1"></a>    xticks <span class="op">=</span> np.linspace(<span class="dv">0</span>, <span class="bu">len</span>(e1), num<span class="op">=</span><span class="dv">5</span>)</span>
<span id="cb71-26"><a href="#cb71-26" aria-hidden="true" tabindex="-1"></a>    xticks <span class="op">=</span> np.append(xticks, median_pos)  <span class="co"># Add median position to xticks</span></span>
<span id="cb71-27"><a href="#cb71-27" aria-hidden="true" tabindex="-1"></a>    ax[i].set_xticks(xticks)</span>
<span id="cb71-28"><a href="#cb71-28" aria-hidden="true" tabindex="-1"></a>    xticklabels <span class="op">=</span> np.linspace(<span class="dv">0</span>, <span class="bu">len</span>(e1) <span class="op">*</span> <span class="fl">0.5</span>, num<span class="op">=</span><span class="dv">5</span>, dtype <span class="op">=</span> <span class="st">'int'</span>).tolist()</span>
<span id="cb71-29"><a href="#cb71-29" aria-hidden="true" tabindex="-1"></a>    xticklabels.append(median_pos<span class="op">*</span><span class="fl">0.5</span>)  <span class="co"># Add median label</span></span>
<span id="cb71-30"><a href="#cb71-30" aria-hidden="true" tabindex="-1"></a>    ax[i].set_xticklabels(xticklabels, rotation<span class="op">=</span><span class="dv">45</span>)</span>
<span id="cb71-31"><a href="#cb71-31" aria-hidden="true" tabindex="-1"></a>    ax[i].set_xlabel(<span class="st">'Position (Mbp)'</span>)</span>
<span id="cb71-32"><a href="#cb71-32" aria-hidden="true" tabindex="-1"></a>                     </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display page-columns page-full">
<div id="e1_nan_hist" class="quarto-figure quarto-figure-center anchored page-columns page-full">
<figure class="figure page-columns page-full">
<p class="page-columns page-full"><img src="03_compartments_files/figure-html/e1_nan_hist-output-1.png" class="img-fluid figure-img column-page-inset"></p>
<figcaption>Histogram of NaN values in the E1 eigenvector for each cell type.</figcaption>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="plot-the-e1-compartments" class="level4 page-columns page-full">
<h4 class="anchored" data-anchor-id="plot-the-e1-compartments">Plot the E1 compartments</h4>
<div id="cell-e1-fullmerged-all-500kb" class="cell page-columns page-full" data-execution_count="48">
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb72-2"><a href="#cb72-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb72-3"><a href="#cb72-3" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> matplotlib.colors <span class="im">import</span> LogNorm</span>
<span id="cb72-4"><a href="#cb72-4" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> mpl_toolkits.axes_grid1 <span class="im">import</span> make_axes_locatable</span>
<span id="cb72-5"><a href="#cb72-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-6"><a href="#cb72-6" aria-hidden="true" tabindex="-1"></a>names_abbr <span class="op">=</span> {<span class="st">'fibroblast'</span>: <span class="st">'Fibroblast'</span>, <span class="st">'spermatogonia'</span>: <span class="st">'Spermatogonia'</span>, <span class="st">'pachytene_spermatocyte'</span>: <span class="st">'Pachytene Spermatogonia'</span>, <span class="st">'round_spermatid'</span>: <span class="st">'Round Spermatid'</span>, <span class="st">'sperm'</span>: <span class="st">'Sperm'</span>}</span>
<span id="cb72-7"><a href="#cb72-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-8"><a href="#cb72-8" aria-hidden="true" tabindex="-1"></a>chrom_start <span class="op">=</span> e1_track_full[<span class="st">'start'</span>].values</span>
<span id="cb72-9"><a href="#cb72-9" aria-hidden="true" tabindex="-1"></a>chrom_end <span class="op">=</span> e1_track_full[<span class="st">'end'</span>].values<span class="op">-</span><span class="dv">1</span></span>
<span id="cb72-10"><a href="#cb72-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-11"><a href="#cb72-11" aria-hidden="true" tabindex="-1"></a>f, axs <span class="op">=</span> plt.subplots(<span class="dv">5</span>, <span class="dv">3</span>, figsize<span class="op">=</span>(<span class="dv">18</span>, <span class="dv">8</span>), sharex<span class="op">=</span><span class="va">True</span>, sharey<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb72-12"><a href="#cb72-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-13"><a href="#cb72-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Populate the first column</span></span>
<span id="cb72-14"><a href="#cb72-14" aria-hidden="true" tabindex="-1"></a>axs[<span class="dv">0</span>,<span class="dv">0</span>].set_title(<span class="st">'E1: Full-chromosome restricted'</span>)</span>
<span id="cb72-15"><a href="#cb72-15" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i, (name, e1) <span class="kw">in</span> <span class="bu">enumerate</span>(e1_values_full.items()):</span>
<span id="cb72-16"><a href="#cb72-16" aria-hidden="true" tabindex="-1"></a>    ax <span class="op">=</span> axs[i,<span class="dv">0</span>]</span>
<span id="cb72-17"><a href="#cb72-17" aria-hidden="true" tabindex="-1"></a>    ax.set_ylabel(names_abbr[name])</span>
<span id="cb72-18"><a href="#cb72-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-19"><a href="#cb72-19" aria-hidden="true" tabindex="-1"></a>    <span class="co">#ax.set_title(name)</span></span>
<span id="cb72-20"><a href="#cb72-20" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb72-21"><a href="#cb72-21" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create stairs</span></span>
<span id="cb72-22"><a href="#cb72-22" aria-hidden="true" tabindex="-1"></a>    x <span class="op">=</span> np.zeros(<span class="dv">2</span><span class="op">*</span>chrom_start.size)</span>
<span id="cb72-23"><a href="#cb72-23" aria-hidden="true" tabindex="-1"></a>    y <span class="op">=</span> np.zeros(<span class="dv">2</span><span class="op">*</span>chrom_start.size)</span>
<span id="cb72-24"><a href="#cb72-24" aria-hidden="true" tabindex="-1"></a>    x[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> chrom_start</span>
<span id="cb72-25"><a href="#cb72-25" aria-hidden="true" tabindex="-1"></a>    x[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> chrom_end</span>
<span id="cb72-26"><a href="#cb72-26" aria-hidden="true" tabindex="-1"></a>    y[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> e1</span>
<span id="cb72-27"><a href="#cb72-27" aria-hidden="true" tabindex="-1"></a>    y[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> e1</span>
<span id="cb72-28"><a href="#cb72-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-29"><a href="#cb72-29" aria-hidden="true" tabindex="-1"></a>    ax.fill_between(x, y, <span class="dv">0</span>, where<span class="op">=</span>(y <span class="op">&gt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:red'</span>)</span>
<span id="cb72-30"><a href="#cb72-30" aria-hidden="true" tabindex="-1"></a>    ax.fill_between(x, y, <span class="dv">0</span>, where<span class="op">=</span>(y <span class="op">&lt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:blue'</span>)</span>
<span id="cb72-31"><a href="#cb72-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-32"><a href="#cb72-32" aria-hidden="true" tabindex="-1"></a><span class="co"># Populate the second column</span></span>
<span id="cb72-33"><a href="#cb72-33" aria-hidden="true" tabindex="-1"></a>axs[<span class="dv">0</span>,<span class="dv">1</span>].set_title(<span class="st">'E1: Chromosome-arms restricted'</span>)</span>
<span id="cb72-34"><a href="#cb72-34" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i, (name, e1) <span class="kw">in</span> <span class="bu">enumerate</span>(e1_values.items()):</span>
<span id="cb72-35"><a href="#cb72-35" aria-hidden="true" tabindex="-1"></a>    ax <span class="op">=</span> axs[i,<span class="dv">1</span>]</span>
<span id="cb72-36"><a href="#cb72-36" aria-hidden="true" tabindex="-1"></a>    <span class="co">#ax.set_title(name)</span></span>
<span id="cb72-37"><a href="#cb72-37" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb72-38"><a href="#cb72-38" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create stairs</span></span>
<span id="cb72-39"><a href="#cb72-39" aria-hidden="true" tabindex="-1"></a>    x <span class="op">=</span> np.zeros(<span class="dv">2</span><span class="op">*</span>chrom_start.size)</span>
<span id="cb72-40"><a href="#cb72-40" aria-hidden="true" tabindex="-1"></a>    y <span class="op">=</span> np.zeros(<span class="dv">2</span><span class="op">*</span>chrom_start.size)</span>
<span id="cb72-41"><a href="#cb72-41" aria-hidden="true" tabindex="-1"></a>    x[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> chrom_start</span>
<span id="cb72-42"><a href="#cb72-42" aria-hidden="true" tabindex="-1"></a>    x[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> chrom_end</span>
<span id="cb72-43"><a href="#cb72-43" aria-hidden="true" tabindex="-1"></a>    y[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> e1</span>
<span id="cb72-44"><a href="#cb72-44" aria-hidden="true" tabindex="-1"></a>    y[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> e1</span>
<span id="cb72-45"><a href="#cb72-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-46"><a href="#cb72-46" aria-hidden="true" tabindex="-1"></a>    ax.fill_between(x, y, <span class="dv">0</span>, where<span class="op">=</span>(y <span class="op">&gt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:red'</span>)</span>
<span id="cb72-47"><a href="#cb72-47" aria-hidden="true" tabindex="-1"></a>    ax.fill_between(x, y, <span class="dv">0</span>, where<span class="op">=</span>(y <span class="op">&lt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:blue'</span>)</span>
<span id="cb72-48"><a href="#cb72-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-49"><a href="#cb72-49" aria-hidden="true" tabindex="-1"></a><span class="co"># Populate the third column</span></span>
<span id="cb72-50"><a href="#cb72-50" aria-hidden="true" tabindex="-1"></a>axs[<span class="dv">0</span>,<span class="dv">2</span>].set_title(<span class="st">'E1: 10Mb restricted (refined)'</span>)</span>
<span id="cb72-51"><a href="#cb72-51" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i, (name, e1) <span class="kw">in</span> <span class="bu">enumerate</span>(e1_values_10mb.items()):</span>
<span id="cb72-52"><a href="#cb72-52" aria-hidden="true" tabindex="-1"></a>    ax <span class="op">=</span> axs[i,<span class="dv">2</span>]</span>
<span id="cb72-53"><a href="#cb72-53" aria-hidden="true" tabindex="-1"></a>    <span class="co">#ax.set_title(name)</span></span>
<span id="cb72-54"><a href="#cb72-54" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb72-55"><a href="#cb72-55" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create stairs</span></span>
<span id="cb72-56"><a href="#cb72-56" aria-hidden="true" tabindex="-1"></a>    x <span class="op">=</span> np.zeros(<span class="dv">2</span><span class="op">*</span>chrom_start.size)</span>
<span id="cb72-57"><a href="#cb72-57" aria-hidden="true" tabindex="-1"></a>    y <span class="op">=</span> np.zeros(<span class="dv">2</span><span class="op">*</span>chrom_start.size)</span>
<span id="cb72-58"><a href="#cb72-58" aria-hidden="true" tabindex="-1"></a>    x[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> chrom_start</span>
<span id="cb72-59"><a href="#cb72-59" aria-hidden="true" tabindex="-1"></a>    x[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> chrom_end</span>
<span id="cb72-60"><a href="#cb72-60" aria-hidden="true" tabindex="-1"></a>    y[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> e1</span>
<span id="cb72-61"><a href="#cb72-61" aria-hidden="true" tabindex="-1"></a>    y[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> e1</span>
<span id="cb72-62"><a href="#cb72-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-63"><a href="#cb72-63" aria-hidden="true" tabindex="-1"></a>    ax.fill_between(x, y, <span class="dv">0</span>, where<span class="op">=</span>(y <span class="op">&gt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:red'</span>)</span>
<span id="cb72-64"><a href="#cb72-64" aria-hidden="true" tabindex="-1"></a>    ax.fill_between(x, y, <span class="dv">0</span>, where<span class="op">=</span>(y <span class="op">&lt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:blue'</span>)</span>
<span id="cb72-65"><a href="#cb72-65" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb72-66"><a href="#cb72-66" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb72-67"><a href="#cb72-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-68"><a href="#cb72-68" aria-hidden="true" tabindex="-1"></a><span class="co"># Set y-limits for all subplots</span></span>
<span id="cb72-69"><a href="#cb72-69" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> ax <span class="kw">in</span> axs.flat:</span>
<span id="cb72-70"><a href="#cb72-70" aria-hidden="true" tabindex="-1"></a>    ax.set_ylim(<span class="op">-</span><span class="fl">0.8</span>, <span class="fl">0.8</span>)</span>
<span id="cb72-71"><a href="#cb72-71" aria-hidden="true" tabindex="-1"></a>    ax.spines[:].set_visible(<span class="va">False</span>)</span>
<span id="cb72-72"><a href="#cb72-72" aria-hidden="true" tabindex="-1"></a>    ax.set_xticks([])</span>
<span id="cb72-73"><a href="#cb72-73" aria-hidden="true" tabindex="-1"></a>    ax.set_yticks([])</span>
<span id="cb72-74"><a href="#cb72-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-75"><a href="#cb72-75" aria-hidden="true" tabindex="-1"></a>plt.tight_layout()</span>
<span id="cb72-76"><a href="#cb72-76" aria-hidden="true" tabindex="-1"></a>plt.savefig(<span class="st">'../steps/e1_plot_full.svg'</span>, bbox_inches<span class="op">=</span><span class="st">'tight'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display page-columns page-full">
<div id="e1-fullmerged-all-500kb" class="quarto-figure quarto-figure-center anchored page-columns page-full">
<figure class="figure page-columns page-full">
<p class="page-columns page-full"><img src="03_compartments_files/figure-html/e1-fullmerged-all-500kb-output-1.png" class="img-fluid figure-img column-page-inset"></p>
<figcaption>E1 eigenvector values for all merged samples at 500kb resolution. Left: Chromosome-arm restricted E1. Right: 10Mb window restricted E1.</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="plot-matrices-with-compartments-round-spermatid" class="level4">
<h4 class="anchored" data-anchor-id="plot-matrices-with-compartments-round-spermatid">Plot matrices with compartments (round spermatid)</h4>
<div id="cell-e1-matrix-500kb-full-arms-10mb-round_spermatid" class="cell" data-execution_count="55">
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooltools.lib.plotting</span>
<span id="cb73-2"><a href="#cb73-2" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> matplotlib.colors <span class="im">import</span> LogNorm</span>
<span id="cb73-3"><a href="#cb73-3" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> mpl_toolkits.axes_grid1 <span class="im">import</span> make_axes_locatable</span>
<span id="cb73-4"><a href="#cb73-4" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> cooltools.lib.numutils <span class="im">import</span> adaptive_coarsegrain, interp_nan</span>
<span id="cb73-5"><a href="#cb73-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-6"><a href="#cb73-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-7"><a href="#cb73-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-8"><a href="#cb73-8" aria-hidden="true" tabindex="-1"></a>clr <span class="op">=</span> clrs[<span class="st">'round_spermatid'</span>]</span>
<span id="cb73-9"><a href="#cb73-9" aria-hidden="true" tabindex="-1"></a>chrom_start, chrom_end <span class="op">=</span> e1_track_full[<span class="st">'start'</span>].values, e1_track_full[<span class="st">'end'</span>].values<span class="op">-</span><span class="dv">1</span></span>
<span id="cb73-10"><a href="#cb73-10" aria-hidden="true" tabindex="-1"></a>e1 <span class="op">=</span> e1_values_10mb[<span class="st">'round_spermatid'</span>]</span>
<span id="cb73-11"><a href="#cb73-11" aria-hidden="true" tabindex="-1"></a>nbins <span class="op">=</span> <span class="bu">len</span>(clr.bins().fetch(<span class="st">'chrX'</span>))</span>
<span id="cb73-12"><a href="#cb73-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-13"><a href="#cb73-13" aria-hidden="true" tabindex="-1"></a>f, ax <span class="op">=</span> plt.subplots(</span>
<span id="cb73-14"><a href="#cb73-14" aria-hidden="true" tabindex="-1"></a>    figsize<span class="op">=</span>(<span class="dv">9</span>,<span class="dv">6</span>),</span>
<span id="cb73-15"><a href="#cb73-15" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb73-16"><a href="#cb73-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-17"><a href="#cb73-17" aria-hidden="true" tabindex="-1"></a>norm <span class="op">=</span> LogNorm(vmax<span class="op">=</span><span class="fl">0.1</span>)</span>
<span id="cb73-18"><a href="#cb73-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-19"><a href="#cb73-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-20"><a href="#cb73-20" aria-hidden="true" tabindex="-1"></a>e1_list <span class="op">=</span> [e1_values_10mb[<span class="st">'round_spermatid'</span>], e1_values[<span class="st">'round_spermatid'</span>], e1_values_full[<span class="st">'round_spermatid'</span>]]</span>
<span id="cb73-21"><a href="#cb73-21" aria-hidden="true" tabindex="-1"></a>e1_names <span class="op">=</span> [<span class="st">'10Mb'</span>, <span class="st">'Arms'</span>, <span class="st">'Full'</span>]</span>
<span id="cb73-22"><a href="#cb73-22" aria-hidden="true" tabindex="-1"></a>colors <span class="op">=</span> [<span class="st">'tab:red'</span>, <span class="st">'tab:blue'</span>, <span class="st">'tab:green'</span>]</span>
<span id="cb73-23"><a href="#cb73-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-24"><a href="#cb73-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-25"><a href="#cb73-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-26"><a href="#cb73-26" aria-hidden="true" tabindex="-1"></a>im <span class="op">=</span> ax.matshow(</span>
<span id="cb73-27"><a href="#cb73-27" aria-hidden="true" tabindex="-1"></a>    clr.matrix().fetch(<span class="st">'chrX'</span>),</span>
<span id="cb73-28"><a href="#cb73-28" aria-hidden="true" tabindex="-1"></a>    norm<span class="op">=</span>norm,</span>
<span id="cb73-29"><a href="#cb73-29" aria-hidden="true" tabindex="-1"></a>    cmap<span class="op">=</span><span class="st">'fall'</span>,</span>
<span id="cb73-30"><a href="#cb73-30" aria-hidden="true" tabindex="-1"></a>)<span class="op">;</span></span>
<span id="cb73-31"><a href="#cb73-31" aria-hidden="true" tabindex="-1"></a>plt.axis([<span class="dv">0</span>,nbins,nbins,<span class="dv">0</span>])</span>
<span id="cb73-32"><a href="#cb73-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-33"><a href="#cb73-33" aria-hidden="true" tabindex="-1"></a>divider <span class="op">=</span> make_axes_locatable(ax)</span>
<span id="cb73-34"><a href="#cb73-34" aria-hidden="true" tabindex="-1"></a>cax <span class="op">=</span> divider.append_axes(<span class="st">"right"</span>, size<span class="op">=</span><span class="st">"2%"</span>, pad<span class="op">=</span><span class="fl">0.05</span>)</span>
<span id="cb73-35"><a href="#cb73-35" aria-hidden="true" tabindex="-1"></a>plt.colorbar(im, cax<span class="op">=</span>cax, label<span class="op">=</span><span class="st">'corrected frequencies'</span>)<span class="op">;</span></span>
<span id="cb73-36"><a href="#cb73-36" aria-hidden="true" tabindex="-1"></a>ax.set_ylabel(<span class="st">'chrX:500kb bins, #bin'</span>)</span>
<span id="cb73-37"><a href="#cb73-37" aria-hidden="true" tabindex="-1"></a>ax.xaxis.set_visible(<span class="va">False</span>)</span>
<span id="cb73-38"><a href="#cb73-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-39"><a href="#cb73-39" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i, e1 <span class="kw">in</span> <span class="bu">enumerate</span>(e1_list):</span>
<span id="cb73-40"><a href="#cb73-40" aria-hidden="true" tabindex="-1"></a>    ax1 <span class="op">=</span> divider.append_axes(<span class="st">"top"</span>, size<span class="op">=</span><span class="st">"10%"</span>, pad<span class="op">=</span><span class="fl">0.05</span>, sharex <span class="op">=</span> ax)</span>
<span id="cb73-41"><a href="#cb73-41" aria-hidden="true" tabindex="-1"></a>    <span class="co"># weights = clr.bins()[:]['weight'].values</span></span>
<span id="cb73-42"><a href="#cb73-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-43"><a href="#cb73-43" aria-hidden="true" tabindex="-1"></a>    <span class="co"># ax1.plot(e1, label='E1')</span></span>
<span id="cb73-44"><a href="#cb73-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-45"><a href="#cb73-45" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create stairs</span></span>
<span id="cb73-46"><a href="#cb73-46" aria-hidden="true" tabindex="-1"></a>    x <span class="op">=</span> np.zeros(<span class="dv">2</span><span class="op">*</span>chrom_start.size)</span>
<span id="cb73-47"><a href="#cb73-47" aria-hidden="true" tabindex="-1"></a>    y <span class="op">=</span> np.zeros(<span class="dv">2</span><span class="op">*</span>chrom_start.size)</span>
<span id="cb73-48"><a href="#cb73-48" aria-hidden="true" tabindex="-1"></a>    <span class="co">#smooth_y = np.zeros(2*chrom_start.size)</span></span>
<span id="cb73-49"><a href="#cb73-49" aria-hidden="true" tabindex="-1"></a>    x[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> chrom_start<span class="op">/</span><span class="dv">500_000</span></span>
<span id="cb73-50"><a href="#cb73-50" aria-hidden="true" tabindex="-1"></a>    x[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> chrom_end<span class="op">/</span><span class="dv">500_000</span></span>
<span id="cb73-51"><a href="#cb73-51" aria-hidden="true" tabindex="-1"></a>    y[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> e1</span>
<span id="cb73-52"><a href="#cb73-52" aria-hidden="true" tabindex="-1"></a>    y[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> e1</span>
<span id="cb73-53"><a href="#cb73-53" aria-hidden="true" tabindex="-1"></a>    <span class="co"># smooth_e1 = (lambda x: x.rolling(5, 1, center=True).sum())(pd.DataFrame(e1, columns=['value']))['value'].values</span></span>
<span id="cb73-54"><a href="#cb73-54" aria-hidden="true" tabindex="-1"></a>    <span class="co"># smooth_y[0::2] = smooth_e1</span></span>
<span id="cb73-55"><a href="#cb73-55" aria-hidden="true" tabindex="-1"></a>    <span class="co"># smooth_y[1::2] = smooth_e1</span></span>
<span id="cb73-56"><a href="#cb73-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-57"><a href="#cb73-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-58"><a href="#cb73-58" aria-hidden="true" tabindex="-1"></a>    ax1.fill_between(x, y, <span class="dv">0</span>, where<span class="op">=</span>(y <span class="op">&gt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:red'</span>)</span>
<span id="cb73-59"><a href="#cb73-59" aria-hidden="true" tabindex="-1"></a>    ax1.fill_between(x, y, <span class="dv">0</span>, where<span class="op">=</span>(y <span class="op">&lt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:blue'</span>)</span>
<span id="cb73-60"><a href="#cb73-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-61"><a href="#cb73-61" aria-hidden="true" tabindex="-1"></a>    ax1.set_ylabel(e1_names[i], rotation<span class="op">=</span><span class="dv">70</span>)</span>
<span id="cb73-62"><a href="#cb73-62" aria-hidden="true" tabindex="-1"></a>    ax1.set_ylim(<span class="op">-</span><span class="fl">0.8</span>, <span class="fl">0.8</span>)</span>
<span id="cb73-63"><a href="#cb73-63" aria-hidden="true" tabindex="-1"></a>    ax1.set_xticks([])</span>
<span id="cb73-64"><a href="#cb73-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-65"><a href="#cb73-65" aria-hidden="true" tabindex="-1"></a>ax1.set_title(<span class="st">'Round Spermatid: 500kb bins'</span>)</span>
<span id="cb73-66"><a href="#cb73-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-67"><a href="#cb73-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-68"><a href="#cb73-68" aria-hidden="true" tabindex="-1"></a><span class="co">#    # Plot the sign changes on the matrix </span></span>
<span id="cb73-69"><a href="#cb73-69" aria-hidden="true" tabindex="-1"></a><span class="co">#     col = colors[i]</span></span>
<span id="cb73-70"><a href="#cb73-70" aria-hidden="true" tabindex="-1"></a><span class="co">#     for i in np.where(np.diff( (pd.Series(e1)&gt;0).astype(int)))[0]:</span></span>
<span id="cb73-71"><a href="#cb73-71" aria-hidden="true" tabindex="-1"></a><span class="co">#         # Horisontal lines where E1 intersects 0</span></span>
<span id="cb73-72"><a href="#cb73-72" aria-hidden="true" tabindex="-1"></a><span class="co">#         ax.plot([0,nbins],[i,i],col,lw=0.5)</span></span>
<span id="cb73-73"><a href="#cb73-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-74"><a href="#cb73-74" aria-hidden="true" tabindex="-1"></a><span class="co">#         # Vertical lines where E1 intersects 0</span></span>
<span id="cb73-75"><a href="#cb73-75" aria-hidden="true" tabindex="-1"></a><span class="co">#         ax.plot([i,i],[0,nbins],col,lw=0.5)</span></span>
<span id="cb73-76"><a href="#cb73-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-77"><a href="#cb73-77" aria-hidden="true" tabindex="-1"></a>plt.tight_layout()</span>
<span id="cb73-78"><a href="#cb73-78" aria-hidden="true" tabindex="-1"></a>f.canvas.draw()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div id="e1-matrix-500kb-full-arms-10mb-round_spermatid" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="03_compartments_files/figure-html/e1-matrix-500kb-full-arms-10mb-round_spermatid-output-1.png" class="img-fluid figure-img"></p>
<figcaption>E1 eigenvector values for all merged samples at 500kb resolution for round spermatid, as well as the interaction matrix. E1 was restricted to top: Full-chromosome, middle: Chromosome-arms, bottom: 10Mb window.</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="sliding-window-summed-e1-compartments" class="level4 page-columns page-full">
<h4 class="anchored" data-anchor-id="sliding-window-summed-e1-compartments">Sliding window summed E1 compartments</h4>
<p>To mimic the smoothing applied in the Wang et al.&nbsp;2019 paper, where they slide a 400kb window in 100kb steps on the obs/exp matrix, we will similarly slide a 400kb window in 100kb steps directly on the E1 compartments.</p>
<div id="cell-e1-fullmerged-all-500kb-running-avg" class="cell page-columns page-full">
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb74-3"><a href="#cb74-3" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> matplotlib.colors <span class="im">import</span> LogNorm</span>
<span id="cb74-4"><a href="#cb74-4" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> mpl_toolkits.axes_grid1 <span class="im">import</span> make_axes_locatable</span>
<span id="cb74-5"><a href="#cb74-5" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb74-6"><a href="#cb74-6" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> scipy.signal.windows <span class="im">import</span> triang</span>
<span id="cb74-7"><a href="#cb74-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-8"><a href="#cb74-8" aria-hidden="true" tabindex="-1"></a>resolution <span class="op">=</span> <span class="dv">500_000</span></span>
<span id="cb74-9"><a href="#cb74-9" aria-hidden="true" tabindex="-1"></a>window_size <span class="op">=</span> <span class="dv">2_500_000</span></span>
<span id="cb74-10"><a href="#cb74-10" aria-hidden="true" tabindex="-1"></a>step_size <span class="op">=</span> window_size <span class="op">//</span> resolution</span>
<span id="cb74-11"><a href="#cb74-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-12"><a href="#cb74-12" aria-hidden="true" tabindex="-1"></a>chrom_start <span class="op">=</span> e1_track[<span class="st">'start'</span>].values</span>
<span id="cb74-13"><a href="#cb74-13" aria-hidden="true" tabindex="-1"></a>chrom_end <span class="op">=</span> e1_track[<span class="st">'end'</span>].values<span class="op">-</span><span class="dv">1</span></span>
<span id="cb74-14"><a href="#cb74-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-15"><a href="#cb74-15" aria-hidden="true" tabindex="-1"></a>f, axs <span class="op">=</span> plt.subplots(<span class="dv">5</span>, <span class="dv">2</span>, figsize<span class="op">=</span>(<span class="dv">30</span>, <span class="dv">10</span>), sharex<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb74-16"><a href="#cb74-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-17"><a href="#cb74-17" aria-hidden="true" tabindex="-1"></a>axs[<span class="dv">0</span>, <span class="dv">0</span>].set_title(<span class="st">'Chromosome-arm E1 (arm restricted)'</span>)</span>
<span id="cb74-18"><a href="#cb74-18" aria-hidden="true" tabindex="-1"></a>axs[<span class="dv">0</span>, <span class="dv">1</span>].set_title(<span class="st">'Smoothed by summation'</span>)</span>
<span id="cb74-19"><a href="#cb74-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-20"><a href="#cb74-20" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i, (name, e1) <span class="kw">in</span> <span class="bu">enumerate</span>(e1_values.items()):</span>
<span id="cb74-21"><a href="#cb74-21" aria-hidden="true" tabindex="-1"></a>    <span class="co"># print(i, name, e1.size)</span></span>
<span id="cb74-22"><a href="#cb74-22" aria-hidden="true" tabindex="-1"></a>    smooth_e1 <span class="op">=</span> (<span class="kw">lambda</span> x: x.rolling(<span class="dv">5</span>, <span class="dv">1</span>, center<span class="op">=</span><span class="va">True</span>).<span class="bu">sum</span>())(pd.DataFrame(e1, columns<span class="op">=</span>[<span class="st">'value'</span>]))</span>
<span id="cb74-23"><a href="#cb74-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-24"><a href="#cb74-24" aria-hidden="true" tabindex="-1"></a>    ax0 <span class="op">=</span> axs[i, <span class="dv">0</span>]</span>
<span id="cb74-25"><a href="#cb74-25" aria-hidden="true" tabindex="-1"></a>    ax1 <span class="op">=</span> axs[i, <span class="dv">1</span>]</span>
<span id="cb74-26"><a href="#cb74-26" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb74-27"><a href="#cb74-27" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create stairs</span></span>
<span id="cb74-28"><a href="#cb74-28" aria-hidden="true" tabindex="-1"></a>    x <span class="op">=</span> np.zeros(<span class="dv">2</span><span class="op">*</span>chrom_start.size)</span>
<span id="cb74-29"><a href="#cb74-29" aria-hidden="true" tabindex="-1"></a>    y <span class="op">=</span> np.zeros(<span class="dv">2</span><span class="op">*</span>chrom_start.size)</span>
<span id="cb74-30"><a href="#cb74-30" aria-hidden="true" tabindex="-1"></a>    smooth_y <span class="op">=</span> np.zeros(<span class="dv">2</span><span class="op">*</span>chrom_start.size)</span>
<span id="cb74-31"><a href="#cb74-31" aria-hidden="true" tabindex="-1"></a>    x[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> chrom_start</span>
<span id="cb74-32"><a href="#cb74-32" aria-hidden="true" tabindex="-1"></a>    x[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> chrom_end</span>
<span id="cb74-33"><a href="#cb74-33" aria-hidden="true" tabindex="-1"></a>    y[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> e1</span>
<span id="cb74-34"><a href="#cb74-34" aria-hidden="true" tabindex="-1"></a>    y[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> e1</span>
<span id="cb74-35"><a href="#cb74-35" aria-hidden="true" tabindex="-1"></a>    smooth_y[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> smooth_e1[<span class="st">'value'</span>].values</span>
<span id="cb74-36"><a href="#cb74-36" aria-hidden="true" tabindex="-1"></a>    smooth_y[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> smooth_e1[<span class="st">'value'</span>].values</span>
<span id="cb74-37"><a href="#cb74-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-38"><a href="#cb74-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-39"><a href="#cb74-39" aria-hidden="true" tabindex="-1"></a>    ax0.fill_between(x, y, <span class="dv">0</span>, where<span class="op">=</span>(y <span class="op">&gt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:red'</span>)</span>
<span id="cb74-40"><a href="#cb74-40" aria-hidden="true" tabindex="-1"></a>    ax0.fill_between(x, y, <span class="dv">0</span>, where<span class="op">=</span>(y <span class="op">&lt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:blue'</span>)</span>
<span id="cb74-41"><a href="#cb74-41" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb74-42"><a href="#cb74-42" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Overlay the smoothed line (divided by 5 to make it a mean)</span></span>
<span id="cb74-43"><a href="#cb74-43" aria-hidden="true" tabindex="-1"></a>    ax0.plot(x, smooth_y<span class="op">/</span><span class="dv">5</span>, color<span class="op">=</span><span class="st">'C1'</span>)</span>
<span id="cb74-44"><a href="#cb74-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-45"><a href="#cb74-45" aria-hidden="true" tabindex="-1"></a>    ax1.fill_between(x, smooth_y, <span class="dv">0</span>, where<span class="op">=</span>(smooth_y <span class="op">&gt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:red'</span>)</span>
<span id="cb74-46"><a href="#cb74-46" aria-hidden="true" tabindex="-1"></a>    ax1.fill_between(x, smooth_y, <span class="dv">0</span>, where<span class="op">=</span>(smooth_y <span class="op">&lt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:blue'</span>)</span>
<span id="cb74-47"><a href="#cb74-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-48"><a href="#cb74-48" aria-hidden="true" tabindex="-1"></a>    ax0.set_ylabel(name)</span>
<span id="cb74-49"><a href="#cb74-49" aria-hidden="true" tabindex="-1"></a>    ylim <span class="op">=</span> <span class="fl">1.5</span></span>
<span id="cb74-50"><a href="#cb74-50" aria-hidden="true" tabindex="-1"></a>    <span class="co">#ax0.set_ylim(-ylim, ylim)</span></span>
<span id="cb74-51"><a href="#cb74-51" aria-hidden="true" tabindex="-1"></a>    <span class="co">#ax1.set_ylim(-ylim*4, ylim*4)</span></span>
<span id="cb74-52"><a href="#cb74-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-53"><a href="#cb74-53" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> ax <span class="kw">in</span> axs.flat:</span>
<span id="cb74-54"><a href="#cb74-54" aria-hidden="true" tabindex="-1"></a>    ax.spines[:].set_visible(<span class="va">False</span>)</span>
<span id="cb74-55"><a href="#cb74-55" aria-hidden="true" tabindex="-1"></a>    ax.set_xticks([])</span>
<span id="cb74-56"><a href="#cb74-56" aria-hidden="true" tabindex="-1"></a>    <span class="co">#ax.set_yticks([])</span></span>
<span id="cb74-57"><a href="#cb74-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-58"><a href="#cb74-58" aria-hidden="true" tabindex="-1"></a><span class="co"># plt.tight_layout()</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display page-columns page-full">
<div id="e1-fullmerged-all-500kb-running-avg" class="quarto-figure quarto-figure-center anchored page-columns page-full">
<figure class="figure page-columns page-full">
<p class="page-columns page-full"><img src="03_compartments_files/figure-html/e1-fullmerged-all-500kb-running-avg-output-1.png" class="img-fluid figure-img column-body-outset"></p>
<figcaption>E1 eigenvector values for all merged samples at 500kb resolution with rolling summation, with window size 2.5Mb, step size 500Kb: Each value is now the sum of the surrounding n=5 bins.</figcaption>
</figure>
</div>
</div>
</div>
</section>
</section>
</section>
<section id="kb-resolution-1" class="level2 page-columns page-full">
<h2 class="anchored" data-anchor-id="kb-resolution-1">100kb resolution</h2>
<section id="all-5-full-merges-1" class="level3 page-columns page-full">
<h3 class="anchored" data-anchor-id="all-5-full-merges-1">All 5 full merges</h3>
<section id="load-coolers-1" class="level4">
<h4 class="anchored" data-anchor-id="load-coolers-1">Load coolers</h4>
<div id="cell-78" class="cell" data-execution_count="1">
<div class="sourceCode cell-code" id="cb75"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> glob</span>
<span id="cb75-2"><a href="#cb75-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os.path <span class="im">as</span> op</span>
<span id="cb75-3"><a href="#cb75-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooler</span>
<span id="cb75-4"><a href="#cb75-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-5"><a href="#cb75-5" aria-hidden="true" tabindex="-1"></a>mcools <span class="op">=</span> glob.glob(<span class="st">"../steps/bwa/PE/cool/*/*.mcool"</span>)</span>
<span id="cb75-6"><a href="#cb75-6" aria-hidden="true" tabindex="-1"></a>res <span class="op">=</span> <span class="st">"::resolutions/100000"</span></span>
<span id="cb75-7"><a href="#cb75-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-8"><a href="#cb75-8" aria-hidden="true" tabindex="-1"></a>clrs <span class="op">=</span> {op.basename(op.dirname(mcool)): cooler.Cooler(mcool<span class="op">+</span>res) <span class="cf">for</span> mcool <span class="kw">in</span> mcools}</span>
<span id="cb75-9"><a href="#cb75-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-10"><a href="#cb75-10" aria-hidden="true" tabindex="-1"></a>chron_order <span class="op">=</span> [<span class="st">'fibroblast'</span>, <span class="st">'spermatogonia'</span>, <span class="st">'pachytene_spermatocyte'</span>, <span class="st">'round_spermatid'</span>, <span class="st">'sperm'</span>]</span>
<span id="cb75-11"><a href="#cb75-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-12"><a href="#cb75-12" aria-hidden="true" tabindex="-1"></a>clrs <span class="op">=</span> {key: clrs[key] <span class="cf">for</span> key <span class="kw">in</span> chron_order}</span>
<span id="cb75-13"><a href="#cb75-13" aria-hidden="true" tabindex="-1"></a>clrs</span>
<span id="cb75-14"><a href="#cb75-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-15"><a href="#cb75-15" aria-hidden="true" tabindex="-1"></a>names_abbr <span class="op">=</span> {<span class="st">'fibroblast'</span>: <span class="st">'Fibroblast'</span>, <span class="st">'spermatogonia'</span>: <span class="st">'Spermatogonia'</span>, <span class="st">'pachytene_spermatocyte'</span>: <span class="st">'Pachytene Spermatogonia'</span>, <span class="st">'round_spermatid'</span>: <span class="st">'Round Spermatid'</span>, <span class="st">'sperm'</span>: <span class="st">'Sperm'</span>}</span>
<span id="cb75-16"><a href="#cb75-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-17"><a href="#cb75-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate chromstart and chromend for each bin on chrX</span></span>
<span id="cb75-18"><a href="#cb75-18" aria-hidden="true" tabindex="-1"></a>chrX_size <span class="op">=</span> clrs[<span class="st">'fibroblast'</span>].chromsizes[<span class="st">'chrX'</span>]</span>
<span id="cb75-19"><a href="#cb75-19" aria-hidden="true" tabindex="-1"></a>chrom_start <span class="op">=</span> clrs[<span class="st">'fibroblast'</span>].bins().fetch(<span class="st">'chrX'</span>)[<span class="st">'start'</span>].values</span>
<span id="cb75-20"><a href="#cb75-20" aria-hidden="true" tabindex="-1"></a>chrom_end <span class="op">=</span> clrs[<span class="st">'fibroblast'</span>].bins().fetch(<span class="st">'chrX'</span>)[<span class="st">'end'</span>].values<span class="op">-</span><span class="dv">1</span></span>
<span id="cb75-21"><a href="#cb75-21" aria-hidden="true" tabindex="-1"></a>nbins <span class="op">=</span> <span class="bu">len</span>(clrs[<span class="st">'fibroblast'</span>].bins().fetch(<span class="st">'chrX'</span>))</span>
<span id="cb75-22"><a href="#cb75-22" aria-hidden="true" tabindex="-1"></a>binsize <span class="op">=</span> clrs[<span class="st">'fibroblast'</span>].binsize</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="calculate-gc-covariance-from-the-reference-genome-2" class="level4">
<h4 class="anchored" data-anchor-id="calculate-gc-covariance-from-the-reference-genome-2">Calculate gc covariance (from the reference genome)</h4>
<p>Do this with any of the clrs - it just needs the bins positions.</p>
<div id="cell-80" class="cell" data-execution_count="2">
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Try with only the gc_cov for chrX</span></span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-3"><a href="#cb76-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> bioframe</span>
<span id="cb76-4"><a href="#cb76-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb76-5"><a href="#cb76-5" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os.path <span class="im">as</span> op</span>
<span id="cb76-6"><a href="#cb76-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-7"><a href="#cb76-7" aria-hidden="true" tabindex="-1"></a>bins <span class="op">=</span> clrs[<span class="st">'fibroblast'</span>].bins().fetch(<span class="st">'chrX'</span>)[:]</span>
<span id="cb76-8"><a href="#cb76-8" aria-hidden="true" tabindex="-1"></a>out_name <span class="op">=</span> <span class="st">'../steps/rheMac10_gc_cov_X_100kb.tsv'</span></span>
<span id="cb76-9"><a href="#cb76-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-10"><a href="#cb76-10" aria-hidden="true" tabindex="-1"></a>rheMac10 <span class="op">=</span> bioframe.load_fasta(<span class="st">'../data/links/ucsc_ref/rheMac10.fa'</span>)</span>
<span id="cb76-11"><a href="#cb76-11" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> <span class="kw">not</span> op.exists(out_name):</span>
<span id="cb76-12"><a href="#cb76-12" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">'Calculate the fraction of GC basepairs for each bin'</span>)</span>
<span id="cb76-13"><a href="#cb76-13" aria-hidden="true" tabindex="-1"></a>    gc_cov <span class="op">=</span> bioframe.frac_gc(bins[[<span class="st">'chrom'</span>, <span class="st">'start'</span>, <span class="st">'end'</span>]], rheMac10)</span>
<span id="cb76-14"><a href="#cb76-14" aria-hidden="true" tabindex="-1"></a>    gc_cov.to_csv(out_name, index<span class="op">=</span><span class="va">False</span>,sep<span class="op">=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>)</span>
<span id="cb76-15"><a href="#cb76-15" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(gc_cov.info())</span>
<span id="cb76-16"><a href="#cb76-16" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>: </span>
<span id="cb76-17"><a href="#cb76-17" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"Already exists, read from file"</span>)</span>
<span id="cb76-18"><a href="#cb76-18" aria-hidden="true" tabindex="-1"></a>    gc_cov <span class="op">=</span> pd.read_csv(out_name, sep<span class="op">=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>)</span>
<span id="cb76-19"><a href="#cb76-19" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(gc_cov.info())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Already exists, read from file
&lt;class 'pandas.core.frame.DataFrame'&gt;
RangeIndex: 1534 entries, 0 to 1533
Data columns (total 4 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   chrom   1534 non-null   object 
 1   start   1534 non-null   int64  
 2   end     1534 non-null   int64  
 3   GC      1533 non-null   float64
dtypes: float64(1), int64(2), object(1)
memory usage: 48.1+ KB
None</code></pre>
</div>
</div>
</section>
<section id="calculate-the-e1-compartments-1" class="level4">
<h4 class="anchored" data-anchor-id="calculate-the-e1-compartments-1">Calculate the E1 compartments</h4>
<p>Loop: view_df, cis_eigs, e1_values</p>
<section id="plot-gc-covariance-1" class="level5">
<h5 class="anchored" data-anchor-id="plot-gc-covariance-1">Plot GC covariance</h5>
<div id="cell-83" class="cell" data-execution_count="3">
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-3"><a href="#cb78-3" aria-hidden="true" tabindex="-1"></a>f, ax <span class="op">=</span> plt.subplots(figsize<span class="op">=</span>(<span class="dv">10</span>, <span class="dv">2</span>))</span>
<span id="cb78-4"><a href="#cb78-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-5"><a href="#cb78-5" aria-hidden="true" tabindex="-1"></a>ax.plot(gc_cov[<span class="st">'start'</span>],gc_cov[<span class="st">'GC'</span>])</span>
<span id="cb78-6"><a href="#cb78-6" aria-hidden="true" tabindex="-1"></a>plt.tight_layout()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="03_compartments_files/figure-html/cell-46-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="create-viewframes-full-arms-10mb-windows" class="level5">
<h5 class="anchored" data-anchor-id="create-viewframes-full-arms-10mb-windows">Create viewframes: full, arms, 10Mb windows</h5>
<div id="cell-85" class="cell" data-execution_count="4">
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-3"><a href="#cb79-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Fetch the chromsize of X from one of the coolers</span></span>
<span id="cb79-4"><a href="#cb79-4" aria-hidden="true" tabindex="-1"></a>chrX_size <span class="op">=</span> clrs[<span class="st">'fibroblast'</span>].chromsizes[<span class="st">'chrX'</span>]</span>
<span id="cb79-5"><a href="#cb79-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-6"><a href="#cb79-6" aria-hidden="true" tabindex="-1"></a>views <span class="op">=</span> {}</span>
<span id="cb79-7"><a href="#cb79-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-8"><a href="#cb79-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Make the full view frame</span></span>
<span id="cb79-9"><a href="#cb79-9" aria-hidden="true" tabindex="-1"></a>views[<span class="st">'full'</span>] <span class="op">=</span> pd.DataFrame({</span>
<span id="cb79-10"><a href="#cb79-10" aria-hidden="true" tabindex="-1"></a>    <span class="st">'chrom'</span>: <span class="st">'chrX'</span>,</span>
<span id="cb79-11"><a href="#cb79-11" aria-hidden="true" tabindex="-1"></a>    <span class="st">'start'</span>: <span class="dv">0</span>,</span>
<span id="cb79-12"><a href="#cb79-12" aria-hidden="true" tabindex="-1"></a>    <span class="st">'end'</span>: chrX_size,</span>
<span id="cb79-13"><a href="#cb79-13" aria-hidden="true" tabindex="-1"></a>    <span class="st">'name'</span>: <span class="st">'chrX'</span>}, index<span class="op">=</span>[<span class="dv">0</span>])</span>
<span id="cb79-14"><a href="#cb79-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-15"><a href="#cb79-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Divide into chromosome arms</span></span>
<span id="cb79-16"><a href="#cb79-16" aria-hidden="true" tabindex="-1"></a>views[<span class="st">'arms'</span>] <span class="op">=</span> pd.DataFrame({</span>
<span id="cb79-17"><a href="#cb79-17" aria-hidden="true" tabindex="-1"></a>    <span class="st">'chrom'</span>: <span class="st">'chrX'</span>,</span>
<span id="cb79-18"><a href="#cb79-18" aria-hidden="true" tabindex="-1"></a>    <span class="st">'start'</span>: [<span class="dv">0</span>, <span class="dv">59_000_001</span>],</span>
<span id="cb79-19"><a href="#cb79-19" aria-hidden="true" tabindex="-1"></a>    <span class="st">'end'</span>: [<span class="dv">59_000_000</span>, chrX_size],</span>
<span id="cb79-20"><a href="#cb79-20" aria-hidden="true" tabindex="-1"></a>    <span class="st">'name'</span>: [<span class="st">'X_short'</span>, <span class="st">'X_long'</span>]}, index<span class="op">=</span>[<span class="dv">0</span>,<span class="dv">1</span>])</span>
<span id="cb79-21"><a href="#cb79-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-22"><a href="#cb79-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate in 10Mb windows</span></span>
<span id="cb79-23"><a href="#cb79-23" aria-hidden="true" tabindex="-1"></a>window_size <span class="op">=</span> <span class="dv">10_000_000</span></span>
<span id="cb79-24"><a href="#cb79-24" aria-hidden="true" tabindex="-1"></a>start_positions <span class="op">=</span> <span class="bu">list</span>(<span class="bu">range</span>(<span class="dv">0</span>, chrX_size, window_size))</span>
<span id="cb79-25"><a href="#cb79-25" aria-hidden="true" tabindex="-1"></a>end_positions <span class="op">=</span> [<span class="bu">min</span>(start <span class="op">+</span> window_size, chrX_size) <span class="cf">for</span> start <span class="kw">in</span> start_positions]</span>
<span id="cb79-26"><a href="#cb79-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-27"><a href="#cb79-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the DataFrame</span></span>
<span id="cb79-28"><a href="#cb79-28" aria-hidden="true" tabindex="-1"></a>views[<span class="st">'10Mb'</span>] <span class="op">=</span> pd.DataFrame({</span>
<span id="cb79-29"><a href="#cb79-29" aria-hidden="true" tabindex="-1"></a>    <span class="st">'chrom'</span>: [<span class="st">'chrX'</span>] <span class="op">*</span> <span class="bu">len</span>(start_positions),</span>
<span id="cb79-30"><a href="#cb79-30" aria-hidden="true" tabindex="-1"></a>    <span class="st">'start'</span>: start_positions,</span>
<span id="cb79-31"><a href="#cb79-31" aria-hidden="true" tabindex="-1"></a>    <span class="st">'end'</span>: end_positions,</span>
<span id="cb79-32"><a href="#cb79-32" aria-hidden="true" tabindex="-1"></a>    <span class="st">'name'</span>: [<span class="ss">f'X_</span><span class="sc">{</span>i<span class="sc">}</span><span class="ss">'</span> <span class="cf">for</span> i <span class="kw">in</span> <span class="bu">range</span>(<span class="bu">len</span>(start_positions))]</span>
<span id="cb79-33"><a href="#cb79-33" aria-hidden="true" tabindex="-1"></a>})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="calulate-the-e1-compartments" class="level5">
<h5 class="anchored" data-anchor-id="calulate-the-e1-compartments">Calulate the E1 compartments</h5>
<div id="cell-87" class="cell" data-execution_count="5">
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooltools</span>
<span id="cb80-2"><a href="#cb80-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-3"><a href="#cb80-3" aria-hidden="true" tabindex="-1"></a>eigs <span class="op">=</span> {}</span>
<span id="cb80-4"><a href="#cb80-4" aria-hidden="true" tabindex="-1"></a>e1_values <span class="op">=</span> {}</span>
<span id="cb80-5"><a href="#cb80-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-6"><a href="#cb80-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> name, clr <span class="kw">in</span> clrs.items():</span>
<span id="cb80-7"><a href="#cb80-7" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> name <span class="kw">not</span> <span class="kw">in</span> eigs <span class="kw">or</span> name <span class="kw">not</span> <span class="kw">in</span> e1_values:</span>
<span id="cb80-8"><a href="#cb80-8" aria-hidden="true" tabindex="-1"></a>        eigs[name] <span class="op">=</span> {} </span>
<span id="cb80-9"><a href="#cb80-9" aria-hidden="true" tabindex="-1"></a>        e1_values[name] <span class="op">=</span> {}</span>
<span id="cb80-10"><a href="#cb80-10" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> view, view_df <span class="kw">in</span> views.items():</span>
<span id="cb80-11"><a href="#cb80-11" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Calculating eigenvectors for </span><span class="sc">{</span>name<span class="sc">}</span><span class="ss"> at </span><span class="sc">{</span>view<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb80-12"><a href="#cb80-12" aria-hidden="true" tabindex="-1"></a>        cis_eigs <span class="op">=</span> cooltools.eigs_cis(</span>
<span id="cb80-13"><a href="#cb80-13" aria-hidden="true" tabindex="-1"></a>                            clr,</span>
<span id="cb80-14"><a href="#cb80-14" aria-hidden="true" tabindex="-1"></a>                            gc_cov,</span>
<span id="cb80-15"><a href="#cb80-15" aria-hidden="true" tabindex="-1"></a>                            view_df<span class="op">=</span>view_df,</span>
<span id="cb80-16"><a href="#cb80-16" aria-hidden="true" tabindex="-1"></a>                            n_eigs<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb80-17"><a href="#cb80-17" aria-hidden="true" tabindex="-1"></a>        eigs[name][view] <span class="op">=</span> cis_eigs[<span class="dv">1</span>]</span>
<span id="cb80-18"><a href="#cb80-18" aria-hidden="true" tabindex="-1"></a>        e1_values[name][view] <span class="op">=</span> cis_eigs[<span class="dv">1</span>][<span class="st">'E1'</span>].values</span>
<span id="cb80-19"><a href="#cb80-19" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb80-20"><a href="#cb80-20" aria-hidden="true" tabindex="-1"></a></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for fibroblast at full</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for fibroblast at arms</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for fibroblast at 10Mb</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for spermatogonia at full</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for spermatogonia at arms</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for spermatogonia at 10Mb</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for pachytene_spermatocyte at full</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for pachytene_spermatocyte at arms</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for pachytene_spermatocyte at 10Mb</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for round_spermatid at full</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for round_spermatid at arms</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for round_spermatid at 10Mb</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for sperm at full</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for sperm at arms</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for sperm at 10Mb</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
</div>
</section>
</section>
<section id="plot-the-e1-compartments-1" class="level4 page-columns page-full">
<h4 class="anchored" data-anchor-id="plot-the-e1-compartments-1">Plot the E1 compartments</h4>
<div id="cell-e1-fullmerged-all-100kb-5smoothed" class="cell page-columns page-full" data-execution_count="6">
<div class="sourceCode cell-code" id="cb111"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb111-1"><a href="#cb111-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb111-2"><a href="#cb111-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb111-3"><a href="#cb111-3" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> matplotlib.colors <span class="im">import</span> LogNorm</span>
<span id="cb111-4"><a href="#cb111-4" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> mpl_toolkits.axes_grid1 <span class="im">import</span> make_axes_locatable</span>
<span id="cb111-5"><a href="#cb111-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-6"><a href="#cb111-6" aria-hidden="true" tabindex="-1"></a>f, axs <span class="op">=</span> plt.subplots(<span class="dv">5</span>, <span class="dv">3</span>, figsize<span class="op">=</span>(<span class="dv">18</span>, <span class="dv">8</span>), sharex<span class="op">=</span><span class="va">True</span>, sharey<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb111-7"><a href="#cb111-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-8"><a href="#cb111-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Loop through eigs and e1_values to plot the E1 values</span></span>
<span id="cb111-9"><a href="#cb111-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i, (name, e1_dict) <span class="kw">in</span> <span class="bu">enumerate</span>(e1_values.items()):</span>
<span id="cb111-10"><a href="#cb111-10" aria-hidden="true" tabindex="-1"></a>    axs[i,<span class="dv">0</span>].set_ylabel(names_abbr[name])</span>
<span id="cb111-11"><a href="#cb111-11" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> j, (view, e1) <span class="kw">in</span> <span class="bu">enumerate</span>(e1_dict.items()):</span>
<span id="cb111-12"><a href="#cb111-12" aria-hidden="true" tabindex="-1"></a>        ax <span class="op">=</span> axs[i, j]</span>
<span id="cb111-13"><a href="#cb111-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-14"><a href="#cb111-14" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> i<span class="op">==</span><span class="dv">0</span>:</span>
<span id="cb111-15"><a href="#cb111-15" aria-hidden="true" tabindex="-1"></a>            ax.set_title(view)</span>
<span id="cb111-16"><a href="#cb111-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-17"><a href="#cb111-17" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Create stairs</span></span>
<span id="cb111-18"><a href="#cb111-18" aria-hidden="true" tabindex="-1"></a>        x <span class="op">=</span> np.zeros(<span class="dv">2</span><span class="op">*</span>chrom_start.size)</span>
<span id="cb111-19"><a href="#cb111-19" aria-hidden="true" tabindex="-1"></a>        y <span class="op">=</span> np.zeros(<span class="dv">2</span><span class="op">*</span>chrom_start.size)</span>
<span id="cb111-20"><a href="#cb111-20" aria-hidden="true" tabindex="-1"></a>        smooth_y <span class="op">=</span> np.zeros(<span class="dv">2</span><span class="op">*</span>chrom_start.size)</span>
<span id="cb111-21"><a href="#cb111-21" aria-hidden="true" tabindex="-1"></a>        x[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> chrom_start</span>
<span id="cb111-22"><a href="#cb111-22" aria-hidden="true" tabindex="-1"></a>        x[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> chrom_end</span>
<span id="cb111-23"><a href="#cb111-23" aria-hidden="true" tabindex="-1"></a>        y[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> e1</span>
<span id="cb111-24"><a href="#cb111-24" aria-hidden="true" tabindex="-1"></a>        y[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> e1</span>
<span id="cb111-25"><a href="#cb111-25" aria-hidden="true" tabindex="-1"></a>        smooth_e1 <span class="op">=</span> (<span class="kw">lambda</span> x: x.rolling(<span class="dv">5</span>, <span class="dv">1</span>, center<span class="op">=</span><span class="va">True</span>).<span class="bu">sum</span>())(pd.DataFrame(e1, columns<span class="op">=</span>[<span class="st">'value'</span>]))[<span class="st">'value'</span>].values</span>
<span id="cb111-26"><a href="#cb111-26" aria-hidden="true" tabindex="-1"></a>        smooth_y[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> smooth_e1</span>
<span id="cb111-27"><a href="#cb111-27" aria-hidden="true" tabindex="-1"></a>        smooth_y[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> smooth_e1</span>
<span id="cb111-28"><a href="#cb111-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-29"><a href="#cb111-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-30"><a href="#cb111-30" aria-hidden="true" tabindex="-1"></a>        <span class="co">#ax.fill_between(x, y, 0, where=(y &gt; 0), color='tab:red')</span></span>
<span id="cb111-31"><a href="#cb111-31" aria-hidden="true" tabindex="-1"></a>        <span class="co">#ax.fill_between(x, y, 0, where=(y &lt; 0), color='tab:blue')</span></span>
<span id="cb111-32"><a href="#cb111-32" aria-hidden="true" tabindex="-1"></a>        <span class="co">#ax.plot(x, smooth_y/5, color='C1')</span></span>
<span id="cb111-33"><a href="#cb111-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-34"><a href="#cb111-34" aria-hidden="true" tabindex="-1"></a>        ax.fill_between(x, smooth_y, <span class="dv">0</span>, where<span class="op">=</span>(smooth_y <span class="op">&gt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:red'</span>)</span>
<span id="cb111-35"><a href="#cb111-35" aria-hidden="true" tabindex="-1"></a>        ax.fill_between(x, smooth_y, <span class="dv">0</span>, where<span class="op">=</span>(smooth_y <span class="op">&lt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:blue'</span>)</span>
<span id="cb111-36"><a href="#cb111-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-37"><a href="#cb111-37" aria-hidden="true" tabindex="-1"></a><span class="co"># Set y-limits for all subplots</span></span>
<span id="cb111-38"><a href="#cb111-38" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> ax <span class="kw">in</span> axs.flat:</span>
<span id="cb111-39"><a href="#cb111-39" aria-hidden="true" tabindex="-1"></a>    ax.set_ylim(<span class="op">-</span><span class="dv">4</span>, <span class="dv">4</span>)</span>
<span id="cb111-40"><a href="#cb111-40" aria-hidden="true" tabindex="-1"></a>    ax.spines[:].set_visible(<span class="va">False</span>)</span>
<span id="cb111-41"><a href="#cb111-41" aria-hidden="true" tabindex="-1"></a>    ax.set_xticks([])</span>
<span id="cb111-42"><a href="#cb111-42" aria-hidden="true" tabindex="-1"></a>    ax.set_yticks([])</span>
<span id="cb111-43"><a href="#cb111-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-44"><a href="#cb111-44" aria-hidden="true" tabindex="-1"></a>plt.tight_layout()</span>
<span id="cb111-45"><a href="#cb111-45" aria-hidden="true" tabindex="-1"></a>plt.savefig(<span class="st">'../steps/e1_plot_full_100kb_smoothed.svg'</span>, bbox_inches<span class="op">=</span><span class="st">'tight'</span>)</span>
<span id="cb111-46"><a href="#cb111-46" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb111-47"><a href="#cb111-47" aria-hidden="true" tabindex="-1"></a>    </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display page-columns page-full">
<div id="e1-fullmerged-all-100kb-5smoothed" class="quarto-figure quarto-figure-center anchored page-columns page-full">
<figure class="figure page-columns page-full">
<p class="page-columns page-full"><img src="03_compartments_files/figure-html/e1-fullmerged-all-100kb-5smoothed-output-1.png" class="img-fluid figure-img column-page-inset"></p>
<figcaption>E1 eigenvector values for all merged samples at 100kb resolution. Left: Chromosomes (not restricted) Middle: Chromosome-arm restricted E1. Right: 10Mb window restricted E1.Values are smoothed with a sliding window of 5 bins, step size 1 bin.</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="plot-matrices-with-compartments" class="level4">
<h4 class="anchored" data-anchor-id="plot-matrices-with-compartments">Plot matrices with compartments</h4>
<div id="cell-e1-matrix-100kb-full-arms-10mb-all-5smoothed" class="cell">
<div class="sourceCode cell-code" id="cb112"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb112-1"><a href="#cb112-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooltools.lib.plotting</span>
<span id="cb112-2"><a href="#cb112-2" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> matplotlib.colors <span class="im">import</span> LogNorm</span>
<span id="cb112-3"><a href="#cb112-3" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> mpl_toolkits.axes_grid1 <span class="im">import</span> make_axes_locatable</span>
<span id="cb112-4"><a href="#cb112-4" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> cooltools.lib.numutils <span class="im">import</span> adaptive_coarsegrain, interp_nan</span>
<span id="cb112-5"><a href="#cb112-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb112-6"><a href="#cb112-6" aria-hidden="true" tabindex="-1"></a>f, axs <span class="op">=</span> plt.subplots(<span class="dv">1</span>,<span class="dv">5</span>,</span>
<span id="cb112-7"><a href="#cb112-7" aria-hidden="true" tabindex="-1"></a>    figsize<span class="op">=</span>(<span class="dv">25</span>,<span class="dv">10</span>)</span>
<span id="cb112-8"><a href="#cb112-8" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb112-9"><a href="#cb112-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb112-10"><a href="#cb112-10" aria-hidden="true" tabindex="-1"></a>norm <span class="op">=</span> LogNorm(vmax<span class="op">=</span><span class="fl">0.1</span>)</span>
<span id="cb112-11"><a href="#cb112-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb112-12"><a href="#cb112-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Loop through the clrs and its matrix: plot the matrix on its axis</span></span>
<span id="cb112-13"><a href="#cb112-13" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i, (name, e1_dict) <span class="kw">in</span> <span class="bu">enumerate</span>(e1_values.items()):</span>
<span id="cb112-14"><a href="#cb112-14" aria-hidden="true" tabindex="-1"></a>    ax <span class="op">=</span> axs[i]</span>
<span id="cb112-15"><a href="#cb112-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb112-16"><a href="#cb112-16" aria-hidden="true" tabindex="-1"></a>    im <span class="op">=</span> ax.matshow(</span>
<span id="cb112-17"><a href="#cb112-17" aria-hidden="true" tabindex="-1"></a>        clr.matrix().fetch(<span class="st">'chrX'</span>),</span>
<span id="cb112-18"><a href="#cb112-18" aria-hidden="true" tabindex="-1"></a>        norm<span class="op">=</span>norm,</span>
<span id="cb112-19"><a href="#cb112-19" aria-hidden="true" tabindex="-1"></a>        cmap<span class="op">=</span><span class="st">'fall'</span>,</span>
<span id="cb112-20"><a href="#cb112-20" aria-hidden="true" tabindex="-1"></a>    )<span class="op">;</span></span>
<span id="cb112-21"><a href="#cb112-21" aria-hidden="true" tabindex="-1"></a>    ax.set_xlim(<span class="dv">0</span>, nbins)</span>
<span id="cb112-22"><a href="#cb112-22" aria-hidden="true" tabindex="-1"></a>    ax.set_ylim(nbins, <span class="dv">0</span>)</span>
<span id="cb112-23"><a href="#cb112-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb112-24"><a href="#cb112-24" aria-hidden="true" tabindex="-1"></a>    divider <span class="op">=</span> make_axes_locatable(ax)</span>
<span id="cb112-25"><a href="#cb112-25" aria-hidden="true" tabindex="-1"></a>    cax <span class="op">=</span> divider.append_axes(<span class="st">"right"</span>, size<span class="op">=</span><span class="st">"2%"</span>, pad<span class="op">=</span><span class="fl">0.05</span>)</span>
<span id="cb112-26"><a href="#cb112-26" aria-hidden="true" tabindex="-1"></a>    plt.colorbar(im, cax<span class="op">=</span>cax, label<span class="op">=</span><span class="st">'corrected frequencies'</span>)<span class="op">;</span></span>
<span id="cb112-27"><a href="#cb112-27" aria-hidden="true" tabindex="-1"></a>    ax.set_ylabel(<span class="st">'chrX:100kb bins, #bin'</span>)</span>
<span id="cb112-28"><a href="#cb112-28" aria-hidden="true" tabindex="-1"></a>    ax.xaxis.set_visible(<span class="va">False</span>)</span>
<span id="cb112-29"><a href="#cb112-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb112-30"><a href="#cb112-30" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> j, (view, e1) <span class="kw">in</span> <span class="bu">enumerate</span>(e1_dict.items()):</span>
<span id="cb112-31"><a href="#cb112-31" aria-hidden="true" tabindex="-1"></a>        ax1 <span class="op">=</span> divider.append_axes(<span class="st">"top"</span>, size<span class="op">=</span><span class="st">"10%"</span>, pad<span class="op">=</span><span class="fl">0.1</span>, sharex <span class="op">=</span> ax)</span>
<span id="cb112-32"><a href="#cb112-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb112-33"><a href="#cb112-33" aria-hidden="true" tabindex="-1"></a>        <span class="co"># ax1.plot(e1, label='E1')</span></span>
<span id="cb112-34"><a href="#cb112-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb112-35"><a href="#cb112-35" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Create stairs</span></span>
<span id="cb112-36"><a href="#cb112-36" aria-hidden="true" tabindex="-1"></a>        x <span class="op">=</span> np.zeros(<span class="dv">2</span><span class="op">*</span>chrom_start.size)</span>
<span id="cb112-37"><a href="#cb112-37" aria-hidden="true" tabindex="-1"></a>        y <span class="op">=</span> np.zeros(<span class="dv">2</span><span class="op">*</span>chrom_start.size)</span>
<span id="cb112-38"><a href="#cb112-38" aria-hidden="true" tabindex="-1"></a>        <span class="co">#smooth_y = np.zeros(2*chrom_start.size)</span></span>
<span id="cb112-39"><a href="#cb112-39" aria-hidden="true" tabindex="-1"></a>        x[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> chrom_start<span class="op">/</span>binsize</span>
<span id="cb112-40"><a href="#cb112-40" aria-hidden="true" tabindex="-1"></a>        x[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> chrom_end<span class="op">/</span>binsize</span>
<span id="cb112-41"><a href="#cb112-41" aria-hidden="true" tabindex="-1"></a>        y[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> e1</span>
<span id="cb112-42"><a href="#cb112-42" aria-hidden="true" tabindex="-1"></a>        y[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> e1</span>
<span id="cb112-43"><a href="#cb112-43" aria-hidden="true" tabindex="-1"></a>        <span class="co"># smooth_e1 = (lambda x: x.rolling(5, 1, center=True).sum())(pd.DataFrame(e1, columns=['value']))['value'].values</span></span>
<span id="cb112-44"><a href="#cb112-44" aria-hidden="true" tabindex="-1"></a>        <span class="co"># smooth_y[0::2] = smooth_e1</span></span>
<span id="cb112-45"><a href="#cb112-45" aria-hidden="true" tabindex="-1"></a>        <span class="co"># smooth_y[1::2] = smooth_e1</span></span>
<span id="cb112-46"><a href="#cb112-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb112-47"><a href="#cb112-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb112-48"><a href="#cb112-48" aria-hidden="true" tabindex="-1"></a>        ax1.fill_between(x, y, <span class="dv">0</span>, where<span class="op">=</span>(y <span class="op">&gt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:red'</span>)</span>
<span id="cb112-49"><a href="#cb112-49" aria-hidden="true" tabindex="-1"></a>        ax1.fill_between(x, y, <span class="dv">0</span>, where<span class="op">=</span>(y <span class="op">&lt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:blue'</span>)</span>
<span id="cb112-50"><a href="#cb112-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb112-51"><a href="#cb112-51" aria-hidden="true" tabindex="-1"></a>        ax1.set_ylabel(view, rotation<span class="op">=-</span><span class="dv">70</span>)</span>
<span id="cb112-52"><a href="#cb112-52" aria-hidden="true" tabindex="-1"></a>        ax1.set_ylim(<span class="op">-</span><span class="fl">0.8</span>, <span class="fl">0.8</span>)</span>
<span id="cb112-53"><a href="#cb112-53" aria-hidden="true" tabindex="-1"></a>        ax1.set_xticks([])</span>
<span id="cb112-54"><a href="#cb112-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb112-55"><a href="#cb112-55" aria-hidden="true" tabindex="-1"></a>    ax1.set_title(name)</span>
<span id="cb112-56"><a href="#cb112-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb112-57"><a href="#cb112-57" aria-hidden="true" tabindex="-1"></a><span class="co">#    # Plot the sign changes on the matrix </span></span>
<span id="cb112-58"><a href="#cb112-58" aria-hidden="true" tabindex="-1"></a><span class="co">#     col = colors[i]</span></span>
<span id="cb112-59"><a href="#cb112-59" aria-hidden="true" tabindex="-1"></a><span class="co">#     for i in np.where(np.diff( (pd.Series(e1)&gt;0).astype(int)))[0]:</span></span>
<span id="cb112-60"><a href="#cb112-60" aria-hidden="true" tabindex="-1"></a><span class="co">#         # Horisontal lines where E1 intersects 0</span></span>
<span id="cb112-61"><a href="#cb112-61" aria-hidden="true" tabindex="-1"></a><span class="co">#         ax.plot([0,nbins],[i,i],col,lw=0.5)</span></span>
<span id="cb112-62"><a href="#cb112-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb112-63"><a href="#cb112-63" aria-hidden="true" tabindex="-1"></a><span class="co">#         # Vertical lines where E1 intersects 0</span></span>
<span id="cb112-64"><a href="#cb112-64" aria-hidden="true" tabindex="-1"></a><span class="co">#         ax.plot([i,i],[0,nbins],col,lw=0.5)</span></span>
<span id="cb112-65"><a href="#cb112-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb112-66"><a href="#cb112-66" aria-hidden="true" tabindex="-1"></a><span class="co"># Now show the plot</span></span>
<span id="cb112-67"><a href="#cb112-67" aria-hidden="true" tabindex="-1"></a>f.set_layout_engine(<span class="st">'compressed'</span>)</span>
<span id="cb112-68"><a href="#cb112-68" aria-hidden="true" tabindex="-1"></a>f.canvas.draw()</span>
<span id="cb112-69"><a href="#cb112-69" aria-hidden="true" tabindex="-1"></a>plt.show()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div id="e1-matrix-100kb-full-arms-10mb-all-5smoothed" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="03_compartments_files/figure-html/e1-matrix-100kb-full-arms-10mb-all-5smoothed-output-1.png" class="img-fluid figure-img"></p>
<figcaption>E1 eigenvector values for all merged samples at 500kb resolution for round spermatid, as well as the interaction matrix. E1 was restricted to top: Full-chromosome, middle: Chromosome-arms, bottom: 10Mb window.</figcaption>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="smooth-the-observedexpected-matrix" class="level3">
<h3 class="anchored" data-anchor-id="smooth-the-observedexpected-matrix">Smooth the Observed/Expected matrix</h3>
<p>Wang et al.&nbsp;2019 used a 400Kb window with a step size of 100Kb to smooth the O/E matrix before calculating the E1. I have tried various different methods to smooth the interaction matrix, but I can’t get dimensions of the <code>pixels</code> to match with the <code>bins</code>. Also, <code>cooltools.eigs_cis</code> need a full <code>cooler</code> to fetch data from, and I can’t get it to work without a <code>pixels</code> object.</p>
<p>Thus, I will try smoothing the E1 values instead, as I can’t really rationalize what biological difference it would make. I think I need help figuring that out.</p>
<div id="cell-94" class="cell">
<div class="sourceCode cell-code" id="cb113"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb113-1"><a href="#cb113-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> glob</span>
<span id="cb113-2"><a href="#cb113-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os.path <span class="im">as</span> op</span>
<span id="cb113-3"><a href="#cb113-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooler</span>
<span id="cb113-4"><a href="#cb113-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-5"><a href="#cb113-5" aria-hidden="true" tabindex="-1"></a>mcools <span class="op">=</span> glob.glob(<span class="st">"../steps/bwa/PE/cool/*/*.mcool"</span>)</span>
<span id="cb113-6"><a href="#cb113-6" aria-hidden="true" tabindex="-1"></a>res <span class="op">=</span> <span class="st">"::resolutions/100000"</span></span>
<span id="cb113-7"><a href="#cb113-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-8"><a href="#cb113-8" aria-hidden="true" tabindex="-1"></a>clrs100kb <span class="op">=</span> {op.basename(op.dirname(mcool)): cooler.Cooler(mcool<span class="op">+</span>res) <span class="cf">for</span> mcool <span class="kw">in</span> mcools}</span>
<span id="cb113-9"><a href="#cb113-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-10"><a href="#cb113-10" aria-hidden="true" tabindex="-1"></a>chron_order <span class="op">=</span> [<span class="st">'fibroblast'</span>, <span class="st">'spermatogonia'</span>, <span class="st">'pachytene_spermatocyte'</span>, <span class="st">'round_spermatid'</span>, <span class="st">'sperm'</span>]</span>
<span id="cb113-11"><a href="#cb113-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-12"><a href="#cb113-12" aria-hidden="true" tabindex="-1"></a>clrs100kb <span class="op">=</span> {key: clrs100kb[key] <span class="cf">for</span> key <span class="kw">in</span> chron_order}</span>
<span id="cb113-13"><a href="#cb113-13" aria-hidden="true" tabindex="-1"></a>clrs100kb</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<pre><code>{'fibroblast': &lt;Cooler "fibroblast.mcool::/resolutions/100000"&gt;,
 'spermatogonia': &lt;Cooler "spermatogonia.mcool::/resolutions/100000"&gt;,
 'pachytene_spermatocyte': &lt;Cooler "pachytene_spermatocyte.mcool::/resolutions/100000"&gt;,
 'round_spermatid': &lt;Cooler "round_spermatid.mcool::/resolutions/100000"&gt;,
 'sperm': &lt;Cooler "sperm.mcool::/resolutions/100000"&gt;}</code></pre>
</div>
</div>
<section id="calculate-gc-covariance-from-the-reference-genome-3" class="level4">
<h4 class="anchored" data-anchor-id="calculate-gc-covariance-from-the-reference-genome-3">Calculate gc covariance (from the reference genome)</h4>
<p>Do this with any of the clrs - it just needs the bins positions.</p>
<div id="cell-96" class="cell">
<div class="sourceCode cell-code" id="cb115"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb115-1"><a href="#cb115-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> bioframe</span>
<span id="cb115-2"><a href="#cb115-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb115-3"><a href="#cb115-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os.path <span class="im">as</span> op</span>
<span id="cb115-4"><a href="#cb115-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb115-5"><a href="#cb115-5" aria-hidden="true" tabindex="-1"></a>bins <span class="op">=</span> clrs100kb[<span class="st">'fibroblast'</span>].bins().fetch(<span class="st">'chrX'</span>)[:]</span>
<span id="cb115-6"><a href="#cb115-6" aria-hidden="true" tabindex="-1"></a>out_name <span class="op">=</span> <span class="st">'rheMac10_gc_cov_X_100kb.tsv'</span></span>
<span id="cb115-7"><a href="#cb115-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb115-8"><a href="#cb115-8" aria-hidden="true" tabindex="-1"></a>rheMac10 <span class="op">=</span> bioframe.load_fasta(<span class="st">'../data/links/ucsc_ref/rheMac10.fa'</span>)</span>
<span id="cb115-9"><a href="#cb115-9" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> <span class="kw">not</span> op.exists(out_name):</span>
<span id="cb115-10"><a href="#cb115-10" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">'Calculate the fraction of GC basepairs for each bin'</span>)</span>
<span id="cb115-11"><a href="#cb115-11" aria-hidden="true" tabindex="-1"></a>    gc_cov <span class="op">=</span> bioframe.frac_gc(bins[[<span class="st">'chrom'</span>, <span class="st">'start'</span>, <span class="st">'end'</span>]], rheMac10)</span>
<span id="cb115-12"><a href="#cb115-12" aria-hidden="true" tabindex="-1"></a>    gc_cov.to_csv(out_name, index<span class="op">=</span><span class="va">False</span>,sep<span class="op">=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>)</span>
<span id="cb115-13"><a href="#cb115-13" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(gc_cov.info())</span>
<span id="cb115-14"><a href="#cb115-14" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>: </span>
<span id="cb115-15"><a href="#cb115-15" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"Already exists, read from file"</span>)</span>
<span id="cb115-16"><a href="#cb115-16" aria-hidden="true" tabindex="-1"></a>    gc_cov <span class="op">=</span> pd.read_csv(out_name, sep<span class="op">=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>)</span>
<span id="cb115-17"><a href="#cb115-17" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(gc_cov.info())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Already exists, read from file
&lt;class 'pandas.core.frame.DataFrame'&gt;
RangeIndex: 1534 entries, 0 to 1533
Data columns (total 4 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   chrom   1534 non-null   object 
 1   start   1534 non-null   int64  
 2   end     1534 non-null   int64  
 3   GC      1533 non-null   float64
dtypes: float64(1), int64(2), object(1)
memory usage: 48.1+ KB
None</code></pre>
</div>
</div>
<div id="cell-97" class="cell">
<div class="sourceCode cell-code" id="cb117"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb117-1"><a href="#cb117-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooler</span>
<span id="cb117-2"><a href="#cb117-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb117-3"><a href="#cb117-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb117-4"><a href="#cb117-4" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> smooth_matrix(matrix, window_size<span class="op">=</span><span class="dv">4</span>, step_size<span class="op">=</span><span class="dv">1</span>):</span>
<span id="cb117-5"><a href="#cb117-5" aria-hidden="true" tabindex="-1"></a>    n <span class="op">=</span> matrix.shape[<span class="dv">0</span>]</span>
<span id="cb117-6"><a href="#cb117-6" aria-hidden="true" tabindex="-1"></a>    smoothed_matrix <span class="op">=</span> np.zeros_like(matrix)</span>
<span id="cb117-7"><a href="#cb117-7" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb117-8"><a href="#cb117-8" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> i <span class="kw">in</span> <span class="bu">range</span>(<span class="dv">0</span>, n <span class="op">-</span> window_size <span class="op">+</span> <span class="dv">1</span>, step_size):</span>
<span id="cb117-9"><a href="#cb117-9" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> j <span class="kw">in</span> <span class="bu">range</span>(<span class="dv">0</span>, n <span class="op">-</span> window_size <span class="op">+</span> <span class="dv">1</span>, step_size):</span>
<span id="cb117-10"><a href="#cb117-10" aria-hidden="true" tabindex="-1"></a>            window <span class="op">=</span> matrix[i:i<span class="op">+</span>window_size, j:j<span class="op">+</span>window_size]</span>
<span id="cb117-11"><a href="#cb117-11" aria-hidden="true" tabindex="-1"></a>            smoothed_matrix[i:i<span class="op">+</span>window_size, j:j<span class="op">+</span>window_size] <span class="op">=</span> np.<span class="bu">sum</span>(window)</span>
<span id="cb117-12"><a href="#cb117-12" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb117-13"><a href="#cb117-13" aria-hidden="true" tabindex="-1"></a>    <span class="cf">return</span> smoothed_matrix</span>
<span id="cb117-14"><a href="#cb117-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb117-15"><a href="#cb117-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Load the Hi-C interaction matrix from a multi-resolution .mcool file</span></span>
<span id="cb117-16"><a href="#cb117-16" aria-hidden="true" tabindex="-1"></a>cooler_file <span class="op">=</span> <span class="st">'../steps/bwa/PE/cool/fibroblast/fibroblast.mcool'</span></span>
<span id="cb117-17"><a href="#cb117-17" aria-hidden="true" tabindex="-1"></a>clr <span class="op">=</span> cooler.Cooler(cooler_file <span class="op">+</span> <span class="st">'::resolutions/100000'</span>)</span>
<span id="cb117-18"><a href="#cb117-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb117-19"><a href="#cb117-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract the matrix</span></span>
<span id="cb117-20"><a href="#cb117-20" aria-hidden="true" tabindex="-1"></a>matrix <span class="op">=</span> clr.matrix(balance<span class="op">=</span><span class="va">True</span>).fetch(<span class="st">'chrX'</span>)[:]</span>
<span id="cb117-21"><a href="#cb117-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb117-22"><a href="#cb117-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Smooth the matrix</span></span>
<span id="cb117-23"><a href="#cb117-23" aria-hidden="true" tabindex="-1"></a>smoothed_matrix <span class="op">=</span> smooth_matrix(matrix, window_size<span class="op">=</span><span class="dv">4</span>, step_size<span class="op">=</span><span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div id="cell-98" class="cell">
<div class="sourceCode cell-code" id="cb118"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb118-1"><a href="#cb118-1" aria-hidden="true" tabindex="-1"></a><span class="co">#clr.bins().fetch('chrX')[:]</span></span>
<span id="cb118-2"><a href="#cb118-2" aria-hidden="true" tabindex="-1"></a>clr.pixels().fetch(<span class="st">'chrX'</span>)[:]</span>
<span id="cb118-3"><a href="#cb118-3" aria-hidden="true" tabindex="-1"></a><span class="co">#print(matrix.shape)</span></span>
<span id="cb118-4"><a href="#cb118-4" aria-hidden="true" tabindex="-1"></a><span class="co">#print(smoothed_matrix.shape)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">bin1_id</th>
<th data-quarto-table-cell-role="th">bin2_id</th>
<th data-quarto-table-cell-role="th">count</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">145469195</td>
<td>26898</td>
<td>26898</td>
<td>2013</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">145469196</td>
<td>26898</td>
<td>26899</td>
<td>171</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">145469197</td>
<td>26898</td>
<td>26901</td>
<td>75</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">145469198</td>
<td>26898</td>
<td>26902</td>
<td>100</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">145469199</td>
<td>26898</td>
<td>26903</td>
<td>187</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">146369705</td>
<td>28431</td>
<td>28532</td>
<td>2</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">146369706</td>
<td>28431</td>
<td>28533</td>
<td>1</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">146369707</td>
<td>28431</td>
<td>28534</td>
<td>2</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">146369708</td>
<td>28431</td>
<td>28539</td>
<td>1</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">146369709</td>
<td>28431</td>
<td>28544</td>
<td>1</td>
</tr>
</tbody>
</table>

<p>900515 rows × 3 columns</p>
</div>
</div>
</div>
<div id="cell-99" class="cell">
<div class="sourceCode cell-code" id="cb119"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb119-1"><a href="#cb119-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb119-2"><a href="#cb119-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooler</span>
<span id="cb119-3"><a href="#cb119-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb119-4"><a href="#cb119-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-5"><a href="#cb119-5" aria-hidden="true" tabindex="-1"></a>pixels <span class="op">=</span> pd.DataFrame(smoothed_matrix).stack().rename_axis([<span class="st">'bin1_id'</span>, <span class="st">'bin2_id'</span>]).reset_index(name<span class="op">=</span><span class="st">'count'</span>)</span>
<span id="cb119-6"><a href="#cb119-6" aria-hidden="true" tabindex="-1"></a>pixels <span class="op">=</span> pixels[pixels[<span class="st">'bin1_id'</span>] <span class="op">&lt;=</span> pixels[<span class="st">'bin2_id'</span>]]</span>
<span id="cb119-7"><a href="#cb119-7" aria-hidden="true" tabindex="-1"></a>pixels.sort_values([<span class="st">'bin1_id'</span>, <span class="st">'bin2_id'</span>])[[<span class="st">'bin1_id'</span>, <span class="st">'bin2_id'</span>, <span class="st">'count'</span>]].reset_index(drop<span class="op">=</span><span class="va">True</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">bin1_id</th>
<th data-quarto-table-cell-role="th">bin2_id</th>
<th data-quarto-table-cell-role="th">count</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>5</td>
<td>5</td>
<td>2.555538</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>5</td>
<td>6</td>
<td>2.141584</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>5</td>
<td>7</td>
<td>1.550878</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3</td>
<td>5</td>
<td>8</td>
<td>0.864845</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">4</td>
<td>5</td>
<td>9</td>
<td>0.269873</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">1049071</td>
<td>1531</td>
<td>1532</td>
<td>4.241950</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1049072</td>
<td>1531</td>
<td>1533</td>
<td>4.241950</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">1049073</td>
<td>1532</td>
<td>1532</td>
<td>4.241950</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1049074</td>
<td>1532</td>
<td>1533</td>
<td>4.241950</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">1049075</td>
<td>1533</td>
<td>1533</td>
<td>4.241950</td>
</tr>
</tbody>
</table>

<p>1049076 rows × 3 columns</p>
</div>
</div>
</div>
<div id="cell-100" class="cell">
<div class="sourceCode cell-code" id="cb120"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb120-1"><a href="#cb120-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the eigs</span></span>
<span id="cb120-2"><a href="#cb120-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-3"><a href="#cb120-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooltools</span>
<span id="cb120-4"><a href="#cb120-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb120-5"><a href="#cb120-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-6"><a href="#cb120-6" aria-hidden="true" tabindex="-1"></a>eigvecs, eigvals <span class="op">=</span> cooltools.lib.numutils.get_eig(matrix, n<span class="op">=</span><span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-error">
<div class="ansi-escaped-output">
<pre><span class="ansi-red-fg">---------------------------------------------------------------------------</span>

<span class="ansi-red-fg">ArpackNoConvergence</span>                       Traceback (most recent call last)

Cell <span class="ansi-green-fg">In[6], line 6</span>

<span class="ansi-green-fg ansi-bold">      3</span> <span style="font-weight:bold;color:rgb(0,135,0)">import</span> <span style="font-weight:bold;color:rgb(0,0,255)">cooltools</span>

<span class="ansi-green-fg ansi-bold">      4</span> <span style="font-weight:bold;color:rgb(0,135,0)">import</span> <span style="font-weight:bold;color:rgb(0,0,255)">pandas</span> <span style="font-weight:bold;color:rgb(0,135,0)">as</span> <span style="font-weight:bold;color:rgb(0,0,255)">pd</span>

<span class="ansi-green-fg">----&gt; 6</span> eigvecs, eigvals <span style="color:rgb(98,98,98)">=</span> cooltools<span style="color:rgb(98,98,98)">.</span>lib<span style="color:rgb(98,98,98)">.</span>numutils<span style="color:rgb(98,98,98)">.</span>get_eig(matrix, n<span style="color:rgb(98,98,98)">=</span><span style="color:rgb(98,98,98)">1</span>)



File <span class="ansi-green-fg">~/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/numutils.py:519</span>, in <span class="ansi-cyan-fg">get_eig</span><span class="ansi-blue-fg">(mat, n, mask_zero_rows, subtract_mean, divide_by_mean)</span>

<span class="ansi-green-fg ansi-bold">    516</span> <span style="font-weight:bold;color:rgb(0,135,0)">else</span>:

<span class="ansi-green-fg ansi-bold">    517</span>     <span style="font-style:italic;color:rgb(95,135,135)"># adjust requested number of eigvals for "eigs"</span>

<span class="ansi-green-fg ansi-bold">    518</span>     _n <span style="color:rgb(98,98,98)">=</span> n <span style="font-weight:bold;color:rgb(0,135,0)">if</span> n <span style="color:rgb(98,98,98)">&lt;</span> (n_rows <span style="color:rgb(98,98,98)">-</span> <span style="color:rgb(98,98,98)">1</span>) <span style="font-weight:bold;color:rgb(0,135,0)">else</span> (n_rows <span style="color:rgb(98,98,98)">-</span> <span style="color:rgb(98,98,98)">2</span>)

<span class="ansi-green-fg">--&gt; 519</span>     _eigvals, _eigvecs <span style="color:rgb(98,98,98)">=</span> scipy<span style="color:rgb(98,98,98)">.</span>sparse<span style="color:rgb(98,98,98)">.</span>linalg<span style="color:rgb(98,98,98)">.</span>eigs(mat, _n)

<span class="ansi-green-fg ansi-bold">    521</span> <span style="font-style:italic;color:rgb(95,135,135)"># reorder according to eigvals and copy into output arrays</span>

<span class="ansi-green-fg ansi-bold">    522</span> order <span style="color:rgb(98,98,98)">=</span> np<span style="color:rgb(98,98,98)">.</span>argsort(<span style="color:rgb(98,98,98)">-</span>np<span style="color:rgb(98,98,98)">.</span>abs(_eigvals))



File <span class="ansi-green-fg">~/miniconda3/envs/hic/lib/python3.12/site-packages/scipy/sparse/linalg/_eigen/arpack/arpack.py:1355</span>, in <span class="ansi-cyan-fg">eigs</span><span class="ansi-blue-fg">(A, k, M, sigma, which, v0, ncv, maxiter, tol, return_eigenvectors, Minv, OPinv, OPpart)</span>

<span class="ansi-green-fg ansi-bold">   1353</span> <span style="font-weight:bold;color:rgb(0,135,0)">with</span> _ARPACK_LOCK:

<span class="ansi-green-fg ansi-bold">   1354</span>     <span style="font-weight:bold;color:rgb(0,135,0)">while</span> <span style="font-weight:bold;color:rgb(175,0,255)">not</span> params<span style="color:rgb(98,98,98)">.</span>converged:

<span class="ansi-green-fg">-&gt; 1355</span>         params<span style="color:rgb(98,98,98)">.</span>iterate()

<span class="ansi-green-fg ansi-bold">   1357</span>     <span style="font-weight:bold;color:rgb(0,135,0)">return</span> params<span style="color:rgb(98,98,98)">.</span>extract(return_eigenvectors)



File <span class="ansi-green-fg">~/miniconda3/envs/hic/lib/python3.12/site-packages/scipy/sparse/linalg/_eigen/arpack/arpack.py:760</span>, in <span class="ansi-cyan-fg">_UnsymmetricArpackParams.iterate</span><span class="ansi-blue-fg">(self)</span>

<span class="ansi-green-fg ansi-bold">    758</span>     <span style="font-weight:bold;color:rgb(0,135,0)">pass</span>

<span class="ansi-green-fg ansi-bold">    759</span> <span style="font-weight:bold;color:rgb(0,135,0)">elif</span> <span style="color:rgb(0,135,0)">self</span><span style="color:rgb(98,98,98)">.</span>info <span style="color:rgb(98,98,98)">==</span> <span style="color:rgb(98,98,98)">1</span>:

<span class="ansi-green-fg">--&gt; 760</span>     <span style="color:rgb(0,135,0)">self</span><span style="color:rgb(98,98,98)">.</span>_raise_no_convergence()

<span class="ansi-green-fg ansi-bold">    761</span> <span style="font-weight:bold;color:rgb(0,135,0)">else</span>:

<span class="ansi-green-fg ansi-bold">    762</span>     <span style="font-weight:bold;color:rgb(0,135,0)">raise</span> ArpackError(<span style="color:rgb(0,135,0)">self</span><span style="color:rgb(98,98,98)">.</span>info, infodict<span style="color:rgb(98,98,98)">=</span><span style="color:rgb(0,135,0)">self</span><span style="color:rgb(98,98,98)">.</span>iterate_infodict)



File <span class="ansi-green-fg">~/miniconda3/envs/hic/lib/python3.12/site-packages/scipy/sparse/linalg/_eigen/arpack/arpack.py:377</span>, in <span class="ansi-cyan-fg">_ArpackParams._raise_no_convergence</span><span class="ansi-blue-fg">(self)</span>

<span class="ansi-green-fg ansi-bold">    375</span>     vec <span style="color:rgb(98,98,98)">=</span> np<span style="color:rgb(98,98,98)">.</span>zeros((<span style="color:rgb(0,135,0)">self</span><span style="color:rgb(98,98,98)">.</span>n, <span style="color:rgb(98,98,98)">0</span>))

<span class="ansi-green-fg ansi-bold">    376</span>     k_ok <span style="color:rgb(98,98,98)">=</span> <span style="color:rgb(98,98,98)">0</span>

<span class="ansi-green-fg">--&gt; 377</span> <span style="font-weight:bold;color:rgb(0,135,0)">raise</span> ArpackNoConvergence(msg <span style="color:rgb(98,98,98)">%</span> (num_iter, k_ok, <span style="color:rgb(0,135,0)">self</span><span style="color:rgb(98,98,98)">.</span>k), ev, vec)



<span class="ansi-red-fg">ArpackNoConvergence</span>: ARPACK error -1: No convergence (15341 iterations, 0/1 eigenvectors converged) [ARPACK error -14: DNAUPD  did not find any eigenvalues to sufficient accuracy.]</pre>
</div>
</div>
</div>
<div id="cell-101" class="cell">
<div class="sourceCode cell-code" id="cb121"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb121-1"><a href="#cb121-1" aria-hidden="true" tabindex="-1"></a><span class="co"># # create a cooler for the smoothed matrix</span></span>
<span id="cb121-2"><a href="#cb121-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-3"><a href="#cb121-3" aria-hidden="true" tabindex="-1"></a><span class="co"># import cooler</span></span>
<span id="cb121-4"><a href="#cb121-4" aria-hidden="true" tabindex="-1"></a><span class="co"># import numpy as np</span></span>
<span id="cb121-5"><a href="#cb121-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-6"><a href="#cb121-6" aria-hidden="true" tabindex="-1"></a><span class="co"># # Create a new cooler file for the smoothed matrix</span></span>
<span id="cb121-7"><a href="#cb121-7" aria-hidden="true" tabindex="-1"></a><span class="co"># smoothed_cooler_file = cooler_file.replace('.mcool', '.smoothed.mcool')</span></span>
<span id="cb121-8"><a href="#cb121-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-9"><a href="#cb121-9" aria-hidden="true" tabindex="-1"></a><span class="co"># # Create a new cooler object</span></span>
<span id="cb121-10"><a href="#cb121-10" aria-hidden="true" tabindex="-1"></a><span class="co"># cooler.create_cooler(smoothed_cooler_file, bins=clr.bins()[:], pixels=smoothed_matrix) </span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>


<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list" style="display: none">
<div id="ref-wang_reprogramming_2019" class="csl-entry" role="listitem">
Wang, Yao, Hanben Wang, Yu Zhang, Zhenhai Du, Wei Si, Suixing Fan, Dongdong Qin, et al. 2019. <span>“Reprogramming of <span>Meiotic</span> <span>Chromatin</span> <span>Architecture</span> During <span>Spermatogenesis</span>.”</span> <em>Molecular Cell</em> 73 (3): 547–561.e6. <a href="https://doi.org/10.1016/j.molcel.2018.11.019">https://doi.org/10.1016/j.molcel.2018.11.019</a>.
</div>
</div>
</section>
</section>
</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../notebooks/02_open2c_framework.html" class="pagination-link" aria-label="Open2C framework">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-title">Open2C framework</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../master_workflow.html" class="pagination-link" aria-label="master_workflow">
        <span class="nav-page-text"><span class="chapter-title">master_workflow</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->




<footer class="footer"><div class="nav-footer"><div class="nav-footer-center"><div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.dev/munch-group/hic-spermatogenesis/blob/main/notebooks/03_compartments.ipynb" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/munch-group/hic-spermatogenesis/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li><li><a href="https://github.com/munch-group/hic-spermatogenesis/blob/main/notebooks/03_compartments.ipynb" class="toc-action"><i class="bi empty"></i>View source</a></li></ul></div></div></div></footer></body></html>