<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>4&nbsp; Compartments Analysis – Søren's MSc project about Hi-C and spermatogenesis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js" integrity="sha512-bLT0Qm9VnAYZDflyKcBaQ2gg0hSYNQrJ8RilYldYQ1FxQYoCLtUjuuRuZo+fjqhx/qtq/1itJ0C2ejDxltZVFg==" crossorigin="anonymous"></script><script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../scripts/master_workflow.html" rel="next">
<link href="../notebooks/02_open2c_framework.html" rel="prev">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://hypothes.is/embed.js"></script>
<script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" integrity="sha512-c3Nl8+7g4LMSTdrm621y7kf9v3SDPnhxLNhcjFJbKECVnmZHTdo+IRO05sNLTH/D3vA6u1X32ehoLC7WFVdheg==" crossorigin="anonymous"></script>

<script type="application/javascript">define('jquery', [],function() {return window.jQuery;})</script>


</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../notebooks/01_hicexplorer.html">Notebooks</a></li><li class="breadcrumb-item"><a href="../notebooks/03_compartments.html"><span class="chapter-title">Compartments Analysis</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">Søren’s MSc project about Hi-C and spermatogenesis</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/munch-group/hic-spermatogenesis" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Project</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">Notebooks</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/01_hicexplorer.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">HiCExplorer using macaque data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/02_open2c_framework.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Open2C framework</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/03_compartments.html" class="sidebar-item-text sidebar-link active"><span class="chapter-title">Compartments Analysis</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false">
 <span class="menu-text">Workflows</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../scripts/master_workflow.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">master_workflow</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false">
 <span class="menu-text">Reports</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../reports/thoughts.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Thoughts and Prayers</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../results/Readme.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Result files</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false">
 <span class="menu-text">Slides</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../slides/index.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Project Progress Day: Msc. Project 2024</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="false">
 <span class="menu-text">Thesis</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../thesis/index.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Chromatin Compartments and Selection on X</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../thesis/supplementary.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Supplementary material</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#working-with-coolers" id="toc-working-with-coolers" class="nav-link active" data-scroll-target="#working-with-coolers"><span class="header-section-number">5</span> Working with coolers</a>
  <ul>
  <li><a href="#overview" id="toc-overview" class="nav-link" data-scroll-target="#overview"><span class="header-section-number">5.1</span> Overview</a>
  <ul class="collapse">
  <li><a href="#data-accessions" id="toc-data-accessions" class="nav-link" data-scroll-target="#data-accessions"><span class="header-section-number">5.1.1</span> Data (Accessions)</a></li>
  <li><a href="#folder-structure" id="toc-folder-structure" class="nav-link" data-scroll-target="#folder-structure"><span class="header-section-number">5.1.2</span> Folder structure</a></li>
  </ul></li>
  <li><a href="#fullmerge-pool-all-from-each-source_name" id="toc-fullmerge-pool-all-from-each-source_name" class="nav-link" data-scroll-target="#fullmerge-pool-all-from-each-source_name"><span class="header-section-number">5.2</span> FullMerge (pool all from each source_name)</a>
  <ul class="collapse">
  <li><a href="#create-cooler-dictionary-glob" id="toc-create-cooler-dictionary-glob" class="nav-link" data-scroll-target="#create-cooler-dictionary-glob"><span class="header-section-number">5.2.1</span> Create cooler dictionary (<code>glob</code>)</a></li>
  <li><a href="#merge-coolers" id="toc-merge-coolers" class="nav-link" data-scroll-target="#merge-coolers"><span class="header-section-number">5.2.2</span> Merge coolers</a></li>
  <li><a href="#zoomify-the-merged-cooler-files" id="toc-zoomify-the-merged-cooler-files" class="nav-link" data-scroll-target="#zoomify-the-merged-cooler-files"><span class="header-section-number">5.2.3</span> Zoomify the merged cooler files</a></li>
  <li><a href="#balance-the-matrices" id="toc-balance-the-matrices" class="nav-link" data-scroll-target="#balance-the-matrices"><span class="header-section-number">5.2.4</span> Balance the matrices</a></li>
  </ul></li>
  <li><a href="#repmerge-pool-all-from-each-biosample-id" id="toc-repmerge-pool-all-from-each-biosample-id" class="nav-link" data-scroll-target="#repmerge-pool-all-from-each-biosample-id"><span class="header-section-number">5.3</span> RepMerge (pool all from each BioSample ID)</a></li>
  </ul></li>
  <li><a href="#plotting" id="toc-plotting" class="nav-link" data-scroll-target="#plotting"><span class="header-section-number">6</span> Plotting</a>
  <ul>
  <li><a href="#kb-resolution" id="toc-kb-resolution" class="nav-link" data-scroll-target="#kb-resolution"><span class="header-section-number">6.1</span> 500kb resolution</a>
  <ul class="collapse">
  <li><a href="#example-with-a-single-sample" id="toc-example-with-a-single-sample" class="nav-link" data-scroll-target="#example-with-a-single-sample"><span class="header-section-number">6.1.1</span> Example with a single sample</a>
  <ul class="collapse">
  <li><a href="#imports" id="toc-imports" class="nav-link" data-scroll-target="#imports"><span class="header-section-number">6.1.1.1</span> Imports</a></li>
  <li><a href="#load-cooler" id="toc-load-cooler" class="nav-link" data-scroll-target="#load-cooler"><span class="header-section-number">6.1.1.2</span> Load cooler</a></li>
  <li><a href="#calculate-gc-covariance-from-the-reference-genome" id="toc-calculate-gc-covariance-from-the-reference-genome" class="nav-link" data-scroll-target="#calculate-gc-covariance-from-the-reference-genome"><span class="header-section-number">6.1.1.3</span> Calculate gc covariance (from the reference genome)</a></li>
  <li><a href="#calculate-the-e1-compartments" id="toc-calculate-the-e1-compartments" class="nav-link" data-scroll-target="#calculate-the-e1-compartments"><span class="header-section-number">6.1.1.4</span> Calculate the E1 compartments</a></li>
  <li><a href="#plot-e1-and-matrix" id="toc-plot-e1-and-matrix" class="nav-link" data-scroll-target="#plot-e1-and-matrix"><span class="header-section-number">6.1.1.5</span> Plot E1 and matrix</a></li>
  <li><a href="#stairs-plot-of-the-e1-compartments" id="toc-stairs-plot-of-the-e1-compartments" class="nav-link" data-scroll-target="#stairs-plot-of-the-e1-compartments"><span class="header-section-number">6.1.1.6</span> Stairs plot of the E1 compartments</a></li>
  </ul></li>
  <li><a href="#all-5-full-merges" id="toc-all-5-full-merges" class="nav-link" data-scroll-target="#all-5-full-merges"><span class="header-section-number">6.1.2</span> All 5 full merges</a>
  <ul class="collapse">
  <li><a href="#load-coolers" id="toc-load-coolers" class="nav-link" data-scroll-target="#load-coolers"><span class="header-section-number">6.1.2.1</span> Load coolers</a></li>
  <li><a href="#calculate-gc-covariance-from-the-reference-genome-1" id="toc-calculate-gc-covariance-from-the-reference-genome-1" class="nav-link" data-scroll-target="#calculate-gc-covariance-from-the-reference-genome-1"><span class="header-section-number">6.1.2.2</span> Calculate gc covariance (from the reference genome)</a></li>
  <li><a href="#calculate-the-e1-compartments-1" id="toc-calculate-the-e1-compartments-1" class="nav-link" data-scroll-target="#calculate-the-e1-compartments-1"><span class="header-section-number">6.1.2.3</span> Calculate the E1 compartments</a></li>
  <li><a href="#plot-the-e1-compartments" id="toc-plot-the-e1-compartments" class="nav-link" data-scroll-target="#plot-the-e1-compartments"><span class="header-section-number">6.1.2.4</span> Plot the E1 compartments</a></li>
  <li><a href="#plot-matrices-with-compartments-round-spermatid" id="toc-plot-matrices-with-compartments-round-spermatid" class="nav-link" data-scroll-target="#plot-matrices-with-compartments-round-spermatid"><span class="header-section-number">6.1.2.5</span> Plot matrices with compartments (round spermatid)</a></li>
  <li><a href="#sliding-window-summed-e1-compartments" id="toc-sliding-window-summed-e1-compartments" class="nav-link" data-scroll-target="#sliding-window-summed-e1-compartments"><span class="header-section-number">6.1.2.6</span> Sliding window summed E1 compartments</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#kb-resolution-1" id="toc-kb-resolution-1" class="nav-link" data-scroll-target="#kb-resolution-1"><span class="header-section-number">6.2</span> 100kb resolution</a>
  <ul class="collapse">
  <li><a href="#all-5-full-merges-1" id="toc-all-5-full-merges-1" class="nav-link" data-scroll-target="#all-5-full-merges-1"><span class="header-section-number">6.2.1</span> All 5 full merges</a>
  <ul class="collapse">
  <li><a href="#load-coolers-1" id="toc-load-coolers-1" class="nav-link" data-scroll-target="#load-coolers-1"><span class="header-section-number">6.2.1.1</span> Load coolers</a></li>
  <li><a href="#calculate-gc-covariance-from-the-reference-genome-2" id="toc-calculate-gc-covariance-from-the-reference-genome-2" class="nav-link" data-scroll-target="#calculate-gc-covariance-from-the-reference-genome-2"><span class="header-section-number">6.2.1.2</span> Calculate gc covariance (from the reference genome)</a></li>
  <li><a href="#calculate-the-e1-compartments-2" id="toc-calculate-the-e1-compartments-2" class="nav-link" data-scroll-target="#calculate-the-e1-compartments-2"><span class="header-section-number">6.2.1.3</span> Calculate the E1 compartments</a></li>
  <li><a href="#plot-the-e1-compartments-1" id="toc-plot-the-e1-compartments-1" class="nav-link" data-scroll-target="#plot-the-e1-compartments-1"><span class="header-section-number">6.2.1.4</span> Plot the E1 compartments</a></li>
  <li><a href="#plot-matrices-with-compartments" id="toc-plot-matrices-with-compartments" class="nav-link" data-scroll-target="#plot-matrices-with-compartments"><span class="header-section-number">6.2.1.5</span> Plot matrices with compartments</a></li>
  </ul></li>
  <li><a href="#smooth-the-observedexpected-matrix" id="toc-smooth-the-observedexpected-matrix" class="nav-link" data-scroll-target="#smooth-the-observedexpected-matrix"><span class="header-section-number">6.2.2</span> Smooth the Observed/Expected matrix</a>
  <ul class="collapse">
  <li><a href="#calculate-gc-covariance-from-the-reference-genome-3" id="toc-calculate-gc-covariance-from-the-reference-genome-3" class="nav-link" data-scroll-target="#calculate-gc-covariance-from-the-reference-genome-3"><span class="header-section-number">6.2.2.1</span> Calculate gc covariance (from the reference genome)</a></li>
  </ul></li>
  </ul></li>
  </ul></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.dev/munch-group/hic-spermatogenesis/blob/main/notebooks/03_compartments.ipynb" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/munch-group/hic-spermatogenesis/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li><li><a href="https://github.com/munch-group/hic-spermatogenesis/blob/main/notebooks/03_compartments.ipynb" class="toc-action"><i class="bi empty"></i>View source</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content page-columns page-full" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../notebooks/01_hicexplorer.html">Notebooks</a></li><li class="breadcrumb-item"><a href="../notebooks/03_compartments.html"><span class="chapter-title">Compartments Analysis</span></a></li></ol></nav>
<div class="quarto-title">
<h1 class="title"><span class="chapter-title">Compartments Analysis</span></h1>
<p class="subtitle lead">Analysis of Chromatin Compartments using the Open2C Ecosystem</p>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="working-with-coolers" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> Working with coolers</h1>
<p>In this notebook, we use files generated by the workflow <code>master_workflow</code> that, in short, does the following:</p>
<ol type="1">
<li>Download the data from the source</li>
<li>Index the reference genome with <code>bwa index</code> and <code>samtools faidx</code></li>
<li>Align the reads to the reference genome with <code>bwa mem</code></li>
<li>Pair and sort the reads to <code>.pairs</code> files with <code>pairtools parse | pairtools sort</code></li>
<li>Deduplicate the pairs with <code>pairtools dedup</code></li>
<li>Convert the .pairs to cooler files with <code>cooler cload pairs</code></li>
</ol>
<p>We will:</p>
<ol type="1">
<li>Load the cooler files</li>
<li>Merge the coolers from the same BioSample ID –&gt; Create ‘replicates’</li>
<li>Zoomify the merged cooler files (coarsen) to create a multicooler (.mcool) file</li>
<li>Balance the matrices (use commandline <code>cooler balance</code>)</li>
<li>Calculate E1 compartments with <code>cooltools compute_cis_eig</code></li>
</ol>
<section id="overview" class="level2" data-number="5.1">
<h2 data-number="5.1" class="anchored" data-anchor-id="overview"><span class="header-section-number">5.1</span> Overview</h2>
<section id="data-accessions" class="level3" data-number="5.1.1">
<h3 data-number="5.1.1" class="anchored" data-anchor-id="data-accessions"><span class="header-section-number">5.1.1</span> Data (Accessions)</h3>
<div id="cell-md-data-accessions" class="cell" data-execution_count="3">
<div id="md-data-accessions" class="cell-output cell-output-display cell-output-markdown" data-execution_count="3">
<p>To get an overview of the data accessions used in this analysis, we will first summarize the <code>SRA-runtable.tsv</code> that contains the accession numbers and some metadata for each sample (<a href="../thesis/index.html#tbl-runtable-summary" class="quarto-xref">Table&nbsp;<span>9.1</span></a>).</p>
</div>
</div>
<div class="cell" data-execution_count="5">
<div id="tbl-runtable" class="cell quarto-float quarto-figure quarto-figure-center anchored" data-execution_count="5">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-runtable-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;5.1: The most relevant columns of the <code>SRA-runtable.tsv</code> file
</figcaption>
<div aria-describedby="tbl-runtable-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output cell-output-display">
<style type="text/css">
</style>

<table id="T_bbc3f" class="do-not-create-environment cell caption-top table table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th class="blank level0" data-quarto-table-cell-role="th">&nbsp;</th>
<th id="T_bbc3f_level0_col0" class="col_heading level0 col0" data-quarto-table-cell-role="th">source_name</th>
<th id="T_bbc3f_level0_col1" class="col_heading level0 col1" data-quarto-table-cell-role="th">BioSample</th>
<th id="T_bbc3f_level0_col2" class="col_heading level0 col2" data-quarto-table-cell-role="th">Run</th>
<th id="T_bbc3f_level0_col3" class="col_heading level0 col3" data-quarto-table-cell-role="th">GB</th>
<th id="T_bbc3f_level0_col4" class="col_heading level0 col4" data-quarto-table-cell-role="th">Bases</th>
<th id="T_bbc3f_level0_col5" class="col_heading level0 col5" data-quarto-table-cell-role="th">Reads</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td id="T_bbc3f_level0_row0" class="row_heading level0 row0" data-quarto-table-cell-role="th">16</td>
<td id="T_bbc3f_row0_col0" class="data row0 col0">fibroblast</td>
<td id="T_bbc3f_row0_col1" class="data row0 col1">SAMN08375237</td>
<td id="T_bbc3f_row0_col2" class="data row0 col2">SRR6502335</td>
<td id="T_bbc3f_row0_col3" class="data row0 col3">29.771059</td>
<td id="T_bbc3f_row0_col4" class="data row0 col4">73,201,141,800</td>
<td id="T_bbc3f_row0_col5" class="data row0 col5">244,003,806</td>
</tr>
<tr class="even">
<td id="T_bbc3f_level0_row1" class="row_heading level0 row1" data-quarto-table-cell-role="th">17</td>
<td id="T_bbc3f_row1_col0" class="data row1 col0">fibroblast</td>
<td id="T_bbc3f_row1_col1" class="data row1 col1">SAMN08375237</td>
<td id="T_bbc3f_row1_col2" class="data row1 col2">SRR6502336</td>
<td id="T_bbc3f_row1_col3" class="data row1 col3">22.755361</td>
<td id="T_bbc3f_row1_col4" class="data row1 col4">65,119,970,100</td>
<td id="T_bbc3f_row1_col5" class="data row1 col5">217,066,567</td>
</tr>
<tr class="odd">
<td id="T_bbc3f_level0_row2" class="row_heading level0 row2" data-quarto-table-cell-role="th">18</td>
<td id="T_bbc3f_row2_col0" class="data row2 col0">fibroblast</td>
<td id="T_bbc3f_row2_col1" class="data row2 col1">SAMN08375236</td>
<td id="T_bbc3f_row2_col2" class="data row2 col2">SRR6502337</td>
<td id="T_bbc3f_row2_col3" class="data row2 col3">21.434722</td>
<td id="T_bbc3f_row2_col4" class="data row2 col4">52,769,196,300</td>
<td id="T_bbc3f_row2_col5" class="data row2 col5">175,897,321</td>
</tr>
<tr class="even">
<td id="T_bbc3f_level0_row3" class="row_heading level0 row3" data-quarto-table-cell-role="th">19</td>
<td id="T_bbc3f_row3_col0" class="data row3 col0">fibroblast</td>
<td id="T_bbc3f_row3_col1" class="data row3 col1">SAMN08375236</td>
<td id="T_bbc3f_row3_col2" class="data row3 col2">SRR6502338</td>
<td id="T_bbc3f_row3_col3" class="data row3 col3">21.420030</td>
<td id="T_bbc3f_row3_col4" class="data row3 col4">52,378,949,100</td>
<td id="T_bbc3f_row3_col5" class="data row3 col5">174,596,497</td>
</tr>
<tr class="odd">
<td id="T_bbc3f_level0_row4" class="row_heading level0 row4" data-quarto-table-cell-role="th">20</td>
<td id="T_bbc3f_row4_col0" class="data row4 col0">fibroblast</td>
<td id="T_bbc3f_row4_col1" class="data row4 col1">SAMN08375236</td>
<td id="T_bbc3f_row4_col2" class="data row4 col2">SRR6502339</td>
<td id="T_bbc3f_row4_col3" class="data row4 col3">10.207410</td>
<td id="T_bbc3f_row4_col4" class="data row4 col4">28,885,941,600</td>
<td id="T_bbc3f_row4_col5" class="data row4 col5">96,286,472</td>
</tr>
<tr class="even">
<td id="T_bbc3f_level0_row5" class="row_heading level0 row5" data-quarto-table-cell-role="th">9</td>
<td id="T_bbc3f_row5_col0" class="data row5 col0">fibroblast</td>
<td id="T_bbc3f_row5_col1" class="data row5 col1">SAMN08375237</td>
<td id="T_bbc3f_row5_col2" class="data row5 col2">SRR7349189</td>
<td id="T_bbc3f_row5_col3" class="data row5 col3">52.729173</td>
<td id="T_bbc3f_row5_col4" class="data row5 col4">139,604,854,200</td>
<td id="T_bbc3f_row5_col5" class="data row5 col5">465,349,514</td>
</tr>
<tr class="odd">
<td id="T_bbc3f_level0_row6" class="row_heading level0 row6" data-quarto-table-cell-role="th">10</td>
<td id="T_bbc3f_row6_col0" class="data row6 col0">fibroblast</td>
<td id="T_bbc3f_row6_col1" class="data row6 col1">SAMN08375236</td>
<td id="T_bbc3f_row6_col2" class="data row6 col2">SRR7349190</td>
<td id="T_bbc3f_row6_col3" class="data row6 col3">53.085520</td>
<td id="T_bbc3f_row6_col4" class="data row6 col4">142,008,353,400</td>
<td id="T_bbc3f_row6_col5" class="data row6 col5">473,361,178</td>
</tr>
<tr class="even">
<td id="T_bbc3f_level0_row7" class="row_heading level0 row7" data-quarto-table-cell-role="th">21</td>
<td id="T_bbc3f_row7_col0" class="data row7 col0">pachytene spermatocyte</td>
<td id="T_bbc3f_row7_col1" class="data row7 col1">SAMN08375234</td>
<td id="T_bbc3f_row7_col2" class="data row7 col2">SRR6502342</td>
<td id="T_bbc3f_row7_col3" class="data row7 col3">60.258880</td>
<td id="T_bbc3f_row7_col4" class="data row7 col4">150,370,993,500</td>
<td id="T_bbc3f_row7_col5" class="data row7 col5">501,236,645</td>
</tr>
<tr class="odd">
<td id="T_bbc3f_level0_row8" class="row_heading level0 row8" data-quarto-table-cell-role="th">22</td>
<td id="T_bbc3f_row8_col0" class="data row8 col0">pachytene spermatocyte</td>
<td id="T_bbc3f_row8_col1" class="data row8 col1">SAMN08375234</td>
<td id="T_bbc3f_row8_col2" class="data row8 col2">SRR6502344</td>
<td id="T_bbc3f_row8_col3" class="data row8 col3">27.146048</td>
<td id="T_bbc3f_row8_col4" class="data row8 col4">65,697,684,300</td>
<td id="T_bbc3f_row8_col5" class="data row8 col5">218,992,281</td>
</tr>
<tr class="even">
<td id="T_bbc3f_level0_row9" class="row_heading level0 row9" data-quarto-table-cell-role="th">23</td>
<td id="T_bbc3f_row9_col0" class="data row9 col0">pachytene spermatocyte</td>
<td id="T_bbc3f_row9_col1" class="data row9 col1">SAMN08375234</td>
<td id="T_bbc3f_row9_col2" class="data row9 col2">SRR6502345</td>
<td id="T_bbc3f_row9_col3" class="data row9 col3">26.202707</td>
<td id="T_bbc3f_row9_col4" class="data row9 col4">63,490,538,700</td>
<td id="T_bbc3f_row9_col5" class="data row9 col5">211,635,129</td>
</tr>
<tr class="odd">
<td id="T_bbc3f_level0_row10" class="row_heading level0 row10" data-quarto-table-cell-role="th">0</td>
<td id="T_bbc3f_row10_col0" class="data row10 col0">pachytene spermatocyte</td>
<td id="T_bbc3f_row10_col1" class="data row10 col1">SAMN09427370</td>
<td id="T_bbc3f_row10_col2" class="data row10 col2">SRR7345458</td>
<td id="T_bbc3f_row10_col3" class="data row10 col3">55.970557</td>
<td id="T_bbc3f_row10_col4" class="data row10 col4">153,281,577,900</td>
<td id="T_bbc3f_row10_col5" class="data row10 col5">510,938,593</td>
</tr>
<tr class="even">
<td id="T_bbc3f_level0_row11" class="row_heading level0 row11" data-quarto-table-cell-role="th">1</td>
<td id="T_bbc3f_row11_col0" class="data row11 col0">pachytene spermatocyte</td>
<td id="T_bbc3f_row11_col1" class="data row11 col1">SAMN09427370</td>
<td id="T_bbc3f_row11_col2" class="data row11 col2">SRR7345459</td>
<td id="T_bbc3f_row11_col3" class="data row11 col3">53.982492</td>
<td id="T_bbc3f_row11_col4" class="data row11 col4">144,993,841,200</td>
<td id="T_bbc3f_row11_col5" class="data row11 col5">483,312,804</td>
</tr>
<tr class="odd">
<td id="T_bbc3f_level0_row12" class="row_heading level0 row12" data-quarto-table-cell-role="th">11</td>
<td id="T_bbc3f_row12_col0" class="data row12 col0">pachytene spermatocyte</td>
<td id="T_bbc3f_row12_col1" class="data row12 col1">SAMN08375235</td>
<td id="T_bbc3f_row12_col2" class="data row12 col2">SRR7349191</td>
<td id="T_bbc3f_row12_col3" class="data row12 col3">51.274476</td>
<td id="T_bbc3f_row12_col4" class="data row12 col4">137,821,979,100</td>
<td id="T_bbc3f_row12_col5" class="data row12 col5">459,406,597</td>
</tr>
<tr class="even">
<td id="T_bbc3f_level0_row13" class="row_heading level0 row13" data-quarto-table-cell-role="th">24</td>
<td id="T_bbc3f_row13_col0" class="data row13 col0">round spermatid</td>
<td id="T_bbc3f_row13_col1" class="data row13 col1">SAMN08375232</td>
<td id="T_bbc3f_row13_col2" class="data row13 col2">SRR6502351</td>
<td id="T_bbc3f_row13_col3" class="data row13 col3">20.924497</td>
<td id="T_bbc3f_row13_col4" class="data row13 col4">55,095,075,300</td>
<td id="T_bbc3f_row13_col5" class="data row13 col5">183,650,251</td>
</tr>
<tr class="odd">
<td id="T_bbc3f_level0_row14" class="row_heading level0 row14" data-quarto-table-cell-role="th">25</td>
<td id="T_bbc3f_row14_col0" class="data row14 col0">round spermatid</td>
<td id="T_bbc3f_row14_col1" class="data row14 col1">SAMN08375232</td>
<td id="T_bbc3f_row14_col2" class="data row14 col2">SRR6502352</td>
<td id="T_bbc3f_row14_col3" class="data row14 col3">41.133960</td>
<td id="T_bbc3f_row14_col4" class="data row14 col4">115,578,475,800</td>
<td id="T_bbc3f_row14_col5" class="data row14 col5">385,261,586</td>
</tr>
<tr class="even">
<td id="T_bbc3f_level0_row15" class="row_heading level0 row15" data-quarto-table-cell-role="th">26</td>
<td id="T_bbc3f_row15_col0" class="data row15 col0">round spermatid</td>
<td id="T_bbc3f_row15_col1" class="data row15 col1">SAMN08375232</td>
<td id="T_bbc3f_row15_col2" class="data row15 col2">SRR6502353</td>
<td id="T_bbc3f_row15_col3" class="data row15 col3">36.444117</td>
<td id="T_bbc3f_row15_col4" class="data row15 col4">96,195,161,400</td>
<td id="T_bbc3f_row15_col5" class="data row15 col5">320,650,538</td>
</tr>
<tr class="odd">
<td id="T_bbc3f_level0_row16" class="row_heading level0 row16" data-quarto-table-cell-role="th">2</td>
<td id="T_bbc3f_row16_col0" class="data row16 col0">round spermatid</td>
<td id="T_bbc3f_row16_col1" class="data row16 col1">SAMN09427369</td>
<td id="T_bbc3f_row16_col2" class="data row16 col2">SRR7345460</td>
<td id="T_bbc3f_row16_col3" class="data row16 col3">38.244654</td>
<td id="T_bbc3f_row16_col4" class="data row16 col4">104,105,827,200</td>
<td id="T_bbc3f_row16_col5" class="data row16 col5">347,019,424</td>
</tr>
<tr class="even">
<td id="T_bbc3f_level0_row17" class="row_heading level0 row17" data-quarto-table-cell-role="th">3</td>
<td id="T_bbc3f_row17_col0" class="data row17 col0">round spermatid</td>
<td id="T_bbc3f_row17_col1" class="data row17 col1">SAMN09427369</td>
<td id="T_bbc3f_row17_col2" class="data row17 col2">SRR7345461</td>
<td id="T_bbc3f_row17_col3" class="data row17 col3">53.996261</td>
<td id="T_bbc3f_row17_col4" class="data row17 col4">144,532,309,500</td>
<td id="T_bbc3f_row17_col5" class="data row17 col5">481,774,365</td>
</tr>
<tr class="odd">
<td id="T_bbc3f_level0_row18" class="row_heading level0 row18" data-quarto-table-cell-role="th">12</td>
<td id="T_bbc3f_row18_col0" class="data row18 col0">round spermatid</td>
<td id="T_bbc3f_row18_col1" class="data row18 col1">SAMN08375232</td>
<td id="T_bbc3f_row18_col2" class="data row18 col2">SRR7349192</td>
<td id="T_bbc3f_row18_col3" class="data row18 col3">52.384556</td>
<td id="T_bbc3f_row18_col4" class="data row18 col4">140,431,608,000</td>
<td id="T_bbc3f_row18_col5" class="data row18 col5">468,105,360</td>
</tr>
<tr class="even">
<td id="T_bbc3f_level0_row19" class="row_heading level0 row19" data-quarto-table-cell-role="th">29</td>
<td id="T_bbc3f_row19_col0" class="data row19 col0">sperm</td>
<td id="T_bbc3f_row19_col1" class="data row19 col1">SAMN08375229</td>
<td id="T_bbc3f_row19_col2" class="data row19 col2">SRR6502360</td>
<td id="T_bbc3f_row19_col3" class="data row19 col3">26.653940</td>
<td id="T_bbc3f_row19_col4" class="data row19 col4">64,752,370,800</td>
<td id="T_bbc3f_row19_col5" class="data row19 col5">215,841,236</td>
</tr>
<tr class="odd">
<td id="T_bbc3f_level0_row20" class="row_heading level0 row20" data-quarto-table-cell-role="th">30</td>
<td id="T_bbc3f_row20_col0" class="data row20 col0">sperm</td>
<td id="T_bbc3f_row20_col1" class="data row20 col1">SAMN08375228</td>
<td id="T_bbc3f_row20_col2" class="data row20 col2">SRR6502362</td>
<td id="T_bbc3f_row20_col3" class="data row20 col3">23.973440</td>
<td id="T_bbc3f_row20_col4" class="data row20 col4">58,369,232,700</td>
<td id="T_bbc3f_row20_col5" class="data row20 col5">194,564,109</td>
</tr>
<tr class="even">
<td id="T_bbc3f_level0_row21" class="row_heading level0 row21" data-quarto-table-cell-role="th">13</td>
<td id="T_bbc3f_row21_col0" class="data row21 col0">sperm</td>
<td id="T_bbc3f_row21_col1" class="data row21 col1">SAMN08375229</td>
<td id="T_bbc3f_row21_col2" class="data row21 col2">SRR7349193</td>
<td id="T_bbc3f_row21_col3" class="data row21 col3">52.806276</td>
<td id="T_bbc3f_row21_col4" class="data row21 col4">141,148,572,300</td>
<td id="T_bbc3f_row21_col5" class="data row21 col5">470,495,241</td>
</tr>
<tr class="odd">
<td id="T_bbc3f_level0_row22" class="row_heading level0 row22" data-quarto-table-cell-role="th">14</td>
<td id="T_bbc3f_row22_col0" class="data row22 col0">sperm</td>
<td id="T_bbc3f_row22_col1" class="data row22 col1">SAMN08375229</td>
<td id="T_bbc3f_row22_col2" class="data row22 col2">SRR7349195</td>
<td id="T_bbc3f_row22_col3" class="data row22 col3">22.444378</td>
<td id="T_bbc3f_row22_col4" class="data row22 col4">60,523,788,600</td>
<td id="T_bbc3f_row22_col5" class="data row22 col5">201,745,962</td>
</tr>
<tr class="even">
<td id="T_bbc3f_level0_row23" class="row_heading level0 row23" data-quarto-table-cell-role="th">15</td>
<td id="T_bbc3f_row23_col0" class="data row23 col0">sperm</td>
<td id="T_bbc3f_row23_col1" class="data row23 col1">SAMN08375229</td>
<td id="T_bbc3f_row23_col2" class="data row23 col2">SRR7349196</td>
<td id="T_bbc3f_row23_col3" class="data row23 col3">38.253606</td>
<td id="T_bbc3f_row23_col4" class="data row23 col4">104,119,671,000</td>
<td id="T_bbc3f_row23_col5" class="data row23 col5">347,065,570</td>
</tr>
<tr class="odd">
<td id="T_bbc3f_level0_row24" class="row_heading level0 row24" data-quarto-table-cell-role="th">27</td>
<td id="T_bbc3f_row24_col0" class="data row24 col0">spermatogonia</td>
<td id="T_bbc3f_row24_col1" class="data row24 col1">SAMN08375231</td>
<td id="T_bbc3f_row24_col2" class="data row24 col2">SRR6502356</td>
<td id="T_bbc3f_row24_col3" class="data row24 col3">22.845286</td>
<td id="T_bbc3f_row24_col4" class="data row24 col4">58,909,579,800</td>
<td id="T_bbc3f_row24_col5" class="data row24 col5">196,365,266</td>
</tr>
<tr class="even">
<td id="T_bbc3f_level0_row25" class="row_heading level0 row25" data-quarto-table-cell-role="th">28</td>
<td id="T_bbc3f_row25_col0" class="data row25 col0">spermatogonia</td>
<td id="T_bbc3f_row25_col1" class="data row25 col1">SAMN08375231</td>
<td id="T_bbc3f_row25_col2" class="data row25 col2">SRR6502357</td>
<td id="T_bbc3f_row25_col3" class="data row25 col3">17.947471</td>
<td id="T_bbc3f_row25_col4" class="data row25 col4">46,888,332,900</td>
<td id="T_bbc3f_row25_col5" class="data row25 col5">156,294,443</td>
</tr>
<tr class="odd">
<td id="T_bbc3f_level0_row26" class="row_heading level0 row26" data-quarto-table-cell-role="th">4</td>
<td id="T_bbc3f_row26_col0" class="data row26 col0">spermatogonia</td>
<td id="T_bbc3f_row26_col1" class="data row26 col1">SAMN09427379</td>
<td id="T_bbc3f_row26_col2" class="data row26 col2">SRR7345462</td>
<td id="T_bbc3f_row26_col3" class="data row26 col3">18.686342</td>
<td id="T_bbc3f_row26_col4" class="data row26 col4">52,032,780,000</td>
<td id="T_bbc3f_row26_col5" class="data row26 col5">173,442,600</td>
</tr>
<tr class="even">
<td id="T_bbc3f_level0_row27" class="row_heading level0 row27" data-quarto-table-cell-role="th">5</td>
<td id="T_bbc3f_row27_col0" class="data row27 col0">spermatogonia</td>
<td id="T_bbc3f_row27_col1" class="data row27 col1">SAMN09427379</td>
<td id="T_bbc3f_row27_col2" class="data row27 col2">SRR7345463</td>
<td id="T_bbc3f_row27_col3" class="data row27 col3">29.956561</td>
<td id="T_bbc3f_row27_col4" class="data row27 col4">82,384,836,000</td>
<td id="T_bbc3f_row27_col5" class="data row27 col5">274,616,120</td>
</tr>
<tr class="odd">
<td id="T_bbc3f_level0_row28" class="row_heading level0 row28" data-quarto-table-cell-role="th">6</td>
<td id="T_bbc3f_row28_col0" class="data row28 col0">spermatogonia</td>
<td id="T_bbc3f_row28_col1" class="data row28 col1">SAMN09427379</td>
<td id="T_bbc3f_row28_col2" class="data row28 col2">SRR7345464</td>
<td id="T_bbc3f_row28_col3" class="data row28 col3">39.145759</td>
<td id="T_bbc3f_row28_col4" class="data row28 col4">105,153,716,100</td>
<td id="T_bbc3f_row28_col5" class="data row28 col5">350,512,387</td>
</tr>
<tr class="even">
<td id="T_bbc3f_level0_row29" class="row_heading level0 row29" data-quarto-table-cell-role="th">7</td>
<td id="T_bbc3f_row29_col0" class="data row29 col0">spermatogonia</td>
<td id="T_bbc3f_row29_col1" class="data row29 col1">SAMN09427378</td>
<td id="T_bbc3f_row29_col2" class="data row29 col2">SRR7345465</td>
<td id="T_bbc3f_row29_col3" class="data row29 col3">35.816184</td>
<td id="T_bbc3f_row29_col4" class="data row29 col4">96,048,594,600</td>
<td id="T_bbc3f_row29_col5" class="data row29 col5">320,161,982</td>
</tr>
<tr class="odd">
<td id="T_bbc3f_level0_row30" class="row_heading level0 row30" data-quarto-table-cell-role="th">8</td>
<td id="T_bbc3f_row30_col0" class="data row30 col0">spermatogonia</td>
<td id="T_bbc3f_row30_col1" class="data row30 col1">SAMN09427378</td>
<td id="T_bbc3f_row30_col2" class="data row30 col2">SRR7345467</td>
<td id="T_bbc3f_row30_col3" class="data row30 col3">28.396816</td>
<td id="T_bbc3f_row30_col4" class="data row30 col4">77,248,140,900</td>
<td id="T_bbc3f_row30_col5" class="data row30 col5">257,493,803</td>
</tr>
</tbody>
</table>
</div>
</div>
</figure>
</div>
</div>
<div class="cell">
<div id="tbl-runtable-summary" class="cell quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-runtable-summary-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;5.2: Summary of the data accessions used in this analysis
</figcaption>
<div aria-describedby="tbl-runtable-summary-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output cell-output-display">
<style type="text/css">
</style>

<table id="T_0aa9c" class="do-not-create-environment cell caption-top table table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th class="blank level0" data-quarto-table-cell-role="th">&nbsp;</th>
<th id="T_0aa9c_level0_col0" class="col_heading level0 col0" data-quarto-table-cell-role="th">source_name</th>
<th id="T_0aa9c_level0_col1" class="col_heading level0 col1" data-quarto-table-cell-role="th">GB</th>
<th id="T_0aa9c_level0_col2" class="col_heading level0 col2" data-quarto-table-cell-role="th">Bases</th>
<th id="T_0aa9c_level0_col3" class="col_heading level0 col3" data-quarto-table-cell-role="th">Reads</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td id="T_0aa9c_level0_row0" class="row_heading level0 row0" data-quarto-table-cell-role="th">0</td>
<td id="T_0aa9c_row0_col0" class="data row0 col0">fibroblast</td>
<td id="T_0aa9c_row0_col1" class="data row0 col1">211.403275</td>
<td id="T_0aa9c_row0_col2" class="data row0 col2">553,968,406,500</td>
<td id="T_0aa9c_row0_col3" class="data row0 col3">1,846,561,355</td>
</tr>
<tr class="even">
<td id="T_0aa9c_level0_row1" class="row_heading level0 row1" data-quarto-table-cell-role="th">1</td>
<td id="T_0aa9c_row1_col0" class="data row1 col0">pachytene spermatocyte</td>
<td id="T_0aa9c_row1_col1" class="data row1 col1">274.835160</td>
<td id="T_0aa9c_row1_col2" class="data row1 col2">715,656,614,700</td>
<td id="T_0aa9c_row1_col3" class="data row1 col3">2,385,522,049</td>
</tr>
<tr class="odd">
<td id="T_0aa9c_level0_row2" class="row_heading level0 row2" data-quarto-table-cell-role="th">2</td>
<td id="T_0aa9c_row2_col0" class="data row2 col0">round spermatid</td>
<td id="T_0aa9c_row2_col1" class="data row2 col1">243.128044</td>
<td id="T_0aa9c_row2_col2" class="data row2 col2">655,938,457,200</td>
<td id="T_0aa9c_row2_col3" class="data row2 col3">2,186,461,524</td>
</tr>
<tr class="even">
<td id="T_0aa9c_level0_row3" class="row_heading level0 row3" data-quarto-table-cell-role="th">3</td>
<td id="T_0aa9c_row3_col0" class="data row3 col0">sperm</td>
<td id="T_0aa9c_row3_col1" class="data row3 col1">164.131640</td>
<td id="T_0aa9c_row3_col2" class="data row3 col2">428,913,635,400</td>
<td id="T_0aa9c_row3_col3" class="data row3 col3">1,429,712,118</td>
</tr>
<tr class="odd">
<td id="T_0aa9c_level0_row4" class="row_heading level0 row4" data-quarto-table-cell-role="th">4</td>
<td id="T_0aa9c_row4_col0" class="data row4 col0">spermatogonia</td>
<td id="T_0aa9c_row4_col1" class="data row4 col1">192.794420</td>
<td id="T_0aa9c_row4_col2" class="data row4 col2">518,665,980,300</td>
<td id="T_0aa9c_row4_col3" class="data row4 col3">1,728,886,601</td>
</tr>
</tbody>
</table>
</div>
</div>
</figure>
</div>
</div>
</section>
<section id="folder-structure" class="level3" data-number="5.1.2">
<h3 data-number="5.1.2" class="anchored" data-anchor-id="folder-structure"><span class="header-section-number">5.1.2</span> Folder structure</h3>
<div id="cell-md-folder-structure" class="cell" data-execution_count="7">
<div id="md-folder-structure" class="cell-output cell-output-display cell-output-markdown" data-execution_count="7">
<p>For ease of mind, here is the folder structure of the project. <code>../steps/bwa/PE/</code> is the base directory artificially defined in the <code>master_workflow.py</code>. It ccould be any other directory inside <code>steps</code>. It is defined relative to the <code>master_workflow.py</code> file (inside the worklow), and converted to an absolute path by python.</p>
</div>
</div>
<div id="fig-folder-structure" class="cell quarto-float quarto-figure quarto-figure-center anchored" data-execution_count="12">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-folder-structure-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output cell-output-stdout">
<pre><code>../steps/bwa/PE
├── bamfiles
│&nbsp;&nbsp; ├── fibroblast
│&nbsp;&nbsp; ├── pachytene_spermatocyte
│&nbsp;&nbsp; ├── round_spermatid
│&nbsp;&nbsp; ├── sperm
│&nbsp;&nbsp; └── spermatogonia
├── cool
│&nbsp;&nbsp; ├── fibroblast
│&nbsp;&nbsp; ├── pachytene_spermatocyte
│&nbsp;&nbsp; ├── round_spermatid
│&nbsp;&nbsp; ├── sperm
│&nbsp;&nbsp; └── spermatogonia
└── pairs
    ├── fibroblast
    ├── pachytene_spermatocyte
    ├── round_spermatid
    ├── sperm
    └── spermatogonia

18 directories</code></pre>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-folder-structure-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5.1
</figcaption>
</figure>
</div>
</section>
</section>
<section id="fullmerge-pool-all-from-each-source_name" class="level2" data-number="5.2">
<h2 data-number="5.2" class="anchored" data-anchor-id="fullmerge-pool-all-from-each-source_name"><span class="header-section-number">5.2</span> FullMerge (pool all from each source_name)</h2>
<div id="cell-md-fullmerge-coolers" class="cell" data-execution_count="10">
<div id="md-fullmerge-coolers" class="cell-output cell-output-display cell-output-markdown" data-execution_count="10">
<p>We will use <code>cooler merge</code> to merge all samples in each sub-folder (cell type) to just one interaction matrix for each cell type. The reason for that is that we choose to trust <span class="citation" data-cites="wang_reprogramming_2019">(<a href="../thesis/index.html#ref-wang_reprogramming_2019" role="doc-biblioref">Wang et al. 2019</a>)</span> when they say that compartments are highly reproducible between replicates, and by merging all replicates, we will have a more robust signal.</p>
<p>Thus, we will merge all samples from the same source_name into a single cooler file. The coolers are produced to only contain the real chromosomes (1-22, X, Y), and not the mitochondrial DNA or the unplaced contigs (~2,900), as they would not bring any information to the analysis.</p>
<p><strong>Overview of this section:</strong></p>
<ul>
<li>Locate the coolers (<code>glob</code> –&gt; dictionary)</li>
<li>Merge the coolers (<code>cooler.merge_coolers</code>)</li>
<li>Zoomify the merged cooler (<code>cooler.zoomify_cooler</code>) to resolutions: 10kb, 50kb, 100kb, 500kb.</li>
<li>Balance the matrices (<code>!cooler balance</code>) (use the CLI, as it is more easily parallelized)</li>
</ul>
</div>
</div>
<section id="create-cooler-dictionary-glob" class="level3" data-number="5.2.1">
<h3 data-number="5.2.1" class="anchored" data-anchor-id="create-cooler-dictionary-glob"><span class="header-section-number">5.2.1</span> Create cooler dictionary (<code>glob</code>)</h3>
<div id="cell-md-create-cooler-dict" class="cell" data-execution_count="33">
<div id="md-create-cooler-dict" class="cell-output cell-output-display cell-output-markdown" data-execution_count="33">
<p>First, we will create a dictionary with the paths to the coolers for each sample. We use <code>glob.glob</code> to fetch all the coolers in each sub-folder that remain after filtering and (automatic) quality control (those are the ones with <code>nodups</code> in their names).</p>
</div>
</div>
<div class="cell" data-execution_count="27">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> glob</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os.path <span class="im">as</span> op</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> pprint <span class="im">import</span> pprint <span class="im">as</span> pp</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the list of cell type dirs</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>base_dir <span class="op">=</span> <span class="st">'../steps/bwa/PE/cool'</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>folders <span class="op">=</span> glob.glob(op.join(base_dir, <span class="st">'*'</span>))</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>files_dict <span class="op">=</span> {f:glob.glob(<span class="ss">f"</span><span class="sc">{</span>f<span class="sc">}</span><span class="ss">/*.nodups.*"</span>) <span class="cf">for</span> f <span class="kw">in</span> folders}</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>cooler_dict <span class="op">=</span> {op.basename(k): [op.basename(f) <span class="cf">for</span> f <span class="kw">in</span> v] <span class="cf">for</span> k,v <span class="kw">in</span> files_dict.items()}</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a><span class="co">#pp(cooler_dict)</span></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>df <span class="op">=</span> pd.DataFrame.from_dict(cooler_dict, orient<span class="op">=</span><span class="st">'index'</span>).T.fillna(<span class="st">'-'</span>)</span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>df[[<span class="st">'fibroblast'</span>, <span class="st">'spermatogonia'</span>, <span class="st">'pachytene_spermatocyte'</span>, <span class="st">'round_spermatid'</span>, <span class="st">'sperm'</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div id="tbl-cooler_dict" class="cell quarto-float quarto-figure quarto-figure-center anchored" data-execution_count="27">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-cooler_dict-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;5.3: Dictionary of coolers for each cell type
</figcaption>
<div aria-describedby="tbl-cooler_dict-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output cell-output-display" data-execution_count="27">
<div>


<table class="dataframe do-not-create-environment cell caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">fibroblast</th>
<th data-quarto-table-cell-role="th">spermatogonia</th>
<th data-quarto-table-cell-role="th">pachytene_spermatocyte</th>
<th data-quarto-table-cell-role="th">round_spermatid</th>
<th data-quarto-table-cell-role="th">sperm</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>SRR6502339.nodups.10000.cool</td>
<td>SRR6502357.nodups.10000.cool</td>
<td>SRR7345459.nodups.10000.cool</td>
<td>SRR7349192.nodups.10000.cool</td>
<td>SRR7349196.nodups.10000.cool</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>SRR7349190.nodups.10000.cool</td>
<td>SRR7345467.nodups.10000.cool</td>
<td>SRR6502344.nodups.10000.cool</td>
<td>SRR6502353.nodups.10000.cool</td>
<td>SRR6502362.nodups.10000.cool</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>SRR7349189.nodups.10000.cool</td>
<td>SRR6502356.nodups.10000.cool</td>
<td>SRR6502342.nodups.10000.cool</td>
<td>SRR6502352.nodups.10000.cool</td>
<td>SRR7349193.nodups.10000.cool</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3</td>
<td>SRR6502335.nodups.10000.cool</td>
<td>SRR7345464.nodups.10000.cool</td>
<td>SRR7345458.nodups.10000.cool</td>
<td>SRR6502351.nodups.10000.cool</td>
<td>SRR6502360.nodups.10000.cool</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">4</td>
<td>SRR6502338.nodups.10000.cool</td>
<td>SRR7345462.nodups.10000.cool</td>
<td>SRR6502345.nodups.10000.cool</td>
<td>SRR7345460.nodups.10000.cool</td>
<td>SRR7349195.nodups.10000.cool</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">5</td>
<td>SRR6502336.nodups.10000.cool</td>
<td>SRR7345465.nodups.10000.cool</td>
<td>SRR7349191.nodups.10000.cool</td>
<td>SRR7345461.nodups.10000.cool</td>
<td>-</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">6</td>
<td>SRR6502337.nodups.10000.cool</td>
<td>SRR7345463.nodups.10000.cool</td>
<td>-</td>
<td>-</td>
<td>-</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
</figure>
</div>
</div>
<p>We evidently see from the <a href="#tbl-cooler_dict" class="quarto-xref">Table&nbsp;<span>5.3</span></a> that…</p>
</section>
<section id="merge-coolers" class="level3" data-number="5.2.2">
<h3 data-number="5.2.2" class="anchored" data-anchor-id="merge-coolers"><span class="header-section-number">5.2.2</span> Merge coolers</h3>
<div id="cell-md-merge-coolers" class="cell" data-execution_count="3">
<div id="md-merge-coolers" class="cell-output cell-output-display cell-output-markdown" data-execution_count="3">
<p>The coolers are merged by summing each bin in the matrices, meaning we can only merge matrices with same dimensions. We iterate through the dictionary and merge the coolers with <code>cooler merge</code>. The <code>mergebuf</code> parameter should be adjusted if you don’t have 32G memory. Default: <code>mergebuf = 20000000</code>. Below, we also check if the output file already exists. If it does, we skip the merge.</p>
</div>
</div>
<div id="cell-21" class="cell" data-execution_count="5">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># NB adjust `mergebuf` if you don't have 32G of RAM</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooler</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> folder,cooler_list <span class="kw">in</span> cooler_dict.items():</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>    in_uris <span class="op">=</span> [op.join(base_dir, folder, <span class="bu">file</span>) <span class="cf">for</span> <span class="bu">file</span> <span class="kw">in</span> cooler_list]</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>    out_uri <span class="op">=</span> op.join(base_dir, folder, <span class="ss">f'</span><span class="sc">{</span>folder<span class="sc">}</span><span class="ss">.fullmerge.cool'</span>)</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> op.exists(out_uri):</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Skipping </span><span class="sc">{</span>out_uri<span class="sc">}</span><span class="ss">: exists..."</span>)</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>        <span class="cf">continue</span></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Creating </span><span class="sc">{</span>out_uri<span class="sc">}</span><span class="ss"> by </span><span class="ch">\n</span><span class="ss">Merging </span><span class="sc">{</span><span class="bu">len</span>(cooler_list)<span class="sc">}</span><span class="ss"> coolers into one:"</span>, end<span class="op">=</span><span class="st">" "</span>)</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"</span><span class="ch">\t</span><span class="st">"</span>,[<span class="bu">file</span>.split(<span class="st">'.'</span>)[<span class="dv">0</span>] <span class="cf">for</span> <span class="bu">file</span> <span class="kw">in</span> cooler_list])</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>    cooler.merge_coolers(output_uri<span class="op">=</span>out_uri,</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>                         input_uris<span class="op">=</span>in_uris,</span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>                         mergebuf<span class="op">=</span><span class="bu">int</span>(<span class="fl">5e7</span>),</span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>                         )</span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"... Done!"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Skipping ../steps/bwa/PE/cool/round_spermatid/round_spermatid.fullmerge.cool: exists...
Skipping ../steps/bwa/PE/cool/spermatogonia/spermatogonia.fullmerge.cool: exists...
Skipping ../steps/bwa/PE/cool/sperm/sperm.fullmerge.cool: exists...
Skipping ../steps/bwa/PE/cool/fibroblast/fibroblast.fullmerge.cool: exists...
Skipping ../steps/bwa/PE/cool/pachytene_spermatocyte/pachytene_spermatocyte.fullmerge.cool: exists...</code></pre>
</div>
</div>
</section>
<section id="zoomify-the-merged-cooler-files" class="level3" data-number="5.2.3">
<h3 data-number="5.2.3" class="anchored" data-anchor-id="zoomify-the-merged-cooler-files"><span class="header-section-number">5.2.3</span> Zoomify the merged cooler files</h3>
<div id="cell-md-zoomify-coolers" class="cell" data-execution_count="9">
<div id="md-zoomify-coolers" class="cell-output cell-output-display cell-output-markdown" data-execution_count="9">
<p>After merging the coolers, we will zoomify them to resolutions: 10kb, 50kb, 100kb, 500kb. It will recursively create the zoom levels for the cooler file from the ‘base’ resolution (in our case 10kb bins) to the ‘max’ resolution (in our case 500kb bins), and save all zoom levels in a multi-resolution cooler (<code>.mcool</code>) file. The resolutions are stored under the <code>resolutions</code> key in the cooler file (e.g.&nbsp;cell_type.mcool::/resolutions/10000).</p>
<p>Here, we also check if the output file already exists. If it does, we skip the zoomify. Again, can tailor the <code>chunksize</code> (pixels loaded <em>per process</em>) parameter according to memory availability. I found that the command stalled completely (without quitting) when the memory allocation was not sufficient, as I started out with 32 cores and 32G of RAM. I did not test what was more time-efficient; increasing number of processes or chunksize. After all, it didn’t take long to run anyways.</p>
<p>Finally, I list the resolutions in the mcool file.</p>
</div>
</div>
<div id="cell-24" class="cell" data-execution_count="7">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># NB 8 cores and 32G of RAM was used</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> glob</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooler</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os.path <span class="im">as</span> op</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>base_dir <span class="op">=</span> <span class="st">'../steps/bwa/PE/cool'</span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>merged_coolers <span class="op">=</span> glob.glob(op.join(base_dir, <span class="st">'*/*.fullmerge.cool'</span>))</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> clr <span class="kw">in</span> merged_coolers:</span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>    out_uri <span class="op">=</span> clr.replace(<span class="st">'.fullmerge.cool'</span>, <span class="st">'.mcool'</span>)</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> op.exists(out_uri):</span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Skipping </span><span class="sc">{</span>out_uri<span class="sc">}</span><span class="ss">: exists..."</span>)</span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>        <span class="cf">continue</span></span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Zoomifying cooler: </span><span class="ch">\n\t</span><span class="ss">   </span><span class="sc">{</span>clr<span class="sc">}</span><span class="ch">\n\t</span><span class="ss">-&gt; </span><span class="sc">{</span>out_uri<span class="sc">}</span><span class="ss">"</span>, end<span class="op">=</span><span class="st">""</span>)</span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>    cooler.zoomify_cooler(base_uris <span class="op">=</span> clr,</span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a>                          outfile <span class="op">=</span> out_uri,</span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a>                          resolutions <span class="op">=</span> [<span class="dv">10000</span>,<span class="dv">50000</span>,<span class="dv">100000</span>,<span class="dv">500000</span>],</span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a>                          chunksize <span class="op">=</span> <span class="dv">10000000</span>,</span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a>                          nproc <span class="op">=</span> <span class="dv">8</span>)</span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">" --&gt; done"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Skipping ../steps/bwa/PE/cool/round_spermatid/round_spermatid.mcool: exists...
Skipping ../steps/bwa/PE/cool/spermatogonia/spermatogonia.mcool: exists...
Skipping ../steps/bwa/PE/cool/sperm/sperm.mcool: exists...
Zoomifying cooler: 
       ../steps/bwa/PE/cool/fibroblast/fibroblast.fullmerge.cool
    -&gt; ../steps/bwa/PE/cool/fibroblast/fibroblast.mcool --&gt; done
Skipping ../steps/bwa/PE/cool/pachytene_spermatocyte/pachytene_spermatocyte.mcool: exists...</code></pre>
</div>
</div>
<div id="cell-25" class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> glob</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooler</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>mcools <span class="op">=</span> glob.glob(<span class="st">"../steps/bwa/PE/cool/*/*.mcool"</span>)</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> mcool <span class="kw">in</span> mcools:</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"</span><span class="sc">{</span>mcool<span class="sc">}</span><span class="ss">:"</span>)</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(cooler.fileops.list_coolers(mcool))</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>../steps/bwa/PE/cool/round_spermatid/round_spermatid.mcool:
['/resolutions/10000', '/resolutions/50000', '/resolutions/100000', '/resolutions/500000']

../steps/bwa/PE/cool/spermatogonia/spermatogonia.mcool:
['/resolutions/10000', '/resolutions/50000', '/resolutions/100000', '/resolutions/500000']

../steps/bwa/PE/cool/sperm/sperm.mcool:
['/resolutions/10000', '/resolutions/50000', '/resolutions/100000', '/resolutions/500000']

../steps/bwa/PE/cool/fibroblast/fibroblast.mcool:
['/resolutions/10000', '/resolutions/50000', '/resolutions/100000', '/resolutions/500000']

../steps/bwa/PE/cool/pachytene_spermatocyte/pachytene_spermatocyte.mcool:
['/resolutions/10000', '/resolutions/50000', '/resolutions/100000', '/resolutions/500000']
</code></pre>
</div>
</div>
</section>
<section id="balance-the-matrices" class="level3" data-number="5.2.4">
<h3 data-number="5.2.4" class="anchored" data-anchor-id="balance-the-matrices"><span class="header-section-number">5.2.4</span> Balance the matrices</h3>
<div id="cell-md-balance-coolers" class="cell" data-execution_count="40">
<div id="md-balance-coolers" class="cell-output cell-output-display cell-output-markdown" data-execution_count="40">
<p>Finally, we balance the matrices using the cooler CLI. We use the <code>cooler balance</code> command with the default options using 32 cores, which iteratively balances the matrix (Iterative Corecction). It is first described as a method for bias correction of Hi-C matrices in <span class="citation" data-cites="imakaev_iterative_2012">(<a href="../thesis/index.html#ref-imakaev_iterative_2012" role="doc-biblioref">Imakaev et al. 2012</a>)</span>, where it is paired with eigenvector decomposition (ICE). Here the eigenvector decomposition of the obtained maps is shown to provide insights into local chromatin states.</p>
<p>According to <code>cooler</code> documentation, we have to balance the matrices on each resolution, and thus it cannot be done prior to zoomifying. The argument is that the balancing weights are resolution-specific and will no longer retain its meaning when binned with other weights. Therefore, we use a nested for-loop that iterates through all the <code>.mcools</code> and all the resolutions in each <code>.mcool</code>. <code>cooler balance</code> will create a new column in the <code>bins</code> group of each cooler , <code>weight</code>, which can then be included or not in the downstream analysis. This means we will have access to both the balanced and the unbalanced matrix.</p>
<p>The default mode uses genome-wide data to calculate the weights for each bin. It would maybe be more suitable to calculate the weights for <em>cis</em> contacts only, and that is possible through the <code>--cis-only</code> flag, and that can be added to another column, so that we can compare the difference between the two methods easily. However, we will only use the default mode for now. The default options are:</p>
<ul>
<li><code>ignore-diags: 2</code> (ignore the diagonals (-1,0,1))</li>
<li><code>mad-max: 5</code> (median absolute deviation threshold)</li>
<li><code>min-nnz: 10</code> (minimum number of non-zero entries before exclusion)</li>
</ul>
<p>Conveniently, <code>cooler balance</code> automatically checks if there is already a <code>weight</code> column, and skips the balancing if it already exists. We can overwrite with <code>--force</code>.</p>
</div>
</div>
<div id="cell-28" class="cell" data-execution_count="27">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="op">%%</span>capture balance_out <span class="op">--</span>no<span class="op">-</span>stderr</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> sys</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> glob</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>mcools <span class="op">=</span> glob.glob(<span class="st">"../steps/bwa/PE/cool/*/*.mcool"</span>)</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>resolutions <span class="op">=</span> [<span class="dv">10000</span>, <span class="dv">50000</span>, <span class="dv">100000</span>, <span class="dv">500000</span>]</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> mcool <span class="kw">in</span> mcools:</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Balancing </span><span class="sc">{</span>mcool<span class="sc">}</span><span class="ss">:"</span>, <span class="bu">file</span><span class="op">=</span>sys.stderr)</span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> res <span class="kw">in</span> resolutions:</span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>        full_name <span class="op">=</span> <span class="ss">f"</span><span class="sc">{</span>mcool<span class="sc">}</span><span class="ss">::resolutions/</span><span class="sc">{</span>res<span class="sc">}</span><span class="ss">"</span></span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\t</span><span class="ss">resolution </span><span class="sc">{</span>res<span class="sc">}</span><span class="ss">..."</span>, end<span class="op">=</span><span class="st">" "</span>, <span class="bu">file</span><span class="op">=</span>sys.stderr)</span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a>        <span class="co"># First, just default values</span></span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a>        <span class="op">!</span>cooler balance <span class="op">-</span>p <span class="dv">32</span> {full_name}</span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a>        <span class="co"># With only cis-contacts, save column seprately in the cooler file</span></span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a>        <span class="co">#!cooler balance -p 32 -c 20000000 --cis-only -n cis_weights {full_name}</span></span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"--&gt; Done!"</span>, <span class="bu">file</span><span class="op">=</span>sys.stderr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Balancing ../steps/bwa/PE/cool/round_spermatid/round_spermatid.mcool:
    resolution 10000... --&gt; Done!
    resolution 50000... --&gt; Done!
    resolution 100000... --&gt; Done!
    resolution 500000... --&gt; Done!
Balancing ../steps/bwa/PE/cool/spermatogonia/spermatogonia.mcool:
    resolution 10000... --&gt; Done!
    resolution 50000... --&gt; Done!
    resolution 100000... --&gt; Done!
    resolution 500000... --&gt; Done!
Balancing ../steps/bwa/PE/cool/sperm/sperm.mcool:
    resolution 10000... --&gt; Done!
    resolution 50000... --&gt; Done!
    resolution 100000... --&gt; Done!
    resolution 500000... --&gt; Done!
Balancing ../steps/bwa/PE/cool/fibroblast/fibroblast.mcool:
    resolution 10000... --&gt; Done!
    resolution 50000... --&gt; Done!
    resolution 100000... --&gt; Done!
    resolution 500000... --&gt; Done!
Balancing ../steps/bwa/PE/cool/pachytene_spermatocyte/pachytene_spermatocyte.mcool:
    resolution 10000... --&gt; Done!
    resolution 50000... --&gt; Done!
    resolution 100000... --&gt; Done!
    resolution 500000... --&gt; Done!</code></pre>
</div>
</div>
<div id="cell-29" class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the balancing stdout if you want (but it is up to 200 lines per cooler)</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>balance_out()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>'weight' column already exists. Use --force option to overwrite.
'weight' column already exists. Use --force option to overwrite.
'weight' column already exists. Use --force option to overwrite.
'weight' column already exists. Use --force option to overwrite.
'weight' column already exists. Use --force option to overwrite.
'weight' column already exists. Use --force option to overwrite.
'weight' column already exists. Use --force option to overwrite.
'weight' column already exists. Use --force option to overwrite.
'weight' column already exists. Use --force option to overwrite.
'weight' column already exists. Use --force option to overwrite.
'weight' column already exists. Use --force option to overwrite.
'weight' column already exists. Use --force option to overwrite.
'weight' column already exists. Use --force option to overwrite.
'weight' column already exists. Use --force option to overwrite.
'weight' column already exists. Use --force option to overwrite.
'weight' column already exists. Use --force option to overwrite.
'weight' column already exists. Use --force option to overwrite.
'weight' column already exists. Use --force option to overwrite.
'weight' column already exists. Use --force option to overwrite.
'weight' column already exists. Use --force option to overwrite.</code></pre>
</div>
</div>
</section>
</section>
<section id="repmerge-pool-all-from-each-biosample-id" class="level2" data-number="5.3">
<h2 data-number="5.3" class="anchored" data-anchor-id="repmerge-pool-all-from-each-biosample-id"><span class="header-section-number">5.3</span> RepMerge (pool all from each BioSample ID)</h2>
<div id="cell-md-merge-replicates" class="cell" data-execution_count="21">
<div id="md-merge-replicates" class="cell-output cell-output-display cell-output-markdown" data-execution_count="21">
<p>Initially, I planned to re-create the replicates that they used in the paper, but I reason that it is not necessary and might even be better to pool all the samples in stead. They state alredy that their compartments are highly repdoducible between replicates, so I choose to trust that and not bother with the replicates.</p>
<p>Therefore, this section was only briefly initialized, and now it is commented out.</p>
</div>
</div>
<div id="cell-32" class="cell" data-execution_count="1">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co">#df.groupby(['source_name','BioSample'])['Reads'].sum()</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div id="cell-33" class="cell" data-execution_count="2">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># from pprintpp import pprint as pp</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a><span class="co"># grouped_df = df.groupby(['source_name','BioSample'])</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a><span class="co"># # Initialize an empty dictionary</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a><span class="co"># rep_dict = {}</span></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a><span class="co"># # Iterate over each group</span></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a><span class="co"># for (source_name, BioSample), group in grouped_df:</span></span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a><span class="co">#     # Extract the 'Run' column and convert it to a list</span></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a><span class="co">#     run_list = group['Run'].tolist()</span></span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a><span class="co">#     # Populate the nested dictionary</span></span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a><span class="co">#     if source_name not in rep_dict:</span></span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a><span class="co">#         rep_dict[source_name] = {}</span></span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a><span class="co">#     rep_dict[source_name][BioSample] = run_list</span></span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a><span class="co"># pp(rep_dict)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div id="cell-34" class="cell" data-execution_count="3">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># for source_name, BioSample_dict in rep_dict.items():</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="co">#     print(f"source_name: {source_name}")</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="co">#     print(f"Changing working dir: {source_name}/")</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a><span class="co">#     for BioSample, run_list in BioSample_dict.items():</span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="co">#         print(f"Merging samples for BioSample: {BioSample}: {run_list}")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="plotting" class="level1 page-columns page-full" data-number="6">
<h1 data-number="6"><span class="header-section-number">6</span> Plotting</h1>
<div id="cell-md-plotting-main" class="cell">
<div id="md-plotting-main" class="cell-output cell-output-display cell-output-markdown" data-execution_count="1">
<p>The following section contains the visualization of the matrices and includes the calculation of the E1 compartments and their visualization. I discuss differents methods for construction and try to get both methodologically and result-wise close to <span class="citation" data-cites="wang_reprogramming_2019">(<a href="../thesis/index.html#ref-wang_reprogramming_2019" role="doc-biblioref">Wang et al. 2019</a>)</span>.</p>
<p>First, we will work on matrices in 500kb resolution, as documentation [maybe it was HiCExplorer??] states it is sufficient for chromosome-wide analysis and plotting. We will follow the <code>cooltools</code>-recommended pipeline (except that they use a 100kb cooler) for <a href="https://cooltools.readthedocs.io/en/latest/notebooks/viz.html">visualization</a> and <a href="https://cooltools.readthedocs.io/en/latest/notebooks/compartments_and_saddles.html">compartment calling</a> with one example cooler (the merged fibroblast cooler at 500kb resolution).</p>
<p>Then, the most relevant parts will be generalized to run in a loop over all the coolers. First at 500kb resolution, then at 100kb resolution.</p>
<p>To accomodate the approach (barely) described in the paper, we will discuss a smoothing step to the observed/expected matrices before compartment calling, and we will apply a smoothing step to the E1 compartments and compare to the raw compartments.</p>
</div>
</div>
<section id="kb-resolution" class="level2 page-columns page-full" data-number="6.1">
<h2 data-number="6.1" class="anchored" data-anchor-id="kb-resolution"><span class="header-section-number">6.1</span> 500kb resolution</h2>
<section id="example-with-a-single-sample" class="level3" data-number="6.1.1">
<h3 data-number="6.1.1" class="anchored" data-anchor-id="example-with-a-single-sample"><span class="header-section-number">6.1.1</span> Example with a single sample</h3>
<div id="cell-md-plotting-500kb-single" class="cell" data-execution_count="2">
<div id="md-plotting-500kb-single" class="cell-output cell-output-display cell-output-markdown" data-execution_count="2">
<p>First, I will explore the visualization pipeline for a single cooler at 500kb resolution. I will modify the plot to be ‘stairs’ in stead of just a regular line plot, as it is both a more accurate representation of the data and it is more aesthetically pleasing with less spiky lines and holes.</p>
<p>In practice, the length of the dataframe is doubled, as it now contains an E1 value for both the start and end position for each bin in stead of only for the start. However, I first make the regular line plot to show the difference.</p>
</div>
</div>
<section id="imports" class="level4" data-number="6.1.1.1">
<h4 data-number="6.1.1.1" class="anchored" data-anchor-id="imports"><span class="header-section-number">6.1.1.1</span> Imports</h4>
<div id="cell-41" class="cell" data-execution_count="3">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># import standard python libraries</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a><span class="op">%</span>matplotlib inline</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os, subprocess</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div id="cell-42" class="cell" data-execution_count="4">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Import python package for working with cooler files and tools for analysis</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooler</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooltools.lib.plotting</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooltools</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-cooler" class="level4" data-number="6.1.1.2">
<h4 data-number="6.1.1.2" class="anchored" data-anchor-id="load-cooler"><span class="header-section-number">6.1.1.2</span> Load cooler</h4>
<div id="cell-44" class="cell" data-execution_count="5">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>mclr <span class="op">=</span> <span class="st">"../steps/bwa/PE/cool/fibroblast/fibroblast.mcool"</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>clr <span class="op">=</span> cooler.Cooler(<span class="ss">f"</span><span class="sc">{</span>mclr<span class="sc">}</span><span class="ss">::resolutions/500000"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="calculate-gc-covariance-from-the-reference-genome" class="level4" data-number="6.1.1.3">
<h4 data-number="6.1.1.3" class="anchored" data-anchor-id="calculate-gc-covariance-from-the-reference-genome"><span class="header-section-number">6.1.1.3</span> Calculate gc covariance (from the reference genome)</h4>
<div id="cell-md-plotting-500kb-single-gc" class="cell" data-execution_count="6">
<div id="md-plotting-500kb-single-gc" class="cell-output cell-output-display cell-output-markdown" data-execution_count="6">
<p>I will use <code>bioframe</code> to load the reference and calculate the GC content of each bin in the cooler file. The convention in Hi-C is to use GC content as a phasing track to orient eigenvector track to positively correlated with the GC content. In this subsection, I calculate the GC content for all chromosomes and filter afterwards to only include the ‘X’ chromosome. I will do that smarter in the next subsection.</p>
</div>
</div>
<div id="cell-47" class="cell" data-execution_count="32">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> bioframe</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>bins <span class="op">=</span> clr.bins()[:]</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>rheMac10 <span class="op">=</span> bioframe.load_fasta(<span class="st">'../data/links/ucsc_ref/rheMac10.fa'</span>)</span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>gc_cov_csv <span class="op">=</span> <span class="st">'../steps/rheMac10_gc_cov_500kb.tsv'</span></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> <span class="kw">not</span> os.path.exists(gc_cov_csv):</span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">'Calculate the fraction of GC basepairs for each bin'</span>)</span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>    gc_cov <span class="op">=</span> bioframe.frac_gc(bins[[<span class="st">'chrom'</span>, <span class="st">'start'</span>, <span class="st">'end'</span>]], rheMac10)</span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>    gc_cov.to_csv(gc_cov_csv, index<span class="op">=</span><span class="va">False</span>, sep<span class="op">=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>)</span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>    display(gc_cov)</span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>: </span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"Already exists, just read from file:"</span>)</span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a>    gc_cov <span class="op">=</span> pd.read_csv(gc_cov_csv, sep<span class="op">=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>)</span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a>    display(gc_cov)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Already exists, just read from file:</code></pre>
</div>
<div class="cell-output cell-output-display">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">chrom</th>
<th data-quarto-table-cell-role="th">start</th>
<th data-quarto-table-cell-role="th">end</th>
<th data-quarto-table-cell-role="th">GC</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>chr1</td>
<td>0</td>
<td>500000</td>
<td>0.424508</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>chr1</td>
<td>500000</td>
<td>1000000</td>
<td>0.378836</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>chr1</td>
<td>1000000</td>
<td>1500000</td>
<td>0.388272</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3</td>
<td>chr1</td>
<td>1500000</td>
<td>2000000</td>
<td>0.445226</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">4</td>
<td>chr1</td>
<td>2000000</td>
<td>2500000</td>
<td>0.439485</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">5715</td>
<td>chrY</td>
<td>9500000</td>
<td>10000000</td>
<td>0.399518</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">5716</td>
<td>chrY</td>
<td>10000000</td>
<td>10500000</td>
<td>0.396028</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">5717</td>
<td>chrY</td>
<td>10500000</td>
<td>11000000</td>
<td>0.395280</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">5718</td>
<td>chrY</td>
<td>11000000</td>
<td>11500000</td>
<td>0.382006</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">5719</td>
<td>chrY</td>
<td>11500000</td>
<td>11753682</td>
<td>NaN</td>
</tr>
</tbody>
</table>

<p>5720 rows × 4 columns</p>
</div>
</div>
</div>
</section>
<section id="calculate-the-e1-compartments" class="level4" data-number="6.1.1.4">
<h4 data-number="6.1.1.4" class="anchored" data-anchor-id="calculate-the-e1-compartments"><span class="header-section-number">6.1.1.4</span> Calculate the E1 compartments</h4>
<div id="cell-md-plotting-500kb-single-e1" class="cell" data-execution_count="8">
<div id="md-plotting-500kb-single-e1" class="cell-output cell-output-display cell-output-markdown" data-execution_count="8">
<p>Here, I will calculate the E1 compartments for the ‘X’ chromosome in the fibroblast cooler at 500kb resolution. I will use the <code>cooltools</code> package to do eigendecomposition and calculate the E1 compartments. I use the GC content (<code>gc_cov</code> obtained above) as a phasing track.</p>
<p>I will only calculate the within-chromosomes compartmentalization (<em>cis</em> contacts). I use <code>cooltools.eigs_cis</code> that decorrelate the contact-frequency by distance before performing the eigendecomposition.</p>
<p>I this example, I will use the simplest ‘view’ (chromosome-wide) of the chromosome to calculate the E1 values, but later I will explore how partitioning the chromosome affects the E1 values. As the GC content was calculated for all chromosomes, the eigenvectors are also calculated for the whole matrix. This will <em>also</em> be optimized in the next subsection, as we only ever look at the ‘X’ chromosome.</p>
</div>
</div>
<div id="cell-50" class="cell" data-execution_count="9">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Make the full view frame</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>view_df <span class="op">=</span> pd.DataFrame(</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>    {</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>    <span class="st">'chrom'</span>: clr.chromnames,</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>    <span class="st">'start'</span>: <span class="dv">0</span>,</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>    <span class="st">'end'</span>: clr.chromsizes.values,</span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>    <span class="st">'name'</span>: clr.chromnames</span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>                      )</span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a><span class="co">#display(view_df)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div id="cell-51" class="cell" data-execution_count="10">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># obtain first 3 eigenvectors</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>cis_eigs <span class="op">=</span> cooltools.eigs_cis(</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>                        clr,</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>                        gc_cov,</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>                        view_df<span class="op">=</span>view_df,</span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>                        n_eigs<span class="op">=</span><span class="dv">3</span>,</span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>                        )</span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a><span class="co"># cis_eigs[0] returns eigenvalues, here we focus on eigenvectors</span></span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>eigenvector_track <span class="op">=</span> cis_eigs[<span class="dv">1</span>][[<span class="st">'chrom'</span>,<span class="st">'start'</span>,<span class="st">'end'</span>,<span class="st">'E1'</span>]]</span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a><span class="bu">len</span>(eigenvector_track)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="10">
<pre><code>5720</code></pre>
</div>
</div>
<div id="cell-52" class="cell" data-execution_count="11">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># full track</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>eigenvector_track</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="11">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">chrom</th>
<th data-quarto-table-cell-role="th">start</th>
<th data-quarto-table-cell-role="th">end</th>
<th data-quarto-table-cell-role="th">E1</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>chr1</td>
<td>0</td>
<td>500000</td>
<td>-0.756052</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>chr1</td>
<td>500000</td>
<td>1000000</td>
<td>-1.001063</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>chr1</td>
<td>1000000</td>
<td>1500000</td>
<td>-0.928042</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3</td>
<td>chr1</td>
<td>1500000</td>
<td>2000000</td>
<td>-0.507579</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">4</td>
<td>chr1</td>
<td>2000000</td>
<td>2500000</td>
<td>0.123708</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">5715</td>
<td>chrY</td>
<td>9500000</td>
<td>10000000</td>
<td>-0.499798</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">5716</td>
<td>chrY</td>
<td>10000000</td>
<td>10500000</td>
<td>-0.588820</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">5717</td>
<td>chrY</td>
<td>10500000</td>
<td>11000000</td>
<td>-0.988677</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">5718</td>
<td>chrY</td>
<td>11000000</td>
<td>11500000</td>
<td>-0.566585</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">5719</td>
<td>chrY</td>
<td>11500000</td>
<td>11753682</td>
<td>NaN</td>
</tr>
</tbody>
</table>

<p>5720 rows × 4 columns</p>
</div>
</div>
</div>
<div id="cell-md-plotting-500kb-single-e1-X" class="cell" data-execution_count="12">
<div id="md-plotting-500kb-single-e1-x" class="cell-output cell-output-display cell-output-markdown" data-execution_count="12">
<p>As I have made myself a detour and calculated the eigenvectors for all chromosomes, I will now filter the eigenvector track to only include the ‘X’ chromosome. Then, I will (in the name of explicitness) construct an array of all the indices where the E1 values change sign. This is the simplest way to save the coordinates of the compartment boundaries. It can later on be saved to .csv or whatever for comparison between different samples. Also, it will be simplified in the next subsection.</p>
</div>
</div>
<div id="cell-54" class="cell" data-execution_count="13">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co"># subset the chrX   </span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>eigenvector_track_chrX <span class="op">=</span> eigenvector_track.loc[eigenvector_track[<span class="st">'chrom'</span>] <span class="op">==</span> <span class="st">'chrX'</span>]</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>nbins <span class="op">=</span> <span class="bu">len</span>(eigenvector_track_chrX)</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>eigenvector_track_chrX</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>e1X_values <span class="op">=</span> eigenvector_track_chrX[<span class="st">'E1'</span>].values</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div id="cell-55" class="cell" data-execution_count="14">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Where does the E1 change sign</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>np.where(np.diff( (cis_eigs[<span class="dv">1</span>][cis_eigs[<span class="dv">1</span>][<span class="st">'chrom'</span>]<span class="op">==</span><span class="st">'chrX'</span>][<span class="st">'E1'</span>]<span class="op">&gt;</span><span class="dv">0</span>).astype(<span class="bu">int</span>)))[<span class="dv">0</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="14">
<pre><code>array([  1,   6,  16,  26,  29,  40,  42,  49,  54,  55,  59,  60,  61,
        62,  74,  83,  86, 101, 102, 103, 104, 108, 123, 124, 129, 138,
       141, 142, 147, 149, 185, 188, 194, 196, 197, 201, 206, 209, 211,
       215, 222, 224, 228, 233, 238, 240, 250, 253, 256, 265, 273, 274,
       288, 289, 290, 293, 296, 302])</code></pre>
</div>
</div>
</section>
<section id="plot-e1-and-matrix" class="level4" data-number="6.1.1.5">
<h4 data-number="6.1.1.5" class="anchored" data-anchor-id="plot-e1-and-matrix"><span class="header-section-number">6.1.1.5</span> Plot E1 and matrix</h4>
<div id="cell-md-plotting-500kb-single-plot" class="cell" data-execution_count="15">
<div id="md-plotting-500kb-single-plot" class="cell-output cell-output-display cell-output-markdown" data-execution_count="15">
<p>Finally, I will plot the matrix, the GC content, and the E1 compartments for the ‘X’ chromosome in the fibroblast cooler at 500kb resolution. I will use the <code>cooltools</code> package to plot the matrix and the compartments. I will use the <code>matplotlib</code> package to plot the GC content.</p>
<p>I will plot the matrix as a heatmap, the GC content as a line plot, and the E1 compartments as a line plot. I will also plot the compartment boundaries as vertical lines. The compartments are colored according to the sign of the E1 values.</p>
<p>Following the <code>cooltools</code> recommendation, the colorbar itself will be log-transformed, as otherwise the balanced interaction matrix (values from 0 to 1) will be hard to interpret. The E1 values are (as is the colorbar) added to the axis with an <code>AxesDivider.append_axes</code> object to make them match the axes of the matrix.</p>
<p>I’m testing out the <code>%%capture</code> magic to capture the plot to a variable, then displaying it in the next cell. Initially, the plot took a while to generate, and I wanted to avoid re-generating when adjusting graphics on cell level with the YAML options. Now it is not necessary, but I keep it for the sake of the example.</p>
</div>
</div>
<div id="cell-58" class="cell" data-execution_count="16">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="op">%%</span>capture chrX_matrix_e1_500kb</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> matplotlib.colors <span class="im">import</span> LogNorm</span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> mpl_toolkits.axes_grid1 <span class="im">import</span> make_axes_locatable</span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>f, ax <span class="op">=</span> plt.subplots(</span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a>    figsize<span class="op">=</span>(<span class="dv">15</span>, <span class="dv">10</span>),</span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a>norm <span class="op">=</span> LogNorm(vmax<span class="op">=</span><span class="fl">0.1</span>)</span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a>im <span class="op">=</span> ax.matshow(</span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a>    clr.matrix().fetch(<span class="st">'chrX'</span>),</span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a>    norm<span class="op">=</span>norm,</span>
<span id="cb28-15"><a href="#cb28-15" aria-hidden="true" tabindex="-1"></a>    cmap<span class="op">=</span><span class="st">'fall'</span>,</span>
<span id="cb28-16"><a href="#cb28-16" aria-hidden="true" tabindex="-1"></a>)<span class="op">;</span></span>
<span id="cb28-17"><a href="#cb28-17" aria-hidden="true" tabindex="-1"></a>plt.axis([<span class="dv">0</span>,nbins,nbins,<span class="dv">0</span>])</span>
<span id="cb28-18"><a href="#cb28-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-19"><a href="#cb28-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-20"><a href="#cb28-20" aria-hidden="true" tabindex="-1"></a>divider <span class="op">=</span> make_axes_locatable(ax)</span>
<span id="cb28-21"><a href="#cb28-21" aria-hidden="true" tabindex="-1"></a>cax <span class="op">=</span> divider.append_axes(<span class="st">"right"</span>, size<span class="op">=</span><span class="st">"5%"</span>, pad<span class="op">=</span><span class="fl">0.05</span>)</span>
<span id="cb28-22"><a href="#cb28-22" aria-hidden="true" tabindex="-1"></a>plt.colorbar(im, cax<span class="op">=</span>cax, label<span class="op">=</span><span class="st">'corrected frequencies'</span>)<span class="op">;</span></span>
<span id="cb28-23"><a href="#cb28-23" aria-hidden="true" tabindex="-1"></a>ax.set_ylabel(<span class="st">'chrX:500kb bins, #bin'</span>)</span>
<span id="cb28-24"><a href="#cb28-24" aria-hidden="true" tabindex="-1"></a>ax.xaxis.set_visible(<span class="va">False</span>)</span>
<span id="cb28-25"><a href="#cb28-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-26"><a href="#cb28-26" aria-hidden="true" tabindex="-1"></a>ax1 <span class="op">=</span> divider.append_axes(<span class="st">"top"</span>, size<span class="op">=</span><span class="st">"20%"</span>, pad<span class="op">=</span><span class="fl">0.05</span>, sharex<span class="op">=</span>ax)</span>
<span id="cb28-27"><a href="#cb28-27" aria-hidden="true" tabindex="-1"></a><span class="co">#weights = clr.bins()[:]['weight'].values</span></span>
<span id="cb28-28"><a href="#cb28-28" aria-hidden="true" tabindex="-1"></a><span class="co">#ax1.plot([0,nbins],[0,0],'k',lw=0.25)</span></span>
<span id="cb28-29"><a href="#cb28-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-30"><a href="#cb28-30" aria-hidden="true" tabindex="-1"></a><span class="co">#ax1.plot(e1X_values, label='E1')</span></span>
<span id="cb28-31"><a href="#cb28-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-32"><a href="#cb28-32" aria-hidden="true" tabindex="-1"></a><span class="co"># Fill between the line and 0</span></span>
<span id="cb28-33"><a href="#cb28-33" aria-hidden="true" tabindex="-1"></a>ax1.fill_between(<span class="bu">range</span>(<span class="bu">len</span>(e1X_values)), e1X_values, <span class="dv">0</span>, where<span class="op">=</span>(e1X_values <span class="op">&gt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:red'</span>)</span>
<span id="cb28-34"><a href="#cb28-34" aria-hidden="true" tabindex="-1"></a>ax1.fill_between(<span class="bu">range</span>(<span class="bu">len</span>(e1X_values)), e1X_values, <span class="dv">0</span>, where<span class="op">=</span>(e1X_values <span class="op">&lt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:blue'</span>)</span>
<span id="cb28-35"><a href="#cb28-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-36"><a href="#cb28-36" aria-hidden="true" tabindex="-1"></a>ax1.set_ylabel(<span class="st">'E1'</span>)</span>
<span id="cb28-37"><a href="#cb28-37" aria-hidden="true" tabindex="-1"></a>ax1.set_xticks([])<span class="op">;</span></span>
<span id="cb28-38"><a href="#cb28-38" aria-hidden="true" tabindex="-1"></a>ax1.get_subplotspec()</span>
<span id="cb28-39"><a href="#cb28-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-40"><a href="#cb28-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-41"><a href="#cb28-41" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i <span class="kw">in</span> np.where(np.diff( (cis_eigs[<span class="dv">1</span>][cis_eigs[<span class="dv">1</span>][<span class="st">'chrom'</span>]<span class="op">==</span><span class="st">'chrX'</span>][<span class="st">'E1'</span>]<span class="op">&gt;</span><span class="dv">0</span>).astype(<span class="bu">int</span>)))[<span class="dv">0</span>]:</span>
<span id="cb28-42"><a href="#cb28-42" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Horisontal lines where E1 intersects 0</span></span>
<span id="cb28-43"><a href="#cb28-43" aria-hidden="true" tabindex="-1"></a>    ax.plot([<span class="dv">0</span>,nbins],[i,i],<span class="st">'k'</span>,lw<span class="op">=</span><span class="fl">0.4</span>)</span>
<span id="cb28-44"><a href="#cb28-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-45"><a href="#cb28-45" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Vertical lines where E1 intersects 0</span></span>
<span id="cb28-46"><a href="#cb28-46" aria-hidden="true" tabindex="-1"></a>    ax.plot([i,i],[<span class="dv">0</span>,nbins],<span class="st">'k'</span>,lw<span class="op">=</span><span class="fl">0.4</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div id="cell-fig-matrix_e1_500kb" class="cell" data-execution_count="43">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>chrX_matrix_e1_500kb.outputs[<span class="dv">0</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="43">
<div id="fig-matrix_e1_500kb" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-matrix_e1_500kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="03_compartments_files/figure-html/fig-matrix_e1_500kb-output-1.png" class="img-fluid figure-img">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-matrix_e1_500kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6.1: chrX interaction matrix with E1 eigenvector values in 500kb resolution. The sign change of the E1 is overlayed as thin black lines, making it more easily interpretable. We should qualitatively determine how well the E1 values capture the <em>plaid</em> pattern in the matrix.
</figcaption>
</figure>
</div>
</div>
</div>
<p>Markdown: <a href="../slides/index.html#fig-matrix_e1_500kb" class="quarto-xref">Figure&nbsp;<span>9.2</span></a></p>
<div id="cell-md-plotting-500kb-single-plot-discussion" class="cell" data-execution_count="45">
<div id="md-plotting-500kb-single-plot-discussion" class="cell-output cell-output-display cell-output-markdown" data-execution_count="45">
<p>We observe that the E1 values only somewhat captures the plaid pattern in the matrix <a href="../slides/index.html#fig-matrix_e1_500kb" class="quarto-xref">Figure&nbsp;<span>9.2</span></a>, but it is not related to the size of the compartments, as both small and large compartments can be observed. The B-compartment that starts from around bin 150 (75.000.000 bp) seems to have squares that are not captured by the E1 values. Maybe it could be captures by TAD calling, but that is not the scope of this analysis.</p>
</div>
</div>
<div id="cell-fig-e1_500kb_plot" class="cell" data-execution_count="44">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>f, ax1 <span class="op">=</span> plt.subplots(</span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>    figsize<span class="op">=</span>(<span class="dv">10</span>, <span class="dv">2</span>),</span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Fill between the line and 0</span></span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a>ax1.fill_between(<span class="bu">range</span>(<span class="bu">len</span>(e1X_values)), e1X_values, <span class="dv">0</span>, where<span class="op">=</span>(e1X_values <span class="op">&gt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:red'</span>)</span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a>ax1.fill_between(<span class="bu">range</span>(<span class="bu">len</span>(e1X_values)), e1X_values, <span class="dv">0</span>, where<span class="op">=</span>(e1X_values <span class="op">&lt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:blue'</span>)</span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-11"><a href="#cb30-11" aria-hidden="true" tabindex="-1"></a><span class="co">#ax1.set_ylabel('E1')</span></span>
<span id="cb30-12"><a href="#cb30-12" aria-hidden="true" tabindex="-1"></a>ax1.set_xticks([])</span>
<span id="cb30-13"><a href="#cb30-13" aria-hidden="true" tabindex="-1"></a>ax1.set_yticks([])</span>
<span id="cb30-14"><a href="#cb30-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-15"><a href="#cb30-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove borders</span></span>
<span id="cb30-16"><a href="#cb30-16" aria-hidden="true" tabindex="-1"></a>ax1.spines[<span class="st">'top'</span>].set_visible(<span class="va">False</span>)</span>
<span id="cb30-17"><a href="#cb30-17" aria-hidden="true" tabindex="-1"></a>ax1.spines[<span class="st">'right'</span>].set_visible(<span class="va">False</span>)</span>
<span id="cb30-18"><a href="#cb30-18" aria-hidden="true" tabindex="-1"></a>ax1.spines[<span class="st">'left'</span>].set_visible(<span class="va">False</span>)</span>
<span id="cb30-19"><a href="#cb30-19" aria-hidden="true" tabindex="-1"></a>ax1.spines[<span class="st">'bottom'</span>].set_visible(<span class="va">False</span>)</span>
<span id="cb30-20"><a href="#cb30-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-21"><a href="#cb30-21" aria-hidden="true" tabindex="-1"></a>plt.tight_layout()</span>
<span id="cb30-22"><a href="#cb30-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-23"><a href="#cb30-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the plot as a SVG file</span></span>
<span id="cb30-24"><a href="#cb30-24" aria-hidden="true" tabindex="-1"></a><span class="co">#plt.savefig('e1_plot.svg', bbox_inches='tight')</span></span>
<span id="cb30-25"><a href="#cb30-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-26"><a href="#cb30-26" aria-hidden="true" tabindex="-1"></a>plt.show()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div id="fig-e1_500kb_plot" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-e1_500kb_plot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="03_compartments_files/figure-html/fig-e1_500kb_plot-output-1.png" class="img-fluid figure-img">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-e1_500kb_plot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6.2: 500kb binned E1 eigenvector values for chrX. Freshly calculated from the cooler file.
</figcaption>
</figure>
</div>
</div>
</div>
<p>Markdown the next plot: <a href="../reports/thoughts.html#fig-e1_500kb_plot" class="quarto-xref">Figure&nbsp;<span>6.3</span></a> that is</p>
</section>
<section id="stairs-plot-of-the-e1-compartments" class="level4" data-number="6.1.1.6">
<h4 data-number="6.1.1.6" class="anchored" data-anchor-id="stairs-plot-of-the-e1-compartments"><span class="header-section-number">6.1.1.6</span> Stairs plot of the E1 compartments</h4>
<p>(Less spiky, more smooth)</p>
<div id="cell-chrX-e1-500kb-stairs" class="cell" data-fig-height="2">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>f, (ax1, ax2) <span class="op">=</span> plt.subplots(<span class="dv">2</span>,<span class="dv">1</span>,</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>    figsize<span class="op">=</span>(<span class="dv">10</span>, <span class="dv">4</span>)</span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a>chrom_start <span class="op">=</span> eigenvector_track_chrX[<span class="st">'start'</span>].values</span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a>window_size <span class="op">=</span> chrom_start[<span class="dv">1</span>] <span class="op">-</span> chrom_start[<span class="dv">0</span>]</span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-10"><a href="#cb31-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Fill between the line and 0</span></span>
<span id="cb31-11"><a href="#cb31-11" aria-hidden="true" tabindex="-1"></a>ax1.fill_between(<span class="bu">range</span>(<span class="bu">len</span>(e1X_values)), e1X_values, <span class="dv">0</span>, where<span class="op">=</span>(e1X_values <span class="op">&gt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:red'</span>)</span>
<span id="cb31-12"><a href="#cb31-12" aria-hidden="true" tabindex="-1"></a>ax1.fill_between(<span class="bu">range</span>(<span class="bu">len</span>(e1X_values)), e1X_values, <span class="dv">0</span>, where<span class="op">=</span>(e1X_values <span class="op">&lt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:blue'</span>)</span>
<span id="cb31-13"><a href="#cb31-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-14"><a href="#cb31-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Create stairs</span></span>
<span id="cb31-15"><a href="#cb31-15" aria-hidden="true" tabindex="-1"></a>x <span class="op">=</span> np.zeros(<span class="dv">2</span><span class="op">*</span>chrom_start.size)</span>
<span id="cb31-16"><a href="#cb31-16" aria-hidden="true" tabindex="-1"></a>y <span class="op">=</span> np.zeros(<span class="dv">2</span><span class="op">*</span>chrom_start.size)</span>
<span id="cb31-17"><a href="#cb31-17" aria-hidden="true" tabindex="-1"></a>x[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> chrom_start</span>
<span id="cb31-18"><a href="#cb31-18" aria-hidden="true" tabindex="-1"></a>x[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> chrom_start <span class="op">+</span> window_size</span>
<span id="cb31-19"><a href="#cb31-19" aria-hidden="true" tabindex="-1"></a>y[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> e1X_values</span>
<span id="cb31-20"><a href="#cb31-20" aria-hidden="true" tabindex="-1"></a>y[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> e1X_values</span>
<span id="cb31-21"><a href="#cb31-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-22"><a href="#cb31-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Layout</span></span>
<span id="cb31-23"><a href="#cb31-23" aria-hidden="true" tabindex="-1"></a>ax1.set_ylabel(<span class="st">'Spiky E1'</span>)</span>
<span id="cb31-24"><a href="#cb31-24" aria-hidden="true" tabindex="-1"></a>ax2.set_ylabel(<span class="st">'Stairs E1'</span>)</span>
<span id="cb31-25"><a href="#cb31-25" aria-hidden="true" tabindex="-1"></a>ax1.set_xticks([])</span>
<span id="cb31-26"><a href="#cb31-26" aria-hidden="true" tabindex="-1"></a>ax1.set_yticks([])</span>
<span id="cb31-27"><a href="#cb31-27" aria-hidden="true" tabindex="-1"></a>ax2.set_xticks([])</span>
<span id="cb31-28"><a href="#cb31-28" aria-hidden="true" tabindex="-1"></a>ax2.set_yticks([])</span>
<span id="cb31-29"><a href="#cb31-29" aria-hidden="true" tabindex="-1"></a>ax2.set_ylim(<span class="op">-</span><span class="dv">1</span>, <span class="dv">1</span>)</span>
<span id="cb31-30"><a href="#cb31-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-31"><a href="#cb31-31" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove borders</span></span>
<span id="cb31-32"><a href="#cb31-32" aria-hidden="true" tabindex="-1"></a>ax1.spines[:].set_visible(<span class="va">False</span>)</span>
<span id="cb31-33"><a href="#cb31-33" aria-hidden="true" tabindex="-1"></a>ax2.spines[:].set_visible(<span class="va">False</span>)</span>
<span id="cb31-34"><a href="#cb31-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-35"><a href="#cb31-35" aria-hidden="true" tabindex="-1"></a>ax2.fill_between(x, y, <span class="dv">0</span>, where<span class="op">=</span>(y <span class="op">&gt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:red'</span>)</span>
<span id="cb31-36"><a href="#cb31-36" aria-hidden="true" tabindex="-1"></a>ax2.fill_between(x, y, <span class="dv">0</span>, where<span class="op">=</span>(y <span class="op">&lt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:blue'</span>)</span>
<span id="cb31-37"><a href="#cb31-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-38"><a href="#cb31-38" aria-hidden="true" tabindex="-1"></a>plt.tight_layout()</span>
<span id="cb31-39"><a href="#cb31-39" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the plot as a high-resolution PNG file</span></span>
<span id="cb31-40"><a href="#cb31-40" aria-hidden="true" tabindex="-1"></a><span class="co">#plt.savefig('../steps/e1_plot.png', dpi=320, bbox_inches='tight')</span></span>
<span id="cb31-41"><a href="#cb31-41" aria-hidden="true" tabindex="-1"></a></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div id="chrx-e1-500kb-stairs" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="03_compartments_files/figure-html/chrx-e1-500kb-stairs-output-1.png" class="img-fluid figure-img"></p>
<figcaption>500kb binned E1 eigenvector values for chrX.</figcaption>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="all-5-full-merges" class="level3 page-columns page-full" data-number="6.1.2">
<h3 data-number="6.1.2" class="anchored" data-anchor-id="all-5-full-merges"><span class="header-section-number">6.1.2</span> All 5 full merges</h3>
<section id="load-coolers" class="level4" data-number="6.1.2.1">
<h4 data-number="6.1.2.1" class="anchored" data-anchor-id="load-coolers"><span class="header-section-number">6.1.2.1</span> Load coolers</h4>
<div id="cell-69" class="cell" data-execution_count="41">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> glob</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os.path <span class="im">as</span> op</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooler</span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>mcools <span class="op">=</span> glob.glob(<span class="st">"../steps/bwa/PE/cool/*/*.mcool"</span>)</span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>res <span class="op">=</span> <span class="st">"::resolutions/500000"</span></span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a>clrs <span class="op">=</span> {op.basename(op.dirname(mcool)): cooler.Cooler(mcool<span class="op">+</span>res) <span class="cf">for</span> mcool <span class="kw">in</span> mcools}</span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-10"><a href="#cb32-10" aria-hidden="true" tabindex="-1"></a>chron_order <span class="op">=</span> [<span class="st">'fibroblast'</span>, <span class="st">'spermatogonia'</span>, <span class="st">'pachytene_spermatocyte'</span>, <span class="st">'round_spermatid'</span>, <span class="st">'sperm'</span>]</span>
<span id="cb32-11"><a href="#cb32-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-12"><a href="#cb32-12" aria-hidden="true" tabindex="-1"></a>clrs <span class="op">=</span> {key: clrs[key] <span class="cf">for</span> key <span class="kw">in</span> chron_order}</span>
<span id="cb32-13"><a href="#cb32-13" aria-hidden="true" tabindex="-1"></a>clrs</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="41">
<pre><code>{'fibroblast': &lt;Cooler "fibroblast.mcool::/resolutions/500000"&gt;,
 'spermatogonia': &lt;Cooler "spermatogonia.mcool::/resolutions/500000"&gt;,
 'pachytene_spermatocyte': &lt;Cooler "pachytene_spermatocyte.mcool::/resolutions/500000"&gt;,
 'round_spermatid': &lt;Cooler "round_spermatid.mcool::/resolutions/500000"&gt;,
 'sperm': &lt;Cooler "sperm.mcool::/resolutions/500000"&gt;}</code></pre>
</div>
</div>
</section>
<section id="calculate-gc-covariance-from-the-reference-genome-1" class="level4" data-number="6.1.2.2">
<h4 data-number="6.1.2.2" class="anchored" data-anchor-id="calculate-gc-covariance-from-the-reference-genome-1"><span class="header-section-number">6.1.2.2</span> Calculate gc covariance (from the reference genome)</h4>
<p>Do this with any of the clrs - it just needs the bins positions.</p>
<div id="cell-71" class="cell" data-execution_count="42">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Try with only the gc_cov for chrX</span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> bioframe</span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os.path <span class="im">as</span> op</span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a>bins <span class="op">=</span> clrs[<span class="st">'fibroblast'</span>].bins().fetch(<span class="st">'chrX'</span>)[:]</span>
<span id="cb34-8"><a href="#cb34-8" aria-hidden="true" tabindex="-1"></a>out_name <span class="op">=</span> <span class="st">'../steps/rheMac10_gc_cov_X_500kb.tsv'</span></span>
<span id="cb34-9"><a href="#cb34-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-10"><a href="#cb34-10" aria-hidden="true" tabindex="-1"></a>rheMac10 <span class="op">=</span> bioframe.load_fasta(<span class="st">'../data/links/ucsc_ref/rheMac10.fa'</span>)</span>
<span id="cb34-11"><a href="#cb34-11" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> <span class="kw">not</span> op.exists(out_name):</span>
<span id="cb34-12"><a href="#cb34-12" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">'Calculate the fraction of GC basepairs for each bin'</span>)</span>
<span id="cb34-13"><a href="#cb34-13" aria-hidden="true" tabindex="-1"></a>    gc_cov <span class="op">=</span> bioframe.frac_gc(bins[[<span class="st">'chrom'</span>, <span class="st">'start'</span>, <span class="st">'end'</span>]], rheMac10)</span>
<span id="cb34-14"><a href="#cb34-14" aria-hidden="true" tabindex="-1"></a>    gc_cov.to_csv(out_name, index<span class="op">=</span><span class="va">False</span>,sep<span class="op">=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>)</span>
<span id="cb34-15"><a href="#cb34-15" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(gc_cov.info())</span>
<span id="cb34-16"><a href="#cb34-16" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>: </span>
<span id="cb34-17"><a href="#cb34-17" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"Already exists, read from file"</span>)</span>
<span id="cb34-18"><a href="#cb34-18" aria-hidden="true" tabindex="-1"></a>    gc_cov <span class="op">=</span> pd.read_csv(out_name, sep<span class="op">=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>)</span>
<span id="cb34-19"><a href="#cb34-19" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(gc_cov.info())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Already exists, read from file
&lt;class 'pandas.core.frame.DataFrame'&gt;
RangeIndex: 307 entries, 0 to 306
Data columns (total 4 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   chrom   307 non-null    object 
 1   start   307 non-null    int64  
 2   end     307 non-null    int64  
 3   GC      307 non-null    float64
dtypes: float64(1), int64(2), object(1)
memory usage: 9.7+ KB
None</code></pre>
</div>
</div>
</section>
<section id="calculate-the-e1-compartments-1" class="level4 page-columns page-full" data-number="6.1.2.3">
<h4 data-number="6.1.2.3" class="anchored" data-anchor-id="calculate-the-e1-compartments-1"><span class="header-section-number">6.1.2.3</span> Calculate the E1 compartments</h4>
<p>Loop: view_df, cis_eigs, e1_values</p>
<section id="plot-gc-covariance" class="level5" data-number="6.1.2.3.1">
<h5 data-number="6.1.2.3.1" class="anchored" data-anchor-id="plot-gc-covariance"><span class="header-section-number">6.1.2.3.1</span> Plot GC covariance</h5>
<div id="cell-74" class="cell" data-execution_count="43">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>f, ax <span class="op">=</span> plt.subplots(</span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>    figsize<span class="op">=</span>(<span class="dv">10</span>, <span class="dv">2</span>),</span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a>ax.plot(gc_cov[<span class="st">'start'</span>],gc_cov[<span class="st">'GC'</span>])</span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-9"><a href="#cb36-9" aria-hidden="true" tabindex="-1"></a>plt.tight_layout()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="03_compartments_files/figure-html/cell-47-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="chromosome-restricted-e1-compartments" class="level5" data-number="6.1.2.3.2">
<h5 data-number="6.1.2.3.2" class="anchored" data-anchor-id="chromosome-restricted-e1-compartments"><span class="header-section-number">6.1.2.3.2</span> Chromosome restricted E1 compartments</h5>
<div id="cell-76" class="cell" data-execution_count="44">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Use gc_cov to calculate eigenvectors with cooltools.eigs_cis</span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooltools</span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a>eigs_full <span class="op">=</span> {}</span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a>e1_values_full <span class="op">=</span> {}</span>
<span id="cb37-8"><a href="#cb37-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-9"><a href="#cb37-9" aria-hidden="true" tabindex="-1"></a>chrX_size <span class="op">=</span> clrs[<span class="st">'fibroblast'</span>].chromsizes[<span class="st">'chrX'</span>]</span>
<span id="cb37-10"><a href="#cb37-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-11"><a href="#cb37-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Divide into chromosome arms</span></span>
<span id="cb37-12"><a href="#cb37-12" aria-hidden="true" tabindex="-1"></a>clr <span class="op">=</span> clrs[<span class="st">'fibroblast'</span>]</span>
<span id="cb37-13"><a href="#cb37-13" aria-hidden="true" tabindex="-1"></a>view_df_full <span class="op">=</span> pd.DataFrame(</span>
<span id="cb37-14"><a href="#cb37-14" aria-hidden="true" tabindex="-1"></a>    {</span>
<span id="cb37-15"><a href="#cb37-15" aria-hidden="true" tabindex="-1"></a>    <span class="st">'chrom'</span>: <span class="st">'chrX'</span>,</span>
<span id="cb37-16"><a href="#cb37-16" aria-hidden="true" tabindex="-1"></a>    <span class="st">'start'</span>: <span class="dv">0</span>,</span>
<span id="cb37-17"><a href="#cb37-17" aria-hidden="true" tabindex="-1"></a>    <span class="st">'end'</span>: chrX_size,</span>
<span id="cb37-18"><a href="#cb37-18" aria-hidden="true" tabindex="-1"></a>    <span class="st">'name'</span>: <span class="st">'chrX'</span></span>
<span id="cb37-19"><a href="#cb37-19" aria-hidden="true" tabindex="-1"></a>    }, index<span class="op">=</span>[<span class="dv">0</span>]</span>
<span id="cb37-20"><a href="#cb37-20" aria-hidden="true" tabindex="-1"></a>                      )</span>
<span id="cb37-21"><a href="#cb37-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-22"><a href="#cb37-22" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> name, clr <span class="kw">in</span> clrs.items():</span>
<span id="cb37-23"><a href="#cb37-23" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Calculating eigenvectors for </span><span class="sc">{</span>name<span class="sc">}</span><span class="ss">, size </span><span class="sc">{</span>clr<span class="sc">.</span>binsize<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb37-24"><a href="#cb37-24" aria-hidden="true" tabindex="-1"></a>    cis_eigs_full <span class="op">=</span> cooltools.eigs_cis(</span>
<span id="cb37-25"><a href="#cb37-25" aria-hidden="true" tabindex="-1"></a>                        clr,</span>
<span id="cb37-26"><a href="#cb37-26" aria-hidden="true" tabindex="-1"></a>                        gc_cov,</span>
<span id="cb37-27"><a href="#cb37-27" aria-hidden="true" tabindex="-1"></a>                        view_df<span class="op">=</span>view_df_full,</span>
<span id="cb37-28"><a href="#cb37-28" aria-hidden="true" tabindex="-1"></a>                        n_eigs<span class="op">=</span><span class="dv">3</span>,</span>
<span id="cb37-29"><a href="#cb37-29" aria-hidden="true" tabindex="-1"></a>                        )</span>
<span id="cb37-30"><a href="#cb37-30" aria-hidden="true" tabindex="-1"></a>    eigs_full[name] <span class="op">=</span> cis_eigs_full[<span class="dv">1</span>]</span>
<span id="cb37-31"><a href="#cb37-31" aria-hidden="true" tabindex="-1"></a>    e1_track_full <span class="op">=</span> cis_eigs_full[<span class="dv">1</span>][[<span class="st">'chrom'</span>,<span class="st">'start'</span>,<span class="st">'end'</span>,<span class="st">'E1'</span>]]</span>
<span id="cb37-32"><a href="#cb37-32" aria-hidden="true" tabindex="-1"></a>    e1_values_full[name] <span class="op">=</span> e1_track_full[<span class="st">'E1'</span>].values</span>
<span id="cb37-33"><a href="#cb37-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-34"><a href="#cb37-34" aria-hidden="true" tabindex="-1"></a><span class="co">#eigs</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for fibroblast, size 500000</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for spermatogonia, size 500000</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for pachytene_spermatocyte, size 500000</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for round_spermatid, size 500000</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for sperm, size 500000</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
</div>
</section>
<section id="chromosome-arms-restricted" class="level5" data-number="6.1.2.3.3">
<h5 data-number="6.1.2.3.3" class="anchored" data-anchor-id="chromosome-arms-restricted"><span class="header-section-number">6.1.2.3.3</span> Chromosome arms restricted</h5>
<div id="cell-78" class="cell" data-execution_count="45">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Use gc_cov to calculate eigenvectors with cooltools.eigs_cis</span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooltools</span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a>eigs <span class="op">=</span> {}</span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a>e1_values <span class="op">=</span> {}</span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-9"><a href="#cb48-9" aria-hidden="true" tabindex="-1"></a>chrX_size <span class="op">=</span> clrs[<span class="st">'fibroblast'</span>].chromsizes[<span class="st">'chrX'</span>]</span>
<span id="cb48-10"><a href="#cb48-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-11"><a href="#cb48-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Divide into chromosome arms</span></span>
<span id="cb48-12"><a href="#cb48-12" aria-hidden="true" tabindex="-1"></a>view_df <span class="op">=</span> pd.DataFrame(</span>
<span id="cb48-13"><a href="#cb48-13" aria-hidden="true" tabindex="-1"></a>    {</span>
<span id="cb48-14"><a href="#cb48-14" aria-hidden="true" tabindex="-1"></a>    <span class="st">'chrom'</span>: <span class="st">'chrX'</span>,</span>
<span id="cb48-15"><a href="#cb48-15" aria-hidden="true" tabindex="-1"></a>    <span class="st">'start'</span>: [<span class="dv">0</span>, <span class="dv">59_000_001</span>],</span>
<span id="cb48-16"><a href="#cb48-16" aria-hidden="true" tabindex="-1"></a>    <span class="st">'end'</span>: [<span class="dv">59_000_000</span>, chrX_size],</span>
<span id="cb48-17"><a href="#cb48-17" aria-hidden="true" tabindex="-1"></a>    <span class="st">'name'</span>: [<span class="st">'X_short'</span>, <span class="st">'X_long'</span>]</span>
<span id="cb48-18"><a href="#cb48-18" aria-hidden="true" tabindex="-1"></a>    }, index<span class="op">=</span>[<span class="dv">0</span>,<span class="dv">1</span>]</span>
<span id="cb48-19"><a href="#cb48-19" aria-hidden="true" tabindex="-1"></a>                      )</span>
<span id="cb48-20"><a href="#cb48-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-21"><a href="#cb48-21" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> name, clr <span class="kw">in</span> clrs.items():</span>
<span id="cb48-22"><a href="#cb48-22" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Calculating eigenvectors for </span><span class="sc">{</span>name<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb48-23"><a href="#cb48-23" aria-hidden="true" tabindex="-1"></a>    cis_eigs <span class="op">=</span> cooltools.eigs_cis(</span>
<span id="cb48-24"><a href="#cb48-24" aria-hidden="true" tabindex="-1"></a>                        clr,</span>
<span id="cb48-25"><a href="#cb48-25" aria-hidden="true" tabindex="-1"></a>                        gc_cov,</span>
<span id="cb48-26"><a href="#cb48-26" aria-hidden="true" tabindex="-1"></a>                        view_df<span class="op">=</span>view_df,</span>
<span id="cb48-27"><a href="#cb48-27" aria-hidden="true" tabindex="-1"></a>                        n_eigs<span class="op">=</span><span class="dv">3</span>,</span>
<span id="cb48-28"><a href="#cb48-28" aria-hidden="true" tabindex="-1"></a>                        )</span>
<span id="cb48-29"><a href="#cb48-29" aria-hidden="true" tabindex="-1"></a>    eigs[name] <span class="op">=</span> cis_eigs[<span class="dv">1</span>]</span>
<span id="cb48-30"><a href="#cb48-30" aria-hidden="true" tabindex="-1"></a>    e1_track <span class="op">=</span> cis_eigs[<span class="dv">1</span>][[<span class="st">'chrom'</span>,<span class="st">'start'</span>,<span class="st">'end'</span>,<span class="st">'E1'</span>]]</span>
<span id="cb48-31"><a href="#cb48-31" aria-hidden="true" tabindex="-1"></a>    e1_values[name] <span class="op">=</span> e1_track[<span class="st">'E1'</span>].values</span>
<span id="cb48-32"><a href="#cb48-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-33"><a href="#cb48-33" aria-hidden="true" tabindex="-1"></a><span class="co">#eigs</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for fibroblast
Calculating eigenvectors for spermatogonia</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")
/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for pachytene_spermatocyte
Calculating eigenvectors for round_spermatid</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")
/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for sperm</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
</div>
</section>
<section id="refined-10mb-windows-restricted" class="level5" data-number="6.1.2.3.4">
<h5 data-number="6.1.2.3.4" class="anchored" data-anchor-id="refined-10mb-windows-restricted"><span class="header-section-number">6.1.2.3.4</span> Refined: 10Mb windows restricted</h5>
<div id="cell-80" class="cell" data-execution_count="46">
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Use gc_cov to calculate eigenvectors with cooltools.eigs_cis</span></span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooltools</span>
<span id="cb55-4"><a href="#cb55-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-5"><a href="#cb55-5" aria-hidden="true" tabindex="-1"></a>eigs_10mb <span class="op">=</span> {}</span>
<span id="cb55-6"><a href="#cb55-6" aria-hidden="true" tabindex="-1"></a>e1_values_10mb <span class="op">=</span> {}</span>
<span id="cb55-7"><a href="#cb55-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-8"><a href="#cb55-8" aria-hidden="true" tabindex="-1"></a>chrX_size <span class="op">=</span> clrs[<span class="st">'fibroblast'</span>].chromsizes[<span class="st">'chrX'</span>]</span>
<span id="cb55-9"><a href="#cb55-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-10"><a href="#cb55-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate in 1Mb windows</span></span>
<span id="cb55-11"><a href="#cb55-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the window size (1Mb)</span></span>
<span id="cb55-12"><a href="#cb55-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-13"><a href="#cb55-13" aria-hidden="true" tabindex="-1"></a>window_size <span class="op">=</span> <span class="dv">10_000_000</span></span>
<span id="cb55-14"><a href="#cb55-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-15"><a href="#cb55-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate the start and end positions for each window</span></span>
<span id="cb55-16"><a href="#cb55-16" aria-hidden="true" tabindex="-1"></a>start_positions <span class="op">=</span> <span class="bu">list</span>(<span class="bu">range</span>(<span class="dv">0</span>, chrX_size, window_size))</span>
<span id="cb55-17"><a href="#cb55-17" aria-hidden="true" tabindex="-1"></a>end_positions <span class="op">=</span> [<span class="bu">min</span>(start <span class="op">+</span> window_size, chrX_size) <span class="cf">for</span> start <span class="kw">in</span> start_positions]</span>
<span id="cb55-18"><a href="#cb55-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the DataFrame</span></span>
<span id="cb55-19"><a href="#cb55-19" aria-hidden="true" tabindex="-1"></a>view_df_10mb <span class="op">=</span> pd.DataFrame({</span>
<span id="cb55-20"><a href="#cb55-20" aria-hidden="true" tabindex="-1"></a>    <span class="st">'chrom'</span>: [<span class="st">'chrX'</span>] <span class="op">*</span> <span class="bu">len</span>(start_positions),</span>
<span id="cb55-21"><a href="#cb55-21" aria-hidden="true" tabindex="-1"></a>    <span class="st">'start'</span>: start_positions,</span>
<span id="cb55-22"><a href="#cb55-22" aria-hidden="true" tabindex="-1"></a>    <span class="st">'end'</span>: end_positions,</span>
<span id="cb55-23"><a href="#cb55-23" aria-hidden="true" tabindex="-1"></a>    <span class="st">'name'</span>: [<span class="ss">f'X_</span><span class="sc">{</span>i<span class="sc">}</span><span class="ss">'</span> <span class="cf">for</span> i <span class="kw">in</span> <span class="bu">range</span>(<span class="bu">len</span>(start_positions))]</span>
<span id="cb55-24"><a href="#cb55-24" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb55-25"><a href="#cb55-25" aria-hidden="true" tabindex="-1"></a><span class="co">#display(view_df)</span></span>
<span id="cb55-26"><a href="#cb55-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-27"><a href="#cb55-27" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> name, clr <span class="kw">in</span> clrs.items():</span>
<span id="cb55-28"><a href="#cb55-28" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Calculating eigenvectors for </span><span class="sc">{</span>name<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb55-29"><a href="#cb55-29" aria-hidden="true" tabindex="-1"></a>    cis_eigs_10mb <span class="op">=</span> cooltools.eigs_cis(</span>
<span id="cb55-30"><a href="#cb55-30" aria-hidden="true" tabindex="-1"></a>                        clr,</span>
<span id="cb55-31"><a href="#cb55-31" aria-hidden="true" tabindex="-1"></a>                        gc_cov,</span>
<span id="cb55-32"><a href="#cb55-32" aria-hidden="true" tabindex="-1"></a>                        view_df<span class="op">=</span>view_df_10mb,</span>
<span id="cb55-33"><a href="#cb55-33" aria-hidden="true" tabindex="-1"></a>                        n_eigs<span class="op">=</span><span class="dv">3</span>,</span>
<span id="cb55-34"><a href="#cb55-34" aria-hidden="true" tabindex="-1"></a>                        )</span>
<span id="cb55-35"><a href="#cb55-35" aria-hidden="true" tabindex="-1"></a>    eigs_10mb[name] <span class="op">=</span> cis_eigs_10mb[<span class="dv">1</span>]</span>
<span id="cb55-36"><a href="#cb55-36" aria-hidden="true" tabindex="-1"></a>    e1_track_10mb <span class="op">=</span> cis_eigs_10mb[<span class="dv">1</span>][[<span class="st">'chrom'</span>,<span class="st">'start'</span>,<span class="st">'end'</span>,<span class="st">'E1'</span>]]</span>
<span id="cb55-37"><a href="#cb55-37" aria-hidden="true" tabindex="-1"></a>    e1_values_10mb[name] <span class="op">=</span> e1_track_10mb[<span class="st">'E1'</span>].values</span>
<span id="cb55-38"><a href="#cb55-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-39"><a href="#cb55-39" aria-hidden="true" tabindex="-1"></a><span class="co">#eigs</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for fibroblast</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for spermatogonia</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for pachytene_spermatocyte</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for round_spermatid</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for sperm</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
</div>
</section>
<section id="plot-nan-histogram" class="level5 page-columns page-full" data-number="6.1.2.3.5">
<h5 data-number="6.1.2.3.5" class="anchored" data-anchor-id="plot-nan-histogram"><span class="header-section-number">6.1.2.3.5</span> Plot NaN histogram</h5>
<div id="cell-e1_nan_hist" class="cell page-columns page-full" data-execution_count="47">
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Check the number of NaN values in the E1 column</span></span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb66-3"><a href="#cb66-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-4"><a href="#cb66-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Check the number of NaN values in the E1 column and create a DataFrame</span></span>
<span id="cb66-5"><a href="#cb66-5" aria-hidden="true" tabindex="-1"></a>nan_counts <span class="op">=</span> {k: {<span class="st">'length'</span>: <span class="bu">len</span>(v), <span class="st">'NaNs'</span>: np.isnan(v).<span class="bu">sum</span>()} <span class="cf">for</span> k, v <span class="kw">in</span> e1_values.items()}</span>
<span id="cb66-6"><a href="#cb66-6" aria-hidden="true" tabindex="-1"></a><span class="co">#display(pd.DataFrame.from_dict(nan_counts, orient='index'))</span></span>
<span id="cb66-7"><a href="#cb66-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-8"><a href="#cb66-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Locate the NaN values (histogram)</span></span>
<span id="cb66-9"><a href="#cb66-9" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb66-10"><a href="#cb66-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-11"><a href="#cb66-11" aria-hidden="true" tabindex="-1"></a>f, ax <span class="op">=</span> plt.subplots(<span class="dv">1</span>, <span class="dv">5</span>, figsize<span class="op">=</span>(<span class="dv">20</span>,<span class="dv">2</span>))</span>
<span id="cb66-12"><a href="#cb66-12" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i, (name, track) <span class="kw">in</span> <span class="bu">enumerate</span>(eigs.items()):</span>
<span id="cb66-13"><a href="#cb66-13" aria-hidden="true" tabindex="-1"></a>    e1 <span class="op">=</span> track[<span class="st">'E1'</span>].values</span>
<span id="cb66-14"><a href="#cb66-14" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Locate NaN values</span></span>
<span id="cb66-15"><a href="#cb66-15" aria-hidden="true" tabindex="-1"></a>    e1_nan <span class="op">=</span> np.where(np.isnan(e1))</span>
<span id="cb66-16"><a href="#cb66-16" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Plot histogram</span></span>
<span id="cb66-17"><a href="#cb66-17" aria-hidden="true" tabindex="-1"></a>    ax[i].hist(e1_nan, bins<span class="op">=</span><span class="dv">100</span>)</span>
<span id="cb66-18"><a href="#cb66-18" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb66-19"><a href="#cb66-19" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Plot median line</span></span>
<span id="cb66-20"><a href="#cb66-20" aria-hidden="true" tabindex="-1"></a>    median_pos <span class="op">=</span> np.median(e1_nan)</span>
<span id="cb66-21"><a href="#cb66-21" aria-hidden="true" tabindex="-1"></a>    ax[i].axvline(median_pos, color<span class="op">=</span><span class="st">'r'</span>)</span>
<span id="cb66-22"><a href="#cb66-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-23"><a href="#cb66-23" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Layout</span></span>
<span id="cb66-24"><a href="#cb66-24" aria-hidden="true" tabindex="-1"></a>    ax[i].set_title(name)</span>
<span id="cb66-25"><a href="#cb66-25" aria-hidden="true" tabindex="-1"></a>    xticks <span class="op">=</span> np.linspace(<span class="dv">0</span>, <span class="bu">len</span>(e1), num<span class="op">=</span><span class="dv">5</span>)</span>
<span id="cb66-26"><a href="#cb66-26" aria-hidden="true" tabindex="-1"></a>    xticks <span class="op">=</span> np.append(xticks, median_pos)  <span class="co"># Add median position to xticks</span></span>
<span id="cb66-27"><a href="#cb66-27" aria-hidden="true" tabindex="-1"></a>    ax[i].set_xticks(xticks)</span>
<span id="cb66-28"><a href="#cb66-28" aria-hidden="true" tabindex="-1"></a>    xticklabels <span class="op">=</span> np.linspace(<span class="dv">0</span>, <span class="bu">len</span>(e1) <span class="op">*</span> <span class="fl">0.5</span>, num<span class="op">=</span><span class="dv">5</span>, dtype <span class="op">=</span> <span class="st">'int'</span>).tolist()</span>
<span id="cb66-29"><a href="#cb66-29" aria-hidden="true" tabindex="-1"></a>    xticklabels.append(median_pos<span class="op">*</span><span class="fl">0.5</span>)  <span class="co"># Add median label</span></span>
<span id="cb66-30"><a href="#cb66-30" aria-hidden="true" tabindex="-1"></a>    ax[i].set_xticklabels(xticklabels, rotation<span class="op">=</span><span class="dv">45</span>)</span>
<span id="cb66-31"><a href="#cb66-31" aria-hidden="true" tabindex="-1"></a>    ax[i].set_xlabel(<span class="st">'Position (Mbp)'</span>)</span>
<span id="cb66-32"><a href="#cb66-32" aria-hidden="true" tabindex="-1"></a>                     </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display page-columns page-full">
<div id="e1_nan_hist" class="quarto-figure quarto-figure-center anchored page-columns page-full">
<figure class="figure page-columns page-full">
<p class="page-columns page-full"><img src="03_compartments_files/figure-html/e1_nan_hist-output-1.png" class="img-fluid figure-img column-page-inset"></p>
<figcaption>Histogram of NaN values in the E1 eigenvector for each cell type.</figcaption>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="plot-the-e1-compartments" class="level4 page-columns page-full" data-number="6.1.2.4">
<h4 data-number="6.1.2.4" class="anchored" data-anchor-id="plot-the-e1-compartments"><span class="header-section-number">6.1.2.4</span> Plot the E1 compartments</h4>
<div id="cell-e1-fullmerged-all-500kb" class="cell page-columns page-full" data-execution_count="48">
<div class="sourceCode cell-code" id="cb67"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb67-2"><a href="#cb67-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb67-3"><a href="#cb67-3" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> matplotlib.colors <span class="im">import</span> LogNorm</span>
<span id="cb67-4"><a href="#cb67-4" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> mpl_toolkits.axes_grid1 <span class="im">import</span> make_axes_locatable</span>
<span id="cb67-5"><a href="#cb67-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-6"><a href="#cb67-6" aria-hidden="true" tabindex="-1"></a>names_abbr <span class="op">=</span> {<span class="st">'fibroblast'</span>: <span class="st">'Fibroblast'</span>, <span class="st">'spermatogonia'</span>: <span class="st">'Spermatogonia'</span>, <span class="st">'pachytene_spermatocyte'</span>: <span class="st">'Pachytene Spermatogonia'</span>, <span class="st">'round_spermatid'</span>: <span class="st">'Round Spermatid'</span>, <span class="st">'sperm'</span>: <span class="st">'Sperm'</span>}</span>
<span id="cb67-7"><a href="#cb67-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-8"><a href="#cb67-8" aria-hidden="true" tabindex="-1"></a>chrom_start <span class="op">=</span> e1_track_full[<span class="st">'start'</span>].values</span>
<span id="cb67-9"><a href="#cb67-9" aria-hidden="true" tabindex="-1"></a>chrom_end <span class="op">=</span> e1_track_full[<span class="st">'end'</span>].values<span class="op">-</span><span class="dv">1</span></span>
<span id="cb67-10"><a href="#cb67-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-11"><a href="#cb67-11" aria-hidden="true" tabindex="-1"></a>f, axs <span class="op">=</span> plt.subplots(<span class="dv">5</span>, <span class="dv">3</span>, figsize<span class="op">=</span>(<span class="dv">18</span>, <span class="dv">8</span>), sharex<span class="op">=</span><span class="va">True</span>, sharey<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb67-12"><a href="#cb67-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-13"><a href="#cb67-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Populate the first column</span></span>
<span id="cb67-14"><a href="#cb67-14" aria-hidden="true" tabindex="-1"></a>axs[<span class="dv">0</span>,<span class="dv">0</span>].set_title(<span class="st">'E1: Full-chromosome restricted'</span>)</span>
<span id="cb67-15"><a href="#cb67-15" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i, (name, e1) <span class="kw">in</span> <span class="bu">enumerate</span>(e1_values_full.items()):</span>
<span id="cb67-16"><a href="#cb67-16" aria-hidden="true" tabindex="-1"></a>    ax <span class="op">=</span> axs[i,<span class="dv">0</span>]</span>
<span id="cb67-17"><a href="#cb67-17" aria-hidden="true" tabindex="-1"></a>    ax.set_ylabel(names_abbr[name])</span>
<span id="cb67-18"><a href="#cb67-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-19"><a href="#cb67-19" aria-hidden="true" tabindex="-1"></a>    <span class="co">#ax.set_title(name)</span></span>
<span id="cb67-20"><a href="#cb67-20" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb67-21"><a href="#cb67-21" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create stairs</span></span>
<span id="cb67-22"><a href="#cb67-22" aria-hidden="true" tabindex="-1"></a>    x <span class="op">=</span> np.zeros(<span class="dv">2</span><span class="op">*</span>chrom_start.size)</span>
<span id="cb67-23"><a href="#cb67-23" aria-hidden="true" tabindex="-1"></a>    y <span class="op">=</span> np.zeros(<span class="dv">2</span><span class="op">*</span>chrom_start.size)</span>
<span id="cb67-24"><a href="#cb67-24" aria-hidden="true" tabindex="-1"></a>    x[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> chrom_start</span>
<span id="cb67-25"><a href="#cb67-25" aria-hidden="true" tabindex="-1"></a>    x[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> chrom_end</span>
<span id="cb67-26"><a href="#cb67-26" aria-hidden="true" tabindex="-1"></a>    y[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> e1</span>
<span id="cb67-27"><a href="#cb67-27" aria-hidden="true" tabindex="-1"></a>    y[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> e1</span>
<span id="cb67-28"><a href="#cb67-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-29"><a href="#cb67-29" aria-hidden="true" tabindex="-1"></a>    ax.fill_between(x, y, <span class="dv">0</span>, where<span class="op">=</span>(y <span class="op">&gt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:red'</span>)</span>
<span id="cb67-30"><a href="#cb67-30" aria-hidden="true" tabindex="-1"></a>    ax.fill_between(x, y, <span class="dv">0</span>, where<span class="op">=</span>(y <span class="op">&lt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:blue'</span>)</span>
<span id="cb67-31"><a href="#cb67-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-32"><a href="#cb67-32" aria-hidden="true" tabindex="-1"></a><span class="co"># Populate the second column</span></span>
<span id="cb67-33"><a href="#cb67-33" aria-hidden="true" tabindex="-1"></a>axs[<span class="dv">0</span>,<span class="dv">1</span>].set_title(<span class="st">'E1: Chromosome-arms restricted'</span>)</span>
<span id="cb67-34"><a href="#cb67-34" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i, (name, e1) <span class="kw">in</span> <span class="bu">enumerate</span>(e1_values.items()):</span>
<span id="cb67-35"><a href="#cb67-35" aria-hidden="true" tabindex="-1"></a>    ax <span class="op">=</span> axs[i,<span class="dv">1</span>]</span>
<span id="cb67-36"><a href="#cb67-36" aria-hidden="true" tabindex="-1"></a>    <span class="co">#ax.set_title(name)</span></span>
<span id="cb67-37"><a href="#cb67-37" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb67-38"><a href="#cb67-38" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create stairs</span></span>
<span id="cb67-39"><a href="#cb67-39" aria-hidden="true" tabindex="-1"></a>    x <span class="op">=</span> np.zeros(<span class="dv">2</span><span class="op">*</span>chrom_start.size)</span>
<span id="cb67-40"><a href="#cb67-40" aria-hidden="true" tabindex="-1"></a>    y <span class="op">=</span> np.zeros(<span class="dv">2</span><span class="op">*</span>chrom_start.size)</span>
<span id="cb67-41"><a href="#cb67-41" aria-hidden="true" tabindex="-1"></a>    x[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> chrom_start</span>
<span id="cb67-42"><a href="#cb67-42" aria-hidden="true" tabindex="-1"></a>    x[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> chrom_end</span>
<span id="cb67-43"><a href="#cb67-43" aria-hidden="true" tabindex="-1"></a>    y[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> e1</span>
<span id="cb67-44"><a href="#cb67-44" aria-hidden="true" tabindex="-1"></a>    y[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> e1</span>
<span id="cb67-45"><a href="#cb67-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-46"><a href="#cb67-46" aria-hidden="true" tabindex="-1"></a>    ax.fill_between(x, y, <span class="dv">0</span>, where<span class="op">=</span>(y <span class="op">&gt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:red'</span>)</span>
<span id="cb67-47"><a href="#cb67-47" aria-hidden="true" tabindex="-1"></a>    ax.fill_between(x, y, <span class="dv">0</span>, where<span class="op">=</span>(y <span class="op">&lt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:blue'</span>)</span>
<span id="cb67-48"><a href="#cb67-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-49"><a href="#cb67-49" aria-hidden="true" tabindex="-1"></a><span class="co"># Populate the third column</span></span>
<span id="cb67-50"><a href="#cb67-50" aria-hidden="true" tabindex="-1"></a>axs[<span class="dv">0</span>,<span class="dv">2</span>].set_title(<span class="st">'E1: 10Mb restricted (refined)'</span>)</span>
<span id="cb67-51"><a href="#cb67-51" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i, (name, e1) <span class="kw">in</span> <span class="bu">enumerate</span>(e1_values_10mb.items()):</span>
<span id="cb67-52"><a href="#cb67-52" aria-hidden="true" tabindex="-1"></a>    ax <span class="op">=</span> axs[i,<span class="dv">2</span>]</span>
<span id="cb67-53"><a href="#cb67-53" aria-hidden="true" tabindex="-1"></a>    <span class="co">#ax.set_title(name)</span></span>
<span id="cb67-54"><a href="#cb67-54" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb67-55"><a href="#cb67-55" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create stairs</span></span>
<span id="cb67-56"><a href="#cb67-56" aria-hidden="true" tabindex="-1"></a>    x <span class="op">=</span> np.zeros(<span class="dv">2</span><span class="op">*</span>chrom_start.size)</span>
<span id="cb67-57"><a href="#cb67-57" aria-hidden="true" tabindex="-1"></a>    y <span class="op">=</span> np.zeros(<span class="dv">2</span><span class="op">*</span>chrom_start.size)</span>
<span id="cb67-58"><a href="#cb67-58" aria-hidden="true" tabindex="-1"></a>    x[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> chrom_start</span>
<span id="cb67-59"><a href="#cb67-59" aria-hidden="true" tabindex="-1"></a>    x[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> chrom_end</span>
<span id="cb67-60"><a href="#cb67-60" aria-hidden="true" tabindex="-1"></a>    y[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> e1</span>
<span id="cb67-61"><a href="#cb67-61" aria-hidden="true" tabindex="-1"></a>    y[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> e1</span>
<span id="cb67-62"><a href="#cb67-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-63"><a href="#cb67-63" aria-hidden="true" tabindex="-1"></a>    ax.fill_between(x, y, <span class="dv">0</span>, where<span class="op">=</span>(y <span class="op">&gt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:red'</span>)</span>
<span id="cb67-64"><a href="#cb67-64" aria-hidden="true" tabindex="-1"></a>    ax.fill_between(x, y, <span class="dv">0</span>, where<span class="op">=</span>(y <span class="op">&lt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:blue'</span>)</span>
<span id="cb67-65"><a href="#cb67-65" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb67-66"><a href="#cb67-66" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb67-67"><a href="#cb67-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-68"><a href="#cb67-68" aria-hidden="true" tabindex="-1"></a><span class="co"># Set y-limits for all subplots</span></span>
<span id="cb67-69"><a href="#cb67-69" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> ax <span class="kw">in</span> axs.flat:</span>
<span id="cb67-70"><a href="#cb67-70" aria-hidden="true" tabindex="-1"></a>    ax.set_ylim(<span class="op">-</span><span class="fl">0.8</span>, <span class="fl">0.8</span>)</span>
<span id="cb67-71"><a href="#cb67-71" aria-hidden="true" tabindex="-1"></a>    ax.spines[:].set_visible(<span class="va">False</span>)</span>
<span id="cb67-72"><a href="#cb67-72" aria-hidden="true" tabindex="-1"></a>    ax.set_xticks([])</span>
<span id="cb67-73"><a href="#cb67-73" aria-hidden="true" tabindex="-1"></a>    ax.set_yticks([])</span>
<span id="cb67-74"><a href="#cb67-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-75"><a href="#cb67-75" aria-hidden="true" tabindex="-1"></a>plt.tight_layout()</span>
<span id="cb67-76"><a href="#cb67-76" aria-hidden="true" tabindex="-1"></a>plt.savefig(<span class="st">'../steps/e1_plot_full.svg'</span>, bbox_inches<span class="op">=</span><span class="st">'tight'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display page-columns page-full">
<div id="e1-fullmerged-all-500kb" class="quarto-figure quarto-figure-center anchored page-columns page-full">
<figure class="figure page-columns page-full">
<p class="page-columns page-full"><img src="03_compartments_files/figure-html/e1-fullmerged-all-500kb-output-1.png" class="img-fluid figure-img column-page-inset"></p>
<figcaption>E1 eigenvector values for all merged samples at 500kb resolution. Left: Chromosome-arm restricted E1. Right: 10Mb window restricted E1.</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="plot-matrices-with-compartments-round-spermatid" class="level4" data-number="6.1.2.5">
<h4 data-number="6.1.2.5" class="anchored" data-anchor-id="plot-matrices-with-compartments-round-spermatid"><span class="header-section-number">6.1.2.5</span> Plot matrices with compartments (round spermatid)</h4>
<div id="cell-e1-matrix-500kb-full-arms-10mb-round_spermatid" class="cell" data-execution_count="55">
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooltools.lib.plotting</span>
<span id="cb68-2"><a href="#cb68-2" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> matplotlib.colors <span class="im">import</span> LogNorm</span>
<span id="cb68-3"><a href="#cb68-3" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> mpl_toolkits.axes_grid1 <span class="im">import</span> make_axes_locatable</span>
<span id="cb68-4"><a href="#cb68-4" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> cooltools.lib.numutils <span class="im">import</span> adaptive_coarsegrain, interp_nan</span>
<span id="cb68-5"><a href="#cb68-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-6"><a href="#cb68-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-7"><a href="#cb68-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-8"><a href="#cb68-8" aria-hidden="true" tabindex="-1"></a>clr <span class="op">=</span> clrs[<span class="st">'round_spermatid'</span>]</span>
<span id="cb68-9"><a href="#cb68-9" aria-hidden="true" tabindex="-1"></a>chrom_start, chrom_end <span class="op">=</span> e1_track_full[<span class="st">'start'</span>].values, e1_track_full[<span class="st">'end'</span>].values<span class="op">-</span><span class="dv">1</span></span>
<span id="cb68-10"><a href="#cb68-10" aria-hidden="true" tabindex="-1"></a>e1 <span class="op">=</span> e1_values_10mb[<span class="st">'round_spermatid'</span>]</span>
<span id="cb68-11"><a href="#cb68-11" aria-hidden="true" tabindex="-1"></a>nbins <span class="op">=</span> <span class="bu">len</span>(clr.bins().fetch(<span class="st">'chrX'</span>))</span>
<span id="cb68-12"><a href="#cb68-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-13"><a href="#cb68-13" aria-hidden="true" tabindex="-1"></a>f, ax <span class="op">=</span> plt.subplots(</span>
<span id="cb68-14"><a href="#cb68-14" aria-hidden="true" tabindex="-1"></a>    figsize<span class="op">=</span>(<span class="dv">9</span>,<span class="dv">6</span>),</span>
<span id="cb68-15"><a href="#cb68-15" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb68-16"><a href="#cb68-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-17"><a href="#cb68-17" aria-hidden="true" tabindex="-1"></a>norm <span class="op">=</span> LogNorm(vmax<span class="op">=</span><span class="fl">0.1</span>)</span>
<span id="cb68-18"><a href="#cb68-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-19"><a href="#cb68-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-20"><a href="#cb68-20" aria-hidden="true" tabindex="-1"></a>e1_list <span class="op">=</span> [e1_values_10mb[<span class="st">'round_spermatid'</span>], e1_values[<span class="st">'round_spermatid'</span>], e1_values_full[<span class="st">'round_spermatid'</span>]]</span>
<span id="cb68-21"><a href="#cb68-21" aria-hidden="true" tabindex="-1"></a>e1_names <span class="op">=</span> [<span class="st">'10Mb'</span>, <span class="st">'Arms'</span>, <span class="st">'Full'</span>]</span>
<span id="cb68-22"><a href="#cb68-22" aria-hidden="true" tabindex="-1"></a>colors <span class="op">=</span> [<span class="st">'tab:red'</span>, <span class="st">'tab:blue'</span>, <span class="st">'tab:green'</span>]</span>
<span id="cb68-23"><a href="#cb68-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-24"><a href="#cb68-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-25"><a href="#cb68-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-26"><a href="#cb68-26" aria-hidden="true" tabindex="-1"></a>im <span class="op">=</span> ax.matshow(</span>
<span id="cb68-27"><a href="#cb68-27" aria-hidden="true" tabindex="-1"></a>    clr.matrix().fetch(<span class="st">'chrX'</span>),</span>
<span id="cb68-28"><a href="#cb68-28" aria-hidden="true" tabindex="-1"></a>    norm<span class="op">=</span>norm,</span>
<span id="cb68-29"><a href="#cb68-29" aria-hidden="true" tabindex="-1"></a>    cmap<span class="op">=</span><span class="st">'fall'</span>,</span>
<span id="cb68-30"><a href="#cb68-30" aria-hidden="true" tabindex="-1"></a>)<span class="op">;</span></span>
<span id="cb68-31"><a href="#cb68-31" aria-hidden="true" tabindex="-1"></a>plt.axis([<span class="dv">0</span>,nbins,nbins,<span class="dv">0</span>])</span>
<span id="cb68-32"><a href="#cb68-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-33"><a href="#cb68-33" aria-hidden="true" tabindex="-1"></a>divider <span class="op">=</span> make_axes_locatable(ax)</span>
<span id="cb68-34"><a href="#cb68-34" aria-hidden="true" tabindex="-1"></a>cax <span class="op">=</span> divider.append_axes(<span class="st">"right"</span>, size<span class="op">=</span><span class="st">"2%"</span>, pad<span class="op">=</span><span class="fl">0.05</span>)</span>
<span id="cb68-35"><a href="#cb68-35" aria-hidden="true" tabindex="-1"></a>plt.colorbar(im, cax<span class="op">=</span>cax, label<span class="op">=</span><span class="st">'corrected frequencies'</span>)<span class="op">;</span></span>
<span id="cb68-36"><a href="#cb68-36" aria-hidden="true" tabindex="-1"></a>ax.set_ylabel(<span class="st">'chrX:500kb bins, #bin'</span>)</span>
<span id="cb68-37"><a href="#cb68-37" aria-hidden="true" tabindex="-1"></a>ax.xaxis.set_visible(<span class="va">False</span>)</span>
<span id="cb68-38"><a href="#cb68-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-39"><a href="#cb68-39" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i, e1 <span class="kw">in</span> <span class="bu">enumerate</span>(e1_list):</span>
<span id="cb68-40"><a href="#cb68-40" aria-hidden="true" tabindex="-1"></a>    ax1 <span class="op">=</span> divider.append_axes(<span class="st">"top"</span>, size<span class="op">=</span><span class="st">"10%"</span>, pad<span class="op">=</span><span class="fl">0.05</span>, sharex <span class="op">=</span> ax)</span>
<span id="cb68-41"><a href="#cb68-41" aria-hidden="true" tabindex="-1"></a>    <span class="co"># weights = clr.bins()[:]['weight'].values</span></span>
<span id="cb68-42"><a href="#cb68-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-43"><a href="#cb68-43" aria-hidden="true" tabindex="-1"></a>    <span class="co"># ax1.plot(e1, label='E1')</span></span>
<span id="cb68-44"><a href="#cb68-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-45"><a href="#cb68-45" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create stairs</span></span>
<span id="cb68-46"><a href="#cb68-46" aria-hidden="true" tabindex="-1"></a>    x <span class="op">=</span> np.zeros(<span class="dv">2</span><span class="op">*</span>chrom_start.size)</span>
<span id="cb68-47"><a href="#cb68-47" aria-hidden="true" tabindex="-1"></a>    y <span class="op">=</span> np.zeros(<span class="dv">2</span><span class="op">*</span>chrom_start.size)</span>
<span id="cb68-48"><a href="#cb68-48" aria-hidden="true" tabindex="-1"></a>    <span class="co">#smooth_y = np.zeros(2*chrom_start.size)</span></span>
<span id="cb68-49"><a href="#cb68-49" aria-hidden="true" tabindex="-1"></a>    x[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> chrom_start<span class="op">/</span><span class="dv">500_000</span></span>
<span id="cb68-50"><a href="#cb68-50" aria-hidden="true" tabindex="-1"></a>    x[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> chrom_end<span class="op">/</span><span class="dv">500_000</span></span>
<span id="cb68-51"><a href="#cb68-51" aria-hidden="true" tabindex="-1"></a>    y[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> e1</span>
<span id="cb68-52"><a href="#cb68-52" aria-hidden="true" tabindex="-1"></a>    y[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> e1</span>
<span id="cb68-53"><a href="#cb68-53" aria-hidden="true" tabindex="-1"></a>    <span class="co"># smooth_e1 = (lambda x: x.rolling(5, 1, center=True).sum())(pd.DataFrame(e1, columns=['value']))['value'].values</span></span>
<span id="cb68-54"><a href="#cb68-54" aria-hidden="true" tabindex="-1"></a>    <span class="co"># smooth_y[0::2] = smooth_e1</span></span>
<span id="cb68-55"><a href="#cb68-55" aria-hidden="true" tabindex="-1"></a>    <span class="co"># smooth_y[1::2] = smooth_e1</span></span>
<span id="cb68-56"><a href="#cb68-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-57"><a href="#cb68-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-58"><a href="#cb68-58" aria-hidden="true" tabindex="-1"></a>    ax1.fill_between(x, y, <span class="dv">0</span>, where<span class="op">=</span>(y <span class="op">&gt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:red'</span>)</span>
<span id="cb68-59"><a href="#cb68-59" aria-hidden="true" tabindex="-1"></a>    ax1.fill_between(x, y, <span class="dv">0</span>, where<span class="op">=</span>(y <span class="op">&lt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:blue'</span>)</span>
<span id="cb68-60"><a href="#cb68-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-61"><a href="#cb68-61" aria-hidden="true" tabindex="-1"></a>    ax1.set_ylabel(e1_names[i], rotation<span class="op">=</span><span class="dv">70</span>)</span>
<span id="cb68-62"><a href="#cb68-62" aria-hidden="true" tabindex="-1"></a>    ax1.set_ylim(<span class="op">-</span><span class="fl">0.8</span>, <span class="fl">0.8</span>)</span>
<span id="cb68-63"><a href="#cb68-63" aria-hidden="true" tabindex="-1"></a>    ax1.set_xticks([])</span>
<span id="cb68-64"><a href="#cb68-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-65"><a href="#cb68-65" aria-hidden="true" tabindex="-1"></a>ax1.set_title(<span class="st">'Round Spermatid: 500kb bins'</span>)</span>
<span id="cb68-66"><a href="#cb68-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-67"><a href="#cb68-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-68"><a href="#cb68-68" aria-hidden="true" tabindex="-1"></a><span class="co">#    # Plot the sign changes on the matrix </span></span>
<span id="cb68-69"><a href="#cb68-69" aria-hidden="true" tabindex="-1"></a><span class="co">#     col = colors[i]</span></span>
<span id="cb68-70"><a href="#cb68-70" aria-hidden="true" tabindex="-1"></a><span class="co">#     for i in np.where(np.diff( (pd.Series(e1)&gt;0).astype(int)))[0]:</span></span>
<span id="cb68-71"><a href="#cb68-71" aria-hidden="true" tabindex="-1"></a><span class="co">#         # Horisontal lines where E1 intersects 0</span></span>
<span id="cb68-72"><a href="#cb68-72" aria-hidden="true" tabindex="-1"></a><span class="co">#         ax.plot([0,nbins],[i,i],col,lw=0.5)</span></span>
<span id="cb68-73"><a href="#cb68-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-74"><a href="#cb68-74" aria-hidden="true" tabindex="-1"></a><span class="co">#         # Vertical lines where E1 intersects 0</span></span>
<span id="cb68-75"><a href="#cb68-75" aria-hidden="true" tabindex="-1"></a><span class="co">#         ax.plot([i,i],[0,nbins],col,lw=0.5)</span></span>
<span id="cb68-76"><a href="#cb68-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-77"><a href="#cb68-77" aria-hidden="true" tabindex="-1"></a>plt.tight_layout()</span>
<span id="cb68-78"><a href="#cb68-78" aria-hidden="true" tabindex="-1"></a>f.canvas.draw()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div id="e1-matrix-500kb-full-arms-10mb-round_spermatid" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="03_compartments_files/figure-html/e1-matrix-500kb-full-arms-10mb-round_spermatid-output-1.png" class="img-fluid figure-img"></p>
<figcaption>E1 eigenvector values for all merged samples at 500kb resolution for round spermatid, as well as the interaction matrix. E1 was restricted to top: Full-chromosome, middle: Chromosome-arms, bottom: 10Mb window.</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="sliding-window-summed-e1-compartments" class="level4 page-columns page-full" data-number="6.1.2.6">
<h4 data-number="6.1.2.6" class="anchored" data-anchor-id="sliding-window-summed-e1-compartments"><span class="header-section-number">6.1.2.6</span> Sliding window summed E1 compartments</h4>
<p>To mimic the smoothing applied in the Wang et al.&nbsp;2019 paper, where they slide a 400kb window in 100kb steps on the obs/exp matrix, we will similarly slide a 400kb window in 100kb steps directly on the E1 compartments.</p>
<div id="cell-e1-fullmerged-all-500kb-running-avg" class="cell page-columns page-full">
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb69-3"><a href="#cb69-3" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> matplotlib.colors <span class="im">import</span> LogNorm</span>
<span id="cb69-4"><a href="#cb69-4" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> mpl_toolkits.axes_grid1 <span class="im">import</span> make_axes_locatable</span>
<span id="cb69-5"><a href="#cb69-5" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb69-6"><a href="#cb69-6" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> scipy.signal.windows <span class="im">import</span> triang</span>
<span id="cb69-7"><a href="#cb69-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-8"><a href="#cb69-8" aria-hidden="true" tabindex="-1"></a>resolution <span class="op">=</span> <span class="dv">500_000</span></span>
<span id="cb69-9"><a href="#cb69-9" aria-hidden="true" tabindex="-1"></a>window_size <span class="op">=</span> <span class="dv">2_500_000</span></span>
<span id="cb69-10"><a href="#cb69-10" aria-hidden="true" tabindex="-1"></a>step_size <span class="op">=</span> window_size <span class="op">//</span> resolution</span>
<span id="cb69-11"><a href="#cb69-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-12"><a href="#cb69-12" aria-hidden="true" tabindex="-1"></a>chrom_start <span class="op">=</span> e1_track[<span class="st">'start'</span>].values</span>
<span id="cb69-13"><a href="#cb69-13" aria-hidden="true" tabindex="-1"></a>chrom_end <span class="op">=</span> e1_track[<span class="st">'end'</span>].values<span class="op">-</span><span class="dv">1</span></span>
<span id="cb69-14"><a href="#cb69-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-15"><a href="#cb69-15" aria-hidden="true" tabindex="-1"></a>f, axs <span class="op">=</span> plt.subplots(<span class="dv">5</span>, <span class="dv">2</span>, figsize<span class="op">=</span>(<span class="dv">30</span>, <span class="dv">10</span>), sharex<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb69-16"><a href="#cb69-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-17"><a href="#cb69-17" aria-hidden="true" tabindex="-1"></a>axs[<span class="dv">0</span>, <span class="dv">0</span>].set_title(<span class="st">'Chromosome-arm E1 (arm restricted)'</span>)</span>
<span id="cb69-18"><a href="#cb69-18" aria-hidden="true" tabindex="-1"></a>axs[<span class="dv">0</span>, <span class="dv">1</span>].set_title(<span class="st">'Smoothed by summation'</span>)</span>
<span id="cb69-19"><a href="#cb69-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-20"><a href="#cb69-20" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i, (name, e1) <span class="kw">in</span> <span class="bu">enumerate</span>(e1_values.items()):</span>
<span id="cb69-21"><a href="#cb69-21" aria-hidden="true" tabindex="-1"></a>    <span class="co"># print(i, name, e1.size)</span></span>
<span id="cb69-22"><a href="#cb69-22" aria-hidden="true" tabindex="-1"></a>    smooth_e1 <span class="op">=</span> (<span class="kw">lambda</span> x: x.rolling(<span class="dv">5</span>, <span class="dv">1</span>, center<span class="op">=</span><span class="va">True</span>).<span class="bu">sum</span>())(pd.DataFrame(e1, columns<span class="op">=</span>[<span class="st">'value'</span>]))</span>
<span id="cb69-23"><a href="#cb69-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-24"><a href="#cb69-24" aria-hidden="true" tabindex="-1"></a>    ax0 <span class="op">=</span> axs[i, <span class="dv">0</span>]</span>
<span id="cb69-25"><a href="#cb69-25" aria-hidden="true" tabindex="-1"></a>    ax1 <span class="op">=</span> axs[i, <span class="dv">1</span>]</span>
<span id="cb69-26"><a href="#cb69-26" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb69-27"><a href="#cb69-27" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create stairs</span></span>
<span id="cb69-28"><a href="#cb69-28" aria-hidden="true" tabindex="-1"></a>    x <span class="op">=</span> np.zeros(<span class="dv">2</span><span class="op">*</span>chrom_start.size)</span>
<span id="cb69-29"><a href="#cb69-29" aria-hidden="true" tabindex="-1"></a>    y <span class="op">=</span> np.zeros(<span class="dv">2</span><span class="op">*</span>chrom_start.size)</span>
<span id="cb69-30"><a href="#cb69-30" aria-hidden="true" tabindex="-1"></a>    smooth_y <span class="op">=</span> np.zeros(<span class="dv">2</span><span class="op">*</span>chrom_start.size)</span>
<span id="cb69-31"><a href="#cb69-31" aria-hidden="true" tabindex="-1"></a>    x[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> chrom_start</span>
<span id="cb69-32"><a href="#cb69-32" aria-hidden="true" tabindex="-1"></a>    x[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> chrom_end</span>
<span id="cb69-33"><a href="#cb69-33" aria-hidden="true" tabindex="-1"></a>    y[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> e1</span>
<span id="cb69-34"><a href="#cb69-34" aria-hidden="true" tabindex="-1"></a>    y[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> e1</span>
<span id="cb69-35"><a href="#cb69-35" aria-hidden="true" tabindex="-1"></a>    smooth_y[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> smooth_e1[<span class="st">'value'</span>].values</span>
<span id="cb69-36"><a href="#cb69-36" aria-hidden="true" tabindex="-1"></a>    smooth_y[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> smooth_e1[<span class="st">'value'</span>].values</span>
<span id="cb69-37"><a href="#cb69-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-38"><a href="#cb69-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-39"><a href="#cb69-39" aria-hidden="true" tabindex="-1"></a>    ax0.fill_between(x, y, <span class="dv">0</span>, where<span class="op">=</span>(y <span class="op">&gt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:red'</span>)</span>
<span id="cb69-40"><a href="#cb69-40" aria-hidden="true" tabindex="-1"></a>    ax0.fill_between(x, y, <span class="dv">0</span>, where<span class="op">=</span>(y <span class="op">&lt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:blue'</span>)</span>
<span id="cb69-41"><a href="#cb69-41" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb69-42"><a href="#cb69-42" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Overlay the smoothed line (divided by 5 to make it a mean)</span></span>
<span id="cb69-43"><a href="#cb69-43" aria-hidden="true" tabindex="-1"></a>    ax0.plot(x, smooth_y<span class="op">/</span><span class="dv">5</span>, color<span class="op">=</span><span class="st">'C1'</span>)</span>
<span id="cb69-44"><a href="#cb69-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-45"><a href="#cb69-45" aria-hidden="true" tabindex="-1"></a>    ax1.fill_between(x, smooth_y, <span class="dv">0</span>, where<span class="op">=</span>(smooth_y <span class="op">&gt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:red'</span>)</span>
<span id="cb69-46"><a href="#cb69-46" aria-hidden="true" tabindex="-1"></a>    ax1.fill_between(x, smooth_y, <span class="dv">0</span>, where<span class="op">=</span>(smooth_y <span class="op">&lt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:blue'</span>)</span>
<span id="cb69-47"><a href="#cb69-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-48"><a href="#cb69-48" aria-hidden="true" tabindex="-1"></a>    ax0.set_ylabel(name)</span>
<span id="cb69-49"><a href="#cb69-49" aria-hidden="true" tabindex="-1"></a>    ylim <span class="op">=</span> <span class="fl">1.5</span></span>
<span id="cb69-50"><a href="#cb69-50" aria-hidden="true" tabindex="-1"></a>    <span class="co">#ax0.set_ylim(-ylim, ylim)</span></span>
<span id="cb69-51"><a href="#cb69-51" aria-hidden="true" tabindex="-1"></a>    <span class="co">#ax1.set_ylim(-ylim*4, ylim*4)</span></span>
<span id="cb69-52"><a href="#cb69-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-53"><a href="#cb69-53" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> ax <span class="kw">in</span> axs.flat:</span>
<span id="cb69-54"><a href="#cb69-54" aria-hidden="true" tabindex="-1"></a>    ax.spines[:].set_visible(<span class="va">False</span>)</span>
<span id="cb69-55"><a href="#cb69-55" aria-hidden="true" tabindex="-1"></a>    ax.set_xticks([])</span>
<span id="cb69-56"><a href="#cb69-56" aria-hidden="true" tabindex="-1"></a>    <span class="co">#ax.set_yticks([])</span></span>
<span id="cb69-57"><a href="#cb69-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-58"><a href="#cb69-58" aria-hidden="true" tabindex="-1"></a><span class="co"># plt.tight_layout()</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display page-columns page-full">
<div id="e1-fullmerged-all-500kb-running-avg" class="quarto-figure quarto-figure-center anchored page-columns page-full">
<figure class="figure page-columns page-full">
<p class="page-columns page-full"><img src="03_compartments_files/figure-html/e1-fullmerged-all-500kb-running-avg-output-1.png" class="img-fluid figure-img column-body-outset"></p>
<figcaption>E1 eigenvector values for all merged samples at 500kb resolution with rolling summation, with window size 2.5Mb, step size 500Kb: Each value is now the sum of the surrounding n=5 bins.</figcaption>
</figure>
</div>
</div>
</div>
</section>
</section>
</section>
<section id="kb-resolution-1" class="level2 page-columns page-full" data-number="6.2">
<h2 data-number="6.2" class="anchored" data-anchor-id="kb-resolution-1"><span class="header-section-number">6.2</span> 100kb resolution</h2>
<section id="all-5-full-merges-1" class="level3 page-columns page-full" data-number="6.2.1">
<h3 data-number="6.2.1" class="anchored" data-anchor-id="all-5-full-merges-1"><span class="header-section-number">6.2.1</span> All 5 full merges</h3>
<section id="load-coolers-1" class="level4" data-number="6.2.1.1">
<h4 data-number="6.2.1.1" class="anchored" data-anchor-id="load-coolers-1"><span class="header-section-number">6.2.1.1</span> Load coolers</h4>
<div id="cell-94" class="cell" data-execution_count="1">
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> glob</span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os.path <span class="im">as</span> op</span>
<span id="cb70-3"><a href="#cb70-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooler</span>
<span id="cb70-4"><a href="#cb70-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-5"><a href="#cb70-5" aria-hidden="true" tabindex="-1"></a>mcools <span class="op">=</span> glob.glob(<span class="st">"../steps/bwa/PE/cool/*/*.mcool"</span>)</span>
<span id="cb70-6"><a href="#cb70-6" aria-hidden="true" tabindex="-1"></a>res <span class="op">=</span> <span class="st">"::resolutions/100000"</span></span>
<span id="cb70-7"><a href="#cb70-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-8"><a href="#cb70-8" aria-hidden="true" tabindex="-1"></a>clrs <span class="op">=</span> {op.basename(op.dirname(mcool)): cooler.Cooler(mcool<span class="op">+</span>res) <span class="cf">for</span> mcool <span class="kw">in</span> mcools}</span>
<span id="cb70-9"><a href="#cb70-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-10"><a href="#cb70-10" aria-hidden="true" tabindex="-1"></a>chron_order <span class="op">=</span> [<span class="st">'fibroblast'</span>, <span class="st">'spermatogonia'</span>, <span class="st">'pachytene_spermatocyte'</span>, <span class="st">'round_spermatid'</span>, <span class="st">'sperm'</span>]</span>
<span id="cb70-11"><a href="#cb70-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-12"><a href="#cb70-12" aria-hidden="true" tabindex="-1"></a>clrs <span class="op">=</span> {key: clrs[key] <span class="cf">for</span> key <span class="kw">in</span> chron_order}</span>
<span id="cb70-13"><a href="#cb70-13" aria-hidden="true" tabindex="-1"></a>clrs</span>
<span id="cb70-14"><a href="#cb70-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-15"><a href="#cb70-15" aria-hidden="true" tabindex="-1"></a>names_abbr <span class="op">=</span> {<span class="st">'fibroblast'</span>: <span class="st">'Fibroblast'</span>, <span class="st">'spermatogonia'</span>: <span class="st">'Spermatogonia'</span>, <span class="st">'pachytene_spermatocyte'</span>: <span class="st">'Pachytene Spermatogonia'</span>, <span class="st">'round_spermatid'</span>: <span class="st">'Round Spermatid'</span>, <span class="st">'sperm'</span>: <span class="st">'Sperm'</span>}</span>
<span id="cb70-16"><a href="#cb70-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-17"><a href="#cb70-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate chromstart and chromend for each bin on chrX</span></span>
<span id="cb70-18"><a href="#cb70-18" aria-hidden="true" tabindex="-1"></a>chrX_size <span class="op">=</span> clrs[<span class="st">'fibroblast'</span>].chromsizes[<span class="st">'chrX'</span>]</span>
<span id="cb70-19"><a href="#cb70-19" aria-hidden="true" tabindex="-1"></a>chrom_start <span class="op">=</span> clrs[<span class="st">'fibroblast'</span>].bins().fetch(<span class="st">'chrX'</span>)[<span class="st">'start'</span>].values</span>
<span id="cb70-20"><a href="#cb70-20" aria-hidden="true" tabindex="-1"></a>chrom_end <span class="op">=</span> clrs[<span class="st">'fibroblast'</span>].bins().fetch(<span class="st">'chrX'</span>)[<span class="st">'end'</span>].values<span class="op">-</span><span class="dv">1</span></span>
<span id="cb70-21"><a href="#cb70-21" aria-hidden="true" tabindex="-1"></a>nbins <span class="op">=</span> <span class="bu">len</span>(clrs[<span class="st">'fibroblast'</span>].bins().fetch(<span class="st">'chrX'</span>))</span>
<span id="cb70-22"><a href="#cb70-22" aria-hidden="true" tabindex="-1"></a>binsize <span class="op">=</span> clrs[<span class="st">'fibroblast'</span>].binsize</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="calculate-gc-covariance-from-the-reference-genome-2" class="level4" data-number="6.2.1.2">
<h4 data-number="6.2.1.2" class="anchored" data-anchor-id="calculate-gc-covariance-from-the-reference-genome-2"><span class="header-section-number">6.2.1.2</span> Calculate gc covariance (from the reference genome)</h4>
<p>Do this with any of the clrs - it just needs the bins positions.</p>
<div id="cell-96" class="cell" data-execution_count="2">
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Try with only the gc_cov for chrX</span></span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-3"><a href="#cb71-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> bioframe</span>
<span id="cb71-4"><a href="#cb71-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb71-5"><a href="#cb71-5" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os.path <span class="im">as</span> op</span>
<span id="cb71-6"><a href="#cb71-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-7"><a href="#cb71-7" aria-hidden="true" tabindex="-1"></a>bins <span class="op">=</span> clrs[<span class="st">'fibroblast'</span>].bins().fetch(<span class="st">'chrX'</span>)[:]</span>
<span id="cb71-8"><a href="#cb71-8" aria-hidden="true" tabindex="-1"></a>out_name <span class="op">=</span> <span class="st">'../steps/rheMac10_gc_cov_X_100kb.tsv'</span></span>
<span id="cb71-9"><a href="#cb71-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-10"><a href="#cb71-10" aria-hidden="true" tabindex="-1"></a>rheMac10 <span class="op">=</span> bioframe.load_fasta(<span class="st">'../data/links/ucsc_ref/rheMac10.fa'</span>)</span>
<span id="cb71-11"><a href="#cb71-11" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> <span class="kw">not</span> op.exists(out_name):</span>
<span id="cb71-12"><a href="#cb71-12" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">'Calculate the fraction of GC basepairs for each bin'</span>)</span>
<span id="cb71-13"><a href="#cb71-13" aria-hidden="true" tabindex="-1"></a>    gc_cov <span class="op">=</span> bioframe.frac_gc(bins[[<span class="st">'chrom'</span>, <span class="st">'start'</span>, <span class="st">'end'</span>]], rheMac10)</span>
<span id="cb71-14"><a href="#cb71-14" aria-hidden="true" tabindex="-1"></a>    gc_cov.to_csv(out_name, index<span class="op">=</span><span class="va">False</span>,sep<span class="op">=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>)</span>
<span id="cb71-15"><a href="#cb71-15" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(gc_cov.info())</span>
<span id="cb71-16"><a href="#cb71-16" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>: </span>
<span id="cb71-17"><a href="#cb71-17" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"Already exists, read from file"</span>)</span>
<span id="cb71-18"><a href="#cb71-18" aria-hidden="true" tabindex="-1"></a>    gc_cov <span class="op">=</span> pd.read_csv(out_name, sep<span class="op">=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>)</span>
<span id="cb71-19"><a href="#cb71-19" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(gc_cov.info())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Already exists, read from file
&lt;class 'pandas.core.frame.DataFrame'&gt;
RangeIndex: 1534 entries, 0 to 1533
Data columns (total 4 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   chrom   1534 non-null   object 
 1   start   1534 non-null   int64  
 2   end     1534 non-null   int64  
 3   GC      1533 non-null   float64
dtypes: float64(1), int64(2), object(1)
memory usage: 48.1+ KB
None</code></pre>
</div>
</div>
</section>
<section id="calculate-the-e1-compartments-2" class="level4" data-number="6.2.1.3">
<h4 data-number="6.2.1.3" class="anchored" data-anchor-id="calculate-the-e1-compartments-2"><span class="header-section-number">6.2.1.3</span> Calculate the E1 compartments</h4>
<p>Loop: view_df, cis_eigs, e1_values</p>
<section id="plot-gc-covariance-1" class="level5" data-number="6.2.1.3.1">
<h5 data-number="6.2.1.3.1" class="anchored" data-anchor-id="plot-gc-covariance-1"><span class="header-section-number">6.2.1.3.1</span> Plot GC covariance</h5>
<div id="cell-99" class="cell" data-execution_count="3">
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb73-2"><a href="#cb73-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-3"><a href="#cb73-3" aria-hidden="true" tabindex="-1"></a>f, ax <span class="op">=</span> plt.subplots(figsize<span class="op">=</span>(<span class="dv">10</span>, <span class="dv">2</span>))</span>
<span id="cb73-4"><a href="#cb73-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-5"><a href="#cb73-5" aria-hidden="true" tabindex="-1"></a>ax.plot(gc_cov[<span class="st">'start'</span>],gc_cov[<span class="st">'GC'</span>])</span>
<span id="cb73-6"><a href="#cb73-6" aria-hidden="true" tabindex="-1"></a>plt.tight_layout()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="03_compartments_files/figure-html/cell-57-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="create-viewframes-full-arms-10mb-windows" class="level5" data-number="6.2.1.3.2">
<h5 data-number="6.2.1.3.2" class="anchored" data-anchor-id="create-viewframes-full-arms-10mb-windows"><span class="header-section-number">6.2.1.3.2</span> Create viewframes: full, arms, 10Mb windows</h5>
<div id="cell-101" class="cell" data-execution_count="4">
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-3"><a href="#cb74-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Fetch the chromsize of X from one of the coolers</span></span>
<span id="cb74-4"><a href="#cb74-4" aria-hidden="true" tabindex="-1"></a>chrX_size <span class="op">=</span> clrs[<span class="st">'fibroblast'</span>].chromsizes[<span class="st">'chrX'</span>]</span>
<span id="cb74-5"><a href="#cb74-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-6"><a href="#cb74-6" aria-hidden="true" tabindex="-1"></a>views <span class="op">=</span> {}</span>
<span id="cb74-7"><a href="#cb74-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-8"><a href="#cb74-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Make the full view frame</span></span>
<span id="cb74-9"><a href="#cb74-9" aria-hidden="true" tabindex="-1"></a>views[<span class="st">'full'</span>] <span class="op">=</span> pd.DataFrame({</span>
<span id="cb74-10"><a href="#cb74-10" aria-hidden="true" tabindex="-1"></a>    <span class="st">'chrom'</span>: <span class="st">'chrX'</span>,</span>
<span id="cb74-11"><a href="#cb74-11" aria-hidden="true" tabindex="-1"></a>    <span class="st">'start'</span>: <span class="dv">0</span>,</span>
<span id="cb74-12"><a href="#cb74-12" aria-hidden="true" tabindex="-1"></a>    <span class="st">'end'</span>: chrX_size,</span>
<span id="cb74-13"><a href="#cb74-13" aria-hidden="true" tabindex="-1"></a>    <span class="st">'name'</span>: <span class="st">'chrX'</span>}, index<span class="op">=</span>[<span class="dv">0</span>])</span>
<span id="cb74-14"><a href="#cb74-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-15"><a href="#cb74-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Divide into chromosome arms</span></span>
<span id="cb74-16"><a href="#cb74-16" aria-hidden="true" tabindex="-1"></a>views[<span class="st">'arms'</span>] <span class="op">=</span> pd.DataFrame({</span>
<span id="cb74-17"><a href="#cb74-17" aria-hidden="true" tabindex="-1"></a>    <span class="st">'chrom'</span>: <span class="st">'chrX'</span>,</span>
<span id="cb74-18"><a href="#cb74-18" aria-hidden="true" tabindex="-1"></a>    <span class="st">'start'</span>: [<span class="dv">0</span>, <span class="dv">59_000_001</span>],</span>
<span id="cb74-19"><a href="#cb74-19" aria-hidden="true" tabindex="-1"></a>    <span class="st">'end'</span>: [<span class="dv">59_000_000</span>, chrX_size],</span>
<span id="cb74-20"><a href="#cb74-20" aria-hidden="true" tabindex="-1"></a>    <span class="st">'name'</span>: [<span class="st">'X_short'</span>, <span class="st">'X_long'</span>]}, index<span class="op">=</span>[<span class="dv">0</span>,<span class="dv">1</span>])</span>
<span id="cb74-21"><a href="#cb74-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-22"><a href="#cb74-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate in 10Mb windows</span></span>
<span id="cb74-23"><a href="#cb74-23" aria-hidden="true" tabindex="-1"></a>window_size <span class="op">=</span> <span class="dv">10_000_000</span></span>
<span id="cb74-24"><a href="#cb74-24" aria-hidden="true" tabindex="-1"></a>start_positions <span class="op">=</span> <span class="bu">list</span>(<span class="bu">range</span>(<span class="dv">0</span>, chrX_size, window_size))</span>
<span id="cb74-25"><a href="#cb74-25" aria-hidden="true" tabindex="-1"></a>end_positions <span class="op">=</span> [<span class="bu">min</span>(start <span class="op">+</span> window_size, chrX_size) <span class="cf">for</span> start <span class="kw">in</span> start_positions]</span>
<span id="cb74-26"><a href="#cb74-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-27"><a href="#cb74-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the DataFrame</span></span>
<span id="cb74-28"><a href="#cb74-28" aria-hidden="true" tabindex="-1"></a>views[<span class="st">'10Mb'</span>] <span class="op">=</span> pd.DataFrame({</span>
<span id="cb74-29"><a href="#cb74-29" aria-hidden="true" tabindex="-1"></a>    <span class="st">'chrom'</span>: [<span class="st">'chrX'</span>] <span class="op">*</span> <span class="bu">len</span>(start_positions),</span>
<span id="cb74-30"><a href="#cb74-30" aria-hidden="true" tabindex="-1"></a>    <span class="st">'start'</span>: start_positions,</span>
<span id="cb74-31"><a href="#cb74-31" aria-hidden="true" tabindex="-1"></a>    <span class="st">'end'</span>: end_positions,</span>
<span id="cb74-32"><a href="#cb74-32" aria-hidden="true" tabindex="-1"></a>    <span class="st">'name'</span>: [<span class="ss">f'X_</span><span class="sc">{</span>i<span class="sc">}</span><span class="ss">'</span> <span class="cf">for</span> i <span class="kw">in</span> <span class="bu">range</span>(<span class="bu">len</span>(start_positions))]</span>
<span id="cb74-33"><a href="#cb74-33" aria-hidden="true" tabindex="-1"></a>})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="calulate-the-e1-compartments" class="level5" data-number="6.2.1.3.3">
<h5 data-number="6.2.1.3.3" class="anchored" data-anchor-id="calulate-the-e1-compartments"><span class="header-section-number">6.2.1.3.3</span> Calulate the E1 compartments</h5>
<div id="cell-103" class="cell" data-execution_count="5">
<div class="sourceCode cell-code" id="cb75"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooltools</span>
<span id="cb75-2"><a href="#cb75-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-3"><a href="#cb75-3" aria-hidden="true" tabindex="-1"></a>eigs <span class="op">=</span> {}</span>
<span id="cb75-4"><a href="#cb75-4" aria-hidden="true" tabindex="-1"></a>e1_values <span class="op">=</span> {}</span>
<span id="cb75-5"><a href="#cb75-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-6"><a href="#cb75-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> name, clr <span class="kw">in</span> clrs.items():</span>
<span id="cb75-7"><a href="#cb75-7" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> name <span class="kw">not</span> <span class="kw">in</span> eigs <span class="kw">or</span> name <span class="kw">not</span> <span class="kw">in</span> e1_values:</span>
<span id="cb75-8"><a href="#cb75-8" aria-hidden="true" tabindex="-1"></a>        eigs[name] <span class="op">=</span> {} </span>
<span id="cb75-9"><a href="#cb75-9" aria-hidden="true" tabindex="-1"></a>        e1_values[name] <span class="op">=</span> {}</span>
<span id="cb75-10"><a href="#cb75-10" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> view, view_df <span class="kw">in</span> views.items():</span>
<span id="cb75-11"><a href="#cb75-11" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Calculating eigenvectors for </span><span class="sc">{</span>name<span class="sc">}</span><span class="ss"> at </span><span class="sc">{</span>view<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb75-12"><a href="#cb75-12" aria-hidden="true" tabindex="-1"></a>        cis_eigs <span class="op">=</span> cooltools.eigs_cis(</span>
<span id="cb75-13"><a href="#cb75-13" aria-hidden="true" tabindex="-1"></a>                            clr,</span>
<span id="cb75-14"><a href="#cb75-14" aria-hidden="true" tabindex="-1"></a>                            gc_cov,</span>
<span id="cb75-15"><a href="#cb75-15" aria-hidden="true" tabindex="-1"></a>                            view_df<span class="op">=</span>view_df,</span>
<span id="cb75-16"><a href="#cb75-16" aria-hidden="true" tabindex="-1"></a>                            n_eigs<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb75-17"><a href="#cb75-17" aria-hidden="true" tabindex="-1"></a>        eigs[name][view] <span class="op">=</span> cis_eigs[<span class="dv">1</span>]</span>
<span id="cb75-18"><a href="#cb75-18" aria-hidden="true" tabindex="-1"></a>        e1_values[name][view] <span class="op">=</span> cis_eigs[<span class="dv">1</span>][<span class="st">'E1'</span>].values</span>
<span id="cb75-19"><a href="#cb75-19" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb75-20"><a href="#cb75-20" aria-hidden="true" tabindex="-1"></a></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for fibroblast at full</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for fibroblast at arms</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for fibroblast at 10Mb</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for spermatogonia at full</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for spermatogonia at arms</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for spermatogonia at 10Mb</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for pachytene_spermatocyte at full</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for pachytene_spermatocyte at arms</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for pachytene_spermatocyte at 10Mb</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for round_spermatid at full</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for round_spermatid at arms</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for round_spermatid at 10Mb</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for sperm at full</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for sperm at arms</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Calculating eigenvectors for sperm at 10Mb</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/home/sojern/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/common.py:489: UserWarning: less than 50% of valid bins have been assigned a value
  warnings.warn("less than 50% of valid bins have been assigned a value")</code></pre>
</div>
</div>
</section>
</section>
<section id="plot-the-e1-compartments-1" class="level4 page-columns page-full" data-number="6.2.1.4">
<h4 data-number="6.2.1.4" class="anchored" data-anchor-id="plot-the-e1-compartments-1"><span class="header-section-number">6.2.1.4</span> Plot the E1 compartments</h4>
<div id="cell-e1-fullmerged-all-100kb-5smoothed" class="cell page-columns page-full" data-execution_count="6">
<div class="sourceCode cell-code" id="cb106"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb106-2"><a href="#cb106-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb106-3"><a href="#cb106-3" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> matplotlib.colors <span class="im">import</span> LogNorm</span>
<span id="cb106-4"><a href="#cb106-4" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> mpl_toolkits.axes_grid1 <span class="im">import</span> make_axes_locatable</span>
<span id="cb106-5"><a href="#cb106-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-6"><a href="#cb106-6" aria-hidden="true" tabindex="-1"></a>f, axs <span class="op">=</span> plt.subplots(<span class="dv">5</span>, <span class="dv">3</span>, figsize<span class="op">=</span>(<span class="dv">18</span>, <span class="dv">8</span>), sharex<span class="op">=</span><span class="va">True</span>, sharey<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb106-7"><a href="#cb106-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-8"><a href="#cb106-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Loop through eigs and e1_values to plot the E1 values</span></span>
<span id="cb106-9"><a href="#cb106-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i, (name, e1_dict) <span class="kw">in</span> <span class="bu">enumerate</span>(e1_values.items()):</span>
<span id="cb106-10"><a href="#cb106-10" aria-hidden="true" tabindex="-1"></a>    axs[i,<span class="dv">0</span>].set_ylabel(names_abbr[name])</span>
<span id="cb106-11"><a href="#cb106-11" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> j, (view, e1) <span class="kw">in</span> <span class="bu">enumerate</span>(e1_dict.items()):</span>
<span id="cb106-12"><a href="#cb106-12" aria-hidden="true" tabindex="-1"></a>        ax <span class="op">=</span> axs[i, j]</span>
<span id="cb106-13"><a href="#cb106-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-14"><a href="#cb106-14" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> i<span class="op">==</span><span class="dv">0</span>:</span>
<span id="cb106-15"><a href="#cb106-15" aria-hidden="true" tabindex="-1"></a>            ax.set_title(view)</span>
<span id="cb106-16"><a href="#cb106-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-17"><a href="#cb106-17" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Create stairs</span></span>
<span id="cb106-18"><a href="#cb106-18" aria-hidden="true" tabindex="-1"></a>        x <span class="op">=</span> np.zeros(<span class="dv">2</span><span class="op">*</span>chrom_start.size)</span>
<span id="cb106-19"><a href="#cb106-19" aria-hidden="true" tabindex="-1"></a>        y <span class="op">=</span> np.zeros(<span class="dv">2</span><span class="op">*</span>chrom_start.size)</span>
<span id="cb106-20"><a href="#cb106-20" aria-hidden="true" tabindex="-1"></a>        smooth_y <span class="op">=</span> np.zeros(<span class="dv">2</span><span class="op">*</span>chrom_start.size)</span>
<span id="cb106-21"><a href="#cb106-21" aria-hidden="true" tabindex="-1"></a>        x[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> chrom_start</span>
<span id="cb106-22"><a href="#cb106-22" aria-hidden="true" tabindex="-1"></a>        x[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> chrom_end</span>
<span id="cb106-23"><a href="#cb106-23" aria-hidden="true" tabindex="-1"></a>        y[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> e1</span>
<span id="cb106-24"><a href="#cb106-24" aria-hidden="true" tabindex="-1"></a>        y[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> e1</span>
<span id="cb106-25"><a href="#cb106-25" aria-hidden="true" tabindex="-1"></a>        smooth_e1 <span class="op">=</span> (<span class="kw">lambda</span> x: x.rolling(<span class="dv">5</span>, <span class="dv">1</span>, center<span class="op">=</span><span class="va">True</span>).<span class="bu">sum</span>())(pd.DataFrame(e1, columns<span class="op">=</span>[<span class="st">'value'</span>]))[<span class="st">'value'</span>].values</span>
<span id="cb106-26"><a href="#cb106-26" aria-hidden="true" tabindex="-1"></a>        smooth_y[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> smooth_e1</span>
<span id="cb106-27"><a href="#cb106-27" aria-hidden="true" tabindex="-1"></a>        smooth_y[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> smooth_e1</span>
<span id="cb106-28"><a href="#cb106-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-29"><a href="#cb106-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-30"><a href="#cb106-30" aria-hidden="true" tabindex="-1"></a>        <span class="co">#ax.fill_between(x, y, 0, where=(y &gt; 0), color='tab:red')</span></span>
<span id="cb106-31"><a href="#cb106-31" aria-hidden="true" tabindex="-1"></a>        <span class="co">#ax.fill_between(x, y, 0, where=(y &lt; 0), color='tab:blue')</span></span>
<span id="cb106-32"><a href="#cb106-32" aria-hidden="true" tabindex="-1"></a>        <span class="co">#ax.plot(x, smooth_y/5, color='C1')</span></span>
<span id="cb106-33"><a href="#cb106-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-34"><a href="#cb106-34" aria-hidden="true" tabindex="-1"></a>        ax.fill_between(x, smooth_y, <span class="dv">0</span>, where<span class="op">=</span>(smooth_y <span class="op">&gt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:red'</span>)</span>
<span id="cb106-35"><a href="#cb106-35" aria-hidden="true" tabindex="-1"></a>        ax.fill_between(x, smooth_y, <span class="dv">0</span>, where<span class="op">=</span>(smooth_y <span class="op">&lt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:blue'</span>)</span>
<span id="cb106-36"><a href="#cb106-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-37"><a href="#cb106-37" aria-hidden="true" tabindex="-1"></a><span class="co"># Set y-limits for all subplots</span></span>
<span id="cb106-38"><a href="#cb106-38" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> ax <span class="kw">in</span> axs.flat:</span>
<span id="cb106-39"><a href="#cb106-39" aria-hidden="true" tabindex="-1"></a>    ax.set_ylim(<span class="op">-</span><span class="dv">4</span>, <span class="dv">4</span>)</span>
<span id="cb106-40"><a href="#cb106-40" aria-hidden="true" tabindex="-1"></a>    ax.spines[:].set_visible(<span class="va">False</span>)</span>
<span id="cb106-41"><a href="#cb106-41" aria-hidden="true" tabindex="-1"></a>    ax.set_xticks([])</span>
<span id="cb106-42"><a href="#cb106-42" aria-hidden="true" tabindex="-1"></a>    ax.set_yticks([])</span>
<span id="cb106-43"><a href="#cb106-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-44"><a href="#cb106-44" aria-hidden="true" tabindex="-1"></a>plt.tight_layout()</span>
<span id="cb106-45"><a href="#cb106-45" aria-hidden="true" tabindex="-1"></a>plt.savefig(<span class="st">'../steps/e1_plot_full_100kb_smoothed.svg'</span>, bbox_inches<span class="op">=</span><span class="st">'tight'</span>)</span>
<span id="cb106-46"><a href="#cb106-46" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb106-47"><a href="#cb106-47" aria-hidden="true" tabindex="-1"></a>    </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display page-columns page-full">
<div id="e1-fullmerged-all-100kb-5smoothed" class="quarto-figure quarto-figure-center anchored page-columns page-full">
<figure class="figure page-columns page-full">
<p class="page-columns page-full"><img src="03_compartments_files/figure-html/e1-fullmerged-all-100kb-5smoothed-output-1.png" class="img-fluid figure-img column-page-inset"></p>
<figcaption>E1 eigenvector values for all merged samples at 100kb resolution. Left: Chromosomes (not restricted) Middle: Chromosome-arm restricted E1. Right: 10Mb window restricted E1.Values are smoothed with a sliding window of 5 bins, step size 1 bin.</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="plot-matrices-with-compartments" class="level4" data-number="6.2.1.5">
<h4 data-number="6.2.1.5" class="anchored" data-anchor-id="plot-matrices-with-compartments"><span class="header-section-number">6.2.1.5</span> Plot matrices with compartments</h4>
<div id="cell-e1-matrix-100kb-full-arms-10mb-all-5smoothed" class="cell">
<div class="sourceCode cell-code" id="cb107"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb107-1"><a href="#cb107-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooltools.lib.plotting</span>
<span id="cb107-2"><a href="#cb107-2" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> matplotlib.colors <span class="im">import</span> LogNorm</span>
<span id="cb107-3"><a href="#cb107-3" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> mpl_toolkits.axes_grid1 <span class="im">import</span> make_axes_locatable</span>
<span id="cb107-4"><a href="#cb107-4" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> cooltools.lib.numutils <span class="im">import</span> adaptive_coarsegrain, interp_nan</span>
<span id="cb107-5"><a href="#cb107-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-6"><a href="#cb107-6" aria-hidden="true" tabindex="-1"></a>f, axs <span class="op">=</span> plt.subplots(<span class="dv">1</span>,<span class="dv">5</span>,</span>
<span id="cb107-7"><a href="#cb107-7" aria-hidden="true" tabindex="-1"></a>    figsize<span class="op">=</span>(<span class="dv">25</span>,<span class="dv">10</span>)</span>
<span id="cb107-8"><a href="#cb107-8" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb107-9"><a href="#cb107-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-10"><a href="#cb107-10" aria-hidden="true" tabindex="-1"></a>norm <span class="op">=</span> LogNorm(vmax<span class="op">=</span><span class="fl">0.1</span>)</span>
<span id="cb107-11"><a href="#cb107-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-12"><a href="#cb107-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Loop through the clrs and its matrix: plot the matrix on its axis</span></span>
<span id="cb107-13"><a href="#cb107-13" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i, (name, e1_dict) <span class="kw">in</span> <span class="bu">enumerate</span>(e1_values.items()):</span>
<span id="cb107-14"><a href="#cb107-14" aria-hidden="true" tabindex="-1"></a>    ax <span class="op">=</span> axs[i]</span>
<span id="cb107-15"><a href="#cb107-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-16"><a href="#cb107-16" aria-hidden="true" tabindex="-1"></a>    im <span class="op">=</span> ax.matshow(</span>
<span id="cb107-17"><a href="#cb107-17" aria-hidden="true" tabindex="-1"></a>        clr.matrix().fetch(<span class="st">'chrX'</span>),</span>
<span id="cb107-18"><a href="#cb107-18" aria-hidden="true" tabindex="-1"></a>        norm<span class="op">=</span>norm,</span>
<span id="cb107-19"><a href="#cb107-19" aria-hidden="true" tabindex="-1"></a>        cmap<span class="op">=</span><span class="st">'fall'</span>,</span>
<span id="cb107-20"><a href="#cb107-20" aria-hidden="true" tabindex="-1"></a>    )<span class="op">;</span></span>
<span id="cb107-21"><a href="#cb107-21" aria-hidden="true" tabindex="-1"></a>    ax.set_xlim(<span class="dv">0</span>, nbins)</span>
<span id="cb107-22"><a href="#cb107-22" aria-hidden="true" tabindex="-1"></a>    ax.set_ylim(nbins, <span class="dv">0</span>)</span>
<span id="cb107-23"><a href="#cb107-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-24"><a href="#cb107-24" aria-hidden="true" tabindex="-1"></a>    divider <span class="op">=</span> make_axes_locatable(ax)</span>
<span id="cb107-25"><a href="#cb107-25" aria-hidden="true" tabindex="-1"></a>    cax <span class="op">=</span> divider.append_axes(<span class="st">"right"</span>, size<span class="op">=</span><span class="st">"2%"</span>, pad<span class="op">=</span><span class="fl">0.05</span>)</span>
<span id="cb107-26"><a href="#cb107-26" aria-hidden="true" tabindex="-1"></a>    plt.colorbar(im, cax<span class="op">=</span>cax, label<span class="op">=</span><span class="st">'corrected frequencies'</span>)<span class="op">;</span></span>
<span id="cb107-27"><a href="#cb107-27" aria-hidden="true" tabindex="-1"></a>    ax.set_ylabel(<span class="st">'chrX:100kb bins, #bin'</span>)</span>
<span id="cb107-28"><a href="#cb107-28" aria-hidden="true" tabindex="-1"></a>    ax.xaxis.set_visible(<span class="va">False</span>)</span>
<span id="cb107-29"><a href="#cb107-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-30"><a href="#cb107-30" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> j, (view, e1) <span class="kw">in</span> <span class="bu">enumerate</span>(e1_dict.items()):</span>
<span id="cb107-31"><a href="#cb107-31" aria-hidden="true" tabindex="-1"></a>        ax1 <span class="op">=</span> divider.append_axes(<span class="st">"top"</span>, size<span class="op">=</span><span class="st">"10%"</span>, pad<span class="op">=</span><span class="fl">0.1</span>, sharex <span class="op">=</span> ax)</span>
<span id="cb107-32"><a href="#cb107-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-33"><a href="#cb107-33" aria-hidden="true" tabindex="-1"></a>        <span class="co"># ax1.plot(e1, label='E1')</span></span>
<span id="cb107-34"><a href="#cb107-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-35"><a href="#cb107-35" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Create stairs</span></span>
<span id="cb107-36"><a href="#cb107-36" aria-hidden="true" tabindex="-1"></a>        x <span class="op">=</span> np.zeros(<span class="dv">2</span><span class="op">*</span>chrom_start.size)</span>
<span id="cb107-37"><a href="#cb107-37" aria-hidden="true" tabindex="-1"></a>        y <span class="op">=</span> np.zeros(<span class="dv">2</span><span class="op">*</span>chrom_start.size)</span>
<span id="cb107-38"><a href="#cb107-38" aria-hidden="true" tabindex="-1"></a>        <span class="co">#smooth_y = np.zeros(2*chrom_start.size)</span></span>
<span id="cb107-39"><a href="#cb107-39" aria-hidden="true" tabindex="-1"></a>        x[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> chrom_start<span class="op">/</span>binsize</span>
<span id="cb107-40"><a href="#cb107-40" aria-hidden="true" tabindex="-1"></a>        x[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> chrom_end<span class="op">/</span>binsize</span>
<span id="cb107-41"><a href="#cb107-41" aria-hidden="true" tabindex="-1"></a>        y[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> e1</span>
<span id="cb107-42"><a href="#cb107-42" aria-hidden="true" tabindex="-1"></a>        y[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> e1</span>
<span id="cb107-43"><a href="#cb107-43" aria-hidden="true" tabindex="-1"></a>        <span class="co"># smooth_e1 = (lambda x: x.rolling(5, 1, center=True).sum())(pd.DataFrame(e1, columns=['value']))['value'].values</span></span>
<span id="cb107-44"><a href="#cb107-44" aria-hidden="true" tabindex="-1"></a>        <span class="co"># smooth_y[0::2] = smooth_e1</span></span>
<span id="cb107-45"><a href="#cb107-45" aria-hidden="true" tabindex="-1"></a>        <span class="co"># smooth_y[1::2] = smooth_e1</span></span>
<span id="cb107-46"><a href="#cb107-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-47"><a href="#cb107-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-48"><a href="#cb107-48" aria-hidden="true" tabindex="-1"></a>        ax1.fill_between(x, y, <span class="dv">0</span>, where<span class="op">=</span>(y <span class="op">&gt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:red'</span>)</span>
<span id="cb107-49"><a href="#cb107-49" aria-hidden="true" tabindex="-1"></a>        ax1.fill_between(x, y, <span class="dv">0</span>, where<span class="op">=</span>(y <span class="op">&lt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:blue'</span>)</span>
<span id="cb107-50"><a href="#cb107-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-51"><a href="#cb107-51" aria-hidden="true" tabindex="-1"></a>        ax1.set_ylabel(view, rotation<span class="op">=-</span><span class="dv">70</span>)</span>
<span id="cb107-52"><a href="#cb107-52" aria-hidden="true" tabindex="-1"></a>        ax1.set_ylim(<span class="op">-</span><span class="fl">0.8</span>, <span class="fl">0.8</span>)</span>
<span id="cb107-53"><a href="#cb107-53" aria-hidden="true" tabindex="-1"></a>        ax1.set_xticks([])</span>
<span id="cb107-54"><a href="#cb107-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-55"><a href="#cb107-55" aria-hidden="true" tabindex="-1"></a>    ax1.set_title(name)</span>
<span id="cb107-56"><a href="#cb107-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-57"><a href="#cb107-57" aria-hidden="true" tabindex="-1"></a><span class="co">#    # Plot the sign changes on the matrix </span></span>
<span id="cb107-58"><a href="#cb107-58" aria-hidden="true" tabindex="-1"></a><span class="co">#     col = colors[i]</span></span>
<span id="cb107-59"><a href="#cb107-59" aria-hidden="true" tabindex="-1"></a><span class="co">#     for i in np.where(np.diff( (pd.Series(e1)&gt;0).astype(int)))[0]:</span></span>
<span id="cb107-60"><a href="#cb107-60" aria-hidden="true" tabindex="-1"></a><span class="co">#         # Horisontal lines where E1 intersects 0</span></span>
<span id="cb107-61"><a href="#cb107-61" aria-hidden="true" tabindex="-1"></a><span class="co">#         ax.plot([0,nbins],[i,i],col,lw=0.5)</span></span>
<span id="cb107-62"><a href="#cb107-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-63"><a href="#cb107-63" aria-hidden="true" tabindex="-1"></a><span class="co">#         # Vertical lines where E1 intersects 0</span></span>
<span id="cb107-64"><a href="#cb107-64" aria-hidden="true" tabindex="-1"></a><span class="co">#         ax.plot([i,i],[0,nbins],col,lw=0.5)</span></span>
<span id="cb107-65"><a href="#cb107-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-66"><a href="#cb107-66" aria-hidden="true" tabindex="-1"></a><span class="co"># Now show the plot</span></span>
<span id="cb107-67"><a href="#cb107-67" aria-hidden="true" tabindex="-1"></a>f.set_layout_engine(<span class="st">'compressed'</span>)</span>
<span id="cb107-68"><a href="#cb107-68" aria-hidden="true" tabindex="-1"></a>f.canvas.draw()</span>
<span id="cb107-69"><a href="#cb107-69" aria-hidden="true" tabindex="-1"></a>plt.show()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div id="e1-matrix-100kb-full-arms-10mb-all-5smoothed" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="03_compartments_files/figure-html/e1-matrix-100kb-full-arms-10mb-all-5smoothed-output-1.png" class="img-fluid figure-img"></p>
<figcaption>E1 eigenvector values for all merged samples at 500kb resolution for round spermatid, as well as the interaction matrix. E1 was restricted to top: Full-chromosome, middle: Chromosome-arms, bottom: 10Mb window.</figcaption>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="smooth-the-observedexpected-matrix" class="level3" data-number="6.2.2">
<h3 data-number="6.2.2" class="anchored" data-anchor-id="smooth-the-observedexpected-matrix"><span class="header-section-number">6.2.2</span> Smooth the Observed/Expected matrix</h3>
<p>Wang et al.&nbsp;2019 used a 400Kb window with a step size of 100Kb to smooth the O/E matrix before calculating the E1. I have tried various different methods to smooth the interaction matrix, but I can’t get dimensions of the <code>pixels</code> to match with the <code>bins</code>. Also, <code>cooltools.eigs_cis</code> need a full <code>cooler</code> to fetch data from, and I can’t get it to work without a <code>pixels</code> object.</p>
<p>Thus, I will try smoothing the E1 values instead, as I can’t really rationalize what biological difference it would make. I think I need help figuring that out.</p>
<div id="cell-110" class="cell">
<div class="sourceCode cell-code" id="cb108"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb108-1"><a href="#cb108-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> glob</span>
<span id="cb108-2"><a href="#cb108-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os.path <span class="im">as</span> op</span>
<span id="cb108-3"><a href="#cb108-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooler</span>
<span id="cb108-4"><a href="#cb108-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb108-5"><a href="#cb108-5" aria-hidden="true" tabindex="-1"></a>mcools <span class="op">=</span> glob.glob(<span class="st">"../steps/bwa/PE/cool/*/*.mcool"</span>)</span>
<span id="cb108-6"><a href="#cb108-6" aria-hidden="true" tabindex="-1"></a>res <span class="op">=</span> <span class="st">"::resolutions/100000"</span></span>
<span id="cb108-7"><a href="#cb108-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb108-8"><a href="#cb108-8" aria-hidden="true" tabindex="-1"></a>clrs100kb <span class="op">=</span> {op.basename(op.dirname(mcool)): cooler.Cooler(mcool<span class="op">+</span>res) <span class="cf">for</span> mcool <span class="kw">in</span> mcools}</span>
<span id="cb108-9"><a href="#cb108-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb108-10"><a href="#cb108-10" aria-hidden="true" tabindex="-1"></a>chron_order <span class="op">=</span> [<span class="st">'fibroblast'</span>, <span class="st">'spermatogonia'</span>, <span class="st">'pachytene_spermatocyte'</span>, <span class="st">'round_spermatid'</span>, <span class="st">'sperm'</span>]</span>
<span id="cb108-11"><a href="#cb108-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb108-12"><a href="#cb108-12" aria-hidden="true" tabindex="-1"></a>clrs100kb <span class="op">=</span> {key: clrs100kb[key] <span class="cf">for</span> key <span class="kw">in</span> chron_order}</span>
<span id="cb108-13"><a href="#cb108-13" aria-hidden="true" tabindex="-1"></a>clrs100kb</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<pre><code>{'fibroblast': &lt;Cooler "fibroblast.mcool::/resolutions/100000"&gt;,
 'spermatogonia': &lt;Cooler "spermatogonia.mcool::/resolutions/100000"&gt;,
 'pachytene_spermatocyte': &lt;Cooler "pachytene_spermatocyte.mcool::/resolutions/100000"&gt;,
 'round_spermatid': &lt;Cooler "round_spermatid.mcool::/resolutions/100000"&gt;,
 'sperm': &lt;Cooler "sperm.mcool::/resolutions/100000"&gt;}</code></pre>
</div>
</div>
<section id="calculate-gc-covariance-from-the-reference-genome-3" class="level4" data-number="6.2.2.1">
<h4 data-number="6.2.2.1" class="anchored" data-anchor-id="calculate-gc-covariance-from-the-reference-genome-3"><span class="header-section-number">6.2.2.1</span> Calculate gc covariance (from the reference genome)</h4>
<p>Do this with any of the clrs - it just needs the bins positions.</p>
<div id="cell-112" class="cell">
<div class="sourceCode cell-code" id="cb110"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb110-1"><a href="#cb110-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> bioframe</span>
<span id="cb110-2"><a href="#cb110-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb110-3"><a href="#cb110-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os.path <span class="im">as</span> op</span>
<span id="cb110-4"><a href="#cb110-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-5"><a href="#cb110-5" aria-hidden="true" tabindex="-1"></a>bins <span class="op">=</span> clrs100kb[<span class="st">'fibroblast'</span>].bins().fetch(<span class="st">'chrX'</span>)[:]</span>
<span id="cb110-6"><a href="#cb110-6" aria-hidden="true" tabindex="-1"></a>out_name <span class="op">=</span> <span class="st">'rheMac10_gc_cov_X_100kb.tsv'</span></span>
<span id="cb110-7"><a href="#cb110-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-8"><a href="#cb110-8" aria-hidden="true" tabindex="-1"></a>rheMac10 <span class="op">=</span> bioframe.load_fasta(<span class="st">'../data/links/ucsc_ref/rheMac10.fa'</span>)</span>
<span id="cb110-9"><a href="#cb110-9" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> <span class="kw">not</span> op.exists(out_name):</span>
<span id="cb110-10"><a href="#cb110-10" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">'Calculate the fraction of GC basepairs for each bin'</span>)</span>
<span id="cb110-11"><a href="#cb110-11" aria-hidden="true" tabindex="-1"></a>    gc_cov <span class="op">=</span> bioframe.frac_gc(bins[[<span class="st">'chrom'</span>, <span class="st">'start'</span>, <span class="st">'end'</span>]], rheMac10)</span>
<span id="cb110-12"><a href="#cb110-12" aria-hidden="true" tabindex="-1"></a>    gc_cov.to_csv(out_name, index<span class="op">=</span><span class="va">False</span>,sep<span class="op">=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>)</span>
<span id="cb110-13"><a href="#cb110-13" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(gc_cov.info())</span>
<span id="cb110-14"><a href="#cb110-14" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>: </span>
<span id="cb110-15"><a href="#cb110-15" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"Already exists, read from file"</span>)</span>
<span id="cb110-16"><a href="#cb110-16" aria-hidden="true" tabindex="-1"></a>    gc_cov <span class="op">=</span> pd.read_csv(out_name, sep<span class="op">=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>)</span>
<span id="cb110-17"><a href="#cb110-17" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(gc_cov.info())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Already exists, read from file
&lt;class 'pandas.core.frame.DataFrame'&gt;
RangeIndex: 1534 entries, 0 to 1533
Data columns (total 4 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   chrom   1534 non-null   object 
 1   start   1534 non-null   int64  
 2   end     1534 non-null   int64  
 3   GC      1533 non-null   float64
dtypes: float64(1), int64(2), object(1)
memory usage: 48.1+ KB
None</code></pre>
</div>
</div>
<div id="cell-113" class="cell">
<div class="sourceCode cell-code" id="cb112"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb112-1"><a href="#cb112-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooler</span>
<span id="cb112-2"><a href="#cb112-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb112-3"><a href="#cb112-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb112-4"><a href="#cb112-4" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> smooth_matrix(matrix, window_size<span class="op">=</span><span class="dv">4</span>, step_size<span class="op">=</span><span class="dv">1</span>):</span>
<span id="cb112-5"><a href="#cb112-5" aria-hidden="true" tabindex="-1"></a>    n <span class="op">=</span> matrix.shape[<span class="dv">0</span>]</span>
<span id="cb112-6"><a href="#cb112-6" aria-hidden="true" tabindex="-1"></a>    smoothed_matrix <span class="op">=</span> np.zeros_like(matrix)</span>
<span id="cb112-7"><a href="#cb112-7" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb112-8"><a href="#cb112-8" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> i <span class="kw">in</span> <span class="bu">range</span>(<span class="dv">0</span>, n <span class="op">-</span> window_size <span class="op">+</span> <span class="dv">1</span>, step_size):</span>
<span id="cb112-9"><a href="#cb112-9" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> j <span class="kw">in</span> <span class="bu">range</span>(<span class="dv">0</span>, n <span class="op">-</span> window_size <span class="op">+</span> <span class="dv">1</span>, step_size):</span>
<span id="cb112-10"><a href="#cb112-10" aria-hidden="true" tabindex="-1"></a>            window <span class="op">=</span> matrix[i:i<span class="op">+</span>window_size, j:j<span class="op">+</span>window_size]</span>
<span id="cb112-11"><a href="#cb112-11" aria-hidden="true" tabindex="-1"></a>            smoothed_matrix[i:i<span class="op">+</span>window_size, j:j<span class="op">+</span>window_size] <span class="op">=</span> np.<span class="bu">sum</span>(window)</span>
<span id="cb112-12"><a href="#cb112-12" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb112-13"><a href="#cb112-13" aria-hidden="true" tabindex="-1"></a>    <span class="cf">return</span> smoothed_matrix</span>
<span id="cb112-14"><a href="#cb112-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb112-15"><a href="#cb112-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Load the Hi-C interaction matrix from a multi-resolution .mcool file</span></span>
<span id="cb112-16"><a href="#cb112-16" aria-hidden="true" tabindex="-1"></a>cooler_file <span class="op">=</span> <span class="st">'../steps/bwa/PE/cool/fibroblast/fibroblast.mcool'</span></span>
<span id="cb112-17"><a href="#cb112-17" aria-hidden="true" tabindex="-1"></a>clr <span class="op">=</span> cooler.Cooler(cooler_file <span class="op">+</span> <span class="st">'::resolutions/100000'</span>)</span>
<span id="cb112-18"><a href="#cb112-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb112-19"><a href="#cb112-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract the matrix</span></span>
<span id="cb112-20"><a href="#cb112-20" aria-hidden="true" tabindex="-1"></a>matrix <span class="op">=</span> clr.matrix(balance<span class="op">=</span><span class="va">True</span>).fetch(<span class="st">'chrX'</span>)[:]</span>
<span id="cb112-21"><a href="#cb112-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb112-22"><a href="#cb112-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Smooth the matrix</span></span>
<span id="cb112-23"><a href="#cb112-23" aria-hidden="true" tabindex="-1"></a>smoothed_matrix <span class="op">=</span> smooth_matrix(matrix, window_size<span class="op">=</span><span class="dv">4</span>, step_size<span class="op">=</span><span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div id="cell-114" class="cell">
<div class="sourceCode cell-code" id="cb113"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb113-1"><a href="#cb113-1" aria-hidden="true" tabindex="-1"></a><span class="co">#clr.bins().fetch('chrX')[:]</span></span>
<span id="cb113-2"><a href="#cb113-2" aria-hidden="true" tabindex="-1"></a>clr.pixels().fetch(<span class="st">'chrX'</span>)[:]</span>
<span id="cb113-3"><a href="#cb113-3" aria-hidden="true" tabindex="-1"></a><span class="co">#print(matrix.shape)</span></span>
<span id="cb113-4"><a href="#cb113-4" aria-hidden="true" tabindex="-1"></a><span class="co">#print(smoothed_matrix.shape)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">bin1_id</th>
<th data-quarto-table-cell-role="th">bin2_id</th>
<th data-quarto-table-cell-role="th">count</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">145469195</td>
<td>26898</td>
<td>26898</td>
<td>2013</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">145469196</td>
<td>26898</td>
<td>26899</td>
<td>171</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">145469197</td>
<td>26898</td>
<td>26901</td>
<td>75</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">145469198</td>
<td>26898</td>
<td>26902</td>
<td>100</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">145469199</td>
<td>26898</td>
<td>26903</td>
<td>187</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">146369705</td>
<td>28431</td>
<td>28532</td>
<td>2</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">146369706</td>
<td>28431</td>
<td>28533</td>
<td>1</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">146369707</td>
<td>28431</td>
<td>28534</td>
<td>2</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">146369708</td>
<td>28431</td>
<td>28539</td>
<td>1</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">146369709</td>
<td>28431</td>
<td>28544</td>
<td>1</td>
</tr>
</tbody>
</table>

<p>900515 rows × 3 columns</p>
</div>
</div>
</div>
<div id="cell-115" class="cell">
<div class="sourceCode cell-code" id="cb114"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb114-1"><a href="#cb114-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb114-2"><a href="#cb114-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooler</span>
<span id="cb114-3"><a href="#cb114-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb114-4"><a href="#cb114-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-5"><a href="#cb114-5" aria-hidden="true" tabindex="-1"></a>pixels <span class="op">=</span> pd.DataFrame(smoothed_matrix).stack().rename_axis([<span class="st">'bin1_id'</span>, <span class="st">'bin2_id'</span>]).reset_index(name<span class="op">=</span><span class="st">'count'</span>)</span>
<span id="cb114-6"><a href="#cb114-6" aria-hidden="true" tabindex="-1"></a>pixels <span class="op">=</span> pixels[pixels[<span class="st">'bin1_id'</span>] <span class="op">&lt;=</span> pixels[<span class="st">'bin2_id'</span>]]</span>
<span id="cb114-7"><a href="#cb114-7" aria-hidden="true" tabindex="-1"></a>pixels.sort_values([<span class="st">'bin1_id'</span>, <span class="st">'bin2_id'</span>])[[<span class="st">'bin1_id'</span>, <span class="st">'bin2_id'</span>, <span class="st">'count'</span>]].reset_index(drop<span class="op">=</span><span class="va">True</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">bin1_id</th>
<th data-quarto-table-cell-role="th">bin2_id</th>
<th data-quarto-table-cell-role="th">count</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>5</td>
<td>5</td>
<td>2.555538</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>5</td>
<td>6</td>
<td>2.141584</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>5</td>
<td>7</td>
<td>1.550878</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3</td>
<td>5</td>
<td>8</td>
<td>0.864845</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">4</td>
<td>5</td>
<td>9</td>
<td>0.269873</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">1049071</td>
<td>1531</td>
<td>1532</td>
<td>4.241950</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1049072</td>
<td>1531</td>
<td>1533</td>
<td>4.241950</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">1049073</td>
<td>1532</td>
<td>1532</td>
<td>4.241950</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1049074</td>
<td>1532</td>
<td>1533</td>
<td>4.241950</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">1049075</td>
<td>1533</td>
<td>1533</td>
<td>4.241950</td>
</tr>
</tbody>
</table>

<p>1049076 rows × 3 columns</p>
</div>
</div>
</div>
<div id="cell-116" class="cell">
<div class="sourceCode cell-code" id="cb115"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb115-1"><a href="#cb115-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the eigs</span></span>
<span id="cb115-2"><a href="#cb115-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb115-3"><a href="#cb115-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooltools</span>
<span id="cb115-4"><a href="#cb115-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb115-5"><a href="#cb115-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb115-6"><a href="#cb115-6" aria-hidden="true" tabindex="-1"></a>eigvecs, eigvals <span class="op">=</span> cooltools.lib.numutils.get_eig(matrix, n<span class="op">=</span><span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-error">
<div class="ansi-escaped-output">
<pre><span class="ansi-red-fg">---------------------------------------------------------------------------</span>

<span class="ansi-red-fg">ArpackNoConvergence</span>                       Traceback (most recent call last)

Cell <span class="ansi-green-fg">In[6], line 6</span>

<span class="ansi-green-fg ansi-bold">      3</span> <span style="font-weight:bold;color:rgb(0,135,0)">import</span> <span style="font-weight:bold;color:rgb(0,0,255)">cooltools</span>

<span class="ansi-green-fg ansi-bold">      4</span> <span style="font-weight:bold;color:rgb(0,135,0)">import</span> <span style="font-weight:bold;color:rgb(0,0,255)">pandas</span> <span style="font-weight:bold;color:rgb(0,135,0)">as</span> <span style="font-weight:bold;color:rgb(0,0,255)">pd</span>

<span class="ansi-green-fg">----&gt; 6</span> eigvecs, eigvals <span style="color:rgb(98,98,98)">=</span> cooltools<span style="color:rgb(98,98,98)">.</span>lib<span style="color:rgb(98,98,98)">.</span>numutils<span style="color:rgb(98,98,98)">.</span>get_eig(matrix, n<span style="color:rgb(98,98,98)">=</span><span style="color:rgb(98,98,98)">1</span>)



File <span class="ansi-green-fg">~/miniconda3/envs/hic/lib/python3.12/site-packages/cooltools/lib/numutils.py:519</span>, in <span class="ansi-cyan-fg">get_eig</span><span class="ansi-blue-fg">(mat, n, mask_zero_rows, subtract_mean, divide_by_mean)</span>

<span class="ansi-green-fg ansi-bold">    516</span> <span style="font-weight:bold;color:rgb(0,135,0)">else</span>:

<span class="ansi-green-fg ansi-bold">    517</span>     <span style="font-style:italic;color:rgb(95,135,135)"># adjust requested number of eigvals for "eigs"</span>

<span class="ansi-green-fg ansi-bold">    518</span>     _n <span style="color:rgb(98,98,98)">=</span> n <span style="font-weight:bold;color:rgb(0,135,0)">if</span> n <span style="color:rgb(98,98,98)">&lt;</span> (n_rows <span style="color:rgb(98,98,98)">-</span> <span style="color:rgb(98,98,98)">1</span>) <span style="font-weight:bold;color:rgb(0,135,0)">else</span> (n_rows <span style="color:rgb(98,98,98)">-</span> <span style="color:rgb(98,98,98)">2</span>)

<span class="ansi-green-fg">--&gt; 519</span>     _eigvals, _eigvecs <span style="color:rgb(98,98,98)">=</span> scipy<span style="color:rgb(98,98,98)">.</span>sparse<span style="color:rgb(98,98,98)">.</span>linalg<span style="color:rgb(98,98,98)">.</span>eigs(mat, _n)

<span class="ansi-green-fg ansi-bold">    521</span> <span style="font-style:italic;color:rgb(95,135,135)"># reorder according to eigvals and copy into output arrays</span>

<span class="ansi-green-fg ansi-bold">    522</span> order <span style="color:rgb(98,98,98)">=</span> np<span style="color:rgb(98,98,98)">.</span>argsort(<span style="color:rgb(98,98,98)">-</span>np<span style="color:rgb(98,98,98)">.</span>abs(_eigvals))



File <span class="ansi-green-fg">~/miniconda3/envs/hic/lib/python3.12/site-packages/scipy/sparse/linalg/_eigen/arpack/arpack.py:1355</span>, in <span class="ansi-cyan-fg">eigs</span><span class="ansi-blue-fg">(A, k, M, sigma, which, v0, ncv, maxiter, tol, return_eigenvectors, Minv, OPinv, OPpart)</span>

<span class="ansi-green-fg ansi-bold">   1353</span> <span style="font-weight:bold;color:rgb(0,135,0)">with</span> _ARPACK_LOCK:

<span class="ansi-green-fg ansi-bold">   1354</span>     <span style="font-weight:bold;color:rgb(0,135,0)">while</span> <span style="font-weight:bold;color:rgb(175,0,255)">not</span> params<span style="color:rgb(98,98,98)">.</span>converged:

<span class="ansi-green-fg">-&gt; 1355</span>         params<span style="color:rgb(98,98,98)">.</span>iterate()

<span class="ansi-green-fg ansi-bold">   1357</span>     <span style="font-weight:bold;color:rgb(0,135,0)">return</span> params<span style="color:rgb(98,98,98)">.</span>extract(return_eigenvectors)



File <span class="ansi-green-fg">~/miniconda3/envs/hic/lib/python3.12/site-packages/scipy/sparse/linalg/_eigen/arpack/arpack.py:760</span>, in <span class="ansi-cyan-fg">_UnsymmetricArpackParams.iterate</span><span class="ansi-blue-fg">(self)</span>

<span class="ansi-green-fg ansi-bold">    758</span>     <span style="font-weight:bold;color:rgb(0,135,0)">pass</span>

<span class="ansi-green-fg ansi-bold">    759</span> <span style="font-weight:bold;color:rgb(0,135,0)">elif</span> <span style="color:rgb(0,135,0)">self</span><span style="color:rgb(98,98,98)">.</span>info <span style="color:rgb(98,98,98)">==</span> <span style="color:rgb(98,98,98)">1</span>:

<span class="ansi-green-fg">--&gt; 760</span>     <span style="color:rgb(0,135,0)">self</span><span style="color:rgb(98,98,98)">.</span>_raise_no_convergence()

<span class="ansi-green-fg ansi-bold">    761</span> <span style="font-weight:bold;color:rgb(0,135,0)">else</span>:

<span class="ansi-green-fg ansi-bold">    762</span>     <span style="font-weight:bold;color:rgb(0,135,0)">raise</span> ArpackError(<span style="color:rgb(0,135,0)">self</span><span style="color:rgb(98,98,98)">.</span>info, infodict<span style="color:rgb(98,98,98)">=</span><span style="color:rgb(0,135,0)">self</span><span style="color:rgb(98,98,98)">.</span>iterate_infodict)



File <span class="ansi-green-fg">~/miniconda3/envs/hic/lib/python3.12/site-packages/scipy/sparse/linalg/_eigen/arpack/arpack.py:377</span>, in <span class="ansi-cyan-fg">_ArpackParams._raise_no_convergence</span><span class="ansi-blue-fg">(self)</span>

<span class="ansi-green-fg ansi-bold">    375</span>     vec <span style="color:rgb(98,98,98)">=</span> np<span style="color:rgb(98,98,98)">.</span>zeros((<span style="color:rgb(0,135,0)">self</span><span style="color:rgb(98,98,98)">.</span>n, <span style="color:rgb(98,98,98)">0</span>))

<span class="ansi-green-fg ansi-bold">    376</span>     k_ok <span style="color:rgb(98,98,98)">=</span> <span style="color:rgb(98,98,98)">0</span>

<span class="ansi-green-fg">--&gt; 377</span> <span style="font-weight:bold;color:rgb(0,135,0)">raise</span> ArpackNoConvergence(msg <span style="color:rgb(98,98,98)">%</span> (num_iter, k_ok, <span style="color:rgb(0,135,0)">self</span><span style="color:rgb(98,98,98)">.</span>k), ev, vec)



<span class="ansi-red-fg">ArpackNoConvergence</span>: ARPACK error -1: No convergence (15341 iterations, 0/1 eigenvectors converged) [ARPACK error -14: DNAUPD  did not find any eigenvalues to sufficient accuracy.]</pre>
</div>
</div>
</div>
<div id="cell-117" class="cell">
<div class="sourceCode cell-code" id="cb116"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb116-1"><a href="#cb116-1" aria-hidden="true" tabindex="-1"></a><span class="co"># # create a cooler for the smoothed matrix</span></span>
<span id="cb116-2"><a href="#cb116-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-3"><a href="#cb116-3" aria-hidden="true" tabindex="-1"></a><span class="co"># import cooler</span></span>
<span id="cb116-4"><a href="#cb116-4" aria-hidden="true" tabindex="-1"></a><span class="co"># import numpy as np</span></span>
<span id="cb116-5"><a href="#cb116-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-6"><a href="#cb116-6" aria-hidden="true" tabindex="-1"></a><span class="co"># # Create a new cooler file for the smoothed matrix</span></span>
<span id="cb116-7"><a href="#cb116-7" aria-hidden="true" tabindex="-1"></a><span class="co"># smoothed_cooler_file = cooler_file.replace('.mcool', '.smoothed.mcool')</span></span>
<span id="cb116-8"><a href="#cb116-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-9"><a href="#cb116-9" aria-hidden="true" tabindex="-1"></a><span class="co"># # Create a new cooler object</span></span>
<span id="cb116-10"><a href="#cb116-10" aria-hidden="true" tabindex="-1"></a><span class="co"># cooler.create_cooler(smoothed_cooler_file, bins=clr.bins()[:], pixels=smoothed_matrix) </span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>


<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list" style="display: none">
<div id="ref-imakaev_iterative_2012" class="csl-entry" role="listitem">
Imakaev, Maxim, Geoffrey Fudenberg, Rachel Patton McCord, Natalia Naumova, Anton Goloborodko, Bryan R Lajoie, Job Dekker, and Leonid A Mirny. 2012. <span>“Iterative Correction of <span>Hi</span>-<span>C</span> Data Reveals Hallmarks of Chromosome Organization.”</span> <em>Nature Methods</em> 9 (10): 999–1003. <a href="https://doi.org/10.1038/nmeth.2148">https://doi.org/10.1038/nmeth.2148</a>.
</div>
<div id="ref-wang_reprogramming_2019" class="csl-entry" role="listitem">
Wang, Yao, Hanben Wang, Yu Zhang, Zhenhai Du, Wei Si, Suixing Fan, Dongdong Qin, et al. 2019. <span>“Reprogramming of <span>Meiotic</span> <span>Chromatin</span> <span>Architecture</span> During <span>Spermatogenesis</span>.”</span> <em>Molecular Cell</em> 73 (3): 547–561.e6. <a href="https://doi.org/10.1016/j.molcel.2018.11.019">https://doi.org/10.1016/j.molcel.2018.11.019</a>.
</div>
</div>
</section>
</section>
</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../notebooks/02_open2c_framework.html" class="pagination-link" aria-label="Open2C framework">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-title">Open2C framework</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../scripts/master_workflow.html" class="pagination-link" aria-label="master_workflow">
        <span class="nav-page-text"><span class="chapter-title">master_workflow</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->




<footer class="footer"><div class="nav-footer"><div class="nav-footer-center"><div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.dev/munch-group/hic-spermatogenesis/blob/main/notebooks/03_compartments.ipynb" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/munch-group/hic-spermatogenesis/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li><li><a href="https://github.com/munch-group/hic-spermatogenesis/blob/main/notebooks/03_compartments.ipynb" class="toc-action"><i class="bi empty"></i>View source</a></li></ul></div></div></div></footer></body></html>