<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.39">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>E1 vs.&nbsp;ECH (recommended params) – Søren's MSc project about Hi-C and spermatogenesis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js" integrity="sha512-bLT0Qm9VnAYZDflyKcBaQ2gg0hSYNQrJ8RilYldYQ1FxQYoCLtUjuuRuZo+fjqhx/qtq/1itJ0C2ejDxltZVFg==" crossorigin="anonymous"></script><script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../workflow.html" rel="next">
<link href="../notebooks/05_rec_compartments.html" rel="prev">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-e26003cea8cd680ca0c55a263523d882.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap-6a801e7181b8b24ef9cb8ae289dd8d09.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script src="../site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="../site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="../site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://hypothes.is/embed.js"></script>
<script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" integrity="sha512-c3Nl8+7g4LMSTdrm621y7kf9v3SDPnhxLNhcjFJbKECVnmZHTdo+IRO05sNLTH/D3vA6u1X32ehoLC7WFVdheg==" crossorigin="anonymous"></script>

<script type="application/javascript">define('jquery', [],function() {return window.jQuery;})</script>


</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../notebooks/01_hicexplorer.html">Notebooks</a></li><li class="breadcrumb-item"><a href="../notebooks/06_rec_genomicintervals.html"><span class="chapter-title">E1 vs.&nbsp;ECH (recommended params)</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">Søren’s MSc project about Hi-C and spermatogenesis</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/munch-group/hic-spermatogenesis" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Project</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">Notebooks</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/01_hicexplorer.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">HiCExplorer using macaque data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/02_open2c_framework.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Open2C framework</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/03_compartments.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Compartments Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/04_genomicintervals.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">E1 vs.&nbsp;ECH</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/05_rec_compartments.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Compartments Analysis w. recommended parameters</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/06_rec_genomicintervals.html" class="sidebar-item-text sidebar-link active"><span class="chapter-title">E1 vs.&nbsp;ECH (recommended params)</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">Workflows</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../workflow.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">master_workflow</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true">
 <span class="menu-text">Reports</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../reports/thoughts.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Thoughts and Prayers</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../results/Readme.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Result files</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true">
 <span class="menu-text">Slides</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../slides/index.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Project Progress Day: Msc. Project 2024</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="true">
 <span class="menu-text">Thesis</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../thesis/index.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Chromatin Compartments and Selection on X</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#the-genomic-regions-in-question" id="toc-the-genomic-regions-in-question" class="nav-link active" data-scroll-target="#the-genomic-regions-in-question">The genomic regions in question</a>
  <ul>
  <li><a href="#load-the-data" id="toc-load-the-data" class="nav-link" data-scroll-target="#load-the-data">Load the data</a></li>
  </ul></li>
  <li><a href="#time-to-unleash-genominterv-on-the-.csv-files" id="toc-time-to-unleash-genominterv-on-the-.csv-files" class="nav-link" data-scroll-target="#time-to-unleash-genominterv-on-the-.csv-files">Time to unleash <code>genominterv</code> on the .csv files</a>
  <ul>
  <li><a href="#define-a-plotting-function" id="toc-define-a-plotting-function" class="nav-link" data-scroll-target="#define-a-plotting-function">Define a plotting function</a></li>
  <li><a href="#test-with-a-subsample-of-the-data" id="toc-test-with-a-subsample-of-the-data" class="nav-link" data-scroll-target="#test-with-a-subsample-of-the-data">Test with a subsample of the data</a></li>
  <li><a href="#bootstrap-to-get-a-p-value" id="toc-bootstrap-to-get-a-p-value" class="nav-link" data-scroll-target="#bootstrap-to-get-a-p-value">Bootstrap to get a p-value</a></li>
  <li><a href="#partition-the-a-compartments-into-regions-around-the-edges" id="toc-partition-the-a-compartments-into-regions-around-the-edges" class="nav-link" data-scroll-target="#partition-the-a-compartments-into-regions-around-the-edges">Partition the A-compartments into regions <em>around</em> the edges</a></li>
  </ul></li>
  <li><a href="#import-edges" id="toc-import-edges" class="nav-link" data-scroll-target="#import-edges">Import edges</a>
  <ul>
  <li><a href="#do-testing-of-the-edges" id="toc-do-testing-of-the-edges" class="nav-link" data-scroll-target="#do-testing-of-the-edges">Do testing of the edges</a>
  <ul class="collapse">
  <li><a href="#parallelize-the-jaccard-test" id="toc-parallelize-the-jaccard-test" class="nav-link" data-scroll-target="#parallelize-the-jaccard-test">Parallelize the Jaccard test</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#plot-some-summary-statistics" id="toc-plot-some-summary-statistics" class="nav-link" data-scroll-target="#plot-some-summary-statistics">Plot some summary statistics</a></li>
  <li><a href="#plot-proximity-result-with-jaccard-result" id="toc-plot-proximity-result-with-jaccard-result" class="nav-link" data-scroll-target="#plot-proximity-result-with-jaccard-result">Plot proximity result with Jaccard result</a></li>
  <li><a href="#plot-all-cominations-of-the-parameters" id="toc-plot-all-cominations-of-the-parameters" class="nav-link" data-scroll-target="#plot-all-cominations-of-the-parameters">Plot all cominations of the parameters</a></li>
  <li><a href="#genominterv" id="toc-genominterv" class="nav-link" data-scroll-target="#genominterv">genominterv</a></li>
  <li><a href="#geneinfo" id="toc-geneinfo" class="nav-link" data-scroll-target="#geneinfo">Geneinfo</a>
  <ul>
  <li><a href="#get-the-geneinfo-for-all-intersections-between-edges-and-ech90" id="toc-get-the-geneinfo-for-all-intersections-between-edges-and-ech90" class="nav-link" data-scroll-target="#get-the-geneinfo-for-all-intersections-between-edges-and-ech90">Get the geneinfo for all intersections between edges and ECH90</a></li>
  <li><a href="#what-can-i-do-with-the-list-of-genes-on-the-edges" id="toc-what-can-i-do-with-the-list-of-genes-on-the-edges" class="nav-link" data-scroll-target="#what-can-i-do-with-the-list-of-genes-on-the-edges">What can I do with the list of genes on the edges?</a>
  <ul class="collapse">
  <li><a href="#go-enrichment" id="toc-go-enrichment" class="nav-link" data-scroll-target="#go-enrichment">GO enrichment?</a></li>
  </ul></li>
  </ul></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.dev/munch-group/hic-spermatogenesis/blob/main/notebooks/06_rec_genomicintervals.ipynb" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/munch-group/hic-spermatogenesis/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li><li><a href="https://github.com/munch-group/hic-spermatogenesis/blob/main/notebooks/06_rec_genomicintervals.ipynb" class="toc-action"><i class="bi empty"></i>View source</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../notebooks/01_hicexplorer.html">Notebooks</a></li><li class="breadcrumb-item"><a href="../notebooks/06_rec_genomicintervals.html"><span class="chapter-title">E1 vs.&nbsp;ECH (recommended params)</span></a></li></ol></nav>
<div class="quarto-title">
<h1 class="title"><span class="chapter-title">E1 vs.&nbsp;ECH (recommended params)</span></h1>
<p class="subtitle lead">Overlaying the A-Compartments with Extended Common Haplotypes</p>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="the-genomic-regions-in-question" class="level2">
<h2 class="anchored" data-anchor-id="the-genomic-regions-in-question">The genomic regions in question</h2>
<p>In <code>05_rec_compartments.ipynb</code> we extracted the genomic intervals of A compartments on all cell types in all combinations of the following parameters:</p>
<ul>
<li>Cell type: fibroblast, spermatocyte, pachytene spermatocyte, round spermatid, sperm</li>
<li>Chromosome: X</li>
<li>E1 restriction: full-chromosome, chromosome arms, 10Mb windows</li>
<li>Resolution: 100 kb, 500 kb, ps500kb (smoothed 100kb)</li>
</ul>
<p>Resulting in <strong>45</strong> .csv files. They are saved to <code>../results/rec_compartments/</code>.</p>
<section id="load-the-data" class="level3">
<h3 class="anchored" data-anchor-id="load-the-data">Load the data</h3>
<div id="cell-3" class="cell" data-execution_count="1">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Directory containing your .csv files</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>csv_dir <span class="op">=</span> <span class="st">'../results/rec_compartments/'</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a dictionary to store the DataFrames</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>dataframes <span class="op">=</span> {}</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Iterate over all .csv files in the directory</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> filename <span class="kw">in</span> os.listdir(csv_dir):</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> filename.endswith(<span class="st">'.csv'</span>) <span class="kw">and</span> <span class="st">'e1'</span> <span class="kw">not</span> <span class="kw">in</span> filename:  <span class="co"># Check for .csv files</span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Construct the full file path</span></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>        filepath <span class="op">=</span> os.path.join(csv_dir, filename)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Load the CSV into a DataFrame</span></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Use the filename (without extension) as the dictionary key</span></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>        key <span class="op">=</span> filename.replace(<span class="st">'_a_comp_coords_'</span>, <span class="st">'_'</span>)</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>        key <span class="op">=</span> os.path.splitext(key)[<span class="dv">0</span>]</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>        dataframes[key] <span class="op">=</span> pd.read_csv(filepath)</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>        dataframes[key][<span class="st">'length'</span>] <span class="op">=</span> dataframes[key][<span class="st">'end'</span>] <span class="op">-</span> dataframes[key][<span class="st">'start'</span>]</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a><span class="co"># The `dataframes` dictionary now contains the DataFrames</span></span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a>dataframes.keys()</span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a>ech90 <span class="op">=</span> pd.read_csv(<span class="st">'../data/ech90_human_Mmul_10.csv'</span>)</span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a>ech90[<span class="st">'length'</span>] <span class="op">=</span> ech90[<span class="st">'end'</span>] <span class="op">-</span> ech90[<span class="st">'start'</span>]</span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a><span class="co"># Load the chromosome sizes</span></span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a>chromsizes <span class="op">=</span> (pd.read_csv(</span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a>    <span class="st">'../data/rheMac10.filtered.chrom.sizes'</span>, </span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a>    sep<span class="op">=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>, </span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a>    index_col<span class="op">=</span><span class="st">'chrom'</span>, </span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a>    header<span class="op">=</span><span class="va">None</span>, </span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a>    names<span class="op">=</span>[<span class="st">'chrom'</span>,<span class="st">'size'</span>])</span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a>.to_dict()[<span class="st">'size'</span>]</span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="time-to-unleash-genominterv-on-the-.csv-files" class="level2">
<h2 class="anchored" data-anchor-id="time-to-unleash-genominterv-on-the-.csv-files">Time to unleash <code>genominterv</code> on the .csv files</h2>
<section id="define-a-plotting-function" class="level3">
<h3 class="anchored" data-anchor-id="define-a-plotting-function">Define a plotting function</h3>
<div id="cell-6" class="cell" data-execution_count="2">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Kaspers plotting function</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="op">%</span>config InlineBackend.figure_format <span class="op">=</span> <span class="st">'svg'</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> plot_intervals(query<span class="op">=</span><span class="va">None</span>, annot<span class="op">=</span><span class="va">None</span>, <span class="op">**</span>kwargs):</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>    tups <span class="op">=</span> []</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> query <span class="kw">is</span> <span class="kw">not</span> <span class="va">None</span>:</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>        tups.append((<span class="st">'query'</span>, query))</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> annot <span class="kw">is</span> <span class="kw">not</span> <span class="va">None</span>:</span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>        tups.append((<span class="st">'annot'</span>, annot))</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>    tups.extend(kwargs.items())</span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>    tups <span class="op">=</span> <span class="bu">reversed</span>(tups)</span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>    df_list <span class="op">=</span> []</span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>    labels <span class="op">=</span> []</span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> label, df <span class="kw">in</span> tups:</span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>        labels.append(label)</span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>        df[<span class="st">'label'</span>] <span class="op">=</span> label</span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a>        df_list.append(df)</span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a>    bigdf <span class="op">=</span> pd.concat(df_list)</span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a>    bigdf[<span class="st">'chrom'</span>] <span class="op">=</span> pd.Categorical(bigdf[<span class="st">'chrom'</span>], bigdf[<span class="st">'chrom'</span>].unique())</span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a>    bigdf[<span class="st">'label'</span>] <span class="op">=</span> pd.Categorical(bigdf[<span class="st">'label'</span>], bigdf[<span class="st">'label'</span>].unique())</span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a>    gr <span class="op">=</span> bigdf.groupby(<span class="st">'chrom'</span>, observed<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb2-30"><a href="#cb2-30" aria-hidden="true" tabindex="-1"></a>    fig, axes <span class="op">=</span> plt.subplots(gr.ngroups, <span class="dv">1</span>, figsize<span class="op">=</span>(<span class="dv">10</span>, <span class="fl">1.5</span><span class="op">*</span>gr.ngroups), </span>
<span id="cb2-31"><a href="#cb2-31" aria-hidden="true" tabindex="-1"></a>                            sharey<span class="op">=</span><span class="va">True</span></span>
<span id="cb2-32"><a href="#cb2-32" aria-hidden="true" tabindex="-1"></a>                            <span class="co">#  sharex=True</span></span>
<span id="cb2-33"><a href="#cb2-33" aria-hidden="true" tabindex="-1"></a>                             )</span>
<span id="cb2-34"><a href="#cb2-34" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> <span class="bu">type</span>(axes) <span class="kw">is</span> <span class="kw">not</span> np.ndarray:</span>
<span id="cb2-35"><a href="#cb2-35" aria-hidden="true" tabindex="-1"></a>        <span class="co"># in case there is only one axis so it not returned as a list</span></span>
<span id="cb2-36"><a href="#cb2-36" aria-hidden="true" tabindex="-1"></a>        axes <span class="op">=</span> np.array([axes])</span>
<span id="cb2-37"><a href="#cb2-37" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb2-38"><a href="#cb2-38" aria-hidden="true" tabindex="-1"></a>    <span class="co"># with plt.style.context(('default')):</span></span>
<span id="cb2-39"><a href="#cb2-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-40"><a href="#cb2-40" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> i, chrom <span class="kw">in</span> <span class="bu">enumerate</span>(gr.groups):</span>
<span id="cb2-41"><a href="#cb2-41" aria-hidden="true" tabindex="-1"></a>        _df <span class="op">=</span> gr.get_group(chrom)</span>
<span id="cb2-42"><a href="#cb2-42" aria-hidden="true" tabindex="-1"></a>        _gr <span class="op">=</span> _df.groupby(<span class="st">'label'</span>, observed<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb2-43"><a href="#cb2-43" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> y, label <span class="kw">in</span> <span class="bu">enumerate</span>(_gr.groups):</span>
<span id="cb2-44"><a href="#cb2-44" aria-hidden="true" tabindex="-1"></a>            <span class="cf">try</span>:</span>
<span id="cb2-45"><a href="#cb2-45" aria-hidden="true" tabindex="-1"></a>                df <span class="op">=</span> _gr.get_group(label)</span>
<span id="cb2-46"><a href="#cb2-46" aria-hidden="true" tabindex="-1"></a>            <span class="cf">except</span> <span class="pp">KeyError</span>:</span>
<span id="cb2-47"><a href="#cb2-47" aria-hidden="true" tabindex="-1"></a>                <span class="cf">continue</span></span>
<span id="cb2-48"><a href="#cb2-48" aria-hidden="true" tabindex="-1"></a>            y <span class="op">=</span> np.repeat(y, df.index.size)</span>
<span id="cb2-49"><a href="#cb2-49" aria-hidden="true" tabindex="-1"></a>            axes[i].hlines(y, df.start.tolist(), df.end.tolist(), alpha<span class="op">=</span><span class="fl">0.5</span>, lw<span class="op">=</span><span class="dv">5</span>, colors<span class="op">=</span><span class="ss">f'C</span><span class="sc">{</span>y[<span class="dv">0</span>]<span class="sc">}</span><span class="ss">'</span>)</span>
<span id="cb2-50"><a href="#cb2-50" aria-hidden="true" tabindex="-1"></a>            delta <span class="op">=</span> <span class="bu">len</span>(labels)<span class="op">/</span><span class="dv">10</span></span>
<span id="cb2-51"><a href="#cb2-51" aria-hidden="true" tabindex="-1"></a>            axes[i].vlines(df.start.tolist(), y<span class="op">-</span>delta, y<span class="op">+</span>delta, alpha<span class="op">=</span><span class="fl">0.5</span>, lw<span class="op">=</span><span class="dv">2</span>, colors<span class="op">=</span><span class="ss">f'C</span><span class="sc">{</span>y[<span class="dv">0</span>]<span class="sc">}</span><span class="ss">'</span>)</span>
<span id="cb2-52"><a href="#cb2-52" aria-hidden="true" tabindex="-1"></a>            axes[i].vlines(df.end.tolist(), y<span class="op">-</span>delta, y<span class="op">+</span>delta, alpha<span class="op">=</span><span class="fl">0.5</span>, lw<span class="op">=</span><span class="dv">2</span>, colors<span class="op">=</span><span class="ss">f'C</span><span class="sc">{</span>y[<span class="dv">0</span>]<span class="sc">}</span><span class="ss">'</span>)</span>
<span id="cb2-53"><a href="#cb2-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-54"><a href="#cb2-54" aria-hidden="true" tabindex="-1"></a>        axes[i].spines[<span class="st">'top'</span>].set_visible(<span class="va">False</span>)</span>
<span id="cb2-55"><a href="#cb2-55" aria-hidden="true" tabindex="-1"></a>        axes[i].spines[<span class="st">'left'</span>].set_visible(<span class="va">False</span>)</span>
<span id="cb2-56"><a href="#cb2-56" aria-hidden="true" tabindex="-1"></a>        axes[i].spines[<span class="st">'right'</span>].set_visible(<span class="va">False</span>)</span>
<span id="cb2-57"><a href="#cb2-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-58"><a href="#cb2-58" aria-hidden="true" tabindex="-1"></a>        axes[i].set_yticks(<span class="bu">list</span>(<span class="bu">range</span>(<span class="bu">len</span>(labels))), labels)</span>
<span id="cb2-59"><a href="#cb2-59" aria-hidden="true" tabindex="-1"></a>        axes[i].tick_params(axis<span class="op">=</span><span class="st">'y'</span>, which<span class="op">=</span><span class="st">'both'</span>, left<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb2-60"><a href="#cb2-60" aria-hidden="true" tabindex="-1"></a>        axes[i].set_ylim(<span class="op">-</span><span class="dv">1</span>, <span class="bu">len</span>(labels)<span class="op">-</span><span class="fl">0.7</span>)</span>
<span id="cb2-61"><a href="#cb2-61" aria-hidden="true" tabindex="-1"></a>        <span class="co"># axes[i].set_xlim(df.start.min()-delta, df.end.max()+delta)</span></span>
<span id="cb2-62"><a href="#cb2-62" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> i <span class="op">!=</span> gr.ngroups<span class="op">-</span><span class="dv">1</span>:</span>
<span id="cb2-63"><a href="#cb2-63" aria-hidden="true" tabindex="-1"></a>            axes[i].tick_params(axis<span class="op">=</span><span class="st">'x'</span>, which<span class="op">=</span><span class="st">'both'</span>, bottom<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb2-64"><a href="#cb2-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-65"><a href="#cb2-65" aria-hidden="true" tabindex="-1"></a>        axes[i].set_title(chrom, loc<span class="op">=</span><span class="st">'left'</span>, fontsize<span class="op">=</span><span class="dv">10</span>)</span>
<span id="cb2-66"><a href="#cb2-66" aria-hidden="true" tabindex="-1"></a>    plt.tight_layout()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div id="cell-7" class="cell" data-execution_count="34">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># My plotting function</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> matplotlib.patches <span class="im">import</span> Rectangle</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> mpl_toolkits.axes_grid1 <span class="im">import</span> make_axes_locatable</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> genominterv <span class="im">import</span> interval_intersect</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> plot_regions(query<span class="op">=</span><span class="va">None</span>, annot<span class="op">=</span><span class="va">None</span>, intersect<span class="op">=</span><span class="va">None</span>, </span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>                 track_titles<span class="op">=</span>[<span class="st">'query'</span>, <span class="st">'annot'</span>, <span class="st">'intersect'</span>], </span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>                 figsize<span class="op">=</span>(<span class="dv">10</span>, <span class="dv">1</span>)):</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>    chrom <span class="op">=</span> annot[<span class="st">'chrom'</span>].unique()[<span class="dv">0</span>]</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>    chromsize <span class="op">=</span> pd.read_csv(<span class="st">'../data/rheMac10.filtered.chrom.sizes'</span>, sep<span class="op">=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>, header<span class="op">=</span><span class="va">None</span>, names<span class="op">=</span>[<span class="st">'chrom'</span>, <span class="st">'size'</span>])</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>    chromsize <span class="op">=</span> chromsize[chromsize[<span class="st">'chrom'</span>] <span class="op">==</span> chrom][<span class="st">'size'</span>].values[<span class="dv">0</span>]</span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Define the plot size</span></span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a>    f, ax <span class="op">=</span> plt.subplots(figsize<span class="op">=</span>figsize, sharex<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a>    ax.spines[:].set_visible(<span class="va">False</span>)</span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Plot the annot</span></span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Iterate over each interval in the DataFrame</span></span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> start, end <span class="kw">in</span> <span class="bu">zip</span>(annot[<span class="st">'start'</span>], annot[<span class="st">'end'</span>]):</span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a>        rect <span class="op">=</span> Rectangle((start, <span class="fl">0.1</span>), width<span class="op">=</span>end<span class="op">-</span>start, height<span class="op">=</span><span class="fl">0.9</span>, color<span class="op">=</span><span class="st">'tab:red'</span>, linewidth<span class="op">=</span><span class="dv">0</span>, alpha<span class="op">=</span><span class="fl">0.8</span>)</span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a>        ax.add_patch(rect)</span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a>        ax.spines[<span class="st">'bottom'</span>].set_visible(<span class="va">True</span>)</span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true" tabindex="-1"></a>        ax.set_ylabel(track_titles[<span class="dv">1</span>], rotation<span class="op">=</span><span class="dv">0</span>, labelpad<span class="op">=</span><span class="dv">10</span>, ha<span class="op">=</span><span class="st">'right'</span>)</span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb3-32"><a href="#cb3-32" aria-hidden="true" tabindex="-1"></a>    divider <span class="op">=</span> make_axes_locatable(ax)</span>
<span id="cb3-33"><a href="#cb3-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-34"><a href="#cb3-34" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> query <span class="kw">is</span> <span class="kw">not</span> <span class="va">None</span>:</span>
<span id="cb3-35"><a href="#cb3-35" aria-hidden="true" tabindex="-1"></a>        qax <span class="op">=</span> divider.append_axes(<span class="st">"top"</span>, size<span class="op">=</span><span class="st">"100%"</span>, pad<span class="op">=</span><span class="fl">0.05</span>, sharex<span class="op">=</span>ax)</span>
<span id="cb3-36"><a href="#cb3-36" aria-hidden="true" tabindex="-1"></a>        qax.xaxis.set_visible(<span class="va">False</span>)</span>
<span id="cb3-37"><a href="#cb3-37" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Plot the query</span></span>
<span id="cb3-38"><a href="#cb3-38" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> start, end <span class="kw">in</span> <span class="bu">zip</span>(query[<span class="st">'start'</span>], query[<span class="st">'end'</span>]):</span>
<span id="cb3-39"><a href="#cb3-39" aria-hidden="true" tabindex="-1"></a>            rect <span class="op">=</span> Rectangle((start, <span class="fl">0.1</span>), width<span class="op">=</span>end<span class="op">-</span>start, height<span class="op">=</span><span class="fl">0.9</span>, color<span class="op">=</span><span class="st">'tab:blue'</span>, linewidth<span class="op">=</span><span class="dv">0</span>, alpha<span class="op">=</span><span class="fl">0.8</span>)</span>
<span id="cb3-40"><a href="#cb3-40" aria-hidden="true" tabindex="-1"></a>            qax.add_patch(rect)</span>
<span id="cb3-41"><a href="#cb3-41" aria-hidden="true" tabindex="-1"></a>            qax.spines[:].set_visible(<span class="va">False</span>)</span>
<span id="cb3-42"><a href="#cb3-42" aria-hidden="true" tabindex="-1"></a>            qax.set_yticks([]) </span>
<span id="cb3-43"><a href="#cb3-43" aria-hidden="true" tabindex="-1"></a>            qax.set_title(chrom, loc<span class="op">=</span><span class="st">'left'</span>)</span>
<span id="cb3-44"><a href="#cb3-44" aria-hidden="true" tabindex="-1"></a>            qax.set_ylabel(track_titles[<span class="dv">0</span>], rotation<span class="op">=</span><span class="dv">0</span>, labelpad<span class="op">=</span><span class="dv">10</span>, ha<span class="op">=</span><span class="st">'right'</span>)</span>
<span id="cb3-45"><a href="#cb3-45" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb3-46"><a href="#cb3-46" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> intersect <span class="kw">is</span> <span class="kw">not</span> <span class="va">None</span>:</span>
<span id="cb3-47"><a href="#cb3-47" aria-hidden="true" tabindex="-1"></a>        iax <span class="op">=</span> divider.append_axes(<span class="st">"bottom"</span>, size<span class="op">=</span><span class="st">"100%"</span>, pad<span class="op">=</span><span class="fl">0.05</span>, sharex<span class="op">=</span>ax)</span>
<span id="cb3-48"><a href="#cb3-48" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Invisible x-axis for 'annot' (intersect ie below) </span></span>
<span id="cb3-49"><a href="#cb3-49" aria-hidden="true" tabindex="-1"></a>        ax.xaxis.set_visible(<span class="va">False</span>)</span>
<span id="cb3-50"><a href="#cb3-50" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Plot the intersect</span></span>
<span id="cb3-51"><a href="#cb3-51" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> start, end <span class="kw">in</span> <span class="bu">zip</span>(intersect[<span class="st">'start'</span>], intersect[<span class="st">'end'</span>]):</span>
<span id="cb3-52"><a href="#cb3-52" aria-hidden="true" tabindex="-1"></a>            rect <span class="op">=</span> Rectangle((start, <span class="fl">0.1</span>), width<span class="op">=</span>end<span class="op">-</span>start, height<span class="op">=</span><span class="fl">0.9</span>, color<span class="op">=</span><span class="st">'tab:green'</span>, linewidth<span class="op">=</span><span class="dv">0</span>, alpha<span class="op">=</span><span class="fl">0.8</span>)</span>
<span id="cb3-53"><a href="#cb3-53" aria-hidden="true" tabindex="-1"></a>            iax.add_patch(rect)</span>
<span id="cb3-54"><a href="#cb3-54" aria-hidden="true" tabindex="-1"></a>            iax.spines[:].set_visible(<span class="va">False</span>)</span>
<span id="cb3-55"><a href="#cb3-55" aria-hidden="true" tabindex="-1"></a>            iax.set_yticks([]) </span>
<span id="cb3-56"><a href="#cb3-56" aria-hidden="true" tabindex="-1"></a>            ax.spines[<span class="st">'bottom'</span>].set_visible(<span class="va">False</span>)</span>
<span id="cb3-57"><a href="#cb3-57" aria-hidden="true" tabindex="-1"></a>            iax.spines[<span class="st">'bottom'</span>].set_visible(<span class="va">True</span>)</span>
<span id="cb3-58"><a href="#cb3-58" aria-hidden="true" tabindex="-1"></a>            iax.set_ylabel(track_titles[<span class="dv">2</span>], rotation<span class="op">=</span><span class="dv">0</span>, labelpad<span class="op">=</span><span class="dv">10</span>, ha<span class="op">=</span><span class="st">'right'</span>)</span>
<span id="cb3-59"><a href="#cb3-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-60"><a href="#cb3-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-61"><a href="#cb3-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-62"><a href="#cb3-62" aria-hidden="true" tabindex="-1"></a>    ax.set_yticks([])</span>
<span id="cb3-63"><a href="#cb3-63" aria-hidden="true" tabindex="-1"></a>    ax.set_xlim(<span class="dv">0</span>, chromsize)</span>
<span id="cb3-64"><a href="#cb3-64" aria-hidden="true" tabindex="-1"></a>    ticks <span class="op">=</span> np.linspace(<span class="dv">0</span>, chromsize, num<span class="op">=</span><span class="dv">5</span>)</span>
<span id="cb3-65"><a href="#cb3-65" aria-hidden="true" tabindex="-1"></a>    ax.set_xticks(ticks) </span>
<span id="cb3-66"><a href="#cb3-66" aria-hidden="true" tabindex="-1"></a>    ax.set_xticklabels([<span class="ss">f'</span><span class="sc">{</span><span class="bu">int</span>(t<span class="op">/</span><span class="fl">1e6</span>)<span class="sc">}</span><span class="ss"> Mbp'</span> <span class="cf">for</span> t <span class="kw">in</span> ticks])</span>
<span id="cb3-67"><a href="#cb3-67" aria-hidden="true" tabindex="-1"></a>    plt.tight_layout()</span>
<span id="cb3-68"><a href="#cb3-68" aria-hidden="true" tabindex="-1"></a></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div id="cell-8" class="cell" data-execution_count="4">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Define my params:</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="co">## Params will comply with my desired layout for the Manuscript (PDF)</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>notebook_rcparams <span class="op">=</span> {</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>    <span class="st">'font.size'</span>: <span class="dv">7</span>,</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>    <span class="st">'axes.titlesize'</span>: <span class="dv">8</span>,</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>    <span class="st">'axes.labelsize'</span>: <span class="dv">7</span>,</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>    <span class="st">'xtick.labelsize'</span>: <span class="dv">6</span>,</span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>    <span class="st">'ytick.labelsize'</span>: <span class="dv">6</span>,</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>    <span class="st">'figure.titlesize'</span>: <span class="dv">9</span>,</span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>    <span class="st">'figure.figsize'</span>: [<span class="fl">6.0</span>, <span class="fl">2.0</span>],</span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>    <span class="st">'figure.labelsize'</span>: <span class="dv">7</span></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply config</span></span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a><span class="op">%</span>matplotlib inline</span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a><span class="op">%</span>config InlineBackend.figure_formats <span class="op">=</span> [<span class="st">'retina'</span>]</span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a><span class="op">%</span>config InlineBackend.rc <span class="op">=</span> notebook_rcparams</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div id="cell-9" class="cell" data-execution_count="5">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>({key:plt.rcParams[key] <span class="cf">for</span> key <span class="kw">in</span> notebook_rcparams.keys()})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>{'font.size': 10.0, 'axes.titlesize': 'large', 'axes.labelsize': 'medium', 'xtick.labelsize': 'medium', 'ytick.labelsize': 'medium', 'figure.titlesize': 'large', 'figure.figsize': [6.4, 4.8], 'figure.labelsize': 'large'}</code></pre>
</div>
</div>
<div id="cell-10" class="cell" data-execution_count="6">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Define my params:</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="co">## Params will comply with my desired layout for the Manuscript (PDF)</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>notebook_rcparams <span class="op">=</span> {</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>    <span class="st">'font.size'</span>: <span class="dv">7</span>,</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>    <span class="st">'axes.titlesize'</span>: <span class="dv">8</span>,</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>    <span class="st">'axes.labelsize'</span>: <span class="dv">7</span>,</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>    <span class="st">'xtick.labelsize'</span>: <span class="dv">6</span>,</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>    <span class="st">'ytick.labelsize'</span>: <span class="dv">6</span>,</span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>    <span class="st">'figure.titlesize'</span>: <span class="dv">9</span>,</span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>    <span class="st">'figure.figsize'</span>: [<span class="fl">6.0</span>, <span class="fl">2.0</span>],</span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>    <span class="st">'figure.labelsize'</span>: <span class="dv">7</span></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply config</span></span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a><span class="op">%</span>matplotlib inline</span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a><span class="op">%</span>config InlineBackend.figure_formats <span class="op">=</span> [<span class="st">'retina'</span>]</span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a><span class="op">%</span>config InlineBackend.rc <span class="op">=</span> notebook_rcparams</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div id="cell-11" class="cell" data-execution_count="7">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>({key:plt.rcParams[key] <span class="cf">for</span> key <span class="kw">in</span> notebook_rcparams.keys()})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>{'font.size': 7.0, 'axes.titlesize': 8.0, 'axes.labelsize': 7.0, 'xtick.labelsize': 6.0, 'ytick.labelsize': 6.0, 'figure.titlesize': 9.0, 'figure.figsize': [6.0, 2.0], 'figure.labelsize': 7.0}</code></pre>
</div>
</div>
</section>
<section id="test-with-a-subsample-of-the-data" class="level3">
<h3 class="anchored" data-anchor-id="test-with-a-subsample-of-the-data">Test with a subsample of the data</h3>
<div id="cell-13" class="cell" data-execution_count="16">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>annot <span class="op">=</span> dataframes[<span class="st">'round_spermatid_100kb_arms'</span>]</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>query <span class="op">=</span> ech90</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>intersect <span class="op">=</span> interval_intersect(annot, query)</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>plot_intervals(query, annot, intersection<span class="op">=</span>intersect)</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>plot_regions(query, annot, intersect)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><a href="06_rec_genomicintervals_files/figure-html/cell-9-output-1.svg" class="lightbox" data-gallery="quarto-lightbox-gallery-1"><img src="06_rec_genomicintervals_files/figure-html/cell-9-output-1.svg" class="img-fluid figure-img"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><a href="06_rec_genomicintervals_files/figure-html/cell-9-output-2.svg" class="lightbox" data-gallery="quarto-lightbox-gallery-2"><img src="06_rec_genomicintervals_files/figure-html/cell-9-output-2.svg" class="img-fluid figure-img"></a></p>
</figure>
</div>
</div>
</div>
<div id="cell-14" class="cell" data-execution_count="17">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> genominterv <span class="im">import</span> proximity_test, interval_collapse, interval_diff, interval_intersect, jaccard_stat</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>annot <span class="op">=</span> dataframes[<span class="st">'round_spermatid_100kb_arms'</span>]</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>query <span class="op">=</span> ech90</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a><span class="co">#plot_intervals(query=query, annot=annot)</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> key,annot <span class="kw">in</span> dataframes.items():</span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Filter out subset</span></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="st">'round_spermatid_100'</span>) <span class="kw">in</span> key <span class="kw">and</span> <span class="kw">not</span> <span class="st">'full'</span> <span class="kw">in</span> key:</span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Plot the intervals</span></span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>        intersection <span class="op">=</span> interval_intersect(query, annot)</span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>        plot_intervals(query<span class="op">=</span>query, annot<span class="op">=</span>annot, intersection<span class="op">=</span>intersection)</span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a>        plt.title(key)</span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Do a proximity test</span></span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Tests for </span><span class="sc">{</span>key<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a>        annot_collapsed <span class="op">=</span> interval_collapse(annot)</span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a>        non_ovl_query <span class="op">=</span> interval_diff(query, annot_collapsed)</span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"Proximity:"</span>, proximity_test(non_ovl_query, annot_collapsed))</span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"Jaccard:"</span>, jaccard_stat(query, annot))</span>
<span id="cb11-22"><a href="#cb11-22" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>()</span>
<span id="cb11-23"><a href="#cb11-23" aria-hidden="true" tabindex="-1"></a></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Tests for round_spermatid_100kb_arms_smoothed
Proximity: TestResult(statistic=0.3711999999999999, pvalue=0.0204)
Jaccard: 0.03771152626674943

Tests for round_spermatid_100kb_10Mb
Proximity: TestResult(statistic=0.51525, pvalue=0.0)
Jaccard: 0.046023479610810235

Tests for round_spermatid_100kb_10Mb_smoothed
Proximity: TestResult(statistic=0.46520000000000006, pvalue=0.0049)
Jaccard: 0.042623832902054265

Tests for round_spermatid_100kb_arms
Proximity: TestResult(statistic=0.37199999999999966, pvalue=0.0121)
Jaccard: 0.046543092859550536
</code></pre>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><a href="06_rec_genomicintervals_files/figure-html/cell-10-output-2.svg" class="lightbox" data-gallery="quarto-lightbox-gallery-3"><img src="06_rec_genomicintervals_files/figure-html/cell-10-output-2.svg" class="img-fluid figure-img"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><a href="06_rec_genomicintervals_files/figure-html/cell-10-output-3.svg" class="lightbox" data-gallery="quarto-lightbox-gallery-4"><img src="06_rec_genomicintervals_files/figure-html/cell-10-output-3.svg" class="img-fluid figure-img"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><a href="06_rec_genomicintervals_files/figure-html/cell-10-output-4.svg" class="lightbox" data-gallery="quarto-lightbox-gallery-5"><img src="06_rec_genomicintervals_files/figure-html/cell-10-output-4.svg" class="img-fluid figure-img"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><a href="06_rec_genomicintervals_files/figure-html/cell-10-output-5.svg" class="lightbox" data-gallery="quarto-lightbox-gallery-6"><img src="06_rec_genomicintervals_files/figure-html/cell-10-output-5.svg" class="img-fluid figure-img"></a></p>
</figure>
</div>
</div>
</div>
</section>
<section id="bootstrap-to-get-a-p-value" class="level3">
<h3 class="anchored" data-anchor-id="bootstrap-to-get-a-p-value">Bootstrap to get a p-value</h3>
<div id="cell-16" class="cell" data-execution_count="6">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> genominterv <span class="im">import</span> bootstrap</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>annot <span class="op">=</span> dataframes[<span class="st">'round_spermatid_100kb_arms'</span>]</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>chromsizes <span class="op">=</span> pd.read_csv(<span class="st">'../data/rheMac10.filtered.chrom.sizes'</span>, sep<span class="op">=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>, index_col<span class="op">=</span><span class="st">'chrom'</span>, header<span class="op">=</span><span class="va">None</span>, names<span class="op">=</span>[<span class="st">'chrom'</span>,<span class="st">'size'</span>]).to_dict()[<span class="st">'size'</span>]</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="co">#display(chromsizes)</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a><span class="at">@bootstrap</span>(chromsizes, samples<span class="op">=</span><span class="dv">1000</span>)</span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> jaccard_bootstrap(query, annot):</span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>    <span class="cf">return</span> jaccard_stat(query, annot)</span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>jacccard_stat, p_value <span class="op">=</span> jaccard_bootstrap(query, annot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div id="cell-17" class="cell" data-execution_count="7">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>jacccard_stat, p_value</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="7">
<pre><code>(0.046543092859550536, 0.112)</code></pre>
</div>
</div>
</section>
<section id="partition-the-a-compartments-into-regions-around-the-edges" class="level3">
<h3 class="anchored" data-anchor-id="partition-the-a-compartments-into-regions-around-the-edges">Partition the A-compartments into regions <em>around</em> the edges</h3>
<div id="cell-19" class="cell" data-execution_count="8">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> genominterv <span class="im">import</span> interval_collapse, interval_union</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>df <span class="op">=</span> dataframes[<span class="st">'round_spermatid_100kb_arms'</span>]</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>start_edge <span class="op">=</span> pd.DataFrame({</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>    <span class="st">'chrom'</span>: df[<span class="st">'chrom'</span>],</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>    <span class="st">'start'</span>: df.<span class="bu">apply</span>(<span class="kw">lambda</span> x: <span class="bu">max</span>(x[<span class="st">'start'</span>] <span class="op">-</span> x[<span class="st">'resolution'</span>], <span class="dv">0</span>), axis<span class="op">=</span><span class="dv">1</span>),</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>    <span class="st">'end'</span>: df[<span class="st">'start'</span>]<span class="op">+</span><span class="dv">1</span><span class="op">*</span>df[<span class="st">'resolution'</span>],</span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>    <span class="st">'resolution'</span>: df[<span class="st">'resolution'</span>],</span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>    <span class="st">'label'</span>: <span class="st">'start_edge'</span></span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>end_edge <span class="op">=</span> pd.DataFrame({</span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>    <span class="st">'chrom'</span>: df[<span class="st">'chrom'</span>],</span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>    <span class="st">'start'</span>: df[<span class="st">'end'</span>]<span class="op">-</span><span class="dv">1</span><span class="op">*</span>df[<span class="st">'resolution'</span>],</span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a>    <span class="st">'end'</span>: df.<span class="bu">apply</span>(<span class="kw">lambda</span> x: <span class="bu">min</span>(x[<span class="st">'end'</span>] <span class="op">+</span> x[<span class="st">'resolution'</span>], chromsizes[<span class="st">'chrX'</span>]), axis<span class="op">=</span><span class="dv">1</span>),</span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a>    <span class="st">'resolution'</span>: df[<span class="st">'resolution'</span>],</span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a>    <span class="st">'label'</span>: <span class="st">'end_edge'</span></span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-20"><a href="#cb16-20" aria-hidden="true" tabindex="-1"></a><span class="co"># The end cannot exceed the chromosome size</span></span>
<span id="cb16-21"><a href="#cb16-21" aria-hidden="true" tabindex="-1"></a>test_df <span class="op">=</span> pd.concat([start_edge, end_edge]).sort_values([<span class="st">'chrom'</span>, <span class="st">'start'</span>, <span class="st">'end'</span>])</span>
<span id="cb16-22"><a href="#cb16-22" aria-hidden="true" tabindex="-1"></a>interval_collapse(test_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="8">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">start</th>
<th data-quarto-table-cell-role="th">end</th>
<th data-quarto-table-cell-role="th">chrom</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>800000</td>
<td>1000000</td>
<td>chrX</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>1500000</td>
<td>1800000</td>
<td>chrX</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>2500000</td>
<td>2700000</td>
<td>chrX</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3</td>
<td>2800000</td>
<td>3300000</td>
<td>chrX</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">4</td>
<td>3400000</td>
<td>3600000</td>
<td>chrX</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">91</td>
<td>148700000</td>
<td>148900000</td>
<td>chrX</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">92</td>
<td>149000000</td>
<td>149400000</td>
<td>chrX</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">93</td>
<td>151100000</td>
<td>151300000</td>
<td>chrX</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">94</td>
<td>152400000</td>
<td>152600000</td>
<td>chrX</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">95</td>
<td>153200000</td>
<td>153388924</td>
<td>chrX</td>
</tr>
</tbody>
</table>

<p>96 rows × 3 columns</p>
</div>
</div>
</div>
<div id="cell-20" class="cell" data-execution_count="9">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> genominterv <span class="im">import</span> interval_collapse</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> key, df <span class="kw">in</span> dataframes.items():</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>    outdir <span class="op">=</span> <span class="st">'../results/rec_edges'</span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>    edge_csv_name <span class="op">=</span> os.path.join(outdir,<span class="ss">f'</span><span class="sc">{</span>key<span class="op">+</span><span class="st">'_edges.csv'</span><span class="sc">}</span><span class="ss">'</span>)</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> <span class="kw">not</span> os.path.exists(edge_csv_name):</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>        res <span class="op">=</span> df[<span class="st">'resolution'</span>].unique()[<span class="dv">0</span>]</span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>        start_edge <span class="op">=</span> pd.DataFrame({</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>            <span class="st">'chrom'</span>: df[<span class="st">'chrom'</span>],</span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>            <span class="st">'start'</span>: df.<span class="bu">apply</span>(<span class="kw">lambda</span> x: <span class="bu">max</span>(x[<span class="st">'start'</span>] <span class="op">-</span> x[<span class="st">'resolution'</span>], <span class="dv">0</span>), axis<span class="op">=</span><span class="dv">1</span>),</span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>            <span class="st">'end'</span>: df[<span class="st">'start'</span>]<span class="op">+</span><span class="dv">1</span><span class="op">*</span>df[<span class="st">'resolution'</span>],</span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>            <span class="st">'resolution'</span>: df[<span class="st">'resolution'</span>],</span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a>            <span class="st">'label'</span>: <span class="st">'start_edge'</span></span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a>        })</span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a>        end_edge <span class="op">=</span> pd.DataFrame({</span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a>            <span class="st">'chrom'</span>: df[<span class="st">'chrom'</span>],</span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a>            <span class="st">'start'</span>: df[<span class="st">'end'</span>]<span class="op">-</span><span class="dv">1</span><span class="op">*</span>df[<span class="st">'resolution'</span>],</span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a>            <span class="st">'end'</span>: df.<span class="bu">apply</span>(<span class="kw">lambda</span> x: <span class="bu">min</span>(x[<span class="st">'end'</span>] <span class="op">+</span> x[<span class="st">'resolution'</span>], chromsizes[<span class="st">'chrX'</span>]), axis<span class="op">=</span><span class="dv">1</span>),</span>
<span id="cb17-21"><a href="#cb17-21" aria-hidden="true" tabindex="-1"></a>            <span class="st">'resolution'</span>: df[<span class="st">'resolution'</span>],</span>
<span id="cb17-22"><a href="#cb17-22" aria-hidden="true" tabindex="-1"></a>            <span class="st">'label'</span>: <span class="st">'end_edge'</span></span>
<span id="cb17-23"><a href="#cb17-23" aria-hidden="true" tabindex="-1"></a>        })</span>
<span id="cb17-24"><a href="#cb17-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-25"><a href="#cb17-25" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> <span class="kw">not</span> os.path.exists(outdir):</span>
<span id="cb17-26"><a href="#cb17-26" aria-hidden="true" tabindex="-1"></a>            os.makedirs(outdir)</span>
<span id="cb17-27"><a href="#cb17-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-28"><a href="#cb17-28" aria-hidden="true" tabindex="-1"></a>        tmp <span class="op">=</span> pd.concat([start_edge, end_edge]).sort_values([<span class="st">'chrom'</span>, <span class="st">'start'</span>, <span class="st">'end'</span>])</span>
<span id="cb17-29"><a href="#cb17-29" aria-hidden="true" tabindex="-1"></a>        interval_collapse(tmp).assign(resolution<span class="op">=</span>res).to_csv(edge_csv_name, index<span class="op">=</span><span class="va">False</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="import-edges" class="level2">
<h2 class="anchored" data-anchor-id="import-edges">Import edges</h2>
<div id="cell-22" class="cell" data-execution_count="21">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Directory containing your .csv files</span></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>csv_dir <span class="op">=</span> <span class="st">'../results/rec_edges/'</span></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a dictionary to store the DataFrames</span></span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>edge_df <span class="op">=</span> {}</span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Iterate over all .csv files in the directory</span></span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> filename <span class="kw">in</span> os.listdir(csv_dir):</span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> filename.endswith(<span class="st">'.csv'</span>):  <span class="co"># Check for .csv files</span></span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Construct the full file path</span></span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>        filepath <span class="op">=</span> os.path.join(csv_dir, filename)</span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Load the CSV into a DataFrame</span></span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Use the filename (without extension) as the dictionary key</span></span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a>        key <span class="op">=</span> filename.replace(<span class="st">'_edges_'</span>, <span class="st">''</span>)</span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a>        key <span class="op">=</span> os.path.splitext(key)[<span class="dv">0</span>]</span>
<span id="cb18-20"><a href="#cb18-20" aria-hidden="true" tabindex="-1"></a>        edge_df[key] <span class="op">=</span> pd.read_csv(filepath)</span>
<span id="cb18-21"><a href="#cb18-21" aria-hidden="true" tabindex="-1"></a>        edge_df[key][<span class="st">'length'</span>] <span class="op">=</span> edge_df[key][<span class="st">'end'</span>] <span class="op">-</span> edge_df[key][<span class="st">'start'</span>]</span>
<span id="cb18-22"><a href="#cb18-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-23"><a href="#cb18-23" aria-hidden="true" tabindex="-1"></a><span class="co"># The `edges` dictionary now contains the DataFrames</span></span>
<span id="cb18-24"><a href="#cb18-24" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(edge_df.keys())</span>
<span id="cb18-25"><a href="#cb18-25" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(edge_df[<span class="st">'fibroblast_100kb_10Mb_edges'</span>].columns)</span>
<span id="cb18-26"><a href="#cb18-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-27"><a href="#cb18-27" aria-hidden="true" tabindex="-1"></a><span class="co">#ech90 = pd.read_csv('../data/ech90_human_Mmul_10.csv')</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>dict_keys(['pachytene_spermatocyte_100kb_arms_smoothed_edges', 'round_spermatid_100kb_arms_smoothed_edges', 'fibroblast_500kb_arms_edges', 'round_spermatid_500kb_10Mb_edges', 'fibroblast_100kb_full_edges', 'round_spermatid_500kb_arms_edges', 'spermatogonia_100kb_arms_smoothed_edges', 'fibroblast_500kb_10Mb_edges', 'round_spermatid_100kb_full_edges', 'spermatogonia_500kb_arms_edges', 'spermatogonia_100kb_full_edges', 'pachytene_spermatocyte_100kb_arms_edges', 'sperm_500kb_full_edges', 'pachytene_spermatocyte_500kb_full_edges', 'sperm_100kb_arms_edges', 'sperm_100kb_arms_smoothed_edges', 'sperm_100kb_10Mb_edges', 'pachytene_spermatocyte_100kb_10Mb_edges', 'fibroblast_100kb_arms_smoothed_edges', 'spermatogonia_500kb_10Mb_edges', 'round_spermatid_100kb_full_smoothed_edges', 'fibroblast_500kb_full_edges', 'round_spermatid_100kb_10Mb_edges', 'pachytene_spermatocyte_100kb_full_smoothed_edges', 'fibroblast_100kb_arms_edges', 'sperm_100kb_10Mb_smoothed_edges', 'round_spermatid_500kb_full_edges', 'fibroblast_100kb_10Mb_edges', 'round_spermatid_100kb_arms_edges', 'fibroblast_100kb_10Mb_smoothed_edges', 'spermatogonia_100kb_full_smoothed_edges', 'spermatogonia_500kb_full_edges', 'spermatogonia_100kb_arms_edges', 'pachytene_spermatocyte_100kb_full_edges', 'sperm_500kb_arms_edges', 'sperm_100kb_full_smoothed_edges', 'pachytene_spermatocyte_100kb_10Mb_smoothed_edges', 'pachytene_spermatocyte_500kb_arms_edges', 'sperm_100kb_full_edges', 'round_spermatid_100kb_10Mb_smoothed_edges', 'spermatogonia_100kb_10Mb_smoothed_edges', 'fibroblast_100kb_full_smoothed_edges', 'pachytene_spermatocyte_500kb_10Mb_edges', 'sperm_500kb_10Mb_edges', 'spermatogonia_100kb_10Mb_edges'])
Index(['start', 'end', 'chrom', 'resolution', 'length'], dtype='object')</code></pre>
</div>
</div>
<div id="cell-23" class="cell" data-execution_count="22">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> genominterv <span class="im">import</span> interval_intersect</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>samples <span class="op">=</span> dataframes.keys()</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a><span class="co">#samples = ['round_spermatid_100kb_arms', 'round_spermatid_100kb_10Mb', 'sperm_100kb_arms', 'sperm_100kb_10Mb']</span></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>comps <span class="op">=</span> {key:dataframes[key] <span class="cf">for</span> key <span class="kw">in</span> samples}</span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>edges <span class="op">=</span> {key:edge_df[<span class="ss">f'</span><span class="sc">{</span>key<span class="sc">}</span><span class="ss">_edges'</span>] <span class="cf">for</span> key <span class="kw">in</span> samples}</span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a>comp_intersects <span class="op">=</span> {key:interval_intersect(comp, ech90).assign(length <span class="op">=</span> <span class="kw">lambda</span> x: x[<span class="st">'end'</span>] <span class="op">-</span> x[<span class="st">'start'</span>]) <span class="cf">for</span> key, comp <span class="kw">in</span> comps.items()}</span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a>edge_intersects <span class="op">=</span> {key:interval_intersect(edge, ech90).assign(length <span class="op">=</span> <span class="kw">lambda</span> x: x[<span class="st">'end'</span>] <span class="op">-</span> x[<span class="st">'start'</span>]) <span class="cf">for</span> key, edge <span class="kw">in</span> edges.items()}</span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-15"><a href="#cb20-15" aria-hidden="true" tabindex="-1"></a><span class="co">## For a single sample only ##</span></span>
<span id="cb20-16"><a href="#cb20-16" aria-hidden="true" tabindex="-1"></a><span class="co"># sample = 'round_spermatid_100kb_arms'</span></span>
<span id="cb20-17"><a href="#cb20-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-18"><a href="#cb20-18" aria-hidden="true" tabindex="-1"></a><span class="co"># full_df = dataframes[sample]</span></span>
<span id="cb20-19"><a href="#cb20-19" aria-hidden="true" tabindex="-1"></a><span class="co"># full_intersect = interval_intersect(full_df, ech90).assign(length=lambda x: x['end'] - x['start'])</span></span>
<span id="cb20-20"><a href="#cb20-20" aria-hidden="true" tabindex="-1"></a><span class="co"># edge_intersect = interval_intersect(edge_df[f'{sample}_edges'], ech90).assign(length=lambda x: x['end'] - x['start'])</span></span>
<span id="cb20-21"><a href="#cb20-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-22"><a href="#cb20-22" aria-hidden="true" tabindex="-1"></a><span class="co"># edge_df = edge_df[f'{sample}_edges']</span></span>
<span id="cb20-23"><a href="#cb20-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-24"><a href="#cb20-24" aria-hidden="true" tabindex="-1"></a><span class="co"># # Plot full</span></span>
<span id="cb20-25"><a href="#cb20-25" aria-hidden="true" tabindex="-1"></a><span class="co"># plot_regions(ech90, full_df, full_intersect)</span></span>
<span id="cb20-26"><a href="#cb20-26" aria-hidden="true" tabindex="-1"></a><span class="co"># plt.suptitle('All edges')</span></span>
<span id="cb20-27"><a href="#cb20-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-28"><a href="#cb20-28" aria-hidden="true" tabindex="-1"></a><span class="co"># # Plot edge</span></span>
<span id="cb20-29"><a href="#cb20-29" aria-hidden="true" tabindex="-1"></a><span class="co"># plot_regions(ech90, edge_df, edge_intersect)</span></span>
<span id="cb20-30"><a href="#cb20-30" aria-hidden="true" tabindex="-1"></a><span class="co"># plt.suptitle('Edges only')</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div id="cell-24" class="cell" data-execution_count="23">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co">### Some stats about the data and intersections </span><span class="al">###</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>stats <span class="op">=</span> pd.DataFrame({</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>    <span class="st">'Sample'</span>: samples,</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>    <span class="st">'Total regions'</span>: [comp.shape[<span class="dv">0</span>] <span class="cf">for</span> comp <span class="kw">in</span> comps.values()],</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>    <span class="st">'Total regions on ECH90'</span>: [comp_intersect.shape[<span class="dv">0</span>] <span class="cf">for</span> comp_intersect <span class="kw">in</span> comp_intersects.values()],</span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>    <span class="st">'Total regions on edges'</span>: [edge_intersect.shape[<span class="dv">0</span>] <span class="cf">for</span> edge_intersect <span class="kw">in</span> edge_intersects.values()],</span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>    <span class="st">'Prop regions on ECH90'</span>: [comp_intersect.shape[<span class="dv">0</span>] <span class="op">/</span> ech90.shape[<span class="dv">0</span>] <span class="cf">for</span> comp, comp_intersect <span class="kw">in</span> <span class="bu">zip</span>(comps.values(), comp_intersects.values())],</span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>    <span class="st">'Total bp'</span>: [comp[<span class="st">'length'</span>].<span class="bu">sum</span>() <span class="cf">for</span> comp <span class="kw">in</span> comps.values()],</span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a>    <span class="st">'Total bp on ECH90'</span>: [comp_intersect[<span class="st">'length'</span>].<span class="bu">sum</span>() <span class="cf">for</span> comp_intersect <span class="kw">in</span> comp_intersects.values()],</span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>    <span class="st">'Total bp on edges'</span>: [edge_intersect[<span class="st">'length'</span>].<span class="bu">sum</span>() <span class="cf">for</span> edge_intersect <span class="kw">in</span> edge_intersects.values()],</span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a>    <span class="st">'Prop bp on ECH90'</span>: [comp_intersect[<span class="st">'length'</span>].<span class="bu">sum</span>() <span class="op">/</span> ech90[<span class="st">'length'</span>].<span class="bu">sum</span>() <span class="cf">for</span> comp, comp_intersect <span class="kw">in</span> <span class="bu">zip</span>(comps.values(), comp_intersects.values())],</span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a><span class="co">#df[['Source', 'Rest']] = df['Sample'].str.extract(r'^(.*?)(_\d+\w+.*)')</span></span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a>stats[[<span class="st">'source'</span>, <span class="st">'res'</span>, <span class="st">'view'</span>, <span class="st">'smoothed'</span>]] <span class="op">=</span> stats[<span class="st">'Sample'</span>].<span class="bu">str</span>.extract(<span class="vs">r'^(.*?)_(100kb|500kb)_(full|arms|10Mb)(?:_(smoothed))?$'</span>)</span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a>stats[<span class="st">'smoothed'</span>] <span class="op">=</span> stats[<span class="st">'smoothed'</span>].notna() <span class="co"># Convert to boolean</span></span>
<span id="cb21-19"><a href="#cb21-19" aria-hidden="true" tabindex="-1"></a>stats[[<span class="st">'Sample'</span>,<span class="st">'source'</span>, <span class="st">'res'</span>, <span class="st">'view'</span>, <span class="st">'smoothed'</span>,<span class="st">'Prop regions on ECH90'</span>, <span class="st">'Prop bp on ECH90'</span>]]</span>
<span id="cb21-20"><a href="#cb21-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-21"><a href="#cb21-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-22"><a href="#cb21-22" aria-hidden="true" tabindex="-1"></a><span class="co"># # Determine the proportion of total regions in ECH90 that lies on compartments</span></span>
<span id="cb21-23"><a href="#cb21-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-24"><a href="#cb21-24" aria-hidden="true" tabindex="-1"></a><span class="co"># print("Proportion of ECH90 on: (#count)")</span></span>
<span id="cb21-25"><a href="#cb21-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-26"><a href="#cb21-26" aria-hidden="true" tabindex="-1"></a><span class="co"># # Proportion of ECH90 on full regions</span></span>
<span id="cb21-27"><a href="#cb21-27" aria-hidden="true" tabindex="-1"></a><span class="co"># print(f'\tFull: {full_intersect.shape[0] / ech90.shape[0]}')</span></span>
<span id="cb21-28"><a href="#cb21-28" aria-hidden="true" tabindex="-1"></a><span class="co"># print(f'\tEdge: {edge_intersect.shape[0] / ech90.shape[0]}')</span></span>
<span id="cb21-29"><a href="#cb21-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-30"><a href="#cb21-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-31"><a href="#cb21-31" aria-hidden="true" tabindex="-1"></a><span class="co"># print("\nProportion of ECH90: (#bpairs)")</span></span>
<span id="cb21-32"><a href="#cb21-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-33"><a href="#cb21-33" aria-hidden="true" tabindex="-1"></a><span class="co"># # Proportion of ECH90 on full regions</span></span>
<span id="cb21-34"><a href="#cb21-34" aria-hidden="true" tabindex="-1"></a><span class="co"># print(f'\tFull: {full_intersect['length'].sum() / ech90['length'].sum()}')</span></span>
<span id="cb21-35"><a href="#cb21-35" aria-hidden="true" tabindex="-1"></a><span class="co"># print(f'\tEdge: {edge_intersect['length'].sum() / ech90['length'].sum()}')</span></span>
<span id="cb21-36"><a href="#cb21-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-37"><a href="#cb21-37" aria-hidden="true" tabindex="-1"></a><span class="co"># # Determine the proportion of total regions in ECH90 that lies on compartment edges</span></span>
<span id="cb21-38"><a href="#cb21-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-39"><a href="#cb21-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-40"><a href="#cb21-40" aria-hidden="true" tabindex="-1"></a><span class="co"># # What is the total length of ech90 regions and edges</span></span>
<span id="cb21-41"><a href="#cb21-41" aria-hidden="true" tabindex="-1"></a><span class="co"># print("\nTotal length of:")</span></span>
<span id="cb21-42"><a href="#cb21-42" aria-hidden="true" tabindex="-1"></a><span class="co"># print(f'\tECH90:   {(ech90['end'] - ech90['start']).sum()} bp')</span></span>
<span id="cb21-43"><a href="#cb21-43" aria-hidden="true" tabindex="-1"></a><span class="co"># print(f'\tEdge:   {(edge_df["end"] - edge_df["start"]).sum()} bp')</span></span>
<span id="cb21-44"><a href="#cb21-44" aria-hidden="true" tabindex="-1"></a><span class="co"># print(f'\tFull:   {(full_df["end"] - full_df["start"]).sum()} bp')</span></span>
<span id="cb21-45"><a href="#cb21-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-46"><a href="#cb21-46" aria-hidden="true" tabindex="-1"></a><span class="co">## Plot</span></span>
<span id="cb21-47"><a href="#cb21-47" aria-hidden="true" tabindex="-1"></a><span class="co"># f, ax = plot_regions(ech90, edge_df, full_intersect)</span></span>
<span id="cb21-48"><a href="#cb21-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-49"><a href="#cb21-49" aria-hidden="true" tabindex="-1"></a><span class="co"># for ax in list(f.axes):</span></span>
<span id="cb21-50"><a href="#cb21-50" aria-hidden="true" tabindex="-1"></a><span class="co">#     if ax.get_ylabel() == 'ECH90':</span></span>
<span id="cb21-51"><a href="#cb21-51" aria-hidden="true" tabindex="-1"></a><span class="co">#         #ax.set_ylabel('Edges', rotation=0, fontsize=10, labelpad=30)</span></span>
<span id="cb21-52"><a href="#cb21-52" aria-hidden="true" tabindex="-1"></a><span class="co">#         pass</span></span>
<span id="cb21-53"><a href="#cb21-53" aria-hidden="true" tabindex="-1"></a><span class="co">#     if ax.get_ylabel() in ("query", "A-Comp"):</span></span>
<span id="cb21-54"><a href="#cb21-54" aria-hidden="true" tabindex="-1"></a><span class="co">#         ax.set_ylabel('Edges', rotation=0, fontsize=10, labelpad=30)</span></span>
<span id="cb21-55"><a href="#cb21-55" aria-hidden="true" tabindex="-1"></a><span class="co">#     if ax.get_ylabel() == 'Intersect':</span></span>
<span id="cb21-56"><a href="#cb21-56" aria-hidden="true" tabindex="-1"></a><span class="co">#         #ax.set_ylabel('end_edge \nintersect', rotation=0, fontsize=10, labelpad=30)</span></span>
<span id="cb21-57"><a href="#cb21-57" aria-hidden="true" tabindex="-1"></a><span class="co">#         pass</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="23">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">Sample</th>
<th data-quarto-table-cell-role="th">source</th>
<th data-quarto-table-cell-role="th">res</th>
<th data-quarto-table-cell-role="th">view</th>
<th data-quarto-table-cell-role="th">smoothed</th>
<th data-quarto-table-cell-role="th">Prop regions on ECH90</th>
<th data-quarto-table-cell-role="th">Prop bp on ECH90</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>spermatogonia_500kb_arms</td>
<td>spermatogonia</td>
<td>500kb</td>
<td>arms</td>
<td>False</td>
<td>0.421053</td>
<td>0.489351</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>round_spermatid_500kb_full</td>
<td>round_spermatid</td>
<td>500kb</td>
<td>full</td>
<td>False</td>
<td>0.368421</td>
<td>0.418981</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>fibroblast_500kb_full</td>
<td>fibroblast</td>
<td>500kb</td>
<td>full</td>
<td>False</td>
<td>0.684211</td>
<td>0.529916</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3</td>
<td>pachytene_spermatocyte_100kb_arms_smoothed</td>
<td>pachytene_spermatocyte</td>
<td>100kb</td>
<td>arms</td>
<td>True</td>
<td>0.684211</td>
<td>0.632145</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">4</td>
<td>pachytene_spermatocyte_500kb_full</td>
<td>pachytene_spermatocyte</td>
<td>500kb</td>
<td>full</td>
<td>False</td>
<td>0.368421</td>
<td>0.362651</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">5</td>
<td>pachytene_spermatocyte_100kb_full</td>
<td>pachytene_spermatocyte</td>
<td>100kb</td>
<td>full</td>
<td>False</td>
<td>0.473684</td>
<td>0.365265</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">6</td>
<td>round_spermatid_100kb_arms_smoothed</td>
<td>round_spermatid</td>
<td>100kb</td>
<td>arms</td>
<td>True</td>
<td>0.631579</td>
<td>0.412828</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">7</td>
<td>fibroblast_100kb_arms_smoothed</td>
<td>fibroblast</td>
<td>100kb</td>
<td>arms</td>
<td>True</td>
<td>0.684211</td>
<td>0.499297</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">8</td>
<td>fibroblast_100kb_full</td>
<td>fibroblast</td>
<td>100kb</td>
<td>full</td>
<td>False</td>
<td>0.842105</td>
<td>0.550167</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">9</td>
<td>round_spermatid_100kb_full</td>
<td>round_spermatid</td>
<td>100kb</td>
<td>full</td>
<td>False</td>
<td>0.421053</td>
<td>0.472204</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">10</td>
<td>spermatogonia_100kb_arms</td>
<td>spermatogonia</td>
<td>100kb</td>
<td>arms</td>
<td>False</td>
<td>0.526316</td>
<td>0.550008</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">11</td>
<td>spermatogonia_100kb_10Mb</td>
<td>spermatogonia</td>
<td>100kb</td>
<td>10Mb</td>
<td>False</td>
<td>0.631579</td>
<td>0.567910</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">12</td>
<td>sperm_100kb_full</td>
<td>sperm</td>
<td>100kb</td>
<td>full</td>
<td>False</td>
<td>0.631579</td>
<td>0.674746</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">13</td>
<td>sperm_100kb_arms_smoothed</td>
<td>sperm</td>
<td>100kb</td>
<td>arms</td>
<td>True</td>
<td>0.578947</td>
<td>0.572887</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">14</td>
<td>spermatogonia_500kb_10Mb</td>
<td>spermatogonia</td>
<td>500kb</td>
<td>10Mb</td>
<td>False</td>
<td>0.684211</td>
<td>0.689415</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">15</td>
<td>sperm_500kb_full</td>
<td>sperm</td>
<td>500kb</td>
<td>full</td>
<td>False</td>
<td>0.631579</td>
<td>0.559567</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">16</td>
<td>spermatogonia_100kb_arms_smoothed</td>
<td>spermatogonia</td>
<td>100kb</td>
<td>arms</td>
<td>True</td>
<td>0.526316</td>
<td>0.549381</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">17</td>
<td>pachytene_spermatocyte_100kb_10Mb</td>
<td>pachytene_spermatocyte</td>
<td>100kb</td>
<td>10Mb</td>
<td>False</td>
<td>0.789474</td>
<td>0.630109</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">18</td>
<td>round_spermatid_100kb_10Mb</td>
<td>round_spermatid</td>
<td>100kb</td>
<td>10Mb</td>
<td>False</td>
<td>0.684211</td>
<td>0.519772</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">19</td>
<td>round_spermatid_100kb_10Mb_smoothed</td>
<td>round_spermatid</td>
<td>100kb</td>
<td>10Mb</td>
<td>True</td>
<td>0.578947</td>
<td>0.483689</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">20</td>
<td>fibroblast_100kb_10Mb</td>
<td>fibroblast</td>
<td>100kb</td>
<td>10Mb</td>
<td>False</td>
<td>0.684211</td>
<td>0.494509</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">21</td>
<td>fibroblast_100kb_10Mb_smoothed</td>
<td>fibroblast</td>
<td>100kb</td>
<td>10Mb</td>
<td>True</td>
<td>0.526316</td>
<td>0.414530</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">22</td>
<td>fibroblast_100kb_full_smoothed</td>
<td>fibroblast</td>
<td>100kb</td>
<td>full</td>
<td>True</td>
<td>0.736842</td>
<td>0.591105</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">23</td>
<td>sperm_100kb_arms</td>
<td>sperm</td>
<td>100kb</td>
<td>arms</td>
<td>False</td>
<td>0.736842</td>
<td>0.578219</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">24</td>
<td>round_spermatid_100kb_full_smoothed</td>
<td>round_spermatid</td>
<td>100kb</td>
<td>full</td>
<td>True</td>
<td>0.368421</td>
<td>0.418981</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">25</td>
<td>fibroblast_500kb_10Mb</td>
<td>fibroblast</td>
<td>500kb</td>
<td>10Mb</td>
<td>False</td>
<td>0.473684</td>
<td>0.359088</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">26</td>
<td>round_spermatid_500kb_10Mb</td>
<td>round_spermatid</td>
<td>500kb</td>
<td>10Mb</td>
<td>False</td>
<td>0.578947</td>
<td>0.496303</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">27</td>
<td>sperm_500kb_arms</td>
<td>sperm</td>
<td>500kb</td>
<td>arms</td>
<td>False</td>
<td>0.421053</td>
<td>0.395138</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">28</td>
<td>pachytene_spermatocyte_100kb_10Mb_smoothed</td>
<td>pachytene_spermatocyte</td>
<td>100kb</td>
<td>10Mb</td>
<td>True</td>
<td>0.631579</td>
<td>0.665339</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">29</td>
<td>pachytene_spermatocyte_500kb_10Mb</td>
<td>pachytene_spermatocyte</td>
<td>500kb</td>
<td>10Mb</td>
<td>False</td>
<td>0.631579</td>
<td>0.538153</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">30</td>
<td>pachytene_spermatocyte_100kb_full_smoothed</td>
<td>pachytene_spermatocyte</td>
<td>100kb</td>
<td>full</td>
<td>True</td>
<td>0.368421</td>
<td>0.401552</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">31</td>
<td>round_spermatid_500kb_arms</td>
<td>round_spermatid</td>
<td>500kb</td>
<td>arms</td>
<td>False</td>
<td>0.684211</td>
<td>0.655953</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">32</td>
<td>spermatogonia_500kb_full</td>
<td>spermatogonia</td>
<td>500kb</td>
<td>full</td>
<td>False</td>
<td>0.421053</td>
<td>0.392944</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">33</td>
<td>fibroblast_500kb_arms</td>
<td>fibroblast</td>
<td>500kb</td>
<td>arms</td>
<td>False</td>
<td>0.684211</td>
<td>0.550729</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">34</td>
<td>spermatogonia_100kb_10Mb_smoothed</td>
<td>spermatogonia</td>
<td>100kb</td>
<td>10Mb</td>
<td>True</td>
<td>0.631579</td>
<td>0.580640</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">35</td>
<td>spermatogonia_100kb_full_smoothed</td>
<td>spermatogonia</td>
<td>100kb</td>
<td>full</td>
<td>True</td>
<td>0.368421</td>
<td>0.386720</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">36</td>
<td>sperm_500kb_10Mb</td>
<td>sperm</td>
<td>500kb</td>
<td>10Mb</td>
<td>False</td>
<td>0.526316</td>
<td>0.438818</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">37</td>
<td>pachytene_spermatocyte_500kb_arms</td>
<td>pachytene_spermatocyte</td>
<td>500kb</td>
<td>arms</td>
<td>False</td>
<td>0.631579</td>
<td>0.650289</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">38</td>
<td>pachytene_spermatocyte_100kb_arms</td>
<td>pachytene_spermatocyte</td>
<td>100kb</td>
<td>arms</td>
<td>False</td>
<td>0.789474</td>
<td>0.584619</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">39</td>
<td>sperm_100kb_10Mb_smoothed</td>
<td>sperm</td>
<td>100kb</td>
<td>10Mb</td>
<td>True</td>
<td>0.578947</td>
<td>0.549366</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">40</td>
<td>sperm_100kb_full_smoothed</td>
<td>sperm</td>
<td>100kb</td>
<td>full</td>
<td>True</td>
<td>0.578947</td>
<td>0.677841</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">41</td>
<td>fibroblast_100kb_arms</td>
<td>fibroblast</td>
<td>100kb</td>
<td>arms</td>
<td>False</td>
<td>0.736842</td>
<td>0.510641</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">42</td>
<td>spermatogonia_100kb_full</td>
<td>spermatogonia</td>
<td>100kb</td>
<td>full</td>
<td>False</td>
<td>0.473684</td>
<td>0.374801</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">43</td>
<td>round_spermatid_100kb_arms</td>
<td>round_spermatid</td>
<td>100kb</td>
<td>arms</td>
<td>False</td>
<td>0.789474</td>
<td>0.500366</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">44</td>
<td>sperm_100kb_10Mb</td>
<td>sperm</td>
<td>100kb</td>
<td>10Mb</td>
<td>False</td>
<td>0.684211</td>
<td>0.545798</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<div id="cell-25" class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>        </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="do-testing-of-the-edges" class="level3">
<h3 class="anchored" data-anchor-id="do-testing-of-the-edges">Do testing of the edges</h3>
<div id="cell-27" class="cell" data-execution_count="5">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="op">%%</span>capture</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Define what we are testing</span></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"""</span></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a><span class="st">Goal: To test whether ECH90 regions are enriched in compartment edges</span></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a><span class="st">Query: ECH90</span></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a><span class="st">Annotation: Start and end edges of compartments</span></span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a><span class="st">Hypothesis: </span></span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a><span class="st">      ECH90 regions are enriched in compartment edges</span></span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a><span class="st">Null hypothesis: </span></span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a><span class="st">      ECH90 regions are not enriched in compartment edges</span></span>
<span id="cb23-13"><a href="#cb23-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-14"><a href="#cb23-14" aria-hidden="true" tabindex="-1"></a><span class="st">Tests: </span></span>
<span id="cb23-15"><a href="#cb23-15" aria-hidden="true" tabindex="-1"></a><span class="st">* Proximity test: </span></span>
<span id="cb23-16"><a href="#cb23-16" aria-hidden="true" tabindex="-1"></a><span class="st">      tests whether the query regions are closer to </span></span>
<span id="cb23-17"><a href="#cb23-17" aria-hidden="true" tabindex="-1"></a><span class="st">      the annotation regions than expected by chance. </span></span>
<span id="cb23-18"><a href="#cb23-18" aria-hidden="true" tabindex="-1"></a><span class="st">      NB regions can not overlap, so we need to collapse the annotation regions</span></span>
<span id="cb23-19"><a href="#cb23-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-20"><a href="#cb23-20" aria-hidden="true" tabindex="-1"></a><span class="st">* Jaccard index: </span></span>
<span id="cb23-21"><a href="#cb23-21" aria-hidden="true" tabindex="-1"></a><span class="st">      tests the similarity between the query and annotation regions, </span></span>
<span id="cb23-22"><a href="#cb23-22" aria-hidden="true" tabindex="-1"></a><span class="st">      where a value of 1 indicates perfect overlap</span></span>
<span id="cb23-23"><a href="#cb23-23" aria-hidden="true" tabindex="-1"></a><span class="st">"""</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div id="cell-28" class="cell" data-execution_count="6">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co">### Proximity test </span><span class="al">###</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os.path <span class="im">as</span> op</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> genominterv <span class="im">import</span> proximity_test, interval_diff</span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>proximity_file <span class="op">=</span> <span class="st">'../results/proximity_test_results.csv'</span></span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> <span class="kw">not</span> op.exists(proximity_file):</span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>    proximity_res <span class="op">=</span> pd.DataFrame({})</span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> sample <span class="kw">in</span> samples:</span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>        query_comp <span class="op">=</span> ech90</span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a>        query_edge <span class="op">=</span> ech90</span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a>        annot_comp <span class="op">=</span> comps[sample]</span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a>        annot_edge <span class="op">=</span> edges[sample]</span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Get non-overlapping query regions</span></span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a>        non_ovl_query_comp <span class="op">=</span> interval_diff(query_comp, annot_comp)</span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a>        non_ovl_query_edge <span class="op">=</span> interval_diff(query_edge, annot_edge)</span>
<span id="cb24-19"><a href="#cb24-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-20"><a href="#cb24-20" aria-hidden="true" tabindex="-1"></a>        proximity_comp <span class="op">=</span> proximity_test(non_ovl_query_comp, annot_comp, two_sided<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb24-21"><a href="#cb24-21" aria-hidden="true" tabindex="-1"></a>        proximity_edge <span class="op">=</span> proximity_test(non_ovl_query_edge, annot_edge, two_sided<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb24-22"><a href="#cb24-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-23"><a href="#cb24-23" aria-hidden="true" tabindex="-1"></a>        proximity_res <span class="op">=</span> pd.concat([proximity_res, pd.DataFrame({</span>
<span id="cb24-24"><a href="#cb24-24" aria-hidden="true" tabindex="-1"></a>            <span class="st">'Sample'</span>: sample,</span>
<span id="cb24-25"><a href="#cb24-25" aria-hidden="true" tabindex="-1"></a>            <span class="st">'Query'</span>: <span class="st">'ECH90'</span>,</span>
<span id="cb24-26"><a href="#cb24-26" aria-hidden="true" tabindex="-1"></a>            <span class="st">'Comp Statistic'</span>: proximity_comp.statistic,</span>
<span id="cb24-27"><a href="#cb24-27" aria-hidden="true" tabindex="-1"></a>            <span class="st">'Edge Statistic'</span>: proximity_edge.statistic,</span>
<span id="cb24-28"><a href="#cb24-28" aria-hidden="true" tabindex="-1"></a>            <span class="st">'Comp P-value'</span>: proximity_comp.pvalue,</span>
<span id="cb24-29"><a href="#cb24-29" aria-hidden="true" tabindex="-1"></a>            <span class="st">'Edge P-value'</span>: proximity_edge.pvalue</span>
<span id="cb24-30"><a href="#cb24-30" aria-hidden="true" tabindex="-1"></a>        }, index<span class="op">=</span>[<span class="dv">0</span>])</span>
<span id="cb24-31"><a href="#cb24-31" aria-hidden="true" tabindex="-1"></a>        ])</span>
<span id="cb24-32"><a href="#cb24-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-33"><a href="#cb24-33" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Swap query and annotation</span></span>
<span id="cb24-34"><a href="#cb24-34" aria-hidden="true" tabindex="-1"></a>        query_comp, annot_comp <span class="op">=</span> annot_comp, query_comp</span>
<span id="cb24-35"><a href="#cb24-35" aria-hidden="true" tabindex="-1"></a>        query_edge, annot_edge <span class="op">=</span> annot_edge, query_edge</span>
<span id="cb24-36"><a href="#cb24-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-37"><a href="#cb24-37" aria-hidden="true" tabindex="-1"></a>        non_ovl_query_comp <span class="op">=</span> interval_diff(query_comp, annot_comp)</span>
<span id="cb24-38"><a href="#cb24-38" aria-hidden="true" tabindex="-1"></a>        non_ovl_query_edge <span class="op">=</span> interval_diff(query_edge, annot_edge)</span>
<span id="cb24-39"><a href="#cb24-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-40"><a href="#cb24-40" aria-hidden="true" tabindex="-1"></a>        proximity_comp <span class="op">=</span> proximity_test(non_ovl_query_comp, annot_comp, two_sided<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb24-41"><a href="#cb24-41" aria-hidden="true" tabindex="-1"></a>        proximity_edge <span class="op">=</span> proximity_test(non_ovl_query_edge, annot_edge, two_sided<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb24-42"><a href="#cb24-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-43"><a href="#cb24-43" aria-hidden="true" tabindex="-1"></a>        proximity_res <span class="op">=</span> pd.concat([proximity_res, pd.DataFrame({</span>
<span id="cb24-44"><a href="#cb24-44" aria-hidden="true" tabindex="-1"></a>            <span class="st">'Sample'</span>: sample,</span>
<span id="cb24-45"><a href="#cb24-45" aria-hidden="true" tabindex="-1"></a>            <span class="st">'Query'</span>: <span class="st">'Edge'</span>,</span>
<span id="cb24-46"><a href="#cb24-46" aria-hidden="true" tabindex="-1"></a>            <span class="st">'Comp Statistic'</span>: proximity_comp.statistic,</span>
<span id="cb24-47"><a href="#cb24-47" aria-hidden="true" tabindex="-1"></a>            <span class="st">'Edge Statistic'</span>: proximity_edge.statistic,</span>
<span id="cb24-48"><a href="#cb24-48" aria-hidden="true" tabindex="-1"></a>            <span class="st">'Comp P-value'</span>: proximity_comp.pvalue,</span>
<span id="cb24-49"><a href="#cb24-49" aria-hidden="true" tabindex="-1"></a>            <span class="st">'Edge P-value'</span>: proximity_edge.pvalue</span>
<span id="cb24-50"><a href="#cb24-50" aria-hidden="true" tabindex="-1"></a>        }, index<span class="op">=</span>[<span class="dv">0</span>])</span>
<span id="cb24-51"><a href="#cb24-51" aria-hidden="true" tabindex="-1"></a>        ])</span>
<span id="cb24-52"><a href="#cb24-52" aria-hidden="true" tabindex="-1"></a>    proximity_res.to_csv(proximity_file, index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb24-53"><a href="#cb24-53" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>: </span>
<span id="cb24-54"><a href="#cb24-54" aria-hidden="true" tabindex="-1"></a>    proximity_res <span class="op">=</span> pd.read_csv(proximity_file)</span>
<span id="cb24-55"><a href="#cb24-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-56"><a href="#cb24-56" aria-hidden="true" tabindex="-1"></a><span class="co">## For a single sample only ##</span></span>
<span id="cb24-57"><a href="#cb24-57" aria-hidden="true" tabindex="-1"></a><span class="co"># # Define the query and annotation</span></span>
<span id="cb24-58"><a href="#cb24-58" aria-hidden="true" tabindex="-1"></a><span class="co"># query = ech90</span></span>
<span id="cb24-59"><a href="#cb24-59" aria-hidden="true" tabindex="-1"></a><span class="co"># annot = edge_df</span></span>
<span id="cb24-60"><a href="#cb24-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-61"><a href="#cb24-61" aria-hidden="true" tabindex="-1"></a><span class="co"># # display(interval_diff(query, annot), interval_diff(annot, query))</span></span>
<span id="cb24-62"><a href="#cb24-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-63"><a href="#cb24-63" aria-hidden="true" tabindex="-1"></a><span class="co"># # Calculate the non-overlapping query regions</span></span>
<span id="cb24-64"><a href="#cb24-64" aria-hidden="true" tabindex="-1"></a><span class="co"># non_ovl_query_full = interval_diff(query, annot)</span></span>
<span id="cb24-65"><a href="#cb24-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-66"><a href="#cb24-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-67"><a href="#cb24-67" aria-hidden="true" tabindex="-1"></a><span class="co"># # Perform the proximity test</span></span>
<span id="cb24-68"><a href="#cb24-68" aria-hidden="true" tabindex="-1"></a><span class="co"># proximity_full = proximity_test(non_ovl_query_full, annot, two_sided=True)</span></span>
<span id="cb24-69"><a href="#cb24-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-70"><a href="#cb24-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-71"><a href="#cb24-71" aria-hidden="true" tabindex="-1"></a><span class="co"># print("Proximity test results: query=ECH90, annotation=edge_df")</span></span>
<span id="cb24-72"><a href="#cb24-72" aria-hidden="true" tabindex="-1"></a><span class="co"># print("non_ovl_query: the parts of ECH90 that do not overlap with edge_df")</span></span>
<span id="cb24-73"><a href="#cb24-73" aria-hidden="true" tabindex="-1"></a><span class="co"># print(f"\tstatistic: {proximity_full.statistic}, \n\tp-value: {proximity_full.pvalue}")</span></span>
<span id="cb24-74"><a href="#cb24-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-75"><a href="#cb24-75" aria-hidden="true" tabindex="-1"></a><span class="co"># query, annot = annot, query</span></span>
<span id="cb24-76"><a href="#cb24-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-77"><a href="#cb24-77" aria-hidden="true" tabindex="-1"></a><span class="co"># # Calculate the non-overlapping query regions</span></span>
<span id="cb24-78"><a href="#cb24-78" aria-hidden="true" tabindex="-1"></a><span class="co"># non_ovl_query_full = interval_diff(query, annot)</span></span>
<span id="cb24-79"><a href="#cb24-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-80"><a href="#cb24-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-81"><a href="#cb24-81" aria-hidden="true" tabindex="-1"></a><span class="co"># print("\nProximity test results: query=edge_df, annotation=ECH90")</span></span>
<span id="cb24-82"><a href="#cb24-82" aria-hidden="true" tabindex="-1"></a><span class="co"># print("non_ovl_query: the parts of edge_df that do not overlap with ECH90")</span></span>
<span id="cb24-83"><a href="#cb24-83" aria-hidden="true" tabindex="-1"></a><span class="co"># proximity_full = proximity_test(non_ovl_query_full, annot, two_sided=True)</span></span>
<span id="cb24-84"><a href="#cb24-84" aria-hidden="true" tabindex="-1"></a><span class="co"># print(f"\tstatistic: {proximity_full.statistic}, \n\tp-value: {proximity_full.pvalue}")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Parallelize the proximity ttest, set samples = 100_000</p>
<div id="cell-30" class="cell" data-execution_count="6">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> joblib <span class="im">import</span> Parallel, delayed</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> tqdm <span class="im">import</span> tqdm</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os.path <span class="im">as</span> op</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> genominterv <span class="im">import</span> proximity_test, interval_diff</span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> process_proximity(sample, nsamples <span class="op">=</span> <span class="dv">100000</span>):</span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>    query_comp <span class="op">=</span> ech90</span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>    query_edge <span class="op">=</span> ech90</span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>    annot_comp <span class="op">=</span> comps[sample]</span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a>    annot_edge <span class="op">=</span> edges[sample]</span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Get non-overlapping query regions</span></span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a>    non_ovl_query_comp <span class="op">=</span> interval_diff(query_comp, annot_comp)</span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a>    non_ovl_query_edge <span class="op">=</span> interval_diff(query_edge, annot_edge)</span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a>    proximity_comp <span class="op">=</span> proximity_test(non_ovl_query_comp, annot_comp, nsamples, two_sided<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb25-18"><a href="#cb25-18" aria-hidden="true" tabindex="-1"></a>    proximity_edge <span class="op">=</span> proximity_test(non_ovl_query_edge, annot_edge, nsamples, two_sided<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb25-19"><a href="#cb25-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-20"><a href="#cb25-20" aria-hidden="true" tabindex="-1"></a>    results <span class="op">=</span> [</span>
<span id="cb25-21"><a href="#cb25-21" aria-hidden="true" tabindex="-1"></a>        {<span class="st">'Sample'</span>: sample, <span class="st">'Query'</span>: <span class="st">'ECH90'</span>, <span class="st">'Comp Statistic'</span>: proximity_comp.statistic, </span>
<span id="cb25-22"><a href="#cb25-22" aria-hidden="true" tabindex="-1"></a>         <span class="st">'Edge Statistic'</span>: proximity_edge.statistic, <span class="st">'Comp P-value'</span>: proximity_comp.pvalue, </span>
<span id="cb25-23"><a href="#cb25-23" aria-hidden="true" tabindex="-1"></a>         <span class="st">'Edge P-value'</span>: proximity_edge.pvalue}</span>
<span id="cb25-24"><a href="#cb25-24" aria-hidden="true" tabindex="-1"></a>    ]</span>
<span id="cb25-25"><a href="#cb25-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-26"><a href="#cb25-26" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Swap query and annotation</span></span>
<span id="cb25-27"><a href="#cb25-27" aria-hidden="true" tabindex="-1"></a>    query_comp, annot_comp <span class="op">=</span> annot_comp, query_comp</span>
<span id="cb25-28"><a href="#cb25-28" aria-hidden="true" tabindex="-1"></a>    query_edge, annot_edge <span class="op">=</span> annot_edge, query_edge</span>
<span id="cb25-29"><a href="#cb25-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-30"><a href="#cb25-30" aria-hidden="true" tabindex="-1"></a>    non_ovl_query_comp <span class="op">=</span> interval_diff(query_comp, annot_comp)</span>
<span id="cb25-31"><a href="#cb25-31" aria-hidden="true" tabindex="-1"></a>    non_ovl_query_edge <span class="op">=</span> interval_diff(query_edge, annot_edge)</span>
<span id="cb25-32"><a href="#cb25-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-33"><a href="#cb25-33" aria-hidden="true" tabindex="-1"></a>    proximity_comp <span class="op">=</span> proximity_test(non_ovl_query_comp, annot_comp, nsamples, two_sided<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb25-34"><a href="#cb25-34" aria-hidden="true" tabindex="-1"></a>    proximity_edge <span class="op">=</span> proximity_test(non_ovl_query_edge, annot_edge, nsamples, two_sided<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb25-35"><a href="#cb25-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-36"><a href="#cb25-36" aria-hidden="true" tabindex="-1"></a>    results.append(</span>
<span id="cb25-37"><a href="#cb25-37" aria-hidden="true" tabindex="-1"></a>        {<span class="st">'Sample'</span>: sample, <span class="st">'Query'</span>: <span class="st">'Edge'</span>, <span class="st">'Comp Statistic'</span>: proximity_comp.statistic, </span>
<span id="cb25-38"><a href="#cb25-38" aria-hidden="true" tabindex="-1"></a>         <span class="st">'Edge Statistic'</span>: proximity_edge.statistic, <span class="st">'Comp P-value'</span>: proximity_comp.pvalue, </span>
<span id="cb25-39"><a href="#cb25-39" aria-hidden="true" tabindex="-1"></a>         <span class="st">'Edge P-value'</span>: proximity_edge.pvalue}</span>
<span id="cb25-40"><a href="#cb25-40" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb25-41"><a href="#cb25-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-42"><a href="#cb25-42" aria-hidden="true" tabindex="-1"></a>    <span class="cf">return</span> results</span>
<span id="cb25-43"><a href="#cb25-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-44"><a href="#cb25-44" aria-hidden="true" tabindex="-1"></a><span class="co">### End of function definition </span><span class="al">###</span></span>
<span id="cb25-45"><a href="#cb25-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-46"><a href="#cb25-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-47"><a href="#cb25-47" aria-hidden="true" tabindex="-1"></a>nsamples <span class="op">=</span> <span class="dv">100_000</span></span>
<span id="cb25-48"><a href="#cb25-48" aria-hidden="true" tabindex="-1"></a>proximity_file <span class="op">=</span> <span class="ss">f'../results/proximity_test_</span><span class="sc">{</span>nsamples<span class="sc">}</span><span class="ss">.csv'</span></span>
<span id="cb25-49"><a href="#cb25-49" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> <span class="kw">not</span> op.exists(proximity_file):</span>
<span id="cb25-50"><a href="#cb25-50" aria-hidden="true" tabindex="-1"></a>    proximity_res <span class="op">=</span> Parallel(n_jobs<span class="op">=-</span><span class="dv">1</span>)(</span>
<span id="cb25-51"><a href="#cb25-51" aria-hidden="true" tabindex="-1"></a>        delayed(process_proximity)(sample, nsamples) <span class="cf">for</span> sample <span class="kw">in</span> tqdm(samples)</span>
<span id="cb25-52"><a href="#cb25-52" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb25-53"><a href="#cb25-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-54"><a href="#cb25-54" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Flatten the results</span></span>
<span id="cb25-55"><a href="#cb25-55" aria-hidden="true" tabindex="-1"></a>    proximity_res <span class="op">=</span> pd.DataFrame([item <span class="cf">for</span> sublist <span class="kw">in</span> proximity_res <span class="cf">for</span> item <span class="kw">in</span> sublist])</span>
<span id="cb25-56"><a href="#cb25-56" aria-hidden="true" tabindex="-1"></a>    proximity_res.to_csv(proximity_file, index<span class="op">=</span><span class="va">False</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>100%|██████████| 45/45 [00:00&lt;00:00, 2343.88it/s]</code></pre>
</div>
</div>
<div id="cell-31" class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co">#### This will take 20 hours to run with b = 100_000</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a><span class="co">### Jaccard index </span><span class="al">###</span></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os.path <span class="im">as</span> op</span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> genominterv <span class="im">import</span> jaccard_stat, bootstrap</span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> datetime <span class="im">import</span> datetime</span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a>b <span class="op">=</span> <span class="dv">100_000</span></span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a>jaccard_file <span class="op">=</span> <span class="ss">f'../results/jaccard_test_</span><span class="sc">{</span>b<span class="sc">}</span><span class="ss">.csv'</span></span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> <span class="kw">not</span> op.exists(jaccard_file):</span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">@bootstrap</span>(chromsizes, samples<span class="op">=</span>b)</span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a>    <span class="kw">def</span> jaccard_bootstrap(query, annot):</span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a>        <span class="cf">return</span> jaccard_stat(query, annot)</span>
<span id="cb27-14"><a href="#cb27-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-15"><a href="#cb27-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-16"><a href="#cb27-16" aria-hidden="true" tabindex="-1"></a>    jaccard_res <span class="op">=</span> pd.DataFrame({})</span>
<span id="cb27-17"><a href="#cb27-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-18"><a href="#cb27-18" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> sample <span class="kw">in</span> samples:</span>
<span id="cb27-19"><a href="#cb27-19" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Running bootstrap (</span><span class="sc">{</span>b<span class="sc">}</span><span class="ss">) for </span><span class="sc">{</span>sample<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb27-20"><a href="#cb27-20" aria-hidden="true" tabindex="-1"></a>        query_comp <span class="op">=</span> ech90</span>
<span id="cb27-21"><a href="#cb27-21" aria-hidden="true" tabindex="-1"></a>        query_edge <span class="op">=</span> ech90</span>
<span id="cb27-22"><a href="#cb27-22" aria-hidden="true" tabindex="-1"></a>        annot_comp <span class="op">=</span> comps[sample]</span>
<span id="cb27-23"><a href="#cb27-23" aria-hidden="true" tabindex="-1"></a>        annot_edge <span class="op">=</span> edges[sample]</span>
<span id="cb27-24"><a href="#cb27-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-25"><a href="#cb27-25" aria-hidden="true" tabindex="-1"></a>        start <span class="op">=</span> datetime.now()</span>
<span id="cb27-26"><a href="#cb27-26" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Timestamp start Comp: </span><span class="sc">{</span>start<span class="sc">.</span>strftime(<span class="st">"%Y-%m-</span><span class="sc">%d</span><span class="st"> %H:%M:%S"</span>)<span class="sc">}</span><span class="ss">"</span>, end<span class="op">=</span><span class="st">""</span>)</span>
<span id="cb27-27"><a href="#cb27-27" aria-hidden="true" tabindex="-1"></a>        jaccard_comp <span class="op">=</span> jaccard_bootstrap(query_comp, annot_comp)</span>
<span id="cb27-28"><a href="#cb27-28" aria-hidden="true" tabindex="-1"></a>        jaccard_edge <span class="op">=</span> jaccard_bootstrap(query_edge, annot_edge)</span>
<span id="cb27-29"><a href="#cb27-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-30"><a href="#cb27-30" aria-hidden="true" tabindex="-1"></a>        jaccard_res <span class="op">=</span> pd.concat([jaccard_res, pd.DataFrame({</span>
<span id="cb27-31"><a href="#cb27-31" aria-hidden="true" tabindex="-1"></a>            <span class="st">'Sample'</span>: sample,</span>
<span id="cb27-32"><a href="#cb27-32" aria-hidden="true" tabindex="-1"></a>            <span class="st">'Query'</span>: <span class="st">'ECH90'</span>,</span>
<span id="cb27-33"><a href="#cb27-33" aria-hidden="true" tabindex="-1"></a>            <span class="st">'Comp Index'</span>: jaccard_comp[<span class="dv">0</span>],</span>
<span id="cb27-34"><a href="#cb27-34" aria-hidden="true" tabindex="-1"></a>            <span class="st">'Edge Index'</span>: jaccard_edge[<span class="dv">0</span>],</span>
<span id="cb27-35"><a href="#cb27-35" aria-hidden="true" tabindex="-1"></a>            <span class="st">'Comp P-value'</span>: jaccard_comp[<span class="dv">1</span>],</span>
<span id="cb27-36"><a href="#cb27-36" aria-hidden="true" tabindex="-1"></a>            <span class="st">'Edge P-value'</span>: jaccard_edge[<span class="dv">1</span>]</span>
<span id="cb27-37"><a href="#cb27-37" aria-hidden="true" tabindex="-1"></a>        }, index<span class="op">=</span>[<span class="dv">0</span>])</span>
<span id="cb27-38"><a href="#cb27-38" aria-hidden="true" tabindex="-1"></a>        ])</span>
<span id="cb27-39"><a href="#cb27-39" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"--&gt; Done in </span><span class="sc">{}</span><span class="st"> minutes"</span>.<span class="bu">format</span>((datetime.now()<span class="op">-</span>start).seconds<span class="op">/</span><span class="dv">60</span>))</span>
<span id="cb27-40"><a href="#cb27-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-41"><a href="#cb27-41" aria-hidden="true" tabindex="-1"></a>        query_comp, annot_comp <span class="op">=</span> annot_comp, query_comp</span>
<span id="cb27-42"><a href="#cb27-42" aria-hidden="true" tabindex="-1"></a>        query_edge, annot_edge <span class="op">=</span> annot_edge, query_edge</span>
<span id="cb27-43"><a href="#cb27-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-44"><a href="#cb27-44" aria-hidden="true" tabindex="-1"></a>        start <span class="op">=</span> datetime.now()</span>
<span id="cb27-45"><a href="#cb27-45" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Timestamp start Edge: </span><span class="sc">{</span>start<span class="sc">.</span>strftime(<span class="st">"%Y-%m-</span><span class="sc">%d</span><span class="st"> %H:%M:%S"</span>)<span class="sc">}</span><span class="ss">"</span>, end<span class="op">=</span><span class="st">""</span>)</span>
<span id="cb27-46"><a href="#cb27-46" aria-hidden="true" tabindex="-1"></a>        jaccard_comp <span class="op">=</span> jaccard_bootstrap(query_comp, annot_comp)</span>
<span id="cb27-47"><a href="#cb27-47" aria-hidden="true" tabindex="-1"></a>        jaccard_edge <span class="op">=</span> jaccard_bootstrap(query_edge, annot_edge)</span>
<span id="cb27-48"><a href="#cb27-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-49"><a href="#cb27-49" aria-hidden="true" tabindex="-1"></a>        jaccard_res <span class="op">=</span> pd.concat([jaccard_res, pd.DataFrame({</span>
<span id="cb27-50"><a href="#cb27-50" aria-hidden="true" tabindex="-1"></a>            <span class="st">'Sample'</span>: sample,</span>
<span id="cb27-51"><a href="#cb27-51" aria-hidden="true" tabindex="-1"></a>            <span class="st">'Query'</span>: <span class="st">'Edge'</span>,</span>
<span id="cb27-52"><a href="#cb27-52" aria-hidden="true" tabindex="-1"></a>            <span class="st">'Comp Index'</span>: jaccard_comp[<span class="dv">0</span>],</span>
<span id="cb27-53"><a href="#cb27-53" aria-hidden="true" tabindex="-1"></a>            <span class="st">'Edge Index'</span>: jaccard_edge[<span class="dv">0</span>],</span>
<span id="cb27-54"><a href="#cb27-54" aria-hidden="true" tabindex="-1"></a>            <span class="st">'Comp P-value'</span>: jaccard_comp[<span class="dv">1</span>],</span>
<span id="cb27-55"><a href="#cb27-55" aria-hidden="true" tabindex="-1"></a>            <span class="st">'Edge P-value'</span>: jaccard_edge[<span class="dv">1</span>]</span>
<span id="cb27-56"><a href="#cb27-56" aria-hidden="true" tabindex="-1"></a>        }, index<span class="op">=</span>[<span class="dv">0</span>])</span>
<span id="cb27-57"><a href="#cb27-57" aria-hidden="true" tabindex="-1"></a>        ])</span>
<span id="cb27-58"><a href="#cb27-58" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"--&gt; Done in </span><span class="sc">{}</span><span class="st"> minutes"</span>.<span class="bu">format</span>((datetime.now()<span class="op">-</span>start).seconds<span class="op">/</span><span class="dv">60</span>))</span>
<span id="cb27-59"><a href="#cb27-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-60"><a href="#cb27-60" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Write</span></span>
<span id="cb27-61"><a href="#cb27-61" aria-hidden="true" tabindex="-1"></a>    jaccard_res.to_csv(jaccard_file, index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb27-62"><a href="#cb27-62" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb27-63"><a href="#cb27-63" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Read</span></span>
<span id="cb27-64"><a href="#cb27-64" aria-hidden="true" tabindex="-1"></a>    jaccard_res <span class="op">=</span> pd.read_csv(jaccard_file)</span>
<span id="cb27-65"><a href="#cb27-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-66"><a href="#cb27-66" aria-hidden="true" tabindex="-1"></a><span class="co">## For a single sample only ##</span></span>
<span id="cb27-67"><a href="#cb27-67" aria-hidden="true" tabindex="-1"></a><span class="co"># # Calculate the Jaccard index</span></span>
<span id="cb27-68"><a href="#cb27-68" aria-hidden="true" tabindex="-1"></a><span class="co"># query = ech90</span></span>
<span id="cb27-69"><a href="#cb27-69" aria-hidden="true" tabindex="-1"></a><span class="co"># annot = edge_df</span></span>
<span id="cb27-70"><a href="#cb27-70" aria-hidden="true" tabindex="-1"></a><span class="co"># jaccard_stat_full, p_value_full = jaccard_bootstrap(query,annot)</span></span>
<span id="cb27-71"><a href="#cb27-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-72"><a href="#cb27-72" aria-hidden="true" tabindex="-1"></a><span class="co"># print("\nJaccard results: query=ECH90, annotation=edge_df")</span></span>
<span id="cb27-73"><a href="#cb27-73" aria-hidden="true" tabindex="-1"></a><span class="co"># print(f"ECH90 is bootstrapped n={b}")</span></span>
<span id="cb27-74"><a href="#cb27-74" aria-hidden="true" tabindex="-1"></a><span class="co"># print(f"\tIndex: {jaccard_stat_full}, \n\tp-value: {p_value_full}")</span></span>
<span id="cb27-75"><a href="#cb27-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-76"><a href="#cb27-76" aria-hidden="true" tabindex="-1"></a><span class="co"># query, annot = annot, query</span></span>
<span id="cb27-77"><a href="#cb27-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-78"><a href="#cb27-78" aria-hidden="true" tabindex="-1"></a><span class="co"># jaccard_stat_full, p_value_full = jaccard_bootstrap(query,annot)</span></span>
<span id="cb27-79"><a href="#cb27-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-80"><a href="#cb27-80" aria-hidden="true" tabindex="-1"></a><span class="co"># print("\nJaccard results: query=edge_df, annotation=ECH90")</span></span>
<span id="cb27-81"><a href="#cb27-81" aria-hidden="true" tabindex="-1"></a><span class="co"># print(f"Edge_df is bootstrapped n={b}")</span></span>
<span id="cb27-82"><a href="#cb27-82" aria-hidden="true" tabindex="-1"></a><span class="co"># print(f"\tIndex: {jaccard_stat_full}, \n\tp-value: {p_value_full}")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="parallelize-the-jaccard-test" class="level4">
<h4 class="anchored" data-anchor-id="parallelize-the-jaccard-test">Parallelize the Jaccard test</h4>
<div id="cell-33" class="cell" data-execution_count="7">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> joblib <span class="im">import</span> Parallel, delayed</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> tqdm <span class="im">import</span> tqdm</span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os.path <span class="im">as</span> op</span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> genominterv <span class="im">import</span> jaccard_stat, bootstrap</span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> datetime <span class="im">import</span> datetime</span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>b <span class="op">=</span> <span class="dv">100_000</span></span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a><span class="at">@bootstrap</span>(chromsizes, samples<span class="op">=</span>b)</span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> jaccard_bootstrap(query, annot):</span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a>    <span class="cf">return</span> jaccard_stat(query, annot)</span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> process_sample(sample):</span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a>    <span class="co">"""Function to process a single sample."""</span></span>
<span id="cb28-15"><a href="#cb28-15" aria-hidden="true" tabindex="-1"></a>    query_comp <span class="op">=</span> ech90</span>
<span id="cb28-16"><a href="#cb28-16" aria-hidden="true" tabindex="-1"></a>    query_edge <span class="op">=</span> ech90</span>
<span id="cb28-17"><a href="#cb28-17" aria-hidden="true" tabindex="-1"></a>    annot_comp <span class="op">=</span> comps[sample]</span>
<span id="cb28-18"><a href="#cb28-18" aria-hidden="true" tabindex="-1"></a>    annot_edge <span class="op">=</span> edges[sample]</span>
<span id="cb28-19"><a href="#cb28-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-20"><a href="#cb28-20" aria-hidden="true" tabindex="-1"></a>    jaccard_comp <span class="op">=</span> jaccard_bootstrap(query_comp, annot_comp)</span>
<span id="cb28-21"><a href="#cb28-21" aria-hidden="true" tabindex="-1"></a>    jaccard_edge <span class="op">=</span> jaccard_bootstrap(query_edge, annot_edge)</span>
<span id="cb28-22"><a href="#cb28-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-23"><a href="#cb28-23" aria-hidden="true" tabindex="-1"></a>    result <span class="op">=</span> [{</span>
<span id="cb28-24"><a href="#cb28-24" aria-hidden="true" tabindex="-1"></a>        <span class="st">'Sample'</span>: sample, </span>
<span id="cb28-25"><a href="#cb28-25" aria-hidden="true" tabindex="-1"></a>        <span class="st">'Query'</span>: <span class="st">'ECH90'</span>, </span>
<span id="cb28-26"><a href="#cb28-26" aria-hidden="true" tabindex="-1"></a>        <span class="st">'Comp Index'</span>: jaccard_comp[<span class="dv">0</span>], </span>
<span id="cb28-27"><a href="#cb28-27" aria-hidden="true" tabindex="-1"></a>        <span class="st">'Edge Index'</span>: jaccard_edge[<span class="dv">0</span>], </span>
<span id="cb28-28"><a href="#cb28-28" aria-hidden="true" tabindex="-1"></a>        <span class="st">'Comp P-value'</span>: jaccard_comp[<span class="dv">1</span>], </span>
<span id="cb28-29"><a href="#cb28-29" aria-hidden="true" tabindex="-1"></a>        <span class="st">'Edge P-value'</span>: jaccard_edge[<span class="dv">1</span>]</span>
<span id="cb28-30"><a href="#cb28-30" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb28-31"><a href="#cb28-31" aria-hidden="true" tabindex="-1"></a>    ]</span>
<span id="cb28-32"><a href="#cb28-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-33"><a href="#cb28-33" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Swap query and annotation for the Edge test</span></span>
<span id="cb28-34"><a href="#cb28-34" aria-hidden="true" tabindex="-1"></a>    query_comp, annot_comp <span class="op">=</span> annot_comp, query_comp</span>
<span id="cb28-35"><a href="#cb28-35" aria-hidden="true" tabindex="-1"></a>    query_edge, annot_edge <span class="op">=</span> annot_edge, query_edge</span>
<span id="cb28-36"><a href="#cb28-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-37"><a href="#cb28-37" aria-hidden="true" tabindex="-1"></a>    jaccard_comp <span class="op">=</span> jaccard_bootstrap(query_comp, annot_comp)</span>
<span id="cb28-38"><a href="#cb28-38" aria-hidden="true" tabindex="-1"></a>    jaccard_edge <span class="op">=</span> jaccard_bootstrap(query_edge, annot_edge)</span>
<span id="cb28-39"><a href="#cb28-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-40"><a href="#cb28-40" aria-hidden="true" tabindex="-1"></a>    result.append({</span>
<span id="cb28-41"><a href="#cb28-41" aria-hidden="true" tabindex="-1"></a>        <span class="st">'Sample'</span>: sample, </span>
<span id="cb28-42"><a href="#cb28-42" aria-hidden="true" tabindex="-1"></a>        <span class="st">'Query'</span>: <span class="st">'Edge'</span>, </span>
<span id="cb28-43"><a href="#cb28-43" aria-hidden="true" tabindex="-1"></a>        <span class="st">'Comp Index'</span>: jaccard_comp[<span class="dv">0</span>], </span>
<span id="cb28-44"><a href="#cb28-44" aria-hidden="true" tabindex="-1"></a>        <span class="st">'Edge Index'</span>: jaccard_edge[<span class="dv">0</span>], </span>
<span id="cb28-45"><a href="#cb28-45" aria-hidden="true" tabindex="-1"></a>        <span class="st">'Comp P-value'</span>: jaccard_comp[<span class="dv">1</span>], </span>
<span id="cb28-46"><a href="#cb28-46" aria-hidden="true" tabindex="-1"></a>        <span class="st">'Edge P-value'</span>: jaccard_edge[<span class="dv">1</span>]</span>
<span id="cb28-47"><a href="#cb28-47" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb28-48"><a href="#cb28-48" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb28-49"><a href="#cb28-49" aria-hidden="true" tabindex="-1"></a>    <span class="cf">return</span> result</span>
<span id="cb28-50"><a href="#cb28-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-51"><a href="#cb28-51" aria-hidden="true" tabindex="-1"></a>jaccard_file <span class="op">=</span> <span class="ss">f'../results/jaccard_test_</span><span class="sc">{</span>b<span class="sc">}</span><span class="ss">.csv'</span></span>
<span id="cb28-52"><a href="#cb28-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-53"><a href="#cb28-53" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> <span class="kw">not</span> op.exists(jaccard_file):</span>
<span id="cb28-54"><a href="#cb28-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-55"><a href="#cb28-55" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Run parallel computation</span></span>
<span id="cb28-56"><a href="#cb28-56" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Running bootstrap (</span><span class="sc">{</span>b<span class="sc">}</span><span class="ss">) for all samples in parallel"</span>)</span>
<span id="cb28-57"><a href="#cb28-57" aria-hidden="true" tabindex="-1"></a>    results <span class="op">=</span> Parallel(n_jobs<span class="op">=-</span><span class="dv">1</span>)(</span>
<span id="cb28-58"><a href="#cb28-58" aria-hidden="true" tabindex="-1"></a>        delayed(process_sample)(sample) <span class="cf">for</span> sample <span class="kw">in</span> tqdm(samples)</span>
<span id="cb28-59"><a href="#cb28-59" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb28-60"><a href="#cb28-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-61"><a href="#cb28-61" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Flatten and create DataFrame</span></span>
<span id="cb28-62"><a href="#cb28-62" aria-hidden="true" tabindex="-1"></a>    jaccard_res <span class="op">=</span> pd.DataFrame([item <span class="cf">for</span> sublist <span class="kw">in</span> results <span class="cf">for</span> item <span class="kw">in</span> sublist])</span>
<span id="cb28-63"><a href="#cb28-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-64"><a href="#cb28-64" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Write to CSV</span></span>
<span id="cb28-65"><a href="#cb28-65" aria-hidden="true" tabindex="-1"></a>    jaccard_res.to_csv(jaccard_file, index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb28-66"><a href="#cb28-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-67"><a href="#cb28-67" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb28-68"><a href="#cb28-68" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Read</span></span>
<span id="cb28-69"><a href="#cb28-69" aria-hidden="true" tabindex="-1"></a>    jaccard_res <span class="op">=</span> pd.read_csv(jaccard_file)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
</section>
<section id="plot-some-summary-statistics" class="level2">
<h2 class="anchored" data-anchor-id="plot-some-summary-statistics">Plot some summary statistics</h2>
<div id="cell-35" class="cell" data-execution_count="8">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Split sample column:</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>proximity_res <span class="op">=</span> pd.read_csv(proximity_file)</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>proximity_res[[<span class="st">'source'</span>, <span class="st">'resolution'</span>, <span class="st">'view'</span>, <span class="st">'smoothed'</span>]] <span class="op">=</span> proximity_res[<span class="st">'Sample'</span>].<span class="bu">str</span>.extract(<span class="vs">r'^(.*?)_(100kb|500kb)_(full|arms|10Mb)(?:_(smoothed))?$'</span>)</span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>proximity_res[<span class="st">'source'</span>] <span class="op">=</span> proximity_res[<span class="st">'source'</span>].<span class="bu">map</span>({<span class="st">'fibroblast'</span>:<span class="st">'Fib'</span>, <span class="st">'spermatogonia'</span>: <span class="st">'SP'</span>, <span class="st">'pachytene_spermatocyte'</span>: <span class="st">'PAC'</span>, <span class="st">'round_spermatid'</span>: <span class="st">'RS'</span>, <span class="st">'sperm'</span>: <span class="st">'Sperm'</span>})</span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>proximity_res[<span class="st">'smoothed'</span>] <span class="op">=</span> proximity_res[<span class="st">'smoothed'</span>].notna()</span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a>proximity_res[<span class="st">'resolution'</span>] <span class="op">=</span> proximity_res.<span class="bu">apply</span>(<span class="kw">lambda</span> x: <span class="st">'ps500kb'</span> <span class="cf">if</span> x[<span class="st">'smoothed'</span>] <span class="cf">else</span> x[<span class="st">'resolution'</span>], axis<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a>proximity_res <span class="op">=</span> proximity_res.melt(</span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a>    id_vars<span class="op">=</span>[<span class="st">'Sample'</span>, <span class="st">'source'</span>, <span class="st">'resolution'</span>, <span class="st">'view'</span>, <span class="st">'Query'</span>],</span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a>    value_vars<span class="op">=</span>[<span class="st">'Comp P-value'</span>, <span class="st">'Edge P-value'</span>], </span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a>     var_name<span class="op">=</span><span class="st">'type'</span>, value_name<span class="op">=</span><span class="st">'p-value'</span>,).query(<span class="st">'Query == "ECH90"'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div id="cell-36" class="cell" data-execution_count="10">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> seaborn <span class="im">as</span> sns</span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>source_order <span class="op">=</span> [<span class="st">'Fib'</span>, <span class="st">'SP'</span>, <span class="st">'PAC'</span>, <span class="st">'RS'</span>, <span class="st">'Sperm'</span>]</span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Choose the combination og parameters to plot</span></span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a>source <span class="op">=</span> [<span class="st">'fibroblast'</span>,<span class="st">'round_spermatid'</span>]</span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a>res <span class="op">=</span> [<span class="st">'100kb'</span>]</span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a>view <span class="op">=</span> [<span class="st">'arms'</span>]</span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a>annot <span class="op">=</span> [<span class="st">'Edge P-value'</span>]</span>
<span id="cb30-11"><a href="#cb30-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-12"><a href="#cb30-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot the results from @group</span></span>
<span id="cb30-13"><a href="#cb30-13" aria-hidden="true" tabindex="-1"></a>plot_group_prox <span class="op">=</span> proximity_res.query(<span class="st">'resolution in @res and view in @view and type in @annot'</span>)</span>
<span id="cb30-14"><a href="#cb30-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-15"><a href="#cb30-15" aria-hidden="true" tabindex="-1"></a>plot_group_prox <span class="op">=</span> plot_group_prox.assign(</span>
<span id="cb30-16"><a href="#cb30-16" aria-hidden="true" tabindex="-1"></a>    minuslog10p <span class="op">=</span> <span class="kw">lambda</span> x: <span class="op">-</span>np.log10(x[<span class="st">'p-value'</span>]),</span>
<span id="cb30-17"><a href="#cb30-17" aria-hidden="true" tabindex="-1"></a>    <span class="bu">type</span> <span class="op">=</span> <span class="st">'Proximity'</span>)</span>
<span id="cb30-18"><a href="#cb30-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-19"><a href="#cb30-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot the results</span></span>
<span id="cb30-20"><a href="#cb30-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-21"><a href="#cb30-21" aria-hidden="true" tabindex="-1"></a>sns.set_theme(style<span class="op">=</span><span class="st">"whitegrid"</span>)</span>
<span id="cb30-22"><a href="#cb30-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-23"><a href="#cb30-23" aria-hidden="true" tabindex="-1"></a>g <span class="op">=</span> sns.catplot(</span>
<span id="cb30-24"><a href="#cb30-24" aria-hidden="true" tabindex="-1"></a>    plot_group_prox, kind<span class="op">=</span><span class="st">'bar'</span>,</span>
<span id="cb30-25"><a href="#cb30-25" aria-hidden="true" tabindex="-1"></a>    x<span class="op">=</span><span class="st">'source'</span>, y<span class="op">=</span><span class="st">'minuslog10p'</span>, hue<span class="op">=</span><span class="st">'type'</span>, </span>
<span id="cb30-26"><a href="#cb30-26" aria-hidden="true" tabindex="-1"></a>    order<span class="op">=</span>source_order,</span>
<span id="cb30-27"><a href="#cb30-27" aria-hidden="true" tabindex="-1"></a>    errorbar<span class="op">=</span><span class="va">None</span>, margin_titles<span class="op">=</span><span class="va">True</span>, sharex<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb30-28"><a href="#cb30-28" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb30-29"><a href="#cb30-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-30"><a href="#cb30-30" aria-hidden="true" tabindex="-1"></a>g.despine(left<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb30-31"><a href="#cb30-31" aria-hidden="true" tabindex="-1"></a>g.set_axis_labels(<span class="st">""</span>, <span class="st">"-log10(p)"</span>)</span>
<span id="cb30-32"><a href="#cb30-32" aria-hidden="true" tabindex="-1"></a>g.legend.set_title(<span class="st">''</span>)</span>
<span id="cb30-33"><a href="#cb30-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-34"><a href="#cb30-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-35"><a href="#cb30-35" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> axis <span class="kw">in</span> g.axes.flat:</span>
<span id="cb30-36"><a href="#cb30-36" aria-hidden="true" tabindex="-1"></a>    axis.tick_params(labelbottom<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb30-37"><a href="#cb30-37" aria-hidden="true" tabindex="-1"></a>    <span class="co"># rotate said labels</span></span>
<span id="cb30-38"><a href="#cb30-38" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> item <span class="kw">in</span> axis.get_xticklabels():</span>
<span id="cb30-39"><a href="#cb30-39" aria-hidden="true" tabindex="-1"></a>        item.set_rotation(<span class="dv">45</span>)</span>
<span id="cb30-40"><a href="#cb30-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-41"><a href="#cb30-41" aria-hidden="true" tabindex="-1"></a>g.tight_layout()</span>
<span id="cb30-42"><a href="#cb30-42" aria-hidden="true" tabindex="-1"></a></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><a href="06_rec_genomicintervals_files/figure-html/cell-25-output-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-7"><img src="06_rec_genomicintervals_files/figure-html/cell-25-output-1.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
</div>
</div>
<div id="cell-37" class="cell" data-execution_count="11">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Now do the same for the Jaccard index</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>jaccard_res <span class="op">=</span> pd.read_csv(jaccard_file)</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>jaccard_res[[<span class="st">'source'</span>, <span class="st">'resolution'</span>, <span class="st">'view'</span>, <span class="st">'smoothed'</span>]] <span class="op">=</span> jaccard_res[<span class="st">'Sample'</span>].<span class="bu">str</span>.extract(<span class="vs">r'^(.*?)_(100kb|500kb)_(full|arms|10Mb)(?:_(smoothed))?$'</span>)</span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>jaccard_res[<span class="st">'source'</span>] <span class="op">=</span> jaccard_res[<span class="st">'source'</span>].<span class="bu">map</span>({<span class="st">'fibroblast'</span>:<span class="st">'Fib'</span>, <span class="st">'spermatogonia'</span>: <span class="st">'SP'</span>, <span class="st">'pachytene_spermatocyte'</span>: <span class="st">'PAC'</span>, <span class="st">'round_spermatid'</span>: <span class="st">'RS'</span>, <span class="st">'sperm'</span>: <span class="st">'Sperm'</span>})</span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a>jaccard_res[<span class="st">'smoothed'</span>] <span class="op">=</span> jaccard_res[<span class="st">'smoothed'</span>].notna()</span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a>jaccard_res[<span class="st">'resolution'</span>] <span class="op">=</span> jaccard_res.<span class="bu">apply</span>(<span class="kw">lambda</span> x: <span class="st">'ps500kb'</span> <span class="cf">if</span> x[<span class="st">'smoothed'</span>] <span class="cf">else</span> x[<span class="st">'resolution'</span>], axis<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a>jaccard_res <span class="op">=</span> jaccard_res.melt(</span>
<span id="cb31-10"><a href="#cb31-10" aria-hidden="true" tabindex="-1"></a>    id_vars<span class="op">=</span>[<span class="st">'Sample'</span>, <span class="st">'source'</span>, <span class="st">'resolution'</span>, <span class="st">'view'</span>, <span class="st">'Query'</span>],</span>
<span id="cb31-11"><a href="#cb31-11" aria-hidden="true" tabindex="-1"></a>    value_vars<span class="op">=</span>[<span class="st">'Comp P-value'</span>, <span class="st">'Edge P-value'</span>], </span>
<span id="cb31-12"><a href="#cb31-12" aria-hidden="true" tabindex="-1"></a>    var_name<span class="op">=</span><span class="st">'type'</span>, value_name<span class="op">=</span><span class="st">'p-value'</span>,).query(<span class="st">'Query == "ECH90"'</span>)</span>
<span id="cb31-13"><a href="#cb31-13" aria-hidden="true" tabindex="-1"></a></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div id="cell-38" class="cell" data-execution_count="12">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>source_order <span class="op">=</span> [<span class="st">'Fib'</span>, <span class="st">'SP'</span>, <span class="st">'PAC'</span>, <span class="st">'RS'</span>, <span class="st">'Sperm'</span>]</span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Choose the combination og parameters to plot</span></span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>source <span class="op">=</span> [<span class="st">'fibroblast'</span>,<span class="st">'round_spermatid'</span>]</span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>res <span class="op">=</span> [<span class="st">'100kb'</span>]</span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>view <span class="op">=</span> [<span class="st">'arms'</span>]</span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a>annot <span class="op">=</span> [<span class="st">'Edge P-value'</span>]</span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-10"><a href="#cb32-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot the results from @group</span></span>
<span id="cb32-11"><a href="#cb32-11" aria-hidden="true" tabindex="-1"></a>plot_group_jacc <span class="op">=</span> jaccard_res.query(<span class="st">'resolution in @res and view in @view and type in @annot'</span>)</span>
<span id="cb32-12"><a href="#cb32-12" aria-hidden="true" tabindex="-1"></a>plot_group_jacc <span class="op">=</span> plot_group_jacc.assign(</span>
<span id="cb32-13"><a href="#cb32-13" aria-hidden="true" tabindex="-1"></a>    minuslog10p <span class="op">=</span> <span class="kw">lambda</span> x: <span class="op">-</span>np.log10(x[<span class="st">'p-value'</span>]),</span>
<span id="cb32-14"><a href="#cb32-14" aria-hidden="true" tabindex="-1"></a>    <span class="bu">type</span> <span class="op">=</span> <span class="st">'Jaccard'</span>)</span>
<span id="cb32-15"><a href="#cb32-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-16"><a href="#cb32-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot the results</span></span>
<span id="cb32-17"><a href="#cb32-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-18"><a href="#cb32-18" aria-hidden="true" tabindex="-1"></a>sns.set_theme(style<span class="op">=</span><span class="st">"whitegrid"</span>)</span>
<span id="cb32-19"><a href="#cb32-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-20"><a href="#cb32-20" aria-hidden="true" tabindex="-1"></a>g <span class="op">=</span> sns.catplot(</span>
<span id="cb32-21"><a href="#cb32-21" aria-hidden="true" tabindex="-1"></a>    plot_group_jacc, kind<span class="op">=</span><span class="st">'bar'</span>,</span>
<span id="cb32-22"><a href="#cb32-22" aria-hidden="true" tabindex="-1"></a>    x<span class="op">=</span><span class="st">'source'</span>, y<span class="op">=</span><span class="st">'minuslog10p'</span>, hue<span class="op">=</span><span class="st">'type'</span>, </span>
<span id="cb32-23"><a href="#cb32-23" aria-hidden="true" tabindex="-1"></a>    order<span class="op">=</span>source_order,</span>
<span id="cb32-24"><a href="#cb32-24" aria-hidden="true" tabindex="-1"></a>    errorbar<span class="op">=</span><span class="va">None</span>, margin_titles<span class="op">=</span><span class="va">True</span>, sharex<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb32-25"><a href="#cb32-25" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb32-26"><a href="#cb32-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-27"><a href="#cb32-27" aria-hidden="true" tabindex="-1"></a>g.despine(left<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb32-28"><a href="#cb32-28" aria-hidden="true" tabindex="-1"></a>g.set_axis_labels(<span class="st">""</span>, <span class="st">"-log10(p)"</span>)</span>
<span id="cb32-29"><a href="#cb32-29" aria-hidden="true" tabindex="-1"></a>g.legend.set_title(<span class="st">''</span>)</span>
<span id="cb32-30"><a href="#cb32-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-31"><a href="#cb32-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-32"><a href="#cb32-32" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> axis <span class="kw">in</span> g.axes.flat:</span>
<span id="cb32-33"><a href="#cb32-33" aria-hidden="true" tabindex="-1"></a>    axis.tick_params(labelbottom<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb32-34"><a href="#cb32-34" aria-hidden="true" tabindex="-1"></a>    <span class="co"># rotate said labels</span></span>
<span id="cb32-35"><a href="#cb32-35" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> item <span class="kw">in</span> axis.get_xticklabels():</span>
<span id="cb32-36"><a href="#cb32-36" aria-hidden="true" tabindex="-1"></a>        item.set_rotation(<span class="dv">45</span>)</span>
<span id="cb32-37"><a href="#cb32-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-38"><a href="#cb32-38" aria-hidden="true" tabindex="-1"></a>g.tight_layout()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><a href="06_rec_genomicintervals_files/figure-html/cell-27-output-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-8"><img src="06_rec_genomicintervals_files/figure-html/cell-27-output-1.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
</div>
</div>
</section>
<section id="plot-proximity-result-with-jaccard-result" class="level2">
<h2 class="anchored" data-anchor-id="plot-proximity-result-with-jaccard-result">Plot proximity result with Jaccard result</h2>
<div id="cell-40" class="cell" data-execution_count="13">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Concatenate the two plot_groups</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>group <span class="op">=</span> pd.concat([plot_group_prox, plot_group_jacc])</span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>display(group[[<span class="st">'source'</span>, <span class="st">'resolution'</span>, <span class="st">'type'</span>, <span class="st">'p-value'</span>, <span class="st">'minuslog10p'</span>]].query(</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>    <span class="st">'source in ("Fib", "RS")'</span></span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>).reset_index(drop<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a>g <span class="op">=</span> sns.catplot(</span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a>    group, kind<span class="op">=</span><span class="st">'bar'</span>,</span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a>    x<span class="op">=</span><span class="st">'source'</span>, y<span class="op">=</span><span class="st">'minuslog10p'</span>, hue<span class="op">=</span><span class="st">'type'</span>, </span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a>    order<span class="op">=</span>source_order,</span>
<span id="cb33-12"><a href="#cb33-12" aria-hidden="true" tabindex="-1"></a>    errorbar<span class="op">=</span><span class="va">None</span>, margin_titles<span class="op">=</span><span class="va">True</span>, sharex<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb33-13"><a href="#cb33-13" aria-hidden="true" tabindex="-1"></a>    height<span class="op">=</span><span class="dv">3</span>, aspect<span class="op">=</span><span class="fl">1.5</span></span>
<span id="cb33-14"><a href="#cb33-14" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb33-15"><a href="#cb33-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-16"><a href="#cb33-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot a horisontal line at p=0.05</span></span>
<span id="cb33-17"><a href="#cb33-17" aria-hidden="true" tabindex="-1"></a>g.ax.axhline(<span class="op">-</span>np.log10(<span class="fl">0.05</span>), color<span class="op">=</span><span class="st">'tab:red'</span>, linestyle<span class="op">=</span><span class="st">'--'</span>)</span>
<span id="cb33-18"><a href="#cb33-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-19"><a href="#cb33-19" aria-hidden="true" tabindex="-1"></a>g.despine(left<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb33-20"><a href="#cb33-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-21"><a href="#cb33-21" aria-hidden="true" tabindex="-1"></a>g.set_axis_labels(<span class="st">""</span>, <span class="st">"-log10(p)"</span>)</span>
<span id="cb33-22"><a href="#cb33-22" aria-hidden="true" tabindex="-1"></a>g.legend.set_title(<span class="st">''</span>)</span>
<span id="cb33-23"><a href="#cb33-23" aria-hidden="true" tabindex="-1"></a>sns.move_legend(g, <span class="st">'upper center'</span>)</span>
<span id="cb33-24"><a href="#cb33-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-25"><a href="#cb33-25" aria-hidden="true" tabindex="-1"></a>g.tight_layout()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">source</th>
<th data-quarto-table-cell-role="th">resolution</th>
<th data-quarto-table-cell-role="th">type</th>
<th data-quarto-table-cell-role="th">p-value</th>
<th data-quarto-table-cell-role="th">minuslog10p</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>Fib</td>
<td>100kb</td>
<td>Proximity</td>
<td>0.00642</td>
<td>2.192465</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>RS</td>
<td>100kb</td>
<td>Proximity</td>
<td>0.03227</td>
<td>1.491201</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>Fib</td>
<td>100kb</td>
<td>Jaccard</td>
<td>0.01385</td>
<td>1.858550</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3</td>
<td>RS</td>
<td>100kb</td>
<td>Jaccard</td>
<td>0.00718</td>
<td>2.143876</td>
</tr>
</tbody>
</table>

</div>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><a href="06_rec_genomicintervals_files/figure-html/cell-28-output-2.png" class="lightbox" data-gallery="quarto-lightbox-gallery-9"><img src="06_rec_genomicintervals_files/figure-html/cell-28-output-2.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
</div>
</div>
</section>
<section id="plot-all-cominations-of-the-parameters" class="level2">
<h2 class="anchored" data-anchor-id="plot-all-cominations-of-the-parameters">Plot all cominations of the parameters</h2>
<p>Avoid this to avoid p-hacking.</p>
<div id="cell-43" class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a><span class="co"># # Define the order of the x-axis if we plot all combinations</span></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a><span class="co"># view_order = ['full', 'arms', '10Mb']</span></span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a><span class="co"># row_order = ['Fib', 'SP', 'PAC', 'RS', 'Sperm']</span></span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a><span class="co"># sns.set_theme(style="whitegrid")</span></span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb34-8"><a href="#cb34-8" aria-hidden="true" tabindex="-1"></a><span class="co"># g = sns.catplot(</span></span>
<span id="cb34-9"><a href="#cb34-9" aria-hidden="true" tabindex="-1"></a><span class="co">#     proximity_res, kind='bar', row='source',</span></span>
<span id="cb34-10"><a href="#cb34-10" aria-hidden="true" tabindex="-1"></a><span class="co">#     x='view', y='p-value', hue='type', col='resolution', </span></span>
<span id="cb34-11"><a href="#cb34-11" aria-hidden="true" tabindex="-1"></a><span class="co">#     order=view_order, row_order=row_order,</span></span>
<span id="cb34-12"><a href="#cb34-12" aria-hidden="true" tabindex="-1"></a><span class="co">#     errorbar=None, margin_titles=True, sharex=False,</span></span>
<span id="cb34-13"><a href="#cb34-13" aria-hidden="true" tabindex="-1"></a><span class="co">#     )</span></span>
<span id="cb34-14"><a href="#cb34-14" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb34-15"><a href="#cb34-15" aria-hidden="true" tabindex="-1"></a><span class="co"># g.despine(left=True)</span></span>
<span id="cb34-16"><a href="#cb34-16" aria-hidden="true" tabindex="-1"></a><span class="co"># g.set_axis_labels("", "P-value")</span></span>
<span id="cb34-17"><a href="#cb34-17" aria-hidden="true" tabindex="-1"></a><span class="co"># g.legend.set_title('')</span></span>
<span id="cb34-18"><a href="#cb34-18" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb34-19"><a href="#cb34-19" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb34-20"><a href="#cb34-20" aria-hidden="true" tabindex="-1"></a><span class="co"># for axis in g.axes.flat:</span></span>
<span id="cb34-21"><a href="#cb34-21" aria-hidden="true" tabindex="-1"></a><span class="co">#     axis.tick_params(labelbottom=True)</span></span>
<span id="cb34-22"><a href="#cb34-22" aria-hidden="true" tabindex="-1"></a><span class="co">#     # rotate said labels</span></span>
<span id="cb34-23"><a href="#cb34-23" aria-hidden="true" tabindex="-1"></a><span class="co">#     for item in axis.get_xticklabels():</span></span>
<span id="cb34-24"><a href="#cb34-24" aria-hidden="true" tabindex="-1"></a><span class="co">#         item.set_rotation(45)</span></span>
<span id="cb34-25"><a href="#cb34-25" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb34-26"><a href="#cb34-26" aria-hidden="true" tabindex="-1"></a><span class="co"># g.tight_layout()</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div id="cell-44" class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="co"># # # Group the results by source and resolution</span></span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a><span class="co"># # group = ["fibroblast", 'round_spermatid']</span></span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a><span class="co"># # # Plot the results from @group</span></span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a><span class="co"># # plot_group = jaccard_res.query(f'source in @group')</span></span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a><span class="co"># sns.set_theme(style="whitegrid")</span></span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a><span class="co"># view_order = ['full', 'arms', '10Mb']</span></span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a><span class="co"># row_order = ['Fib', 'SP', 'PAC', 'RS', 'Sperm']</span></span>
<span id="cb35-10"><a href="#cb35-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-11"><a href="#cb35-11" aria-hidden="true" tabindex="-1"></a><span class="co"># g = sns.catplot(</span></span>
<span id="cb35-12"><a href="#cb35-12" aria-hidden="true" tabindex="-1"></a><span class="co">#     jaccard_res, kind='bar', row='source',</span></span>
<span id="cb35-13"><a href="#cb35-13" aria-hidden="true" tabindex="-1"></a><span class="co">#     x='view', y='p-value', hue='type', col='resolution', </span></span>
<span id="cb35-14"><a href="#cb35-14" aria-hidden="true" tabindex="-1"></a><span class="co">#     margin_titles=True, sharex=False, </span></span>
<span id="cb35-15"><a href="#cb35-15" aria-hidden="true" tabindex="-1"></a><span class="co">#     order=view_order, row_order=row_order,</span></span>
<span id="cb35-16"><a href="#cb35-16" aria-hidden="true" tabindex="-1"></a><span class="co">#     errorbar=None)</span></span>
<span id="cb35-17"><a href="#cb35-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-18"><a href="#cb35-18" aria-hidden="true" tabindex="-1"></a><span class="co"># g.despine(left=True)</span></span>
<span id="cb35-19"><a href="#cb35-19" aria-hidden="true" tabindex="-1"></a><span class="co"># g.set_axis_labels("", "P-value")</span></span>
<span id="cb35-20"><a href="#cb35-20" aria-hidden="true" tabindex="-1"></a><span class="co"># g.legend.set_title('')</span></span>
<span id="cb35-21"><a href="#cb35-21" aria-hidden="true" tabindex="-1"></a><span class="co"># for axis in g.axes.flat:</span></span>
<span id="cb35-22"><a href="#cb35-22" aria-hidden="true" tabindex="-1"></a><span class="co">#     axis.tick_params(labelbottom=True)</span></span>
<span id="cb35-23"><a href="#cb35-23" aria-hidden="true" tabindex="-1"></a><span class="co">#     # rotate said labels</span></span>
<span id="cb35-24"><a href="#cb35-24" aria-hidden="true" tabindex="-1"></a><span class="co">#     for item in axis.get_xticklabels():</span></span>
<span id="cb35-25"><a href="#cb35-25" aria-hidden="true" tabindex="-1"></a><span class="co">#         item.set_rotation(45)</span></span>
<span id="cb35-26"><a href="#cb35-26" aria-hidden="true" tabindex="-1"></a><span class="co">#         item.set_va('top')</span></span>
<span id="cb35-27"><a href="#cb35-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-28"><a href="#cb35-28" aria-hidden="true" tabindex="-1"></a><span class="co"># g.tight_layout()</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="genominterv" class="level2">
<h2 class="anchored" data-anchor-id="genominterv">genominterv</h2>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> genominterv <span class="im">import</span> proximity_test, interval_collapse, interval_diff, interval_intersect, jaccard_stat</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>query <span class="op">=</span> ech90</span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>fb_ann_comp <span class="op">=</span> dataframes[<span class="st">'fibroblast_100kb_arms'</span>]</span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>fb_ann_edge <span class="op">=</span> edge_df[<span class="st">'fibroblast_100kb_arms_edges'</span>]</span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a>rs_ann_comp <span class="op">=</span> dataframes[<span class="st">'round_spermatid_100kb_arms'</span>]</span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a>rs_ann_edge <span class="op">=</span> edge_df[<span class="st">'round_spermatid_100kb_arms_edges'</span>]</span>
<span id="cb36-9"><a href="#cb36-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-10"><a href="#cb36-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Fib and RS compartments</span></span>
<span id="cb36-11"><a href="#cb36-11" aria-hidden="true" tabindex="-1"></a>plot_regions(query<span class="op">=</span>query, annot<span class="op">=</span>fb_ann_comp,</span>
<span id="cb36-12"><a href="#cb36-12" aria-hidden="true" tabindex="-1"></a>             intersect<span class="op">=</span>interval_intersect(query, fb_ann_comp), </span>
<span id="cb36-13"><a href="#cb36-13" aria-hidden="true" tabindex="-1"></a>             track_titles<span class="op">=</span>[<span class="st">'ECH90'</span>, <span class="st">'A-Comp'</span>, <span class="st">'Intersect'</span>],</span>
<span id="cb36-14"><a href="#cb36-14" aria-hidden="true" tabindex="-1"></a>             figsize<span class="op">=</span>(<span class="dv">4</span>,<span class="fl">1.3</span>))</span>
<span id="cb36-15"><a href="#cb36-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-16"><a href="#cb36-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-17"><a href="#cb36-17" aria-hidden="true" tabindex="-1"></a>plot_regions(query<span class="op">=</span>query, annot<span class="op">=</span>rs_ann_comp, </span>
<span id="cb36-18"><a href="#cb36-18" aria-hidden="true" tabindex="-1"></a>             intersect<span class="op">=</span>interval_intersect(query, rs_ann_comp), </span>
<span id="cb36-19"><a href="#cb36-19" aria-hidden="true" tabindex="-1"></a>             track_titles<span class="op">=</span>[<span class="st">'ECH90'</span>, <span class="st">'A-Comp'</span>, <span class="st">'Intersect'</span>], </span>
<span id="cb36-20"><a href="#cb36-20" aria-hidden="true" tabindex="-1"></a>             figsize<span class="op">=</span>(<span class="dv">4</span>,<span class="fl">1.3</span>))</span>
<span id="cb36-21"><a href="#cb36-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-22"><a href="#cb36-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Fib and RS edges</span></span>
<span id="cb36-23"><a href="#cb36-23" aria-hidden="true" tabindex="-1"></a>plot_regions(query<span class="op">=</span>query, annot<span class="op">=</span>fb_ann_edge,</span>
<span id="cb36-24"><a href="#cb36-24" aria-hidden="true" tabindex="-1"></a>             intersect<span class="op">=</span>interval_intersect(query, fb_ann_edge), </span>
<span id="cb36-25"><a href="#cb36-25" aria-hidden="true" tabindex="-1"></a>             track_titles<span class="op">=</span>[<span class="st">'ECH90'</span>, <span class="st">'Edges'</span>, <span class="st">'Intersect'</span>],</span>
<span id="cb36-26"><a href="#cb36-26" aria-hidden="true" tabindex="-1"></a>             figsize<span class="op">=</span>(<span class="dv">4</span>,<span class="fl">1.3</span>))</span>
<span id="cb36-27"><a href="#cb36-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-28"><a href="#cb36-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-29"><a href="#cb36-29" aria-hidden="true" tabindex="-1"></a>plot_regions(query<span class="op">=</span>query, annot<span class="op">=</span>rs_ann_edge, </span>
<span id="cb36-30"><a href="#cb36-30" aria-hidden="true" tabindex="-1"></a>             intersect<span class="op">=</span>interval_intersect(query, rs_ann_edge), </span>
<span id="cb36-31"><a href="#cb36-31" aria-hidden="true" tabindex="-1"></a>             track_titles<span class="op">=</span>[<span class="st">'ECH90'</span>, <span class="st">'Edges'</span>, <span class="st">'Intersect'</span>],</span>
<span id="cb36-32"><a href="#cb36-32" aria-hidden="true" tabindex="-1"></a>             figsize<span class="op">=</span>(<span class="dv">4</span>,<span class="fl">1.3</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div id="fig-comps-edges-ech" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-comps-edges-ech-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-comps-edges-ech" style="flex-basis: 50.0%;justify-content: flex-start;">
<div id="fig-comps-edges-ech-1" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-comps-edges-ech-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="06_rec_genomicintervals_files/figure-html/fig-comps-edges-ech-output-1.png" class="lightbox" data-gallery="fig-comps-edges-ech" title="Figure&nbsp;7.1&nbsp;(a): Visual representation of the genomic intervals of ECH90, A-compartments, edges, and their intersections. Shown fibroblast and round spermatid at 100kb resolution and arms view."><img src="06_rec_genomicintervals_files/figure-html/fig-comps-edges-ech-output-1.png" data-ref-parent="fig-comps-edges-ech" width="398" height="128" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-comps-edges-ech-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) Visual representation of the genomic intervals of ECH90, A-compartments, edges, and their intersections. Shown fibroblast and round spermatid at 100kb resolution and arms view.
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-comps-edges-ech" style="flex-basis: 50.0%;justify-content: flex-start;">
<div id="fig-comps-edges-ech-2" class="quarto-float quarto-figure quarto-figure-center anchored" width="398" height="128">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-comps-edges-ech-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="06_rec_genomicintervals_files/figure-html/fig-comps-edges-ech-output-2.png" class="lightbox" data-gallery="fig-comps-edges-ech" title="Figure&nbsp;7.1&nbsp;(b): "><img src="06_rec_genomicintervals_files/figure-html/fig-comps-edges-ech-output-2.png" id="fig-comps-edges-ech-2" data-ref-parent="fig-comps-edges-ech" width="398" height="128" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-comps-edges-ech-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b)
</figcaption>
</figure>
</div>
</div>
</div>
<div class="quarto-layout-row">
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-comps-edges-ech" style="flex-basis: 50.0%;justify-content: flex-start;">
<div id="fig-comps-edges-ech-3" class="quarto-float quarto-figure quarto-figure-center anchored" width="398" height="128">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-comps-edges-ech-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="06_rec_genomicintervals_files/figure-html/fig-comps-edges-ech-output-3.png" class="lightbox" data-gallery="fig-comps-edges-ech" title="Figure&nbsp;7.1&nbsp;(c): "><img src="06_rec_genomicintervals_files/figure-html/fig-comps-edges-ech-output-3.png" id="fig-comps-edges-ech-3" data-ref-parent="fig-comps-edges-ech" width="398" height="128" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-comps-edges-ech-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(c)
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-comps-edges-ech" style="flex-basis: 50.0%;justify-content: flex-start;">
<div id="fig-comps-edges-ech-4" class="quarto-float quarto-figure quarto-figure-center anchored" width="398" height="128">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-comps-edges-ech-4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="06_rec_genomicintervals_files/figure-html/fig-comps-edges-ech-output-4.png" class="lightbox" data-gallery="fig-comps-edges-ech" title="Figure&nbsp;7.1&nbsp;(d): "><img src="06_rec_genomicintervals_files/figure-html/fig-comps-edges-ech-output-4.png" id="fig-comps-edges-ech-4" data-ref-parent="fig-comps-edges-ech" width="398" height="128" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-comps-edges-ech-4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(d)
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-comps-edges-ech-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7.1
</figcaption>
</figure>
</div>
<div id="cell-48" class="cell" data-execution_count="43">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Determine how many bases edges are reduced compared to the full compartment</span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>fb_comp_tot_length <span class="op">=</span> fb_ann_comp[<span class="st">'length'</span>].<span class="bu">sum</span>()</span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>fb_edge_tot_length <span class="op">=</span> fb_ann_edge[<span class="st">'length'</span>].<span class="bu">sum</span>()</span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a>fb_edge_tot_length <span class="op">/</span> fb_comp_tot_length</span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a><span class="co"># one liner for RS</span></span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a>rs_ann_edge[<span class="st">'length'</span>].<span class="bu">sum</span>() <span class="op">/</span> rs_ann_comp[<span class="st">'length'</span>].<span class="bu">sum</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="43">
<pre><code>0.479449982300885</code></pre>
</div>
</div>
</section>
<section id="geneinfo" class="level2">
<h2 class="anchored" data-anchor-id="geneinfo">Geneinfo</h2>
<p>How does the edges align with genes?</p>
<p>This first plot is just to figure out how to plot with <code>gene_plot</code>.</p>
<div id="cell-51" class="cell" data-execution_count="15">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> geneinfo <span class="im">as</span> gi</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> matplotlib.patches <span class="im">import</span> Rectangle</span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> matplotlib.collections <span class="im">import</span> PatchCollection</span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Use the proximity test results to plot the ECH90 regions and the compartment edges</span></span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a>start <span class="op">=</span> full_intersect[<span class="st">'start'</span>][<span class="dv">2</span>]</span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a>end <span class="op">=</span> full_intersect[<span class="st">'end'</span>][<span class="dv">5</span>]</span>
<span id="cb39-10"><a href="#cb39-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-11"><a href="#cb39-11" aria-hidden="true" tabindex="-1"></a>rectangles <span class="op">=</span> [Rectangle((start, <span class="fl">0.1</span>), width<span class="op">=</span>end<span class="op">-</span>start, height<span class="op">=</span><span class="fl">0.9</span>, color<span class="op">=</span><span class="st">'tab:green'</span>, linewidth<span class="op">=</span><span class="dv">0</span>, alpha<span class="op">=</span><span class="fl">0.6</span>) <span class="cf">for</span> start, end <span class="kw">in</span> <span class="bu">zip</span>(full_intersect[<span class="st">'start'</span>][<span class="dv">2</span>:<span class="dv">6</span>], full_intersect[<span class="st">'end'</span>][<span class="dv">2</span>:<span class="dv">6</span>])]</span>
<span id="cb39-12"><a href="#cb39-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-13"><a href="#cb39-13" aria-hidden="true" tabindex="-1"></a>pc <span class="op">=</span> PatchCollection(rectangles, match_original<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb39-14"><a href="#cb39-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-15"><a href="#cb39-15" aria-hidden="true" tabindex="-1"></a>ax <span class="op">=</span> gi.gene_plot(<span class="st">'chrX'</span>, start<span class="op">-</span><span class="dv">100_000</span>, end<span class="op">+</span><span class="dv">100_000</span>, assembly<span class="op">=</span><span class="st">'rheMac10'</span>)</span>
<span id="cb39-16"><a href="#cb39-16" aria-hidden="true" tabindex="-1"></a>ax.add_collection(pc)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><a href="06_rec_genomicintervals_files/figure-html/cell-35-output-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-14"><img src="06_rec_genomicintervals_files/figure-html/cell-35-output-1.png" width="547" height="436" class="figure-img"></a></p>
</figure>
</div>
</div>
</div>
<section id="get-the-geneinfo-for-all-intersections-between-edges-and-ech90" class="level4">
<h4 class="anchored" data-anchor-id="get-the-geneinfo-for-all-intersections-between-edges-and-ech90">Get the geneinfo for all intersections between edges and ECH90</h4>
<p>And write to a csv file. If the file exists, read it with pandas.</p>
<div id="cell-53" class="cell">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Use get_genes_region</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os.path <span class="im">as</span> op</span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> geneinfo <span class="im">as</span> gi</span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> os <span class="im">import</span> makedirs <span class="im">as</span> mkdir</span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a>genes_dir <span class="op">=</span> <span class="st">'../results/rec_edge_genes/'</span></span>
<span id="cb40-8"><a href="#cb40-8" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> <span class="kw">not</span> op.exists(genes_dir):</span>
<span id="cb40-9"><a href="#cb40-9" aria-hidden="true" tabindex="-1"></a>    mkdir(genes_dir)</span>
<span id="cb40-10"><a href="#cb40-10" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb40-11"><a href="#cb40-11" aria-hidden="true" tabindex="-1"></a>genes_file <span class="op">=</span> op.join(genes_dir,<span class="st">'rs_edges_100kb_genes.csv'</span>)</span>
<span id="cb40-12"><a href="#cb40-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-13"><a href="#cb40-13" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> <span class="kw">not</span> op.exists(genes_file):</span>
<span id="cb40-14"><a href="#cb40-14" aria-hidden="true" tabindex="-1"></a>    genes <span class="op">=</span> pd.concat(</span>
<span id="cb40-15"><a href="#cb40-15" aria-hidden="true" tabindex="-1"></a>        full_intersect.<span class="bu">apply</span>(</span>
<span id="cb40-16"><a href="#cb40-16" aria-hidden="true" tabindex="-1"></a>            <span class="kw">lambda</span> x: gi.get_genes_region_dataframe(<span class="st">'chrX'</span>, x[<span class="st">'start'</span>], x[<span class="st">'end'</span>], assembly<span class="op">=</span><span class="st">'rheMac10'</span>), </span>
<span id="cb40-17"><a href="#cb40-17" aria-hidden="true" tabindex="-1"></a>            axis <span class="op">=</span><span class="dv">1</span></span>
<span id="cb40-18"><a href="#cb40-18" aria-hidden="true" tabindex="-1"></a>            ).to_list(),</span>
<span id="cb40-19"><a href="#cb40-19" aria-hidden="true" tabindex="-1"></a>        ignore_index<span class="op">=</span><span class="va">True</span></span>
<span id="cb40-20"><a href="#cb40-20" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb40-21"><a href="#cb40-21" aria-hidden="true" tabindex="-1"></a>    genes.to_csv(genes_file, index<span class="op">=</span><span class="va">False</span>) </span>
<span id="cb40-22"><a href="#cb40-22" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>: </span>
<span id="cb40-23"><a href="#cb40-23" aria-hidden="true" tabindex="-1"></a>    genes <span class="op">=</span> pd.read_csv(genes_file)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div id="cell-54" class="cell" data-execution_count="18">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>genes_list <span class="op">=</span> genes[<span class="st">'name'</span>].unique().tolist()</span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>genes_list</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="18">
<pre><code>['SH3KBP1',
 'MIR7206',
 'LANCL3',
 'XK',
 'CYBB',
 'LOC696657',
 'DYNLT3',
 'PAGE4',
 'USP27X',
 'CLCN5',
 'LOC114675180',
 'MIR532',
 'MIR188',
 'MIR500A',
 'MIR362',
 'MIR501',
 'MIR500B',
 'MIR660',
 'MIR502',
 'AKAP4',
 'CCNB3',
 'LOC114675218',
 'LOC695959',
 'CENPVL3',
 'FAM120C',
 'WNK3',
 'LOC114675302',
 'ZC3H12B',
 'LAS1L',
 'MSN',
 'ATRX',
 'MAGT1',
 'LOC114675151',
 'COX7B',
 'ATP7A',
 'ALG13',
 'LOC706958',
 'TRPC5',
 'ENOX2',
 'RAP2C',
 'LOC114675176',
 'DKC1',
 'LOC114675231',
 'MPP1',
 'SMIM9',
 'F8',
 'H2AFB3',
 'FUNDC2',
 'CMC4',
 'MTCP1',
 'BRCC3',
 'LOC703257']</code></pre>
</div>
</div>
<div id="cell-55" class="cell" data-execution_count="27">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> geneinfo <span class="im">as</span> gi</span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> matplotlib.patches <span class="im">import</span> Rectangle</span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> matplotlib.collections <span class="im">import</span> PatchCollection</span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-6"><a href="#cb43-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Use the proximity test results to plot the ECH90 regions and the compartment edges</span></span>
<span id="cb43-7"><a href="#cb43-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-8"><a href="#cb43-8" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i <span class="kw">in</span> full_intersect.index:</span>
<span id="cb43-9"><a href="#cb43-9" aria-hidden="true" tabindex="-1"></a>    start_idx <span class="op">=</span> i</span>
<span id="cb43-10"><a href="#cb43-10" aria-hidden="true" tabindex="-1"></a>    end_idx <span class="op">=</span> i</span>
<span id="cb43-11"><a href="#cb43-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-12"><a href="#cb43-12" aria-hidden="true" tabindex="-1"></a>    start <span class="op">=</span> full_intersect[<span class="st">'start'</span>][start_idx]</span>
<span id="cb43-13"><a href="#cb43-13" aria-hidden="true" tabindex="-1"></a>    end <span class="op">=</span> full_intersect[<span class="st">'end'</span>][end_idx]</span>
<span id="cb43-14"><a href="#cb43-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-15"><a href="#cb43-15" aria-hidden="true" tabindex="-1"></a>    rectangles <span class="op">=</span> [Rectangle(</span>
<span id="cb43-16"><a href="#cb43-16" aria-hidden="true" tabindex="-1"></a>        (start, <span class="fl">0.1</span>), width<span class="op">=</span>end<span class="op">-</span>start, height<span class="op">=</span><span class="fl">0.9</span>, color<span class="op">=</span><span class="st">'tab:green'</span>, linewidth<span class="op">=</span><span class="dv">0</span>, alpha<span class="op">=</span><span class="fl">0.6</span>) <span class="cf">for</span> start, end <span class="kw">in</span> <span class="bu">zip</span>(full_intersect[<span class="st">'start'</span>][start_idx:end_idx<span class="op">+</span><span class="dv">1</span>], full_intersect[<span class="st">'end'</span>][start_idx:end_idx<span class="op">+</span><span class="dv">1</span>])]</span>
<span id="cb43-17"><a href="#cb43-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-18"><a href="#cb43-18" aria-hidden="true" tabindex="-1"></a>    pc <span class="op">=</span> PatchCollection(rectangles, match_original<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb43-19"><a href="#cb43-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-20"><a href="#cb43-20" aria-hidden="true" tabindex="-1"></a>    ax <span class="op">=</span> gi.gene_plot(<span class="st">'chrX'</span>, start<span class="op">-</span><span class="dv">100_000</span>, end<span class="op">+</span><span class="dv">100_000</span>, assembly<span class="op">=</span><span class="st">'rheMac10'</span>, </span>
<span id="cb43-21"><a href="#cb43-21" aria-hidden="true" tabindex="-1"></a>                    highlight<span class="op">=</span>genes_list,</span>
<span id="cb43-22"><a href="#cb43-22" aria-hidden="true" tabindex="-1"></a>                    despine<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb43-23"><a href="#cb43-23" aria-hidden="true" tabindex="-1"></a>                    figsize<span class="op">=</span>(<span class="dv">5</span>, <span class="dv">5</span>),</span>
<span id="cb43-24"><a href="#cb43-24" aria-hidden="true" tabindex="-1"></a>                    aspect<span class="op">=</span><span class="dv">5</span>,</span>
<span id="cb43-25"><a href="#cb43-25" aria-hidden="true" tabindex="-1"></a>                    )</span>
<span id="cb43-26"><a href="#cb43-26" aria-hidden="true" tabindex="-1"></a>    ax.add_collection(pc)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><a href="06_rec_genomicintervals_files/figure-html/cell-38-output-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-15"><img src="06_rec_genomicintervals_files/figure-html/cell-38-output-1.png" width="439" height="452" class="figure-img"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><a href="06_rec_genomicintervals_files/figure-html/cell-38-output-2.png" class="lightbox" data-gallery="quarto-lightbox-gallery-16"><img src="06_rec_genomicintervals_files/figure-html/cell-38-output-2.png" width="439" height="452" class="figure-img"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><a href="06_rec_genomicintervals_files/figure-html/cell-38-output-3.png" class="lightbox" data-gallery="quarto-lightbox-gallery-17"><img src="06_rec_genomicintervals_files/figure-html/cell-38-output-3.png" width="439" height="452" class="figure-img"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><a href="06_rec_genomicintervals_files/figure-html/cell-38-output-4.png" class="lightbox" data-gallery="quarto-lightbox-gallery-18"><img src="06_rec_genomicintervals_files/figure-html/cell-38-output-4.png" width="439" height="452" class="figure-img"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><a href="06_rec_genomicintervals_files/figure-html/cell-38-output-5.png" class="lightbox" data-gallery="quarto-lightbox-gallery-19"><img src="06_rec_genomicintervals_files/figure-html/cell-38-output-5.png" width="439" height="452" class="figure-img"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><a href="06_rec_genomicintervals_files/figure-html/cell-38-output-6.png" class="lightbox" data-gallery="quarto-lightbox-gallery-20"><img src="06_rec_genomicintervals_files/figure-html/cell-38-output-6.png" width="439" height="452" class="figure-img"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><a href="06_rec_genomicintervals_files/figure-html/cell-38-output-7.png" class="lightbox" data-gallery="quarto-lightbox-gallery-21"><img src="06_rec_genomicintervals_files/figure-html/cell-38-output-7.png" width="459" height="452" class="figure-img"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><a href="06_rec_genomicintervals_files/figure-html/cell-38-output-8.png" class="lightbox" data-gallery="quarto-lightbox-gallery-22"><img src="06_rec_genomicintervals_files/figure-html/cell-38-output-8.png" width="439" height="452" class="figure-img"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><a href="06_rec_genomicintervals_files/figure-html/cell-38-output-9.png" class="lightbox" data-gallery="quarto-lightbox-gallery-23"><img src="06_rec_genomicintervals_files/figure-html/cell-38-output-9.png" width="439" height="452" class="figure-img"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><a href="06_rec_genomicintervals_files/figure-html/cell-38-output-10.png" class="lightbox" data-gallery="quarto-lightbox-gallery-24"><img src="06_rec_genomicintervals_files/figure-html/cell-38-output-10.png" width="463" height="452" class="figure-img"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><a href="06_rec_genomicintervals_files/figure-html/cell-38-output-11.png" class="lightbox" data-gallery="quarto-lightbox-gallery-25"><img src="06_rec_genomicintervals_files/figure-html/cell-38-output-11.png" width="445" height="452" class="figure-img"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><a href="06_rec_genomicintervals_files/figure-html/cell-38-output-12.png" class="lightbox" data-gallery="quarto-lightbox-gallery-26"><img src="06_rec_genomicintervals_files/figure-html/cell-38-output-12.png" width="439" height="452" class="figure-img"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><a href="06_rec_genomicintervals_files/figure-html/cell-38-output-13.png" class="lightbox" data-gallery="quarto-lightbox-gallery-27"><img src="06_rec_genomicintervals_files/figure-html/cell-38-output-13.png" width="463" height="452" class="figure-img"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><a href="06_rec_genomicintervals_files/figure-html/cell-38-output-14.png" class="lightbox" data-gallery="quarto-lightbox-gallery-28"><img src="06_rec_genomicintervals_files/figure-html/cell-38-output-14.png" width="439" height="452" class="figure-img"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><a href="06_rec_genomicintervals_files/figure-html/cell-38-output-15.png" class="lightbox" data-gallery="quarto-lightbox-gallery-29"><img src="06_rec_genomicintervals_files/figure-html/cell-38-output-15.png" width="442" height="452" class="figure-img"></a></p>
</figure>
</div>
</div>
</div>
</section>
<section id="what-can-i-do-with-the-list-of-genes-on-the-edges" class="level3">
<h3 class="anchored" data-anchor-id="what-can-i-do-with-the-list-of-genes-on-the-edges">What can I do with the list of genes on the edges?</h3>
<section id="go-enrichment" class="level4">
<h4 class="anchored" data-anchor-id="go-enrichment">GO enrichment?</h4>
<div id="cell-58" class="cell" data-execution_count="49">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a>mmul_x_genes <span class="op">=</span> gi.get_genes_region_dataframe(<span class="st">'chrX'</span>, <span class="dv">0</span>, <span class="dv">155_000_000</span>, assembly<span class="op">=</span><span class="st">'rheMac10'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div id="cell-59" class="cell" data-execution_count="50">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>mmul_x_genelist <span class="op">=</span> mmul_x_genes[<span class="st">'name'</span>].unique().tolist()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div id="cell-60" class="cell" data-execution_count="51">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>gene_list <span class="op">=</span> genes[<span class="st">'name'</span>].unique().tolist()</span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>taxid <span class="op">=</span> <span class="dv">9544</span></span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>gi.email(<span class="st">'sojernj@gmail.com'</span>)</span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a><span class="co">#gi.go_annotation_table(taxid=taxid)</span></span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-6"><a href="#cb46-6" aria-hidden="true" tabindex="-1"></a><span class="co">#gi.show_go_evidence_codes()</span></span>
<span id="cb46-7"><a href="#cb46-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-8"><a href="#cb46-8" aria-hidden="true" tabindex="-1"></a>go_terms <span class="op">=</span> gi.get_go_terms_for_genes(gene_list, taxid<span class="op">=</span>taxid)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div id="cell-61" class="cell" data-execution_count="53">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="bu">len</span>(go_terms)</span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a><span class="co">#gene_list[:5]</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="53">
<pre><code>127</code></pre>
</div>
</div>
<div id="cell-62" class="cell" data-execution_count="55">
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a>results <span class="op">=</span> gi.go_enrichment(</span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Use human taxid as a start</span></span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a>    gene_list, </span>
<span id="cb49-4"><a href="#cb49-4" aria-hidden="true" tabindex="-1"></a>    alpha<span class="op">=</span><span class="fl">0.05</span>,</span>
<span id="cb49-5"><a href="#cb49-5" aria-hidden="true" tabindex="-1"></a>    terms<span class="op">=</span>go_terms)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>geneinfo_cache/go-basic.obo: fmt(1.2) rel(2024-10-27) 44,017 Terms; optional_attrs(def relationship)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Could not map gene symbol "MIR7206" to ncbi id
Could not map gene symbol "LOC696657" to ncbi id
Could not map gene symbol "LOC114675180" to ncbi id</code></pre>
</div>
<div class="cell-output cell-output-error">
<div class="ansi-escaped-output">
<pre><span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">KeyError</span>                                  Traceback (most recent call last)
File <span class="ansi-green-fg">~/miniconda3/envs/hic/lib/python3.12/site-packages/geneinfo/__init__.py:1050</span>, in <span class="ansi-cyan-fg">_cached_symbol2ncbi</span><span class="ansi-blue-fg">(symbols, taxid)</span>
<span class="ansi-green-fg ansi-bold">   1049</span> <span style="font-weight:bold;color:rgb(0,135,0)">try</span>:    
<span class="ansi-green-fg">-&gt; 1050</span>     <span style="font-weight:bold;color:rgb(0,135,0)">return</span> symbol2ncbi<span style="color:rgb(98,98,98)">.</span>loc[symbols]<span style="color:rgb(98,98,98)">.</span>tolist() 
<span class="ansi-green-fg ansi-bold">   1051</span> <span style="font-weight:bold;color:rgb(0,135,0)">except</span> <span style="font-weight:bold;color:rgb(215,95,95)">KeyError</span>:

File <span class="ansi-green-fg">~/miniconda3/envs/hic/lib/python3.12/site-packages/pandas/core/indexing.py:1191</span>, in <span class="ansi-cyan-fg">_LocationIndexer.__getitem__</span><span class="ansi-blue-fg">(self, key)</span>
<span class="ansi-green-fg ansi-bold">   1190</span> maybe_callable <span style="color:rgb(98,98,98)">=</span> <span style="color:rgb(0,135,0)">self</span><span style="color:rgb(98,98,98)">.</span>_check_deprecated_callable_usage(key, maybe_callable)
<span class="ansi-green-fg">-&gt; 1191</span> <span style="font-weight:bold;color:rgb(0,135,0)">return</span> <span style="color:rgb(0,135,0)">self</span><span style="color:rgb(98,98,98)">.</span>_getitem_axis(maybe_callable, axis<span style="color:rgb(98,98,98)">=</span>axis)

File <span class="ansi-green-fg">~/miniconda3/envs/hic/lib/python3.12/site-packages/pandas/core/indexing.py:1420</span>, in <span class="ansi-cyan-fg">_LocIndexer._getitem_axis</span><span class="ansi-blue-fg">(self, key, axis)</span>
<span class="ansi-green-fg ansi-bold">   1418</span>         <span style="font-weight:bold;color:rgb(0,135,0)">raise</span> <span style="font-weight:bold;color:rgb(215,95,95)">ValueError</span>(<span style="color:rgb(175,0,0)">"</span><span style="color:rgb(175,0,0)">Cannot index with multidimensional key</span><span style="color:rgb(175,0,0)">"</span>)
<span class="ansi-green-fg">-&gt; 1420</span>     <span style="font-weight:bold;color:rgb(0,135,0)">return</span> <span style="color:rgb(0,135,0)">self</span><span style="color:rgb(98,98,98)">.</span>_getitem_iterable(key, axis<span style="color:rgb(98,98,98)">=</span>axis)
<span class="ansi-green-fg ansi-bold">   1422</span> <span style="font-style:italic;color:rgb(95,135,135)"># nested tuple slicing</span>

File <span class="ansi-green-fg">~/miniconda3/envs/hic/lib/python3.12/site-packages/pandas/core/indexing.py:1360</span>, in <span class="ansi-cyan-fg">_LocIndexer._getitem_iterable</span><span class="ansi-blue-fg">(self, key, axis)</span>
<span class="ansi-green-fg ansi-bold">   1359</span> <span style="font-style:italic;color:rgb(95,135,135)"># A collection of keys</span>
<span class="ansi-green-fg">-&gt; 1360</span> keyarr, indexer <span style="color:rgb(98,98,98)">=</span> <span style="color:rgb(0,135,0)">self</span><span style="color:rgb(98,98,98)">.</span>_get_listlike_indexer(key, axis)
<span class="ansi-green-fg ansi-bold">   1361</span> <span style="font-weight:bold;color:rgb(0,135,0)">return</span> <span style="color:rgb(0,135,0)">self</span><span style="color:rgb(98,98,98)">.</span>obj<span style="color:rgb(98,98,98)">.</span>_reindex_with_indexers(
<span class="ansi-green-fg ansi-bold">   1362</span>     {axis: [keyarr, indexer]}, copy<span style="color:rgb(98,98,98)">=</span><span style="font-weight:bold;color:rgb(0,135,0)">True</span>, allow_dups<span style="color:rgb(98,98,98)">=</span><span style="font-weight:bold;color:rgb(0,135,0)">True</span>
<span class="ansi-green-fg ansi-bold">   1363</span> )

File <span class="ansi-green-fg">~/miniconda3/envs/hic/lib/python3.12/site-packages/pandas/core/indexing.py:1558</span>, in <span class="ansi-cyan-fg">_LocIndexer._get_listlike_indexer</span><span class="ansi-blue-fg">(self, key, axis)</span>
<span class="ansi-green-fg ansi-bold">   1556</span> axis_name <span style="color:rgb(98,98,98)">=</span> <span style="color:rgb(0,135,0)">self</span><span style="color:rgb(98,98,98)">.</span>obj<span style="color:rgb(98,98,98)">.</span>_get_axis_name(axis)
<span class="ansi-green-fg">-&gt; 1558</span> keyarr, indexer <span style="color:rgb(98,98,98)">=</span> ax<span style="color:rgb(98,98,98)">.</span>_get_indexer_strict(key, axis_name)
<span class="ansi-green-fg ansi-bold">   1560</span> <span style="font-weight:bold;color:rgb(0,135,0)">return</span> keyarr, indexer

File <span class="ansi-green-fg">~/miniconda3/envs/hic/lib/python3.12/site-packages/pandas/core/indexes/base.py:6200</span>, in <span class="ansi-cyan-fg">Index._get_indexer_strict</span><span class="ansi-blue-fg">(self, key, axis_name)</span>
<span class="ansi-green-fg ansi-bold">   6198</span>     keyarr, indexer, new_indexer <span style="color:rgb(98,98,98)">=</span> <span style="color:rgb(0,135,0)">self</span><span style="color:rgb(98,98,98)">.</span>_reindex_non_unique(keyarr)
<span class="ansi-green-fg">-&gt; 6200</span> <span style="color:rgb(0,135,0)">self</span><span style="color:rgb(98,98,98)">.</span>_raise_if_missing(keyarr, indexer, axis_name)
<span class="ansi-green-fg ansi-bold">   6202</span> keyarr <span style="color:rgb(98,98,98)">=</span> <span style="color:rgb(0,135,0)">self</span><span style="color:rgb(98,98,98)">.</span>take(indexer)

File <span class="ansi-green-fg">~/miniconda3/envs/hic/lib/python3.12/site-packages/pandas/core/indexes/base.py:6252</span>, in <span class="ansi-cyan-fg">Index._raise_if_missing</span><span class="ansi-blue-fg">(self, key, indexer, axis_name)</span>
<span class="ansi-green-fg ansi-bold">   6251</span> not_found <span style="color:rgb(98,98,98)">=</span> <span style="color:rgb(0,135,0)">list</span>(ensure_index(key)[missing_mask<span style="color:rgb(98,98,98)">.</span>nonzero()[<span style="color:rgb(98,98,98)">0</span>]]<span style="color:rgb(98,98,98)">.</span>unique())
<span class="ansi-green-fg">-&gt; 6252</span> <span style="font-weight:bold;color:rgb(0,135,0)">raise</span> <span style="font-weight:bold;color:rgb(215,95,95)">KeyError</span>(<span style="color:rgb(175,0,0)">f</span><span style="color:rgb(175,0,0)">"</span><span style="font-weight:bold;color:rgb(175,95,135)">{</span>not_found<span style="font-weight:bold;color:rgb(175,95,135)">}</span><span style="color:rgb(175,0,0)"> not in index</span><span style="color:rgb(175,0,0)">"</span>)

<span class="ansi-red-fg">KeyError</span>: "['MIR7206', 'LOC696657', 'LOC114675180', 'MIR532', 'MIR188', 'MIR500A', 'MIR362', 'MIR501', 'MIR500B', 'MIR660', 'MIR502', 'LOC114675218', 'LOC695959', 'LOC114675302', 'LOC114675151', 'LOC706958', 'LOC114675176', 'LOC114675231', 'H2AFB3', 'LOC703257'] not in index"

During handling of the above exception, another exception occurred:

<span class="ansi-red-fg">KeyError</span>                                  Traceback (most recent call last)
File <span class="ansi-green-fg">~/miniconda3/envs/hic/lib/python3.12/site-packages/pandas/core/indexes/base.py:3805</span>, in <span class="ansi-cyan-fg">Index.get_loc</span><span class="ansi-blue-fg">(self, key)</span>
<span class="ansi-green-fg ansi-bold">   3804</span> <span style="font-weight:bold;color:rgb(0,135,0)">try</span>:
<span class="ansi-green-fg">-&gt; 3805</span>     <span style="font-weight:bold;color:rgb(0,135,0)">return</span> <span style="color:rgb(0,135,0)">self</span><span style="color:rgb(98,98,98)">.</span>_engine<span style="color:rgb(98,98,98)">.</span>get_loc(casted_key)
<span class="ansi-green-fg ansi-bold">   3806</span> <span style="font-weight:bold;color:rgb(0,135,0)">except</span> <span style="font-weight:bold;color:rgb(215,95,95)">KeyError</span> <span style="font-weight:bold;color:rgb(0,135,0)">as</span> err:

File <span class="ansi-green-fg">index.pyx:167</span>, in <span class="ansi-cyan-fg">pandas._libs.index.IndexEngine.get_loc</span><span class="ansi-blue-fg">()</span>

File <span class="ansi-green-fg">index.pyx:196</span>, in <span class="ansi-cyan-fg">pandas._libs.index.IndexEngine.get_loc</span><span class="ansi-blue-fg">()</span>

File <span class="ansi-green-fg">pandas/_libs/hashtable_class_helper.pxi:7081</span>, in <span class="ansi-cyan-fg">pandas._libs.hashtable.PyObjectHashTable.get_item</span><span class="ansi-blue-fg">()</span>

File <span class="ansi-green-fg">pandas/_libs/hashtable_class_helper.pxi:7089</span>, in <span class="ansi-cyan-fg">pandas._libs.hashtable.PyObjectHashTable.get_item</span><span class="ansi-blue-fg">()</span>

<span class="ansi-red-fg">KeyError</span>: 'MIR532'

The above exception was the direct cause of the following exception:

<span class="ansi-red-fg">KeyError</span>                                  Traceback (most recent call last)
File <span class="ansi-green-fg">~/miniconda3/envs/hic/lib/python3.12/site-packages/geneinfo/__init__.py:1055</span>, in <span class="ansi-cyan-fg">_cached_symbol2ncbi</span><span class="ansi-blue-fg">(symbols, taxid)</span>
<span class="ansi-green-fg ansi-bold">   1054</span> <span style="font-weight:bold;color:rgb(0,135,0)">try</span>:
<span class="ansi-green-fg">-&gt; 1055</span>     geneids<span style="color:rgb(98,98,98)">.</span>append(symbol2ncbi<span style="color:rgb(98,98,98)">.</span>loc[symbol])
<span class="ansi-green-fg ansi-bold">   1056</span> <span style="font-weight:bold;color:rgb(0,135,0)">except</span> <span style="font-weight:bold;color:rgb(215,95,95)">KeyError</span>:

File <span class="ansi-green-fg">~/miniconda3/envs/hic/lib/python3.12/site-packages/pandas/core/indexing.py:1191</span>, in <span class="ansi-cyan-fg">_LocationIndexer.__getitem__</span><span class="ansi-blue-fg">(self, key)</span>
<span class="ansi-green-fg ansi-bold">   1190</span> maybe_callable <span style="color:rgb(98,98,98)">=</span> <span style="color:rgb(0,135,0)">self</span><span style="color:rgb(98,98,98)">.</span>_check_deprecated_callable_usage(key, maybe_callable)
<span class="ansi-green-fg">-&gt; 1191</span> <span style="font-weight:bold;color:rgb(0,135,0)">return</span> <span style="color:rgb(0,135,0)">self</span><span style="color:rgb(98,98,98)">.</span>_getitem_axis(maybe_callable, axis<span style="color:rgb(98,98,98)">=</span>axis)

File <span class="ansi-green-fg">~/miniconda3/envs/hic/lib/python3.12/site-packages/pandas/core/indexing.py:1431</span>, in <span class="ansi-cyan-fg">_LocIndexer._getitem_axis</span><span class="ansi-blue-fg">(self, key, axis)</span>
<span class="ansi-green-fg ansi-bold">   1430</span> <span style="color:rgb(0,135,0)">self</span><span style="color:rgb(98,98,98)">.</span>_validate_key(key, axis)
<span class="ansi-green-fg">-&gt; 1431</span> <span style="font-weight:bold;color:rgb(0,135,0)">return</span> <span style="color:rgb(0,135,0)">self</span><span style="color:rgb(98,98,98)">.</span>_get_label(key, axis<span style="color:rgb(98,98,98)">=</span>axis)

File <span class="ansi-green-fg">~/miniconda3/envs/hic/lib/python3.12/site-packages/pandas/core/indexing.py:1381</span>, in <span class="ansi-cyan-fg">_LocIndexer._get_label</span><span class="ansi-blue-fg">(self, label, axis)</span>
<span class="ansi-green-fg ansi-bold">   1379</span> <span style="font-weight:bold;color:rgb(0,135,0)">def</span> <span style="color:rgb(0,0,255)">_get_label</span>(<span style="color:rgb(0,135,0)">self</span>, label, axis: AxisInt):
<span class="ansi-green-fg ansi-bold">   1380</span>     <span style="font-style:italic;color:rgb(95,135,135)"># GH#5567 this will fail if the label is not present in the axis.</span>
<span class="ansi-green-fg">-&gt; 1381</span>     <span style="font-weight:bold;color:rgb(0,135,0)">return</span> <span style="color:rgb(0,135,0)">self</span><span style="color:rgb(98,98,98)">.</span>obj<span style="color:rgb(98,98,98)">.</span>xs(label, axis<span style="color:rgb(98,98,98)">=</span>axis)

File <span class="ansi-green-fg">~/miniconda3/envs/hic/lib/python3.12/site-packages/pandas/core/generic.py:4301</span>, in <span class="ansi-cyan-fg">NDFrame.xs</span><span class="ansi-blue-fg">(self, key, axis, level, drop_level)</span>
<span class="ansi-green-fg ansi-bold">   4300</span> <span style="font-weight:bold;color:rgb(0,135,0)">else</span>:
<span class="ansi-green-fg">-&gt; 4301</span>     loc <span style="color:rgb(98,98,98)">=</span> index<span style="color:rgb(98,98,98)">.</span>get_loc(key)
<span class="ansi-green-fg ansi-bold">   4303</span>     <span style="font-weight:bold;color:rgb(0,135,0)">if</span> <span style="color:rgb(0,135,0)">isinstance</span>(loc, np<span style="color:rgb(98,98,98)">.</span>ndarray):

File <span class="ansi-green-fg">~/miniconda3/envs/hic/lib/python3.12/site-packages/pandas/core/indexes/base.py:3812</span>, in <span class="ansi-cyan-fg">Index.get_loc</span><span class="ansi-blue-fg">(self, key)</span>
<span class="ansi-green-fg ansi-bold">   3811</span>         <span style="font-weight:bold;color:rgb(0,135,0)">raise</span> InvalidIndexError(key)
<span class="ansi-green-fg">-&gt; 3812</span>     <span style="font-weight:bold;color:rgb(0,135,0)">raise</span> <span style="font-weight:bold;color:rgb(215,95,95)">KeyError</span>(key) <span style="font-weight:bold;color:rgb(0,135,0)">from</span> <span style="font-weight:bold;color:rgb(0,0,255)">err</span>
<span class="ansi-green-fg ansi-bold">   3813</span> <span style="font-weight:bold;color:rgb(0,135,0)">except</span> <span style="font-weight:bold;color:rgb(215,95,95)">TypeError</span>:
<span class="ansi-green-fg ansi-bold">   3814</span>     <span style="font-style:italic;color:rgb(95,135,135)"># If we have a listlike key, _check_indexing_error will raise</span>
<span class="ansi-green-fg ansi-bold">   3815</span>     <span style="font-style:italic;color:rgb(95,135,135)">#  InvalidIndexError. Otherwise we fall through and re-raise</span>
<span class="ansi-green-fg ansi-bold">   3816</span>     <span style="font-style:italic;color:rgb(95,135,135)">#  the TypeError.</span>

<span class="ansi-red-fg">KeyError</span>: 'MIR532'

During handling of the above exception, another exception occurred:

<span class="ansi-red-fg">HTTPError</span>                                 Traceback (most recent call last)
Cell <span class="ansi-green-fg">In[55], line 1</span>
<span class="ansi-green-fg">----&gt; 1</span> results <span style="color:rgb(98,98,98)">=</span> gi<span style="color:rgb(98,98,98)">.</span>go_enrichment(
<span class="ansi-green-fg ansi-bold">      2</span>     <span style="font-style:italic;color:rgb(95,135,135)"># Use human taxid as a start</span>
<span class="ansi-green-fg ansi-bold">      3</span>     gene_list, 
<span class="ansi-green-fg ansi-bold">      4</span>     alpha<span style="color:rgb(98,98,98)">=</span><span style="color:rgb(98,98,98)">0.05</span>,
<span class="ansi-green-fg ansi-bold">      5</span>     terms<span style="color:rgb(98,98,98)">=</span>go_terms)

File <span class="ansi-green-fg">~/miniconda3/envs/hic/lib/python3.12/site-packages/geneinfo/__init__.py:1780</span>, in <span class="ansi-cyan-fg">go_enrichment</span><span class="ansi-blue-fg">(gene_list, taxid, background_chrom, background_genes, terms, list_study_genes, alpha)</span>
<span class="ansi-green-fg ansi-bold">   1777</span> GeneID2nt <span style="color:rgb(98,98,98)">=</span> background_genes<span style="color:rgb(98,98,98)">.</span>GENEID2NT
<span class="ansi-green-fg ansi-bold">   1779</span> <span style="font-weight:bold;color:rgb(0,135,0)">if</span> <span style="font-weight:bold;color:rgb(175,0,255)">not</span> <span style="color:rgb(0,135,0)">all</span>(<span style="color:rgb(0,135,0)">type</span>(x) <span style="font-weight:bold;color:rgb(175,0,255)">is</span> <span style="color:rgb(0,135,0)">int</span> <span style="font-weight:bold;color:rgb(0,135,0)">for</span> x <span style="font-weight:bold;color:rgb(175,0,255)">in</span> gene_list):
<span class="ansi-green-fg">-&gt; 1780</span>     gene_list <span style="color:rgb(98,98,98)">=</span> _cached_symbol2ncbi(gene_list, taxid<span style="color:rgb(98,98,98)">=</span>taxid)
<span class="ansi-green-fg ansi-bold">   1782</span> goeaobj <span style="color:rgb(98,98,98)">=</span> GOEnrichmentStudyNS(
<span class="ansi-green-fg ansi-bold">   1783</span>         GeneID2nt, <span style="font-style:italic;color:rgb(95,135,135)"># List of mouse protein-coding genes</span>
<span class="ansi-green-fg ansi-bold">   1784</span>         ns2assoc, <span style="font-style:italic;color:rgb(95,135,135)"># geneid/GO associations</span>
<span class="ansi-green-fg">   (...)</span>
<span class="ansi-green-fg ansi-bold">   1788</span>         methods<span style="color:rgb(98,98,98)">=</span>[<span style="color:rgb(175,0,0)">'</span><span style="color:rgb(175,0,0)">fdr_bh</span><span style="color:rgb(175,0,0)">'</span>],
<span class="ansi-green-fg ansi-bold">   1789</span>         pvalcalc<span style="color:rgb(98,98,98)">=</span><span style="color:rgb(175,0,0)">'</span><span style="color:rgb(175,0,0)">fisher_scipy_stats</span><span style="color:rgb(175,0,0)">'</span>) 
<span class="ansi-green-fg ansi-bold">   1791</span> goea_results_all <span style="color:rgb(98,98,98)">=</span> goeaobj<span style="color:rgb(98,98,98)">.</span>run_study(gene_list)

File <span class="ansi-green-fg">~/miniconda3/envs/hic/lib/python3.12/site-packages/geneinfo/__init__.py:1058</span>, in <span class="ansi-cyan-fg">_cached_symbol2ncbi</span><span class="ansi-blue-fg">(symbols, taxid)</span>
<span class="ansi-green-fg ansi-bold">   1056</span> <span style="font-weight:bold;color:rgb(0,135,0)">except</span> <span style="font-weight:bold;color:rgb(215,95,95)">KeyError</span>:
<span class="ansi-green-fg ansi-bold">   1057</span>     <span style="font-weight:bold;color:rgb(0,135,0)">try</span>:
<span class="ansi-green-fg">-&gt; 1058</span>         ncbi_id <span style="color:rgb(98,98,98)">=</span> hgcn_symbol(symbol)
<span class="ansi-green-fg ansi-bold">   1059</span>         <span style="font-weight:bold;color:rgb(0,135,0)">if</span> ncbi_id <span style="font-weight:bold;color:rgb(175,0,255)">not</span> <span style="font-weight:bold;color:rgb(175,0,255)">in</span> symbol2ncbi<span style="color:rgb(98,98,98)">.</span>index:
<span class="ansi-green-fg ansi-bold">   1060</span>             <span style="color:rgb(0,135,0)">print</span>(ncbi_id, <span style="color:rgb(175,0,0)">'</span><span style="color:rgb(175,0,0)">not in symbol2ncbi index</span><span style="color:rgb(175,0,0)">'</span>)

File <span class="ansi-green-fg">~/miniconda3/envs/hic/lib/python3.12/site-packages/geneinfo/__init__.py:256</span>, in <span class="ansi-cyan-fg">hgcn_symbol</span><span class="ansi-blue-fg">(name)</span>
<span class="ansi-green-fg ansi-bold">    254</span>     <span style="font-weight:bold;color:rgb(0,135,0)">return</span> [ensembl2symbol(ensembl_id(n)) <span style="font-weight:bold;color:rgb(0,135,0)">for</span> n <span style="font-weight:bold;color:rgb(175,0,255)">in</span> name]
<span class="ansi-green-fg ansi-bold">    255</span> <span style="font-weight:bold;color:rgb(0,135,0)">else</span>:
<span class="ansi-green-fg">--&gt; 256</span>     <span style="font-weight:bold;color:rgb(0,135,0)">return</span> ensembl2symbol(ensembl_id(name))

File <span class="ansi-green-fg">~/miniconda3/envs/hic/lib/python3.12/site-packages/geneinfo/__init__.py:227</span>, in <span class="ansi-cyan-fg">ensembl2symbol</span><span class="ansi-blue-fg">(ensembl_id)</span>
<span class="ansi-green-fg ansi-bold">    225</span> r <span style="color:rgb(98,98,98)">=</span> requests<span style="color:rgb(98,98,98)">.</span>get(server<span style="color:rgb(98,98,98)">+</span>ext, headers<span style="color:rgb(98,98,98)">=</span>{ <span style="color:rgb(175,0,0)">"</span><span style="color:rgb(175,0,0)">Content-Type</span><span style="color:rgb(175,0,0)">"</span> : <span style="color:rgb(175,0,0)">"</span><span style="color:rgb(175,0,0)">application/json</span><span style="color:rgb(175,0,0)">"</span>})
<span class="ansi-green-fg ansi-bold">    226</span> <span style="font-weight:bold;color:rgb(0,135,0)">if</span> <span style="font-weight:bold;color:rgb(175,0,255)">not</span> r<span style="color:rgb(98,98,98)">.</span>ok:
<span class="ansi-green-fg">--&gt; 227</span>     r<span style="color:rgb(98,98,98)">.</span>raise_for_status()
<span class="ansi-green-fg ansi-bold">    228</span> decoded <span style="color:rgb(98,98,98)">=</span> r<span style="color:rgb(98,98,98)">.</span>json()
<span class="ansi-green-fg ansi-bold">    229</span> symbols <span style="color:rgb(98,98,98)">=</span> [x[<span style="color:rgb(175,0,0)">'</span><span style="color:rgb(175,0,0)">display_id</span><span style="color:rgb(175,0,0)">'</span>] <span style="font-weight:bold;color:rgb(0,135,0)">for</span> x <span style="font-weight:bold;color:rgb(175,0,255)">in</span> decoded <span style="font-weight:bold;color:rgb(0,135,0)">if</span> x[<span style="color:rgb(175,0,0)">'</span><span style="color:rgb(175,0,0)">dbname</span><span style="color:rgb(175,0,0)">'</span>] <span style="color:rgb(98,98,98)">==</span> <span style="color:rgb(175,0,0)">'</span><span style="color:rgb(175,0,0)">HGNC</span><span style="color:rgb(175,0,0)">'</span>]

File <span class="ansi-green-fg">~/miniconda3/envs/hic/lib/python3.12/site-packages/requests/models.py:1024</span>, in <span class="ansi-cyan-fg">Response.raise_for_status</span><span class="ansi-blue-fg">(self)</span>
<span class="ansi-green-fg ansi-bold">   1019</span>     http_error_msg <span style="color:rgb(98,98,98)">=</span> (
<span class="ansi-green-fg ansi-bold">   1020</span>         <span style="color:rgb(175,0,0)">f</span><span style="color:rgb(175,0,0)">"</span><span style="font-weight:bold;color:rgb(175,95,135)">{</span><span style="color:rgb(0,135,0)">self</span><span style="color:rgb(98,98,98)">.</span>status_code<span style="font-weight:bold;color:rgb(175,95,135)">}</span><span style="color:rgb(175,0,0)"> Server Error: </span><span style="font-weight:bold;color:rgb(175,95,135)">{</span>reason<span style="font-weight:bold;color:rgb(175,95,135)">}</span><span style="color:rgb(175,0,0)"> for url: </span><span style="font-weight:bold;color:rgb(175,95,135)">{</span><span style="color:rgb(0,135,0)">self</span><span style="color:rgb(98,98,98)">.</span>url<span style="font-weight:bold;color:rgb(175,95,135)">}</span><span style="color:rgb(175,0,0)">"</span>
<span class="ansi-green-fg ansi-bold">   1021</span>     )
<span class="ansi-green-fg ansi-bold">   1023</span> <span style="font-weight:bold;color:rgb(0,135,0)">if</span> http_error_msg:
<span class="ansi-green-fg">-&gt; 1024</span>     <span style="font-weight:bold;color:rgb(0,135,0)">raise</span> HTTPError(http_error_msg, response<span style="color:rgb(98,98,98)">=</span><span style="color:rgb(0,135,0)">self</span>)

<span class="ansi-red-fg">HTTPError</span>: 400 Client Error: Bad Request for url: https://rest.ensembl.org/xrefs/id/%5B'ENSG00000207758'%5D</pre>
</div>
</div>
</div>


</section>
</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../notebooks/05_rec_compartments.html" class="pagination-link" aria-label="Compartments Analysis w. recommended parameters">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-title">Compartments Analysis w. recommended parameters</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../workflow.html" class="pagination-link" aria-label="master_workflow">
        <span class="nav-page-text"><span class="chapter-title">master_workflow</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer"><div class="nav-footer-center"><div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.dev/munch-group/hic-spermatogenesis/blob/main/notebooks/06_rec_genomicintervals.ipynb" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/munch-group/hic-spermatogenesis/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li><li><a href="https://github.com/munch-group/hic-spermatogenesis/blob/main/notebooks/06_rec_genomicintervals.ipynb" class="toc-action"><i class="bi empty"></i>View source</a></li></ul></div></div></div></footer><script>var lightboxQuarto = GLightbox({"closeEffect":"zoom","descPosition":"bottom","loop":false,"openEffect":"zoom","selector":".lightbox"});
(function() {
  let previousOnload = window.onload;
  window.onload = () => {
    if (previousOnload) {
      previousOnload();
    }
    lightboxQuarto.on('slide_before_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      const href = trigger.getAttribute('href');
      if (href !== null) {
        const imgEl = window.document.querySelector(`a[href="${href}"] img`);
        if (imgEl !== null) {
          const srcAttr = imgEl.getAttribute("src");
          if (srcAttr && srcAttr.startsWith("data:")) {
            slideConfig.href = srcAttr;
          }
        }
      } 
    });
  
    lightboxQuarto.on('slide_after_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(slideNode);
      }
    });
  
  };
  
})();
          </script>




</body></html>