<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.40">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Various Comparisons – Søren's MSc project about Hi-C and spermatogenesis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js" integrity="sha512-bLT0Qm9VnAYZDflyKcBaQ2gg0hSYNQrJ8RilYldYQ1FxQYoCLtUjuuRuZo+fjqhx/qtq/1itJ0C2ejDxltZVFg==" crossorigin="anonymous"></script><script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../workflow.html" rel="next">
<link href="../notebooks/06_rec_genomicintervals.html" rel="prev">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-549806ee2085284f45b00abea8c6df48.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap-6a801e7181b8b24ef9cb8ae289dd8d09.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script src="../site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="../site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="../site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://hypothes.is/embed.js"></script>
<script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" integrity="sha512-c3Nl8+7g4LMSTdrm621y7kf9v3SDPnhxLNhcjFJbKECVnmZHTdo+IRO05sNLTH/D3vA6u1X32ehoLC7WFVdheg==" crossorigin="anonymous"></script>

<script type="application/javascript">define('jquery', [],function() {return window.jQuery;})</script>


</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../notebooks/01_hicexplorer.html">Notebooks</a></li><li class="breadcrumb-item"><a href="../notebooks/07_various_plotting.html"><span class="chapter-title">Various Comparisons</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">Søren’s MSc project about Hi-C and spermatogenesis</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/munch-group/hic-spermatogenesis" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Project</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">Notebooks</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/01_hicexplorer.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">HiCExplorer using macaque data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/02_open2c_framework.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Open2C framework</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/03_compartments.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Compartments Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/04_genomicintervals.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">E1 vs.&nbsp;ECH</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/05_rec_compartments.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Compartments Analysis w. recommended parameters</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/06_rec_genomicintervals.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">E1 vs.&nbsp;ECH (recommended params)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/07_various_plotting.html" class="sidebar-item-text sidebar-link active"><span class="chapter-title">Various Comparisons</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">Workflows</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../workflow.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">master_workflow</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true">
 <span class="menu-text">Reports</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../reports/thoughts.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Thoughts and Prayers</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../results/Readme.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Result files</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true">
 <span class="menu-text">Slides</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../slides/index.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Project Progress Day: Msc. Project 2024</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="true">
 <span class="menu-text">Thesis</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../thesis/index.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Chromatin Compartments and Selection on X</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#notes-before-we-start" id="toc-notes-before-we-start" class="nav-link active" data-scroll-target="#notes-before-we-start">Notes before we start</a></li>
  <li><a href="#setup" id="toc-setup" class="nav-link" data-scroll-target="#setup">Setup</a>
  <ul>
  <li><a href="#imports-for-plots-etc" id="toc-imports-for-plots-etc" class="nav-link" data-scroll-target="#imports-for-plots-etc">Imports for plots etc</a></li>
  <li><a href="#inlinebackend.rc" id="toc-inlinebackend.rc" class="nav-link" data-scroll-target="#inlinebackend.rc"><code>InlineBackend.rc</code></a></li>
  <li><a href="#load-all-the-coolers-and-eigenvectors" id="toc-load-all-the-coolers-and-eigenvectors" class="nav-link" data-scroll-target="#load-all-the-coolers-and-eigenvectors">Load all the coolers and eigenvectors</a></li>
  <li><a href="#load-the-genomic-intervals" id="toc-load-the-genomic-intervals" class="nav-link" data-scroll-target="#load-the-genomic-intervals">Load the genomic intervals</a>
  <ul class="collapse">
  <li><a href="#a-compartments" id="toc-a-compartments" class="nav-link" data-scroll-target="#a-compartments">A-compartments</a></li>
  <li><a href="#edge-intervals" id="toc-edge-intervals" class="nav-link" data-scroll-target="#edge-intervals">Edge intervals</a></li>
  <li><a href="#baboons" id="toc-baboons" class="nav-link" data-scroll-target="#baboons">Baboons</a></li>
  </ul></li>
  <li><a href="#define-a-plotting-utility-for-compartments-and-e1" id="toc-define-a-plotting-utility-for-compartments-and-e1" class="nav-link" data-scroll-target="#define-a-plotting-utility-for-compartments-and-e1">Define a plotting utility for compartments and E1</a></li>
  <li><a href="#plotting-matrices-separately" id="toc-plotting-matrices-separately" class="nav-link" data-scroll-target="#plotting-matrices-separately">Plotting matrices separately</a></li>
  <li><a href="#plot-intervals" id="toc-plot-intervals" class="nav-link" data-scroll-target="#plot-intervals">Plot intervals</a></li>
  <li><a href="#svedig-tabel" id="toc-svedig-tabel" class="nav-link" data-scroll-target="#svedig-tabel">Svedig tabel</a></li>
  </ul></li>
  <li><a href="#matrices-e1" id="toc-matrices-e1" class="nav-link" data-scroll-target="#matrices-e1">Matrices / E1</a>
  <ul>
  <li><a href="#e1-rs100500" id="toc-e1-rs100500" class="nav-link" data-scroll-target="#e1-rs100500">E1: RS100/500</a></li>
  <li><a href="#pe-vs-recpe-rs100kb" id="toc-pe-vs-recpe-rs100kb" class="nav-link" data-scroll-target="#pe-vs-recpe-rs100kb">PE vs recPE (RS100kb)</a></li>
  <li><a href="#pe-vs.-recpe-rs100kb" id="toc-pe-vs.-recpe-rs100kb" class="nav-link" data-scroll-target="#pe-vs.-recpe-rs100kb">PE vs.&nbsp;recPE: RS100kb</a></li>
  </ul></li>
  <li><a href="#intervals" id="toc-intervals" class="nav-link" data-scroll-target="#intervals">Intervals</a>
  <ul>
  <li><a href="#perecpe-compare-compartments-100kb" id="toc-perecpe-compare-compartments-100kb" class="nav-link" data-scroll-target="#perecpe-compare-compartments-100kb">PE/recPE compare compartments (100kb)</a></li>
  <li><a href="#visualize-ech-enrichment" id="toc-visualize-ech-enrichment" class="nav-link" data-scroll-target="#visualize-ech-enrichment">Visualize ECH enrichment</a></li>
  <li><a href="#all-3-sets" id="toc-all-3-sets" class="nav-link" data-scroll-target="#all-3-sets">All 3 sets</a></li>
  <li><a href="#liftover-comparison" id="toc-liftover-comparison" class="nav-link" data-scroll-target="#liftover-comparison">LiftOver comparison</a></li>
  <li><a href="#baboon-rs100-intersect" id="toc-baboon-rs100-intersect" class="nav-link" data-scroll-target="#baboon-rs100-intersect">Baboon-rs100-intersect</a></li>
  <li><a href="#table-test-1bp-limits" id="toc-table-test-1bp-limits" class="nav-link" data-scroll-target="#table-test-1bp-limits">Table: Test 1bp limits</a></li>
  </ul></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.dev/munch-group/hic-spermatogenesis/blob/main/notebooks/07_various_plotting.ipynb" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/munch-group/hic-spermatogenesis/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li><li><a href="https://github.com/munch-group/hic-spermatogenesis/blob/main/notebooks/07_various_plotting.ipynb" class="toc-action"><i class="bi empty"></i>View source</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../notebooks/01_hicexplorer.html">Notebooks</a></li><li class="breadcrumb-item"><a href="../notebooks/07_various_plotting.html"><span class="chapter-title">Various Comparisons</span></a></li></ol></nav>
<div class="quarto-title">
<h1 class="title"><span class="chapter-title">Various Comparisons</span></h1>
<p class="subtitle lead">Various plots using homemade plotting utility comparing different parameters and methods for the manuscript</p>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="notes-before-we-start" class="level3">
<h3 class="anchored" data-anchor-id="notes-before-we-start">Notes before we start</h3>
<p>We want to create a plotting utility so we can easily plot and compare different matrices. We might move them to <code>hicstuff.py</code> for import, but as for now, they are defined only in this notebook.</p>
</section>
<section id="setup" class="level2">
<h2 class="anchored" data-anchor-id="setup">Setup</h2>
<section id="imports-for-plots-etc" class="level3">
<h3 class="anchored" data-anchor-id="imports-for-plots-etc">Imports for plots etc</h3>
<div id="cell-4" class="cell" data-execution_count="1">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># import standard python libraries</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Import packages for working with cooler files and tools for plots</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooler</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooltools.lib.plotting</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooltools</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> seaborn <span class="im">as</span> sns</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> cooltools.lib.numutils <span class="im">import</span> adaptive_coarsegrain, interp_nan</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> matplotlib.colors <span class="im">import</span> LogNorm</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> mpl_toolkits.axes_grid1 <span class="im">import</span> make_axes_locatable</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="inlinebackend.rc" class="level3">
<h3 class="anchored" data-anchor-id="inlinebackend.rc"><code>InlineBackend.rc</code></h3>
<p>For consistent plots that fit with pdf manus.</p>
<div id="cell-6" class="cell" data-execution_count="2">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Define my params:</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="co">## Params will comply with my desired layout for the Manuscript (PDF)</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>notebook_rcparams <span class="op">=</span> {</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>    <span class="st">'font.size'</span>: <span class="dv">7</span>,</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>    <span class="st">'axes.titlesize'</span>: <span class="dv">8</span>,</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>    <span class="st">'axes.labelsize'</span>: <span class="dv">7</span>,</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>    <span class="st">'xtick.labelsize'</span>: <span class="dv">6</span>,</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>    <span class="st">'ytick.labelsize'</span>: <span class="dv">6</span>,</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>    <span class="st">'figure.titlesize'</span>: <span class="dv">9</span>,</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>    <span class="st">'figure.figsize'</span>: [<span class="fl">6.0</span>, <span class="fl">2.0</span>],</span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>    <span class="st">'figure.labelsize'</span>: <span class="dv">7</span>,</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply config</span></span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a><span class="op">%</span>matplotlib inline</span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a><span class="op">%</span>config InlineBackend.figure_formats <span class="op">=</span> [<span class="st">'retina'</span>]</span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a><span class="op">%</span>config InlineBackend.rc <span class="op">=</span> notebook_rcparams</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div id="cell-7" class="cell" data-execution_count="3">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>({key:plt.rcParams[key] <span class="cf">for</span> key <span class="kw">in</span> notebook_rcparams.keys()})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>{'font.size': 10.0, 'axes.titlesize': 'large', 'axes.labelsize': 'medium', 'xtick.labelsize': 'medium', 'ytick.labelsize': 'medium', 'figure.titlesize': 'large', 'figure.figsize': [6.4, 4.8], 'figure.labelsize': 'large'}</code></pre>
</div>
</div>
<div id="cell-8" class="cell" data-execution_count="4">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co">## Apparently IPython needs this twice to understand?? </span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Define my params:</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="co">## Params will comply with my desired layout for the Manuscript (PDF)</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>notebook_rcparams <span class="op">=</span> {</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>    <span class="st">'font.size'</span>: <span class="dv">7</span>,</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>    <span class="st">'axes.titlesize'</span>: <span class="dv">8</span>,</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>    <span class="st">'axes.labelsize'</span>: <span class="dv">7</span>,</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>    <span class="st">'xtick.labelsize'</span>: <span class="dv">6</span>,</span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>    <span class="st">'ytick.labelsize'</span>: <span class="dv">6</span>,</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>    <span class="st">'figure.titlesize'</span>: <span class="dv">9</span>,</span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>    <span class="st">'figure.figsize'</span>: [<span class="fl">6.0</span>, <span class="fl">2.0</span>],</span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>    <span class="st">'figure.labelsize'</span>: <span class="dv">7</span>,</span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply config</span></span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a><span class="op">%</span>matplotlib inline</span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a><span class="op">%</span>config InlineBackend.figure_formats <span class="op">=</span> [<span class="st">'retina'</span>]</span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a><span class="op">%</span>config InlineBackend.rc <span class="op">=</span> notebook_rcparams</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div id="cell-9" class="cell" data-execution_count="5">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>({key:plt.rcParams[key] <span class="cf">for</span> key <span class="kw">in</span> notebook_rcparams.keys()})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>{'font.size': 7.0, 'axes.titlesize': 8.0, 'axes.labelsize': 7.0, 'xtick.labelsize': 6.0, 'ytick.labelsize': 6.0, 'figure.titlesize': 9.0, 'figure.figsize': [6.0, 2.0], 'figure.labelsize': 7.0}</code></pre>
</div>
</div>
</section>
<section id="load-all-the-coolers-and-eigenvectors" class="level3">
<h3 class="anchored" data-anchor-id="load-all-the-coolers-and-eigenvectors">Load all the coolers and eigenvectors</h3>
<div id="cell-11" class="cell" data-execution_count="6">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co">## Loading all coolers and their compartments</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> glob</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os.path <span class="im">as</span> op</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cooler</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Abbreviate the names (almost the same abbreviations as Wang et al. 2019)</span></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>names_abbr <span class="op">=</span> {<span class="st">'fibroblast'</span>: <span class="st">'Fib'</span>, <span class="st">'spermatogonia'</span>: <span class="st">'SPA'</span>, <span class="st">'pachytene_spermatocyte'</span>: <span class="st">'PAC'</span>, <span class="st">'round_spermatid'</span>: <span class="st">'RS'</span>, <span class="st">'sperm'</span>: <span class="st">'Sp'</span>}</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Paths and resolutions</span></span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>mcools <span class="op">=</span> pd.Series(glob.glob(<span class="st">"../steps/bwa/*/cool/*/*.mcool"</span>))</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>res_100 <span class="op">=</span> <span class="st">"::resolutions/100000"</span></span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>res_500 <span class="op">=</span> <span class="st">"::resolutions/500000"</span></span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a DataFrame</span></span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>clr_df <span class="op">=</span> pd.DataFrame({</span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a>    <span class="st">'name'</span>: pd.concat([mcools.<span class="bu">apply</span>(<span class="kw">lambda</span> x: op.basename(op.dirname(x)))] <span class="op">*</span> <span class="dv">3</span>, ignore_index<span class="op">=</span><span class="va">True</span>),</span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a>    <span class="st">'resolution'</span>: [<span class="st">'100'</span>] <span class="op">*</span> <span class="bu">len</span>(mcools) <span class="op">+</span> [<span class="st">'ps500'</span>] <span class="op">*</span> <span class="bu">len</span>(mcools) <span class="op">+</span> [<span class="st">'500'</span>] <span class="op">*</span> <span class="bu">len</span>(mcools),</span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a>    <span class="st">'mcool_path'</span>: pd.concat([mcools] <span class="op">*</span> <span class="dv">3</span>, ignore_index<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-21"><a href="#cb8-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Insert the 'sname' column (shortname/abbreviation) </span></span>
<span id="cb8-22"><a href="#cb8-22" aria-hidden="true" tabindex="-1"></a>clr_df[<span class="st">'sname'</span>] <span class="op">=</span> clr_df[<span class="st">'name'</span>].<span class="bu">apply</span>(<span class="kw">lambda</span> x: names_abbr[x])</span>
<span id="cb8-23"><a href="#cb8-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-24"><a href="#cb8-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the 'cooler_id' column</span></span>
<span id="cb8-25"><a href="#cb8-25" aria-hidden="true" tabindex="-1"></a>clr_df[<span class="st">'cooler_id'</span>] <span class="op">=</span> clr_df.<span class="bu">apply</span>(</span>
<span id="cb8-26"><a href="#cb8-26" aria-hidden="true" tabindex="-1"></a>    <span class="kw">lambda</span> row: cooler.Cooler(</span>
<span id="cb8-27"><a href="#cb8-27" aria-hidden="true" tabindex="-1"></a>        row[<span class="st">'mcool_path'</span>] <span class="op">+</span> (res_100 <span class="cf">if</span> row[<span class="st">'resolution'</span>] <span class="kw">in</span> [<span class="st">'100'</span>, <span class="st">'ps500'</span>] <span class="cf">else</span> res_500)</span>
<span id="cb8-28"><a href="#cb8-28" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb8-29"><a href="#cb8-29" aria-hidden="true" tabindex="-1"></a>    axis<span class="op">=</span><span class="dv">1</span></span>
<span id="cb8-30"><a href="#cb8-30" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb8-31"><a href="#cb8-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-32"><a href="#cb8-32" aria-hidden="true" tabindex="-1"></a><span class="co"># Determine parsing run (dirname 3 levels up PE or recPE)</span></span>
<span id="cb8-33"><a href="#cb8-33" aria-hidden="true" tabindex="-1"></a>clr_df[<span class="st">'run'</span>] <span class="op">=</span> clr_df[<span class="st">'mcool_path'</span>].<span class="bu">apply</span>(<span class="kw">lambda</span> x: <span class="st">'recPE'</span> <span class="cf">if</span> <span class="st">'rec'</span> <span class="kw">in</span> x <span class="cf">else</span> <span class="st">'PE'</span>)</span>
<span id="cb8-34"><a href="#cb8-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-35"><a href="#cb8-35" aria-hidden="true" tabindex="-1"></a><span class="co"># Add the 'nbins' column</span></span>
<span id="cb8-36"><a href="#cb8-36" aria-hidden="true" tabindex="-1"></a>clr_df[<span class="st">'nbins'</span>] <span class="op">=</span> clr_df[<span class="st">'cooler_id'</span>].<span class="bu">apply</span>(<span class="kw">lambda</span> x: <span class="bu">len</span>(x.bins().fetch(<span class="st">'chrX'</span>)))</span>
<span id="cb8-37"><a href="#cb8-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-38"><a href="#cb8-38" aria-hidden="true" tabindex="-1"></a><span class="co"># Add the path to the compartments (../results/{compartments,rec_compartments})</span></span>
<span id="cb8-39"><a href="#cb8-39" aria-hidden="true" tabindex="-1"></a><span class="co"># make a comp_dir dict</span></span>
<span id="cb8-40"><a href="#cb8-40" aria-hidden="true" tabindex="-1"></a>comp_dirs <span class="op">=</span> {</span>
<span id="cb8-41"><a href="#cb8-41" aria-hidden="true" tabindex="-1"></a>    <span class="st">'PE'</span>: <span class="st">'../results/compartments/'</span>,</span>
<span id="cb8-42"><a href="#cb8-42" aria-hidden="true" tabindex="-1"></a>    <span class="st">'recPE'</span>: <span class="st">'../results/rec_compartments/'</span></span>
<span id="cb8-43"><a href="#cb8-43" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb8-44"><a href="#cb8-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-45"><a href="#cb8-45" aria-hidden="true" tabindex="-1"></a>clr_df[<span class="st">'e1_dict'</span>] <span class="op">=</span> clr_df.<span class="bu">apply</span>(</span>
<span id="cb8-46"><a href="#cb8-46" aria-hidden="true" tabindex="-1"></a>    <span class="kw">lambda</span> row: {view: pd.read_csv(op.join(</span>
<span id="cb8-47"><a href="#cb8-47" aria-hidden="true" tabindex="-1"></a>        comp_dirs[row[<span class="st">'run'</span>]], <span class="ss">f"</span><span class="sc">{</span>row[<span class="st">'name'</span>]<span class="sc">}</span><span class="ss">_e1_</span><span class="sc">{</span>row[<span class="st">'resolution'</span>]<span class="sc">}</span><span class="ss">kb_</span><span class="sc">{</span>view<span class="sc">}</span><span class="ss">.csv"</span></span>
<span id="cb8-48"><a href="#cb8-48" aria-hidden="true" tabindex="-1"></a>        ))[<span class="st">'e1'</span>]</span>
<span id="cb8-49"><a href="#cb8-49" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> row[<span class="st">'resolution'</span>] <span class="op">!=</span> <span class="st">'ps500'</span> </span>
<span id="cb8-50"><a href="#cb8-50" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span>  pd.read_csv(op.join(</span>
<span id="cb8-51"><a href="#cb8-51" aria-hidden="true" tabindex="-1"></a>        comp_dirs[row[<span class="st">'run'</span>]], <span class="ss">f"</span><span class="sc">{</span>row[<span class="st">'name'</span>]<span class="sc">}</span><span class="ss">_e1_100kb_</span><span class="sc">{</span>view<span class="sc">}</span><span class="ss">_smoothed.csv"</span></span>
<span id="cb8-52"><a href="#cb8-52" aria-hidden="true" tabindex="-1"></a>        ))[<span class="st">'e1'</span>] </span>
<span id="cb8-53"><a href="#cb8-53" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> view <span class="kw">in</span> [<span class="st">'full'</span>, <span class="st">'arms'</span>, <span class="st">'10Mb'</span>]},</span>
<span id="cb8-54"><a href="#cb8-54" aria-hidden="true" tabindex="-1"></a>    axis<span class="op">=</span><span class="dv">1</span></span>
<span id="cb8-55"><a href="#cb8-55" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb8-56"><a href="#cb8-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-57"><a href="#cb8-57" aria-hidden="true" tabindex="-1"></a><span class="co"># Now we have all the coolers and compartments ready for plotting (in semi-long format)</span></span>
<span id="cb8-58"><a href="#cb8-58" aria-hidden="true" tabindex="-1"></a><span class="co">#clr_df</span></span>
<span id="cb8-59"><a href="#cb8-59" aria-hidden="true" tabindex="-1"></a>clr_df[[<span class="st">'sname'</span>, <span class="st">'resolution'</span>, <span class="st">'nbins'</span>, <span class="st">'run'</span>, <span class="st">'e1_dict'</span>]].query(<span class="st">'resolution == "ps500"'</span>)</span>
<span id="cb8-60"><a href="#cb8-60" aria-hidden="true" tabindex="-1"></a><span class="co">#print(clr_df['e1_dict'].values)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="6">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">sname</th>
<th data-quarto-table-cell-role="th">resolution</th>
<th data-quarto-table-cell-role="th">nbins</th>
<th data-quarto-table-cell-role="th">run</th>
<th data-quarto-table-cell-role="th">e1_dict</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">10</td>
<td>PAC</td>
<td>ps500</td>
<td>1534</td>
<td>recPE</td>
<td>{'full': [nan, nan, nan, nan, nan, nan, nan, n...</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">11</td>
<td>SPA</td>
<td>ps500</td>
<td>1534</td>
<td>recPE</td>
<td>{'full': [nan, nan, nan, nan, nan, nan, nan, n...</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">12</td>
<td>Fib</td>
<td>ps500</td>
<td>1534</td>
<td>recPE</td>
<td>{'full': [nan, nan, nan, nan, nan, -0.15077432...</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">13</td>
<td>RS</td>
<td>ps500</td>
<td>1534</td>
<td>recPE</td>
<td>{'full': [nan, nan, nan, nan, nan, nan, nan, n...</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">14</td>
<td>Sp</td>
<td>ps500</td>
<td>1534</td>
<td>recPE</td>
<td>{'full': [nan, nan, nan, nan, nan, -0.00698684...</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">15</td>
<td>RS</td>
<td>ps500</td>
<td>1534</td>
<td>PE</td>
<td>{'full': [nan, nan, nan, nan, nan, -0.17383142...</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">16</td>
<td>SPA</td>
<td>ps500</td>
<td>1534</td>
<td>PE</td>
<td>{'full': [nan, nan, nan, nan, nan, nan, nan, -...</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">17</td>
<td>Sp</td>
<td>ps500</td>
<td>1534</td>
<td>PE</td>
<td>{'full': [nan, nan, nan, -0.102536873416233, -...</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">18</td>
<td>Fib</td>
<td>ps500</td>
<td>1534</td>
<td>PE</td>
<td>{'full': [nan, nan, nan, -0.2342462339281897, ...</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">19</td>
<td>PAC</td>
<td>ps500</td>
<td>1534</td>
<td>PE</td>
<td>{'full': [nan, nan, nan, nan, nan, 0.595328434...</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
</section>
<section id="load-the-genomic-intervals" class="level3">
<h3 class="anchored" data-anchor-id="load-the-genomic-intervals">Load the genomic intervals</h3>
<section id="a-compartments" class="level4">
<h4 class="anchored" data-anchor-id="a-compartments">A-compartments</h4>
<div id="cell-14" class="cell" data-execution_count="7">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="co"># ECH regions and chromsizes</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>ech90 <span class="op">=</span> pd.read_csv(<span class="st">'../data/ech90_human_Mmul_10.csv'</span>).assign(length<span class="op">=</span><span class="kw">lambda</span> x: x.end <span class="op">-</span> x.start)</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>chromsizes <span class="op">=</span> (pd.read_csv(</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>    <span class="st">'../data/rheMac10.filtered.chrom.sizes'</span>, </span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>    sep<span class="op">=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>, </span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>    index_col<span class="op">=</span><span class="st">'chrom'</span>, </span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>    header<span class="op">=</span><span class="va">None</span>, </span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>    names<span class="op">=</span>[<span class="st">'chrom'</span>,<span class="st">'size'</span>])</span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>.to_dict()[<span class="st">'size'</span>]</span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Directory containing your .csv files</span></span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a>comp_dirs <span class="op">=</span> {</span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a>    <span class="st">'PE'</span>: <span class="st">'../results/compartments/'</span>,</span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a>    <span class="st">'recPE'</span>: <span class="st">'../results/rec_compartments/'</span></span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb9-22"><a href="#cb9-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-23"><a href="#cb9-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a dictionary to store the comps</span></span>
<span id="cb9-24"><a href="#cb9-24" aria-hidden="true" tabindex="-1"></a>comps <span class="op">=</span> {</span>
<span id="cb9-25"><a href="#cb9-25" aria-hidden="true" tabindex="-1"></a>    <span class="st">'PE'</span>: {},</span>
<span id="cb9-26"><a href="#cb9-26" aria-hidden="true" tabindex="-1"></a>    <span class="st">'recPE'</span>: {}</span>
<span id="cb9-27"><a href="#cb9-27" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb9-28"><a href="#cb9-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-29"><a href="#cb9-29" aria-hidden="true" tabindex="-1"></a><span class="co"># Iterate over all .csv files in the directory</span></span>
<span id="cb9-30"><a href="#cb9-30" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> run <span class="kw">in</span> comps.keys():</span>
<span id="cb9-31"><a href="#cb9-31" aria-hidden="true" tabindex="-1"></a>    comp_dir <span class="op">=</span> comp_dirs[run]</span>
<span id="cb9-32"><a href="#cb9-32" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> filename <span class="kw">in</span> os.listdir(comp_dir):</span>
<span id="cb9-33"><a href="#cb9-33" aria-hidden="true" tabindex="-1"></a>        <span class="co"># We only want the .csv files and not e1 tracks</span></span>
<span id="cb9-34"><a href="#cb9-34" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> filename.endswith(<span class="st">'.csv'</span>) <span class="kw">and</span> <span class="st">'e1'</span> <span class="kw">not</span> <span class="kw">in</span> filename:</span>
<span id="cb9-35"><a href="#cb9-35" aria-hidden="true" tabindex="-1"></a>            <span class="co"># Construct the full file path</span></span>
<span id="cb9-36"><a href="#cb9-36" aria-hidden="true" tabindex="-1"></a>            filepath <span class="op">=</span> os.path.join(comp_dir, filename)</span>
<span id="cb9-37"><a href="#cb9-37" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb9-38"><a href="#cb9-38" aria-hidden="true" tabindex="-1"></a>            <span class="co"># Load the CSV into a DataFrame</span></span>
<span id="cb9-39"><a href="#cb9-39" aria-hidden="true" tabindex="-1"></a>            <span class="co"># Use the filename (without extension) as the dictionary key</span></span>
<span id="cb9-40"><a href="#cb9-40" aria-hidden="true" tabindex="-1"></a>            key <span class="op">=</span> filename.replace(<span class="st">'_a_comp_coords_'</span>, <span class="st">'_'</span>)</span>
<span id="cb9-41"><a href="#cb9-41" aria-hidden="true" tabindex="-1"></a>            key <span class="op">=</span> os.path.splitext(key)[<span class="dv">0</span>]</span>
<span id="cb9-42"><a href="#cb9-42" aria-hidden="true" tabindex="-1"></a>            comps[run][key] <span class="op">=</span> pd.read_csv(filepath).assign(length<span class="op">=</span><span class="kw">lambda</span> x: x.end <span class="op">-</span> x.start)</span>
<span id="cb9-43"><a href="#cb9-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-44"><a href="#cb9-44" aria-hidden="true" tabindex="-1"></a><span class="co"># The `comps` dictionary now contains the DataFrames</span></span>
<span id="cb9-45"><a href="#cb9-45" aria-hidden="true" tabindex="-1"></a><span class="co"># Example usage:</span></span>
<span id="cb9-46"><a href="#cb9-46" aria-hidden="true" tabindex="-1"></a>comps[<span class="st">'recPE'</span>][<span class="st">'round_spermatid_100kb_arms'</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="7">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">chrom</th>
<th data-quarto-table-cell-role="th">start</th>
<th data-quarto-table-cell-role="th">end</th>
<th data-quarto-table-cell-role="th">resolution</th>
<th data-quarto-table-cell-role="th">length</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>chrX</td>
<td>900000</td>
<td>1600000</td>
<td>100000</td>
<td>700000</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>chrX</td>
<td>1700000</td>
<td>2600000</td>
<td>100000</td>
<td>900000</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>chrX</td>
<td>2900000</td>
<td>3100000</td>
<td>100000</td>
<td>200000</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3</td>
<td>chrX</td>
<td>3200000</td>
<td>3500000</td>
<td>100000</td>
<td>300000</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">4</td>
<td>chrX</td>
<td>6600000</td>
<td>6700000</td>
<td>100000</td>
<td>100000</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">72</td>
<td>chrX</td>
<td>145400000</td>
<td>147100000</td>
<td>100000</td>
<td>1700000</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">73</td>
<td>chrX</td>
<td>147300000</td>
<td>147400000</td>
<td>100000</td>
<td>100000</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">74</td>
<td>chrX</td>
<td>148800000</td>
<td>149100000</td>
<td>100000</td>
<td>300000</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">75</td>
<td>chrX</td>
<td>149300000</td>
<td>151200000</td>
<td>100000</td>
<td>1900000</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">76</td>
<td>chrX</td>
<td>152500000</td>
<td>153300000</td>
<td>100000</td>
<td>800000</td>
</tr>
</tbody>
</table>

<p>77 rows × 5 columns</p>
</div>
</div>
</div>
</section>
<section id="edge-intervals" class="level4">
<h4 class="anchored" data-anchor-id="edge-intervals">Edge intervals</h4>
<div id="cell-16" class="cell" data-execution_count="8">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Directory containing your .csv files</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>edge_dirs <span class="op">=</span> {</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>    <span class="st">'PE'</span>: <span class="st">'../results/edges/'</span>,</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>    <span class="st">'recPE'</span>: <span class="st">'../results/rec_edges/'</span></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>edges <span class="op">=</span> {</span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>    <span class="st">'PE'</span>: {},</span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>    <span class="st">'recPE'</span>: {}</span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Iterate over all .csv files in the directories</span></span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> run <span class="kw">in</span> edges.keys():</span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a>    edge_dir <span class="op">=</span> edge_dirs[run]    </span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> filename <span class="kw">in</span> os.listdir(edge_dir):</span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> filename.endswith(<span class="st">'.csv'</span>):  <span class="co"># Check for .csv files</span></span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a>            <span class="co"># Construct the full file path</span></span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a>            filepath <span class="op">=</span> os.path.join(edge_dir, filename)</span>
<span id="cb10-22"><a href="#cb10-22" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb10-23"><a href="#cb10-23" aria-hidden="true" tabindex="-1"></a>            <span class="co"># Load the CSV into a DataFrame</span></span>
<span id="cb10-24"><a href="#cb10-24" aria-hidden="true" tabindex="-1"></a>            <span class="co"># Use the filename (without extension) as the dictionary key</span></span>
<span id="cb10-25"><a href="#cb10-25" aria-hidden="true" tabindex="-1"></a>            key <span class="op">=</span> filename.replace(<span class="st">'_edges'</span>, <span class="st">''</span>)</span>
<span id="cb10-26"><a href="#cb10-26" aria-hidden="true" tabindex="-1"></a>            key <span class="op">=</span> os.path.splitext(key)[<span class="dv">0</span>]</span>
<span id="cb10-27"><a href="#cb10-27" aria-hidden="true" tabindex="-1"></a>            edges[run][key] <span class="op">=</span> pd.read_csv(filepath).assign(length<span class="op">=</span><span class="kw">lambda</span> x: x.end <span class="op">-</span> x.start)</span>
<span id="cb10-28"><a href="#cb10-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-29"><a href="#cb10-29" aria-hidden="true" tabindex="-1"></a><span class="co"># The `edges` dictionary now contains the DataFrames</span></span>
<span id="cb10-30"><a href="#cb10-30" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(edges.keys())</span>
<span id="cb10-31"><a href="#cb10-31" aria-hidden="true" tabindex="-1"></a>display(edges[<span class="st">'recPE'</span>][<span class="st">'fibroblast_100kb_10Mb'</span>])</span>
<span id="cb10-32"><a href="#cb10-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-33"><a href="#cb10-33" aria-hidden="true" tabindex="-1"></a><span class="co">#ech90 = pd.read_csv('../data/ech90_human_Mmul_10.csv')</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>dict_keys(['PE', 'recPE'])</code></pre>
</div>
<div class="cell-output cell-output-display">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">start</th>
<th data-quarto-table-cell-role="th">end</th>
<th data-quarto-table-cell-role="th">chrom</th>
<th data-quarto-table-cell-role="th">resolution</th>
<th data-quarto-table-cell-role="th">length</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>700000</td>
<td>900000</td>
<td>chrX</td>
<td>100000</td>
<td>200000</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>3300000</td>
<td>3500000</td>
<td>chrX</td>
<td>100000</td>
<td>200000</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>8200000</td>
<td>8400000</td>
<td>chrX</td>
<td>100000</td>
<td>200000</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3</td>
<td>13300000</td>
<td>13500000</td>
<td>chrX</td>
<td>100000</td>
<td>200000</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">4</td>
<td>14900000</td>
<td>15100000</td>
<td>chrX</td>
<td>100000</td>
<td>200000</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">87</td>
<td>144400000</td>
<td>144800000</td>
<td>chrX</td>
<td>100000</td>
<td>400000</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">88</td>
<td>145400000</td>
<td>145600000</td>
<td>chrX</td>
<td>100000</td>
<td>200000</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">89</td>
<td>146700000</td>
<td>147100000</td>
<td>chrX</td>
<td>100000</td>
<td>400000</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">90</td>
<td>148800000</td>
<td>149000000</td>
<td>chrX</td>
<td>100000</td>
<td>200000</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">91</td>
<td>152100000</td>
<td>152300000</td>
<td>chrX</td>
<td>100000</td>
<td>200000</td>
</tr>
</tbody>
</table>

<p>92 rows × 5 columns</p>
</div>
</div>
</div>
</section>
<section id="baboons" class="level4">
<h4 class="anchored" data-anchor-id="baboons">Baboons</h4>
<div id="cell-18" class="cell" data-execution_count="9">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load baboons</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> genominterv <span class="im">import</span> interval_collapse</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>hama_panu3 <span class="op">=</span> pd.read_csv(<span class="st">'../data/lift/papanu4/high_hama_papanu4.bed'</span>, sep<span class="op">=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>, index_col<span class="op">=</span><span class="va">False</span>, header<span class="op">=</span><span class="va">None</span>, names<span class="op">=</span>[<span class="st">'group'</span>,<span class="st">'chrom'</span>, <span class="st">'start'</span>, <span class="st">'end'</span>])</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>olive_panu3 <span class="op">=</span> pd.read_csv(<span class="st">'../data/lift/papanu4/high_olive_papanu4.bed'</span>, sep<span class="op">=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>, index_col<span class="op">=</span><span class="va">False</span>, header<span class="op">=</span><span class="va">None</span>, names<span class="op">=</span>[<span class="st">'group'</span>,<span class="st">'chrom'</span>, <span class="st">'start'</span>, <span class="st">'end'</span>])</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>baboon_df <span class="op">=</span> pd.read_csv(<span class="st">'../results/high_baboon_rhemac10.bed'</span>, sep<span class="op">=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>, index_col<span class="op">=</span><span class="va">False</span>, header<span class="op">=</span><span class="va">None</span>, names<span class="op">=</span>[<span class="st">'chrom'</span>, <span class="st">'start'</span>, <span class="st">'end'</span>, <span class="st">'group'</span>, <span class="st">'score'</span>])</span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>high_hama_ucsc <span class="op">=</span> (interval_collapse(baboon_df.query(<span class="st">'group == "high_hama"'</span>)).</span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>             assign(group <span class="op">=</span> <span class="st">"high_hama"</span>).</span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>             query(<span class="st">'chrom == "chrX"'</span>)</span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a>             )</span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a>high_olive_ucsc <span class="op">=</span> (interval_collapse(baboon_df.query(<span class="st">'group == "high_olive"'</span>)).</span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a>              assign(group <span class="op">=</span> <span class="st">"high_olive"</span>).</span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a>              query(<span class="st">'chrom == "chrX"'</span>)</span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a>              )</span>
<span id="cb12-20"><a href="#cb12-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-21"><a href="#cb12-21" aria-hidden="true" tabindex="-1"></a>high_hama <span class="op">=</span> pd.read_csv(<span class="st">'../data/lift/rheMac10/high_hama_rhemac10.bed'</span>, sep<span class="op">=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>, index_col<span class="op">=</span><span class="va">False</span>, header<span class="op">=</span><span class="va">None</span>, names<span class="op">=</span>[<span class="st">'group'</span>,<span class="st">'chrom'</span>, <span class="st">'start'</span>, <span class="st">'end'</span>])</span>
<span id="cb12-22"><a href="#cb12-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-23"><a href="#cb12-23" aria-hidden="true" tabindex="-1"></a>high_olive <span class="op">=</span> pd.read_csv(<span class="st">'../data/lift/rheMac10/high_olive_rhemac10.bed'</span>, sep<span class="op">=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>, index_col<span class="op">=</span><span class="va">False</span>, header<span class="op">=</span><span class="va">None</span>, names<span class="op">=</span>[<span class="st">'group'</span>,<span class="st">'chrom'</span>, <span class="st">'start'</span>, <span class="st">'end'</span>])</span>
<span id="cb12-24"><a href="#cb12-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-25"><a href="#cb12-25" aria-hidden="true" tabindex="-1"></a>baboon_dict <span class="op">=</span> {<span class="st">"P.hama"</span>: high_hama, <span class="st">"P.anubis"</span>: high_olive}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="define-a-plotting-utility-for-compartments-and-e1" class="level3">
<h3 class="anchored" data-anchor-id="define-a-plotting-utility-for-compartments-and-e1">Define a plotting utility for compartments and E1</h3>
<p>This plots a group/subset of coolers and their respective E1 compartments.</p>
<div id="cell-20" class="cell" data-execution_count="10">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># We can vary the number of subplots based on the number of coolers</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="co"># We can vary the number of views for each cooler</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="co"># We can define the region (tuple) of the chromosome to plot (default chrX)</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> plot_grouped(</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>    group: pd.DataFrame, </span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>    region: <span class="bu">str</span> <span class="op">=</span> <span class="st">'chrX'</span>, <span class="co"># or (chrom, start, end)</span></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>    include_compartments: <span class="bu">bool</span> <span class="op">=</span> <span class="va">True</span>,</span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>    view_order: <span class="bu">list</span>[<span class="bu">str</span>] <span class="op">=</span> [<span class="st">'10Mb'</span>, <span class="st">'arms'</span>, <span class="st">'full'</span>], </span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>    figsize: <span class="bu">tuple</span>[<span class="bu">float</span>, <span class="bu">float</span>] <span class="op">=</span> (<span class="dv">6</span>, <span class="dv">6</span>)</span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>) <span class="op">-&gt;</span> <span class="va">None</span>:</span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>    group <span class="op">=</span> group.reset_index(drop<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Define the number of subplots based on the number of coolers</span></span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a>    f, axs <span class="op">=</span> plt.subplots(<span class="dv">1</span>,group.shape[<span class="dv">0</span>],</span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a>        figsize<span class="op">=</span>figsize, </span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a>        sharex<span class="op">=</span><span class="va">True</span></span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Logtransform the colorscale</span></span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a>    norm <span class="op">=</span> LogNorm(vmax<span class="op">=</span><span class="fl">0.1</span>)</span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-24"><a href="#cb13-24" aria-hidden="true" tabindex="-1"></a>    <span class="co"># If there is only one cooler, axs is not a list, so we need to make it a list</span></span>
<span id="cb13-25"><a href="#cb13-25" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> <span class="kw">not</span> <span class="bu">isinstance</span>(axs, np.ndarray):</span>
<span id="cb13-26"><a href="#cb13-26" aria-hidden="true" tabindex="-1"></a>        axs <span class="op">=</span> [axs]</span>
<span id="cb13-27"><a href="#cb13-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-28"><a href="#cb13-28" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb13-29"><a href="#cb13-29" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Loop through the clrs and its matrix: plot the matrix on its axis</span></span>
<span id="cb13-30"><a href="#cb13-30" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> row <span class="kw">in</span> group.itertuples():</span>
<span id="cb13-31"><a href="#cb13-31" aria-hidden="true" tabindex="-1"></a>        ax <span class="op">=</span> axs[row.Index]</span>
<span id="cb13-32"><a href="#cb13-32" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Define the y-axis label</span></span>
<span id="cb13-33"><a href="#cb13-33" aria-hidden="true" tabindex="-1"></a>        title <span class="op">=</span> <span class="ss">f'</span><span class="sc">{</span>row<span class="sc">.</span>sname<span class="sc">}{</span>row<span class="sc">.</span>resolution<span class="sc">}</span><span class="ss">kb: </span><span class="sc">{</span>row<span class="sc">.</span>run<span class="sc">}</span><span class="ss">'</span></span>
<span id="cb13-34"><a href="#cb13-34" aria-hidden="true" tabindex="-1"></a>        ylab <span class="op">=</span> <span class="ss">f'</span><span class="sc">{</span>region<span class="sc">}</span><span class="ss">:</span><span class="sc">{</span>row<span class="sc">.</span>resolution<span class="sc">}</span><span class="ss">kb [Mb]'</span></span>
<span id="cb13-35"><a href="#cb13-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-36"><a href="#cb13-36" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Extract some variables</span></span>
<span id="cb13-37"><a href="#cb13-37" aria-hidden="true" tabindex="-1"></a>        e1_dict <span class="op">=</span> {k: v <span class="cf">for</span> k, v <span class="kw">in</span> row.e1_dict.items() <span class="cf">if</span> k <span class="kw">in</span> view_order}</span>
<span id="cb13-38"><a href="#cb13-38" aria-hidden="true" tabindex="-1"></a>        chrom_start <span class="op">=</span> row.cooler_id.bins().fetch(region)[<span class="st">'start'</span>].values</span>
<span id="cb13-39"><a href="#cb13-39" aria-hidden="true" tabindex="-1"></a>        chrom_end <span class="op">=</span> row.cooler_id.bins().fetch(region)[<span class="st">'end'</span>].values</span>
<span id="cb13-40"><a href="#cb13-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-41"><a href="#cb13-41" aria-hidden="true" tabindex="-1"></a>        im <span class="op">=</span> ax.matshow(</span>
<span id="cb13-42"><a href="#cb13-42" aria-hidden="true" tabindex="-1"></a>            row.cooler_id.matrix().fetch(region),</span>
<span id="cb13-43"><a href="#cb13-43" aria-hidden="true" tabindex="-1"></a>            norm<span class="op">=</span>norm,</span>
<span id="cb13-44"><a href="#cb13-44" aria-hidden="true" tabindex="-1"></a>            cmap<span class="op">=</span><span class="st">'fall'</span>,</span>
<span id="cb13-45"><a href="#cb13-45" aria-hidden="true" tabindex="-1"></a>        )<span class="op">;</span></span>
<span id="cb13-46"><a href="#cb13-46" aria-hidden="true" tabindex="-1"></a>        ax.set_xlim(<span class="dv">0</span>, row.nbins)</span>
<span id="cb13-47"><a href="#cb13-47" aria-hidden="true" tabindex="-1"></a>        ax.set_ylim(row.nbins, <span class="dv">0</span>)</span>
<span id="cb13-48"><a href="#cb13-48" aria-hidden="true" tabindex="-1"></a>        ax.set_ylabel(ylab)</span>
<span id="cb13-49"><a href="#cb13-49" aria-hidden="true" tabindex="-1"></a>        ticks <span class="op">=</span> np.linspace(<span class="dv">0</span>, row.nbins, <span class="dv">5</span>)</span>
<span id="cb13-50"><a href="#cb13-50" aria-hidden="true" tabindex="-1"></a>        ax.set_yticks(ticks)</span>
<span id="cb13-51"><a href="#cb13-51" aria-hidden="true" tabindex="-1"></a>        ax.set_yticklabels([<span class="ss">f'</span><span class="sc">{</span>x<span class="op">*</span>row<span class="sc">.</span>cooler_id<span class="sc">.</span>binsize<span class="op">/</span><span class="dv">1_000_000</span><span class="sc">:.0f}</span><span class="ss">'</span> <span class="cf">for</span> x <span class="kw">in</span> ticks])</span>
<span id="cb13-52"><a href="#cb13-52" aria-hidden="true" tabindex="-1"></a>        ax.xaxis.set_visible(<span class="va">False</span>)</span>
<span id="cb13-53"><a href="#cb13-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-54"><a href="#cb13-54" aria-hidden="true" tabindex="-1"></a>        divider <span class="op">=</span> make_axes_locatable(ax)</span>
<span id="cb13-55"><a href="#cb13-55" aria-hidden="true" tabindex="-1"></a>        cax <span class="op">=</span> divider.append_axes(<span class="st">"right"</span>, size<span class="op">=</span><span class="st">"2%"</span>, pad<span class="op">=</span><span class="fl">0.05</span>)</span>
<span id="cb13-56"><a href="#cb13-56" aria-hidden="true" tabindex="-1"></a>        plt.colorbar(im, cax<span class="op">=</span>cax, label<span class="op">=</span><span class="st">'corrected frequencies'</span>)<span class="op">;</span></span>
<span id="cb13-57"><a href="#cb13-57" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Only show the colorbar on the last plot and ytitle on first plot</span></span>
<span id="cb13-58"><a href="#cb13-58" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> row.Index <span class="op">!=</span> group.shape[<span class="dv">0</span>] <span class="op">-</span> <span class="dv">1</span>:</span>
<span id="cb13-59"><a href="#cb13-59" aria-hidden="true" tabindex="-1"></a>            cax.set_visible(<span class="va">False</span>)</span>
<span id="cb13-60"><a href="#cb13-60" aria-hidden="true" tabindex="-1"></a>        <span class="cf">elif</span> row.Index <span class="op">!=</span> <span class="dv">0</span>:</span>
<span id="cb13-61"><a href="#cb13-61" aria-hidden="true" tabindex="-1"></a>            ax.set_ylabel(<span class="st">''</span>)</span>
<span id="cb13-62"><a href="#cb13-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-63"><a href="#cb13-63" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Skip the E1 plot when</span></span>
<span id="cb13-64"><a href="#cb13-64" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> <span class="kw">not</span> include_compartments:</span>
<span id="cb13-65"><a href="#cb13-65" aria-hidden="true" tabindex="-1"></a>            ax.set_title(title)</span>
<span id="cb13-66"><a href="#cb13-66" aria-hidden="true" tabindex="-1"></a>            <span class="cf">continue</span></span>
<span id="cb13-67"><a href="#cb13-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-68"><a href="#cb13-68" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> j, (view, e1) <span class="kw">in</span> <span class="bu">enumerate</span>(e1_dict.items()):</span>
<span id="cb13-69"><a href="#cb13-69" aria-hidden="true" tabindex="-1"></a>            ax1 <span class="op">=</span> divider.append_axes(<span class="st">"top"</span>, size<span class="op">=</span><span class="st">"10%"</span>, pad<span class="op">=</span><span class="fl">0.1</span>, sharex <span class="op">=</span> ax)</span>
<span id="cb13-70"><a href="#cb13-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-71"><a href="#cb13-71" aria-hidden="true" tabindex="-1"></a>            <span class="co"># Create stairs</span></span>
<span id="cb13-72"><a href="#cb13-72" aria-hidden="true" tabindex="-1"></a>            x <span class="op">=</span> np.zeros(<span class="dv">2</span><span class="op">*</span>chrom_start.size)</span>
<span id="cb13-73"><a href="#cb13-73" aria-hidden="true" tabindex="-1"></a>            y <span class="op">=</span> np.zeros(<span class="dv">2</span><span class="op">*</span>chrom_start.size)</span>
<span id="cb13-74"><a href="#cb13-74" aria-hidden="true" tabindex="-1"></a>            x[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> chrom_start<span class="op">/</span>row.cooler_id.binsize</span>
<span id="cb13-75"><a href="#cb13-75" aria-hidden="true" tabindex="-1"></a>            x[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> chrom_end<span class="op">/</span>row.cooler_id.binsize</span>
<span id="cb13-76"><a href="#cb13-76" aria-hidden="true" tabindex="-1"></a>            y[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> e1</span>
<span id="cb13-77"><a href="#cb13-77" aria-hidden="true" tabindex="-1"></a>            y[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> e1</span>
<span id="cb13-78"><a href="#cb13-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-79"><a href="#cb13-79" aria-hidden="true" tabindex="-1"></a>            ax1.fill_between(x, y, <span class="dv">0</span>, where<span class="op">=</span>(y <span class="op">&gt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:red'</span>, ec <span class="op">=</span> <span class="st">'None'</span>)</span>
<span id="cb13-80"><a href="#cb13-80" aria-hidden="true" tabindex="-1"></a>            ax1.fill_between(x, y, <span class="dv">0</span>, where<span class="op">=</span>(y <span class="op">&lt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:blue'</span>, ec <span class="op">=</span> <span class="st">'None'</span>)</span>
<span id="cb13-81"><a href="#cb13-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-82"><a href="#cb13-82" aria-hidden="true" tabindex="-1"></a>            ax1.set_ylabel(view, rotation<span class="op">=</span><span class="dv">0</span>, ha<span class="op">=</span><span class="st">'right'</span>, va<span class="op">=</span><span class="st">'center'</span>)</span>
<span id="cb13-83"><a href="#cb13-83" aria-hidden="true" tabindex="-1"></a>            ax1.set_ylim(<span class="op">-</span><span class="fl">0.8</span>, <span class="fl">0.8</span>)</span>
<span id="cb13-84"><a href="#cb13-84" aria-hidden="true" tabindex="-1"></a>            ax1.set_yticks([])</span>
<span id="cb13-85"><a href="#cb13-85" aria-hidden="true" tabindex="-1"></a>            ax1.set_xticks([])</span>
<span id="cb13-86"><a href="#cb13-86" aria-hidden="true" tabindex="-1"></a>            ax1.spines[:].set_visible(<span class="va">False</span>)</span>
<span id="cb13-87"><a href="#cb13-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-88"><a href="#cb13-88" aria-hidden="true" tabindex="-1"></a>        ax1.set_title(title)</span>
<span id="cb13-89"><a href="#cb13-89" aria-hidden="true" tabindex="-1"></a>    f.set_layout_engine(<span class="st">'compressed'</span>)</span>
<span id="cb13-90"><a href="#cb13-90" aria-hidden="true" tabindex="-1"></a>    plt.show()</span>
<span id="cb13-91"><a href="#cb13-91" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb13-92"><a href="#cb13-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-93"><a href="#cb13-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-94"><a href="#cb13-94" aria-hidden="true" tabindex="-1"></a></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="plotting-matrices-separately" class="level3">
<h3 class="anchored" data-anchor-id="plotting-matrices-separately">Plotting matrices separately</h3>
<p>Plotting function that makes separate plots in stead of mpl.subplots. This makes it compatible with YAML header subfigures and layout options for embedding into Quarto</p>
<div id="cell-22" class="cell" data-execution_count="11">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> plot_for_quarto(</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>    group: pd.DataFrame, </span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>    region: <span class="bu">str</span> <span class="op">|</span> <span class="bu">tuple</span>[<span class="bu">str</span>, <span class="bu">int</span>, <span class="bu">int</span>] <span class="op">=</span> <span class="st">'chrX'</span>,</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>    include_compartments: <span class="bu">bool</span> <span class="op">=</span> <span class="va">True</span>,</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>    view_order: <span class="bu">list</span>[<span class="bu">str</span>] <span class="op">=</span> [<span class="st">'10Mb'</span>, <span class="st">'arms'</span>, <span class="st">'full'</span>], </span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>    figsize: <span class="bu">tuple</span>[<span class="bu">float</span>, <span class="bu">float</span>] <span class="op">=</span> (<span class="dv">6</span>, <span class="dv">6</span>),</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>    col_vmax: <span class="bu">float</span> <span class="op">=</span> <span class="fl">0.1</span></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>) <span class="op">-&gt;</span> <span class="va">None</span>:</span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>    <span class="co">"""</span></span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a><span class="co">    Plot each cooler as a separate figure to allow Quarto YAML options for subfigure plots and captions.</span></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a><span class="co">    """</span></span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>    group <span class="op">=</span> group.reset_index(drop<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Scale figsize based on the number of coolers</span></span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a>    figsize <span class="op">=</span> (figsize[<span class="dv">0</span>]<span class="op">/</span>group.shape[<span class="dv">0</span>], figsize[<span class="dv">1</span>])</span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Set the region to a tuple if it is a string</span></span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> <span class="bu">isinstance</span>(region, <span class="bu">str</span>):</span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a>        region <span class="op">=</span> (region, <span class="dv">0</span>, group.iloc[<span class="dv">0</span>].cooler_id.chromsizes[region])</span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Define start and end for the matrix plot</span></span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a>    start, end <span class="op">=</span> region[<span class="dv">1</span>], region[<span class="dv">2</span>]</span>
<span id="cb14-23"><a href="#cb14-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-24"><a href="#cb14-24" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Loop through each cooler (row) in the DataFrame</span></span>
<span id="cb14-25"><a href="#cb14-25" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> row <span class="kw">in</span> group.itertuples():</span>
<span id="cb14-26"><a href="#cb14-26" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Create a new figure for each cooler</span></span>
<span id="cb14-27"><a href="#cb14-27" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Scale figsize</span></span>
<span id="cb14-28"><a href="#cb14-28" aria-hidden="true" tabindex="-1"></a>        fig, ax <span class="op">=</span> plt.subplots(figsize<span class="op">=</span>figsize)</span>
<span id="cb14-29"><a href="#cb14-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-30"><a href="#cb14-30" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Logtransform the colorscale</span></span>
<span id="cb14-31"><a href="#cb14-31" aria-hidden="true" tabindex="-1"></a>        norm <span class="op">=</span> LogNorm(vmax<span class="op">=</span>col_vmax)</span>
<span id="cb14-32"><a href="#cb14-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-33"><a href="#cb14-33" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Define the y-axis label</span></span>
<span id="cb14-34"><a href="#cb14-34" aria-hidden="true" tabindex="-1"></a>        <span class="co"># title = f'{row.sname}{row.resolution}kb: {row.run}'</span></span>
<span id="cb14-35"><a href="#cb14-35" aria-hidden="true" tabindex="-1"></a>        ylab <span class="op">=</span> <span class="ss">f'</span><span class="sc">{</span>row<span class="sc">.</span>resolution<span class="sc">}</span><span class="ss">kb'</span></span>
<span id="cb14-36"><a href="#cb14-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-37"><a href="#cb14-37" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Extract some variables</span></span>
<span id="cb14-38"><a href="#cb14-38" aria-hidden="true" tabindex="-1"></a>        e1_dict <span class="op">=</span> {k: row.e1_dict[k] <span class="cf">for</span> k <span class="kw">in</span> view_order}</span>
<span id="cb14-39"><a href="#cb14-39" aria-hidden="true" tabindex="-1"></a>        chrom_start <span class="op">=</span> row.cooler_id.bins().fetch(region)[<span class="st">'start'</span>].values</span>
<span id="cb14-40"><a href="#cb14-40" aria-hidden="true" tabindex="-1"></a>        chrom_end <span class="op">=</span> row.cooler_id.bins().fetch(region)[<span class="st">'end'</span>].values</span>
<span id="cb14-41"><a href="#cb14-41" aria-hidden="true" tabindex="-1"></a>        nbins <span class="op">=</span> chrom_start.size <span class="co"># != row.nbins when subsetting chr</span></span>
<span id="cb14-42"><a href="#cb14-42" aria-hidden="true" tabindex="-1"></a>        binsize <span class="op">=</span> row.cooler_id.binsize <span class="co"># is used several times</span></span>
<span id="cb14-43"><a href="#cb14-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-44"><a href="#cb14-44" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Plot the Hi-C matrix</span></span>
<span id="cb14-45"><a href="#cb14-45" aria-hidden="true" tabindex="-1"></a>        im <span class="op">=</span> ax.matshow(</span>
<span id="cb14-46"><a href="#cb14-46" aria-hidden="true" tabindex="-1"></a>            row.cooler_id.matrix().fetch(region),</span>
<span id="cb14-47"><a href="#cb14-47" aria-hidden="true" tabindex="-1"></a>            norm<span class="op">=</span>norm,</span>
<span id="cb14-48"><a href="#cb14-48" aria-hidden="true" tabindex="-1"></a>            cmap<span class="op">=</span><span class="st">'fall'</span>,</span>
<span id="cb14-49"><a href="#cb14-49" aria-hidden="true" tabindex="-1"></a>            extent<span class="op">=</span>(<span class="dv">0</span>, nbins, nbins, <span class="dv">0</span>)</span>
<span id="cb14-50"><a href="#cb14-50" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb14-51"><a href="#cb14-51" aria-hidden="true" tabindex="-1"></a>        ax.set_xlim(<span class="dv">0</span>, nbins)</span>
<span id="cb14-52"><a href="#cb14-52" aria-hidden="true" tabindex="-1"></a>        ax.set_ylim(nbins, <span class="dv">0</span>)</span>
<span id="cb14-53"><a href="#cb14-53" aria-hidden="true" tabindex="-1"></a>        ax.set_ylabel(ylab)</span>
<span id="cb14-54"><a href="#cb14-54" aria-hidden="true" tabindex="-1"></a>        ticks <span class="op">=</span> np.linspace(<span class="dv">0</span>, nbins, <span class="dv">5</span>)</span>
<span id="cb14-55"><a href="#cb14-55" aria-hidden="true" tabindex="-1"></a>        ax.set_yticks(ticks)</span>
<span id="cb14-56"><a href="#cb14-56" aria-hidden="true" tabindex="-1"></a>        ax.set_yticklabels([<span class="ss">f'</span><span class="sc">{</span>(start<span class="op">/</span>binsize<span class="op">+</span>x)<span class="op">*</span>binsize<span class="op">/</span><span class="dv">1_000_000</span><span class="sc">:.0f}</span><span class="ss">'</span> <span class="cf">for</span> x <span class="kw">in</span> ticks])</span>
<span id="cb14-57"><a href="#cb14-57" aria-hidden="true" tabindex="-1"></a>        ax.xaxis.set_visible(<span class="va">False</span>)</span>
<span id="cb14-58"><a href="#cb14-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-59"><a href="#cb14-59" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Add a colorbar</span></span>
<span id="cb14-60"><a href="#cb14-60" aria-hidden="true" tabindex="-1"></a>        divider <span class="op">=</span> make_axes_locatable(ax)</span>
<span id="cb14-61"><a href="#cb14-61" aria-hidden="true" tabindex="-1"></a>        cax <span class="op">=</span> divider.append_axes(<span class="st">"right"</span>, size<span class="op">=</span><span class="st">"2%"</span>, pad<span class="op">=</span><span class="fl">0.05</span>)</span>
<span id="cb14-62"><a href="#cb14-62" aria-hidden="true" tabindex="-1"></a>        plt.colorbar(im, cax<span class="op">=</span>cax, label<span class="op">=</span><span class="st">'corrected frequencies'</span>)</span>
<span id="cb14-63"><a href="#cb14-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-64"><a href="#cb14-64" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Skip E1 plot if not including compartments</span></span>
<span id="cb14-65"><a href="#cb14-65" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> include_compartments:</span>
<span id="cb14-66"><a href="#cb14-66" aria-hidden="true" tabindex="-1"></a>            <span class="cf">for</span> view, e1 <span class="kw">in</span> e1_dict.items():</span>
<span id="cb14-67"><a href="#cb14-67" aria-hidden="true" tabindex="-1"></a>                <span class="co"># subset the e1 to region (chrX:start-end)/binsize</span></span>
<span id="cb14-68"><a href="#cb14-68" aria-hidden="true" tabindex="-1"></a>                start_bin <span class="op">=</span> <span class="bu">int</span>(chrom_start[<span class="dv">0</span>]<span class="op">/</span>binsize)</span>
<span id="cb14-69"><a href="#cb14-69" aria-hidden="true" tabindex="-1"></a>                end_bin <span class="op">=</span> <span class="bu">int</span>(chrom_start[<span class="op">-</span><span class="dv">1</span>]<span class="op">/</span>binsize)</span>
<span id="cb14-70"><a href="#cb14-70" aria-hidden="true" tabindex="-1"></a>                e1 <span class="op">=</span> e1.iloc[start_bin:end_bin<span class="op">+</span><span class="dv">1</span>].reset_index(drop<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb14-71"><a href="#cb14-71" aria-hidden="true" tabindex="-1"></a>                ax1 <span class="op">=</span> divider.append_axes(<span class="st">"top"</span>, size<span class="op">=</span><span class="st">"10%"</span>, pad<span class="op">=</span><span class="fl">0.05</span>, sharex<span class="op">=</span>ax)</span>
<span id="cb14-72"><a href="#cb14-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-73"><a href="#cb14-73" aria-hidden="true" tabindex="-1"></a>                <span class="co"># Create stairs for E1 values</span></span>
<span id="cb14-74"><a href="#cb14-74" aria-hidden="true" tabindex="-1"></a>                x <span class="op">=</span> np.zeros(<span class="dv">2</span> <span class="op">*</span> chrom_start.size)</span>
<span id="cb14-75"><a href="#cb14-75" aria-hidden="true" tabindex="-1"></a>                y <span class="op">=</span> np.zeros(<span class="dv">2</span> <span class="op">*</span> chrom_start.size)</span>
<span id="cb14-76"><a href="#cb14-76" aria-hidden="true" tabindex="-1"></a>                x[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> np.arange(chrom_start.size)</span>
<span id="cb14-77"><a href="#cb14-77" aria-hidden="true" tabindex="-1"></a>                x[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> np.arange(chrom_start.size)</span>
<span id="cb14-78"><a href="#cb14-78" aria-hidden="true" tabindex="-1"></a>                y[<span class="dv">0</span>::<span class="dv">2</span>] <span class="op">=</span> e1</span>
<span id="cb14-79"><a href="#cb14-79" aria-hidden="true" tabindex="-1"></a>                y[<span class="dv">1</span>::<span class="dv">2</span>] <span class="op">=</span> e1</span>
<span id="cb14-80"><a href="#cb14-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-81"><a href="#cb14-81" aria-hidden="true" tabindex="-1"></a>                ax1.fill_between(x, y, <span class="dv">0</span>, where<span class="op">=</span>(y <span class="op">&gt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:red'</span>, ec<span class="op">=</span><span class="st">'None'</span>)</span>
<span id="cb14-82"><a href="#cb14-82" aria-hidden="true" tabindex="-1"></a>                ax1.fill_between(x, y, <span class="dv">0</span>, where<span class="op">=</span>(y <span class="op">&lt;</span> <span class="dv">0</span>), color<span class="op">=</span><span class="st">'tab:blue'</span>, ec<span class="op">=</span><span class="st">'None'</span>)</span>
<span id="cb14-83"><a href="#cb14-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-84"><a href="#cb14-84" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span> <span class="bu">len</span>(e1_dict) <span class="op">&gt;</span> <span class="dv">1</span>:</span>
<span id="cb14-85"><a href="#cb14-85" aria-hidden="true" tabindex="-1"></a>                    ax1.set_ylabel(view, rotation<span class="op">=</span><span class="dv">0</span>, ha<span class="op">=</span><span class="st">'right'</span>, va<span class="op">=</span><span class="st">'center'</span>, fontsize<span class="op">=</span><span class="dv">7</span>)</span>
<span id="cb14-86"><a href="#cb14-86" aria-hidden="true" tabindex="-1"></a>                <span class="cf">else</span>:</span>
<span id="cb14-87"><a href="#cb14-87" aria-hidden="true" tabindex="-1"></a>                    ax1.set_ylabel(<span class="st">'E1'</span>, rotation<span class="op">=</span><span class="dv">0</span>, ha<span class="op">=</span><span class="st">'right'</span>, va<span class="op">=</span><span class="st">'center'</span>, fontsize<span class="op">=</span><span class="dv">7</span>)</span>
<span id="cb14-88"><a href="#cb14-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-89"><a href="#cb14-89" aria-hidden="true" tabindex="-1"></a>                ax1.set_ylim(<span class="op">-</span><span class="fl">0.8</span>, <span class="fl">0.8</span>)</span>
<span id="cb14-90"><a href="#cb14-90" aria-hidden="true" tabindex="-1"></a>                ax1.set_yticks([])</span>
<span id="cb14-91"><a href="#cb14-91" aria-hidden="true" tabindex="-1"></a>                ax1.set_xticks([])</span>
<span id="cb14-92"><a href="#cb14-92" aria-hidden="true" tabindex="-1"></a>                ax1.spines[:].set_visible(<span class="va">False</span>)</span>
<span id="cb14-93"><a href="#cb14-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-94"><a href="#cb14-94" aria-hidden="true" tabindex="-1"></a>        <span class="co"># # Set the figure title</span></span>
<span id="cb14-95"><a href="#cb14-95" aria-hidden="true" tabindex="-1"></a>            <span class="co"># ax1.set_title(title)</span></span>
<span id="cb14-96"><a href="#cb14-96" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb14-97"><a href="#cb14-97" aria-hidden="true" tabindex="-1"></a>        fig.tight_layout()</span>
<span id="cb14-98"><a href="#cb14-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-99"><a href="#cb14-99" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Display the figure for Quarto to capture</span></span>
<span id="cb14-100"><a href="#cb14-100" aria-hidden="true" tabindex="-1"></a>        plt.show()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="plot-intervals" class="level3">
<h3 class="anchored" data-anchor-id="plot-intervals">Plot intervals</h3>
<p>Plot genomic intervals as tracks (up to 3)</p>
<div id="cell-24" class="cell" data-execution_count="12">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> matplotlib.patches <span class="im">import</span> Rectangle</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> mpl_toolkits.axes_grid1 <span class="im">import</span> make_axes_locatable</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> genominterv <span class="im">import</span> interval_intersect</span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> plot_regions(query<span class="op">=</span><span class="va">None</span>, annot<span class="op">=</span><span class="va">None</span>, intersect<span class="op">=</span><span class="va">None</span>, </span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>                 track_titles<span class="op">=</span>[<span class="st">'query'</span>, <span class="st">'annot'</span>, <span class="st">'intersect'</span>], </span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>                 figsize<span class="op">=</span>(<span class="dv">10</span>, <span class="dv">1</span>),</span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>                 title<span class="op">=</span><span class="st">'chrom'</span>,</span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>                 alpha<span class="op">=</span><span class="fl">0.8</span>, lw<span class="op">=</span><span class="dv">0</span>):</span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a>    chrom <span class="op">=</span> annot[<span class="st">'chrom'</span>].unique()[<span class="dv">0</span>]</span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a>    chromsizes <span class="op">=</span> pd.read_csv(<span class="st">'../data/rheMac10.filtered.chrom.sizes'</span>, sep<span class="op">=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>, header<span class="op">=</span><span class="va">None</span>, names<span class="op">=</span>[<span class="st">'chrom'</span>, <span class="st">'size'</span>])</span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a>    base_size <span class="op">=</span> chromsizes[chromsizes[<span class="st">'chrom'</span>] <span class="op">==</span> chrom][<span class="st">'size'</span>].values[<span class="dv">0</span>]</span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a>    annot_size <span class="op">=</span> annot[<span class="st">'end'</span>].<span class="bu">max</span>()</span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a>    query_size <span class="op">=</span> query[<span class="st">'end'</span>].<span class="bu">max</span>()</span>
<span id="cb15-21"><a href="#cb15-21" aria-hidden="true" tabindex="-1"></a>    chromsize <span class="op">=</span> <span class="bu">max</span>(base_size, annot_size, query_size)</span>
<span id="cb15-22"><a href="#cb15-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-23"><a href="#cb15-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-24"><a href="#cb15-24" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> title <span class="op">==</span> <span class="st">'chrom'</span>:</span>
<span id="cb15-25"><a href="#cb15-25" aria-hidden="true" tabindex="-1"></a>        title <span class="op">=</span> chrom</span>
<span id="cb15-26"><a href="#cb15-26" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb15-27"><a href="#cb15-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-28"><a href="#cb15-28" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Define the plot size</span></span>
<span id="cb15-29"><a href="#cb15-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-30"><a href="#cb15-30" aria-hidden="true" tabindex="-1"></a>    f, ax <span class="op">=</span> plt.subplots(figsize<span class="op">=</span>figsize, sharex<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb15-31"><a href="#cb15-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-32"><a href="#cb15-32" aria-hidden="true" tabindex="-1"></a>    ax.spines[:].set_visible(<span class="va">False</span>)</span>
<span id="cb15-33"><a href="#cb15-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-34"><a href="#cb15-34" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Plot the annot</span></span>
<span id="cb15-35"><a href="#cb15-35" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Iterate over each interval in the DataFrame</span></span>
<span id="cb15-36"><a href="#cb15-36" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> start, end <span class="kw">in</span> <span class="bu">zip</span>(annot[<span class="st">'start'</span>], annot[<span class="st">'end'</span>]):</span>
<span id="cb15-37"><a href="#cb15-37" aria-hidden="true" tabindex="-1"></a>        rect <span class="op">=</span> Rectangle((start, <span class="fl">0.1</span>), width<span class="op">=</span>end<span class="op">-</span>start, height<span class="op">=</span><span class="fl">0.9</span>, color<span class="op">=</span><span class="st">'tab:red'</span>, linewidth<span class="op">=</span>lw, alpha<span class="op">=</span>alpha)</span>
<span id="cb15-38"><a href="#cb15-38" aria-hidden="true" tabindex="-1"></a>        ax.add_patch(rect)</span>
<span id="cb15-39"><a href="#cb15-39" aria-hidden="true" tabindex="-1"></a>        ax.spines[<span class="st">'bottom'</span>].set_visible(<span class="va">True</span>)</span>
<span id="cb15-40"><a href="#cb15-40" aria-hidden="true" tabindex="-1"></a>        ax.set_ylabel(track_titles[<span class="dv">1</span>], rotation<span class="op">=</span><span class="dv">0</span>, labelpad<span class="op">=</span><span class="dv">10</span>, ha<span class="op">=</span><span class="st">'right'</span>, va<span class="op">=</span><span class="st">'center'</span>)</span>
<span id="cb15-41"><a href="#cb15-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-42"><a href="#cb15-42" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb15-43"><a href="#cb15-43" aria-hidden="true" tabindex="-1"></a>    divider <span class="op">=</span> make_axes_locatable(ax)</span>
<span id="cb15-44"><a href="#cb15-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-45"><a href="#cb15-45" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> query <span class="kw">is</span> <span class="kw">not</span> <span class="va">None</span>:</span>
<span id="cb15-46"><a href="#cb15-46" aria-hidden="true" tabindex="-1"></a>        qax <span class="op">=</span> divider.append_axes(<span class="st">"top"</span>, size<span class="op">=</span><span class="st">"100%"</span>, pad<span class="op">=</span><span class="fl">0.05</span>, sharex<span class="op">=</span>ax)</span>
<span id="cb15-47"><a href="#cb15-47" aria-hidden="true" tabindex="-1"></a>        qax.xaxis.set_visible(<span class="va">False</span>)</span>
<span id="cb15-48"><a href="#cb15-48" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Plot the query</span></span>
<span id="cb15-49"><a href="#cb15-49" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> start, end <span class="kw">in</span> <span class="bu">zip</span>(query[<span class="st">'start'</span>], query[<span class="st">'end'</span>]):</span>
<span id="cb15-50"><a href="#cb15-50" aria-hidden="true" tabindex="-1"></a>            rect <span class="op">=</span> Rectangle((start, <span class="fl">0.1</span>), width<span class="op">=</span>end<span class="op">-</span>start, height<span class="op">=</span><span class="fl">0.9</span>, color<span class="op">=</span><span class="st">'tab:blue'</span>, linewidth<span class="op">=</span>lw, alpha<span class="op">=</span>alpha)</span>
<span id="cb15-51"><a href="#cb15-51" aria-hidden="true" tabindex="-1"></a>            qax.add_patch(rect)</span>
<span id="cb15-52"><a href="#cb15-52" aria-hidden="true" tabindex="-1"></a>            qax.spines[:].set_visible(<span class="va">False</span>)</span>
<span id="cb15-53"><a href="#cb15-53" aria-hidden="true" tabindex="-1"></a>            qax.set_yticks([]) </span>
<span id="cb15-54"><a href="#cb15-54" aria-hidden="true" tabindex="-1"></a>            qax.set_title(title, loc<span class="op">=</span><span class="st">'left'</span>)</span>
<span id="cb15-55"><a href="#cb15-55" aria-hidden="true" tabindex="-1"></a>            qax.set_ylabel(track_titles[<span class="dv">0</span>], rotation<span class="op">=</span><span class="dv">0</span>, labelpad<span class="op">=</span><span class="dv">10</span>, ha<span class="op">=</span><span class="st">'right'</span>, va<span class="op">=</span><span class="st">'center'</span>)</span>
<span id="cb15-56"><a href="#cb15-56" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb15-57"><a href="#cb15-57" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> intersect <span class="kw">is</span> <span class="kw">not</span> <span class="va">None</span>:</span>
<span id="cb15-58"><a href="#cb15-58" aria-hidden="true" tabindex="-1"></a>        iax <span class="op">=</span> divider.append_axes(<span class="st">"bottom"</span>, size<span class="op">=</span><span class="st">"100%"</span>, pad<span class="op">=</span><span class="fl">0.05</span>, sharex<span class="op">=</span>ax)</span>
<span id="cb15-59"><a href="#cb15-59" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Invisible x-axis for 'annot' (intersect ie below) </span></span>
<span id="cb15-60"><a href="#cb15-60" aria-hidden="true" tabindex="-1"></a>        ax.xaxis.set_visible(<span class="va">False</span>)</span>
<span id="cb15-61"><a href="#cb15-61" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Plot the intersect</span></span>
<span id="cb15-62"><a href="#cb15-62" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> start, end <span class="kw">in</span> <span class="bu">zip</span>(intersect[<span class="st">'start'</span>], intersect[<span class="st">'end'</span>]):</span>
<span id="cb15-63"><a href="#cb15-63" aria-hidden="true" tabindex="-1"></a>            rect <span class="op">=</span> Rectangle((start, <span class="fl">0.1</span>), width<span class="op">=</span>end<span class="op">-</span>start, height<span class="op">=</span><span class="fl">0.9</span>, color<span class="op">=</span><span class="st">'tab:green'</span>, linewidth<span class="op">=</span>lw, alpha<span class="op">=</span>alpha)</span>
<span id="cb15-64"><a href="#cb15-64" aria-hidden="true" tabindex="-1"></a>            iax.add_patch(rect)</span>
<span id="cb15-65"><a href="#cb15-65" aria-hidden="true" tabindex="-1"></a>            iax.spines[:].set_visible(<span class="va">False</span>)</span>
<span id="cb15-66"><a href="#cb15-66" aria-hidden="true" tabindex="-1"></a>            iax.set_yticks([]) </span>
<span id="cb15-67"><a href="#cb15-67" aria-hidden="true" tabindex="-1"></a>            ax.spines[<span class="st">'bottom'</span>].set_visible(<span class="va">False</span>)</span>
<span id="cb15-68"><a href="#cb15-68" aria-hidden="true" tabindex="-1"></a>            iax.spines[<span class="st">'bottom'</span>].set_visible(<span class="va">True</span>)</span>
<span id="cb15-69"><a href="#cb15-69" aria-hidden="true" tabindex="-1"></a>            iax.set_ylabel(track_titles[<span class="dv">2</span>], rotation<span class="op">=</span><span class="dv">0</span>, labelpad<span class="op">=</span><span class="dv">10</span>, ha<span class="op">=</span><span class="st">'right'</span>, va<span class="op">=</span><span class="st">'center'</span>)</span>
<span id="cb15-70"><a href="#cb15-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-71"><a href="#cb15-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-72"><a href="#cb15-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-73"><a href="#cb15-73" aria-hidden="true" tabindex="-1"></a>    ax.set_yticks([])</span>
<span id="cb15-74"><a href="#cb15-74" aria-hidden="true" tabindex="-1"></a>    ax.set_xlim(<span class="dv">0</span>, chromsize)</span>
<span id="cb15-75"><a href="#cb15-75" aria-hidden="true" tabindex="-1"></a>    ticks <span class="op">=</span> np.linspace(<span class="dv">0</span>, chromsize, num<span class="op">=</span><span class="dv">5</span>)</span>
<span id="cb15-76"><a href="#cb15-76" aria-hidden="true" tabindex="-1"></a>    ax.set_xticks(ticks) </span>
<span id="cb15-77"><a href="#cb15-77" aria-hidden="true" tabindex="-1"></a>    ax.set_xticklabels([<span class="ss">f'</span><span class="sc">{</span><span class="bu">int</span>(t<span class="op">/</span><span class="fl">1e6</span>)<span class="sc">}</span><span class="ss"> Mbp'</span> <span class="cf">for</span> t <span class="kw">in</span> ticks])</span>
<span id="cb15-78"><a href="#cb15-78" aria-hidden="true" tabindex="-1"></a>    f.tight_layout()</span>
<span id="cb15-79"><a href="#cb15-79" aria-hidden="true" tabindex="-1"></a>    plt.show()</span>
<span id="cb15-80"><a href="#cb15-80" aria-hidden="true" tabindex="-1"></a>    <span class="co">#plt.tight_layout()</span></span>
<span id="cb15-81"><a href="#cb15-81" aria-hidden="true" tabindex="-1"></a></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="svedig-tabel" class="level3">
<h3 class="anchored" data-anchor-id="svedig-tabel">Svedig tabel</h3>
<div id="cell-26" class="cell" data-execution_count="13">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>all_tests <span class="op">=</span> pd.read_csv(<span class="st">'../results/all_tests.csv'</span>, index_col<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> svedig_tabel(orig_df, index, columns, values, </span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>                 cmap<span class="op">=</span><span class="st">'Reds'</span>, col_order<span class="op">=</span><span class="va">None</span>, col_snames<span class="op">=</span><span class="va">None</span>):</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>    <span class="co">"""</span></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a><span class="co">    Creates a styled pivot table from a DataFrame, filtering by significance, </span></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a><span class="co">    applying a log transformation, and adding visual styling.</span></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a><span class="co">    Parameters:</span></span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a><span class="co">    ----------</span></span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a><span class="co">    orig_df : pd.DataFrame</span></span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a><span class="co">        The original DataFrame containing data to be processed.</span></span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a><span class="co">    index : str</span></span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a><span class="co">        Column name to use as the row index in the pivot table.</span></span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a><span class="co">    columns : str</span></span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a><span class="co">        Column name to use as the column headers in the pivot table.</span></span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a><span class="co">    values : str</span></span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a><span class="co">        Column name whose values populate the pivot table cells.</span></span>
<span id="cb16-20"><a href="#cb16-20" aria-hidden="true" tabindex="-1"></a><span class="co">    cmap : str, optional</span></span>
<span id="cb16-21"><a href="#cb16-21" aria-hidden="true" tabindex="-1"></a><span class="co">        Colormap to use for the background gradient styling (default is 'Reds').</span></span>
<span id="cb16-22"><a href="#cb16-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-23"><a href="#cb16-23" aria-hidden="true" tabindex="-1"></a><span class="co">    Returns:</span></span>
<span id="cb16-24"><a href="#cb16-24" aria-hidden="true" tabindex="-1"></a><span class="co">    -------</span></span>
<span id="cb16-25"><a href="#cb16-25" aria-hidden="true" tabindex="-1"></a><span class="co">    pd.io.formats.style.Styler</span></span>
<span id="cb16-26"><a href="#cb16-26" aria-hidden="true" tabindex="-1"></a><span class="co">        A styled DataFrame object with the pivot table, including gradient-based</span></span>
<span id="cb16-27"><a href="#cb16-27" aria-hidden="true" tabindex="-1"></a><span class="co">        cell coloring, transparent NaN values, formatted numbers, and table styles.</span></span>
<span id="cb16-28"><a href="#cb16-28" aria-hidden="true" tabindex="-1"></a><span class="co">    """</span></span>
<span id="cb16-29"><a href="#cb16-29" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-30"><a href="#cb16-30" aria-hidden="true" tabindex="-1"></a>    df <span class="op">=</span> (orig_df</span>
<span id="cb16-31"><a href="#cb16-31" aria-hidden="true" tabindex="-1"></a>     .assign(log10p<span class="op">=</span>np.log10(all_tests.p))</span>
<span id="cb16-32"><a href="#cb16-32" aria-hidden="true" tabindex="-1"></a>     .loc[(all_tests.p <span class="op">&lt;</span> <span class="fl">0.05</span>)]</span>
<span id="cb16-33"><a href="#cb16-33" aria-hidden="true" tabindex="-1"></a>     .pivot(index<span class="op">=</span>index, columns<span class="op">=</span>columns, values<span class="op">=</span>values)</span>
<span id="cb16-34"><a href="#cb16-34" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb16-35"><a href="#cb16-35" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> col_order:</span>
<span id="cb16-36"><a href="#cb16-36" aria-hidden="true" tabindex="-1"></a>        col_bool <span class="op">=</span> [x <span class="kw">in</span> df.columns <span class="cf">for</span> x <span class="kw">in</span> col_order]</span>
<span id="cb16-37"><a href="#cb16-37" aria-hidden="true" tabindex="-1"></a>        col_order <span class="op">=</span> [x <span class="cf">for</span> x <span class="kw">in</span> col_order <span class="cf">if</span> x <span class="kw">in</span> df.columns]</span>
<span id="cb16-38"><a href="#cb16-38" aria-hidden="true" tabindex="-1"></a>        df <span class="op">=</span> df[col_order]</span>
<span id="cb16-39"><a href="#cb16-39" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> col_snames:</span>
<span id="cb16-40"><a href="#cb16-40" aria-hidden="true" tabindex="-1"></a>        col_snames <span class="op">=</span> [col_snames[i] <span class="cf">for</span> i,x <span class="kw">in</span> <span class="bu">enumerate</span>(col_bool) <span class="cf">if</span> x]</span>
<span id="cb16-41"><a href="#cb16-41" aria-hidden="true" tabindex="-1"></a>        df <span class="op">=</span> df.rename(columns <span class="op">=</span> {x:col_snames[i] <span class="cf">for</span> i,x <span class="kw">in</span> <span class="bu">enumerate</span>(df.columns.tolist())})</span>
<span id="cb16-42"><a href="#cb16-42" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb16-43"><a href="#cb16-43" aria-hidden="true" tabindex="-1"></a>        df <span class="op">=</span> df.rename(columns <span class="op">=</span> {x:x.replace(<span class="st">'_'</span>, <span class="st">'&lt;br&gt;'</span>) <span class="cf">for</span> x <span class="kw">in</span> df.columns.tolist()})</span>
<span id="cb16-44"><a href="#cb16-44" aria-hidden="true" tabindex="-1"></a>    df <span class="op">=</span> (df.style</span>
<span id="cb16-45"><a href="#cb16-45" aria-hidden="true" tabindex="-1"></a>     .background_gradient(subset<span class="op">=</span>df.columns, axis<span class="op">=</span><span class="va">None</span>, cmap<span class="op">=</span>cmap, vmin<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb16-46"><a href="#cb16-46" aria-hidden="true" tabindex="-1"></a>     .<span class="bu">map</span>(<span class="kw">lambda</span> x: <span class="st">'color: transparent; background-color: transparent'</span> <span class="cf">if</span> np.isnan(x) <span class="cf">else</span> <span class="st">''</span>)</span>
<span id="cb16-47"><a href="#cb16-47" aria-hidden="true" tabindex="-1"></a>     .<span class="bu">format</span>(<span class="st">'</span><span class="sc">{:.3f}</span><span class="st">'</span>)</span>
<span id="cb16-48"><a href="#cb16-48" aria-hidden="true" tabindex="-1"></a>     .set_table_styles(</span>
<span id="cb16-49"><a href="#cb16-49" aria-hidden="true" tabindex="-1"></a>                {c: [{<span class="st">'selector'</span>: <span class="st">''</span>, </span>
<span id="cb16-50"><a href="#cb16-50" aria-hidden="true" tabindex="-1"></a>                      <span class="st">'props'</span>: [(<span class="st">'min-width'</span>, <span class="st">'20px'</span>)],</span>
<span id="cb16-51"><a href="#cb16-51" aria-hidden="true" tabindex="-1"></a>                     }] <span class="cf">for</span> c <span class="kw">in</span> df.columns}, overwrite<span class="op">=</span><span class="va">False</span></span>
<span id="cb16-52"><a href="#cb16-52" aria-hidden="true" tabindex="-1"></a>     )</span>
<span id="cb16-53"><a href="#cb16-53" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb16-54"><a href="#cb16-54" aria-hidden="true" tabindex="-1"></a>    <span class="cf">return</span> df</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div id="cell-27" class="cell" data-execution_count="14">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"That should be all the setup :-)"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>That should be all the setup :-)</code></pre>
</div>
</div>
</section>
</section>
<section id="matrices-e1" class="level2">
<h2 class="anchored" data-anchor-id="matrices-e1">Matrices / E1</h2>
<section id="e1-rs100500" class="level3">
<h3 class="anchored" data-anchor-id="e1-rs100500">E1: RS100/500</h3>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Compare the compartments of the round_spermatid, 500kb, all views</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>res <span class="op">=</span> [<span class="st">'100'</span>, <span class="st">'500'</span>]</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>name <span class="op">=</span> [<span class="st">'round_spermatid'</span>]</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>run <span class="op">=</span> [<span class="st">'recPE'</span>]</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>group <span class="op">=</span> clr_df.query(<span class="st">'name in @name and resolution in @res and run in @run'</span>)</span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>plot_for_quarto(group)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div id="fig-e1-matrix-full-arms-10mb-round_spermatid" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-e1-matrix-full-arms-10mb-round_spermatid-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-e1-matrix-full-arms-10mb-round_spermatid" style="flex-basis: 50.0%;justify-content: flex-start;">
<div id="fig-e1-matrix-full-arms-10mb-round_spermatid-1" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-e1-matrix-full-arms-10mb-round_spermatid-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="07_various_plotting_files/figure-html/fig-e1-matrix-full-arms-10mb-round_spermatid-output-1.png" class="lightbox" data-gallery="fig-e1-matrix-full-arms-10mb-round_spermatid" title="Figure&nbsp;8.1&nbsp;(a): 100kb resolution"><img src="07_various_plotting_files/figure-html/fig-e1-matrix-full-arms-10mb-round_spermatid-output-1.png" data-ref-parent="fig-e1-matrix-full-arms-10mb-round_spermatid" width="297" height="277" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-e1-matrix-full-arms-10mb-round_spermatid-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) 100kb resolution
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-e1-matrix-full-arms-10mb-round_spermatid" style="flex-basis: 50.0%;justify-content: flex-start;">
<div id="fig-e1-matrix-full-arms-10mb-round_spermatid-2" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-e1-matrix-full-arms-10mb-round_spermatid-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="07_various_plotting_files/figure-html/fig-e1-matrix-full-arms-10mb-round_spermatid-output-2.png" class="lightbox" data-gallery="fig-e1-matrix-full-arms-10mb-round_spermatid" title="Figure&nbsp;8.1&nbsp;(b): 500kb resolution"><img src="07_various_plotting_files/figure-html/fig-e1-matrix-full-arms-10mb-round_spermatid-output-2.png" data-ref-parent="fig-e1-matrix-full-arms-10mb-round_spermatid" width="297" height="277" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-e1-matrix-full-arms-10mb-round_spermatid-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) 500kb resolution
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-e1-matrix-full-arms-10mb-round_spermatid-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;8.1: E1 eigenvector values for merged round spermatid samples at a) 100kb or b) 500kb resolution, as well as the interaction matrix. E1 was restricted to either Full-chromosome (top), Chromosome-arms (middle), or 10Mb windows (bottom)
</figcaption>
</figure>
</div>
</section>
<section id="pe-vs-recpe-rs100kb" class="level3">
<h3 class="anchored" data-anchor-id="pe-vs-recpe-rs100kb">PE vs recPE (RS100kb)</h3>
<div id="cell-32" class="cell" data-execution_count="18">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Compare PE and recPE compartments for round_spermatid, 100kb</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>group <span class="op">=</span> clr_df.query(<span class="st">'resolution == "100" and name == "round_spermatid"'</span>)</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>plot_grouped(group, figsize<span class="op">=</span>(<span class="dv">6</span>, <span class="dv">6</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><a href="07_various_plotting_files/figure-html/cell-17-output-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-3"><img src="07_various_plotting_files/figure-html/cell-17-output-1.png" width="612" height="331" class="figure-img"></a></p>
</figure>
</div>
</div>
</div>
<div id="cell-33" class="cell" data-execution_count="41">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>df <span class="op">=</span> clr_df.query(<span class="st">'resolution == "500" and name == "round_spermatid" and run == "PE"'</span>)</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>plot_grouped(df, include_compartments<span class="op">=</span><span class="va">False</span>, figsize<span class="op">=</span>(<span class="dv">3</span>, <span class="dv">3</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><a href="07_various_plotting_files/figure-html/cell-18-output-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-4"><img src="07_various_plotting_files/figure-html/cell-18-output-1.png" width="356" height="282" class="figure-img"></a></p>
</figure>
</div>
</div>
</div>
</section>
<section id="pe-vs.-recpe-rs100kb" class="level3">
<h3 class="anchored" data-anchor-id="pe-vs.-recpe-rs100kb">PE vs.&nbsp;recPE: RS100kb</h3>
<div class="cell" data-layout="[[-0.5,5,5,-0.5],[-0.5,5,5,-0.5]]" data-execution_count="21">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>df <span class="op">=</span> clr_df.query(<span class="st">'resolution == "100" and name == "round_spermatid"'</span>)</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Check the order of the group</span></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a><span class="co">#print(df[['name', 'resolution', 'run']].reset_index(drop=True))</span></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>plot_for_quarto(df, view_order<span class="op">=</span>[<span class="st">'10Mb'</span>,<span class="st">'arms'</span>], figsize<span class="op">=</span>(<span class="dv">5</span>, <span class="dv">5</span>))</span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>start, end <span class="op">=</span> <span class="dv">70_000_000</span>, <span class="dv">78_000_000</span></span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>region <span class="op">=</span> (<span class="st">'chrX'</span>, start, end)</span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>plot_for_quarto(df, region<span class="op">=</span>region, include_compartments<span class="op">=</span><span class="va">True</span>, view_order<span class="op">=</span>[<span class="st">'10Mb'</span>,<span class="st">'arms'</span>], figsize<span class="op">=</span>(<span class="dv">5</span>, <span class="dv">5</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div id="fig-rs100-recpe-pe" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-rs100-recpe-pe-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 4.6%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-rs100-recpe-pe" style="flex-basis: 45.5%;justify-content: flex-start;">
<div id="fig-rs100-recpe-pe-1" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-rs100-recpe-pe-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="07_various_plotting_files/figure-html/fig-rs100-recpe-pe-output-1.png" class="lightbox" data-gallery="fig-rs100-recpe-pe" title="Figure&nbsp;8.2&nbsp;(a): 5unique: chrX:start-end"><img src="07_various_plotting_files/figure-html/fig-rs100-recpe-pe-output-1.png" data-ref-parent="fig-rs100-recpe-pe" width="247" height="195" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-rs100-recpe-pe-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) <code>5unique</code>: chrX:start-end
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-rs100-recpe-pe" style="flex-basis: 45.5%;justify-content: flex-start;">
<div id="fig-rs100-recpe-pe-2" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-rs100-recpe-pe-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="07_various_plotting_files/figure-html/fig-rs100-recpe-pe-output-2.png" class="lightbox" data-gallery="fig-rs100-recpe-pe" title="Figure&nbsp;8.2&nbsp;(b): mask: chrX:start-end"><img src="07_various_plotting_files/figure-html/fig-rs100-recpe-pe-output-2.png" data-ref-parent="fig-rs100-recpe-pe" width="247" height="195" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-rs100-recpe-pe-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) <code>mask</code>: chrX:start-end
</figcaption>
</figure>
</div>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 4.6%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
</div>
<div class="quarto-layout-row">
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 4.6%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-rs100-recpe-pe" style="flex-basis: 45.5%;justify-content: flex-start;">
<div id="fig-rs100-recpe-pe-3" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-rs100-recpe-pe-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="07_various_plotting_files/figure-html/fig-rs100-recpe-pe-output-3.png" class="lightbox" data-gallery="fig-rs100-recpe-pe" title="Figure&nbsp;8.2&nbsp;(c): 5unique: chrX:70Mb-78Mb"><img src="07_various_plotting_files/figure-html/fig-rs100-recpe-pe-output-3.png" data-ref-parent="fig-rs100-recpe-pe" width="247" height="201" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-rs100-recpe-pe-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(c) <code>5unique</code>: chrX:70Mb-78Mb
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-rs100-recpe-pe" style="flex-basis: 45.5%;justify-content: flex-start;">
<div id="fig-rs100-recpe-pe-4" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-rs100-recpe-pe-4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="07_various_plotting_files/figure-html/fig-rs100-recpe-pe-output-4.png" class="lightbox" data-gallery="fig-rs100-recpe-pe" title="Figure&nbsp;8.2&nbsp;(d): mask: chrX:70Mb-78Mb"><img src="07_various_plotting_files/figure-html/fig-rs100-recpe-pe-output-4.png" data-ref-parent="fig-rs100-recpe-pe" width="247" height="201" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-rs100-recpe-pe-4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(d) <code>mask</code>: chrX:70Mb-78Mb
</figcaption>
</figure>
</div>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 4.6%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-rs100-recpe-pe-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;8.2: Round Spermatid (RS) at 100kb, comparing the impact of parsing parameters
</figcaption>
</figure>
</div>
</div>
</section>
</section>
<section id="intervals" class="level2">
<h2 class="anchored" data-anchor-id="intervals">Intervals</h2>
<section id="perecpe-compare-compartments-100kb" class="level3">
<h3 class="anchored" data-anchor-id="perecpe-compare-compartments-100kb">PE/recPE compare compartments (100kb)</h3>
<div class="cell" data-layout="[[-0.6,2.5,2.5,-0.6],[-0.6,2.5,2.5,-0.6]]" data-execution_count="103">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> genominterv <span class="im">import</span> interval_diff</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a><span class="co">### Plot the intervals and the diff between </span><span class="al">###</span><span class="co"> </span></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Round Spermatid arms</span></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>query <span class="op">=</span> comps[<span class="st">'PE'</span>][<span class="st">'round_spermatid_100kb_arms'</span>]</span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>annot <span class="op">=</span> comps[<span class="st">'recPE'</span>][<span class="st">'round_spermatid_100kb_arms'</span>]</span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a>plot_regions(query<span class="op">=</span>query, annot<span class="op">=</span>annot, </span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a>             intersect<span class="op">=</span>interval_diff(query, annot), </span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a>             track_titles<span class="op">=</span>[<span class="st">'PE'</span>, <span class="st">'recPE'</span>, <span class="st">'Diff'</span>],</span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a>             figsize<span class="op">=</span>(<span class="fl">2.5</span>, <span class="fl">0.8</span>),</span>
<span id="cb23-13"><a href="#cb23-13" aria-hidden="true" tabindex="-1"></a>             title<span class="op">=</span><span class="va">None</span>)</span>
<span id="cb23-14"><a href="#cb23-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-15"><a href="#cb23-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Fibroblast arms</span></span>
<span id="cb23-16"><a href="#cb23-16" aria-hidden="true" tabindex="-1"></a>query <span class="op">=</span> comps[<span class="st">'PE'</span>][<span class="st">'fibroblast_100kb_arms'</span>]</span>
<span id="cb23-17"><a href="#cb23-17" aria-hidden="true" tabindex="-1"></a>annot <span class="op">=</span> comps[<span class="st">'recPE'</span>][<span class="st">'fibroblast_100kb_arms'</span>]</span>
<span id="cb23-18"><a href="#cb23-18" aria-hidden="true" tabindex="-1"></a>plot_regions(query<span class="op">=</span>query, annot<span class="op">=</span>annot, </span>
<span id="cb23-19"><a href="#cb23-19" aria-hidden="true" tabindex="-1"></a>             intersect<span class="op">=</span>interval_diff(query, annot), </span>
<span id="cb23-20"><a href="#cb23-20" aria-hidden="true" tabindex="-1"></a>             track_titles<span class="op">=</span>[<span class="st">'PE'</span>, <span class="st">'recPE'</span>, <span class="st">'Diff'</span>],</span>
<span id="cb23-21"><a href="#cb23-21" aria-hidden="true" tabindex="-1"></a>             figsize<span class="op">=</span>(<span class="fl">2.5</span>, <span class="fl">0.8</span>),</span>
<span id="cb23-22"><a href="#cb23-22" aria-hidden="true" tabindex="-1"></a>             title<span class="op">=</span><span class="va">None</span>)</span>
<span id="cb23-23"><a href="#cb23-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-24"><a href="#cb23-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Round Spermatid 10Mb</span></span>
<span id="cb23-25"><a href="#cb23-25" aria-hidden="true" tabindex="-1"></a>query <span class="op">=</span> comps[<span class="st">'PE'</span>][<span class="st">'round_spermatid_100kb_10Mb'</span>]</span>
<span id="cb23-26"><a href="#cb23-26" aria-hidden="true" tabindex="-1"></a>annot <span class="op">=</span> comps[<span class="st">'recPE'</span>][<span class="st">'round_spermatid_100kb_10Mb'</span>]</span>
<span id="cb23-27"><a href="#cb23-27" aria-hidden="true" tabindex="-1"></a>plot_regions(query<span class="op">=</span>query, annot<span class="op">=</span>annot, </span>
<span id="cb23-28"><a href="#cb23-28" aria-hidden="true" tabindex="-1"></a>             intersect<span class="op">=</span>interval_diff(query, annot), </span>
<span id="cb23-29"><a href="#cb23-29" aria-hidden="true" tabindex="-1"></a>             track_titles<span class="op">=</span>[<span class="st">'PE'</span>, <span class="st">'recPE'</span>, <span class="st">'Diff'</span>],</span>
<span id="cb23-30"><a href="#cb23-30" aria-hidden="true" tabindex="-1"></a>             figsize<span class="op">=</span>(<span class="fl">2.5</span>, <span class="fl">0.8</span>),</span>
<span id="cb23-31"><a href="#cb23-31" aria-hidden="true" tabindex="-1"></a>             title<span class="op">=</span><span class="va">None</span>)</span>
<span id="cb23-32"><a href="#cb23-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-33"><a href="#cb23-33" aria-hidden="true" tabindex="-1"></a><span class="co"># Fibroblast 10Mb</span></span>
<span id="cb23-34"><a href="#cb23-34" aria-hidden="true" tabindex="-1"></a>query <span class="op">=</span> comps[<span class="st">'PE'</span>][<span class="st">'fibroblast_100kb_10Mb'</span>]</span>
<span id="cb23-35"><a href="#cb23-35" aria-hidden="true" tabindex="-1"></a>annot <span class="op">=</span> comps[<span class="st">'recPE'</span>][<span class="st">'fibroblast_100kb_10Mb'</span>]</span>
<span id="cb23-36"><a href="#cb23-36" aria-hidden="true" tabindex="-1"></a>plot_regions(query<span class="op">=</span>query, annot<span class="op">=</span>annot, </span>
<span id="cb23-37"><a href="#cb23-37" aria-hidden="true" tabindex="-1"></a>             intersect<span class="op">=</span>interval_diff(query, annot), </span>
<span id="cb23-38"><a href="#cb23-38" aria-hidden="true" tabindex="-1"></a>             track_titles<span class="op">=</span>[<span class="st">'PE'</span>, <span class="st">'recPE'</span>, <span class="st">'Diff'</span>],</span>
<span id="cb23-39"><a href="#cb23-39" aria-hidden="true" tabindex="-1"></a>             figsize<span class="op">=</span>(<span class="fl">2.5</span>, <span class="fl">.8</span>),</span>
<span id="cb23-40"><a href="#cb23-40" aria-hidden="true" tabindex="-1"></a>             title<span class="op">=</span><span class="va">None</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div id="fig-rs-fb-100-pe-recpe-intervals" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-rs-fb-100-pe-recpe-intervals-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 9.7%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-rs-fb-100-pe-recpe-intervals" style="flex-basis: 40.3%;justify-content: flex-start;">
<div id="fig-rs-fb-100-pe-recpe-intervals-1" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-rs-fb-100-pe-recpe-intervals-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="07_various_plotting_files/figure-html/fig-rs-fb-100-pe-recpe-intervals-output-1.png" class="lightbox" data-gallery="fig-rs-fb-100-pe-recpe-intervals" title="Figure&nbsp;8.3&nbsp;(a): RS: arms"><img src="07_various_plotting_files/figure-html/fig-rs-fb-100-pe-recpe-intervals-output-1.png" data-ref-parent="fig-rs-fb-100-pe-recpe-intervals" width="249" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-rs-fb-100-pe-recpe-intervals-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) RS: arms
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-rs-fb-100-pe-recpe-intervals" style="flex-basis: 40.3%;justify-content: flex-start;">
<div id="fig-rs-fb-100-pe-recpe-intervals-2" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-rs-fb-100-pe-recpe-intervals-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="07_various_plotting_files/figure-html/fig-rs-fb-100-pe-recpe-intervals-output-2.png" class="lightbox" data-gallery="fig-rs-fb-100-pe-recpe-intervals" title="Figure&nbsp;8.3&nbsp;(b): Fib: arms"><img src="07_various_plotting_files/figure-html/fig-rs-fb-100-pe-recpe-intervals-output-2.png" data-ref-parent="fig-rs-fb-100-pe-recpe-intervals" width="249" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-rs-fb-100-pe-recpe-intervals-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) Fib: arms
</figcaption>
</figure>
</div>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 9.7%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
</div>
<div class="quarto-layout-row">
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 9.7%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-rs-fb-100-pe-recpe-intervals" style="flex-basis: 40.3%;justify-content: flex-start;">
<div id="fig-rs-fb-100-pe-recpe-intervals-3" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-rs-fb-100-pe-recpe-intervals-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="07_various_plotting_files/figure-html/fig-rs-fb-100-pe-recpe-intervals-output-3.png" class="lightbox" data-gallery="fig-rs-fb-100-pe-recpe-intervals" title="Figure&nbsp;8.3&nbsp;(c): RS: 10Mb"><img src="07_various_plotting_files/figure-html/fig-rs-fb-100-pe-recpe-intervals-output-3.png" data-ref-parent="fig-rs-fb-100-pe-recpe-intervals" width="249" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-rs-fb-100-pe-recpe-intervals-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(c) RS: 10Mb
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-rs-fb-100-pe-recpe-intervals" style="flex-basis: 40.3%;justify-content: flex-start;">
<div id="fig-rs-fb-100-pe-recpe-intervals-4" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-rs-fb-100-pe-recpe-intervals-4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="07_various_plotting_files/figure-html/fig-rs-fb-100-pe-recpe-intervals-output-4.png" class="lightbox" data-gallery="fig-rs-fb-100-pe-recpe-intervals" title="Figure&nbsp;8.3&nbsp;(d): Fib: 10Mb"><img src="07_various_plotting_files/figure-html/fig-rs-fb-100-pe-recpe-intervals-output-4.png" data-ref-parent="fig-rs-fb-100-pe-recpe-intervals" width="249" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-rs-fb-100-pe-recpe-intervals-4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(d) Fib: 10Mb
</figcaption>
</figure>
</div>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 9.7%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-rs-fb-100-pe-recpe-intervals-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;8.3: Round Spermatid (RS) and Fibroblast (Fb) at 100kb, comparing the impact of parsing parameters on A-compartment calling at different viewframes; <em>arms</em>, <em>10Mb</em>. PE: initial parse (masking complex walks); recPE: recommended parse (reporting the 5’most unique alignment of a complex walk).
</figcaption>
</figure>
</div>
</div>
<div class="cell" data-layout="[[-0.05,3.2,3.2,-0.05],[-0.05,3.2,3.2,-0.05]]" data-execution_count="20">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> genominterv <span class="im">import</span> interval_intersect</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>query <span class="op">=</span> ech90</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>fb_ann_comp <span class="op">=</span> comps[<span class="st">'recPE'</span>][<span class="st">'fibroblast_100kb_arms'</span>]</span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>rs_ann_comp <span class="op">=</span> comps[<span class="st">'recPE'</span>][<span class="st">'round_spermatid_100kb_arms'</span>]</span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>fb_ann_edge <span class="op">=</span> edges[<span class="st">'recPE'</span>][<span class="st">'fibroblast_100kb_arms'</span>]</span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>rs_ann_edge <span class="op">=</span> edges[<span class="st">'recPE'</span>][<span class="st">'round_spermatid_100kb_arms'</span>]</span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>size<span class="op">=</span>(<span class="fl">3.2</span>, <span class="fl">0.8</span>)</span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Fib and RS compartments</span></span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a>plot_regions(query<span class="op">=</span>query, annot<span class="op">=</span>fb_ann_comp,</span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a>             intersect<span class="op">=</span>interval_intersect(query, fb_ann_comp), </span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a>             track_titles<span class="op">=</span>[<span class="st">'ECH90'</span>, <span class="st">'A-Comp'</span>, <span class="st">'Intersect'</span>],</span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a>             figsize<span class="op">=</span>size,</span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a>             title<span class="op">=</span><span class="va">None</span>,</span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a>             lw<span class="op">=</span><span class="fl">0.3</span>)</span>
<span id="cb24-19"><a href="#cb24-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-20"><a href="#cb24-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-21"><a href="#cb24-21" aria-hidden="true" tabindex="-1"></a>plot_regions(query<span class="op">=</span>query, annot<span class="op">=</span>rs_ann_comp, </span>
<span id="cb24-22"><a href="#cb24-22" aria-hidden="true" tabindex="-1"></a>             intersect<span class="op">=</span>interval_intersect(query, rs_ann_comp), </span>
<span id="cb24-23"><a href="#cb24-23" aria-hidden="true" tabindex="-1"></a>             track_titles<span class="op">=</span>[<span class="st">'ECH90'</span>, <span class="st">'A-Comp'</span>, <span class="st">'Intersect'</span>], </span>
<span id="cb24-24"><a href="#cb24-24" aria-hidden="true" tabindex="-1"></a>             figsize<span class="op">=</span>size,</span>
<span id="cb24-25"><a href="#cb24-25" aria-hidden="true" tabindex="-1"></a>             title<span class="op">=</span><span class="va">None</span>,</span>
<span id="cb24-26"><a href="#cb24-26" aria-hidden="true" tabindex="-1"></a>             lw<span class="op">=</span><span class="fl">0.3</span>)</span>
<span id="cb24-27"><a href="#cb24-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-28"><a href="#cb24-28" aria-hidden="true" tabindex="-1"></a><span class="co"># Fib and RS edges</span></span>
<span id="cb24-29"><a href="#cb24-29" aria-hidden="true" tabindex="-1"></a>plot_regions(query<span class="op">=</span>query, annot<span class="op">=</span>fb_ann_edge,</span>
<span id="cb24-30"><a href="#cb24-30" aria-hidden="true" tabindex="-1"></a>             intersect<span class="op">=</span>interval_intersect(query, fb_ann_edge), </span>
<span id="cb24-31"><a href="#cb24-31" aria-hidden="true" tabindex="-1"></a>             track_titles<span class="op">=</span>[<span class="st">'ECH90'</span>, <span class="st">'Edges'</span>, <span class="st">'Intersect'</span>],</span>
<span id="cb24-32"><a href="#cb24-32" aria-hidden="true" tabindex="-1"></a>             figsize<span class="op">=</span>size,</span>
<span id="cb24-33"><a href="#cb24-33" aria-hidden="true" tabindex="-1"></a>             title<span class="op">=</span><span class="va">None</span>,</span>
<span id="cb24-34"><a href="#cb24-34" aria-hidden="true" tabindex="-1"></a>             lw<span class="op">=</span><span class="fl">0.3</span>, alpha<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb24-35"><a href="#cb24-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-36"><a href="#cb24-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-37"><a href="#cb24-37" aria-hidden="true" tabindex="-1"></a>plot_regions(query<span class="op">=</span>query, annot<span class="op">=</span>rs_ann_edge, </span>
<span id="cb24-38"><a href="#cb24-38" aria-hidden="true" tabindex="-1"></a>             intersect<span class="op">=</span>interval_intersect(query, rs_ann_edge), </span>
<span id="cb24-39"><a href="#cb24-39" aria-hidden="true" tabindex="-1"></a>             track_titles<span class="op">=</span>[<span class="st">'ECH90'</span>, <span class="st">'Edges'</span>, <span class="st">'Intersect'</span>],</span>
<span id="cb24-40"><a href="#cb24-40" aria-hidden="true" tabindex="-1"></a>             figsize<span class="op">=</span>size,</span>
<span id="cb24-41"><a href="#cb24-41" aria-hidden="true" tabindex="-1"></a>             title<span class="op">=</span><span class="va">None</span>,</span>
<span id="cb24-42"><a href="#cb24-42" aria-hidden="true" tabindex="-1"></a>             lw<span class="op">=</span><span class="fl">0.3</span>, alpha<span class="op">=</span><span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div id="fig-comps-edges-ech" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-comps-edges-ech-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 0.8%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-comps-edges-ech" style="flex-basis: 49.2%;justify-content: flex-start;">
<div id="fig-comps-edges-ech-1" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-comps-edges-ech-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="07_various_plotting_files/figure-html/fig-comps-edges-ech-output-1.png" class="lightbox" data-gallery="fig-comps-edges-ech" title="Figure&nbsp;8.4&nbsp;(a): Fibroblast A-compartments"><img src="07_various_plotting_files/figure-html/fig-comps-edges-ech-output-1.png" data-ref-parent="fig-comps-edges-ech" width="318" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-comps-edges-ech-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) Fibroblast A-compartments
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-comps-edges-ech" style="flex-basis: 49.2%;justify-content: flex-start;">
<div id="fig-comps-edges-ech-2" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-comps-edges-ech-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="07_various_plotting_files/figure-html/fig-comps-edges-ech-output-2.png" class="lightbox" data-gallery="fig-comps-edges-ech" title="Figure&nbsp;8.4&nbsp;(b): Round Spermatid A-compartments"><img src="07_various_plotting_files/figure-html/fig-comps-edges-ech-output-2.png" data-ref-parent="fig-comps-edges-ech" width="318" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-comps-edges-ech-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) Round Spermatid A-compartments
</figcaption>
</figure>
</div>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 0.8%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
</div>
<div class="quarto-layout-row">
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 0.8%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-comps-edges-ech" style="flex-basis: 49.2%;justify-content: flex-start;">
<div id="fig-comps-edges-ech-3" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-comps-edges-ech-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="07_various_plotting_files/figure-html/fig-comps-edges-ech-output-3.png" class="lightbox" data-gallery="fig-comps-edges-ech" title="Figure&nbsp;8.4&nbsp;(c): Fibroblast edges"><img src="07_various_plotting_files/figure-html/fig-comps-edges-ech-output-3.png" data-ref-parent="fig-comps-edges-ech" width="318" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-comps-edges-ech-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(c) Fibroblast edges
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-comps-edges-ech" style="flex-basis: 49.2%;justify-content: flex-start;">
<div id="fig-comps-edges-ech-4" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-comps-edges-ech-4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="07_various_plotting_files/figure-html/fig-comps-edges-ech-output-4.png" class="lightbox" data-gallery="fig-comps-edges-ech" title="Figure&nbsp;8.4&nbsp;(d): Round Spermatid edges"><img src="07_various_plotting_files/figure-html/fig-comps-edges-ech-output-4.png" data-ref-parent="fig-comps-edges-ech" width="318" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-comps-edges-ech-4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(d) Round Spermatid edges
</figcaption>
</figure>
</div>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 0.8%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-comps-edges-ech-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;8.4: Visual representation of the genomic intervals of ECH90, A-compartments (a+b), edges (c+d), and their intersections. Shown fibroblast (a+c) and round spermatid (b+d) at 100kb resolution and arms viewframe.
</figcaption>
</figure>
</div>
</div>
</section>
<section id="visualize-ech-enrichment" class="level3">
<h3 class="anchored" data-anchor-id="visualize-ech-enrichment">Visualize ECH enrichment</h3>
<div class="cell" data-layout="[-0.05,3.2,3.2,-0.05]" data-execution_count="15">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> genominterv <span class="im">import</span> interval_diff, interval_intersect, interval_union</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>complist <span class="op">=</span> [comps[<span class="st">'recPE'</span>][<span class="st">'round_spermatid_100kb_arms'</span>],</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>            comps[<span class="st">'recPE'</span>][<span class="st">'fibroblast_100kb_arms'</span>]]</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>edgelist <span class="op">=</span> [edges[<span class="st">'recPE'</span>][<span class="st">'round_spermatid_100kb_arms'</span>],</span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>            edges[<span class="st">'recPE'</span>][<span class="st">'fibroblast_100kb_arms'</span>]]</span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> comp, edge <span class="kw">in</span> <span class="bu">zip</span>(complist, edgelist):</span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>    int_comp <span class="op">=</span> interval_intersect(ech90, comp)</span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>    int_edge <span class="op">=</span> interval_intersect(ech90, edge)</span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a>    diff_int_compedge <span class="op">=</span> interval_diff(int_edge, int_comp)</span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a>    plot_regions(int_comp, int_edge, diff_int_compedge,</span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a>                track_titles<span class="op">=</span>[<span class="st">'CompInt'</span>, <span class="st">'EdgeInt'</span>, <span class="st">'DiffInt'</span>],</span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a>                title<span class="op">=</span><span class="st">"ECH90 intersect"</span>,</span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a>                figsize<span class="op">=</span>(<span class="fl">3.2</span>, <span class="dv">1</span>),</span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a>                alpha<span class="op">=</span><span class="dv">1</span>, lw<span class="op">=</span><span class="fl">0.5</span>)</span>
<span id="cb25-18"><a href="#cb25-18" aria-hidden="true" tabindex="-1"></a></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div id="fig-edge-enrichment" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-edge-enrichment-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 0.8%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-edge-enrichment" style="flex-basis: 49.2%;justify-content: flex-start;">
<div id="fig-edge-enrichment-1" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-edge-enrichment-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="07_various_plotting_files/figure-html/fig-edge-enrichment-output-1.png" class="lightbox" data-gallery="fig-edge-enrichment" title="Figure&nbsp;8.5&nbsp;(a): Round Spermatid"><img src="07_various_plotting_files/figure-html/fig-edge-enrichment-output-1.png" data-ref-parent="fig-edge-enrichment" width="318" height="98" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-edge-enrichment-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) Round Spermatid
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-edge-enrichment" style="flex-basis: 49.2%;justify-content: flex-start;">
<div id="fig-edge-enrichment-2" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-edge-enrichment-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="07_various_plotting_files/figure-html/fig-edge-enrichment-output-2.png" class="lightbox" data-gallery="fig-edge-enrichment" title="Figure&nbsp;8.5&nbsp;(b): Fibroblast"><img src="07_various_plotting_files/figure-html/fig-edge-enrichment-output-2.png" data-ref-parent="fig-edge-enrichment" width="318" height="98" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-edge-enrichment-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) Fibroblast
</figcaption>
</figure>
</div>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 0.8%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-edge-enrichment-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;8.5: Visual representation of the enrichment of edges in the intersection of ECH90 and A-compartments. Shown round spermatid (a) and fibroblast (b) at 100kb resolution and arms viewframe. Note that the edge-regions are too small to be distinguished visually from the compartment on the graph, making it look like they overlap, even though the difference is reported.
</figcaption>
</figure>
</div>
</div>
</section>
<section id="all-3-sets" class="level3">
<h3 class="anchored" data-anchor-id="all-3-sets">All 3 sets</h3>
<div id="cell-fig-introduce-selected-regions" class="cell" data-execution_count="33">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>plot_regions(ech90, high_olive, high_hama,</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>             track_titles<span class="op">=</span>[<span class="st">'ECH90'</span>, <span class="vs">r"$\it</span><span class="sc">{P.anubis}</span><span class="vs">$"</span>, <span class="vs">r"$\it</span><span class="sc">{P.hama}</span><span class="vs">$"</span>],</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>             figsize<span class="op">=</span>(<span class="dv">5</span>, <span class="fl">0.8</span>),</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>             title<span class="op">=</span><span class="va">None</span>,</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>             lw<span class="op">=</span><span class="fl">0.5</span>, alpha<span class="op">=</span><span class="fl">0.8</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div id="fig-introduce-selected-regions" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-introduce-selected-regions-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="07_various_plotting_files/figure-html/fig-introduce-selected-regions-output-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-19" title="Figure&nbsp;8.6: Visual representation of the selected regions for the baboon comparison. Shown are the ECH90 (top), high-olive (middle) and high-hama (bottom) regions, the basis of the comparison of genomic intervals in this analysis. All coordinates are lifted to the rheMac10 genome."><img src="07_various_plotting_files/figure-html/fig-introduce-selected-regions-output-1.png" width="498" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-introduce-selected-regions-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;8.6: Visual representation of the selected regions for the baboon comparison. Shown are the ECH90 (top), high-olive (middle) and high-hama (bottom) regions, the basis of the comparison of genomic intervals in this analysis. All coordinates are lifted to the rheMac10 genome.
</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="liftover-comparison" class="level3">
<h3 class="anchored" data-anchor-id="liftover-comparison">LiftOver comparison</h3>
<div class="cell" data-execution_count="21">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>track0 <span class="op">=</span> [olive_panu3, hama_panu3]</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>track1 <span class="op">=</span> [high_olive, high_hama]</span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>track2 <span class="op">=</span> [high_olive_ucsc, high_hama_ucsc]</span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> t0,t1,t2 <span class="kw">in</span> <span class="bu">zip</span>(track0, track1, track2):</span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>    plot_regions(t0,t1,t2,</span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a>                track_titles<span class="op">=</span>[<span class="st">'Panu_3.0'</span>, <span class="st">'segment_lift'</span>,<span class="st">'UCSCliftOver'</span>],</span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a>    <span class="co">#            title='High Olive Ancestry Intervals',</span></span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a>                title<span class="op">=</span><span class="va">None</span>,</span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a>                figsize<span class="op">=</span>(<span class="dv">6</span>, <span class="fl">0.8</span>),</span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a>                alpha<span class="op">=</span><span class="fl">0.8</span>, lw<span class="op">=</span><span class="fl">0.1</span>)</span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div id="fig-compare-liftover" class="cell quarto-float quarto-figure quarto-figure-center anchored" data-execution_count="21">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-compare-liftover-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output cell-output-display">
<div id="fig-compare-liftover-1" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-compare-liftover-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="07_various_plotting_files/figure-html/fig-compare-liftover-output-1.png" class="lightbox" data-gallery="fig-compare-liftover" title="Figure&nbsp;8.7&nbsp;(a): high-olive"><img src="07_various_plotting_files/figure-html/fig-compare-liftover-output-1.png" data-ref-parent="fig-compare-liftover" width="599" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-compare-liftover-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) high-olive
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display">
<div id="fig-compare-liftover-2" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-compare-liftover-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="07_various_plotting_files/figure-html/fig-compare-liftover-output-2.png" class="lightbox" data-gallery="fig-compare-liftover" title="Figure&nbsp;8.7&nbsp;(b): high-hama"><img src="07_various_plotting_files/figure-html/fig-compare-liftover-output-2.png" data-ref-parent="fig-compare-liftover" width="599" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-compare-liftover-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) high-hama
</figcaption>
</figure>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-compare-liftover-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;8.7: Comparison of a) high-olive and b) high-hama intervals between Panu_3.0, segment_liftover, or UCSC liftOver coordinates when lifting from PapAnu4–&gt;PapAnu3–&gt;rheMac10.
</figcaption>
</figure>
</div>
</div>
</section>
<section id="baboon-rs100-intersect" class="level3">
<h3 class="anchored" data-anchor-id="baboon-rs100-intersect">Baboon-rs100-intersect</h3>
<div class="cell" data-layout="[[-0.05,3.2,3.2,-0.05],[-0.05,3.2,3.2,-0.05]]" data-execution_count="26">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot </span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>size <span class="op">=</span> (<span class="fl">3.2</span>, <span class="fl">0.8</span>)</span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>edge_rs <span class="op">=</span> edges[<span class="st">'recPE'</span>][<span class="st">'round_spermatid_100kb_arms'</span>]</span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> name,group <span class="kw">in</span> baboon_dict.items():</span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a>    plot_regions(query<span class="op">=</span>edge_rs, annot<span class="op">=</span>group, </span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>                 intersect<span class="op">=</span>interval_intersect(edge_rs, group), </span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a>                 <span class="co">#intersect=ech90,</span></span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a>                 track_titles<span class="op">=</span>[<span class="st">'RS100'</span>, <span class="vs">fr"$\it</span><span class="ch">{{</span><span class="sc">{</span>name<span class="sc">}</span><span class="ch">}}</span><span class="vs">$"</span>, <span class="st">'Intersect'</span>],</span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a>                 figsize<span class="op">=</span>size,</span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a>                 title<span class="op">=</span><span class="va">None</span>,</span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a>                 lw<span class="op">=</span><span class="fl">0.3</span>, alpha<span class="op">=</span><span class="fl">0.9</span>)</span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb28-15"><a href="#cb28-15" aria-hidden="true" tabindex="-1"></a>edge_rs <span class="op">=</span> edges[<span class="st">'recPE'</span>][<span class="st">'round_spermatid_100kb_10Mb'</span>]</span>
<span id="cb28-16"><a href="#cb28-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-17"><a href="#cb28-17" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> name,group <span class="kw">in</span> baboon_dict.items():</span>
<span id="cb28-18"><a href="#cb28-18" aria-hidden="true" tabindex="-1"></a>    plot_regions(query<span class="op">=</span>edge_rs, annot<span class="op">=</span>group, </span>
<span id="cb28-19"><a href="#cb28-19" aria-hidden="true" tabindex="-1"></a>                 intersect<span class="op">=</span>interval_intersect(edge_rs, group), </span>
<span id="cb28-20"><a href="#cb28-20" aria-hidden="true" tabindex="-1"></a>                 <span class="co">#intersect=ech90,</span></span>
<span id="cb28-21"><a href="#cb28-21" aria-hidden="true" tabindex="-1"></a>                 track_titles<span class="op">=</span>[<span class="st">'RS100'</span>, <span class="vs">fr"$\it</span><span class="ch">{{</span><span class="sc">{</span>name<span class="sc">}</span><span class="ch">}}</span><span class="vs">$"</span>, <span class="st">'Intersect'</span>],</span>
<span id="cb28-22"><a href="#cb28-22" aria-hidden="true" tabindex="-1"></a>                 figsize<span class="op">=</span>size,</span>
<span id="cb28-23"><a href="#cb28-23" aria-hidden="true" tabindex="-1"></a>                 title<span class="op">=</span><span class="va">None</span>,</span>
<span id="cb28-24"><a href="#cb28-24" aria-hidden="true" tabindex="-1"></a>                 lw<span class="op">=</span><span class="fl">0.2</span>, alpha<span class="op">=</span><span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div id="fig-baboon-rs100-intersect" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-baboon-rs100-intersect-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 0.8%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-baboon-rs100-intersect" style="flex-basis: 49.2%;justify-content: flex-start;">
<div id="fig-baboon-rs100-intersect-1" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-baboon-rs100-intersect-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="07_various_plotting_files/figure-html/fig-baboon-rs100-intersect-output-1.png" class="lightbox" data-gallery="fig-baboon-rs100-intersect" title="Figure&nbsp;8.8&nbsp;(a): P. hamadryas + arms E1"><img src="07_various_plotting_files/figure-html/fig-baboon-rs100-intersect-output-1.png" data-ref-parent="fig-baboon-rs100-intersect" width="318" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-baboon-rs100-intersect-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) <em>P. hamadryas</em> + arms E1
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-baboon-rs100-intersect" style="flex-basis: 49.2%;justify-content: flex-start;">
<div id="fig-baboon-rs100-intersect-2" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-baboon-rs100-intersect-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="07_various_plotting_files/figure-html/fig-baboon-rs100-intersect-output-2.png" class="lightbox" data-gallery="fig-baboon-rs100-intersect" title="Figure&nbsp;8.8&nbsp;(b): P. anubis + arms E1"><img src="07_various_plotting_files/figure-html/fig-baboon-rs100-intersect-output-2.png" data-ref-parent="fig-baboon-rs100-intersect" width="318" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-baboon-rs100-intersect-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) <em>P. anubis</em> + arms E1
</figcaption>
</figure>
</div>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 0.8%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
</div>
<div class="quarto-layout-row">
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 0.8%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-baboon-rs100-intersect" style="flex-basis: 49.2%;justify-content: flex-start;">
<div id="fig-baboon-rs100-intersect-3" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-baboon-rs100-intersect-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="07_various_plotting_files/figure-html/fig-baboon-rs100-intersect-output-3.png" class="lightbox" data-gallery="fig-baboon-rs100-intersect" title="Figure&nbsp;8.8&nbsp;(c): P. hamadryas + 10Mb E1"><img src="07_various_plotting_files/figure-html/fig-baboon-rs100-intersect-output-3.png" data-ref-parent="fig-baboon-rs100-intersect" width="318" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-baboon-rs100-intersect-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(c) <em>P. hamadryas</em> + 10Mb E1
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-baboon-rs100-intersect" style="flex-basis: 49.2%;justify-content: flex-start;">
<div id="fig-baboon-rs100-intersect-4" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-baboon-rs100-intersect-4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="07_various_plotting_files/figure-html/fig-baboon-rs100-intersect-output-4.png" class="lightbox" data-gallery="fig-baboon-rs100-intersect" title="Figure&nbsp;8.8&nbsp;(d): P. anubis + 10Mb E1"><img src="07_various_plotting_files/figure-html/fig-baboon-rs100-intersect-output-4.png" data-ref-parent="fig-baboon-rs100-intersect" width="318" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-baboon-rs100-intersect-4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(d) <em>P. anubis</em> + 10Mb E1
</figcaption>
</figure>
</div>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 0.8%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-baboon-rs100-intersect-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;8.8: Comparing the A-compartment edges of round spermatid (RS) with regions in baboons from hybrid population where a) 95% of sampled individuals have <em>Papio hamadryas</em> ancestry or b) 100% of the samples have <em>Papio anubis</em> ancestry. The regions are extracted and lifted from PapAnu4 to rhemac10 using segment_liftover.
</figcaption>
</figure>
</div>
</div>
</section>
<section id="table-test-1bp-limits" class="level3">
<h3 class="anchored" data-anchor-id="table-test-1bp-limits">Table: Test 1bp limits</h3>
<div id="cell-50" class="cell" data-execution_count="77">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>all_tests.head() <span class="co">#.query('query == "hama_edge_1bp"')</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="77">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">tissue</th>
<th data-quarto-table-cell-role="th">pc_scale</th>
<th data-quarto-table-cell-role="th">query</th>
<th data-quarto-table-cell-role="th">annot</th>
<th data-quarto-table-cell-role="th">test</th>
<th data-quarto-table-cell-role="th">value</th>
<th data-quarto-table-cell-role="th">p</th>
<th data-quarto-table-cell-role="th">-log10p</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>fibroblast</td>
<td>arms</td>
<td>ECH90</td>
<td>comp_edge_1bp</td>
<td>jaccard</td>
<td>0.000002</td>
<td>0.01656</td>
<td>1.780940</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>fibroblast</td>
<td>arms</td>
<td>hama_edge_1bp</td>
<td>comp_edge_1bp</td>
<td>proximity</td>
<td>0.221717</td>
<td>0.88421</td>
<td>0.053445</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>fibroblast</td>
<td>arms</td>
<td>olive_edge_1bp</td>
<td>comp_edge_1bp</td>
<td>proximity</td>
<td>0.243643</td>
<td>0.67235</td>
<td>0.172405</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3</td>
<td>fibroblast</td>
<td>arms</td>
<td>olivehama_edge_1bp</td>
<td>comp_edge_1bp</td>
<td>proximity</td>
<td>0.243643</td>
<td>0.74099</td>
<td>0.130188</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">4</td>
<td>fibroblast</td>
<td>10Mb</td>
<td>ECH90</td>
<td>comp_edge_1bp</td>
<td>jaccard</td>
<td>0.080105</td>
<td>0.02146</td>
<td>1.668370</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<div id="cell-51" class="cell" data-execution_count="29">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>tissue_order <span class="op">=</span> [<span class="st">'fibroblast'</span>, <span class="st">'spermatogonia'</span>, <span class="st">'pachytene_spermatocyte'</span>, <span class="st">'round_spermatid'</span>, <span class="st">'sperm'</span>]</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>tissue_sname <span class="op">=</span> [<span class="st">'Fb'</span>, <span class="st">'Spa'</span>, <span class="st">'Pac'</span>, <span class="st">'RS'</span>, <span class="st">'Sperm'</span>]</span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a>t0 <span class="op">=</span> (svedig_tabel(all_tests</span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a>                   .<span class="bu">map</span>(<span class="kw">lambda</span> x: x.replace(<span class="st">'_edge_1bp'</span>, <span class="st">''</span>) <span class="cf">if</span> <span class="bu">isinstance</span>(x, <span class="bu">str</span>) <span class="cf">else</span> x),</span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a>             index<span class="op">=</span>[<span class="st">'pc_scale'</span>, <span class="st">'test'</span>,<span class="st">'query'</span>],</span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a>             columns<span class="op">=</span>[<span class="st">"tissue"</span>],</span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a>             col_order <span class="op">=</span> tissue_order,</span>
<span id="cb30-11"><a href="#cb30-11" aria-hidden="true" tabindex="-1"></a>             col_snames <span class="op">=</span> tissue_sname,</span>
<span id="cb30-12"><a href="#cb30-12" aria-hidden="true" tabindex="-1"></a>             values<span class="op">=</span><span class="st">'p'</span>,</span>
<span id="cb30-13"><a href="#cb30-13" aria-hidden="true" tabindex="-1"></a>             cmap<span class="op">=</span><span class="st">'Reds_r'</span>)</span>
<span id="cb30-14"><a href="#cb30-14" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb30-15"><a href="#cb30-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-16"><a href="#cb30-16" aria-hidden="true" tabindex="-1"></a>t1 <span class="op">=</span> (svedig_tabel(</span>
<span id="cb30-17"><a href="#cb30-17" aria-hidden="true" tabindex="-1"></a>    all_tests</span>
<span id="cb30-18"><a href="#cb30-18" aria-hidden="true" tabindex="-1"></a>    .<span class="bu">map</span>(<span class="kw">lambda</span> x: x.replace(<span class="st">'_edge_1bp'</span>, <span class="st">''</span>) <span class="cf">if</span> <span class="bu">isinstance</span>(x, <span class="bu">str</span>) <span class="cf">else</span> x)</span>
<span id="cb30-19"><a href="#cb30-19" aria-hidden="true" tabindex="-1"></a>    .query(<span class="st">'pc_scale == "arms"'</span>),</span>
<span id="cb30-20"><a href="#cb30-20" aria-hidden="true" tabindex="-1"></a>    index<span class="op">=</span>[<span class="st">'test'</span>,<span class="st">'query'</span>],</span>
<span id="cb30-21"><a href="#cb30-21" aria-hidden="true" tabindex="-1"></a>    columns<span class="op">=</span>[<span class="st">"tissue"</span>],</span>
<span id="cb30-22"><a href="#cb30-22" aria-hidden="true" tabindex="-1"></a>    col_order <span class="op">=</span> tissue_order,</span>
<span id="cb30-23"><a href="#cb30-23" aria-hidden="true" tabindex="-1"></a>    col_snames <span class="op">=</span> tissue_sname,</span>
<span id="cb30-24"><a href="#cb30-24" aria-hidden="true" tabindex="-1"></a>    values<span class="op">=</span><span class="st">'p'</span>,</span>
<span id="cb30-25"><a href="#cb30-25" aria-hidden="true" tabindex="-1"></a>    cmap<span class="op">=</span><span class="st">'Reds_r'</span>)</span>
<span id="cb30-26"><a href="#cb30-26" aria-hidden="true" tabindex="-1"></a><span class="co">#.relabel_index(labels=[lbl for lbl in tissue_sname if lbl in t1.columns], axis=1)</span></span>
<span id="cb30-27"><a href="#cb30-27" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb30-28"><a href="#cb30-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-29"><a href="#cb30-29" aria-hidden="true" tabindex="-1"></a>t2 <span class="op">=</span> (svedig_tabel(all_tests</span>
<span id="cb30-30"><a href="#cb30-30" aria-hidden="true" tabindex="-1"></a>                   .<span class="bu">map</span>(<span class="kw">lambda</span> x: x.replace(<span class="st">'_edge_1bp'</span>, <span class="st">''</span>) <span class="cf">if</span> <span class="bu">isinstance</span>(x, <span class="bu">str</span>) <span class="cf">else</span> x)</span>
<span id="cb30-31"><a href="#cb30-31" aria-hidden="true" tabindex="-1"></a>                   .query(<span class="st">'pc_scale == "10Mb"'</span>),</span>
<span id="cb30-32"><a href="#cb30-32" aria-hidden="true" tabindex="-1"></a>             index<span class="op">=</span>[<span class="st">'test'</span>,<span class="st">'query'</span>],</span>
<span id="cb30-33"><a href="#cb30-33" aria-hidden="true" tabindex="-1"></a>             columns<span class="op">=</span>[<span class="st">"tissue"</span>],</span>
<span id="cb30-34"><a href="#cb30-34" aria-hidden="true" tabindex="-1"></a>             col_order <span class="op">=</span> tissue_order,</span>
<span id="cb30-35"><a href="#cb30-35" aria-hidden="true" tabindex="-1"></a>             col_snames <span class="op">=</span> tissue_sname,</span>
<span id="cb30-36"><a href="#cb30-36" aria-hidden="true" tabindex="-1"></a>             values<span class="op">=</span><span class="st">'p'</span>,</span>
<span id="cb30-37"><a href="#cb30-37" aria-hidden="true" tabindex="-1"></a>             cmap<span class="op">=</span><span class="st">'Reds_r'</span>)</span>
<span id="cb30-38"><a href="#cb30-38" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb30-39"><a href="#cb30-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-40"><a href="#cb30-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-41"><a href="#cb30-41" aria-hidden="true" tabindex="-1"></a>display(t1,t2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<style type="text/css">
#T_bd4ec .col0 {
  min-width: 20px;
}
#T_bd4ec .col1 {
  min-width: 20px;
}
#T_bd4ec .col2 {
  min-width: 20px;
}
#T_bd4ec .col3 {
  min-width: 20px;
}
#T_bd4ec_row0_col0 {
  background-color: #f75c41;
  color: #f1f1f1;
}
#T_bd4ec_row0_col1 {
  background-color: #b21218;
  color: #f1f1f1;
}
#T_bd4ec_row0_col2 {
  background-color: #ac1117;
  color: #f1f1f1;
}
#T_bd4ec_row0_col3 {
  background-color: #fcc3ab;
  color: #000000;
}
#T_bd4ec_row1_col0, #T_bd4ec_row1_col1, #T_bd4ec_row1_col3 {
  background-color: #000000;
  color: #f1f1f1;
  color: transparent;
  background-color: transparent;
}
#T_bd4ec_row1_col2 {
  background-color: #fff5f0;
  color: #000000;
}
</style>

<table id="T_bd4ec" class="caption-top table table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th class="blank" data-quarto-table-cell-role="th">&nbsp;</th>
<th class="index_name level0" data-quarto-table-cell-role="th">tissue</th>
<th id="T_bd4ec_level0_col0" class="col_heading level0 col0" data-quarto-table-cell-role="th">Fb</th>
<th id="T_bd4ec_level0_col1" class="col_heading level0 col1" data-quarto-table-cell-role="th">Spa</th>
<th id="T_bd4ec_level0_col2" class="col_heading level0 col2" data-quarto-table-cell-role="th">Pac</th>
<th id="T_bd4ec_level0_col3" class="col_heading level0 col3" data-quarto-table-cell-role="th">RS</th>
</tr>
<tr class="even">
<th class="index_name level0" data-quarto-table-cell-role="th">test</th>
<th class="index_name level1" data-quarto-table-cell-role="th">query</th>
<th class="blank col0" data-quarto-table-cell-role="th">&nbsp;</th>
<th class="blank col1" data-quarto-table-cell-role="th">&nbsp;</th>
<th class="blank col2" data-quarto-table-cell-role="th">&nbsp;</th>
<th class="blank col3" data-quarto-table-cell-role="th">&nbsp;</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td id="T_bd4ec_level0_row0" class="row_heading level0 row0" data-quarto-table-cell-role="th">jaccard</td>
<td id="T_bd4ec_level1_row0" class="row_heading level1 row0" data-quarto-table-cell-role="th">ECH90</td>
<td id="T_bd4ec_row0_col0" class="data row0 col0">0.017</td>
<td id="T_bd4ec_row0_col1" class="data row0 col1">0.006</td>
<td id="T_bd4ec_row0_col2" class="data row0 col2">0.005</td>
<td id="T_bd4ec_row0_col3" class="data row0 col3">0.028</td>
</tr>
<tr class="even">
<td id="T_bd4ec_level0_row1" class="row_heading level0 row1" data-quarto-table-cell-role="th">proximity</td>
<td id="T_bd4ec_level1_row1" class="row_heading level1 row1" data-quarto-table-cell-role="th">olivehama</td>
<td id="T_bd4ec_row1_col0" class="data row1 col0">nan</td>
<td id="T_bd4ec_row1_col1" class="data row1 col1">nan</td>
<td id="T_bd4ec_row1_col2" class="data row1 col2">0.036</td>
<td id="T_bd4ec_row1_col3" class="data row1 col3">nan</td>
</tr>
</tbody>
</table>
</div>
<div class="cell-output cell-output-display">
<style type="text/css">
#T_29a48 .col0 {
  min-width: 20px;
}
#T_29a48 .col1 {
  min-width: 20px;
}
#T_29a48 .col2 {
  min-width: 20px;
}
#T_29a48 .col3 {
  min-width: 20px;
}
#T_29a48_row0_col0 {
  background-color: #f6583e;
  color: #f1f1f1;
}
#T_29a48_row0_col1 {
  background-color: #a50f15;
  color: #f1f1f1;
}
#T_29a48_row0_col2, #T_29a48_row1_col0, #T_29a48_row1_col1, #T_29a48_row1_col2, #T_29a48_row2_col0 {
  background-color: #000000;
  color: #f1f1f1;
  color: transparent;
  background-color: transparent;
}
#T_29a48_row0_col3 {
  background-color: #fff5f0;
  color: #000000;
}
#T_29a48_row1_col3 {
  background-color: #e43027;
  color: #f1f1f1;
}
#T_29a48_row2_col1 {
  background-color: #f14331;
  color: #f1f1f1;
}
#T_29a48_row2_col2 {
  background-color: #ce1a1e;
  color: #f1f1f1;
}
#T_29a48_row2_col3 {
  background-color: #73030f;
  color: #f1f1f1;
}
</style>

<table id="T_29a48" class="caption-top table table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th class="blank" data-quarto-table-cell-role="th">&nbsp;</th>
<th class="index_name level0" data-quarto-table-cell-role="th">tissue</th>
<th id="T_29a48_level0_col0" class="col_heading level0 col0" data-quarto-table-cell-role="th">Fb</th>
<th id="T_29a48_level0_col1" class="col_heading level0 col1" data-quarto-table-cell-role="th">Spa</th>
<th id="T_29a48_level0_col2" class="col_heading level0 col2" data-quarto-table-cell-role="th">Pac</th>
<th id="T_29a48_level0_col3" class="col_heading level0 col3" data-quarto-table-cell-role="th">Sperm</th>
</tr>
<tr class="even">
<th class="index_name level0" data-quarto-table-cell-role="th">test</th>
<th class="index_name level1" data-quarto-table-cell-role="th">query</th>
<th class="blank col0" data-quarto-table-cell-role="th">&nbsp;</th>
<th class="blank col1" data-quarto-table-cell-role="th">&nbsp;</th>
<th class="blank col2" data-quarto-table-cell-role="th">&nbsp;</th>
<th class="blank col3" data-quarto-table-cell-role="th">&nbsp;</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td id="T_29a48_level0_row0" class="row_heading level0 row0" data-quarto-table-cell-role="th">jaccard</td>
<td id="T_29a48_level1_row0" class="row_heading level1 row0" data-quarto-table-cell-role="th">ECH90</td>
<td id="T_29a48_row0_col0" class="data row0 col0">0.021</td>
<td id="T_29a48_row0_col1" class="data row0 col1">0.006</td>
<td id="T_29a48_row0_col2" class="data row0 col2">nan</td>
<td id="T_29a48_row0_col3" class="data row0 col3">0.048</td>
</tr>
<tr class="even">
<td rowspan="2" id="T_29a48_level0_row1" class="row_heading level0 row1" data-quarto-table-cell-role="th">proximity</td>
<td id="T_29a48_level1_row1" class="row_heading level1 row1" data-quarto-table-cell-role="th">olive</td>
<td id="T_29a48_row1_col0" class="data row1 col0">nan</td>
<td id="T_29a48_row1_col1" class="data row1 col1">nan</td>
<td id="T_29a48_row1_col2" class="data row1 col2">nan</td>
<td id="T_29a48_row1_col3" class="data row1 col3">0.016</td>
</tr>
<tr class="odd">
<td id="T_29a48_level1_row2" class="row_heading level1 row2" data-quarto-table-cell-role="th">olivehama</td>
<td id="T_29a48_row2_col0" class="data row2 col0">nan</td>
<td id="T_29a48_row2_col1" class="data row2 col1">0.019</td>
<td id="T_29a48_row2_col2" class="data row2 col2">0.012</td>
<td id="T_29a48_row2_col3" class="data row2 col3">0.001</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="cell" data-execution_count="38">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>tissue_order <span class="op">=</span> [<span class="st">'fibroblast'</span>, <span class="st">'spermatogonia'</span>, <span class="st">'pachytene_spermatocyte'</span>, <span class="st">'round_spermatid'</span>, <span class="st">'sperm'</span>]</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>tissue_sname <span class="op">=</span> [<span class="st">'Fb'</span>, <span class="st">'Spa'</span>, <span class="st">'Pac'</span>, <span class="st">'RS'</span>, <span class="st">'Sperm'</span>]</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>df <span class="op">=</span> (all_tests</span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a>      .<span class="bu">map</span>(<span class="kw">lambda</span> x: x.replace(<span class="st">'_edge_1bp'</span>, <span class="st">'_1bp'</span>) <span class="cf">if</span> <span class="bu">isinstance</span>(x, <span class="bu">str</span>) <span class="cf">else</span> x)</span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a>      .assign(log10p<span class="op">=</span>np.log10(all_tests.p))</span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a>      .loc[(all_tests.p <span class="op">&lt;</span> <span class="fl">0.05</span>)]</span>
<span id="cb31-10"><a href="#cb31-10" aria-hidden="true" tabindex="-1"></a>      .pivot(index<span class="op">=</span>[<span class="st">'pc_scale'</span>, <span class="st">'query'</span>, <span class="st">'test'</span> ], columns<span class="op">=</span>[<span class="st">"tissue"</span>], values<span class="op">=</span><span class="st">"p"</span>).</span>
<span id="cb31-11"><a href="#cb31-11" aria-hidden="true" tabindex="-1"></a>      <span class="bu">filter</span>(tissue_order)</span>
<span id="cb31-12"><a href="#cb31-12" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb31-13"><a href="#cb31-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-14"><a href="#cb31-14" aria-hidden="true" tabindex="-1"></a>df <span class="op">=</span> df.rename(columns <span class="op">=</span> {x:tissue_sname[i] <span class="cf">for</span> i,x <span class="kw">in</span> <span class="bu">enumerate</span>(df.columns.tolist())})</span>
<span id="cb31-15"><a href="#cb31-15" aria-hidden="true" tabindex="-1"></a>df.columns.name <span class="op">=</span> <span class="va">None</span></span>
<span id="cb31-16"><a href="#cb31-16" aria-hidden="true" tabindex="-1"></a>df <span class="op">=</span> df.reset_index().fillna(<span class="st">'-'</span>)</span>
<span id="cb31-17"><a href="#cb31-17" aria-hidden="true" tabindex="-1"></a>df <span class="op">=</span> df.rename(columns<span class="op">=</span>{<span class="st">'pc_scale'</span>: <span class="st">'View'</span>, <span class="st">'query'</span>: <span class="st">'Query'</span>, <span class="st">'test'</span>: <span class="st">'Test'</span>})</span>
<span id="cb31-18"><a href="#cb31-18" aria-hidden="true" tabindex="-1"></a>df.style.hide(axis<span class="op">=</span><span class="st">'index'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div id="tbl-svedig-tabel-significant" class="cell quarto-float quarto-figure quarto-figure-center anchored" data-execution_count="38">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-svedig-tabel-significant-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;8.1: Test results from proximity and Jaccard, comparing ECH (full regions) or baboon 1bp limits with compartment 1bp limits from all cell types. Significant p-values across tissue types. All insignificant p-values are filtered out. <em>Fb: Fibroblast, Spa: Spermatogonia, Pac: Pachytene Spermatocyte, RS: Round Spermatid, Sperm: Sperm</em>
</figcaption>
<div aria-describedby="tbl-svedig-tabel-significant-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output cell-output-display" data-execution_count="38">
<style type="text/css">
</style>

<table id="T_a1cfd" class="do-not-create-environment cell caption-top table table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th id="T_a1cfd_level0_col0" class="col_heading level0 col0" data-quarto-table-cell-role="th">View</th>
<th id="T_a1cfd_level0_col1" class="col_heading level0 col1" data-quarto-table-cell-role="th">Query</th>
<th id="T_a1cfd_level0_col2" class="col_heading level0 col2" data-quarto-table-cell-role="th">Test</th>
<th id="T_a1cfd_level0_col3" class="col_heading level0 col3" data-quarto-table-cell-role="th">Fb</th>
<th id="T_a1cfd_level0_col4" class="col_heading level0 col4" data-quarto-table-cell-role="th">Spa</th>
<th id="T_a1cfd_level0_col5" class="col_heading level0 col5" data-quarto-table-cell-role="th">Pac</th>
<th id="T_a1cfd_level0_col6" class="col_heading level0 col6" data-quarto-table-cell-role="th">RS</th>
<th id="T_a1cfd_level0_col7" class="col_heading level0 col7" data-quarto-table-cell-role="th">Sperm</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td id="T_a1cfd_row0_col0" class="data row0 col0">10Mb</td>
<td id="T_a1cfd_row0_col1" class="data row0 col1">ECH90</td>
<td id="T_a1cfd_row0_col2" class="data row0 col2">jaccard</td>
<td id="T_a1cfd_row0_col3" class="data row0 col3">0.021460</td>
<td id="T_a1cfd_row0_col4" class="data row0 col4">0.006110</td>
<td id="T_a1cfd_row0_col5" class="data row0 col5">-</td>
<td id="T_a1cfd_row0_col6" class="data row0 col6">-</td>
<td id="T_a1cfd_row0_col7" class="data row0 col7">0.047500</td>
</tr>
<tr class="even">
<td id="T_a1cfd_row1_col0" class="data row1 col0">10Mb</td>
<td id="T_a1cfd_row1_col1" class="data row1 col1">olive_1bp</td>
<td id="T_a1cfd_row1_col2" class="data row1 col2">proximity</td>
<td id="T_a1cfd_row1_col3" class="data row1 col3">-</td>
<td id="T_a1cfd_row1_col4" class="data row1 col4">-</td>
<td id="T_a1cfd_row1_col5" class="data row1 col5">-</td>
<td id="T_a1cfd_row1_col6" class="data row1 col6">-</td>
<td id="T_a1cfd_row1_col7" class="data row1 col7">0.016140</td>
</tr>
<tr class="odd">
<td id="T_a1cfd_row2_col0" class="data row2 col0">10Mb</td>
<td id="T_a1cfd_row2_col1" class="data row2 col1">olivehama_1bp</td>
<td id="T_a1cfd_row2_col2" class="data row2 col2">proximity</td>
<td id="T_a1cfd_row2_col3" class="data row2 col3">-</td>
<td id="T_a1cfd_row2_col4" class="data row2 col4">0.018790</td>
<td id="T_a1cfd_row2_col5" class="data row2 col5">0.012300</td>
<td id="T_a1cfd_row2_col6" class="data row2 col6">-</td>
<td id="T_a1cfd_row2_col7" class="data row2 col7">0.001280</td>
</tr>
<tr class="even">
<td id="T_a1cfd_row3_col0" class="data row3 col0">arms</td>
<td id="T_a1cfd_row3_col1" class="data row3 col1">ECH90</td>
<td id="T_a1cfd_row3_col2" class="data row3 col2">jaccard</td>
<td id="T_a1cfd_row3_col3" class="data row3 col3">0.016560</td>
<td id="T_a1cfd_row3_col4" class="data row3 col4">0.006120</td>
<td id="T_a1cfd_row3_col5" class="data row3 col5">0.005340</td>
<td id="T_a1cfd_row3_col6" class="data row3 col6">0.027690</td>
<td id="T_a1cfd_row3_col7" class="data row3 col7">-</td>
</tr>
<tr class="odd">
<td id="T_a1cfd_row4_col0" class="data row4 col0">arms</td>
<td id="T_a1cfd_row4_col1" class="data row4 col1">olivehama_1bp</td>
<td id="T_a1cfd_row4_col2" class="data row4 col2">proximity</td>
<td id="T_a1cfd_row4_col3" class="data row4 col3">-</td>
<td id="T_a1cfd_row4_col4" class="data row4 col4">-</td>
<td id="T_a1cfd_row4_col5" class="data row4 col5">0.035640</td>
<td id="T_a1cfd_row4_col6" class="data row4 col6">-</td>
<td id="T_a1cfd_row4_col7" class="data row4 col7">-</td>
</tr>
</tbody>
</table>
</div>
</div>
</figure>
</div>
</div>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../notebooks/06_rec_genomicintervals.html" class="pagination-link" aria-label="E1 vs. ECH (recommended params)">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-title">E1 vs.&nbsp;ECH (recommended params)</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../workflow.html" class="pagination-link" aria-label="master_workflow">
        <span class="nav-page-text"><span class="chapter-title">master_workflow</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer"><div class="nav-footer-center"><div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.dev/munch-group/hic-spermatogenesis/blob/main/notebooks/07_various_plotting.ipynb" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/munch-group/hic-spermatogenesis/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li><li><a href="https://github.com/munch-group/hic-spermatogenesis/blob/main/notebooks/07_various_plotting.ipynb" class="toc-action"><i class="bi empty"></i>View source</a></li></ul></div></div></div></footer><script>var lightboxQuarto = GLightbox({"closeEffect":"zoom","descPosition":"bottom","loop":false,"openEffect":"zoom","selector":".lightbox"});
(function() {
  let previousOnload = window.onload;
  window.onload = () => {
    if (previousOnload) {
      previousOnload();
    }
    lightboxQuarto.on('slide_before_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      const href = trigger.getAttribute('href');
      if (href !== null) {
        const imgEl = window.document.querySelector(`a[href="${href}"] img`);
        if (imgEl !== null) {
          const srcAttr = imgEl.getAttribute("src");
          if (srcAttr && srcAttr.startsWith("data:")) {
            slideConfig.href = srcAttr;
          }
        }
      } 
    });
  
    lightboxQuarto.on('slide_after_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(slideNode);
      }
    });
  
  };
  
})();
          </script>




</body></html>