<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.39">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2024-12-28">
<meta name="keywords" content="Hi-C, Chromatin Compartments, Evolutionary pressures, Selfish gene drive">

<title>Chromatin Compartments and Selection on X</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js" integrity="sha512-bLT0Qm9VnAYZDflyKcBaQ2gg0hSYNQrJ8RilYldYQ1FxQYoCLtUjuuRuZo+fjqhx/qtq/1itJ0C2ejDxltZVFg==" crossorigin="anonymous"></script><script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-e26003cea8cd680ca0c55a263523d882.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-7bfb9ce031f0150bcdd0059ffcd3ab40.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script src="site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet">
<script async="" src="https://hypothes.is/embed.js"></script>
<script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" integrity="sha512-c3Nl8+7g4LMSTdrm621y7kf9v3SDPnhxLNhcjFJbKECVnmZHTdo+IRO05sNLTH/D3vA6u1X32ehoLC7WFVdheg==" crossorigin="anonymous"></script>

<script type="application/javascript">define('jquery', [],function() {return window.jQuery;})</script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<meta name="citation_title" content="Chromatin Compartments and Selection on X">
<meta name="citation_abstract" content="Here, the 3D chromatin architecture of the X chromosome in rhesus macaque (*Macaca mulata*) is investigated in the context of evolutionary pressures and genetic drivers. We redo Hi-C analyses from a 2019 paper on the latest reference genome (*rheMac10* or *Mmul_10*). We compare two Hi-C analysis frameworks, *HiCExplorer* and *cooler/cooltools* (Open2C), on a subset, finding Open2C to be most flexible and intuitive. The ICE method (Iterative Correction and Eigendecomposition) was used to infer conventional and refined A/B compartments for fibroblast and four stages of spermatogenesis. We find that 200 kbp transition-zones between A/B-compartments in both fibroblasts and round spermatids align well with human selective sweeps where meiotic drive can explain the selection (ECH-regions), and with hybrid incompatility in baboons (genus *Papio*). We discuss the biological meaning of these findings, where conserved chromatin features may help to retain non-advantageous alleles, hinting to the role of selfish genetic elements in genome evolution.
">
<meta name="citation_keywords" content="Hi-C,Chromatin Compartments,Evolutionary pressures,Selfish gene drive">
<meta name="citation_author" content="Søren Jørgensen">
<meta name="citation_publication_date" content="2024-12-28">
<meta name="citation_cover_date" content="2024-12-28">
<meta name="citation_year" content="2024">
<meta name="citation_online_date" content="2024-12-28">
<meta name="citation_language" content="en">
<meta name="citation_reference" content="citation_title=Principal components analysis corrects for stratification in genome-wide association studies;,citation_author=Alkes L Price;,citation_author=Nick J Patterson;,citation_author=Robert M Plenge;,citation_author=Michael E Weinblatt;,citation_author=Nancy A Shadick;,citation_author=David Reich;,citation_publication_date=2006-08;,citation_cover_date=2006-08;,citation_year=2006;,citation_fulltext_html_url=https://www.nature.com/articles/ng1847;,citation_issue=8;,citation_doi=10.1038/ng1847;,citation_issn=1061-4036, 1546-1718;,citation_volume=38;,citation_journal_title=Nature Genetics;">
<meta name="citation_reference" content="citation_title=Accurate assembly of the olive baboon ( &amp;amp;amp;lt;i&amp;gt;Papio anubis&amp;lt;/i&amp;gt; ) genome using long-read and Hi-C data;,citation_abstract=AbstractBackground Baboons are a widely used nonhuman primate model for biomedical, evolutionary, and basic genetics research. Despite this importance, the genomic resources for baboons are limited. In particular, the current baboon reference genome Panu_3.0 is a highly fragmented, reference-guided (i.e., not fully de novo) assembly, and its poor quality inhibits our ability to conduct downstream genomic analyses.Findings Here we present a de novo genome assembly of the olive baboon (Papio anubis) that uses data from several recently developed single-molecule technologies. Our assembly, Panubis1.0, has an N50 contig size of ∼1.46 Mb (as opposed to 139 kb for Panu_3.0) and has single scaffolds that span each of the 20 autosomes and the X chromosome.Conclusions We highlight multiple lines of evidence (including Bionano Genomics data, pedigree linkage information, and linkage disequilibrium data) suggesting that there are several large assembly errors in Panu_3.0, which have been corrected in Panubis1.0.;,citation_author=Sanjit Singh Batra;,citation_author=Michal Levy-Sakin;,citation_author=Jacqueline Robinson;,citation_author=Joseph Guillory;,citation_author=Steffen Durinck;,citation_author=Tauras P Vilgalys;,citation_author=Pui-Yan Kwok;,citation_author=Laura A Cox;,citation_author=Somasekar Seshagiri;,citation_author=Yun S Song;,citation_author=Jeffrey D Wall;,citation_publication_date=2020-12;,citation_cover_date=2020-12;,citation_year=2020;,citation_fulltext_html_url=https://academic.oup.com/gigascience/article/doi/10.1093/gigascience/giaa134/6025063;,citation_issue=12;,citation_doi=10.1093/gigascience/giaa134;,citation_issn=2047-217X;,citation_volume=9;,citation_journal_title=GigaScience;">
<meta name="citation_reference" content="citation_title=HiC-Pro: An optimized and flexible pipeline for Hi-C data processing;,citation_author=Nicolas Servant;,citation_author=Nelle Varoquaux;,citation_author=Bryan R. Lajoie;,citation_author=Eric Viara;,citation_author=Chong-Jian Chen;,citation_author=Jean-Philippe Vert;,citation_author=Edith Heard;,citation_author=Job Dekker;,citation_author=Emmanuel Barillot;,citation_publication_date=2015-12;,citation_cover_date=2015-12;,citation_year=2015;,citation_fulltext_html_url=https://genomebiology.biomedcentral.com/articles/10.1186/s13059-015-0831-x;,citation_issue=1;,citation_doi=10.1186/s13059-015-0831-x;,citation_issn=1474-760X;,citation_volume=16;,citation_journal_title=Genome Biology;">
<meta name="citation_reference" content="citation_title=Comprehensive Mapping of Long-Range Interactions Reveals Folding Principles of the Human Genome;,citation_abstract=We describe Hi-C, a method that probes the three-dimensional architecture of whole genomes by coupling proximity-based ligation with massively parallel sequencing. We constructed spatial proximity maps of the human genome with Hi-C at a resolution of 1 megabase. These maps confirm the presence of chromosome territories and the spatial proximity of small, gene-rich chromosomes. We identified an additional level of genome organization that is characterized by the spatial segregation of open and closed chromatin to form two genome-wide compartments. At the megabase scale, the chromatin conformation is consistent with a fractal globule, a knot-free, polymer conformation that enables maximally dense packing while preserving the ability to easily fold and unfold any genomic locus. The fractal globule is distinct from the more commonly used globular equilibrium model. Our results demonstrate the power of Hi-C to map the dynamic conformations of whole genomes.;,citation_author=Erez Lieberman-Aiden;,citation_author=Nynke L. Van Berkum;,citation_author=Louise Williams;,citation_author=Maxim Imakaev;,citation_author=Tobias Ragoczy;,citation_author=Agnes Telling;,citation_author=Ido Amit;,citation_author=Bryan R. Lajoie;,citation_author=Peter J. Sabo;,citation_author=Michael O. Dorschner;,citation_author=Richard Sandstrom;,citation_author=Bradley Bernstein;,citation_author=M. A. Bender;,citation_author=Mark Groudine;,citation_author=Andreas Gnirke;,citation_author=John Stamatoyannopoulos;,citation_author=Leonid A. Mirny;,citation_author=Eric S. Lander;,citation_author=Job Dekker;,citation_publication_date=2009-10;,citation_cover_date=2009-10;,citation_year=2009;,citation_fulltext_html_url=https://www.science.org/doi/10.1126/science.1181369;,citation_issue=5950;,citation_doi=10.1126/science.1181369;,citation_issn=0036-8075, 1095-9203;,citation_volume=326;,citation_journal_title=Science;">
<meta name="citation_reference" content="citation_title=Stage-resolved Hi-C analyses reveal meiotic chromosome organizational features influencing homolog alignment;,citation_abstract=Abstract During meiosis, chromosomes exhibit dramatic changes in morphology and intranuclear positioning. How these changes influence homolog pairing, alignment, and recombination remain elusive. Using Hi-C, we systematically mapped 3D genome architecture throughout all meiotic prophase substages during mouse spermatogenesis. Our data uncover two major chromosome organizational features varying along the chromosome axis during early meiotic prophase, when homolog alignment occurs. First, transcriptionally active and inactive genomic regions form alternating domains consisting of shorter and longer chromatin loops, respectively. Second, the force-transmitting LINC complex promotes the alignment of ends of different chromosomes over a range of up to 20% of chromosome length. Both features correlate with the pattern of homolog interactions and the distribution of recombination events. Collectively, our data reveal the influences of transcription and force on meiotic chromosome structure and suggest chromosome organization may provide an infrastructure for the modulation of meiotic recombination in higher eukaryotes.;,citation_author=Wu Zuo;,citation_author=Guangming Chen;,citation_author=Zhimei Gao;,citation_author=Shuai Li;,citation_author=Yanyan Chen;,citation_author=Chenhui Huang;,citation_author=Juan Chen;,citation_author=Zhengjun Chen;,citation_author=Ming Lei;,citation_author=Qian Bian;,citation_publication_date=2021-10;,citation_cover_date=2021-10;,citation_year=2021;,citation_fulltext_html_url=https://www.nature.com/articles/s41467-021-26033-0;,citation_issue=1;,citation_doi=10.1038/s41467-021-26033-0;,citation_issn=2041-1723;,citation_volume=12;,citation_journal_title=Nature Communications;">
<meta name="citation_reference" content="citation_title=Single-Cell RNA Sequencing of Human, Macaque, and Mouse Testes Uncovers Conserved and Divergent Features of Mammalian Spermatogenesis;,citation_author=Adrienne Niederriter Shami;,citation_author=Xianing Zheng;,citation_author=Sarah K. Munyoki;,citation_author=Qianyi Ma;,citation_author=Gabriel L. Manske;,citation_author=Christopher D. Green;,citation_author=Meena Sukhwani;,citation_author=Kyle E. Orwig;,citation_author=Jun Z. Li;,citation_author=Saher Sue Hammoud;,citation_publication_date=2020-08;,citation_cover_date=2020-08;,citation_year=2020;,citation_fulltext_html_url=https://linkinghub.elsevier.com/retrieve/pii/S1534580720303993;,citation_issue=4;,citation_doi=10.1016/j.devcel.2020.05.010;,citation_issn=15345807;,citation_volume=54;,citation_journal_title=Developmental Cell;">
<meta name="citation_reference" content="citation_title=Reprogramming of Meiotic Chromatin Architecture during Spermatogenesis;,citation_author=Yao Wang;,citation_author=Hanben Wang;,citation_author=Yu Zhang;,citation_author=Zhenhai Du;,citation_author=Wei Si;,citation_author=Suixing Fan;,citation_author=Dongdong Qin;,citation_author=Mei Wang;,citation_author=Yanchao Duan;,citation_author=Lufan Li;,citation_author=Yuying Jiao;,citation_author=Yuanyuan Li;,citation_author=Qiujun Wang;,citation_author=Qinghua Shi;,citation_author=Xin Wu;,citation_author=Wei Xie;,citation_publication_date=2019-02;,citation_cover_date=2019-02;,citation_year=2019;,citation_fulltext_html_url=https://linkinghub.elsevier.com/retrieve/pii/S1097276518309894;,citation_issue=3;,citation_doi=10.1016/j.molcel.2018.11.019;,citation_issn=10972765;,citation_volume=73;,citation_journal_title=Molecular Cell;">
<meta name="citation_reference" content="citation_title=dcHiC detects differential compartments across multiple Hi-C datasets;,citation_abstract=AbstractThe compartmental organization of mammalian genomes and its changes play important roles in distinct biological processes. Here, we introduce dcHiC, which utilizes a multivariate distance measure to identify significant changes in compartmentalization among multiple contact maps. Evaluating dcHiC on four collections of bulk and single-cell contact maps from in vitro mouse neural differentiation (n=3), mouse hematopoiesis (n=10), human LCLs (n=20) and post-natal mouse brain development (n=3 stages), we show its effectiveness and sensitivity in detecting biologically relevant changes, including those orthogonally validated. dcHiC reported regions with dynamically regulated genes associated with cell identity, along with correlated changes in chromatin states, subcompartments, replication timing and lamin association. With its efficient implementation, dcHiC enables high-resolution compartment analysis as well as standalone browser visualization, differential interaction identification and time-series clustering. dcHiC is an essential addition to the Hi-C analysis toolbox for the ever-growing number of bulk and single-cell contact maps. Available at: https://github.com/ay-lab/dcHiC .;,citation_author=Abhijit Chakraborty;,citation_author=Jeffrey G. Wang;,citation_author=Ferhat Ay;,citation_publication_date=2022-11;,citation_cover_date=2022-11;,citation_year=2022;,citation_fulltext_html_url=https://www.nature.com/articles/s41467-022-34626-6;,citation_issue=1;,citation_doi=10.1038/s41467-022-34626-6;,citation_issn=2041-1723;,citation_volume=13;,citation_journal_title=Nature Communications;">
<meta name="citation_reference" content="citation_title=Hi-C Data Analysis: Methods and Protocols;,citation_editor=Silvio Bicciato;,citation_editor=Francesco Ferrari;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_fulltext_html_url=https://link.springer.com/10.1007/978-1-0716-1390-0;,citation_doi=10.1007/978-1-0716-1390-0;,citation_isbn=978-1-07-161389-4 978-1-07-161390-0;,citation_volume=2301;,citation_series_title=Methods in Molecular Biology;">
<meta name="citation_reference" content="citation_title=Extraordinary selection on the human X chromosome associated with archaic admixture;,citation_author=Laurits Skov;,citation_author=Moisès Coll Macià;,citation_author=Elise Anne Lucotte;,citation_author=Maria Izabel Alves Cavassim;,citation_author=David Castellano;,citation_author=Mikkel Heide Schierup;,citation_author=Kasper Munch;,citation_publication_date=2023-03;,citation_cover_date=2023-03;,citation_year=2023;,citation_fulltext_html_url=https://linkinghub.elsevier.com/retrieve/pii/S2666979X23000344;,citation_issue=3;,citation_doi=10.1016/j.xgen.2023.100274;,citation_issn=2666979X;,citation_volume=3;,citation_journal_title=Cell Genomics;">
<meta name="citation_reference" content="citation_title=Genetic Villains: Killer Meiotic Drivers;,citation_abstract=Unbiased allele transmission into progeny is a fundamental genetic concept canonized as Mendel’s Law of Segregation. Not all alleles, however, abide by the law. Killer meiotic drivers are ultra-selfish DNA sequences that are transmitted into more than half (sometimes all) of the meiotic products generated by a heterozygote. As their name implies, these loci gain a transmission advantage in heterozygotes by destroying otherwise viable meiotic products that do not inherit the driver. We review and classify killer meiotic drive genes across a wide spectrum of eukaryotes. We discuss how analyses of these ultra-selfish genes can lead to greater insight into the mechanisms of gametogenesis and the causes of infertility.;,citation_author=María Angélica Bravo Núñez;,citation_author=Nicole L. Nuckolls;,citation_author=Sarah E. Zanders;,citation_publication_date=2018;,citation_cover_date=2018;,citation_year=2018;,citation_fulltext_html_url=https://www.sciencedirect.com/science/article/pii/S016895251830026X;,citation_issue=6;,citation_doi=https://doi.org/10.1016/j.tig.2018.02.003;,citation_issn=0168-9525;,citation_volume=34;,citation_journal_title=Trends in Genetics;">
<meta name="citation_reference" content="citation_title=Chromatin architecture reorganization during stem cell differentiation;,citation_abstract=Higher-order chromatin structure is emerging as an important regulator of gene expression. Although dynamic chromatin structures have been identified in the genome, the full scope of chromatin dynamics during mammalian development and lineage specification remains to be determined. By mapping genome-wide chromatin interactions in human embryonic stem (ES) cells and four human ES-cell-derived lineages, we uncover extensive chromatin reorganization during lineage specification. We observe that although self-associating chromatin domains are stable during differentiation, chromatin interactions both within and between domains change in a striking manner, altering 36% of active and inactive chromosomal compartments throughout the genome. By integrating chromatin interaction maps with haplotype-resolved epigenome and transcriptome data sets, we find widespread allelic bias in gene expression correlated with allele-biased chromatin states of linked promoters and distal enhancers. Our results therefore provide a global view of chromatin dynamics and a resource for studying long-range control of gene expression in distinct human cell lineages.;,citation_author=Jesse R. Dixon;,citation_author=Inkyung Jung;,citation_author=Siddarth Selvaraj;,citation_author=Yin Shen;,citation_author=Jessica E. Antosiewicz-Bourget;,citation_author=Ah Young Lee;,citation_author=Zhen Ye;,citation_author=Audrey Kim;,citation_author=Nisha Rajagopal;,citation_author=Wei Xie;,citation_author=Yarui Diao;,citation_author=Jing Liang;,citation_author=Huimin Zhao;,citation_author=Victor V. Lobanenkov;,citation_author=Joseph R. Ecker;,citation_author=James A. Thomson;,citation_author=Bing Ren;,citation_publication_date=2015-02;,citation_cover_date=2015-02;,citation_year=2015;,citation_fulltext_html_url=https://doi.org/10.1038/nature14222;,citation_issue=7539;,citation_doi=10.1038/nature14222;,citation_issn=1476-4687;,citation_volume=518;,citation_journal_title=Nature;">
<meta name="citation_reference" content="citation_title=The Hitchhiker’s Guide to Hi-C Analysis: Practical guidelines;,citation_abstract=Over the last decade, development and application of a set of molecular genomic approaches based on the chromosome conformation capture method (3C), combined with increasingly powerful imaging approaches, have enabled high resolution and genome-wide analysis of the spatial organization of chromosomes. The aim of this paper is to provide guidelines for analyzing and interpreting data obtained with genome-wide 3C methods such as Hi-C and 3C-seq that rely on deep sequencing to detect and quantify pairwise chromatin interactions genome-wide.;,citation_author=Bryan R. Lajoie;,citation_author=Job Dekker;,citation_author=Noam Kaplan;,citation_publication_date=2015-01;,citation_cover_date=2015-01;,citation_year=2015;,citation_fulltext_html_url=https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4347522/;,citation_doi=10.1016/j.ymeth.2014.10.031;,citation_issn=1046-2023;,citation_pmid=25448293;,citation_volume=72;,citation_journal_title=Methods (San Diego, Calif.);">
<meta name="citation_reference" content="citation_title=Chromatin alternates between A and B compartments at kilobase scale for subgenic organization;,citation_abstract=Abstract Nuclear compartments are prominent features of 3D chromatin organization, but sequencing depth limitations have impeded investigation at ultra fine-scale. CTCF loops are generally studied at a finer scale, but the impact of looping on proximal interactions remains enigmatic. Here, we critically examine nuclear compartments and CTCF loop-proximal interactions using a combination of in situ Hi-C at unparalleled depth, algorithm development, and biophysical modeling. Producing a large Hi-C map with 33 billion contacts in conjunction with an algorithm for performing principal component analysis on sparse, super massive matrices&nbsp;(POSSUMM), we resolve compartments to 500 bp. Our results demonstrate that essentially all active promoters and distal enhancers localize in the A compartment, even when flanking sequences do not. Furthermore, we find that the TSS and TTS of paused genes are often segregated into separate compartments. We then identify diffuse interactions that radiate from CTCF loop anchors, which correlate with strong enhancer-promoter interactions and proximal transcription. We also find that these diffuse interactions depend on CTCF’s RNA binding domains. In this work, we demonstrate features of fine-scale chromatin organization consistent with a revised model in which compartments are more precise than commonly thought while CTCF loops are more protracted.;,citation_author=Hannah L. Harris;,citation_author=Huiya Gu;,citation_author=Moshe Olshansky;,citation_author=Ailun Wang;,citation_author=Irene Farabella;,citation_author=Yossi Eliaz;,citation_author=Achyuth Kalluchi;,citation_author=Akshay Krishna;,citation_author=Mozes Jacobs;,citation_author=Gesine Cauer;,citation_author=Melanie Pham;,citation_author=Suhas S. P. Rao;,citation_author=Olga Dudchenko;,citation_author=Arina Omer;,citation_author=Kiana Mohajeri;,citation_author=Sungjae Kim;,citation_author=Michael H. Nichols;,citation_author=Eric S. Davis;,citation_author=Dimos Gkountaroulis;,citation_author=Devika Udupa;,citation_author=Aviva Presser Aiden;,citation_author=Victor G. Corces;,citation_author=Douglas H. Phanstiel;,citation_author=William Stafford Noble;,citation_author=Guy Nir;,citation_author=Michele Di Pierro;,citation_author=Jeong-Sun Seo;,citation_author=Michael E. Talkowski;,citation_author=Erez Lieberman Aiden;,citation_author=M. Jordan Rowley;,citation_publication_date=2023-06;,citation_cover_date=2023-06;,citation_year=2023;,citation_fulltext_html_url=https://www.nature.com/articles/s41467-023-38429-1;,citation_issue=1;,citation_doi=10.1038/s41467-023-38429-1;,citation_issn=2041-1723;,citation_volume=14;,citation_journal_title=Nature Communications;">
<meta name="citation_reference" content="citation_title=Iterative correction of Hi-C data reveals hallmarks of chromosome organization;,citation_author=Maxim Imakaev;,citation_author=Geoffrey Fudenberg;,citation_author=Rachel Patton McCord;,citation_author=Natalia Naumova;,citation_author=Anton Goloborodko;,citation_author=Bryan R Lajoie;,citation_author=Job Dekker;,citation_author=Leonid A Mirny;,citation_publication_date=2012-10;,citation_cover_date=2012-10;,citation_year=2012;,citation_fulltext_html_url=https://www.nature.com/articles/nmeth.2148;,citation_issue=10;,citation_doi=10.1038/nmeth.2148;,citation_issn=1548-7091, 1548-7105;,citation_volume=9;,citation_journal_title=Nature Methods;">
<meta name="citation_reference" content="citation_title=Open Chromosome Collective (Open2C);,citation_abstract=We are the Open Chromosome Collective (Open2C). We are a group of computational biologists working on chromosome structure and interested in developing open source tools for the wider community.We are particularly interested in the 3D organization of chromosomes, and most of the tools are focused on the analysis of data obtained using Hi-C and related technologies. We like our tools to be easy to use, flexible, to facilitate active development of novel analytical approaches, and scalable, to make use of the latest and largest datasets.;,citation_author=Open Chromosome Collective;,citation_publication_date=2024-11-13;,citation_cover_date=2024-11-13;,citation_year=2024;,citation_fulltext_html_url=https://open2c.github.io/;,citation_journal_title=Open2C;">
<meta name="citation_reference" content="citation_title=High resolution maps of chromatin reorganization through mouse meiosis reveal novel features of the 3D meiotic structure;,citation_abstract=Abstract When germ cells transition from the mitotic cycle into meiotic prophase I (MPI), chromosomes condense into an array of chromatin loops that are required to promote homolog pairing and genetic recombination. To identify the changes in chromosomal conformation, we isolated nuclei on a trajectory from spermatogonia to the end of MPI. At each stage along this trajectory, we built genomic interaction maps with the highest temporal and spatial resolution to date. The changes in chromatin folding coincided with a concurrent decline in mitotic cohesion and a rise in meiotic cohesin complexes. We found that the stereotypical large-scale A and B compartmentalization was lost during meiotic prophase I alongside the loss of topological associating domains (TADs). Still, local subcompartments were detected and maintained throughout meiosis. The enhanced Micro-C resolution revealed that, despite the loss of TADs, higher frequency contact sites between two loci were detectable during meiotic prophase I coinciding with CTCF bound sites. The pattern of interactions around these CTCF sites with their neighboring loci showed that CTCF sites were often anchoring the meiotic loops. Additionally, the localization of CTCF to the meiotic axes indicated that these anchors were at the base of loops. Strikingly, even in the face of the dramatic reconfiguration of interphase chromatin into a condensed loop-array, the interactions between regulatory elements remained well preserved. This establishes a potential mechanism for how the meiotic chromatin maintains active transcription within a highly structured genome. In summary, the high temporal and spatial resolution of these data revealed previously unappreciated aspects of mammalian meiotic chromatin organization.;,citation_author=Gang Cheng;,citation_author=Florencia Pratto;,citation_author=Kevin Brick;,citation_author=Xin Li;,citation_author=Benjamin Alleva;,citation_author=Mini Huang;,citation_author=Gabriel Lam;,citation_author=R. Daniel Camerini-Otero;,citation_publication_date=2024-03;,citation_cover_date=2024-03;,citation_year=2024;,citation_fulltext_html_url=http://biorxiv.org/lookup/doi/10.1101/2024.03.25.586627;,citation_doi=10.1101/2024.03.25.586627;">
<meta name="citation_reference" content="citation_title=3D chromatin structure changes during spermatogenesis and oogenesis;,citation_author=Shiqiang Zhang;,citation_author=Wanyu Tao;,citation_author=Jing-Dong J. Han;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_fulltext_html_url=https://linkinghub.elsevier.com/retrieve/pii/S200103702200188X;,citation_doi=10.1016/j.csbj.2022.05.032;,citation_issn=20010370;,citation_volume=20;,citation_journal_title=Computational and Structural Biotechnology Journal;">
<meta name="citation_reference" content="citation_title=Sex Chromosome Meiotic Drive;,citation_abstract=▪ Abstract  Sex chromosome drive refers to the unequal transmission of X and Y chromosomes from individuals of the heterogametic sex, resulting in biased sex ratios among progeny and within populations. The presence of driving sex chromosomes can reduce mean fitness within a population, bring about intragenomic conflict between the X chromosome, the Y, and the autosomes, and alter the intensity or mode of sexual selection within species. Sex chromosome drive, or its genetic equivalent, is known in plants, mammals, and flies. Many species harboring driving X chromosomes have evolved Y-linked and autosomal suppressors of drive. If a drive polymorphism is not stable, then driving chromosomes may spread to fixation and cause the extinction of a species. Certain characteristics of species, such as population density and female mating rate, may affect the probability of fixation of driving chromosomes. Thus, sex chromosome drive could be an agent of species-level selection.;,citation_author=John Jaenike;,citation_publication_date=2001-11;,citation_cover_date=2001-11;,citation_year=2001;,citation_fulltext_html_url=https://www.annualreviews.org/doi/10.1146/annurev.ecolsys.32.081501.113958;,citation_issue=1;,citation_doi=10.1146/annurev.ecolsys.32.081501.113958;,citation_issn=0066-4162;,citation_volume=32;,citation_journal_title=Annual Review of Ecology and Systematics;">
<meta name="citation_reference" content="citation_title=Do sex-linked male meiotic drivers contribute to intrinsic hybrid incompatibilities? Recent empirical studies from flies and rodents;,citation_author=Jun Kitano;,citation_author=Kohta Yoshida;,citation_publication_date=2023-08;,citation_cover_date=2023-08;,citation_year=2023;,citation_fulltext_html_url=https://linkinghub.elsevier.com/retrieve/pii/S0959437X23000485;,citation_doi=10.1016/j.gde.2023.102068;,citation_issn=0959437X;,citation_volume=81;,citation_journal_title=Current Opinion in Genetics &amp;amp;amp; Development;">
<meta name="citation_reference" content="citation_title=Munch-group;,citation_author=Kasper Munch;,citation_publication_date=2024-11;,citation_cover_date=2024-11;,citation_year=2024;,citation_fulltext_html_url=https://munch-group.org/research.html;">
<meta name="citation_reference" content="citation_title=Genome-wide coancestry reveals details of ancient and recent male-driven reticulation in baboons;,citation_abstract=Baboons (genus Papio ) are a morphologically and behaviorally diverse clade of catarrhine monkeys that have experienced hybridization between phenotypically and genetically distinct phylogenetic species. We used high-coverage whole-genome sequences from 225 wild baboons representing 19 geographic localities to investigate population genomics and interspecies gene flow. Our analyses provide an expanded picture of evolutionary reticulation among species and reveal patterns of population structure within and among species, including differential admixture among conspecific populations. We describe the first example of a baboon population with a genetic composition that is derived from three distinct lineages. The results reveal processes, both ancient and recent, that produced the observed mismatch between phylogenetic relationships based on matrilineal, patrilineal, and biparental inheritance. We also identified several candidate genes that may contribute to species-specific phenotypes.;,citation_author=Erik F. Sørensen;,citation_author=R. Alan Harris;,citation_author=Liye Zhang;,citation_author=Muthuswamy Raveendran;,citation_author=Lukas F. K. Kuderna;,citation_author=Jerilyn A. Walker;,citation_author=Jessica M. Storer;,citation_author=Martin Kuhlwilm;,citation_author=Claudia Fontsere;,citation_author=Lakshmi Seshadri;,citation_author=Christina M. Bergey;,citation_author=Andrew S. Burrell;,citation_author=Juraj Bergman;,citation_author=Jane E. Phillips-Conroy;,citation_author=Fekadu Shiferaw;,citation_author=Kenneth L. Chiou;,citation_author=Idrissa S. Chuma;,citation_author=Julius D. Keyyu;,citation_author=Julia Fischer;,citation_author=Marie-Claude Gingras;,citation_author=Sejal Salvi;,citation_author=Harshavardhan Doddapaneni;,citation_author=Mikkel H. Schierup;,citation_author=Mark A. Batzer;,citation_author=Clifford J. Jolly;,citation_author=Sascha Knauf;,citation_author=Dietmar Zinner;,citation_author=Kyle K.-H. Farh;,citation_author=Tomas Marques-Bonet;,citation_author=Kasper Munch;,citation_author=Christian Roos;,citation_author=Jeffrey Rogers;,citation_publication_date=2023-06;,citation_cover_date=2023-06;,citation_year=2023;,citation_fulltext_html_url=https://www.science.org/doi/10.1126/science.abn8153;,citation_issue=6648;,citation_doi=10.1126/science.abn8153;,citation_issn=0036-8075, 1095-9203;,citation_volume=380;,citation_journal_title=Science;">
<meta name="citation_reference" content="citation_title=Loops, topologically associating domains, compartments, and territories are elastic and robust to dramatic nuclear volume swelling;,citation_abstract=Abstract Layers of genome organization are becoming increasingly better characterized, but less is known about how these structures respond to perturbation or shape changes. Low-salt swelling of isolated chromatin fibers or nuclei has been used for decades to investigate the structural properties of chromatin. But, visible changes in chromatin appearance have not been linked to known building blocks of genome structure or features along the genome sequence. We combine low-salt swelling of isolated nuclei with genome-wide chromosome conformation capture (Hi-C) and imaging approaches to probe the effects of chromatin extension genome-wide. Photoconverted patterns on nuclei during expansion and contraction indicate that global genome structure is preserved after dramatic nuclear volume swelling, suggesting a highly elastic chromosome topology. Hi-C experiments before, during, and after nuclear swelling show changes in average contact probabilities at short length scales, reflecting the extension of the local chromatin fiber. But, surprisingly, during this large increase in nuclear volume, there is a striking maintenance of loops, TADs, active and inactive compartments, and chromosome territories. Subtle differences after expansion are observed, suggesting that the local chromatin state, protein interactions, and location in the nucleus can affect how strongly a given structure is maintained under stress. From these observations, we propose that genome topology is robust to extension of the chromatin fiber and isotropic shape change, and that this elasticity may be beneficial in physiological circumstances of changes in nuclear size and volume.;,citation_author=Jacob T. Sanders;,citation_author=Rosela Golloshi;,citation_author=Priyojit Das;,citation_author=Yang Xu;,citation_author=Peyton H. Terry;,citation_author=Darrian G. Nash;,citation_author=Job Dekker;,citation_author=Rachel Patton McCord;,citation_publication_date=2022-03;,citation_cover_date=2022-03;,citation_year=2022;,citation_fulltext_html_url=https://www.nature.com/articles/s41598-022-08602-5;,citation_issue=1;,citation_doi=10.1038/s41598-022-08602-5;,citation_issn=2045-2322;,citation_volume=12;,citation_journal_title=Scientific Reports;">
<meta name="citation_reference" content="citation_title=Application of hi-c and other omics data analysis in human cancer and cell differentiation research;,citation_abstract=With the development of 3C (chromosome conformation capture) and its derivative technology Hi-C (High-throughput chromosome conformation capture) research, the study of the spatial structure of the genomic sequence in the nucleus helps researchers understand the functions of biological processes such as gene transcription, replication, repair, and regulation. In this paper, we first introduce the research background and purpose of Hi-C data visualization analysis. After that, we discuss the Hi-C data analysis methods from genome 3D structure, A/B compartment, TADs (topologically associated domain), and loop detection. We also discuss how to apply genome visualization technologies to the identification of chromosome feature structures. We continue with a review of correlation analysis differences among multi-omics data, and how to apply Hi-C and other omics data analysis into cancer and cell differentiation research. Finally, we summarize the various problems in joint analyses based on Hi-C and other multi-omics data. We believe this review can help researchers better understand the progress and applications of 3D genome technology.;,citation_author=Haiyan Gong;,citation_author=Yi Yang;,citation_author=Sichen Zhang;,citation_author=Minghong Li;,citation_author=Xiaotong Zhang;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_fulltext_html_url=https://www.sciencedirect.com/science/article/pii/S2001037021001276;,citation_doi=https://doi.org/10.1016/j.csbj.2021.04.016;,citation_issn=2001-0370;,citation_volume=19;,citation_journal_title=Computational and Structural Biotechnology Journal;">
</head>

<body>

<header id="title-block-header" class="quarto-title-block default toc-left page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Chromatin Compartments and Selection on X</h1>
            <p class="subtitle lead">How Edges of Active Chromatin Align with Selection Regions in Primates</p>
          </div>

    
    <div class="quarto-title-meta-container">
      <div class="quarto-title-meta-column-start">
        
        <div class="quarto-title-meta">

                <div>
            <div class="quarto-title-meta-heading">Author</div>
            <div class="quarto-title-meta-contents">
                        <p>Søren Jørgensen <a href="mailto:201906763@post.au.dk" class="quarto-title-author-email"><i class="bi bi-envelope"></i></a> </p>
                      </div>
          </div>
                
                <div>
            <div class="quarto-title-meta-heading">Published</div>
            <div class="quarto-title-meta-contents">
              <p class="date">December 28, 2024</p>
            </div>
          </div>
          
                
              </div>
      </div>
      <div class="quarto-title-meta-column-end quarto-other-formats-target">
      <div class="quarto-alternate-formats"><div class="quarto-title-meta-heading">Other Formats</div><div class="quarto-title-meta-contents"><p><a href="index.pdf"><i class="bi bi-file-pdf"></i>PDF</a></p></div></div></div>
    </div>

    <div>
      <div class="abstract">
        <div class="block-title">Abstract</div>
        <p>Here, the 3D chromatin architecture of the X chromosome in rhesus macaque (<em>Macaca mulata</em>) is investigated in the context of evolutionary pressures and genetic drivers. We redo Hi-C analyses from a 2019 paper on the latest reference genome (<em>rheMac10</em> or <em>Mmul_10</em>). We compare two Hi-C analysis frameworks, <em>HiCExplorer</em> and <em>cooler/cooltools</em> (Open2C), on a subset, finding Open2C to be most flexible and intuitive. The ICE method (Iterative Correction and Eigendecomposition) was used to infer conventional and refined A/B compartments for fibroblast and four stages of spermatogenesis. We find that 200 kbp transition-zones between A/B-compartments in both fibroblasts and round spermatids align well with human selective sweeps where meiotic drive can explain the selection (ECH-regions), and with hybrid incompatility in baboons (genus <em>Papio</em>). We discuss the biological meaning of these findings, where conserved chromatin features may help to retain non-advantageous alleles, hinting to the role of selfish genetic elements in genome evolution.</p>
      </div>
    </div>

    <div>
      <div class="keywords">
        <div class="block-title">Keywords</div>
        <p>Hi-C, Chromatin Compartments, Evolutionary pressures, Selfish gene drive</p>
      </div>
    </div>

    <div class="quarto-other-links-text-target">
    </div>  </div>
</header><div id="quarto-content" class="page-columns page-rows-contents page-layout-article toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction">Introduction</a>
  <ul class="collapse">
  <li><a href="#reproducible-research-and-quarto" id="toc-reproducible-research-and-quarto" class="nav-link" data-scroll-target="#reproducible-research-and-quarto">Reproducible Research and Quarto</a></li>
  <li><a href="#sexual-reproduction-spermatogenesis-meiosis" id="toc-sexual-reproduction-spermatogenesis-meiosis" class="nav-link" data-scroll-target="#sexual-reproduction-spermatogenesis-meiosis">Sexual reproduction (spermatogenesis, meiosis)</a></li>
  <li><a href="#selfish-genes-and-randomness" id="toc-selfish-genes-and-randomness" class="nav-link" data-scroll-target="#selfish-genes-and-randomness">Selfish genes (and randomness)</a></li>
  <li><a href="#our-organism-of-interest-wang-et-al.-and-the-references" id="toc-our-organism-of-interest-wang-et-al.-and-the-references" class="nav-link" data-scroll-target="#our-organism-of-interest-wang-et-al.-and-the-references">Our Organism of Interest, Wang et al., and the references</a></li>
  <li><a href="#extended-common-haplotypes-discovered-in-humans" id="toc-extended-common-haplotypes-discovered-in-humans" class="nav-link" data-scroll-target="#extended-common-haplotypes-discovered-in-humans">Extended Common Haplotypes Discovered in Humans</a></li>
  <li><a href="#c-chromatin-conformation-capture" id="toc-c-chromatin-conformation-capture" class="nav-link" data-scroll-target="#c-chromatin-conformation-capture">3C: Chromatin Conformation Capture</a></li>
  <li><a href="#hi-c-high-throughput-3c" id="toc-hi-c-high-throughput-3c" class="nav-link" data-scroll-target="#hi-c-high-throughput-3c">Hi-C: High-Throughput 3C</a>
  <ul class="collapse">
  <li><a href="#hi-c-library-preparation" id="toc-hi-c-library-preparation" class="nav-link" data-scroll-target="#hi-c-library-preparation">Hi-C Library preparation</a></li>
  <li><a href="#hi-c-data-analysis" id="toc-hi-c-data-analysis" class="nav-link" data-scroll-target="#hi-c-data-analysis">Hi-C Data Analysis</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#methods" id="toc-methods" class="nav-link" data-scroll-target="#methods">Methods</a>
  <ul class="collapse">
  <li><a href="#initial-exploration-with-hicexplorer" id="toc-initial-exploration-with-hicexplorer" class="nav-link" data-scroll-target="#initial-exploration-with-hicexplorer">Initial Exploration with HiCExplorer</a></li>
  <li><a href="#downloading-data-and-project-structure" id="toc-downloading-data-and-project-structure" class="nav-link" data-scroll-target="#downloading-data-and-project-structure">Downloading Data and Project Structure</a></li>
  <li><a href="#handling-coolers-or-preparing-coolers" id="toc-handling-coolers-or-preparing-coolers" class="nav-link" data-scroll-target="#handling-coolers-or-preparing-coolers">Handling coolers (Or: preparing coolers)</a>
  <ul class="collapse">
  <li><a href="#the-gwf-workflow-targets" id="toc-the-gwf-workflow-targets" class="nav-link" data-scroll-target="#the-gwf-workflow-targets">The <em>gwf</em> workflow targets</a></li>
  <li><a href="#notebook-edits" id="toc-notebook-edits" class="nav-link" data-scroll-target="#notebook-edits">Notebook edits</a></li>
  <li><a href="#compartments-and-their-edges-transitional-regions" id="toc-compartments-and-their-edges-transitional-regions" class="nav-link" data-scroll-target="#compartments-and-their-edges-transitional-regions">Compartments and Their Edges (Transitional Regions)</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#results" id="toc-results" class="nav-link" data-scroll-target="#results">Results</a>
  <ul class="collapse">
  <li><a href="#exploration-with-hicexplorer" id="toc-exploration-with-hicexplorer" class="nav-link" data-scroll-target="#exploration-with-hicexplorer">Exploration with HicExplorer</a>
  <ul class="collapse">
  <li><a href="#quality-control" id="toc-quality-control" class="nav-link" data-scroll-target="#quality-control">Quality Control</a></li>
  <li><a href="#correction" id="toc-correction" class="nav-link" data-scroll-target="#correction">Correction</a></li>
  <li><a href="#eigenvectors" id="toc-eigenvectors" class="nav-link" data-scroll-target="#eigenvectors">Eigenvectors</a></li>
  </ul></li>
  <li><a href="#open2c-ecosystem" id="toc-open2c-ecosystem" class="nav-link" data-scroll-target="#open2c-ecosystem">Open2c ecosystem</a>
  <ul class="collapse">
  <li><a href="#quality-control-1" id="toc-quality-control-1" class="nav-link" data-scroll-target="#quality-control-1">Quality Control</a></li>
  <li><a href="#correction-1" id="toc-correction-1" class="nav-link" data-scroll-target="#correction-1">Correction</a></li>
  <li><a href="#compartments-eigenvectors" id="toc-compartments-eigenvectors" class="nav-link" data-scroll-target="#compartments-eigenvectors">Compartments (Eigenvectors)</a></li>
  <li><a href="#compartment-edges-transition-zones" id="toc-compartment-edges-transition-zones" class="nav-link" data-scroll-target="#compartment-edges-transition-zones">Compartment Edges (transition zones)</a></li>
  </ul></li>
  <li><a href="#testing-against-hybrid-incompatibility-in-baboons" id="toc-testing-against-hybrid-incompatibility-in-baboons" class="nav-link" data-scroll-target="#testing-against-hybrid-incompatibility-in-baboons">Testing against hybrid incompatibility in baboons</a></li>
  </ul></li>
  <li><a href="#discussion" id="toc-discussion" class="nav-link" data-scroll-target="#discussion">Discussion</a></li>
  <li><a href="#bibliography" id="toc-bibliography" class="nav-link" data-scroll-target="#bibliography">Bibliography</a></li>
  </ul>
</nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
</div>
<main class="content quarto-banner-title-block" id="quarto-document-content">



  


<section id="introduction" class="level1">
<h1>Introduction</h1>
<section id="reproducible-research-and-quarto" class="level2">
<h2 class="anchored" data-anchor-id="reproducible-research-and-quarto">Reproducible Research and Quarto</h2>
</section>
<section id="sexual-reproduction-spermatogenesis-meiosis" class="level2">
<h2 class="anchored" data-anchor-id="sexual-reproduction-spermatogenesis-meiosis">Sexual reproduction (spermatogenesis, meiosis)</h2>
<p>The production of gametes in a sexually reproducing organism is a highly complex process that involves numeruous elements. Spermatogenesis, the process of forming male gametes, involves four stages of differentiation from a germ cell through <em>spermatogonia</em>, <em>pachytene spermatocyte</em>, and <em>round spermatids</em> to <em>spermatozoa</em>, or <em>sperm</em> <span class="citation" data-cites="wang_reprogramming_2019">(<a href="#ref-wang_reprogramming_2019" role="doc-biblioref">Wang et al. 2019</a>)</span>, and it is the very basis of male reproduction. The specialized cell division of meiosis neatly handles the pairing, recombination, and segregation of homologous chromosomes, thereby ensuring proper genetic distribution. Deeply understanding the steps of molecular steps of reproduction and how our genetic material is inherited is essential in biology, bringing insight to areas such as speciation, population diversity, and (male) infertility.</p>
</section>
<section id="selfish-genes-and-randomness" class="level2">
<h2 class="anchored" data-anchor-id="selfish-genes-and-randomness">Selfish genes (and randomness)</h2>
<p>The conventional story of meiosis in gametogenesis is one of random segregation of the sex chromosomes. They split into haploid gametes, where each chromosome has an equal chance of being passed on to a gamete. That seems like a fair game, but what if some genes are cheating the system by making others less viable. A meiotic driver is a selfish gene element that modulates meiosis and preferentially transmits its own allele through meiosis, regardless of the downstream fitness effects it may have (good or bad) on the organism it is part of. This phenomenon challenges the traditional understanding of selection, extending its scope beyond the fitness effects on an organism to include selective pressures at the molecular level. For example, if some genes on the X chromosome create a disadvantage for gametes that <em>do not</em> contain those genes, making sure the Y chromosome is not as viable as the X, resulting in a sex imbalance and possibly numerous other downstream effects. That is exactly what is coined <em>sex chromosome meiotic drive</em> <span class="citation" data-cites="jaenike_sex_2001">(<a href="#ref-jaenike_sex_2001" role="doc-biblioref">Jaenike 2001</a>)</span>, a result of selfish genetic elements.</p>
<p>Motivated by previous results in the Munch Research group <span class="citation" data-cites="munch_group_2024">(<a href="#ref-munch_group_2024" role="doc-biblioref">Munch 2024</a>)</span> on hybrid incompatibility and extended common haplotypes <span class="citation" data-cites="skov_extraordinary_2023 sorensen_genome_wide_2023">(<a href="#ref-skov_extraordinary_2023" role="doc-biblioref">Skov et al. 2023</a>; <a href="#ref-sorensen_genome_wide_2023" role="doc-biblioref">Sørensen et al. 2023</a>)</span> that could be explained by meiotic drive, we wanted to investigate how these patterns correlate with chromatin compartments.</p>
</section>
<section id="our-organism-of-interest-wang-et-al.-and-the-references" class="level2">
<h2 class="anchored" data-anchor-id="our-organism-of-interest-wang-et-al.-and-the-references">Our Organism of Interest, Wang et al., and the references</h2>
</section>
<section id="extended-common-haplotypes-discovered-in-humans" class="level2">
<h2 class="anchored" data-anchor-id="extended-common-haplotypes-discovered-in-humans">Extended Common Haplotypes Discovered in Humans</h2>
</section>
<section id="c-chromatin-conformation-capture" class="level2">
<h2 class="anchored" data-anchor-id="c-chromatin-conformation-capture">3C: Chromatin Conformation Capture</h2>
</section>
<section id="hi-c-high-throughput-3c" class="level2">
<h2 class="anchored" data-anchor-id="hi-c-high-throughput-3c">Hi-C: High-Throughput 3C</h2>
<p>Our DNA can be divided into different orders of structure. <em>3C</em> focus on identifying the highest orders of organization inside the nucleus, that is, when the 30 nm thick coil of chromatin fibers folds into loops, Topologically Associating Domains (TADs), and chromatin compartments. Here, we narrow our focus on the largest of the structures, <em>compartments</em>, that is known to determine availability to transcription factors, thus making an <em>A</em> compartment <em>active</em>—and the <em>B</em> compartment <em>inactive</em>. The introduction of the Hi-C (high-throughput 3C) method <span class="citation" data-cites="lieberman_aiden_comprehensive_2009">(<a href="#ref-lieberman_aiden_comprehensive_2009" role="doc-biblioref">Lieberman-Aiden et al. 2009</a>)</span> opened new possibilities for exploring the three-dimensional organization of the genome.</p>
<section id="hi-c-library-preparation" class="level3">
<h3 class="anchored" data-anchor-id="hi-c-library-preparation">Hi-C Library preparation</h3>
<p>A specialized protocol for preparing the DNA library is necessary [<span class="citation" data-cites="lieberman_aiden_comprehensive_2009">Lieberman-Aiden et al. (<a href="#ref-lieberman_aiden_comprehensive_2009" role="doc-biblioref">2009</a>)</span>; Fig. 1a]. Briefly, formaldehyde is used to crosslink spatially adjecent chromatin. Restriction enzymes are used to digest the crosslinked chromatin, leaving sticky ends that are filled and biotinylated with a polymerase. Then chimeric molecules are ligated in highly dilute conditions, where the restriction site is lost and a ligation site (also referred to as <em>ligation junction</em>) is created. Lastly, the ligation junction are isolated with streptavidin beads and sequenced as a paired-end library.</p>
<p>To be able to create stage-resolved Hi-C library of spermatogenetis, additional steps have to be performed. First, the samples have to be treated immediately after harvesting to ensure viable cells. Secondly, the samples have to be purified to accurately represent each stage og spermatogenesis. Specifically, the data for this project (<span class="citation" data-cites="wang_reprogramming_2019">Wang et al. (<a href="#ref-wang_reprogramming_2019" role="doc-biblioref">2019</a>)</span>; acc. GSE109344) preluded the library preparation protocol by sedimentation-based cell sorting to separate live spermatogenic cells into different stages of differentiation, namely spermatogonia, pachytene spermatocyte, round spermatid, and spermatozoa. Then, the cells were fixed in their respective state before crosslinking. They use their own derived method for library preparation, termed small-scale <em>in-situ</em> Hi-C, allegedly producing a high-quality Hi-C libary from as little as 500 cells (capturing the variance of millions of cells).</p>
</section>
<section id="hi-c-data-analysis" class="level3">
<h3 class="anchored" data-anchor-id="hi-c-data-analysis">Hi-C Data Analysis</h3>
<p>The analysis of the read-pairs of a Hi-C library is divided into several smaller tasks: We must align the reads to the reference in a way such that the <em>intentional</em> chimeric read-pairs are rescued, and <em>unintentional</em> read-pairs are discarded. That is, they represent ligation junction of adjecent chromatin segments, or they represent technical artefacts or unintentional fusions of unrelated DNA.</p>
<section id="aligning-the-hi-c-reads" class="level4">
<h4 class="anchored" data-anchor-id="aligning-the-hi-c-reads">Aligning the Hi-C reads</h4>
<p>Any software for local alignment can be used for aligning reads from a Hi-C library. However, one should make sure to disable paired-end mode if possible, otherwise each read in a pair (each mate) should be aligned separately <span class="citation" data-cites="lajoie_hitchhikers_2015">(<a href="#ref-lajoie_hitchhikers_2015" role="doc-biblioref">Lajoie, Dekker, and Kaplan 2015</a>)</span>. This removes the assumption that the distance between mates fits a known distribution because the genomic sequences originate from a continuous DNA-fragment. For example, the <em>bwa-mem</em> [ref] implementation of this (the <code>-P</code> option) activates the Smith-Waterman algorithm to rescue missing hits, but disables the search of hits that fit a ‘proper’ pair. After alignment, each read is assigned to the nearest restriction fragment.</p>
</section>
<section id="identifying-and-storing-valid-hi-c-pairs" class="level4">
<h4 class="anchored" data-anchor-id="identifying-and-storing-valid-hi-c-pairs">Identifying and Storing Valid Hi-C Pairs</h4>
<p>One should be cautious when filtering invalid from valid pairs, as they are not easily distinguished. We should be able to put a detected ligation event into one of five categories: dangling-end, self-circle, weird, intrachromosomal, and interchromosomal (<a href="#tbl-ligation-events" class="quarto-xref">Table&nbsp;1</a>) (<span class="citation" data-cites="bicciato_hic_2022">Bicciato and Ferrari (<a href="#ref-bicciato_hic_2022" role="doc-biblioref">2022</a>)</span>; Ch. 1).</p>
<div id="tbl-ligation-events" class="striped quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-ligation-events-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;1: Five categories of ligation events.
</figcaption>
<div aria-describedby="tbl-ligation-events-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<table class="table-striped caption-top table">
<colgroup>
<col style="width: 25%">
<col style="width: 75%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Event name</th>
<th style="text-align: left;">Explanation</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Dangling-end</td>
<td style="text-align: left;">Non-digested collinear fragments. Fraction can be high.</td>
</tr>
<tr class="even">
<td style="text-align: left;">Self-circle</td>
<td style="text-align: left;">Collinear fragment(s) have circularized. Very low fraction could indicate unsuccesful ligation.</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Weird</td>
<td style="text-align: left;">Mates have the same orientation on the reference. Is not possible with single copy fragment. Either sequencing errors or diploid fragments<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a>.</td>
</tr>
<tr class="even">
<td style="text-align: left;">Intrachromosomal</td>
<td style="text-align: left;">Pairs from the same chromosome</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Interchromosomal</td>
<td style="text-align: left;">Pairs from distinct chromosomes</td>
</tr>
</tbody>
</table>
</div>
</figure>
</div>
</section>
<section id="interaction-matrices-and-quality-control" class="level4">
<h4 class="anchored" data-anchor-id="interaction-matrices-and-quality-control">Interaction Matrices and Quality Control</h4>
</section>
<section id="calling-compartments-ice" class="level4">
<h4 class="anchored" data-anchor-id="calling-compartments-ice">Calling Compartments (ICE)</h4>
</section>
<section id="edges" class="level4">
<h4 class="anchored" data-anchor-id="edges">Edges</h4>
</section>
</section>
</section>
</section>
<section id="methods" class="level1">
<h1>Methods</h1>
<p>In this project, we formulate two objectives:</p>
<p><strong>A</strong>: Reproduce the Hi-C interaction maps and eigendecomposition from <span class="citation" data-cites="wang_reprogramming_2019">(<a href="#ref-wang_reprogramming_2019" role="doc-biblioref">Wang et al. 2019</a>)</span>, with some modifications. We briefly use <em>HiCExplorer</em>, but change the analyses to use the <em>Open2C Ecosystem</em> <span class="citation" data-cites="open2c">(<a href="#ref-open2c" role="doc-biblioref">Open Chromosome Collective 2024</a>)</span> which have a Pyton API as well as command-line functions, which can be paired very well with Jupyter Notebooks. The majority of the data analysis was run with a <em>gwf</em> workflow, and the commands that were visually inspected were run in Jupyter Notebooks.</p>
<p><strong>B</strong> Compare with regions of selection that are found in <em>human</em>. Investigate the biological meaning of the results.</p>
<p>All computations were performed on GenomeDK (GDK) [ref], an HPC cluster located on Aarhus Uninversity, and most of the processing of the data was nested into a <em>gwf</em> workflow [ref], a workflow manager developed at GDK. I would like to thank GDK and Aarhus University for providing computational resources and support that contributed to these research results.</p>
<p>The whole of this project is carried out with reproducibility in mind, so an effort (and quite a significant amount of time) has been put into documenting code and organizing the project for readbility and transparency through a Quarto project [ref]. Therefore, all code, virtual environments and text is made available as a Quarto book, rendered directly from the GitHub repository with GitHub Pages [ref]. To make this possible, the Quarto documentation has been extensively studied and discussed with <em>KMT</em> [ref, aknowledge].</p>
<section id="initial-exploration-with-hicexplorer" class="level2">
<h2 class="anchored" data-anchor-id="initial-exploration-with-hicexplorer">Initial Exploration with HiCExplorer</h2>
<p>Here moves most of the text about HiCExplorer…</p>
<p>For the initial exploration of methods with <em>HiCExplorer</em>, the 5 first samples in ‘fibroblast’ were chosen (<a href="#tbl-hic-exploration" class="quarto-xref">Table&nbsp;2</a>).</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/01_hicexplorer.ipynb" data-notebook-title="HiCExplorer using macaque data" data-notebook-cellid="cell-tbl-hic-exploration">
<div class="cell">
<div id="tbl-hic-exploration" class="cell quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-hic-exploration-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;2: The samples chosen for initial data exploration with HiCExplorer. From NCBI SRA Portal.
</figcaption>
<div aria-describedby="tbl-hic-exploration-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output cell-output-display" data-execution_count="3">
<div>


<table class="dataframe do-not-create-environment cell caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">Run</th>
<th data-quarto-table-cell-role="th">Bases</th>
<th data-quarto-table-cell-role="th">Bytes</th>
<th data-quarto-table-cell-role="th">source_name</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>SRR6502335</td>
<td>73201141800</td>
<td>31966430779</td>
<td>fibroblast</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>SRR6502336</td>
<td>65119970100</td>
<td>24433383054</td>
<td>fibroblast</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>SRR6502337</td>
<td>52769196300</td>
<td>23015357755</td>
<td>fibroblast</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3</td>
<td>SRR6502338</td>
<td>52378949100</td>
<td>22999581685</td>
<td>fibroblast</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">4</td>
<td>SRR6502339</td>
<td>28885941600</td>
<td>10960123150</td>
<td>fibroblast</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
</figure>
</div>
</div>
</div>
<p>The goal was to replicate some of the figures from <span class="citation" data-cites="wang_reprogramming_2019">Wang et al. (<a href="#ref-wang_reprogramming_2019" role="doc-biblioref">2019</a>)</span> using <em>HiCExplorer</em>, especially to reconstruct interaction matrices and E1 graphs from macaque data.</p>
<p>The matrices are constructed with <code>hicBuildMatrix</code> from separately mapped read-pairs. Along with the matrix <em>.h5</em> file, a <em>.log</em> file it outputted, documenting the quality control for the sample. Multiple logs can be aggregated and visualized with <code>hicQC</code>.</p>
<p>Before correction (or balancing) of the interaction matrix, a pre-correction filter is applied, filtering out low-count bins and very high-count bins. A threshold for Mean Absolute Deviation (<em>MAD</em>) is estimated by <code>hicCorrect diagnostic_plot</code>, followed by iterative correction with <code>hicCorrect correct --correctionMethod ICE</code>.</p>
<p>The PCA was performed with <code>hicPCA</code> on the correcte matrices, yielding the first 3 PCs.</p>
</section>
<section id="downloading-data-and-project-structure" class="level2">
<h2 class="anchored" data-anchor-id="downloading-data-and-project-structure">Downloading Data and Project Structure</h2>
<p>To reproduce the results from <span class="citation" data-cites="wang_reprogramming_2019">(<a href="#ref-wang_reprogramming_2019" role="doc-biblioref">Wang et al. 2019</a>)</span>, I chose to use their raw data directly from the SRA portal [ref]. I filtered the data to contain all their paired-end Hi-C reads, and included only macaque samples. The data set also contains RNAseq data, and the same tissues for both macaque and mouse. The meta data for the data set was extracted into a runtable <code>SRA-runtable.tsv</code>. To get an overview of the data accessions used in this analysis, we will first summarize the runtable that contains the accession numbers and some metadata for each sample (<a href="#tbl-runtable-summary" class="quarto-xref">Table&nbsp;3</a>). It adds up to ~1Tb of compressed <code>fastq</code> files, holding ~9.5 billion reads, roughly evenly spread on the 5 tissue types.</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/03_compartments.ipynb" data-notebook-title="Compartments Analysis" data-notebook-cellid="cell-tbl-runtable-summary">
<div class="cell" data-execution_count="47">
<div id="tbl-runtable-summary" class="cell quarto-float quarto-figure quarto-figure-center anchored" data-execution_count="47">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-runtable-summary-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;3: Summary of the data accessions used in this analysis
</figcaption>
<div aria-describedby="tbl-runtable-summary-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output cell-output-display">
<div>


<table class="dataframe do-not-create-environment cell caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">source_name</th>
<th data-quarto-table-cell-role="th">GB</th>
<th data-quarto-table-cell-role="th">Bases</th>
<th data-quarto-table-cell-role="th">Reads</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>fibroblast</td>
<td>211.403275</td>
<td>553,968,406,500</td>
<td>1,846,561,355</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>pachytene spermatocyte</td>
<td>274.835160</td>
<td>715,656,614,700</td>
<td>2,385,522,049</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>round spermatid</td>
<td>243.128044</td>
<td>655,938,457,200</td>
<td>2,186,461,524</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3</td>
<td>sperm</td>
<td>164.131640</td>
<td>428,913,635,400</td>
<td>1,429,712,118</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">4</td>
<td>spermatogonia</td>
<td>192.794420</td>
<td>518,665,980,300</td>
<td>1,728,886,601</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
</figure>
</div>
</div>
</div>
</section>
<section id="handling-coolers-or-preparing-coolers" class="level2">
<h2 class="anchored" data-anchor-id="handling-coolers-or-preparing-coolers">Handling coolers (Or: preparing coolers)</h2>
<div id="fig-flowchart-handling-coolers" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-flowchart-handling-coolers-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="illustrations/placeholder2000x360.png" class="lightbox" data-gallery="quarto-lightbox-gallery-1" title="Figure&nbsp;1: A flowchart showing the pipeline from .fastq to .mcool. The first 6 steps were done with a Probably BioRender or Inkscape."><img src="illustrations/placeholder2000x360.png" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-flowchart-handling-coolers-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1: A flowchart showing the pipeline from <code>.fastq</code> to <code>.mcool</code>. The first 6 steps were done with a Probably BioRender or Inkscape.
</figcaption>
</figure>
</div>
<section id="the-gwf-workflow-targets" class="level3">
<h3 class="anchored" data-anchor-id="the-gwf-workflow-targets">The <em>gwf</em> workflow targets</h3>
<p>A <em>gwf</em> workflow was created to handle the first part of the data processing, and each accesion number (read pair, mate pair) from the Hi-C sequencing was processed in parallel, so their execution was independent from each other.</p>
<section id="downloading-the-reads" class="level4">
<h4 class="anchored" data-anchor-id="downloading-the-reads">Downloading the reads</h4>
<p>The reads were downloaded from NCBI SRA portal [ref] directly to GDK using <code>sra-downloader</code> [ref] as <code>.fastq.gz</code> files.</p>
</section>
<section id="handling-the-reference" class="level4">
<h4 class="anchored" data-anchor-id="handling-the-reference">Handling the reference</h4>
<p>The latest reference genome for rhesus macaque (<em>macaca mulata</em>), <em>rheMac10</em> (or <em>Mmul_10</em>, UCSC or NCBI naming conventions, respectively) was downloaded to GDK from UCSC web servers with <code>wget</code> [ref]. To use <code>bwa</code> (Burrow Wheeler’s Aligner) [ref] for mapping, rheMac10 needs to be indexed with both <code>bwa index</code> with the <code>--bwtsw</code> option and <code>samtools faidx</code>, which results in six indexing files for <code>bwa mem</code> to use.</p>
<p>Since <span class="citation" data-cites="wang_reprogramming_2019">(<a href="#ref-wang_reprogramming_2019" role="doc-biblioref">2019</a>)</span>, the reference genome for rhesus macaque has changed several times from <em>rheMac2</em> to <em>rheMac10</em>, each time resulting in a much less fragmented reference assembly. Part of the reasoning for reproducing their results was doing so on the latest assembly of the Macaca mulata genome, which arguably will result in a more accurate mapping of the reads, and a better inference of the chromatin compartments as well.</p>
<p>Several mappers were used in different configurations (described in below), and <code>bowtie2</code> requires its own indexing of the reference, using <code>bowtie2-build --large-index</code>, which creates six index files for <code>bowtie2</code> to use. <code>--large-index</code> creates the special indexing format required for large genomes such as macaque.</p>
</section>
<section id="mapping-hi-c-reads" class="level4">
<h4 class="anchored" data-anchor-id="mapping-hi-c-reads">Mapping Hi-C reads</h4>
<p>paragraph will be restructured.</p>
<p>The main difference between Hi-C libraries and standard paired-end libraries is the high fraction of chimeric reads in Hi-C. As a contact pair is crosslinked and ligated before sequencing, chimeric reads occur as a feature, and standard mapping techniques seeks to filter out this type of reads [ref]. Thus, we need specialized tools for rescuing chimeric reads. That said, we have to be cautious distinguishing the intended chimerism for Hi-C and that of technical artefacts.</p>
<p>It was not feasible to follow the same approach as <span class="citation" data-cites="wang_reprogramming_2019">(<a href="#ref-wang_reprogramming_2019" role="doc-biblioref">Wang et al. 2019</a>)</span> with either <em>HiCExplorer</em> or <em>Open2C</em>, as they use a third software, <em>HiC-Pro</em>. Hic-Pro uses bowtie2 in end-to-end mode, followed by remapping of 5’-ends of the unmapped reads to rescue chimeric fragments along with another approach. I mapped the reads using <code>bowtie2 --end-to-end</code> without the rescue-remapping, and it returned a very high fraction of discarded reads. I argue that even when trying to reproduce results, it is nonsensical to use methods that are not state-of-the-art. The HiC-Pro pipeline stops at a normalized contact map, and is thus not sufficient for downstream analysis.</p>
<p><strong>HiCExplorer</strong> Initially, recommendations from HiCExplorer were used. According to their documentation [ref] it is crucial to 1) align reads locally, and 2) map mates separately. They recommend either of <code>bwa</code> or <code>bowtie2</code>, so I tested both with their recommended settings. <code>bowtie2</code> turned out to be a lot more resource-intensive and to produce almost no mapped reads [ref sup-fig-bowtie2-stats], so I suspect some settings was not set correctly. The mapped reads was converted to a Hi-C Matrix (<code>.h5</code>) with <em>HiCExplorers</em> <code>hicBuildMatrix</code>, which is extremely memory-intensive, using ~120 GB memory for the biggest matrix. I followed <em>HiCExplorer</em> pipeline to plot and explore the matrices created from this mapping. However, the work was laborious for experimentation, as, even though written in Python, <code>HicExplorer</code> only comes with a command-line interface and provided functions all write plots to files. I did not manage to make an efficient implementation for plotting the <code>.h5</code> files produced by the pipeline, as would be required for utilizing Jupyter Notebooks for customizing plots. I relatively quickly shifted to <em>Open2C</em> for their promises of the greener grass (a Python API).</p>
<p><strong>Open2C</strong> Suspiciously, <span class="citation" data-cites="open2c">(<a href="#ref-open2c" role="doc-biblioref">Open Chromosome Collective 2024</a>)</span> never mentions any problems with aligning the Hi-C reads, they just provide an example using <code>bwa mem</code> in paired-end mode and with the <code>-P</code> option set, which activates the Smith-Waterman [ref] algorithm to rescue missing hits, by focusing on assigning only of of the mates to a good mapping and escape mate-rescue. The documentation of <code>bwa</code> <a href="https://bio-bwa.sourceforge.net">ref</a> state that both bwa-mem and bwa-sw will rescue chimeric reads. Consequently, Open2C does not have a builtin way of pairing the reads after mapping, and I was left with two options: <strong>1)</strong> to re(-)pair the individually mapped read-mates (.bam) with <code>samtools-fixmate</code> into one of the specific input formats required for <code>cooler</code> to create an interaction matrix <em>cooler</em>, or <strong>2)</strong> re-map the reads using Open2C’s recommendations and use their established pipeline for producing a cooler. I chose the latter, where I mapped the fastq files to <em>rheMac10</em> in paired end mode for a pair (m1, m2) with <code>bwa mem -SP rheMac10 m1 m2</code>.</p>
</section>
<section id="parse-and-sort-the-reads" class="level4">
<h4 class="anchored" data-anchor-id="parse-and-sort-the-reads">Parse and sort the reads</h4>
<p>With <em>HiCExplorer</em> No action is needed, as this step is done implicitly when building the matrix</p>
<p><strong>Open2C</strong> We need to convert the alignments into ligation events, and distinguish between several types of ligation events. The simplest event is when each side only maps to one unique segment in the genome ‘UU’. Other events, where one or both sides map to multiple segments or the reads are long enough (&gt;150bp) to contain two alignments (multiple ligations) have to be considered as well. Multiple ligations (walks) are treated according to the <code>--walks-policy</code> when parsing the alignments into valid pairs (or valid Hi-C contacts). Here, <code>mask</code> is the most conservative and masks all complex walks, whereas <code>5unique</code> reports the 5’-most unique alignment on each side. The pairs are piped directly into <code>pairtools sort</code> after parsing, as the deduplication step requires a sorted set of pairs. The <em>.pairs</em>-format produced by <code>pairtools</code> is an extension the <a href="https://data.4dnucleome.org/file-formats/pairs/">4DN Consortium</a>-specified format, storing Hi-C pairs as in <a href="#tbl-pairsformat" class="quarto-xref">Table&nbsp;4</a>.</p>
<div id="tbl-pairsformat" class="striped quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-pairsformat-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;4: Column specification of the .pairs format as extended by <code>pairtools</code> [ref].
</figcaption>
<div aria-describedby="tbl-pairsformat-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<table class="table-striped caption-top table">
<colgroup>
<col style="width: 7%">
<col style="width: 12%">
<col style="width: 80%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: right;">Index</th>
<th style="text-align: left;">Name</th>
<th style="text-align: left;">Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: right;">1</td>
<td style="text-align: left;">read_id</td>
<td style="text-align: left;">the ID of the read as defined in fastq files</td>
</tr>
<tr class="even">
<td style="text-align: right;">2</td>
<td style="text-align: left;">chrom1</td>
<td style="text-align: left;">the chromosome of the alignment on side 1</td>
</tr>
<tr class="odd">
<td style="text-align: right;">3</td>
<td style="text-align: left;">pos1</td>
<td style="text-align: left;">the 1-based genomic position of the outer-most (5’) mapped bp on side 1</td>
</tr>
<tr class="even">
<td style="text-align: right;">4</td>
<td style="text-align: left;">chrom2</td>
<td style="text-align: left;">the chromosome of the alignment on side 2</td>
</tr>
<tr class="odd">
<td style="text-align: right;">5</td>
<td style="text-align: left;">pos2</td>
<td style="text-align: left;">the 1-based genomic position of the outer-most (5’) mapped bp on side 2</td>
</tr>
<tr class="even">
<td style="text-align: right;">6</td>
<td style="text-align: left;">strand1</td>
<td style="text-align: left;">the strand of the alignment on side 1</td>
</tr>
<tr class="odd">
<td style="text-align: right;">7</td>
<td style="text-align: left;">strand2</td>
<td style="text-align: left;">the strand of the alignment on side 2</td>
</tr>
<tr class="even">
<td style="text-align: right;">8</td>
<td style="text-align: left;">pair_type</td>
<td style="text-align: left;">the type of a Hi-C pair</td>
</tr>
<tr class="odd">
<td style="text-align: right;">9</td>
<td style="text-align: left;">mapq1</td>
<td style="text-align: left;">mapq of the first mate</td>
</tr>
<tr class="even">
<td style="text-align: right;">10</td>
<td style="text-align: left;">mapq2</td>
<td style="text-align: left;">mapq of the second mate</td>
</tr>
</tbody>
</table>
</div>
</figure>
</div>
<p>I initially used <code>--walks-policy mask</code>, reasoning I had plenty of data points and could handle complex walks in a conservative way. Only later I realized the recommendations from <em>pairtools</em>, specifically informing that longer reads might have a significant proportion of reads that contain complex walks. With this in mind, I decided to re-parse the alignments into a new set of pairs, and equally apply the recommended filter (next section). As both results are saved, we can compare the two approaches.</p>
</section>
<section id="filter-deduplicate-pairs" class="level4">
<h4 class="anchored" data-anchor-id="filter-deduplicate-pairs">Filter (deduplicate) pairs</h4>
<p>With <em>HiCExplorer</em>, no action is needed, as this step is done implicitly when building the matrix.</p>
<p><em>pairtools</em> comes with a de-duplication function, <code>dedup</code>, to detect PCR duplication artefacts. At this point we will remove all reads that are mapped to an unplaced scaffold. Even though the publication of <em>rhemac10</em> assembly states they have closed 99% of the gaps since <em>rhemac8</em> [ref], <em>rheMac10</em> still contain more than 2,500 unplaced scaffolds, which are all uninformative when calculating the chromatin compartments as is the goal of this analysis. Therefore, we simply only include the list of conventional chromosomes (1..22, X, Y) when doing the deduplication. Initially, the default values were used to remove duplicates, where pairs with both sides mapped within 3 base pairs from each other are considered duplicates.</p>
<p><code>cooler</code> recommend to store the most comprehensive and unfiltered list of pairs, and then applying a filter it on the fly by piping from <code>pairtools select</code>. Initially, I missed this step and I did not filter for mapping quality. After reparsing the alignments and applying the same analysis, we compare the two pipelines.</p>
<p>A quality control report is generated by <code>pairtools dedup</code>, and the reports are merged with <code>MultiQC</code> [ref] for each cell type.</p>
</section>
<section id="create-interaction-matrices-coolers" class="level4">
<h4 class="anchored" data-anchor-id="create-interaction-matrices-coolers">Create interaction matrices (coolers)</h4>
<p>With <em>HiCExplorer</em> <code>hicBuildMatrix</code> both parse and filter the mapped reads. The default value was used, where alignments with <span class="math inline">\(mapq &lt; 15\)</span> are discarded.</p>
<p><strong>Open2C</strong> The final part of the <em>gwf</em> workflow takes <code>.pairs</code> as input and outputs a <code>.cool</code> file (<em>cooler</em>). Initially, we read directly from the newly generated deduplicated pairs without additional filtering, but here, the official recommendation is to filter out everything below <span class="math inline">\(mapq = 30\)</span> by piping the pairs through <code>pairtools select "(mapq1&gt;=30) and (mapq2&gt;=30)"</code> to <code>cooler cload pairs</code>.</p>
<p>We should have plenty of data to do the filtering, but I argue it is not strictly necessary. I will show a histogram of the <em>mapq</em> scores to convince you [ref].</p>
<p>I have re-parsed the alignments and created new coolers, including only the Hi-C contacts where <span class="math inline">\(mapq \leq 30\)</span>, following the current recommendations from <code>cooler</code>.</p>
</section>
</section>
<section id="notebook-edits" class="level3">
<h3 class="anchored" data-anchor-id="notebook-edits">Notebook edits</h3>
<p>As <code>cooler</code> and <code>cooltools</code> have a Python API, the more experimental parts of the analysis were moved to Jupyter Notebooks. <code>cooltools</code> comes with a helper library for operations on genomic intervals called <code>bioframe</code>.</p>
<section id="pooling-samples-merging-coolers" class="level4">
<h4 class="anchored" data-anchor-id="pooling-samples-merging-coolers">Pooling samples (Merging coolers)</h4>
<p>The samples are grouped into <em>replicates</em> with a unique <strong>BioSample</strong> ID, but we chose to pool all the interaction matrices for each cell type. We argue that when <span class="citation" data-cites="wang_reprogramming_2019">Wang et al. (<a href="#ref-wang_reprogramming_2019" role="doc-biblioref">2019</a>)</span> determine compartments to be highly reproducible between replicates, by merging the replicates we can get a more robust signal.</p>
<p><code>cooler merge</code> was used to merge all samples in each sub-folder (cell type) to just one interaction matrix for each cell type. The function merges matrices of the same dimensions by simply adding the interaction frequencies of each genomic position together, resulting in less empty positions by chance.</p>
</section>
<section id="create-multi-resolution-coolers-zoomify" class="level4">
<h4 class="anchored" data-anchor-id="create-multi-resolution-coolers-zoomify">Create multi-resolution coolers (zoomify)</h4>
<p>A feature of working inside the ecosystem of <em>Open2C</em> [ref] is that it natively provides support for storing sparse interaction matrices in multiple resolutions in the same file by adding groups to the cooler [ref]. We can then efficiently store resolutions (i.e., different bin sizes) that is multiples of the smallest bin size. We chose to use 10kb, 50kb, 100kb, and 500kb bins, and the resolutions are made by recursively binning the base resolution. We call this process zoomifying.</p>
</section>
<section id="matrix-balancing-iterative-correction" class="level4">
<h4 class="anchored" data-anchor-id="matrix-balancing-iterative-correction">Matrix balancing (Iterative correction)</h4>
<p>Finally, we balance the matrices using the cooler CLI. We use <code>cooler balance</code> with the default options which iteratively balances the matrix (Iterative Correction). It is first described as a method for bias correction of Hi-C matrices in <span class="citation" data-cites="imakaev_iterative_2012">(<a href="#ref-imakaev_iterative_2012" role="doc-biblioref">Imakaev et al. 2012</a>)</span>, where it is paired with eigenvector decomposition, coining the combined analysis <strong>ICE</strong>. Here, the eigenvector decomposition of the obtained maps is experimentally validated to provide insights into local chromatin states.</p>
<p>[According to <code>cooler</code> documentation] We have to balance the matrices on each resolution, and thus it cannot be done prior to zoomifying. They state that the balancing weights are resolution-specific and will no longer retain its biological meaning when binned with other weights. Therefore, we apply <code>cooler balance</code> to each resolution separately. <code>cooler balance</code> will create a new column in the <code>bins</code> group of each cooler , <code>weight</code>, which can then be included or not in the downstream analysis. This means we will have access to both the balanced and the unbalanced matrix.</p>
<p>The default mode uses genome-wide data to calculate the weights for each bin. It would maybe be more suitable to calculate the weights for <em>cis</em> contacts only, and that is possible through the <code>--cis-only</code> flag, and that can be added to another column, so that we can compare the difference between the two methods easily. However, we will only use the default mode for now.</p>
</section>
<section id="eigendecomposition" class="level4">
<h4 class="anchored" data-anchor-id="eigendecomposition">Eigendecomposition</h4>
<p>The eigendecomposition of a Hi-C interaction matrix is performed in multiple steps. As value of the eigenvector is only <em>significant</em> up to a sign, it is convention [ref] to use GC content as a phasing track to orient the vector. E1 is arbitrarily defined to be positively correlated with GC content, meaning a positive E1 value signifies an active chromatin state, which we denote a A-type compartment (or simply A-compartment). We performed eigendecomposition of two resolutions, 100 Kbp and 500 Kbp. <span class="citation" data-cites="wang_reprogramming_2019">Wang et al. (<a href="#ref-wang_reprogramming_2019" role="doc-biblioref">2019</a>)</span> briefly describes their method to calculate the eigenvectors as a sliding window approach on the observed/expected matrix in 100 kb resolution summing over 400 kb bins with 100 kb step size, a method I was not able to replicate in the <em>Open2C</em> ecosystem. I decided to mimic this by smoothing the 100 kb E1 values by summing to 500 kb bins in steps of 100 kb, yielding a comparable resolution which I denote ‘<em>pseudo</em>-500 kb’ resolution (<em>ps500kb</em>).</p>
<p>First, we calculate the GC content of each bin of the reference genome, <em>rheMac10</em>, which is binned to the resolution of the Hi-C matrix we are handling. It is done with <code>bioframe.frac_gc</code> (<em>Open2C</em>). To calculate the E1 compartments, we use only within-chromosome contacts (<em>cis</em>), as we are not interested in the genome-wide contacts. <code>cooltools.eigs_cis</code> will decorrelate the contact-frequency by distance before performing the eigendecomposition. <code>eigs_cis</code> needs a <em>viewframe</em> (view) to calculate E1 values, the simplest view being the full chromosome. However, when there is more variance between chromosome arms than within arms, the sign of the first eigenvector will be determined largely by the chromosome arm it sits on, and not by the chromatin compartments. To mitigate this, we apply a chromosome-arm-partitioned view of the chromosome (as a bedlike format, described in <code>bioframe</code> docs [ref]).</p>
<p>Additionally, to mimic the <em>Local PCA</em> from <span class="citation" data-cites="wang_reprogramming_2019">(<a href="#ref-wang_reprogramming_2019" role="doc-biblioref">Wang et al. 2019</a>)</span>, I also defined a view of 10 Mb bins. Thoughout the project, I will compare results from each of the three views and resolutions.</p>
</section>
<section id="plotting-matrices" class="level4">
<h4 class="anchored" data-anchor-id="plotting-matrices">Plotting matrices</h4>
<p><em>HiCExplorer</em> plots matrices to .png from the command-line. When plots were generated (with <code>hicPlotMatrix</code>), it produces a .png output that has to be loaded back into the notebook. There is limited support for modifying the plot (from command-line options), such as to add spacing for a bigWig track with E1 values, add plot titles, and define the size and resolution of the plot. I briefly tried to implement a plotting function on the .h5 matrices and bigWig tracks, but it could not fetch regions from a matrix on the fly and had to load the full matrix into memory (that is, all full-length chromosomes). To better visualise differences in the interaction matrix, the interaction frequency <span class="math inline">\(f_i\)</span>, is transformed to <span class="math inline">\(\log\mathrm{1p}(f_i) = \log\left(1 + f_i\right)\)</span>. It is required to visualize the normalized interaction frequency, as there are mostly values very close to 0, but it also aids the visibility in the raw counts matrix.</p>
<p>We use matplotlib and seaborn to plot in the <em>Open2C</em> framework. Utilizing the <code>cooler</code> class, we can fetch regions of the matrix without modifying the file. As my analysis is centered around the X chromosome, it is efficiently handled by simply fetching ‘chrX’ from the matrix with <code>cooler.Cooler.matrix().fetch('chrX')</code>. Many methods of the cooler class returns data selectors, which do not retrieve data before it is queried [ref]. This means we can create many selectors at once without overflowing memory, enabling us to plot multiple interaction matrices side-by-side, e.g.&nbsp;the corrected and un-corrected matrices. This is easily done with the <code>balance</code> parameter of the matrix selector (<code>.matrix()</code>), which determines if it should apply the balancing weights to the coordinates and defaults to <code>True</code>.</p>
<p>The matrix is retrieved an plotted with <code>matplotlib.pyplot.matshow</code>, which automatically produces a heatmap image of the matrix. Here, in stead of transforming the interaction matrix, the color scale is log-transformed with <code>matplotlib.colors.LogNorm</code>. Additionally, <code>cooltools</code> comes with more tools to aid visualization: <em>adative coarsegrain</em> and <em>interpolation</em>, which can be chained. <code>adaptive_coarsegrain</code> iteratively coarsens an array to the nearest power of two and refines it back to the original resolution, replacing low-count pixels with NaN-aware averages to ensure no zeros in the output, unless there are very large regions that exceed the <code>max_levels</code> threshold, such as the peri-centromeric region.</p>
<p>I implemented a plotting utility, <code>plot_for_quarto</code> in notebook <code>07_various_plotting.ipynb</code> that is compatible with the YAML cell-options read by Quarto’s <code>embed</code> shortcode. It will take an arbitrary number of samples and plot a chromosome (or region) with or without its respective E1 value for either of the three viewframes that has been created. The input is a (subsetted) <em>pandas DataFrame</em>, defined from a file search matching a pattern specified to the <code>glob</code> Python module.</p>
</section>
</section>
<section id="compartments-and-their-edges-transitional-regions" class="level3">
<h3 class="anchored" data-anchor-id="compartments-and-their-edges-transitional-regions">Compartments and Their Edges (Transitional Regions)</h3>
<p>From the eigenvectors, the A-compartments were extracted in bedgraph-format (<code>['chrom', 'start', 'end']</code>) and compared with ECH90 regions lifted to <em>rheMac10</em> from human [ref what reference?]. We perform visual inspection of the genomic intervals and test whether ECH90 regions are enriched near the edges of the compartments by defining a 200 kilobase transition-zone centered at each sign change of E1 (referred to as <em>compartment edge</em>). We compare genomic intervals (or sets) both visually by plotting the regions, and by a proximity test and bootstrapping the Jaccard index.</p>
<section id="proximity-test" class="level4">
<h4 class="anchored" data-anchor-id="proximity-test">Proximity test</h4>
<p>Determines whether the non-overlapping parts of the sets are more proximal than expected by chance. We define the <em>annotation</em> set and the <em>query</em> set, and the distance from each interval on the <em>query</em> to the most proximal interval on the <em>annotation</em> is used to generate an index of proximity. It uses bootstrapping (<span class="math inline">\(b = 100000\)</span>) to generate the null distribution, and finally, the fraction of the <em>null</em> as or more extreme as our observed proximity is reported as the p-value.</p>
</section>
<section id="jaccard-test" class="level4">
<h4 class="anchored" data-anchor-id="jaccard-test">Jaccard test</h4>
<p>Measures the significance of the observed Jaccard index (intersection over union) between two sets. The index is a measure of <em>similarity</em> between two sets (between 0 and 1), which is very sensitive to the size difference between the sets, as even when comparing a set of intervals to a small subset of itself will yield a very small Jaccard index. When we use bootstrapping to generate a null distribution (shuffling the intervals of the <em>query</em>), we generate the probability that the two sets (with their respective number and size of intervals), are as similar or more than what we observe. The ratio is reported as the p-value. However, this approach is still sensitive to flipping of query/annotation, as only the query is bootstrapped.</p>
</section>
<section id="multiple-testing" class="level4">
<h4 class="anchored" data-anchor-id="multiple-testing">Multiple testing</h4>
<p>Careful considerations were made to avoid multiple testing biases (p-hacking): Performing tests on all combinations of variables (cell type, resolution, viewframe, flip annot, query, test) will yield 260 p-values, and we would have to adjust the significance threshold (with <span class="math inline">\(\alpha = 0.05\)</span>, we expect 13 ‘significant’ tests by chance alone). However, if we choose only</p>
</section>
</section>
</section>
</section>
<section id="results" class="level1">
<h1>Results</h1>
<section id="exploration-with-hicexplorer" class="level2">
<h2 class="anchored" data-anchor-id="exploration-with-hicexplorer">Exploration with HicExplorer</h2>
<section id="quality-control" class="level3">
<h3 class="anchored" data-anchor-id="quality-control">Quality Control</h3>
<p>The separately mapped read-mates were parsed into a <em>.h5</em> interaction matrix by <code>hicBuildMatrix</code>, which include a <em>.log</em> file documenting the builtin quality control (hereafter, <em>QC</em>). Log files from the 5 samples were merged with <code>hicQC</code> (<a href="#fig-explorer-qc" class="quarto-xref">Figure&nbsp;2</a>). We observe showed equal fractions of the read-orientation of read-pairs (<a href="#fig-explorer-read-orientation" class="quarto-xref">Figure&nbsp;2 (a)</a>), which is expected for a good Hi-C library. Additionally, it determines between 40% to 50% of the total reads to be valid Hi-C contacts (<a href="#fig-explorer-unique-pairs" class="quarto-xref">Figure&nbsp;2 (b)</a>), which is <a href="https://hicexplorer.readthedocs.io/en/latest/content/example_usage.html#creation-of-a-hi-c-matrix">allegedly</a> [ref] usually only 25%-40%. <del>Overall a solid set of Hi-C samples until now</del>. <a href="#fig-explorer-contact-distance" class="quarto-xref">Figure&nbsp;2 (e)</a> shows, however, unusually high fractions of <em>inter</em>-chromosomal contacts (up to 30%) compared to <em>intra</em>-chromosomal contacts (also denoted <em>trans</em> and <em>cis</em> contacts, respectively). It is expected that <em>cis</em> contacts are orders of magnitude more frequent than <em>trans</em> contacts <span class="citation" data-cites="bicciato_hic_2022 lieberman_aiden_comprehensive_2009">(<a href="#ref-bicciato_hic_2022" role="doc-biblioref">Bicciato and Ferrari 2022, 2301:236</a>; <a href="#ref-lieberman_aiden_comprehensive_2009" role="doc-biblioref">Lieberman-Aiden et al. 2009</a>)</span>, and <em>HiCExplorer</em> states it should be below 10% [ref]. The high fraction may be mitigated by enforcing a stricter <em>mapq</em> threshold for a valid Hi-C pair, as we also observe higher-than expected valid contacts. However, we continue without the current matrices.</p>
<div id="fig-explorer-qc" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-explorer-qc-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-qc" style="flex-basis: 33.3%;justify-content: flex-start;">
<div id="fig-explorer-read-orientation" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-read-orientation-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../steps/bwa/QC_all_samples/read_orientation.png" class="lightbox" data-gallery="fig-explorer-qc" title="Figure&nbsp;2&nbsp;(a): Read orientation"><img src="../steps/bwa/QC_all_samples/read_orientation.png" class="img-fluid figure-img" data-ref-parent="fig-explorer-qc"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-read-orientation-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) Read orientation
</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-qc" style="flex-basis: 33.3%;justify-content: flex-start;">
<div id="fig-explorer-unique-pairs" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-unique-pairs-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../steps/bwa/QC_all_samples/unmappable_and_non_unique.png" class="lightbox" data-gallery="fig-explorer-qc" title="Figure&nbsp;2&nbsp;(b): Unique pairs"><img src="../steps/bwa/QC_all_samples/unmappable_and_non_unique.png" class="img-fluid figure-img" data-ref-parent="fig-explorer-qc"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-unique-pairs-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) Unique pairs
</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-qc" style="flex-basis: 33.3%;justify-content: flex-start;">
<div id="fig-explorer-discarded-pairs" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-discarded-pairs-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../steps/bwa/QC_all_samples/pairs_discarded.png" class="lightbox" data-gallery="fig-explorer-qc" title="Figure&nbsp;2&nbsp;(c): Discarded pairs"><img src="../steps/bwa/QC_all_samples/pairs_discarded.png" class="img-fluid figure-img" data-ref-parent="fig-explorer-qc"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-discarded-pairs-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(c) Discarded pairs
</figcaption>
</figure>
</div>
</div>
</div>
<div class="quarto-layout-row">
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-qc" style="flex-basis: 33.3%;justify-content: flex-start;">
<div id="fig-explorer-pairs-sequenced" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-pairs-sequenced-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../steps/bwa/QC_all_samples/pairs_sequenced.png" class="lightbox" data-gallery="fig-explorer-qc" title="Figure&nbsp;2&nbsp;(d): Pairs sequenced"><img src="../steps/bwa/QC_all_samples/pairs_sequenced.png" class="img-fluid figure-img" data-ref-parent="fig-explorer-qc"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-pairs-sequenced-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(d) Pairs sequenced
</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-qc" style="flex-basis: 33.3%;justify-content: flex-start;">
<div id="fig-explorer-contact-distance" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-contact-distance-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../steps/bwa/QC_all_samples/distance.png" class="lightbox" data-gallery="fig-explorer-qc" title="Figure&nbsp;2&nbsp;(e): Contact distance"><img src="../steps/bwa/QC_all_samples/distance.png" class="img-fluid figure-img" data-ref-parent="fig-explorer-qc"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-contact-distance-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(e) Contact distance
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-explorer-qc-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2: Quality control of the mapped Hi-C reads using <em>HiCExplorer</em> <code>hicQC</code>. The figures should be moved to Supplementary/Appendix because they are ugly and un-alignable. But that is the fault of HiCExplorer, not me. <del>Or I should spend a couple of hours to plot them manually</del>.
</figcaption>
</figure>
</div>
</section>
<section id="correction" class="level3">
<h3 class="anchored" data-anchor-id="correction">Correction</h3>
<p>The correction diagnostic tool yielded a similar <em>mad</em> threshold within the range <span class="math inline">\([-3,-2]\)</span>. Even so, I followed the <em>HicExplorer</em> recommendation to set the lower threshold to at least -2 and the upper threshold to 5 in the pre-normalization filter. I argue that with a high number of valid contacts, it is safer to err on the side of caution and maybe filter out bad data.</p>
<div id="fig-explorer-pre-correction" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-explorer-pre-correction-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 5.9%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-pre-correction" style="flex-basis: 29.4%;justify-content: flex-start;">
<div id="fig-explorer-pre-correction-SRR6502335" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-pre-correction-SRR6502335-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/bwa/SRR6502335_diag_plot.png" class="lightbox" data-gallery="fig-explorer-pre-correction" title="Figure&nbsp;3&nbsp;(a): SRR6502335"><img src="../figures/bwa/SRR6502335_diag_plot.png" class="img-fluid figure-img" data-ref-parent="fig-explorer-pre-correction"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-pre-correction-SRR6502335-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) SRR6502335
</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-pre-correction" style="flex-basis: 29.4%;justify-content: flex-start;">
<div id="fig-explorer-pre-correction-SRR6502336" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-pre-correction-SRR6502336-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/bwa/SRR6502336_diag_plot.png" class="lightbox" data-gallery="fig-explorer-pre-correction" title="Figure&nbsp;3&nbsp;(b): SRR6502336"><img src="../figures/bwa/SRR6502336_diag_plot.png" class="img-fluid figure-img" data-ref-parent="fig-explorer-pre-correction"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-pre-correction-SRR6502336-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) SRR6502336
</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-pre-correction" style="flex-basis: 29.4%;justify-content: flex-start;">
<div id="fig-explorer-pre-correction-SRR6502337" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-pre-correction-SRR6502337-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/bwa/SRR6502337_diag_plot.png" class="lightbox" data-gallery="fig-explorer-pre-correction" title="Figure&nbsp;3&nbsp;(c): SRR6502337"><img src="../figures/bwa/SRR6502337_diag_plot.png" class="img-fluid figure-img" data-ref-parent="fig-explorer-pre-correction"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-pre-correction-SRR6502337-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(c) SRR6502337
</figcaption>
</figure>
</div>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 5.9%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
</div>
<div class="quarto-layout-row">
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 20.6%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-pre-correction" style="flex-basis: 29.4%;justify-content: flex-start;">
<div id="fig-explorer-pre-correction-SRR6502338" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-pre-correction-SRR6502338-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/bwa/SRR6502338_diag_plot.png" class="lightbox" data-gallery="fig-explorer-pre-correction" title="Figure&nbsp;3&nbsp;(d): SRR6502338"><img src="../figures/bwa/SRR6502338_diag_plot.png" class="img-fluid figure-img" data-ref-parent="fig-explorer-pre-correction"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-pre-correction-SRR6502338-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(d) SRR6502338
</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-pre-correction" style="flex-basis: 29.4%;justify-content: flex-start;">
<div id="fig-explorer-pre-correction-SRR6502339" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-pre-correction-SRR6502339-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/bwa/SRR6502339_diag_plot.png" class="lightbox" data-gallery="fig-explorer-pre-correction" title="Figure&nbsp;3&nbsp;(e): SRR6502339"><img src="../figures/bwa/SRR6502339_diag_plot.png" class="img-fluid figure-img" data-ref-parent="fig-explorer-pre-correction"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-pre-correction-SRR6502339-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(e) SRR6502339
</figcaption>
</figure>
</div>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 20.6%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-explorer-pre-correction-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3: Histograms of the number of counts per bin (bottom x-axis) and the modified z-score (top x-axis) from which the <em>mad</em> threshold is defined.
</figcaption>
</figure>
</div>
<blockquote class="blockquote">
<p>NB: when I say that a mapper performs poorly in finding Hi-C contacts, it is <code>hicBuildMatrix</code> that performs badly when reads are mapped with that mapper.</p>
</blockquote>
<p>To compare these mappings with others, the QC results is an easy way. Therefore, the reads were mapped with <em>bowtie2</em> in both end-to-end- and local-mode followed by <code>hiCBuildMatrix</code>, and the QC from each method was plotted next to each other (<a href="#fig-explorer-all-3-qc" class="quarto-xref">Figure&nbsp;4</a>). Interestingly, <em>bowtie2</em> was much more computer-intensive in both modes, perhaps because of the <code>--very-sensitive</code> option. In any case, the QC reveals a major difference in the total number of reads that are determined to be valid Hi-C contacts by <code>hicBuildMatrix</code>. As expected, <em>end-to-end-bowtie2</em> performs worse at locating Hi-C contacts than the other methods [ref row1], finding a very low amount of mappable, unique pairs passing the quality threshold. In contrast, <em>local-bowtie2</em> performs similarly to <em>bwa</em> in finding mappable, unique, high-quality pairs, but calls only approximately half the number of valid Hi-C contacts (&gt;20%), resulting in a fraction of valid Hi-C pairs that hits the expectation from <em>HicExplorer</em> docs [ref row3]. With <em>bwa</em>, the reads were discarded either due to low mapping quality or non-unique mates, whereas with <em>local-bowtie2</em>, the reads were almost exclusively filtered out due to low mapping quality. This must be a result of how the mappers assign mapping quality, and I believe <em>local-bowtie2</em> looks suspiciously selective in finding unique but low quality alignments. <em>end-to-end-bowtie</em> almost exclusively filters out read-pairs where one mate is unmapped, which is expected when the majority of reads are unmapped.</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/01_hicexplorer.ipynb" data-notebook-title="HiCExplorer using macaque data" data-notebook-cellid="cell-fig-explorer-all-3-qc">
<div id="cell-fig-explorer-all-3-qc" class="cell" data-execution_count="1">
<div class="cell-output cell-output-display">
<div id="fig-explorer-all-3-qc" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-explorer-all-3-qc-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-01_hicexplorer-fig-explorer-all-3-qc-output-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-12" title="Figure&nbsp;4: Comparison of HiCExplorer QC plots for all samples using different alignment tools."><img src="index_files/figure-html/..-notebooks-01_hicexplorer-fig-explorer-all-3-qc-output-1.png" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-explorer-all-3-qc-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;4: Comparison of HiCExplorer QC plots for all samples using different alignment tools.
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<p>As discussed, the five samples were pooled with <code>hicSumMatrices</code>, and the non-standard contigs (unplaced scaffolds) were filtered out, and the different resolutions were created (<code>hicMergeMatrixBins</code>). <em>HiCExplorer</em> also comes with a normalization function prior to correcting the matrix, which should be applied if different samples should have comparable bin counts. It has no effect when having only one matrix. Nevertheless, the pooled matrix was normalized and then corrected compared in <a href="#fig-explorer-pooled-norm-normcorr" class="quarto-xref">Figure&nbsp;5</a>.</p>
<div id="fig-explorer-pooled-norm-normcorr" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-explorer-pooled-norm-normcorr-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-pooled-norm-normcorr" style="flex-basis: 50.0%;justify-content: flex-start;">
<div id="fig-explorer-pooled-chrX-norm" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-pooled-chrX-norm-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/bowtie2/local/filter_pooled_50kb_chrX.png" class="lightbox" data-gallery="fig-explorer-pooled-norm-normcorr" title="Figure&nbsp;5&nbsp;(a): Normalized matrix chrX"><img src="../figures/bowtie2/local/filter_pooled_50kb_chrX.png" class="img-fluid figure-img" data-ref-parent="fig-explorer-pooled-norm-normcorr"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-pooled-chrX-norm-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) Normalized matrix chrX
</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-pooled-norm-normcorr" style="flex-basis: 50.0%;justify-content: flex-start;">
<div id="fig-explorer-pooled-chrX-normcorr" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-pooled-chrX-normcorr-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/bowtie2/local/normalized/normsm_filter_pooled_100kb_corrected_chrX-full.png" class="lightbox" data-gallery="fig-explorer-pooled-norm-normcorr" title="Figure&nbsp;5&nbsp;(b): Normalized and corrected chrX"><img src="../figures/bowtie2/local/normalized/normsm_filter_pooled_100kb_corrected_chrX-full.png" class="img-fluid figure-img" data-ref-parent="fig-explorer-pooled-norm-normcorr"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-pooled-chrX-normcorr-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) Normalized and corrected chrX
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-explorer-pooled-norm-normcorr-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5: A comparison of interaction matrices before/after iterative correction (<em>HiCExplorer</em>).
</figcaption>
</figure>
</div>
<p>It is now obvious why we have to correct the matrix. The uncorrected (<a href="#fig-explorer-pooled-chrX-norm" class="quarto-xref">Figure&nbsp;5 (a)</a>) has no signal apart from the diagonal. Even though some bins have been filtered out, the expected <em>plaid</em> pattern of a contact matrix is visible along the diagonal after the correction (<a href="#fig-explorer-pooled-chrX-normcorr" class="quarto-xref">Figure&nbsp;5 (b)</a>), leaving evidence for chromatin structure, especially in the first 50 million bases of the chromosome. There is a wide region of empty values at the place of the centromere.</p>
</section>
<section id="eigenvectors" class="level3">
<h3 class="anchored" data-anchor-id="eigenvectors">Eigenvectors</h3>
<p>The PCA performed by <code>hicPCA</code> on the pooled samples at both 50kb and 100kb resolution yielded the first 3 principal components. For PC1 on both resolutions (<a href="#fig-explorer-pc1-50kb" class="quarto-xref">Figure&nbsp;6 (a)</a>, <a href="#fig-explorer-pc1-100kb" class="quarto-xref">Figure&nbsp;6 (d)</a>) we observe only a single sign change which occurs at around 60 Mbp, the region of the centromere. It means the PCA has captured more variance between the chromosome arms than within them, making it uninformative about chromatin compartments. Upon visual inspection, it is clear that neither of the PC graphs capture the pattern of the interaction matrix by its change of sign. It seems the PCs capture variance from a bias that varies slowly and predictably along the chromosome. The first PC that is supposed to capture the compartments very suspiciously changes sign at the region of the centromere, a classic problem that could be solved by restricting the values from which the PC is calculated along the chromosome. Unimpressed, I rationalize that the option <code>--extra-track</code> to provide a gene track or histone coverage should not affect this result much. It should be provided as a phasing track to orient the eigenvector to positively correlate with gene density or histone marks, and could possibly muddle the compartments if not included. At this point, I stoppped using <em>HiCExplorer</em>, as I assessed that a more flexible tool was needed.</p>
<div id="fig-explorer-pca" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-explorer-pca-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-pca" style="flex-basis: 33.3%;justify-content: flex-start;">
<div id="fig-explorer-pc1-50kb" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-pc1-50kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/bowtie2/local/normalized/pc1_50kb_corrected_chrX.png" class="lightbox" data-gallery="fig-explorer-pca" title="Figure&nbsp;6&nbsp;(a): "><img src="../figures/bowtie2/local/normalized/pc1_50kb_corrected_chrX.png" id="fig-explorer-pc1-50kb" class="img-fluid figure-img" data-ref-parent="fig-explorer-pca"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-pc1-50kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a)
</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-pca" style="flex-basis: 33.3%;justify-content: flex-start;">
<div id="fig-explorer-pc2-50kb" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-pc2-50kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/bowtie2/local/normalized/pc2_50kb_corrected_chrX.png" class="lightbox" data-gallery="fig-explorer-pca" title="Figure&nbsp;6&nbsp;(b): "><img src="../figures/bowtie2/local/normalized/pc2_50kb_corrected_chrX.png" id="fig-explorer-pc2-50kb" class="img-fluid figure-img" data-ref-parent="fig-explorer-pca"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-pc2-50kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b)
</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-pca" style="flex-basis: 33.3%;justify-content: flex-start;">
<div id="fig-explorer-pc2-50kb" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-pc2-50kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/bowtie2/local/normalized/pc3_50kb_corrected_chrX.png" class="lightbox" data-gallery="fig-explorer-pca" title="Figure&nbsp;6&nbsp;(c): "><img src="../figures/bowtie2/local/normalized/pc3_50kb_corrected_chrX.png" id="fig-explorer-pc2-50kb" class="img-fluid figure-img" data-ref-parent="fig-explorer-pca"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-pc2-50kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(c)
</figcaption>
</figure>
</div>
</div>
</div>
<div class="quarto-layout-row">
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-pca" style="flex-basis: 33.3%;justify-content: flex-start;">
<div id="fig-explorer-pc1-100kb" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-pc1-100kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/bowtie2/local/normalized/pc1_100kb_corrected_chrX.png" class="lightbox" data-gallery="fig-explorer-pca" title="Figure&nbsp;6&nbsp;(d): "><img src="../figures/bowtie2/local/normalized/pc1_100kb_corrected_chrX.png" id="fig-explorer-pc1-100kb" class="img-fluid figure-img" data-ref-parent="fig-explorer-pca"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-pc1-100kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(d)
</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-pca" style="flex-basis: 33.3%;justify-content: flex-start;">
<div id="fig-explorer-pc2-100kb" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-pc2-100kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/bowtie2/local/normalized/pc2_100kb_corrected_chrX.png" class="lightbox" data-gallery="fig-explorer-pca" title="Figure&nbsp;6&nbsp;(e): "><img src="../figures/bowtie2/local/normalized/pc2_100kb_corrected_chrX.png" id="fig-explorer-pc2-100kb" class="img-fluid figure-img" data-ref-parent="fig-explorer-pca"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-pc2-100kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(e)
</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-pca" style="flex-basis: 33.3%;justify-content: flex-start;">
<div id="fig-explorer-pc3-100kb" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-pc3-100kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/bowtie2/local/normalized/pc3_100kb_corrected_chrX.png" class="lightbox" data-gallery="fig-explorer-pca" title="Figure&nbsp;6&nbsp;(f): "><img src="../figures/bowtie2/local/normalized/pc3_100kb_corrected_chrX.png" id="fig-explorer-pc3-100kb" class="img-fluid figure-img" data-ref-parent="fig-explorer-pca"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-pc3-100kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(f)
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-explorer-pca-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6: Corrected interaction matrix for chromosome X along with PC1, 2, or 3, respectively. a-c: 50kb resolution, d-f: 100kb resolution. <em>HiCExplorer</em>.
</figcaption>
</figure>
</div>
</section>
</section>
<section id="open2c-ecosystem" class="level2">
<h2 class="anchored" data-anchor-id="open2c-ecosystem">Open2c ecosystem</h2>
<section id="quality-control-1" class="level3">
<h3 class="anchored" data-anchor-id="quality-control-1">Quality Control</h3>
<section id="initial-run---walks-policy-mask" class="level4">
<h4 class="anchored" data-anchor-id="initial-run---walks-policy-mask">Initial run (<code>--walks-policy mask</code>)</h4>
<p>Comparing the multiQC report for each of the cell sources show similar distributions of <em>unmapped</em> (both sides unmapped), <em>one-sided</em> (one side mapped), <em>two-sided</em> (both sides mapped), and <em>duplicated</em> (w.r.t. total mapped) reads. The percentage of <em>cis</em> pairs w.r.t. mapped pairs is around 70% for all samples. [ref supptbl-qc-all-samples-mask]. The valid pairs also show similar distributions of pair types divided into 10 categories. The <span class="math inline">\(P(s)\)</span> curve looks similar as well, peaking between 270 bp and 320 bp separation (ref suppfig-multiqc-ps-curve). The QC does not show any information about mapping quality of the reads. Note that the <span class="math inline">\(P(s)\)</span> ccurve arise from pre-filtered pairs, and should be compcared with the <span class="math inline">\(P(s)\)</span> of the cooler after filtering.</p>
<p>[ref if there is time, include a histogram of mapq values for pairs]</p>
</section>
<section id="recommended---walks-policy-5unique" class="level4">
<h4 class="anchored" data-anchor-id="recommended---walks-policy-5unique">Recommended (<code>--walks-policy 5unique</code>)</h4>
<p>Parsing alignments with the recommended walks-policy aproximately halves the percentage of <em>unmapped</em> reads, and <em>one-</em> and <em>two-sided</em> reads as well <em>duplicated</em> reads are slightly increased. Overall number of unique pairs are increased with more than 20% increase. The percentage of <em>cis</em> pairs are only decreased by a percentage point at most [ref supptbl-qc-all-samples-5unique].</p>
</section>
</section>
<section id="correction-1" class="level3">
<h3 class="anchored" data-anchor-id="correction-1">Correction</h3>
<p>Matrix balancing did not show major improvement in the plaid pattern, as it was already pretty good. It does, however, filter out bins that are deemed too low-count to be informative, for example peri-centromeric regions. The matrix was expected to be smoother after balancing (for chromosomes), as regions along a chromosome should only vary slowly in contact frequency with other regions [ref].</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/05_rec_compartments.ipynb" data-notebook-title="Compartments Analysis w. recommended parameters" data-notebook-cellid="cell-fig-rs-chrx-raw-balanced-cgi">
<div id="cell-fig-rs-chrx-raw-balanced-cgi" class="cell" data-execution_count="100">
<div class="cell-output cell-output-display">
<div id="fig-rs-chrx-raw-balanced-cgi" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-rs-chrx-raw-balanced-cgi-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-05_rec_compartments-fig-rs-chrx-raw-balanced-cgi-output-2.png" class="lightbox" data-gallery="quarto-lightbox-gallery-21" title="Figure&nbsp;7: Raw, balanced, and interpolated chrX interaction matrix in 500kb resolution. The interpolation is done to make the matrix more visually appealing, but it is not necessary for the analysis."><img src="index_files/figure-html/..-notebooks-05_rec_compartments-fig-rs-chrx-raw-balanced-cgi-output-2.png" width="593" height="164" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-rs-chrx-raw-balanced-cgi-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7: Raw, balanced, and interpolated chrX interaction matrix in 500kb resolution. The interpolation is done to make the matrix more visually appealing, but it is not necessary for the analysis.
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/05_rec_compartments.ipynb" data-notebook-title="Compartments Analysis w. recommended parameters" data-notebook-cellid="cell-fig-rs-chrx-raw-balanced-cgi-subset">
<div id="cell-fig-rs-chrx-raw-balanced-cgi-subset" class="cell" data-execution_count="102">
<div class="cell-output cell-output-display">
<div id="fig-rs-chrx-raw-balanced-cgi-subset" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-rs-chrx-raw-balanced-cgi-subset-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-05_rec_compartments-fig-rs-chrx-raw-balanced-cgi-subset-output-2.png" class="lightbox" data-gallery="quarto-lightbox-gallery-22" title="Figure&nbsp;8: Raw, balanced, and interpolated chrX interaction matrix in 50kb resolution. The interpolation is done to make the matrix more visually appealing, but it is not necessary for the analysis."><img src="index_files/figure-html/..-notebooks-05_rec_compartments-fig-rs-chrx-raw-balanced-cgi-subset-output-2.png" width="593" height="167" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-rs-chrx-raw-balanced-cgi-subset-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;8: Raw, balanced, and interpolated chrX interaction matrix in 50kb resolution. The interpolation is done to make the matrix more visually appealing, but it is not necessary for the analysis.
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<p>The coarsegrained and interpolated matrix is useful to make a good-looking interaction matrix, but is not that useful for analysis purposes. It might get easier to visually inspect the matrix, but it is not clear how well the interpolated matrix reflects the structure of the chromatin. The regions that are coarsgrained are small zero- or low-count bins which are averaged, effectively reducing the resolution of those regions until the count is sufficient. They get more frequent the longer genomic distance (the further we travel from the diagonal), and effectively enables us to get some intuition about the interactions. The coarsegrain, however, does not interpolate the <code>NaN</code>s created when filtering out whole bins in the balancing step (horisontal and vertical lines in <a href="#fig-rs-chrx-raw-balanced-cgi" class="quarto-xref">Figure&nbsp;7</a> and <a href="#fig-rs-chrx-raw-balanced-cgi-subset" class="quarto-xref">Figure&nbsp;8</a>; middle). This is done in a subsequent step by linearly interpolating the <code>NaN</code>s. Examining the interpolated matrix on full chrX (<a href="#fig-rs-chrx-raw-balanced-cgi" class="quarto-xref">Figure&nbsp;7</a>; right) gives the impression that the pericentromeric (at ~60 Mbp) region harbours a <em>very</em> strong compartment, but that is clearly an artefact of the interpolation on the very large empty region of the centromere, where the diagonal is somehow extended in a square. On the thinner lines, the interpolation seem to be more smooth, and barely noticable on the diagonal.</p>
<section id="nan-histograms" class="level4">
<h4 class="anchored" data-anchor-id="nan-histograms"><code>NaN</code> histograms</h4>
<p>As expected, most of the low quality bins are located on the edges of the chromosome arms, especially the region around the centromere [ref litterature], as they contain many repetitive sequences. The low-quality bins are filtered out by the balancing algorithm, those bins are <code>NaN</code> in the Hi-C matrix. The median position of the <code>NaN</code> values (<a href="#fig-e1_nan_hist" class="quarto-xref">Figure&nbsp;9</a>) ranges between <span class="math inline">\(58\)</span> and <span class="math inline">\(63.5\)</span>, which is within the estimate of the centromeric region of <em>rhemac10</em>.</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/05_rec_compartments.ipynb" data-notebook-title="Compartments Analysis w. recommended parameters" data-notebook-cellid="cell-fig-e1_nan_hist">
<div id="cell-fig-e1_nan_hist" class="cell" data-execution_count="108">
<div class="cell-output cell-output-display">
<div id="fig-e1_nan_hist" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-e1_nan_hist-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-05_rec_compartments-fig-e1_nan_hist-output-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-23" title="Figure&nbsp;9: Histogram of NaN values in the E1 eigenvector for each cell type. Median position is marked with a red dashed line."><img src="index_files/figure-html/..-notebooks-05_rec_compartments-fig-e1_nan_hist-output-1.png" width="595" height="176" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-e1_nan_hist-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;9: Histogram of NaN values in the E1 eigenvector for each cell type. Median position is marked with a red dashed line.
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<p>The fact that the medians lie within the centromeric region on all cell sources shows both that the majority of the bad bins are in the (peri)centromeric region <em>and</em> there are approximately equally many on each side.</p>
</section>
</section>
<section id="compartments-eigenvectors" class="level3">
<h3 class="anchored" data-anchor-id="compartments-eigenvectors">Compartments (Eigenvectors)</h3>
<p>The three viewframes (<em>Full</em>, <em>Arms</em>, <em>10Mb</em>) for the calculation of the eigenvectors captured different variability in the data (<a href="#fig-e1-matrix-500kb-full-arms-10mb-round_spermatid" class="quarto-xref">Figure&nbsp;10 (a)</a>), and as expected, the inferred compartments (colored red on the E1 tracks) are more abundant and smaller with smaller viewframes. To determine how well each of the E1 tracks capture the pattern in the interaction matrix, we can overlay the matrix with the E1 sign-change and visually determine if the squares reflect the E1 sign change (<a href="#fig-e1-matrix-500kb-full-arms-10mb-round_spermatid" class="quarto-xref">Figure&nbsp;10 (a)</a>).</p>
<div id="fig-e1-matrix-full-arms-10mb-round-spermatid" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-e1-matrix-full-arms-10mb-round-spermatid-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-embed-nb-cell quarto-layout-cell-subref quarto-layout-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/05_rec_compartments.ipynb" data-notebook-title="Compartments Analysis w. recommended parameters" data-notebook-cellid="cell-fig-e1-matrix-500kb-full-arms-10mb-round_spermatid" data-ref-parent="fig-e1-matrix-full-arms-10mb-round-spermatid" style="flex-basis: 48.4%;justify-content: flex-start;">
<div id="cell-fig-e1-matrix-500kb-full-arms-10mb-round_spermatid" class="cell quarto-layout-cell-subref" data-execution_count="89" data-ref-parent="fig-e1-matrix-full-arms-10mb-round-spermatid">
<div class="cell-output cell-output-display quarto-layout-cell-subref" data-ref-parent="fig-e1-matrix-full-arms-10mb-round-spermatid">
<div id="fig-e1-matrix-500kb-full-arms-10mb-round_spermatid" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-e1-matrix-500kb-full-arms-10mb-round_spermatid-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-05_rec_compartments-fig-e1-matrix-500kb-full-arms-10mb-round_spermatid-output-1.png" class="lightbox" data-gallery="fig-e1-matrix-full-arms-10mb-round-spermatid" title="Figure&nbsp;10&nbsp;(a): E1 eigenvector values for merged round spermatid samples at 500kb resolution, as well as the interaction matrix. E1 was restricted to either Full-chromosome (top), Chromosome-arms (middle), or 10Mb windows (bottom)."><img src="index_files/figure-html/..-notebooks-05_rec_compartments-fig-e1-matrix-500kb-full-arms-10mb-round_spermatid-output-1.png" data-ref-parent="fig-e1-matrix-full-arms-10mb-round-spermatid" width="316" height="296" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-e1-matrix-500kb-full-arms-10mb-round_spermatid-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) E1 eigenvector values for merged round spermatid samples at 500kb resolution, as well as the interaction matrix. E1 was restricted to either Full-chromosome (top), Chromosome-arms (middle), or 10Mb windows (bottom).
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 3.2%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="quarto-embed-nb-cell quarto-layout-cell-subref quarto-layout-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/05_rec_compartments.ipynb" data-notebook-title="Compartments Analysis w. recommended parameters" data-notebook-cellid="cell-fig-e1-matrix-100kb-full-arms-10mb-round_spermatid" data-ref-parent="fig-e1-matrix-full-arms-10mb-round-spermatid" style="flex-basis: 48.4%;justify-content: flex-start;">
<div id="cell-fig-e1-matrix-100kb-full-arms-10mb-round_spermatid" class="cell quarto-layout-cell-subref" data-execution_count="83" data-ref-parent="fig-e1-matrix-full-arms-10mb-round-spermatid">
<div class="cell-output cell-output-display quarto-layout-cell-subref" data-ref-parent="fig-e1-matrix-full-arms-10mb-round-spermatid">
<div id="fig-e1-matrix-100kb-full-arms-10mb-round_spermatid" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-e1-matrix-100kb-full-arms-10mb-round_spermatid-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-05_rec_compartments-fig-e1-matrix-100kb-full-arms-10mb-round_spermatid-output-1.png" class="lightbox" data-gallery="fig-e1-matrix-full-arms-10mb-round-spermatid" title="Figure&nbsp;10&nbsp;(b): E1 eigenvector values for merged round spermatid samples at 500kb resolution, as well as the interaction matrix. E1 was restricted to either Full-chromosome (top), Chromosome-arms (middle), or 10Mb windows (bottom)."><img src="index_files/figure-html/..-notebooks-05_rec_compartments-fig-e1-matrix-100kb-full-arms-10mb-round_spermatid-output-1.png" data-ref-parent="fig-e1-matrix-full-arms-10mb-round-spermatid" width="316" height="296" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-e1-matrix-100kb-full-arms-10mb-round_spermatid-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) E1 eigenvector values for merged round spermatid samples at 500kb resolution, as well as the interaction matrix. E1 was restricted to either Full-chromosome (top), Chromosome-arms (middle), or 10Mb windows (bottom).
</figcaption>
</figure>
</div>
</div>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-e1-matrix-full-arms-10mb-round-spermatid-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;10: Super caption, subcaptions should be moved here (from notebook). They now fit on the page, but it would be nice to make this as one plt.subplots with shared axis title etc. <em>Update:</em> The notebook is ready for making these plots (<code>07_various_plotting.ipynb</code>, with a new plotting function) should be same size as the following matrix plots.
</figcaption>
</figure>
</div>
<p>I argue that without more finescaled knowledge than the position of the centromeres, the arbitrary size of the 10 Mb windowed E1 can not fully be justified. Also, <span class="citation" data-cites="wang_reprogramming_2019">Wang et al. (<a href="#ref-wang_reprogramming_2019" role="doc-biblioref">2019</a>)</span> concludes that only the pachytene spermatocyte showed local interactions in that viewframe (what they refer to as <em>refined A/B-compartments</em> ), and all the other stages of spermatogenesis were consistent with the conventional A/B compartments. The reasonable thing to do is therefore to continue the analysis, focusing on the arms-restricted eigendecomposition. Nevertheless, we also keep <em>refined</em> compartments in the analysis.</p>
<p>Additionally, as I created coolers with two different sets of parsing parameters we will compare the resulting matrices and their compartments (<a href="#fig-rs100-recpe-pe" class="quarto-xref">Figure&nbsp;11</a>). As expected, we observe more empty bins in the Hi-C matrix, but otherwise, the interaction pattern is indestinguishable when going from the initial run (PE) to the recommended parameters (redPE). The effect on the E1 is more noticable, where the absolute magnitude of the E1 values is generally smaller. There is, however, a small region that changes sign (from A to B) on the 10Mb-windowed (‘refined’) E1 track (<a href="#fig-rs100-recpe-pe" class="quarto-xref">Figure&nbsp;11</a>;c+d, zoomed-in region). This region is surrounded by added empty bins, which could mean that too many low quality pairs in PE were introducing bias and swapped the sign of E1. It is supported by the fact that the sign change <em>only</em> occured in <em>refined</em> E1, and that the sign after filtering weak pairs (<span class="math inline">\(mapq &lt; 30\)</span>) is consistent with the <em>arms</em> view. It supports my previous postulate that it is better to use a viewframe where with explicit molecular meaning than one of an arbitrary window size. That said, the <code>mapq</code> threshold should really be determined taking both coverage and resolution into account. For our purposes, and with the <em>arms</em> view, the mapping- and parsing parameters do not seem to be too sensitive.</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/07_various_plotting.ipynb" data-notebook-title="Various Comparisons" data-notebook-cellid="cell-fig-rs100-recpe-pe">
<div class="cell" data-layout="[[-0.5,5,5,-0.5],[-0.5,5,5,-0.5]]" data-execution_count="42">
<div id="fig-rs100-recpe-pe" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-rs100-recpe-pe-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 4.6%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-rs100-recpe-pe" style="flex-basis: 45.5%;justify-content: flex-start;">
<div id="fig-rs100-recpe-pe-1" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-rs100-recpe-pe-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs100-recpe-pe-output-1.png" class="lightbox" data-gallery="fig-rs100-recpe-pe" title="Figure&nbsp;11&nbsp;(a): recPE RS 100kb full chrX"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs100-recpe-pe-output-1.png" data-ref-parent="fig-rs100-recpe-pe" width="247" height="195" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-rs100-recpe-pe-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) recPE RS 100kb full chrX
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-rs100-recpe-pe" style="flex-basis: 45.5%;justify-content: flex-start;">
<div id="fig-rs100-recpe-pe-2" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-rs100-recpe-pe-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs100-recpe-pe-output-2.png" class="lightbox" data-gallery="fig-rs100-recpe-pe" title="Figure&nbsp;11&nbsp;(b): PE RS 100kb full chrX"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs100-recpe-pe-output-2.png" data-ref-parent="fig-rs100-recpe-pe" width="247" height="195" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-rs100-recpe-pe-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) PE RS 100kb full chrX
</figcaption>
</figure>
</div>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 4.6%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
</div>
<div class="quarto-layout-row">
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 4.6%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-rs100-recpe-pe" style="flex-basis: 45.5%;justify-content: flex-start;">
<div id="fig-rs100-recpe-pe-3" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-rs100-recpe-pe-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs100-recpe-pe-output-3.png" class="lightbox" data-gallery="fig-rs100-recpe-pe" title="Figure&nbsp;11&nbsp;(c): recPE RS 100kb chrX:70Mb-78Mb"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs100-recpe-pe-output-3.png" data-ref-parent="fig-rs100-recpe-pe" width="247" height="201" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-rs100-recpe-pe-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(c) recPE RS 100kb chrX:70Mb-78Mb
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-rs100-recpe-pe" style="flex-basis: 45.5%;justify-content: flex-start;">
<div id="fig-rs100-recpe-pe-4" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-rs100-recpe-pe-4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs100-recpe-pe-output-4.png" class="lightbox" data-gallery="fig-rs100-recpe-pe" title="Figure&nbsp;11&nbsp;(d): PE RS 100kb chrX:70Mb-78Mb"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs100-recpe-pe-output-4.png" data-ref-parent="fig-rs100-recpe-pe" width="247" height="201" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-rs100-recpe-pe-4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(d) PE RS 100kb chrX:70Mb-78Mb
</figcaption>
</figure>
</div>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 4.6%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-rs100-recpe-pe-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;11: Round Spermatid (RS) at 100kb, comparing the impact of parsing parameters
</figcaption>
</figure>
</div>
</div>
</div>
<p>To emphasize the findings, the sets of A-compartments were compared between the two parsing runs, showing almost identical compartment calls. Additionally, the set difference was 8 bins between PE and recPE for round spermatid 100kb and 5 bins for fibroblast for <em>arms</em> viewframe (<a href="#fig-rs-fb-100-pe-recpe-intervals" class="quarto-xref">Figure&nbsp;12</a>; a+b, respectively). We observe a high number of differences around 76Mb for the refined compartments (10Mb) of round spermatid, which is consistent with the sign-flip of E1 values discussed earlier. Anything else would be surprising, as it is the same data, but visualized in a different way.</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/07_various_plotting.ipynb" data-notebook-title="Various Comparisons" data-notebook-cellid="cell-fig-rs-fb-100-pe-recpe-intervals">
<div class="cell" data-layout="[[-0.6,2.5,2.5,-0.6],[-0.6,2.5,2.5,-0.6]]" data-execution_count="103">
<div id="fig-rs-fb-100-pe-recpe-intervals" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-rs-fb-100-pe-recpe-intervals-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 9.7%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-rs-fb-100-pe-recpe-intervals" style="flex-basis: 40.3%;justify-content: flex-start;">
<div id="fig-rs-fb-100-pe-recpe-intervals-1" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-rs-fb-100-pe-recpe-intervals-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs-fb-100-pe-recpe-intervals-output-1.png" class="lightbox" data-gallery="fig-rs-fb-100-pe-recpe-intervals" title="Figure&nbsp;12&nbsp;(a): RS: arms"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs-fb-100-pe-recpe-intervals-output-1.png" data-ref-parent="fig-rs-fb-100-pe-recpe-intervals" width="249" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-rs-fb-100-pe-recpe-intervals-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) RS: arms
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-rs-fb-100-pe-recpe-intervals" style="flex-basis: 40.3%;justify-content: flex-start;">
<div id="fig-rs-fb-100-pe-recpe-intervals-2" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-rs-fb-100-pe-recpe-intervals-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs-fb-100-pe-recpe-intervals-output-2.png" class="lightbox" data-gallery="fig-rs-fb-100-pe-recpe-intervals" title="Figure&nbsp;12&nbsp;(b): Fib: arms"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs-fb-100-pe-recpe-intervals-output-2.png" data-ref-parent="fig-rs-fb-100-pe-recpe-intervals" width="249" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-rs-fb-100-pe-recpe-intervals-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) Fib: arms
</figcaption>
</figure>
</div>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 9.7%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
</div>
<div class="quarto-layout-row">
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 9.7%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-rs-fb-100-pe-recpe-intervals" style="flex-basis: 40.3%;justify-content: flex-start;">
<div id="fig-rs-fb-100-pe-recpe-intervals-3" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-rs-fb-100-pe-recpe-intervals-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs-fb-100-pe-recpe-intervals-output-3.png" class="lightbox" data-gallery="fig-rs-fb-100-pe-recpe-intervals" title="Figure&nbsp;12&nbsp;(c): RS: 10Mb"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs-fb-100-pe-recpe-intervals-output-3.png" data-ref-parent="fig-rs-fb-100-pe-recpe-intervals" width="249" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-rs-fb-100-pe-recpe-intervals-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(c) RS: 10Mb
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-rs-fb-100-pe-recpe-intervals" style="flex-basis: 40.3%;justify-content: flex-start;">
<div id="fig-rs-fb-100-pe-recpe-intervals-4" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-rs-fb-100-pe-recpe-intervals-4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs-fb-100-pe-recpe-intervals-output-4.png" class="lightbox" data-gallery="fig-rs-fb-100-pe-recpe-intervals" title="Figure&nbsp;12&nbsp;(d): Fib: 10Mb"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs-fb-100-pe-recpe-intervals-output-4.png" data-ref-parent="fig-rs-fb-100-pe-recpe-intervals" width="249" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-rs-fb-100-pe-recpe-intervals-4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(d) Fib: 10Mb
</figcaption>
</figure>
</div>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 9.7%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-rs-fb-100-pe-recpe-intervals-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;12: Round Spermatid (RS) and Fibroblast (Fb) at 100kb, comparing the impact of parsing parameters on A-compartment calling at different viewframes; <em>arms</em>, <em>10Mb</em>. PE: initial parse (masking complex walks); recPE: recommended parse (reporting the 5’most unique alignment of a complex walk).
</figcaption>
</figure>
</div>
</div>
</div>
<p>The observed difference between the sets can for our data be attributed to chance, but we cannot draw general conclusions about the parameters in general. I argue that the quality and size of the Hi-C library will influence sensitive to parsing parameters. In that case, the most flexible approach is still to follow the recommendations from <code>cooler</code> to report more pairs as valid contacts, and then create coolers with different <em>mapq</em> filters if issues are encountered.</p>
</section>
<section id="compartment-edges-transition-zones" class="level3">
<h3 class="anchored" data-anchor-id="compartment-edges-transition-zones">Compartment Edges (transition zones)</h3>
<p>We compare how the ECH90 regions fit when queried on top of the A-compartments and equivalently for the edges, for fibroblasts and round spermatids at 100kb resolution. When queried against the edges in stead, the the total set size is reduced to less than 50%. Interestingly, some of the intersections between A-compartments and ECH90 remain, and new ones appear as we move to the outside edge of the compartment (<a href="#fig-comps-edges-ech" class="quarto-xref">Figure&nbsp;13</a>). This indicates that most, but not all, of the intersection between ECH90 regions and the A-compartments are within 100kb of the compartment edge, and additional overlap is gained if we define a transition zone on the outside of the edge as well. To visualize this (outside) edge enrichment, we find the set difference of the ECH-intersection to compartments and edges, respectively (<a href="#fig-edge-enrichment" class="quarto-xref">Figure&nbsp;14</a>), thus removing all the ‘inside’ edges. We observe that in almost all of the of the regions of <span class="math inline">\(ECH \cap Comp\)</span> are accompanied by an edge also intersecting ECH (<span class="math inline">\(ECH \cap Edge\)</span>), localized where the <em>Diff</em> track aligns (within 100kb) with both <span class="math inline">\(CompInt\)</span> and <span class="math inline">\(EdgeInt\)</span>.</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/07_various_plotting.ipynb" data-notebook-title="Various Comparisons" data-notebook-cellid="cell-fig-comps-edges-ech">
<div class="cell" data-layout="[[-0.6,2.5,2.5,-0.6],[-0.6,2.5,2.5,-0.6]]">
<div id="fig-comps-edges-ech" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-comps-edges-ech-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 9.7%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-comps-edges-ech" style="flex-basis: 40.3%;justify-content: flex-start;">
<div id="fig-comps-edges-ech-1" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-comps-edges-ech-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-comps-edges-ech-output-1.png" class="lightbox" data-gallery="fig-comps-edges-ech" title="Figure&nbsp;13&nbsp;(a): Fibroblast A-compartments"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-comps-edges-ech-output-1.png" data-ref-parent="fig-comps-edges-ech" width="248" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-comps-edges-ech-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) Fibroblast A-compartments
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-comps-edges-ech" style="flex-basis: 40.3%;justify-content: flex-start;">
<div id="fig-comps-edges-ech-2" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-comps-edges-ech-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-comps-edges-ech-output-2.png" class="lightbox" data-gallery="fig-comps-edges-ech" title="Figure&nbsp;13&nbsp;(b): Round Spermatid A-compartments"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-comps-edges-ech-output-2.png" data-ref-parent="fig-comps-edges-ech" width="248" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-comps-edges-ech-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) Round Spermatid A-compartments
</figcaption>
</figure>
</div>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 9.7%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
</div>
<div class="quarto-layout-row">
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 9.7%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-comps-edges-ech" style="flex-basis: 40.3%;justify-content: flex-start;">
<div id="fig-comps-edges-ech-3" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-comps-edges-ech-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-comps-edges-ech-output-3.png" class="lightbox" data-gallery="fig-comps-edges-ech" title="Figure&nbsp;13&nbsp;(c): Fibroblast edges"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-comps-edges-ech-output-3.png" data-ref-parent="fig-comps-edges-ech" width="248" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-comps-edges-ech-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(c) Fibroblast edges
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-comps-edges-ech" style="flex-basis: 40.3%;justify-content: flex-start;">
<div id="fig-comps-edges-ech-4" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-comps-edges-ech-4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-comps-edges-ech-output-4.png" class="lightbox" data-gallery="fig-comps-edges-ech" title="Figure&nbsp;13&nbsp;(d): Round Spermatid edges"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-comps-edges-ech-output-4.png" data-ref-parent="fig-comps-edges-ech" width="248" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-comps-edges-ech-4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(d) Round Spermatid edges
</figcaption>
</figure>
</div>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 9.7%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-comps-edges-ech-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;13: Visual representation of the genomic intervals of ECH90, A-compartments (a+b), edges (c+d), and their intersections. Shown fibroblast (a+c) and round spermatid (b+d) at 100kb resolution and arms viewframe.
</figcaption>
</figure>
</div>
</div>
</div>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/07_various_plotting.ipynb" data-notebook-title="Various Comparisons" data-notebook-cellid="cell-fig-edge-enrichment">
<div id="fig-edge-enrichment" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-edge-enrichment-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-edge-enrichment" style="flex-basis: 50.0%;justify-content: flex-start;">
<div id="fig-edge-enrichment-1" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-edge-enrichment-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-edge-enrichment-output-1.png" class="lightbox" data-gallery="fig-edge-enrichment" title="Figure&nbsp;14&nbsp;(a): Round Spermatid"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-edge-enrichment-output-1.png" data-ref-parent="fig-edge-enrichment" width="248" height="98" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-edge-enrichment-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) Round Spermatid
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-edge-enrichment" style="flex-basis: 50.0%;justify-content: flex-start;">
<div id="fig-edge-enrichment-2" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-edge-enrichment-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-edge-enrichment-output-2.png" class="lightbox" data-gallery="fig-edge-enrichment" title="Figure&nbsp;14&nbsp;(b): Fibroblast"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-edge-enrichment-output-2.png" data-ref-parent="fig-edge-enrichment" width="248" height="98" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-edge-enrichment-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) Fibroblast
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-edge-enrichment-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;14: Visual representation of the enrichment of edges in the intersection of ECH90 and A-compartments. Shown round spermatid (a) and fibroblast (b) at 100kb resolution and arms viewframe. Note that the edge-regions are too small to be distinguished visually from the compartment on the graph, making it look like they overlap, even though the difference is reported.
</figcaption>
</figure>
</div>
</div>
<p>We apply both proximity test and Jaccard test, to see how well the results could occur by chance (<a href="#fig-proximity-jaccard-bar" class="quarto-xref">Figure&nbsp;15</a>). For completeness, the tests are included for all cell types, but we only use 100kb resolution arms viewframe. We observe that both fibroblast and round spermatid have <span class="math inline">\(p &lt; 0.05\)</span> for both tests, meaning the two cell type have both more intersection with ECH regions than expected by chance (Jaccard) <em>and</em> the non-overlapping intervals are more proximal to compartment edges than expected by chance (proximity test). I argue that a significant Jaccard statistic should be interpreted as a significant amount of overlap between the two sets, i.e.&nbsp;compartment edges and ECH90 regions, and the proximity test (when performed on the edges) gives us information about the potential of expanding or moving the transition window. That is, if the non-overlapping regions are <em>very</em> proximal, a larger (or shifted) window to only capture the 200kb region outside of the edge might be favourable.</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/06_rec_genomicintervals.ipynb" data-notebook-title="E1 vs. ECH (recommended params)" data-notebook-cellid="cell-fig-proximity-jaccard-bar">
<div id="cell-fig-proximity-jaccard-bar" class="cell" data-execution_count="59">
<div class="cell-output cell-output-display">
<div id="fig-proximity-jaccard-bar" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-proximity-jaccard-bar-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-06_rec_genomicintervals-fig-proximity-jaccard-bar-output-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-40" title="Figure&nbsp;15: Proximity and Jaccard index p-values for ECH90 regions on compartment edges for all cell types at 100kb resolution at arms view p=0.05 is marked as a red line."><img src="index_files/figure-html/..-notebooks-06_rec_genomicintervals-fig-proximity-jaccard-bar-output-1.png" width="298" height="197" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-proximity-jaccard-bar-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;15: Proximity and Jaccard index p-values for ECH90 regions on compartment edges for all cell types at 100kb resolution at arms view <span class="math inline">\(p=0.05\)</span> is marked as a red line.
</figcaption>
</figure>
</div>
</div>
</div>
</div>
</section>
</section>
<section id="testing-against-hybrid-incompatibility-in-baboons" class="level2">
<h2 class="anchored" data-anchor-id="testing-against-hybrid-incompatibility-in-baboons">Testing against hybrid incompatibility in baboons</h2>
<p>Initially, the compartment edges of round spermatid at 100kb resolution (RS100) were plotted against the lifted coordinates from a <em>P.</em> anubis-hamadryas hybrid population, where either all the sampled individuals have <em>Papio anubis</em> ancestry or 95% of the sampled individuals have <em>Papio hamadryas</em> ancestry. Their respective intersections were plotted undeneath. We expect less intersection for <em>hamadryas</em> than for <em>anubis</em> as the total set size is much smaller. The compartment edges and <em>Papio anubis</em>-derived regions(<a href="#fig-baboon-rs100-intersect" class="quarto-xref">Figure&nbsp;16</a>; b) seem to be highly enriched in the first 25 Mbp, and thus it has a high degree of intersection with the compartment edges. Interestingly, the ECH90 set is nearly empty in that region, making the finding outside the scope of this analysis, although it could be useful for determining the mechanism for selecting the <em>P.anubis</em> ancestral allele in the hybrid baboon population. The <em>P.hamadryas</em>-derived regions seem intersect the compartment edges more centered on the chromosome (<a href="#fig-baboon-rs100-intersect" class="quarto-xref">Figure&nbsp;16</a>; a). The proximity test ruled out that the non-intersecting parts of the respective regions were this proximal by chance. However, the Jaccard test revealed that the intersection between the RS100 and both Hi-<em>P.hama</em> and Hi-<em>P.anu</em> can be explained by chance alone .</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/07_various_plotting.ipynb" data-notebook-title="Various Comparisons" data-notebook-cellid="cell-fig-baboon-rs100-intersect">
<div id="fig-baboon-rs100-intersect" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-baboon-rs100-intersect-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-baboon-rs100-intersect" style="flex-basis: 50.0%;justify-content: flex-start;">
<div id="fig-baboon-rs100-intersect-1" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-baboon-rs100-intersect-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-baboon-rs100-intersect-output-1.png" class="lightbox" data-gallery="fig-baboon-rs100-intersect" title="Figure&nbsp;16&nbsp;(a): P. hamadryas"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-baboon-rs100-intersect-output-1.png" data-ref-parent="fig-baboon-rs100-intersect" width="248" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-baboon-rs100-intersect-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) <em>P. hamadryas</em>
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-baboon-rs100-intersect" style="flex-basis: 50.0%;justify-content: flex-start;">
<div id="fig-baboon-rs100-intersect-2" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-baboon-rs100-intersect-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-baboon-rs100-intersect-output-2.png" class="lightbox" data-gallery="fig-baboon-rs100-intersect" title="Figure&nbsp;16&nbsp;(b): P. anubis"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-baboon-rs100-intersect-output-2.png" data-ref-parent="fig-baboon-rs100-intersect" width="248" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-baboon-rs100-intersect-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) <em>P. anubis</em>
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-baboon-rs100-intersect-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;16: Comparing the A-compartments of round spermatid (RS) with regions in baboons from hybrid population where a) 95% of sampled individuals have <em>Papio hamadryas</em> ancestry or b) 100% of the samples have <em>Papio anubis</em> ancestry. The regions are extracted and lifted from panu4.0 reference to rhemac10, then collapsed to non-overlapping intervals with <code>genominterv</code>.
</figcaption>
</figure>
</div>
</div>
<div style="page-break-after: always;"></div>
</section>
</section>
<section id="discussion" class="level1">
<h1>Discussion</h1>
<p>Here is the discussion</p>
</section>
<section id="bibliography" class="level1 unnumbered">
<h1 class="unnumbered">Bibliography</h1>
<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-bicciato_hic_2022" class="csl-entry" role="listitem">
Bicciato, Silvio, and Francesco Ferrari, eds. 2022. <em>Hi-<span>C</span> <span>Data</span> <span>Analysis</span>: <span>Methods</span> and <span>Protocols</span></em>. Vol. 2301. Methods in <span>Molecular</span> <span>Biology</span>. New York, NY: Springer US. <a href="https://doi.org/10.1007/978-1-0716-1390-0">https://doi.org/10.1007/978-1-0716-1390-0</a>.
</div>
<div id="ref-imakaev_iterative_2012" class="csl-entry" role="listitem">
Imakaev, Maxim, Geoffrey Fudenberg, Rachel Patton McCord, Natalia Naumova, Anton Goloborodko, Bryan R Lajoie, Job Dekker, and Leonid A Mirny. 2012. <span>“Iterative Correction of <span>Hi</span>-<span>C</span> Data Reveals Hallmarks of Chromosome Organization.”</span> <em>Nature Methods</em> 9 (10): 999–1003. <a href="https://doi.org/10.1038/nmeth.2148">https://doi.org/10.1038/nmeth.2148</a>.
</div>
<div id="ref-jaenike_sex_2001" class="csl-entry" role="listitem">
Jaenike, John. 2001. <span>“Sex <span>Chromosome</span> <span>Meiotic</span> <span>Drive</span>.”</span> <em>Annual Review of Ecology and Systematics</em> 32 (1): 25–49. <a href="https://doi.org/10.1146/annurev.ecolsys.32.081501.113958">https://doi.org/10.1146/annurev.ecolsys.32.081501.113958</a>.
</div>
<div id="ref-lajoie_hitchhikers_2015" class="csl-entry" role="listitem">
Lajoie, Bryan R., Job Dekker, and Noam Kaplan. 2015. <span>“The <span>Hitchhiker</span>’s <span>Guide</span> to <span>Hi</span>-<span>C</span> <span>Analysis</span>: <span>Practical</span> Guidelines.”</span> <em>Methods (San Diego, Calif.)</em> 72 (January): 65–75. <a href="https://doi.org/10.1016/j.ymeth.2014.10.031">https://doi.org/10.1016/j.ymeth.2014.10.031</a>.
</div>
<div id="ref-lieberman_aiden_comprehensive_2009" class="csl-entry" role="listitem">
Lieberman-Aiden, Erez, Nynke L. Van Berkum, Louise Williams, Maxim Imakaev, Tobias Ragoczy, Agnes Telling, Ido Amit, et al. 2009. <span>“Comprehensive <span>Mapping</span> of <span>Long</span>-<span>Range</span> <span>Interactions</span> <span>Reveals</span> <span>Folding</span> <span>Principles</span> of the <span>Human</span> <span>Genome</span>.”</span> <em>Science</em> 326 (5950): 289–93. <a href="https://doi.org/10.1126/science.1181369">https://doi.org/10.1126/science.1181369</a>.
</div>
<div id="ref-munch_group_2024" class="csl-entry" role="listitem">
Munch, Kasper. 2024. <span>“Munch-Group.”</span> <a href="https://munch-group.org/research.html">https://munch-group.org/research.html</a>.
</div>
<div id="ref-open2c" class="csl-entry" role="listitem">
Open Chromosome Collective. 2024. <span>“Open <span>Chromosome</span> <span>Collective</span> (<span>Open2C</span>).”</span> Resource. <em>Open2C</em>. <a href="https://open2c.github.io/">https://open2c.github.io/</a>.
</div>
<div id="ref-skov_extraordinary_2023" class="csl-entry" role="listitem">
Skov, Laurits, Moisès Coll Macià, Elise Anne Lucotte, Maria Izabel Alves Cavassim, David Castellano, Mikkel Heide Schierup, and Kasper Munch. 2023. <span>“Extraordinary Selection on the Human <span>X</span> Chromosome Associated with Archaic Admixture.”</span> <em>Cell Genomics</em> 3 (3): 100274. <a href="https://doi.org/10.1016/j.xgen.2023.100274">https://doi.org/10.1016/j.xgen.2023.100274</a>.
</div>
<div id="ref-sorensen_genome_wide_2023" class="csl-entry" role="listitem">
Sørensen, Erik F., R. Alan Harris, Liye Zhang, Muthuswamy Raveendran, Lukas F. K. Kuderna, Jerilyn A. Walker, Jessica M. Storer, et al. 2023. <span>“Genome-Wide Coancestry Reveals Details of Ancient and Recent Male-Driven Reticulation in Baboons.”</span> <em>Science</em> 380 (6648): eabn8153. <a href="https://doi.org/10.1126/science.abn8153">https://doi.org/10.1126/science.abn8153</a>.
</div>
<div id="ref-wang_reprogramming_2019" class="csl-entry" role="listitem">
Wang, Yao, Hanben Wang, Yu Zhang, Zhenhai Du, Wei Si, Suixing Fan, Dongdong Qin, et al. 2019. <span>“Reprogramming of <span>Meiotic</span> <span>Chromatin</span> <span>Architecture</span> During <span>Spermatogenesis</span>.”</span> <em>Molecular Cell</em> 73 (3): 547–561.e6. <a href="https://doi.org/10.1016/j.molcel.2018.11.019">https://doi.org/10.1016/j.molcel.2018.11.019</a>.
</div>
</div>
</section>


<div id="quarto-appendix" class="default"><section id="footnotes" class="footnotes footnotes-end-of-document" role="doc-endnotes"><h2 class="anchored quarto-appendix-heading">Footnotes</h2>

<ol>
<li id="fn1"><p><span class="citation" data-cites="bicciato_hic_2022">Bicciato and Ferrari (<a href="#ref-bicciato_hic_2022" role="doc-biblioref">2022</a>)</span> mentions that <em>Drosophila</em> has been studied where this type of ligations are used to infer structure of sister-chromatid post-replication.<a href="#fnref1" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
</ol>
</section></div></main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<script>var lightboxQuarto = GLightbox({"closeEffect":"zoom","descPosition":"bottom","loop":false,"openEffect":"zoom","selector":".lightbox"});
(function() {
  let previousOnload = window.onload;
  window.onload = () => {
    if (previousOnload) {
      previousOnload();
    }
    lightboxQuarto.on('slide_before_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      const href = trigger.getAttribute('href');
      if (href !== null) {
        const imgEl = window.document.querySelector(`a[href="${href}"] img`);
        if (imgEl !== null) {
          const srcAttr = imgEl.getAttribute("src");
          if (srcAttr && srcAttr.startsWith("data:")) {
            slideConfig.href = srcAttr;
          }
        }
      } 
    });
  
    lightboxQuarto.on('slide_after_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(slideNode);
      }
    });
  
  };
  
})();
          </script>




</body></html>