<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.40">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2025-01-29">
<meta name="keywords" content="Hi-C, Chromatin Compartments, Selection on X, Selfish gene drive">

<title>Chromatin Compartments and Selection on X</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js" integrity="sha512-bLT0Qm9VnAYZDflyKcBaQ2gg0hSYNQrJ8RilYldYQ1FxQYoCLtUjuuRuZo+fjqhx/qtq/1itJ0C2ejDxltZVFg==" crossorigin="anonymous"></script><script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-549806ee2085284f45b00abea8c6df48.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-2ecafa609d29e6b5df2c2179b389d5c3.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script src="site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet">
<script async="" src="https://hypothes.is/embed.js"></script>
<script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" integrity="sha512-c3Nl8+7g4LMSTdrm621y7kf9v3SDPnhxLNhcjFJbKECVnmZHTdo+IRO05sNLTH/D3vA6u1X32ehoLC7WFVdheg==" crossorigin="anonymous"></script>

<script type="application/javascript">define('jquery', [],function() {return window.jQuery;})</script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<meta name="citation_title" content="Chromatin Compartments and Selection on X">
<meta name="citation_abstract" content="The X chromosome is uniquely exposed to selective pressures due to the lack of a second copy in the hemizygous sex, leaving no buffer against deleterious mutations, and giving unique inheritance patterns. Combined with its high density of essential genes related to reproduction and brain function, this suggests the presence of biological mechanisms that safeguard the integrity of the X chromosome.

In this study, the 3D chromatin architecture of the X chromosome in rhesus macaque (*Macaca mulata*) is investigated in the context of evolutionary pressures and genetic drivers. To ensure transparency and reproducibility, we adopt a comprehensive computational framework for publishing a version-controlled, fully reproducible analysis.

We compare two Hi-C analysis frameworks, *HiCExplorer* and *cooler/cooltools* (Open2C), on a subset, finding Open2C to be most flexible. The ICE method (Iterative Correction and Eigendecomposition) was used to infer conventional and refined A/B compartments for fibroblast and four stages of spermatogenesis.

We find 200 kbp transition-zones between A/B-compartments on the X chromosomes in both fibroblasts and round spermatids that align well with strong selective sweeps in humans (ECH-regions), but not with strong negative selection in baboons (*Papio* spp.). We find that most edges either overlap or are in significant proximity of each other when comparing regions under selection in human and baboons with A/B-compartments inferred Hi-C matrices at 100kb resolution and restricting eigendecomposition along the X chromosome to 10Mb windows. We discuss the biological meaning of these findings, where conserved chromatin features may help to retain non-advantageous alleles, hinting to the role of structural features aiding in genome evolution.
">
<meta name="citation_keywords" content="Hi-C,Chromatin Compartments,Selection on X,Selfish gene drive">
<meta name="citation_author" content="Søren Jørgensen">
<meta name="citation_author" content="Kasper Munch">
<meta name="citation_publication_date" content="2025-01-29">
<meta name="citation_cover_date" content="2025-01-29">
<meta name="citation_year" content="2025">
<meta name="citation_online_date" content="2025-01-29">
<meta name="citation_language" content="en">
<meta name="citation_reference" content="citation_title=Cooler: Scalable storage for Hi-C data and other genomically labeled arrays;,citation_abstract=Abstract Motivation Most existing coverage-based (epi)genomic datasets are one-dimensional, but newer technologies probing interactions (physical, genetic, etc.) produce quantitative maps with two-dimensional genomic coordinate systems. Storage and computational costs mount sharply with data resolution when such maps are stored in dense form. Hence, there is a pressing need to develop data storage strategies that handle the full range of useful resolutions in multidimensional genomic datasets by taking advantage of their sparse nature, while supporting efficient compression and providing fast random access to facilitate development of scalable algorithms for data analysis. Results We developed a file format called cooler, based on a sparse data model, that can support genomically labeled matrices at any resolution. It has the flexibility to accommodate various descriptions of the data axes (genomic coordinates, tracks and bin annotations), resolutions, data density patterns and metadata. Cooler is based on HDF5 and is supported by a Python library and command line suite to create, read, inspect and manipulate cooler data collections. The format has been adopted as a standard by the NIH 4D Nucleome Consortium. Availability and implementation Cooler is cross-platform, BSD-licensed and can be installed from the Python package index or the bioconda repository. The source code is maintained on Github at https://github.com/mirnylab/cooler. Supplementary information Supplementary data are available at Bioinformatics online.;,citation_author=Nezar Abdennur;,citation_author=Leonid A Mirny;,citation_editor=Jonathan Wren;,citation_publication_date=2020-01;,citation_cover_date=2020-01;,citation_year=2020;,citation_issue=1;,citation_doi=10.1093/bioinformatics/btz540;,citation_issn=1367-4803, 1367-4811;,citation_volume=36;,citation_language=en-US;,citation_journal_title=Bioinformatics;">
<meta name="citation_reference" content="citation_title=Accurate assembly of the olive baboon ( &amp;amp;amp;lt;i&amp;gt;Papio&amp;lt;/i&amp;gt; <i>Anubis</i> ) genome using long-read and Hi-C data;,citation_abstract=Abstract Background Baboons are a widely used nonhuman primate model for biomedical, evolutionary, and basic genetics research. Despite this importance, the genomic resources for baboons are limited. In particular, the current baboon reference genome Panu_3.0 is a highly fragmented, reference-guided (i.e., not fully de novo) assembly, and its poor quality inhibits our ability to conduct downstream genomic analyses. Findings Here we present a de novo genome assembly of the olive baboon (Papio anubis) that uses data from several recently developed single-molecule technologies. Our assembly, Panubis1.0, has an N50 contig size of $\sim$1.46 Mb (as opposed to 139 kb for Panu_3.0) and has single scaffolds that span each of the 20 autosomes and the X chromosome. Conclusions We highlight multiple lines of evidence (including Bionano Genomics data, pedigree linkage information, and linkage disequilibrium data) suggesting that there are several large assembly errors in Panu_3.0, which have been corrected in Panubis1.0.;,citation_author=Sanjit Singh Batra;,citation_author=Michal Levy-Sakin;,citation_author=Jacqueline Robinson;,citation_author=Joseph Guillory;,citation_author=Steffen Durinck;,citation_author=Tauras P Vilgalys;,citation_author=Pui-Yan Kwok;,citation_author=Laura A Cox;,citation_author=Somasekar Seshagiri;,citation_author=Yun S Song;,citation_author=Jeffrey D Wall;,citation_publication_date=2020-12;,citation_cover_date=2020-12;,citation_year=2020;,citation_issue=12;,citation_doi=10.1093/gigascience/giaa134;,citation_issn=2047-217X;,citation_volume=9;,citation_language=en-US;,citation_journal_title=GigaScience;">
<meta name="citation_reference" content="citation_title=Hi-C Data Analysis: Methods and Protocols;,citation_editor=Silvio Bicciato;,citation_editor=Francesco Ferrari;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_doi=10.1007/978-1-0716-1390-0;,citation_isbn=978-1-0716-1389-4 978-1-0716-1390-0;,citation_volume=2301;,citation_language=en-US;,citation_series_title=Methods in Molecular Biology;">
<meta name="citation_reference" content="citation_title=Genetic Villains: Killer Meiotic Drivers;,citation_author=María Angélica Bravo Núñez;,citation_author=Nicole L. Nuckolls;,citation_author=Sarah E. Zanders;,citation_publication_date=2018-06;,citation_cover_date=2018-06;,citation_year=2018;,citation_issue=6;,citation_doi=10.1016/j.tig.2018.02.003;,citation_issn=01689525;,citation_volume=34;,citation_language=en-US;,citation_journal_title=Trends in Genetics;">
<meta name="citation_reference" content="citation_title=dcHiC detects differential compartments across multiple Hi-C datasets;,citation_abstract=Abstract The compartmental organization of mammalian genomes and its changes play important roles in distinct biological processes. Here, we introduce dcHiC, which utilizes a multivariate distance measure to identify significant changes in compartmentalization among multiple contact maps. Evaluating dcHiC on four collections of bulk and single-cell contact maps from in vitro mouse neural differentiation (n=3), mouse hematopoiesis (n=10), human LCLs (n=20) and post-natal mouse brain development (n=3 stages), we show its effectiveness and sensitivity in detecting biologically relevant changes, including those orthogonally validated. dcHiC reported regions with dynamically regulated genes associated with cell identity, along with correlated changes in chromatin states, subcompartments, replication timing and lamin association. With its efficient implementation, dcHiC enables high-resolution compartment analysis as well as standalone browser visualization, differential interaction identification and time-series clustering. dcHiC is an essential addition to the Hi-C analysis toolbox for the ever-growing number of bulk and single-cell contact maps. Available at: https://github.com/ay-lab/dcHiC .;,citation_author=Abhijit Chakraborty;,citation_author=Jeffrey G. Wang;,citation_author=Ferhat Ay;,citation_publication_date=2022-11;,citation_cover_date=2022-11;,citation_year=2022;,citation_issue=1;,citation_doi=10.1038/s41467-022-34626-6;,citation_issn=2041-1723;,citation_volume=13;,citation_language=en-US;,citation_journal_title=Nature Communications;">
<meta name="citation_reference" content="citation_title=High resolution maps of chromatin reorganization through mouse meiosis reveal novel features of the 3D meiotic structure;,citation_abstract=Abstract When germ cells transition from the mitotic cycle into meiotic prophase I (MPI), chromosomes condense into an array of chromatin loops that are required to promote homolog pairing and genetic recombination. To identify the changes in chromosomal conformation, we isolated nuclei on a trajectory from spermatogonia to the end of MPI. At each stage along this trajectory, we built genomic interaction maps with the highest temporal and spatial resolution to date. The changes in chromatin folding coincided with a concurrent decline in mitotic cohesion and a rise in meiotic cohesin complexes. We found that the stereotypical large-scale A and B compartmentalization was lost during meiotic prophase I alongside the loss of topological associating domains (TADs). Still, local subcompartments were detected and maintained throughout meiosis. The enhanced Micro-C resolution revealed that, despite the loss of TADs, higher frequency contact sites between two loci were detectable during meiotic prophase I coinciding with CTCF bound sites. The pattern of interactions around these CTCF sites with their neighboring loci showed that CTCF sites were often anchoring the meiotic loops. Additionally, the localization of CTCF to the meiotic axes indicated that these anchors were at the base of loops. Strikingly, even in the face of the dramatic reconfiguration of interphase chromatin into a condensed loop-array, the interactions between regulatory elements remained well preserved. This establishes a potential mechanism for how the meiotic chromatin maintains active transcription within a highly structured genome. In summary, the high temporal and spatial resolution of these data revealed previously unappreciated aspects of mammalian meiotic chromatin organization.;,citation_author=Gang Cheng;,citation_author=Florencia Pratto;,citation_author=Kevin Brick;,citation_author=Xin Li;,citation_author=Benjamin Alleva;,citation_author=Mini Huang;,citation_author=Gabriel Lam;,citation_author=R. Daniel Camerini-Otero;,citation_publication_date=2024-03;,citation_cover_date=2024-03;,citation_year=2024;,citation_doi=10.1101/2024.03.25.586627;,citation_language=en-US;">
<meta name="citation_reference" content="citation_title=100 years of Haldane’s rule;,citation_abstract=Abstract Haldane’s rule is one of the “two rules of speciation”. It states that if one sex is “absent, rare or sterile” in a hybrid population, then that sex will be heterogametic. Since Haldane first made this observation, 100&nbsp;years have passed and still questions arise over how many independent examples exist and what the underlying causes of Haldane’s rule are. This review aims to examine research that has occurred over the last century. It seeks to do so by discussing possible causes of Haldane’s rule, as well as gaps in the research of these causes that could be readily addressed today. After 100&nbsp;years of research, it can be concluded that Haldane’s rule is a complicated one, and much current knowledge has been accrued by studying the model organisms of speciation. This has led to the primacy of dominance theory and faster-male theory as explanations for Haldane’s rule. However, some of the most interesting findings of the 21st century with regard to Haldane’s rule have involved investigating a wider range of taxa emphasizing the need to continue using comparative methods, including ever more taxa as new cases are discovered.;,citation_author=Finn Cowell;,citation_publication_date=2023-02;,citation_cover_date=2023-02;,citation_year=2023;,citation_issue=2;,citation_doi=10.1111/jeb.14112;,citation_issn=1010-061X, 1420-9101;,citation_volume=36;,citation_language=en-US;,citation_journal_title=Journal of Evolutionary Biology;">
<meta name="citation_reference" content="citation_title=Chromatin architecture reorganization during stem cell differentiation;,citation_abstract=Higher-order chromatin structure is emerging as an important regulator of gene expression. Although dynamic chromatin structures have been identified in the genome, the full scope of chromatin dynamics during mammalian development and lineage specification remains to be determined. By mapping genome-wide chromatin interactions in human embryonic stem (ES) cells and four human ES-cell-derived lineages, we uncover extensive chromatin reorganization during lineage specification. We observe that although self-associating chromatin domains are stable during differentiation, chromatin interactions both within and between domains change in a striking manner, altering 36% of active and inactive chromosomal compartments throughout the genome. By integrating chromatin interaction maps with haplotype-resolved epigenome and transcriptome data sets, we find widespread allelic bias in gene expression correlated with allele-biased chromatin states of linked promoters and distal enhancers. Our results therefore provide a global view of chromatin dynamics and a resource for studying long-range control of gene expression in distinct human cell lineages.;,citation_author=Jesse R. Dixon;,citation_author=Inkyung Jung;,citation_author=Siddarth Selvaraj;,citation_author=Yin Shen;,citation_author=Jessica E. Antosiewicz-Bourget;,citation_author=Ah Young Lee;,citation_author=Zhen Ye;,citation_author=Audrey Kim;,citation_author=Nisha Rajagopal;,citation_author=Wei Xie;,citation_author=Yarui Diao;,citation_author=Jing Liang;,citation_author=Huimin Zhao;,citation_author=Victor V. Lobanenkov;,citation_author=Joseph R. Ecker;,citation_author=James A. Thomson;,citation_author=Bing Ren;,citation_publication_date=2015-02;,citation_cover_date=2015-02;,citation_year=2015;,citation_issue=7539;,citation_doi=10.1038/nature14222;,citation_issn=1476-4687;,citation_volume=518;,citation_journal_title=Nature;">
<meta name="citation_reference" content="citation_title=Sex ratio and unisexual sterility in hybrid animals;,citation_author=J. B. S. Haldane;,citation_publication_date=1922-10;,citation_cover_date=1922-10;,citation_year=1922;,citation_issue=2;,citation_doi=10.1007/BF02983075;,citation_issn=0022-1333;,citation_volume=12;,citation_language=en-US;,citation_journal_title=Journal of Genetics;">
<meta name="citation_reference" content="citation_title=Chromatin alternates between A and B compartments at kilobase scale for subgenic organization;,citation_abstract=Abstract Nuclear compartments are prominent features of 3D chromatin organization, but sequencing depth limitations have impeded investigation at ultra fine-scale. CTCF loops are generally studied at a finer scale, but the impact of looping on proximal interactions remains enigmatic. Here, we critically examine nuclear compartments and CTCF loop-proximal interactions using a combination of in situ Hi-C at unparalleled depth, algorithm development, and biophysical modeling. Producing a large Hi-C map with 33 billion contacts in conjunction with an algorithm for performing principal component analysis on sparse, super massive matrices&nbsp;(POSSUMM), we resolve compartments to 500 bp. Our results demonstrate that essentially all active promoters and distal enhancers localize in the A compartment, even when flanking sequences do not. Furthermore, we find that the TSS and TTS of paused genes are often segregated into separate compartments. We then identify diffuse interactions that radiate from CTCF loop anchors, which correlate with strong enhancer-promoter interactions and proximal transcription. We also find that these diffuse interactions depend on CTCF’s RNA binding domains. In this work, we demonstrate features of fine-scale chromatin organization consistent with a revised model in which compartments are more precise than commonly thought while CTCF loops are more protracted.;,citation_author=Hannah L. Harris;,citation_author=Huiya Gu;,citation_author=Moshe Olshansky;,citation_author=Ailun Wang;,citation_author=Irene Farabella;,citation_author=Yossi Eliaz;,citation_author=Achyuth Kalluchi;,citation_author=Akshay Krishna;,citation_author=Mozes Jacobs;,citation_author=Gesine Cauer;,citation_author=Melanie Pham;,citation_author=Suhas S. P. Rao;,citation_author=Olga Dudchenko;,citation_author=Arina Omer;,citation_author=Kiana Mohajeri;,citation_author=Sungjae Kim;,citation_author=Michael H. Nichols;,citation_author=Eric S. Davis;,citation_author=Dimos Gkountaroulis;,citation_author=Devika Udupa;,citation_author=Aviva Presser Aiden;,citation_author=Victor G. Corces;,citation_author=Douglas H. Phanstiel;,citation_author=William Stafford Noble;,citation_author=Guy Nir;,citation_author=Michele Di Pierro;,citation_author=Jeong-Sun Seo;,citation_author=Michael E. Talkowski;,citation_author=Erez Lieberman Aiden;,citation_author=M. Jordan Rowley;,citation_publication_date=2023-06;,citation_cover_date=2023-06;,citation_year=2023;,citation_issue=1;,citation_doi=10.1038/s41467-023-38429-1;,citation_issn=2041-1723;,citation_volume=14;,citation_language=en-US;,citation_journal_title=Nature Communications;">
<meta name="citation_reference" content="citation_title=Iterative correction of Hi-C data reveals hallmarks of chromosome organization;,citation_author=Maxim Imakaev;,citation_author=Geoffrey Fudenberg;,citation_author=Rachel Patton McCord;,citation_author=Natalia Naumova;,citation_author=Anton Goloborodko;,citation_author=Bryan R Lajoie;,citation_author=Job Dekker;,citation_author=Leonid A Mirny;,citation_publication_date=2012-10;,citation_cover_date=2012-10;,citation_year=2012;,citation_issue=10;,citation_doi=10.1038/nmeth.2148;,citation_issn=1548-7091, 1548-7105;,citation_volume=9;,citation_language=en-US;,citation_journal_title=Nature Methods;">
<meta name="citation_reference" content="citation_title=Sex Chromosome Meiotic Drive;,citation_abstract=▪ Abstract Sex chromosome drive refers to the unequal transmission of X and Y chromosomes from individuals of the heterogametic sex, resulting in biased sex ratios among progeny and within populations. The presence of driving sex chromosomes can reduce mean fitness within a population, bring about intragenomic conflict between the X chromosome, the Y, and the autosomes, and alter the intensity or mode of sexual selection within species. Sex chromosome drive, or its genetic equivalent, is known in plants, mammals, and flies. Many species harboring driving X chromosomes have evolved Y-linked and autosomal suppressors of drive. If a drive polymorphism is not stable, then driving chromosomes may spread to fixation and cause the extinction of a species. Certain characteristics of species, such as population density and female mating rate, may affect the probability of fixation of driving chromosomes. Thus, sex chromosome drive could be an agent of species-level selection.;,citation_author=John Jaenike;,citation_publication_date=2001-11;,citation_cover_date=2001-11;,citation_year=2001;,citation_issue=1;,citation_doi=10.1146/annurev.ecolsys.32.081501.113958;,citation_issn=0066-4162;,citation_volume=32;,citation_language=en-US;,citation_journal_title=Annual Review of Ecology and Systematics;">
<meta name="citation_reference" content="citation_title=Do sex-linked male meiotic drivers contribute to intrinsic hybrid incompatibilities? Recent empirical studies from flies and rodents;,citation_author=Jun Kitano;,citation_author=Kohta Yoshida;,citation_publication_date=2023-08;,citation_cover_date=2023-08;,citation_year=2023;,citation_doi=10.1016/j.gde.2023.102068;,citation_issn=0959437X;,citation_volume=81;,citation_language=en-US;,citation_journal_title=Current Opinion in Genetics &amp;amp;amp; Development;">
<meta name="citation_reference" content="citation_title=The Hitchhiker’s Guide to Hi-C Analysis: Practical guidelines;,citation_abstract=Over the last decade, development and application of a set of molecular genomic approaches based on the chromosome conformation capture method (3C), combined with increasingly powerful imaging approaches, have enabled high resolution and genome-wide analysis of the spatial organization of chromosomes. The aim of this paper is to provide guidelines for analyzing and interpreting data obtained with genome-wide 3C methods such as Hi-C and 3C-seq that rely on deep sequencing to detect and quantify pairwise chromatin interactions genome-wide.;,citation_author=Bryan R. Lajoie;,citation_author=Job Dekker;,citation_author=Noam Kaplan;,citation_publication_date=2015-01;,citation_cover_date=2015-01;,citation_year=2015;,citation_doi=10.1016/j.ymeth.2014.10.031;,citation_issn=1046-2023;,citation_pmid=25448293;,citation_volume=72;,citation_journal_title=Methods (San Diego, Calif.);">
<meta name="citation_reference" content="citation_title=Comprehensive Mapping of Long-Range Interactions Reveals Folding Principles of the Human Genome;,citation_abstract=Chromosomal Mapping The conformation of the genome in the nucleus and contacts between both proximal and distal loci influence gene expression. In order to map genomic contacts, Lieberman-Aiden et al. (p. 289 , see the cover) developed a technique to allow the detection of all interactions between genomic loci in the eukaryotic nucleus followed by deep sequencing. This technology was used to map the organization of the human genome and to examine the spatial proximity of chromosomal loci at one megabase resolution. The map suggests that the genome is partitioned into two spatial compartments that are related to local chromatin state and whose remodeling correlates with changes in the chromatin state. , Chromosomes are organized in a fractal knot-free conformation that is densely packed while easily folded and unfolded. , We describe Hi-C, a method that probes the three-dimensional architecture of whole genomes by coupling proximity-based ligation with massively parallel sequencing. We constructed spatial proximity maps of the human genome with Hi-C at a resolution of 1 megabase. These maps confirm the presence of chromosome territories and the spatial proximity of small, gene-rich chromosomes. We identified an additional level of genome organization that is characterized by the spatial segregation of open and closed chromatin to form two genome-wide compartments. At the megabase scale, the chromatin conformation is consistent with a fractal globule, a knot-free, polymer conformation that enables maximally dense packing while preserving the ability to easily fold and unfold any genomic locus. The fractal globule is distinct from the more commonly used globular equilibrium model. Our results demonstrate the power of Hi-C to map the dynamic conformations of whole genomes.;,citation_author=Erez Lieberman-Aiden;,citation_author=Nynke L. Van Berkum;,citation_author=Louise Williams;,citation_author=Maxim Imakaev;,citation_author=Tobias Ragoczy;,citation_author=Agnes Telling;,citation_author=Ido Amit;,citation_author=Bryan R. Lajoie;,citation_author=Peter J. Sabo;,citation_author=Michael O. Dorschner;,citation_author=Richard Sandstrom;,citation_author=Bradley Bernstein;,citation_author=M. A. Bender;,citation_author=Mark Groudine;,citation_author=Andreas Gnirke;,citation_author=John Stamatoyannopoulos;,citation_author=Leonid A. Mirny;,citation_author=Eric S. Lander;,citation_author=Job Dekker;,citation_publication_date=2009-10;,citation_cover_date=2009-10;,citation_year=2009;,citation_issue=5950;,citation_doi=10.1126/science.1181369;,citation_issn=0036-8075, 1095-9203;,citation_volume=326;,citation_language=en-US;,citation_journal_title=Science;">
<meta name="citation_reference" content="citation_title=The Ecology and Evolutionary Dynamics of Meiotic Drive;,citation_author=Anna K. Lindholm;,citation_author=Kelly A. Dyer;,citation_author=Renée C. Firman;,citation_author=Lila Fishman;,citation_author=Wolfgang Forstmeier;,citation_author=Luke Holman;,citation_author=Hanna Johannesson;,citation_author=Ulrich Knief;,citation_author=Hanna Kokko;,citation_author=Amanda M. Larracuente;,citation_author=Andri Manser;,citation_author=Catherine Montchamp-Moreau;,citation_author=Varos G. Petrosyan;,citation_author=Andrew Pomiankowski;,citation_author=Daven C. Presgraves;,citation_author=Larisa D. Safronova;,citation_author=Andreas Sutter;,citation_author=Robert L. Unckless;,citation_author=Rudi L. Verspoor;,citation_author=Nina Wedell;,citation_author=Gerald S. Wilkinson;,citation_author=Tom A. R. Price;,citation_publication_date=2016-04;,citation_cover_date=2016-04;,citation_year=2016;,citation_issue=4;,citation_doi=10.1016/j.tree.2016.02.001;,citation_issn=01695347;,citation_volume=31;,citation_language=en-US;,citation_journal_title=Trends in Ecology &amp;amp;amp; Evolution;">
<meta name="citation_reference" content="citation_title=Munch-group;,citation_author=Kasper Munch;,citation_publication_date=2024-11;,citation_cover_date=2024-11;,citation_year=2024;,citation_publisher=https://munch-group.org/research.html;">
<meta name="citation_reference" content="citation_title=Cooltools: Enabling high-resolution Hi-C analysis in Python;,citation_abstract=Abstract Chromosome conformation capture (3C) technologies reveal the incredible complexity of genome organization. Maps of increasing size, depth, and resolution are now used to probe genome architecture across cell states, types, and organisms. Larger datasets add challenges at each step of computational analysis, from storage and memory constraints to researchers’ time; however, analysis tools that meet these increased resource demands have not kept pace. Furthermore, existing tools offer limited support for customizing analysis for specific use cases or new biology. Here we introduce cooltools ( https://github.com/open2c/cooltools ), a suite of computational tools that enables flexible, scalable, and reproducible analysis of high-resolution contact frequency data. Cooltools leverages the widely-adopted cooler format which handles storage and access for high-resolution datasets. Cooltools provides a paired command line interface (CLI) and Python application programming interface (API), which respectively facilitate workflows on high-performance computing clusters and in interactive analysis environments. In short, cooltools enables the effective use of the latest and largest genome folding datasets.;,citation_author=Open2C;,citation_author=Nezar Abdennur;,citation_author=Sameer Abraham;,citation_author=Geoffrey Fudenberg;,citation_author=Ilya M. Flyamer;,citation_author=Aleksandra A. Galitsyna;,citation_author=Anton Goloborodko;,citation_author=Maxim Imakaev;,citation_author=Betul A. Oksuz;,citation_author=Sergey V. Venev;,citation_publication_date=2022-11;,citation_cover_date=2022-11;,citation_year=2022;,citation_doi=10.1101/2022.10.31.514564;,citation_language=en-US;,citation_publisher=Bioinformatics;">
<meta name="citation_reference" content="citation_title=Pairtools: From sequencing data to chromosome contacts;,citation_abstract=The field of 3D genome organization produces large amounts of sequencing data from Hi-C and a rapidly-expanding set of other chromosome conformation protocols (3C+). Massive and heterogeneous 3C+ data require high-performance and flexible processing of sequenced reads into contact pairs. To meet these challenges, we present pairtools –a flexible suite of tools for contact extraction from sequencing data. Pairtools provides modular command-line interface (CLI) tools that can be flexibly chained into data processing pipelines. The core operations provided by pairtools are parsing of.sam alignments into Hi-C pairs, sorting and removal of PCR duplicates. In addition, pairtools provides auxiliary tools for building feature-rich 3C+ pipelines, including contact pair manipulation, filtration, and quality control. Benchmarking pairtools against popular 3C+ data pipelines shows advantages of pairtools for high-performance and flexible 3C+ analysis. Finally, pairtools provides protocol-specific tools for restriction-based protocols, haplotype-resolved contacts, and single-cell Hi-C. The combination of CLI tools and tight integration with Python data analysis libraries makes pairtools a versatile foundation for a broad range of 3C+ pipelines.;,citation_author=Open2C;,citation_author=Nezar Abdennur;,citation_author=Geoffrey Fudenberg;,citation_author=Ilya M. Flyamer;,citation_author=Aleksandra A. Galitsyna;,citation_author=Anton Goloborodko;,citation_author=Maxim Imakaev;,citation_author=Sergey V. Venev;,citation_editor=Ferhat Ay;,citation_publication_date=2024-05;,citation_cover_date=2024-05;,citation_year=2024;,citation_issue=5;,citation_doi=10.1371/journal.pcbi.1012164;,citation_issn=1553-7358;,citation_volume=20;,citation_language=en-US;,citation_journal_title=PLOS Computational Biology;">
<meta name="citation_reference" content="citation_title=Open2C;,citation_abstract=We are the Open Chromosome Collective (Open2C). We are a group of computational biologists working on chromosome structure and interested in developing open source tools for the wider community. We are particularly interested in the 3D organization of chromosomes, and most of the tools are focused on the analysis of data obtained using Hi-C and related technologies. We like our tools to be easy to use, flexible, to facilitate active development of novel analytical approaches, and scalable, to make use of the latest and largest datasets.;,citation_language=en-US;,citation_journal_title=Open Chromosome Collective;,citation_publisher=https://open2c.github.io/;">
<meta name="citation_reference" content="citation_title=Principal components analysis corrects for stratification in genome-wide association studies;,citation_author=Alkes L Price;,citation_author=Nick J Patterson;,citation_author=Robert M Plenge;,citation_author=Michael E Weinblatt;,citation_author=Nancy A Shadick;,citation_author=David Reich;,citation_publication_date=2006-08;,citation_cover_date=2006-08;,citation_year=2006;,citation_issue=8;,citation_doi=10.1038/ng1847;,citation_issn=1061-4036, 1546-1718;,citation_volume=38;,citation_language=en-US;,citation_journal_title=Nature Genetics;">
<meta name="citation_reference" content="citation_title=High-resolution TADs reveal DNA sequences underlying genome organization in flies;,citation_abstract=Abstract Despite an abundance of new studies about topologically associating domains (TADs), the role of genetic information in TAD formation is still not fully understood. Here we use our software, HiCExplorer ( hicexplorer.readthedocs.io ) to annotate $&amp;amp;amp;gt;$2800 high-resolution (570 bp) TAD boundaries&nbsp;in Drosophila&nbsp;melanogaster . We identify eight DNA motifs enriched at boundaries, including a motif bound by the M1BP protein, and two new boundary motifs. In contrast to mammals, the CTCF motif is only enriched on a small fraction of boundaries flanking inactive chromatin while most active boundaries contain the motifs bound by the M1BP or Beaf-32 proteins. We demonstrate that boundaries can be accurately predicted using only the motif sequences at open chromatin sites. We propose that DNA sequence guides the genome architecture by allocation of boundary proteins in the genome. Finally, we present an interactive online database to access and explore the spatial organization of fly, mouse and human genomes, available at http://chorogenome.ie-freiburg.mpg.de .;,citation_author=Fidel Ramírez;,citation_author=Vivek Bhardwaj;,citation_author=Laura Arrigoni;,citation_author=Kin Chung Lam;,citation_author=Björn A. Grüning;,citation_author=José Villaveces;,citation_author=Bianca Habermann;,citation_author=Asifa Akhtar;,citation_author=Thomas Manke;,citation_publication_date=2018-01;,citation_cover_date=2018-01;,citation_year=2018;,citation_issue=1;,citation_doi=10.1038/s41467-017-02525-w;,citation_issn=2041-1723;,citation_volume=9;,citation_language=en-US;,citation_journal_title=Nature Communications;">
<meta name="citation_reference" content="citation_title=Loops, topologically associating domains, compartments, and territories are elastic and robust to dramatic nuclear volume swelling;,citation_abstract=Abstract Layers of genome organization are becoming increasingly better characterized, but less is known about how these structures respond to perturbation or shape changes. Low-salt swelling of isolated chromatin fibers or nuclei has been used for decades to investigate the structural properties of chromatin. But, visible changes in chromatin appearance have not been linked to known building blocks of genome structure or features along the genome sequence. We combine low-salt swelling of isolated nuclei with genome-wide chromosome conformation capture (Hi-C) and imaging approaches to probe the effects of chromatin extension genome-wide. Photoconverted patterns on nuclei during expansion and contraction indicate that global genome structure is preserved after dramatic nuclear volume swelling, suggesting a highly elastic chromosome topology. Hi-C experiments before, during, and after nuclear swelling show changes in average contact probabilities at short length scales, reflecting the extension of the local chromatin fiber. But, surprisingly, during this large increase in nuclear volume, there is a striking maintenance of loops, TADs, active and inactive compartments, and chromosome territories. Subtle differences after expansion are observed, suggesting that the local chromatin state, protein interactions, and location in the nucleus can affect how strongly a given structure is maintained under stress. From these observations, we propose that genome topology is robust to extension of the chromatin fiber and isotropic shape change, and that this elasticity may be beneficial in physiological circumstances of changes in nuclear size and volume.;,citation_author=Jacob T. Sanders;,citation_author=Rosela Golloshi;,citation_author=Priyojit Das;,citation_author=Yang Xu;,citation_author=Peyton H. Terry;,citation_author=Darrian G. Nash;,citation_author=Job Dekker;,citation_author=Rachel Patton McCord;,citation_publication_date=2022-03;,citation_cover_date=2022-03;,citation_year=2022;,citation_issue=1;,citation_doi=10.1038/s41598-022-08602-5;,citation_issn=2045-2322;,citation_volume=12;,citation_language=en-US;,citation_journal_title=Scientific Reports;">
<meta name="citation_reference" content="citation_title=HiC-Pro: An optimized and flexible pipeline for Hi-C data processing;,citation_author=Nicolas Servant;,citation_author=Nelle Varoquaux;,citation_author=Bryan R. Lajoie;,citation_author=Eric Viara;,citation_author=Chong-Jian Chen;,citation_author=Jean-Philippe Vert;,citation_author=Edith Heard;,citation_author=Job Dekker;,citation_author=Emmanuel Barillot;,citation_publication_date=2015-12;,citation_cover_date=2015-12;,citation_year=2015;,citation_issue=1;,citation_doi=10.1186/s13059-015-0831-x;,citation_issn=1474-760X;,citation_volume=16;,citation_language=en-US;,citation_journal_title=Genome Biology;">
<meta name="citation_reference" content="citation_title=Single-Cell RNA Sequencing of Human, Macaque, and Mouse Testes Uncovers Conserved and Divergent Features of Mammalian Spermatogenesis;,citation_author=Adrienne Niederriter Shami;,citation_author=Xianing Zheng;,citation_author=Sarah K. Munyoki;,citation_author=Qianyi Ma;,citation_author=Gabriel L. Manske;,citation_author=Christopher D. Green;,citation_author=Meena Sukhwani;,citation_author=Kyle E. Orwig;,citation_author=Jun Z. Li;,citation_author=Saher Sue Hammoud;,citation_publication_date=2020-08;,citation_cover_date=2020-08;,citation_year=2020;,citation_issue=4;,citation_doi=10.1016/j.devcel.2020.05.010;,citation_issn=15345807;,citation_volume=54;,citation_language=en-US;,citation_journal_title=Developmental Cell;">
<meta name="citation_reference" content="citation_title=Extraordinary selection on the human X chromosome associated with archaic admixture;,citation_author=Laurits Skov;,citation_author=Moisès Coll Macià;,citation_author=Elise Anne Lucotte;,citation_author=Maria Izabel Alves Cavassim;,citation_author=David Castellano;,citation_author=Mikkel Heide Schierup;,citation_author=Kasper Munch;,citation_publication_date=2023-03;,citation_cover_date=2023-03;,citation_year=2023;,citation_issue=3;,citation_doi=10.1016/j.xgen.2023.100274;,citation_issn=2666979X;,citation_volume=3;,citation_language=en-US;,citation_journal_title=Cell Genomics;">
<meta name="citation_reference" content="citation_title=Genome-wide coancestry reveals details of ancient and recent male-driven reticulation in baboons;,citation_abstract=Baboons (genus Papio ) are a morphologically and behaviorally diverse clade of catarrhine monkeys that have experienced hybridization between phenotypically and genetically distinct phylogenetic species. We used high-coverage whole-genome sequences from 225 wild baboons representing 19 geographic localities to investigate population genomics and interspecies gene flow. Our analyses provide an expanded picture of evolutionary reticulation among species and reveal patterns of population structure within and among species, including differential admixture among conspecific populations. We describe the first example of a baboon population with a genetic composition that is derived from three distinct lineages. The results reveal processes, both ancient and recent, that produced the observed mismatch between phylogenetic relationships based on matrilineal, patrilineal, and biparental inheritance. We also identified several candidate genes that may contribute to species-specific phenotypes. , INTRODUCTION As a widespread but comparatively young clade of six parapatric species, the baboons ( Papio sp.) exemplify a frequently observed pattern of mammalian diversity. In particular, they provide analogs for the population structure of the multibranched prehuman lineage that occupied a similar geographic range before the hegemony of “modern” humans, Homo sapiens . Despite phenotypic and genetic differences, interspecies hybridization has been described between baboons at several locations, and population relationships based on mitochondrial DNA (mtDNA) do not correspond with relationships based on phenotype. These previous studies captured the broad outlines of baboon population genetic structure and evolutionary history but necessarily used data that were limited in genomic and geographical coverage and therefore could not adequately document inter- and intrapopulation variation. In this study, we analyzed whole-genome sequences of 225 baboons representing all six species and 19 geographic sites, with 18 local populations represented by multiple individuals. RATIONALE Recent studies have identified several mammalian species groups in which genetically distinct lineages have hybridized to generate complex reticulate phylogenies. Baboons provide a valuable context for studying processes generating such population and phylogenetic complexity because extant parapatric species form hybrid zones in several regions of Africa, allowing for direct observation of ongoing introgression. Furthermore, prior studies of nuclear and mtDNA and phenotypic diversity have demonstrated gene flow among differentiated lineages but were unable to develop the detailed picture of process and history that is now possible using whole-genome sequences and modern computational methods. To address these questions, we designed a study that would provide a more fine-grained picture of recent and ancient genetic reticulation by comparing phenotypes and autosomal, X and Y chromosomal, and mtDNA sequences, along with polymorphic insertions of repetitive elements across multiple baboon populations. RESULTS Using deep whole-genome sequence data from 225 baboons representing multiple populations, we identified several previously unknown geographic sites of gene flow between genetically distinct populations. We report that yellow baboons ( P. cynocephalus ) from western Tanzania are the first nonhuman primate found to have received genetic input from three distinct lineages. We compared the ancestry shared among individuals, estimated separately from the X chromosome and autosomes, to distinguish shared ancestry due to ancestral population relationships from coancestry as a result of recent male-biased immigration and gene flow. This reveals directionality and sex bias of recent gene flow in several locations. Analyses of population differences within species quantified different degrees of interspecies introgression among populations with an essentially identical phenotype. CONCLUSION The population genetic structure and history of introgression among baboon lineages are even more complex than predicted from observed phenotypic diversity and prior studies of limited genetic data. Single populations can carry genetic contributions from more than two ancestral sources. Populations that appear homogeneous on the basis of observable phenotype can display different levels of interspecies introgression. The evolutionary dynamics and current structure of baboon population diversity indicate that other mammals displaying differentiated and geographically separate species may also have more-complex histories than anticipated. This may also be true for the morphologically defined hominin taxa from the past 4 million years. Ancient and recent admixture among baboons: Complex population substructure and reticulation revealed by whole-genome sequencing. Pie charts represent recent ancestry of East African populations, with species contributions colored as in the inset map. Patterns of mixed ancestry differ substantially, even among conspecific populations. This suggests a complex history of recurrent interpopulational gene flow, driven predominantly by male migration. Comparably complex admixture probably also occurred among early hominins.;,citation_author=Erik F. Sørensen;,citation_author=R. Alan Harris;,citation_author=Liye Zhang;,citation_author=Muthuswamy Raveendran;,citation_author=Lukas F. K. Kuderna;,citation_author=Jerilyn A. Walker;,citation_author=Jessica M. Storer;,citation_author=Martin Kuhlwilm;,citation_author=Claudia Fontsere;,citation_author=Lakshmi Seshadri;,citation_author=Christina M. Bergey;,citation_author=Andrew S. Burrell;,citation_author=Juraj Bergman;,citation_author=Jane E. Phillips-Conroy;,citation_author=Fekadu Shiferaw;,citation_author=Kenneth L. Chiou;,citation_author=Idrissa S. Chuma;,citation_author=Julius D. Keyyu;,citation_author=Julia Fischer;,citation_author=Marie-Claude Gingras;,citation_author=Sejal Salvi;,citation_author=Harshavardhan Doddapaneni;,citation_author=Mikkel H. Schierup;,citation_author=Mark A. Batzer;,citation_author=Clifford J. Jolly;,citation_author=Sascha Knauf;,citation_author=Dietmar Zinner;,citation_author=Kyle K.-H. Farh;,citation_author=Tomas Marques-Bonet;,citation_author=Kasper Munch;,citation_author=Christian Roos;,citation_author=Jeffrey Rogers;,citation_publication_date=2023-06;,citation_cover_date=2023-06;,citation_year=2023;,citation_issue=6648;,citation_doi=10.1126/science.abn8153;,citation_issn=0036-8075, 1095-9203;,citation_volume=380;,citation_language=en-US;,citation_journal_title=Science;">
<meta name="citation_reference" content="citation_title=Reprogramming of Meiotic Chromatin Architecture during Spermatogenesis;,citation_author=Yao Wang;,citation_author=Hanben Wang;,citation_author=Yu Zhang;,citation_author=Zhenhai Du;,citation_author=Wei Si;,citation_author=Suixing Fan;,citation_author=Dongdong Qin;,citation_author=Mei Wang;,citation_author=Yanchao Duan;,citation_author=Lufan Li;,citation_author=Yuying Jiao;,citation_author=Yuanyuan Li;,citation_author=Qiujun Wang;,citation_author=Qinghua Shi;,citation_author=Xin Wu;,citation_author=Wei Xie;,citation_publication_date=2019-02;,citation_cover_date=2019-02;,citation_year=2019;,citation_issue=3;,citation_doi=10.1016/j.molcel.2018.11.019;,citation_issn=10972765;,citation_volume=73;,citation_language=en-US;,citation_journal_title=Molecular Cell;">
<meta name="citation_reference" content="citation_title=3D chromatin structure changes during spermatogenesis and oogenesis;,citation_author=Shiqiang Zhang;,citation_author=Wanyu Tao;,citation_author=Jing-Dong J. Han;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_doi=10.1016/j.csbj.2022.05.032;,citation_issn=20010370;,citation_volume=20;,citation_language=en-US;,citation_journal_title=Computational and Structural Biotechnology Journal;">
<meta name="citation_reference" content="citation_title=Stage-resolved Hi-C analyses reveal meiotic chromosome organizational features influencing homolog alignment;,citation_abstract=Abstract During meiosis, chromosomes exhibit dramatic changes in morphology and intranuclear positioning. How these changes influence homolog pairing, alignment, and recombination remain elusive. Using Hi-C, we systematically mapped 3D genome architecture throughout all meiotic prophase substages during mouse spermatogenesis. Our data uncover two major chromosome organizational features varying along the chromosome axis during early meiotic prophase, when homolog alignment occurs. First, transcriptionally active and inactive genomic regions form alternating domains consisting of shorter and longer chromatin loops, respectively. Second, the force-transmitting LINC complex promotes the alignment of ends of different chromosomes over a range of up to 20% of chromosome length. Both features correlate with the pattern of homolog interactions and the distribution of recombination events. Collectively, our data reveal the influences of transcription and force on meiotic chromosome structure and suggest chromosome organization may provide an infrastructure for the modulation of meiotic recombination in higher eukaryotes.;,citation_author=Wu Zuo;,citation_author=Guangming Chen;,citation_author=Zhimei Gao;,citation_author=Shuai Li;,citation_author=Yanyan Chen;,citation_author=Chenhui Huang;,citation_author=Juan Chen;,citation_author=Zhengjun Chen;,citation_author=Ming Lei;,citation_author=Qian Bian;,citation_publication_date=2021-10;,citation_cover_date=2021-10;,citation_year=2021;,citation_issue=1;,citation_doi=10.1038/s41467-021-26033-0;,citation_issn=2041-1723;,citation_volume=12;,citation_language=en-US;,citation_journal_title=Nature Communications;">
<meta name="citation_reference" content="citation_title=SRA Toolkit;,citation_abstract=SRA Tools. Contribute to ncbi/sra-tools development by creating an account on GitHub.;,citation_author=SRA Toolkit DevTeam;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_fulltext_html_url=https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=software;,citation_language=en-US;,citation_journal_title=GitHub;,citation_publisher=https://github.com/ncbi/sra-tools/wiki/01.-Downloading-SRA-Toolkit;">
<meta name="citation_reference" content="citation_title=Aligning sequence reads, clone sequences and assembly contigs with BWA-MEM;,citation_author=Heng Li;,citation_publication_date=2013;,citation_cover_date=2013;,citation_year=2013;,citation_fulltext_html_url=https://arxiv.org/abs/1303.3997;">
<meta name="citation_reference" content="citation_title=Sequence diversity analyses of an improved rhesus macaque genome enhance its biomedical utility;,citation_abstract=A high-quality rhesus macaque genome. Genome technology has improved substantially since the first full organismal genomes were generated. Applying new technology, Warren et al. refined the genome of the rhesus macaque, a model nonhuman primate. Long-read technology and other recent advances in sequencing technology were applied to generate a genome with far fewer gaps and helped to refine the locations and numbers of repetitive elements. Furthermore, the authors performed resequencing among populations to identify the genetic variability of the rhesus macaque. Thus, a previously incomplete and inaccurate set of sequence information is now fully resolved, improving gene mapping for biomedical and comparative genetic studies.;,citation_author=Wesley C. Warren;,citation_author=R. Alan Harris;,citation_author=Marina Haukness;,citation_author=Ian T. Fiddes;,citation_author=Shwetha C. Murali;,citation_author=Jason Fernandes;,citation_author=Philip C. Dishuck;,citation_author=Jessica M. Storer;,citation_author=Muthuswamy Raveendran;,citation_author=LaDeana W. Hillier;,citation_author=David Porubsky;,citation_author=Yafei Mao;,citation_author=David Gordon;,citation_author=Mitchell R. Vollger;,citation_author=Alexandra P. Lewis;,citation_author=Katherine M. Munson;,citation_author=Elizabeth DeVogelaere;,citation_author=Joel Armstrong;,citation_author=Mark Diekhans;,citation_author=Jerilyn A. Walker;,citation_author=Chad Tomlinson;,citation_author=Tina A. Graves-Lindsay;,citation_author=Milinn Kremitzki;,citation_author=Sofie R. Salama;,citation_author=Peter A. Audano;,citation_author=Merly Escalona;,citation_author=Nicholas W. Maurer;,citation_author=Francesca Antonacci;,citation_author=Ludovica Mercuri;,citation_author=Flavia A. M. Maggiolini;,citation_author=Claudia Rita Catacchio;,citation_author=Jason G. Underwood;,citation_author=David H. O’Connor;,citation_author=Ashley D. Sanders;,citation_author=Jan O. Korbel;,citation_author=Betsy Ferguson;,citation_author=H. Michael Kubisch;,citation_author=Louis Picker;,citation_author=Ned H. Kalin;,citation_author=Douglas Rosene;,citation_author=Jon Levine;,citation_author=David H. Abbott;,citation_author=Stanton B. Gray;,citation_author=Mar M. Sanchez;,citation_author=Zsofia A. Kovacs-Balint;,citation_author=Joseph W. Kemnitz;,citation_author=Sara M. Thomasy;,citation_author=Jeffrey A. Roberts;,citation_author=Erin L. Kinnally;,citation_author=John P. Capitanio;,citation_author=J. H. Pate Skene;,citation_author=Michael Platt;,citation_author=Shelley A. Cole;,citation_author=Richard E. Green;,citation_author=Mario Ventura;,citation_author=Roger W. Wiseman;,citation_author=Benedict Paten;,citation_author=Mark A. Batzer;,citation_author=Jeffrey Rogers;,citation_author=Evan E. Eichler;,citation_publication_date=2020-12;,citation_cover_date=2020-12;,citation_year=2020;,citation_issue=6523;,citation_doi=10.1126/science.abc6617;,citation_issn=0036-8075, 1095-9203;,citation_volume=370;,citation_journal_title=Science;">
<meta name="citation_reference" content="citation_title=Application of Hi-C and other omics data analysis in human cancer and cell differentiation research;,citation_author=Haiyan Gong;,citation_author=Yi Yang;,citation_author=Sichen Zhang;,citation_author=Minghong Li;,citation_author=Xiaotong Zhang;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_doi=10.1016/j.csbj.2021.04.016;,citation_issn=20010370;,citation_volume=19;,citation_language=en-US;,citation_journal_title=Computational and Structural Biotechnology Journal;">
<meta name="citation_reference" content="citation_title=MultiQC: Summarize analysis results for multiple tools and samples in a single report;,citation_abstract=Abstract Motivation: Fast and accurate quality control is essential for studies involving next-generation sequencing data. Whilst numerous tools exist to quantify QC metrics, there is no common approach to flexibly integrate these across tools and large sample sets. Assessing analysis results across an entire project can be time consuming and error prone; batch effects and outlier samples can easily be missed in the early stages of analysis. Results: We present MultiQC, a tool to create a single report visualising output from multiple tools across many samples, enabling global trends and biases to be quickly identified. MultiQC can plot data from many common bioinformatics tools and is built to allow easy extension and customization. Availability and implementation: MultiQC is available with an GNU GPLv3 license on GitHub, the Python Package Index and Bioconda. Documentation and example reports are available at http://multiqc.info Contact: &nbsp;phil.ewels@scilifelab.se;,citation_author=Philip Ewels;,citation_author=Måns Magnusson;,citation_author=Sverker Lundin;,citation_author=Max Käller;,citation_publication_date=2016-10;,citation_cover_date=2016-10;,citation_year=2016;,citation_issue=19;,citation_doi=10.1093/bioinformatics/btw354;,citation_issn=1367-4811, 1367-4803;,citation_volume=32;,citation_language=en-US;,citation_journal_title=Bioinformatics;">
<meta name="citation_reference" content="citation_title=Segment_liftover : A Python tool to convert segments between genome assemblies;,citation_abstract=The process of assembling a species’ reference genome may be performed in a number of iterations, with subsequent genome assemblies differing in the coordinates of mapped elements. The conversion of genome coordinates between different assemblies is required for many integrative and comparative studies. While currently a number of bioinformatics tools are available to accomplish this task, most of them are tailored towards the conversion of single genome coordinates. When converting the boundary positions of segments spanning larger genome regions, segments may be mapped into smaller sub-segments if the original segment’s continuity is disrupted in the target assembly. Such a conversion may lead to a relevant degree of data loss in some circumstances such as copy number variation (CNV) analysis, where the quantitative representation of a genomic region takes precedence over base-specific accuracy. segment_liftover aims at continuity-preserving remapping of genome segments between assemblies and provides features such as approximate locus conversion, automated batch processing and comprehensive logging to facilitate processing of datasets containing large numbers of structural genome variation data.;,citation_author=Bo Gao;,citation_author=Qingyao Huang;,citation_author=Michael Baudis;,citation_publication_date=2018-06;,citation_cover_date=2018-06;,citation_year=2018;,citation_doi=10.12688/f1000research.14148.2;,citation_issn=2046-1402;,citation_volume=7;,citation_language=en-US;,citation_journal_title=F1000Research;">
<meta name="citation_reference" content="citation_title=The UCSC Genome Browser Database: Update 2006;,citation_author=A. S. Hinrichs;,citation_publication_date=2006-01;,citation_cover_date=2006-01;,citation_year=2006;,citation_issue=90001;,citation_doi=10.1093/nar/gkj144;,citation_issn=0305-1048, 1362-4962;,citation_volume=34;,citation_language=en-US;,citation_journal_title=Nucleic Acids Research;">
<meta name="citation_reference" content="citation_title=Your guide to GAI responsibly;,citation_author=AU IT;,citation_language=en-US;,citation_publisher=https://medarbejdere.au.dk/en/gai;">
<meta name="citation_reference" content="citation_title=Analysis of 62 hybrid assembled human Y chromosomes exposes rapid structural changes and high rates of gene conversion;,citation_author=Laurits Skov;,citation_author=The Danish Pan Genome Consortium;,citation_author=Mikkel Heide Schierup;,citation_editor=Kateryna D. Makova;,citation_publication_date=2017-08;,citation_cover_date=2017-08;,citation_year=2017;,citation_issue=8;,citation_doi=10.1371/journal.pgen.1006834;,citation_issn=1553-7404;,citation_volume=13;,citation_language=en-US;,citation_journal_title=PLOS Genetics;">
<meta name="citation_reference" content="citation_title=Strong Selective Sweeps on the X Chromosome in the Human-Chimpanzee Ancestor Explain Its Low Divergence;,citation_author=Julien Y. Dutheil;,citation_author=Kasper Munch;,citation_author=Kiwoong Nam;,citation_author=Thomas Mailund;,citation_author=Mikkel H. Schierup;,citation_editor=Nick H. Barton;,citation_publication_date=2015-08;,citation_cover_date=2015-08;,citation_year=2015;,citation_issue=8;,citation_doi=10.1371/journal.pgen.1005451;,citation_issn=1553-7404;,citation_volume=11;,citation_language=en-US;,citation_journal_title=PLOS Genetics;">
<meta name="citation_reference" content="citation_title=Selective Sweeps across Twenty Millions Years of Primate Evolution;,citation_author=Kasper Munch;,citation_author=Kiwoong Nam;,citation_author=Mikkel Heide Schierup;,citation_author=Thomas Mailund;,citation_publication_date=2016-12;,citation_cover_date=2016-12;,citation_year=2016;,citation_issue=12;,citation_doi=10.1093/molbev/msw199;,citation_issn=0737-4038, 1537-1719;,citation_volume=33;,citation_language=en-US;,citation_journal_title=Molecular Biology and Evolution;">
<meta name="citation_reference" content="citation_title=An in-depth investigation on the behavior of measures to quantify reproducibility;,citation_author=Maria Maistro;,citation_author=Timo Breuer;,citation_author=Philipp Schaer;,citation_author=Nicola Ferro;,citation_publication_date=2023-05;,citation_cover_date=2023-05;,citation_year=2023;,citation_issue=3;,citation_doi=10.1016/j.ipm.2023.103332;,citation_issn=03064573;,citation_volume=60;,citation_language=en-US;,citation_journal_title=Information Processing &amp;amp;amp; Management;">
<meta name="citation_reference" content="citation_title=1,500 scientists lift the lid on reproducibility;,citation_author=Monya Baker;,citation_publication_date=2016-05;,citation_cover_date=2016-05;,citation_year=2016;,citation_issue=7604;,citation_doi=10.1038/533452a;,citation_issn=0028-0836, 1476-4687;,citation_volume=533;,citation_language=en-US;,citation_journal_title=Nature;">
<meta name="citation_reference" content="citation_title=The art of UNIX programming: With contributions from thirteen UNIX pioneers, including its inventor, Ken Thompson;,citation_author=Eric S. Raymond;,citation_publication_date=2008;,citation_cover_date=2008;,citation_year=2008;,citation_isbn=978-0-13-142901-7;,citation_language=en-US;,citation_series_title=Addison-Wesley professional computing series;">
<meta name="citation_reference" content="citation_title=Jupyter notebooks ? A publishing format for reproducible computational workflows;,citation_abstract=It is increasingly necessary for researchers in all fields to write computer code, and in order to reproduce research results, it is important that this code is published. We present Jupyter notebooks, a document format for publishing code, results and explanations in a form that is both readable and executable. We discuss various tools and use cases for notebook documents.;,citation_author=Thomas Kluyver;,citation_author=Benjamin Ragan-Kelley;,citation_author=Fernando Pérez;,citation_author=Brian Granger;,citation_author=Matthias Bussonnier;,citation_author=Jonathan Frederic;,citation_author=Kyle Kelley;,citation_author=Jessica Hamrick;,citation_author=Jason Grout;,citation_author=Sylvain Corlay;,citation_author=Paul Ivanov;,citation_author=Damián Avila;,citation_author=Safia Abdalla;,citation_author=Carol Willing;,citation_author=Jupyter team;,citation_editor=Fernando Loizides;,citation_editor=Birgit Scmidt;,citation_publication_date=2016;,citation_cover_date=2016;,citation_year=2016;,citation_fulltext_html_url=https://eprints.soton.ac.uk/403913/;,citation_conference_title=Positioning and power in academic publishing: Players, agents and agendas;,citation_conference=IOS Press;">
<meta name="citation_reference" content="citation_title=The Galaxy platform for accessible, reproducible, and collaborative data analyses: 2024 update;,citation_abstract=Abstract Galaxy (https://galaxyproject.org) is deployed globally, predominantly through free-to-use services, supporting user-driven research that broadens in scope each year. Users are attracted to public Galaxy services by platform stability, tool and reference dataset diversity, training, support&nbsp;and integration, which enables complex, reproducible, shareable data analysis. Applying the principles of user experience design (UXD), has driven improvements in accessibility, tool discoverability through Galaxy Labs/subdomains, and a redesigned Galaxy ToolShed. Galaxy tool capabilities are progressing in two strategic directions: integrating general purpose graphical processing units (GPGPU) access for cutting-edge methods, and licensed tool support. Engagement with global research consortia is being increased by developing more workflows in Galaxy and by resourcing the public Galaxy services to run them. The Galaxy Training Network (GTN) portfolio has grown in both size, and accessibility, through learning paths and direct integration with Galaxy tools that feature in training courses. Code development continues in line with the Galaxy Project roadmap, with improvements to job scheduling and the user interface. Environmental impact assessment is also helping engage users and developers, reminding them of their role in sustainability, by displaying estimated CO2 emissions generated by each Galaxy job.;,citation_author=The Galaxy Community;,citation_author=Linelle Ann L Abueg;,citation_author=Enis Afgan;,citation_author=Olivier Allart;,citation_author=Ahmed H Awan;,citation_author=Wendi A Bacon;,citation_author=Dannon Baker;,citation_author=Madeline Bassetti;,citation_author=Bérénice Batut;,citation_author=Matthias Bernt;,citation_author=Daniel Blankenberg;,citation_author=Aureliano Bombarely;,citation_author=Anthony Bretaudeau;,citation_author=Catherine J Bromhead;,citation_author=Melissa L Burke;,citation_author=Patrick K Capon;,citation_author=Martin Čech;,citation_author=María Chavero-Díez;,citation_author=John M Chilton;,citation_author=Tyler J Collins;,citation_author=Frederik Coppens;,citation_author=Nate Coraor;,citation_author=Gianmauro Cuccuru;,citation_author=Fabio Cumbo;,citation_author=John Davis;,citation_author=Paul F De Geest;,citation_author=Willem De Koning;,citation_author=Martin Demko;,citation_author=Assunta DeSanto;,citation_author=José Manuel Domínguez Begines;,citation_author=Maria A Doyle;,citation_author=Bert Droesbeke;,citation_author=Anika Erxleben-Eggenhofer;,citation_author=Melanie C Föll;,citation_author=Giulio Formenti;,citation_author=Anne Fouilloux;,citation_author=Rendani Gangazhe;,citation_author=Tanguy Genthon;,citation_author=Jeremy Goecks;,citation_author=Alejandra N Gonzalez Beltran;,citation_author=Nuwan A Goonasekera;,citation_author=Nadia Goué;,citation_author=Timothy J Griffin;,citation_author=Björn A Grüning;,citation_author=Aysam Guerler;,citation_author=Sveinung Gundersen;,citation_author=Ove Johan Ragnar Gustafsson;,citation_author=Christina Hall;,citation_author=Thomas W Harrop;,citation_author=Helge Hecht;,citation_author=Alireza Heidari;,citation_author=Tillman Heisner;,citation_author=Florian Heyl;,citation_author=Saskia Hiltemann;,citation_author=Hans-Rudolf Hotz;,citation_author=Cameron J Hyde;,citation_author=Pratik D Jagtap;,citation_author=Julia Jakiela;,citation_author=James E Johnson;,citation_author=Jayadev Joshi;,citation_author=Marie Jossé;,citation_author=Khaled Jum’ah;,citation_author=Matúš Kalaš;,citation_author=Katarzyna Kamieniecka;,citation_author=Tunc Kayikcioglu;,citation_author=Markus Konkol;,citation_author=Leonid Kostrykin;,citation_author=Natalie Kucher;,citation_author=Anup Kumar;,citation_author=Mira Kuntz;,citation_author=Delphine Lariviere;,citation_author=Ross Lazarus;,citation_author=Yvan Le Bras;,citation_author=Gildas Le Corguillé;,citation_author=Justin Lee;,citation_author=Simone Leo;,citation_author=Leandro Liborio;,citation_author=Romane Libouban;,citation_author=David López Tabernero;,citation_author=Lucille Lopez-Delisle;,citation_author=Laila S Los;,citation_author=Alexandru Mahmoud;,citation_author=Igor Makunin;,citation_author=Pierre Marin;,citation_author=Subina Mehta;,citation_author=Winnie Mok;,citation_author=Pablo A Moreno;,citation_author=François Morier-Genoud;,citation_author=Stephen Mosher;,citation_author=Teresa Müller;,citation_author=Engy Nasr;,citation_author=Anton Nekrutenko;,citation_author=Tiffanie M Nelson;,citation_author=Asime J Oba;,citation_author=Alexander Ostrovsky;,citation_author=Polina V Polunina;,citation_author=Krzysztof Poterlowicz;,citation_author=Elliott J Price;,citation_author=Gareth R Price;,citation_author=Helena Rasche;,citation_author=Bryan Raubenolt;,citation_author=Coline Royaux;,citation_author=Luke Sargent;,citation_author=Michelle T Savage;,citation_author=Volodymyr Savchenko;,citation_author=Denys Savchenko;,citation_author=Michael C Schatz;,citation_author=Pauline Seguineau;,citation_author=Beatriz Serrano-Solano;,citation_author=Nicola Soranzo;,citation_author=Sanjay Kumar Srikakulam;,citation_author=Keith Suderman;,citation_author=Anna E Syme;,citation_author=Marco Antonio Tangaro;,citation_author=Jonathan A Tedds;,citation_author=Mehmet Tekman;,citation_author=Wai Cheng (Mike) Thang;,citation_author=Anil S Thanki;,citation_author=Michael Uhl;,citation_author=Marius Van Den Beek;,citation_author=Deepti Varshney;,citation_author=Jenn Vessio;,citation_author=Pavankumar Videm;,citation_author=Greg Von Kuster;,citation_author=Gregory R Watson;,citation_author=Natalie Whitaker-Allen;,citation_author=Uwe Winter;,citation_author=Martin Wolstencroft;,citation_author=Federico Zambelli;,citation_author=Paul Zierep;,citation_author=Rand Zoabi;,citation_publication_date=2024-07;,citation_cover_date=2024-07;,citation_year=2024;,citation_issue=W1;,citation_doi=10.1093/nar/gkae410;,citation_issn=0305-1048, 1362-4962;,citation_volume=52;,citation_language=en-US;,citation_journal_title=Nucleic Acids Research;">
<meta name="citation_reference" content="citation_title=Galaxy HiCExplorer 3: A web server for reproducible Hi-C, capture Hi-C and single-cell Hi-C data analysis, quality control and visualization;,citation_abstract=Abstract The Galaxy HiCExplorer provides a web service at https://hicexplorer.usegalaxy.eu. It enables the integrative analysis of chromosome conformation by providing tools and computational resources to pre-process, analyse and visualize Hi-C, Capture Hi-C (cHi-C) and single-cell Hi-C (scHi-C) data. Since the last publication, Galaxy HiCExplorer has been expanded considerably with new tools to facilitate the analysis of cHi-C and to provide an in-depth analysis of Hi-C data. Moreover, it supports the analysis of scHi-C data by offering a broad range of tools. With the help of the standard graphical user interface of Galaxy, presented workflows, extensive documentation and tutorials, novices as well as Hi-C experts are supported in their Hi-C data analysis with Galaxy HiCExplorer.;,citation_author=Joachim Wolff;,citation_author=Leily Rabbani;,citation_author=Ralf Gilsbach;,citation_author=Gautier Richard;,citation_author=Thomas Manke;,citation_author=Rolf Backofen;,citation_author=Björn A Grüning;,citation_publication_date=2020-07;,citation_cover_date=2020-07;,citation_year=2020;,citation_issue=W1;,citation_doi=10.1093/nar/gkaa220;,citation_issn=0305-1048, 1362-4962;,citation_volume=48;,citation_language=en-US;,citation_journal_title=Nucleic Acids Research;">
<meta name="citation_reference" content="citation_title=Galaxy HiCExplorer: A web server for reproducible Hi-C data analysis, quality control and visualization;,citation_author=Joachim Wolff;,citation_author=Vivek Bhardwaj;,citation_author=Stephan Nothjunge;,citation_author=Gautier Richard;,citation_author=Gina Renschler;,citation_author=Ralf Gilsbach;,citation_author=Thomas Manke;,citation_author=Rolf Backofen;,citation_author=Fidel Ramírez;,citation_author=Björn A Grüning;,citation_publication_date=2018-07;,citation_cover_date=2018-07;,citation_year=2018;,citation_issue=W1;,citation_doi=10.1093/nar/gky504;,citation_issn=0305-1048, 1362-4962;,citation_volume=46;,citation_language=en-US;,citation_journal_title=Nucleic Acids Research;">
<meta name="citation_reference" content="citation_title=Lineage Sorting in Apes;,citation_abstract=Recombination allows different parts of the genome to have different genealogical histories. When a species splits in two, allelic lineages sort into the two descendant species, and this lineage sorting varies along the genome. If speciation events are close in time, the lineage sorting process may be incomplete at the second speciation event and lead to gene genealogies that do not match the species phylogeny. We review different recent approaches to model lineage sorting along the genome and show how it is possible to learn about population sizes, natural selection, and recombination rates in ancestral species from application of these models to genome alignments of great ape species.;,citation_author=Thomas Mailund;,citation_author=Kasper Munch;,citation_author=Mikkel Heide Schierup;,citation_publication_date=2014-11;,citation_cover_date=2014-11;,citation_year=2014;,citation_issue=1;,citation_doi=10.1146/annurev-genet-120213-092532;,citation_issn=0066-4197, 1545-2948;,citation_volume=48;,citation_language=en-US;,citation_journal_title=Annual Review of Genetics;">
<meta name="citation_reference" content="citation_title=Genetic Signatures of Strong Recent Positive Selection at the Lactase Gene;,citation_author=Todd Bersaglieri;,citation_author=Pardis C. Sabeti;,citation_author=Nick Patterson;,citation_author=Trisha Vanderploeg;,citation_author=Steve F. Schaffner;,citation_author=Jared A. Drake;,citation_author=Matthew Rhodes;,citation_author=David E. Reich;,citation_author=Joel N. Hirschhorn;,citation_publication_date=2004-06;,citation_cover_date=2004-06;,citation_year=2004;,citation_issue=6;,citation_doi=10.1086/421051;,citation_issn=00029297;,citation_volume=74;,citation_language=en-US;,citation_journal_title=The American Journal of Human Genetics;">
<meta name="citation_reference" content="citation_title=Matplotlib: Visualization with Python;,citation_abstract=This is the fourth micro release of the 3.8 series. Highlights of the 3.8.4 release include: Enable building against numpy 2.0; released wheels are built against numpy 2 macosx: Clean up single-shot timers correctly Add a draw during show for macos backend Fix color sequence data for Set2 and Set3 gtk: Ensure pending draws are done before GTK draw Update &amp;amp;amp;quot;Created with&amp;quot; url in hand.svg Avoid modifying user input to Axes.bar fix quiver3d incorrect arrow colors;,citation_author=The Matplotlib Development Team;,citation_publication_date=2024-04;,citation_cover_date=2024-04;,citation_year=2024;,citation_doi=10.5281/ZENODO.10916799;,citation_publisher=Zenodo;">
<meta name="citation_reference" content="citation_title=Seaborn: Statistical data visualization;,citation_author=Michael Waskom;,citation_publication_date=2021-04;,citation_cover_date=2021-04;,citation_year=2021;,citation_issue=60;,citation_doi=10.21105/joss.03021;,citation_issn=2475-9066;,citation_volume=6;,citation_journal_title=Journal of Open Source Software;">
<meta name="citation_reference" content="citation_title=Database resources of the national center for biotechnology information;,citation_abstract=Abstract The National Center for Biotechnology Information (NCBI) produces a variety of online information resources for biology, including the GenBank nucleic acid sequence database and the PubMed database of citations and abstracts published in life science journals. NCBI provides search and retrieval operations for most of these data from 35 distinct databases. The E-utilities serve as the programming interface for the most of these databases. Resources receiving significant updates in the past year include PubMed, PMC, Bookshelf, RefSeq, SRA, Virus, dbSNP, dbVar, ClinicalTrials.gov, MMDB, iCn3D&nbsp;and PubChem. These resources can be accessed through the NCBI home page at https://www.ncbi.nlm.nih.gov.;,citation_author=Eric W Sayers;,citation_author=Evan E Bolton;,citation_author=J Rodney Brister;,citation_author=Kathi Canese;,citation_author=Jessica Chan;,citation_author=Donald C Comeau;,citation_author=Ryan Connor;,citation_author=Kathryn Funk;,citation_author=Chris Kelly;,citation_author=Sunghwan Kim;,citation_author=Tom Madej;,citation_author=Aron Marchler-Bauer;,citation_author=Christopher Lanczycki;,citation_author=Stacy Lathrop;,citation_author=Zhiyong Lu;,citation_author=Francoise Thibaud-Nissen;,citation_author=Terence Murphy;,citation_author=Lon Phan;,citation_author=Yuri Skripchenko;,citation_author=Tony Tse;,citation_author=Jiyao Wang;,citation_author=Rebecca Williams;,citation_author=Barton W Trawick;,citation_author=Kim D Pruitt;,citation_author=Stephen T Sherry;,citation_publication_date=2022-01;,citation_cover_date=2022-01;,citation_year=2022;,citation_issue=D1;,citation_doi=10.1093/nar/gkab1112;,citation_issn=0305-1048, 1362-4962;,citation_volume=50;,citation_language=en-US;,citation_journal_title=Nucleic Acids Research;">
<meta name="citation_reference" content="citation_title=The Self-Organizing Genome: Principles of Genome Architecture and Function;,citation_author=Tom Misteli;,citation_publication_date=2020-10;,citation_cover_date=2020-10;,citation_year=2020;,citation_issue=1;,citation_doi=10.1016/j.cell.2020.09.014;,citation_issn=00928674;,citation_volume=183;,citation_language=en-US;,citation_journal_title=Cell;">
<meta name="citation_reference" content="citation_title=Git: A distributed version control system;,citation_author=Linus Torvalds;,citation_author=Junio C Hamano;,citation_publication_date=2005;,citation_cover_date=2005;,citation_year=2005;,citation_fulltext_html_url=https://git-scm.com;">
<meta name="citation_reference" content="citation_title=GitHub;,citation_author=GitHub, Inc.;,citation_publisher=https://github.com;">
<meta name="citation_reference" content="citation_title=Hic-spermatogenesis;,citation_author=Søren Jørgensen;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_publisher=https://github.com/munch-group/hic-spermatogenesis.git;">
<meta name="citation_reference" content="citation_title=Quarto;,citation_author=J. J. Allaire;,citation_author=Charles Teague;,citation_author=Carlos Scheidegger;,citation_author=Yihui Xie;,citation_author=Christophe Dervieux;,citation_publication_date=2024-02;,citation_cover_date=2024-02;,citation_year=2024;,citation_fulltext_html_url=https://github.com/quarto-dev/quarto-cli;,citation_doi=10.5281/zenodo.5960048;">
<meta name="citation_reference" content="citation_title=Anaconda software distribution;,citation_author=Anaconda Software Distribution;,citation_publication_date=2016-11;,citation_cover_date=2016-11;,citation_year=2016;,citation_fulltext_html_url=https://anaconda.com;,citation_publisher=Computer software;">
<meta name="citation_reference" content="citation_title=Gwf workflow manager;,citation_author=undefined GenomeDK;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_publisher=https://github.com/gwforg/gwf.git;">
<meta name="citation_reference" content="citation_title=Announcing Quarto, a new scientific and technical publishing system;,citation_abstract=Today we’re excited to announce Quarto, the next generation of R Markdown.;,citation_author=Posit Quarto Team;,citation_publication_date=2022-07;,citation_cover_date=2022-07;,citation_year=2022;,citation_language=en-US;,citation_journal_title=Posit;">
<meta name="citation_reference" content="citation_title=HiCschematic (cutout);,citation_author=WikiCommons:Prakrutiuday;,citation_publication_date=00;,citation_cover_date=00;,citation_year=0;,citation_fulltext_html_url=https://commons.wikimedia.org/wiki/File:HiCschematic.png;">
</head>

<body>

<header id="title-block-header" class="quarto-title-block default toc-left page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Chromatin Compartments and Selection on X</h1>
            <p class="subtitle lead">How Chromatin Compartments Align with Regions Under Selection in Primates</p>
          </div>

    
    <div class="quarto-title-meta-container">
      <div class="quarto-title-meta-column-start">
            <div class="quarto-title-meta-author">
          <div class="quarto-title-meta-heading">Authors</div>
          <div class="quarto-title-meta-heading">Affiliation</div>
          
                <div class="quarto-title-meta-contents">
            <p class="author">Søren Jørgensen <a href="mailto:201906763@post.au.dk" class="quarto-title-author-email"><i class="bi bi-envelope"></i></a> </p>
          </div>
                      <div class="quarto-title-meta-contents">
            <p class="author">Kasper Munch </p>
          </div>
                <div class="quarto-title-meta-contents">
                    <p class="affiliation">
                        Aarhus University, Denmark
                      </p>
                  </div>
                    </div>
        
        <div class="quarto-title-meta">

                      
                <div>
            <div class="quarto-title-meta-heading">Published</div>
            <div class="quarto-title-meta-contents">
              <p class="date">January 29, 2025</p>
            </div>
          </div>
          
                
              </div>
      </div>
      <div class="quarto-title-meta-column-end quarto-other-formats-target">
      <div class="quarto-alternate-formats"><div class="quarto-title-meta-heading">Other Formats</div><div class="quarto-title-meta-contents"><p><a href="index.pdf"><i class="bi bi-file-pdf"></i>PDF</a></p></div></div></div>
    </div>

    <div>
      <div class="abstract">
        <div class="block-title">Abstract</div>
        <p>The X chromosome is uniquely exposed to selective pressures due to the lack of a second copy in the hemizygous sex, leaving no buffer against deleterious mutations, and giving unique inheritance patterns. Combined with its high density of essential genes related to reproduction and brain function, this suggests the presence of biological mechanisms that safeguard the integrity of the X chromosome.</p>
        <p>In this study, the 3D chromatin architecture of the X chromosome in rhesus macaque (<em>Macaca mulata</em>) is investigated in the context of evolutionary pressures and genetic drivers. To ensure transparency and reproducibility, we adopt a comprehensive computational framework for publishing a version-controlled, fully reproducible analysis.</p>
        <p>We compare two Hi-C analysis frameworks, <em>HiCExplorer</em> and <em>cooler/cooltools</em> (Open2C), on a subset, finding Open2C to be most flexible. The ICE method (Iterative Correction and Eigendecomposition) was used to infer conventional and refined A/B compartments for fibroblast and four stages of spermatogenesis.</p>
        <p>We find 200 kbp transition-zones between A/B-compartments on the X chromosomes in both fibroblasts and round spermatids that align well with strong selective sweeps in humans (ECH-regions), but not with strong negative selection in baboons (<em>Papio</em> spp.). We find that most edges either overlap or are in significant proximity of each other when comparing regions under selection in human and baboons with A/B-compartments inferred Hi-C matrices at 100kb resolution and restricting eigendecomposition along the X chromosome to 10Mb windows. We discuss the biological meaning of these findings, where conserved chromatin features may help to retain non-advantageous alleles, hinting to the role of structural features aiding in genome evolution.</p>
      </div>
    </div>

    <div>
      <div class="keywords">
        <div class="block-title">Keywords</div>
        <p>Hi-C, Chromatin Compartments, Selection on X, Selfish gene drive</p>
      </div>
    </div>

    <div class="quarto-other-links-text-target">
    </div>  </div>
</header><div id="quarto-content" class="page-columns page-rows-contents page-layout-article toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction">Introduction</a>
  <ul class="collapse">
  <li><a href="#evolution-on-x" id="toc-evolution-on-x" class="nav-link" data-scroll-target="#evolution-on-x">Evolution on X</a>
  <ul class="collapse">
  <li><a href="#extended-common-haplotypes-ech-on-human-x" id="toc-extended-common-haplotypes-ech-on-human-x" class="nav-link" data-scroll-target="#extended-common-haplotypes-ech-on-human-x">Extended Common Haplotypes (ECH) on human X</a></li>
  <li><a href="#disproportionate-minor-parent-ancestry-on-baboon-x" id="toc-disproportionate-minor-parent-ancestry-on-baboon-x" class="nav-link" data-scroll-target="#disproportionate-minor-parent-ancestry-on-baboon-x">Disproportionate minor parent ancestry on baboon X</a></li>
  </ul></li>
  <li><a href="#chromatin-architecture" id="toc-chromatin-architecture" class="nav-link" data-scroll-target="#chromatin-architecture">Chromatin Architecture</a></li>
  <li><a href="#c-chromatin-conformation-capture" id="toc-c-chromatin-conformation-capture" class="nav-link" data-scroll-target="#c-chromatin-conformation-capture">3C: Chromatin Conformation Capture</a></li>
  <li><a href="#hi-c-high-throughput-3c" id="toc-hi-c-high-throughput-3c" class="nav-link" data-scroll-target="#hi-c-high-throughput-3c">Hi-C: High-Throughput 3C</a>
  <ul class="collapse">
  <li><a href="#hi-c-library-preparation" id="toc-hi-c-library-preparation" class="nav-link" data-scroll-target="#hi-c-library-preparation">Hi-C Library preparation</a></li>
  <li><a href="#hi-c-data-analysis" id="toc-hi-c-data-analysis" class="nav-link" data-scroll-target="#hi-c-data-analysis">Hi-C Data Analysis</a></li>
  </ul></li>
  <li><a href="#reproducibility-infrastructure" id="toc-reproducibility-infrastructure" class="nav-link" data-scroll-target="#reproducibility-infrastructure">Reproducibility Infrastructure</a>
  <ul class="collapse">
  <li><a href="#gwf-workflow-management-for-high-performance-computing-hpc" id="toc-gwf-workflow-management-for-high-performance-computing-hpc" class="nav-link" data-scroll-target="#gwf-workflow-management-for-high-performance-computing-hpc">GWF: workflow management for High-Performance Computing (HPC)</a></li>
  <li><a href="#project-initialization" id="toc-project-initialization" class="nav-link" data-scroll-target="#project-initialization">Project Initialization</a></li>
  <li><a href="#git-and-github" id="toc-git-and-github" class="nav-link" data-scroll-target="#git-and-github">git and GitHub</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#methods" id="toc-methods" class="nav-link" data-scroll-target="#methods">Methods</a>
  <ul class="collapse">
  <li><a href="#fetching-raw-data" id="toc-fetching-raw-data" class="nav-link" data-scroll-target="#fetching-raw-data">Fetching raw data</a>
  <ul class="collapse">
  <li><a href="#fetching-and-indexing-the-reference" id="toc-fetching-and-indexing-the-reference" class="nav-link" data-scroll-target="#fetching-and-indexing-the-reference">Fetching and indexing the reference</a></li>
  </ul></li>
  <li><a href="#hicexplorer-trials" id="toc-hicexplorer-trials" class="nav-link" data-scroll-target="#hicexplorer-trials">HiCExplorer trials</a></li>
  <li><a href="#open2c-pipeline" id="toc-open2c-pipeline" class="nav-link" data-scroll-target="#open2c-pipeline">Open2C pipeline</a></li>
  <li><a href="#genomic-intervals" id="toc-genomic-intervals" class="nav-link" data-scroll-target="#genomic-intervals">Genomic Intervals</a>
  <ul class="collapse">
  <li><a href="#full-regions-edges-and-limits" id="toc-full-regions-edges-and-limits" class="nav-link" data-scroll-target="#full-regions-edges-and-limits">Full Regions, Edges, and Limits</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#results" id="toc-results" class="nav-link" data-scroll-target="#results">Results</a>
  <ul class="collapse">
  <li><a href="#hicexplorer-trials-1" id="toc-hicexplorer-trials-1" class="nav-link" data-scroll-target="#hicexplorer-trials-1">HicExplorer Trials</a>
  <ul class="collapse">
  <li><a href="#quality-control" id="toc-quality-control" class="nav-link" data-scroll-target="#quality-control">Quality Control</a></li>
  <li><a href="#correction" id="toc-correction" class="nav-link" data-scroll-target="#correction">Correction</a></li>
  <li><a href="#eigenvectors" id="toc-eigenvectors" class="nav-link" data-scroll-target="#eigenvectors">Eigenvectors</a></li>
  </ul></li>
  <li><a href="#open2c-ecosystem" id="toc-open2c-ecosystem" class="nav-link" data-scroll-target="#open2c-ecosystem">Open2c ecosystem</a>
  <ul class="collapse">
  <li><a href="#quality-control-1" id="toc-quality-control-1" class="nav-link" data-scroll-target="#quality-control-1">Quality Control</a></li>
  <li><a href="#correction-1" id="toc-correction-1" class="nav-link" data-scroll-target="#correction-1">Correction</a></li>
  <li><a href="#sec-results-eigenvectors" id="toc-sec-results-eigenvectors" class="nav-link" data-scroll-target="#sec-results-eigenvectors">Compartments (Eigenvectors)</a></li>
  </ul></li>
  <li><a href="#comparing-genomic-intervals" id="toc-comparing-genomic-intervals" class="nav-link" data-scroll-target="#comparing-genomic-intervals">Comparing Genomic Intervals</a>
  <ul class="collapse">
  <li><a href="#compartment-edges-transition-zones" id="toc-compartment-edges-transition-zones" class="nav-link" data-scroll-target="#compartment-edges-transition-zones">Compartment Edges (transition zones)</a></li>
  <li><a href="#testing-against-baboon-regions-under-strong-selection" id="toc-testing-against-baboon-regions-under-strong-selection" class="nav-link" data-scroll-target="#testing-against-baboon-regions-under-strong-selection">Testing against baboon regions under strong selection</a></li>
  <li><a href="#testing-limits" id="toc-testing-limits" class="nav-link" data-scroll-target="#testing-limits">Testing limits</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#sec-discussion" id="toc-sec-discussion" class="nav-link" data-scroll-target="#sec-discussion">Discussion</a>
  <ul class="collapse">
  <li><a href="#on-reproducibility" id="toc-on-reproducibility" class="nav-link" data-scroll-target="#on-reproducibility">On Reproducibility</a></li>
  <li><a href="#on-methodology" id="toc-on-methodology" class="nav-link" data-scroll-target="#on-methodology">On Methodology</a></li>
  <li><a href="#on-results" id="toc-on-results" class="nav-link" data-scroll-target="#on-results">On Results</a>
  <ul class="collapse">
  <li><a href="#narrowing-the-parameter-space-for-testing" id="toc-narrowing-the-parameter-space-for-testing" class="nav-link" data-scroll-target="#narrowing-the-parameter-space-for-testing">Narrowing the parameter space for testing</a></li>
  <li><a href="#correlating-genomic-regions" id="toc-correlating-genomic-regions" class="nav-link" data-scroll-target="#correlating-genomic-regions">Correlating genomic regions</a></li>
  <li><a href="#gene-drive-selfishness-and-its-effect-on-selection" id="toc-gene-drive-selfishness-and-its-effect-on-selection" class="nav-link" data-scroll-target="#gene-drive-selfishness-and-its-effect-on-selection">Gene Drive, Selfishness and its Effect on Selection</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#conclusion-and-future-work" id="toc-conclusion-and-future-work" class="nav-link" data-scroll-target="#conclusion-and-future-work">Conclusion and future work</a></li>
  <li><a href="#bibliography" id="toc-bibliography" class="nav-link" data-scroll-target="#bibliography">Bibliography</a></li>
  <li><a href="#a-appendix" id="toc-a-appendix" class="nav-link" data-scroll-target="#a-appendix">A Appendix</a>
  <ul class="collapse">
  <li><a href="#data-availability" id="toc-data-availability" class="nav-link" data-scroll-target="#data-availability">Data Availability</a></li>
  <li><a href="#aknowledgements" id="toc-aknowledgements" class="nav-link" data-scroll-target="#aknowledgements">Aknowledgements</a>
  <ul class="collapse">
  <li><a href="#people" id="toc-people" class="nav-link" data-scroll-target="#people">People</a></li>
  <li><a href="#compute" id="toc-compute" class="nav-link" data-scroll-target="#compute">Compute</a></li>
  <li><a href="#use-of-generative-artificial-intelligence" id="toc-use-of-generative-artificial-intelligence" class="nav-link" data-scroll-target="#use-of-generative-artificial-intelligence">Use of Generative Artificial Intelligence</a></li>
  </ul></li>
  </ul></li>
  </ul>
</nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
</div>
<main class="content quarto-banner-title-block" id="quarto-document-content">



  


<section id="introduction" class="level1">
<h1>Introduction</h1>
<p>The X chromosome plays a unique role in evolution, being directly exposed to selection pressures due to its hemizygosity in males, leading to unique inheritance patterns. Regions of reduced diversity on the X chromosome could reflect the influence of strong evolutionary forces that are aided by structural features of the genome. Chromatin architecture, particularly the organization into A/B compartments, may provide insights into the functional basis of these patterns.</p>
<p>Here, I analyze the 3D chromatin structure of the X chromosome in rhesus macaque (<em>Macaca mulata</em>) using Hi-C data on the latest reference genome (rheMac10). The analysis extensively combines software tools that emphasize reproducibility, ensuring that the analyses are transparent, portable, and replicable.</p>
<section id="evolution-on-x" class="level2">
<h2 class="anchored" data-anchor-id="evolution-on-x">Evolution on X</h2>
<p>The production of gametes in a sexually reproducing organism is a highly complex process that involves numeruous elements. Spermatogenesis, the process of forming male gametes, involves four stages of differentiation from a germ cell through <em>spermatogonia</em>, <em>pachytene spermatocyte</em>, and <em>round spermatids</em> to <em>spermatozoa</em> <span class="citation" data-cites="wang2019reprogrammingmeioticchromatin">(<a href="#ref-wang2019reprogrammingmeioticchromatin" role="doc-biblioref">Wang et al. 2019</a>)</span>, and is the basis of male reproduction. The specialized cell division of meiosis neatly handles the pairing, recombination, and segregation of homologous chromosomes, thereby ensuring proper genetic distribution. A thorough understanding of the molecular steps of reproduction and how genetic material is inherited is essential in biology, bringing insight to areas such as speciation, population diversity, and infertility.</p>
<p>Sex chromosomes differ from autosomes in several ways, primarily due to their unique inheritance patterns and copy number. The Y chromosome is present only in males, with a single copy per individual. The X chromosome, on the other hand, has a more complex inheritance pattern: males have one copy, while females have two. As a result, X chromosomes spend two-thirds of their time in females and only one-third in males. This skewed ratio influences the dynamics of selection on the X chromosome. Furthermore, in males, the single-copy nature of the X chromosome (hemizygosity) means that any mutations or loss of function are directly exposed, as there is no second copy to compensate. This concept also underpins Haldane’s rule <span class="citation" data-cites="haldane1922sexratiounisexual">(<a href="#ref-haldane1922sexratiounisexual" role="doc-biblioref">Haldane 1922</a>)</span>, which states that in hybrids of two species, the heterogametic sex (e.g., XY in mammals, ZW in birds) is the first to exhibit reduced fitness, such as sterility, or to disappear entirely. Even a century later, the exact reasons for this phenomenon remain debated, with several hypotheses proposed. These include:</p>
<ul>
<li>Y-incompatibility: The Y chromosome must remain compatible with the X chromosome or autosomes.</li>
<li>Dosage compensation: Hybridization may disrupt crucial dosage compensation mechanisms in heterogametic individuals.</li>
<li>Dominance: Recessive deleterious alleles may cause sterility when expressed in the heterogametic sex.</li>
<li>Faster-male evolution: Male reproductive genes may evolve more rapidly than female ones, leading to incompatibilities.</li>
<li>Faster-X evolution: X-linked loci may diverge more quickly than autosomal loci, contributing to hybrid sterility.</li>
<li>Meiotic drive: Conflicts between drivers and suppressors on sex chromosomes may result in sterility.</li>
</ul>
<p>These hypotheses highlight the intricate interplay between sex chromosomes, selection, and hybridization <span class="citation" data-cites="cowell2023100yearshaldanes">(<a href="#ref-cowell2023100yearshaldanes" role="doc-biblioref">Cowell 2023</a>)</span>. Furthermore, although this phenomenon is observed across various taxa and even kingdoms, the underlying explanations differ, and there is no universal consensus for all species. In many cases, multiple mechanisms from the listed explanations are thought to act in concert <span class="citation" data-cites="lindholm2016ecologyevolutionarydynamics">(<a href="#ref-lindholm2016ecologyevolutionarydynamics" role="doc-biblioref">Lindholm et al. 2016</a>)</span>. The complexity of selection on the X chromosome remains an area of active research, with numerous studies suggesting strong selection pressures on the X chromosome across primates. This topic is explored in greater detail in the following sections.</p>
<section id="extended-common-haplotypes-ech-on-human-x" class="level3">
<h3 class="anchored" data-anchor-id="extended-common-haplotypes-ech-on-human-x">Extended Common Haplotypes (ECH) on human X</h3>
<p>Incomplete lineage sorting (ILS) occur when all lineages in a population are not completely sorted between two speciation events, resulting in a gene tree incongruent with the species tree <span class="citation" data-cites="mailund2014lineagesortingapes">(<a href="#ref-mailund2014lineagesortingapes" role="doc-biblioref">Mailund, Munch, and Schierup 2014</a>)</span>. Briefly, it complicates phylogenetic inference as it implies that divergence time does represent speciation time. But, by solving the incongruency with a maximum likelihood approach <span class="citation" data-cites="mailund2014lineagesortingapes">(see <a href="#ref-mailund2014lineagesortingapes" role="doc-biblioref">Mailund, Munch, and Schierup 2014</a>)</span>, the relation between the ILS proportion, <span class="math inline">\(p\)</span>, time between two speciation events, <span class="math inline">\(\Delta\tau\)</span>, and effective population size, <span class="math inline">\(N_e\)</span>, is given by the formula</p>
<p><span class="math display">\[
N_e = \frac{\Delta\tau}{2} \ln \frac{2p}{3}.
\]</span></p>
<p>Then, by comparing the observed (local) ILS with the expected (e.g.&nbsp;a genomic average), a reduction in <span class="math inline">\(N_e\)</span> can be used to infer selection. Additionally, as selective sweeps force lineages to coalesce, sweeps will also cause a reduction in ILS. <span class="citation" data-cites="dutheil2015strongselectivesweeps">Dutheil et al. (<a href="#ref-dutheil2015strongselectivesweeps" role="doc-biblioref">2015</a>)</span> found that the human X chromosome had reduced ILS compared to autosomes on a third of its sequence, fully explaining the low divergence between human and chimpanzees. Reduced ILS co-occur with reduced diversity across the X chromosomes of great apes, including human. Additionally, Neanderthal introgression was depleted in the same regions, and the authors suggest that they are a target of selection, as they rule background selection to be responsible for reduced ILS.</p>
<p>Low-diversity regions on the human X chromosome that also overlap with reduced human-chimp ILS have been hypothesized to arise from selective sweeps between 55,000 (archaic introgression after out-of-Africa event) to 45,000 (Ust’-Ishim man) years ago <span class="citation" data-cites="skov2023extraordinaryselectionhuman">(<a href="#ref-skov2023extraordinaryselectionhuman" role="doc-biblioref">Skov et al. 2023</a>)</span>. The authors define the time-span by comparing haplotypes of unadmixed African genomes with admixed non-African genomes. Here, they locate megabase-spanning regions on the X chromosome where a large proportion of the non-African population have reduced archaic introgression, hypothesizing that selective sweeps have created these extended common haplotypes (ECHs). The ECHs span 11% of the X chromosome and are shared across all non-African populations. Similar to the low-ILS regions, ECHs exhibit a complete absence of archaic introgression (Neanderthal and Denisovan). Notably, the largest continuous ECH spans 1.8 Mb—more than double the size of the selective sweep associated with the lactase persistence gene, which represents strongest selective sweeps documented in humans <span class="citation" data-cites="skov2023extraordinaryselectionhuman bersaglieri2004geneticsignaturesstrong">(<a href="#ref-skov2023extraordinaryselectionhuman" role="doc-biblioref">Skov et al. 2023</a>; <a href="#ref-bersaglieri2004geneticsignaturesstrong" role="doc-biblioref">Bersaglieri et al. 2004</a>)</span>. If such strong selection underlies the observed low diversity, it suggests that factors beyond fitness effects alone may be contributing to this pattern. This raises the possibility that the observed patterns of low diversity may result not only from strong selective sweeps but also from several factors in combination. Structural elements, such as physical linkage between the regions within ECHs, may facilitate coordinated selection, maintaining reduced diversity over extended genomic stretches. Additionally, sex-specific pressures, such as those linked to the unique transmission and evolutionary dynamics of the X chromosome, could contribute to this pattern. Further investigation is needed to disentangle the roles of these mechanisms in shaping the genomic landscape of the X chromosome.</p>
</section>
<section id="disproportionate-minor-parent-ancestry-on-baboon-x" class="level3">
<h3 class="anchored" data-anchor-id="disproportionate-minor-parent-ancestry-on-baboon-x">Disproportionate minor parent ancestry on baboon X</h3>
<p>As mentioned above, low ILS and reduced diversity is consistent across all great apes, and thus the regions have been investigated in baboons as well. <span class="citation" data-cites="sorensen2023genomewidecoancestryreveals">Sørensen et al. (<a href="#ref-sorensen2023genomewidecoancestryreveals" role="doc-biblioref">2023</a>)</span> have mapped the interspecies gene flow between baboon genomes in their evolutionary history, which revealed male-driven admixture patterns in six different baboon species. The authors infer a male-driven admixture from mismatching phylogenies of the mitochondrial DNA and the nuclear DNA. This indicates a role of nuclear swamping, where the nuclear DNA from one species progressively replaces that of another through repeated hybridization and backcrossing, while the mitochondrial DNA, inherited maternally, remains unaffected. This pattern suggests that male-driven gene flow plays a significant role in shaping the genomic landscape of these regions, potentially impacting diversity and co-ancestry across species. <span class="citation" data-cites="sorensen2023genomewidecoancestryreveals">Sørensen et al. (<a href="#ref-sorensen2023genomewidecoancestryreveals" role="doc-biblioref">2023</a>)</span> compare the ratio of admixture between chromosome 8/X across the populations, which support the same pattern. Recently, in a yet unpublished analysis by <span class="citation" data-cites="munch2024munchgroup">Munch (<a href="#ref-munch2024munchgroup" role="doc-biblioref">2024</a>)</span>, exceptionally large regions (up to 6 Mb, averaging 1 Mb) were identified on X in olive baboon (<em>Papio anubis</em>), where either all individuals had <em>olive</em> ancestry, or 95% of the individuals had <em>hamadryas</em> ancestry, significantly diverging from the background proportions, suggesting that these regions may be under strong selective pressures, are influenced by structural genomic features that limit recombination, or potentially both. Such patterns suggest that certain chromosomal regions are either preserved due to adaptive advantages or subjected to sex-biased gene flow, further reinforcing the impact of male-driven admixture on shaping genomic variation. These findings highlight the complexity of admixture dynamics and their potential to create localized genomic signatures that deviate from overall ancestry proportions.</p>
<p><span class="citation" data-cites="sorensen2023genomewidecoancestryreveals">Sørensen et al. (<a href="#ref-sorensen2023genomewidecoancestryreveals" role="doc-biblioref">2023</a>)</span> highlights the importance of studying these populations, as they serve as models for understanding genomic separation in archaic hominin lineages while inhabiting comparable geographic ranges. Studying closely related, extant primate species provides a unique opportunity to gain insight into how <em>hominin</em> (human, Neanderthal, Denisovan) phylogenies were formed. By analyzing the intricate effects of population structure, migration, and selection in these species, we can draw parallels and infer events in our own archaic evolutionary history. See <a href="#fig-introduce-selected-regions" class="quarto-xref">Figure&nbsp;1</a> for a genomic overview of the abovementioned sets of regions.</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/07_various_plotting.ipynb" data-notebook-title="Various Comparisons" data-notebook-cellid="cell-fig-introduce-selected-regions">
<div id="cell-fig-introduce-selected-regions" class="cell" data-execution_count="33">
<div class="cell-output cell-output-display">
<div id="fig-introduce-selected-regions" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-introduce-selected-regions-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-introduce-selected-regions-output-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-1" title="Figure&nbsp;1: Visual representation of the selected regions for the baboon comparison. Shown are the ECH90 (top), high-olive (middle) and high-hama (bottom) regions, the basis of the comparison of genomic intervals in this analysis. All coordinates are lifted to the rheMac10 genome."><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-introduce-selected-regions-output-1.png" width="498" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-introduce-selected-regions-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1: Visual representation of the selected regions for the baboon comparison. Shown are the ECH90 (top), high-olive (middle) and high-hama (bottom) regions, the basis of the comparison of genomic intervals in this analysis. All coordinates are lifted to the rheMac10 genome.
</figcaption>
</figure>
</div>
</div>
</div>
</div>
</section>
</section>
<section id="chromatin-architecture" class="level2">
<h2 class="anchored" data-anchor-id="chromatin-architecture">Chromatin Architecture</h2>
<p>The specific regions under strong selection identified across primate species, including humans and baboons, are all located on the X chromosome and span megabase-scale genomic regions. As mentioned above, alternative mechanisms, such as the chromosomal architecture and its rearrangements, could play critical roles in facilitating, regulating, or buffering these regions from selective forces.</p>
<div id="fig-chromatin-architecture-overview" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-chromatin-architecture-overview-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="illustrations/chromatin-intro.jpg" class="lightbox" data-gallery="quarto-lightbox-gallery-2" title="Figure&nbsp;2: DNA is organized at multiple hierarchical structures. Figure from @misteli2020selforganizinggenomeprinciples."><img src="illustrations/chromatin-intro.jpg" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-chromatin-architecture-overview-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2: DNA is organized at multiple hierarchical structures. Figure from <span class="citation" data-cites="misteli2020selforganizinggenomeprinciples">Misteli (<a href="#ref-misteli2020selforganizinggenomeprinciples" role="doc-biblioref">2020</a>)</span>.
</figcaption>
</figure>
</div>
<p>Therefore, understanding the genome organization and variation could gain insights regarding mechanisms of selection as well. Chromatin has long since been implicated in gene regulation and the functional state of the cell <span class="citation" data-cites="lieberman-aiden2009comprehensivemappinglongrange">(<a href="#ref-lieberman-aiden2009comprehensivemappinglongrange" role="doc-biblioref">Lieberman-Aiden et al. 2009</a>)</span>, as the three-dimensional structure of the chromosome can bring distant regions in close proximity, and disrupting the organization can lead to development abnormalities <span class="citation" data-cites="dixon2015chromatinarchitecturereorganization">(<a href="#ref-dixon2015chromatinarchitecturereorganization" role="doc-biblioref">Dixon et al. 2015</a>)</span>. Chromatin is hierarchically compartmentalized, meaning multiple orders of organizaion can be nested under each other (see <a href="#fig-chromatin-architecture-overview" class="quarto-xref">Figure&nbsp;2</a>). At the large end, compartments can span multiple megabases <span class="citation" data-cites="lieberman-aiden2009comprehensivemappinglongrange">(<a href="#ref-lieberman-aiden2009comprehensivemappinglongrange" role="doc-biblioref">Lieberman-Aiden et al. 2009</a>)</span>, and in the small end as little as 500 base pairs have been showed to aid subgenic structural organization. Even at sub-megabase scales, at the level of topologically associating domains (TADs) or chromatin loops <span class="citation" data-cites="ramirez2018highresolutiontadsreveal zuo2021stageresolvedhicanalyses">(<a href="#ref-ramirez2018highresolutiontadsreveal" role="doc-biblioref">Ramírez et al. 2018</a>; <a href="#ref-zuo2021stageresolvedhicanalyses" role="doc-biblioref">Zuo et al. 2021</a>)</span>, the structure helps segregate regulatory elements and ensure proper function.</p>
<p>Analyzing 3D chromatin data from rhesus macaques, a widely used primate in experimental studies, could answer the question of whether regions under selection are intertwined with chromosomal organizational features. Studying the chromosomal architecture through gametogenesis, particularly spermatogenesis, could provide valuable insights about meiotic recombination and its relationship to genome organization. If they correlate with regions under selection across primates, we have new mystery to explain.</p>
<p><span class="citation" data-cites="wang2019reprogrammingmeioticchromatin">Wang et al. (<a href="#ref-wang2019reprogrammingmeioticchromatin" role="doc-biblioref">2019</a>)</span> found that through the stages of spermatogenesis, chromatin compartmentalization undergo massive reprogramming, going from megabase-spanning compartments through smaller, <em>refined</em> compartments and back to the megabase compartments. The reprogramming is conserved between rhesus macaque and mouse, indicating a very important feature of the organization, but the author does not test whether genomic positions of the compartments are conserved between species. Extending their analysis of chromatin compartmentalization through spermatogenesis could make an obvious starting point for investigating the relationship between yet another set of megabase-spanning regions and those of strong selection identified above.</p>
</section>
<section id="c-chromatin-conformation-capture" class="level2">
<h2 class="anchored" data-anchor-id="c-chromatin-conformation-capture">3C: Chromatin Conformation Capture</h2>
<p>The first method developed to capture long-range interactions between pairs of loci was 3C, which uses spatially constrained ligation followed by locus-specific PCR <span class="citation" data-cites="lieberman-aiden2009comprehensivemappinglongrange">(<a href="#ref-lieberman-aiden2009comprehensivemappinglongrange" role="doc-biblioref">Lieberman-Aiden et al. 2009</a>)</span>. Subsequent advancements introduced inverse PCR (4C) and multiplexed ligation-mediated amplification (5C). A limitation common to these methods is their inability to perform genome-wide, unbiased analyses, as they require predefined pairs of target loci.</p>
<p>DNA can be organized into different structural levels. 3C focuses on identifying the higher-order organization within the nucleus, such as when the 30 nm chromatin fiber folds into loops, Topologically Associating Domains (TADs), and chromatin compartments.</p>
</section>
<section id="hi-c-high-throughput-3c" class="level2">
<h2 class="anchored" data-anchor-id="hi-c-high-throughput-3c">Hi-C: High-Throughput 3C</h2>
<p>The introduction of the Hi-C (high-throughput 3C) method <span class="citation" data-cites="lieberman-aiden2009comprehensivemappinglongrange">(<a href="#ref-lieberman-aiden2009comprehensivemappinglongrange" role="doc-biblioref">Lieberman-Aiden et al. 2009</a>)</span> opened new possibilities for exploring the three-dimensional organization of the genome, as deep-sequencing technologies combined with high-performance computing allow for genome-wide analysis that is unbiased. <span class="citation" data-cites="lieberman-aiden2009comprehensivemappinglongrange">Lieberman-Aiden et al. (<a href="#ref-lieberman-aiden2009comprehensivemappinglongrange" role="doc-biblioref">2009</a>)</span> show that when combining spatially constrained ligation with deep parallel sequencing and subsequent analysis, we can infer distinct chromosomal territories as intrachromosomal contacts were significantly less abundant than interchromosomal contacts. The method also confirmed the spatial separation of two chromatin conformations in its active (open) state or inactive (closed) state, as they were significantly correlated with distances measured by fluorescence <em>in-situ</em> hybridisation (FISH). The two comformations were termed <em>A</em>- and <em>B</em>-compartments, respectively, and A defined to positively correlate with gene density. Finally, they showed that loci are physically more proximal when they belong to the same compartment implying that Hi-C reads serves well as a proxy for distance. Later, several smaller-order domains were inferred with the same method, such as topologically associating domains (TADs) and chromatin loops. Here, we narrow our focus on the largest of the structures, <em>compartments</em>, that is known to determine availability to transcription factors, thus making an <em>A</em> compartment <em>active</em>—and the <em>B</em> compartment <em>inactive</em>.</p>
<section id="hi-c-library-preparation" class="level3">
<h3 class="anchored" data-anchor-id="hi-c-library-preparation">Hi-C Library preparation</h3>
<p>A specialized protocol for preparing the DNA library is necessary <span class="citation" data-cites="lieberman-aiden2009comprehensivemappinglongrange">(<a href="#ref-lieberman-aiden2009comprehensivemappinglongrange" role="doc-biblioref">Lieberman-Aiden et al. 2009, fig. 1a</a>)</span>. Briefly, formaldehyde is used to crosslink spatially adjecent chromatin. Restriction enzyme <em>HindIII</em> is used to digest the crosslinked chromatin, leaving sticky ends, <code>5-AGCT-3</code>, that are filled and biotinylated with a polymerase (using either biotinylated A, G, C, or T). The strands are ligated in highly dilute conditions, which is favoring the ligation of the two crosslinked strands, forming chimeric, biotinylated strands. Upon ligation, the restriction site is lost as a biotinylated <code>5-CTAG-3</code> site (also referred to as the <em>ligation junction</em>) is formed. Lastly, the ligation junctions are isolated with streptavidin beads and sequenced as a paired-end library.</p>
<p>To be able to create stage-resolved Hi-C library of spermatogenetis, several steps have to be performed on the samples before crosslinking. First, the samples have to be treated immediately after harvesting to ensure viable cells. Secondly, the samples have to be purified to accurately represent each stage og spermatogenesis. Specifically, the data for this project <span class="citation" data-cites="wang2019reprogrammingmeioticchromatin">(<a href="#ref-wang2019reprogrammingmeioticchromatin" role="doc-biblioref">Wang et al. 2019</a>, acc. GSE109344)</span> preluded the library preparation protocol by sedimentation-based cell sorting to separate live spermatogenic cells into different stages of differentiation, namely spermatogonia, pachytene spermatocyte, round spermatid, and spermatozoa. Then, the cells were fixed in their respective state before crosslinking. The authors use their own derived method for library preparation, termed small-scale <em>in-situ</em> Hi-C, allegedly producing a high-quality Hi-C libary from as little as 500 cells (capturing the variance of millions of cells).</p>
<p>Initially, Hi-C library preparation was designed to generate molecules with only a single ligation site in each, but with advancements in sequencing technology (‘short-reads’ can now span several hundreds of base pairs) and the shift to more frequently cutting restriction enzymes for higher resolution results in multiple ligation events per sequenced molecule <span class="citation" data-cites="open2c2024pairtoolssequencingdata">(<a href="#ref-open2c2024pairtoolssequencingdata" role="doc-biblioref">Open2C et al. 2024</a>)</span>, which is adressed in the section below.</p>
</section>
<section id="hi-c-data-analysis" class="level3">
<h3 class="anchored" data-anchor-id="hi-c-data-analysis">Hi-C Data Analysis</h3>
<p>The analysis of the read-pairs of a Hi-C library is divided into several smaller tasks, see <a href="#fig-hic-analysis-flow" class="quarto-xref">Figure&nbsp;3</a>.</p>
<div id="fig-hic-analysis-flow" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-hic-analysis-flow-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="illustrations/fig-hic-data-analysis.png" class="lightbox" data-gallery="quarto-lightbox-gallery-3" title="Figure&nbsp;3: A simplified pipeline for Hi-C data analysis: from raw reads to a Hi-C interaction matrix. See Table&nbsp;1 for details on ligation events."><img src="illustrations/fig-hic-data-analysis.png" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-hic-analysis-flow-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3: A simplified pipeline for Hi-C data analysis: from raw reads to a Hi-C interaction matrix. See <a href="#tbl-ligation-events" class="quarto-xref">Table&nbsp;1</a> for details on ligation events.
</figcaption>
</figure>
</div>
<p>The reads must be aligned to the reference in such a way that the <em>intentional</em> chimeric read-pairs (as per above-mentioned protocol) are rescued, and <em>unintentional</em> read-pairs are discarded. That is, we must make sure they represent ligation junction of adjecent chromatin segments, and not technical artefacts or unintentional (random) fusions of unrelated DNA.</p>
<section id="aligning-the-hi-c-reads" class="level4">
<h4 class="anchored" data-anchor-id="aligning-the-hi-c-reads">Aligning the Hi-C reads</h4>
<p>The main difference between Hi-C libraries and standard paired-end libraries is the high fraction of chimeric reads in Hi-C. As a contact pair is crosslinked and ligated before sequencing, chimeric reads occur as a feature, and standard mapping techniques seeks to filter out this type of reads <span class="citation" data-cites="lajoie2015hitchhikersguidehic">(<a href="#ref-lajoie2015hitchhikersguidehic" role="doc-biblioref">Lajoie, Dekker, and Kaplan 2015</a>)</span>. Thus, we need specialized tools for rescuing chimeric reads. That said, we have to be cautious distinguishing the intended chimerism for Hi-C and that of technical artefacts. Any software for local alignment can be used for aligning reads from a Hi-C library. However, one should make sure to disable paired-end rescue mode if possible, otherwise each read in a pair (each mate) should be aligned separately <span class="citation" data-cites="lajoie2015hitchhikersguidehic">(<a href="#ref-lajoie2015hitchhikersguidehic" role="doc-biblioref">Lajoie, Dekker, and Kaplan 2015</a>)</span>. This removes the assumption that the distance between mates fits a known distribution because the genomic sequences originate from a continuous DNA-fragment. For example, the <em>bwa-mem</em> <span class="citation" data-cites="li2013aligningsequencereadsclone">(<a href="#ref-li2013aligningsequencereadsclone" role="doc-biblioref">Li 2013</a>)</span> implementation of this (the <code>-P</code> option) activates the Smith-Waterman algorithm to rescue missing hits, but disables the search of hits that fit a ‘proper’ pair. After alignment, each read is typically assigned to the nearest restriction fragment to enable categorization of pairs into different categories.</p>
<p>Interestingly, this last step is not included by default in <em>pairtools</em>, as <span class="citation" data-cites="open2c2024pairtoolssequencingdata">Open2C et al. (<a href="#ref-open2c2024pairtoolssequencingdata" role="doc-biblioref">2024</a>)</span> observe very similar statistical properties on pairs that are either close or distant from the nearest restriction site. Thus, restriction fragment filters are not needed, and instead, a simple filter is applied against short-distance pairs that is automatically calibrated.</p>
</section>
<section id="identifying-and-storing-valid-hi-c-pairs" class="level4">
<h4 class="anchored" data-anchor-id="identifying-and-storing-valid-hi-c-pairs">Identifying and Storing Valid Hi-C Pairs</h4>
<p>One should be cautious when filtering invalid from valid pairs, as they are not easily distinguished. A ligation event will be categorized into one of five categories (see <a href="#tbl-ligation-events" class="quarto-xref">Table&nbsp;1</a>): <em>dangling-end</em>, <em>self-circle</em>, <em>weird</em>, <em>intrachromosomal</em> (cis), and <em>interchromosomal</em> (trans) <span class="citation" data-cites="bicciato2022hicdataanalysis">(<a href="#ref-bicciato2022hicdataanalysis" role="doc-biblioref">Bicciato and Ferrari 2022</a>, Ch. 1)</span>. Either <em>dangling-end</em> or <em>self-circle</em> events are reported if a read-pair maps to the same restriction fragment depending on the orientation, and deemed uninformative <span class="citation" data-cites="lajoie2015hitchhikersguidehic">(<a href="#ref-lajoie2015hitchhikersguidehic" role="doc-biblioref">Lajoie, Dekker, and Kaplan 2015</a>)</span>. Usually, <em>weird</em> events are demeed uninformative as well, as it is challenging to distinguish a sequencing error from the result of a diploid fragment. PCR duplicates should be discarded as well, having either identical genomic sequence, or sharing exact 5’ alignment positions of the pair <span class="citation" data-cites="lajoie2015hitchhikersguidehic bicciato2022hicdataanalysis">(<a href="#ref-lajoie2015hitchhikersguidehic" role="doc-biblioref">Lajoie, Dekker, and Kaplan 2015</a>; <a href="#ref-bicciato2022hicdataanalysis" role="doc-biblioref">Bicciato and Ferrari 2022</a>, Ch. 1)</span>. The probability that such pairs are valid (i.e.&nbsp;there are multiple of the same pairs) is very low. We also have to distinguish between molecules with only a single ligation event (one-way contact) or multiple ligation events (multi-way contacts). For that, a descision should be made on whether to 1) discard molecules with multiple ligations, 2) report one of the ligations (e.g.&nbsp;the 5’-most in both directions), or 3) report all events on a molecule.</p>
<div id="tbl-ligation-events" class="striped quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-ligation-events-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;1: Five categories of ligation events and a short explanation. <em>Hi-C Data Analysis: Methods and Protocols Ch. 1</em>.
</figcaption>
<div aria-describedby="tbl-ligation-events-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<table class="table-striped caption-top table">
<colgroup>
<col style="width: 20%">
<col style="width: 79%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Event name</th>
<th style="text-align: left;">Explanation</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Dangling-end</td>
<td style="text-align: left;">Non-digested collinear fragments. Fraction can be high.</td>
</tr>
<tr class="even">
<td style="text-align: left;">Self-circle</td>
<td style="text-align: left;">Collinear fragment(s) have circularized. Very low fraction could indicate unsuccesful ligation.</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Weird</td>
<td style="text-align: left;">Mates have the same orientation on the reference. Is not possible with single copy fragment. Either sequencing errors or diploid fragments<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a>.</td>
</tr>
<tr class="even">
<td style="text-align: left;">Cis</td>
<td style="text-align: left;">Pairs from the same chromosome (intrachromosomal)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Trans</td>
<td style="text-align: left;">Pairs from distinct chromosomes (interchromosomal)</td>
</tr>
</tbody>
</table>
</div>
</figure>
</div>
</section>
<section id="sec-matrix-qc" class="level4">
<h4 class="anchored" data-anchor-id="sec-matrix-qc">Quality Control and Interaction Matrices</h4>
<p>To determine the quality of the Hi-C library, most tools generate quality control log files at some point during the filtering steps, which can then be aggregated and analyzed (with e.g.&nbsp;MultiQC <span class="citation" data-cites="ewels2016multiqcsummarizeanalysis">Ewels et al. (<a href="#ref-ewels2016multiqcsummarizeanalysis" role="doc-biblioref">2016</a>)</span>). The ratios between the different ligation events can be informative about the quality of the Hi-C library. Here, both the distribution of discarded reads across categories, as well as the ratios between <em>cis</em>/<em>trans</em> interactions for a certain organism provide information about the library. For example, the biases of different aligners might be captured by comparing the reason why reads are discarded between two different aligners, as well as whether or not there is a preference of <em>cis</em> or <em>trans</em> in an aligner itself. This allows for evaluating the mapping parameters as well as the filters applied downstream. Additionally, <span class="math inline">\(P(s)\)</span>, the contact probability as a function of genomic separation can be inspected as it should decay with increasing distance. The <em>trans</em>/<em>cis</em>-ratio can sometimes be a good indicator of the noise level in the library, and additionally, the level of random ligation events can be quantified by counting the number of <em>trans</em> events occurring to mitochondrial genome. They should not occur naturally, as the mitochondrial genome is separated from the DNA in the nucleus. This method has some pitfalls that should be controlled for; some parts of the mitochondrial genome can be integrated into the host genome, and mitochondrial count may differ between cell-stages.</p>
<p>Typically, a filter against low mapping quality is applied on the data before constructing the interaction matrix (Hi-C matrix), and a conventional threshold is <span class="math inline">\(mapq &lt; 30\)</span> <span class="citation" data-cites="bicciato2022hicdataanalysis">(<a href="#ref-bicciato2022hicdataanalysis" role="doc-biblioref">Bicciato and Ferrari 2022</a>)</span>. However, a considerable amount of reads do not pass that threshold, and thus we risk discarding potential valid information and should make sure to have enough data. Consequently, <em>HicExplorer</em> defaults to a lower threshold (<span class="math inline">\(mapq &lt; 15\)</span>), and <em>pairtools</em> enforces no filter by default, but recommends setting this manually (starting at <span class="math inline">\(mapq &lt; 30\)</span>).</p>
<p>A Hi-C interaction matrix simply maps the frequency of interactions between genomic positions in a sample. The maximum resolution of a Hi-C matrix is defined by the restriction enzyme, where the size of the restriction site (probabilistically) determines average space between each cut. With a 4 bp restriction site, the fragments will average <span class="math inline">\(4^4 = 256 bp\)</span> and similarly <span class="math inline">\(4^6 = 4096 bp\)</span> for a 6 bp restriction site. This leads to ~12,000,000 and ~800,000 fragments, respectively. Very deep sequencing is required to achieve enough coverage to analyze the interaction matrix at the restriction fragment resolution, but, usually, such high resolution is not required. Therefore, it is common practice to bin the genome into fixed bin sizes, which also enables a more efficient handling of the data if the full resolution is not needed (e.g.&nbsp;when plotting large regions such as a whole chromosome). The conventional format to store a Hi-C matrix, consisting of large multidimensional arrays, is HDF5. Each HDF5 file can store all resolutions and metadata about the sample, resolutions typically ranging from 10kb to 1Mb. Typically, the stored resolutions should be multiples of the chosen base-resolution, as the lower resolutions are constructed by recursive binning of the base resolution. <em>cooler</em> <span class="citation" data-cites="abdennur2020coolerscalablestorage">(<a href="#ref-abdennur2020coolerscalablestorage" role="doc-biblioref">Abdennur and Mirny 2020</a>)</span> neatly offers efficient storage with sparse, upper-triangle symmetric matrices and naming-conventions of the groups in their <em>.h5</em>-based file format, <em>.cool</em>, and they provide a Python class <code>Cooler</code> as well for efficiently fetching and manipulating the matrices in Python.</p>
</section>
<section id="inferring-from-the-matrix-calling-compartments" class="level4">
<h4 class="anchored" data-anchor-id="inferring-from-the-matrix-calling-compartments">Inferring from the matrix (Calling Compartments)</h4>
<p>The raw frequency matrices are generally not very informative, as the contact frequencies vary greatly between bins and contain biases in addition to the <span class="math inline">\(P(s)\)</span> decay, which results in a diagonal-heavy matrix with high amount of noise the further we travel from the diagonal. Therefore, to analyze the three-dimensional structure of the chromatin, a method for correcting (or balancing) the raw Hi-C matrix has to be applied. It is unadvisable to correct low-count bins as it will greatly increase the noise, or to correct very noisy bins, or very high-count bins. Therefore, some bin-level filters are applied before balancing <span class="citation" data-cites="lajoie2015hitchhikersguidehic">(<a href="#ref-lajoie2015hitchhikersguidehic" role="doc-biblioref">Lajoie, Dekker, and Kaplan 2015</a>)</span>;</p>
<ul>
<li>Low-count bins are detected by comparing bin sums to the distribution of bin sums with a percentile cutoff,</li>
<li>Noisy bins are detected by comparing bin variance to the variance distribution of all bins (and percentile cutoff), and</li>
<li>Outlier point-interactions are removed (a top-percentile of bin-bin interactions)</li>
</ul>
<p>Iterative Correction and Eigendecomposition (ICE) <span class="citation" data-cites="imakaev2012iterativecorrectionhic">(<a href="#ref-imakaev2012iterativecorrectionhic" role="doc-biblioref">Imakaev et al. 2012</a>)</span> is a widely used method for correcting multiplicative biases in Hi-C data. ICE operates on the assumption that all loci should have roughly equal visibility across the genome, meaning that the total number of interactions per locus should be uniform. It corrects the raw interaction matrix by iteratively scaling rows and columns until they converge to a consistent coverage. By leveraging pairwise and genome-wide interaction data, ICE calculates a set of biases for each locus and normalizes the interaction frequencies accordingly. This process results in a corrected interaction matrix with uniform coverage, smoother transitions, and reduced visibility-related biases. Notably, ICE does not distinguish between different sources of bias, instead calculating a collective bias for each locus. <span class="citation" data-cites="imakaev2012iterativecorrectionhic">Imakaev et al. (<a href="#ref-imakaev2012iterativecorrectionhic" role="doc-biblioref">2012</a>)</span> show that <em>known</em> biases are factorizable by comparing their results to predictions of restriction fragment biases, GC content, and mappability from a computationally intensive probabilistic approach. By showing that the product of those known biases explain <span class="math inline">\(&gt;99.99%\)</span> of the variability in their bias estimation, they argue that both known and unknown biases will be captured with their iterative correction method (also denoted <em>matrix balancing</em>).</p>
<p>Even with a binned, filtered, and balanced matrix, we are still left with the challenge of translating the matrix into biologically relevant inferations. Importantly, we have to remember that the matrix arise from a collection of cells and that the interaction frequency cannot be translated to a fraction of cells. Additionally, the effect from averaging interaction patterns can cause both individual patterns to be burried and the average pattern to show a pattern that does not exist in any of the single cells. Therefore, when pooling matrices one must make sure that the samples are as similar as possible (e.g.&nbsp;the same differentiation stage and so on). We can also not distinguish interactions that either co-occur in the same cell or ones that are mutually exclusive. Lastly, the way interaction patterns are defined poses a challenge; we define the chromatin compartments to be the output of a method, the ‘E’ in ‘ICE’, eigendecomposition, not as a specific pattern that we can explicitly search for. Although experimentally verified to tightly correlate with chromatin states <span class="citation" data-cites="lieberman-aiden2009comprehensivemappinglongrange">(<a href="#ref-lieberman-aiden2009comprehensivemappinglongrange" role="doc-biblioref">Lieberman-Aiden et al. 2009</a>)</span>, the inferred compartments vary with different methods of calculating the eigenvector, as dicussed in <a href="#sec-methods-eigendecomposition" class="quarto-xref">Section&nbsp;2.3.0.9</a> and <a href="#sec-results-eigenvectors" class="quarto-xref">Section&nbsp;3.2.3</a>. To further complicate the challenge, interaction patterns on different scales co-exist and are difficult to disentangle without simplifying assumptions, such as assuming that small-scale interactions are not visible (or are negligible) at a certain resolution, or restricting the viewframe to eliminate large-scale variance between chromosome arms. It is by definition a speculative exercise to interpret the biological relevance of an observed pattern, but the consensus is to call compartments on interacting regions that arise from the eigendecomposition of a Hi-C matrix without further modifications <span class="citation" data-cites="lajoie2015hitchhikersguidehic">(<a href="#ref-lajoie2015hitchhikersguidehic" role="doc-biblioref">Lajoie, Dekker, and Kaplan 2015</a>)</span>. Briefly, each genomic bin is assigned a value reflecting its compartment identity, where positive values indicate compartment A and negative values indicate compartment B. The interaction score between two loci is represented as the product of their compartment values, resulting in enriched interactions for loci within the same compartment (positive product) and depleted interactions for loci in different compartments (negative product). This model reproduces the checkerboard pattern characteristic of Hi-C data. Principal component analysis (PCA) identifies the first principal component, which optimally captures the strongest compartment signal. The sign of this component determines whether a locus belongs to compartment A or B. As the eigenvector is only unique up to a sign change, a phasing track is used to orient the eigenvector, aiming for a positive correlation with GC content (in mammals), so that A-compartments represent the active euchromatin, while B-compartments represent the closed heterochromatin.</p>
</section>
<section id="compartment-edges-and-genomic-intervals" class="level4">
<h4 class="anchored" data-anchor-id="compartment-edges-and-genomic-intervals">Compartment Edges and Genomic Intervals</h4>
<p>As arbitrarily as a compartment may be defined, we chose to define another genomic interval for analysis. It is well known that CTCF and other structural proteins preferentially binds to Topologically Associating Domains <span class="citation" data-cites="bicciato2022hicdataanalysis">(<a href="#ref-bicciato2022hicdataanalysis" role="doc-biblioref">Bicciato and Ferrari 2022</a>, Ch. 3)</span> (TADs; they were initially defined as sub-Mb chromatin structures <span class="citation" data-cites="lajoie2015hitchhikersguidehic">(<a href="#ref-lajoie2015hitchhikersguidehic" role="doc-biblioref">Lajoie, Dekker, and Kaplan 2015</a>)</span>, but currently the definition seems to vary based on the method of extraction <span class="citation" data-cites="open2c2022cooltoolsenablinghighresolution">(<a href="#ref-open2c2022cooltoolsenablinghighresolution" role="doc-biblioref">Open2C et al. 2022</a>)</span>). Derived from this, we define a transition zone between A/B compartments to look for enrichment of specific regions of interest.</p>
<p>We can test if two sets of genomic intervals correlate (say, compartment edges and ECH regions) by either proximity of the non-intersecting parts of the sets, or by intersection over union (Jaccard index). When the underlying distribution of a statistic (or index) is unknown, a widespread method in statistics for estimating a p-value is by bootstrapping. Here, one of the sets are bootstrapped (the intervals are placed at random positions) a number of times, <span class="math inline">\(b\)</span>, and the fraction of statistics more extreme than the one we observe is reported as the p-value.</p>
</section>
</section>
</section>
<section id="reproducibility-infrastructure" class="level2">
<h2 class="anchored" data-anchor-id="reproducibility-infrastructure">Reproducibility Infrastructure</h2>
<div id="lst-reproduce" class="listing quarto-float quarto-figure quarto-figure-left anchored">
<figure class="quarto-float quarto-float-lst figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-reproduce-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;1: Three simple steps to re-create the project.
</figcaption>
<div aria-describedby="lst-reproduce-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="sourceCode" id="lst-reproduce"><pre class="sourceCode numberSource console number-lines code-with-copy"><code class="sourceCode"><span id="lst-reproduce-1"><a href="#lst-reproduce-1"></a>git clone https://github.com/munch-group/hic-spermatogenesis.git</span>
<span id="lst-reproduce-2"><a href="#lst-reproduce-2"></a>conda env create -f binder/environment.yml</span>
<span id="lst-reproduce-3"><a href="#lst-reproduce-3"></a>gwf run</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</figure>
</div>
<p>Reproducibility is a cornerstone of the scientific method, ensuring that findings can be independently verified and built upon by others. Without it, the credibility of research findings is undermined, hindering scientific progress by becoming anecdotal. The necessity of reproducibility also aligns with scientific skepticism—the practice of critically evaluating evidence and methods to guard against biases or errors. As research grows increasingly computational, robust reproducibility infrastructure is essential to maintain these principles <span class="citation" data-cites="baker20161500scientistslift">(<a href="#ref-baker20161500scientistslift" role="doc-biblioref">Baker 2016</a>)</span>, facilitating transparency, validation, and collaboration across diverse scientific disciplines. Thus, apart from the biological questions we seek to investigate and answer in this thesis, a major goal is to create fully (and easily) reproducible results through a self-contained and version-controlled pipeline using git <span class="citation" data-cites="git">(<a href="#ref-git" role="doc-biblioref">Torvalds and Hamano 2005</a>)</span>, GitHub <span class="citation" data-cites="github">(<a href="#ref-github" role="doc-biblioref">GitHub, Inc. n.d.</a>)</span>, quarto <span class="citation" data-cites="Allaire_Quarto_2024">(<a href="#ref-Allaire_Quarto_2024" role="doc-biblioref">Allaire et al. 2024</a>)</span>, Anaconda <span class="citation" data-cites="anaconda">(<a href="#ref-anaconda" role="doc-biblioref">Anaconda Software Distribution 2016</a>)</span>, gwf <span class="citation" data-cites="gwf">(<a href="#ref-gwf" role="doc-biblioref">GenomeDK 2023</a>)</span>, and Jupyter <span class="citation" data-cites="soton403913jupyter">(<a href="#ref-soton403913jupyter" role="doc-biblioref">Kluyver et al. 2016</a>)</span>. The result is a fully reproducible analysis, including figures and tables in only 3 lines of code<a href="#fn2" class="footnote-ref" id="fnref2" role="doc-noteref"><sup>2</sup></a> (<a href="#lst-reproduce" class="quarto-xref">Listing&nbsp;1</a>) . See <a href="#tbl-reproducibility" class="quarto-xref">Table&nbsp;2</a> for a brief overview:</p>
<div id="tbl-reproducibility" class="striped quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-reproducibility-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;2: Overview of the tools used for reproducibility of this thesis.
</figcaption>
<div aria-describedby="tbl-reproducibility-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<table class="table-striped caption-top table">
<thead>
<tr class="header">
<th style="text-align: right;">Tool</th>
<th style="text-align: left;">Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: right;">Jupyter</td>
<td style="text-align: left;">Interactive coding environment for analysis and development (notebooks are natively rendered with Quarto)</td>
</tr>
<tr class="even">
<td style="text-align: right;">Quarto</td>
<td style="text-align: left;">A Quarto Manuscript project nested inside a Quarto Book for rendering html (website) and PDF (manuscript) from Markdown via Pandoc. Supports direct embedding of output from Jupyter Notebook cells (plots, tables).</td>
</tr>
<tr class="odd">
<td style="text-align: right;">Conda</td>
<td style="text-align: left;">For managing software requirements and dependency versions reproducibly.</td>
</tr>
<tr class="even">
<td style="text-align: right;">git</td>
<td style="text-align: left;">Version control and <code>gh-pages</code> branch for automated render of Quarto project</td>
</tr>
<tr class="odd">
<td style="text-align: right;">GitHub</td>
<td style="text-align: left;">Action was triggered <code>on push</code> to render the project and host on <a href="https://munch-group.org/hic-spermatogenesis/">munch-group.org</a></td>
</tr>
<tr class="even">
<td style="text-align: right;"><em>gwf</em></td>
<td style="text-align: left;">Workflow manager to automate the analysis on a HPC cluster, wrapped in Python code. <code>workflow.py</code> currently does everything from <code>.fastq</code> to <code>.cool</code>, but notebooks can be set to run sequentially as part of the workflow as well.</td>
</tr>
</tbody>
</table>
</div>
</figure>
</div>
<section id="gwf-workflow-management-for-high-performance-computing-hpc" class="level3">
<h3 class="anchored" data-anchor-id="gwf-workflow-management-for-high-performance-computing-hpc">GWF: workflow management for High-Performance Computing (HPC)</h3>
<p>To enable consistently reproducing the analyses, a workflow manager is utilized. Several exist, but the most well-known is likely Snakemake. However, we use the pragmatic (authors’ own words), lightweight workflow manager GWF, which is optimized for the GenomeDK insfrastructure, and has the benefit of in-house support.</p>
<p>Briefly, GWF works on a python script, conventionally <code>workflow.py</code>, that wraps all the jobs (<em>targets</em> in gwf lingo) you will submit to the HPC cluster. Each target is submitted from a template, written as a Python function, which includes <code>inputs</code> and <code>outputs</code> that GWF should look for when building the depency graph, <code>options</code> list of resources that is forwarded to the queueing system (Slurm in our case), and <code>specs</code>, specifying the submission code in Bash as a formatted Python string (meaning we can pass Python variables to the submission code), providing an extremely flexible framework for running large and intensive analyses in a high-performance computing environment.</p>
</section>
<section id="project-initialization" class="level3">
<h3 class="anchored" data-anchor-id="project-initialization">Project Initialization</h3>
<section id="gwf" class="level4">
<h4 class="anchored" data-anchor-id="gwf"><em>gwf</em></h4>
<p>The initialization of the project directory is the basis of reproducibility and transparency, together with <code>workflow.py</code> inhabiting the main directory. Specifically, it includes a subdirectory for (intermediate) files that are produced by the pipeline, <code>steps/</code>. Everything in this directory is reproducible simply by re-running the <em>gwf</em>-workflow. It is thus not tracked by <code>git</code>, as the large files (raw reads, aligned read-pairs, etc.) it contains are already indirectly tracked (<code>workflow.py</code> is tracked). It can be safely deleted if your system administrator tells you to free up disk space, although you would have to run the workflow again to continue the analysis. Several directories are created for files that are not produced by the pipeline, that is, files that the workflow uses, configuration files, figures edited by hand, etc. Ideally, as few files as possible should be outside of <code>steps/</code>, to be as close as possible to an automated analysis.</p>
</section>
<section id="jupyter-notebooks" class="level4">
<h4 class="anchored" data-anchor-id="jupyter-notebooks">Jupyter Notebooks</h4>
<p>A <code>notebooks/</code> subdirectory contains Jupyter notebooks that are named chronologically, meaning they operate on data located in either <code>steps/</code> or generated from a previous notebook. This way, the workflow can also be set up to run the notebooks (in order) to produce the figures, tables, and their captions used in this manuscript.</p>
</section>
<section id="quarto" class="level4">
<h4 class="anchored" data-anchor-id="quarto">Quarto</h4>
<p>Quarto is an open-source scientific and technical publishing system that uses (pandoc) markdown to create and share production quality output, integrating Jupyter Notebooks with Markdown and LaTeX and enabling embedding content across <em>.ipynb</em> and <em>.qmd</em>. In <em>.qmd</em>, code chunks in several programming languages can be executed and rendered, including Python, R, mermaid (JavaScript-based diagramming). A Quarto project is configured with a YAML configuration file (<code>_quarto.yml</code>) that defines how output is rendered. In this project, we use a nested structure, nesting a <em>Slides</em> project and a <em>Manuscript</em> project inside a <em>Book</em> project. To manage the directory as a Quarto Book project, a quarto configuration file was placed at the base, defining how the Book should be rendered. Additionally, configuration files were placed in <code>slides/</code> and <code>thesis/</code>, to render them as Quarto Slides and Quarto Manuscript, respectively. This nested structure lets us render different subprojects with different configurations than the main project, for example to generate the manuscript, a single Quarto Markdown file, in both <em>.html</em> and <em>.pdf</em>, and only including embedded outputs from specified cells from notebooks in the parent directory. Although the Quarto framework is extensive, it is still under development and has several drawbacks worth mentioning. First, one can only embed the output of code cells from notebooks, meaning the only way to embed text with a python variable (e.g.&nbsp;you want the manuscript to reflect the actual value of a variable, sample sizes <code>n = [1000, 10000, 100000]</code>, and their respective outputs) is by converting a formatted python string into Markdown and send it to the output. Second, embedded figures will be copied as-is in the notebook, and thus cannot be post-processed with size or layout. This makes it impractical to e.g.&nbsp;use the same figures in slides and in the manuscript. Third, when rendering large projects that is tracked by git, some output files (that have to be tracked to publish the website) can exeed GitHub size limits. Especially if rendering in the <em>jats</em> format, producing a MECA Bundle that should be the most flexible way to exchange manuscripts. However, as not applicable to this thesis, the option was simply disabled. Fourth, some functionality relies on external dependencies that cannot be installed on a (linux) remote host (GenomeDK), such as relying on a browser for converting <code>mermaid</code> diagrams into png for the pdf-manuscript.</p>
</section>
</section>
<section id="git-and-github" class="level3">
<h3 class="anchored" data-anchor-id="git-and-github">git and GitHub</h3>
<p>To track the project with git and GitHub, the abovementioned structure was initialized as a GitHub repository, including a workflow for GitHub Actions to publish and deploy the website on the <code>gh-pages</code> branch when pushing commits to <code>main</code>. Briefly, it sets up a virtual machine with Quarto and its dependencies, renders the project as specified in the <code>_quarto.yml</code> configuration file(s), and publishes the project on the group website <a href="https://munch-group.org/hic-spermatogenesis">munch-group.org</a>.</p>
</section>
</section>
</section>
<section id="methods" class="level1">
<h1>Methods</h1>
<p>All computations were performed on GenomeDK (GDK), an HPC cluster located on Aarhus Uninversity, and most of the processing of the data was made into a custom GWF workflow, a workflow manager developed at GDK.</p>
<p>With the analysis tools determined in the above section, I decided it was not feasible to follow the exact approach as <span class="citation" data-cites="wang2019reprogrammingmeioticchromatin">Wang et al. (<a href="#ref-wang2019reprogrammingmeioticchromatin" role="doc-biblioref">2019</a>)</span> with any of HiCExplorer and Open2C, as they use a third software, HiC-Pro. For mapping the raw reads, Hic-Pro internally uses bowtie2 in end-to-end mode, followed by trimming the 3’-end of the unmapped reads, then remapping the 5’-ends to rescue chimeric fragments. I initially mapped the reads using end-to-end Bowtie2 without the rescue-remapping feature, resulting in a very high fraction of discarded reads. Manually implementing the remapping approach would be impractical. When reanalyzing data, it is important to use state-of-the-art tools. Considering the release timeline (e.g., HiC-Pro v3.1.0 in 2021), both HiCExplorer and Open2C are more recent and likely offer better support for current methodologies. Additionally, the HiC-Pro pipeline stops at a normalized contact map, and is thus not sufficient for downstream analysis. In hindsight, it would have been more sensical to use HiC-Pro to get normalized contact maps, then continue analyzing with cooler/cooltools, and finally compare the results evenly with the results achieved from using Open2C from start to finish. <a href="#fig-hic-tools-comparison" class="quarto-xref">Figure&nbsp;4</a> gives an overview of the 3 pipelines metioned in this report.</p>
<div id="fig-hic-tools-comparison" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-hic-tools-comparison-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="illustrations/fig-hic-tools-comparison.png" class="lightbox" data-gallery="quarto-lightbox-gallery-4" title="Figure&nbsp;4: A 3-column flowchart of HiC-Pro, HiCExplorer, and Open2C"><img src="illustrations/fig-hic-tools-comparison.png" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-hic-tools-comparison-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;4: A 3-column flowchart of HiC-Pro, HiCExplorer, and Open2C
</figcaption>
</figure>
</div>
<section id="fetching-raw-data" class="level2">
<h2 class="anchored" data-anchor-id="fetching-raw-data">Fetching raw data</h2>
<p>To reproduce the results from <span class="citation" data-cites="wang2019reprogrammingmeioticchromatin">Wang et al. (<a href="#ref-wang2019reprogrammingmeioticchromatin" role="doc-biblioref">2019</a>)</span>, I chose to use their raw data directly from the SRA portal <span class="citation" data-cites="sayers2022databaseresourcesnational">(<a href="#ref-sayers2022databaseresourcesnational" role="doc-biblioref">Sayers et al. 2022</a>)</span>. I filtered the data to contain all their paired-end Hi-C reads, and included only macaque samples. The dataset also contains RNAseq data, and the same tissues for both macaque and mouse. The metadata for the dataset was extracted into a runtable <code>SRA-runtable.tsv</code>. To get an overview of the data accessions used in this analysis, we will first summarize the runtable that contains the accession numbers and some metadata for each sample (<a href="#tbl-runtable-summary" class="quarto-xref">Table&nbsp;3</a>). It adds up to ~1Tb of compressed <code>fastq</code> files, holding ~9.5 billion reads, roughly evenly spread on the 5 tissue types.</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/03_compartments.ipynb" data-notebook-title="Compartments Analysis" data-notebook-cellid="cell-tbl-runtable-summary">
<div class="cell" data-execution_count="47">
<div id="tbl-runtable-summary" class="cell quarto-float quarto-figure quarto-figure-center anchored" data-execution_count="47">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-runtable-summary-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;3: Summary of the data accessions used in this analysis
</figcaption>
<div aria-describedby="tbl-runtable-summary-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output cell-output-display">
<div>


<table class="dataframe do-not-create-environment cell caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">source_name</th>
<th data-quarto-table-cell-role="th">GB</th>
<th data-quarto-table-cell-role="th">Bases</th>
<th data-quarto-table-cell-role="th">Reads</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>fibroblast</td>
<td>211.403275</td>
<td>553,968,406,500</td>
<td>1,846,561,355</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>pachytene spermatocyte</td>
<td>274.835160</td>
<td>715,656,614,700</td>
<td>2,385,522,049</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>round spermatid</td>
<td>243.128044</td>
<td>655,938,457,200</td>
<td>2,186,461,524</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3</td>
<td>sperm</td>
<td>164.131640</td>
<td>428,913,635,400</td>
<td>1,429,712,118</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">4</td>
<td>spermatogonia</td>
<td>192.794420</td>
<td>518,665,980,300</td>
<td>1,728,886,601</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
</figure>
</div>
</div>
</div>
<section id="fetching-and-indexing-the-reference" class="level3">
<h3 class="anchored" data-anchor-id="fetching-and-indexing-the-reference">Fetching and indexing the reference</h3>
<p><span class="citation" data-cites="wang2019reprogrammingmeioticchromatin">Wang et al. (<a href="#ref-wang2019reprogrammingmeioticchromatin" role="doc-biblioref">2019</a>)</span> use the 2006-version of the macaque reference, <em>rheMac2</em>. Supporting my previous sentiment about not using outdated resources I find it most reasonable to use the latest reference, <em>rheMac10</em>, where <span class="citation" data-cites="warren2020sequencediversityanalyses">Warren et al. (<a href="#ref-warren2020sequencediversityanalyses" role="doc-biblioref">2020</a>)</span> have improved contiguity from <em>rhemac8</em> by 120 fold, going from N50 contig size of 107 Kbp to 46 Mbp. Reproducing the analysis on the latest assembly of the macaque genome should therefore gain a more accurate read mapping, and consequently a better inference of the chromatin compartments. Therefore, <em>rheMac10</em> was downloaded to GDK from UCSC web servers. To use bwa for mapping, <em>rheMac10</em> needs to be indexed with both bwa-index with the <code>--bwtsw</code> option and samtools-faidx, which results in six indexing files for bwa-mem to use. Both bwa-mem and bowtie2 were used in different configurations, and bowtie2 requires its own indexing of the reference, using bowtie2-build with the <code>--large-index</code> option, which creates six index files for <code>bowtie2</code> to use. The options <code>--bwtsw</code> and <code>--large-index</code> create the special indexing format required for large genomes such as macaque. As we use the position of the centromere to partition chrX into its two arms, and no comment about it was made by <span class="citation" data-cites="warren2020sequencediversityanalyses">Warren et al. (<a href="#ref-warren2020sequencediversityanalyses" role="doc-biblioref">2020</a>)</span>, the centromeric region was inferred (visually) from the UCSC browser view of <em>rheMac10</em>, where a large continuous region (chrX:57.5Mb-60.2Mb) had no annotation and showed many repeating regions. The region is roughly the same region as inferred by <span class="citation" data-cites="wang2019reprogrammingmeioticchromatin">Wang et al. (<a href="#ref-wang2019reprogrammingmeioticchromatin" role="doc-biblioref">2019</a>)</span> by the same method.</p>
</section>
</section>
<section id="hicexplorer-trials" class="level2">
<h2 class="anchored" data-anchor-id="hicexplorer-trials">HiCExplorer trials</h2>
<p>To get aligned reads in a format compatible with HiCExplorer, the read mates have to be mapped individually to the reference genome. This supports the old convention to avoid the common heuristics of local aligners used for regular paired-end sequencing libraries (<span class="citation" data-cites="lajoie2015hitchhikersguidehic">Lajoie, Dekker, and Kaplan (<a href="#ref-lajoie2015hitchhikersguidehic" role="doc-biblioref">2015</a>)</span>). HiCExplorer provide examples for both <em>bwa</em> and <em>bowtie2</em>, so I used both with recommended settings. In both cases, the aligner outputs a .bam-file for each mate (<code>sample_R1.bam</code> and <code>sample_R2.bam</code>), and HiCExplorer performs the parsing, deduplication, and filtering of the reads and builds the raw interaction matrix in a single command,</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="ex">hicBuildMatrix</span> <span class="at">-s</span> sample_R1.bam sample_R2.bam <span class="at">-o</span> matrix.h5 <span class="pp">[</span><span class="ss">...</span><span class="pp">]</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><a href="#fig-hic-analysis-flow" class="quarto-xref">Figure&nbsp;3</a> (middle) shows an overview of the pipeline. For parsing, the command needs a <code>--restrictionCutFile</code>, locating the restriction sites from the restriction enzyme used on the reference genome, which is generated with <code>hicFindRestSites</code> that operates on the reference genome and restriction sequence. The default filter, <code>--minMappingQuality 15</code>, was applied as described in <a href="#sec-matrix-qc" class="quarto-xref">Section&nbsp;1.4.2.3</a>. Notably, HiCExplorer has no options on handling multiple ligations, and the method is undocumented. I assume that <span class="citation" data-cites="ramirez2018highresolutiontadsreveal">Ramírez et al. (<a href="#ref-ramirez2018highresolutiontadsreveal" role="doc-biblioref">2018</a>)</span> have implemented a conservative handling (masking all multiple ligations), as that would have the least consequence.</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/01_hicexplorer.ipynb" data-notebook-title="HiCExplorer trial pipeline" data-notebook-cellid="cell-tbl-hic-exploration">
<div class="cell">
<div id="tbl-hic-exploration" class="cell quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-hic-exploration-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;4: The samples chosen for initial data exploration with HiCExplorer. From NCBI SRA Portal.
</figcaption>
<div aria-describedby="tbl-hic-exploration-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output cell-output-display" data-execution_count="3">
<div>


<table class="dataframe do-not-create-environment cell caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">Run</th>
<th data-quarto-table-cell-role="th">Bases</th>
<th data-quarto-table-cell-role="th">Bytes</th>
<th data-quarto-table-cell-role="th">source_name</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>SRR6502335</td>
<td>73201141800</td>
<td>31966430779</td>
<td>fibroblast</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>SRR6502336</td>
<td>65119970100</td>
<td>24433383054</td>
<td>fibroblast</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>SRR6502337</td>
<td>52769196300</td>
<td>23015357755</td>
<td>fibroblast</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3</td>
<td>SRR6502338</td>
<td>52378949100</td>
<td>22999581685</td>
<td>fibroblast</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">4</td>
<td>SRR6502339</td>
<td>28885941600</td>
<td>10960123150</td>
<td>fibroblast</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
</figure>
</div>
</div>
</div>
<p>For the initial exploration of methods with HiCExplorer, I chose five fibroblast samples (see <a href="#tbl-hic-exploration" class="quarto-xref">Table&nbsp;4</a>). The goal was to replicate some of the figures from <span class="citation" data-cites="wang2019reprogrammingmeioticchromatin">Wang et al. (<a href="#ref-wang2019reprogrammingmeioticchromatin" role="doc-biblioref">2019</a>)</span> using HiCExplorer, especially to reconstruct interaction matrices and E1 graphs from macaque data. We constructed matrices with <code>hicBuildMatrix</code> as described from the separately mapped read-pairs. Along with the matrix <em>.h5</em> file, a <em>.log</em> file was created as well, documenting the quality control for the sample. Multiple logs were aggregated and visualized with <code>hicQC</code>.</p>
<p>Before correction (or balancing) of the interaction matrix, a pre-correction filter is applied, filtering out low-count bins and very high-count bins. A threshold for Mean Absolute Deviation (MAD) is estimated by <code>hicCorrect diagnostic_plot</code>, followed by iterative correction with <code>hicCorrect correct --correctionMethod ICE</code>. The PCA was performed with <code>hicPCA</code> on the corrected matrices, yielding the first 3 PCs.</p>
<p>The matrix plotting function, <code>hicPlotMatrix</code>, plots matrices directly to .png, and so there is no output sent to the Jupyter display. As I kept the analysis in a Jupyter Notebook (using the buikt-in shell-escape commands to execute <code>bash</code> code), the plot files must be embedded back into the notebook. The command-line options for modifying plots are quite limited, such as adding spacing for a bigWig track with E1 values, including plot titles, or defining the size and resolution of the plot. I made a brief attempt to implement a custom plotting function for the .h5 matrices and bigWig tracks. However, this approach had a major limitation: it could not fetch specific regions from the matrix dynamically and instead required loading entire matrices into memory, including full-length chromosomes.</p>
</section>
<section id="open2c-pipeline" class="level2">
<h2 class="anchored" data-anchor-id="open2c-pipeline">Open2C pipeline</h2>
<p>A GWF workflow was created to handle the first part of the data processing, and each accesion number (read pair, mate pair) from the Hi-C sequencing was processed in parallel, so their execution was independent from each other. See <a href="#fig-hic-analysis-flow" class="quarto-xref">Figure&nbsp;3</a> (right) for an overview of the initial steps.</p>
<section id="downloading-the-reads" class="level4">
<h4 class="anchored" data-anchor-id="downloading-the-reads">Downloading the reads</h4>
<p>The reads were downloaded from NCBI SRA with SRA-toolkit <span class="citation" data-cites="devteam2024sratoolkit">(<a href="#ref-devteam2024sratoolkit" role="doc-biblioref">DevTeam 2024</a>)</span> directly to GDK using the docker image <em>wwydmanski/sra-downloader</em> as gunzipped <code>.fastq</code> files. Although possible to provide a list of accessions to the toolkit, I submitted each accession as a separate target, as SRA-Toolkit acts sequentially, and only starts the next download after all compression tasks were done. It was therefore a low-hanging fruit to parallelize the download for efficiency.</p>
</section>
<section id="mapping-hi-c-reads" class="level4">
<h4 class="anchored" data-anchor-id="mapping-hi-c-reads">Mapping Hi-C reads</h4>
<p>Suspiciously, <span class="citation" data-cites="openchromosomecollective">(<a href="#ref-openchromosomecollective" role="doc-biblioref"><span>“<span>Open2C</span>”</span> n.d.</a>)</span> do not mention any problems with aligning the Hi-C reads, they just provide an example using bwa-mem in paired-end mode and with the <code>-P</code> option set, which activates the Smith-Waterman <span class="citation" data-cites="li2013aligningsequencereadsclone">(<a href="#ref-li2013aligningsequencereadsclone" role="doc-biblioref">Li 2013</a>)</span> algorithm to rescue missing hits, by focusing on assigning only one of the mates to a good mapping and escape mate-rescue. The documentation of <a href="https://bio-bwa.sourceforge.net"><code>bwa</code></a> state that both bwa-mem and bwa-sw will rescue chimeric reads. Consequently, Open2C does not have a buikt-in way of pairing the reads after mapping, and the reads were thus mapped using Open2C’s recommendations, using their established pipeline for producing a cooler. I chose the latter, where I mapped the fastq files to <em>rheMac10</em> in paired end mode for a pair (<span class="math inline">\(m1\)</span>, <span class="math inline">\(m2\)</span>) with <code>bwa mem -SP rheMac10 m1 m2</code>.</p>
</section>
<section id="parse-and-sort-the-reads" class="level4">
<h4 class="anchored" data-anchor-id="parse-and-sort-the-reads">Parse and sort the reads</h4>
<p>We need to convert the alignments into ligation events, and distinguish between several types of ligation events. The simplest event is when each side only maps to one unique segment in the genome ‘UU’. Other events, where one or both sides map to multiple segments or the reads are long enough (<span class="math inline">\(&gt;150bp\)</span>) to contain two alignments (multiple ligations) have to be considered as well. Multiple ligations are called <em>walks</em> by Open2C, and are treated according to the <code>--walks-policy</code> when parsing the alignments into valid pairs (or valid Hi-C contacts). Here, <code>mask</code> is the most conservative and masks all complex walks, whereas <code>5unique</code> and <code>3unique</code> reports the 5’-most or 3’-most unique alignment on each side, respectively, and <code>all</code> reports all the alignments. The pairs are piped directly into <code>pairtools sort</code> after parsing, as the deduplication step requires a sorted set of pairs. The <em>.pairs</em>-format produced by <code>pairtools</code> is an extension the <a href="https://data.4dnucleome.org/file-formats/pairs/">4DN Consortium</a>-specified format, storing Hi-C pairs as in <a href="#tbl-pairsformat" class="quarto-xref">Table&nbsp;5</a>.</p>
<div id="tbl-pairsformat" class="striped quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-pairsformat-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;5: Column specification of the .pairs format as extended by pairtools.
</figcaption>
<div aria-describedby="tbl-pairsformat-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<table class="table-striped caption-top table">
<colgroup>
<col style="width: 16%">
<col style="width: 16%">
<col style="width: 67%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: right;">Index</th>
<th style="text-align: left;">Name</th>
<th style="text-align: left;">Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: right;">1</td>
<td style="text-align: left;">read_id</td>
<td style="text-align: left;">the ID of the read as defined in fastq files</td>
</tr>
<tr class="even">
<td style="text-align: right;">2</td>
<td style="text-align: left;">chrom1</td>
<td style="text-align: left;">the chromosome of the alignment on side 1</td>
</tr>
<tr class="odd">
<td style="text-align: right;">3</td>
<td style="text-align: left;">pos1</td>
<td style="text-align: left;">the 1-based genomic position of the outer-most (5’) mapped bp on side 1</td>
</tr>
<tr class="even">
<td style="text-align: right;">4</td>
<td style="text-align: left;">chrom2</td>
<td style="text-align: left;">the chromosome of the alignment on side 2</td>
</tr>
<tr class="odd">
<td style="text-align: right;">5</td>
<td style="text-align: left;">pos2</td>
<td style="text-align: left;">the 1-based genomic position of the outer-most (5’) mapped bp on side 2</td>
</tr>
<tr class="even">
<td style="text-align: right;">6</td>
<td style="text-align: left;">strand1</td>
<td style="text-align: left;">the strand of the alignment on side 1</td>
</tr>
<tr class="odd">
<td style="text-align: right;">7</td>
<td style="text-align: left;">strand2</td>
<td style="text-align: left;">the strand of the alignment on side 2</td>
</tr>
<tr class="even">
<td style="text-align: right;">8</td>
<td style="text-align: left;">pair_type</td>
<td style="text-align: left;">the type of a Hi-C pair</td>
</tr>
<tr class="odd">
<td style="text-align: right;">9</td>
<td style="text-align: left;">mapq1</td>
<td style="text-align: left;">mapq of the first mate</td>
</tr>
<tr class="even">
<td style="text-align: right;">10</td>
<td style="text-align: left;">mapq2</td>
<td style="text-align: left;">mapq of the second mate</td>
</tr>
</tbody>
</table>
</div>
</figure>
</div>
<p>I initially used <code>--walks-policy mask</code>, and later followed the recommendations from <em>pairtools</em>, specifically informing that longer reads (<span class="math inline">\(&gt;150bp\)</span>) might have a significant proportion of reads that contain complex walks. With this in mind and as the average read-length of our data is 300 bp, I decided to re-parse the alignments into a new set of pairs, and equally apply the recommended filter (next section). As both results are saved, we can compare the two approaches.</p>
</section>
<section id="filter-and-deduplicate-pairs" class="level4">
<h4 class="anchored" data-anchor-id="filter-and-deduplicate-pairs">Filter and deduplicate pairs</h4>
<p>Pairtools comes with a de-duplication function, <code>dedup</code>, to detect PCR duplication artefacts. At this point I removed all reads that mapped to an unplaced scaffold. Even though <em>rhemac10</em> is less fragmented than <em>rhemac8</em> , <em>rheMac10</em> still contains more than 2,500 unplaced scaffolds, which are all uninformative when calculating the chromatin compartments as is the goal of this analysis. Therefore, we simply only include the list of conventional chromosomes (1..22, X, Y) when doing the deduplication. Initially, the default values were used to remove duplicates, where pairs with both sides mapped within 3 base pairs from each other are considered duplicates. <code>cooler</code> recommend to store the most comprehensive and unfiltered list of pairs, and then applying a filter it on the fly by piping from <code>pairtools select</code>. The first run that was parsed with <code>mask</code> was not filtered for mapping quality. After reparsing the alignments and applying the same analysis, we compare the two pipelines. A quality control report is generated by <code>pairtools dedup</code> as well, and the reports are merged and visualized with <code>MultiQC</code> <span class="citation" data-cites="ewels2016multiqcsummarizeanalysis">(<a href="#ref-ewels2016multiqcsummarizeanalysis" role="doc-biblioref">Ewels et al. 2016</a>)</span> for each cell type.</p>
</section>
<section id="create-interaction-matrices-coolers" class="level4">
<h4 class="anchored" data-anchor-id="create-interaction-matrices-coolers">Create interaction matrices (coolers)</h4>
<p>The final part of the GWF workflow takes <code>.pairs</code> as input and outputs a <code>.cool</code> file (a <em>cooler</em>). Initially, I read directly from the newly generated deduplicated pairs without additional filtering, but the official recommendation is to filter out everything below <span class="math inline">\(mapq = 30\)</span> by piping the pairs through <code>pairtools select "(mapq1&gt;=30) and (mapq2&gt;=30)"</code> to <code>cooler cload pairs</code>. I re-parsed the alignments and created new coolers, including only the Hi-C contacts where <span class="math inline">\(mapq \leq 30\)</span>, following the current recommendations from Open2C.</p>
</section>
<section id="pooling-samples-merging-coolers" class="level4">
<h4 class="anchored" data-anchor-id="pooling-samples-merging-coolers">Pooling samples (Merging coolers)</h4>
<p>The samples are grouped into <em>replicates</em> with a unique BioSample ID, and I chose to pool all the interaction matrices for each cell type. Even though it is slightly unclear how their replicates are defined, <span class="citation" data-cites="wang2019reprogrammingmeioticchromatin">Wang et al. (<a href="#ref-wang2019reprogrammingmeioticchromatin" role="doc-biblioref">2019</a>)</span> determine compartments to be highly reproducible between replicates, and they are pooling the replicates as well.</p>
<p><code>cooler merge</code> was used to merge all samples in each cell-type directory to just one interaction matrix for each cell type. The function merges matrices of the same dimensions by simply adding the interaction frequencies of each genomic position together, resulting in less empty or low-count bins.</p>
</section>
<section id="create-multi-resolution-coolers-zoomify" class="level4">
<h4 class="anchored" data-anchor-id="create-multi-resolution-coolers-zoomify">Create multi-resolution coolers (zoomify)</h4>
<p>A feature of working inside the ecosystem of Open2C is that it natively provides support for storing sparse interaction matrices in multiple resolutions in the same file by adding HDF5-groups to the (multires-)cooler. We can then efficiently store resolutions (i.e., different bin sizes) that is multiples of the smallest bin size. We chose to use 10kb, 50kb, 100kb, and 500kb bins, and the resolutions are made by recursively binning the base resolution. <span class="citation" data-cites="abdennur2020coolerscalablestorage">Abdennur and Mirny (<a href="#ref-abdennur2020coolerscalablestorage" role="doc-biblioref">2020</a>)</span> call this process zoomifying, and <code>cooler zoomify</code> does the job (it recursively calls <code>cooler coarsen</code> to merge bins).</p>
</section>
<section id="matrix-balancing-iterative-correction" class="level4">
<h4 class="anchored" data-anchor-id="matrix-balancing-iterative-correction">Matrix balancing (Iterative correction)</h4>
<p>Finally, we balance (or correct) the matrices using the cooler CLI. We use <code>cooler balance</code> with the default options which iteratively balances the matrix (Iterative Correction).</p>
<p>We balance the matrices on each resolution, and thus it cannot be done prior to zoomifying. <span class="citation" data-cites="abdennur2020coolerscalablestorage">Abdennur and Mirny (<a href="#ref-abdennur2020coolerscalablestorage" role="doc-biblioref">2020</a>)</span> state that the balancing weights are resolution-specific and will no longer retain its biological meaning when binned with other weights. Therefore, we apply <code>cooler balance</code> to each resolution separately. <code>cooler balance</code> will create a new column in the <code>bins</code> group of each cooler, <code>weight</code>, which can then be included or not in the downstream analysis. This means we will have access to both the balanced and the unbalanced matrix.</p>
<p>The default mode uses genome-wide data to calculate the weights for each bin. It would maybe be more suitable to calculate the weights for <em>cis</em> contacts only, and that is possible through the <code>--cis-only</code> flag, and that can be added to another column, so that we can compare the difference between the two methods easily. However, when adding the option, the process seemed to stall and had to be terminated manually, and it was not investigated further.</p>
</section>
<section id="sec-methods-eigendecomposition" class="level4">
<h4 class="anchored" data-anchor-id="sec-methods-eigendecomposition">Eigendecomposition</h4>
<p>The eigendecomposition of a Hi-C interaction matrix is performed in multiple steps. As value of the eigenvector is only <em>significant</em> up to a sign, it is convention to use GC content as a phasing track to orient the vector. E1 is defined to be positively correlated with GC content, meaning a positive E1 value signifies an active chromatin state, which we denote a A-type compartment (or simply A-compartment). We performed eigendecomposition of two resolutions, 100 kbp and 500 kbp. <span class="citation" data-cites="wang2019reprogrammingmeioticchromatin">Wang et al. (<a href="#ref-wang2019reprogrammingmeioticchromatin" role="doc-biblioref">2019</a>)</span> briefly describes their method to calculate the eigenvectors as a sliding window approach on the observed/expected matrix in 100 kb resolution summing over 400 kb bins with 100 kb step size, a method I was not able to replicate in the <em>Open2C</em> ecosystem. I decided to mimic this by smoothing the 100 kb E1 values by summing to 500 kb bins in steps of 100 kb, yielding a comparable resolution which I denote ‘<em>pseudo</em>-500 kb’ resolution (<em>ps500kb</em>).</p>
<p>First, we calculate the GC content of each bin of the reference genome, <em>rheMac10</em>, which is binned to the resolution of the Hi-C matrix we are handling. It is done with <code>bioframe.frac_gc</code> (<em>Open2C</em>). To calculate the E1 compartments, we use only within-chromosome contacts (<em>cis</em>), as we are not interested in the genome-wide contacts. <code>cooltools.eigs_cis</code> will decorrelate the contact-frequency by distance before performing the eigendecomposition. <code>eigs_cis</code> needs a <em>viewframe</em> (view) to calculate E1 values, the simplest view being the full chromosome. However, when there is more variance between chromosome arms than within arms, the sign of the first eigenvector will be determined largely by the chromosome arm it sits on, and not by the chromatin compartments. To mitigate this, we apply a chromosome-arm-partitioned view of the chromosome.</p>
<p>Additionally, to mimic the <em>Local PCA</em> from <span class="citation" data-cites="wang2019reprogrammingmeioticchromatin">(<a href="#ref-wang2019reprogrammingmeioticchromatin" role="doc-biblioref">Wang et al. 2019</a>)</span>, I also defined a view of 10 Mb bins. Thoughout the project, I will compare results from each of the three views and resolutions, and they will be referred to as ‘full’, ‘arms’, and ‘10Mb’ views.</p>
</section>
<section id="plotting-matrices" class="level4">
<h4 class="anchored" data-anchor-id="plotting-matrices">Plotting matrices</h4>
<p>We use matplotlib <span class="citation" data-cites="thematplotlibdevelopmentteam2024matplotlibvisualizationpython">(<a href="#ref-thematplotlibdevelopmentteam2024matplotlibvisualizationpython" role="doc-biblioref">Team 2024</a>)</span> and seaborn <span class="citation" data-cites="waskom2021seabornstatisticaldata">(<a href="#ref-waskom2021seabornstatisticaldata" role="doc-biblioref">Waskom 2021</a>)</span> to plot in the Open2C framework. Utilizing the <code>Cooler</code> class, we can fetch regions of the matrix without modifying the file. As my analysis is centered around the X chromosome, it is efficiently handled by simply fetching ‘chrX’ from the matrix with <code>cooler.Cooler.matrix().fetch('chrX')</code>. Many methods of the cooler class returns data selectors, which do not retrieve data before it is queried <span class="citation" data-cites="abdennur2020coolerscalablestorage">(<a href="#ref-abdennur2020coolerscalablestorage" role="doc-biblioref">Abdennur and Mirny 2020</a>)</span>. This means we can create many selectors at once without overflowing memory, enabling us to plot multiple interaction matrices side-by-side, e.g.&nbsp;the corrected and un-corrected matrices. This is easily done with the <code>balance</code> parameter of the matrix selector (<code>.matrix()</code>), which determines if it should apply the balancing weights to the coordinates and defaults to <code>True</code>.</p>
<p>The matrix is retrieved and plotted with <code>matplotlib.pyplot.matshow</code>, which automatically produces a heatmap image of the matrix. Here, instead of transforming the interaction matrix, the color scale is log-transformed with <code>matplotlib.colors.LogNorm</code>. Additionally, <code>cooltools</code> comes with more tools to aid visualization: <em>adative coarsegrain</em> and <em>interpolation</em>, which can be chained. <code>adaptive_coarsegrain</code> iteratively coarsens an array to the nearest power of two and refines it back to the original resolution, replacing low-count pixels with NaN-aware averages to ensure no zeros in the output, unless there are very large regions that exceed the <code>max_levels</code> threshold, such as the peri-centromeric region.</p>
<p>I implemented a plotting utility, <code>plot_for_quarto</code> in notebook <code>07_various_plotting.ipynb</code> that is compatible with the YAML cell-options read by Quarto’s <code>embed</code> shortcode. It will take an arbitrary number of samples and plot a chromosome (or region) with or without its respective E1 value for either of the three viewframes which has been created. The input is a (subsetted) <em>pandas DataFrame</em>, defined from a file search matching a pattern specified to the <code>glob</code> Python module.</p>
</section>
</section>
<section id="genomic-intervals" class="level2">
<h2 class="anchored" data-anchor-id="genomic-intervals">Genomic Intervals</h2>
<section id="full-regions-edges-and-limits" class="level3">
<h3 class="anchored" data-anchor-id="full-regions-edges-and-limits">Full Regions, Edges, and Limits</h3>
<p>From the eigenvectors, the A-compartments were extracted in bedgraph-format (<code>['chrom', 'start', 'end']</code>) and compared with ECH90 regions lifted to <em>rheMac10</em> from human. We perform visual inspection of the genomic intervals and test whether ECH90 regions are enriched near the edges of the compartments by defining a 200 kilobase transition-zone centered at each sign change of E1 (referred to as <em>compartment edge</em>). We compare genomic intervals (or sets) both visually by plotting the regions, and by a proximity test and bootstrapping the Jaccard index. Additionally, the compartment limits were invetigated (only 1 flanking base pair) in the same manner. Additionally, the limits were extracted from the baboon datasets and lifted to <em>rheMac10</em> for comparison.</p>
<section id="proximity-test" class="level4">
<h4 class="anchored" data-anchor-id="proximity-test">Proximity test</h4>
<p>Determines whether the non-overlapping segments of the sets are more proximal than expected by chance. We define the <em>annotation</em> set and the <em>query</em> set, and the distance from each interval on the <em>query</em> to the most proximal interval on the <em>annotation</em> is used to generate an index of proximity by the mean distance to nearest interval in the <em>annotation</em>. Then, bootstrapping (<span class="math inline">\(b = 100000\)</span>) is performed by randomly placing the query intervals to generate the null distribution, and finally, the fraction of the <em>null</em> as extreme or more extreme as our observed proximity is reported as the p-value.</p>
</section>
<section id="jaccard-test" class="level4">
<h4 class="anchored" data-anchor-id="jaccard-test">Jaccard test</h4>
<p>Measures the significance of the observed Jaccard index (intersection over union) between two sets. The index is a measure of <em>similarity</em> (<span class="math inline">\(intersection/union\)</span>) between two sets (between 0 and 1), which is very sensitive to the size difference between the sets, as even when comparing a set of intervals to a small subset of itself will yield a very small Jaccard index. When we use bootstrapping to generate a null distribution (shuffling the intervals of the <em>query</em>), we find the probability that the two sets (with their respective number and size of intervals), are as similar or more than what we observe. The ratio is reported as the p-value. However, this approach is still sensitive to flipping of query/annotation (if the regions are not the same size), as only the query is bootstrapped.</p>
</section>
<section id="multiple-testing" class="level4">
<h4 class="anchored" data-anchor-id="multiple-testing">Multiple testing</h4>
<p>Considerations were made to avoid multiple testing biases (p-hacking): Performing tests on all combinations of variables (cell type, resolution, viewframe, query) will yield 90 p-values for each test, and we would have to adjust the significance threshold (with <span class="math inline">\(\alpha = 0.05\)</span>, we expect 4 tests passing the threshold by chance). However, I will narrow down the parameter space based on visualizations and reasoning, leaving fewer combinations to test. I do not find it necessary to correct the conventional significance threshold, <span class="math inline">\(\alpha=0.05\)</span>.</p>
</section>
</section>
</section>
</section>
<section id="results" class="level1">
<h1>Results</h1>
<section id="hicexplorer-trials-1" class="level2">
<h2 class="anchored" data-anchor-id="hicexplorer-trials-1">HicExplorer Trials</h2>
<section id="quality-control" class="level3">
<h3 class="anchored" data-anchor-id="quality-control">Quality Control</h3>
<p>The separately mapped read-mates were parsed into a <em>.h5</em> interaction matrix by <code>hicBuildMatrix</code>, which include a <em>.log</em> file documenting the buikt-in quality control (hereafter, <em>QC</em>). Log files from the 5 samples were merged with <code>hicQC</code> (<a href="#fig-explorer-all-3-qc" class="quarto-xref">Figure&nbsp;5</a>). I observe equal fractions of the read-orientation of read-pairs (<a href="#fig-explorer-all-3-qc" class="quarto-xref">Figure&nbsp;5</a>, left, row 5), which is expected for a good Hi-C library. Additionally, it determines between 40% to 50% of the total reads to be valid Hi-C contacts (<a href="#fig-explorer-all-3-qc" class="quarto-xref">Figure&nbsp;5</a>, left, row 1), which is usually only 25%-40% (as described in HiCExplorer docs). <a href="#fig-explorer-all-3-qc" class="quarto-xref">Figure&nbsp;5</a> (left, row 4) shows, however, unusually high fractions of <em>inter</em>-chromosomal contacts (up to 30%) compared to <em>intra</em>-chromosomal contacts (also denoted <em>trans</em> and <em>cis</em> contacts, respectively). It is expected that <em>cis</em> contacts are orders of magnitude more frequent than <em>trans</em> contacts <span class="citation" data-cites="bicciato2022hicdataanalysis lieberman-aiden2009comprehensivemappinglongrange">(<a href="#ref-bicciato2022hicdataanalysis" role="doc-biblioref">Bicciato and Ferrari 2022, 2301:236</a>; <a href="#ref-lieberman-aiden2009comprehensivemappinglongrange" role="doc-biblioref">Lieberman-Aiden et al. 2009</a>)</span>, and HiCExplorer states it is usually below 10% for a high-quality library. The high fraction may be mitigated by enforcing a stricter <em>mapq</em> threshold for a valid Hi-C pair, as we also observe higher-than expected valid contacts. However, we continue with the current matrices.</p>
<p>To compare how well these mappings perform, the plotting QC results is an easy way. Therefore, the reads were mapped with <em>bowtie2</em> in both end-to-end- and local-mode followed by <code>hiCBuildMatrix</code>, and the QC from each method was plotted next to each other (<a href="#fig-explorer-all-3-qc" class="quarto-xref">Figure&nbsp;5</a>). Interestingly, <em>bowtie2</em> was much more computer-intensive in both modes, perhaps because of the <code>--very-sensitive</code> option. In any case, the QC reveals a major difference in the total number of reads that are determined to be valid Hi-C contacts by <code>hicBuildMatrix</code>. As expected, mapping with <em>end-to-end-bowtie2</em> makes locating Hi-C contacts more difficult than the other methods (<a href="#fig-explorer-all-3-qc" class="quarto-xref">Figure&nbsp;5</a>, row 1), finding a very low amount of mappable, unique pairs passing the quality threshold. In contrast, mapping with <em>local-bowtie2</em> performs similarly to <em>bwa</em> in finding mappable, unique, high-quality pairs, but calls only approximately half the number of valid Hi-C contacts (&gt;20%), resulting in a fraction of valid Hi-C pairs that hits the expectation from <em>HicExplorer</em> docs (row3). With <em>bwa</em>, the reads were discarded either due to low mapping quality or non-unique mates, whereas with <em>local-bowtie2</em>, the reads were almost exclusively filtered out due to low mapping quality. This must be a result of how the mappers assign mapping quality, and I believe <em>local-bowtie2</em> looks suspiciously selective in finding unique but low quality alignments. <em>end-to-end-bowtie</em> almost exclusively filters out read-pairs where one mate is unmapped, which is expected when the majority of reads are unmapped.</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/01_hicexplorer.ipynb" data-notebook-title="HiCExplorer trial pipeline" data-notebook-cellid="cell-fig-explorer-all-3-qc">
<div id="cell-fig-explorer-all-3-qc" class="cell" data-execution_count="38">
<div class="cell-output cell-output-display">
<div id="fig-explorer-all-3-qc" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-explorer-all-3-qc-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-01_hicexplorer-fig-explorer-all-3-qc-output-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-5" title="Figure&nbsp;5: Comparison of HiCExplorer QC plots for all samples using different alignment tools. The rows represent different QC plots (pairs sequenced, pairs discarded, unmappable/non-unique, distance, and read orientation), and the columns represent the 3 alignments used (BWA, Bowtie2 end-to-end, Bowtie2 local). Generated by hicQC."><img src="index_files/figure-html/..-notebooks-01_hicexplorer-fig-explorer-all-3-qc-output-1.png" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-explorer-all-3-qc-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5: Comparison of HiCExplorer QC plots for all samples using different alignment tools. The rows represent different QC plots (pairs sequenced, pairs discarded, unmappable/non-unique, distance, and read orientation), and the columns represent the 3 alignments used (BWA, Bowtie2 end-to-end, Bowtie2 local). Generated by <code>hicQC</code>.
</figcaption>
</figure>
</div>
</div>
</div>
</div>
</section>
<section id="correction" class="level3">
<h3 class="anchored" data-anchor-id="correction">Correction</h3>
<p>The correction diagnostic tool yielded a similar <em>mad</em> threshold within the range <span class="math inline">\([-3,-2]\)</span>. Even so, I followed the <em>HicExplorer</em> recommendation to set the lower threshold to at least -2 and the upper threshold to 5 in the pre-normalization filter. I argue that with a high number of valid contacts, it is safer to err on the side of caution and maybe filter out bad data.</p>
<div id="fig-explorer-pre-correction" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-explorer-pre-correction-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 5.9%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-pre-correction" style="flex-basis: 29.4%;justify-content: flex-start;">
<div id="fig-explorer-pre-correction-SRR6502335" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-pre-correction-SRR6502335-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/bwa/SRR6502335_diag_plot.png" class="lightbox" data-gallery="fig-explorer-pre-correction" title="Figure&nbsp;6&nbsp;(a): SRR6502335"><img src="../figures/bwa/SRR6502335_diag_plot.png" class="img-fluid figure-img" data-ref-parent="fig-explorer-pre-correction"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-pre-correction-SRR6502335-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) SRR6502335
</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-pre-correction" style="flex-basis: 29.4%;justify-content: flex-start;">
<div id="fig-explorer-pre-correction-SRR6502336" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-pre-correction-SRR6502336-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/bwa/SRR6502336_diag_plot.png" class="lightbox" data-gallery="fig-explorer-pre-correction" title="Figure&nbsp;6&nbsp;(b): SRR6502336"><img src="../figures/bwa/SRR6502336_diag_plot.png" class="img-fluid figure-img" data-ref-parent="fig-explorer-pre-correction"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-pre-correction-SRR6502336-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) SRR6502336
</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-pre-correction" style="flex-basis: 29.4%;justify-content: flex-start;">
<div id="fig-explorer-pre-correction-SRR6502337" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-pre-correction-SRR6502337-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/bwa/SRR6502337_diag_plot.png" class="lightbox" data-gallery="fig-explorer-pre-correction" title="Figure&nbsp;6&nbsp;(c): SRR6502337"><img src="../figures/bwa/SRR6502337_diag_plot.png" class="img-fluid figure-img" data-ref-parent="fig-explorer-pre-correction"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-pre-correction-SRR6502337-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(c) SRR6502337
</figcaption>
</figure>
</div>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 5.9%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
</div>
<div class="quarto-layout-row">
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 20.6%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-pre-correction" style="flex-basis: 29.4%;justify-content: flex-start;">
<div id="fig-explorer-pre-correction-SRR6502338" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-pre-correction-SRR6502338-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/bwa/SRR6502338_diag_plot.png" class="lightbox" data-gallery="fig-explorer-pre-correction" title="Figure&nbsp;6&nbsp;(d): SRR6502338"><img src="../figures/bwa/SRR6502338_diag_plot.png" class="img-fluid figure-img" data-ref-parent="fig-explorer-pre-correction"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-pre-correction-SRR6502338-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(d) SRR6502338
</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-pre-correction" style="flex-basis: 29.4%;justify-content: flex-start;">
<div id="fig-explorer-pre-correction-SRR6502339" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-pre-correction-SRR6502339-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/bwa/SRR6502339_diag_plot.png" class="lightbox" data-gallery="fig-explorer-pre-correction" title="Figure&nbsp;6&nbsp;(e): SRR6502339"><img src="../figures/bwa/SRR6502339_diag_plot.png" class="img-fluid figure-img" data-ref-parent="fig-explorer-pre-correction"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-pre-correction-SRR6502339-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(e) SRR6502339
</figcaption>
</figure>
</div>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 20.6%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-explorer-pre-correction-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6: Histograms of the number of counts per bin (bottom x-axis) and the modified z-score (top x-axis) from which the <em>mad</em> threshold is defined.
</figcaption>
</figure>
</div>
<p>The five samples were pooled with <code>hicSumMatrices</code>, and the non-standard contigs (unplaced scaffolds) were filtered out, and the different resolutions were created (<code>hicMergeMatrixBins</code>). <em>HiCExplorer</em> also comes with a normalization function prior to correcting the matrix, which should be applied if different samples should have comparable bin counts. It has no effect when having only one matrix. Nevertheless, the pooled matrix was normalized and then corrected compared in <a href="#fig-explorer-pooled-norm-normcorr" class="quarto-xref">Figure&nbsp;7</a>. It is now obvious why we have to correct the matrix. The uncorrected (<a href="#fig-explorer-pooled-chrX-norm" class="quarto-xref">Figure&nbsp;7 (a)</a>) has no signal apart from the diagonal. Even though some bins have been filtered out, the expected <em>plaid</em> pattern of a contact matrix is visible along the diagonal after the correction (<a href="#fig-explorer-pooled-chrX-normcorr" class="quarto-xref">Figure&nbsp;7 (b)</a>), leaving evidence for chromatin structure, especially in the first 50 million bases of the chromosome. There is a wide region of empty values at the place of the centromere.</p>
<div id="fig-explorer-pooled-norm-normcorr" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-explorer-pooled-norm-normcorr-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-pooled-norm-normcorr" style="flex-basis: 50.0%;justify-content: flex-start;">
<div id="fig-explorer-pooled-chrX-norm" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-pooled-chrX-norm-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/bowtie2/local/filter_pooled_50kb_chrX.png" class="lightbox" data-gallery="fig-explorer-pooled-norm-normcorr" title="Figure&nbsp;7&nbsp;(a): Normalized matrix chrX"><img src="../figures/bowtie2/local/filter_pooled_50kb_chrX.png" class="img-fluid figure-img" data-ref-parent="fig-explorer-pooled-norm-normcorr"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-pooled-chrX-norm-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) Normalized matrix chrX
</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-pooled-norm-normcorr" style="flex-basis: 50.0%;justify-content: flex-start;">
<div id="fig-explorer-pooled-chrX-normcorr" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-pooled-chrX-normcorr-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/bowtie2/local/normalized/normsm_filter_pooled_100kb_corrected_chrX-full.png" class="lightbox" data-gallery="fig-explorer-pooled-norm-normcorr" title="Figure&nbsp;7&nbsp;(b): Normalized and corrected chrX"><img src="../figures/bowtie2/local/normalized/normsm_filter_pooled_100kb_corrected_chrX-full.png" class="img-fluid figure-img" data-ref-parent="fig-explorer-pooled-norm-normcorr"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-pooled-chrX-normcorr-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) Normalized and corrected chrX
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-explorer-pooled-norm-normcorr-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7: A comparison of interaction matrices before/after iterative correction (<em>HiCExplorer</em>).
</figcaption>
</figure>
</div>
</section>
<section id="eigenvectors" class="level3">
<h3 class="anchored" data-anchor-id="eigenvectors">Eigenvectors</h3>
<p>The PCA performed by <code>hicPCA</code> on the pooled samples at both 50kb and 100kb resolution yielded the first 3 principal components. For PC1 on both resolutions (<a href="#fig-explorer-pc1-50kb" class="quarto-xref">Figure&nbsp;8 (a)</a>, <a href="#fig-explorer-pc1-100kb" class="quarto-xref">Figure&nbsp;8 (d)</a>) we observe only a single sign change which occurs at around 60 Mbp, the region of the centromere. It means the PCA has captured more variance between the chromosome arms than within them, making it uninformative about chromatin compartments. Upon visual inspection, it is clear that neither of the PC graphs capture the pattern of the interaction matrix by its change of sign. It seems the PCs capture variance from a bias that varies slowly and predictably along the chromosome. The first PC that is supposed to capture the compartments very suspiciously changes sign at the region of the centromere, a classic problem that could be solved by restricting the values from which the PC is calculated along the chromosome. Unimpressed, I rationalize that the option <code>--extra-track</code> to provide a gene track or histone coverage should not affect this result much. It should be provided as a phasing track to orient the eigenvector to positively correlate with gene density or histone marks, and could possibly muddle the compartments if not included. I followed <em>HiCExplorer</em> pipeline to plot and explore the matrices. At this point, I stoppped using <em>HiCExplorer</em>, as I assessed that a more flexible tool was needed.</p>
<div id="fig-explorer-pca" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-explorer-pca-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-pca" style="flex-basis: 33.3%;justify-content: flex-start;">
<div id="fig-explorer-pc1-50kb" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-pc1-50kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/bowtie2/local/normalized/pc1_50kb_corrected_chrX.png" class="lightbox" data-gallery="fig-explorer-pca" title="Figure&nbsp;8&nbsp;(a): "><img src="../figures/bowtie2/local/normalized/pc1_50kb_corrected_chrX.png" id="fig-explorer-pc1-50kb" class="img-fluid figure-img" data-ref-parent="fig-explorer-pca"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-pc1-50kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a)
</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-pca" style="flex-basis: 33.3%;justify-content: flex-start;">
<div id="fig-explorer-pc2-50kb" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-pc2-50kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/bowtie2/local/normalized/pc2_50kb_corrected_chrX.png" class="lightbox" data-gallery="fig-explorer-pca" title="Figure&nbsp;8&nbsp;(b): "><img src="../figures/bowtie2/local/normalized/pc2_50kb_corrected_chrX.png" id="fig-explorer-pc2-50kb" class="img-fluid figure-img" data-ref-parent="fig-explorer-pca"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-pc2-50kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b)
</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-pca" style="flex-basis: 33.3%;justify-content: flex-start;">
<div id="fig-explorer-pc3-50kb" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-pc3-50kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/bowtie2/local/normalized/pc3_50kb_corrected_chrX.png" class="lightbox" data-gallery="fig-explorer-pca" title="Figure&nbsp;8&nbsp;(c): "><img src="../figures/bowtie2/local/normalized/pc3_50kb_corrected_chrX.png" id="fig-explorer-pc3-50kb" class="img-fluid figure-img" data-ref-parent="fig-explorer-pca"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-pc3-50kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(c)
</figcaption>
</figure>
</div>
</div>
</div>
<div class="quarto-layout-row">
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-pca" style="flex-basis: 33.3%;justify-content: flex-start;">
<div id="fig-explorer-pc1-100kb" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-pc1-100kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/bowtie2/local/normalized/pc1_100kb_corrected_chrX.png" class="lightbox" data-gallery="fig-explorer-pca" title="Figure&nbsp;8&nbsp;(d): "><img src="../figures/bowtie2/local/normalized/pc1_100kb_corrected_chrX.png" id="fig-explorer-pc1-100kb" class="img-fluid figure-img" data-ref-parent="fig-explorer-pca"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-pc1-100kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(d)
</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-pca" style="flex-basis: 33.3%;justify-content: flex-start;">
<div id="fig-explorer-pc2-100kb" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-pc2-100kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/bowtie2/local/normalized/pc2_100kb_corrected_chrX.png" class="lightbox" data-gallery="fig-explorer-pca" title="Figure&nbsp;8&nbsp;(e): "><img src="../figures/bowtie2/local/normalized/pc2_100kb_corrected_chrX.png" id="fig-explorer-pc2-100kb" class="img-fluid figure-img" data-ref-parent="fig-explorer-pca"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-pc2-100kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(e)
</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-pca" style="flex-basis: 33.3%;justify-content: flex-start;">
<div id="fig-explorer-pc3-100kb" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-pc3-100kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/bowtie2/local/normalized/pc3_100kb_corrected_chrX.png" class="lightbox" data-gallery="fig-explorer-pca" title="Figure&nbsp;8&nbsp;(f): "><img src="../figures/bowtie2/local/normalized/pc3_100kb_corrected_chrX.png" id="fig-explorer-pc3-100kb" class="img-fluid figure-img" data-ref-parent="fig-explorer-pca"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-pc3-100kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(f)
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-explorer-pca-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;8: Corrected interaction matrix for chromosome X along with PC1, 2, or 3, respectively. a-c: 50kb resolution, d-f: 100kb resolution. <em>HiCExplorer</em>.
</figcaption>
</figure>
</div>
</section>
</section>
<section id="open2c-ecosystem" class="level2">
<h2 class="anchored" data-anchor-id="open2c-ecosystem">Open2c ecosystem</h2>
<section id="quality-control-1" class="level3">
<h3 class="anchored" data-anchor-id="quality-control-1">Quality Control</h3>
<p>As described, the <code>pairtools</code> module in MultiQC was used to visualize results from <code>pairtools stats</code> for the two parsing runs, see <a href="#fig-pairtools-qc" class="quarto-xref">Figure&nbsp;9</a>. The peak at around 250 bp is a techical artefact and should not be parsed into valid contacts.</p>
<section id="walks-policy-mask" class="level4">
<h4 class="anchored" data-anchor-id="walks-policy-mask"><code>--walks-policy mask</code></h4>
<p>Comparing the multiQC report for each of the cell sources show similar distributions of <em>unmapped</em> (both sides unmapped), <em>one-sided</em> (one side mapped), <em>two-sided</em> (both sides mapped), and <em>duplicated</em> (w.r.t. total mapped) reads. The percentage of <em>cis</em> pairs w.r.t. mapped pairs is around 70% for all samples (<a href="#fig-pairtools-multiqc-mask-violin" class="quarto-xref">Figure&nbsp;9 (a)</a>). The valid pairs also show similar distributions of pair types divided into 10 categories. The <span class="math inline">\(P(s)\)</span> curve looks similar for all samples as well, peaking around 250 bp separation (<a href="#fig-pairtools-multiqc-mask-ps" class="quarto-xref">Figure&nbsp;9 (c)</a>). The QC does not show any information about mapping quality of the reads. Note that the <span class="math inline">\(P(s)\)</span> curve arise from pre-filtered pairs, meaning it provides information about the Hi-C library.</p>
</section>
<section id="walks-policy-5unique" class="level4">
<h4 class="anchored" data-anchor-id="walks-policy-5unique"><code>--walks-policy 5unique</code></h4>
<p>Parsing alignments with the recommended walks-policy aproximately halves the percentage of <em>unmapped</em> reads, and <em>one-</em> and <em>two-sided</em> reads as well <em>duplicated</em> reads are slightly increased. Overall number of unique pairs are increased with more than 20% increase. The percentage of <em>cis</em> pairs are only decreased by a percentage point at most (<a href="#fig-pairtools-multiqc-5unique-violin" class="quarto-xref">Figure&nbsp;9 (b)</a>). Changing the walks policy does not alter the <span class="math inline">\(P(s)\)</span> curve, meaning the parameter does not bias the parsing w.r.t. genomic separation.</p>
<div id="fig-pairtools-qc" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-pairtools-qc-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-pairtools-qc" style="flex-basis: 50.0%;justify-content: flex-start;">
<div id="fig-pairtools-multiqc-mask-violin" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-pairtools-multiqc-mask-violin-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/fig-pairtools-parse-multiqc-mask-violin.png" class="lightbox" data-gallery="fig-pairtools-qc" title="Figure&nbsp;9&nbsp;(a): --walks-policy mask"><img src="../figures/fig-pairtools-parse-multiqc-mask-violin.png" class="img-fluid figure-img" data-ref-parent="fig-pairtools-qc"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-pairtools-multiqc-mask-violin-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) <code>--walks-policy mask</code>
</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-pairtools-qc" style="flex-basis: 50.0%;justify-content: flex-start;">
<div id="fig-pairtools-multiqc-5unique-violin" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-pairtools-multiqc-5unique-violin-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/fig-pairtools-parse-multiqc-5unique-violin.png" class="lightbox" data-gallery="fig-pairtools-qc" title="Figure&nbsp;9&nbsp;(b): --walks-policy 5unique"><img src="../figures/fig-pairtools-parse-multiqc-5unique-violin.png" class="img-fluid figure-img" data-ref-parent="fig-pairtools-qc"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-pairtools-multiqc-5unique-violin-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) <code>--walks-policy 5unique</code>
</figcaption>
</figure>
</div>
</div>
</div>
<div class="quarto-layout-row">
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-pairtools-qc" style="flex-basis: 50.0%;justify-content: flex-start;">
<div id="fig-pairtools-multiqc-mask-ps" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-pairtools-multiqc-mask-ps-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/fig-pairtools-parse-multiqc-mask-ps.png" class="lightbox" data-gallery="fig-pairtools-qc" title="Figure&nbsp;9&nbsp;(c): --walks-policy mask"><img src="../figures/fig-pairtools-parse-multiqc-mask-ps.png" class="img-fluid figure-img" data-ref-parent="fig-pairtools-qc"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-pairtools-multiqc-mask-ps-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(c) <code>--walks-policy mask</code>
</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-pairtools-qc" style="flex-basis: 50.0%;justify-content: flex-start;">
<div id="fig-pairtools-multiqc-5unique-ps" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-pairtools-multiqc-5unique-ps-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/fig-pairtools-parse-multiqc-5unique-ps.png" class="lightbox" data-gallery="fig-pairtools-qc" title="Figure&nbsp;9&nbsp;(d): --walks-policy 5unique"><img src="../figures/fig-pairtools-parse-multiqc-5unique-ps.png" class="img-fluid figure-img" data-ref-parent="fig-pairtools-qc"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-pairtools-multiqc-5unique-ps-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(d) <code>--walks-policy 5unique</code>
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-pairtools-qc-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;9: Results of <code>pairtools stats</code> run on all samples from the two walks-policies. Left (a+c): <code>mask</code>; right (b+d): <code>5unique</code>. <em>Generated by MultiQC</em> <span class="citation" data-cites="ewels2016multiqcsummarizeanalysis">(<a href="#ref-ewels2016multiqcsummarizeanalysis" role="doc-biblioref">Ewels et al. 2016</a>)</span>. <em>Note: X-axes are not shared in the ‘Genereal Statistics’ plot.</em>
</figcaption>
</figure>
</div>
</section>
</section>
<section id="correction-1" class="level3">
<h3 class="anchored" data-anchor-id="correction-1">Correction</h3>
<p>Matrix balancing did not show major improvement in the plaid pattern, as it already showed the expected pattern. It does, however, filter out bins that are deemed too low-count to be informative, for example peri-centromeric regions. The matrix was expected to be smoother after balancing (for chromosome-wide maps), as regions along a chromosome should only vary slowly in contact frequency with other regions as they are on a continouos molecule. Therefore, sharp contrasts represent a sudden drop in bin count (<a href="#fig-rs-chrx-raw-balanced-cgi-both-1" class="quarto-xref">Figure&nbsp;10 (a)</a>, raw) and should not be interpreted as devoid of interaction, but an indication that the data is not sufficient to interpret. It is then better to simply remove the bins instead of correcting, which will also amplify noise. Even with a high-quality Hi-C library we expect that all bins do not have the same coverage throughout<span class="citation" data-cites="lajoie2015hitchhikersguidehic">(<a href="#ref-lajoie2015hitchhikersguidehic" role="doc-biblioref">Lajoie, Dekker, and Kaplan 2015</a>)</span>, as restriction enzymes do not bind equally to all regions of the genome, and therefore, some bins will be underrepresented as an artefact of binding/cutting efficieny of the restriction enzyme used.</p>
<p>We can try to mitigate the white lines of empty bins that now appear in the matrices. The coarsegrained and interpolated matrix is useful to make a good-looking interaction matrix, but is not that useful for analysis purposes. It might get easier to visually inspect the matrix, but it is not clear how well the interpolated matrix reflects the structure of the chromatin, and it is not transparent which regions are interpolated and which that are not. I find it purposeful for interpolation on high-resolution (zoomed-in) views (<a href="#fig-rs-chrx-raw-balanced-cgi-both-2" class="quarto-xref">Figure&nbsp;10 (b)</a>) with small empty regions, but misleading for chromosome-wide maps, where typically the centromere and extremities of the chromosome have filtered-out bins. Interpolation is further discussed below.</p>
<p>The regions that are coarsegrained are small zero- or low-count bins which are averaged, effectively reducing the resolution of those regions until the count is sufficient. They get more frequent the longer genomic distance (the further we travel from the diagonal), and effectively enables us to get some intuition about the interactions. The coarsegrain, however, does not interpolate the <code>NaN</code>s created when filtering out whole bins in the balancing step (horisontal and vertical lines in <a href="#fig-rs-chrx-raw-balanced-cgi-both-1" class="quarto-xref">Figure&nbsp;10 (a)</a> and <a href="#fig-rs-chrx-raw-balanced-cgi-both-2" class="quarto-xref">Figure&nbsp;10 (b)</a>; middle). This is done in a subsequent step by linearly interpolating the <code>NaN</code>s. Examining the interpolated matrix on full chrX (<a href="#fig-rs-chrx-raw-balanced-cgi-both-1" class="quarto-xref">Figure&nbsp;10 (a)</a>; right) gives the impression that the pericentromeric (at ~60 Mbp) region harbours a <em>very</em> strong compartment, but that is clearly an artefact of the interpolation on the very large empty region of the centromere, where the diagonal is somehow extended in a square. On the thinner lines, the interpolation seem to be more smooth, and barely noticable on the diagonal.</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/05_rec_compartments.ipynb" data-notebook-title="Compartments Analysis w. recommended parameters" data-notebook-cellid="cell-fig-rs-chrx-raw-balanced-cgi-both">
<div class="cell" data-execution_count="54">
<div id="fig-rs-chrx-raw-balanced-cgi-both" class="cell quarto-float quarto-figure quarto-figure-center anchored" data-execution_count="54">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-rs-chrx-raw-balanced-cgi-both-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output cell-output-display">
<div id="fig-rs-chrx-raw-balanced-cgi-both-1" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-rs-chrx-raw-balanced-cgi-both-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-05_rec_compartments-fig-rs-chrx-raw-balanced-cgi-both-output-1.png" class="lightbox" data-gallery="fig-rs-chrx-raw-balanced-cgi-both" title="Figure&nbsp;10&nbsp;(a): Full RS matrix at 500kb resolution"><img src="index_files/figure-html/..-notebooks-05_rec_compartments-fig-rs-chrx-raw-balanced-cgi-both-output-1.png" data-ref-parent="fig-rs-chrx-raw-balanced-cgi-both" width="593" height="164" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-rs-chrx-raw-balanced-cgi-both-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) Full RS matrix at 500kb resolution
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display">
<div id="fig-rs-chrx-raw-balanced-cgi-both-2" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-rs-chrx-raw-balanced-cgi-both-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-05_rec_compartments-fig-rs-chrx-raw-balanced-cgi-both-output-2.png" class="lightbox" data-gallery="fig-rs-chrx-raw-balanced-cgi-both" title="Figure&nbsp;10&nbsp;(b): Subset RS matrix at 50kb resolution"><img src="index_files/figure-html/..-notebooks-05_rec_compartments-fig-rs-chrx-raw-balanced-cgi-both-output-2.png" data-ref-parent="fig-rs-chrx-raw-balanced-cgi-both" width="593" height="167" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-rs-chrx-raw-balanced-cgi-both-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) Subset RS matrix at 50kb resolution
</figcaption>
</figure>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-rs-chrx-raw-balanced-cgi-both-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;10: Raw, balanced, and interpolated chrX interaction matrix in a) 500kb resolution (full chrX) or b) 50kb resolution (chrX:30Mb-39Mb). The interpolation is done to make the matrix more visually appealing, but it is not necessary for the analysis.
</figcaption>
</figure>
</div>
</div>
</div>
<section id="nan-histograms" class="level4">
<h4 class="anchored" data-anchor-id="nan-histograms"><code>NaN</code> histograms</h4>
<p>As expected, most of the low quality bins are located on the edges of the chromosome arms, especially the region around the centromere <span class="citation" data-cites="warren2020sequencediversityanalyses">(<a href="#ref-warren2020sequencediversityanalyses" role="doc-biblioref">Warren et al. 2020</a>)</span>, as they contain many repetitive sequences. The low-quality bins are filtered out by the balancing algorithm, those bins are <code>NaN</code> in the Hi-C matrix. The median position of the <code>NaN</code> values (<a href="#fig-e1_nan_hist" class="quarto-xref">Figure&nbsp;11</a>) ranges between <span class="math inline">\(58\)</span> and <span class="math inline">\(63.5\)</span>, which is within the estimate of the centromeric region of <em>rhemac10</em> (the UCSC browser has a continuously unannotated region at chrX:57,500,000-60,200,000). The fact that the medians lie within the centromeric region on all cell sources shows both that the majority of the bad bins are in the (peri)centromeric region <em>and</em> there are approximately equally many on each side.</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/05_rec_compartments.ipynb" data-notebook-title="Compartments Analysis w. recommended parameters" data-notebook-cellid="cell-fig-e1_nan_hist">
<div id="cell-fig-e1_nan_hist" class="cell" data-execution_count="108">
<div class="cell-output cell-output-display">
<div id="fig-e1_nan_hist" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-e1_nan_hist-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-05_rec_compartments-fig-e1_nan_hist-output-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-25" title="Figure&nbsp;11: Histogram of NaN values in the E1 eigenvector for each cell type. Median position is marked with a red dashed line."><img src="index_files/figure-html/..-notebooks-05_rec_compartments-fig-e1_nan_hist-output-1.png" width="595" height="176" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-e1_nan_hist-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;11: Histogram of NaN values in the E1 eigenvector for each cell type. Median position is marked with a red dashed line.
</figcaption>
</figure>
</div>
</div>
</div>
</div>
</section>
</section>
<section id="sec-results-eigenvectors" class="level3">
<h3 class="anchored" data-anchor-id="sec-results-eigenvectors">Compartments (Eigenvectors)</h3>
<p>The three viewframes (<em>Full</em>, <em>Arms</em>, <em>10Mb</em>) used for the calculation of the eigenvectors captured different variability in the data (<a href="#fig-e1-matrix-full-arms-10mb-round_spermatid" class="quarto-xref">Figure&nbsp;12</a>), and as expected, the inferred compartments (colored red on the E1 tracks) are more abundant and smaller with smaller viewframes. To determine how well each of the E1 tracks capture the pattern in the interaction matrix, we can overlay the matrix with the E1 sign-change and visually determine if the squares reflect the E1 sign change (<a href="#fig-e1-matrix-full-arms-10mb-round_spermatid" class="quarto-xref">Figure&nbsp;12</a>).</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/07_various_plotting.ipynb" data-notebook-title="Various Comparisons" data-notebook-cellid="cell-fig-e1-matrix-full-arms-10mb-round_spermatid">
<div id="fig-e1-matrix-full-arms-10mb-round_spermatid" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-e1-matrix-full-arms-10mb-round_spermatid-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-e1-matrix-full-arms-10mb-round_spermatid" style="flex-basis: 50.0%;justify-content: flex-start;">
<div id="fig-e1-matrix-full-arms-10mb-round_spermatid-1" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-e1-matrix-full-arms-10mb-round_spermatid-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-e1-matrix-full-arms-10mb-round_spermatid-output-1.png" class="lightbox" data-gallery="fig-e1-matrix-full-arms-10mb-round_spermatid" title="Figure&nbsp;12&nbsp;(a): 100kb resolution"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-e1-matrix-full-arms-10mb-round_spermatid-output-1.png" data-ref-parent="fig-e1-matrix-full-arms-10mb-round_spermatid" width="297" height="277" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-e1-matrix-full-arms-10mb-round_spermatid-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) 100kb resolution
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-e1-matrix-full-arms-10mb-round_spermatid" style="flex-basis: 50.0%;justify-content: flex-start;">
<div id="fig-e1-matrix-full-arms-10mb-round_spermatid-2" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-e1-matrix-full-arms-10mb-round_spermatid-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-e1-matrix-full-arms-10mb-round_spermatid-output-2.png" class="lightbox" data-gallery="fig-e1-matrix-full-arms-10mb-round_spermatid" title="Figure&nbsp;12&nbsp;(b): 500kb resolution"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-e1-matrix-full-arms-10mb-round_spermatid-output-2.png" data-ref-parent="fig-e1-matrix-full-arms-10mb-round_spermatid" width="297" height="277" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-e1-matrix-full-arms-10mb-round_spermatid-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) 500kb resolution
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-e1-matrix-full-arms-10mb-round_spermatid-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;12: E1 eigenvector values for merged round spermatid samples at a) 100kb or b) 500kb resolution, as well as the interaction matrix. E1 was restricted to either Full-chromosome (top), Chromosome-arms (middle), or 10Mb windows (bottom)
</figcaption>
</figure>
</div>
</div>
<p>I decide that without more finescaled knowledge than the position of the centromeres, the arbitrary size of the 10 Mb windowed E1 can not fully be justified. That is, we could arbitrarily calculate any windowed E1 track. Also, <span class="citation" data-cites="wang2019reprogrammingmeioticchromatin">Wang et al. (<a href="#ref-wang2019reprogrammingmeioticchromatin" role="doc-biblioref">2019</a>)</span> concludes only for pachytene spermatocyte to show local interactions in the 10Mb viewframe (what they refer to as <em>refined A/B-compartments</em>), and all the other stages of spermatogenesis were consistent with the conventional A/B compartments. The reasonable thing to do is therefore to continue the analysis, focusing on the arms-restricted eigendecomposition. Nevertheless, we also keep <em>refined</em> compartments in the analysis.</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/07_various_plotting.ipynb" data-notebook-title="Various Comparisons" data-notebook-cellid="cell-fig-rs100-recpe-pe">
<div class="cell" data-layout="[[-0.5,5,5,-0.5],[-0.5,5,5,-0.5]]" data-execution_count="21">
<div id="fig-rs100-recpe-pe" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-rs100-recpe-pe-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 4.6%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-rs100-recpe-pe" style="flex-basis: 45.5%;justify-content: flex-start;">
<div id="fig-rs100-recpe-pe-1" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-rs100-recpe-pe-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs100-recpe-pe-output-1.png" class="lightbox" data-gallery="fig-rs100-recpe-pe" title="Figure&nbsp;13&nbsp;(a): 5unique: chrX:start-end"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs100-recpe-pe-output-1.png" data-ref-parent="fig-rs100-recpe-pe" width="247" height="195" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-rs100-recpe-pe-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) <code>5unique</code>: chrX:start-end
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-rs100-recpe-pe" style="flex-basis: 45.5%;justify-content: flex-start;">
<div id="fig-rs100-recpe-pe-2" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-rs100-recpe-pe-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs100-recpe-pe-output-2.png" class="lightbox" data-gallery="fig-rs100-recpe-pe" title="Figure&nbsp;13&nbsp;(b): mask: chrX:start-end"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs100-recpe-pe-output-2.png" data-ref-parent="fig-rs100-recpe-pe" width="247" height="195" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-rs100-recpe-pe-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) <code>mask</code>: chrX:start-end
</figcaption>
</figure>
</div>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 4.6%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
</div>
<div class="quarto-layout-row">
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 4.6%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-rs100-recpe-pe" style="flex-basis: 45.5%;justify-content: flex-start;">
<div id="fig-rs100-recpe-pe-3" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-rs100-recpe-pe-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs100-recpe-pe-output-3.png" class="lightbox" data-gallery="fig-rs100-recpe-pe" title="Figure&nbsp;13&nbsp;(c): 5unique: chrX:70Mb-78Mb"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs100-recpe-pe-output-3.png" data-ref-parent="fig-rs100-recpe-pe" width="247" height="201" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-rs100-recpe-pe-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(c) <code>5unique</code>: chrX:70Mb-78Mb
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-rs100-recpe-pe" style="flex-basis: 45.5%;justify-content: flex-start;">
<div id="fig-rs100-recpe-pe-4" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-rs100-recpe-pe-4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs100-recpe-pe-output-4.png" class="lightbox" data-gallery="fig-rs100-recpe-pe" title="Figure&nbsp;13&nbsp;(d): mask: chrX:70Mb-78Mb"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs100-recpe-pe-output-4.png" data-ref-parent="fig-rs100-recpe-pe" width="247" height="201" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-rs100-recpe-pe-4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(d) <code>mask</code>: chrX:70Mb-78Mb
</figcaption>
</figure>
</div>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 4.6%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-rs100-recpe-pe-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;13: Round Spermatid (RS) at 100kb, comparing the impact of parsing parameters
</figcaption>
</figure>
</div>
</div>
</div>
<p>Additionally, as I created coolers with two different sets of parsing parameters we will compare the resulting matrices and their compartments (<a href="#fig-rs100-recpe-pe" class="quarto-xref">Figure&nbsp;13</a>). As expected, we observe more empty bins in the Hi-C matrix when comparing the initial run (<code>mask</code>) to the recommended parameters (<code>5unique</code>), but otherwise, the interaction pattern is indestinguishable. The effect on the E1 is more noticable, where the absolute magnitude of the E1 values is generally smaller. There is, however, a small region that changes sign (from A to B) on the 10Mb-windowed (‘refined’) E1 track (<a href="#fig-rs100-recpe-pe" class="quarto-xref">Figure&nbsp;13</a>;c+d). This region is surrounded by added empty bins, which could mean that too many low quality pairs in <code>mask</code> were introducing bias and swapped the sign of E1. It is supported by the fact that the sign change <em>only</em> occured in <em>refined</em> E1, and that the sign after filtering weak pairs (<span class="math inline">\(mapq &lt; 30\)</span>) is consistent with the <em>arms</em> view. It supports my previous postulate that it is better to use a viewframe with explicit molecular meaning than one of an arbitrary window size. That said, the <code>mapq</code> threshold should really be determined taking both coverage and resolution into account. For our purposes, and with the <em>arms</em> view, the mapping- and parsing parameters do not seem to be too sensitive.</p>
<p>To emphasize the findings, the sets of A-compartments were compared between the two parsing runs, showing almost identical compartment calls. Additionally, the set difference was 8 bins between PE and recPE for round spermatid 100kb and 5 bins for fibroblast for <em>arms</em> viewframe (<a href="#fig-rs-fb-100-pe-recpe-intervals" class="quarto-xref">Figure&nbsp;14</a>; a+b, respectively). We observe a high number of differences around 76Mb for the refined compartments (10Mb) of round spermatid, which is consistent with the sign-flip of E1 values discussed earlier. Anything else would be surprising, as it is the same data, but visualized in a different way.</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/07_various_plotting.ipynb" data-notebook-title="Various Comparisons" data-notebook-cellid="cell-fig-rs-fb-100-pe-recpe-intervals">
<div class="cell" data-layout="[[-0.6,2.5,2.5,-0.6],[-0.6,2.5,2.5,-0.6]]" data-execution_count="103">
<div id="fig-rs-fb-100-pe-recpe-intervals" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-rs-fb-100-pe-recpe-intervals-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 9.7%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-rs-fb-100-pe-recpe-intervals" style="flex-basis: 40.3%;justify-content: flex-start;">
<div id="fig-rs-fb-100-pe-recpe-intervals-1" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-rs-fb-100-pe-recpe-intervals-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs-fb-100-pe-recpe-intervals-output-1.png" class="lightbox" data-gallery="fig-rs-fb-100-pe-recpe-intervals" title="Figure&nbsp;14&nbsp;(a): RS: arms"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs-fb-100-pe-recpe-intervals-output-1.png" data-ref-parent="fig-rs-fb-100-pe-recpe-intervals" width="249" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-rs-fb-100-pe-recpe-intervals-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) RS: arms
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-rs-fb-100-pe-recpe-intervals" style="flex-basis: 40.3%;justify-content: flex-start;">
<div id="fig-rs-fb-100-pe-recpe-intervals-2" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-rs-fb-100-pe-recpe-intervals-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs-fb-100-pe-recpe-intervals-output-2.png" class="lightbox" data-gallery="fig-rs-fb-100-pe-recpe-intervals" title="Figure&nbsp;14&nbsp;(b): Fib: arms"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs-fb-100-pe-recpe-intervals-output-2.png" data-ref-parent="fig-rs-fb-100-pe-recpe-intervals" width="249" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-rs-fb-100-pe-recpe-intervals-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) Fib: arms
</figcaption>
</figure>
</div>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 9.7%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
</div>
<div class="quarto-layout-row">
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 9.7%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-rs-fb-100-pe-recpe-intervals" style="flex-basis: 40.3%;justify-content: flex-start;">
<div id="fig-rs-fb-100-pe-recpe-intervals-3" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-rs-fb-100-pe-recpe-intervals-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs-fb-100-pe-recpe-intervals-output-3.png" class="lightbox" data-gallery="fig-rs-fb-100-pe-recpe-intervals" title="Figure&nbsp;14&nbsp;(c): RS: 10Mb"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs-fb-100-pe-recpe-intervals-output-3.png" data-ref-parent="fig-rs-fb-100-pe-recpe-intervals" width="249" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-rs-fb-100-pe-recpe-intervals-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(c) RS: 10Mb
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-rs-fb-100-pe-recpe-intervals" style="flex-basis: 40.3%;justify-content: flex-start;">
<div id="fig-rs-fb-100-pe-recpe-intervals-4" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-rs-fb-100-pe-recpe-intervals-4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs-fb-100-pe-recpe-intervals-output-4.png" class="lightbox" data-gallery="fig-rs-fb-100-pe-recpe-intervals" title="Figure&nbsp;14&nbsp;(d): Fib: 10Mb"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs-fb-100-pe-recpe-intervals-output-4.png" data-ref-parent="fig-rs-fb-100-pe-recpe-intervals" width="249" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-rs-fb-100-pe-recpe-intervals-4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(d) Fib: 10Mb
</figcaption>
</figure>
</div>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 9.7%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-rs-fb-100-pe-recpe-intervals-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;14: Round Spermatid (RS) and Fibroblast (Fb) at 100kb, comparing the impact of parsing parameters on A-compartment calling at different viewframes; <em>arms</em>, <em>10Mb</em>. PE: initial parse (masking complex walks); recPE: recommended parse (reporting the 5’most unique alignment of a complex walk).
</figcaption>
</figure>
</div>
</div>
</div>
<p>The observed difference between the sets can for our data be attributed to chance, but we cannot draw general conclusions about the parameters in general. I argue that the quality and size of the Hi-C library will influence sensitive to parsing parameters. In that case, the most flexible approach is still to follow the recommendations from <code>cooler</code> to report more pairs as valid contacts, and then create coolers with different <em>mapq</em> filters if issues are encountered.</p>
</section>
</section>
<section id="comparing-genomic-intervals" class="level2">
<h2 class="anchored" data-anchor-id="comparing-genomic-intervals">Comparing Genomic Intervals</h2>
<p>The comparison of genomic intervals was modified along the way, as we gained more knowledge and caught mistakes in the original implementation (especially of the proxomity test). Here, I only report the results of the final version of the software, but the modifications and implications hereof are discussed in <a href="#sec-discussion" class="quarto-xref">Section&nbsp;4</a>.</p>
<section id="compartment-edges-transition-zones" class="level3">
<h3 class="anchored" data-anchor-id="compartment-edges-transition-zones">Compartment Edges (transition zones)</h3>
<p>We compare how the ECH90 regions fit when queried on top of the A-compartments and equivalently for the edges, for fibroblasts and round spermatids at 100kb resolution. When queried against the edges instead, the the total set size is reduced to less than 50%. Interestingly, some of the intersections between A-compartments and ECH90 remain, and new ones appear as we move to the outside edge of the compartment (<a href="#fig-comps-edges-ech" class="quarto-xref">Figure&nbsp;15</a>). This indicates that most, but not all, of the intersection between ECH90 regions and the A-compartments are within 100kb of the compartment edge, and additional overlap is gained if we define a transition zone on the outside of the edge as well. To visualize this (outside) edge enrichment, we find the set difference of the ECH-intersection to compartments and edges, respectively (<a href="#fig-edge-enrichment" class="quarto-xref">Figure&nbsp;16</a>), thus removing all the ‘inside’ edges. We observe that in almost all of the of the regions of <span class="math inline">\(ECH \cap Comp\)</span> are accompanied by an edge also intersecting ECH (<span class="math inline">\(ECH \cap Edge\)</span>), localized where the <em>Diff</em> track aligns (within 100kb) with both <span class="math inline">\(CompInt\)</span> and <span class="math inline">\(EdgeInt\)</span>.</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/07_various_plotting.ipynb" data-notebook-title="Various Comparisons" data-notebook-cellid="cell-fig-comps-edges-ech">
<div class="cell" data-layout="[[-0.05,3.2,3.2,-0.05],[-0.05,3.2,3.2,-0.05]]" data-execution_count="20">
<div id="fig-comps-edges-ech" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-comps-edges-ech-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 0.8%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-comps-edges-ech" style="flex-basis: 49.2%;justify-content: flex-start;">
<div id="fig-comps-edges-ech-1" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-comps-edges-ech-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-comps-edges-ech-output-1.png" class="lightbox" data-gallery="fig-comps-edges-ech" title="Figure&nbsp;15&nbsp;(a): Fibroblast A-compartments"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-comps-edges-ech-output-1.png" data-ref-parent="fig-comps-edges-ech" width="318" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-comps-edges-ech-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) Fibroblast A-compartments
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-comps-edges-ech" style="flex-basis: 49.2%;justify-content: flex-start;">
<div id="fig-comps-edges-ech-2" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-comps-edges-ech-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-comps-edges-ech-output-2.png" class="lightbox" data-gallery="fig-comps-edges-ech" title="Figure&nbsp;15&nbsp;(b): Round Spermatid A-compartments"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-comps-edges-ech-output-2.png" data-ref-parent="fig-comps-edges-ech" width="318" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-comps-edges-ech-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) Round Spermatid A-compartments
</figcaption>
</figure>
</div>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 0.8%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
</div>
<div class="quarto-layout-row">
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 0.8%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-comps-edges-ech" style="flex-basis: 49.2%;justify-content: flex-start;">
<div id="fig-comps-edges-ech-3" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-comps-edges-ech-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-comps-edges-ech-output-3.png" class="lightbox" data-gallery="fig-comps-edges-ech" title="Figure&nbsp;15&nbsp;(c): Fibroblast edges"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-comps-edges-ech-output-3.png" data-ref-parent="fig-comps-edges-ech" width="318" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-comps-edges-ech-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(c) Fibroblast edges
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-comps-edges-ech" style="flex-basis: 49.2%;justify-content: flex-start;">
<div id="fig-comps-edges-ech-4" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-comps-edges-ech-4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-comps-edges-ech-output-4.png" class="lightbox" data-gallery="fig-comps-edges-ech" title="Figure&nbsp;15&nbsp;(d): Round Spermatid edges"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-comps-edges-ech-output-4.png" data-ref-parent="fig-comps-edges-ech" width="318" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-comps-edges-ech-4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(d) Round Spermatid edges
</figcaption>
</figure>
</div>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 0.8%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-comps-edges-ech-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;15: Visual representation of the genomic intervals of ECH90, A-compartments (a+b), edges (c+d), and their intersections. Shown fibroblast (a+c) and round spermatid (b+d) at 100kb resolution and arms viewframe.
</figcaption>
</figure>
</div>
</div>
</div>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/07_various_plotting.ipynb" data-notebook-title="Various Comparisons" data-notebook-cellid="cell-fig-edge-enrichment">
<div class="cell" data-layout="[-0.05,3.2,3.2,-0.05]" data-execution_count="15">
<div id="fig-edge-enrichment" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-edge-enrichment-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 0.8%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-edge-enrichment" style="flex-basis: 49.2%;justify-content: flex-start;">
<div id="fig-edge-enrichment-1" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-edge-enrichment-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-edge-enrichment-output-1.png" class="lightbox" data-gallery="fig-edge-enrichment" title="Figure&nbsp;16&nbsp;(a): Round Spermatid"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-edge-enrichment-output-1.png" data-ref-parent="fig-edge-enrichment" width="318" height="98" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-edge-enrichment-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) Round Spermatid
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-edge-enrichment" style="flex-basis: 49.2%;justify-content: flex-start;">
<div id="fig-edge-enrichment-2" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-edge-enrichment-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-edge-enrichment-output-2.png" class="lightbox" data-gallery="fig-edge-enrichment" title="Figure&nbsp;16&nbsp;(b): Fibroblast"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-edge-enrichment-output-2.png" data-ref-parent="fig-edge-enrichment" width="318" height="98" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-edge-enrichment-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) Fibroblast
</figcaption>
</figure>
</div>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 0.8%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-edge-enrichment-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;16: Visual representation of the enrichment of edges in the intersection of ECH90 and A-compartments. Shown round spermatid (a) and fibroblast (b) at 100kb resolution and arms viewframe. Note that the edge-regions are too small to be distinguished visually from the compartment on the graph, making it look like they overlap, even though the difference is reported.
</figcaption>
</figure>
</div>
</div>
</div>
<p>I apply both the proximity and Jaccard test to see how well the observations could be explained by chance. For completeness, the tests are performed for all cell types, but only use 100kb resolution and arms viewframe. We observe that both fibroblast and round spermatids have significant overlap (Jaccard p-values <span class="math inline">\(0.012\)</span> and <span class="math inline">\(0.010\)</span>, respectively), meaning the edge zone have more intersection with the ECH regions than expected by chance. None of the samples passed the proximity test, meaning the non-overlapping segments were no more proximal than expected by chance. A significant proximity test (when performed on the edges) might provide information about the potential of expanding or moving the transition window. That is, if the non-overlapping regions are <em>very</em> proximal, a larger (or shifted) window to only capture the 200kb region outside of the edge might be favourable. With the very small amount of observed proximity (high p-values), we might consider shrinking the windows.</p>
</section>
<section id="testing-against-baboon-regions-under-strong-selection" class="level3">
<h3 class="anchored" data-anchor-id="testing-against-baboon-regions-under-strong-selection">Testing against baboon regions under strong selection</h3>
<p>The data for this analysis was provided by Kasper Munch in bed-like format, mapped to <em>panu_3.0</em> (PapAnu4) assembly. The intervals define genomic regions in a hybrid/migrating population of baboon where strong negative selection acts against minor parent ancestry <span class="citation" data-cites="sorensen2023genomewidecoancestryreveals">(<a href="#ref-sorensen2023genomewidecoancestryreveals" role="doc-biblioref">Sørensen et al. 2023</a>)</span>. The segments had to be lifted to <em>rheMac10</em> coordinates to be able to correlate the two sets of intervals. The original UCSC-liftOver <span class="citation" data-cites="hinrichs2006ucscgenomebrowser">(<a href="#ref-hinrichs2006ucscgenomebrowser" role="doc-biblioref">Hinrichs 2006</a>)</span> is very strict and does not try to conserve segments in favor of accuracy e.g.&nbsp;inversions or small indels, which results in highly fragmented regions when lifted to another assembly, if the segments are not continouos on the new assembly. For our analysis, it is not the exact genomic position or order of sub-genic regions that are important, but rather, the start and end coordinates of each segment are quantified. To favor preservation of segments, we use <em>segment_liftover</em> <span class="citation" data-cites="gao2018segment_liftoverpythontool">(<a href="#ref-gao2018segment_liftoverpythontool" role="doc-biblioref">Gao, Huang, and Baudis 2018</a>)</span>, resulting in much more similar segments to the original (<a href="#fig-compare-liftover" class="quarto-xref">Figure&nbsp;17</a>). As no chain file from <em>panu_3.0</em> to <em>Mmul_10</em> was available, we used <em>panu_2.0</em> as intermediate.</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/07_various_plotting.ipynb" data-notebook-title="Various Comparisons" data-notebook-cellid="cell-fig-compare-liftover">
<div class="cell" data-execution_count="21">
<div id="fig-compare-liftover" class="cell quarto-float quarto-figure quarto-figure-center anchored" data-execution_count="21">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-compare-liftover-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output cell-output-display">
<div id="fig-compare-liftover-1" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-compare-liftover-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-compare-liftover-output-1.png" class="lightbox" data-gallery="fig-compare-liftover" title="Figure&nbsp;17&nbsp;(a): high-olive"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-compare-liftover-output-1.png" data-ref-parent="fig-compare-liftover" width="599" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-compare-liftover-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) high-olive
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display">
<div id="fig-compare-liftover-2" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-compare-liftover-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-compare-liftover-output-2.png" class="lightbox" data-gallery="fig-compare-liftover" title="Figure&nbsp;17&nbsp;(b): high-hama"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-compare-liftover-output-2.png" data-ref-parent="fig-compare-liftover" width="599" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-compare-liftover-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) high-hama
</figcaption>
</figure>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-compare-liftover-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;17: Comparison of a) high-olive and b) high-hama intervals between Panu_3.0, segment_liftover, or UCSC liftOver coordinates when lifting from PapAnu4–&gt;PapAnu3–&gt;rheMac10.
</figcaption>
</figure>
</div>
</div>
</div>
<p>Initially, the compartment edges of round spermatid at 100kb resolution (RS100) were plotted against the lifted coordinates from the baboon hybrid population, where either all the sampled individuals have olive (<em>P. anubis</em>) ancestry or 95% of the sampled individuals have hamadryas (<em>P. hamadryas</em> ancestry. Their respective intersections were plotted underneath. We expect less intersection for <em>hamadryas</em> than for <em>anubis</em> as the total set size is much smaller. The compartment edges and <em>Papio anubis</em>-derived regions(<a href="#fig-baboon-rs100-intersect" class="quarto-xref">Figure&nbsp;18</a>; b) seem to be highly enriched in the first 25 Mbp, and thus it has a high degree of intersection with the compartment edges. Interestingly, the ECH90 set is nearly empty in that region, which could be useful for determining the mechanism for selecting against the <em>P.hamadryas</em> ancestral allele in the hybrid baboon population. The <em>P.hamadryas</em>-derived regions seem intersect the compartment edges more centered on the chromosome (<a href="#fig-baboon-rs100-intersect" class="quarto-xref">Figure&nbsp;18</a>; a). The proximity test initially ruled out that the non-intersecting parts of the respective regions were this proximal by chance. However, after updating the method to exclude whole segments that partially overlaps (instead of keeping their non-overlapping parts), there was no statistical significance. Additionally, the Jaccard test revealed that the intersection between the RS100 and both Hi-<em>P.hama</em> and Hi-<em>P.anu</em> can be explained by chance alone with no p-values below <span class="math inline">\(0.05\)</span>.</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/07_various_plotting.ipynb" data-notebook-title="Various Comparisons" data-notebook-cellid="cell-fig-baboon-rs100-intersect">
<div class="cell" data-layout="[[-0.05,3.2,3.2,-0.05],[-0.05,3.2,3.2,-0.05]]" data-execution_count="26">
<div id="fig-baboon-rs100-intersect" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-baboon-rs100-intersect-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 0.8%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-baboon-rs100-intersect" style="flex-basis: 49.2%;justify-content: flex-start;">
<div id="fig-baboon-rs100-intersect-1" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-baboon-rs100-intersect-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-baboon-rs100-intersect-output-1.png" class="lightbox" data-gallery="fig-baboon-rs100-intersect" title="Figure&nbsp;18&nbsp;(a): P. hamadryas + arms E1"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-baboon-rs100-intersect-output-1.png" data-ref-parent="fig-baboon-rs100-intersect" width="318" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-baboon-rs100-intersect-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) <em>P. hamadryas</em> + arms E1
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-baboon-rs100-intersect" style="flex-basis: 49.2%;justify-content: flex-start;">
<div id="fig-baboon-rs100-intersect-2" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-baboon-rs100-intersect-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-baboon-rs100-intersect-output-2.png" class="lightbox" data-gallery="fig-baboon-rs100-intersect" title="Figure&nbsp;18&nbsp;(b): P. anubis + arms E1"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-baboon-rs100-intersect-output-2.png" data-ref-parent="fig-baboon-rs100-intersect" width="318" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-baboon-rs100-intersect-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) <em>P. anubis</em> + arms E1
</figcaption>
</figure>
</div>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 0.8%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
</div>
<div class="quarto-layout-row">
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 0.8%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-baboon-rs100-intersect" style="flex-basis: 49.2%;justify-content: flex-start;">
<div id="fig-baboon-rs100-intersect-3" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-baboon-rs100-intersect-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-baboon-rs100-intersect-output-3.png" class="lightbox" data-gallery="fig-baboon-rs100-intersect" title="Figure&nbsp;18&nbsp;(c): P. hamadryas + 10Mb E1"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-baboon-rs100-intersect-output-3.png" data-ref-parent="fig-baboon-rs100-intersect" width="318" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-baboon-rs100-intersect-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(c) <em>P. hamadryas</em> + 10Mb E1
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-baboon-rs100-intersect" style="flex-basis: 49.2%;justify-content: flex-start;">
<div id="fig-baboon-rs100-intersect-4" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-baboon-rs100-intersect-4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-baboon-rs100-intersect-output-4.png" class="lightbox" data-gallery="fig-baboon-rs100-intersect" title="Figure&nbsp;18&nbsp;(d): P. anubis + 10Mb E1"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-baboon-rs100-intersect-output-4.png" data-ref-parent="fig-baboon-rs100-intersect" width="318" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-baboon-rs100-intersect-4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(d) <em>P. anubis</em> + 10Mb E1
</figcaption>
</figure>
</div>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 0.8%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-baboon-rs100-intersect-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;18: Comparing the A-compartment edges of round spermatid (RS) with regions in baboons from hybrid population where a) 95% of sampled individuals have <em>Papio hamadryas</em> ancestry or b) 100% of the samples have <em>Papio anubis</em> ancestry. The regions are extracted and lifted from PapAnu4 to rhemac10 using segment_liftover.
</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="testing-limits" class="level3">
<h3 class="anchored" data-anchor-id="testing-limits">Testing limits</h3>
<p>We then abandoned the idea of a 200kb transition zone between the compartments. It was introduced as a less stringent edge, allowing the intersection of the selected regions to be within a defined threshold. We hoped a weaker threshold could allow for more variance, but it turned out to be the opposite (<a href="#tbl-svedig-tabel-significant" class="quarto-xref">Table&nbsp;6</a>). After defining the compartment limits (<code>_edge_1bp</code>), we perform tests on combinations of all five sources (fibroblast, spermatogonia, pachytene spermatocyte, round spermatid, sperm) and 2 viewframes (arms, 10Mb) and 5 queries. The queries were modified as such; the full ECH was used unmodified for the tests as it was already shown to overlap (Jaccard) fibroblast and round spermatid. <em>hama</em> and <em>olive</em> limits were used separately and as a concatenated set (<code>hamaolive_edge_1pb</code>), to increase the effect size, especially as the <em>hama</em> set is quite small compared to the rest of the sets. By their definition, <em>olive</em> and <em>hama</em> do not overlap, and can safely be concatenated (see <a href="#fig-introduce-selected-regions" class="quarto-xref">Figure&nbsp;1</a> for confirmation). Now, we observe significant p-values across all cell types.</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/07_various_plotting.ipynb" data-notebook-title="Various Comparisons" data-notebook-cellid="cell-tbl-svedig-tabel-significant">
<div class="cell" data-execution_count="38">
<div id="tbl-svedig-tabel-significant" class="cell quarto-float quarto-figure quarto-figure-center anchored" data-execution_count="38">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-svedig-tabel-significant-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;6: Test results from proximity and Jaccard, comparing ECH (full regions) or baboon 1bp limits with compartment 1bp limits from all cell types. Significant p-values across tissue types. All insignificant p-values are filtered out. <em>Fb: Fibroblast, Spa: Spermatogonia, Pac: Pachytene Spermatocyte, RS: Round Spermatid, Sperm: Sperm</em>
</figcaption>
<div aria-describedby="tbl-svedig-tabel-significant-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output cell-output-display" data-execution_count="38">
<style type="text/css">
</style>

<table id="T_a1cfd" class="do-not-create-environment cell caption-top table table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th id="T_a1cfd_level0_col0" class="col_heading level0 col0" data-quarto-table-cell-role="th">View</th>
<th id="T_a1cfd_level0_col1" class="col_heading level0 col1" data-quarto-table-cell-role="th">Query</th>
<th id="T_a1cfd_level0_col2" class="col_heading level0 col2" data-quarto-table-cell-role="th">Test</th>
<th id="T_a1cfd_level0_col3" class="col_heading level0 col3" data-quarto-table-cell-role="th">Fb</th>
<th id="T_a1cfd_level0_col4" class="col_heading level0 col4" data-quarto-table-cell-role="th">Spa</th>
<th id="T_a1cfd_level0_col5" class="col_heading level0 col5" data-quarto-table-cell-role="th">Pac</th>
<th id="T_a1cfd_level0_col6" class="col_heading level0 col6" data-quarto-table-cell-role="th">RS</th>
<th id="T_a1cfd_level0_col7" class="col_heading level0 col7" data-quarto-table-cell-role="th">Sperm</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td id="T_a1cfd_row0_col0" class="data row0 col0">10Mb</td>
<td id="T_a1cfd_row0_col1" class="data row0 col1">ECH90</td>
<td id="T_a1cfd_row0_col2" class="data row0 col2">jaccard</td>
<td id="T_a1cfd_row0_col3" class="data row0 col3">0.021460</td>
<td id="T_a1cfd_row0_col4" class="data row0 col4">0.006110</td>
<td id="T_a1cfd_row0_col5" class="data row0 col5">-</td>
<td id="T_a1cfd_row0_col6" class="data row0 col6">-</td>
<td id="T_a1cfd_row0_col7" class="data row0 col7">0.047500</td>
</tr>
<tr class="even">
<td id="T_a1cfd_row1_col0" class="data row1 col0">10Mb</td>
<td id="T_a1cfd_row1_col1" class="data row1 col1">olive_1bp</td>
<td id="T_a1cfd_row1_col2" class="data row1 col2">proximity</td>
<td id="T_a1cfd_row1_col3" class="data row1 col3">-</td>
<td id="T_a1cfd_row1_col4" class="data row1 col4">-</td>
<td id="T_a1cfd_row1_col5" class="data row1 col5">-</td>
<td id="T_a1cfd_row1_col6" class="data row1 col6">-</td>
<td id="T_a1cfd_row1_col7" class="data row1 col7">0.016140</td>
</tr>
<tr class="odd">
<td id="T_a1cfd_row2_col0" class="data row2 col0">10Mb</td>
<td id="T_a1cfd_row2_col1" class="data row2 col1">olivehama_1bp</td>
<td id="T_a1cfd_row2_col2" class="data row2 col2">proximity</td>
<td id="T_a1cfd_row2_col3" class="data row2 col3">-</td>
<td id="T_a1cfd_row2_col4" class="data row2 col4">0.018790</td>
<td id="T_a1cfd_row2_col5" class="data row2 col5">0.012300</td>
<td id="T_a1cfd_row2_col6" class="data row2 col6">-</td>
<td id="T_a1cfd_row2_col7" class="data row2 col7">0.001280</td>
</tr>
<tr class="even">
<td id="T_a1cfd_row3_col0" class="data row3 col0">arms</td>
<td id="T_a1cfd_row3_col1" class="data row3 col1">ECH90</td>
<td id="T_a1cfd_row3_col2" class="data row3 col2">jaccard</td>
<td id="T_a1cfd_row3_col3" class="data row3 col3">0.016560</td>
<td id="T_a1cfd_row3_col4" class="data row3 col4">0.006120</td>
<td id="T_a1cfd_row3_col5" class="data row3 col5">0.005340</td>
<td id="T_a1cfd_row3_col6" class="data row3 col6">0.027690</td>
<td id="T_a1cfd_row3_col7" class="data row3 col7">-</td>
</tr>
<tr class="odd">
<td id="T_a1cfd_row4_col0" class="data row4 col0">arms</td>
<td id="T_a1cfd_row4_col1" class="data row4 col1">olivehama_1bp</td>
<td id="T_a1cfd_row4_col2" class="data row4 col2">proximity</td>
<td id="T_a1cfd_row4_col3" class="data row4 col3">-</td>
<td id="T_a1cfd_row4_col4" class="data row4 col4">-</td>
<td id="T_a1cfd_row4_col5" class="data row4 col5">0.035640</td>
<td id="T_a1cfd_row4_col6" class="data row4 col6">-</td>
<td id="T_a1cfd_row4_col7" class="data row4 col7">-</td>
</tr>
</tbody>
</table>
</div>
</div>
</figure>
</div>
</div>
</div>
<p>Reassuringly, both fibroblast and round spermatid still have significant Jaccard p-values for the arms viewframe although the p-values have increased (to <span class="math inline">\(0.017\)</span> and <span class="math inline">\(0.028\)</span>, respectively). Now, all tissues except sperm show significant overlap with ECH regions, especially spermatogonia and pachytene spermatocytes with p-values <span class="math inline">\(0.006\)</span> and <span class="math inline">\(0.005\)</span>, respectively. Only fibroblast and spermatogonia significantly overlap the ECH region for both viewframes, indicating that their predicted compartments shift only minimally compared to the pachytene spermatocyte and round spermatid when narrowing the viewframe. Pachytene spermatocyte and round spermatid loose their significance when narrowing the viewframe, meaning their compartments are highly varied between the two views, and the same conclusion can be drawn about sperm, as it gains significance when the viewframe is narrowed. Interestingly, only sperm overlaps the baboon regions with olive ancestry, and no tissue overlaps with hamadryas ancestry. Conversely, when concatenating the olive and hamadryas sets to capture all selection in baboons, both spermatogonia, pachytene spermatocyte, and sperm are significantly proximal. Here, sperm, which is not proximal in the arms viewframe, has the lowest p-value (<span class="math inline">\(p=0.001\)</span>).</p>
</section>
</section>
</section>
<section id="sec-discussion" class="level1">
<h1>Discussion</h1>
<section id="on-reproducibility" class="level2">
<h2 class="anchored" data-anchor-id="on-reproducibility">On Reproducibility</h2>
<p>As an extensive reproducibility infrastructere became a major part of the project, I have to reflect upon the advantages it has, but also the consequences it inevitably brings along. First, reproducibility is the basis of scientific progression, and idealisticly it should come before the discovery itself. But let us not pretend that reproducibility is favored above hypothesis testing to be the driving forces of progression, and let us not pretend that it has a singular definition. There are many layers to reproducibility, and we do, of course, not need to host a website or to construct a single-command pipeline for reproducing the analysis to call a study <em>reproducible</em>. Additionally, the option only exist for analyses that are strictly computational. It is, however, very useful and can greatly improve downstream efficiency, and I find it important to at least explore the available options. If such a pipeline had existed, I would not have had to spend weeks to update the results of <span class="citation" data-cites="wang2019reprogrammingmeioticchromatin">Wang et al. (<a href="#ref-wang2019reprogrammingmeioticchromatin" role="doc-biblioref">2019</a>)</span> to reflect a more recent reference genome. With such a pipeline, it is now easy to extend the analysis to include more samples and other species. Additionally, if they had made the coordinates of the compartments available, I could probably have used them directly and start by comparing genomic intervals. Instead, this became a study more of Hi-C data analysis than comparing genomic intervals.</p>
<p>At the time of writing, the repository is <em>almost there</em> w.r.t. reproducibility, but still not quite. It only reproduces the final analysis done with <em>bwa</em>, <em>pairtools</em>, <em>cooler</em>, <em>cooltools</em>, and not the preliminary analyses comparing aligners and using <em>HicExplorer</em>. This is the intention, as most use-cases would not need the results of a preliminary methodological comparison. However, the workflows are saved under <code>scripts/</code> and can be run from there, or they can be added to the main <code>workflow.py</code> as separate targets. A small amount of refactoring of the directory structure is required as well, because some files are fetched from a parent folder to the project directory. Additionally, the main workflow is intensive in I/O and could be heavily optimized by piping between commands instead of writing the outputs to file and reading the files from the next command. However, that would make it harder to debug, as no checkpoints are saved. It would not be as efficient use of resources, which are requested on target level, as the analysis would have to run as a single target. The different steps of the pipeline are either CPU-bound or RAM-bound, but rarely both, and it would take some experimenting to find the optimal compromise. Additionally, the computational steps between a raw <em>.cool</em> file and a balanced, multi-resolution <em>.mcool</em> are performed in a notebook, and the steps should be moved to the workflow. Then, a final target should be added to the workflow, executing all the notebooks (they are named in order), to generate the figures and tables that are used in this manuscript and on the main project pages.</p>
<p>There is a major downside when trying to conform to an idealistic framework—the generation of figures and tables. Throughout this project, I have adhered to the rule that none of the figures and tables should be created or even annotated manually. It is very time-consuming to create production-quality figures that needs no further enhancements before embedding into a manuscript, and even more so do create tables that are both compatible with HTML and PDF. Here, Quarto inherits the limitations of Pandoc (which it uses internally to convert Markdown to its output formats), and it does not work well with complex grouped tables generated in Python to translate into LaTeX (which again is an idealistic typesetting system). As Quarto is a relatively new framework (first release 2022-07-28 <span class="citation" data-cites="quartoteam2022announcingquartonew">Quarto Team (<a href="#ref-quartoteam2022announcingquartonew" role="doc-biblioref">2022</a>)</span>), there are contiously released updates to fix bugs. Therefore, one can spend hours on a workaround for a feature, which could be included in week’s release.</p>
<p>I believe it is an important exercise to force oneself to be cautious about design choices and explore the advantages and limitations of the tools we have available to communicate what we want to communicate.</p>
</section>
<section id="on-methodology" class="level2">
<h2 class="anchored" data-anchor-id="on-methodology">On Methodology</h2>
<p>HiCExplorer is a comprehensive software that consist of an extensive list of ready-made command line functions that calls Python source code. It is modular, except for the parsing of the aligned reads into a raw Hi-C matrix. The documentation provide an extensive list of options to pass to the commands. Nevertheless, they do not provide a Python API (or at least not any documentation thereof). The modules can be imported into a Python-session with standard import syntax, but there are no guiding help pages. Looking through source code revealed that one would have to do something like</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode numberSource python number-lines code-with-copy"><code class="sourceCode python"><span id="cb2-1"><a href="#cb2-1"></a><span class="im">from</span> hicexplorer.hicPlotMatrix <span class="im">import</span> plotHeatmap</span>
<span id="cb2-2"><a href="#cb2-2"></a><span class="im">from</span> hicmatrix <span class="im">import</span> HiCMatrix</span>
<span id="cb2-3"><a href="#cb2-3"></a>plotHeatmap(HiCMatrix(matrix_file)) <span class="co"># and test the required parameters</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>to use the plotting function directly within a notebook. However, it would have to modified to stop saving the plots as files instead of sending them to the display, and it would still require reading from files. Clearly, it is not meant to be used directly within Python, as additionally, the CLI function simply is a wrapper for the <code>main()</code>-function of the module:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode numberSource python number-lines code-with-copy"><code class="sourceCode python"><span id="cb3-1"><a href="#cb3-1"></a><span class="co">#!path/to/python</span></span>
<span id="cb3-2"><a href="#cb3-2"></a><span class="co"># -*- coding: utf-8 -*-</span></span>
<span id="cb3-3"><a href="#cb3-3"></a><span class="im">from</span> hicexplorer.hicPlotMatrix <span class="im">import</span> main</span>
<span id="cb3-4"><a href="#cb3-4"></a><span class="cf">if</span> <span class="va">__name__</span> <span class="op">==</span> <span class="st">"__main__"</span>:</span>
<span id="cb3-5"><a href="#cb3-5"></a>    main()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>HiCExplorer is, however, also integrated into Galaxy <span class="citation" data-cites="thegalaxycommunity2024galaxyplatformaccessible wolff2020galaxyhicexplorer3 wolff2018galaxyhicexplorerweb">(<a href="#ref-thegalaxycommunity2024galaxyplatformaccessible" role="doc-biblioref">The Galaxy Community et al. 2024</a>; <a href="#ref-wolff2020galaxyhicexplorer3" role="doc-biblioref">Wolff et al. 2020</a>, <a href="#ref-wolff2018galaxyhicexplorerweb" role="doc-biblioref">2018</a>)</span>, a web-based platform that has computationally demanding softwares executable through their servers. Here, one can make a drag-and-drop workflow with HicExplorer’s commands, get integrated visualizations, among other cool stuff. Although it was not suitable for this project, they should be given credit for such an extensive integration. Additionally, following a Galaxy-walkthrough would have given me a better intuition of the pipeline they recommend, and faster as well.</p>
<p>The Open2C Ecosystem consists of three major modules—pairtools, cooler, and cooltools. They all conform to the same standards; they are highly modular, and highly flexible and extensively documented. They have an interactive cooler-demo running on Jupyter notebooks on a server, which is easily adaptable to other data. They include both a command-line interface and a (well-documented) Python API, where a <code>Cooler</code> class efficiently handles the Hi-C matrix. They provide tutorials on all 6 of their intended use cases—Visualization, Contacts vs Distance, Compartments and Saddleplots, Insulation and Boundaries, Dots and Focal Enrichment, and Pileups and Average Features—of which we only use visualization and compartments. The software is fully compatible with Pandas and Matplotlib packages, making the analysis highly flexible.</p>
<p>The drawback with having a very flexible tool is that the conventions are unclear. The latest methodological review is from 2015 <span class="citation" data-cites="lajoie2015hitchhikersguidehic">(<a href="#ref-lajoie2015hitchhikersguidehic" role="doc-biblioref">Lajoie, Dekker, and Kaplan 2015</a>)</span>, and many things have progressed since then—a previously mentioned example being the update to <em>bwa</em> so mapping reads seperately is no longer necessary. Additionally, a Springer Protocol about Hi-C Data analysis <span class="citation" data-cites="bicciato2022hicdataanalysis">(<a href="#ref-bicciato2022hicdataanalysis" role="doc-biblioref">Bicciato and Ferrari 2022</a>)</span> is more recently published, but I find that it more so establishes a range of possibilities than a practical guideline.</p>
</section>
<section id="on-results" class="level2">
<h2 class="anchored" data-anchor-id="on-results">On Results</h2>
<p>The differences in the results the two tools produced were crucial, as I could not reproduce the eigenvector compartments reported in <span class="citation" data-cites="wang2019reprogrammingmeioticchromatin">Wang et al. (<a href="#ref-wang2019reprogrammingmeioticchromatin" role="doc-biblioref">2019</a>)</span> with HiCExplorer, as HiCExplorer has no (apparent) way of restricting the viewframe of the eigendecomposition. It resulted in highly biased PC-tracks that only resembled 2 or 3 compartments (<a href="#fig-explorer-pca" class="quarto-xref">Figure&nbsp;8</a>). While it is a strong limitation to the tool (either documentation-wise, or capability-wise), the Hi-C matrices were constructed in only a few steps and passed their the provided quality control, even though the mapping parameters were only explored superficially. The only notable deviance in quality control was a high fraction of valid contacts, but this may be relieved by increasing the default <span class="math inline">\(mapq\)</span> threshold from <span class="math inline">\(15\)</span> to <span class="math inline">\(30\)</span> as the noted convention in <span class="citation" data-cites="lajoie2015hitchhikersguidehic">Lajoie, Dekker, and Kaplan (<a href="#ref-lajoie2015hitchhikersguidehic" role="doc-biblioref">2015</a>)</span>. The contact matrices generated from HiCExplorer improved when correcting, but still, the visible pattern closely delimited by low-count bins. This undesirable feature may be mitigated by passing a stricter filtering threshold to <code>hicCorrectMatrix</code>.</p>
<p>The eigenvectors resulting from Open2C line up very well with the ones from <span class="citation" data-cites="wang2019reprogrammingmeioticchromatin">Wang et al. (<a href="#ref-wang2019reprogrammingmeioticchromatin" role="doc-biblioref">2019</a>)</span>. Here, some of the maps were lined up and overlayed manually in a sketching software, as no coordinates were available from the authors. In hindsight, it could have been informative to request the analyzed datat from the authors. The predicted compartments from cooler qualitatively aligned very well with the ones predicted by <span class="citation" data-cites="wang2019reprogrammingmeioticchromatin">Wang et al. (<a href="#ref-wang2019reprogrammingmeioticchromatin" role="doc-biblioref">2019</a>)</span> if their track was split in the region of the centromere. I confidently rationalize that the reason I have to split in the centromeric region is the newly, more accurate reference genome. As it <em>rheMac10</em> has closed a high amount of gaps since last reference (most of which are expected around the centromere), we expect more sequences to align in that region, resulting in a longer genome as well. This could readily have been tested by lifting the newly calculated coordinates to <em>rheMac2</em>, but I decided it was not worth the effort for a slightly more accurate (yet still qualitative) validation. Additionally, I wanted to shift the focus from <em>reproducing their results</em> to <em>re-running their analysis on rheMac10</em>. With similar reasoning, only a small effort was put into performing their exact eigendecomposition of the Hi-C matrices before abandoning, as it was deemed unnecessary to force a protocol that was not meant for the tool I was using.</p>
<p>Comparing the two separate runs for parsing parameters revealed only minor differences in parsing statistics, the most notable being the fraction of unmapped pairs, which was expected—a higher amount for <code>mask</code> than for <code>5unique</code>. The lower amount of unmapped reads also seem to trigger a higher amount of duplicates. As a result, the two runs end up with the same distribution of cis pairs. Supporting the claim that the parsing parameters made no significant difference, the compartments called for each run were qualitatively compared (<a href="#fig-rs-fb-100-pe-recpe-intervals" class="quarto-xref">Figure&nbsp;14</a>), revealing only minor differences</p>
<p>The brief exploration of matrix plotting with Open2C shows how different measures that can be taken to improve visibility in some regions, but end up misrepresenting the data. Especially, for large genomic regions of missing values, the interpolation is very unreliable (<a href="#fig-rs-chrx-raw-balanced-cgi-both-1" class="quarto-xref">Figure&nbsp;10 (a)</a>, right), where it might be interpreted that very high-frequency interactions occur at the centromeric region. Additionally, as it does not add to the analysis, but only is for visualization, it looses transparency by hiding empty bins. The problem is not as evident in smaller regions (<a href="#fig-rs-chrx-raw-balanced-cgi-both-2" class="quarto-xref">Figure&nbsp;10 (b)</a>), where mostly single empty bins are interpolated.</p>
<section id="narrowing-the-parameter-space-for-testing" class="level3">
<h3 class="anchored" data-anchor-id="narrowing-the-parameter-space-for-testing">Narrowing the parameter space for testing</h3>
<p>To mitigate the effects of multiple testing and reduce my own biases, I try to reason about the different parameters used to pick a subset of combinations to test for significant overlap (Jaccard test) or proximity. Comparing the corrected matrices with the calculated E1 values is initally a subjective task. I find that the predicted compartments are highly similar between arms and 10Mb views at both 100kb and 500kb resolutions, but with the full view of the chromosome, fewer compartments are determined. Bear in mind that none of the compartments can be said to be <em>wrong</em>, they all just capture the variance on different scales. Remembering that the size of the regions under selection were spanning 6 megabases at most informs the decision to discard the full genomic view, which is also supported by the litterature <span class="citation" data-cites="lajoie2015hitchhikersguidehic">(<a href="#ref-lajoie2015hitchhikersguidehic" role="doc-biblioref">Lajoie, Dekker, and Kaplan 2015</a>)</span>, recommending partitioning the chromosome into its arms <em>if the first eigenvector only captures the arms</em>.</p>
<p>Similarly, the choice of resolution should be made cautiously. Here, the resolution of the matrix is the same resolution as the resolution of the eigenvector. Therefore, a resolution can be chosen to reflect the genomic scale that we want to investigate, and the minimum resolution required to test if the compartments overlap with regions under selection. Although some of these regions are exceptionally large, the shortest segment is just below 100 kb. Thus, the 100 kb resolution is deemed sufficient to capture compartments that are within the range of the regions under selection.</p>
<p>However, it is important to acknowledge the trade-offs involved in selecting a resolution. Higher resolutions (e.g., 50 kb or 10 kb) can provide finer detail and may help identify sub-compartmental structures or smaller features of chromatin organization. As <span class="citation" data-cites="wang2019reprogrammingmeioticchromatin">Wang et al. (<a href="#ref-wang2019reprogrammingmeioticchromatin" role="doc-biblioref">2019</a>)</span> used 40 kb matrices for TAD calling, resolution should not be an issue for 50 kb matrices in this context. Although this study did not experiment with higher resolutions than 100 kb, it is plausible that 50 kb resolution could better capture the structural variance. The sequencing depth and coverage of the library appear sufficient to support such analyses, further justifying the exploration of higher resolutions. On the other hand, lower resolutions (e.g., 500 kb or 1 Mb) are more robust for sparse datasets but may oversimplify chromatin structure and fail to resolve smaller regions under selection. The choice of 100 kb strikes a balance between these extremes, ensuring that biologically relevant features of both the compartments and the regions under selection can be detected while minimizing the risk of false positives due to noise.</p>
<p>It is thus apparent that further analysis at higher resolutions, such as 50 kb, could provide additional insights into the structural variance of the regions under selection. This study largely avoids performing in-depth quality control of the Hi-C library and the resulting interaction matrices, instead relying on qualitative evaluation. However, this approach is justified as I use the same resolution as <span class="citation" data-cites="wang2019reprogrammingmeioticchromatin">Wang et al. (<a href="#ref-wang2019reprogrammingmeioticchromatin" role="doc-biblioref">2019</a>)</span>, whose work includes comprehensive quality control measures such as coverage profiles along the chromosomes, <span class="math inline">\(P(s)\)</span> curves, and compartment segregation strength.</p>
</section>
<section id="correlating-genomic-regions" class="level3">
<h3 class="anchored" data-anchor-id="correlating-genomic-regions">Correlating genomic regions</h3>
<p>The extracted compartments and their 200 kb edge regions were qualitatively compared for overlap with the ECH region (<a href="#fig-comps-edges-ech" class="quarto-xref">Figure&nbsp;15</a>). For fibroblast (Fb) and round spermatid (RS), I observe promising overlap with ECH90. Additionally, visualization of the difference in overlaps (“DiffInt”) between (ECH, full compartments) and (ECH, edges) confirmed that the ‘outside’ edge captured additional overlap (see <a href="#fig-edge-enrichment" class="quarto-xref">Figure&nbsp;16</a>). It would have been informative to compare the intersection of the same overlaps (“IntInt”) to determine whether the overlap occurs more frequently on the inside edge than the outside edge.</p>
<p>The Jaccard test confirmed significant ECH overlap with Fb and RS, with <span class="math inline">\(p=0.012\)</span> and <span class="math inline">\(p=0.010\)</span>, respectively. In contrast, the proximity test showed no significant results, which is unsurprising given the design of the test. Specifically, when calculating the proximity index between a query and an annotation (<code>query</code> and <code>annot</code>, respectively), overlapping segments are masked from <code>query</code> before calculating the mean distance to the nearest non-overlapping segments. This means that substantial intersection, as confirmed by the Jaccard test, leads to many segments being removed in the proximity test. This design choice—–removing entire overlapping segments rather than just the overlapping bases–—creates a clearer separation between the two test statistics. Without this distinction, the proximity test could capture some of the variance inherent to the Jaccard statistic, making it harder to isolate proximal, non-overlapping segments.</p>
<p>The significant overlap between ECH regions, Fb, and RS at arms view is reproduced when testing at the compartment 1bp limits instead of the 200kb edges, although the significance is weakened a bit. The incresed p-value indicates that the full regions are important for for relationship. Surprisingly, the limits of both spermatogonia and pachytene spermatocyte are significantly overlapping the ECH regions, more so than both fibroblast and round spermatid, supporting the findings from <span class="citation" data-cites="wang2019reprogrammingmeioticchromatin">Wang et al. (<a href="#ref-wang2019reprogrammingmeioticchromatin" role="doc-biblioref">2019</a>)</span> that chromatin undergoes remodelling through spermatogenesis.</p>
<p>The visualization of the overlap between baboon regions and round spermatid showed notable amounts of intersection (<a href="#fig-baboon-rs100-intersect" class="quarto-xref">Figure&nbsp;18</a>), but both significant overlap <em>and</em> proximity was rejected by the tests. Conversely, the tests performed on the 1bp limits of baboon regions against the 1bp limits of A/B compartments revealed a significant proximity between olive and spermatozoa. Additionally, significant proximity was observed between spermatogonia, pachytene spermatocyte by defining baboon regions under selection as a single set, <code>olivehama</code>, and here the significance was strengthened an order of magnitude for spermatozoa. This indicate that the effect size of the hama regions is too small for statistical inference, but also that the two sets of genomic intervals can be viewed as a single set, possibly indicating that similar forces have shaped the genomic landscape.</p>
<p>Notably, three out of five cell types both overlap the ECH regions <em>and</em> are proximal to the baboon limits—spermatogonia and spermatozoa in the 10Mb viewframe, and pachytene spermatocyte in the arms viewframe, indicating an indirect relationship through chromatin architecture. As neither proximity nor overlap is observed between the baboon regions and ECH, this observation could indicate that some structural features of the chromosome are aiding in reducing diversity. These findings propose an intricate relationship between chromatin architecture and evolutionary pressures. The intersection and proximity between regions under strong selection and chromatin compartments on the X chromosome of primates, along with their location on the X chromosome, suggest that structural genomic features influence patterns of diversity and selection. While these results are promising, further testing of the compartments is needed—both at higher resolutions and in other primates—to refine our understanding of these dynamics.</p>
<p>Notably, <span class="citation" data-cites="skov2023extraordinaryselectionhuman">Skov et al. (<a href="#ref-skov2023extraordinaryselectionhuman" role="doc-biblioref">2023</a>)</span> describe a compelling scenario where meiotic drive could explain the reduced diversity on ECH regions, and these structural features might also play a role in facilitating or constraining mechanisms like gene drive, which can significantly alter allele frequencies across generations by bypassing classical Mendelian inheritance. However, caution is warranted when inferring the role of gene drive, as it remains a contentious concept among biologists.</p>
</section>
<section id="gene-drive-selfishness-and-its-effect-on-selection" class="level3">
<h3 class="anchored" data-anchor-id="gene-drive-selfishness-and-its-effect-on-selection">Gene Drive, Selfishness and its Effect on Selection</h3>
<p>Gene drive occur when a particular collection of genes is propagated through a population by increasing the probability of transmitting genes to the offspring from random (Mendelian) inheritance, resulting in a biased gene transmission against its alternative (referred to as <em>transmission advantage</em>). Two categories of gene drivers exist <span class="citation" data-cites="bravonunez2018geneticvillainskiller">(<a href="#ref-bravonunez2018geneticvillainskiller" role="doc-biblioref">Bravo Núñez, Nuckolls, and Zanders 2018</a>)</span>; <em>class one drivers</em> affect chromosome segregation in meiosis, and <em>killer meiotic drivers</em> will sabotage meiotic product that have not inherited the driving allele <span class="citation" data-cites="bravonunez2018geneticvillainskiller">(<a href="#ref-bravonunez2018geneticvillainskiller" role="doc-biblioref">Bravo Núñez, Nuckolls, and Zanders 2018</a>)</span>. This happens regardless of the fitness effects of the developed organism, and is there often a factor offsetting classical selection. Even though the implications of such systems are potentially detrimental, they are notoriously difficult to detect. A circumstance contributing to the difficulty is that most genetic experiments are done in homozygotes. <span class="citation" data-cites="bravonunez2018geneticvillainskiller">Bravo Núñez, Nuckolls, and Zanders (<a href="#ref-bravonunez2018geneticvillainskiller" role="doc-biblioref">2018</a>)</span> state that the general choice of experimental system may have biased our understanding of sexual reproduction. A key point is; meiotic drivers can only be observed in heterozygotes, where a genetic driver has a competitor.</p>
<p>If an autosomal driver reaches fixation, it no longer has a target to act against, and consequently the driving phenotype will not be observed <span class="citation" data-cites="bravonunez2018geneticvillainskiller">(<a href="#ref-bravonunez2018geneticvillainskiller" role="doc-biblioref">Bravo Núñez, Nuckolls, and Zanders 2018</a>)</span>. Over time, as there is neither selfish nor selective pressure to maintain the driving mechanism of a fixed driver, the mechanism will decay as it accumulates inactivating mutations. As a result, genetic drivers are said to be transient <span class="citation" data-cites="bravonunez2018geneticvillainskiller">(<a href="#ref-bravonunez2018geneticvillainskiller" role="doc-biblioref">Bravo Núñez, Nuckolls, and Zanders 2018</a>)</span> on an evolutionary timescale, unless it is linked to another positively selected allele <span class="citation" data-cites="jaenike2001sexchromosomemeiotic">(<a href="#ref-jaenike2001sexchromosomemeiotic" role="doc-biblioref">Jaenike 2001</a>)</span>. Interestingly, gene drive is much more well-documented on sex chromosomes than for autosomes. Possibly because the sex chromosome meiotic drive inherently causes a skewed sex ratio, more notably raising a flag for further analysis. Or, because X and Y do not recombine, allowing for killer-supressor gene pairs to stay on separate chromosomes. The consequences of sex chromosome drive and skewed sex ratio can be widespread. A fully driving gene on X can to lead to the extinction of the population that fixes the allele <span class="citation" data-cites="jaenike2001sexchromosomemeiotic">(<a href="#ref-jaenike2001sexchromosomemeiotic" role="doc-biblioref">Jaenike 2001</a>)</span>, as only one (fertile) offspring sex will be produced. Therefore, sex chromosome drivers usually exist in equilibrium with a supressor. When a population becomes female-biased, autosomal suppressors will be favored with a process termed <em>Fisher’s sex ratio selection</em> <span class="citation" data-cites="lindholm2016ecologyevolutionarydynamics">(<a href="#ref-lindholm2016ecologyevolutionarydynamics" role="doc-biblioref">Lindholm et al. 2016</a>.)</span>, as the individuals that produce offspring of the minor-sex have higher fitness.</p>
<p>A driving element on X may target a region on Y, mitigating the risk of self-destruction and increasing its frequency at the same time. However, as it induces a skewed sex-ratio by limiting the number of functioning Y-bearing spermatozoa, the selection on Y to counter this effect is strong, engaging an evolutionary arms race between driving allele and its supressor, potentially explaining stable Y chromosome polymorphisms <span class="citation" data-cites="jaenike2001sexchromosomemeiotic">(<a href="#ref-jaenike2001sexchromosomemeiotic" role="doc-biblioref">Jaenike 2001</a>)</span>. However, the Y chromosome only exist in the male population and a supressor on Y has only limited effect on the X chromosome on a population-scale compared to its autosomal cousins, which consequently often harbour supressors of X-drive, as they are passed down to more offspring than the Y chromosome. The fight between a driver and a supressor in their coevolution is linked to the chromosomal architecture, as they have been shown to be protected by low-recombining regions or chromosomal inversions, and to be mediating the evolution of karyotype <span class="citation" data-cites="lindholm2016ecologyevolutionarydynamics">(<a href="#ref-lindholm2016ecologyevolutionarydynamics" role="doc-biblioref">Lindholm et al. 2016</a>)</span>. Thus, the structural features of a chromosome not only shield genetic drivers but influence the patterns of synteny, which, in turn, are shaped by both selection and genomic constraints.</p>
<p>The interplay between selfish genetic elements, such as gene drivers, and their suppression through coevolutionary processes emphasizes the importance of chromosomal architecture in shaping evolutionary dynamics. This evolutionary “arms race” between drivers and suppressors is not only driven by selection but also by the structural constraints of the chromosomes themselves. In particular, features such as low-recombining regions and chromosomal inversions offer a protective shield for these genetic elements, influencing both their persistence and their evolutionary trajectories. Given this, understanding the role of chromosomal organization in these dynamics can provide deeper insights into the mechanisms that govern selection on sex chromosomes and their broader genomic implications.</p>
</section>
</section>
</section>
<section id="conclusion-and-future-work" class="level1">
<h1>Conclusion and future work</h1>
<p>In summary, I find that previously identified regions on X under strong selection in humans (referred to as ECHs) and in baboons (disproportionate <em>P.hamadryas</em> or <em>P.anubis</em> ancestry) strongly correlate with chromatin A/B-compartments in some stages of spermatogenesis in rhesus macaque (<em>Macaca mulata</em>) inferred by PCA from either a chromosome-arm (conventional) viewframe or a 10 Mb viewframe (local). The regions under selection are associated with regions of low diversity and low ILS on the X chromosome across all the great apes.</p>
<p>Briefly, an array of significant intersection and proximity is observed when comparing ECH regions baboon regions with the 1bp limit of A/B compartments. Most notably, spermatogonia, spermatozoa, and pachytene spermatocyte was both significantly proximal to baboon regions <em>and</em> intersecting ECH regions. This suggests relationship between strong selection and chromosomal architecture, and as the selection is specific on X, it could indicate sex-chromosome related mechanisms. A point against a meiosis-specific mechanisms is that fibroblast intersects the ECH regions, where fibroblast was used as a control-group for spermatogenesis in <span class="citation" data-cites="wang2019reprogrammingmeioticchromatin">Wang et al. (<a href="#ref-wang2019reprogrammingmeioticchromatin" role="doc-biblioref">2019</a>)</span>.</p>
<p>I duly note that this is purely an explorative and correlational study, requiring further testing to conclude a relationship between chromosomal organization and selection, even if my analysis yields statistically significant results. To provide a more solid foundation for this inference, the next steps could include:</p>
<ul>
<li>Include compartment metadata in the correlation, such as weighing by insulation score, or filter out edges that have adjecent missing values</li>
<li>Recall compartments using higher-resolution matrices, or call TADs</li>
<li>Analyze Hi-C spermatogenesis data in other great ape species (e.g.&nbsp;baboons)</li>
<li>Use a more distant relative as a control group (e.g.&nbsp;the mouse data from <span class="citation" data-cites="wang2019reprogrammingmeioticchromatin">Wang et al. (<a href="#ref-wang2019reprogrammingmeioticchromatin" role="doc-biblioref">2019</a>)</span>)</li>
</ul>
<p>Until then, let us remain cautious in our interpretations, recognizing the complexity of the systems at play and the need for further investigation to validate the potential links between chromosomal architecture and selective forces. In the meantime, let us appreciate the complexity of a multi-species correlation between selection and the 3D architecture of our chromosomes, a region of considerable interest with many aspects still to be understood and explored.</p>
</section>
<section id="bibliography" class="level1 unnumbered">
<h1 class="unnumbered">Bibliography</h1>
<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-abdennur2020coolerscalablestorage" class="csl-entry" role="listitem">
Abdennur, Nezar, and Leonid A Mirny. 2020. <span>“Cooler: Scalable Storage for <span>Hi-C</span> Data and Other Genomically Labeled Arrays.”</span> Edited by Jonathan Wren. <em>Bioinformatics</em> 36 (1): 311–16. <a href="https://doi.org/10.1093/bioinformatics/btz540">https://doi.org/10.1093/bioinformatics/btz540</a>.
</div>
<div id="ref-Allaire_Quarto_2024" class="csl-entry" role="listitem">
Allaire, J. J., Charles Teague, Carlos Scheidegger, Yihui Xie, and Christophe Dervieux. 2024. <span>“<span>Quarto</span>.”</span> <a href="https://doi.org/10.5281/zenodo.5960048">https://doi.org/10.5281/zenodo.5960048</a>.
</div>
<div id="ref-anaconda" class="csl-entry" role="listitem">
Anaconda Software Distribution. 2016. <span>“Anaconda Software Distribution.”</span> Computer software. <a href="https://anaconda.com">https://anaconda.com</a>.
</div>
<div id="ref-baker20161500scientistslift" class="csl-entry" role="listitem">
Baker, Monya. 2016. <span>“1,500 Scientists Lift the Lid on Reproducibility.”</span> <em>Nature</em> 533 (7604): 452–54. <a href="https://doi.org/10.1038/533452a">https://doi.org/10.1038/533452a</a>.
</div>
<div id="ref-bersaglieri2004geneticsignaturesstrong" class="csl-entry" role="listitem">
Bersaglieri, Todd, Pardis C. Sabeti, Nick Patterson, Trisha Vanderploeg, Steve F. Schaffner, Jared A. Drake, Matthew Rhodes, David E. Reich, and Joel N. Hirschhorn. 2004. <span>“Genetic <span>Signatures</span> of <span>Strong Recent Positive Selection</span> at the <span>Lactase Gene</span>.”</span> <em>The American Journal of Human Genetics</em> 74 (6): 1111–20. <a href="https://doi.org/10.1086/421051">https://doi.org/10.1086/421051</a>.
</div>
<div id="ref-bicciato2022hicdataanalysis" class="csl-entry" role="listitem">
Bicciato, Silvio, and Francesco Ferrari, eds. 2022. <em>Hi-<span>C Data Analysis</span>: <span>Methods</span> and <span>Protocols</span></em>. Vol. 2301. Methods in <span>Molecular Biology</span>. New York, NY: Springer US. <a href="https://doi.org/10.1007/978-1-0716-1390-0">https://doi.org/10.1007/978-1-0716-1390-0</a>.
</div>
<div id="ref-bravonunez2018geneticvillainskiller" class="csl-entry" role="listitem">
Bravo Núñez, María Angélica, Nicole L. Nuckolls, and Sarah E. Zanders. 2018. <span>“Genetic <span>Villains</span>: <span>Killer Meiotic Drivers</span>.”</span> <em>Trends in Genetics</em> 34 (6): 424–33. <a href="https://doi.org/10.1016/j.tig.2018.02.003">https://doi.org/10.1016/j.tig.2018.02.003</a>.
</div>
<div id="ref-cowell2023100yearshaldanes" class="csl-entry" role="listitem">
Cowell, Finn. 2023. <span>“100 Years of <span>Haldane</span>’s Rule.”</span> <em>Journal of Evolutionary Biology</em> 36 (2): 337–46. <a href="https://doi.org/10.1111/jeb.14112">https://doi.org/10.1111/jeb.14112</a>.
</div>
<div id="ref-devteam2024sratoolkit" class="csl-entry" role="listitem">
DevTeam, SRA Toolkit. 2024. <span>“<span>SRA Toolkit</span>.”</span> Wiki. <em>GitHub</em>. https://github.com/ncbi/sra-tools/wiki/01.-Downloading-SRA-Toolkit. <a href="https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=software">https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=software</a>.
</div>
<div id="ref-dixon2015chromatinarchitecturereorganization" class="csl-entry" role="listitem">
Dixon, Jesse R., Inkyung Jung, Siddarth Selvaraj, Yin Shen, Jessica E. Antosiewicz-Bourget, Ah Young Lee, Zhen Ye, et al. 2015. <span>“Chromatin Architecture Reorganization During Stem Cell Differentiation.”</span> <em>Nature</em> 518 (7539): 331–36. <a href="https://doi.org/10.1038/nature14222">https://doi.org/10.1038/nature14222</a>.
</div>
<div id="ref-dutheil2015strongselectivesweeps" class="csl-entry" role="listitem">
Dutheil, Julien Y., Kasper Munch, Kiwoong Nam, Thomas Mailund, and Mikkel H. Schierup. 2015. <span>“Strong <span>Selective Sweeps</span> on the <span>X Chromosome</span> in the <span>Human-Chimpanzee Ancestor Explain Its Low Divergence</span>.”</span> Edited by Nick H. Barton. <em>PLOS Genetics</em> 11 (8): e1005451. <a href="https://doi.org/10.1371/journal.pgen.1005451">https://doi.org/10.1371/journal.pgen.1005451</a>.
</div>
<div id="ref-ewels2016multiqcsummarizeanalysis" class="csl-entry" role="listitem">
Ewels, Philip, Måns Magnusson, Sverker Lundin, and Max Käller. 2016. <span>“<span>MultiQC</span>: Summarize Analysis Results for Multiple Tools and Samples in a Single Report.”</span> <em>Bioinformatics</em> 32 (19): 3047–48. <a href="https://doi.org/10.1093/bioinformatics/btw354">https://doi.org/10.1093/bioinformatics/btw354</a>.
</div>
<div id="ref-gao2018segment_liftoverpythontool" class="csl-entry" role="listitem">
Gao, Bo, Qingyao Huang, and Michael Baudis. 2018. <span>“Segment_liftover : A <span>Python</span> Tool to Convert Segments Between Genome Assemblies.”</span> <em>F1000Research</em> 7 (June): 319. <a href="https://doi.org/10.12688/f1000research.14148.2">https://doi.org/10.12688/f1000research.14148.2</a>.
</div>
<div id="ref-gwf" class="csl-entry" role="listitem">
GenomeDK. 2023. <span>“Gwf Workflow Manager.”</span> <a href="https://github.com/gwforg/gwf.git" class="uri">https://github.com/gwforg/gwf.git</a>.
</div>
<div id="ref-github" class="csl-entry" role="listitem">
GitHub, Inc. n.d. <span>“GitHub.”</span> <a href="https://github.com" class="uri">https://github.com</a>.
</div>
<div id="ref-haldane1922sexratiounisexual" class="csl-entry" role="listitem">
Haldane, J. B. S. 1922. <span>“Sex Ratio and Unisexual Sterility in Hybrid Animals.”</span> <em>Journal of Genetics</em> 12 (2): 101–9. <a href="https://doi.org/10.1007/BF02983075">https://doi.org/10.1007/BF02983075</a>.
</div>
<div id="ref-hinrichs2006ucscgenomebrowser" class="csl-entry" role="listitem">
Hinrichs, A. S. 2006. <span>“The <span>UCSC Genome Browser Database</span>: Update 2006.”</span> <em>Nucleic Acids Research</em> 34 (90001): D590–98. <a href="https://doi.org/10.1093/nar/gkj144">https://doi.org/10.1093/nar/gkj144</a>.
</div>
<div id="ref-imakaev2012iterativecorrectionhic" class="csl-entry" role="listitem">
Imakaev, Maxim, Geoffrey Fudenberg, Rachel Patton McCord, Natalia Naumova, Anton Goloborodko, Bryan R Lajoie, Job Dekker, and Leonid A Mirny. 2012. <span>“Iterative Correction of <span>Hi-C</span> Data Reveals Hallmarks of Chromosome Organization.”</span> <em>Nature Methods</em> 9 (10): 999–1003. <a href="https://doi.org/10.1038/nmeth.2148">https://doi.org/10.1038/nmeth.2148</a>.
</div>
<div id="ref-auityourguidegai" class="csl-entry" role="listitem">
IT, AU. n.d. <span>“Your Guide to <span>GAI</span> Responsibly.”</span> https://medarbejdere.au.dk/en/gai. Accessed January 9, 2025.
</div>
<div id="ref-jaenike2001sexchromosomemeiotic" class="csl-entry" role="listitem">
Jaenike, John. 2001. <span>“Sex <span>Chromosome Meiotic Drive</span>.”</span> <em>Annual Review of Ecology and Systematics</em> 32 (1): 25–49. <a href="https://doi.org/10.1146/annurev.ecolsys.32.081501.113958">https://doi.org/10.1146/annurev.ecolsys.32.081501.113958</a>.
</div>
<div id="ref-soton403913jupyter" class="csl-entry" role="listitem">
Kluyver, Thomas, Benjamin Ragan-Kelley, Fernando Pérez, Brian Granger, Matthias Bussonnier, Jonathan Frederic, Kyle Kelley, et al. 2016. <span>“Jupyter Notebooks ? A Publishing Format for Reproducible Computational Workflows.”</span> In <em>Positioning and Power in Academic Publishing: Players, Agents and Agendas</em>, edited by Fernando Loizides and Birgit Scmidt, 87–90. IOS Press. <a href="https://eprints.soton.ac.uk/403913/">https://eprints.soton.ac.uk/403913/</a>.
</div>
<div id="ref-lajoie2015hitchhikersguidehic" class="csl-entry" role="listitem">
Lajoie, Bryan R., Job Dekker, and Noam Kaplan. 2015. <span>“The <span>Hitchhiker</span>’s <span>Guide</span> to <span>Hi-C Analysis</span>: <span>Practical</span> Guidelines.”</span> <em>Methods (San Diego, Calif.)</em> 72 (January): 65–75. <a href="https://doi.org/10.1016/j.ymeth.2014.10.031">https://doi.org/10.1016/j.ymeth.2014.10.031</a>.
</div>
<div id="ref-li2013aligningsequencereadsclone" class="csl-entry" role="listitem">
Li, Heng. 2013. <span>“Aligning Sequence Reads, Clone Sequences and Assembly Contigs with BWA-MEM.”</span> <a href="https://arxiv.org/abs/1303.3997">https://arxiv.org/abs/1303.3997</a>.
</div>
<div id="ref-lieberman-aiden2009comprehensivemappinglongrange" class="csl-entry" role="listitem">
Lieberman-Aiden, Erez, Nynke L. Van Berkum, Louise Williams, Maxim Imakaev, Tobias Ragoczy, Agnes Telling, Ido Amit, et al. 2009. <span>“Comprehensive <span>Mapping</span> of <span>Long-Range Interactions Reveals Folding Principles</span> of the <span>Human Genome</span>.”</span> <em>Science</em> 326 (5950): 289–93. <a href="https://doi.org/10.1126/science.1181369">https://doi.org/10.1126/science.1181369</a>.
</div>
<div id="ref-lindholm2016ecologyevolutionarydynamics" class="csl-entry" role="listitem">
Lindholm, Anna K., Kelly A. Dyer, Renée C. Firman, Lila Fishman, Wolfgang Forstmeier, Luke Holman, Hanna Johannesson, et al. 2016. <span>“The <span>Ecology</span> and <span>Evolutionary Dynamics</span> of <span>Meiotic Drive</span>.”</span> <em>Trends in Ecology &amp; Evolution</em> 31 (4): 315–26. <a href="https://doi.org/10.1016/j.tree.2016.02.001">https://doi.org/10.1016/j.tree.2016.02.001</a>.
</div>
<div id="ref-mailund2014lineagesortingapes" class="csl-entry" role="listitem">
Mailund, Thomas, Kasper Munch, and Mikkel Heide Schierup. 2014. <span>“Lineage <span>Sorting</span> in <span>Apes</span>.”</span> <em>Annual Review of Genetics</em> 48 (1): 519–35. <a href="https://doi.org/10.1146/annurev-genet-120213-092532">https://doi.org/10.1146/annurev-genet-120213-092532</a>.
</div>
<div id="ref-misteli2020selforganizinggenomeprinciples" class="csl-entry" role="listitem">
Misteli, Tom. 2020. <span>“The <span>Self-Organizing Genome</span>: <span>Principles</span> of <span>Genome Architecture</span> and <span>Function</span>.”</span> <em>Cell</em> 183 (1): 28–45. <a href="https://doi.org/10.1016/j.cell.2020.09.014">https://doi.org/10.1016/j.cell.2020.09.014</a>.
</div>
<div id="ref-munch2024munchgroup" class="csl-entry" role="listitem">
Munch, Kasper. 2024. <span>“Munch-Group.”</span> https://munch-group.org/research.html.
</div>
<div id="ref-openchromosomecollective" class="csl-entry" role="listitem">
<span>“<span>Open2C</span>.”</span> n.d. Resource. <em>Open <span>Chromosome Collective</span></em>. https://open2c.github.io/. Accessed November 13, 2024.
</div>
<div id="ref-open2c2022cooltoolsenablinghighresolution" class="csl-entry" role="listitem">
Open2C, Nezar Abdennur, Sameer Abraham, Geoffrey Fudenberg, Ilya M. Flyamer, Aleksandra A. Galitsyna, Anton Goloborodko, Maxim Imakaev, Betul A. Oksuz, and Sergey V. Venev. 2022. <span>“Cooltools: Enabling High-Resolution <span>Hi-C</span> Analysis in <span>Python</span>.”</span> Bioinformatics. <a href="https://doi.org/10.1101/2022.10.31.514564">https://doi.org/10.1101/2022.10.31.514564</a>.
</div>
<div id="ref-open2c2024pairtoolssequencingdata" class="csl-entry" role="listitem">
Open2C, Nezar Abdennur, Geoffrey Fudenberg, Ilya M. Flyamer, Aleksandra A. Galitsyna, Anton Goloborodko, Maxim Imakaev, and Sergey V. Venev. 2024. <span>“Pairtools: <span>From</span> Sequencing Data to Chromosome Contacts.”</span> Edited by Ferhat Ay. <em>PLOS Computational Biology</em> 20 (5): e1012164. <a href="https://doi.org/10.1371/journal.pcbi.1012164">https://doi.org/10.1371/journal.pcbi.1012164</a>.
</div>
<div id="ref-quartoteam2022announcingquartonew" class="csl-entry" role="listitem">
Quarto Team, Posit. 2022. <span>“Announcing <span>Quarto</span>, a New Scientific and Technical Publishing System.”</span> <em>Posit</em>.
</div>
<div id="ref-ramirez2018highresolutiontadsreveal" class="csl-entry" role="listitem">
Ramírez, Fidel, Vivek Bhardwaj, Laura Arrigoni, Kin Chung Lam, Björn A. Grüning, José Villaveces, Bianca Habermann, Asifa Akhtar, and Thomas Manke. 2018. <span>“High-Resolution <span>TADs</span> Reveal <span>DNA</span> Sequences Underlying Genome Organization in Flies.”</span> <em>Nature Communications</em> 9 (1): 189. <a href="https://doi.org/10.1038/s41467-017-02525-w">https://doi.org/10.1038/s41467-017-02525-w</a>.
</div>
<div id="ref-sayers2022databaseresourcesnational" class="csl-entry" role="listitem">
Sayers, Eric W, Evan E Bolton, J Rodney Brister, Kathi Canese, Jessica Chan, Donald C Comeau, Ryan Connor, et al. 2022. <span>“Database Resources of the National Center for Biotechnology Information.”</span> <em>Nucleic Acids Research</em> 50 (D1): D20–26. <a href="https://doi.org/10.1093/nar/gkab1112">https://doi.org/10.1093/nar/gkab1112</a>.
</div>
<div id="ref-skov2023extraordinaryselectionhuman" class="csl-entry" role="listitem">
Skov, Laurits, Moisès Coll Macià, Elise Anne Lucotte, Maria Izabel Alves Cavassim, David Castellano, Mikkel Heide Schierup, and Kasper Munch. 2023. <span>“Extraordinary Selection on the Human <span>X</span> Chromosome Associated with Archaic Admixture.”</span> <em>Cell Genomics</em> 3 (3): 100274. <a href="https://doi.org/10.1016/j.xgen.2023.100274">https://doi.org/10.1016/j.xgen.2023.100274</a>.
</div>
<div id="ref-sorensen2023genomewidecoancestryreveals" class="csl-entry" role="listitem">
Sørensen, Erik F., R. Alan Harris, Liye Zhang, Muthuswamy Raveendran, Lukas F. K. Kuderna, Jerilyn A. Walker, Jessica M. Storer, et al. 2023. <span>“Genome-Wide Coancestry Reveals Details of Ancient and Recent Male-Driven Reticulation in Baboons.”</span> <em>Science</em> 380 (6648): eabn8153. <a href="https://doi.org/10.1126/science.abn8153">https://doi.org/10.1126/science.abn8153</a>.
</div>
<div id="ref-thematplotlibdevelopmentteam2024matplotlibvisualizationpython" class="csl-entry" role="listitem">
Team, The Matplotlib Development. 2024. <span>“Matplotlib: <span>Visualization</span> with <span>Python</span>.”</span> Zenodo. <a href="https://doi.org/10.5281/ZENODO.10916799">https://doi.org/10.5281/ZENODO.10916799</a>.
</div>
<div id="ref-thegalaxycommunity2024galaxyplatformaccessible" class="csl-entry" role="listitem">
The Galaxy Community, Linelle Ann L Abueg, Enis Afgan, Olivier Allart, Ahmed H Awan, Wendi A Bacon, Dannon Baker, et al. 2024. <span>“The <span>Galaxy</span> Platform for Accessible, Reproducible, and Collaborative Data Analyses: 2024 Update.”</span> <em>Nucleic Acids Research</em> 52 (W1): W83–94. <a href="https://doi.org/10.1093/nar/gkae410">https://doi.org/10.1093/nar/gkae410</a>.
</div>
<div id="ref-git" class="csl-entry" role="listitem">
Torvalds, Linus, and Junio C Hamano. 2005. <em>Git: A Distributed Version Control System</em>. <a href="https://git-scm.com">https://git-scm.com</a>.
</div>
<div id="ref-wang2019reprogrammingmeioticchromatin" class="csl-entry" role="listitem">
Wang, Yao, Hanben Wang, Yu Zhang, Zhenhai Du, Wei Si, Suixing Fan, Dongdong Qin, et al. 2019. <span>“Reprogramming of <span>Meiotic Chromatin Architecture</span> During <span>Spermatogenesis</span>.”</span> <em>Molecular Cell</em> 73 (3): 547–561.e6. <a href="https://doi.org/10.1016/j.molcel.2018.11.019">https://doi.org/10.1016/j.molcel.2018.11.019</a>.
</div>
<div id="ref-warren2020sequencediversityanalyses" class="csl-entry" role="listitem">
Warren, Wesley C., R. Alan Harris, Marina Haukness, Ian T. Fiddes, Shwetha C. Murali, Jason Fernandes, Philip C. Dishuck, et al. 2020. <span>“Sequence Diversity Analyses of an Improved Rhesus Macaque Genome Enhance Its Biomedical Utility.”</span> <em>Science</em> 370 (6523): eabc6617. <a href="https://doi.org/10.1126/science.abc6617">https://doi.org/10.1126/science.abc6617</a>.
</div>
<div id="ref-waskom2021seabornstatisticaldata" class="csl-entry" role="listitem">
Waskom, Michael. 2021. <span>“Seaborn: Statistical Data Visualization.”</span> <em>Journal of Open Source Software</em> 6 (60): 3021. <a href="https://doi.org/10.21105/joss.03021">https://doi.org/10.21105/joss.03021</a>.
</div>
<div id="ref-wolff2018galaxyhicexplorerweb" class="csl-entry" role="listitem">
Wolff, Joachim, Vivek Bhardwaj, Stephan Nothjunge, Gautier Richard, Gina Renschler, Ralf Gilsbach, Thomas Manke, Rolf Backofen, Fidel Ramírez, and Björn A Grüning. 2018. <span>“Galaxy <span>HiCExplorer</span>: A Web Server for Reproducible <span>Hi-C</span> Data Analysis, Quality Control and Visualization.”</span> <em>Nucleic Acids Research</em> 46 (W1): W11–16. <a href="https://doi.org/10.1093/nar/gky504">https://doi.org/10.1093/nar/gky504</a>.
</div>
<div id="ref-wolff2020galaxyhicexplorer3" class="csl-entry" role="listitem">
Wolff, Joachim, Leily Rabbani, Ralf Gilsbach, Gautier Richard, Thomas Manke, Rolf Backofen, and Björn A Grüning. 2020. <span>“Galaxy <span>HiCExplorer</span> 3: A Web Server for Reproducible <span>Hi-C</span>, Capture <span>Hi-C</span> and Single-Cell <span>Hi-C</span> Data Analysis, Quality Control and Visualization.”</span> <em>Nucleic Acids Research</em> 48 (W1): W177–84. <a href="https://doi.org/10.1093/nar/gkaa220">https://doi.org/10.1093/nar/gkaa220</a>.
</div>
<div id="ref-zuo2021stageresolvedhicanalyses" class="csl-entry" role="listitem">
Zuo, Wu, Guangming Chen, Zhimei Gao, Shuai Li, Yanyan Chen, Chenhui Huang, Juan Chen, Zhengjun Chen, Ming Lei, and Qian Bian. 2021. <span>“Stage-Resolved <span>Hi-C</span> Analyses Reveal Meiotic Chromosome Organizational Features Influencing Homolog Alignment.”</span> <em>Nature Communications</em> 12 (1): 5827. <a href="https://doi.org/10.1038/s41467-021-26033-0">https://doi.org/10.1038/s41467-021-26033-0</a>.
</div>
</div>
</section>
<section id="a-appendix" class="level1 unnumbered">
<h1 class="unnumbered">A Appendix</h1>
<section id="data-availability" class="level2 unnumbered">
<h2 class="unnumbered anchored" data-anchor-id="data-availability">Data Availability</h2>
<p>All code, text, and files used for this project are available on <a href="https://github.com/munch-group/hic-spermatogenesis">https://github.com/munch-group/hic-spermatogenesis</a> with instructions on how to reproduce.</p>
</section>
<section id="aknowledgements" class="level2 unnumbered">
<h2 class="unnumbered anchored" data-anchor-id="aknowledgements">Aknowledgements</h2>
<section id="people" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="people">People</h3>
<p>I would like to extend my sincere gratitude to my supervisor, Kasper Munch, for his guidance and support throughout this project. His sparring and scientific literacy have been invaluable in refining my ideas and shaping the direction of this work. I am especially thankful for Kasper’s enthusiasm, which has been a source of motivation during this process. His expertise and insights were instrumental in setting up the Quarto templates and providing support for the discussion. His constructive feedback on the manuscript has greatly improved its quality. I feel fortunate to have had the opportunity to learn from and work with such a knowledgeable and dedicated mentor.</p>
<p>I would also like to thank my brothers, Jakob and Andreas, for their help with proofreading of the manuscript.</p>
</section>
<section id="compute" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="compute">Compute</h3>
<p>All computations were performed on GenomeDK (GDK), an HPC cluster located on Aarhus Uninversity, and most of the processing of the data was handled by a <em>gwf</em> workflow <span class="citation" data-cites="gwf">(<a href="#ref-gwf" role="doc-biblioref">GenomeDK 2023</a>)</span>, a workflow manager developed at GDK. Developing the manuscript and all associated writing, editing, and rendering the project was also performed on GDK so work seemlesly with the analysis. I would like to thank GDK and Aarhus University for providing computational resources and support that contributed to these research results. None of this would have been possible on my own laptop from 2013.</p>
</section>
<section id="use-of-generative-artificial-intelligence" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="use-of-generative-artificial-intelligence">Use of Generative Artificial Intelligence</h3>
<p>As per Aarhus University guidelines <span class="citation" data-cites="auityourguidegai">(<a href="#ref-auityourguidegai" role="doc-biblioref">IT n.d.</a>)</span>, I hereby disclose my use of Generative Artificial Intelligence (GAI) tools during the preparation of this thesis. All usage was conducted critically and responsibly, ensuring that the final work remained my own and adhered to academic integrity standards. These tools supplemented my expertise and accelerated routine tasks, but all creative, analytical, and scientific contributions are my own.</p>
<section id="copilot" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="copilot">Copilot</h4>
<p>I used Copilot for code completion and debugging, as well as modifying existing code to reflect updated variables and experimental needs.</p>
</section>
<section id="chatgpt" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="chatgpt">ChatGPT</h4>
<p>ChatGPT provided syntax support for Bash, YAML, JSON, Mermaid diagrams, and Pandoc Templates, to accelerate my learning of new programming languages. It hallucinates a lot about Quarto.</p>
</section>
<section id="notebooklm" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="notebooklm">NotebookLM</h4>
<p>NotebookLM supported the re-identification of references for a small amount of statements where placeholders (<code>[ref]</code>) were initially used before setting up BibTex references. Also to look up contradicting statements from other sources. NotebookLM only ‘knows’ the sources you provide it.</p>
</section>
</section>
</section>
</section>


<div id="quarto-appendix" class="default"><section id="footnotes" class="footnotes footnotes-end-of-document" role="doc-endnotes"><h2 class="anchored quarto-appendix-heading">Footnotes</h2>

<ol>
<li id="fn1"><p><span class="citation" data-cites="bicciato2022hicdataanalysis">Bicciato and Ferrari (<a href="#ref-bicciato2022hicdataanalysis" role="doc-biblioref">2022</a>)</span> mentions that this type of ligations had been used to model interaction between sister-chromatids post-replication in <em>Drosophila</em>.<a href="#fnref1" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn2"><p>And access to a high-performance computer with at least 12 TB storage and 32 CPU-cores, and a couple of days waiting time (else <code>gwf run</code> will fail as it asks for a lot of resources).<a href="#fnref2" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
</ol>
</section></div></main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<script>var lightboxQuarto = GLightbox({"closeEffect":"zoom","descPosition":"bottom","loop":false,"openEffect":"zoom","selector":".lightbox"});
(function() {
  let previousOnload = window.onload;
  window.onload = () => {
    if (previousOnload) {
      previousOnload();
    }
    lightboxQuarto.on('slide_before_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      const href = trigger.getAttribute('href');
      if (href !== null) {
        const imgEl = window.document.querySelector(`a[href="${href}"] img`);
        if (imgEl !== null) {
          const srcAttr = imgEl.getAttribute("src");
          if (srcAttr && srcAttr.startsWith("data:")) {
            slideConfig.href = srcAttr;
          }
        }
      } 
    });
  
    lightboxQuarto.on('slide_after_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(slideNode);
      }
    });
  
  };
  
})();
          </script>




</body></html>