<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.40">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2025-01-06">
<meta name="keywords" content="Hi-C, Chromatin Compartments, Selection on X, Selfish gene drive">

<title>Chromatin Compartments and Selection on X</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js" integrity="sha512-bLT0Qm9VnAYZDflyKcBaQ2gg0hSYNQrJ8RilYldYQ1FxQYoCLtUjuuRuZo+fjqhx/qtq/1itJ0C2ejDxltZVFg==" crossorigin="anonymous"></script><script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-549806ee2085284f45b00abea8c6df48.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-7bfb9ce031f0150bcdd0059ffcd3ab40.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script src="site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet">
<script async="" src="https://hypothes.is/embed.js"></script>
<script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" integrity="sha512-c3Nl8+7g4LMSTdrm621y7kf9v3SDPnhxLNhcjFJbKECVnmZHTdo+IRO05sNLTH/D3vA6u1X32ehoLC7WFVdheg==" crossorigin="anonymous"></script>

<script type="application/javascript">define('jquery', [],function() {return window.jQuery;})</script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<meta name="citation_title" content="Chromatin Compartments and Selection on X">
<meta name="citation_abstract" content="The hemizygosity of male mammals makes the X chromosome uniquely exposed to selective pressures, as the lack of a second copy provides no buffer against deleterious mutations. This, combined with a high density of essential genes related to reproduction and brain function, suggests the existence of biological mechanisms that protect the integrity of the X chromosome. One such mechanism is meiotic sex chromosome inactivation (MSCI) in males, while another could involve chromatin architecture.

In this study, the 3D chromatin architecture of the X chromosome in rhesus macaque (*Macaca mulata*) is investigated in the context of evolutionary pressures and genetic drivers. We redo Hi-C analyses from a 2019 paper on the latest reference genome (*rheMac10* or *Mmul_10*). We compare two Hi-C analysis frameworks, *HiCExplorer* and *cooler/cooltools* (Open2C), on a subset, finding Open2C to be most flexible and intuitive. The ICE method (Iterative Correction and Eigendecomposition) was used to infer conventional and refined A/B compartments for fibroblast and four stages of spermatogenesis.

We find that 200 kbp transition-zones between A/B-compartments in both fibroblasts and round spermatids align well with strong selective sweeps in humans (ECH-regions), and with strong negative selection in baboons (*Papio* spp.). We discuss the biological meaning of these findings, where conserved chromatin features may help to retain non-advantageous alleles, hinting to the role of selfish genetic elements in genome evolution.
">
<meta name="citation_keywords" content="Hi-C,Chromatin Compartments,Selection on X,Selfish gene drive">
<meta name="citation_author" content="Søren Jørgensen">
<meta name="citation_publication_date" content="2025-01-06">
<meta name="citation_cover_date" content="2025-01-06">
<meta name="citation_year" content="2025">
<meta name="citation_online_date" content="2025-01-06">
<meta name="citation_language" content="en">
<meta name="citation_reference" content="citation_title=Cooler: Scalable storage for Hi-C data and other genomically labeled arrays;,citation_abstract=Abstract Motivation Most existing coverage-based (epi)genomic datasets are one-dimensional, but newer technologies probing interactions (physical, genetic, etc.) produce quantitative maps with two-dimensional genomic coordinate systems. Storage and computational costs mount sharply with data resolution when such maps are stored in dense form. Hence, there is a pressing need to develop data storage strategies that handle the full range of useful resolutions in multidimensional genomic datasets by taking advantage of their sparse nature, while supporting efficient compression and providing fast random access to facilitate development of scalable algorithms for data analysis. Results We developed a file format called cooler, based on a sparse data model, that can support genomically labeled matrices at any resolution. It has the flexibility to accommodate various descriptions of the data axes (genomic coordinates, tracks and bin annotations), resolutions, data density patterns and metadata. Cooler is based on HDF5 and is supported by a Python library and command line suite to create, read, inspect and manipulate cooler data collections. The format has been adopted as a standard by the NIH 4D Nucleome Consortium. Availability and implementation Cooler is cross-platform, BSD-licensed and can be installed from the Python package index or the bioconda repository. The source code is maintained on Github at https://github.com/mirnylab/cooler. Supplementary information Supplementary data are available at Bioinformatics online.;,citation_author=Nezar Abdennur;,citation_author=Leonid A Mirny;,citation_editor=Jonathan Wren;,citation_publication_date=2020-01;,citation_cover_date=2020-01;,citation_year=2020;,citation_issue=1;,citation_doi=10.1093/bioinformatics/btz540;,citation_issn=1367-4803, 1367-4811;,citation_volume=36;,citation_language=en-US;,citation_journal_title=Bioinformatics;">
<meta name="citation_reference" content="citation_title=Accurate assembly of the olive baboon ( &amp;amp;amp;lt;i&amp;gt;Papio&amp;lt;/i&amp;gt; <i>Anubis</i> ) genome using long-read and Hi-C data;,citation_abstract=Abstract Background Baboons are a widely used nonhuman primate model for biomedical, evolutionary, and basic genetics research. Despite this importance, the genomic resources for baboons are limited. In particular, the current baboon reference genome Panu_3.0 is a highly fragmented, reference-guided (i.e., not fully de novo) assembly, and its poor quality inhibits our ability to conduct downstream genomic analyses. Findings Here we present a de novo genome assembly of the olive baboon (Papio anubis) that uses data from several recently developed single-molecule technologies. Our assembly, Panubis1.0, has an N50 contig size of $\sim$1.46 Mb (as opposed to 139 kb for Panu_3.0) and has single scaffolds that span each of the 20 autosomes and the X chromosome. Conclusions We highlight multiple lines of evidence (including Bionano Genomics data, pedigree linkage information, and linkage disequilibrium data) suggesting that there are several large assembly errors in Panu_3.0, which have been corrected in Panubis1.0.;,citation_author=Sanjit Singh Batra;,citation_author=Michal Levy-Sakin;,citation_author=Jacqueline Robinson;,citation_author=Joseph Guillory;,citation_author=Steffen Durinck;,citation_author=Tauras P Vilgalys;,citation_author=Pui-Yan Kwok;,citation_author=Laura A Cox;,citation_author=Somasekar Seshagiri;,citation_author=Yun S Song;,citation_author=Jeffrey D Wall;,citation_publication_date=2020-12;,citation_cover_date=2020-12;,citation_year=2020;,citation_issue=12;,citation_doi=10.1093/gigascience/giaa134;,citation_issn=2047-217X;,citation_volume=9;,citation_language=en-US;,citation_journal_title=GigaScience;">
<meta name="citation_reference" content="citation_title=Hi-C Data Analysis: Methods and Protocols;,citation_editor=Silvio Bicciato;,citation_editor=Francesco Ferrari;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_doi=10.1007/978-1-0716-1390-0;,citation_isbn=978-1-0716-1389-4 978-1-0716-1390-0;,citation_volume=2301;,citation_language=en-US;,citation_series_title=Methods in Molecular Biology;">
<meta name="citation_reference" content="citation_title=Genetic Villains: Killer Meiotic Drivers;,citation_author=María Angélica Bravo Núñez;,citation_author=Nicole L. Nuckolls;,citation_author=Sarah E. Zanders;,citation_publication_date=2018-06;,citation_cover_date=2018-06;,citation_year=2018;,citation_issue=6;,citation_doi=10.1016/j.tig.2018.02.003;,citation_issn=01689525;,citation_volume=34;,citation_language=en-US;,citation_journal_title=Trends in Genetics;">
<meta name="citation_reference" content="citation_title=dcHiC detects differential compartments across multiple Hi-C datasets;,citation_abstract=Abstract The compartmental organization of mammalian genomes and its changes play important roles in distinct biological processes. Here, we introduce dcHiC, which utilizes a multivariate distance measure to identify significant changes in compartmentalization among multiple contact maps. Evaluating dcHiC on four collections of bulk and single-cell contact maps from in vitro mouse neural differentiation (n=3), mouse hematopoiesis (n=10), human LCLs (n=20) and post-natal mouse brain development (n=3 stages), we show its effectiveness and sensitivity in detecting biologically relevant changes, including those orthogonally validated. dcHiC reported regions with dynamically regulated genes associated with cell identity, along with correlated changes in chromatin states, subcompartments, replication timing and lamin association. With its efficient implementation, dcHiC enables high-resolution compartment analysis as well as standalone browser visualization, differential interaction identification and time-series clustering. dcHiC is an essential addition to the Hi-C analysis toolbox for the ever-growing number of bulk and single-cell contact maps. Available at: https://github.com/ay-lab/dcHiC .;,citation_author=Abhijit Chakraborty;,citation_author=Jeffrey G. Wang;,citation_author=Ferhat Ay;,citation_publication_date=2022-11;,citation_cover_date=2022-11;,citation_year=2022;,citation_issue=1;,citation_doi=10.1038/s41467-022-34626-6;,citation_issn=2041-1723;,citation_volume=13;,citation_language=en-US;,citation_journal_title=Nature Communications;">
<meta name="citation_reference" content="citation_title=High resolution maps of chromatin reorganization through mouse meiosis reveal novel features of the 3D meiotic structure;,citation_abstract=Abstract When germ cells transition from the mitotic cycle into meiotic prophase I (MPI), chromosomes condense into an array of chromatin loops that are required to promote homolog pairing and genetic recombination. To identify the changes in chromosomal conformation, we isolated nuclei on a trajectory from spermatogonia to the end of MPI. At each stage along this trajectory, we built genomic interaction maps with the highest temporal and spatial resolution to date. The changes in chromatin folding coincided with a concurrent decline in mitotic cohesion and a rise in meiotic cohesin complexes. We found that the stereotypical large-scale A and B compartmentalization was lost during meiotic prophase I alongside the loss of topological associating domains (TADs). Still, local subcompartments were detected and maintained throughout meiosis. The enhanced Micro-C resolution revealed that, despite the loss of TADs, higher frequency contact sites between two loci were detectable during meiotic prophase I coinciding with CTCF bound sites. The pattern of interactions around these CTCF sites with their neighboring loci showed that CTCF sites were often anchoring the meiotic loops. Additionally, the localization of CTCF to the meiotic axes indicated that these anchors were at the base of loops. Strikingly, even in the face of the dramatic reconfiguration of interphase chromatin into a condensed loop-array, the interactions between regulatory elements remained well preserved. This establishes a potential mechanism for how the meiotic chromatin maintains active transcription within a highly structured genome. In summary, the high temporal and spatial resolution of these data revealed previously unappreciated aspects of mammalian meiotic chromatin organization.;,citation_author=Gang Cheng;,citation_author=Florencia Pratto;,citation_author=Kevin Brick;,citation_author=Xin Li;,citation_author=Benjamin Alleva;,citation_author=Mini Huang;,citation_author=Gabriel Lam;,citation_author=R. Daniel Camerini-Otero;,citation_publication_date=2024-03;,citation_cover_date=2024-03;,citation_year=2024;,citation_doi=10.1101/2024.03.25.586627;,citation_language=en-US;">
<meta name="citation_reference" content="citation_title=100 years of Haldane’s rule;,citation_abstract=Abstract Haldane’s rule is one of the “two rules of speciation”. It states that if one sex is “absent, rare or sterile” in a hybrid population, then that sex will be heterogametic. Since Haldane first made this observation, 100&nbsp;years have passed and still questions arise over how many independent examples exist and what the underlying causes of Haldane’s rule are. This review aims to examine research that has occurred over the last century. It seeks to do so by discussing possible causes of Haldane’s rule, as well as gaps in the research of these causes that could be readily addressed today. After 100&nbsp;years of research, it can be concluded that Haldane’s rule is a complicated one, and much current knowledge has been accrued by studying the model organisms of speciation. This has led to the primacy of dominance theory and faster-male theory as explanations for Haldane’s rule. However, some of the most interesting findings of the 21st century with regard to Haldane’s rule have involved investigating a wider range of taxa emphasizing the need to continue using comparative methods, including ever more taxa as new cases are discovered.;,citation_author=Finn Cowell;,citation_publication_date=2023-02;,citation_cover_date=2023-02;,citation_year=2023;,citation_issue=2;,citation_doi=10.1111/jeb.14112;,citation_issn=1010-061X, 1420-9101;,citation_volume=36;,citation_language=en-US;,citation_journal_title=Journal of Evolutionary Biology;">
<meta name="citation_reference" content="citation_title=Chromatin architecture reorganization during stem cell differentiation;,citation_abstract=Higher-order chromatin structure is emerging as an important regulator of gene expression. Although dynamic chromatin structures have been identified in the genome, the full scope of chromatin dynamics during mammalian development and lineage specification remains to be determined. By mapping genome-wide chromatin interactions in human embryonic stem (ES) cells and four human ES-cell-derived lineages, we uncover extensive chromatin reorganization during lineage specification. We observe that although self-associating chromatin domains are stable during differentiation, chromatin interactions both within and between domains change in a striking manner, altering 36% of active and inactive chromosomal compartments throughout the genome. By integrating chromatin interaction maps with haplotype-resolved epigenome and transcriptome data sets, we find widespread allelic bias in gene expression correlated with allele-biased chromatin states of linked promoters and distal enhancers. Our results therefore provide a global view of chromatin dynamics and a resource for studying long-range control of gene expression in distinct human cell lineages.;,citation_author=Jesse R. Dixon;,citation_author=Inkyung Jung;,citation_author=Siddarth Selvaraj;,citation_author=Yin Shen;,citation_author=Jessica E. Antosiewicz-Bourget;,citation_author=Ah Young Lee;,citation_author=Zhen Ye;,citation_author=Audrey Kim;,citation_author=Nisha Rajagopal;,citation_author=Wei Xie;,citation_author=Yarui Diao;,citation_author=Jing Liang;,citation_author=Huimin Zhao;,citation_author=Victor V. Lobanenkov;,citation_author=Joseph R. Ecker;,citation_author=James A. Thomson;,citation_author=Bing Ren;,citation_publication_date=2015-02;,citation_cover_date=2015-02;,citation_year=2015;,citation_issue=7539;,citation_doi=10.1038/nature14222;,citation_issn=1476-4687;,citation_volume=518;,citation_journal_title=Nature;">
<meta name="citation_reference" content="citation_title=Sex ratio and unisexual sterility in hybrid animals;,citation_author=J. B. S. Haldane;,citation_publication_date=1922-10;,citation_cover_date=1922-10;,citation_year=1922;,citation_issue=2;,citation_doi=10.1007/BF02983075;,citation_issn=0022-1333;,citation_volume=12;,citation_language=en-US;,citation_journal_title=Journal of Genetics;">
<meta name="citation_reference" content="citation_title=Chromatin alternates between A and B compartments at kilobase scale for subgenic organization;,citation_abstract=Abstract Nuclear compartments are prominent features of 3D chromatin organization, but sequencing depth limitations have impeded investigation at ultra fine-scale. CTCF loops are generally studied at a finer scale, but the impact of looping on proximal interactions remains enigmatic. Here, we critically examine nuclear compartments and CTCF loop-proximal interactions using a combination of in situ Hi-C at unparalleled depth, algorithm development, and biophysical modeling. Producing a large Hi-C map with 33 billion contacts in conjunction with an algorithm for performing principal component analysis on sparse, super massive matrices&nbsp;(POSSUMM), we resolve compartments to 500 bp. Our results demonstrate that essentially all active promoters and distal enhancers localize in the A compartment, even when flanking sequences do not. Furthermore, we find that the TSS and TTS of paused genes are often segregated into separate compartments. We then identify diffuse interactions that radiate from CTCF loop anchors, which correlate with strong enhancer-promoter interactions and proximal transcription. We also find that these diffuse interactions depend on CTCF’s RNA binding domains. In this work, we demonstrate features of fine-scale chromatin organization consistent with a revised model in which compartments are more precise than commonly thought while CTCF loops are more protracted.;,citation_author=Hannah L. Harris;,citation_author=Huiya Gu;,citation_author=Moshe Olshansky;,citation_author=Ailun Wang;,citation_author=Irene Farabella;,citation_author=Yossi Eliaz;,citation_author=Achyuth Kalluchi;,citation_author=Akshay Krishna;,citation_author=Mozes Jacobs;,citation_author=Gesine Cauer;,citation_author=Melanie Pham;,citation_author=Suhas S. P. Rao;,citation_author=Olga Dudchenko;,citation_author=Arina Omer;,citation_author=Kiana Mohajeri;,citation_author=Sungjae Kim;,citation_author=Michael H. Nichols;,citation_author=Eric S. Davis;,citation_author=Dimos Gkountaroulis;,citation_author=Devika Udupa;,citation_author=Aviva Presser Aiden;,citation_author=Victor G. Corces;,citation_author=Douglas H. Phanstiel;,citation_author=William Stafford Noble;,citation_author=Guy Nir;,citation_author=Michele Di Pierro;,citation_author=Jeong-Sun Seo;,citation_author=Michael E. Talkowski;,citation_author=Erez Lieberman Aiden;,citation_author=M. Jordan Rowley;,citation_publication_date=2023-06;,citation_cover_date=2023-06;,citation_year=2023;,citation_issue=1;,citation_doi=10.1038/s41467-023-38429-1;,citation_issn=2041-1723;,citation_volume=14;,citation_language=en-US;,citation_journal_title=Nature Communications;">
<meta name="citation_reference" content="citation_title=Iterative correction of Hi-C data reveals hallmarks of chromosome organization;,citation_author=Maxim Imakaev;,citation_author=Geoffrey Fudenberg;,citation_author=Rachel Patton McCord;,citation_author=Natalia Naumova;,citation_author=Anton Goloborodko;,citation_author=Bryan R Lajoie;,citation_author=Job Dekker;,citation_author=Leonid A Mirny;,citation_publication_date=2012-10;,citation_cover_date=2012-10;,citation_year=2012;,citation_issue=10;,citation_doi=10.1038/nmeth.2148;,citation_issn=1548-7091, 1548-7105;,citation_volume=9;,citation_language=en-US;,citation_journal_title=Nature Methods;">
<meta name="citation_reference" content="citation_title=Sex Chromosome Meiotic Drive;,citation_abstract=▪ Abstract Sex chromosome drive refers to the unequal transmission of X and Y chromosomes from individuals of the heterogametic sex, resulting in biased sex ratios among progeny and within populations. The presence of driving sex chromosomes can reduce mean fitness within a population, bring about intragenomic conflict between the X chromosome, the Y, and the autosomes, and alter the intensity or mode of sexual selection within species. Sex chromosome drive, or its genetic equivalent, is known in plants, mammals, and flies. Many species harboring driving X chromosomes have evolved Y-linked and autosomal suppressors of drive. If a drive polymorphism is not stable, then driving chromosomes may spread to fixation and cause the extinction of a species. Certain characteristics of species, such as population density and female mating rate, may affect the probability of fixation of driving chromosomes. Thus, sex chromosome drive could be an agent of species-level selection.;,citation_author=John Jaenike;,citation_publication_date=2001-11;,citation_cover_date=2001-11;,citation_year=2001;,citation_issue=1;,citation_doi=10.1146/annurev.ecolsys.32.081501.113958;,citation_issn=0066-4162;,citation_volume=32;,citation_language=en-US;,citation_journal_title=Annual Review of Ecology and Systematics;">
<meta name="citation_reference" content="citation_title=Do sex-linked male meiotic drivers contribute to intrinsic hybrid incompatibilities? Recent empirical studies from flies and rodents;,citation_author=Jun Kitano;,citation_author=Kohta Yoshida;,citation_publication_date=2023-08;,citation_cover_date=2023-08;,citation_year=2023;,citation_doi=10.1016/j.gde.2023.102068;,citation_issn=0959437X;,citation_volume=81;,citation_language=en-US;,citation_journal_title=Current Opinion in Genetics &amp;amp;amp; Development;">
<meta name="citation_reference" content="citation_title=The Hitchhiker’s Guide to Hi-C Analysis: Practical guidelines;,citation_abstract=Over the last decade, development and application of a set of molecular genomic approaches based on the chromosome conformation capture method (3C), combined with increasingly powerful imaging approaches, have enabled high resolution and genome-wide analysis of the spatial organization of chromosomes. The aim of this paper is to provide guidelines for analyzing and interpreting data obtained with genome-wide 3C methods such as Hi-C and 3C-seq that rely on deep sequencing to detect and quantify pairwise chromatin interactions genome-wide.;,citation_author=Bryan R. Lajoie;,citation_author=Job Dekker;,citation_author=Noam Kaplan;,citation_publication_date=2015-01;,citation_cover_date=2015-01;,citation_year=2015;,citation_doi=10.1016/j.ymeth.2014.10.031;,citation_issn=1046-2023;,citation_pmid=25448293;,citation_volume=72;,citation_journal_title=Methods (San Diego, Calif.);">
<meta name="citation_reference" content="citation_title=Comprehensive Mapping of Long-Range Interactions Reveals Folding Principles of the Human Genome;,citation_abstract=Chromosomal Mapping The conformation of the genome in the nucleus and contacts between both proximal and distal loci influence gene expression. In order to map genomic contacts, Lieberman-Aiden et al. (p. 289 , see the cover) developed a technique to allow the detection of all interactions between genomic loci in the eukaryotic nucleus followed by deep sequencing. This technology was used to map the organization of the human genome and to examine the spatial proximity of chromosomal loci at one megabase resolution. The map suggests that the genome is partitioned into two spatial compartments that are related to local chromatin state and whose remodeling correlates with changes in the chromatin state. , Chromosomes are organized in a fractal knot-free conformation that is densely packed while easily folded and unfolded. , We describe Hi-C, a method that probes the three-dimensional architecture of whole genomes by coupling proximity-based ligation with massively parallel sequencing. We constructed spatial proximity maps of the human genome with Hi-C at a resolution of 1 megabase. These maps confirm the presence of chromosome territories and the spatial proximity of small, gene-rich chromosomes. We identified an additional level of genome organization that is characterized by the spatial segregation of open and closed chromatin to form two genome-wide compartments. At the megabase scale, the chromatin conformation is consistent with a fractal globule, a knot-free, polymer conformation that enables maximally dense packing while preserving the ability to easily fold and unfold any genomic locus. The fractal globule is distinct from the more commonly used globular equilibrium model. Our results demonstrate the power of Hi-C to map the dynamic conformations of whole genomes.;,citation_author=Erez Lieberman-Aiden;,citation_author=Nynke L. Van Berkum;,citation_author=Louise Williams;,citation_author=Maxim Imakaev;,citation_author=Tobias Ragoczy;,citation_author=Agnes Telling;,citation_author=Ido Amit;,citation_author=Bryan R. Lajoie;,citation_author=Peter J. Sabo;,citation_author=Michael O. Dorschner;,citation_author=Richard Sandstrom;,citation_author=Bradley Bernstein;,citation_author=M. A. Bender;,citation_author=Mark Groudine;,citation_author=Andreas Gnirke;,citation_author=John Stamatoyannopoulos;,citation_author=Leonid A. Mirny;,citation_author=Eric S. Lander;,citation_author=Job Dekker;,citation_publication_date=2009-10;,citation_cover_date=2009-10;,citation_year=2009;,citation_issue=5950;,citation_doi=10.1126/science.1181369;,citation_issn=0036-8075, 1095-9203;,citation_volume=326;,citation_language=en-US;,citation_journal_title=Science;">
<meta name="citation_reference" content="citation_title=The Ecology and Evolutionary Dynamics of Meiotic Drive;,citation_author=Anna K. Lindholm;,citation_author=Kelly A. Dyer;,citation_author=Renée C. Firman;,citation_author=Lila Fishman;,citation_author=Wolfgang Forstmeier;,citation_author=Luke Holman;,citation_author=Hanna Johannesson;,citation_author=Ulrich Knief;,citation_author=Hanna Kokko;,citation_author=Amanda M. Larracuente;,citation_author=Andri Manser;,citation_author=Catherine Montchamp-Moreau;,citation_author=Varos G. Petrosyan;,citation_author=Andrew Pomiankowski;,citation_author=Daven C. Presgraves;,citation_author=Larisa D. Safronova;,citation_author=Andreas Sutter;,citation_author=Robert L. Unckless;,citation_author=Rudi L. Verspoor;,citation_author=Nina Wedell;,citation_author=Gerald S. Wilkinson;,citation_author=Tom A. R. Price;,citation_publication_date=2016-04;,citation_cover_date=2016-04;,citation_year=2016;,citation_issue=4;,citation_doi=10.1016/j.tree.2016.02.001;,citation_issn=01695347;,citation_volume=31;,citation_language=en-US;,citation_journal_title=Trends in Ecology &amp;amp;amp; Evolution;">
<meta name="citation_reference" content="citation_title=Munch-group;,citation_author=Kasper Munch;,citation_publication_date=2024-11;,citation_cover_date=2024-11;,citation_year=2024;,citation_publisher=https://munch-group.org/research.html;">
<meta name="citation_reference" content="citation_title=Cooltools: Enabling high-resolution Hi-C analysis in Python;,citation_abstract=Abstract Chromosome conformation capture (3C) technologies reveal the incredible complexity of genome organization. Maps of increasing size, depth, and resolution are now used to probe genome architecture across cell states, types, and organisms. Larger datasets add challenges at each step of computational analysis, from storage and memory constraints to researchers’ time; however, analysis tools that meet these increased resource demands have not kept pace. Furthermore, existing tools offer limited support for customizing analysis for specific use cases or new biology. Here we introduce cooltools ( https://github.com/open2c/cooltools ), a suite of computational tools that enables flexible, scalable, and reproducible analysis of high-resolution contact frequency data. Cooltools leverages the widely-adopted cooler format which handles storage and access for high-resolution datasets. Cooltools provides a paired command line interface (CLI) and Python application programming interface (API), which respectively facilitate workflows on high-performance computing clusters and in interactive analysis environments. In short, cooltools enables the effective use of the latest and largest genome folding datasets.;,citation_author=Open2C;,citation_author=Nezar Abdennur;,citation_author=Sameer Abraham;,citation_author=Geoffrey Fudenberg;,citation_author=Ilya M. Flyamer;,citation_author=Aleksandra A. Galitsyna;,citation_author=Anton Goloborodko;,citation_author=Maxim Imakaev;,citation_author=Betul A. Oksuz;,citation_author=Sergey V. Venev;,citation_publication_date=2022-11;,citation_cover_date=2022-11;,citation_year=2022;,citation_doi=10.1101/2022.10.31.514564;,citation_language=en-US;,citation_publisher=Bioinformatics;">
<meta name="citation_reference" content="citation_title=Pairtools: From sequencing data to chromosome contacts;,citation_abstract=The field of 3D genome organization produces large amounts of sequencing data from Hi-C and a rapidly-expanding set of other chromosome conformation protocols (3C+). Massive and heterogeneous 3C+ data require high-performance and flexible processing of sequenced reads into contact pairs. To meet these challenges, we present pairtools –a flexible suite of tools for contact extraction from sequencing data. Pairtools provides modular command-line interface (CLI) tools that can be flexibly chained into data processing pipelines. The core operations provided by pairtools are parsing of.sam alignments into Hi-C pairs, sorting and removal of PCR duplicates. In addition, pairtools provides auxiliary tools for building feature-rich 3C+ pipelines, including contact pair manipulation, filtration, and quality control. Benchmarking pairtools against popular 3C+ data pipelines shows advantages of pairtools for high-performance and flexible 3C+ analysis. Finally, pairtools provides protocol-specific tools for restriction-based protocols, haplotype-resolved contacts, and single-cell Hi-C. The combination of CLI tools and tight integration with Python data analysis libraries makes pairtools a versatile foundation for a broad range of 3C+ pipelines.;,citation_author=Open2C;,citation_author=Nezar Abdennur;,citation_author=Geoffrey Fudenberg;,citation_author=Ilya M. Flyamer;,citation_author=Aleksandra A. Galitsyna;,citation_author=Anton Goloborodko;,citation_author=Maxim Imakaev;,citation_author=Sergey V. Venev;,citation_editor=Ferhat Ay;,citation_publication_date=2024-05;,citation_cover_date=2024-05;,citation_year=2024;,citation_issue=5;,citation_doi=10.1371/journal.pcbi.1012164;,citation_issn=1553-7358;,citation_volume=20;,citation_language=en-US;,citation_journal_title=PLOS Computational Biology;">
<meta name="citation_reference" content="citation_title=Open Chromosome Collective (Open2C);,citation_abstract=We are the Open Chromosome Collective (Open2C). We are a group of computational biologists working on chromosome structure and interested in developing open source tools for the wider community. We are particularly interested in the 3D organization of chromosomes, and most of the tools are focused on the analysis of data obtained using Hi-C and related technologies. We like our tools to be easy to use, flexible, to facilitate active development of novel analytical approaches, and scalable, to make use of the latest and largest datasets.;,citation_language=en-US;,citation_journal_title=Open2C;,citation_publisher=https://open2c.github.io/;">
<meta name="citation_reference" content="citation_title=Principal components analysis corrects for stratification in genome-wide association studies;,citation_author=Alkes L Price;,citation_author=Nick J Patterson;,citation_author=Robert M Plenge;,citation_author=Michael E Weinblatt;,citation_author=Nancy A Shadick;,citation_author=David Reich;,citation_publication_date=2006-08;,citation_cover_date=2006-08;,citation_year=2006;,citation_issue=8;,citation_doi=10.1038/ng1847;,citation_issn=1061-4036, 1546-1718;,citation_volume=38;,citation_language=en-US;,citation_journal_title=Nature Genetics;">
<meta name="citation_reference" content="citation_title=High-resolution TADs reveal DNA sequences underlying genome organization in flies;,citation_abstract=Abstract Despite an abundance of new studies about topologically associating domains (TADs), the role of genetic information in TAD formation is still not fully understood. Here we use our software, HiCExplorer ( hicexplorer.readthedocs.io ) to annotate $&amp;amp;amp;gt;$2800 high-resolution (570 bp) TAD boundaries&nbsp;in Drosophila&nbsp;melanogaster . We identify eight DNA motifs enriched at boundaries, including a motif bound by the M1BP protein, and two new boundary motifs. In contrast to mammals, the CTCF motif is only enriched on a small fraction of boundaries flanking inactive chromatin while most active boundaries contain the motifs bound by the M1BP or Beaf-32 proteins. We demonstrate that boundaries can be accurately predicted using only the motif sequences at open chromatin sites. We propose that DNA sequence guides the genome architecture by allocation of boundary proteins in the genome. Finally, we present an interactive online database to access and explore the spatial organization of fly, mouse and human genomes, available at http://chorogenome.ie-freiburg.mpg.de .;,citation_author=Fidel Ramírez;,citation_author=Vivek Bhardwaj;,citation_author=Laura Arrigoni;,citation_author=Kin Chung Lam;,citation_author=Björn A. Grüning;,citation_author=José Villaveces;,citation_author=Bianca Habermann;,citation_author=Asifa Akhtar;,citation_author=Thomas Manke;,citation_publication_date=2018-01;,citation_cover_date=2018-01;,citation_year=2018;,citation_issue=1;,citation_doi=10.1038/s41467-017-02525-w;,citation_issn=2041-1723;,citation_volume=9;,citation_language=en-US;,citation_journal_title=Nature Communications;">
<meta name="citation_reference" content="citation_title=Loops, topologically associating domains, compartments, and territories are elastic and robust to dramatic nuclear volume swelling;,citation_abstract=Abstract Layers of genome organization are becoming increasingly better characterized, but less is known about how these structures respond to perturbation or shape changes. Low-salt swelling of isolated chromatin fibers or nuclei has been used for decades to investigate the structural properties of chromatin. But, visible changes in chromatin appearance have not been linked to known building blocks of genome structure or features along the genome sequence. We combine low-salt swelling of isolated nuclei with genome-wide chromosome conformation capture (Hi-C) and imaging approaches to probe the effects of chromatin extension genome-wide. Photoconverted patterns on nuclei during expansion and contraction indicate that global genome structure is preserved after dramatic nuclear volume swelling, suggesting a highly elastic chromosome topology. Hi-C experiments before, during, and after nuclear swelling show changes in average contact probabilities at short length scales, reflecting the extension of the local chromatin fiber. But, surprisingly, during this large increase in nuclear volume, there is a striking maintenance of loops, TADs, active and inactive compartments, and chromosome territories. Subtle differences after expansion are observed, suggesting that the local chromatin state, protein interactions, and location in the nucleus can affect how strongly a given structure is maintained under stress. From these observations, we propose that genome topology is robust to extension of the chromatin fiber and isotropic shape change, and that this elasticity may be beneficial in physiological circumstances of changes in nuclear size and volume.;,citation_author=Jacob T. Sanders;,citation_author=Rosela Golloshi;,citation_author=Priyojit Das;,citation_author=Yang Xu;,citation_author=Peyton H. Terry;,citation_author=Darrian G. Nash;,citation_author=Job Dekker;,citation_author=Rachel Patton McCord;,citation_publication_date=2022-03;,citation_cover_date=2022-03;,citation_year=2022;,citation_issue=1;,citation_doi=10.1038/s41598-022-08602-5;,citation_issn=2045-2322;,citation_volume=12;,citation_language=en-US;,citation_journal_title=Scientific Reports;">
<meta name="citation_reference" content="citation_title=HiC-Pro: An optimized and flexible pipeline for Hi-C data processing;,citation_author=Nicolas Servant;,citation_author=Nelle Varoquaux;,citation_author=Bryan R. Lajoie;,citation_author=Eric Viara;,citation_author=Chong-Jian Chen;,citation_author=Jean-Philippe Vert;,citation_author=Edith Heard;,citation_author=Job Dekker;,citation_author=Emmanuel Barillot;,citation_publication_date=2015-12;,citation_cover_date=2015-12;,citation_year=2015;,citation_issue=1;,citation_doi=10.1186/s13059-015-0831-x;,citation_issn=1474-760X;,citation_volume=16;,citation_language=en-US;,citation_journal_title=Genome Biology;">
<meta name="citation_reference" content="citation_title=Single-Cell RNA Sequencing of Human, Macaque, and Mouse Testes Uncovers Conserved and Divergent Features of Mammalian Spermatogenesis;,citation_author=Adrienne Niederriter Shami;,citation_author=Xianing Zheng;,citation_author=Sarah K. Munyoki;,citation_author=Qianyi Ma;,citation_author=Gabriel L. Manske;,citation_author=Christopher D. Green;,citation_author=Meena Sukhwani;,citation_author=Kyle E. Orwig;,citation_author=Jun Z. Li;,citation_author=Saher Sue Hammoud;,citation_publication_date=2020-08;,citation_cover_date=2020-08;,citation_year=2020;,citation_issue=4;,citation_doi=10.1016/j.devcel.2020.05.010;,citation_issn=15345807;,citation_volume=54;,citation_language=en-US;,citation_journal_title=Developmental Cell;">
<meta name="citation_reference" content="citation_title=Extraordinary selection on the human X chromosome associated with archaic admixture;,citation_author=Laurits Skov;,citation_author=Moisès Coll Macià;,citation_author=Elise Anne Lucotte;,citation_author=Maria Izabel Alves Cavassim;,citation_author=David Castellano;,citation_author=Mikkel Heide Schierup;,citation_author=Kasper Munch;,citation_publication_date=2023-03;,citation_cover_date=2023-03;,citation_year=2023;,citation_issue=3;,citation_doi=10.1016/j.xgen.2023.100274;,citation_issn=2666979X;,citation_volume=3;,citation_language=en-US;,citation_journal_title=Cell Genomics;">
<meta name="citation_reference" content="citation_title=Genome-wide coancestry reveals details of ancient and recent male-driven reticulation in baboons;,citation_abstract=Baboons (genus Papio ) are a morphologically and behaviorally diverse clade of catarrhine monkeys that have experienced hybridization between phenotypically and genetically distinct phylogenetic species. We used high-coverage whole-genome sequences from 225 wild baboons representing 19 geographic localities to investigate population genomics and interspecies gene flow. Our analyses provide an expanded picture of evolutionary reticulation among species and reveal patterns of population structure within and among species, including differential admixture among conspecific populations. We describe the first example of a baboon population with a genetic composition that is derived from three distinct lineages. The results reveal processes, both ancient and recent, that produced the observed mismatch between phylogenetic relationships based on matrilineal, patrilineal, and biparental inheritance. We also identified several candidate genes that may contribute to species-specific phenotypes. , INTRODUCTION As a widespread but comparatively young clade of six parapatric species, the baboons ( Papio sp.) exemplify a frequently observed pattern of mammalian diversity. In particular, they provide analogs for the population structure of the multibranched prehuman lineage that occupied a similar geographic range before the hegemony of “modern” humans, Homo sapiens . Despite phenotypic and genetic differences, interspecies hybridization has been described between baboons at several locations, and population relationships based on mitochondrial DNA (mtDNA) do not correspond with relationships based on phenotype. These previous studies captured the broad outlines of baboon population genetic structure and evolutionary history but necessarily used data that were limited in genomic and geographical coverage and therefore could not adequately document inter- and intrapopulation variation. In this study, we analyzed whole-genome sequences of 225 baboons representing all six species and 19 geographic sites, with 18 local populations represented by multiple individuals. RATIONALE Recent studies have identified several mammalian species groups in which genetically distinct lineages have hybridized to generate complex reticulate phylogenies. Baboons provide a valuable context for studying processes generating such population and phylogenetic complexity because extant parapatric species form hybrid zones in several regions of Africa, allowing for direct observation of ongoing introgression. Furthermore, prior studies of nuclear and mtDNA and phenotypic diversity have demonstrated gene flow among differentiated lineages but were unable to develop the detailed picture of process and history that is now possible using whole-genome sequences and modern computational methods. To address these questions, we designed a study that would provide a more fine-grained picture of recent and ancient genetic reticulation by comparing phenotypes and autosomal, X and Y chromosomal, and mtDNA sequences, along with polymorphic insertions of repetitive elements across multiple baboon populations. RESULTS Using deep whole-genome sequence data from 225 baboons representing multiple populations, we identified several previously unknown geographic sites of gene flow between genetically distinct populations. We report that yellow baboons ( P. cynocephalus ) from western Tanzania are the first nonhuman primate found to have received genetic input from three distinct lineages. We compared the ancestry shared among individuals, estimated separately from the X chromosome and autosomes, to distinguish shared ancestry due to ancestral population relationships from coancestry as a result of recent male-biased immigration and gene flow. This reveals directionality and sex bias of recent gene flow in several locations. Analyses of population differences within species quantified different degrees of interspecies introgression among populations with an essentially identical phenotype. CONCLUSION The population genetic structure and history of introgression among baboon lineages are even more complex than predicted from observed phenotypic diversity and prior studies of limited genetic data. Single populations can carry genetic contributions from more than two ancestral sources. Populations that appear homogeneous on the basis of observable phenotype can display different levels of interspecies introgression. The evolutionary dynamics and current structure of baboon population diversity indicate that other mammals displaying differentiated and geographically separate species may also have more-complex histories than anticipated. This may also be true for the morphologically defined hominin taxa from the past 4 million years. Ancient and recent admixture among baboons: Complex population substructure and reticulation revealed by whole-genome sequencing. Pie charts represent recent ancestry of East African populations, with species contributions colored as in the inset map. Patterns of mixed ancestry differ substantially, even among conspecific populations. This suggests a complex history of recurrent interpopulational gene flow, driven predominantly by male migration. Comparably complex admixture probably also occurred among early hominins.;,citation_author=Erik F. Sørensen;,citation_author=R. Alan Harris;,citation_author=Liye Zhang;,citation_author=Muthuswamy Raveendran;,citation_author=Lukas F. K. Kuderna;,citation_author=Jerilyn A. Walker;,citation_author=Jessica M. Storer;,citation_author=Martin Kuhlwilm;,citation_author=Claudia Fontsere;,citation_author=Lakshmi Seshadri;,citation_author=Christina M. Bergey;,citation_author=Andrew S. Burrell;,citation_author=Juraj Bergman;,citation_author=Jane E. Phillips-Conroy;,citation_author=Fekadu Shiferaw;,citation_author=Kenneth L. Chiou;,citation_author=Idrissa S. Chuma;,citation_author=Julius D. Keyyu;,citation_author=Julia Fischer;,citation_author=Marie-Claude Gingras;,citation_author=Sejal Salvi;,citation_author=Harshavardhan Doddapaneni;,citation_author=Mikkel H. Schierup;,citation_author=Mark A. Batzer;,citation_author=Clifford J. Jolly;,citation_author=Sascha Knauf;,citation_author=Dietmar Zinner;,citation_author=Kyle K.-H. Farh;,citation_author=Tomas Marques-Bonet;,citation_author=Kasper Munch;,citation_author=Christian Roos;,citation_author=Jeffrey Rogers;,citation_publication_date=2023-06;,citation_cover_date=2023-06;,citation_year=2023;,citation_issue=6648;,citation_doi=10.1126/science.abn8153;,citation_issn=0036-8075, 1095-9203;,citation_volume=380;,citation_language=en-US;,citation_journal_title=Science;">
<meta name="citation_reference" content="citation_title=Reprogramming of Meiotic Chromatin Architecture during Spermatogenesis;,citation_author=Yao Wang;,citation_author=Hanben Wang;,citation_author=Yu Zhang;,citation_author=Zhenhai Du;,citation_author=Wei Si;,citation_author=Suixing Fan;,citation_author=Dongdong Qin;,citation_author=Mei Wang;,citation_author=Yanchao Duan;,citation_author=Lufan Li;,citation_author=Yuying Jiao;,citation_author=Yuanyuan Li;,citation_author=Qiujun Wang;,citation_author=Qinghua Shi;,citation_author=Xin Wu;,citation_author=Wei Xie;,citation_publication_date=2019-02;,citation_cover_date=2019-02;,citation_year=2019;,citation_issue=3;,citation_doi=10.1016/j.molcel.2018.11.019;,citation_issn=10972765;,citation_volume=73;,citation_language=en-US;,citation_journal_title=Molecular Cell;">
<meta name="citation_reference" content="citation_title=3D chromatin structure changes during spermatogenesis and oogenesis;,citation_author=Shiqiang Zhang;,citation_author=Wanyu Tao;,citation_author=Jing-Dong J. Han;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_doi=10.1016/j.csbj.2022.05.032;,citation_issn=20010370;,citation_volume=20;,citation_language=en-US;,citation_journal_title=Computational and Structural Biotechnology Journal;">
<meta name="citation_reference" content="citation_title=Stage-resolved Hi-C analyses reveal meiotic chromosome organizational features influencing homolog alignment;,citation_abstract=Abstract During meiosis, chromosomes exhibit dramatic changes in morphology and intranuclear positioning. How these changes influence homolog pairing, alignment, and recombination remain elusive. Using Hi-C, we systematically mapped 3D genome architecture throughout all meiotic prophase substages during mouse spermatogenesis. Our data uncover two major chromosome organizational features varying along the chromosome axis during early meiotic prophase, when homolog alignment occurs. First, transcriptionally active and inactive genomic regions form alternating domains consisting of shorter and longer chromatin loops, respectively. Second, the force-transmitting LINC complex promotes the alignment of ends of different chromosomes over a range of up to 20% of chromosome length. Both features correlate with the pattern of homolog interactions and the distribution of recombination events. Collectively, our data reveal the influences of transcription and force on meiotic chromosome structure and suggest chromosome organization may provide an infrastructure for the modulation of meiotic recombination in higher eukaryotes.;,citation_author=Wu Zuo;,citation_author=Guangming Chen;,citation_author=Zhimei Gao;,citation_author=Shuai Li;,citation_author=Yanyan Chen;,citation_author=Chenhui Huang;,citation_author=Juan Chen;,citation_author=Zhengjun Chen;,citation_author=Ming Lei;,citation_author=Qian Bian;,citation_publication_date=2021-10;,citation_cover_date=2021-10;,citation_year=2021;,citation_issue=1;,citation_doi=10.1038/s41467-021-26033-0;,citation_issn=2041-1723;,citation_volume=12;,citation_language=en-US;,citation_journal_title=Nature Communications;">
<meta name="citation_reference" content="citation_title=SRA Toolkit;,citation_abstract=SRA Tools. Contribute to ncbi/sra-tools development by creating an account on GitHub.;,citation_author=SRA Toolkit DevTeam;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_fulltext_html_url=https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=software;,citation_language=en-US;,citation_journal_title=GitHub;,citation_publisher=https://github.com/ncbi/sra-tools/wiki/01.-Downloading-SRA-Toolkit;">
<meta name="citation_reference" content="citation_title=Aligning sequence reads, clone sequences and assembly contigs with BWA-MEM;,citation_author=Heng Li;,citation_publication_date=2013;,citation_cover_date=2013;,citation_year=2013;,citation_fulltext_html_url=https://arxiv.org/abs/1303.3997;">
<meta name="citation_reference" content="citation_title=Sequence diversity analyses of an improved rhesus macaque genome enhance its biomedical utility;,citation_abstract=A high-quality rhesus macaque genome. Genome technology has improved substantially since the first full organismal genomes were generated. Applying new technology, Warren et al. refined the genome of the rhesus macaque, a model nonhuman primate. Long-read technology and other recent advances in sequencing technology were applied to generate a genome with far fewer gaps and helped to refine the locations and numbers of repetitive elements. Furthermore, the authors performed resequencing among populations to identify the genetic variability of the rhesus macaque. Thus, a previously incomplete and inaccurate set of sequence information is now fully resolved, improving gene mapping for biomedical and comparative genetic studies.;,citation_author=Wesley C. Warren;,citation_author=R. Alan Harris;,citation_author=Marina Haukness;,citation_author=Ian T. Fiddes;,citation_author=Shwetha C. Murali;,citation_author=Jason Fernandes;,citation_author=Philip C. Dishuck;,citation_author=Jessica M. Storer;,citation_author=Muthuswamy Raveendran;,citation_author=LaDeana W. Hillier;,citation_author=David Porubsky;,citation_author=Yafei Mao;,citation_author=David Gordon;,citation_author=Mitchell R. Vollger;,citation_author=Alexandra P. Lewis;,citation_author=Katherine M. Munson;,citation_author=Elizabeth DeVogelaere;,citation_author=Joel Armstrong;,citation_author=Mark Diekhans;,citation_author=Jerilyn A. Walker;,citation_author=Chad Tomlinson;,citation_author=Tina A. Graves-Lindsay;,citation_author=Milinn Kremitzki;,citation_author=Sofie R. Salama;,citation_author=Peter A. Audano;,citation_author=Merly Escalona;,citation_author=Nicholas W. Maurer;,citation_author=Francesca Antonacci;,citation_author=Ludovica Mercuri;,citation_author=Flavia A. M. Maggiolini;,citation_author=Claudia Rita Catacchio;,citation_author=Jason G. Underwood;,citation_author=David H. O’Connor;,citation_author=Ashley D. Sanders;,citation_author=Jan O. Korbel;,citation_author=Betsy Ferguson;,citation_author=H. Michael Kubisch;,citation_author=Louis Picker;,citation_author=Ned H. Kalin;,citation_author=Douglas Rosene;,citation_author=Jon Levine;,citation_author=David H. Abbott;,citation_author=Stanton B. Gray;,citation_author=Mar M. Sanchez;,citation_author=Zsofia A. Kovacs-Balint;,citation_author=Joseph W. Kemnitz;,citation_author=Sara M. Thomasy;,citation_author=Jeffrey A. Roberts;,citation_author=Erin L. Kinnally;,citation_author=John P. Capitanio;,citation_author=J. H. Pate Skene;,citation_author=Michael Platt;,citation_author=Shelley A. Cole;,citation_author=Richard E. Green;,citation_author=Mario Ventura;,citation_author=Roger W. Wiseman;,citation_author=Benedict Paten;,citation_author=Mark A. Batzer;,citation_author=Jeffrey Rogers;,citation_author=Evan E. Eichler;,citation_publication_date=2020-12;,citation_cover_date=2020-12;,citation_year=2020;,citation_issue=6523;,citation_doi=10.1126/science.abc6617;,citation_issn=0036-8075, 1095-9203;,citation_volume=370;,citation_journal_title=Science;">
</head>

<body>

<header id="title-block-header" class="quarto-title-block default toc-left page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Chromatin Compartments and Selection on X</h1>
            <p class="subtitle lead">How Edges of Active Chromatin Align with Selection Regions in Primates</p>
          </div>

    
    <div class="quarto-title-meta-container">
      <div class="quarto-title-meta-column-start">
        
        <div class="quarto-title-meta">

                <div>
            <div class="quarto-title-meta-heading">Author</div>
            <div class="quarto-title-meta-contents">
                        <p>Søren Jørgensen <a href="mailto:201906763@post.au.dk" class="quarto-title-author-email"><i class="bi bi-envelope"></i></a> </p>
                      </div>
          </div>
                
                <div>
            <div class="quarto-title-meta-heading">Published</div>
            <div class="quarto-title-meta-contents">
              <p class="date">January 6, 2025</p>
            </div>
          </div>
          
                
              </div>
      </div>
      <div class="quarto-title-meta-column-end quarto-other-formats-target">
      <div class="quarto-alternate-formats"><div class="quarto-title-meta-heading">Other Formats</div><div class="quarto-title-meta-contents"><p><a href="index.pdf"><i class="bi bi-file-pdf"></i>PDF</a></p></div></div></div>
    </div>

    <div>
      <div class="abstract">
        <div class="block-title">Abstract</div>
        <p>The hemizygosity of male mammals makes the X chromosome uniquely exposed to selective pressures, as the lack of a second copy provides no buffer against deleterious mutations. This, combined with a high density of essential genes related to reproduction and brain function, suggests the existence of biological mechanisms that protect the integrity of the X chromosome. One such mechanism is meiotic sex chromosome inactivation (MSCI) in males, while another could involve chromatin architecture.</p>
        <p>In this study, the 3D chromatin architecture of the X chromosome in rhesus macaque (<em>Macaca mulata</em>) is investigated in the context of evolutionary pressures and genetic drivers. We redo Hi-C analyses from a 2019 paper on the latest reference genome (<em>rheMac10</em> or <em>Mmul_10</em>). We compare two Hi-C analysis frameworks, <em>HiCExplorer</em> and <em>cooler/cooltools</em> (Open2C), on a subset, finding Open2C to be most flexible and intuitive. The ICE method (Iterative Correction and Eigendecomposition) was used to infer conventional and refined A/B compartments for fibroblast and four stages of spermatogenesis.</p>
        <p>We find that 200 kbp transition-zones between A/B-compartments in both fibroblasts and round spermatids align well with strong selective sweeps in humans (ECH-regions), and with strong negative selection in baboons (<em>Papio</em> spp.). We discuss the biological meaning of these findings, where conserved chromatin features may help to retain non-advantageous alleles, hinting to the role of selfish genetic elements in genome evolution.</p>
      </div>
    </div>

    <div>
      <div class="keywords">
        <div class="block-title">Keywords</div>
        <p>Hi-C, Chromatin Compartments, Selection on X, Selfish gene drive</p>
      </div>
    </div>

    <div class="quarto-other-links-text-target">
    </div>  </div>
</header><div id="quarto-content" class="page-columns page-rows-contents page-layout-article toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction">Introduction</a>
  <ul class="collapse">
  <li><a href="#sexual-reproduction-and-sex-chromosomes" id="toc-sexual-reproduction-and-sex-chromosomes" class="nav-link" data-scroll-target="#sexual-reproduction-and-sex-chromosomes">Sexual reproduction and Sex Chromosomes</a></li>
  <li><a href="#known-selection-on-x" id="toc-known-selection-on-x" class="nav-link" data-scroll-target="#known-selection-on-x">Known selection on X</a></li>
  <li><a href="#gene-drivers" id="toc-gene-drivers" class="nav-link" data-scroll-target="#gene-drivers">Gene drivers</a></li>
  <li><a href="#selfish-genes-and-randomness" id="toc-selfish-genes-and-randomness" class="nav-link" data-scroll-target="#selfish-genes-and-randomness">Selfish genes (and randomness)</a></li>
  <li><a href="#chromatin-architecture" id="toc-chromatin-architecture" class="nav-link" data-scroll-target="#chromatin-architecture">Chromatin Architecture</a></li>
  <li><a href="#c-chromatin-conformation-capture" id="toc-c-chromatin-conformation-capture" class="nav-link" data-scroll-target="#c-chromatin-conformation-capture">3C: Chromatin Conformation Capture</a></li>
  <li><a href="#hi-c-high-throughput-3c" id="toc-hi-c-high-throughput-3c" class="nav-link" data-scroll-target="#hi-c-high-throughput-3c">Hi-C: High-Throughput 3C</a>
  <ul class="collapse">
  <li><a href="#hi-c-library-preparation" id="toc-hi-c-library-preparation" class="nav-link" data-scroll-target="#hi-c-library-preparation">Hi-C Library preparation</a></li>
  <li><a href="#hi-c-data-analysis" id="toc-hi-c-data-analysis" class="nav-link" data-scroll-target="#hi-c-data-analysis">Hi-C Data Analysis</a></li>
  </ul></li>
  <li><a href="#reproducibility-infrastructure" id="toc-reproducibility-infrastructure" class="nav-link" data-scroll-target="#reproducibility-infrastructure">Reproducibility Infrastructure</a>
  <ul class="collapse">
  <li><a href="#gwf-workflow-management-for-high-performance-computing-hpc" id="toc-gwf-workflow-management-for-high-performance-computing-hpc" class="nav-link" data-scroll-target="#gwf-workflow-management-for-high-performance-computing-hpc">GWF: workflow management for High-Performance Computing (HPC)</a></li>
  <li><a href="#project-initialization" id="toc-project-initialization" class="nav-link" data-scroll-target="#project-initialization">Project Initialization</a></li>
  <li><a href="#git-and-github" id="toc-git-and-github" class="nav-link" data-scroll-target="#git-and-github">git and GitHub</a></li>
  </ul></li>
  <li><a href="#our-research-question" id="toc-our-research-question" class="nav-link" data-scroll-target="#our-research-question">Our research question</a></li>
  </ul></li>
  <li><a href="#methods" id="toc-methods" class="nav-link" data-scroll-target="#methods">Methods</a>
  <ul class="collapse">
  <li><a href="#fetching-raw-data" id="toc-fetching-raw-data" class="nav-link" data-scroll-target="#fetching-raw-data">Fetching raw data</a>
  <ul class="collapse">
  <li><a href="#fetching-and-indexing-the-reference" id="toc-fetching-and-indexing-the-reference" class="nav-link" data-scroll-target="#fetching-and-indexing-the-reference">Fetching and indexing the reference</a></li>
  </ul></li>
  <li><a href="#hicexplorer-trials" id="toc-hicexplorer-trials" class="nav-link" data-scroll-target="#hicexplorer-trials">HiCExplorer trials</a></li>
  <li><a href="#open2c-pipeline" id="toc-open2c-pipeline" class="nav-link" data-scroll-target="#open2c-pipeline">Open2C pipeline</a>
  <ul class="collapse">
  <li><a href="#notebook-edits" id="toc-notebook-edits" class="nav-link" data-scroll-target="#notebook-edits">Notebook edits</a></li>
  <li><a href="#compartments-and-their-edges-transitional-regions" id="toc-compartments-and-their-edges-transitional-regions" class="nav-link" data-scroll-target="#compartments-and-their-edges-transitional-regions">Compartments and Their Edges (Transitional Regions)</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#results" id="toc-results" class="nav-link" data-scroll-target="#results">Results</a>
  <ul class="collapse">
  <li><a href="#exploration-with-hicexplorer" id="toc-exploration-with-hicexplorer" class="nav-link" data-scroll-target="#exploration-with-hicexplorer">Exploration with HicExplorer</a>
  <ul class="collapse">
  <li><a href="#quality-control" id="toc-quality-control" class="nav-link" data-scroll-target="#quality-control">Quality Control</a></li>
  <li><a href="#correction" id="toc-correction" class="nav-link" data-scroll-target="#correction">Correction</a></li>
  <li><a href="#eigenvectors" id="toc-eigenvectors" class="nav-link" data-scroll-target="#eigenvectors">Eigenvectors</a></li>
  </ul></li>
  <li><a href="#open2c-ecosystem" id="toc-open2c-ecosystem" class="nav-link" data-scroll-target="#open2c-ecosystem">Open2c ecosystem</a>
  <ul class="collapse">
  <li><a href="#quality-control-1" id="toc-quality-control-1" class="nav-link" data-scroll-target="#quality-control-1">Quality Control</a></li>
  <li><a href="#correction-1" id="toc-correction-1" class="nav-link" data-scroll-target="#correction-1">Correction</a></li>
  <li><a href="#sec-results-eigenvectors" id="toc-sec-results-eigenvectors" class="nav-link" data-scroll-target="#sec-results-eigenvectors">Compartments (Eigenvectors)</a></li>
  <li><a href="#compartment-edges-transition-zones" id="toc-compartment-edges-transition-zones" class="nav-link" data-scroll-target="#compartment-edges-transition-zones">Compartment Edges (transition zones)</a></li>
  </ul></li>
  <li><a href="#testing-against-hybrid-incompatibility-in-baboons" id="toc-testing-against-hybrid-incompatibility-in-baboons" class="nav-link" data-scroll-target="#testing-against-hybrid-incompatibility-in-baboons">Testing against hybrid incompatibility in baboons</a></li>
  </ul></li>
  <li><a href="#discussion" id="toc-discussion" class="nav-link" data-scroll-target="#discussion">Discussion</a></li>
  <li><a href="#bibliography" id="toc-bibliography" class="nav-link" data-scroll-target="#bibliography">Bibliography</a></li>
  </ul>
</nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
</div>
<main class="content quarto-banner-title-block" id="quarto-document-content">



  


<section id="introduction" class="level1">
<h1>Introduction</h1>
<section id="sexual-reproduction-and-sex-chromosomes" class="level2">
<h2 class="anchored" data-anchor-id="sexual-reproduction-and-sex-chromosomes">Sexual reproduction and Sex Chromosomes</h2>
<p>The production of gametes in a sexually reproducing organism is a highly complex process that involves numeruous elements. Spermatogenesis, the process of forming male gametes, involves four stages of differentiation from a germ cell through <em>spermatogonia</em>, <em>pachytene spermatocyte</em>, and <em>round spermatids</em> to <em>spermatozoa</em>, or <em>sperm</em> <span class="citation" data-cites="wang2019reprogrammingmeioticchromatin">(<a href="#ref-wang2019reprogrammingmeioticchromatin" role="doc-biblioref">Wang et al. 2019</a>)</span>, and it is the very basis of male reproduction. The specialized cell division of meiosis neatly handles the pairing, recombination, and segregation of homologous chromosomes, thereby ensuring proper genetic distribution. Deeply understanding the steps of molecular steps of reproduction and how our genetic material is inherited is essential in biology, bringing insight to areas such as speciation, population diversity, and (male) infertility.</p>
<p>Sex chromosomes behave differently than autosomes for a long list of reasons. One reason is hemizygosity, where there is only one copy of a chromosome. The Y chromosome exist only in males, and (generally) never more than one copy is in the same individual. The copmlexity increases when we tend to the X chromosome, which exist both alone (in males) and as two copies (in females). This skewed ratio means that X chromosomes exist <span class="math inline">\(2/3\)</span> of the time in females and only 1/3 of the time in males, and it is more exposed in males as there is no other copy to take over loss of function. It is also the underlying reasoning for Haldane’s rule [ref], postulating that the heterogametic sex (XY, ZW) is the first to dissapear or have severely lowered fitness (e.g.&nbsp;sterility) when crossing different species. Even today, a hundred years later, there is only hypotheses as to why this is observed, including <em>Y-incompatibility</em> (Y has to be compatible to X or autosomes), <em>dosage compensation</em> (hybridisation deregulates crucial dosage compensation in heterogametes), <em>dominance</em> (recessive genes causes sterility), <em>faster-male</em> (male reproductive genes diverge faster than female), <em>faster-X</em> (X-linked loci diverge faster than autosomal ones), and <em>meiotic drive</em> (discrepancy between drivers and surpressors on sex chromosomes leads to sterility) [ref]. While is appears to be the same phenomenon across multiple <em>taxa</em> (even <em>kingdoms</em>), it has received different explanations, and consensus for different species are not the same. Often times even, several of the listed explanations are described to be acting in collaboration [ref].</p>
<ul class="task-list">
<li><label><input type="checkbox">include something about conserved synteny on chrX here</label></li>
</ul>
<p>The complexity of selection on the X chromosome therefore still an area of active investigation, and several studies infers selective strong selection on the X chromosomes across primates [ref]</p>
</section>
<section id="known-selection-on-x" class="level2">
<h2 class="anchored" data-anchor-id="known-selection-on-x">Known selection on X</h2>
<ul class="task-list">
<li><label><input type="checkbox">Selective sweeps (or is it?) in humans (ECH90; <span class="citation" data-cites="skov2023extraordinaryselectionhuman">Skov et al. (<a href="#ref-skov2023extraordinaryselectionhuman" role="doc-biblioref">2023</a>)</span>)</label>
<ul class="task-list">
<li><label><input type="checkbox">Explain why it makes sense to compare these data sets</label></li>
</ul></li>
<li><label><input type="checkbox">Negative selection in baboons w.r.t. minor parent ancestry</label>
<ul class="task-list">
<li><label><input type="checkbox">Explain why it makes sense to compare these datasets</label></li>
<li><label><input type="checkbox">The low-diveristy in <em>p.hamadryas</em> are very wide - maybe chromatin</label></li>
</ul></li>
</ul>
</section>
<section id="gene-drivers" class="level2">
<h2 class="anchored" data-anchor-id="gene-drivers">Gene drivers</h2>
<ul class="task-list">
<li><label><input type="checkbox">Talk about what makes a genetic driver, what criteria</label></li>
</ul>
<p>Gene drive occur when a particular collection of genes is propagated through a population by increasing the probability of transmitting the genes to the offspring from random (Mendelian) inheritance, resulting in a biased gene transmission against its alternative. Two categories of gene drivers exist <span class="citation" data-cites="bravonunez2018geneticvillainskiller">(<a href="#ref-bravonunez2018geneticvillainskiller" role="doc-biblioref">Bravo Núñez, Nuckolls, and Zanders 2018</a>)</span>; <em>class one drivers</em> affect chromosome segregation in meiosis, and <em>killer meiotic drivers</em> will sabotage meiotic product that have not inherited the driving allele [ref]. This happens regardless of the fitness effects of the developed organism, and is there often a factor offsetting classical selection. Even though the implications of such systems are potentially detrimental, they are notoriously difficult to detect. A circumstance contributing to the difficulty is that most genetic experiments are done in homozygotes. <span class="citation" data-cites="bravonunez2018geneticvillainskiller">Bravo Núñez, Nuckolls, and Zanders (<a href="#ref-bravonunez2018geneticvillainskiller" role="doc-biblioref">2018</a>)</span> states that the general choice of experimental system may have biased our understanding of sexual reproduction. A key point is; meiotic drivers can only be observed in heterozygotes, where a genetic driver has a competitor.</p>
<p>Interestingly, gene drive is much more well-documented on sex chromosomes than for autosomes. Possibly because the sex chromosome meiotic drive inherently causes a skewed sex ratio, more notably raising a flag for further analysis. Another point to consider is, in the case of sex chromosome drive, a fully driving gene will lead to the extinction of the species [ref], as only one (fertile) offspring will be produced.</p>
</section>
<section id="selfish-genes-and-randomness" class="level2">
<h2 class="anchored" data-anchor-id="selfish-genes-and-randomness">Selfish genes (and randomness)</h2>
<p><strong>The conventional story of meiosis in gametogenesis is one of random segregation of the sex chromosomes. They split into haploid gametes, where each chromosome has an equal chance of being passed on to a gamete. That seems like a fair game, but what if some genes are cheating the system by making others less viable. A meiotic driver is a selfish gene element that modulates meiosis and preferentially transmits its own allele through meiosis, regardless of the downstream fitness effects it may have (good or bad) on the organism it is part of. This phenomenon challenges the traditional understanding of selection, extending its scope beyond the fitness effects on an organism to include selective pressures at the molecular level. For example, if some genes on the X chromosome create a disadvantage for gametes that <em>do not</em> contain those genes, making sure the Y chromosome is not as viable as the X, resulting in a sex imbalance and possibly numerous other downstream effects. That is exactly what is coined <em>sex chromosome meiotic drive</em><span class="citation" data-cites="jaenike2001sexchromosomemeiotic">(<a href="#ref-jaenike2001sexchromosomemeiotic" role="doc-biblioref">Jaenike 2001</a>)</span>, a result of selfish genetic elements. Motivated by previous results in the Munch Research group <span class="citation" data-cites="munch2024munchgroup">(<a href="#ref-munch2024munchgroup" role="doc-biblioref">Munch 2024</a>)</span> on hybrid incompatibility and extended common haplotypes <span class="citation" data-cites="skov2023extraordinaryselectionhuman sorensen2023genomewidecoancestryreveals">(<a href="#ref-skov2023extraordinaryselectionhuman" role="doc-biblioref">Skov et al. 2023</a>; <a href="#ref-sorensen2023genomewidecoancestryreveals" role="doc-biblioref">Sørensen et al. 2023</a>)</span> that could be explained by meiotic drive, we wanted to investigate how these patterns correlate with chromatin compartments.</strong></p>
</section>
<section id="chromatin-architecture" class="level2">
<h2 class="anchored" data-anchor-id="chromatin-architecture">Chromatin Architecture</h2>
<ul>
<li><label><input type="checkbox">Chromatin organization is highly conserved between species</label></li>
<li><label><input type="checkbox">Explain why take offset in <span class="citation" data-cites="wang2019reprogrammingmeioticchromatin">Wang et al. (<a href="#ref-wang2019reprogrammingmeioticchromatin" role="doc-biblioref">2019</a>)</span></label></li>
<li><label><input type="checkbox">Exlpain the rationale of reproducing results with the latest reference</label></li>
<li>[ ]</li>
</ul>
</section>
<section id="c-chromatin-conformation-capture" class="level2">
<h2 class="anchored" data-anchor-id="c-chromatin-conformation-capture">3C: Chromatin Conformation Capture</h2>
<ul class="task-list">
<li><label><input type="checkbox">This might not be needed or will be merged with Hi-C section</label></li>
</ul>
</section>
<section id="hi-c-high-throughput-3c" class="level2">
<h2 class="anchored" data-anchor-id="hi-c-high-throughput-3c">Hi-C: High-Throughput 3C</h2>
<p>Our DNA can be divided into different orders of structure. <em>3C</em> focus on identifying the highest orders of organization inside the nucleus, that is, when the 30 nm thick coil of chromatin fibers folds into loops, Topologically Associating Domains (TADs), and chromatin compartments. Here, we narrow our focus on the largest of the structures, <em>compartments</em>, that is known to determine availability to transcription factors, thus making an <em>A</em> compartment <em>active</em>—and the <em>B</em> compartment <em>inactive</em>. The introduction of the Hi-C (high-throughput 3C) method <span class="citation" data-cites="lieberman-aiden2009comprehensivemappinglongrange">(<a href="#ref-lieberman-aiden2009comprehensivemappinglongrange" role="doc-biblioref">Lieberman-Aiden et al. 2009</a>)</span> opened new possibilities for exploring the three-dimensional organization of the genome.</p>
<section id="hi-c-library-preparation" class="level3">
<h3 class="anchored" data-anchor-id="hi-c-library-preparation">Hi-C Library preparation</h3>
<p>A specialized protocol for preparing the DNA library is necessary <span class="citation" data-cites="lieberman-aiden2009comprehensivemappinglongrange">(<a href="#ref-lieberman-aiden2009comprehensivemappinglongrange" role="doc-biblioref">Lieberman-Aiden et al. 2009, fig. 1a</a>)</span>. Briefly, formaldehyde is used to crosslink spatially adjecent chromatin. Restriction enzyme <em>HindIII</em> is used to digest the crosslinked chromatin, leaving sticky ends, <code>5-AGCT-3</code>, that are filled and biotinylated with a polymerase (using either biotinylated A, G, C, or T). The strands are ligated in highly dilute conditions, which is favoring the ligation of the two crosslinked strands, forming chimeric, biotinylated strands. Upon ligation, the restriction site is lost as a biotinylated <code>5-CTAG-3</code> site (also referred to as the <em>ligation junction</em>) is formed. Lastly, the ligation junctions are isolated with streptavidin beads and sequenced as a paired-end library.</p>
<p>To be able to create stage-resolved Hi-C library of spermatogenetis, several steps have to be performed on the samples before crosslinking. First, the samples have to be treated immediately after harvesting to ensure viable cells. Secondly, the samples have to be purified to accurately represent each stage og spermatogenesis. Specifically, the data for this project <span class="citation" data-cites="wang2019reprogrammingmeioticchromatin">(<a href="#ref-wang2019reprogrammingmeioticchromatin" role="doc-biblioref">Wang et al. 2019</a>, acc. GSE109344)</span> preluded the library preparation protocol by sedimentation-based cell sorting to separate live spermatogenic cells into different stages of differentiation, namely spermatogonia, pachytene spermatocyte, round spermatid, and spermatozoa. Then, the cells were fixed in their respective state before crosslinking. They use their own derived method for library preparation, termed small-scale <em>in-situ</em> Hi-C, allegedly producing a high-quality Hi-C libary from as little as 500 cells (capturing the variance of millions of cells).</p>
</section>
<section id="hi-c-data-analysis" class="level3">
<h3 class="anchored" data-anchor-id="hi-c-data-analysis">Hi-C Data Analysis</h3>
<p>The analysis of the read-pairs of a Hi-C library is divided into several smaller tasks, see [ref fig-hic-analysis-flow].</p>
<div id="fig-hic-analysis-flow" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-hic-analysis-flow-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="illustrations/fig-hic-data-analysis.png" class="lightbox" data-gallery="quarto-lightbox-gallery-1" title="Figure&nbsp;1: A simplified pipeline for Hi-C data analysis: from raw reads to a Hi-C interaction matrix. See Table&nbsp;1 for details on ligation events."><img src="illustrations/fig-hic-data-analysis.png" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-hic-analysis-flow-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1: A simplified pipeline for Hi-C data analysis: from raw reads to a Hi-C interaction matrix. See <a href="#tbl-ligation-events" class="quarto-xref">Table&nbsp;1</a> for details on ligation events.
</figcaption>
</figure>
</div>
<p>We must align the reads to the reference in such a way that the <em>intentional</em> chimeric read-pairs (as per above-mentioned protocol) are rescued, and <em>unintentional</em> read-pairs are discarded. That is, we must make sure they represent ligation junction of adjecent chromatin segments, and not technical artefacts or unintentional (random) fusions of unrelated DNA.</p>
<section id="aligning-the-hi-c-reads" class="level4">
<h4 class="anchored" data-anchor-id="aligning-the-hi-c-reads">Aligning the Hi-C reads</h4>
<p>The main difference between Hi-C libraries and standard paired-end libraries is the high fraction of chimeric reads in Hi-C. As a contact pair is crosslinked and ligated before sequencing, chimeric reads occur as a feature, and standard mapping techniques seeks to filter out this type of reads [ref]. Thus, we need specialized tools for rescuing chimeric reads. That said, we have to be cautious distinguishing the intended chimerism for Hi-C and that of technical artefacts. Any software for local alignment can be used for aligning reads from a Hi-C library. However, one should make sure to disable paired-end rescue mode if possible, otherwise each read in a pair (each mate) should be aligned separately <span class="citation" data-cites="lajoie2015hitchhikersguidehic">(<a href="#ref-lajoie2015hitchhikersguidehic" role="doc-biblioref">Lajoie, Dekker, and Kaplan 2015</a>)</span>. This removes the assumption that the distance between mates fits a known distribution because the genomic sequences originate from a continuous DNA-fragment. For example, the <em>bwa-mem</em> [ref] implementation of this (the <code>-P</code> option) activates the Smith-Waterman algorithm to rescue missing hits, but disables the search of hits that fit a ‘proper’ pair. After alignment, each read is typically assigned to the nearest restriction fragment to enable categorization of pairs into different categories.</p>
<p>Interestingly, this last step is not included by default in <em>pairtools</em>, as [ref] observe very similar statistical properties on pairs that are either close or distant from the nearest restriction site. Thus, restriction fragment filters are not needed, and in stead, a simple filter is applied against short-distance pairs that is automatically calibrated.</p>
</section>
<section id="identifying-and-storing-valid-hi-c-pairs" class="level4">
<h4 class="anchored" data-anchor-id="identifying-and-storing-valid-hi-c-pairs">Identifying and Storing Valid Hi-C Pairs</h4>
<p>One should be cautious when filtering invalid from valid pairs, as they are not easily distinguished. A ligation event will be categorized into one of five categories (see <a href="#tbl-ligation-events" class="quarto-xref">Table&nbsp;1</a>): <em>dangling-end</em>, <em>self-circle</em>, <em>weird</em>, <em>intrachromosomal</em> (cis), and <em>interchromosomal</em> (trans) <span class="citation" data-cites="bicciato2022hicdataanalysis">(<a href="#ref-bicciato2022hicdataanalysis" role="doc-biblioref">Bicciato and Ferrari 2022</a>, Ch. 1)</span>. Either <em>dangling-end</em> or <em>self-circle</em> events are reported if a read-pair maps to the same restriction fragment depending on the orientation, and deemed uninformative <span class="citation" data-cites="lajoie2015hitchhikersguidehic">(<a href="#ref-lajoie2015hitchhikersguidehic" role="doc-biblioref">Lajoie, Dekker, and Kaplan 2015</a>)</span>. Usually, <em>weird</em> events are demeed uninformative as well, as it is challenging to distinguish a sequencing error from the result of a diploid fragment.</p>
<p>PCR duplicates should be discarded as well, having either identical genomic sequence, or sharing exact 5’ alignment positions of the pair <span class="citation" data-cites="lajoie2015hitchhikersguidehic bicciato2022hicdataanalysis">(<a href="#ref-lajoie2015hitchhikersguidehic" role="doc-biblioref">Lajoie, Dekker, and Kaplan 2015</a>; <a href="#ref-bicciato2022hicdataanalysis" role="doc-biblioref">Bicciato and Ferrari 2022</a>, Ch. 1)</span>. The probability that such pairs are valid (i.e.&nbsp;there are multiple of the same pairs) is very low.</p>
<p>The <em>trans</em>/<em>cis</em>-ratio can be a good indicator of the noise level in the library, and additionally, the level of random ligation events can be quantified by counting the number of <em>trans</em> events occurring to mitochondrial genome. They should not occur naturally, as the mitochondrial genome is separated from the DNA in the nucleus. This method has some pitfalls that should be controlled for; some parts of the mitochondrial genome can be integrated into the host genome, and mitochondrial count may differ between states.</p>
<div id="tbl-ligation-events" class="striped quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-ligation-events-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;1: Five categories of ligation events and a short explanation. <em>Hi-C Data Analysis: Methods and Protocols Ch. 1</em>.
</figcaption>
<div aria-describedby="tbl-ligation-events-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<table class="table-striped caption-top table">
<colgroup>
<col style="width: 20%">
<col style="width: 79%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Event name</th>
<th style="text-align: left;">Explanation</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Dangling-end</td>
<td style="text-align: left;">Non-digested collinear fragments. Fraction can be high.</td>
</tr>
<tr class="even">
<td style="text-align: left;">Self-circle</td>
<td style="text-align: left;">Collinear fragment(s) have circularized. Very low fraction could indicate unsuccesful ligation.</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Weird</td>
<td style="text-align: left;">Mates have the same orientation on the reference. Is not possible with single copy fragment. Either sequencing errors or diploid fragments<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a>.</td>
</tr>
<tr class="even">
<td style="text-align: left;">Cis</td>
<td style="text-align: left;">Pairs from the same chromosome (intrachromosomal)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Trans</td>
<td style="text-align: left;">Pairs from distinct chromosomes (interchromosomal)</td>
</tr>
</tbody>
</table>
</div>
</figure>
</div>
</section>
<section id="sec-matrix-qc" class="level4">
<h4 class="anchored" data-anchor-id="sec-matrix-qc">Quality Control and Interaction Matrices</h4>
<p>To determine the quality of the Hi-C library, most tools generate quality control log files at some point during the filtering steps, which can then be aggregated and analyzed (with e.g.&nbsp;MultiQC [ref]). The ratios between the different ligation events can be informative about the quality of the Hi-C library. Here, both the distribution of discarded reads across categories, as well as the ratios between <em>cis</em>/<em>trans</em> interactions for a certain organism provide information about the library. For example, the biases of different aligners might be captured by comparing the reason why reads are discarded between two different aligners, as well as if there is a preference of <em>cis</em> or <em>trans</em> in an aligner itself. This allows for evaluating the mapping parameters as well as the filters applied downstream. Additionally, <span class="math inline">\(P(s)\)</span>, the contact probability as a function of genomic separation can be inspected as it should decay with increasing distance.</p>
<p>Typically, a filter against low mapping quality is applied on the data before constructing the interaction matrix (Hi-C matrix), and a conventional threshold is <span class="math inline">\(mapq &lt; 30\)</span> <span class="citation" data-cites="bicciato2022hicdataanalysis">(<a href="#ref-bicciato2022hicdataanalysis" role="doc-biblioref">Bicciato and Ferrari 2022</a>)</span>. However, a considerable amount of reads do not pass that threshold, and thus we risk discarding potential valid information and should make sure to have enough data. Consequently, <em>HicExplorer</em> defaults a lower threshold (<span class="math inline">\(mapq &lt; 15\)</span>), and <em>pairtools</em> enforces no filter by default, but recommends setting this manually (starting at <span class="math inline">\(mapq &lt; 30\)</span>).</p>
<p>A Hi-C interaction matrix simply maps the frequency of interactions between genomic positions in a sample. The maximum resolution of a Hi-C matrix is defined by the restriction enzyme, where the size of the restriction site (probabilistically) determines average space between each cut. With a 4 bp restriction site, the fragments will average <span class="math inline">\(4^4 = 256 bp\)</span> and similarly <span class="math inline">\(4^6 = 4096 bp\)</span> for a 6 bp restriction site. This leads to ~12,000,000 and ~800,000 fragments, respectively. Very deep sequencing is required to achieve enough coverage to analyze the interaction matrix at the restriction fragment resolution, but, usually, such high resolution is not required. Therefore, it is practice to bin the genome into fixed bin sizes, which also enables a more efficient handling of the data if the full resolution is not needed (e.g.&nbsp;when plotting large regions such as a whole chromosome). The conventional format to store a Hi-C matrix, consisting of large multidimensional arrays, is HDF5. Each HDF5 file can store all resolutions and metadata about the sample, resolutions typically ranging from 10kb to 1Mb. Typically, the stored resolutions should be multiples of the chosen base-resolution, as the lower resolutions are constructed by recursive binning of the base resolution. <em>cooler</em> [ref] neatly offers efficient storage with sparse, upper-triangle symmetric matrices and naming-conventions of the groups in their <em>.h5</em>-based file format, <em>.cool</em>, and they provide a Python class <code>Cooler</code> as well for efficiently fetching and manipulating the matrices in Python.</p>
</section>
<section id="inferring-from-the-matrix-calling-compartments" class="level4">
<h4 class="anchored" data-anchor-id="inferring-from-the-matrix-calling-compartments">Inferring from the matrix (Calling Compartments)</h4>
<p>The raw frequency matrices are generally not very informative, as the contact frequencies vary greatly between bins and contain biases in addition to the <span class="math inline">\(P(s)\)</span> decay, which results in a diagonal-heavy matrix with high amount of noise the further we travel from the diagonal. Therefore, to analyze the three-dimensional structure of the chromatin, a method for correcting (or balancing) the raw Hi-C matrix has to be applied. It is unadvisable to correct low-count bins as it will greatly increase the noise, or to correct very noisy bins, or very high-count bins. Therefore, some bin-level filters are applied before balancing <span class="citation" data-cites="lajoie2015hitchhikersguidehic">(<a href="#ref-lajoie2015hitchhikersguidehic" role="doc-biblioref">Lajoie, Dekker, and Kaplan 2015</a>)</span>;</p>
<ul>
<li>Low-count bins are detected by comparing bin sums to the distribution of bin sums with a percentile cutoff,</li>
<li>Noisy bins are detected by comparing bin variance to the variance distribution of all bins (and percentile cutoff), and</li>
<li>Outlier point-interactions are removed (a top-percentile of bin-bin interactions)</li>
</ul>
<p>A widely used balancing method is Iterative Correction and Eigendecomposition (ICE) <span class="citation" data-cites="imakaev2012iterativecorrectionhic">(<a href="#ref-imakaev2012iterativecorrectionhic" role="doc-biblioref">Imakaev et al. 2012</a>)</span>, which utilizes a data-driven approach for correcting multiplicative biases. Briefly, is based on an assumption of equal visibility of all loci, and uses the pairwise and genome-wide structure to generate a set of biases along with a map of relative interaction frequencies by iteratively dividing each row, then each column, by its mean until convergence. This results in a uniform coverage profile (corrected coverage), yielding a smoother interaction matrix with slower transitions, thus greatly reduces visibility-induced biases. It does not distinguish between the sources of biases, and thus calculates a collective bias for each position. <span class="citation" data-cites="imakaev2012iterativecorrectionhic">Imakaev et al. (<a href="#ref-imakaev2012iterativecorrectionhic" role="doc-biblioref">2012</a>)</span> show that <em>known</em> biases are factorizable by comparing their results to predictions of restriction fragment biases, GC content, and mappability from a computationally intensive probabilistic approach. By showing that the product of those known biases explain <span class="math inline">\(&gt;99.99%\)</span> of the variability in their bias estimation, they argue both known and unknown biases will be captured with their iterative correction method (also denoted <em>matrix balancing</em>).</p>
<p>Even with a binned, filtered, and balanced matrix, we are still left with the challenge of translating the matrix into biologically relevant inferations. Importantly, we have to remember that the matrix arise from a collection of cells and that the interaction frequency cannot be translated to a fraction of cells. Additionally, the effect from averaging interaction patterns can cause both individual patterns to be burried and the average pattern to show a pattern that does not exist in any of the single cells. Therefore, when pooling matrices one must make sure that the samples are as similar as possible (e.g.&nbsp;the same differentiation stage and so on). We can also not distinguish interactions that either co-occur in the same cell or ones that are mutually exclusive. Lastly, the way interaction patterns are defined poses a challenge; we define the chromatin compartments to be the output of a method, the ‘E’ in ‘ICE’, eigendecomposition, not as a specific pattern that we can explicitly search for. Although experimentally verified to tightly correlate with chromatin states, the inferred compartments vary with different methods of calculating the eigenvector, as dicussed in <a href="#sec-methods-eigendecomposition" class="quarto-xref">Section&nbsp;2.3.1.4</a> and <a href="#sec-results-eigenvectors" class="quarto-xref">Section&nbsp;3.2.3</a>. To further complicate the challenge, interaction patterns on different scales co-exist and are difficult to disentangle without simplifying assumptions such as small-scale interactions are not visible (or they are negligible) at a certain resolution, or restricting the viewframe to eliminate large-scale variance between chromosome arms. It is by definition a speculative exercise to interpret the biological relevance of an observed pattern, but the consensus is to call compartments on interacting regions that arise from the eigendecomposition of a Hi-C matrix without further modifications <span class="citation" data-cites="lajoie2015hitchhikersguidehic">(<a href="#ref-lajoie2015hitchhikersguidehic" role="doc-biblioref">Lajoie, Dekker, and Kaplan 2015</a>)</span>. As the eigenvector is only unique up to a sign, a phasing track is used to orient the eigenvector, aiming for a positive correlation with GC content (in mammals), making A-compartments represent the active euchromatin, and B-compartments the closed heterochromatin.</p>
</section>
<section id="compartment-edges-and-genomic-intervals" class="level4">
<h4 class="anchored" data-anchor-id="compartment-edges-and-genomic-intervals">Compartment Edges and Genomic Intervals</h4>
<p>As arbitrary as a compartment may be defined, we chose to define another genomic interval for analysis. It is well <span class="citation" data-cites="bicciato2022hicdataanalysis">(<a href="#ref-bicciato2022hicdataanalysis" role="doc-biblioref">Bicciato and Ferrari 2022</a>, Ch. 3)</span> known that CTCF and other structural proteins preferentially binds to Topologically Associating Domains (TADs; they were initially defined as sub-Mb chromatin structures <span class="citation" data-cites="lajoie2015hitchhikersguidehic">(<a href="#ref-lajoie2015hitchhikersguidehic" role="doc-biblioref">Lajoie, Dekker, and Kaplan 2015</a>)</span>, but now the definition seems to vary based on the method of extraction [ref cooltools]). Derived from this, we define a transition zone between A/B compartments to look for enrichment of specific regions of interest.</p>
<p>We can test if two sets of genomic intervals correlate (say, compartment edges and ECH regions) by either proximity of the non-intersecting parts of the sets, or by intersection over union (Jaccard index). When the underlying distribution of a statistic (or index) is unknown, a widespread method in bioinformatics for estimating a p-value is by bootstrapping. Here, one of the sets are bootstrapped (the intervals are placed at random positions) a number of times, <span class="math inline">\(b\)</span>, and the fraction of statistics more extreme than the one we observe is reported as the p-value.</p>
</section>
</section>
</section>
<section id="reproducibility-infrastructure" class="level2">
<h2 class="anchored" data-anchor-id="reproducibility-infrastructure">Reproducibility Infrastructure</h2>
<p><strong>First, describe the importance of reproducibility and the scientific method. Include a bit of scientific skeptiscism of there is time.</strong></p>
<p>Thus, apart from the biological questions we seek to investigate and answer in this thesis, a major goal of the thesis is to create fully (and easily) reproducible results through a self-contained and version-controlled pipeline using git [ref], GitHub [ref], quarto [ref], Conda [ref], gwf [ref], and Jupyter [ref]. See <a href="#tbl-reproducibility" class="quarto-xref">Table&nbsp;2</a> for a brief introduction.</p>
<div id="tbl-reproducibility" class="striped quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-reproducibility-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;2: Overview of the tools used for reproducibility of this thesis.
</figcaption>
<div aria-describedby="tbl-reproducibility-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<table class="table-striped caption-top table">
<thead>
<tr class="header">
<th style="text-align: left;">Tool</th>
<th style="text-align: left;">Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Jupyter</td>
<td style="text-align: left;">Interactive coding environment for analysis and development (notebooks are natively rendered with Quarto)</td>
</tr>
<tr class="even">
<td style="text-align: left;">Quarto</td>
<td style="text-align: left;">A Quarto Manuscript project nested inside a Quarto Book for rendering html (website) and PDF (manuscript) from Markdown via Pandoc. Supports direct embedding of output from Jupyter Notebook cells (plots, tables).</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Conda</td>
<td style="text-align: left;">For managing software requirements and dependency versions reproducibly.</td>
</tr>
<tr class="even">
<td style="text-align: left;">git</td>
<td style="text-align: left;">Version control and <code>gh-pages</code> branch for automated render of Quarto project</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GitHub</td>
<td style="text-align: left;">Action was triggered <code>on push</code> to render the project and host on <a href="https://munch-group.org/hic-spermatogenesis/">munch-group.org</a></td>
</tr>
<tr class="even">
<td style="text-align: left;"><em>gwf</em></td>
<td style="text-align: left;">Workflow manager to automate the analysis on a HPC cluster, wrapped in Python code. <code>workflow.py</code> currently does everything from <code>.fastq</code> to <code>.cool</code>, but notebooks can be set to run sequentially as part of the workflow as well.</td>
</tr>
</tbody>
</table>
</div>
</figure>
</div>
<section id="gwf-workflow-management-for-high-performance-computing-hpc" class="level3">
<h3 class="anchored" data-anchor-id="gwf-workflow-management-for-high-performance-computing-hpc">GWF: workflow management for High-Performance Computing (HPC)</h3>
<p>To enable consistently reproducing the analyses, a workflow manager is used. Several exist, but the most well-known must be <em>snakemake</em> [ref]. However, we use the pragmatic (their own words), lightweight workflow manager <em>gwf</em>, which is optimized for the GenomeDK insfrastructure, and has the benefit of in-house support.</p>
<p>Briefly, <em>gwf</em> works on a python script, conventionally <code>workflow.py</code>, that wraps all the jobs (<em>targets</em> in gwf lingo) you will submit to the HPC cluster. Each target is submitted from a template, written as a Python function, which includes <code>inputs</code> and <code>outputs</code> that <em>gwf</em> should look for when building the depency graph, <code>options</code> list of resources that is forwarded to the queueing system (<em>Slurm</em> in our case), and <code>specs</code>, specifying the submission code in Bash as a formatted Python string (meaning we can pass Python variables to the submission code), providing an extremely flexible framework for running large and intensive analyses in a high-performance computing environment.</p>
</section>
<section id="project-initialization" class="level3">
<h3 class="anchored" data-anchor-id="project-initialization">Project Initialization</h3>
<section id="gwf" class="level4">
<h4 class="anchored" data-anchor-id="gwf"><em>gwf</em></h4>
<p>The initialization of the project directory is the basis of reproducibility and transparency, together with <code>workflow.py</code> inhabiting the main directory. Specifically, it includes a subdirectory for (intermediate) files that are produced by the pipeline, <code>steps/</code>. Everything in this directory is reproducible simply by re-running the <em>gwf</em>-workflow. It is thus not tracked by <code>git</code>, as the large files (raw reads, aligned read-pairs, etc.) it contains are already indirectly tracked (<code>workflow.py</code> is tracked). It can be safely deleted if your system administrator tells you to free up disk space, although you would have to run the workflow again to continue the analysis. Several directories are created for files that are not produced by the pipeline, that is, files that the workflow uses, configuration files, figures edited by hand, etc. Ideally, as few files as possible should be outside of <code>steps/</code>, to be as close as possible to an automated analysis.</p>
</section>
<section id="jupyter-notebooks" class="level4">
<h4 class="anchored" data-anchor-id="jupyter-notebooks">Jupyter Notebooks</h4>
<p>A <code>notebooks/</code> subdirectory contains Jupyter notebooks that are named chronologically, meaning they operate on data located in either <code>steps/</code> or generated from a previous notebook. This way, the workflow can also be set up to run the notebooks (in order) to produce the figures, tables, and their captions used in this manuscript.</p>
</section>
<section id="quarto" class="level4">
<h4 class="anchored" data-anchor-id="quarto">Quarto</h4>
<p>Quarto is an open-source scientific and technical publishing system that uses (pandoc) markdown to create and share production quality output, integrating Jupyter Notebooks with Markdown and LaTeX and enabling embedding content across <em>.ipynb</em> and <em>.qmd</em>. In <em>.qmd</em>, code chunks in several programming languages can be executed and rendered, including Python, R, mermaid (JavaScript-based diagramming). A Quarto project is configured with a YAML configuration file (<code>_quarto.yml</code>) that defines how output is rendered. In this project, we use a nested structure, nesting a <em>Slides</em> project and a <em>Manuscript</em> project inside a <em>Book</em> project. To manage the directory as a Quarto Book project, a quarto configuration file was placed at the base, defining how the Book should be rendered. Additionally, configuration files were placed in <code>slides/</code> and <code>thesis/</code>, to render them as Quarto Slides and Quarto Manuscript, respectively. This nested structure lets us render different subprojects with different configurations than the main project, for example to generate the manuscript, a single Quarto Markdown file, in both <em>.html</em> and <em>.pdf</em>, and only including embedded outputs from specified cells from notebooks in the parent directory. Although the Quarto framework is extensive, it is still under development and has several drawbacks worth mentioning. First, one can only embed the output of code cells from notebooks, meaning the only way to embed text with a python variable (e.g.&nbsp;you want the manuscript to reflect the actual value of a variable, sample sizes <code>n = [1000, 10000, 100000]</code>, and their respective outputs) is by converting a formatted python string into Markdown and send it to the output. Second, embedded figures will be copied as-is in the notebook, and thus cannot be post-processed with size or layout. This makes it impractical to e.g.&nbsp;use the same figures in slides and in the manuscript. Third, when rendering large projects that is tracked by git, some output files (that have to be tracked to publish the website) can exeed GitHub size limits. Especially if rendering in the <em>jats</em> format, producing a MECA Bundle that should be the most flexible way to exchange manuscripts [ref]. However, as not applicable to this thesis, the option was simply disabled. Fourth, some functionality relies on external dependencies that cannot be installed on a (linux) remote host (GenomeDK), such as relying on a browser for converting <code>mermaid</code> diagrams into png for the pdf-manuscript.</p>
</section>
</section>
<section id="git-and-github" class="level3">
<h3 class="anchored" data-anchor-id="git-and-github">git and GitHub</h3>
<p>To track the project with git and GitHub, the abovementioned structure was initialized as a GitHub repository, including a workflow for GitHub Actions to publish and deploy the website on the <code>gh-pages</code> branch when pushing commits to <code>main</code>. Briefly, it sets up a virtual machine with Quarto and its dependencies, renders the project as specified in the <code>_quarto.yml</code> configuration file(s), and publishes the project on the group website <a href="https://munch-group.org/hic-spermatogenesis">munch-group.org</a>.</p>
</section>
</section>
<section id="our-research-question" class="level2">
<h2 class="anchored" data-anchor-id="our-research-question">Our research question</h2>
<p>In this project, we formulate two main objectives:</p>
<section id="a" class="level4">
<h4 class="anchored" data-anchor-id="a">A</h4>
<p>Redo the Hi-C analyses from <span class="citation" data-cites="wang2019reprogrammingmeioticchromatin">(<a href="#ref-wang2019reprogrammingmeioticchromatin" role="doc-biblioref">Wang et al. 2019</a>)</span> using the latest macaque reference genome, <em>rheMac10</em>, with some modifications. We decided to use <em>HiCExplorer</em>, a Python-based software for command line use, and supplement the analyses with the <em>Open2C Ecosystem</em> <span class="citation" data-cites="openchromosomecollective">(<a href="#ref-openchromosomecollective" role="doc-biblioref"><span>“Open <span>Chromosome Collective</span> (<span>Open2C</span>)”</span> n.d.</a>)</span> that have a Pyton API as well as command-line functions, which can be paired very well with Jupyter Notebooks. The majority of the data analysis was run with a <em>gwf</em> workflow, and the commands that were visually inspected were run in Jupyter Notebooks.</p>
</section>
<section id="b" class="level4">
<h4 class="anchored" data-anchor-id="b">B</h4>
<p>Compare with regions of extended common haplotypes (strong selective sweeps) that are found in <em>human</em>, and with regions of negative selection of minor parent ancestry in baboons. Investigate the biological meaning of the results. We use in-house software to compare genomic intervals.</p>
</section>
</section>
</section>
<section id="methods" class="level1">
<h1>Methods</h1>
<p>All computations were performed on GenomeDK (GDK) [ref], an HPC cluster located on Aarhus Uninversity, and most of the processing of the data was made into a custom <em>gwf</em> workflow [ref], a workflow manager developed at GDK. I would like to thank GDK and Aarhus University for providing computational resources and support that contributed to these research results.</p>
<p>With the analysis tools determined in the above section, we decided it was not feasible to follow the exact approach as <span class="citation" data-cites="wang2019reprogrammingmeioticchromatin">Wang et al. (<a href="#ref-wang2019reprogrammingmeioticchromatin" role="doc-biblioref">2019</a>)</span> with any of <em>HiCExplorer</em> and <em>Open2C</em>, as they use a third software, <em>HiC-Pro</em>. For mapping the raw reads, Hic-Pro internally uses bowtie2 in end-to-end mode, followed by trimming the 3’-end of the unmapped reads, then remapping the 5’-ends to rescue chimeric fragments. I mapped the reads using <code>bowtie2 --end-to-end</code> without the rescue-remapping, and it returned a very high fraction of discarded reads, and it would be silly to spend time on implementing the remapping approach manually. When redoing analyses, it is not sensical to use methods that are not state-of-the-art, and judged by the time since last release (HiC-Pro v3.1.0 in 2021), both HiCExplorer and Open2C are more recent. Additionally, the HiC-Pro pipeline stops at a normalized contact map, and is thus not sufficient for downstream analysis. In hindsight, it would have been more sensical to use <em>HiC-Pro</em> to get normalized contact maps, then continue analyzing with <em>cooler</em>/<em>cooltools</em>, then compare the results evenly with the results achieved from using Open2C from start to finish. <a href="#fig-hic-tools-comparison" class="quarto-xref">Figure&nbsp;2</a> gives an overview of the 3 pipelines metioned in this report.</p>
<div id="fig-hic-tools-comparison" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-hic-tools-comparison-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="illustrations/placeholder2000x360.png" class="lightbox" data-gallery="quarto-lightbox-gallery-2" title="Figure&nbsp;2: A 3-column flow of HiC-Pro, HiCExplorer, and Open2C"><img src="illustrations/placeholder2000x360.png" style="width:50.0%;height:300.0%" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-hic-tools-comparison-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2: A 3-column flow of HiC-Pro, HiCExplorer, and Open2C
</figcaption>
</figure>
</div>
<section id="fetching-raw-data" class="level2">
<h2 class="anchored" data-anchor-id="fetching-raw-data">Fetching raw data</h2>
<p>To reproduce the results from <span class="citation" data-cites="wang2019reprogrammingmeioticchromatin">(<a href="#ref-wang2019reprogrammingmeioticchromatin" role="doc-biblioref">Wang et al. 2019</a>)</span>, I chose to use their raw data directly from the SRA portal [ref]. I filtered the data to contain all their paired-end Hi-C reads, and included only macaque samples. The data set also contains RNAseq data, and the same tissues for both macaque and mouse. The meta data for the data set was extracted into a runtable <code>SRA-runtable.tsv</code>. To get an overview of the data accessions used in this analysis, we will first summarize the runtable that contains the accession numbers and some metadata for each sample (<a href="#tbl-runtable-summary" class="quarto-xref">Table&nbsp;3</a>). It adds up to ~1Tb of compressed <code>fastq</code> files, holding ~9.5 billion reads, roughly evenly spread on the 5 tissue types.</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/03_compartments.ipynb" data-notebook-title="Compartments Analysis" data-notebook-cellid="cell-tbl-runtable-summary">
<div class="cell" data-execution_count="47">
<div id="tbl-runtable-summary" class="cell quarto-float quarto-figure quarto-figure-center anchored" data-execution_count="47">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-runtable-summary-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;3: Summary of the data accessions used in this analysis
</figcaption>
<div aria-describedby="tbl-runtable-summary-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output cell-output-display">
<div>


<table class="dataframe do-not-create-environment cell caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">source_name</th>
<th data-quarto-table-cell-role="th">GB</th>
<th data-quarto-table-cell-role="th">Bases</th>
<th data-quarto-table-cell-role="th">Reads</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>fibroblast</td>
<td>211.403275</td>
<td>553,968,406,500</td>
<td>1,846,561,355</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>pachytene spermatocyte</td>
<td>274.835160</td>
<td>715,656,614,700</td>
<td>2,385,522,049</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>round spermatid</td>
<td>243.128044</td>
<td>655,938,457,200</td>
<td>2,186,461,524</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3</td>
<td>sperm</td>
<td>164.131640</td>
<td>428,913,635,400</td>
<td>1,429,712,118</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">4</td>
<td>spermatogonia</td>
<td>192.794420</td>
<td>518,665,980,300</td>
<td>1,728,886,601</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
</figure>
</div>
</div>
</div>
<section id="fetching-and-indexing-the-reference" class="level3">
<h3 class="anchored" data-anchor-id="fetching-and-indexing-the-reference">Fetching and indexing the reference</h3>
<p><span class="citation" data-cites="wang2019reprogrammingmeioticchromatin">Wang et al. (<a href="#ref-wang2019reprogrammingmeioticchromatin" role="doc-biblioref">2019</a>)</span> use the 2006-version of the macaque reference, <em>rheMac2</em>. Supporting my previous sentiment about not using outdated resources I find it reasonable to use the latest reference, <em>rheMac10</em>. <span class="citation" data-cites="warren2020sequencediversityanalyses">Warren et al. (<a href="#ref-warren2020sequencediversityanalyses" role="doc-biblioref">2020</a>)</span> have improved contiguity from <em>rhemac8</em> by 120 fold, going from N50 contig size of 107 Kbp to 46 Mbp. Part of the reasoning for reproducing their results was doing so on the latest assembly of the <em>Macaca mulata</em> genome, which arguably will result in a more accurate mapping of the reads, and a better inference of the chromatin compartments as well.</p>
<p>Therefore, the latest reference genome for rhesus macaque/<em>Macaca mulata</em>, <em>rheMac10</em>/<em>Mmul_10</em> (UCSC or NCBI naming conventions, respectively) was downloaded to GDK from UCSC web servers with <code>wget</code>. To use <code>bwa</code> for mapping, rheMac10 needs to be indexed with both <code>bwa index</code> with the <code>--bwtsw</code> option and <code>samtools faidx</code>, which results in six indexing files for <code>bwa mem</code> to use.</p>
<p>Several mappers were used in different configurations (described in below), and <code>bowtie2</code> requires its own indexing of the reference, using <code>bowtie2-build --large-index</code>, which creates six index files for <code>bowtie2</code> to use. <code>--large-index</code> creates the special indexing format required for large genomes such as macaque.</p>
</section>
</section>
<section id="hicexplorer-trials" class="level2">
<h2 class="anchored" data-anchor-id="hicexplorer-trials">HiCExplorer trials</h2>
<p>To get aligned reads in a format compatible with HiCExplorer, the read mates have to be mapped individually to the reference genome. This supports the old convention to avoid the common heuristics of local aligners used for regular paired-end sequencing libraries (<span class="citation" data-cites="lajoie2015hitchhikersguidehic">Lajoie, Dekker, and Kaplan (<a href="#ref-lajoie2015hitchhikersguidehic" role="doc-biblioref">2015</a>)</span>). HiCExplorer provide examples for both <em>bwa</em> and <em>bowtie2</em>, so I used both with recommended settings. <em>bowtie2</em> was more resource-intensive, and only succesfully aligned a small fraction of the reads [ref sup-fig-bowtie2-stats], but likely some parameters could be tuned for better alignment. In both cases, the aligner outputs a .bam-file for each mate (<code>sample_R1.bam</code> and <code>sample_R2.bam</code>), and HiCExplorer performs the parsing, deduplication, and filtering of the reads and builds the raw interaction matrix in a single command,</p>
<p><code>hicBuildMatrix -s sample_R1.bam sample_R2.bam -o matrix.h5 [...]</code>,</p>
<p>For parsing, the command needs a <code>restrictionCutFile</code>, locating the restriction sites from the restriction enzyme used on the reference genome, which is generated with <code>hicFindRestSites</code> that operates on the reference genome and restriction sequence. The default filter, <code>--minMappingQuality 15</code>, was applied as described in <a href="#sec-matrix-qc" class="quarto-xref">Section&nbsp;1.7.2.3</a>.</p>
<div class="{fig-hicexplorer-workflow}">
<p><a href="illustrations/placeholder2000x360.png" class="lightbox" data-gallery="quarto-lightbox-gallery-3"><img src="illustrations/placeholder2000x360.png" style="width:70.0%;height:100.0%"></a></p>
<p>Overview of the target templates used for hicexplorer. As most operations are handled by <code>hicBuildMatrix</code>, it is rather simple.</p>
</div>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/01_hicexplorer.ipynb" data-notebook-title="HiCExplorer using macaque data" data-notebook-cellid="cell-tbl-hic-exploration">
<div class="cell">
<div id="tbl-hic-exploration" class="cell quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-hic-exploration-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;4: The samples chosen for initial data exploration with HiCExplorer. From NCBI SRA Portal.
</figcaption>
<div aria-describedby="tbl-hic-exploration-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output cell-output-display" data-execution_count="3">
<div>


<table class="dataframe do-not-create-environment cell caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">Run</th>
<th data-quarto-table-cell-role="th">Bases</th>
<th data-quarto-table-cell-role="th">Bytes</th>
<th data-quarto-table-cell-role="th">source_name</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>SRR6502335</td>
<td>73201141800</td>
<td>31966430779</td>
<td>fibroblast</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>SRR6502336</td>
<td>65119970100</td>
<td>24433383054</td>
<td>fibroblast</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>SRR6502337</td>
<td>52769196300</td>
<td>23015357755</td>
<td>fibroblast</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3</td>
<td>SRR6502338</td>
<td>52378949100</td>
<td>22999581685</td>
<td>fibroblast</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">4</td>
<td>SRR6502339</td>
<td>28885941600</td>
<td>10960123150</td>
<td>fibroblast</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
</figure>
</div>
</div>
</div>
<p>For the initial exploration of methods with <em>HiCExplorer</em>, we chose five fibroblast samples (see <a href="#tbl-hic-exploration" class="quarto-xref">Table&nbsp;4</a>). The goal was to replicate some of the figures from <span class="citation" data-cites="wang2019reprogrammingmeioticchromatin">Wang et al. (<a href="#ref-wang2019reprogrammingmeioticchromatin" role="doc-biblioref">2019</a>)</span> using <em>HiCExplorer</em>, especially to reconstruct interaction matrices and E1 graphs from macaque data. We constructed matrices with <code>hicBuildMatrix</code> as described from the separately mapped read-pairs. Along with the matrix <em>.h5</em> file, a <em>.log</em> file was created as well, documenting the quality control for the sample. Multiple logs were aggregated and visualized with <code>hicQC</code>.</p>
<p>Before correction (or balancing) of the interaction matrix, a pre-correction filter is applied, filtering out low-count bins and very high-count bins. A threshold for Mean Absolute Deviation (<em>MAD</em>) is estimated by <code>hicCorrect diagnostic_plot</code>, followed by iterative correction with <code>hicCorrect correct --correctionMethod ICE</code>. The PCA was performed with <code>hicPCA</code> on the corrected matrices, yielding the first 3 PCs.</p>
<p><code>hicPlotMatrix</code> plots matrices directly to .png (no display). When keeping the analysis in a Jupyter Notebook (using the builtin shell-escape commands to execute <code>bash</code> code), the plot files must be embedded back into the notebook. There is limited support for modifying the plot (from command-line options), such as to add spacing for a bigWig track with E1 values, add plot titles, and define the size and resolution of the plot. I briefly tried to implement a plotting function on the .h5 matrices and bigWig tracks, but it could not fetch regions from a matrix on the fly and had to load the full matrix into memory (that is, all full-length chromosomes).</p>
</section>
<section id="open2c-pipeline" class="level2">
<h2 class="anchored" data-anchor-id="open2c-pipeline">Open2C pipeline</h2>
<div id="fig-flowchart-handling-coolers" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-flowchart-handling-coolers-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="illustrations/placeholder2000x360.png" class="lightbox" data-gallery="quarto-lightbox-gallery-4" title="Figure&nbsp;3: Showing the gwf target templates used with the Open2C pipeline. As it is highly modular, it is also a bit elaborate."><img src="illustrations/placeholder2000x360.png" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-flowchart-handling-coolers-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3: Showing the <em>gwf</em> target templates used with the <em>Open2C</em> pipeline. As it is highly modular, it is also a bit elaborate.
</figcaption>
</figure>
</div>
<p>A <em>gwf</em> workflow was created to handle the first part of the data processing, and each accesion number (read pair, mate pair) from the Hi-C sequencing was processed in parallel, so their execution was independent from each other.</p>
<section id="downloading-the-reads" class="level4">
<h4 class="anchored" data-anchor-id="downloading-the-reads">Downloading the reads</h4>
<p>The reads were downloaded from NCBI SRA with SRA-toolkit <span class="citation" data-cites="devteam2024sratoolkit">(<a href="#ref-devteam2024sratoolkit" role="doc-biblioref">DevTeam 2024</a>)</span> directly to GDK using a docker image of <code>sra-downloader</code> [ref <em>wwydmanski/sra-downloader</em>] as gunzipped <code>.fastq</code> files. Although possible to provide a list of accessions to the toolkit, I submitted each accession as a separate target, as <code>SRA Toolkit</code> acts sequentially, and only starts the next download after all compression tasks were done. It was therefore a low-hanging fruit to parallelize the download for efficiency.</p>
</section>
<section id="mapping-hi-c-reads" class="level4">
<h4 class="anchored" data-anchor-id="mapping-hi-c-reads">Mapping Hi-C reads</h4>
<p><strong>Open2C</strong> Suspiciously, <span class="citation" data-cites="openchromosomecollective">(<a href="#ref-openchromosomecollective" role="doc-biblioref"><span>“Open <span>Chromosome Collective</span> (<span>Open2C</span>)”</span> n.d.</a>)</span> never mentions any problems with aligning the Hi-C reads, they just provide an example using <code>bwa mem</code> in paired-end mode and with the <code>-P</code> option set, which activates the Smith-Waterman [ref] algorithm to rescue missing hits, by focusing on assigning only of of the mates to a good mapping and escape mate-rescue. The documentation of <code>bwa</code> <a href="https://bio-bwa.sourceforge.net">ref</a> state that both bwa-mem and bwa-sw will rescue chimeric reads. Consequently, Open2C does not have a builtin way of pairing the reads after mapping, and I was left with two options: <strong>1)</strong> to re(-)pair the individually mapped read-mates (.bam) with <code>samtools-fixmate</code> into one of the specific input formats required for <code>cooler</code> to create an interaction matrix <em>cooler</em>, or <strong>2)</strong> re-map the reads using Open2C’s recommendations and use their established pipeline for producing a cooler. I chose the latter, where I mapped the fastq files to <em>rheMac10</em> in paired end mode for a pair (<span class="math inline">\(m1\)</span>, <span class="math inline">\(m2\)</span>) with <code>bwa mem -SP rheMac10 m1 m2</code>.</p>
</section>
<section id="parse-and-sort-the-reads" class="level4">
<h4 class="anchored" data-anchor-id="parse-and-sort-the-reads">Parse and sort the reads</h4>
<p>We need to convert the alignments into ligation events, and distinguish between several types of ligation events. The simplest event is when each side only maps to one unique segment in the genome ‘UU’. Other events, where one or both sides map to multiple segments or the reads are long enough (&gt;150bp) to contain two alignments (multiple ligations) have to be considered as well. Multiple ligations (walks) are treated according to the <code>--walks-policy</code> when parsing the alignments into valid pairs (or valid Hi-C contacts). Here, <code>mask</code> is the most conservative and masks all complex walks, whereas <code>5unique</code> reports the 5’-most unique alignment on each side. The pairs are piped directly into <code>pairtools sort</code> after parsing, as the deduplication step requires a sorted set of pairs. The <em>.pairs</em>-format produced by <code>pairtools</code> is an extension the <a href="https://data.4dnucleome.org/file-formats/pairs/">4DN Consortium</a>-specified format, storing Hi-C pairs as in <a href="#tbl-pairsformat" class="quarto-xref">Table&nbsp;5</a>.</p>
<div id="tbl-pairsformat" class="striped quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-pairsformat-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;5: Column specification of the .pairs format as extended by <code>pairtools</code> [ref].
</figcaption>
<div aria-describedby="tbl-pairsformat-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<table class="table-striped caption-top table">
<colgroup>
<col style="width: 16%">
<col style="width: 16%">
<col style="width: 67%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: right;">Index</th>
<th style="text-align: left;">Name</th>
<th style="text-align: left;">Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: right;">1</td>
<td style="text-align: left;">read_id</td>
<td style="text-align: left;">the ID of the read as defined in fastq files</td>
</tr>
<tr class="even">
<td style="text-align: right;">2</td>
<td style="text-align: left;">chrom1</td>
<td style="text-align: left;">the chromosome of the alignment on side 1</td>
</tr>
<tr class="odd">
<td style="text-align: right;">3</td>
<td style="text-align: left;">pos1</td>
<td style="text-align: left;">the 1-based genomic position of the outer-most (5’) mapped bp on side 1</td>
</tr>
<tr class="even">
<td style="text-align: right;">4</td>
<td style="text-align: left;">chrom2</td>
<td style="text-align: left;">the chromosome of the alignment on side 2</td>
</tr>
<tr class="odd">
<td style="text-align: right;">5</td>
<td style="text-align: left;">pos2</td>
<td style="text-align: left;">the 1-based genomic position of the outer-most (5’) mapped bp on side 2</td>
</tr>
<tr class="even">
<td style="text-align: right;">6</td>
<td style="text-align: left;">strand1</td>
<td style="text-align: left;">the strand of the alignment on side 1</td>
</tr>
<tr class="odd">
<td style="text-align: right;">7</td>
<td style="text-align: left;">strand2</td>
<td style="text-align: left;">the strand of the alignment on side 2</td>
</tr>
<tr class="even">
<td style="text-align: right;">8</td>
<td style="text-align: left;">pair_type</td>
<td style="text-align: left;">the type of a Hi-C pair</td>
</tr>
<tr class="odd">
<td style="text-align: right;">9</td>
<td style="text-align: left;">mapq1</td>
<td style="text-align: left;">mapq of the first mate</td>
</tr>
<tr class="even">
<td style="text-align: right;">10</td>
<td style="text-align: left;">mapq2</td>
<td style="text-align: left;">mapq of the second mate</td>
</tr>
</tbody>
</table>
</div>
</figure>
</div>
<p>I initially used <code>--walks-policy mask</code>, reasoning I had plenty of data points and could handle complex walks in a conservative way. Only later I realized the recommendations from <em>pairtools</em>, specifically informing that longer reads might have a significant proportion of reads that contain complex walks. With this in mind, I decided to re-parse the alignments into a new set of pairs, and equally apply the recommended filter (next section). As both results are saved, we can compare the two approaches.</p>
</section>
<section id="filter-deduplicate-pairs" class="level4">
<h4 class="anchored" data-anchor-id="filter-deduplicate-pairs">Filter (deduplicate) pairs</h4>
<p>Pairtools comes with a de-duplication function, <code>dedup</code>, to detect PCR duplication artefacts. At this point we will remove all reads that are mapped to an unplaced scaffold. Even though the publication of <em>rhemac10</em> assembly states they have closed 99% of the gaps since <em>rhemac8</em> [ref], <em>rheMac10</em> still contain more than 2,500 unplaced scaffolds, which are all uninformative when calculating the chromatin compartments as is the goal of this analysis. Therefore, we simply only include the list of conventional chromosomes (1..22, X, Y) when doing the deduplication. Initially, the default values were used to remove duplicates, where pairs with both sides mapped within 3 base pairs from each other are considered duplicates.</p>
<p><code>cooler</code> recommend to store the most comprehensive and unfiltered list of pairs, and then applying a filter it on the fly by piping from <code>pairtools select</code>. Initially, I missed this step and I did not filter for mapping quality. After reparsing the alignments and applying the same analysis, we compare the two pipelines.</p>
<p>A quality control report is generated by <code>pairtools dedup</code>, and the reports are merged with <code>MultiQC</code> [ref] for each cell type.</p>
</section>
<section id="create-interaction-matrices-coolers" class="level4">
<h4 class="anchored" data-anchor-id="create-interaction-matrices-coolers">Create interaction matrices (coolers)</h4>
<p>The final part of the <em>gwf</em> workflow takes <code>.pairs</code> as input and outputs a <code>.cool</code> file (<em>cooler</em>). Initially, we read directly from the newly generated deduplicated pairs without additional filtering, but here, the official recommendation is to filter out everything below <span class="math inline">\(mapq = 30\)</span> by piping the pairs through <code>pairtools select "(mapq1&gt;=30) and (mapq2&gt;=30)"</code> to <code>cooler cload pairs</code>.</p>
<p>We should have plenty of data to do the filtering, but I decided it is not strictly necessary. I will show a histogram of the <em>mapq</em> scores to convince you [ref].</p>
<p>I have re-parsed the alignments and created new coolers, including only the Hi-C contacts where <span class="math inline">\(mapq \leq 30\)</span>, following the current recommendations from <code>cooler</code>.</p>
</section>
<section id="notebook-edits" class="level3">
<h3 class="anchored" data-anchor-id="notebook-edits">Notebook edits</h3>
<p>As <code>cooler</code> and <code>cooltools</code> have a Python API, the more experimental parts of the analysis were moved to Jupyter Notebooks. <code>cooltools</code> comes with a helper library for operations on genomic intervals called <code>bioframe</code>.</p>
<section id="pooling-samples-merging-coolers" class="level4">
<h4 class="anchored" data-anchor-id="pooling-samples-merging-coolers">Pooling samples (Merging coolers)</h4>
<p>The samples are grouped into <em>replicates</em> with a unique <strong>BioSample</strong> ID, but we chose to pool all the interaction matrices for each cell type. We argue that when <span class="citation" data-cites="wang2019reprogrammingmeioticchromatin">Wang et al. (<a href="#ref-wang2019reprogrammingmeioticchromatin" role="doc-biblioref">2019</a>)</span> determine compartments to be highly reproducible between replicates, by merging the replicates we can get a more robust signal.</p>
<p><code>cooler merge</code> was used to merge all samples in each sub-folder (cell type) to just one interaction matrix for each cell type. The function merges matrices of the same dimensions by simply adding the interaction frequencies of each genomic position together, resulting in less empty positions by chance.</p>
</section>
<section id="create-multi-resolution-coolers-zoomify" class="level4">
<h4 class="anchored" data-anchor-id="create-multi-resolution-coolers-zoomify">Create multi-resolution coolers (zoomify)</h4>
<p>A feature of working inside the ecosystem of <em>Open2C</em> [ref] is that it natively provides support for storing sparse interaction matrices in multiple resolutions in the same file by adding groups to the cooler [ref]. We can then efficiently store resolutions (i.e., different bin sizes) that is multiples of the smallest bin size. We chose to use 10kb, 50kb, 100kb, and 500kb bins, and the resolutions are made by recursively binning the base resolution. We call this process zoomifying.</p>
</section>
<section id="matrix-balancing-iterative-correction" class="level4">
<h4 class="anchored" data-anchor-id="matrix-balancing-iterative-correction">Matrix balancing (Iterative correction)</h4>
<p>Finally, we balance the matrices using the cooler CLI. We use <code>cooler balance</code> with the default options which iteratively balances the matrix (Iterative Correction). It is first described as a method for bias correction of Hi-C matrices in <span class="citation" data-cites="imakaev2012iterativecorrectionhic">(<a href="#ref-imakaev2012iterativecorrectionhic" role="doc-biblioref">Imakaev et al. 2012</a>)</span>, where it is paired with eigenvector decomposition, coining the combined analysis <strong>ICE</strong>. Here, the eigenvector decomposition of the obtained maps is experimentally validated to provide insights into local chromatin states.</p>
<p>[According to <code>cooler</code> documentation] We have to balance the matrices on each resolution, and thus it cannot be done prior to zoomifying. They state that the balancing weights are resolution-specific and will no longer retain its biological meaning when binned with other weights. Therefore, we apply <code>cooler balance</code> to each resolution separately. <code>cooler balance</code> will create a new column in the <code>bins</code> group of each cooler , <code>weight</code>, which can then be included or not in the downstream analysis. This means we will have access to both the balanced and the unbalanced matrix.</p>
<p>The default mode uses genome-wide data to calculate the weights for each bin. It would maybe be more suitable to calculate the weights for <em>cis</em> contacts only, and that is possible through the <code>--cis-only</code> flag, and that can be added to another column, so that we can compare the difference between the two methods easily. However, we will only use the default mode for now.</p>
</section>
<section id="sec-methods-eigendecomposition" class="level4">
<h4 class="anchored" data-anchor-id="sec-methods-eigendecomposition">Eigendecomposition</h4>
<p>The eigendecomposition of a Hi-C interaction matrix is performed in multiple steps. As value of the eigenvector is only <em>significant</em> up to a sign, it is convention [ref] to use GC content as a phasing track to orient the vector. E1 is arbitrarily defined to be positively correlated with GC content, meaning a positive E1 value signifies an active chromatin state, which we denote a A-type compartment (or simply A-compartment). We performed eigendecomposition of two resolutions, 100 Kbp and 500 Kbp. <span class="citation" data-cites="wang2019reprogrammingmeioticchromatin">Wang et al. (<a href="#ref-wang2019reprogrammingmeioticchromatin" role="doc-biblioref">2019</a>)</span> briefly describes their method to calculate the eigenvectors as a sliding window approach on the observed/expected matrix in 100 kb resolution summing over 400 kb bins with 100 kb step size, a method I was not able to replicate in the <em>Open2C</em> ecosystem. I decided to mimic this by smoothing the 100 kb E1 values by summing to 500 kb bins in steps of 100 kb, yielding a comparable resolution which I denote ‘<em>pseudo</em>-500 kb’ resolution (<em>ps500kb</em>).</p>
<p>First, we calculate the GC content of each bin of the reference genome, <em>rheMac10</em>, which is binned to the resolution of the Hi-C matrix we are handling. It is done with <code>bioframe.frac_gc</code> (<em>Open2C</em>). To calculate the E1 compartments, we use only within-chromosome contacts (<em>cis</em>), as we are not interested in the genome-wide contacts. <code>cooltools.eigs_cis</code> will decorrelate the contact-frequency by distance before performing the eigendecomposition. <code>eigs_cis</code> needs a <em>viewframe</em> (view) to calculate E1 values, the simplest view being the full chromosome. However, when there is more variance between chromosome arms than within arms, the sign of the first eigenvector will be determined largely by the chromosome arm it sits on, and not by the chromatin compartments. To mitigate this, we apply a chromosome-arm-partitioned view of the chromosome (as a bedlike format, described in <code>bioframe</code> docs [ref]).</p>
<p>Additionally, to mimic the <em>Local PCA</em> from <span class="citation" data-cites="wang2019reprogrammingmeioticchromatin">(<a href="#ref-wang2019reprogrammingmeioticchromatin" role="doc-biblioref">Wang et al. 2019</a>)</span>, I also defined a view of 10 Mb bins. Thoughout the project, I will compare results from each of the three views and resolutions.</p>
</section>
<section id="plotting-matrices" class="level4">
<h4 class="anchored" data-anchor-id="plotting-matrices">Plotting matrices</h4>
<p>We use matplotlib and seaborn to plot in the <em>Open2C</em> framework. Utilizing the <code>cooler</code> class, we can fetch regions of the matrix without modifying the file. As my analysis is centered around the X chromosome, it is efficiently handled by simply fetching ‘chrX’ from the matrix with <code>cooler.Cooler.matrix().fetch('chrX')</code>. Many methods of the cooler class returns data selectors, which do not retrieve data before it is queried [ref]. This means we can create many selectors at once without overflowing memory, enabling us to plot multiple interaction matrices side-by-side, e.g.&nbsp;the corrected and un-corrected matrices. This is easily done with the <code>balance</code> parameter of the matrix selector (<code>.matrix()</code>), which determines if it should apply the balancing weights to the coordinates and defaults to <code>True</code>.</p>
<p>The matrix is retrieved an plotted with <code>matplotlib.pyplot.matshow</code>, which automatically produces a heatmap image of the matrix. Here, in stead of transforming the interaction matrix, the color scale is log-transformed with <code>matplotlib.colors.LogNorm</code>. Additionally, <code>cooltools</code> comes with more tools to aid visualization: <em>adative coarsegrain</em> and <em>interpolation</em>, which can be chained. <code>adaptive_coarsegrain</code> iteratively coarsens an array to the nearest power of two and refines it back to the original resolution, replacing low-count pixels with NaN-aware averages to ensure no zeros in the output, unless there are very large regions that exceed the <code>max_levels</code> threshold, such as the peri-centromeric region.</p>
<p>I implemented a plotting utility, <code>plot_for_quarto</code> in notebook <code>07_various_plotting.ipynb</code> that is compatible with the YAML cell-options read by Quarto’s <code>embed</code> shortcode. It will take an arbitrary number of samples and plot a chromosome (or region) with or without its respective E1 value for either of the three viewframes that has been created. The input is a (subsetted) <em>pandas DataFrame</em>, defined from a file search matching a pattern specified to the <code>glob</code> Python module.</p>
</section>
</section>
<section id="compartments-and-their-edges-transitional-regions" class="level3">
<h3 class="anchored" data-anchor-id="compartments-and-their-edges-transitional-regions">Compartments and Their Edges (Transitional Regions)</h3>
<p>From the eigenvectors, the A-compartments were extracted in bedgraph-format (<code>['chrom', 'start', 'end']</code>) and compared with ECH90 regions lifted to <em>rheMac10</em> from human [ref what reference?]. We perform visual inspection of the genomic intervals and test whether ECH90 regions are enriched near the edges of the compartments by defining a 200 kilobase transition-zone centered at each sign change of E1 (referred to as <em>compartment edge</em>). We compare genomic intervals (or sets) both visually by plotting the regions, and by a proximity test and bootstrapping the Jaccard index.</p>
<section id="proximity-test" class="level4">
<h4 class="anchored" data-anchor-id="proximity-test">Proximity test</h4>
<p>Determines whether the non-overlapping parts of the sets are more proximal than expected by chance. We define the <em>annotation</em> set and the <em>query</em> set, and the distance from each interval on the <em>query</em> to the most proximal interval on the <em>annotation</em> is used to generate an index of proximity. It uses bootstrapping (<span class="math inline">\(b = 100000\)</span>) to generate the null distribution, and finally, the fraction of the <em>null</em> as or more extreme as our observed proximity is reported as the p-value.</p>
</section>
<section id="jaccard-test" class="level4">
<h4 class="anchored" data-anchor-id="jaccard-test">Jaccard test</h4>
<p>Measures the significance of the observed Jaccard index (intersection over union) between two sets. The index is a measure of <em>similarity</em> between two sets (between 0 and 1), which is very sensitive to the size difference between the sets, as even when comparing a set of intervals to a small subset of itself will yield a very small Jaccard index. When we use bootstrapping to generate a null distribution (shuffling the intervals of the <em>query</em>), we generate the probability that the two sets (with their respective number and size of intervals), are as similar or more than what we observe. The ratio is reported as the p-value. However, this approach is still sensitive to flipping of query/annotation, as only the query is bootstrapped.</p>
</section>
<section id="multiple-testing" class="level4">
<h4 class="anchored" data-anchor-id="multiple-testing">Multiple testing</h4>
<p>Careful considerations were made to avoid multiple testing biases (p-hacking): Performing tests on all combinations of variables (cell type, resolution, viewframe, flip annot, query, test) will yield 260 p-values, and we would have to adjust the significance threshold (with <span class="math inline">\(\alpha = 0.05\)</span>, we expect 13 ‘significant’ tests by chance alone). However, if we choose only</p>
</section>
</section>
</section>
</section>
<section id="results" class="level1">
<h1>Results</h1>
<section id="exploration-with-hicexplorer" class="level2">
<h2 class="anchored" data-anchor-id="exploration-with-hicexplorer">Exploration with HicExplorer</h2>
<section id="quality-control" class="level3">
<h3 class="anchored" data-anchor-id="quality-control">Quality Control</h3>
<p>The separately mapped read-mates were parsed into a <em>.h5</em> interaction matrix by <code>hicBuildMatrix</code>, which include a <em>.log</em> file documenting the builtin quality control (hereafter, <em>QC</em>). Log files from the 5 samples were merged with <code>hicQC</code> (<a href="#fig-explorer-qc" class="quarto-xref">Figure&nbsp;4</a>). We observe showed equal fractions of the read-orientation of read-pairs (<a href="#fig-explorer-read-orientation" class="quarto-xref">Figure&nbsp;4 (a)</a>), which is expected for a good Hi-C library. Additionally, it determines between 40% to 50% of the total reads to be valid Hi-C contacts (<a href="#fig-explorer-unique-pairs" class="quarto-xref">Figure&nbsp;4 (b)</a>), which is <a href="https://hicexplorer.readthedocs.io/en/latest/content/example_usage.html#creation-of-a-hi-c-matrix">allegedly</a> [ref] usually only 25%-40%. <del>Overall a solid set of Hi-C samples until now</del>. <a href="#fig-explorer-contact-distance" class="quarto-xref">Figure&nbsp;4 (e)</a> shows, however, unusually high fractions of <em>inter</em>-chromosomal contacts (up to 30%) compared to <em>intra</em>-chromosomal contacts (also denoted <em>trans</em> and <em>cis</em> contacts, respectively). It is expected that <em>cis</em> contacts are orders of magnitude more frequent than <em>trans</em> contacts <span class="citation" data-cites="bicciato2022hicdataanalysis lieberman-aiden2009comprehensivemappinglongrange">(<a href="#ref-bicciato2022hicdataanalysis" role="doc-biblioref">Bicciato and Ferrari 2022, 2301:236</a>; <a href="#ref-lieberman-aiden2009comprehensivemappinglongrange" role="doc-biblioref">Lieberman-Aiden et al. 2009</a>)</span>, and <em>HiCExplorer</em> states it should be below 10% [ref]. The high fraction may be mitigated by enforcing a stricter <em>mapq</em> threshold for a valid Hi-C pair, as we also observe higher-than expected valid contacts. However, we continue without the current matrices.</p>
<div id="fig-explorer-qc" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-explorer-qc-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-qc" style="flex-basis: 33.3%;justify-content: flex-start;">
<div id="fig-explorer-read-orientation" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-read-orientation-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../steps/bwa/QC_all_samples/read_orientation.png" class="lightbox" data-gallery="fig-explorer-qc" title="Figure&nbsp;4&nbsp;(a): Read orientation"><img src="../steps/bwa/QC_all_samples/read_orientation.png" class="img-fluid figure-img" data-ref-parent="fig-explorer-qc"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-read-orientation-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) Read orientation
</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-qc" style="flex-basis: 33.3%;justify-content: flex-start;">
<div id="fig-explorer-unique-pairs" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-unique-pairs-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../steps/bwa/QC_all_samples/unmappable_and_non_unique.png" class="lightbox" data-gallery="fig-explorer-qc" title="Figure&nbsp;4&nbsp;(b): Unique pairs"><img src="../steps/bwa/QC_all_samples/unmappable_and_non_unique.png" class="img-fluid figure-img" data-ref-parent="fig-explorer-qc"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-unique-pairs-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) Unique pairs
</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-qc" style="flex-basis: 33.3%;justify-content: flex-start;">
<div id="fig-explorer-discarded-pairs" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-discarded-pairs-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../steps/bwa/QC_all_samples/pairs_discarded.png" class="lightbox" data-gallery="fig-explorer-qc" title="Figure&nbsp;4&nbsp;(c): Discarded pairs"><img src="../steps/bwa/QC_all_samples/pairs_discarded.png" class="img-fluid figure-img" data-ref-parent="fig-explorer-qc"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-discarded-pairs-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(c) Discarded pairs
</figcaption>
</figure>
</div>
</div>
</div>
<div class="quarto-layout-row">
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-qc" style="flex-basis: 33.3%;justify-content: flex-start;">
<div id="fig-explorer-pairs-sequenced" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-pairs-sequenced-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../steps/bwa/QC_all_samples/pairs_sequenced.png" class="lightbox" data-gallery="fig-explorer-qc" title="Figure&nbsp;4&nbsp;(d): Pairs sequenced"><img src="../steps/bwa/QC_all_samples/pairs_sequenced.png" class="img-fluid figure-img" data-ref-parent="fig-explorer-qc"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-pairs-sequenced-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(d) Pairs sequenced
</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-qc" style="flex-basis: 33.3%;justify-content: flex-start;">
<div id="fig-explorer-contact-distance" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-contact-distance-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../steps/bwa/QC_all_samples/distance.png" class="lightbox" data-gallery="fig-explorer-qc" title="Figure&nbsp;4&nbsp;(e): Contact distance"><img src="../steps/bwa/QC_all_samples/distance.png" class="img-fluid figure-img" data-ref-parent="fig-explorer-qc"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-contact-distance-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(e) Contact distance
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-explorer-qc-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;4: Quality control of the mapped Hi-C reads using <em>HiCExplorer</em> <code>hicQC</code>. The figures should be moved to Supplementary/Appendix because they are ugly and un-alignable. But that is the fault of HiCExplorer, not me. <del>Or I should spend a couple of hours to plot them manually</del>.
</figcaption>
</figure>
</div>
</section>
<section id="correction" class="level3">
<h3 class="anchored" data-anchor-id="correction">Correction</h3>
<p>The correction diagnostic tool yielded a similar <em>mad</em> threshold within the range <span class="math inline">\([-3,-2]\)</span>. Even so, I followed the <em>HicExplorer</em> recommendation to set the lower threshold to at least -2 and the upper threshold to 5 in the pre-normalization filter. I argue that with a high number of valid contacts, it is safer to err on the side of caution and maybe filter out bad data.</p>
<div id="fig-explorer-pre-correction" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-explorer-pre-correction-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 5.9%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-pre-correction" style="flex-basis: 29.4%;justify-content: flex-start;">
<div id="fig-explorer-pre-correction-SRR6502335" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-pre-correction-SRR6502335-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/bwa/SRR6502335_diag_plot.png" class="lightbox" data-gallery="fig-explorer-pre-correction" title="Figure&nbsp;5&nbsp;(a): SRR6502335"><img src="../figures/bwa/SRR6502335_diag_plot.png" class="img-fluid figure-img" data-ref-parent="fig-explorer-pre-correction"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-pre-correction-SRR6502335-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) SRR6502335
</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-pre-correction" style="flex-basis: 29.4%;justify-content: flex-start;">
<div id="fig-explorer-pre-correction-SRR6502336" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-pre-correction-SRR6502336-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/bwa/SRR6502336_diag_plot.png" class="lightbox" data-gallery="fig-explorer-pre-correction" title="Figure&nbsp;5&nbsp;(b): SRR6502336"><img src="../figures/bwa/SRR6502336_diag_plot.png" class="img-fluid figure-img" data-ref-parent="fig-explorer-pre-correction"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-pre-correction-SRR6502336-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) SRR6502336
</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-pre-correction" style="flex-basis: 29.4%;justify-content: flex-start;">
<div id="fig-explorer-pre-correction-SRR6502337" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-pre-correction-SRR6502337-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/bwa/SRR6502337_diag_plot.png" class="lightbox" data-gallery="fig-explorer-pre-correction" title="Figure&nbsp;5&nbsp;(c): SRR6502337"><img src="../figures/bwa/SRR6502337_diag_plot.png" class="img-fluid figure-img" data-ref-parent="fig-explorer-pre-correction"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-pre-correction-SRR6502337-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(c) SRR6502337
</figcaption>
</figure>
</div>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 5.9%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
</div>
<div class="quarto-layout-row">
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 20.6%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-pre-correction" style="flex-basis: 29.4%;justify-content: flex-start;">
<div id="fig-explorer-pre-correction-SRR6502338" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-pre-correction-SRR6502338-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/bwa/SRR6502338_diag_plot.png" class="lightbox" data-gallery="fig-explorer-pre-correction" title="Figure&nbsp;5&nbsp;(d): SRR6502338"><img src="../figures/bwa/SRR6502338_diag_plot.png" class="img-fluid figure-img" data-ref-parent="fig-explorer-pre-correction"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-pre-correction-SRR6502338-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(d) SRR6502338
</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-pre-correction" style="flex-basis: 29.4%;justify-content: flex-start;">
<div id="fig-explorer-pre-correction-SRR6502339" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-pre-correction-SRR6502339-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/bwa/SRR6502339_diag_plot.png" class="lightbox" data-gallery="fig-explorer-pre-correction" title="Figure&nbsp;5&nbsp;(e): SRR6502339"><img src="../figures/bwa/SRR6502339_diag_plot.png" class="img-fluid figure-img" data-ref-parent="fig-explorer-pre-correction"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-pre-correction-SRR6502339-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(e) SRR6502339
</figcaption>
</figure>
</div>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 20.6%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-explorer-pre-correction-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5: Histograms of the number of counts per bin (bottom x-axis) and the modified z-score (top x-axis) from which the <em>mad</em> threshold is defined.
</figcaption>
</figure>
</div>
<blockquote class="blockquote">
<p>NB: when I say that a mapper performs poorly in finding Hi-C contacts, it is <code>hicBuildMatrix</code> that performs badly when reads are mapped with that mapper.</p>
</blockquote>
<p>To compare these mappings with others, the QC results is an easy way. Therefore, the reads were mapped with <em>bowtie2</em> in both end-to-end- and local-mode followed by <code>hiCBuildMatrix</code>, and the QC from each method was plotted next to each other (<a href="#fig-explorer-all-3-qc" class="quarto-xref">Figure&nbsp;6</a>). Interestingly, <em>bowtie2</em> was much more computer-intensive in both modes, perhaps because of the <code>--very-sensitive</code> option. In any case, the QC reveals a major difference in the total number of reads that are determined to be valid Hi-C contacts by <code>hicBuildMatrix</code>. As expected, <em>end-to-end-bowtie2</em> performs worse at locating Hi-C contacts than the other methods [ref row1], finding a very low amount of mappable, unique pairs passing the quality threshold. In contrast, <em>local-bowtie2</em> performs similarly to <em>bwa</em> in finding mappable, unique, high-quality pairs, but calls only approximately half the number of valid Hi-C contacts (&gt;20%), resulting in a fraction of valid Hi-C pairs that hits the expectation from <em>HicExplorer</em> docs [ref row3]. With <em>bwa</em>, the reads were discarded either due to low mapping quality or non-unique mates, whereas with <em>local-bowtie2</em>, the reads were almost exclusively filtered out due to low mapping quality. This must be a result of how the mappers assign mapping quality, and I believe <em>local-bowtie2</em> looks suspiciously selective in finding unique but low quality alignments. <em>end-to-end-bowtie</em> almost exclusively filters out read-pairs where one mate is unmapped, which is expected when the majority of reads are unmapped.</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/01_hicexplorer.ipynb" data-notebook-title="HiCExplorer using macaque data" data-notebook-cellid="cell-fig-explorer-all-3-qc">
<div id="cell-fig-explorer-all-3-qc" class="cell" data-execution_count="1">
<div class="cell-output cell-output-display">
<div id="fig-explorer-all-3-qc" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-explorer-all-3-qc-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-01_hicexplorer-fig-explorer-all-3-qc-output-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-15" title="Figure&nbsp;6: Comparison of HiCExplorer QC plots for all samples using different alignment tools."><img src="index_files/figure-html/..-notebooks-01_hicexplorer-fig-explorer-all-3-qc-output-1.png" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-explorer-all-3-qc-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6: Comparison of HiCExplorer QC plots for all samples using different alignment tools.
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<p>As discussed, the five samples were pooled with <code>hicSumMatrices</code>, and the non-standard contigs (unplaced scaffolds) were filtered out, and the different resolutions were created (<code>hicMergeMatrixBins</code>). <em>HiCExplorer</em> also comes with a normalization function prior to correcting the matrix, which should be applied if different samples should have comparable bin counts. It has no effect when having only one matrix. Nevertheless, the pooled matrix was normalized and then corrected compared in <a href="#fig-explorer-pooled-norm-normcorr" class="quarto-xref">Figure&nbsp;7</a>.</p>
<div id="fig-explorer-pooled-norm-normcorr" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-explorer-pooled-norm-normcorr-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-pooled-norm-normcorr" style="flex-basis: 50.0%;justify-content: flex-start;">
<div id="fig-explorer-pooled-chrX-norm" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-pooled-chrX-norm-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/bowtie2/local/filter_pooled_50kb_chrX.png" class="lightbox" data-gallery="fig-explorer-pooled-norm-normcorr" title="Figure&nbsp;7&nbsp;(a): Normalized matrix chrX"><img src="../figures/bowtie2/local/filter_pooled_50kb_chrX.png" class="img-fluid figure-img" data-ref-parent="fig-explorer-pooled-norm-normcorr"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-pooled-chrX-norm-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) Normalized matrix chrX
</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-pooled-norm-normcorr" style="flex-basis: 50.0%;justify-content: flex-start;">
<div id="fig-explorer-pooled-chrX-normcorr" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-pooled-chrX-normcorr-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/bowtie2/local/normalized/normsm_filter_pooled_100kb_corrected_chrX-full.png" class="lightbox" data-gallery="fig-explorer-pooled-norm-normcorr" title="Figure&nbsp;7&nbsp;(b): Normalized and corrected chrX"><img src="../figures/bowtie2/local/normalized/normsm_filter_pooled_100kb_corrected_chrX-full.png" class="img-fluid figure-img" data-ref-parent="fig-explorer-pooled-norm-normcorr"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-pooled-chrX-normcorr-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) Normalized and corrected chrX
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-explorer-pooled-norm-normcorr-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7: A comparison of interaction matrices before/after iterative correction (<em>HiCExplorer</em>).
</figcaption>
</figure>
</div>
<p>It is now obvious why we have to correct the matrix. The uncorrected (<a href="#fig-explorer-pooled-chrX-norm" class="quarto-xref">Figure&nbsp;7 (a)</a>) has no signal apart from the diagonal. Even though some bins have been filtered out, the expected <em>plaid</em> pattern of a contact matrix is visible along the diagonal after the correction (<a href="#fig-explorer-pooled-chrX-normcorr" class="quarto-xref">Figure&nbsp;7 (b)</a>), leaving evidence for chromatin structure, especially in the first 50 million bases of the chromosome. There is a wide region of empty values at the place of the centromere.</p>
</section>
<section id="eigenvectors" class="level3">
<h3 class="anchored" data-anchor-id="eigenvectors">Eigenvectors</h3>
<p>The PCA performed by <code>hicPCA</code> on the pooled samples at both 50kb and 100kb resolution yielded the first 3 principal components. For PC1 on both resolutions (<a href="#fig-explorer-pc1-50kb" class="quarto-xref">Figure&nbsp;8 (a)</a>, <a href="#fig-explorer-pc1-100kb" class="quarto-xref">Figure&nbsp;8 (c)</a>) we observe only a single sign change which occurs at around 60 Mbp, the region of the centromere. It means the PCA has captured more variance between the chromosome arms than within them, making it uninformative about chromatin compartments. Upon visual inspection, it is clear that neither of the PC graphs capture the pattern of the interaction matrix by its change of sign. It seems the PCs capture variance from a bias that varies slowly and predictably along the chromosome. The first PC that is supposed to capture the compartments very suspiciously changes sign at the region of the centromere, a classic problem that could be solved by restricting the values from which the PC is calculated along the chromosome. Unimpressed, I rationalize that the option <code>--extra-track</code> to provide a gene track or histone coverage should not affect this result much. It should be provided as a phasing track to orient the eigenvector to positively correlate with gene density or histone marks, and could possibly muddle the compartments if not included. I followed <em>HiCExplorer</em> pipeline to plot and explore the matrices. At this point, I stoppped using <em>HiCExplorer</em>, as I assessed that a more flexible tool was needed.</p>
<div id="fig-explorer-pca" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-explorer-pca-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-pca" style="flex-basis: 33.3%;justify-content: flex-start;">
<div id="fig-explorer-pc1-50kb" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-pc1-50kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/bowtie2/local/normalized/pc1_50kb_corrected_chrX.png" class="lightbox" data-gallery="fig-explorer-pca" title="Figure&nbsp;8&nbsp;(a): "><img src="../figures/bowtie2/local/normalized/pc1_50kb_corrected_chrX.png" id="fig-explorer-pc1-50kb" class="img-fluid figure-img" data-ref-parent="fig-explorer-pca"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-pc1-50kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a)
</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-pca" style="flex-basis: 33.3%;justify-content: flex-start;">
<div id="fig-explorer-pc2-50kb" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-pc2-50kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/bowtie2/local/normalized/pc2_50kb_corrected_chrX.png" class="lightbox" data-gallery="fig-explorer-pca" title="Figure&nbsp;8&nbsp;(b): "><img src="../figures/bowtie2/local/normalized/pc2_50kb_corrected_chrX.png" id="fig-explorer-pc2-50kb" class="img-fluid figure-img" data-ref-parent="fig-explorer-pca"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-pc2-50kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b)
</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell" style="flex-basis: 33.3%;justify-content: center;">
<p><a href="../figures/bowtie2/local/normalized/pc3_50kb_corrected_chrX.png" class="lightbox" data-gallery="quarto-lightbox-gallery-20" title="Figure&nbsp;8: Corrected interaction matrix for chromosome X along with PC1, 2, or 3, respectively. a-c: 50kb resolution, d-f: 100kb resolution. HiCExplorer."><img src="../figures/bowtie2/local/normalized/pc3_50kb_corrected_chrX.png" class="img-fluid figure-img"></a></p>
</div>
</div>
<div class="quarto-layout-row">
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-pca" style="flex-basis: 33.3%;justify-content: flex-start;">
<div id="fig-explorer-pc1-100kb" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-pc1-100kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/bowtie2/local/normalized/pc1_100kb_corrected_chrX.png" class="lightbox" data-gallery="fig-explorer-pca" title="Figure&nbsp;8&nbsp;(c): "><img src="../figures/bowtie2/local/normalized/pc1_100kb_corrected_chrX.png" id="fig-explorer-pc1-100kb" class="img-fluid figure-img" data-ref-parent="fig-explorer-pca"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-pc1-100kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(c)
</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-pca" style="flex-basis: 33.3%;justify-content: flex-start;">
<div id="fig-explorer-pc2-100kb" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-pc2-100kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/bowtie2/local/normalized/pc2_100kb_corrected_chrX.png" class="lightbox" data-gallery="fig-explorer-pca" title="Figure&nbsp;8&nbsp;(d): "><img src="../figures/bowtie2/local/normalized/pc2_100kb_corrected_chrX.png" id="fig-explorer-pc2-100kb" class="img-fluid figure-img" data-ref-parent="fig-explorer-pca"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-pc2-100kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(d)
</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-explorer-pca" style="flex-basis: 33.3%;justify-content: flex-start;">
<div id="fig-explorer-pc3-100kb" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-explorer-pc3-100kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/bowtie2/local/normalized/pc3_100kb_corrected_chrX.png" class="lightbox" data-gallery="fig-explorer-pca" title="Figure&nbsp;8&nbsp;(e): "><img src="../figures/bowtie2/local/normalized/pc3_100kb_corrected_chrX.png" id="fig-explorer-pc3-100kb" class="img-fluid figure-img" data-ref-parent="fig-explorer-pca"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-explorer-pc3-100kb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(e)
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-explorer-pca-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;8: Corrected interaction matrix for chromosome X along with PC1, 2, or 3, respectively. a-c: 50kb resolution, d-f: 100kb resolution. <em>HiCExplorer</em>.
</figcaption>
</figure>
</div>
</section>
</section>
<section id="open2c-ecosystem" class="level2">
<h2 class="anchored" data-anchor-id="open2c-ecosystem">Open2c ecosystem</h2>
<section id="quality-control-1" class="level3">
<h3 class="anchored" data-anchor-id="quality-control-1">Quality Control</h3>
<section id="initial-run---walks-policy-mask" class="level4">
<h4 class="anchored" data-anchor-id="initial-run---walks-policy-mask">Initial run (<code>--walks-policy mask</code>)</h4>
<p>Comparing the multiQC report for each of the cell sources show similar distributions of <em>unmapped</em> (both sides unmapped), <em>one-sided</em> (one side mapped), <em>two-sided</em> (both sides mapped), and <em>duplicated</em> (w.r.t. total mapped) reads. The percentage of <em>cis</em> pairs w.r.t. mapped pairs is around 70% for all samples. [ref supptbl-qc-all-samples-mask]. The valid pairs also show similar distributions of pair types divided into 10 categories. The <span class="math inline">\(P(s)\)</span> curve looks similar as well, peaking between 270 bp and 320 bp separation (ref suppfig-multiqc-ps-curve). The QC does not show any information about mapping quality of the reads. Note that the <span class="math inline">\(P(s)\)</span> ccurve arise from pre-filtered pairs, and should be compcared with the <span class="math inline">\(P(s)\)</span> of the cooler after filtering.</p>
<p>[ref if there is time, include a histogram of mapq values for pairs]</p>
</section>
<section id="recommended---walks-policy-5unique" class="level4">
<h4 class="anchored" data-anchor-id="recommended---walks-policy-5unique">Recommended (<code>--walks-policy 5unique</code>)</h4>
<p>Parsing alignments with the recommended walks-policy aproximately halves the percentage of <em>unmapped</em> reads, and <em>one-</em> and <em>two-sided</em> reads as well <em>duplicated</em> reads are slightly increased. Overall number of unique pairs are increased with more than 20% increase. The percentage of <em>cis</em> pairs are only decreased by a percentage point at most [ref supptbl-qc-all-samples-5unique].</p>
</section>
</section>
<section id="correction-1" class="level3">
<h3 class="anchored" data-anchor-id="correction-1">Correction</h3>
<p>Matrix balancing did not show major improvement in the plaid pattern, as it was already pretty good. It does, however, filter out bins that are deemed too low-count to be informative, for example peri-centromeric regions. The matrix was expected to be smoother after balancing (for chromosomes), as regions along a chromosome should only vary slowly in contact frequency with other regions [ref].</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/05_rec_compartments.ipynb" data-notebook-title="Compartments Analysis w. recommended parameters" data-notebook-cellid="cell-fig-rs-chrx-raw-balanced-cgi">
<div id="cell-fig-rs-chrx-raw-balanced-cgi" class="cell" data-execution_count="100">
<div class="cell-output cell-output-display">
<div id="fig-rs-chrx-raw-balanced-cgi" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-rs-chrx-raw-balanced-cgi-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-05_rec_compartments-fig-rs-chrx-raw-balanced-cgi-output-2.png" class="lightbox" data-gallery="quarto-lightbox-gallery-24" title="Figure&nbsp;9: Raw, balanced, and interpolated chrX interaction matrix in 500kb resolution. The interpolation is done to make the matrix more visually appealing, but it is not necessary for the analysis."><img src="index_files/figure-html/..-notebooks-05_rec_compartments-fig-rs-chrx-raw-balanced-cgi-output-2.png" width="593" height="164" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-rs-chrx-raw-balanced-cgi-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;9: Raw, balanced, and interpolated chrX interaction matrix in 500kb resolution. The interpolation is done to make the matrix more visually appealing, but it is not necessary for the analysis.
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/05_rec_compartments.ipynb" data-notebook-title="Compartments Analysis w. recommended parameters" data-notebook-cellid="cell-fig-rs-chrx-raw-balanced-cgi-subset">
<div id="cell-fig-rs-chrx-raw-balanced-cgi-subset" class="cell" data-execution_count="102">
<div class="cell-output cell-output-display">
<div id="fig-rs-chrx-raw-balanced-cgi-subset" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-rs-chrx-raw-balanced-cgi-subset-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-05_rec_compartments-fig-rs-chrx-raw-balanced-cgi-subset-output-2.png" class="lightbox" data-gallery="quarto-lightbox-gallery-25" title="Figure&nbsp;10: Raw, balanced, and interpolated chrX interaction matrix in 50kb resolution. The interpolation is done to make the matrix more visually appealing, but it is not necessary for the analysis."><img src="index_files/figure-html/..-notebooks-05_rec_compartments-fig-rs-chrx-raw-balanced-cgi-subset-output-2.png" width="593" height="167" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-rs-chrx-raw-balanced-cgi-subset-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;10: Raw, balanced, and interpolated chrX interaction matrix in 50kb resolution. The interpolation is done to make the matrix more visually appealing, but it is not necessary for the analysis.
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<p>The coarsegrained and interpolated matrix is useful to make a good-looking interaction matrix, but is not that useful for analysis purposes. It might get easier to visually inspect the matrix, but it is not clear how well the interpolated matrix reflects the structure of the chromatin. The regions that are coarsgrained are small zero- or low-count bins which are averaged, effectively reducing the resolution of those regions until the count is sufficient. They get more frequent the longer genomic distance (the further we travel from the diagonal), and effectively enables us to get some intuition about the interactions. The coarsegrain, however, does not interpolate the <code>NaN</code>s created when filtering out whole bins in the balancing step (horisontal and vertical lines in <a href="#fig-rs-chrx-raw-balanced-cgi" class="quarto-xref">Figure&nbsp;9</a> and <a href="#fig-rs-chrx-raw-balanced-cgi-subset" class="quarto-xref">Figure&nbsp;10</a>; middle). This is done in a subsequent step by linearly interpolating the <code>NaN</code>s. Examining the interpolated matrix on full chrX (<a href="#fig-rs-chrx-raw-balanced-cgi" class="quarto-xref">Figure&nbsp;9</a>; right) gives the impression that the pericentromeric (at ~60 Mbp) region harbours a <em>very</em> strong compartment, but that is clearly an artefact of the interpolation on the very large empty region of the centromere, where the diagonal is somehow extended in a square. On the thinner lines, the interpolation seem to be more smooth, and barely noticable on the diagonal.</p>
<section id="nan-histograms" class="level4">
<h4 class="anchored" data-anchor-id="nan-histograms"><code>NaN</code> histograms</h4>
<p>As expected, most of the low quality bins are located on the edges of the chromosome arms, especially the region around the centromere [ref litterature], as they contain many repetitive sequences. The low-quality bins are filtered out by the balancing algorithm, those bins are <code>NaN</code> in the Hi-C matrix. The median position of the <code>NaN</code> values (<a href="#fig-e1_nan_hist" class="quarto-xref">Figure&nbsp;11</a>) ranges between <span class="math inline">\(58\)</span> and <span class="math inline">\(63.5\)</span>, which is within the estimate of the centromeric region of <em>rhemac10</em>.</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/05_rec_compartments.ipynb" data-notebook-title="Compartments Analysis w. recommended parameters" data-notebook-cellid="cell-fig-e1_nan_hist">
<div id="cell-fig-e1_nan_hist" class="cell" data-execution_count="108">
<div class="cell-output cell-output-display">
<div id="fig-e1_nan_hist" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-e1_nan_hist-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-05_rec_compartments-fig-e1_nan_hist-output-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-26" title="Figure&nbsp;11: Histogram of NaN values in the E1 eigenvector for each cell type. Median position is marked with a red dashed line."><img src="index_files/figure-html/..-notebooks-05_rec_compartments-fig-e1_nan_hist-output-1.png" width="595" height="176" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-e1_nan_hist-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;11: Histogram of NaN values in the E1 eigenvector for each cell type. Median position is marked with a red dashed line.
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<p>The fact that the medians lie within the centromeric region on all cell sources shows both that the majority of the bad bins are in the (peri)centromeric region <em>and</em> there are approximately equally many on each side.</p>
</section>
</section>
<section id="sec-results-eigenvectors" class="level3">
<h3 class="anchored" data-anchor-id="sec-results-eigenvectors">Compartments (Eigenvectors)</h3>
<p>The three viewframes (<em>Full</em>, <em>Arms</em>, <em>10Mb</em>) for the calculation of the eigenvectors captured different variability in the data (<a href="#fig-e1-matrix-500kb-full-arms-10mb-round_spermatid" class="quarto-xref">Figure&nbsp;12 (a)</a>), and as expected, the inferred compartments (colored red on the E1 tracks) are more abundant and smaller with smaller viewframes. To determine how well each of the E1 tracks capture the pattern in the interaction matrix, we can overlay the matrix with the E1 sign-change and visually determine if the squares reflect the E1 sign change (<a href="#fig-e1-matrix-500kb-full-arms-10mb-round_spermatid" class="quarto-xref">Figure&nbsp;12 (a)</a>).</p>
<div id="fig-e1-matrix-full-arms-10mb-round-spermatid" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-e1-matrix-full-arms-10mb-round-spermatid-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-embed-nb-cell quarto-layout-cell-subref quarto-layout-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/05_rec_compartments.ipynb" data-notebook-title="Compartments Analysis w. recommended parameters" data-notebook-cellid="cell-fig-e1-matrix-500kb-full-arms-10mb-round_spermatid" data-ref-parent="fig-e1-matrix-full-arms-10mb-round-spermatid" style="flex-basis: 48.4%;justify-content: flex-start;">
<div id="cell-fig-e1-matrix-500kb-full-arms-10mb-round_spermatid" class="cell quarto-layout-cell-subref" data-execution_count="89" data-ref-parent="fig-e1-matrix-full-arms-10mb-round-spermatid">
<div class="cell-output cell-output-display quarto-layout-cell-subref" data-ref-parent="fig-e1-matrix-full-arms-10mb-round-spermatid">
<div id="fig-e1-matrix-500kb-full-arms-10mb-round_spermatid" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-e1-matrix-500kb-full-arms-10mb-round_spermatid-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-05_rec_compartments-fig-e1-matrix-500kb-full-arms-10mb-round_spermatid-output-1.png" class="lightbox" data-gallery="fig-e1-matrix-full-arms-10mb-round-spermatid" title="Figure&nbsp;12&nbsp;(a): E1 eigenvector values for merged round spermatid samples at 500kb resolution, as well as the interaction matrix. E1 was restricted to either Full-chromosome (top), Chromosome-arms (middle), or 10Mb windows (bottom)."><img src="index_files/figure-html/..-notebooks-05_rec_compartments-fig-e1-matrix-500kb-full-arms-10mb-round_spermatid-output-1.png" data-ref-parent="fig-e1-matrix-full-arms-10mb-round-spermatid" width="316" height="296" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-e1-matrix-500kb-full-arms-10mb-round_spermatid-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) E1 eigenvector values for merged round spermatid samples at 500kb resolution, as well as the interaction matrix. E1 was restricted to either Full-chromosome (top), Chromosome-arms (middle), or 10Mb windows (bottom).
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 3.2%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="quarto-embed-nb-cell quarto-layout-cell-subref quarto-layout-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/05_rec_compartments.ipynb" data-notebook-title="Compartments Analysis w. recommended parameters" data-notebook-cellid="cell-fig-e1-matrix-100kb-full-arms-10mb-round_spermatid" data-ref-parent="fig-e1-matrix-full-arms-10mb-round-spermatid" style="flex-basis: 48.4%;justify-content: flex-start;">
<div id="cell-fig-e1-matrix-100kb-full-arms-10mb-round_spermatid" class="cell quarto-layout-cell-subref" data-execution_count="83" data-ref-parent="fig-e1-matrix-full-arms-10mb-round-spermatid">
<div class="cell-output cell-output-display quarto-layout-cell-subref" data-ref-parent="fig-e1-matrix-full-arms-10mb-round-spermatid">
<div id="fig-e1-matrix-100kb-full-arms-10mb-round_spermatid" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-e1-matrix-100kb-full-arms-10mb-round_spermatid-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-05_rec_compartments-fig-e1-matrix-100kb-full-arms-10mb-round_spermatid-output-1.png" class="lightbox" data-gallery="fig-e1-matrix-full-arms-10mb-round-spermatid" title="Figure&nbsp;12&nbsp;(b): E1 eigenvector values for merged round spermatid samples at 500kb resolution, as well as the interaction matrix. E1 was restricted to either Full-chromosome (top), Chromosome-arms (middle), or 10Mb windows (bottom)."><img src="index_files/figure-html/..-notebooks-05_rec_compartments-fig-e1-matrix-100kb-full-arms-10mb-round_spermatid-output-1.png" data-ref-parent="fig-e1-matrix-full-arms-10mb-round-spermatid" width="316" height="296" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-e1-matrix-100kb-full-arms-10mb-round_spermatid-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) E1 eigenvector values for merged round spermatid samples at 500kb resolution, as well as the interaction matrix. E1 was restricted to either Full-chromosome (top), Chromosome-arms (middle), or 10Mb windows (bottom).
</figcaption>
</figure>
</div>
</div>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-e1-matrix-full-arms-10mb-round-spermatid-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;12: Super caption, subcaptions should be moved here (from notebook). They now fit on the page, but it would be nice to make this as one plt.subplots with shared axis title etc. <em>Update:</em> The notebook is ready for making these plots (<code>07_various_plotting.ipynb</code>, with a new plotting function) should be same size as the following matrix plots.
</figcaption>
</figure>
</div>
<p>I argue that without more finescaled knowledge than the position of the centromeres, the arbitrary size of the 10 Mb windowed E1 can not fully be justified. Also, <span class="citation" data-cites="wang2019reprogrammingmeioticchromatin">Wang et al. (<a href="#ref-wang2019reprogrammingmeioticchromatin" role="doc-biblioref">2019</a>)</span> concludes that only the pachytene spermatocyte showed local interactions in that viewframe (what they refer to as <em>refined A/B-compartments</em> ), and all the other stages of spermatogenesis were consistent with the conventional A/B compartments. The reasonable thing to do is therefore to continue the analysis, focusing on the arms-restricted eigendecomposition. Nevertheless, we also keep <em>refined</em> compartments in the analysis.</p>
<p>Additionally, as I created coolers with two different sets of parsing parameters we will compare the resulting matrices and their compartments (<a href="#fig-rs100-recpe-pe" class="quarto-xref">Figure&nbsp;13</a>). As expected, we observe more empty bins in the Hi-C matrix, but otherwise, the interaction pattern is indestinguishable when going from the initial run (PE) to the recommended parameters (redPE). The effect on the E1 is more noticable, where the absolute magnitude of the E1 values is generally smaller. There is, however, a small region that changes sign (from A to B) on the 10Mb-windowed (‘refined’) E1 track (<a href="#fig-rs100-recpe-pe" class="quarto-xref">Figure&nbsp;13</a>;c+d, zoomed-in region). This region is surrounded by added empty bins, which could mean that too many low quality pairs in PE were introducing bias and swapped the sign of E1. It is supported by the fact that the sign change <em>only</em> occured in <em>refined</em> E1, and that the sign after filtering weak pairs (<span class="math inline">\(mapq &lt; 30\)</span>) is consistent with the <em>arms</em> view. It supports my previous postulate that it is better to use a viewframe where with explicit molecular meaning than one of an arbitrary window size. That said, the <code>mapq</code> threshold should really be determined taking both coverage and resolution into account. For our purposes, and with the <em>arms</em> view, the mapping- and parsing parameters do not seem to be too sensitive.</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/07_various_plotting.ipynb" data-notebook-title="Various Comparisons" data-notebook-cellid="cell-fig-rs100-recpe-pe">
<div class="cell" data-layout="[[-0.5,5,5,-0.5],[-0.5,5,5,-0.5]]" data-execution_count="42">
<div id="fig-rs100-recpe-pe" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-rs100-recpe-pe-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 4.6%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-rs100-recpe-pe" style="flex-basis: 45.5%;justify-content: flex-start;">
<div id="fig-rs100-recpe-pe-1" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-rs100-recpe-pe-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs100-recpe-pe-output-1.png" class="lightbox" data-gallery="fig-rs100-recpe-pe" title="Figure&nbsp;13&nbsp;(a): recPE RS 100kb full chrX"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs100-recpe-pe-output-1.png" data-ref-parent="fig-rs100-recpe-pe" width="247" height="195" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-rs100-recpe-pe-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) recPE RS 100kb full chrX
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-rs100-recpe-pe" style="flex-basis: 45.5%;justify-content: flex-start;">
<div id="fig-rs100-recpe-pe-2" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-rs100-recpe-pe-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs100-recpe-pe-output-2.png" class="lightbox" data-gallery="fig-rs100-recpe-pe" title="Figure&nbsp;13&nbsp;(b): PE RS 100kb full chrX"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs100-recpe-pe-output-2.png" data-ref-parent="fig-rs100-recpe-pe" width="247" height="195" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-rs100-recpe-pe-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) PE RS 100kb full chrX
</figcaption>
</figure>
</div>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 4.6%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
</div>
<div class="quarto-layout-row">
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 4.6%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-rs100-recpe-pe" style="flex-basis: 45.5%;justify-content: flex-start;">
<div id="fig-rs100-recpe-pe-3" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-rs100-recpe-pe-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs100-recpe-pe-output-3.png" class="lightbox" data-gallery="fig-rs100-recpe-pe" title="Figure&nbsp;13&nbsp;(c): recPE RS 100kb chrX:70Mb-78Mb"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs100-recpe-pe-output-3.png" data-ref-parent="fig-rs100-recpe-pe" width="247" height="201" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-rs100-recpe-pe-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(c) recPE RS 100kb chrX:70Mb-78Mb
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-rs100-recpe-pe" style="flex-basis: 45.5%;justify-content: flex-start;">
<div id="fig-rs100-recpe-pe-4" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-rs100-recpe-pe-4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs100-recpe-pe-output-4.png" class="lightbox" data-gallery="fig-rs100-recpe-pe" title="Figure&nbsp;13&nbsp;(d): PE RS 100kb chrX:70Mb-78Mb"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs100-recpe-pe-output-4.png" data-ref-parent="fig-rs100-recpe-pe" width="247" height="201" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-rs100-recpe-pe-4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(d) PE RS 100kb chrX:70Mb-78Mb
</figcaption>
</figure>
</div>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 4.6%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-rs100-recpe-pe-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;13: Round Spermatid (RS) at 100kb, comparing the impact of parsing parameters
</figcaption>
</figure>
</div>
</div>
</div>
<p>To emphasize the findings, the sets of A-compartments were compared between the two parsing runs, showing almost identical compartment calls. Additionally, the set difference was 8 bins between PE and recPE for round spermatid 100kb and 5 bins for fibroblast for <em>arms</em> viewframe (<a href="#fig-rs-fb-100-pe-recpe-intervals" class="quarto-xref">Figure&nbsp;14</a>; a+b, respectively). We observe a high number of differences around 76Mb for the refined compartments (10Mb) of round spermatid, which is consistent with the sign-flip of E1 values discussed earlier. Anything else would be surprising, as it is the same data, but visualized in a different way.</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/07_various_plotting.ipynb" data-notebook-title="Various Comparisons" data-notebook-cellid="cell-fig-rs-fb-100-pe-recpe-intervals">
<div class="cell" data-layout="[[-0.6,2.5,2.5,-0.6],[-0.6,2.5,2.5,-0.6]]" data-execution_count="103">
<div id="fig-rs-fb-100-pe-recpe-intervals" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-rs-fb-100-pe-recpe-intervals-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 9.7%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-rs-fb-100-pe-recpe-intervals" style="flex-basis: 40.3%;justify-content: flex-start;">
<div id="fig-rs-fb-100-pe-recpe-intervals-1" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-rs-fb-100-pe-recpe-intervals-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs-fb-100-pe-recpe-intervals-output-1.png" class="lightbox" data-gallery="fig-rs-fb-100-pe-recpe-intervals" title="Figure&nbsp;14&nbsp;(a): RS: arms"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs-fb-100-pe-recpe-intervals-output-1.png" data-ref-parent="fig-rs-fb-100-pe-recpe-intervals" width="249" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-rs-fb-100-pe-recpe-intervals-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) RS: arms
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-rs-fb-100-pe-recpe-intervals" style="flex-basis: 40.3%;justify-content: flex-start;">
<div id="fig-rs-fb-100-pe-recpe-intervals-2" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-rs-fb-100-pe-recpe-intervals-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs-fb-100-pe-recpe-intervals-output-2.png" class="lightbox" data-gallery="fig-rs-fb-100-pe-recpe-intervals" title="Figure&nbsp;14&nbsp;(b): Fib: arms"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs-fb-100-pe-recpe-intervals-output-2.png" data-ref-parent="fig-rs-fb-100-pe-recpe-intervals" width="249" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-rs-fb-100-pe-recpe-intervals-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) Fib: arms
</figcaption>
</figure>
</div>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 9.7%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
</div>
<div class="quarto-layout-row">
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 9.7%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-rs-fb-100-pe-recpe-intervals" style="flex-basis: 40.3%;justify-content: flex-start;">
<div id="fig-rs-fb-100-pe-recpe-intervals-3" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-rs-fb-100-pe-recpe-intervals-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs-fb-100-pe-recpe-intervals-output-3.png" class="lightbox" data-gallery="fig-rs-fb-100-pe-recpe-intervals" title="Figure&nbsp;14&nbsp;(c): RS: 10Mb"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs-fb-100-pe-recpe-intervals-output-3.png" data-ref-parent="fig-rs-fb-100-pe-recpe-intervals" width="249" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-rs-fb-100-pe-recpe-intervals-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(c) RS: 10Mb
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-rs-fb-100-pe-recpe-intervals" style="flex-basis: 40.3%;justify-content: flex-start;">
<div id="fig-rs-fb-100-pe-recpe-intervals-4" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-rs-fb-100-pe-recpe-intervals-4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs-fb-100-pe-recpe-intervals-output-4.png" class="lightbox" data-gallery="fig-rs-fb-100-pe-recpe-intervals" title="Figure&nbsp;14&nbsp;(d): Fib: 10Mb"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-rs-fb-100-pe-recpe-intervals-output-4.png" data-ref-parent="fig-rs-fb-100-pe-recpe-intervals" width="249" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-rs-fb-100-pe-recpe-intervals-4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(d) Fib: 10Mb
</figcaption>
</figure>
</div>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 9.7%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-rs-fb-100-pe-recpe-intervals-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;14: Round Spermatid (RS) and Fibroblast (Fb) at 100kb, comparing the impact of parsing parameters on A-compartment calling at different viewframes; <em>arms</em>, <em>10Mb</em>. PE: initial parse (masking complex walks); recPE: recommended parse (reporting the 5’most unique alignment of a complex walk).
</figcaption>
</figure>
</div>
</div>
</div>
<p>The observed difference between the sets can for our data be attributed to chance, but we cannot draw general conclusions about the parameters in general. I argue that the quality and size of the Hi-C library will influence sensitive to parsing parameters. In that case, the most flexible approach is still to follow the recommendations from <code>cooler</code> to report more pairs as valid contacts, and then create coolers with different <em>mapq</em> filters if issues are encountered.</p>
</section>
<section id="compartment-edges-transition-zones" class="level3">
<h3 class="anchored" data-anchor-id="compartment-edges-transition-zones">Compartment Edges (transition zones)</h3>
<p>We compare how the ECH90 regions fit when queried on top of the A-compartments and equivalently for the edges, for fibroblasts and round spermatids at 100kb resolution. When queried against the edges in stead, the the total set size is reduced to less than 50%. Interestingly, some of the intersections between A-compartments and ECH90 remain, and new ones appear as we move to the outside edge of the compartment (<a href="#fig-comps-edges-ech" class="quarto-xref">Figure&nbsp;15</a>). This indicates that most, but not all, of the intersection between ECH90 regions and the A-compartments are within 100kb of the compartment edge, and additional overlap is gained if we define a transition zone on the outside of the edge as well. To visualize this (outside) edge enrichment, we find the set difference of the ECH-intersection to compartments and edges, respectively (<a href="#fig-edge-enrichment" class="quarto-xref">Figure&nbsp;16</a>), thus removing all the ‘inside’ edges. We observe that in almost all of the of the regions of <span class="math inline">\(ECH \cap Comp\)</span> are accompanied by an edge also intersecting ECH (<span class="math inline">\(ECH \cap Edge\)</span>), localized where the <em>Diff</em> track aligns (within 100kb) with both <span class="math inline">\(CompInt\)</span> and <span class="math inline">\(EdgeInt\)</span>.</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/07_various_plotting.ipynb" data-notebook-title="Various Comparisons" data-notebook-cellid="cell-fig-comps-edges-ech">
<div class="cell" data-layout="[[-0.6,2.5,2.5,-0.6],[-0.6,2.5,2.5,-0.6]]">
<div id="fig-comps-edges-ech" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-comps-edges-ech-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 9.7%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-comps-edges-ech" style="flex-basis: 40.3%;justify-content: flex-start;">
<div id="fig-comps-edges-ech-1" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-comps-edges-ech-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-comps-edges-ech-output-1.png" class="lightbox" data-gallery="fig-comps-edges-ech" title="Figure&nbsp;15&nbsp;(a): Fibroblast A-compartments"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-comps-edges-ech-output-1.png" data-ref-parent="fig-comps-edges-ech" width="248" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-comps-edges-ech-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) Fibroblast A-compartments
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-comps-edges-ech" style="flex-basis: 40.3%;justify-content: flex-start;">
<div id="fig-comps-edges-ech-2" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-comps-edges-ech-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-comps-edges-ech-output-2.png" class="lightbox" data-gallery="fig-comps-edges-ech" title="Figure&nbsp;15&nbsp;(b): Round Spermatid A-compartments"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-comps-edges-ech-output-2.png" data-ref-parent="fig-comps-edges-ech" width="248" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-comps-edges-ech-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) Round Spermatid A-compartments
</figcaption>
</figure>
</div>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 9.7%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
</div>
<div class="quarto-layout-row">
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 9.7%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-comps-edges-ech" style="flex-basis: 40.3%;justify-content: flex-start;">
<div id="fig-comps-edges-ech-3" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-comps-edges-ech-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-comps-edges-ech-output-3.png" class="lightbox" data-gallery="fig-comps-edges-ech" title="Figure&nbsp;15&nbsp;(c): Fibroblast edges"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-comps-edges-ech-output-3.png" data-ref-parent="fig-comps-edges-ech" width="248" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-comps-edges-ech-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(c) Fibroblast edges
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-comps-edges-ech" style="flex-basis: 40.3%;justify-content: flex-start;">
<div id="fig-comps-edges-ech-4" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-comps-edges-ech-4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-comps-edges-ech-output-4.png" class="lightbox" data-gallery="fig-comps-edges-ech" title="Figure&nbsp;15&nbsp;(d): Round Spermatid edges"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-comps-edges-ech-output-4.png" data-ref-parent="fig-comps-edges-ech" width="248" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-comps-edges-ech-4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(d) Round Spermatid edges
</figcaption>
</figure>
</div>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 9.7%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-comps-edges-ech-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;15: Visual representation of the genomic intervals of ECH90, A-compartments (a+b), edges (c+d), and their intersections. Shown fibroblast (a+c) and round spermatid (b+d) at 100kb resolution and arms viewframe.
</figcaption>
</figure>
</div>
</div>
</div>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/07_various_plotting.ipynb" data-notebook-title="Various Comparisons" data-notebook-cellid="cell-fig-edge-enrichment">
<div id="fig-edge-enrichment" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-edge-enrichment-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-edge-enrichment" style="flex-basis: 50.0%;justify-content: flex-start;">
<div id="fig-edge-enrichment-1" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-edge-enrichment-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-edge-enrichment-output-1.png" class="lightbox" data-gallery="fig-edge-enrichment" title="Figure&nbsp;16&nbsp;(a): Round Spermatid"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-edge-enrichment-output-1.png" data-ref-parent="fig-edge-enrichment" width="248" height="98" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-edge-enrichment-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) Round Spermatid
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-edge-enrichment" style="flex-basis: 50.0%;justify-content: flex-start;">
<div id="fig-edge-enrichment-2" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-edge-enrichment-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-edge-enrichment-output-2.png" class="lightbox" data-gallery="fig-edge-enrichment" title="Figure&nbsp;16&nbsp;(b): Fibroblast"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-edge-enrichment-output-2.png" data-ref-parent="fig-edge-enrichment" width="248" height="98" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-edge-enrichment-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) Fibroblast
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-edge-enrichment-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;16: Visual representation of the enrichment of edges in the intersection of ECH90 and A-compartments. Shown round spermatid (a) and fibroblast (b) at 100kb resolution and arms viewframe. Note that the edge-regions are too small to be distinguished visually from the compartment on the graph, making it look like they overlap, even though the difference is reported.
</figcaption>
</figure>
</div>
</div>
<p>We apply both proximity test and Jaccard test, to see how well the results could occur by chance (<a href="#fig-proximity-jaccard-bar" class="quarto-xref">Figure&nbsp;17</a>). For completeness, the tests are included for all cell types, but we only use 100kb resolution arms viewframe. We observe that both fibroblast and round spermatid have <span class="math inline">\(p &lt; 0.05\)</span> for both tests, meaning the two cell type have both more intersection with ECH regions than expected by chance (Jaccard) <em>and</em> the non-overlapping intervals are more proximal to compartment edges than expected by chance (proximity test). I argue that a significant Jaccard statistic should be interpreted as a significant amount of overlap between the two sets, i.e.&nbsp;compartment edges and ECH90 regions, and the proximity test (when performed on the edges) gives us information about the potential of expanding or moving the transition window. That is, if the non-overlapping regions are <em>very</em> proximal, a larger (or shifted) window to only capture the 200kb region outside of the edge might be favourable.</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/06_rec_genomicintervals.ipynb" data-notebook-title="E1 vs. ECH (recommended params)" data-notebook-cellid="cell-fig-proximity-jaccard-bar">
<div id="cell-fig-proximity-jaccard-bar" class="cell" data-execution_count="59">
<div class="cell-output cell-output-display">
<div id="fig-proximity-jaccard-bar" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-proximity-jaccard-bar-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-06_rec_genomicintervals-fig-proximity-jaccard-bar-output-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-43" title="Figure&nbsp;17: Proximity and Jaccard index p-values for ECH90 regions on compartment edges for all cell types at 100kb resolution at arms view p=0.05 is marked as a red line."><img src="index_files/figure-html/..-notebooks-06_rec_genomicintervals-fig-proximity-jaccard-bar-output-1.png" width="298" height="197" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-proximity-jaccard-bar-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;17: Proximity and Jaccard index p-values for ECH90 regions on compartment edges for all cell types at 100kb resolution at arms view <span class="math inline">\(p=0.05\)</span> is marked as a red line.
</figcaption>
</figure>
</div>
</div>
</div>
</div>
</section>
</section>
<section id="testing-against-hybrid-incompatibility-in-baboons" class="level2">
<h2 class="anchored" data-anchor-id="testing-against-hybrid-incompatibility-in-baboons">Testing against hybrid incompatibility in baboons</h2>
<p>Initially, the compartment edges of round spermatid at 100kb resolution (RS100) were plotted against the lifted coordinates from a <em>P.</em> anubis-hamadryas hybrid population, where either all the sampled individuals have <em>Papio anubis</em> ancestry or 95% of the sampled individuals have <em>Papio hamadryas</em> ancestry. Their respective intersections were plotted undeneath. We expect less intersection for <em>hamadryas</em> than for <em>anubis</em> as the total set size is much smaller. The compartment edges and <em>Papio anubis</em>-derived regions(<a href="#fig-baboon-rs100-intersect" class="quarto-xref">Figure&nbsp;18</a>; b) seem to be highly enriched in the first 25 Mbp, and thus it has a high degree of intersection with the compartment edges. Interestingly, the ECH90 set is nearly empty in that region, making the finding outside the scope of this analysis, although it could be useful for determining the mechanism for selecting the <em>P.anubis</em> ancestral allele in the hybrid baboon population. The <em>P.hamadryas</em>-derived regions seem intersect the compartment edges more centered on the chromosome (<a href="#fig-baboon-rs100-intersect" class="quarto-xref">Figure&nbsp;18</a>; a). The proximity test ruled out that the non-intersecting parts of the respective regions were this proximal by chance. However, the Jaccard test revealed that the intersection between the RS100 and both Hi-<em>P.hama</em> and Hi-<em>P.anu</em> can be explained by chance alone .</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/07_various_plotting.ipynb" data-notebook-title="Various Comparisons" data-notebook-cellid="cell-fig-baboon-rs100-intersect">
<div class="cell" data-layout="[[-0.6,2.5,2.5,-0.6],[-0.6,2.5,2.5,-0.6]]" data-execution_count="21">
<div id="fig-baboon-rs100-intersect" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-baboon-rs100-intersect-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 9.7%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-baboon-rs100-intersect" style="flex-basis: 40.3%;justify-content: flex-start;">
<div id="fig-baboon-rs100-intersect-1" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-baboon-rs100-intersect-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-baboon-rs100-intersect-output-1.png" class="lightbox" data-gallery="fig-baboon-rs100-intersect" title="Figure&nbsp;18&nbsp;(a): P. hamadryas (arms view)"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-baboon-rs100-intersect-output-1.png" data-ref-parent="fig-baboon-rs100-intersect" width="248" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-baboon-rs100-intersect-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) <em>P. hamadryas</em> (arms view)
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-baboon-rs100-intersect" style="flex-basis: 40.3%;justify-content: flex-start;">
<div id="fig-baboon-rs100-intersect-2" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-baboon-rs100-intersect-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-baboon-rs100-intersect-output-2.png" class="lightbox" data-gallery="fig-baboon-rs100-intersect" title="Figure&nbsp;18&nbsp;(b): P. anubis (arms view)"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-baboon-rs100-intersect-output-2.png" data-ref-parent="fig-baboon-rs100-intersect" width="248" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-baboon-rs100-intersect-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) <em>P. anubis</em> (arms view)
</figcaption>
</figure>
</div>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 9.7%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
</div>
<div class="quarto-layout-row">
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 9.7%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-baboon-rs100-intersect" style="flex-basis: 40.3%;justify-content: flex-start;">
<div id="fig-baboon-rs100-intersect-3" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-baboon-rs100-intersect-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-baboon-rs100-intersect-output-3.png" class="lightbox" data-gallery="fig-baboon-rs100-intersect" title="Figure&nbsp;18&nbsp;(c): P. hamadryas (10Mb view)"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-baboon-rs100-intersect-output-3.png" data-ref-parent="fig-baboon-rs100-intersect" width="248" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-baboon-rs100-intersect-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(c) <em>P. hamadryas</em> (10Mb view)
</figcaption>
</figure>
</div>
</div>
<div class="cell-output cell-output-display quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-baboon-rs100-intersect" style="flex-basis: 40.3%;justify-content: flex-start;">
<div id="fig-baboon-rs100-intersect-4" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-baboon-rs100-intersect-4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/..-notebooks-07_various_plotting-fig-baboon-rs100-intersect-output-4.png" class="lightbox" data-gallery="fig-baboon-rs100-intersect" title="Figure&nbsp;18&nbsp;(d): P. anubis (10Mb view)"><img src="index_files/figure-html/..-notebooks-07_various_plotting-fig-baboon-rs100-intersect-output-4.png" data-ref-parent="fig-baboon-rs100-intersect" width="248" height="77" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-baboon-rs100-intersect-4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(d) <em>P. anubis</em> (10Mb view)
</figcaption>
</figure>
</div>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 9.7%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-baboon-rs100-intersect-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;18: Comparing the A-compartments of round spermatid (RS) with regions in baboons from hybrid population where a) 95% of sampled individuals have <em>Papio hamadryas</em> ancestry or b) 100% of the samples have <em>Papio anubis</em> ancestry. The regions are extracted and lifted from panu4.0 reference to rhemac10, then collapsed to non-overlapping intervals with <code>genominterv</code>.
</figcaption>
</figure>
</div>
</div>
</div>
<div style="page-break-after: always;"></div>
</section>
</section>
<section id="discussion" class="level1">
<h1>Discussion</h1>
<p>Here is the discussion</p>
</section>
<section id="bibliography" class="level1 unnumbered">
<h1 class="unnumbered">Bibliography</h1>
<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-bicciato2022hicdataanalysis" class="csl-entry" role="listitem">
Bicciato, Silvio, and Francesco Ferrari, eds. 2022. <em>Hi-<span>C Data Analysis</span>: <span>Methods</span> and <span>Protocols</span></em>. Vol. 2301. Methods in <span>Molecular Biology</span>. New York, NY: Springer US. <a href="https://doi.org/10.1007/978-1-0716-1390-0">https://doi.org/10.1007/978-1-0716-1390-0</a>.
</div>
<div id="ref-bravonunez2018geneticvillainskiller" class="csl-entry" role="listitem">
Bravo Núñez, María Angélica, Nicole L. Nuckolls, and Sarah E. Zanders. 2018. <span>“Genetic <span>Villains</span>: <span>Killer Meiotic Drivers</span>.”</span> <em>Trends in Genetics</em> 34 (6): 424–33. <a href="https://doi.org/10.1016/j.tig.2018.02.003">https://doi.org/10.1016/j.tig.2018.02.003</a>.
</div>
<div id="ref-devteam2024sratoolkit" class="csl-entry" role="listitem">
DevTeam, SRA Toolkit. 2024. <span>“<span>SRA Toolkit</span>.”</span> Wiki. <em>GitHub</em>. https://github.com/ncbi/sra-tools/wiki/01.-Downloading-SRA-Toolkit. <a href="https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=software">https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=software</a>.
</div>
<div id="ref-imakaev2012iterativecorrectionhic" class="csl-entry" role="listitem">
Imakaev, Maxim, Geoffrey Fudenberg, Rachel Patton McCord, Natalia Naumova, Anton Goloborodko, Bryan R Lajoie, Job Dekker, and Leonid A Mirny. 2012. <span>“Iterative Correction of <span>Hi-C</span> Data Reveals Hallmarks of Chromosome Organization.”</span> <em>Nature Methods</em> 9 (10): 999–1003. <a href="https://doi.org/10.1038/nmeth.2148">https://doi.org/10.1038/nmeth.2148</a>.
</div>
<div id="ref-jaenike2001sexchromosomemeiotic" class="csl-entry" role="listitem">
Jaenike, John. 2001. <span>“Sex <span>Chromosome Meiotic Drive</span>.”</span> <em>Annual Review of Ecology and Systematics</em> 32 (1): 25–49. <a href="https://doi.org/10.1146/annurev.ecolsys.32.081501.113958">https://doi.org/10.1146/annurev.ecolsys.32.081501.113958</a>.
</div>
<div id="ref-lajoie2015hitchhikersguidehic" class="csl-entry" role="listitem">
Lajoie, Bryan R., Job Dekker, and Noam Kaplan. 2015. <span>“The <span>Hitchhiker</span>’s <span>Guide</span> to <span>Hi-C Analysis</span>: <span>Practical</span> Guidelines.”</span> <em>Methods (San Diego, Calif.)</em> 72 (January): 65–75. <a href="https://doi.org/10.1016/j.ymeth.2014.10.031">https://doi.org/10.1016/j.ymeth.2014.10.031</a>.
</div>
<div id="ref-lieberman-aiden2009comprehensivemappinglongrange" class="csl-entry" role="listitem">
Lieberman-Aiden, Erez, Nynke L. Van Berkum, Louise Williams, Maxim Imakaev, Tobias Ragoczy, Agnes Telling, Ido Amit, et al. 2009. <span>“Comprehensive <span>Mapping</span> of <span>Long-Range Interactions Reveals Folding Principles</span> of the <span>Human Genome</span>.”</span> <em>Science</em> 326 (5950): 289–93. <a href="https://doi.org/10.1126/science.1181369">https://doi.org/10.1126/science.1181369</a>.
</div>
<div id="ref-munch2024munchgroup" class="csl-entry" role="listitem">
Munch, Kasper. 2024. <span>“Munch-Group.”</span> https://munch-group.org/research.html.
</div>
<div id="ref-openchromosomecollective" class="csl-entry" role="listitem">
<span>“Open <span>Chromosome Collective</span> (<span>Open2C</span>).”</span> n.d. Resource. <em>Open2C</em>. https://open2c.github.io/. Accessed November 13, 2024.
</div>
<div id="ref-skov2023extraordinaryselectionhuman" class="csl-entry" role="listitem">
Skov, Laurits, Moisès Coll Macià, Elise Anne Lucotte, Maria Izabel Alves Cavassim, David Castellano, Mikkel Heide Schierup, and Kasper Munch. 2023. <span>“Extraordinary Selection on the Human <span>X</span> Chromosome Associated with Archaic Admixture.”</span> <em>Cell Genomics</em> 3 (3): 100274. <a href="https://doi.org/10.1016/j.xgen.2023.100274">https://doi.org/10.1016/j.xgen.2023.100274</a>.
</div>
<div id="ref-sorensen2023genomewidecoancestryreveals" class="csl-entry" role="listitem">
Sørensen, Erik F., R. Alan Harris, Liye Zhang, Muthuswamy Raveendran, Lukas F. K. Kuderna, Jerilyn A. Walker, Jessica M. Storer, et al. 2023. <span>“Genome-Wide Coancestry Reveals Details of Ancient and Recent Male-Driven Reticulation in Baboons.”</span> <em>Science</em> 380 (6648): eabn8153. <a href="https://doi.org/10.1126/science.abn8153">https://doi.org/10.1126/science.abn8153</a>.
</div>
<div id="ref-wang2019reprogrammingmeioticchromatin" class="csl-entry" role="listitem">
Wang, Yao, Hanben Wang, Yu Zhang, Zhenhai Du, Wei Si, Suixing Fan, Dongdong Qin, et al. 2019. <span>“Reprogramming of <span>Meiotic Chromatin Architecture</span> During <span>Spermatogenesis</span>.”</span> <em>Molecular Cell</em> 73 (3): 547–561.e6. <a href="https://doi.org/10.1016/j.molcel.2018.11.019">https://doi.org/10.1016/j.molcel.2018.11.019</a>.
</div>
<div id="ref-warren2020sequencediversityanalyses" class="csl-entry" role="listitem">
Warren, Wesley C., R. Alan Harris, Marina Haukness, Ian T. Fiddes, Shwetha C. Murali, Jason Fernandes, Philip C. Dishuck, et al. 2020. <span>“Sequence Diversity Analyses of an Improved Rhesus Macaque Genome Enhance Its Biomedical Utility.”</span> <em>Science</em> 370 (6523): eabc6617. <a href="https://doi.org/10.1126/science.abc6617">https://doi.org/10.1126/science.abc6617</a>.
</div>
</div>
</section>


<div id="quarto-appendix" class="default"><section id="footnotes" class="footnotes footnotes-end-of-document" role="doc-endnotes"><h2 class="anchored quarto-appendix-heading">Footnotes</h2>

<ol>
<li id="fn1"><p><span class="citation" data-cites="bicciato2022hicdataanalysis">Bicciato and Ferrari (<a href="#ref-bicciato2022hicdataanalysis" role="doc-biblioref">2022</a>)</span> mentions that this type of ligations had been used to infer interaction between sister-chromatids post-replication in <em>Drosophila</em>.<a href="#fnref1" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
</ol>
</section></div></main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<script>var lightboxQuarto = GLightbox({"closeEffect":"zoom","descPosition":"bottom","loop":false,"openEffect":"zoom","selector":".lightbox"});
(function() {
  let previousOnload = window.onload;
  window.onload = () => {
    if (previousOnload) {
      previousOnload();
    }
    lightboxQuarto.on('slide_before_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      const href = trigger.getAttribute('href');
      if (href !== null) {
        const imgEl = window.document.querySelector(`a[href="${href}"] img`);
        if (imgEl !== null) {
          const srcAttr = imgEl.getAttribute("src");
          if (srcAttr && srcAttr.startsWith("data:")) {
            slideConfig.href = srcAttr;
          }
        }
      } 
    });
  
    lightboxQuarto.on('slide_after_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(slideNode);
      }
    });
  
  };
  
})();
          </script>




</body></html>