<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
    <meta charset="utf-8">
    <meta name="generator" content="quarto-1.6.38">

    <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


    <title>methods</title>
    <style>
      code{white-space: pre-wrap;}
      span.smallcaps{font-variant: small-caps;}
      div.columns{display: flex; gap: min(4vw, 1.5em);}
      div.column{flex: auto; overflow-x: auto;}
      div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
      ul.task-list{list-style: none;}
      ul.task-list li input[type="checkbox"] {
        width: 0.8em;
        margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
        vertical-align: middle;
      }
      /* CSS for syntax highlighting */
      pre > code.sourceCode { white-space: pre; position: relative; }
      pre > code.sourceCode > span { line-height: 1.25; }
      pre > code.sourceCode > span:empty { height: 1.2em; }
      .sourceCode { overflow: visible; }
      code.sourceCode > span { color: inherit; text-decoration: inherit; }
      div.sourceCode { margin: 1em 0; }
      pre.sourceCode { margin: 0; }
      @media screen {
      div.sourceCode { overflow: auto; }
      }
      @media print {
      pre > code.sourceCode { white-space: pre-wrap; }
      pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
      }
      pre.numberSource code
        { counter-reset: source-line 0; }
      pre.numberSource code > span
        { position: relative; left: -4em; counter-increment: source-line; }
      pre.numberSource code > span > a:first-child::before
        { content: counter(source-line);
          position: relative; left: -1em; text-align: right; vertical-align: baseline;
          border: none; display: inline-block;
          -webkit-touch-callout: none; -webkit-user-select: none;
          -khtml-user-select: none; -moz-user-select: none;
          -ms-user-select: none; user-select: none;
          padding: 0 4px; width: 4em;
        }
      pre.numberSource { margin-left: 3em;  padding-left: 4px; }
      div.sourceCode
        {   }
      @media screen {
      pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
      }
      /* CSS for citations */
      div.csl-bib-body { }
      div.csl-entry {
        clear: both;
        margin-bottom: 0em;
      }
      .hanging-indent div.csl-entry {
        margin-left:2em;
        text-indent:-2em;
      }
      div.csl-left-margin {
        min-width:2em;
        float:left;
      }
      div.csl-right-inline {
        margin-left:2em;
        padding-left:1em;
      }
      div.csl-indent {
        margin-left: 2em;
      }    </style>

    <style>
      body.hypothesis-enabled #quarto-embed-header {
        padding-right: 36px;
      }

      #quarto-embed-header {
        height: 3em;
        width: 100%;
        display: flex;
        justify-content: space-between;
        align-items: center;
        border-bottom: solid 1px;
      }

      #quarto-embed-header h6 {
        font-size: 1.1em;
        padding-top: 0.6em;
        margin-left: 1em;
        margin-right: 1em;
        font-weight: 400;
      }

      #quarto-embed-header a.quarto-back-link,
      #quarto-embed-header a.quarto-download-embed {
        font-size: 0.8em;
        margin-top: 1em;
        margin-bottom: 1em;
        margin-left: 1em;
        margin-right: 1em;
      }

      .quarto-back-container {
        padding-left: 0.5em;
        display: flex;
      }

      .headroom {
          will-change: transform;
          transition: transform 200ms linear;
      }

      .headroom--pinned {
          transform: translateY(0%);
      }

      .headroom--unpinned {
          transform: translateY(-100%);
      }      
    </style>

    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js" integrity="sha512-bLT0Qm9VnAYZDflyKcBaQ2gg0hSYNQrJ8RilYldYQ1FxQYoCLtUjuuRuZo+fjqhx/qtq/1itJ0C2ejDxltZVFg==" crossorigin="anonymous"></script><script>
    window.document.addEventListener("DOMContentLoaded", function () {

      var header = window.document.querySelector("#quarto-embed-header");
      const titleBannerEl = window.document.querySelector("body > #title-block-header");
      if (titleBannerEl) {
        titleBannerEl.style.paddingTop = header.clientHeight + "px";
      }
      const contentEl = window.document.getElementById('quarto-content');
      for (const child of contentEl.children) {
        child.style.paddingTop = header.clientHeight + "px";
        child.style.marginTop = "1em";
      }

      // Use the article root if the `back` call doesn't work. This isn't perfect
      // but should typically work
      window.quartoBackToArticle = () => {
        var currentUrl = window.location.href;
        window.history.back();
        setTimeout(() => {
            // if location was not changed in 100 ms, then there is no history back
            if(currentUrl === window.location.href){              
                // redirect to site root
                window.location.href = "index.html";
            }
        }, 100);
      }

      const headroom = new window.Headroom(header, {
        tolerance: 5,
        onPin: function () {
        },
        onUnpin: function () {
        },
      });
      headroom.init();
    });
    </script>

    
<script src="site_libs/manuscript-notebook/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-b663525e4b92bb903b2199e2d401bf36.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-86e67f1fcd594f562d62505f92e4b4dd.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
     <script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" integrity="sha512-c3Nl8+7g4LMSTdrm621y7kf9v3SDPnhxLNhcjFJbKECVnmZHTdo+IRO05sNLTH/D3vA6u1X32ehoLC7WFVdheg==" crossorigin="anonymous"></script>

<script type="application/javascript">define('jquery', [],function() {return window.jQuery;})</script>  
      </head>

  <body class="quarto-notebook">
    <div id="quarto-embed-header" class="headroom fixed-top bg-primary">
      
      <a onclick="window.quartoBackToArticle(); return false;" class="btn btn-primary quarto-back-link" href=""><i class="bi bi-caret-left"></i> Back to Article</a>
      <h6><i class="bi bi-journal-code"></i> Methods</h6>

            <a href="./methods.qmd" class="btn btn-primary quarto-download-embed" download="methods.qmd">Download Source</a>
          </div>

     <header id="title-block-header" class="quarto-title-block"></header><div id="quarto-content" class="page-columns page-rows-contents page-layout-article toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#methods" id="toc-methods" class="nav-link active" data-scroll-target="#methods">Methods</a>
  <ul class="collapse">
  <li><a href="#downloading-data-and-project-structure" id="toc-downloading-data-and-project-structure" class="nav-link" data-scroll-target="#downloading-data-and-project-structure">Downloading Data and Project Structure</a></li>
  <li><a href="#handling-coolers-or-preparing-coolers" id="toc-handling-coolers-or-preparing-coolers" class="nav-link" data-scroll-target="#handling-coolers-or-preparing-coolers">Handling coolers (Or: preparing coolers)</a></li>
  </ul></li>
  </ul>
</nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
</div>
<main class="content quarto-banner-title-block" id="quarto-document-content">      <section id="methods" class="level2">
<h2 class="anchored" data-anchor-id="methods">Methods</h2>
<p>In this project, we formulate two objectives:</p>
<p><strong>A</strong>: Reproduce the Hi-C interaction maps and eigendecomposition from <span class="citation" data-cites="wang_reprogramming_2019">(<a href="#ref-wang_reprogramming_2019" role="doc-biblioref">Wang et al. 2019</a>)</span>, with some modifications. We briefly use <em>HiCExplorer</em>, but change the analyses to use the <em>Open2C Ecosystem</em> <span class="citation" data-cites="open2c">(<a href="#ref-open2c" role="doc-biblioref">Open Chromosome Collective 2024</a>)</span> which have a Pyton API as well as command-line functions, which can be paired very well with Jupyter Notebooks. The majority of the data analysis was run with a <em>gwf</em> workflow, and the commands that were visually inspected were run in Jupyter Notebooks.</p>
<p><strong>B</strong> Compare with regions of selection that are found in <em>papio anubis</em>, and maybe in <em>human</em> too. Investigate the biological meaning of the results.</p>
<p>All computations were performed on GenomeDK (GDK) [ref], an HPC cluster located on Aarhus Uninversity, and most of the processing of the data was nested into a <em>gwf</em> workflow [ref], a workflow manager developed at GDK. I would like to thank GDK and Aarhus University for providing computational resources and support that contributed to these research results.</p>
<p>The whole of this project is carried out with reproducibility in mind, so an effort (and quite a significant amount of time) has been put into documenting code and organizing the project for readbility and transparency through a Quarto project [ref]. Therefore, all code, virtual environments and text is made available as a Quarto book, rendered directly from the GitHub repository with GitHub Pages []. To make this possible, the Quarto documentation has been extensively studied and discussed with <em>KMT</em> [ref, aknowledge].</p>
<section id="downloading-data-and-project-structure" class="level3">
<h3 class="anchored" data-anchor-id="downloading-data-and-project-structure">Downloading Data and Project Structure</h3>
<p>To reproduce the results from <span class="citation" data-cites="wang_reprogramming_2019">(<a href="#ref-wang_reprogramming_2019" role="doc-biblioref">Wang et al. 2019</a>)</span>, I chose to use their raw data directly from the SRA portal [ref]. I filtered the data to contain all their paired-end Hi-C reads, and included only macaque samples. The data set also contains RNAseq data, and the same tissues for both macaque and mouse. The meta data for the data set was extracted into a runtable <code>SRA-runtable.tsv</code>. To get an overview of the data accessions used in this analysis, we will first summarize the runtable that contains the accession numbers and some metadata for each sample (<span class="citation" data-cites="tbl-runtable">(<a href="#ref-tbl-runtable" role="doc-biblioref"><strong>tbl-runtable?</strong></a>)</span>). It adds up to ~1Tb of compressed <code>fastq</code> files, holding ~9.5 billion reads, roughly evenly spread on the 5 tissue types.</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/hic-spermatogenesis/hic-spermatogenesis/notebooks/03_compartments.ipynb" data-notebook-title="Compartments Analysis" data-notebook-cellid="cell-tbl-runtable">
<div class="cell-container"><div class="cell-decorator"><pre>In [4]:</pre></div><div class="cell" data-execution_count="4">

<div class="cell-container"><div class="cell-decorator"><pre>In [4]:</pre></div><div><div class="sourceCode cell-code" id="cb1"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>display(df1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div><div id="tbl-runtable" class="cell quarto-float quarto-figure quarto-figure-center anchored" data-execution_count="4">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-runtable-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
The most relevant columns of the <code>SRA-runtable.tsv</code> file
</figcaption>
<div aria-describedby="tbl-runtable-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output cell-output-display">
<style type="text/css">
</style>

<table id="T_c1ff2" class="do-not-create-environment cell caption-top table table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th class="blank level0" data-quarto-table-cell-role="th">&nbsp;</th>
<th id="T_c1ff2_level0_col0" class="col_heading level0 col0" data-quarto-table-cell-role="th">source_name</th>
<th id="T_c1ff2_level0_col1" class="col_heading level0 col1" data-quarto-table-cell-role="th">BioSample</th>
<th id="T_c1ff2_level0_col2" class="col_heading level0 col2" data-quarto-table-cell-role="th">Run</th>
<th id="T_c1ff2_level0_col3" class="col_heading level0 col3" data-quarto-table-cell-role="th">GB</th>
<th id="T_c1ff2_level0_col4" class="col_heading level0 col4" data-quarto-table-cell-role="th">Bases</th>
<th id="T_c1ff2_level0_col5" class="col_heading level0 col5" data-quarto-table-cell-role="th">Reads</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td id="T_c1ff2_level0_row0" class="row_heading level0 row0" data-quarto-table-cell-role="th">16</td>
<td id="T_c1ff2_row0_col0" class="data row0 col0">fibroblast</td>
<td id="T_c1ff2_row0_col1" class="data row0 col1">SAMN08375237</td>
<td id="T_c1ff2_row0_col2" class="data row0 col2">SRR6502335</td>
<td id="T_c1ff2_row0_col3" class="data row0 col3">29.771059</td>
<td id="T_c1ff2_row0_col4" class="data row0 col4">73,201,141,800</td>
<td id="T_c1ff2_row0_col5" class="data row0 col5">244,003,806</td>
</tr>
<tr class="even">
<td id="T_c1ff2_level0_row1" class="row_heading level0 row1" data-quarto-table-cell-role="th">17</td>
<td id="T_c1ff2_row1_col0" class="data row1 col0">fibroblast</td>
<td id="T_c1ff2_row1_col1" class="data row1 col1">SAMN08375237</td>
<td id="T_c1ff2_row1_col2" class="data row1 col2">SRR6502336</td>
<td id="T_c1ff2_row1_col3" class="data row1 col3">22.755361</td>
<td id="T_c1ff2_row1_col4" class="data row1 col4">65,119,970,100</td>
<td id="T_c1ff2_row1_col5" class="data row1 col5">217,066,567</td>
</tr>
<tr class="odd">
<td id="T_c1ff2_level0_row2" class="row_heading level0 row2" data-quarto-table-cell-role="th">18</td>
<td id="T_c1ff2_row2_col0" class="data row2 col0">fibroblast</td>
<td id="T_c1ff2_row2_col1" class="data row2 col1">SAMN08375236</td>
<td id="T_c1ff2_row2_col2" class="data row2 col2">SRR6502337</td>
<td id="T_c1ff2_row2_col3" class="data row2 col3">21.434722</td>
<td id="T_c1ff2_row2_col4" class="data row2 col4">52,769,196,300</td>
<td id="T_c1ff2_row2_col5" class="data row2 col5">175,897,321</td>
</tr>
<tr class="even">
<td id="T_c1ff2_level0_row3" class="row_heading level0 row3" data-quarto-table-cell-role="th">19</td>
<td id="T_c1ff2_row3_col0" class="data row3 col0">fibroblast</td>
<td id="T_c1ff2_row3_col1" class="data row3 col1">SAMN08375236</td>
<td id="T_c1ff2_row3_col2" class="data row3 col2">SRR6502338</td>
<td id="T_c1ff2_row3_col3" class="data row3 col3">21.420030</td>
<td id="T_c1ff2_row3_col4" class="data row3 col4">52,378,949,100</td>
<td id="T_c1ff2_row3_col5" class="data row3 col5">174,596,497</td>
</tr>
<tr class="odd">
<td id="T_c1ff2_level0_row4" class="row_heading level0 row4" data-quarto-table-cell-role="th">20</td>
<td id="T_c1ff2_row4_col0" class="data row4 col0">fibroblast</td>
<td id="T_c1ff2_row4_col1" class="data row4 col1">SAMN08375236</td>
<td id="T_c1ff2_row4_col2" class="data row4 col2">SRR6502339</td>
<td id="T_c1ff2_row4_col3" class="data row4 col3">10.207410</td>
<td id="T_c1ff2_row4_col4" class="data row4 col4">28,885,941,600</td>
<td id="T_c1ff2_row4_col5" class="data row4 col5">96,286,472</td>
</tr>
<tr class="even">
<td id="T_c1ff2_level0_row5" class="row_heading level0 row5" data-quarto-table-cell-role="th">9</td>
<td id="T_c1ff2_row5_col0" class="data row5 col0">fibroblast</td>
<td id="T_c1ff2_row5_col1" class="data row5 col1">SAMN08375237</td>
<td id="T_c1ff2_row5_col2" class="data row5 col2">SRR7349189</td>
<td id="T_c1ff2_row5_col3" class="data row5 col3">52.729173</td>
<td id="T_c1ff2_row5_col4" class="data row5 col4">139,604,854,200</td>
<td id="T_c1ff2_row5_col5" class="data row5 col5">465,349,514</td>
</tr>
<tr class="odd">
<td id="T_c1ff2_level0_row6" class="row_heading level0 row6" data-quarto-table-cell-role="th">10</td>
<td id="T_c1ff2_row6_col0" class="data row6 col0">fibroblast</td>
<td id="T_c1ff2_row6_col1" class="data row6 col1">SAMN08375236</td>
<td id="T_c1ff2_row6_col2" class="data row6 col2">SRR7349190</td>
<td id="T_c1ff2_row6_col3" class="data row6 col3">53.085520</td>
<td id="T_c1ff2_row6_col4" class="data row6 col4">142,008,353,400</td>
<td id="T_c1ff2_row6_col5" class="data row6 col5">473,361,178</td>
</tr>
<tr class="even">
<td id="T_c1ff2_level0_row7" class="row_heading level0 row7" data-quarto-table-cell-role="th">21</td>
<td id="T_c1ff2_row7_col0" class="data row7 col0">pachytene spermatocyte</td>
<td id="T_c1ff2_row7_col1" class="data row7 col1">SAMN08375234</td>
<td id="T_c1ff2_row7_col2" class="data row7 col2">SRR6502342</td>
<td id="T_c1ff2_row7_col3" class="data row7 col3">60.258880</td>
<td id="T_c1ff2_row7_col4" class="data row7 col4">150,370,993,500</td>
<td id="T_c1ff2_row7_col5" class="data row7 col5">501,236,645</td>
</tr>
<tr class="odd">
<td id="T_c1ff2_level0_row8" class="row_heading level0 row8" data-quarto-table-cell-role="th">22</td>
<td id="T_c1ff2_row8_col0" class="data row8 col0">pachytene spermatocyte</td>
<td id="T_c1ff2_row8_col1" class="data row8 col1">SAMN08375234</td>
<td id="T_c1ff2_row8_col2" class="data row8 col2">SRR6502344</td>
<td id="T_c1ff2_row8_col3" class="data row8 col3">27.146048</td>
<td id="T_c1ff2_row8_col4" class="data row8 col4">65,697,684,300</td>
<td id="T_c1ff2_row8_col5" class="data row8 col5">218,992,281</td>
</tr>
<tr class="even">
<td id="T_c1ff2_level0_row9" class="row_heading level0 row9" data-quarto-table-cell-role="th">23</td>
<td id="T_c1ff2_row9_col0" class="data row9 col0">pachytene spermatocyte</td>
<td id="T_c1ff2_row9_col1" class="data row9 col1">SAMN08375234</td>
<td id="T_c1ff2_row9_col2" class="data row9 col2">SRR6502345</td>
<td id="T_c1ff2_row9_col3" class="data row9 col3">26.202707</td>
<td id="T_c1ff2_row9_col4" class="data row9 col4">63,490,538,700</td>
<td id="T_c1ff2_row9_col5" class="data row9 col5">211,635,129</td>
</tr>
<tr class="odd">
<td id="T_c1ff2_level0_row10" class="row_heading level0 row10" data-quarto-table-cell-role="th">0</td>
<td id="T_c1ff2_row10_col0" class="data row10 col0">pachytene spermatocyte</td>
<td id="T_c1ff2_row10_col1" class="data row10 col1">SAMN09427370</td>
<td id="T_c1ff2_row10_col2" class="data row10 col2">SRR7345458</td>
<td id="T_c1ff2_row10_col3" class="data row10 col3">55.970557</td>
<td id="T_c1ff2_row10_col4" class="data row10 col4">153,281,577,900</td>
<td id="T_c1ff2_row10_col5" class="data row10 col5">510,938,593</td>
</tr>
<tr class="even">
<td id="T_c1ff2_level0_row11" class="row_heading level0 row11" data-quarto-table-cell-role="th">1</td>
<td id="T_c1ff2_row11_col0" class="data row11 col0">pachytene spermatocyte</td>
<td id="T_c1ff2_row11_col1" class="data row11 col1">SAMN09427370</td>
<td id="T_c1ff2_row11_col2" class="data row11 col2">SRR7345459</td>
<td id="T_c1ff2_row11_col3" class="data row11 col3">53.982492</td>
<td id="T_c1ff2_row11_col4" class="data row11 col4">144,993,841,200</td>
<td id="T_c1ff2_row11_col5" class="data row11 col5">483,312,804</td>
</tr>
<tr class="odd">
<td id="T_c1ff2_level0_row12" class="row_heading level0 row12" data-quarto-table-cell-role="th">11</td>
<td id="T_c1ff2_row12_col0" class="data row12 col0">pachytene spermatocyte</td>
<td id="T_c1ff2_row12_col1" class="data row12 col1">SAMN08375235</td>
<td id="T_c1ff2_row12_col2" class="data row12 col2">SRR7349191</td>
<td id="T_c1ff2_row12_col3" class="data row12 col3">51.274476</td>
<td id="T_c1ff2_row12_col4" class="data row12 col4">137,821,979,100</td>
<td id="T_c1ff2_row12_col5" class="data row12 col5">459,406,597</td>
</tr>
<tr class="even">
<td id="T_c1ff2_level0_row13" class="row_heading level0 row13" data-quarto-table-cell-role="th">24</td>
<td id="T_c1ff2_row13_col0" class="data row13 col0">round spermatid</td>
<td id="T_c1ff2_row13_col1" class="data row13 col1">SAMN08375232</td>
<td id="T_c1ff2_row13_col2" class="data row13 col2">SRR6502351</td>
<td id="T_c1ff2_row13_col3" class="data row13 col3">20.924497</td>
<td id="T_c1ff2_row13_col4" class="data row13 col4">55,095,075,300</td>
<td id="T_c1ff2_row13_col5" class="data row13 col5">183,650,251</td>
</tr>
<tr class="odd">
<td id="T_c1ff2_level0_row14" class="row_heading level0 row14" data-quarto-table-cell-role="th">25</td>
<td id="T_c1ff2_row14_col0" class="data row14 col0">round spermatid</td>
<td id="T_c1ff2_row14_col1" class="data row14 col1">SAMN08375232</td>
<td id="T_c1ff2_row14_col2" class="data row14 col2">SRR6502352</td>
<td id="T_c1ff2_row14_col3" class="data row14 col3">41.133960</td>
<td id="T_c1ff2_row14_col4" class="data row14 col4">115,578,475,800</td>
<td id="T_c1ff2_row14_col5" class="data row14 col5">385,261,586</td>
</tr>
<tr class="even">
<td id="T_c1ff2_level0_row15" class="row_heading level0 row15" data-quarto-table-cell-role="th">26</td>
<td id="T_c1ff2_row15_col0" class="data row15 col0">round spermatid</td>
<td id="T_c1ff2_row15_col1" class="data row15 col1">SAMN08375232</td>
<td id="T_c1ff2_row15_col2" class="data row15 col2">SRR6502353</td>
<td id="T_c1ff2_row15_col3" class="data row15 col3">36.444117</td>
<td id="T_c1ff2_row15_col4" class="data row15 col4">96,195,161,400</td>
<td id="T_c1ff2_row15_col5" class="data row15 col5">320,650,538</td>
</tr>
<tr class="odd">
<td id="T_c1ff2_level0_row16" class="row_heading level0 row16" data-quarto-table-cell-role="th">2</td>
<td id="T_c1ff2_row16_col0" class="data row16 col0">round spermatid</td>
<td id="T_c1ff2_row16_col1" class="data row16 col1">SAMN09427369</td>
<td id="T_c1ff2_row16_col2" class="data row16 col2">SRR7345460</td>
<td id="T_c1ff2_row16_col3" class="data row16 col3">38.244654</td>
<td id="T_c1ff2_row16_col4" class="data row16 col4">104,105,827,200</td>
<td id="T_c1ff2_row16_col5" class="data row16 col5">347,019,424</td>
</tr>
<tr class="even">
<td id="T_c1ff2_level0_row17" class="row_heading level0 row17" data-quarto-table-cell-role="th">3</td>
<td id="T_c1ff2_row17_col0" class="data row17 col0">round spermatid</td>
<td id="T_c1ff2_row17_col1" class="data row17 col1">SAMN09427369</td>
<td id="T_c1ff2_row17_col2" class="data row17 col2">SRR7345461</td>
<td id="T_c1ff2_row17_col3" class="data row17 col3">53.996261</td>
<td id="T_c1ff2_row17_col4" class="data row17 col4">144,532,309,500</td>
<td id="T_c1ff2_row17_col5" class="data row17 col5">481,774,365</td>
</tr>
<tr class="odd">
<td id="T_c1ff2_level0_row18" class="row_heading level0 row18" data-quarto-table-cell-role="th">12</td>
<td id="T_c1ff2_row18_col0" class="data row18 col0">round spermatid</td>
<td id="T_c1ff2_row18_col1" class="data row18 col1">SAMN08375232</td>
<td id="T_c1ff2_row18_col2" class="data row18 col2">SRR7349192</td>
<td id="T_c1ff2_row18_col3" class="data row18 col3">52.384556</td>
<td id="T_c1ff2_row18_col4" class="data row18 col4">140,431,608,000</td>
<td id="T_c1ff2_row18_col5" class="data row18 col5">468,105,360</td>
</tr>
<tr class="even">
<td id="T_c1ff2_level0_row19" class="row_heading level0 row19" data-quarto-table-cell-role="th">29</td>
<td id="T_c1ff2_row19_col0" class="data row19 col0">sperm</td>
<td id="T_c1ff2_row19_col1" class="data row19 col1">SAMN08375229</td>
<td id="T_c1ff2_row19_col2" class="data row19 col2">SRR6502360</td>
<td id="T_c1ff2_row19_col3" class="data row19 col3">26.653940</td>
<td id="T_c1ff2_row19_col4" class="data row19 col4">64,752,370,800</td>
<td id="T_c1ff2_row19_col5" class="data row19 col5">215,841,236</td>
</tr>
<tr class="odd">
<td id="T_c1ff2_level0_row20" class="row_heading level0 row20" data-quarto-table-cell-role="th">30</td>
<td id="T_c1ff2_row20_col0" class="data row20 col0">sperm</td>
<td id="T_c1ff2_row20_col1" class="data row20 col1">SAMN08375228</td>
<td id="T_c1ff2_row20_col2" class="data row20 col2">SRR6502362</td>
<td id="T_c1ff2_row20_col3" class="data row20 col3">23.973440</td>
<td id="T_c1ff2_row20_col4" class="data row20 col4">58,369,232,700</td>
<td id="T_c1ff2_row20_col5" class="data row20 col5">194,564,109</td>
</tr>
<tr class="even">
<td id="T_c1ff2_level0_row21" class="row_heading level0 row21" data-quarto-table-cell-role="th">13</td>
<td id="T_c1ff2_row21_col0" class="data row21 col0">sperm</td>
<td id="T_c1ff2_row21_col1" class="data row21 col1">SAMN08375229</td>
<td id="T_c1ff2_row21_col2" class="data row21 col2">SRR7349193</td>
<td id="T_c1ff2_row21_col3" class="data row21 col3">52.806276</td>
<td id="T_c1ff2_row21_col4" class="data row21 col4">141,148,572,300</td>
<td id="T_c1ff2_row21_col5" class="data row21 col5">470,495,241</td>
</tr>
<tr class="odd">
<td id="T_c1ff2_level0_row22" class="row_heading level0 row22" data-quarto-table-cell-role="th">14</td>
<td id="T_c1ff2_row22_col0" class="data row22 col0">sperm</td>
<td id="T_c1ff2_row22_col1" class="data row22 col1">SAMN08375229</td>
<td id="T_c1ff2_row22_col2" class="data row22 col2">SRR7349195</td>
<td id="T_c1ff2_row22_col3" class="data row22 col3">22.444378</td>
<td id="T_c1ff2_row22_col4" class="data row22 col4">60,523,788,600</td>
<td id="T_c1ff2_row22_col5" class="data row22 col5">201,745,962</td>
</tr>
<tr class="even">
<td id="T_c1ff2_level0_row23" class="row_heading level0 row23" data-quarto-table-cell-role="th">15</td>
<td id="T_c1ff2_row23_col0" class="data row23 col0">sperm</td>
<td id="T_c1ff2_row23_col1" class="data row23 col1">SAMN08375229</td>
<td id="T_c1ff2_row23_col2" class="data row23 col2">SRR7349196</td>
<td id="T_c1ff2_row23_col3" class="data row23 col3">38.253606</td>
<td id="T_c1ff2_row23_col4" class="data row23 col4">104,119,671,000</td>
<td id="T_c1ff2_row23_col5" class="data row23 col5">347,065,570</td>
</tr>
<tr class="odd">
<td id="T_c1ff2_level0_row24" class="row_heading level0 row24" data-quarto-table-cell-role="th">27</td>
<td id="T_c1ff2_row24_col0" class="data row24 col0">spermatogonia</td>
<td id="T_c1ff2_row24_col1" class="data row24 col1">SAMN08375231</td>
<td id="T_c1ff2_row24_col2" class="data row24 col2">SRR6502356</td>
<td id="T_c1ff2_row24_col3" class="data row24 col3">22.845286</td>
<td id="T_c1ff2_row24_col4" class="data row24 col4">58,909,579,800</td>
<td id="T_c1ff2_row24_col5" class="data row24 col5">196,365,266</td>
</tr>
<tr class="even">
<td id="T_c1ff2_level0_row25" class="row_heading level0 row25" data-quarto-table-cell-role="th">28</td>
<td id="T_c1ff2_row25_col0" class="data row25 col0">spermatogonia</td>
<td id="T_c1ff2_row25_col1" class="data row25 col1">SAMN08375231</td>
<td id="T_c1ff2_row25_col2" class="data row25 col2">SRR6502357</td>
<td id="T_c1ff2_row25_col3" class="data row25 col3">17.947471</td>
<td id="T_c1ff2_row25_col4" class="data row25 col4">46,888,332,900</td>
<td id="T_c1ff2_row25_col5" class="data row25 col5">156,294,443</td>
</tr>
<tr class="odd">
<td id="T_c1ff2_level0_row26" class="row_heading level0 row26" data-quarto-table-cell-role="th">4</td>
<td id="T_c1ff2_row26_col0" class="data row26 col0">spermatogonia</td>
<td id="T_c1ff2_row26_col1" class="data row26 col1">SAMN09427379</td>
<td id="T_c1ff2_row26_col2" class="data row26 col2">SRR7345462</td>
<td id="T_c1ff2_row26_col3" class="data row26 col3">18.686342</td>
<td id="T_c1ff2_row26_col4" class="data row26 col4">52,032,780,000</td>
<td id="T_c1ff2_row26_col5" class="data row26 col5">173,442,600</td>
</tr>
<tr class="even">
<td id="T_c1ff2_level0_row27" class="row_heading level0 row27" data-quarto-table-cell-role="th">5</td>
<td id="T_c1ff2_row27_col0" class="data row27 col0">spermatogonia</td>
<td id="T_c1ff2_row27_col1" class="data row27 col1">SAMN09427379</td>
<td id="T_c1ff2_row27_col2" class="data row27 col2">SRR7345463</td>
<td id="T_c1ff2_row27_col3" class="data row27 col3">29.956561</td>
<td id="T_c1ff2_row27_col4" class="data row27 col4">82,384,836,000</td>
<td id="T_c1ff2_row27_col5" class="data row27 col5">274,616,120</td>
</tr>
<tr class="odd">
<td id="T_c1ff2_level0_row28" class="row_heading level0 row28" data-quarto-table-cell-role="th">6</td>
<td id="T_c1ff2_row28_col0" class="data row28 col0">spermatogonia</td>
<td id="T_c1ff2_row28_col1" class="data row28 col1">SAMN09427379</td>
<td id="T_c1ff2_row28_col2" class="data row28 col2">SRR7345464</td>
<td id="T_c1ff2_row28_col3" class="data row28 col3">39.145759</td>
<td id="T_c1ff2_row28_col4" class="data row28 col4">105,153,716,100</td>
<td id="T_c1ff2_row28_col5" class="data row28 col5">350,512,387</td>
</tr>
<tr class="even">
<td id="T_c1ff2_level0_row29" class="row_heading level0 row29" data-quarto-table-cell-role="th">7</td>
<td id="T_c1ff2_row29_col0" class="data row29 col0">spermatogonia</td>
<td id="T_c1ff2_row29_col1" class="data row29 col1">SAMN09427378</td>
<td id="T_c1ff2_row29_col2" class="data row29 col2">SRR7345465</td>
<td id="T_c1ff2_row29_col3" class="data row29 col3">35.816184</td>
<td id="T_c1ff2_row29_col4" class="data row29 col4">96,048,594,600</td>
<td id="T_c1ff2_row29_col5" class="data row29 col5">320,161,982</td>
</tr>
<tr class="odd">
<td id="T_c1ff2_level0_row30" class="row_heading level0 row30" data-quarto-table-cell-role="th">8</td>
<td id="T_c1ff2_row30_col0" class="data row30 col0">spermatogonia</td>
<td id="T_c1ff2_row30_col1" class="data row30 col1">SAMN09427378</td>
<td id="T_c1ff2_row30_col2" class="data row30 col2">SRR7345467</td>
<td id="T_c1ff2_row30_col3" class="data row30 col3">28.396816</td>
<td id="T_c1ff2_row30_col4" class="data row30 col4">77,248,140,900</td>
<td id="T_c1ff2_row30_col5" class="data row30 col5">257,493,803</td>
</tr>
</tbody>
</table>
</div>
</div>
</figure>
</div></div></div>
</div></div>
</div>
</section>
<section id="handling-coolers-or-preparing-coolers" class="level3">
<h3 class="anchored" data-anchor-id="handling-coolers-or-preparing-coolers">Handling coolers (Or: preparing coolers)</h3>
<div id="fig-flowchart-handling-coolers" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-flowchart-handling-coolers-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<p><strong><em>[A flowchart showing the pipeline from <code>.fastq</code> to <code>.mcool</code>. The first 6 steps were done with a Probably BioRender or Inkscape.]</em></strong></p>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-flowchart-handling-coolers-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
</figcaption>
</figure>
</div>
<section id="the-gwf-workflow-targets" class="level4">
<h4 class="anchored" data-anchor-id="the-gwf-workflow-targets">The <em>gwf</em> workflow targets</h4>
<p>A <em>gwf</em> workflow was created to handle the first part of the data processing, and each accesion number (read pair, mate pair) from the Hi-C sequencing was processed in parallel, so their execution was independen on the other samples.</p>
<section id="downloading-the-reads" class="level5">
<h5 class="anchored" data-anchor-id="downloading-the-reads">Downloading the reads</h5>
<p>The reads were downloaded from NCBI SRA portal [ref] directly to GDK using <code>sra-downloader</code> [ref] through docker [ref] as <code>.fastq.gz</code> files.</p>
</section>
<section id="handling-the-reference" class="level5">
<h5 class="anchored" data-anchor-id="handling-the-reference">Handling the reference</h5>
<p>The latest reference genome for rhesus macaque (<em>macaca mulata</em>), <em>rheMac10</em> (or <em>Mmul_10</em>, UCSC or NCBI naming conventions, respectively) was downloaded to GDK from UCSC web servers with <code>wget</code> [ref]. To use <code>bwa</code> (Burrow Wheeler’s Aligner) [ref] for mapping, rheMac10 needs to be indexed with both <code>bwa index</code> with the <code>--bwtsw</code> option and <code>samtools faidx</code>, which results in six indexing files for <code>bwa mem</code> to use.</p>
</section>
<section id="mapping-paired-end-reads" class="level5">
<h5 class="anchored" data-anchor-id="mapping-paired-end-reads">Mapping paired-end reads</h5>
</section>
<section id="pair-and-sort-the-reads" class="level5">
<h5 class="anchored" data-anchor-id="pair-and-sort-the-reads">Pair and sort the reads</h5>
</section>
<section id="filter-deduplicate-pairs" class="level5">
<h5 class="anchored" data-anchor-id="filter-deduplicate-pairs">Filter (deduplicate) pairs</h5>
</section>
<section id="create-interaction-matrices-coolers" class="level5">
<h5 class="anchored" data-anchor-id="create-interaction-matrices-coolers">Create interaction matrices (coolers)</h5>
</section>
<section id="pooling-samples-merging-coolers" class="level5">
<h5 class="anchored" data-anchor-id="pooling-samples-merging-coolers">Pooling samples (Merging coolers)</h5>
<p>The strategy to get the best signal was by pooling the interaction matrices. <code>cooler merge</code> was used to merge all samples in each sub-folder (cell type) to just one interaction matrix for each cell type. The reason for that is that we choose to trust <span class="citation" data-cites="wang_reprogramming_2019">Wang et al. (<a href="#ref-wang_reprogramming_2019" role="doc-biblioref">2019</a>)</span> when they say that compartments are highly reproducible between replicates, and by merging all replicates, we will have a more robust signal.</p>
</section>
<section id="create-multi-resolution-coolers-zoomify" class="level5">
<h5 class="anchored" data-anchor-id="create-multi-resolution-coolers-zoomify">Create multi-resolution coolers (zoomify)</h5>
</section>
<section id="matrix-balancing-iterative-correction" class="level5">
<h5 class="anchored" data-anchor-id="matrix-balancing-iterative-correction">Matrix balancing (Iterative correction)</h5>
</section>
<section id="eigendecomposition" class="level5 unnumbered">
<h5 class="unnumbered anchored" data-anchor-id="eigendecomposition">Eigendecomposition</h5>
<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-open2c" class="csl-entry" role="listitem">
Open Chromosome Collective. 2024. <span>“Open <span>Chromosome</span> <span>Collective</span> (<span>Open2C</span>).”</span> Resource. <em>Open2C</em>. <a href="https://open2c.github.io/">https://open2c.github.io/</a>.
</div>
<div id="ref-wang_reprogramming_2019" class="csl-entry" role="listitem">
Wang, Yao, Hanben Wang, Yu Zhang, Zhenhai Du, Wei Si, Suixing Fan, Dongdong Qin, et al. 2019. <span>“Reprogramming of <span>Meiotic</span> <span>Chromatin</span> <span>Architecture</span> During <span>Spermatogenesis</span>.”</span> <em>Molecular Cell</em> 73 (3): 547–561.e6. <a href="https://doi.org/10.1016/j.molcel.2018.11.019">https://doi.org/10.1016/j.molcel.2018.11.019</a>.
</div>
</div>
</section>
</section>
</section>
</section>
     </main>
<!-- /main column -->  <script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>  </div> <!-- /content --> 
  
</body></html>